# SRA Download, QC, and Trimming

[Bioinformatics - SRA Download, QC, and Trimming](https://www.youtube.com/watch?v=UkrUTgAKFiE&list=PL-0fKymgD8L8qW7SQDaEf3lrSyUkEkR1k&index=2).

1.  Create environment (avoid spaces or special characters).
1.  Init environment
1.  `conda install sra-tools` This will allow us to pull directly from the sra page. (where the sample ex. Files)

  a.  Go to [page](https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP061386&o=acc_s%3Aa).
  b.  Select only the files needed.
  c.  Then in the Selected > click on Accession List >
  d.  Save (this saves a .txt file with the list of names of the selected files).
  e.  Drag onto the file structure, so its now in system
  f.  `ls` to see the files (SRR_Acc_List.txt)
  g.  Check 'head SRR_Acc_List.txt' command
  h.  Also you can use the `vi SRR_Acc_List.txt` command to see the list in the file. Ex.

    ①. SRR2121770\
    ②. SRR2121771\
    ③. SRR2121774\
    ④. SRR2121778\
    ⑤. Etc.

4.  Now that we have the list of files, we have to individually download each one of them.  

  a. One way to do that is with the `fastq-dump --gzi --split-file SRR2121770 &` command.  

     ①. This command will take the SRR2121770 file, and save it as a .gzi file (to save space)    
     ②. it will split the file in forward (1) and Reverse (2),      
     ③. the "&" sign means that it will run this command in the background.    
     
  b. This will take some time since the file is big. A way to see if the process is finished or not is with the command `jobs`, which gives a list of the jobs in process or nothing if there are not jobs running.\
  c. Another way to do it is with the pipe command, so we don't have to do this process for each file. Use the command `cat SRR_Acc_List.txt | parallel fastq-dump --gzi --split-file {}`.  
  
    ①. This will take the SRR_Acc_List.txt and catalog the list.     
    ②. Then it will take each of the element in the list "{}" slipt them, and save them as gzi.  
    ③. In the rawReads folder???.(create rawReads folder). you will have a list of .fastq.gz_1 and .fastq.gz_2 for each element of the list. `…rawReads$ ls`.  
    ④. Each time something is changed in the files, create a new folder and save there so the originals are not modified. Ex. Trimmed.   

5.  Once all files are downloaded, install multiqc. `conda install multiqc`. (if not installed)  
6.  Before running fastqc, create an output directory where to save the processed files. `mkdir rawFastQC` (make sure you are in the right directory before making this folder).  
7.  Then run fastqc: `fastqc -t 64 rawReads/*fastq.gz -o rawFastQC`  

  a.  This means, process files with fastqc using 64 treads (-t 64) windows/linux virtual machine;  
  b.  by taking all fastqc.gz files in the folder rawReads;  
  c.  and save in the output directory (-o) called rawFastQC.  
  d.  This process will take a long time depending on the number and size of the files.  

8.  Once finished `cd rawFastQC` and check the files `ls`   
9.  Then inside the folder, run multiqc in `multiqc .`.   
10. Once this finished, inside the folder a new "multiqc_report.html" will be created.   
11. Open the files and compare.   

22:29    
**Once this quality control is finished we can pass to the trimming of the samples.**  

1.  Install trimmomatic, `conda install trimmomatic` (if not installed).  

```{bash, eval=FALSE}
conda install trimmomatic
# parallel my also be required for task piping.
conda install parallel
```

2.  If we are going to trim the files, we need to make a new folder, so exit rawFastQC folder.   `cd..` and `mkdir trimmedReads`, we are going to point trimmomatic the this folder.  
```{bash, eval=FALSE}
# exit rawFastQC folder (remember there is a space between cd and .)
cd .. 
# Create new folder for trimmed files
mkdir trimmedReads
```

3.  We are going to pull our true adapter sequencer to our working directory. Then it will be a lot easier to point trimmomatic to that vs having to type in the really long path. `cp ~/miniconda3/envs/name_of_environment/share/trimmomatic-0.39-1/adapters/TrueSeq3-PE-2.fa .` (the last point is important, it means that the command will the done in this folder). This command copy the file TruSeq3-PE-2.fa file to our current folder, why? so the process will be done directly in our folder, instead of the need to type the path to the folder (if not, in the code, instead of the point, type the path to the directory.)   
`cp ~/miniconda3/envs/rna-seq_test/share/trimmomatic-0.39-1/adapters/TrueSeq3-PE-2.fa .`  This gives error.  Sometimes, due to updates in conda and trimmomatic, folder names might change or there is a space in a folders name. Check if this is the case by `open "/usr/local/anaconda3/envs/rna-seq_test/share/"`.  Seems like the version of timmomatic is "0.39-2" and not "0.39-1, and I installed anaconda3 and not miniconda3. Correct code and try again.   
Full path is: /usr/local/anaconda3/envs/rna-seq_test/share/trimmomatic-0.39-2/adapters/TruSeq3-PE-2.fa, then the code would be:
```{bash, eval=FALSE}
`cp /usr/local/anaconda3/envs/rna-seq_test/share/trimmomatic-0.39-2/adapters/TruSeq3-PE-2.fa .`
```

4. Install Parallel, `Conda install paralell`    
5. Create new directory for the trimmed files. `mkdir trimmedReads`.   
5. Now run the trimmomatic.  
For my samples it would be...

```{bash, eval=FALSE}
# Try 01 FAILED
trimmomatic PE -threads 5 rawReads/mPDL_RNA7D_Ko1_S1_L001_R1_001.fastq.gz rawReads/mPDL_RNA7D_Ko1_S1_L001_R2_001.fastq.gz -baseout trimmedReads/mPDL_RNA7D_Ko1_S1_L001 ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:10:2keepBothReads LEADING:3 TRALING:3 MINLEN:36 2> trimmedReads/mPDL_RNA7D_Ko1_S1_L001trimming.log
```

```{bash, eval=FALSE}
# Try 02 FAILED
trimmomatic PE -threads 5 rawReads/mPDL_RNA7D_Ko1_S1_L001_R1_001_1.fastq.gz rawReads/mPDL_RNA7D_Ko1_S1_L001_R2_001_2.fastq.gz -baseout trimmedReads/mPDL_RNA7D_Ko1_S1_L001 ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:10:2keepBothReads LEADING:3 TRALING:3 MINLEN:36 2> trimmedReads/mPDL_RNA7D_Ko1_S1_L001trimming.log
```

```{bash, eval=FALSE}
# Try 03 SUCCESS!!!?
trimmomatic PE rawReads/mPDL_RNA7D_Ko1_S1_L001_R1_001_1.fastq.gz rawReads/mPDL_RNA7D_Ko1_S1_L001_R2_001_2.fastq.gz -baseout trimmedReads/mPDL_RNA7D_Ko1_S1_L001 ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36 2> trimmedReads/mPDL_RNA7D_Ko1_S1_L001trimming.log
```

The **PE** means **"Pair end Read"**, because we have both forward and reverse ends (1 and 2); If we would have only one end read (1 or 2) then it would be only single read, for single reads we type SE (instead of the PE).

Timmomatic usually use up to 5 threads, thus we use **"-threads 5"**. (What if we don't type the # of threads? would it use all threads? Why is this important?....)  

Then type the path to the folder and files containing the files to be trimmed. **rawReads/mPDL_RNA7D_Ko1_S1_L001_R1_001.fastq.gz** (forward), and type also the reverse **"rawReads/mPDL_RNA7D_Ko1_S1_L001_R2_001.fastq.gz"**   
The **"-baseout trimmedReads/mPDL_RNA7D_Ko1_S1_L001"** means that; the base file name of the output will all have the name **" trimmedReads/mPDL_RNA7D_Ko1_S1_L001"**. It will place the file in the trimmedReads folder and all the file with start with the name mPDL_RNA7D_Ko1_S1_L001.  

Then we use the True sequence adapter (TruSeq3-PE-2.fa) that we moved to the working folder by typing **"ILLUMINACLIP:TruSeq3-PE-2.fa:"** (you can use the tap for auto completion coz it is in the same folder)  

Then we place the trimming parameters??... **"2:30:10:2keepBothReads"**. Add more info here.....  
how to understand the trimmomatic flags? [Understanding Trimmomatic](https://youtu.be/Op3W5TEej3k?list=PL-0fKymgD8L8qW7SQDaEf3lrSyUkEkR1k)  
   
The **"LEADING:3 TRALING:3"**means that; if the bases have bad quality, then trimm the leading 3 and trailing 3 bases. [more info](https://youtu.be/UkrUTgAKFiE?list=PL-0fKymgD8L8qW7SQDaEf3lrSyUkEkR1k&t=1729),  If the quality is of very good quality we can remove 10, but we dont want to remove/through away that much information from out reeds.   

Then, since our reads are really short and we are looking at their quality, we want to set a mean length. **"MINLEN:36"**. So if any or our 51 base pair reads have more than 15 removed (51-15= 36), we are just not going to use or keep them because we lost ~33% of the information of that read, so is not going to be a good (quality?) read.  

Then we want to keep the output. **"2> mPDL_RNA7D_Ko1_S1_L001trimming.log`"**, the "2" is to keep output, and then feed that into ">" the "trimmedReads folder, and make (from all the ourput) a nice report like for fastQC and colled it "mPDL_RNA7D_Ko1_S1_L001trimming.log".   

From all the code typed, there will be 5 outputs to be loged, ....
Since this takes really long time, and option will be to do a cat with the reads and do 4 jobs at a time. 

For my files..??
```{bash, eval=FALSE}
cat Filename.txt | parallel -j 4 "trimmomatic PE -threads 5 rawReads/{}_R1_001.fastq.gz rawReads/{}_R2_001.fastq.gz -baseout trimmedReads/{} ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:10:2keepBothReads LEADING:3 TRALING:3 MINLEN:36 2> trimmedReads/{}trimming.log"
```


In the tutorial for auto loading.
```{bash, eval=FALSE}
cat Filename_list.txt | parallel -j 4 "trimmomatic PE -threads 5 rawReads/{}_1.fastq.gz rawReads/{}_2.fastq.gz -baseout trimmedReads/{} ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:10:2keepBothReads LEADING:3 TRALING:3 MINLEN:36 2> trimmedReads/{}trimming.log"
```

Control + Z to stop running and add `bj 1` (in root?) to run in the background. type `jobs` (in terminal) to see the jobs running in the background. 

In the github tutorial:
Conda was used again to run Trimmomatic. This isn't as easy as using the wildcard like with FastQC because each output has to be personalized for the read files that are input into Trimmomatic. Also, we have to make sure that the adapter sequences are in the same folder that we are running so we can refer to them easily when calling the Trimmomatic program. In this case, we are using the TruSeq3-PE-2.fa adapter sequences For example:  

```{bash, eval=FALSE}
~/miniconda2/bin/trimmomatic PE SRR2121770_1.fastq.gz SRR2121770_2.fastq.gz 770_fp.fq.gz 770_fu.fq.gz 770_rp.fq.gz 770_ru.fq.gz ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:10:2keepBothReads LEADING:3 TRALING:3 MINLEN:36 &
```
