<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Background | Bioinformatics_RNA-seq</title>
  <meta name="description" content="RNA-seq notes and trials" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Background | Bioinformatics_RNA-seq" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="RNA-seq notes and trials" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Background | Bioinformatics_RNA-seq" />
  
  <meta name="twitter:description" content="RNA-seq notes and trials" />
  

<meta name="author" content="J. Marcelo Rosales R." />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="rna-seq-for-stat736-alex-soupir.html"/>
<link rel="next" href="cleaning-data.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to RNA sequencing (RNA_seq)</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#downloading-and-setting-up-conda-environments."><i class="fa fa-check"></i><b>1.1</b> Downloading and setting up conda environments.</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#install-miniconda-or-anaconda."><i class="fa fa-check"></i><b>1.2</b> Install Miniconda, or Anaconda.</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#conda-terminal-commands."><i class="fa fa-check"></i><b>1.2.1</b> Conda terminal commands.</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#anacondaminiconda-install-summary-of-steps."><i class="fa fa-check"></i><b>1.2.2</b> Anaconda/Miniconda install Summary of Steps.</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#conda-activate-error"><i class="fa fa-check"></i><b>1.2.3</b> Conda activate error</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#alignment-procedures-and-files."><i class="fa fa-check"></i><b>1.3</b> Alignment Procedures and files.</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#using-tophat-early-version"><i class="fa fa-check"></i><b>1.3.1</b> Using Tophat (early version)</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#using-tophat"><i class="fa fa-check"></i><b>1.3.2</b> Using Tophat</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#using-star"><i class="fa fa-check"></i><b>1.3.3</b> Using STAR</a></li>
<li class="chapter" data-level="1.3.4" data-path="index.html"><a href="index.html#custompersonal-way-to-download-the-mus_musculus_ncbi_build37.2.tar.gz-file."><i class="fa fa-check"></i><b>1.3.4</b> Custom/personal way to Download the Mus_musculus_NCBI_build37.2.tar.gz file.</a></li>
<li class="chapter" data-level="1.3.5" data-path="index.html"><a href="index.html#get-the-sample-files."><i class="fa fa-check"></i><b>1.3.5</b> Get the Sample files.</a></li>
<li class="chapter" data-level="1.3.6" data-path="index.html"><a href="index.html#starting-to-analize-fastq.gz-files"><i class="fa fa-check"></i><b>1.3.6</b> Starting to analize fastq.gz files</a></li>
<li class="chapter" data-level="1.3.7" data-path="index.html"><a href="index.html#trimming-with-trimmomatic"><i class="fa fa-check"></i><b>1.3.7</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="1.3.8" data-path="index.html"><a href="index.html#using-star-to-map-genes"><i class="fa fa-check"></i><b>1.3.8</b> Using STAR to map genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html"><i class="fa fa-check"></i><b>2</b> WGCNA RNA-seq Blog</a>
<ul>
<li class="chapter" data-level="2.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#WGCNA"><i class="fa fa-check"></i><b>2.1</b> Introduction to the WGCNA Rpackage</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-rpackage-installation-procedure."><i class="fa fa-check"></i><b>2.1.1</b> WGCNA Rpackage Installation procedure.</a></li>
<li class="chapter" data-level="2.1.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#final-installation-workflow"><i class="fa fa-check"></i><b>2.1.2</b> Final installation workflow:</a></li>
<li class="chapter" data-level="2.1.3" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-tutorial."><i class="fa fa-check"></i><b>2.1.3</b> WGCNA tutorial.</a></li>
<li class="chapter" data-level="2.1.4" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#gene-expression-analysis"><i class="fa fa-check"></i><b>2.1.4</b> 3.4 Gene expression analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#rna-seq-analisys-with-kakus-data-sample."><i class="fa fa-check"></i><b>2.2</b> RNA-seq Analisys with Kaku’s Data sample.</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#data-analysis."><i class="fa fa-check"></i><b>2.2.1</b> Data Analysis.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="sra-download-qc-and-trimming.html"><a href="sra-download-qc-and-trimming.html"><i class="fa fa-check"></i><b>3</b> SRA Download, QC, and Trimming</a></li>
<li class="chapter" data-level="4" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>4</b> Methods</a></li>
<li class="chapter" data-level="5" data-path="youtube-references-.html"><a href="youtube-references-.html"><i class="fa fa-check"></i><b>5</b> Youtube references.</a></li>
<li class="chapter" data-level="6" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>6</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="7" data-path="rna-seq-for-stat736-alex-soupir.html"><a href="rna-seq-for-stat736-alex-soupir.html"><i class="fa fa-check"></i><b>7</b> RNA Seq for STAT736 Alex Soupir</a></li>
<li class="chapter" data-level="8" data-path="background.html"><a href="background.html"><i class="fa fa-check"></i><b>8</b> <strong>Background</strong></a></li>
<li class="chapter" data-level="9" data-path="cleaning-data.html"><a href="cleaning-data.html"><i class="fa fa-check"></i><b>9</b> <strong>Cleaning Data</strong></a>
<ul>
<li class="chapter" data-level="9.1" data-path="cleaning-data.html"><a href="cleaning-data.html#programs-used"><i class="fa fa-check"></i><b>9.1</b> Programs used?</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="cleaning-data.html"><a href="cleaning-data.html#picking-the-right-node"><i class="fa fa-check"></i><b>9.1.1</b> Picking the right node</a></li>
<li class="chapter" data-level="9.1.2" data-path="cleaning-data.html"><a href="cleaning-data.html#creating-slurm-scripts"><i class="fa fa-check"></i><b>9.1.2</b> Creating slurm scripts</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="cleaning-data.html"><a href="cleaning-data.html#acquiring-sequences"><i class="fa fa-check"></i><b>9.2</b> Acquiring sequences</a></li>
<li class="chapter" data-level="9.3" data-path="cleaning-data.html"><a href="cleaning-data.html#fastqc"><i class="fa fa-check"></i><b>9.3</b> FastQC</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="cleaning-data.html"><a href="cleaning-data.html#multiqc"><i class="fa fa-check"></i><b>9.3.1</b> MultiQC</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="cleaning-data.html"><a href="cleaning-data.html#trimming-with-trimmomatic-1"><i class="fa fa-check"></i><b>9.4</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="9.5" data-path="cleaning-data.html"><a href="cleaning-data.html#alignment"><i class="fa fa-check"></i><b>9.5</b> Alignment</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="cleaning-data.html"><a href="cleaning-data.html#using-tophat-1"><i class="fa fa-check"></i><b>9.5.1</b> Using Tophat</a></li>
<li class="chapter" data-level="9.5.2" data-path="cleaning-data.html"><a href="cleaning-data.html#using-star-1"><i class="fa fa-check"></i><b>9.5.2</b> Using STAR</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="cleaning-data.html"><a href="cleaning-data.html#assembling-transcripts-with-cufflinks"><i class="fa fa-check"></i><b>9.6</b> Assembling transcripts with Cufflinks</a></li>
<li class="chapter" data-level="9.7" data-path="cleaning-data.html"><a href="cleaning-data.html#checking-for-contamination"><i class="fa fa-check"></i><b>9.7</b> Checking for Contamination</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="cleaning-data.html"><a href="cleaning-data.html#phix-contamination"><i class="fa fa-check"></i><b>9.7.1</b> PhiX contamination</a></li>
<li class="chapter" data-level="9.7.2" data-path="cleaning-data.html"><a href="cleaning-data.html#rrna-sequences"><i class="fa fa-check"></i><b>9.7.2</b> rRNA Sequences</a></li>
<li class="chapter" data-level="9.7.3" data-path="cleaning-data.html"><a href="cleaning-data.html#bacterial-contamination"><i class="fa fa-check"></i><b>9.7.3</b> Bacterial contamination</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="cleaning-data.html"><a href="cleaning-data.html#counting-transcripts"><i class="fa fa-check"></i><b>9.8</b> Counting Transcripts</a>
<ul>
<li class="chapter" data-level="9.8.1" data-path="cleaning-data.html"><a href="cleaning-data.html#final-qc-of-cleaning-the-data"><i class="fa fa-check"></i><b>9.8.1</b> Final QC of cleaning the data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="differentially-expressed-sequence-identification.html"><a href="differentially-expressed-sequence-identification.html"><i class="fa fa-check"></i><b>10</b> <strong>Differentially Expressed Sequence Identification</strong></a>
<ul>
<li class="chapter" data-level="10.1" data-path="differentially-expressed-sequence-identification.html"><a href="differentially-expressed-sequence-identification.html#analyzing-reads-counts"><i class="fa fa-check"></i><b>10.1</b> Analyzing Reads Counts</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="differentially-expressed-sequence-identification.html"><a href="differentially-expressed-sequence-identification.html#quick-data-exploration"><i class="fa fa-check"></i><b>10.1.1</b> Quick Data Exploration</a></li>
<li class="chapter" data-level="10.1.2" data-path="differentially-expressed-sequence-identification.html"><a href="differentially-expressed-sequence-identification.html#filtering-normalization-and-trasformation-using-deseq2"><i class="fa fa-check"></i><b>10.1.2</b> Filtering, Normalization, and Trasformation using DESeq2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><a href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><i class="fa fa-check"></i><b>11</b> Interactions cause a difference between the lfc betwen pooled data, e.g. p53+/+ (control and IR) and p53-/- (control and IR)</a>
<ul>
<li class="chapter" data-level="11.0.1" data-path="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><a href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>11.0.1</b> Exploratory Data Analysis</a></li>
<li class="chapter" data-level="11.0.2" data-path="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><a href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html#identification-of-differentially-expressed-genes"><i class="fa fa-check"></i><b>11.0.2</b> Identification of Differentially Expressed Genes</a></li>
<li class="chapter" data-level="11.0.3" data-path="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><a href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html#gene-annotations"><i class="fa fa-check"></i><b>11.0.3</b> Gene Annotations</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="go-enrichment-analysis-using-gostats.html"><a href="go-enrichment-analysis-using-gostats.html"><i class="fa fa-check"></i><b>12</b> 7. GO Enrichment analysis using GOstats</a>
<ul>
<li class="chapter" data-level="12.1" data-path="go-enrichment-analysis-using-gostats.html"><a href="go-enrichment-analysis-using-gostats.html#go-enrichment-analysis-of-downregulated-genes"><i class="fa fa-check"></i><b>12.1</b> GO Enrichment analysis of downregulated genes</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html"><i class="fa fa-check"></i><b>13</b> 8. Pathway analysis using expression data</a>
<ul>
<li class="chapter" data-level="13.1" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#prepare-data"><i class="fa fa-check"></i><b>13.1</b> Prepare data</a></li>
<li class="chapter" data-level="13.2" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#kegg-pathways"><i class="fa fa-check"></i><b>13.2</b> KEGG pathways</a></li>
<li class="chapter" data-level="13.3" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-oxidative-phosphorylation."><i class="fa fa-check"></i><b>13.3</b> Pathway and regulation of genes for Oxidative phosphorylation.</a></li>
<li class="chapter" data-level="13.4" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-glycosylphosphatidylinositolgpi-anchor-biosynthesis."><i class="fa fa-check"></i><b>13.4</b> Pathway and regulation of genes for Glycosylphosphatidylinositol(GPI)-anchor biosynthesis.</a></li>
<li class="chapter" data-level="13.5" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-rna-polymerase."><i class="fa fa-check"></i><b>13.5</b> Pathway and regulation of genes for RNA polymerase.</a></li>
<li class="chapter" data-level="13.6" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-nucleotide-excision-repair."><i class="fa fa-check"></i><b>13.6</b> Pathway and regulation of genes for Nucleotide excision repair.</a></li>
<li class="chapter" data-level="13.7" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-non-homologous-end-joining."><i class="fa fa-check"></i><b>13.7</b> Pathway and regulation of genes for Non-homologous end-joining.</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics_RNA-seq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="background" class="section level1" number="8">
<h1><span class="header-section-number">Chapter 8</span> <strong>Background</strong></h1>
<p>For STAT736-Fall-2019, we are analyzing the RNA-Seq from the publication <a href="https://www.ncbi.nlm.nih.gov/pubmed/26372730?dopt=Abstract">Genome-wide analysis of p53 transcriptional programs in B cells upon exposure to genotoxic stress in vivo.</a> We are only using the <a href="https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP061386">sequences</a> <em>B cells from spleen</em> and not the <em>non-B cells from spleen</em> from the SRA Run Selector on NCBI.</p>
<p>The mice were exposed to whole-body ionizing radiation and sequences were extracted from both Bcells and non-B cells from the spleens of the mice. Two genotypes of mice were used: mice with p53 knocked out and the wild-type C57/Bl6. There were 4 different group combinations including the 2 different genotypes; each genotype was subjected to the ionizing radiation as well as control/mock.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="background.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb42-2"><a href="background.html#cb42-2" aria-hidden="true" tabindex="-1"></a>experimental_design <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="st">&#39;Genotype&#39;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;p53&#39;</span>, <span class="st">&#39;C57/Bl6&#39;</span>, <span class="st">&#39;p53&#39;</span>, <span class="st">&#39;C57/Bl6&#39;</span>), <span class="st">&#39;Treatment&#39;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;Mock&#39;</span>, <span class="st">&#39;Mock&#39;</span>, <span class="st">&#39;IR&#39;</span>, <span class="st">&#39;IR&#39;</span>))</span>
<span id="cb42-3"><a href="background.html#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(experimental_design) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;Group 1&#39;</span>, <span class="st">&#39;Group 2&#39;</span>, <span class="st">&#39;Group 3&#39;</span>, <span class="st">&#39;Group 4&#39;</span>)</span>
<span id="cb42-4"><a href="background.html#cb42-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb42-5"><a href="background.html#cb42-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reticulate)</span>
<span id="cb42-6"><a href="background.html#cb42-6" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(experimental_design[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb42-7"><a href="background.html#cb42-7" aria-hidden="true" tabindex="-1"></a>      <span class="co">#, &quot;latex&quot;</span></span>
<span id="cb42-8"><a href="background.html#cb42-8" aria-hidden="true" tabindex="-1"></a>      , <span class="at">caption =</span> <span class="st">&#39;Treatment groups of the mice that were either controls or treated with ionizing radiation to determine reaction of p53.&#39;</span></span>
<span id="cb42-9"><a href="background.html#cb42-9" aria-hidden="true" tabindex="-1"></a>      <span class="co">#,booktabs = T</span></span>
<span id="cb42-10"><a href="background.html#cb42-10" aria-hidden="true" tabindex="-1"></a>      ) <span class="sc">%&gt;%</span></span>
<span id="cb42-11"><a href="background.html#cb42-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">latex_options =</span> <span class="fu">c</span>(<span class="st">&quot;striped&quot;</span>, <span class="st">&quot;repeat_header&quot;</span>))</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
(#tab:experimental_design)Treatment groups of the mice that were either controls or treated with ionizing radiation to determine reaction of p53.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Genotype
</th>
<th style="text-align:left;">
Treatment
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Group 1
</td>
<td style="text-align:left;">
p53
</td>
<td style="text-align:left;">
Mock
</td>
</tr>
<tr>
<td style="text-align:left;">
Group 2
</td>
<td style="text-align:left;">
C57/Bl6
</td>
<td style="text-align:left;">
Mock
</td>
</tr>
<tr>
<td style="text-align:left;">
Group 3
</td>
<td style="text-align:left;">
p53
</td>
<td style="text-align:left;">
IR
</td>
</tr>
<tr>
<td style="text-align:left;">
Group 4
</td>
<td style="text-align:left;">
C57/Bl6
</td>
<td style="text-align:left;">
IR
</td>
</tr>
</tbody>
</table>
<p>This document will contain 2 different pipelines: The first one is going to be using the genome to map the reads too, and the secod is going to be <em>de novo</em>.</p>
<p><a href="#genome-mapping">Genome</a>
<a href="#denovo-assembly"><em>De novo</em></a></p>
<p>#Genome Mapping {#genome-mapping}</p>
<p>The pipeline used in this analysis used <strong>conda</strong> on South Dakota State University’s High Performance Computing cluster to run the programs <strong>FastQC</strong>, <strong>Trimmomatic</strong>, and <strong>Tophat</strong>.</p>
<p>This is different than previous RNA-Seq analyses where I used my workstation pc with <strong>Ubuntu 18.04</strong> to run <strong>FastQC</strong>, <strong>Trimmomatic</strong>, <strong>HiSat2</strong>, <strong>HTSeq</strong>, and <strong>DESeq2</strong> locally. Also, the previous RNA-Seq alayses were of Soybean with treatment combinations of mycorrhizae and rhizobia inoculation.</p>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="rna-seq-for-stat736-alex-soupir.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="cleaning-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Bioinformatics_RNA-seq.pdf", "Bioinformatics_RNA-seq.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
