<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Cleaning Data | Bioinformatics_RNA-seq</title>
  <meta name="description" content="RNA-seq notes and trials" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Cleaning Data | Bioinformatics_RNA-seq" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="RNA-seq notes and trials" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Cleaning Data | Bioinformatics_RNA-seq" />
  
  <meta name="twitter:description" content="RNA-seq notes and trials" />
  

<meta name="author" content="J. Marcelo Rosales R." />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="background.html"/>
<link rel="next" href="differentially-expressed-sequence-identification.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to RNA sequencing (RNA_seq)</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#downloading-and-setting-up-conda-environments."><i class="fa fa-check"></i><b>1.1</b> Downloading and setting up conda environments.</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#install-miniconda-or-anaconda."><i class="fa fa-check"></i><b>1.2</b> Install Miniconda, or Anaconda.</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#conda-terminal-commands."><i class="fa fa-check"></i><b>1.2.1</b> Conda terminal commands.</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#anacondaminiconda-install-summary-of-steps."><i class="fa fa-check"></i><b>1.2.2</b> Anaconda/Miniconda install Summary of Steps.</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#conda-activate-error"><i class="fa fa-check"></i><b>1.2.3</b> Conda activate error</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#alignment-procedures-and-files."><i class="fa fa-check"></i><b>1.3</b> Alignment Procedures and files.</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#using-tophat-early-version"><i class="fa fa-check"></i><b>1.3.1</b> Using Tophat (early version)</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#using-tophat"><i class="fa fa-check"></i><b>1.3.2</b> Using Tophat</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#using-star"><i class="fa fa-check"></i><b>1.3.3</b> Using STAR</a></li>
<li class="chapter" data-level="1.3.4" data-path="index.html"><a href="index.html#custompersonal-way-to-download-the-mus_musculus_ncbi_build37.2.tar.gz-file."><i class="fa fa-check"></i><b>1.3.4</b> Custom/personal way to Download the Mus_musculus_NCBI_build37.2.tar.gz file.</a></li>
<li class="chapter" data-level="1.3.5" data-path="index.html"><a href="index.html#get-the-sample-files."><i class="fa fa-check"></i><b>1.3.5</b> Get the Sample files.</a></li>
<li class="chapter" data-level="1.3.6" data-path="index.html"><a href="index.html#starting-to-analize-fastq.gz-files"><i class="fa fa-check"></i><b>1.3.6</b> Starting to analize fastq.gz files</a></li>
<li class="chapter" data-level="1.3.7" data-path="index.html"><a href="index.html#trimming-with-trimmomatic"><i class="fa fa-check"></i><b>1.3.7</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="1.3.8" data-path="index.html"><a href="index.html#using-star-to-map-genes"><i class="fa fa-check"></i><b>1.3.8</b> Using STAR to map genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html"><i class="fa fa-check"></i><b>2</b> WGCNA RNA-seq Blog</a>
<ul>
<li class="chapter" data-level="2.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#WGCNA"><i class="fa fa-check"></i><b>2.1</b> Introduction to the WGCNA Rpackage</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-rpackage-installation-procedure."><i class="fa fa-check"></i><b>2.1.1</b> WGCNA Rpackage Installation procedure.</a></li>
<li class="chapter" data-level="2.1.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#final-installation-workflow"><i class="fa fa-check"></i><b>2.1.2</b> Final installation workflow:</a></li>
<li class="chapter" data-level="2.1.3" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-tutorial."><i class="fa fa-check"></i><b>2.1.3</b> WGCNA tutorial.</a></li>
<li class="chapter" data-level="2.1.4" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#gene-expression-analysis"><i class="fa fa-check"></i><b>2.1.4</b> 3.4 Gene expression analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#rna-seq-analisys-with-kakus-data-sample."><i class="fa fa-check"></i><b>2.2</b> RNA-seq Analisys with Kaku’s Data sample.</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#data-analysis."><i class="fa fa-check"></i><b>2.2.1</b> Data Analysis.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="sra-download-qc-and-trimming.html"><a href="sra-download-qc-and-trimming.html"><i class="fa fa-check"></i><b>3</b> SRA Download, QC, and Trimming</a></li>
<li class="chapter" data-level="4" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>4</b> Methods</a></li>
<li class="chapter" data-level="5" data-path="youtube-references-.html"><a href="youtube-references-.html"><i class="fa fa-check"></i><b>5</b> Youtube references.</a></li>
<li class="chapter" data-level="6" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>6</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="7" data-path="rna-seq-for-stat736-alex-soupir.html"><a href="rna-seq-for-stat736-alex-soupir.html"><i class="fa fa-check"></i><b>7</b> RNA Seq for STAT736 Alex Soupir</a></li>
<li class="chapter" data-level="8" data-path="background.html"><a href="background.html"><i class="fa fa-check"></i><b>8</b> <strong>Background</strong></a></li>
<li class="chapter" data-level="9" data-path="cleaning-data.html"><a href="cleaning-data.html"><i class="fa fa-check"></i><b>9</b> <strong>Cleaning Data</strong></a>
<ul>
<li class="chapter" data-level="9.1" data-path="cleaning-data.html"><a href="cleaning-data.html#programs-used"><i class="fa fa-check"></i><b>9.1</b> Programs used?</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="cleaning-data.html"><a href="cleaning-data.html#picking-the-right-node"><i class="fa fa-check"></i><b>9.1.1</b> Picking the right node</a></li>
<li class="chapter" data-level="9.1.2" data-path="cleaning-data.html"><a href="cleaning-data.html#creating-slurm-scripts"><i class="fa fa-check"></i><b>9.1.2</b> Creating slurm scripts</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="cleaning-data.html"><a href="cleaning-data.html#acquiring-sequences"><i class="fa fa-check"></i><b>9.2</b> Acquiring sequences</a></li>
<li class="chapter" data-level="9.3" data-path="cleaning-data.html"><a href="cleaning-data.html#fastqc"><i class="fa fa-check"></i><b>9.3</b> FastQC</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="cleaning-data.html"><a href="cleaning-data.html#multiqc"><i class="fa fa-check"></i><b>9.3.1</b> MultiQC</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="cleaning-data.html"><a href="cleaning-data.html#trimming-with-trimmomatic-1"><i class="fa fa-check"></i><b>9.4</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="9.5" data-path="cleaning-data.html"><a href="cleaning-data.html#alignment"><i class="fa fa-check"></i><b>9.5</b> Alignment</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="cleaning-data.html"><a href="cleaning-data.html#using-tophat-1"><i class="fa fa-check"></i><b>9.5.1</b> Using Tophat</a></li>
<li class="chapter" data-level="9.5.2" data-path="cleaning-data.html"><a href="cleaning-data.html#using-star-1"><i class="fa fa-check"></i><b>9.5.2</b> Using STAR</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="cleaning-data.html"><a href="cleaning-data.html#assembling-transcripts-with-cufflinks"><i class="fa fa-check"></i><b>9.6</b> Assembling transcripts with Cufflinks</a></li>
<li class="chapter" data-level="9.7" data-path="cleaning-data.html"><a href="cleaning-data.html#checking-for-contamination"><i class="fa fa-check"></i><b>9.7</b> Checking for Contamination</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="cleaning-data.html"><a href="cleaning-data.html#phix-contamination"><i class="fa fa-check"></i><b>9.7.1</b> PhiX contamination</a></li>
<li class="chapter" data-level="9.7.2" data-path="cleaning-data.html"><a href="cleaning-data.html#rrna-sequences"><i class="fa fa-check"></i><b>9.7.2</b> rRNA Sequences</a></li>
<li class="chapter" data-level="9.7.3" data-path="cleaning-data.html"><a href="cleaning-data.html#bacterial-contamination"><i class="fa fa-check"></i><b>9.7.3</b> Bacterial contamination</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="cleaning-data.html"><a href="cleaning-data.html#counting-transcripts"><i class="fa fa-check"></i><b>9.8</b> Counting Transcripts</a>
<ul>
<li class="chapter" data-level="9.8.1" data-path="cleaning-data.html"><a href="cleaning-data.html#final-qc-of-cleaning-the-data"><i class="fa fa-check"></i><b>9.8.1</b> Final QC of cleaning the data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="differentially-expressed-sequence-identification.html"><a href="differentially-expressed-sequence-identification.html"><i class="fa fa-check"></i><b>10</b> <strong>Differentially Expressed Sequence Identification</strong></a>
<ul>
<li class="chapter" data-level="10.1" data-path="differentially-expressed-sequence-identification.html"><a href="differentially-expressed-sequence-identification.html#analyzing-reads-counts"><i class="fa fa-check"></i><b>10.1</b> Analyzing Reads Counts</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="differentially-expressed-sequence-identification.html"><a href="differentially-expressed-sequence-identification.html#quick-data-exploration"><i class="fa fa-check"></i><b>10.1.1</b> Quick Data Exploration</a></li>
<li class="chapter" data-level="10.1.2" data-path="differentially-expressed-sequence-identification.html"><a href="differentially-expressed-sequence-identification.html#filtering-normalization-and-trasformation-using-deseq2"><i class="fa fa-check"></i><b>10.1.2</b> Filtering, Normalization, and Trasformation using DESeq2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><a href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><i class="fa fa-check"></i><b>11</b> Interactions cause a difference between the lfc betwen pooled data, e.g. p53+/+ (control and IR) and p53-/- (control and IR)</a>
<ul>
<li class="chapter" data-level="11.0.1" data-path="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><a href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>11.0.1</b> Exploratory Data Analysis</a></li>
<li class="chapter" data-level="11.0.2" data-path="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><a href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html#identification-of-differentially-expressed-genes"><i class="fa fa-check"></i><b>11.0.2</b> Identification of Differentially Expressed Genes</a></li>
<li class="chapter" data-level="11.0.3" data-path="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><a href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html#gene-annotations"><i class="fa fa-check"></i><b>11.0.3</b> Gene Annotations</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="go-enrichment-analysis-using-gostats.html"><a href="go-enrichment-analysis-using-gostats.html"><i class="fa fa-check"></i><b>12</b> 7. GO Enrichment analysis using GOstats</a>
<ul>
<li class="chapter" data-level="12.1" data-path="go-enrichment-analysis-using-gostats.html"><a href="go-enrichment-analysis-using-gostats.html#go-enrichment-analysis-of-downregulated-genes"><i class="fa fa-check"></i><b>12.1</b> GO Enrichment analysis of downregulated genes</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html"><i class="fa fa-check"></i><b>13</b> 8. Pathway analysis using expression data</a>
<ul>
<li class="chapter" data-level="13.1" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#prepare-data"><i class="fa fa-check"></i><b>13.1</b> Prepare data</a></li>
<li class="chapter" data-level="13.2" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#kegg-pathways"><i class="fa fa-check"></i><b>13.2</b> KEGG pathways</a></li>
<li class="chapter" data-level="13.3" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-oxidative-phosphorylation."><i class="fa fa-check"></i><b>13.3</b> Pathway and regulation of genes for Oxidative phosphorylation.</a></li>
<li class="chapter" data-level="13.4" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-glycosylphosphatidylinositolgpi-anchor-biosynthesis."><i class="fa fa-check"></i><b>13.4</b> Pathway and regulation of genes for Glycosylphosphatidylinositol(GPI)-anchor biosynthesis.</a></li>
<li class="chapter" data-level="13.5" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-rna-polymerase."><i class="fa fa-check"></i><b>13.5</b> Pathway and regulation of genes for RNA polymerase.</a></li>
<li class="chapter" data-level="13.6" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-nucleotide-excision-repair."><i class="fa fa-check"></i><b>13.6</b> Pathway and regulation of genes for Nucleotide excision repair.</a></li>
<li class="chapter" data-level="13.7" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-non-homologous-end-joining."><i class="fa fa-check"></i><b>13.7</b> Pathway and regulation of genes for Non-homologous end-joining.</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics_RNA-seq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="cleaning-data" class="section level1" number="9">
<h1><span class="header-section-number">Chapter 9</span> <strong>Cleaning Data</strong></h1>
<div id="programs-used" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> Programs used?</h2>
<ul>
<li>FastQC</li>
<li>Trimmomatic 0.39</li>
<li>Bowtie 2.2.5.0</li>
<li>Tophat 2.1.1</li>
<li>STAR</li>
<li>Cufflinks</li>
<li>Kraken2</li>
<li>MultiQC</li>
<li>featureCount</li>
</ul>
<div id="picking-the-right-node" class="section level3" number="9.1.1">
<h3><span class="header-section-number">9.1.1</span> Picking the right node</h3>
<p>To find a node that we can use on our own, we need to see which nodes are already allocated to jobs and which ones are idle. To do this, we can run <strong>sinfo</strong>. We want to pick one of the nodes that are marked ‘idle’ so we get the whole thing and we aren’t interrupting someone elses job. For the sake of this exercise, lets work on <strong>big-mem</strong>.</p>
<p>Once a node that is idle has been found, you can <em>ssh</em> into it by typing <strong>ssh -X big-mem00#</strong> where # is the node number.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb43-1"><a href="cleaning-data.html#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ssh</span> <span class="at">-X</span> big-mem005</span></code></pre></div>
<p>Once on the node, the modules will have to be pulled from the shared folder again, otherwise we will be left with very basic ones. <strong>NOTE:</strong> if running programs that are in a personal folder such as miniconda (these examples), it is not necessary to add the other modules.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb44-1"><a href="cleaning-data.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> use /cm/shared/modulefiles_local/</span></code></pre></div>
<p>After loading the modules you can use it just as you would any other command line.</p>
</div>
<div id="creating-slurm-scripts" class="section level3" number="9.1.2">
<h3><span class="header-section-number">9.1.2</span> Creating slurm scripts</h3>
<p>When running on a cluster, it can sometime be difficult to find open nodes with the resources needed to run the jobs that we have. Making a slurm script is really easy. Fist we make a new file with the <em>touch</em> command.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb45-1"><a href="cleaning-data.html#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">touch</span> commands.slurm</span></code></pre></div>
<p>Now in our directory we have the file <strong>commands.slurm</strong> which we can edit to hold our code in. We can edit it with the <em>vi</em> command.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb46-1"><a href="cleaning-data.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="ex">vi</span> commands.slurm</span></code></pre></div>
<p>We have a few things that we need to put in the file header so slurm knows what to do with our commands.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb47-1"><a href="cleaning-data.html#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb47-2"><a href="cleaning-data.html#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --job-name=example</span></span>
<span id="cb47-3"><a href="cleaning-data.html#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb47-4"><a href="cleaning-data.html#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --ntasks-per-node=10</span></span>
<span id="cb47-5"><a href="cleaning-data.html#cb47-5" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --output=job-%j-%N.log</span></span>
<span id="cb47-6"><a href="cleaning-data.html#cb47-6" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --partition=bigmem</span></span>
<span id="cb47-7"><a href="cleaning-data.html#cb47-7" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --time=10:00:00</span></span></code></pre></div>
<p>When we break this down, we see <em>–job-name</em> which is what we will see when we look at whats running later, <em>–nodes</em> is the number of nodes we have, <em>–ntasks-per-node</em> is the number of cores that we are requesting to have allocated, <em>–output</em> is the output log file of the job (here it names the output file with the job number and the node that we used), <em>–partition</em> here is requesting a big-mem node but <strong>compute</strong> can also be used, and finally <em>–time</em> is how long we are requesting the allocation for.</p>
<p>If the time runs out before the job is done I believe that it just kills the job even if not finished so we need to think a little about how much time to set. If the time is set too low, the job is killed and if the time is set too long, we may face issues with getting the node allocated to us.</p>
<p>To submit a job we can use <em>sbatch commands.slurm</em> and then we have the job ID. To check the status of our submission we use <em>sbatch</em> and then it shows all of the submitted jobs and how long they have been running along with the name that we set in the script.</p>
</div>
</div>
<div id="acquiring-sequences" class="section level2" number="9.2">
<h2><span class="header-section-number">9.2</span> Acquiring sequences</h2>
<p>To download the sequences from the sequence read archive (SRA), the SRA Toolkit was used. The downloading of the files took a very long time, so this was left to run over night. The <strong>–gzip</strong> was used to keep the files a relatively small, although this can be left out to download uncompressed files, and <strong>–split-files</strong> was used to split the forward read from the reverse read for paired end read trimming through <strong>Trimmomatic</strong>.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb48-1"><a href="cleaning-data.html#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/tools/sratoolkit.2.9.6-1-centos_linux64/bin/fastq-dump</span> <span class="at">--gzip</span> <span class="at">--split-files</span> SRR2121770</span></code></pre></div>
<p>This is an example of the single file, but the above code needed to be ran for all of the following SRA numbers:</p>
<ul>
<li>SRR2121770</li>
<li>SRR2121771</li>
<li>SRR2121774</li>
<li>SRR2121775</li>
<li>SRR2121778</li>
<li>SRR2121779</li>
<li>SRR2121780</li>
<li>SRR2121781</li>
<li>SRR2121786</li>
<li>SRR2121787</li>
<li>SRR2121788</li>
<li>SRR2121789</li>
</ul>
<p>The results from downloading with <strong>–split-files</strong> gives 2 files per SRR, as mentioned before, one forward and one reverse. The suffix of the split files is one with <strong>_1.fastq.gz</strong> and another with <strong>_2.fastq.gz</strong>.</p>
</div>
<div id="fastqc" class="section level2" number="9.3">
<h2><span class="header-section-number">9.3</span> FastQC</h2>
<p>FastQC can be run on all of the read files by using the wild card (*) as in <strong>*.fastq.gz</strong>. This prevents the need to hard code each individual read file into a FastQC command, which saves a lot of time since there are 24 read files in total for these 12 samples.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb49-1"><a href="cleaning-data.html#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/fastqc</span> <span class="pp">*</span>.fastq.gz</span></code></pre></div>
<p>The output from running FastQC is a zipped folder and an HTML file for each of the <strong>.gz</strong> files in the folder. The HTML document looks something like this:</p>
<pre><code>![FastQC of Raw SRR2121770_1.fastq.gz Read](./TopFastQCRaw.PNG)</code></pre>
<p>This is just the top of the file, and every category under the <strong>Summary</strong> heading has a graph that shows how the read quality looks for that particular metric. These reports can give insight into whether the reads are of decent quality or if the quality is poor.</p>
<p>The raw reads we have here all passed for adapter content and sequence length distribution and everything failed per base sequence content. SRR2121770, SRR2121771, SRR2121774, SRR2121775, SRR2121788, SRR2121781-2, and SRR2121789-1 were fairly decent quality reads. SRR2121778, SRR2121779, SRR2121780, SRR2121786, SRR2121787, SRR2121781-1, and SRR2121789-2 were of fairly lower quality (failing 3 or more in both reads. All of them failed both per base sequence quality and per tile sequence quality.</p>
<div id="multiqc" class="section level3" number="9.3.1">
<h3><span class="header-section-number">9.3.1</span> MultiQC</h3>
<p>First lets install <strong>multiqc</strong> with conda. The command for this is <strong><em>conda install -c bioconda multiqc</em></strong>.</p>
<p>When that is finished, we can run MultiQC in the folder with the QC files (they should be moved into a folder alone so things don’t get cluttered later on in the analyses).</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb51-1"><a href="cleaning-data.html#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda/bin/multiqc</span> .</span></code></pre></div>
<p>When MultiQC is finished running, there will be a new folder called <strong>multiqc_data</strong> where the summaries are stored. Now lets go back up a level where our raw data folder and fastqc folder is and make a new folder for all of our MultiQC data. We will copy the FastQC output from MultiQC to this new folder.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb52-1"><a href="cleaning-data.html#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> MultiQC_All</span>
<span id="cb52-2"><a href="cleaning-data.html#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> RawQC/multiqc_data/multiqc_fastqc.txt MultiQC_All/</span></code></pre></div>
</div>
</div>
<div id="trimming-with-trimmomatic-1" class="section level2" number="9.4">
<h2><span class="header-section-number">9.4</span> Trimming with Trimmomatic</h2>
<p>Conda was used again to run Trimmomatic. This isn’t as easy as using the wildcard like with FastQC because each output has to be personalized for the read files that are input into Trimmomatic. Also, we have to make sure that the adapter sequences are in the same folder that we are running so we can refer to them easily when calling the Trimmomatic program. In this case, we are using the TruSeq3-PE-2.fa adapter sequences For example:</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb53-1"><a href="cleaning-data.html#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/trimmomatic</span> PE SRR2121770_1.fastq.gz SRR2121770_2.fastq.gz 770_fp.fq.gz 770_fu.fq.gz 770_rp.fq.gz 770_ru.fq.gz ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:10:2keepBothReads LEADING:3 TRALING:3 MINLEN:36 <span class="kw">&amp;</span></span></code></pre></div>
<p>This would be repeated for each of the pairs (12 in total). We are trimming paired-end reads with the TruSeq3-PE-2 adapters. We are chopping off the first and last 3 bases and if we end up with a sequence less than 36 bases, we get rid of it. We want to make sure that there are enough bases in a read to work with. These parameters can be tweaked for possibly better end results with less being discarded. When Trimmomatic is finished running, it will out put the total number of reads, the total number from both the forward and reverse reads that are kept, the number of only forward reads kept, the number of only reverse reads kept, and the number of discarded reads.</p>
<p>The highest number of reads dropped was from trimming SRR2121786, where 20.55% dropped. Most reads were between 5% and 10% dropped. SRR2121786, SRR2121787, and SRR2121779 had sequence drops greater than 15%.</p>
<p>The trimmed reads can be analyzed again with FastQC to see how well the trimming worked to make the file better quality. After running FastQC on the trimmed files we see that the quality of those that were really bad quality were improved. There were a few different metrics throughout all of the files that bounced from a warning before the failing, or from passing before to a warning, and so forth, overall creating better quality read files.</p>
</div>
<div id="alignment" class="section level2" number="9.5">
<h2><span class="header-section-number">9.5</span> Alignment</h2>
<div id="using-tophat-1" class="section level3" number="9.5.1">
<h3><span class="header-section-number">9.5.1</span> Using Tophat</h3>
<p>Tophat can be installed using the same <strong>conda install</strong> (<strong><em>conda install -c bioconda tophat</em></strong>). When this is finished installing, then we will need to get the mouse genome from the <a href="http://ccb.jhu.edu/software/tophat/igenomes.shtml">Johns Hopkins Univeristy Center for computational BIology</a>. The version of the mouse genome that I am using here is the <a href="ftp://igenome:G3nom3s4u@ussd-ftp.illumina.com/Mus_musculus/NCBI/build37.2/Mus_musculus_NCBI_build37.2.tar.gz">NCBI build37.2</a>. Instead of downloading this from the website and having to move it to the cluser, I will just download it using wget into the folder that has the raw reads, trimmed reads, and the FastQC files.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb54-1"><a href="cleaning-data.html#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> ftp://igenome:G3nom3s4u@ussd-ftp.illumina.com/Mus_musculus/NCBI/build37.2/Mus_musculus_NCBI_build37.2.tar.gz</span></code></pre></div>
<p>This will take a long time to download because the file is a little less than 16GB zipped.</p>
<p>We notice here that we have a zipped tar file. To make this file easier to use, lets unzip it.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb55-1"><a href="cleaning-data.html#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tar</span> zxvf Mus_muculus_NCBI_build37.2.tar.gz</span></code></pre></div>
<p>Since Tophat is requiring <strong>*.bt21</strong> files (large index) and the files downloaded for the genome above are only small index files, we have to create a large index using <strong>bowtie2-build</strong>. For this, lets navigate to the WholeGenomeFasta folder within the extracted folder and then run <strong>bowtie2-build</strong>.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb56-1"><a href="cleaning-data.html#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/bowtie2-build</span> <span class="at">--large-index</span> genome.fa genome</span></code></pre></div>
<p>This process took about 26 minutes to run. Now lets copy the index files to a folder close to our reads so we can access them easier, rather than having to refer to the longer path where we build them. After they are copied to a new folder closer to our working directory, I went ahead and unzipped the trimmed read files to try and make the Tophat faster but it turned out not to work. The multicore call with <strong>-p</strong> didn’t use more cores than 1 until <strong>bowtie2-align-s</strong>, then 20 cores were used.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb57-1"><a href="cleaning-data.html#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/tophat</span> <span class="at">--no-converage-search</span> <span class="at">-p</span> 20 <span class="at">-G</span> Mus_musculus/NCBI/build37.2/Annotation/Archives/archive-2015-07-17-14-32-40/Genes/genes.gtf <span class="at">-0</span> 770_thout ./Index/genome 770_fp.fq.gz 770_rp.fq.gz 770_fu.fq 770_ru.fq</span></code></pre></div>
<p>This run took almost 3 hours to complete.. Running with 80 cores rather than 20 cores took just 4 minutes less, so the whole process must be limited by a single core and the core clock speed. The process does use close to 8,000% at its peak so there is a benefit to multicore, just isn’t very scalable.</p>
</div>
<div id="using-star-1" class="section level3" number="9.5.2">
<h3><span class="header-section-number">9.5.2</span> Using STAR</h3>
<p>STAR can be installed the same way as the previous programs with <strong>conda install</strong> (<strong><em>conda install -c bioconda star</em></strong>). In order to run STAR, we need to creaate indices just like with tophat, but STAR has this built in. I’m going to be using the same genome and GTF file as previously downloaded, but Dr. Ge uses a different zipped genome from the <em>gencode</em> database.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb58-1"><a href="cleaning-data.html#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/STAR</span> <span class="dt">\</span></span>
<span id="cb58-2"><a href="cleaning-data.html#cb58-2" aria-hidden="true" tabindex="-1"></a>--runThreadN 80 <span class="dt">\</span></span>
<span id="cb58-3"><a href="cleaning-data.html#cb58-3" aria-hidden="true" tabindex="-1"></a>--runMode genomeGenerate <span class="dt">\</span></span>
<span id="cb58-4"><a href="cleaning-data.html#cb58-4" aria-hidden="true" tabindex="-1"></a>--genomeDir starIndex <span class="dt">\</span></span>
<span id="cb58-5"><a href="cleaning-data.html#cb58-5" aria-hidden="true" tabindex="-1"></a>--genomeFastaFiles Index/genome.fa <span class="dt">\ </span>#same when we made the bowtie indices </span>
<span id="cb58-6"><a href="cleaning-data.html#cb58-6" aria-hidden="true" tabindex="-1"></a><span class="ex">--sjdbGTFfile</span> Mus_musculus/NCBI/build37.2/Annotation/Archives/archive-2015-07-17-14-32-40/Genes/genes.gtf</span></code></pre></div>
<p>With the index files made, we can start aligning with STAR. It’s important here than we only pick the paired end reads and not use all of the reads. Tophat is able to use all 4 reads but STAR doesn’t allow that, so we need to make sure that we feed in the large files from trimming.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb59-1"><a href="cleaning-data.html#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/STAR</span> <span class="at">--runThreadN</span> 80 <span class="at">--genomeDir</span> starIndex <span class="at">--readFilesIn</span> 770_fp.fq 770_rp.fq <span class="at">--outFilterIntronMotifs</span> RemoveNoncanonical <span class="at">--outFileNamePrefix</span> 2121770 <span class="at">--outSAMtype</span> BAM SortedByCoordinate</span></code></pre></div>
</div>
</div>
<div id="assembling-transcripts-with-cufflinks" class="section level2" number="9.6">
<h2><span class="header-section-number">9.6</span> Assembling transcripts with Cufflinks</h2>
<p>Once STAR is done running, we can assemble the transcripts with Cufflinks. This can also be installed with <strong>conda install</strong> (<strong><em>conda install -c bioconda cufflinks</em></strong>).</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb60-1"><a href="cleaning-data.html#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/cufflinks</span> <span class="at">-p</span> 20 <span class="at">-o</span> SRR2121771_clout <span class="at">--library-type</span> fr-firststrand 2121770Aligned.sortedByCoord.out.bam</span></code></pre></div>
</div>
<div id="checking-for-contamination" class="section level2" number="9.7">
<h2><span class="header-section-number">9.7</span> Checking for Contamination</h2>
<div id="phix-contamination" class="section level3" number="9.7.1">
<h3><span class="header-section-number">9.7.1</span> PhiX contamination</h3>
<p>Now we will look at what kind of contamination we are looking at. When samples are sequenced with Illumina, a PhiX control is run along side them. This control is for cluster generation, sequencing, alignment, and calibration for cross-talk matrix generation. We will use Bowtie to create a file to determine the PhiX contamination level.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb61-1"><a href="cleaning-data.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/bowtie2</span> <span class="at">-p</span> 20 <span class="at">-x</span> PhiX/Illumina/RTA/Sequence/Bowtie2Index/genome <span class="dt">\</span></span>
<span id="cb61-2"><a href="cleaning-data.html#cb61-2" aria-hidden="true" tabindex="-1"></a>-1 TrimmedReads/770_fp.fq  <span class="at">-2</span> TrimmedReads/770_rp.fq <span class="at">-S</span> phix.sam <span class="op">&amp;&gt;</span> PhiXout/SRR2121770_phix.out</span></code></pre></div>
<p>When the job is done running, the output file will show how much PhiX contamination we have. For example, lookin at the <strong>SRR2121770_phix.out</strong> created above, we see that 0.11% of the reads aligned with PhiX. The lower this value the better.</p>
</div>
<div id="rrna-sequences" class="section level3" number="9.7.2">
<h3><span class="header-section-number">9.7.2</span> rRNA Sequences</h3>
<p>To retreive the rRNA sequences for mouse, we need to search the taxonomy database on NCBI for <em>Mus musculus</em>. Click on <em>Mus musculus</em> on the next page, and then the top <em>Mus musculus</em> at the head of the list. Now, select the top subtree link in the <strong>Nucleotide</strong> database. Select rRNA sequences on the left side of the page and download full list just downloading with Send &gt; Complete Record &gt; File &gt; FASTA &gt; Create File. Drag the file using WinSCP to the raw folder on the cluster and rename it to rRNA.fa.</p>
<p>We are going to need to install <strong>bwa</strong> with conda in order to get the alignments to work. This can be done with <strong><em>conda install -c bioconda bwa</em></strong>. Following this, we will need to make indixes for the rRNA that we downloaded. To make this more clean, lets make a directory for the rRNA sequences that we downloaded and the indices that we make.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb62-1"><a href="cleaning-data.html#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> rRNA</span></code></pre></div>
<p>Then we move the <strong>rRNA.fa</strong> to the new <strong>rRNA</strong> folder with WinSCP and then we can run the bwa.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb63-1"><a href="cleaning-data.html#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="bu">time</span> ~/miniconda2/bin/bwa mem <span class="at">-t</span> 20 rRNA/rRNA.fa TrimmedReads/770_fp.fq TrimmedReads/770_rp.fq <span class="op">&gt;</span> rnaAlign/770_rna.sam</span></code></pre></div>
<p>When we are done creating the new <em>*.sam</em> files for all of the forward/reverse read combinations, we can use samtools to convert the <em>*.sam</em> file to <em>*.bam</em> files which are essentially the same file just that <em>sam</em> is easier for us to look at while <em>bam</em> is binary. Samtools can be installed with <strong><em>conda install -c bioconda samtools</em></strong>.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb64-1"><a href="cleaning-data.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/samtools</span> view <span class="at">-@</span> 10 <span class="at">-bS</span> <span class="at">-o</span> rnaAlign/770_rna.bam rnaAlign/770.sam</span></code></pre></div>
<p>Now in the rnaAlign folder we have our sam and bam file for each of the libraries. Lets create an output file with <em>flagstat</em>.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb65-1"><a href="cleaning-data.html#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/samtools</span> flagstat <span class="at">-@</span> 10 rnaAlign/770_rna.out</span></code></pre></div>
<p>Wihtin this file we will be able to see the summary of our alignments to the rRNA file that we downloaded from NCBI.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb66-1"><a href="cleaning-data.html#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co">#From 770_rna.out</span></span>
<span id="cb66-2"><a href="cleaning-data.html#cb66-2" aria-hidden="true" tabindex="-1"></a><span class="ex">205559289</span> + 0 in total <span class="er">(</span><span class="ex">QC-passed</span> reads + QC-failed reads<span class="kw">)</span></span>
<span id="cb66-3"><a href="cleaning-data.html#cb66-3" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> + 0 secondary</span>
<span id="cb66-4"><a href="cleaning-data.html#cb66-4" aria-hidden="true" tabindex="-1"></a><span class="ex">85</span> + 0 supplementary</span>
<span id="cb66-5"><a href="cleaning-data.html#cb66-5" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> + 0 duplicates</span>
<span id="cb66-6"><a href="cleaning-data.html#cb66-6" aria-hidden="true" tabindex="-1"></a><span class="ex">4265179</span> + 0 mapped <span class="er">(</span><span class="ex">2.07%</span> : N/A<span class="kw">)</span></span>
<span id="cb66-7"><a href="cleaning-data.html#cb66-7" aria-hidden="true" tabindex="-1"></a><span class="ex">205559204</span> + 0 paired in sequencing</span>
<span id="cb66-8"><a href="cleaning-data.html#cb66-8" aria-hidden="true" tabindex="-1"></a><span class="ex">102779602</span> + 0 read1</span>
<span id="cb66-9"><a href="cleaning-data.html#cb66-9" aria-hidden="true" tabindex="-1"></a><span class="ex">102779602</span> + 0 read2</span>
<span id="cb66-10"><a href="cleaning-data.html#cb66-10" aria-hidden="true" tabindex="-1"></a><span class="ex">4151684</span> + 0 properly paired <span class="er">(</span><span class="ex">2.02%</span> : N/A<span class="kw">)</span></span>
<span id="cb66-11"><a href="cleaning-data.html#cb66-11" aria-hidden="true" tabindex="-1"></a><span class="ex">4187608</span> + 0 with itself and mate mapped</span>
<span id="cb66-12"><a href="cleaning-data.html#cb66-12" aria-hidden="true" tabindex="-1"></a><span class="ex">77486</span> + 0 singletons <span class="er">(</span><span class="ex">0.04%</span> : N/A<span class="kw">)</span></span>
<span id="cb66-13"><a href="cleaning-data.html#cb66-13" aria-hidden="true" tabindex="-1"></a><span class="ex">4222</span> + 0 with mate mapped to a different chr</span>
<span id="cb66-14"><a href="cleaning-data.html#cb66-14" aria-hidden="true" tabindex="-1"></a><span class="ex">1026</span> + 0 with mate mapped to a different chr <span class="er">(</span><span class="ex">mapQ</span><span class="op">&gt;</span>=5<span class="kw">)</span></span></code></pre></div>
</div>
<div id="bacterial-contamination" class="section level3" number="9.7.3">
<h3><span class="header-section-number">9.7.3</span> Bacterial contamination</h3>
<p>In order to find out the contamination, we need to install Kraken2 with <strong><em>conda install -c bioconda kraken2</em></strong> and download a pre-built database containing bacteria, archaea, and viral sequences. The database we are going to download only contains about 5% of k-mers from the original database (but directions are sort of lacking to build an entirely new database). More information can be found at <a href="https://ccb.jhu.edu/software/kraken/" class="uri">https://ccb.jhu.edu/software/kraken/</a> for the pre-built databases.</p>
<p>Using the code in the next chunk will download the 8GB database and then extract the files so we can use them with the <strong>Kraken2</strong> program. Lets do this in the main project folder.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb67-1"><a href="cleaning-data.html#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> ftp://ftp.ccb.jhu.edu/pub/data/kraken2_dbs/minikraken2_v2_8GB_201904_UPDATE.tgz</span>
<span id="cb67-2"><a href="cleaning-data.html#cb67-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tar</span> xzf minikraken2_v2_8GB_201904_UPDATE.tgz</span></code></pre></div>
<p>Now lets make a directory for the output.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb68-1"><a href="cleaning-data.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> krakenOut</span></code></pre></div>
<p>We can call <strong>Kraken</strong> with the extracted database folder and point it to the location of out paired end reads from trimming and to the output folder that we just created for the outputs.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb69-1"><a href="cleaning-data.html#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/kraken2</span> <span class="at">--db</span> minikraken2_v2_8GB_201904_UPDATE/ <span class="at">--output</span> krakenOut/770.out <span class="at">--threads</span> 10 <span class="at">--paired</span> TrimmedReads/770_fp.fq TrimmedReads/770_rp.fq</span></code></pre></div>
<p>When Kraken is done running, it will print out the number (and percentage) of reads that were classified. In this case, we have used 102779602 sequences, of which 19142843 sequences were classified (18.63%) and 83636759 sequences were unclassified (81.37%). My interpretation of this is that 18.63% of the reads are possibly from microbial cell contamination.</p>
</div>
</div>
<div id="counting-transcripts" class="section level2" number="9.8">
<h2><span class="header-section-number">9.8</span> Counting Transcripts</h2>
<p>Since we have the <strong>bam</strong> files from the alignments of the different samples, we can count the features for each and get the transcipt counts using featureCounts form <strong><em>conda install -c bioconda/label/cf201901 subread</em></strong>. The genome and annotations that we previously downloaded were from genome <strong>mm9</strong> so we have to specify to <em>featureCounts</em> what we want to actually count. FeatureCounts defaults to using <strong>gene_id</strong> which our output bam files don’t have described correctly for <em>featureCounts</em> to read them. This is a single line of code because we can use a wildcard to run through all of the <strong>bam</strong> files.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb70-1"><a href="cleaning-data.html#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Move to the Star Alignment output folder for a working directory</span></span>
<span id="cb70-2"><a href="cleaning-data.html#cb70-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> StarOut</span>
<span id="cb70-3"><a href="cleaning-data.html#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/featureCounts</span> <span class="at">-a</span> /gpfs/scratch/alex.soupir/Mus/raw/Mus_musculus/NCBI/build37.2/Annotation/Archives/archive-2015-07-17-14-32-40/Genes/genes.gtf <span class="at">-g</span> <span class="st">&#39;transcript_id&#39;</span> <span class="at">-o</span> readCounds.txt <span class="pp">*</span>bam</span></code></pre></div>
<p>With the files that we are working with, this will take between 3.5 minutes to 5 minutes per <strong>bam</strong> file. The output will be a file that can be imported into excel and saved as csv which we then can work with in R.</p>
<div id="final-qc-of-cleaning-the-data" class="section level3" number="9.8.1">
<h3><span class="header-section-number">9.8.1</span> Final QC of cleaning the data</h3>
<p>Lets look at the data that we have collected from all of the MultiQC runs that we had with initial FastQC, Trimmomatic, STAR alignment, PhiX contamination, rRNA contamination, and the final feature counts.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="cleaning-data.html#cb71-1" aria-hidden="true" tabindex="-1"></a>qc <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&#39;Whole Data QC.csv&#39;</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">na.strings=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb71-2"><a href="cleaning-data.html#cb71-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(qc) <span class="sc">%&gt;%</span></span>
<span id="cb71-3"><a href="cleaning-data.html#cb71-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb71-4"><a href="cleaning-data.html#cb71-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;800px&quot;</span>)</span></code></pre></div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="background.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differentially-expressed-sequence-identification.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Bioinformatics_RNA-seq.pdf", "Bioinformatics_RNA-seq.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
