<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Differentially Expressed Sequence Identification | Bioinformatics_RNA-seq</title>
  <meta name="description" content="RNA-seq notes and trials" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Differentially Expressed Sequence Identification | Bioinformatics_RNA-seq" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="RNA-seq notes and trials" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Differentially Expressed Sequence Identification | Bioinformatics_RNA-seq" />
  
  <meta name="twitter:description" content="RNA-seq notes and trials" />
  

<meta name="author" content="J. Marcelo Rosales R." />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="cleaning-data.html"/>
<link rel="next" href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to RNA sequencing (RNA_seq)</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#downloading-and-setting-up-conda-environments."><i class="fa fa-check"></i><b>1.1</b> Downloading and setting up conda environments.</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#install-miniconda-or-anaconda."><i class="fa fa-check"></i><b>1.2</b> Install Miniconda, or Anaconda.</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#conda-terminal-commands."><i class="fa fa-check"></i><b>1.2.1</b> Conda terminal commands.</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#anacondaminiconda-install-summary-of-steps."><i class="fa fa-check"></i><b>1.2.2</b> Anaconda/Miniconda install Summary of Steps.</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#conda-activate-error"><i class="fa fa-check"></i><b>1.2.3</b> Conda activate error</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#alignment-procedures-and-files."><i class="fa fa-check"></i><b>1.3</b> Alignment Procedures and files.</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#using-tophat-early-version"><i class="fa fa-check"></i><b>1.3.1</b> Using Tophat (early version)</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#using-tophat"><i class="fa fa-check"></i><b>1.3.2</b> Using Tophat</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#using-star"><i class="fa fa-check"></i><b>1.3.3</b> Using STAR</a></li>
<li class="chapter" data-level="1.3.4" data-path="index.html"><a href="index.html#custompersonal-way-to-download-the-mus_musculus_ncbi_build37.2.tar.gz-file."><i class="fa fa-check"></i><b>1.3.4</b> Custom/personal way to Download the Mus_musculus_NCBI_build37.2.tar.gz file.</a></li>
<li class="chapter" data-level="1.3.5" data-path="index.html"><a href="index.html#get-the-sample-files."><i class="fa fa-check"></i><b>1.3.5</b> Get the Sample files.</a></li>
<li class="chapter" data-level="1.3.6" data-path="index.html"><a href="index.html#starting-to-analize-fastq.gz-files"><i class="fa fa-check"></i><b>1.3.6</b> Starting to analize fastq.gz files</a></li>
<li class="chapter" data-level="1.3.7" data-path="index.html"><a href="index.html#trimming-with-trimmomatic"><i class="fa fa-check"></i><b>1.3.7</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="1.3.8" data-path="index.html"><a href="index.html#using-star-to-map-genes"><i class="fa fa-check"></i><b>1.3.8</b> Using STAR to map genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html"><i class="fa fa-check"></i><b>2</b> WGCNA RNA-seq Blog</a>
<ul>
<li class="chapter" data-level="2.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#WGCNA"><i class="fa fa-check"></i><b>2.1</b> Introduction to the WGCNA Rpackage</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-rpackage-installation-procedure."><i class="fa fa-check"></i><b>2.1.1</b> WGCNA Rpackage Installation procedure.</a></li>
<li class="chapter" data-level="2.1.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#final-installation-workflow"><i class="fa fa-check"></i><b>2.1.2</b> Final installation workflow:</a></li>
<li class="chapter" data-level="2.1.3" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-tutorial."><i class="fa fa-check"></i><b>2.1.3</b> WGCNA tutorial.</a></li>
<li class="chapter" data-level="2.1.4" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#gene-expression-analysis"><i class="fa fa-check"></i><b>2.1.4</b> 3.4 Gene expression analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#rna-seq-analisys-with-kakus-data-sample."><i class="fa fa-check"></i><b>2.2</b> RNA-seq Analisys with Kaku’s Data sample.</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#data-analysis."><i class="fa fa-check"></i><b>2.2.1</b> Data Analysis.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="sra-download-qc-and-trimming.html"><a href="sra-download-qc-and-trimming.html"><i class="fa fa-check"></i><b>3</b> SRA Download, QC, and Trimming</a></li>
<li class="chapter" data-level="4" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>4</b> Methods</a></li>
<li class="chapter" data-level="5" data-path="youtube-references-.html"><a href="youtube-references-.html"><i class="fa fa-check"></i><b>5</b> Youtube references.</a></li>
<li class="chapter" data-level="6" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>6</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="7" data-path="rna-seq-for-stat736-alex-soupir.html"><a href="rna-seq-for-stat736-alex-soupir.html"><i class="fa fa-check"></i><b>7</b> RNA Seq for STAT736 Alex Soupir</a></li>
<li class="chapter" data-level="8" data-path="background.html"><a href="background.html"><i class="fa fa-check"></i><b>8</b> <strong>Background</strong></a></li>
<li class="chapter" data-level="9" data-path="cleaning-data.html"><a href="cleaning-data.html"><i class="fa fa-check"></i><b>9</b> <strong>Cleaning Data</strong></a>
<ul>
<li class="chapter" data-level="9.1" data-path="cleaning-data.html"><a href="cleaning-data.html#programs-used"><i class="fa fa-check"></i><b>9.1</b> Programs used?</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="cleaning-data.html"><a href="cleaning-data.html#picking-the-right-node"><i class="fa fa-check"></i><b>9.1.1</b> Picking the right node</a></li>
<li class="chapter" data-level="9.1.2" data-path="cleaning-data.html"><a href="cleaning-data.html#creating-slurm-scripts"><i class="fa fa-check"></i><b>9.1.2</b> Creating slurm scripts</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="cleaning-data.html"><a href="cleaning-data.html#acquiring-sequences"><i class="fa fa-check"></i><b>9.2</b> Acquiring sequences</a></li>
<li class="chapter" data-level="9.3" data-path="cleaning-data.html"><a href="cleaning-data.html#fastqc"><i class="fa fa-check"></i><b>9.3</b> FastQC</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="cleaning-data.html"><a href="cleaning-data.html#multiqc"><i class="fa fa-check"></i><b>9.3.1</b> MultiQC</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="cleaning-data.html"><a href="cleaning-data.html#trimming-with-trimmomatic-1"><i class="fa fa-check"></i><b>9.4</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="9.5" data-path="cleaning-data.html"><a href="cleaning-data.html#alignment"><i class="fa fa-check"></i><b>9.5</b> Alignment</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="cleaning-data.html"><a href="cleaning-data.html#using-tophat-1"><i class="fa fa-check"></i><b>9.5.1</b> Using Tophat</a></li>
<li class="chapter" data-level="9.5.2" data-path="cleaning-data.html"><a href="cleaning-data.html#using-star-1"><i class="fa fa-check"></i><b>9.5.2</b> Using STAR</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="cleaning-data.html"><a href="cleaning-data.html#assembling-transcripts-with-cufflinks"><i class="fa fa-check"></i><b>9.6</b> Assembling transcripts with Cufflinks</a></li>
<li class="chapter" data-level="9.7" data-path="cleaning-data.html"><a href="cleaning-data.html#checking-for-contamination"><i class="fa fa-check"></i><b>9.7</b> Checking for Contamination</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="cleaning-data.html"><a href="cleaning-data.html#phix-contamination"><i class="fa fa-check"></i><b>9.7.1</b> PhiX contamination</a></li>
<li class="chapter" data-level="9.7.2" data-path="cleaning-data.html"><a href="cleaning-data.html#rrna-sequences"><i class="fa fa-check"></i><b>9.7.2</b> rRNA Sequences</a></li>
<li class="chapter" data-level="9.7.3" data-path="cleaning-data.html"><a href="cleaning-data.html#bacterial-contamination"><i class="fa fa-check"></i><b>9.7.3</b> Bacterial contamination</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="cleaning-data.html"><a href="cleaning-data.html#counting-transcripts"><i class="fa fa-check"></i><b>9.8</b> Counting Transcripts</a>
<ul>
<li class="chapter" data-level="9.8.1" data-path="cleaning-data.html"><a href="cleaning-data.html#final-qc-of-cleaning-the-data"><i class="fa fa-check"></i><b>9.8.1</b> Final QC of cleaning the data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="differentially-expressed-sequence-identification.html"><a href="differentially-expressed-sequence-identification.html"><i class="fa fa-check"></i><b>10</b> <strong>Differentially Expressed Sequence Identification</strong></a>
<ul>
<li class="chapter" data-level="10.1" data-path="differentially-expressed-sequence-identification.html"><a href="differentially-expressed-sequence-identification.html#analyzing-reads-counts"><i class="fa fa-check"></i><b>10.1</b> Analyzing Reads Counts</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="differentially-expressed-sequence-identification.html"><a href="differentially-expressed-sequence-identification.html#quick-data-exploration"><i class="fa fa-check"></i><b>10.1.1</b> Quick Data Exploration</a></li>
<li class="chapter" data-level="10.1.2" data-path="differentially-expressed-sequence-identification.html"><a href="differentially-expressed-sequence-identification.html#filtering-normalization-and-trasformation-using-deseq2"><i class="fa fa-check"></i><b>10.1.2</b> Filtering, Normalization, and Trasformation using DESeq2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><a href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><i class="fa fa-check"></i><b>11</b> Interactions cause a difference between the lfc betwen pooled data, e.g. p53+/+ (control and IR) and p53-/- (control and IR)</a>
<ul>
<li class="chapter" data-level="11.0.1" data-path="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><a href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>11.0.1</b> Exploratory Data Analysis</a></li>
<li class="chapter" data-level="11.0.2" data-path="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><a href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html#identification-of-differentially-expressed-genes"><i class="fa fa-check"></i><b>11.0.2</b> Identification of Differentially Expressed Genes</a></li>
<li class="chapter" data-level="11.0.3" data-path="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><a href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html#gene-annotations"><i class="fa fa-check"></i><b>11.0.3</b> Gene Annotations</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="go-enrichment-analysis-using-gostats.html"><a href="go-enrichment-analysis-using-gostats.html"><i class="fa fa-check"></i><b>12</b> 7. GO Enrichment analysis using GOstats</a>
<ul>
<li class="chapter" data-level="12.1" data-path="go-enrichment-analysis-using-gostats.html"><a href="go-enrichment-analysis-using-gostats.html#go-enrichment-analysis-of-downregulated-genes"><i class="fa fa-check"></i><b>12.1</b> GO Enrichment analysis of downregulated genes</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html"><i class="fa fa-check"></i><b>13</b> 8. Pathway analysis using expression data</a>
<ul>
<li class="chapter" data-level="13.1" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#prepare-data"><i class="fa fa-check"></i><b>13.1</b> Prepare data</a></li>
<li class="chapter" data-level="13.2" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#kegg-pathways"><i class="fa fa-check"></i><b>13.2</b> KEGG pathways</a></li>
<li class="chapter" data-level="13.3" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-oxidative-phosphorylation."><i class="fa fa-check"></i><b>13.3</b> Pathway and regulation of genes for Oxidative phosphorylation.</a></li>
<li class="chapter" data-level="13.4" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-glycosylphosphatidylinositolgpi-anchor-biosynthesis."><i class="fa fa-check"></i><b>13.4</b> Pathway and regulation of genes for Glycosylphosphatidylinositol(GPI)-anchor biosynthesis.</a></li>
<li class="chapter" data-level="13.5" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-rna-polymerase."><i class="fa fa-check"></i><b>13.5</b> Pathway and regulation of genes for RNA polymerase.</a></li>
<li class="chapter" data-level="13.6" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-nucleotide-excision-repair."><i class="fa fa-check"></i><b>13.6</b> Pathway and regulation of genes for Nucleotide excision repair.</a></li>
<li class="chapter" data-level="13.7" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-non-homologous-end-joining."><i class="fa fa-check"></i><b>13.7</b> Pathway and regulation of genes for Non-homologous end-joining.</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics_RNA-seq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differentially-expressed-sequence-identification" class="section level1" number="10">
<h1><span class="header-section-number">Chapter 10</span> <strong>Differentially Expressed Sequence Identification</strong></h1>
<p>Programs Used</p>
<ul>
<li>R</li>
<li>RStudio</li>
</ul>
<p>Packages</p>
<ul>
<li>readr (<strong><em>install.packages(‘readr’)</em></strong>)</li>
<li>limma (<strong><em>BiocManager::install(‘limma’)</em></strong>)</li>
<li>DESeq2 (<strong><em>BiocManager::install(‘DESeq2’)</em></strong>)</li>
<li>dplyr (<strong><em>install.packages(“dplyr”)</em></strong>)</li>
<li>ggplot2 (<strong><em>install.packages(“ggplot2”)</em></strong>)</li>
<li>gplots (<strong><em>install.packages(“gplots”)</em></strong>)</li>
<li>Annotations (<strong><em>BiocManager::install(‘AnnotationDbi’)</em></strong>)</li>
<li>org.Hs.eg.db (<strong><em>BiocManager::install(‘org.Hs.eg.db’)</em></strong>)
<ul>
<li>This is for Human</li>
</ul></li>
<li>org.Mm.eg.db (<strong><em>BiocManager::install(‘org.Mm.eg.db’)</em></strong>)
<ul>
<li>This is for Mouse</li>
</ul></li>
<li>ggrepel (<strong><em>install.packages(“ggrepel”)</em></strong>)</li>
<li>ReportingTools (<strong><em>BiocManager::install(‘ReportingTools’)</em></strong>)</li>
<li>GO.db (<strong><em>BiocManager::install(‘GO.db’)</em></strong>)</li>
<li>GOstats (<strong><em>BiocManager::install(‘GOstats’)</em></strong>)</li>
<li>pathview (<strong><em>BiocManager::install(‘pathview’)</em></strong>)</li>
<li>gage (<strong><em>BiocManager::install(‘gage’)</em></strong>)</li>
<li>gageData (<strong><em>BiocManager::install(‘gageData’)</em></strong>)</li>
<li>select (<strong><em>BiocManager::install(‘Select’)</em></strong>)</li>
</ul>
<p>With these, you most certainly will have to step through each and install extra things when you start calling the packages. Take it step by step to ensure that each dependency is installed.</p>
<div id="analyzing-reads-counts" class="section level2" number="10.1">
<h2><span class="header-section-number">10.1</span> Analyzing Reads Counts</h2>
<p>When the count file is completed, we can import it into R and start working with it to determine differentially expressed genes. First we will import it into R</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="differentially-expressed-sequence-identification.html#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb72-2"><a href="differentially-expressed-sequence-identification.html#cb72-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb72-3"><a href="differentially-expressed-sequence-identification.html#cb72-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb72-4"><a href="differentially-expressed-sequence-identification.html#cb72-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb72-5"><a href="differentially-expressed-sequence-identification.html#cb72-5" aria-hidden="true" tabindex="-1"></a>countData <span class="ot">=</span> <span class="fu">read_csv</span>(<span class="st">&quot;readCounts.csv&quot;</span>, <span class="at">skip =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>This gives us our dataframe from out featureCounts program, but if we look at the data we see that featureCounts added some extra information that characterizes each gene_id.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="differentially-expressed-sequence-identification.html#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(countData)) <span class="sc">%&gt;%</span></span>
<span id="cb73-2"><a href="differentially-expressed-sequence-identification.html#cb73-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb73-3"><a href="differentially-expressed-sequence-identification.html#cb73-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;320px&quot;</span>)</span></code></pre></div>
<p>We also need to set out row names to the gene_id. We will do some data frame manipulation and then look at the data again.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="differentially-expressed-sequence-identification.html#cb74-1" aria-hidden="true" tabindex="-1"></a>countData <span class="ot">=</span> <span class="fu">as.data.frame</span>(countData)</span>
<span id="cb74-2"><a href="differentially-expressed-sequence-identification.html#cb74-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(countData) <span class="ot">=</span> countData<span class="sc">$</span>Geneid</span>
<span id="cb74-3"><a href="differentially-expressed-sequence-identification.html#cb74-3" aria-hidden="true" tabindex="-1"></a>countData <span class="ot">=</span> countData[,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>)]</span>
<span id="cb74-4"><a href="differentially-expressed-sequence-identification.html#cb74-4" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(countData)) <span class="sc">%&gt;%</span></span>
<span id="cb74-5"><a href="differentially-expressed-sequence-identification.html#cb74-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb74-6"><a href="differentially-expressed-sequence-identification.html#cb74-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span></code></pre></div>
<div id="quick-data-exploration" class="section level3" number="10.1.1">
<h3><span class="header-section-number">10.1.1</span> Quick Data Exploration</h3>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="differentially-expressed-sequence-identification.html#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(countData)</span></code></pre></div>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="differentially-expressed-sequence-identification.html#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">summary</span>(countData)) <span class="sc">%&gt;%</span></span>
<span id="cb76-2"><a href="differentially-expressed-sequence-identification.html#cb76-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb76-3"><a href="differentially-expressed-sequence-identification.html#cb76-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span></code></pre></div>
<p>Let’s also go ahead and change the names to describe out data a little better.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="differentially-expressed-sequence-identification.html#cb77-1" aria-hidden="true" tabindex="-1"></a>columns <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;Trp53m_mock_1&#39;</span>, <span class="st">&#39;Trp53m_mock_2&#39;</span>, <span class="st">&#39;Trp53m_4h7Gy_1&#39;</span>, <span class="st">&#39;Trp53m_4h7Gy_2&#39;</span>, <span class="st">&#39;Trp53p_mock_1&#39;</span>, <span class="st">&#39;Trp53p_mock_2&#39;</span>, <span class="st">&#39;Trp53p_mock_3&#39;</span>, <span class="st">&#39;Trp53p_mock_4&#39;</span>, <span class="st">&#39;Trp53p_4h7Gy_1&#39;</span>, <span class="st">&#39;Trp53p_4h7Gy_2&#39;</span>, <span class="st">&#39;Trp53p_4h7Gy_3&#39;</span>, <span class="st">&#39;Trp53p_4h7Gy_4&#39;</span>)</span>
<span id="cb77-2"><a href="differentially-expressed-sequence-identification.html#cb77-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(countData) <span class="ot">=</span> columns</span></code></pre></div>
<p>Here we have to make sure that we convert the <em>+/+</em> and <em>-/-</em> to characters. These characters</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="differentially-expressed-sequence-identification.html#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">8</span>,<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">1</span>)<span class="sc">+</span><span class="fl">0.1</span>)</span>
<span id="cb78-2"><a href="differentially-expressed-sequence-identification.html#cb78-2" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>( <span class="fu">colSums</span>(countData)<span class="sc">/</span><span class="fl">1e6</span>, <span class="at">col=</span><span class="st">&quot;green&quot;</span>,<span class="at">las=</span><span class="dv">3</span>,<span class="at">main=</span><span class="st">&quot;Total read counts (millions)&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;Total read counts in millions&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="differentially-expressed-sequence-identification.html#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(countData[,<span class="dv">1</span>], <span class="at">br=</span><span class="dv">200</span>, <span class="at">xlab=</span><span class="st">&quot;Number of Reads Counts per Feature&quot;</span>, <span class="at">main=</span><span class="st">&quot;Histogram of Read Counts for Trp53-/- Mock&quot;</span>)</span></code></pre></div>
<p>We can see that our count data is highly skewed to the right. This is a great case for using <strong>log</strong> transformation!</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="differentially-expressed-sequence-identification.html#cb80-1" aria-hidden="true" tabindex="-1"></a>logCountData <span class="ot">=</span> <span class="fu">log2</span>(<span class="dv">1</span><span class="sc">+</span>countData)</span>
<span id="cb80-2"><a href="differentially-expressed-sequence-identification.html#cb80-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="at">mar=</span><span class="fu">c</span>(<span class="dv">8</span>,<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">1</span>))  <span class="co"># two columns</span></span>
<span id="cb80-3"><a href="differentially-expressed-sequence-identification.html#cb80-3" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(logCountData[,<span class="dv">1</span>], <span class="at">main=</span><span class="st">&quot;Histogram of Log Read Counts&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;Log transformed counts&quot;</span>)</span>
<span id="cb80-4"><a href="differentially-expressed-sequence-identification.html#cb80-4" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(logCountData,<span class="at">las=</span><span class="dv">3</span>, <span class="at">main=</span><span class="st">&quot;Boxplot of Log Read Counts&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="differentially-expressed-sequence-identification.html#cb81-1" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> logCountData</span>
<span id="cb81-2"><a href="differentially-expressed-sequence-identification.html#cb81-2" aria-hidden="true" tabindex="-1"></a>myColors <span class="ot">=</span> <span class="fu">rainbow</span>(<span class="fu">dim</span>(x)[<span class="dv">2</span>])</span>
<span id="cb81-3"><a href="differentially-expressed-sequence-identification.html#cb81-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">density</span>(x[,<span class="dv">1</span>]),<span class="at">col =</span> myColors[<span class="dv">1</span>], <span class="at">lwd=</span><span class="dv">2</span>,</span>
<span id="cb81-4"><a href="differentially-expressed-sequence-identification.html#cb81-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">&quot;Expresson values&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;Density&quot;</span>, <span class="at">main=</span> <span class="st">&quot;Distribution of transformed data&quot;</span>,</span>
<span id="cb81-5"><a href="differentially-expressed-sequence-identification.html#cb81-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(<span class="fu">density</span>(x[,<span class="dv">1</span>])<span class="sc">$</span>y)<span class="sc">+</span>.<span class="dv">02</span> ) )</span>
<span id="cb81-6"><a href="differentially-expressed-sequence-identification.html#cb81-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb81-7"><a href="differentially-expressed-sequence-identification.html#cb81-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">dim</span>(x)[<span class="dv">2</span>] )</span>
<span id="cb81-8"><a href="differentially-expressed-sequence-identification.html#cb81-8" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">density</span>(x[,i]),<span class="at">col=</span>myColors[i], <span class="at">lwd=</span><span class="dv">2</span>)</span>
<span id="cb81-9"><a href="differentially-expressed-sequence-identification.html#cb81-9" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>, <span class="at">cex=</span><span class="fl">1.1</span>,<span class="fu">colnames</span>(x), <span class="at">lty=</span><span class="fu">rep</span>(<span class="dv">1</span>,<span class="fu">dim</span>(x)[<span class="dv">2</span>]), <span class="at">col=</span>myColors )    </span></code></pre></div>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="differentially-expressed-sequence-identification.html#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(logCountData[,<span class="dv">1</span>],logCountData[,<span class="dv">2</span>], <span class="at">xlab=</span><span class="st">&quot;Trp53-/- mock replication 1&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;Trp53-/- mock replication 2&quot;</span>)</span></code></pre></div>
</div>
<div id="filtering-normalization-and-trasformation-using-deseq2" class="section level3" number="10.1.2">
<h3><span class="header-section-number">10.1.2</span> Filtering, Normalization, and Trasformation using DESeq2</h3>
<p>We have to make the experiment design into a small dataframe so we can tell DESeq how we want to analyze the data. Here will will make a small table that has the rep names that we changed the column names to previously, and then a column for which columns are Trp53+/+ or Trp53-/-, and which columns were control mice and which columns were treated with ionizing radiation.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="differentially-expressed-sequence-identification.html#cb83-1" aria-hidden="true" tabindex="-1"></a>detectGroups <span class="ot">&lt;-</span> <span class="cf">function</span> (x){  <span class="co"># x are col names</span></span>
<span id="cb83-2"><a href="differentially-expressed-sequence-identification.html#cb83-2" aria-hidden="true" tabindex="-1"></a>  tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;[0-9]*$&quot;</span>,<span class="st">&quot;&quot;</span>,x) <span class="co"># Remove all numbers from end</span></span>
<span id="cb83-3"><a href="differentially-expressed-sequence-identification.html#cb83-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#tem = gsub(&quot;_Rep|_rep|_REP&quot;,&quot;&quot;,tem)</span></span>
<span id="cb83-4"><a href="differentially-expressed-sequence-identification.html#cb83-4" aria-hidden="true" tabindex="-1"></a>  tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_$&quot;</span>,<span class="st">&quot;&quot;</span>,tem); <span class="co"># remove &quot;_&quot; from end</span></span>
<span id="cb83-5"><a href="differentially-expressed-sequence-identification.html#cb83-5" aria-hidden="true" tabindex="-1"></a>  tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_Rep$&quot;</span>,<span class="st">&quot;&quot;</span>,tem); <span class="co"># remove &quot;_Rep&quot; from end</span></span>
<span id="cb83-6"><a href="differentially-expressed-sequence-identification.html#cb83-6" aria-hidden="true" tabindex="-1"></a>  tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_rep$&quot;</span>,<span class="st">&quot;&quot;</span>,tem); <span class="co"># remove &quot;_rep&quot; from end</span></span>
<span id="cb83-7"><a href="differentially-expressed-sequence-identification.html#cb83-7" aria-hidden="true" tabindex="-1"></a>  tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_REP$&quot;</span>,<span class="st">&quot;&quot;</span>,tem)  <span class="co"># remove &quot;_REP&quot; from end</span></span>
<span id="cb83-8"><a href="differentially-expressed-sequence-identification.html#cb83-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>( tem )</span>
<span id="cb83-9"><a href="differentially-expressed-sequence-identification.html#cb83-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb83-10"><a href="differentially-expressed-sequence-identification.html#cb83-10" aria-hidden="true" tabindex="-1"></a>groups <span class="ot">=</span> <span class="fu">as.character</span> ( <span class="fu">detectGroups</span>( <span class="fu">colnames</span>( countData ) ) )</span>
<span id="cb83-11"><a href="differentially-expressed-sequence-identification.html#cb83-11" aria-hidden="true" tabindex="-1"></a>groups</span>
<span id="cb83-12"><a href="differentially-expressed-sequence-identification.html#cb83-12" aria-hidden="true" tabindex="-1"></a>p53 <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;m&quot;</span>, <span class="st">&quot;m&quot;</span>, <span class="st">&quot;m&quot;</span>, <span class="st">&quot;m&quot;</span>,</span>
<span id="cb83-13"><a href="differentially-expressed-sequence-identification.html#cb83-13" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;p&quot;</span>, <span class="st">&quot;p&quot;</span>, <span class="st">&quot;p&quot;</span>, <span class="st">&quot;p&quot;</span>, <span class="st">&quot;p&quot;</span>, <span class="st">&quot;p&quot;</span>, <span class="st">&quot;p&quot;</span>, <span class="st">&quot;p&quot;</span>)</span>
<span id="cb83-14"><a href="differentially-expressed-sequence-identification.html#cb83-14" aria-hidden="true" tabindex="-1"></a>treatment <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;control&quot;</span>, <span class="st">&quot;control&quot;</span>, <span class="st">&quot;IR&quot;</span>, <span class="st">&quot;IR&quot;</span>, <span class="st">&quot;control&quot;</span>, <span class="st">&quot;control&quot;</span>, <span class="st">&quot;control&quot;</span>, <span class="st">&quot;control&quot;</span>,</span>
<span id="cb83-15"><a href="differentially-expressed-sequence-identification.html#cb83-15" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;IR&quot;</span>, <span class="st">&quot;IR&quot;</span>, <span class="st">&quot;IR&quot;</span>, <span class="st">&quot;IR&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="differentially-expressed-sequence-identification.html#cb84-1" aria-hidden="true" tabindex="-1"></a>colData <span class="ot">=</span> <span class="fu">cbind</span>(<span class="fu">colnames</span>(countData), p53 )</span>
<span id="cb84-2"><a href="differentially-expressed-sequence-identification.html#cb84-2" aria-hidden="true" tabindex="-1"></a>colData</span></code></pre></div>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="differentially-expressed-sequence-identification.html#cb85-1" aria-hidden="true" tabindex="-1"></a>colData <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="fu">colnames</span>(countData), p53, treatment))</span>
<span id="cb85-2"><a href="differentially-expressed-sequence-identification.html#cb85-2" aria-hidden="true" tabindex="-1"></a>colData</span>
<span id="cb85-3"><a href="differentially-expressed-sequence-identification.html#cb85-3" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(colData)</span></code></pre></div>
<p>Creating a DESeq Dataset</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="differentially-expressed-sequence-identification.html#cb86-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">=</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData=</span>countData,</span>
<span id="cb86-2"><a href="differentially-expressed-sequence-identification.html#cb86-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">colData=</span>colData,</span>
<span id="cb86-3"><a href="differentially-expressed-sequence-identification.html#cb86-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">design=</span> <span class="sc">~</span> p53<span class="sc">+</span>treatment<span class="sc">+</span>p53<span class="sc">*</span>treatment)   <span class="co"># note that the study design is changed.</span></span>
<span id="cb86-4"><a href="differentially-expressed-sequence-identification.html#cb86-4" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">=</span> <span class="fu">DESeq</span>(dds)  <span class="co"># main function</span></span>
<span id="cb86-5"><a href="differentially-expressed-sequence-identification.html#cb86-5" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(dds)</span></code></pre></div>
<p>Filtering: we will only keep rows that have a sum count between all samples greater than 5. This will remove most of the genes that mostly have “0” counts.</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="differentially-expressed-sequence-identification.html#cb87-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[ <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds)) <span class="sc">&gt;</span> <span class="dv">5</span>, ]</span>
<span id="cb87-2"><a href="differentially-expressed-sequence-identification.html#cb87-2" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(dds)</span></code></pre></div>
<p>Regularized log transformation - used for clustering</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="differentially-expressed-sequence-identification.html#cb88-1" aria-hidden="true" tabindex="-1"></a>rld <span class="ot">&lt;-</span> <span class="fu">rlog</span>(dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb88-2"><a href="differentially-expressed-sequence-identification.html#cb88-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">assay</span>(rld), <span class="dv">6</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb88-3"><a href="differentially-expressed-sequence-identification.html#cb88-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb88-4"><a href="differentially-expressed-sequence-identification.html#cb88-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span></code></pre></div>
<p>Variance Stabilizing Transformation</p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="cleaning-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Bioinformatics_RNA-seq.pdf", "Bioinformatics_RNA-seq.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
