<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Bioinformatics_RNA-seq</title>
  <meta name="description" content="RNA-seq notes and trials" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Bioinformatics_RNA-seq" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="RNA-seq notes and trials" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Bioinformatics_RNA-seq" />
  
  <meta name="twitter:description" content="RNA-seq notes and trials" />
  

<meta name="author" content="J. Marcelo Rosales R." />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="wgcna-rna-seq-blog.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to RNA sequencing (RNA_seq)</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#downloading-and-setting-up-conda-environments."><i class="fa fa-check"></i><b>1.1</b> Downloading and setting up conda environments.</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#install-miniconda-or-anaconda."><i class="fa fa-check"></i><b>1.2</b> Install Miniconda, or Anaconda.</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#conda-terminal-commands."><i class="fa fa-check"></i><b>1.2.1</b> Conda terminal commands.</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#anacondaminiconda-install-summary-of-steps."><i class="fa fa-check"></i><b>1.2.2</b> Anaconda/Miniconda install Summary of Steps.</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#conda-activate-error"><i class="fa fa-check"></i><b>1.2.3</b> Conda activate error</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#alignment-procedures-and-files."><i class="fa fa-check"></i><b>1.3</b> Alignment Procedures and files.</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#using-tophat-early-version"><i class="fa fa-check"></i><b>1.3.1</b> Using Tophat (early version)</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#using-tophat"><i class="fa fa-check"></i><b>1.3.2</b> Using Tophat</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#using-star"><i class="fa fa-check"></i><b>1.3.3</b> Using STAR</a></li>
<li class="chapter" data-level="1.3.4" data-path="index.html"><a href="index.html#custompersonal-way-to-download-the-mus_musculus_ncbi_build37.2.tar.gz-file."><i class="fa fa-check"></i><b>1.3.4</b> Custom/personal way to Download the Mus_musculus_NCBI_build37.2.tar.gz file.</a></li>
<li class="chapter" data-level="1.3.5" data-path="index.html"><a href="index.html#get-the-sample-files."><i class="fa fa-check"></i><b>1.3.5</b> Get the Sample files.</a></li>
<li class="chapter" data-level="1.3.6" data-path="index.html"><a href="index.html#starting-to-analize-fastq.gz-files"><i class="fa fa-check"></i><b>1.3.6</b> Starting to analize fastq.gz files</a></li>
<li class="chapter" data-level="1.3.7" data-path="index.html"><a href="index.html#trimming-with-trimmomatic"><i class="fa fa-check"></i><b>1.3.7</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="1.3.8" data-path="index.html"><a href="index.html#using-star-to-map-genes"><i class="fa fa-check"></i><b>1.3.8</b> Using STAR to map genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html"><i class="fa fa-check"></i><b>2</b> WGCNA RNA-seq Blog</a>
<ul>
<li class="chapter" data-level="2.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#WGCNA"><i class="fa fa-check"></i><b>2.1</b> Introduction to the WGCNA Rpackage</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-rpackage-installation-procedure."><i class="fa fa-check"></i><b>2.1.1</b> WGCNA Rpackage Installation procedure.</a></li>
<li class="chapter" data-level="2.1.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#final-installation-workflow"><i class="fa fa-check"></i><b>2.1.2</b> Final installation workflow:</a></li>
<li class="chapter" data-level="2.1.3" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-tutorial."><i class="fa fa-check"></i><b>2.1.3</b> WGCNA tutorial.</a></li>
<li class="chapter" data-level="2.1.4" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#gene-expression-analysis"><i class="fa fa-check"></i><b>2.1.4</b> 3.4 Gene expression analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#rna-seq-analisys-with-kakus-data-sample."><i class="fa fa-check"></i><b>2.2</b> RNA-seq Analisys with Kaku’s Data sample.</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#data-analysis."><i class="fa fa-check"></i><b>2.2.1</b> Data Analysis.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="sra-download-qc-and-trimming.html"><a href="sra-download-qc-and-trimming.html"><i class="fa fa-check"></i><b>3</b> SRA Download, QC, and Trimming</a></li>
<li class="chapter" data-level="4" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>4</b> Methods</a></li>
<li class="chapter" data-level="5" data-path="youtube-references-.html"><a href="youtube-references-.html"><i class="fa fa-check"></i><b>5</b> Youtube references.</a></li>
<li class="chapter" data-level="6" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>6</b> Final Words</a></li>
<li class="chapter" data-level="7" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html"><i class="fa fa-check"></i><b>7</b> RNA Seq for STAT736</a>
<ul>
<li class="chapter" data-level="7.1" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#background"><i class="fa fa-check"></i><b>7.1</b> <strong>Background</strong></a></li>
<li class="chapter" data-level="7.2" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#cleaning-data"><i class="fa fa-check"></i><b>7.2</b> <strong>Cleaning Data</strong></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#programs-used"><i class="fa fa-check"></i><b>7.2.1</b> Programs used?</a></li>
<li class="chapter" data-level="7.2.2" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#acquiring-sequences"><i class="fa fa-check"></i><b>7.2.2</b> Acquiring sequences</a></li>
<li class="chapter" data-level="7.2.3" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#fastqc"><i class="fa fa-check"></i><b>7.2.3</b> FastQC</a></li>
<li class="chapter" data-level="7.2.4" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#trimming-with-trimmomatic-1"><i class="fa fa-check"></i><b>7.2.4</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="7.2.5" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#alignment"><i class="fa fa-check"></i><b>7.2.5</b> Alignment</a></li>
<li class="chapter" data-level="7.2.6" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#assembling-transcripts-with-cufflinks"><i class="fa fa-check"></i><b>7.2.6</b> Assembling transcripts with Cufflinks</a></li>
<li class="chapter" data-level="7.2.7" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#checking-for-contamination"><i class="fa fa-check"></i><b>7.2.7</b> Checking for Contamination</a></li>
<li class="chapter" data-level="7.2.8" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#counting-transcripts"><i class="fa fa-check"></i><b>7.2.8</b> Counting Transcripts</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#differentially-expressed-sequence-identification"><i class="fa fa-check"></i><b>7.3</b> <strong>Differentially Expressed Sequence Identification</strong></a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#analyzing-reads-counts"><i class="fa fa-check"></i><b>7.3.1</b> Analyzing Reads Counts</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e.g.-p53-control-and-ir-and-p53---control-and-ir"><i class="fa fa-check"></i><b>7.4</b> Interactions cause a difference between the lfc betwen pooled data, e.g. p53+/+ (control and IR) and p53-/- (control and IR)</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>7.4.1</b> Exploratory Data Analysis</a></li>
<li class="chapter" data-level="7.4.2" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#identification-of-differentially-expressed-genes"><i class="fa fa-check"></i><b>7.4.2</b> Identification of Differentially Expressed Genes</a></li>
<li class="chapter" data-level="7.4.3" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#gene-annotations"><i class="fa fa-check"></i><b>7.4.3</b> Gene Annotations</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#go-enrichment-analysis-using-gostats"><i class="fa fa-check"></i><b>7.5</b> 7. GO Enrichment analysis using GOstats</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#go-enrichment-analysis-of-downregulated-genes"><i class="fa fa-check"></i><b>7.5.1</b> GO Enrichment analysis of downregulated genes</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#pathway-analysis-using-expression-data"><i class="fa fa-check"></i><b>7.6</b> 8. Pathway analysis using expression data</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#prepare-data"><i class="fa fa-check"></i><b>7.6.1</b> Prepare data</a></li>
<li class="chapter" data-level="7.6.2" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#kegg-pathways"><i class="fa fa-check"></i><b>7.6.2</b> KEGG pathways</a></li>
<li class="chapter" data-level="7.6.3" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#pathway-and-regulation-of-genes-for-oxidative-phosphorylation."><i class="fa fa-check"></i><b>7.6.3</b> Pathway and regulation of genes for Oxidative phosphorylation.</a></li>
<li class="chapter" data-level="7.6.4" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#pathway-and-regulation-of-genes-for-glycosylphosphatidylinositolgpi-anchor-biosynthesis."><i class="fa fa-check"></i><b>7.6.4</b> Pathway and regulation of genes for Glycosylphosphatidylinositol(GPI)-anchor biosynthesis.</a></li>
<li class="chapter" data-level="7.6.5" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#pathway-and-regulation-of-genes-for-rna-polymerase."><i class="fa fa-check"></i><b>7.6.5</b> Pathway and regulation of genes for RNA polymerase.</a></li>
<li class="chapter" data-level="7.6.6" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#pathway-and-regulation-of-genes-for-nucleotide-excision-repair."><i class="fa fa-check"></i><b>7.6.6</b> Pathway and regulation of genes for Nucleotide excision repair.</a></li>
<li class="chapter" data-level="7.6.7" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#pathway-and-regulation-of-genes-for-non-homologous-end-joining."><i class="fa fa-check"></i><b>7.6.7</b> Pathway and regulation of genes for Non-homologous end-joining.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics_RNA-seq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Bioinformatics_RNA-seq</h1>
<p class="author"><em>J. Marcelo Rosales R.</em></p>
<p class="date" style="margin-top: 1.5em;"><em><strong>Created:</strong> 2021-08-10; <strong>Update:</strong> 2021-08-20; <strong>Date:</strong> 2021-09-07</em></p>
</div>
<div id="introduction-to-rna-sequencing-rna_seq" class="section level1" number="1">
<h1><span class="header-section-number">Chapter 1</span> Introduction to RNA sequencing (RNA_seq)</h1>
<p>An over simplification of the Process.</p>
<p>Holds 4 main sections.</p>
<ol style="list-style-type: decimal">
<li>Sample and sample analysis. Samples are taken, prepared and send to a company for analysis. (Illumina !?).<br />
</li>
<li>Raw data (FASTQ) - Expression quantification.<br />
</li>
<li>Differential expression analysis.</li>
<li>Pathway and Enrichment analysis.</li>
</ol>
<p>Each of these sections are subdivided in other steps.</p>
<p><strong>Download or acquire the files</strong><br />
2021/07/02<br />
Copy <strong>Trial data</strong> from:<br />
MK/土橋 to <a href="file:///Volumes/HD-PCFSU3-A/Experiments%Data/Genewiz">folder HD-PCFSU3-A/Experiments Data/Genewiz</a> :<br />
The Report <a href="file:///Volumes/HD-PCFSU3-A/Experiments%20Data/Genewiz/60-499583816_60-501009934/60-499583816_60-501009934_GENEWIZ_RNASeq_Report.html">file</a> is here.<br />
The Summary <a href="file:///Volumes/HD-PCFSU3-A/Experiments%20Data/Genewiz/60-499583816_60-501009934/60-499583816_60-501009934_GENEWIZ_QC_Report.html">file</a> is here.</p>
<p>Raw Data</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="index.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> <span class="st">&quot;Volumes/HD-PCFSU3-A/Experiments Data/Genewiz/60-499583816_60-501009934/Result/03_GeneExpression&quot;</span></span></code></pre></div>
<p>Report file</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="index.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> <span class="st">&quot;/Volumes/HD-PCFSU3-A/Experiments%20Data/Genewiz/60-499583816_60-501009934/60-499583816_60-501009934_GENEWIZ_RNASeq_Report.html&quot;</span></span></code></pre></div>
<p>Summary report file</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="index.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> <span class="st">&quot;/Volumes/HD-PCFSU3-A/Experiments%20Data/Genewiz/60-499583816_60-501009934/60-499583816_60-501009934_GENEWIZ_QC_Report.html&quot;</span></span></code></pre></div>
<p>To open folder in Finder with r code, is similar to terminal commands.:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="index.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> . <span class="co"># To open the current working folder within Finder</span></span>
<span id="cb4-2"><a href="index.html#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> ~ <span class="co"># To open the Home folder </span></span>
<span id="cb4-3"><a href="index.html#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> / <span class="co"># To open the Root directory</span></span>
<span id="cb4-4"><a href="index.html#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> Desktop/</span>
<span id="cb4-5"><a href="index.html#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> Music/</span>
<span id="cb4-6"><a href="index.html#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> /Library/</span>
<span id="cb4-7"><a href="index.html#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> /path/to/Directory/</span>
<span id="cb4-8"><a href="index.html#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co"># If folder have spaces, replace with %20% or other similar, or use &quot;&quot;.</span></span>
<span id="cb4-9"><a href="index.html#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> <span class="st">&quot;/path/to/Directory/&quot;</span></span></code></pre></div>
<p>You can also launch (and update) applications from the Terminal without using Finder. For example, to open Safari, type: <strong>(But seem like it only works in terminal and can not do it with r/rstudio)</strong></p>
<div id="downloading-and-setting-up-conda-environments." class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Downloading and setting up conda environments.</h2>
<p>Benefit: You can install all your tools in “Conda” and export as yml and you can share that environment or create a new one base on it, and by doing so, you can have all those tools automatically installed.</p>
</div>
<div id="install-miniconda-or-anaconda." class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Install Miniconda, or Anaconda.</h2>
<p><strong>Choose Anaconda if you:</strong><br />
- Are new to conda or Python<br />
- Like the convenience of having Python and over 1500 scientific packages automatically installed at once<br />
- Have the time and disk space (a few minutes and 3 GB), and/or<br />
- Don’t want to install each of the packages you want to use individually.</p>
<p><strong>Choose Miniconda if you:</strong><br />
- Do not mind installing each of the packages you want to use individually.<br />
- Do not have time or disk space to install over 1500 packages at once, and/or<br />
- Just want fast access to Python and the conda commands, and wish to sort out the other programs later.</p>
<blockquote>
<p>Mind that Windows and Mac installation and procedures are different.</p>
</blockquote>
<p><em>Does Windows need Linux to be intalled?</em> Yes, a version of Linux kernell for window is available for download.</p>
<p>Install <a href="https://www.anaconda.com/">Anaconda</a> or <a href="https://conda.io/miniconda.html">Miniconda</a>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="index.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">brew</span> install <span class="at">--cask</span> anaconda</span>
<span id="cb5-2"><a href="index.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="ex">brew</span> install <span class="at">--cask</span> miniconda</span>
<span id="cb5-3"><a href="index.html#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="index.html#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co"># To check the version of conda installed.  </span></span>
<span id="cb5-5"><a href="index.html#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> <span class="at">--version</span></span></code></pre></div>
<p>Quick video tutorial on how to use conda.<br />
* <a href="https://youtu.be/1VVCd0eSkYc">(Master) Conda environment</a>.<br />
* <a href="https://youtu.be/ePS6lII4SEg">Set up a data Science environment with Conda-Forge</a>.</p>
<p><strong>To create an environment.</strong><br />
<strong>The easy way:</strong> Open the anaconda app &gt; Environment &gt; New &gt; Name… Ok
To access environment and open in terminal: anaconda app &gt; Environment &gt; Select environment &gt; rna-seq_test ^ &gt; »Open in Terminal.</p>
<div id="conda-terminal-commands." class="section level3" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Conda terminal commands.</h3>
<p>Conda <a href="https://docs.conda.io/projects/conda/en/4.6.0/_downloads/52a95608c49671267e40c689e0bc00ca/conda-cheatsheet.pdf">cheat sheet</a></p>
<p><strong>In Terminal</strong></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="index.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># To open conda app</span></span>
<span id="cb6-2"><a href="index.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> /Applications/Anaconda-Navigator.app</span>
<span id="cb6-3"><a href="index.html#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Or, if any of the container folders have a space character.</span></span>
<span id="cb6-4"><a href="index.html#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> <span class="st">&quot;/Applications/Anaconda-Navigator.app&quot;</span></span>
<span id="cb6-5"><a href="index.html#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="index.html#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co"># To open (activate) conda environment. </span></span>
<span id="cb6-7"><a href="index.html#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="bu">.</span> /usr/local/anaconda3/bin/activate <span class="kw">&amp;&amp;</span> <span class="ex">conda</span> activate /usr/local/anaconda3/envs/rna-seq_test</span>
<span id="cb6-8"><a href="index.html#cb6-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-9"><a href="index.html#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="co"># To see all the environment created.  </span></span>
<span id="cb6-10"><a href="index.html#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> env list  </span>
<span id="cb6-11"><a href="index.html#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="index.html#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="co"># To create a new environment called &quot;rna-seq_test&quot; in conda in terminal:  </span></span>
<span id="cb6-13"><a href="index.html#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> create <span class="at">--rna-seq_test</span></span>
<span id="cb6-14"><a href="index.html#cb6-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-15"><a href="index.html#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="co"># To activate environment.    </span></span>
<span id="cb6-16"><a href="index.html#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate rna-seq_test</span>
<span id="cb6-17"><a href="index.html#cb6-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-18"><a href="index.html#cb6-18" aria-hidden="true" tabindex="-1"></a><span class="co"># To see if we activated the environment and where (path) we are.  </span></span>
<span id="cb6-19"><a href="index.html#cb6-19" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="va">$path</span></span>
<span id="cb6-20"><a href="index.html#cb6-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-21"><a href="index.html#cb6-21" aria-hidden="true" tabindex="-1"></a><span class="co"># To exit the environment and return to base/root. Check with echo $path. </span></span>
<span id="cb6-22"><a href="index.html#cb6-22" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate</span>
<span id="cb6-23"><a href="index.html#cb6-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-24"><a href="index.html#cb6-24" aria-hidden="true" tabindex="-1"></a><span class="co"># To see packages installed in env.:  </span></span>
<span id="cb6-25"><a href="index.html#cb6-25" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--show</span> channels </span>
<span id="cb6-26"><a href="index.html#cb6-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-27"><a href="index.html#cb6-27" aria-hidden="true" tabindex="-1"></a><span class="co"># List all packages and versions installed in active environment</span></span>
<span id="cb6-28"><a href="index.html#cb6-28" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> list</span>
<span id="cb6-29"><a href="index.html#cb6-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-30"><a href="index.html#cb6-30" aria-hidden="true" tabindex="-1"></a><span class="co"># To see the url location of packages (repository):</span></span>
<span id="cb6-31"><a href="index.html#cb6-31" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> info</span>
<span id="cb6-32"><a href="index.html#cb6-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-33"><a href="index.html#cb6-33" aria-hidden="true" tabindex="-1"></a><span class="co"># To add packages to the environment. (only to this environment: --env).  </span></span>
<span id="cb6-34"><a href="index.html#cb6-34" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--env</span> <span class="at">--add</span> channels conda-forge</span>
<span id="cb6-35"><a href="index.html#cb6-35" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--env</span> <span class="at">--add</span> channels bioconda </span>
<span id="cb6-36"><a href="index.html#cb6-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-37"><a href="index.html#cb6-37" aria-hidden="true" tabindex="-1"></a><span class="co"># If a package is repeated in the environment, Conda will always install the package of higher version. If you dont want that, and specifically need a version originally configured at creation of the environment.. type:   </span></span>
<span id="cb6-38"><a href="index.html#cb6-38" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--env</span> <span class="at">--set</span> channel_priority strict.  </span>
<span id="cb6-39"><a href="index.html#cb6-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-40"><a href="index.html#cb6-40" aria-hidden="true" tabindex="-1"></a><span class="co"># To install fastqc</span></span>
<span id="cb6-41"><a href="index.html#cb6-41" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> install fastqc </span>
<span id="cb6-42"><a href="index.html#cb6-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-43"><a href="index.html#cb6-43" aria-hidden="true" tabindex="-1"></a><span class="co">#To make and export a list of the required environment/</span></span>
<span id="cb6-44"><a href="index.html#cb6-44" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> list <span class="at">--export</span> <span class="op">&gt;</span> requirements.txt  </span>
<span id="cb6-45"><a href="index.html#cb6-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-46"><a href="index.html#cb6-46" aria-hidden="true" tabindex="-1"></a><span class="co">#To make and export an yml file to share with others.</span></span>
<span id="cb6-47"><a href="index.html#cb6-47" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> env export <span class="op">&gt;</span> rna-seq_test.yml  </span>
<span id="cb6-48"><a href="index.html#cb6-48" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> env export <span class="at">--file</span>  rna-seq_test.yml </span>
<span id="cb6-49"><a href="index.html#cb6-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-50"><a href="index.html#cb6-50" aria-hidden="true" tabindex="-1"></a><span class="co"># To quickly check the yml file. </span></span>
<span id="cb6-51"><a href="index.html#cb6-51" aria-hidden="true" tabindex="-1"></a><span class="fu">less</span> rna-seq_test.yml </span>
<span id="cb6-52"><a href="index.html#cb6-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-53"><a href="index.html#cb6-53" aria-hidden="true" tabindex="-1"></a><span class="co">#To see the content of a .yml file.</span></span>
<span id="cb6-54"><a href="index.html#cb6-54" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> rna-seq_test.yml</span>
<span id="cb6-55"><a href="index.html#cb6-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-56"><a href="index.html#cb6-56" aria-hidden="true" tabindex="-1"></a><span class="co">#To clear the terminal screen. </span></span>
<span id="cb6-57"><a href="index.html#cb6-57" aria-hidden="true" tabindex="-1"></a><span class="fu">clear</span> </span>
<span id="cb6-58"><a href="index.html#cb6-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-59"><a href="index.html#cb6-59" aria-hidden="true" tabindex="-1"></a><span class="co"># To run a Jupyter Notebook.  </span></span>
<span id="cb6-60"><a href="index.html#cb6-60" aria-hidden="true" tabindex="-1"></a><span class="ex">jupyter</span> notebook   </span>
<span id="cb6-61"><a href="index.html#cb6-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-62"><a href="index.html#cb6-62" aria-hidden="true" tabindex="-1"></a><span class="co"># To Keep Anaconda updated.  </span></span>
<span id="cb6-63"><a href="index.html#cb6-63" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> update conda</span>
<span id="cb6-64"><a href="index.html#cb6-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-65"><a href="index.html#cb6-65" aria-hidden="true" tabindex="-1"></a><span class="co"># Delete an environment and everything in it called rna-seq_test.</span></span>
<span id="cb6-66"><a href="index.html#cb6-66" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> env remove <span class="at">--rna-seq_test</span></span>
<span id="cb6-67"><a href="index.html#cb6-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-68"><a href="index.html#cb6-68" aria-hidden="true" tabindex="-1"></a><span class="co"># To Uninstall Anaconda.  </span></span>
<span id="cb6-69"><a href="index.html#cb6-69" aria-hidden="true" tabindex="-1"></a><span class="co"># [Uninstall page]()</span></span></code></pre></div>
</div>
<div id="anacondaminiconda-install-summary-of-steps." class="section level3" number="1.2.2">
<h3><span class="header-section-number">1.2.2</span> Anaconda/Miniconda install Summary of Steps.</h3>
<ol style="list-style-type: decimal">
<li>Install anaconda/miniconda for your OS from the website.<br />
</li>
<li>Prevent the base environment from automatically activating. <code>conda config --set auto_activate_base false</code><br />
</li>
<li>Create an empty environment. <code>conda create -n rna-seq_test</code>.<br />
</li>
<li>Activate the environment <code>conda activate rna-seq_test</code>.<br />
</li>
<li>Add conda-forge as first channel <code>conda config --env --add channels conda-forge</code>.<br />
</li>
<li>Ensure that conda-forge is used if the package is available. <code>conda config --env --set channel_priority strict</code><br />
</li>
<li>Install packages <code>conda install ....</code><br />
</li>
<li>Install packages not in conda-forge. Search in conda webpage search. <code>conda install -c ....</code><br />
</li>
<li>Verify Jupyter Notebooks in correct environment …!?… In jupyter: <code>sys.executable</code></li>
</ol>
<p>Video reference:</p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/ePS6lII4SEg" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen>
</iframe>
<p>Bioinformatics - Downloading and Setting Up Conda Environments.</p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/ef_-BtMtDG0" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen>
</iframe>
</div>
<div id="conda-activate-error" class="section level3" number="1.2.3">
<h3><span class="header-section-number">1.2.3</span> Conda activate error</h3>
<p>When trying to initialize environment in terminal with the <code>conda activate rna-seq_test</code>command, the following message pops up.</p>
<pre><code>CommandNotFoundError: Your shell has not been properly configured to use &#39;conda activate&#39;.
To initialize your shell, run

    $ conda init &lt;SHELL_NAME&gt;

Currently supported shells are:
  - bash
  - fish
  - tcsh
  - xonsh
  - zsh
  - powershell

See &#39;conda init --help&#39; for more information and options.

IMPORTANT: You may need to close and restart your shell after running &#39;conda init&#39;.</code></pre>
<p>I tried:</p>
<pre><code>conda init powershell
conda init zsh</code></pre>
<p>So far the only way to enter environment is though anaconda app.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="index.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> <span class="st">&quot;/Applications/Anaconda-Navigator.app&quot;</span></span></code></pre></div>
<p>Then Environment &gt; rna-seq_test» &gt; Open in terminal… [Terminal]</p>
<p>Another temporary solution is using the code:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="index.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># For the rna-seq_test environment. </span></span>
<span id="cb10-2"><a href="index.html#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="bu">.</span> /usr/local/anaconda3/bin/activate <span class="kw">&amp;&amp;</span> <span class="ex">conda</span> activate /usr/local/anaconda3/envs/rna-seq_test</span>
<span id="cb10-3"><a href="index.html#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="index.html#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># For anaconda3 base environment. </span></span>
<span id="cb10-5"><a href="index.html#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="bu">.</span> /usr/local/anaconda3/bin/activate <span class="kw">&amp;&amp;</span> <span class="ex">conda</span> activate /usr/local/anaconda3<span class="kw">;</span></span></code></pre></div>
</div>
</div>
<div id="alignment-procedures-and-files." class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Alignment Procedures and files.</h2>
<p>This section was created based in <a href="https://www.youtube.com/watch?v=UkrUTgAKFiE&amp;list=PL-0fKymgD8L8qW7SQDaEf3lrSyUkEkR1k&amp;index=2">this reference</a>.</p>
<p>After installing “conda,” and to gain time, download the genome of the species to be study. In this case, download the mouse (mus muslculus) genome file. This file contains the list of all the mouse genes and their codes.</p>
<p>We need something to map our reads against once we get to the results.<br />
Files for mouse (Mus musculus) are available for download at Johns Hopkins University Center for Computational Biology (CCB) at <a href="https://ccb.jhu.edu/software/tophat/igenomes.shtml">TopHat. A spliced read mapper for RNA-Seq</a>.</p>
<p>There are different ways:</p>
<ol style="list-style-type: decimal">
<li>Using tophat<br />
</li>
<li>Using Star<br />
</li>
<li>Custom/personal code.</li>
</ol>
<div id="using-tophat-early-version" class="section level3" number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> Using Tophat (early version)</h3>
<blockquote>
<p>Not Working!!!</p>
</blockquote>
<p>To download:<br />
1. Look for the <strong>Mus musculus Build37.2</strong> from the NCBI.
1. Right click the link and copy the <a href="ftp://igenome:G3nom3s4u@ussd-ftp.illumina.com/Mus_musculus/NCBI/build37.2/Mus_musculus_NCBI_build37.2.tar.gz">link address</a>.<br />
1. go back to the conda environment and type wget and the paste the link.</p>
<pre><code>wget ftp://igenome:G3nom3s4u@ussd-ftp.illumina.com/Mus_musculus/NCBI/build37.2/Mus_musculus_NCBI_build37.2.tar.gz</code></pre>
</div>
<div id="using-tophat" class="section level3" number="1.3.2">
<h3><span class="header-section-number">1.3.2</span> Using Tophat</h3>
<blockquote>
<p>Not Working!!!</p>
</blockquote>
<p>Tophat can be installed using the same <strong>conda install</strong></p>
<pre><code>conda install \-c bioconda tophat</code></pre>
<p>When this is finished installing, then we will need to get the mouse genome from the <a href="http://ccb.jhu.edu/software/tophat/igenomes.shtml">Johns Hopkins Univeristy Center for computational BIology</a>. The version of the mouse genome that I am using here is the <a href="ftp://igenome:G3nom3s4u@ussd-ftp.illumina.com/Mus_musculus/NCBI/build37.2/Mus_musculus_NCBI_build37.2.tar.gz">NCBI build37.2</a>. Instead of downloading this from the website and having to move it to the cluser, I will just download it using wget into the folder that has the raw reads, trimmed reads, and the FastQC files.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="index.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> ftp://igenome:G3nom3s4u@ussd-ftp.illumina.com/Mus_musculus/NCBI/build37.2/Mus_musculus_NCBI_build37.2.tar.gz</span></code></pre></div>
<p>This will take a long time to download because the file is a little less than 16GB zipped.</p>
<blockquote>
<p>Installation of TopHat failed.!!!!!!!!!!!!!!!!!!!!!!!</p>
</blockquote>
<p>Try STAR</p>
</div>
<div id="using-star" class="section level3" number="1.3.3">
<h3><span class="header-section-number">1.3.3</span> Using STAR</h3>
<blockquote>
<p>Complicated!!!</p>
</blockquote>
<p>STAR can be installed the same way as the previous programs with <strong>conda install</strong> (<strong><em>conda install -c bioconda star</em></strong>). In order to run STAR, we need to creaate indices just like with tophat, but STAR has this built in. I’m going to be using the same genome and GTF file as previously downloaded, but Dr. Ge uses a different zipped genome from the <em>gencode</em> database.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="index.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/STAR</span> <span class="dt">\</span></span>
<span id="cb14-2"><a href="index.html#cb14-2" aria-hidden="true" tabindex="-1"></a>--runThreadN 80 <span class="dt">\</span></span>
<span id="cb14-3"><a href="index.html#cb14-3" aria-hidden="true" tabindex="-1"></a>--runMode genomeGenerate <span class="dt">\</span></span>
<span id="cb14-4"><a href="index.html#cb14-4" aria-hidden="true" tabindex="-1"></a>--genomeDir starIndex <span class="dt">\</span></span>
<span id="cb14-5"><a href="index.html#cb14-5" aria-hidden="true" tabindex="-1"></a>--genomeFastaFiles Index/genome.fa <span class="dt">\ </span>#same when we made the bowtie indices </span>
<span id="cb14-6"><a href="index.html#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="ex">--sjdbGTFfile</span> Mus_musculus/NCBI/build37.2/Annotation/Archives/archive-2015-07-17-14-32-40/Genes/genes.gtf</span></code></pre></div>
<p>With the index files made, we can start aligning with STAR. It’s important here than we only pick the paired end reads and not use all of the reads. Tophat is able to use all 4 reads but STAR doesn’t allow that, so we need to make sure that we feed in the large files from trimming.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="index.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/STAR</span> <span class="at">--runThreadN</span> 80 <span class="at">--genomeDir</span> starIndex <span class="at">--readFilesIn</span> 770_fp.fq 770_rp.fq <span class="at">--outFilterIntronMotifs</span> RemoveNoncanonical <span class="at">--outFileNamePrefix</span> 2121770 <span class="at">--outSAMtype</span> BAM SortedByCoordinate</span>
<span id="cb15-2"><a href="index.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="ex">[here]</span><span class="er">(</span><span class="kw">)</span></span></code></pre></div>
<blockquote>
<p>Complicated.!!!!!!!!!!!!!!!!!!!!!!!</p>
</blockquote>
</div>
<div id="custompersonal-way-to-download-the-mus_musculus_ncbi_build37.2.tar.gz-file." class="section level3" number="1.3.4">
<h3><span class="header-section-number">1.3.4</span> Custom/personal way to Download the Mus_musculus_NCBI_build37.2.tar.gz file.</h3>
<blockquote>
<p>Esaest so far.</p>
</blockquote>
<ol style="list-style-type: decimal">
<li>Go to the iGenomes Ready-To-Use Reference Sequences and Annotations from <a href="https://support.illumina.com/sequencing/sequencing_software/igenome.html">illumina.</a>.<br />
</li>
<li>Copy the Mus musculus (Mouse) NCBI build37.2 <a href="https://support.illumina.com/sequencing/sequencing_software/igenome.html">link</a>.<br />
</li>
<li>Paste the link in a new tab/window on a web browser (chrome) and click enter.</li>
<li>Popup window will ask where to save the file, choose folder and save.</li>
<li>File is 23Gb so it will take about 3 to 4 hrs to download.</li>
</ol>
</div>
<div id="get-the-sample-files." class="section level3" number="1.3.5">
<h3><span class="header-section-number">1.3.5</span> Get the Sample files.</h3>
<p>Samples files come as “x_1.fastq.gz” or “x_2.fastq.gz,” where x is the name of the file, 1 is for forward and 2 for reverse pair reads. example: we will place them in a folder (00_Rawdata)
mPDL_RNA7D_Ko1_S1_L001_R1_001.fastq.gz<br />
mPDL_RNA7D_Ko1_S1_L001_R2_001.fastq.gz<br />
mPDL_RNA7D_Ko2_S1_L001_R1_001.fastq.gz<br />
mPDL_RNA7D_Ko2_S1_L001_R2_001.fastq.gz<br />
mPDL_RNA7D_Ko3_S1_L001_R1_001.fastq.gz<br />
mPDL_RNA7D_Ko3_S1_L001_R2_001.fastq.gz… etc….</p>
<p>Download multiqc</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="index.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> install multiqc</span></code></pre></div>
</div>
<div id="starting-to-analize-fastq.gz-files" class="section level3" number="1.3.6">
<h3><span class="header-section-number">1.3.6</span> Starting to analize fastq.gz files</h3>
<p>Initialize fastqc</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="index.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="ex">fastqc</span> </span></code></pre></div>
<p>Make an output directory named rawFastQC</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="index.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> rawFastQC</span></code></pre></div>
<p>Now we have to process all fastq files contained in the folder (00_Rawdata) and place the results in the directory we just made “rawFastQC.”</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="index.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="ex">fastqc</span> rawFastQC/<span class="pp">*</span>.fastq.gz <span class="at">-o</span> rawFastQC/</span></code></pre></div>
<p>Once analysis is finished. Go to the results folder (rawFastQC)</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="index.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> rawFastQC</span></code></pre></div>
<p>“fastqc” created .zip files and a .html file report for each sample.
To see them individually may be difficult and take time. For a more convenient way to visualize results, merge all sample reports (html file) in one file.
Create the report by reading all the files contained in the results folder (rawFastQC) and combine them in one multiqc_report.html file using “multiqc.”</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="index.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># go to the results folder where the html report files are.</span></span>
<span id="cb21-2"><a href="index.html#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> rawFastQC </span>
<span id="cb21-3"><a href="index.html#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Check if the files are there. </span></span>
<span id="cb21-4"><a href="index.html#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span></span>
<span id="cb21-5"><a href="index.html#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the multiqc report. Type (the dot is important!)</span></span>
<span id="cb21-6"><a href="index.html#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> .</span>
<span id="cb21-7"><a href="index.html#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="co"># check if file was created.</span></span>
<span id="cb21-8"><a href="index.html#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
</div>
<div id="trimming-with-trimmomatic" class="section level3" number="1.3.7">
<h3><span class="header-section-number">1.3.7</span> Trimming with Trimmomatic</h3>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="index.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> install trimmomatic</span>
<span id="cb22-2"><a href="index.html#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="co"># parallel my also be required for task piping.</span></span>
<span id="cb22-3"><a href="index.html#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> install parallel</span></code></pre></div>
<p>24:40
Conda was used again to run Trimmomatic. This isn’t as easy as using the wildcard like with FastQC because each output has to be personalized for the read files that are input into Trimmomatic. Also, we have to make sure that the adapter sequences are in the same folder that we are running so we can refer to them easily when calling the Trimmomatic program. In this case, we are using the TruSeq3-PE-2.fa adapter sequences For example:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="index.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/trimmomatic</span> PE SRR2121770_1.fastq.gz SRR2121770_2.fastq.gz 770_fp.fq.gz 770_fu.fq.gz 770_rp.fq.gz 770_ru.fq.gz ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:10:2keepBothReads LEADING:3 TRALING:3 MINLEN:36 <span class="kw">&amp;</span></span></code></pre></div>
<p>This would be repeated for each of the pairs (12 in total). We are trimming paired-end reads with the TruSeq3-PE-2 adapters. We are chopping off the first and last 3 bases and if we end up with a sequence less than 36 bases, we get rid of it. We want to make sure that there are enough bases in a read to work with. These parameters can be tweaked for possibly better end results with less being discarded. When Trimmomatic is finished running, it will out put the total number of reads, the total number from both the forward and reverse reads that are kept, the number of only forward reads kept, the number of only reverse reads kept, and the number of discarded reads.</p>
<p>The trimmed reads can be analyzed again with FastQC to see how well the trimming worked to make the file better quality. After running FastQC on the trimmed files we see that the quality of those that were really bad quality were improved. There were a few different metrics throughout all of the files that bounced from a warning before the failing, or from passing before to a warning, and so forth, overall creating better quality read files.</p>
<blockquote>
<p>Read Check Undrstantig Trimmomatic tutorial for this section !!!!!!</p>
</blockquote>
<p>after trimmomatic, files will be created, .gz and .log files. Use multiqc to merge all logs in one report file. Then check the timmomatic survival Reads. If all samples are over 55M (million) reads then samples are valid for further analysis.</p>
<blockquote>
<p>Check the order again.
Trimmomatic &gt; alligment -&gt; w/ TopHap or STAR</p>
</blockquote>
</div>
<div id="using-star-to-map-genes" class="section level3" number="1.3.8">
<h3><span class="header-section-number">1.3.8</span> Using STAR to map genes</h3>
<p>Using the Mus_musculus files we will compare our samples and find where in the genome our reads are.
&gt; Build Genome Indexing and useful STAR Flags (quick).</p>
<p>See also:
<a href="http://www.novocraft.com/documentation/novoalign-2/novoalign-ngs-quick-start-tutorial/1040-2/">novocraft aling</a>. *Opens only on safari.<br />
<a href="https://github.com/ACSoupir/SnakeMake">Analyzind RNA-Seq data using Python3 Snakemake</a>. File <a href="https://github.com/ACSoupir/SnakeMake/blob/master/Snakefile">Snake file</a>.</p>
<blockquote>
<p>Look for more information with rRNA and Microbial Contamination.<br />
Bioinformatics - Contamination QC and FeatureCounts
Quick Look at Counts and Setting Up R Project</p>
</blockquote>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="wgcna-rna-seq-blog.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Bioinformatics_RNA-seq.pdf", "Bioinformatics_RNA-seq.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
