<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Methods | Bioinformatics_RNA-seq</title>
  <meta name="description" content="RNA-seq notes and trials" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Methods | Bioinformatics_RNA-seq" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="RNA-seq notes and trials" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Methods | Bioinformatics_RNA-seq" />
  
  <meta name="twitter:description" content="RNA-seq notes and trials" />
  

<meta name="author" content="J. Marcelo Rosales R." />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="file-download-or-sra-qc-and-trimming.html"/>
<link rel="next" href="external-ref.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to RNA sequencing (RNA_seq)</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#downloading-and-setting-up-conda-environments."><i class="fa fa-check"></i><b>1.1</b> Downloading and setting up conda environments.</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#install-miniconda-or-anaconda."><i class="fa fa-check"></i><b>1.2</b> Install Miniconda, or Anaconda.</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#conda-terminal-commands."><i class="fa fa-check"></i><b>1.2.1</b> Conda terminal commands.</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#anacondaminiconda-install-summary-of-steps."><i class="fa fa-check"></i><b>1.2.2</b> Anaconda/Miniconda install Summary of Steps.</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#conda-activate-error"><i class="fa fa-check"></i><b>1.2.3</b> Conda activate error</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#alignment-procedures-and-files."><i class="fa fa-check"></i><b>1.3</b> Alignment Procedures and files.</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#using-tophat-early-version"><i class="fa fa-check"></i><b>1.3.1</b> Using Tophat (early version)</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#using-tophat"><i class="fa fa-check"></i><b>1.3.2</b> Using Tophat</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#using-star"><i class="fa fa-check"></i><b>1.3.3</b> Using STAR</a></li>
<li class="chapter" data-level="1.3.4" data-path="index.html"><a href="index.html#Mus-musculus-NCBI-build37-2"><i class="fa fa-check"></i><b>1.3.4</b> Custom/personal way to Download the Mus_musculus_NCBI_build37.2.tar.gz file.</a></li>
<li class="chapter" data-level="1.3.5" data-path="index.html"><a href="index.html#get-the-sample-files."><i class="fa fa-check"></i><b>1.3.5</b> Get the Sample files.</a></li>
<li class="chapter" data-level="1.3.6" data-path="index.html"><a href="index.html#starting-to-analize-fastq.gz-files"><i class="fa fa-check"></i><b>1.3.6</b> Starting to analize fastq.gz files</a></li>
<li class="chapter" data-level="1.3.7" data-path="index.html"><a href="index.html#trimming-with-trimmomatic"><i class="fa fa-check"></i><b>1.3.7</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="1.3.8" data-path="index.html"><a href="index.html#using-star-to-map-genes"><i class="fa fa-check"></i><b>1.3.8</b> Using STAR to map genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html"><i class="fa fa-check"></i><b>2</b> File Download (or SRA), QC, and Trimming</a>
<ul>
<li class="chapter" data-level="2.1" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#trimmomatic"><i class="fa fa-check"></i><b>2.1</b> Trimmomatic</a></li>
<li class="chapter" data-level="2.2" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#indexing-with-star"><i class="fa fa-check"></i><b>2.2</b> Indexing with STAR</a></li>
<li class="chapter" data-level="2.3" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#microbial-contamination"><i class="fa fa-check"></i><b>2.3</b> Microbial contamination</a></li>
<li class="chapter" data-level="2.4" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#feature-counts."><i class="fa fa-check"></i><b>2.4</b> Feature counts.</a></li>
<li class="chapter" data-level="2.5" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#setting-r-environment."><i class="fa fa-check"></i><b>2.5</b> Setting R environment.</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>3</b> Methods</a></li>
<li class="chapter" data-level="4" data-path="external-ref.html"><a href="external-ref.html"><i class="fa fa-check"></i><b>4</b> External Ref</a></li>
<li class="chapter" data-level="5" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><i class="fa fa-check"></i><b>5</b> RNA Seq Differentially Expressed, GO Enrichment, and Pathway analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#background"><i class="fa fa-check"></i><b>5.1</b> <strong>Background</strong></a></li>
<li class="chapter" data-level="5.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#cleaning-data"><i class="fa fa-check"></i><b>5.2</b> <strong>Cleaning Data</strong></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#programs-used"><i class="fa fa-check"></i><b>5.2.1</b> Programs used?</a></li>
<li class="chapter" data-level="5.2.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#acquiring-sequences"><i class="fa fa-check"></i><b>5.2.2</b> Acquiring sequences</a></li>
<li class="chapter" data-level="5.2.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#fastqc"><i class="fa fa-check"></i><b>5.2.3</b> FastQC</a></li>
<li class="chapter" data-level="5.2.4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#trimming-with-trimmomatic-1"><i class="fa fa-check"></i><b>5.2.4</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="5.2.5" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#alignment"><i class="fa fa-check"></i><b>5.2.5</b> Alignment</a></li>
<li class="chapter" data-level="5.2.6" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#assembling-transcripts-with-cufflinks"><i class="fa fa-check"></i><b>5.2.6</b> Assembling transcripts with Cufflinks</a></li>
<li class="chapter" data-level="5.2.7" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#checking-for-contamination"><i class="fa fa-check"></i><b>5.2.7</b> Checking for Contamination</a></li>
<li class="chapter" data-level="5.2.8" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#counting-transcripts"><i class="fa fa-check"></i><b>5.2.8</b> Counting Transcripts</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#differentially-expressed-sequence-identification"><i class="fa fa-check"></i><b>5.3</b> <strong>Differentially Expressed Sequence Identification</strong></a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#analyzing-reads-counts"><i class="fa fa-check"></i><b>5.3.1</b> Analyzing Reads Counts</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e.g.-p53-control-and-ir-and-p53---control-and-ir"><i class="fa fa-check"></i><b>5.4</b> Interactions cause a difference between the lfc betwen pooled data, e.g. p53+/+ (control and IR) and p53-/- (control and IR)</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>5.4.1</b> Exploratory Data Analysis</a></li>
<li class="chapter" data-level="5.4.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#identification-of-differentially-expressed-genes"><i class="fa fa-check"></i><b>5.4.2</b> Identification of Differentially Expressed Genes</a></li>
<li class="chapter" data-level="5.4.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#gene-annotations"><i class="fa fa-check"></i><b>5.4.3</b> Gene Annotations</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#go-enrichment-analysis-using-gostats"><i class="fa fa-check"></i><b>5.5</b> 7. GO Enrichment analysis using GOstats</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#go-enrichment-analysis-of-downregulated-genes"><i class="fa fa-check"></i><b>5.5.1</b> GO Enrichment analysis of downregulated genes</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-analysis-using-expression-data"><i class="fa fa-check"></i><b>5.6</b> 8. Pathway analysis using expression data</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#prepare-data"><i class="fa fa-check"></i><b>5.6.1</b> Prepare data</a></li>
<li class="chapter" data-level="5.6.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#kegg-pathways"><i class="fa fa-check"></i><b>5.6.2</b> KEGG pathways</a></li>
<li class="chapter" data-level="5.6.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-oxidative-phosphorylation."><i class="fa fa-check"></i><b>5.6.3</b> Pathway and regulation of genes for Oxidative phosphorylation.</a></li>
<li class="chapter" data-level="5.6.4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-glycosylphosphatidylinositolgpi-anchor-biosynthesis."><i class="fa fa-check"></i><b>5.6.4</b> Pathway and regulation of genes for Glycosylphosphatidylinositol(GPI)-anchor biosynthesis.</a></li>
<li class="chapter" data-level="5.6.5" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-rna-polymerase."><i class="fa fa-check"></i><b>5.6.5</b> Pathway and regulation of genes for RNA polymerase.</a></li>
<li class="chapter" data-level="5.6.6" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-nucleotide-excision-repair."><i class="fa fa-check"></i><b>5.6.6</b> Pathway and regulation of genes for Nucleotide excision repair.</a></li>
<li class="chapter" data-level="5.6.7" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-non-homologous-end-joining."><i class="fa fa-check"></i><b>5.6.7</b> Pathway and regulation of genes for Non-homologous end-joining.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html"><i class="fa fa-check"></i><b>6</b> WGCNA RNA-seq Blog</a>
<ul>
<li class="chapter" data-level="6.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#WGCNA"><i class="fa fa-check"></i><b>6.1</b> Introduction to the WGCNA Rpackage</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-rpackage-installation-procedure."><i class="fa fa-check"></i><b>6.1.1</b> WGCNA Rpackage Installation procedure.</a></li>
<li class="chapter" data-level="6.1.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#final-installation-workflow"><i class="fa fa-check"></i><b>6.1.2</b> Final installation workflow:</a></li>
<li class="chapter" data-level="6.1.3" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-tutorial."><i class="fa fa-check"></i><b>6.1.3</b> WGCNA tutorial.</a></li>
<li class="chapter" data-level="6.1.4" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#gene-expression-analysis"><i class="fa fa-check"></i><b>6.1.4</b> 3.4 Gene expression analysis</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#rna-seq-analisys-with-kakus-data-sample."><i class="fa fa-check"></i><b>6.2</b> RNA-seq Analisys with Kaku’s Data sample.</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#data-analysis."><i class="fa fa-check"></i><b>6.2.1</b> Data Analysis.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="youtube-references-.html"><a href="youtube-references-.html"><i class="fa fa-check"></i><b>7</b> Youtube references.</a>
<ul>
<li class="chapter" data-level="7.1" data-path="youtube-references-.html"><a href="youtube-references-.html#anaconda-set-up-robust-data-science-environment-with-miniconda-and-conda-forge."><i class="fa fa-check"></i><b>7.1</b> Anaconda set up, Robust Data Science Environment with Miniconda and Conda-Forge.</a></li>
<li class="chapter" data-level="7.2" data-path="youtube-references-.html"><a href="youtube-references-.html#quality-control-preprocessing-of-raw-reads"><i class="fa fa-check"></i><b>7.2</b> Quality control &amp; preprocessing of raw reads</a></li>
<li class="chapter" data-level="7.3" data-path="youtube-references-.html"><a href="youtube-references-.html#trimmomatic."><i class="fa fa-check"></i><b>7.3</b> Trimmomatic.</a></li>
<li class="chapter" data-level="7.4" data-path="youtube-references-.html"><a href="youtube-references-.html#building-genome-index-and-aligning-with-star"><i class="fa fa-check"></i><b>7.4</b> Building Genome Index and Aligning with STAR</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics_RNA-seq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="methods" class="section level1" number="3">
<h1><span class="header-section-number">Chapter 3</span> Methods</h1>
<p>We describe our methods in this chapter.</p>
<p>Set Environment
QC/multiqc of raw Reads
Trimming with Trimmomatic
QC/multiqc of trimmed Reads
Genome Index and Aligning with STAR</p>
<p>Raw Data file names original:<br />
mPDL_RNA7D_Ko2_S1_L001_R2_001_2.fastq.gz
mPDL_RNA7D_Ko1_S1_L001_R1_001_1.fastq.gz
mPDL_RNA7D_Ko1_S1_L001_R2_001_2.fastq.gz
mPDL_RNA7D_Ko2_S1_L001_R1_001_1.fastq.gz</p>
<p>Change (for easy management) to:
mPDL_RNA7D_Ko1_1.fastq.gz
mPDL_RNA7D_Ko1_2.fastq.gz
mPDL_RNA7D_Ko2_1.fastq.gz
mPDL_RNA7D_Ko2_2.fastq.gz</p>
<blockquote>
<p>Seems like original raw data was already splited</p>
</blockquote>
<div class="sourceCode" id="cb58"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb58-1"><a href="methods.html#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the cat file with the file names.</span></span>
<span id="cb58-2"><a href="methods.html#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="op">&gt;</span> filenames.txt</span>
<span id="cb58-3"><a href="methods.html#cb58-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-4"><a href="methods.html#cb58-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Type or copy paste the name of the files to be analyzed.</span></span>
<span id="cb58-5"><a href="methods.html#cb58-5" aria-hidden="true" tabindex="-1"></a><span class="co"># In last line press enter and then control + D to save the file names in the cat file. </span></span>
<span id="cb58-6"><a href="methods.html#cb58-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-7"><a href="methods.html#cb58-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Check the list</span></span>
<span id="cb58-8"><a href="methods.html#cb58-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> filenames.txt</span>
<span id="cb58-9"><a href="methods.html#cb58-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-10"><a href="methods.html#cb58-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Since already spliced not need to run this section. </span></span>
<span id="cb58-11"><a href="methods.html#cb58-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-12"><a href="methods.html#cb58-12" aria-hidden="true" tabindex="-1"></a><span class="co"># In conda environment</span></span>
<span id="cb58-13"><a href="methods.html#cb58-13" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> rawReads</span>
<span id="cb58-14"><a href="methods.html#cb58-14" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> rawReads</span>
<span id="cb58-15"><a href="methods.html#cb58-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-16"><a href="methods.html#cb58-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Download files from webpage source. (one by one)</span></span>
<span id="cb58-17"><a href="methods.html#cb58-17" aria-hidden="true" tabindex="-1"></a><span class="ex">fastq-dump</span> --gzip --split-file mPDL_RNA7D_Ko1 <span class="kw">&amp;</span></span>
<span id="cb58-18"><a href="methods.html#cb58-18" aria-hidden="true" tabindex="-1"></a><span class="co"># The &amp; indicates to run in the background. To see the jobs in the bkg.</span></span>
<span id="cb58-19"><a href="methods.html#cb58-19" aria-hidden="true" tabindex="-1"></a><span class="bu">jobs</span></span>
<span id="cb58-20"><a href="methods.html#cb58-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-21"><a href="methods.html#cb58-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Download files from webpage source. (all at once). Better to do overnight.</span></span>
<span id="cb58-22"><a href="methods.html#cb58-22" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> filenames.txt <span class="kw">|</span> <span class="ex">parallel</span> <span class="st">&quot;fastq-dump --gzip --split-file {}&quot;</span></span>
<span id="cb58-23"><a href="methods.html#cb58-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-24"><a href="methods.html#cb58-24" aria-hidden="true" tabindex="-1"></a><span class="co"># files will be Forward (1) Reverse (2) or Paired reads. </span></span>
<span id="cb58-25"><a href="methods.html#cb58-25" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> install multiqc</span></code></pre></div>
<div class="sourceCode" id="cb59"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb59-1"><a href="methods.html#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Some git preferences for git contributions.  </span></span>
<span id="cb59-2"><a href="methods.html#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="fu">git</span> status</span>
<span id="cb59-3"><a href="methods.html#cb59-3" aria-hidden="true" tabindex="-1"></a><span class="fu">git</span> config --list</span>
<span id="cb59-4"><a href="methods.html#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="co"># set git email</span></span>
<span id="cb59-5"><a href="methods.html#cb59-5" aria-hidden="true" tabindex="-1"></a><span class="fu">git</span> config user.email <span class="st">&quot;email@example.com&quot;</span></span>
<span id="cb59-6"><a href="methods.html#cb59-6" aria-hidden="true" tabindex="-1"></a><span class="co"># confirm emal</span></span>
<span id="cb59-7"><a href="methods.html#cb59-7" aria-hidden="true" tabindex="-1"></a><span class="fu">git</span> config --global user.email</span>
<span id="cb59-8"><a href="methods.html#cb59-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-9"><a href="methods.html#cb59-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-10"><a href="methods.html#cb59-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-11"><a href="methods.html#cb59-11" aria-hidden="true" tabindex="-1"></a><span class="co"># For the rna-seq_test environment. </span></span>
<span id="cb59-12"><a href="methods.html#cb59-12" aria-hidden="true" tabindex="-1"></a><span class="bu">.</span> <span class="ex">/usr/local/anaconda3/bin/activate</span> <span class="kw">&amp;&amp;</span> <span class="ex">conda</span> activate /usr/local/anaconda3/envs/rna-seq_test</span>
<span id="cb59-13"><a href="methods.html#cb59-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-14"><a href="methods.html#cb59-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-15"><a href="methods.html#cb59-15" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="st">&quot;/Volumes/HD-PCFSU3-A/Experiments Data/Genewiz/Trial02&quot;</span></span>
<span id="cb59-16"><a href="methods.html#cb59-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-17"><a href="methods.html#cb59-17" aria-hidden="true" tabindex="-1"></a><span class="co">## Run FAST QC</span></span>
<span id="cb59-18"><a href="methods.html#cb59-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-19"><a href="methods.html#cb59-19" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> rawFastQC </span>
<span id="cb59-20"><a href="methods.html#cb59-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-21"><a href="methods.html#cb59-21" aria-hidden="true" tabindex="-1"></a><span class="ex">fastqc</span> -t 64 rawReads/*fastq.gz -o rawFastQC</span>
<span id="cb59-22"><a href="methods.html#cb59-22" aria-hidden="true" tabindex="-1"></a><span class="ex">fastqc</span> rawReads/*fastq.gz -o rawFastQC</span>
<span id="cb59-23"><a href="methods.html#cb59-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-24"><a href="methods.html#cb59-24" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> rawFastQC</span>
<span id="cb59-25"><a href="methods.html#cb59-25" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span></span>
<span id="cb59-26"><a href="methods.html#cb59-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-27"><a href="methods.html#cb59-27" aria-hidden="true" tabindex="-1"></a><span class="co">## Run multyQC</span></span>
<span id="cb59-28"><a href="methods.html#cb59-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-29"><a href="methods.html#cb59-29" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> .</span>
<span id="cb59-30"><a href="methods.html#cb59-30" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> </span>
<span id="cb59-31"><a href="methods.html#cb59-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-32"><a href="methods.html#cb59-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-33"><a href="methods.html#cb59-33" aria-hidden="true" tabindex="-1"></a><span class="ex">cd..</span></span>
<span id="cb59-34"><a href="methods.html#cb59-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-35"><a href="methods.html#cb59-35" aria-hidden="true" tabindex="-1"></a><span class="co">## Trimming </span></span>
<span id="cb59-36"><a href="methods.html#cb59-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-37"><a href="methods.html#cb59-37" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="op">&gt;</span> filenames2.txt</span>
<span id="cb59-38"><a href="methods.html#cb59-38" aria-hidden="true" tabindex="-1"></a><span class="ex">mPDL_RNA7D_Ko1</span></span>
<span id="cb59-39"><a href="methods.html#cb59-39" aria-hidden="true" tabindex="-1"></a><span class="ex">mPDL_RNA7D_Ko2</span></span>
<span id="cb59-40"><a href="methods.html#cb59-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-41"><a href="methods.html#cb59-41" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> filenames2.txt</span>
<span id="cb59-42"><a href="methods.html#cb59-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-43"><a href="methods.html#cb59-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-44"><a href="methods.html#cb59-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-45"><a href="methods.html#cb59-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-46"><a href="methods.html#cb59-46" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> trimmedReads</span>
<span id="cb59-47"><a href="methods.html#cb59-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-48"><a href="methods.html#cb59-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Pull TrueSeq to folder. </span></span>
<span id="cb59-49"><a href="methods.html#cb59-49" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> /usr/local/anaconda3/envs/rna-seq_test/share/trimmomatic-0.39-2/adapters/TruSeq3-PE-2.fa .</span>
<span id="cb59-50"><a href="methods.html#cb59-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-51"><a href="methods.html#cb59-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-52"><a href="methods.html#cb59-52" aria-hidden="true" tabindex="-1"></a><span class="co"># Run Trimmomatic </span></span>
<span id="cb59-53"><a href="methods.html#cb59-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-54"><a href="methods.html#cb59-54" aria-hidden="true" tabindex="-1"></a><span class="co"># Original (one by one)</span></span>
<span id="cb59-55"><a href="methods.html#cb59-55" aria-hidden="true" tabindex="-1"></a><span class="ex">trimmomatic</span> PE -threads 5 rawReads/mPDL_RNA7D_Ko1_1.fastq.gz rawReads/mPDL_RNA7D_Ko1_2.fastq.gz -baseout trimmedReads/mPDL_RNA7D_Ko1 ILLUMINACLIP:truSeq-PE-2.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36 <span class="op">2&gt;</span> trimmedReads/mPDL_RNA7D_Ko1trimming.log</span>
<span id="cb59-56"><a href="methods.html#cb59-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-57"><a href="methods.html#cb59-57" aria-hidden="true" tabindex="-1"></a><span class="co">#???????????? HOw to determing leading, trailing and minimun length</span></span>
<span id="cb59-58"><a href="methods.html#cb59-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-59"><a href="methods.html#cb59-59" aria-hidden="true" tabindex="-1"></a><span class="co"># Original (all automatically)</span></span>
<span id="cb59-60"><a href="methods.html#cb59-60" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> filenames2.txt <span class="kw">|</span> <span class="ex">parallel</span> -j 4 <span class="st">&quot;trimmomatic PE -threads 5 rawReads/{}_1.fastq.gz rawReads/{}_2.fastq.gz -baseout trimmedReads/{} ILLUMINACLIP:truSeq-PE-2.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36 2&gt; trimmedReads/{}trimming.log&quot;</span></span>
<span id="cb59-61"><a href="methods.html#cb59-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-62"><a href="methods.html#cb59-62" aria-hidden="true" tabindex="-1"></a><span class="co"># Check if jobs are running</span></span>
<span id="cb59-63"><a href="methods.html#cb59-63" aria-hidden="true" tabindex="-1"></a><span class="bu">jobs</span></span>
<span id="cb59-64"><a href="methods.html#cb59-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-65"><a href="methods.html#cb59-65" aria-hidden="true" tabindex="-1"></a><span class="co"># Problems: Takes a long time and sometimes it stops. </span></span>
<span id="cb59-66"><a href="methods.html#cb59-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-67"><a href="methods.html#cb59-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-68"><a href="methods.html#cb59-68" aria-hidden="true" tabindex="-1"></a><span class="co"># The generic trimmomatic command:</span></span>
<span id="cb59-69"><a href="methods.html#cb59-69" aria-hidden="true" tabindex="-1"></a><span class="ex">java</span> -jar trimmomatic-0.39.jar PE inputforward.fq.gz inputreverse.fq.gz outputforwardpaired.fq.gz outputforwardunpaired.fq.gz outputreversepaired.fq.gz outputreverseunpaired.fq.gz ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36</span>
<span id="cb59-70"><a href="methods.html#cb59-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-71"><a href="methods.html#cb59-71" aria-hidden="true" tabindex="-1"></a><span class="co"># Try 03 SUCCESS!!!?</span></span>
<span id="cb59-72"><a href="methods.html#cb59-72" aria-hidden="true" tabindex="-1"></a><span class="ex">trimmomatic</span> PE rawReads/mPDL_RNA7D_Ko1_S1_L001_R1_001_1.fastq.gz rawReads/mPDL_RNA7D_Ko1_S1_L001_R2_001_2.fastq.gz -baseout trimmedReads/mPDL_RNA7D_Ko1_S1_L001 ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36 <span class="op">2&gt;</span> trimmedReads/mPDL_RNA7D_Ko1_S1_L001trimming.log</span>
<span id="cb59-73"><a href="methods.html#cb59-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-74"><a href="methods.html#cb59-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-75"><a href="methods.html#cb59-75" aria-hidden="true" tabindex="-1"></a><span class="co"># Sample 1 Success!!!!???? error but TrimmomaticPE: Completed successfully message.  # error in TruSeq3-PE.fa (No such file or directory)... Change to TruSeq3-PE-2.fa (not tested yet)</span></span>
<span id="cb59-76"><a href="methods.html#cb59-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-77"><a href="methods.html#cb59-77" aria-hidden="true" tabindex="-1"></a><span class="ex">trimmomatic</span> PE rawReads/mPDL_RNA7D_Ko1_S1_L001_R1_001_1.fastq.gz rawReads/mPDL_RNA7D_Ko1_S1_L001_R2_001_2.fastq.gz -baseout trimmedReads/mPDL_RNA7D_Ko1_S1_L001_R1_001 ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36 <span class="op">2&gt;</span> trimmedReads/mPDL_RNA7D_Ko1_S1_L001trimming.log</span>
<span id="cb59-78"><a href="methods.html#cb59-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-79"><a href="methods.html#cb59-79" aria-hidden="true" tabindex="-1"></a><span class="co"># Sample 2 Success!!!!???? error but TrimmomaticPE: Completed successfully message.  # error in TruSeq3-PE.fa (No such file or directory)... Change to TruSeq3-PE-2.fa (not tested yet)</span></span>
<span id="cb59-80"><a href="methods.html#cb59-80" aria-hidden="true" tabindex="-1"></a><span class="ex">trimmomatic</span> PE rawReads/mPDL_RNA7D_Ko2_S1_L001_R1_001_1.fastq.gz rawReads/mPDL_RNA7D_Ko2_S1_L001_R2_001_2.fastq.gz -baseout trimmedReads/mPDL_RNA7D_Ko2_S1_L001_R2_001 ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:10:2:keepBothReads LEADING:3 TRAILING:3 MINLEN:36 <span class="op">2&gt;</span> trimmedReads/mPDL_RNA7D_Ko2_S1_L001trimming.log</span>
<span id="cb59-81"><a href="methods.html#cb59-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-82"><a href="methods.html#cb59-82" aria-hidden="true" tabindex="-1"></a><span class="co"># Pipe and loop for all files.</span></span>
<span id="cb59-83"><a href="methods.html#cb59-83" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> filenames.txt <span class="kw">|</span> <span class="ex">parallel</span> -j 4 <span class="st">&quot;trimmomatic PE rawReads/{}_1.fastq.gz rawReads/{}_2.fastq.gz -baseout trimmedReads/{} ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:10:2keepBothReads LEADING:3 TRAILING:3 MINLEN:36 2&gt; trimmedReads/{}trimming.log&quot;</span> #Not successful. </span>
<span id="cb59-84"><a href="methods.html#cb59-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-85"><a href="methods.html#cb59-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-86"><a href="methods.html#cb59-86" aria-hidden="true" tabindex="-1"></a><span class="co"># In TrimmedRead run multiqc</span></span>
<span id="cb59-87"><a href="methods.html#cb59-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-88"><a href="methods.html#cb59-88" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span></span>
<span id="cb59-89"><a href="methods.html#cb59-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-90"><a href="methods.html#cb59-90" aria-hidden="true" tabindex="-1"></a> <span class="ex">multiqc</span> .</span>
<span id="cb59-91"><a href="methods.html#cb59-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-92"><a href="methods.html#cb59-92" aria-hidden="true" tabindex="-1"></a><span class="co"># fastQC of trimmed files again?</span></span>
<span id="cb59-93"><a href="methods.html#cb59-93" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> trimmedFastQC</span>
<span id="cb59-94"><a href="methods.html#cb59-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-95"><a href="methods.html#cb59-95" aria-hidden="true" tabindex="-1"></a><span class="co"># rm -d trimmedFastQC # Make sure folder is in main folder, if not delete</span></span>
<span id="cb59-96"><a href="methods.html#cb59-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-97"><a href="methods.html#cb59-97" aria-hidden="true" tabindex="-1"></a><span class="co"># Run fastqc... Trimmed files have no file extension. Why???</span></span>
<span id="cb59-98"><a href="methods.html#cb59-98" aria-hidden="true" tabindex="-1"></a><span class="ex">fastqc</span> trimmedReads/* -o trimmedFastQC</span>
<span id="cb59-99"><a href="methods.html#cb59-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-100"><a href="methods.html#cb59-100" aria-hidden="true" tabindex="-1"></a><span class="co"># .zip and qc html files will be created. Run multiqc</span></span>
<span id="cb59-101"><a href="methods.html#cb59-101" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> trimmedFastQC</span>
<span id="cb59-102"><a href="methods.html#cb59-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-103"><a href="methods.html#cb59-103" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> .</span></code></pre></div>
<p>Alignment with STAR</p>
<ol style="list-style-type: decimal">
<li>Download the Mouse genome form the NCBI page. See <a href="index.html#Mus-musculus-NCBI-build37-2">1.3.4</a></li>
</ol>
<div class="sourceCode" id="cb60"><pre class="sourceCode bash"><code class="sourceCode bash"></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Unzip the tar.gz file. FYI: This processes require a long time.</li>
</ol>
<div class="sourceCode" id="cb61"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb61-1"><a href="methods.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the correct environment.</span></span>
<span id="cb61-2"><a href="methods.html#cb61-2" aria-hidden="true" tabindex="-1"></a><span class="bu">.</span> <span class="ex">/usr/local/anaconda3/bin/activate</span> <span class="kw">&amp;&amp;</span> <span class="ex">conda</span> activate /usr/local/anaconda3/envs/rna-seq_test</span>
<span id="cb61-3"><a href="methods.html#cb61-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-4"><a href="methods.html#cb61-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the correct folder/directory.   </span></span>
<span id="cb61-5"><a href="methods.html#cb61-5" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="st">&quot;/Volumes/HD-PCFSU3-A/Experiments Data/Genewiz/60-499583816_60-501009934/Result/Mus Musculus/Mus_musculus_NCBI_build37.2.tar.gz&quot;</span></span>
<span id="cb61-6"><a href="methods.html#cb61-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-7"><a href="methods.html#cb61-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Data/Genewiz/60-499583816_60-501009934/Result/Mus Musculus/Mus_musculus_NCBI_build37.2.tar.gz: Not a directory</span></span>
<span id="cb61-8"><a href="methods.html#cb61-8" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="st">&quot;/Volumes/HD-PCFSU3-A/Experiments Data/Genewiz/60-499583816_60-501009934/Result/Mus Musculus/&quot;</span></span>
<span id="cb61-9"><a href="methods.html#cb61-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-10"><a href="methods.html#cb61-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Unzip file</span></span>
<span id="cb61-11"><a href="methods.html#cb61-11" aria-hidden="true" tabindex="-1"></a> <span class="fu">tar</span> xvfz Mus_musculus_NCBI_build37.2.tar.gz</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Confirm that STAR is installed if not Install STAR.</li>
</ol>
<div class="sourceCode" id="cb62"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb62-1"><a href="methods.html#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure of enviroment.  </span></span>
<span id="cb62-2"><a href="methods.html#cb62-2" aria-hidden="true" tabindex="-1"></a><span class="ex">STAR</span> --help</span>
<span id="cb62-3"><a href="methods.html#cb62-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-4"><a href="methods.html#cb62-4" aria-hidden="true" tabindex="-1"></a><span class="co">#If not found install.  </span></span>
<span id="cb62-5"><a href="methods.html#cb62-5" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> install STAR </span>
<span id="cb62-6"><a href="methods.html#cb62-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-7"><a href="methods.html#cb62-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure is from &quot;bioconda&quot;.  Preceed([y]/n)? message.</span></span>
<span id="cb62-8"><a href="methods.html#cb62-8" aria-hidden="true" tabindex="-1"></a><span class="ex">y</span></span>
<span id="cb62-9"><a href="methods.html#cb62-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-10"><a href="methods.html#cb62-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Check again. Long list of flags.  </span></span>
<span id="cb62-11"><a href="methods.html#cb62-11" aria-hidden="true" tabindex="-1"></a><span class="ex">STAR</span> --help</span>
<span id="cb62-12"><a href="methods.html#cb62-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-13"><a href="methods.html#cb62-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Create directory for starIndex. </span></span>
<span id="cb62-14"><a href="methods.html#cb62-14" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> starIndex</span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Run STAR</li>
</ol>
<p>1st. Index Mus_Musculus whole genome,
2nd. Index Samples against Mus_Musculus Index.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb63-1"><a href="methods.html#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure that the unzip folder of Mus_musculus_NCBI_build37.2.tar.gz (Mus_musculus) is in the same directory as environment/analysis folder in this case:  </span></span>
<span id="cb63-2"><a href="methods.html#cb63-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-3"><a href="methods.html#cb63-3" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="st">&quot;/Volumes/HD-PCFSU3-A/Experiments Data/Genewiz/Trial02/&quot;</span></span>
<span id="cb63-4"><a href="methods.html#cb63-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-5"><a href="methods.html#cb63-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Run STAR. Takes long time. wait until  ..... finished successfully message appears. if not rerun again. </span></span>
<span id="cb63-6"><a href="methods.html#cb63-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-7"><a href="methods.html#cb63-7" aria-hidden="true" tabindex="-1"></a><span class="ex">STAR</span> --runThreadN 64 --runMode genomeGenerate --genomeDir starIndex --genomeFastaFiles Mus_musculus/NCBI/build37.2/Sequence/WholeGenomeFasta/genome.fa --sjdbGTFfile Mus_musculus/NCBI/build37.2/Annotation/Genes/genes.gtf</span></code></pre></div>
<ol start="5" style="list-style-type: decimal">
<li>Align reads</li>
</ol>
<div class="sourceCode" id="cb64"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb64-1"><a href="methods.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create a new directory</span></span>
<span id="cb64-2"><a href="methods.html#cb64-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-3"><a href="methods.html#cb64-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> starAligned</span></code></pre></div>
<p>Git token</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode bash"><code class="sourceCode bash"></code></pre></div>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="file-download-or-sra-qc-and-trimming.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="external-ref.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Bioinformatics_RNA-seq.pdf", "Bioinformatics_RNA-seq.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
