<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 13 8. Pathway analysis using expression data | Bioinformatics_RNA-seq</title>
  <meta name="description" content="RNA-seq notes and trials" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 13 8. Pathway analysis using expression data | Bioinformatics_RNA-seq" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="RNA-seq notes and trials" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 13 8. Pathway analysis using expression data | Bioinformatics_RNA-seq" />
  
  <meta name="twitter:description" content="RNA-seq notes and trials" />
  

<meta name="author" content="J. Marcelo Rosales R." />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="go-enrichment-analysis-using-gostats.html"/>

<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to RNA sequencing (RNA_seq)</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#downloading-and-setting-up-conda-environments."><i class="fa fa-check"></i><b>1.1</b> Downloading and setting up conda environments.</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#install-miniconda-or-anaconda."><i class="fa fa-check"></i><b>1.2</b> Install Miniconda, or Anaconda.</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#conda-terminal-commands."><i class="fa fa-check"></i><b>1.2.1</b> Conda terminal commands.</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#anacondaminiconda-install-summary-of-steps."><i class="fa fa-check"></i><b>1.2.2</b> Anaconda/Miniconda install Summary of Steps.</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#conda-activate-error"><i class="fa fa-check"></i><b>1.2.3</b> Conda activate error</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#alignment-procedures-and-files."><i class="fa fa-check"></i><b>1.3</b> Alignment Procedures and files.</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#using-tophat-early-version"><i class="fa fa-check"></i><b>1.3.1</b> Using Tophat (early version)</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#using-tophat"><i class="fa fa-check"></i><b>1.3.2</b> Using Tophat</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#using-star"><i class="fa fa-check"></i><b>1.3.3</b> Using STAR</a></li>
<li class="chapter" data-level="1.3.4" data-path="index.html"><a href="index.html#custompersonal-way-to-download-the-mus_musculus_ncbi_build37.2.tar.gz-file."><i class="fa fa-check"></i><b>1.3.4</b> Custom/personal way to Download the Mus_musculus_NCBI_build37.2.tar.gz file.</a></li>
<li class="chapter" data-level="1.3.5" data-path="index.html"><a href="index.html#get-the-sample-files."><i class="fa fa-check"></i><b>1.3.5</b> Get the Sample files.</a></li>
<li class="chapter" data-level="1.3.6" data-path="index.html"><a href="index.html#starting-to-analize-fastq.gz-files"><i class="fa fa-check"></i><b>1.3.6</b> Starting to analize fastq.gz files</a></li>
<li class="chapter" data-level="1.3.7" data-path="index.html"><a href="index.html#trimming-with-trimmomatic"><i class="fa fa-check"></i><b>1.3.7</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="1.3.8" data-path="index.html"><a href="index.html#using-star-to-map-genes"><i class="fa fa-check"></i><b>1.3.8</b> Using STAR to map genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html"><i class="fa fa-check"></i><b>2</b> WGCNA RNA-seq Blog</a>
<ul>
<li class="chapter" data-level="2.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#WGCNA"><i class="fa fa-check"></i><b>2.1</b> Introduction to the WGCNA Rpackage</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-rpackage-installation-procedure."><i class="fa fa-check"></i><b>2.1.1</b> WGCNA Rpackage Installation procedure.</a></li>
<li class="chapter" data-level="2.1.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#final-installation-workflow"><i class="fa fa-check"></i><b>2.1.2</b> Final installation workflow:</a></li>
<li class="chapter" data-level="2.1.3" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-tutorial."><i class="fa fa-check"></i><b>2.1.3</b> WGCNA tutorial.</a></li>
<li class="chapter" data-level="2.1.4" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#gene-expression-analysis"><i class="fa fa-check"></i><b>2.1.4</b> 3.4 Gene expression analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#rna-seq-analisys-with-kakus-data-sample."><i class="fa fa-check"></i><b>2.2</b> RNA-seq Analisys with Kaku’s Data sample.</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#data-analysis."><i class="fa fa-check"></i><b>2.2.1</b> Data Analysis.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="sra-download-qc-and-trimming.html"><a href="sra-download-qc-and-trimming.html"><i class="fa fa-check"></i><b>3</b> SRA Download, QC, and Trimming</a></li>
<li class="chapter" data-level="4" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>4</b> Methods</a></li>
<li class="chapter" data-level="5" data-path="youtube-references-.html"><a href="youtube-references-.html"><i class="fa fa-check"></i><b>5</b> Youtube references.</a></li>
<li class="chapter" data-level="6" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>6</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="7" data-path="rna-seq-for-stat736-alex-soupir.html"><a href="rna-seq-for-stat736-alex-soupir.html"><i class="fa fa-check"></i><b>7</b> RNA Seq for STAT736 Alex Soupir</a></li>
<li class="chapter" data-level="8" data-path="background.html"><a href="background.html"><i class="fa fa-check"></i><b>8</b> <strong>Background</strong></a></li>
<li class="chapter" data-level="9" data-path="cleaning-data.html"><a href="cleaning-data.html"><i class="fa fa-check"></i><b>9</b> <strong>Cleaning Data</strong></a>
<ul>
<li class="chapter" data-level="9.1" data-path="cleaning-data.html"><a href="cleaning-data.html#programs-used"><i class="fa fa-check"></i><b>9.1</b> Programs used?</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="cleaning-data.html"><a href="cleaning-data.html#picking-the-right-node"><i class="fa fa-check"></i><b>9.1.1</b> Picking the right node</a></li>
<li class="chapter" data-level="9.1.2" data-path="cleaning-data.html"><a href="cleaning-data.html#creating-slurm-scripts"><i class="fa fa-check"></i><b>9.1.2</b> Creating slurm scripts</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="cleaning-data.html"><a href="cleaning-data.html#acquiring-sequences"><i class="fa fa-check"></i><b>9.2</b> Acquiring sequences</a></li>
<li class="chapter" data-level="9.3" data-path="cleaning-data.html"><a href="cleaning-data.html#fastqc"><i class="fa fa-check"></i><b>9.3</b> FastQC</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="cleaning-data.html"><a href="cleaning-data.html#multiqc"><i class="fa fa-check"></i><b>9.3.1</b> MultiQC</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="cleaning-data.html"><a href="cleaning-data.html#trimming-with-trimmomatic-1"><i class="fa fa-check"></i><b>9.4</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="9.5" data-path="cleaning-data.html"><a href="cleaning-data.html#alignment"><i class="fa fa-check"></i><b>9.5</b> Alignment</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="cleaning-data.html"><a href="cleaning-data.html#using-tophat-1"><i class="fa fa-check"></i><b>9.5.1</b> Using Tophat</a></li>
<li class="chapter" data-level="9.5.2" data-path="cleaning-data.html"><a href="cleaning-data.html#using-star-1"><i class="fa fa-check"></i><b>9.5.2</b> Using STAR</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="cleaning-data.html"><a href="cleaning-data.html#assembling-transcripts-with-cufflinks"><i class="fa fa-check"></i><b>9.6</b> Assembling transcripts with Cufflinks</a></li>
<li class="chapter" data-level="9.7" data-path="cleaning-data.html"><a href="cleaning-data.html#checking-for-contamination"><i class="fa fa-check"></i><b>9.7</b> Checking for Contamination</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="cleaning-data.html"><a href="cleaning-data.html#phix-contamination"><i class="fa fa-check"></i><b>9.7.1</b> PhiX contamination</a></li>
<li class="chapter" data-level="9.7.2" data-path="cleaning-data.html"><a href="cleaning-data.html#rrna-sequences"><i class="fa fa-check"></i><b>9.7.2</b> rRNA Sequences</a></li>
<li class="chapter" data-level="9.7.3" data-path="cleaning-data.html"><a href="cleaning-data.html#bacterial-contamination"><i class="fa fa-check"></i><b>9.7.3</b> Bacterial contamination</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="cleaning-data.html"><a href="cleaning-data.html#counting-transcripts"><i class="fa fa-check"></i><b>9.8</b> Counting Transcripts</a>
<ul>
<li class="chapter" data-level="9.8.1" data-path="cleaning-data.html"><a href="cleaning-data.html#final-qc-of-cleaning-the-data"><i class="fa fa-check"></i><b>9.8.1</b> Final QC of cleaning the data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="differentially-expressed-sequence-identification.html"><a href="differentially-expressed-sequence-identification.html"><i class="fa fa-check"></i><b>10</b> <strong>Differentially Expressed Sequence Identification</strong></a>
<ul>
<li class="chapter" data-level="10.1" data-path="differentially-expressed-sequence-identification.html"><a href="differentially-expressed-sequence-identification.html#analyzing-reads-counts"><i class="fa fa-check"></i><b>10.1</b> Analyzing Reads Counts</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="differentially-expressed-sequence-identification.html"><a href="differentially-expressed-sequence-identification.html#quick-data-exploration"><i class="fa fa-check"></i><b>10.1.1</b> Quick Data Exploration</a></li>
<li class="chapter" data-level="10.1.2" data-path="differentially-expressed-sequence-identification.html"><a href="differentially-expressed-sequence-identification.html#filtering-normalization-and-trasformation-using-deseq2"><i class="fa fa-check"></i><b>10.1.2</b> Filtering, Normalization, and Trasformation using DESeq2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><a href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><i class="fa fa-check"></i><b>11</b> Interactions cause a difference between the lfc betwen pooled data, e.g. p53+/+ (control and IR) and p53-/- (control and IR)</a>
<ul>
<li class="chapter" data-level="11.0.1" data-path="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><a href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>11.0.1</b> Exploratory Data Analysis</a></li>
<li class="chapter" data-level="11.0.2" data-path="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><a href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html#identification-of-differentially-expressed-genes"><i class="fa fa-check"></i><b>11.0.2</b> Identification of Differentially Expressed Genes</a></li>
<li class="chapter" data-level="11.0.3" data-path="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html"><a href="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e-g-p53-control-and-ir-and-p53-control-and-ir.html#gene-annotations"><i class="fa fa-check"></i><b>11.0.3</b> Gene Annotations</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="go-enrichment-analysis-using-gostats.html"><a href="go-enrichment-analysis-using-gostats.html"><i class="fa fa-check"></i><b>12</b> 7. GO Enrichment analysis using GOstats</a>
<ul>
<li class="chapter" data-level="12.1" data-path="go-enrichment-analysis-using-gostats.html"><a href="go-enrichment-analysis-using-gostats.html#go-enrichment-analysis-of-downregulated-genes"><i class="fa fa-check"></i><b>12.1</b> GO Enrichment analysis of downregulated genes</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html"><i class="fa fa-check"></i><b>13</b> 8. Pathway analysis using expression data</a>
<ul>
<li class="chapter" data-level="13.1" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#prepare-data"><i class="fa fa-check"></i><b>13.1</b> Prepare data</a></li>
<li class="chapter" data-level="13.2" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#kegg-pathways"><i class="fa fa-check"></i><b>13.2</b> KEGG pathways</a></li>
<li class="chapter" data-level="13.3" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-oxidative-phosphorylation."><i class="fa fa-check"></i><b>13.3</b> Pathway and regulation of genes for Oxidative phosphorylation.</a></li>
<li class="chapter" data-level="13.4" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-glycosylphosphatidylinositolgpi-anchor-biosynthesis."><i class="fa fa-check"></i><b>13.4</b> Pathway and regulation of genes for Glycosylphosphatidylinositol(GPI)-anchor biosynthesis.</a></li>
<li class="chapter" data-level="13.5" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-rna-polymerase."><i class="fa fa-check"></i><b>13.5</b> Pathway and regulation of genes for RNA polymerase.</a></li>
<li class="chapter" data-level="13.6" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-nucleotide-excision-repair."><i class="fa fa-check"></i><b>13.6</b> Pathway and regulation of genes for Nucleotide excision repair.</a></li>
<li class="chapter" data-level="13.7" data-path="pathway-analysis-using-expression-data.html"><a href="pathway-analysis-using-expression-data.html#pathway-and-regulation-of-genes-for-non-homologous-end-joining."><i class="fa fa-check"></i><b>13.7</b> Pathway and regulation of genes for Non-homologous end-joining.</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics_RNA-seq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="pathway-analysis-using-expression-data" class="section level1" number="13">
<h1><span class="header-section-number">Chapter 13</span> 8. Pathway analysis using expression data</h1>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="pathway-analysis-using-expression-data.html#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="co"># bioconductor packages</span></span>
<span id="cb119-2"><a href="pathway-analysis-using-expression-data.html#cb119-2" aria-hidden="true" tabindex="-1"></a><span class="co"># source(&quot;https://bioconductor.org/biocLite.R&quot;);</span></span>
<span id="cb119-3"><a href="pathway-analysis-using-expression-data.html#cb119-3" aria-hidden="true" tabindex="-1"></a><span class="co"># biocLite(c(&quot;pathview&quot;,&quot;gage&quot;,&quot;gageData&quot;))</span></span>
<span id="cb119-4"><a href="pathway-analysis-using-expression-data.html#cb119-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pathview) </span>
<span id="cb119-5"><a href="pathway-analysis-using-expression-data.html#cb119-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gage) </span></code></pre></div>
<div id="prepare-data" class="section level2" number="13.1">
<h2><span class="header-section-number">13.1</span> Prepare data</h2>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="pathway-analysis-using-expression-data.html#cb120-1" aria-hidden="true" tabindex="-1"></a>foldchanges <span class="ot">=</span> res<span class="sc">$</span>log2FoldChange</span>
<span id="cb120-2"><a href="pathway-analysis-using-expression-data.html#cb120-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(foldchanges) <span class="ot">=</span> res<span class="sc">$</span>entrez</span>
<span id="cb120-3"><a href="pathway-analysis-using-expression-data.html#cb120-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(foldchanges)</span></code></pre></div>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="pathway-analysis-using-expression-data.html#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gageData)</span>
<span id="cb121-2"><a href="pathway-analysis-using-expression-data.html#cb121-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(go.sets.mm)</span>
<span id="cb121-3"><a href="pathway-analysis-using-expression-data.html#cb121-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(go.subs.mm)</span>
<span id="cb121-4"><a href="pathway-analysis-using-expression-data.html#cb121-4" aria-hidden="true" tabindex="-1"></a>gobpsets <span class="ot">=</span> go.sets.mm[go.subs.mm<span class="sc">$</span>BP]</span>
<span id="cb121-5"><a href="pathway-analysis-using-expression-data.html#cb121-5" aria-hidden="true" tabindex="-1"></a>gobpres <span class="ot">=</span> <span class="fu">gage</span>(foldchanges, <span class="at">gsets=</span>gobpsets, <span class="at">same.dir=</span><span class="cn">TRUE</span>)</span>
<span id="cb121-6"><a href="pathway-analysis-using-expression-data.html#cb121-6" aria-hidden="true" tabindex="-1"></a><span class="co">#lapply(gobpres, head)</span></span>
<span id="cb121-7"><a href="pathway-analysis-using-expression-data.html#cb121-7" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Greater&quot;</span>)</span>
<span id="cb121-8"><a href="pathway-analysis-using-expression-data.html#cb121-8" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(gobpres<span class="sc">$</span>greater)) <span class="sc">%&gt;%</span></span>
<span id="cb121-9"><a href="pathway-analysis-using-expression-data.html#cb121-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb121-10"><a href="pathway-analysis-using-expression-data.html#cb121-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span>
<span id="cb121-11"><a href="pathway-analysis-using-expression-data.html#cb121-11" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Less&quot;</span>)</span>
<span id="cb121-12"><a href="pathway-analysis-using-expression-data.html#cb121-12" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(gobpres<span class="sc">$</span>less)) <span class="sc">%&gt;%</span></span>
<span id="cb121-13"><a href="pathway-analysis-using-expression-data.html#cb121-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb121-14"><a href="pathway-analysis-using-expression-data.html#cb121-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span>
<span id="cb121-15"><a href="pathway-analysis-using-expression-data.html#cb121-15" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Stats&quot;</span>)</span>
<span id="cb121-16"><a href="pathway-analysis-using-expression-data.html#cb121-16" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(gobpres<span class="sc">$</span>stats)) <span class="sc">%&gt;%</span></span>
<span id="cb121-17"><a href="pathway-analysis-using-expression-data.html#cb121-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb121-18"><a href="pathway-analysis-using-expression-data.html#cb121-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span></code></pre></div>
</div>
<div id="kegg-pathways" class="section level2" number="13.2">
<h2><span class="header-section-number">13.2</span> KEGG pathways</h2>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="pathway-analysis-using-expression-data.html#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gageData)</span>
<span id="cb122-2"><a href="pathway-analysis-using-expression-data.html#cb122-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(kegg.sets.mm)</span>
<span id="cb122-3"><a href="pathway-analysis-using-expression-data.html#cb122-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(sigmet.idx.mm)</span>
<span id="cb122-4"><a href="pathway-analysis-using-expression-data.html#cb122-4" aria-hidden="true" tabindex="-1"></a>kegg.sets.mm <span class="ot">=</span> kegg.sets.mm[sigmet.idx.mm]</span>
<span id="cb122-5"><a href="pathway-analysis-using-expression-data.html#cb122-5" aria-hidden="true" tabindex="-1"></a><span class="co">#head(kegg.sets.mm, 3)</span></span>
<span id="cb122-6"><a href="pathway-analysis-using-expression-data.html#cb122-6" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Greater&quot;</span>)</span>
<span id="cb122-7"><a href="pathway-analysis-using-expression-data.html#cb122-7" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(kegg.sets.mm<span class="sc">$</span>greater)) <span class="sc">%&gt;%</span></span>
<span id="cb122-8"><a href="pathway-analysis-using-expression-data.html#cb122-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb122-9"><a href="pathway-analysis-using-expression-data.html#cb122-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span>
<span id="cb122-10"><a href="pathway-analysis-using-expression-data.html#cb122-10" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Less&quot;</span>)</span>
<span id="cb122-11"><a href="pathway-analysis-using-expression-data.html#cb122-11" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(kegg.sets.mm<span class="sc">$</span>less)) <span class="sc">%&gt;%</span></span>
<span id="cb122-12"><a href="pathway-analysis-using-expression-data.html#cb122-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb122-13"><a href="pathway-analysis-using-expression-data.html#cb122-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span>
<span id="cb122-14"><a href="pathway-analysis-using-expression-data.html#cb122-14" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Stats&quot;</span>)</span>
<span id="cb122-15"><a href="pathway-analysis-using-expression-data.html#cb122-15" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(kegg.sets.mm<span class="sc">$</span>stats)) <span class="sc">%&gt;%</span></span>
<span id="cb122-16"><a href="pathway-analysis-using-expression-data.html#cb122-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb122-17"><a href="pathway-analysis-using-expression-data.html#cb122-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span>
<span id="cb122-18"><a href="pathway-analysis-using-expression-data.html#cb122-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the results</span></span>
<span id="cb122-19"><a href="pathway-analysis-using-expression-data.html#cb122-19" aria-hidden="true" tabindex="-1"></a>keggres <span class="ot">=</span> <span class="fu">gage</span>(foldchanges, <span class="at">gsets=</span>kegg.sets.mm, <span class="at">same.dir=</span><span class="cn">TRUE</span>)</span>
<span id="cb122-20"><a href="pathway-analysis-using-expression-data.html#cb122-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Look at both up (greater), down (less), and statatistics.</span></span>
<span id="cb122-21"><a href="pathway-analysis-using-expression-data.html#cb122-21" aria-hidden="true" tabindex="-1"></a><span class="co">#lapply(keggres, head, n=10)</span></span>
<span id="cb122-22"><a href="pathway-analysis-using-expression-data.html#cb122-22" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Greater&quot;</span>)</span>
<span id="cb122-23"><a href="pathway-analysis-using-expression-data.html#cb122-23" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(keggres<span class="sc">$</span>greater)) <span class="sc">%&gt;%</span></span>
<span id="cb122-24"><a href="pathway-analysis-using-expression-data.html#cb122-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb122-25"><a href="pathway-analysis-using-expression-data.html#cb122-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span>
<span id="cb122-26"><a href="pathway-analysis-using-expression-data.html#cb122-26" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Less&quot;</span>)</span>
<span id="cb122-27"><a href="pathway-analysis-using-expression-data.html#cb122-27" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(keggres<span class="sc">$</span>less)) <span class="sc">%&gt;%</span></span>
<span id="cb122-28"><a href="pathway-analysis-using-expression-data.html#cb122-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb122-29"><a href="pathway-analysis-using-expression-data.html#cb122-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span>
<span id="cb122-30"><a href="pathway-analysis-using-expression-data.html#cb122-30" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Stats&quot;</span>)</span>
<span id="cb122-31"><a href="pathway-analysis-using-expression-data.html#cb122-31" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(keggres<span class="sc">$</span>stats)) <span class="sc">%&gt;%</span></span>
<span id="cb122-32"><a href="pathway-analysis-using-expression-data.html#cb122-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb122-33"><a href="pathway-analysis-using-expression-data.html#cb122-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="pathway-analysis-using-expression-data.html#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the pathways</span></span>
<span id="cb123-2"><a href="pathway-analysis-using-expression-data.html#cb123-2" aria-hidden="true" tabindex="-1"></a>keggrespathways <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">id=</span><span class="fu">rownames</span>(keggres<span class="sc">$</span>less), keggres<span class="sc">$</span>less) <span class="sc">%&gt;%</span> </span>
<span id="cb123-3"><a href="pathway-analysis-using-expression-data.html#cb123-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tbl_df</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb123-4"><a href="pathway-analysis-using-expression-data.html#cb123-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>()<span class="sc">&lt;=</span><span class="dv">5</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb123-5"><a href="pathway-analysis-using-expression-data.html#cb123-5" aria-hidden="true" tabindex="-1"></a>  .<span class="sc">$</span>id <span class="sc">%&gt;%</span> </span>
<span id="cb123-6"><a href="pathway-analysis-using-expression-data.html#cb123-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.character</span>()</span>
<span id="cb123-7"><a href="pathway-analysis-using-expression-data.html#cb123-7" aria-hidden="true" tabindex="-1"></a>keggrespathways</span></code></pre></div>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="pathway-analysis-using-expression-data.html#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the IDs.</span></span>
<span id="cb124-2"><a href="pathway-analysis-using-expression-data.html#cb124-2" aria-hidden="true" tabindex="-1"></a>keggresids <span class="ot">=</span> <span class="fu">substr</span>(keggrespathways, <span class="at">start=</span><span class="dv">1</span>, <span class="at">stop=</span><span class="dv">8</span>)</span>
<span id="cb124-3"><a href="pathway-analysis-using-expression-data.html#cb124-3" aria-hidden="true" tabindex="-1"></a>keggresids</span></code></pre></div>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="pathway-analysis-using-expression-data.html#cb125-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define plotting function for applying later</span></span>
<span id="cb125-2"><a href="pathway-analysis-using-expression-data.html#cb125-2" aria-hidden="true" tabindex="-1"></a>plot_pathway <span class="ot">=</span> <span class="cf">function</span>(pid) <span class="fu">pathview</span>(<span class="at">gene.data=</span>foldchanges, <span class="at">pathway.id=</span>pid, <span class="at">species=</span><span class="st">&quot;mmu&quot;</span>, <span class="at">new.signature=</span><span class="cn">FALSE</span>)</span>
<span id="cb125-3"><a href="pathway-analysis-using-expression-data.html#cb125-3" aria-hidden="true" tabindex="-1"></a><span class="co"># plot multiple pathways (plots saved to disk and returns a throwaway list object)</span></span>
<span id="cb125-4"><a href="pathway-analysis-using-expression-data.html#cb125-4" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">=</span> <span class="fu">sapply</span>(keggresids, <span class="cf">function</span>(pid) <span class="fu">pathview</span>(<span class="at">gene.data=</span>foldchanges, <span class="at">pathway.id=</span>pid, <span class="at">species=</span><span class="st">&quot;mmu&quot;</span>))</span></code></pre></div>
</div>
<div id="pathway-and-regulation-of-genes-for-oxidative-phosphorylation." class="section level2" number="13.3">
<h2><span class="header-section-number">13.3</span> Pathway and regulation of genes for Oxidative phosphorylation.</h2>
<p><code>![Oxidative Phosphorylation](./Mouse_RNA_Seq_p53_genotoxic_files/figure-html/mmu00190.pathview.png)</code></p>
</div>
<div id="pathway-and-regulation-of-genes-for-glycosylphosphatidylinositolgpi-anchor-biosynthesis." class="section level2" number="13.4">
<h2><span class="header-section-number">13.4</span> Pathway and regulation of genes for Glycosylphosphatidylinositol(GPI)-anchor biosynthesis.</h2>
<p><code>![Glycosylphosphatidylinositol(GPI)-anchor biosynthesis](./Mouse_RNA_Seq_p53_genotoxic_files/figure-html/mmu00563.pathview.png)</code></p>
</div>
<div id="pathway-and-regulation-of-genes-for-rna-polymerase." class="section level2" number="13.5">
<h2><span class="header-section-number">13.5</span> Pathway and regulation of genes for RNA polymerase.</h2>
<p><code>![RNA polymerase](./Mouse_RNA_Seq_p53_genotoxic_files/figure-html/mmu03020.pathview.png)</code></p>
</div>
<div id="pathway-and-regulation-of-genes-for-nucleotide-excision-repair." class="section level2" number="13.6">
<h2><span class="header-section-number">13.6</span> Pathway and regulation of genes for Nucleotide excision repair.</h2>
<p><code>![Nucleotide excision repair](./Mouse_RNA_Seq_p53_genotoxic_files/figure-html/mmu03420.pathview.png)</code></p>
</div>
<div id="pathway-and-regulation-of-genes-for-non-homologous-end-joining." class="section level2" number="13.7">
<h2><span class="header-section-number">13.7</span> Pathway and regulation of genes for Non-homologous end-joining.</h2>
<p><code>![Non-homologous end-joining](./Mouse_RNA_Seq_p53_genotoxic_files/figure-html/mmu03450.pathview.png)</code></p>
<p>#De novo Assembly {#denovo-assembly}</p>
<p><em>AT THIS TIME, MINICONDA3 WAS INSTALLED SO FURTHER TOOLS ARE ALL INSTALLED UNDER THIS</em></p>
<p>Having the genome for RNA-sequencing analysis is very useful, but sometimes it is not available so <em>de novo</em> assembly is used. <em>de novo</em> assembly is using the reads from sequencing to create longer seqeunces called contigs (contiguous sequences). These sequences are then compared to a protein database to get an idea of what proteins are possibly present. The raw sequences are also mapped back to the contigs, treating the contigs as the “genome,” much like we do when we have the genome from the host available.</p>
<p>Lets jump right into it. We are going to be using <a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki">Trinity</a> for the <em>de novo</em> assembly and analysis because there are many tools built into the Trinity tool that allow for the building of contigs, counting of sequences, and even DeSeq/edgeR analyses.</p>
<p>To install Trinity, all I did was ran <strong>conda install -c bioconda trinity</strong>. This installs all of the previously mentioned tools in one go. In addition to Trinity, BLAST (<strong>conda install -c bioconda blast</strong>) and RSEM (<strong>conda install -c bioconda rsem</strong>) were installed.</p>
<div class="sourceCode" id="cb126"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb126-1"><a href="pathway-analysis-using-expression-data.html#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="ex">Trinity</span> <span class="at">--seqType</span> fq <span class="dt">\</span></span>
<span id="cb126-2"><a href="pathway-analysis-using-expression-data.html#cb126-2" aria-hidden="true" tabindex="-1"></a>--left trimmedReads/SRR2121770_Trimmed_1P.fq.gz,trimmedReads/SRR2121771_Trimmed_1P.fq.gz,trimmedReads/SRR2121774_Trimmed_1P.fq.gz,trimmedReads/SRR2121775_Trimmed_1P.fq.gz,trimmedReads/SRR2121778_Trimmed_1P.fq.gz,trimmedReads/SRR2121779_Trimmed_1P.fq.gz,trimmedReads/SRR2121780_Trimmed_1P.fq.gz,trimmedReads/SRR2121781_Trimmed_1P.fq.gz,trimmedReads/SRR2121786_Trimmed_1P.fq.gz,trimmedReads/SRR2121787_Trimmed_1P.fq.gz,trimmedReads/SRR2121788_Trimmed_1P.fq.gz,trimmedReads/SRR2121789_Trimmed_1P.fq.gz <span class="dt">\</span></span>
<span id="cb126-3"><a href="pathway-analysis-using-expression-data.html#cb126-3" aria-hidden="true" tabindex="-1"></a>--right trimmedReads/SRR2121770_Trimmed_2P.fq.gz,trimmedReads/SRR2121771_Trimmed_2P.fq.gz,trimmedReads/SRR2121774_Trimmed_2P.fq.gz,trimmedReads/SRR2121775_Trimmed_2P.fq.gz,trimmedReads/SRR2121778_Trimmed_2P.fq.gz,trimmedReads/SRR2121779_Trimmed_2P.fq.gz,trimmedReads/SRR2121780_Trimmed_2P.fq.gz,trimmedReads/SRR2121781_Trimmed_2P.fq.gz,trimmedReads/SRR2121786_Trimmed_2P.fq.gz,trimmedReads/SRR2121787_Trimmed_2P.fq.gz,trimmedReads/SRR2121788_Trimmed_2P.fq.gz,trimmedReads/SRR2121789_Trimmed_2P.fq.gz <span class="dt">\</span></span>
<span id="cb126-4"><a href="pathway-analysis-using-expression-data.html#cb126-4" aria-hidden="true" tabindex="-1"></a>--CPU 80 <span class="at">--max_memory</span> 2000G <span class="at">--min_contig_length</span> 150</span></code></pre></div>
<div class="sourceCode" id="cb127"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb127-1"><a href="pathway-analysis-using-expression-data.html#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="ex">TrinityStats.pl</span> trinity_out_dir/Trinity.fasta</span>
<span id="cb127-2"><a href="pathway-analysis-using-expression-data.html#cb127-2" aria-hidden="true" tabindex="-1"></a><span class="co">################################</span></span>
<span id="cb127-3"><a href="pathway-analysis-using-expression-data.html#cb127-3" aria-hidden="true" tabindex="-1"></a><span class="co">## Counts of transcripts, etc.</span></span>
<span id="cb127-4"><a href="pathway-analysis-using-expression-data.html#cb127-4" aria-hidden="true" tabindex="-1"></a><span class="co">################################</span></span>
<span id="cb127-5"><a href="pathway-analysis-using-expression-data.html#cb127-5" aria-hidden="true" tabindex="-1"></a><span class="ex">Total</span> trinity <span class="st">&#39;genes&#39;</span>:  602870</span>
<span id="cb127-6"><a href="pathway-analysis-using-expression-data.html#cb127-6" aria-hidden="true" tabindex="-1"></a><span class="ex">Total</span> trinity transcripts:      730297</span>
<span id="cb127-7"><a href="pathway-analysis-using-expression-data.html#cb127-7" aria-hidden="true" tabindex="-1"></a><span class="ex">Percent</span> GC: 45.89</span>
<span id="cb127-8"><a href="pathway-analysis-using-expression-data.html#cb127-8" aria-hidden="true" tabindex="-1"></a><span class="co">########################################</span></span>
<span id="cb127-9"><a href="pathway-analysis-using-expression-data.html#cb127-9" aria-hidden="true" tabindex="-1"></a><span class="ex">Stats</span> based on ALL transcript contigs:</span>
<span id="cb127-10"><a href="pathway-analysis-using-expression-data.html#cb127-10" aria-hidden="true" tabindex="-1"></a><span class="co">########################################</span></span>
<span id="cb127-11"><a href="pathway-analysis-using-expression-data.html#cb127-11" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N10: 8589</span>
<span id="cb127-12"><a href="pathway-analysis-using-expression-data.html#cb127-12" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N20: 5955</span>
<span id="cb127-13"><a href="pathway-analysis-using-expression-data.html#cb127-13" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N30: 4400</span>
<span id="cb127-14"><a href="pathway-analysis-using-expression-data.html#cb127-14" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N40: 3276</span>
<span id="cb127-15"><a href="pathway-analysis-using-expression-data.html#cb127-15" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N50: 2413</span>
<span id="cb127-16"><a href="pathway-analysis-using-expression-data.html#cb127-16" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Median</span> contig length: 413</span>
<span id="cb127-17"><a href="pathway-analysis-using-expression-data.html#cb127-17" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Average</span> contig: 1028.24</span>
<span id="cb127-18"><a href="pathway-analysis-using-expression-data.html#cb127-18" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Total</span> assembled bases: 750921120</span>
<span id="cb127-19"><a href="pathway-analysis-using-expression-data.html#cb127-19" aria-hidden="true" tabindex="-1"></a><span class="co">#####################################################</span></span>
<span id="cb127-20"><a href="pathway-analysis-using-expression-data.html#cb127-20" aria-hidden="true" tabindex="-1"></a><span class="co">## Stats based on ONLY LONGEST ISOFORM per &#39;GENE&#39;:</span></span>
<span id="cb127-21"><a href="pathway-analysis-using-expression-data.html#cb127-21" aria-hidden="true" tabindex="-1"></a><span class="co">#####################################################</span></span>
<span id="cb127-22"><a href="pathway-analysis-using-expression-data.html#cb127-22" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N10: 5470</span>
<span id="cb127-23"><a href="pathway-analysis-using-expression-data.html#cb127-23" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N20: 3434</span>
<span id="cb127-24"><a href="pathway-analysis-using-expression-data.html#cb127-24" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N30: 2391</span>
<span id="cb127-25"><a href="pathway-analysis-using-expression-data.html#cb127-25" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N40: 1741</span>
<span id="cb127-26"><a href="pathway-analysis-using-expression-data.html#cb127-26" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N50: 1285</span>
<span id="cb127-27"><a href="pathway-analysis-using-expression-data.html#cb127-27" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Median</span> contig length: 343</span>
<span id="cb127-28"><a href="pathway-analysis-using-expression-data.html#cb127-28" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Average</span> contig: 703.04</span>
<span id="cb127-29"><a href="pathway-analysis-using-expression-data.html#cb127-29" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Total</span> assembled bases: 423841627</span></code></pre></div>
<div class="sourceCode" id="cb128"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb128-1"><a href="pathway-analysis-using-expression-data.html#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bowtie2-build</span> trinity_out_dir/Trinity.fasta trinity_out_dir/Trinity.fasta</span>
<span id="cb128-2"><a href="pathway-analysis-using-expression-data.html#cb128-2" aria-hidden="true" tabindex="-1"></a><span class="ex">bowtie2</span> <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out</span></code></pre></div>
<p>#download uniprot swiss-prot db
#<a href="ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz" class="uri">ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz</a></p>
<div class="sourceCode" id="cb129"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb129-1"><a href="pathway-analysis-using-expression-data.html#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gunzip</span> uniprot_sprot.fasta.gz</span>
<span id="cb129-2"><a href="pathway-analysis-using-expression-data.html#cb129-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> blast_protdb</span>
<span id="cb129-3"><a href="pathway-analysis-using-expression-data.html#cb129-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> uniprot_sprot.fasta blast_protdb/.</span>
<span id="cb129-4"><a href="pathway-analysis-using-expression-data.html#cb129-4" aria-hidden="true" tabindex="-1"></a><span class="ex">makeblastdb</span> <span class="at">-in</span> uniprot_sprot.fasta <span class="at">-dbtype</span> prot</span>
<span id="cb129-5"><a href="pathway-analysis-using-expression-data.html#cb129-5" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ..</span></code></pre></div>
<div class="sourceCode" id="cb130"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb130-1"><a href="pathway-analysis-using-expression-data.html#cb130-1" aria-hidden="true" tabindex="-1"></a><span class="ex">blastx</span> <span class="at">-query</span> trinity_out_dir/Trinity.fasta <span class="at">-db</span> blast_protdb/uniprot_sprot.fasta <span class="at">-out</span> blastx.outfmt6 <span class="at">-evalue</span> 1e-20 <span class="at">-num_threads</span> 80 <span class="at">-max_target_seqs</span> 1 <span class="at">-outfmt</span> 6</span>
<span id="cb130-2"><a href="pathway-analysis-using-expression-data.html#cb130-2" aria-hidden="true" tabindex="-1"></a><span class="ex">analyze_blastPlus_topHit_coverage.pl</span> blastx.outfmt6 trinity_out_dir/Trinity.fasta blast_protdb/uniprot_sprot.fasta <span class="kw">|</span> <span class="ex">column</span> <span class="at">-t</span></span>
<span id="cb130-3"><a href="pathway-analysis-using-expression-data.html#cb130-3" aria-hidden="true" tabindex="-1"></a><span class="co">#hit_pct_cov_bin  count_in_bin  &gt;bin_below</span></span>
<span id="cb130-4"><a href="pathway-analysis-using-expression-data.html#cb130-4" aria-hidden="true" tabindex="-1"></a><span class="ex">100</span>               7361          7361</span>
<span id="cb130-5"><a href="pathway-analysis-using-expression-data.html#cb130-5" aria-hidden="true" tabindex="-1"></a><span class="ex">90</span>                1117          8478</span>
<span id="cb130-6"><a href="pathway-analysis-using-expression-data.html#cb130-6" aria-hidden="true" tabindex="-1"></a><span class="ex">80</span>                981           9459</span>
<span id="cb130-7"><a href="pathway-analysis-using-expression-data.html#cb130-7" aria-hidden="true" tabindex="-1"></a><span class="ex">70</span>                971           10430</span>
<span id="cb130-8"><a href="pathway-analysis-using-expression-data.html#cb130-8" aria-hidden="true" tabindex="-1"></a><span class="ex">60</span>                1003          11433</span>
<span id="cb130-9"><a href="pathway-analysis-using-expression-data.html#cb130-9" aria-hidden="true" tabindex="-1"></a><span class="ex">50</span>                865           12298</span>
<span id="cb130-10"><a href="pathway-analysis-using-expression-data.html#cb130-10" aria-hidden="true" tabindex="-1"></a><span class="ex">40</span>                924           13222</span>
<span id="cb130-11"><a href="pathway-analysis-using-expression-data.html#cb130-11" aria-hidden="true" tabindex="-1"></a><span class="ex">30</span>                1102          14324</span>
<span id="cb130-12"><a href="pathway-analysis-using-expression-data.html#cb130-12" aria-hidden="true" tabindex="-1"></a><span class="ex">20</span>                1478          15802</span>
<span id="cb130-13"><a href="pathway-analysis-using-expression-data.html#cb130-13" aria-hidden="true" tabindex="-1"></a><span class="ex">10</span>                932           16734</span></code></pre></div>
<div class="sourceCode" id="cb131"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb131-1"><a href="pathway-analysis-using-expression-data.html#cb131-1" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121770_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121770_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121770.coordSorted.bam</span>
<span id="cb131-2"><a href="pathway-analysis-using-expression-data.html#cb131-2" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121771_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121771_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121771.coordSorted.bam</span>
<span id="cb131-3"><a href="pathway-analysis-using-expression-data.html#cb131-3" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121774_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121774_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121774.coordSorted.bam</span>
<span id="cb131-4"><a href="pathway-analysis-using-expression-data.html#cb131-4" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121775_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121775_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121775.coordSorted.bam</span>
<span id="cb131-5"><a href="pathway-analysis-using-expression-data.html#cb131-5" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121778_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121778_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121778.coordSorted.bam</span>
<span id="cb131-6"><a href="pathway-analysis-using-expression-data.html#cb131-6" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121779_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121779_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121779.coordSorted.bam</span>
<span id="cb131-7"><a href="pathway-analysis-using-expression-data.html#cb131-7" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121780_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121780_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121780.coordSorted.bam</span>
<span id="cb131-8"><a href="pathway-analysis-using-expression-data.html#cb131-8" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121781_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121781_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121781.coordSorted.bam</span>
<span id="cb131-9"><a href="pathway-analysis-using-expression-data.html#cb131-9" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121786_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121786_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121786.coordSorted.bam</span>
<span id="cb131-10"><a href="pathway-analysis-using-expression-data.html#cb131-10" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121787_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121787_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121787.coordSorted.bam</span>
<span id="cb131-11"><a href="pathway-analysis-using-expression-data.html#cb131-11" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121788_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121788_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121788.coordSorted.bam</span>
<span id="cb131-12"><a href="pathway-analysis-using-expression-data.html#cb131-12" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121789_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121789_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121789.coordSorted.bam</span></code></pre></div>
<div class="sourceCode" id="cb132"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb132-1"><a href="pathway-analysis-using-expression-data.html#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121770_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121770_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121770.RSEM</span>
<span id="cb132-2"><a href="pathway-analysis-using-expression-data.html#cb132-2" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121771_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121771_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121771.RSEM</span>
<span id="cb132-3"><a href="pathway-analysis-using-expression-data.html#cb132-3" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121774_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121774_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121774.RSEM</span>
<span id="cb132-4"><a href="pathway-analysis-using-expression-data.html#cb132-4" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121775_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121775_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121775.RSEM</span>
<span id="cb132-5"><a href="pathway-analysis-using-expression-data.html#cb132-5" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121778_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121778_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121778.RSEM</span>
<span id="cb132-6"><a href="pathway-analysis-using-expression-data.html#cb132-6" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121779_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121779_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121779.RSEM</span>
<span id="cb132-7"><a href="pathway-analysis-using-expression-data.html#cb132-7" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121780_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121780_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121780.RSEM</span>
<span id="cb132-8"><a href="pathway-analysis-using-expression-data.html#cb132-8" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121781_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121781_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121781.RSEM</span>
<span id="cb132-9"><a href="pathway-analysis-using-expression-data.html#cb132-9" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121786_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121786_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121786.RSEM</span>
<span id="cb132-10"><a href="pathway-analysis-using-expression-data.html#cb132-10" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121787_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121787_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121787.RSEM</span>
<span id="cb132-11"><a href="pathway-analysis-using-expression-data.html#cb132-11" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121788_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121788_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121788.RSEM</span>
<span id="cb132-12"><a href="pathway-analysis-using-expression-data.html#cb132-12" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121789_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121789_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121789.RSEM</span></code></pre></div>
<div class="sourceCode" id="cb133"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb133-1"><a href="pathway-analysis-using-expression-data.html#cb133-1" aria-hidden="true" tabindex="-1"></a><span class="bu">time</span> abundance_estimates_to_matrix.pl <span class="at">--est_method</span> RSEM <span class="at">--out_prefix</span> Trinity_trans SRR2121770.RSEM/SRR2121770.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb133-2"><a href="pathway-analysis-using-expression-data.html#cb133-2" aria-hidden="true" tabindex="-1"></a>SRR2121771.RSEM/SRR2121771.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb133-3"><a href="pathway-analysis-using-expression-data.html#cb133-3" aria-hidden="true" tabindex="-1"></a>SRR2121774.RSEM/SRR2121774.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb133-4"><a href="pathway-analysis-using-expression-data.html#cb133-4" aria-hidden="true" tabindex="-1"></a>SRR2121775.RSEM/SRR2121775.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb133-5"><a href="pathway-analysis-using-expression-data.html#cb133-5" aria-hidden="true" tabindex="-1"></a>SRR2121778.RSEM/SRR2121778.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb133-6"><a href="pathway-analysis-using-expression-data.html#cb133-6" aria-hidden="true" tabindex="-1"></a>SRR2121779.RSEM/SRR2121779.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb133-7"><a href="pathway-analysis-using-expression-data.html#cb133-7" aria-hidden="true" tabindex="-1"></a>SRR2121780.RSEM/SRR2121780.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb133-8"><a href="pathway-analysis-using-expression-data.html#cb133-8" aria-hidden="true" tabindex="-1"></a>SRR2121781.RSEM/SRR2121781.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb133-9"><a href="pathway-analysis-using-expression-data.html#cb133-9" aria-hidden="true" tabindex="-1"></a>SRR2121786.RSEM/SRR2121786.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb133-10"><a href="pathway-analysis-using-expression-data.html#cb133-10" aria-hidden="true" tabindex="-1"></a>SRR2121787.RSEM/SRR2121787.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb133-11"><a href="pathway-analysis-using-expression-data.html#cb133-11" aria-hidden="true" tabindex="-1"></a>SRR2121788.RSEM/SRR2121788.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb133-12"><a href="pathway-analysis-using-expression-data.html#cb133-12" aria-hidden="true" tabindex="-1"></a>SRR2121789.RSEM/SRR2121789.RSEM.trans.results <span class="at">--gene_trans_map</span> none</span></code></pre></div>
<div class="sourceCode" id="cb134"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb134-1"><a href="pathway-analysis-using-expression-data.html#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="bu">time</span> abundance_estimates_to_matrix.pl <span class="at">--est_method</span> RSEM <span class="at">--out_prefix</span> Trinity_genes SRR2121770.RSEM/SRR2121770.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb134-2"><a href="pathway-analysis-using-expression-data.html#cb134-2" aria-hidden="true" tabindex="-1"></a>SRR2121771.RSEM/SRR2121771.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb134-3"><a href="pathway-analysis-using-expression-data.html#cb134-3" aria-hidden="true" tabindex="-1"></a>SRR2121774.RSEM/SRR2121774.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb134-4"><a href="pathway-analysis-using-expression-data.html#cb134-4" aria-hidden="true" tabindex="-1"></a>SRR2121775.RSEM/SRR2121775.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb134-5"><a href="pathway-analysis-using-expression-data.html#cb134-5" aria-hidden="true" tabindex="-1"></a>SRR2121778.RSEM/SRR2121778.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb134-6"><a href="pathway-analysis-using-expression-data.html#cb134-6" aria-hidden="true" tabindex="-1"></a>SRR2121779.RSEM/SRR2121779.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb134-7"><a href="pathway-analysis-using-expression-data.html#cb134-7" aria-hidden="true" tabindex="-1"></a>SRR2121780.RSEM/SRR2121780.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb134-8"><a href="pathway-analysis-using-expression-data.html#cb134-8" aria-hidden="true" tabindex="-1"></a>SRR2121781.RSEM/SRR2121781.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb134-9"><a href="pathway-analysis-using-expression-data.html#cb134-9" aria-hidden="true" tabindex="-1"></a>SRR2121786.RSEM/SRR2121786.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb134-10"><a href="pathway-analysis-using-expression-data.html#cb134-10" aria-hidden="true" tabindex="-1"></a>SRR2121787.RSEM/SRR2121787.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb134-11"><a href="pathway-analysis-using-expression-data.html#cb134-11" aria-hidden="true" tabindex="-1"></a>SRR2121788.RSEM/SRR2121788.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb134-12"><a href="pathway-analysis-using-expression-data.html#cb134-12" aria-hidden="true" tabindex="-1"></a>SRR2121789.RSEM/SRR2121789.RSEM.genes.results <span class="at">--gene_trans_map</span> none</span></code></pre></div>
<div class="sourceCode" id="cb135"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb135-1"><a href="pathway-analysis-using-expression-data.html#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="ex">contig_Exn50_statistic.pl</span> Trinity_trans.TMM.EXPR.matrix trinity_out_dir/Trinity.fasta <span class="op">&gt;</span> ExN50_trans.stats</span></code></pre></div>
<p>Make samples.txt</p>
<div class="sourceCode" id="cb136"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb136-1"><a href="pathway-analysis-using-expression-data.html#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="ex">mock-</span>   SRR2121770.RSEM</span>
<span id="cb136-2"><a href="pathway-analysis-using-expression-data.html#cb136-2" aria-hidden="true" tabindex="-1"></a><span class="ex">mock-</span>   SRR2121771.RSEM</span>
<span id="cb136-3"><a href="pathway-analysis-using-expression-data.html#cb136-3" aria-hidden="true" tabindex="-1"></a><span class="ex">IR-</span> SRR2121774.RSEM</span>
<span id="cb136-4"><a href="pathway-analysis-using-expression-data.html#cb136-4" aria-hidden="true" tabindex="-1"></a><span class="ex">IR-</span> SRR2121775.RSEM</span>
<span id="cb136-5"><a href="pathway-analysis-using-expression-data.html#cb136-5" aria-hidden="true" tabindex="-1"></a><span class="ex">mock+</span>   SRR2121778.RSEM</span>
<span id="cb136-6"><a href="pathway-analysis-using-expression-data.html#cb136-6" aria-hidden="true" tabindex="-1"></a><span class="ex">mock+</span>   SRR2121779.RSEM</span>
<span id="cb136-7"><a href="pathway-analysis-using-expression-data.html#cb136-7" aria-hidden="true" tabindex="-1"></a><span class="ex">mock+</span>   SRR2121780.RSEM</span>
<span id="cb136-8"><a href="pathway-analysis-using-expression-data.html#cb136-8" aria-hidden="true" tabindex="-1"></a><span class="ex">mock+</span>   SRR2121781.RSEM</span>
<span id="cb136-9"><a href="pathway-analysis-using-expression-data.html#cb136-9" aria-hidden="true" tabindex="-1"></a><span class="ex">IR+</span> SRR2121786.RSEM</span>
<span id="cb136-10"><a href="pathway-analysis-using-expression-data.html#cb136-10" aria-hidden="true" tabindex="-1"></a><span class="ex">IR+</span> SRR2121787.RSEM</span>
<span id="cb136-11"><a href="pathway-analysis-using-expression-data.html#cb136-11" aria-hidden="true" tabindex="-1"></a><span class="ex">IR+</span> SRR2121788.RSEM</span>
<span id="cb136-12"><a href="pathway-analysis-using-expression-data.html#cb136-12" aria-hidden="true" tabindex="-1"></a><span class="ex">IR+</span> SRR2121789.RSEM</span></code></pre></div>
<p>The way that the installed version of <em>run_DE_analysis.pl</em> tries to install or check for edgeR doesn’t work for the new versions of R (3.6.1). In the perl script I just had to edit a few lines under the edgeR section for it to run smoothly, as well as install *BiocManager before hand.</p>
<p>I changed the perl script, under <em>sub run_edgeR_sample_pair</em>, when writing the R script to look like:</p>
<div class="sourceCode" id="cb137"><pre class="sourceCode perl"><code class="sourceCode perl"><span id="cb137-1"><a href="pathway-analysis-using-expression-data.html#cb137-1" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span> <span class="dt">$ofh</span> <span class="ot">&quot;</span><span class="st">if (! require(edgeR)) {</span><span class="ch">\n</span><span class="ot">&quot;</span>;</span>
<span id="cb137-2"><a href="pathway-analysis-using-expression-data.html#cb137-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span> <span class="dt">$ofh</span> <span class="ot">&quot;</span><span class="st">   install.packages(\&quot;BiocManager\&quot;)</span><span class="ch">\n</span><span class="ot">&quot;</span>;</span>
<span id="cb137-3"><a href="pathway-analysis-using-expression-data.html#cb137-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span> <span class="dt">$ofh</span> <span class="ot">&quot;</span><span class="st">   BiocManager::install(\&quot;edgeR\&quot;)</span><span class="ch">\n</span><span class="ot">&quot;</span>;</span>
<span id="cb137-4"><a href="pathway-analysis-using-expression-data.html#cb137-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span> <span class="dt">$ofh</span> <span class="ot">&quot;</span><span class="st">   library(edgeR)</span><span class="ch">\n</span><span class="ot">&quot;</span>;</span>
<span id="cb137-5"><a href="pathway-analysis-using-expression-data.html#cb137-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span> <span class="dt">$ofh</span> <span class="ot">&quot;</span><span class="st">}</span><span class="ch">\n\n</span><span class="ot">&quot;</span>;</span></code></pre></div>
<p>To make sure BiocManager and edgeR work, they were installed in the R terminal.</p>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="pathway-analysis-using-expression-data.html#cb138-1" aria-hidden="true" tabindex="-1"></a>R <span class="co">#runs the installed version of R in terminal mode</span></span>
<span id="cb138-2"><a href="pathway-analysis-using-expression-data.html#cb138-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb138-3"><a href="pathway-analysis-using-expression-data.html#cb138-3" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;edgeR&quot;</span>)</span></code></pre></div>
<p>Since we don’t have at least 3 replicates for each of the treatments, edgeR wants a dispersion parameter. 0.035 seems to be pretty common. If the dispersion parameter has the squareroot taken of it we get ~ 0.19, meaning that we are saying the true abundance for each gene can vary up or down by 19% between replicates.</p>
<div class="sourceCode" id="cb139"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb139-1"><a href="pathway-analysis-using-expression-data.html#cb139-1" aria-hidden="true" tabindex="-1"></a><span class="ex">run_DE_analysis.pl</span> <span class="at">--matrix</span> Trinity_trans.counts.matrix <span class="at">--method</span> edgeR <span class="at">--output</span> edgeR_trans <span class="at">--dispersion</span> 0.035 <span class="at">--samples_file</span> samples.txt</span></code></pre></div>
<div class="sourceCode" id="cb140"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb140-1"><a href="pathway-analysis-using-expression-data.html#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> edgeR_trans/Trinity_trans.counts.matrix.IR-_vs_mock-.edgeR.DE_results <span class="kw">|</span> <span class="ex">column</span> <span class="at">-t</span></span>
<span id="cb140-2"><a href="pathway-analysis-using-expression-data.html#cb140-2" aria-hidden="true" tabindex="-1"></a>                          <span class="ex">sampleA</span>  sampleB  logFC           logCPM             PValue                FDR</span>
<span id="cb140-3"><a href="pathway-analysis-using-expression-data.html#cb140-3" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN2468_c0_g4_i1</span>   IR-      mock-  <span class="at">-14.7753615825754</span>  4.45315041918     5.51021995613131e-115  5.85185359341145e-110</span>
<span id="cb140-4"><a href="pathway-analysis-using-expression-data.html#cb140-4" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN2505_c0_g1_i2</span>   IR-      mock-  <span class="at">-10.1817621187735</span>  4.11815268306374  8.54695600746168e-88   4.53843363996215e-83</span>
<span id="cb140-5"><a href="pathway-analysis-using-expression-data.html#cb140-5" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN3231_c0_g2_i15</span>  IR-      mock-  <span class="at">-13.5922145319911</span>  3.27307743751541  6.29068575237937e-81   2.2269027563423e-76</span>
<span id="cb140-6"><a href="pathway-analysis-using-expression-data.html#cb140-6" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN4975_c0_g1_i7</span>   IR-      mock-  13.4941229058571   3.17132827495398  5.25978730838885e-79   1.39647353037724e-74</span>
<span id="cb140-7"><a href="pathway-analysis-using-expression-data.html#cb140-7" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN4360_c0_g1_i5</span>   IR-      mock-  <span class="at">-13.5215746751431</span>  3.20272600532575  1.0222845072488e-70    2.17133229339645e-66</span>
<span id="cb140-8"><a href="pathway-analysis-using-expression-data.html#cb140-8" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN128_c0_g1_i20</span>   IR-      mock-  13.1582688371709   2.83599676163376  1.8443766792831e-69    3.26454672233108e-65</span>
<span id="cb140-9"><a href="pathway-analysis-using-expression-data.html#cb140-9" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN644_c1_g1_i6</span>    IR-      mock-  <span class="at">-5.91850038396836</span>  4.31456225100214  8.01440426808572e-67   1.21589961895815e-62</span>
<span id="cb140-10"><a href="pathway-analysis-using-expression-data.html#cb140-10" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN2043_c0_g1_i13</span>  IR-      mock-  13.7803059024088   3.45719012794474  3.30519380237759e-66   4.38764477265625e-62</span>
<span id="cb140-11"><a href="pathway-analysis-using-expression-data.html#cb140-11" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN3321_c0_g1_i2</span>   IR-      mock-  12.8215942182179   2.49995284585527  9.5059376972195e-61    1.1217006482719e-56</span></code></pre></div>
<div class="sourceCode" id="cb141"><pre class="sourceCode bash"><code class="sourceCode bash"></code></pre></div>
<div class="sourceCode" id="cb142"><pre class="sourceCode bash"><code class="sourceCode bash"></code></pre></div>
<div class="sourceCode" id="cb143"><pre class="sourceCode bash"><code class="sourceCode bash"></code></pre></div>
<div class="sourceCode" id="cb144"><pre class="sourceCode bash"><code class="sourceCode bash"></code></pre></div>
<div class="sourceCode" id="cb145"><pre class="sourceCode bash"><code class="sourceCode bash"></code></pre></div>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="pathway-analysis-using-expression-data.html#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install(&quot;EnsDb.Mmusculus.v79&quot;)</span></span>
<span id="cb146-2"><a href="pathway-analysis-using-expression-data.html#cb146-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GO.db)</span>
<span id="cb146-3"><a href="pathway-analysis-using-expression-data.html#cb146-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GOstats)</span>
<span id="cb146-4"><a href="pathway-analysis-using-expression-data.html#cb146-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EnsDb.Mmusculus.v79)</span>
<span id="cb146-5"><a href="pathway-analysis-using-expression-data.html#cb146-5" aria-hidden="true" tabindex="-1"></a>res<span class="fl">.3</span> <span class="ot">=</span> <span class="fu">subset</span>(res<span class="fl">.2</span>, <span class="sc">!</span><span class="fu">is.na</span>(res<span class="fl">.2</span><span class="sc">$</span>entrezgene_id))</span>
<span id="cb146-6"><a href="pathway-analysis-using-expression-data.html#cb146-6" aria-hidden="true" tabindex="-1"></a>selectedGenes <span class="ot">=</span> <span class="fu">unique</span>(resSig[resSig<span class="sc">$</span>log2FoldChange<span class="sc">&gt;</span><span class="dv">0</span>,<span class="st">&#39;entrezgene_id&#39;</span>])  <span class="co"># upregulated genes</span></span>
<span id="cb146-7"><a href="pathway-analysis-using-expression-data.html#cb146-7" aria-hidden="true" tabindex="-1"></a>universeGenes <span class="ot">=</span>  <span class="fu">unique</span>( <span class="fu">mapIds</span>(org.Mm.eg.db,</span>
<span id="cb146-8"><a href="pathway-analysis-using-expression-data.html#cb146-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">keys=</span><span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.[0-9]*$&quot;</span>,<span class="st">&quot;&quot;</span>,res<span class="fl">.3</span><span class="sc">$</span>entrezgene_id),  <span class="co"># this is causing problems for mapping</span></span>
<span id="cb146-9"><a href="pathway-analysis-using-expression-data.html#cb146-9" aria-hidden="true" tabindex="-1"></a>                     <span class="at">column=</span><span class="st">&quot;ENSEMBL&quot;</span>,</span>
<span id="cb146-10"><a href="pathway-analysis-using-expression-data.html#cb146-10" aria-hidden="true" tabindex="-1"></a>                     <span class="at">keytype=</span><span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb146-11"><a href="pathway-analysis-using-expression-data.html#cb146-11" aria-hidden="true" tabindex="-1"></a>                     <span class="at">multiVals=</span><span class="st">&quot;first&quot;</span>)</span>
<span id="cb146-12"><a href="pathway-analysis-using-expression-data.html#cb146-12" aria-hidden="true" tabindex="-1"></a>                    )</span>
<span id="cb146-13"><a href="pathway-analysis-using-expression-data.html#cb146-13" aria-hidden="true" tabindex="-1"></a>hgCutoff <span class="ot">&lt;-</span> <span class="fl">0.001</span></span>
<span id="cb146-14"><a href="pathway-analysis-using-expression-data.html#cb146-14" aria-hidden="true" tabindex="-1"></a>params <span class="ot">&lt;-</span> <span class="fu">new</span>(<span class="st">&quot;GOHyperGParams&quot;</span>,</span>
<span id="cb146-15"><a href="pathway-analysis-using-expression-data.html#cb146-15" aria-hidden="true" tabindex="-1"></a>              <span class="at">geneIds=</span>selectedGenes,</span>
<span id="cb146-16"><a href="pathway-analysis-using-expression-data.html#cb146-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">universeGeneIds=</span>universeGenes,</span>
<span id="cb146-17"><a href="pathway-analysis-using-expression-data.html#cb146-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">annotation=</span><span class="st">&quot;org.Hs.eg.db&quot;</span>,</span>
<span id="cb146-18"><a href="pathway-analysis-using-expression-data.html#cb146-18" aria-hidden="true" tabindex="-1"></a>              <span class="at">ontology=</span><span class="st">&quot;BP&quot;</span>,</span>
<span id="cb146-19"><a href="pathway-analysis-using-expression-data.html#cb146-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff=</span>hgCutoff,</span>
<span id="cb146-20"><a href="pathway-analysis-using-expression-data.html#cb146-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">conditional=</span><span class="cn">FALSE</span>,</span>
<span id="cb146-21"><a href="pathway-analysis-using-expression-data.html#cb146-21" aria-hidden="true" tabindex="-1"></a>              <span class="at">testDirection=</span><span class="st">&quot;over&quot;</span>)</span>
<span id="cb146-22"><a href="pathway-analysis-using-expression-data.html#cb146-22" aria-hidden="true" tabindex="-1"></a>hgOver <span class="ot">&lt;-</span> <span class="fu">hyperGTest</span>(params)</span>
<span id="cb146-23"><a href="pathway-analysis-using-expression-data.html#cb146-23" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(hgOver)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,]</span></code></pre></div>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="pathway-analysis-using-expression-data.html#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install(&quot;AnnotationHub&quot;)</span></span>
<span id="cb147-2"><a href="pathway-analysis-using-expression-data.html#cb147-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationHub)</span>
<span id="cb147-3"><a href="pathway-analysis-using-expression-data.html#cb147-3" aria-hidden="true" tabindex="-1"></a>hub <span class="ot">=</span> <span class="fu">AnnotationHub</span>()</span>
<span id="cb147-4"><a href="pathway-analysis-using-expression-data.html#cb147-4" aria-hidden="true" tabindex="-1"></a><span class="fu">query</span>(hub, <span class="fu">c</span>(<span class="st">&quot;Medicago&quot;</span>, <span class="st">&quot;OrgDb&quot;</span>))</span>
<span id="cb147-5"><a href="pathway-analysis-using-expression-data.html#cb147-5" aria-hidden="true" tabindex="-1"></a>Mt.orgdb <span class="ot">=</span> hub[[<span class="st">&quot;AH72235&quot;</span>]]</span>
<span id="cb147-6"><a href="pathway-analysis-using-expression-data.html#cb147-6" aria-hidden="true" tabindex="-1"></a>Mt.orgdb</span></code></pre></div>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="pathway-analysis-using-expression-data.html#cb148-1" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install(&quot;AnnotationHub&quot;)</span></span>
<span id="cb148-2"><a href="pathway-analysis-using-expression-data.html#cb148-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationHub)</span>
<span id="cb148-3"><a href="pathway-analysis-using-expression-data.html#cb148-3" aria-hidden="true" tabindex="-1"></a>hub <span class="ot">=</span> <span class="fu">AnnotationHub</span>()</span>
<span id="cb148-4"><a href="pathway-analysis-using-expression-data.html#cb148-4" aria-hidden="true" tabindex="-1"></a><span class="fu">query</span>(hub, <span class="fu">c</span>(<span class="st">&quot;Glycine&quot;</span>, <span class="st">&quot;OrgDb&quot;</span>))</span>
<span id="cb148-5"><a href="pathway-analysis-using-expression-data.html#cb148-5" aria-hidden="true" tabindex="-1"></a>Gm.orgdb <span class="ot">=</span> hub[[<span class="st">&quot;AH72142&quot;</span>]]</span>
<span id="cb148-6"><a href="pathway-analysis-using-expression-data.html#cb148-6" aria-hidden="true" tabindex="-1"></a>Gm.orgdb</span></code></pre></div>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="pathway-analysis-using-expression-data.html#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationDbi)</span>
<span id="cb149-2"><a href="pathway-analysis-using-expression-data.html#cb149-2" aria-hidden="true" tabindex="-1"></a><span class="fu">keytypes</span>(Mt.orgdb)</span>
<span id="cb149-3"><a href="pathway-analysis-using-expression-data.html#cb149-3" aria-hidden="true" tabindex="-1"></a><span class="co">#key.Gm = keys(Mt.orgdb, keytype=&quot;GENENAME&quot;)</span></span>
<span id="cb149-4"><a href="pathway-analysis-using-expression-data.html#cb149-4" aria-hidden="true" tabindex="-1"></a><span class="co">#head(key.Gm)</span></span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="go-enrichment-analysis-using-gostats.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Bioinformatics_RNA-seq.pdf", "Bioinformatics_RNA-seq.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
