<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 R code modules to be used for stand-alone analysis. | Bioinformatics_RNA-seq</title>
  <meta name="description" content="RNA-seq notes and trials" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 R code modules to be used for stand-alone analysis. | Bioinformatics_RNA-seq" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="RNA-seq notes and trials" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 R code modules to be used for stand-alone analysis. | Bioinformatics_RNA-seq" />
  
  <meta name="twitter:description" content="RNA-seq notes and trials" />
  

<meta name="author" content="J. Marcelo Rosales R." />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="rna-seq-mkdata.html"/>
<link rel="next" href="wgcna-rna-seq-blog.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to RNA sequencing (RNA_seq)</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#downloading-and-setting-up-conda-environments."><i class="fa fa-check"></i><b>1.1</b> Downloading and setting up conda environments.</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#install-miniconda-or-anaconda."><i class="fa fa-check"></i><b>1.2</b> Install Miniconda, or Anaconda.</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#conda-terminal-commands."><i class="fa fa-check"></i><b>1.2.1</b> Conda terminal commands.</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#anacondaminiconda-install-summary-of-steps."><i class="fa fa-check"></i><b>1.2.2</b> Anaconda/Miniconda install Summary of Steps.</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#conda-activate-error"><i class="fa fa-check"></i><b>1.2.3</b> Conda activate error</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#alignment-procedures-and-files."><i class="fa fa-check"></i><b>1.3</b> Alignment Procedures and files.</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#using-tophat-early-version"><i class="fa fa-check"></i><b>1.3.1</b> Using Tophat (early version)</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#using-tophat"><i class="fa fa-check"></i><b>1.3.2</b> Using Tophat</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#using-star"><i class="fa fa-check"></i><b>1.3.3</b> Using STAR</a></li>
<li class="chapter" data-level="1.3.4" data-path="index.html"><a href="index.html#Mus-musculus-NCBI-build37-2"><i class="fa fa-check"></i><b>1.3.4</b> Custom/personal way to Download the Mus_musculus_NCBI_build37.2.tar.gz file.</a></li>
<li class="chapter" data-level="1.3.5" data-path="index.html"><a href="index.html#get-the-sample-files."><i class="fa fa-check"></i><b>1.3.5</b> Get the Sample files.</a></li>
<li class="chapter" data-level="1.3.6" data-path="index.html"><a href="index.html#starting-to-analize-fastq.gz-files"><i class="fa fa-check"></i><b>1.3.6</b> Starting to analize fastq.gz files</a></li>
<li class="chapter" data-level="1.3.7" data-path="index.html"><a href="index.html#trimming-with-trimmomatic"><i class="fa fa-check"></i><b>1.3.7</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="1.3.8" data-path="index.html"><a href="index.html#using-star-to-map-genes"><i class="fa fa-check"></i><b>1.3.8</b> Using STAR to map genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html"><i class="fa fa-check"></i><b>2</b> File Download (or SRA), QC, and Trimming</a>
<ul>
<li class="chapter" data-level="2.1" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#trimmomatic"><i class="fa fa-check"></i><b>2.1</b> Trimmomatic</a></li>
<li class="chapter" data-level="2.2" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#indexing-with-star"><i class="fa fa-check"></i><b>2.2</b> Indexing with STAR</a></li>
<li class="chapter" data-level="2.3" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#microbial-contamination"><i class="fa fa-check"></i><b>2.3</b> Microbial contamination</a></li>
<li class="chapter" data-level="2.4" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#feature-counts."><i class="fa fa-check"></i><b>2.4</b> Feature counts.</a></li>
<li class="chapter" data-level="2.5" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#setting-r-environment."><i class="fa fa-check"></i><b>2.5</b> Setting R environment.</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>3</b> Methods</a></li>
<li class="chapter" data-level="4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><i class="fa fa-check"></i><b>4</b> RNA Seq Differentially Expressed, GO Enrichment, and Pathway analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#background"><i class="fa fa-check"></i><b>4.1</b> <strong>Background</strong></a></li>
<li class="chapter" data-level="4.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#cleaning-data"><i class="fa fa-check"></i><b>4.2</b> <strong>Cleaning Data</strong></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#programs-used"><i class="fa fa-check"></i><b>4.2.1</b> Programs used?</a></li>
<li class="chapter" data-level="4.2.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#acquiring-sequences"><i class="fa fa-check"></i><b>4.2.2</b> Acquiring sequences</a></li>
<li class="chapter" data-level="4.2.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#fastqc"><i class="fa fa-check"></i><b>4.2.3</b> FastQC</a></li>
<li class="chapter" data-level="4.2.4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#trimming-with-trimmomatic-1"><i class="fa fa-check"></i><b>4.2.4</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="4.2.5" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#alignment"><i class="fa fa-check"></i><b>4.2.5</b> Alignment</a></li>
<li class="chapter" data-level="4.2.6" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#assembling-transcripts-with-cufflinks"><i class="fa fa-check"></i><b>4.2.6</b> Assembling transcripts with Cufflinks</a></li>
<li class="chapter" data-level="4.2.7" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#checking-for-contamination"><i class="fa fa-check"></i><b>4.2.7</b> Checking for Contamination</a></li>
<li class="chapter" data-level="4.2.8" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#counting-transcripts"><i class="fa fa-check"></i><b>4.2.8</b> Counting Transcripts</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#differentially-expressed-sequence-identification"><i class="fa fa-check"></i><b>4.3</b> <strong>Differentially Expressed Sequence Identification</strong></a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#analyzing-reads-counts"><i class="fa fa-check"></i><b>4.3.1</b> Analyzing Reads Counts</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e.g.-p53-control-and-ir-and-p53---control-and-ir"><i class="fa fa-check"></i><b>4.4</b> Interactions cause a difference between the lfc betwen pooled data, e.g. p53+/+ (control and IR) and p53-/- (control and IR)</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>4.4.1</b> Exploratory Data Analysis</a></li>
<li class="chapter" data-level="4.4.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#identification-of-differentially-expressed-genes"><i class="fa fa-check"></i><b>4.4.2</b> Identification of Differentially Expressed Genes</a></li>
<li class="chapter" data-level="4.4.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#gene-annotations"><i class="fa fa-check"></i><b>4.4.3</b> Gene Annotations</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#go-enrichment-analysis-using-gostats"><i class="fa fa-check"></i><b>4.5</b> 7. GO Enrichment analysis using GOstats</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#go-enrichment-analysis-of-downregulated-genes"><i class="fa fa-check"></i><b>4.5.1</b> GO Enrichment analysis of downregulated genes</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-analysis-using-expression-data"><i class="fa fa-check"></i><b>4.6</b> 8. Pathway analysis using expression data</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#prepare-data"><i class="fa fa-check"></i><b>4.6.1</b> Prepare data</a></li>
<li class="chapter" data-level="4.6.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#kegg-pathways"><i class="fa fa-check"></i><b>4.6.2</b> KEGG pathways</a></li>
<li class="chapter" data-level="4.6.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-oxidative-phosphorylation."><i class="fa fa-check"></i><b>4.6.3</b> Pathway and regulation of genes for Oxidative phosphorylation.</a></li>
<li class="chapter" data-level="4.6.4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-glycosylphosphatidylinositolgpi-anchor-biosynthesis."><i class="fa fa-check"></i><b>4.6.4</b> Pathway and regulation of genes for Glycosylphosphatidylinositol(GPI)-anchor biosynthesis.</a></li>
<li class="chapter" data-level="4.6.5" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-rna-polymerase."><i class="fa fa-check"></i><b>4.6.5</b> Pathway and regulation of genes for RNA polymerase.</a></li>
<li class="chapter" data-level="4.6.6" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-nucleotide-excision-repair."><i class="fa fa-check"></i><b>4.6.6</b> Pathway and regulation of genes for Nucleotide excision repair.</a></li>
<li class="chapter" data-level="4.6.7" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-non-homologous-end-joining."><i class="fa fa-check"></i><b>4.6.7</b> Pathway and regulation of genes for Non-homologous end-joining.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html"><i class="fa fa-check"></i><b>5</b> RNA-Seq-MKdata</a>
<ul>
<li class="chapter" data-level="5.1" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#pca"><i class="fa fa-check"></i><b>5.1</b> PCA</a></li>
<li class="chapter" data-level="5.2" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#differential-expression"><i class="fa fa-check"></i><b>5.2</b> Differential expression</a></li>
<li class="chapter" data-level="5.3" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#rna-seq-scienceparksg"><i class="fa fa-check"></i><b>5.3</b> RNA-seq (ScienceparkSG)</a></li>
<li class="chapter" data-level="5.4" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#false-discovery-rates-fdr-benjamini-hocheberg-method"><i class="fa fa-check"></i><b>5.4</b> False discovery rates FDR / Benjamini-Hocheberg method</a></li>
<li class="chapter" data-level="5.5" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#volcano-plot"><i class="fa fa-check"></i><b>5.5</b> Volcano plot</a></li>
<li class="chapter" data-level="5.6" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#heatmap-normalization"><i class="fa fa-check"></i><b>5.6</b> Heatmap &amp; normalization</a></li>
<li class="chapter" data-level="5.7" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#ma-plots"><i class="fa fa-check"></i><b>5.7</b> MA plots</a></li>
<li class="chapter" data-level="5.8" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#eastern-new-mexico-university-enmu-method"><i class="fa fa-check"></i><b>5.8</b> Eastern New Mexico University ENMU method</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html"><i class="fa fa-check"></i><b>6</b> R code modules to be used for stand-alone analysis.</a>
<ul>
<li class="chapter" data-level="6.1" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#local-installation"><i class="fa fa-check"></i><b>6.1</b> Local installation</a></li>
<li class="chapter" data-level="6.2" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#install-packages"><i class="fa fa-check"></i><b>6.2</b> Install packages</a></li>
<li class="chapter" data-level="6.3" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#global-variables"><i class="fa fa-check"></i><b>6.3</b> Global variables</a></li>
<li class="chapter" data-level="6.4" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#read-data-and-pre-process"><i class="fa fa-check"></i><b>6.4</b> Read data and pre-process</a></li>
<li class="chapter" data-level="6.5" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#hierarchical-clustering"><i class="fa fa-check"></i><b>6.5</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="6.6" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#pca-1"><i class="fa fa-check"></i><b>6.6</b> PCA</a></li>
<li class="chapter" data-level="6.7" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#k-means-clustering"><i class="fa fa-check"></i><b>6.7</b> K-means clustering</a></li>
<li class="chapter" data-level="6.8" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#differential-expression-1"><i class="fa fa-check"></i><b>6.8</b> Differential expression</a></li>
<li class="chapter" data-level="6.9" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#pathway-analysis"><i class="fa fa-check"></i><b>6.9</b> Pathway analysis</a></li>
<li class="chapter" data-level="6.10" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#genome-wide-view"><i class="fa fa-check"></i><b>6.10</b> Genome-wide view</a></li>
<li class="chapter" data-level="6.11" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#biclustering"><i class="fa fa-check"></i><b>6.11</b> Biclustering</a></li>
<li class="chapter" data-level="6.12" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#co-expression-network"><i class="fa fa-check"></i><b>6.12</b> Co-expression network</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html"><i class="fa fa-check"></i><b>7</b> WGCNA RNA-seq Blog</a>
<ul>
<li class="chapter" data-level="7.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#WGCNA"><i class="fa fa-check"></i><b>7.1</b> Introduction to the WGCNA Rpackage</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-rpackage-installation-procedure."><i class="fa fa-check"></i><b>7.1.1</b> WGCNA Rpackage Installation procedure.</a></li>
<li class="chapter" data-level="7.1.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#final-installation-workflow"><i class="fa fa-check"></i><b>7.1.2</b> Final installation workflow:</a></li>
<li class="chapter" data-level="7.1.3" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-tutorial."><i class="fa fa-check"></i><b>7.1.3</b> WGCNA tutorial.</a></li>
<li class="chapter" data-level="7.1.4" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#gene-expression-analysis"><i class="fa fa-check"></i><b>7.1.4</b> 3.4 Gene expression analysis</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#rna-seq-analisys-with-kakus-data-sample."><i class="fa fa-check"></i><b>7.2</b> RNA-seq Analisys with Kaku’s Data sample.</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#data-analysis."><i class="fa fa-check"></i><b>7.2.1</b> Data Analysis.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="youtube-references..html"><a href="youtube-references..html"><i class="fa fa-check"></i><b>8</b> Youtube references.</a>
<ul>
<li class="chapter" data-level="8.1" data-path="youtube-references..html"><a href="youtube-references..html#anaconda-set-up-robust-data-science-environment-with-miniconda-and-conda-forge."><i class="fa fa-check"></i><b>8.1</b> Anaconda set up, Robust Data Science Environment with Miniconda and Conda-Forge.</a></li>
<li class="chapter" data-level="8.2" data-path="youtube-references..html"><a href="youtube-references..html#quality-control-preprocessing-of-raw-reads"><i class="fa fa-check"></i><b>8.2</b> Quality control &amp; preprocessing of raw reads</a></li>
<li class="chapter" data-level="8.3" data-path="youtube-references..html"><a href="youtube-references..html#trimmomatic."><i class="fa fa-check"></i><b>8.3</b> Trimmomatic.</a></li>
<li class="chapter" data-level="8.4" data-path="youtube-references..html"><a href="youtube-references..html#building-genome-index-and-aligning-with-star"><i class="fa fa-check"></i><b>8.4</b> Building Genome Index and Aligning with STAR</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="external-ref.html"><a href="external-ref.html"><i class="fa fa-check"></i><b>9</b> External Ref</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics_RNA-seq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="r-code-modules-to-be-used-for-stand-alone-analysis." class="section level1" number="6">
<h1><span class="header-section-number">Chapter 6</span> R code modules to be used for stand-alone analysis.</h1>
<p>Base on the South Dakota State University analysis protocol<br />
<a href="http://bioinformatics.sdstate.edu/" class="uri">http://bioinformatics.sdstate.edu/</a></p>
<ol style="list-style-type: decimal">
<li><a href="http://bioinformatics.sdstate.edu/idep/">Integrated Differential Expression and Pathway analysis</a><br />
1.<a href="http://bioinformatics.sdstate.edu/go/">ShinyGO: Gene Ontology Enrichment Analysis</a></li>
</ol>
<p>Github: <a href="https://github.com/iDEP-SDSU/idep#readme">iDEP-SDSU/idep</a></p>
<div id="local-installation" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Local installation</h2>
<p>Local installation of this software is possible through steps below. But it is not supported or updated freqently. Local install is for non-profit organizations only. For-profit businesses please contact us to license the database.</p>
<p>To run iDEP on your local machine (Windows, MacOS, Linux):<br />
Requirements:</p>
<p>More than 250GB available storage<br />
More than 4GB memory<br />
Most recent version of R and RStudio installed.</p>
<ol style="list-style-type: decimal">
<li><p>Upgrade to the most recent version of R and Rstudio.<br />
</p></li>
<li><p>Start RStudio and install all the R packages. As we need so many R packages, this may take several hours. You can let it run and get started on steps 3 and 4 below to save time. From RStudio console window:<br />
source <a href="https://raw.githubusercontent.com/iDEP-SDSU/idep/master/classes/librarySetup.R" class="uri">https://raw.githubusercontent.com/iDEP-SDSU/idep/master/classes/librarySetup.R</a></p></li>
<li><p>Download iDEP source code and example data files from GitHub. The best is to click the green “Code” button and select “Download ZIP” on this <a href="https://github.com/iDEP-SDSU/idep">page</a>. Unzip to a folder such as C:/IDEP, so that it contains all the subfolders such as config, classes, shinyapps, and so on.</p></li>
<li><p>Download the most recent database file from <a href="http://18.235.92.206:8080/data104.tar.gz">here.</a> Unzip it to the same folder (C:/IDEP), so that your database can be found at C:/IDEP/data/data104.</p></li>
<li><p>Start Rstudio and load the ui.R and server.R scripts in the folder C:/IDEP/shinyapps/idep94. And then click on Run app. Similarily, the ShinyGO app could be started at the folder, C:/IDEP/shinyapps/go74/.</p></li>
</ol>
</div>
<div id="install-packages" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Install packages</h2>
<p>source <a href="https://raw.githubusercontent.com/iDEP-SDSU/idep/master/classes/librarySetup.R" class="uri">https://raw.githubusercontent.com/iDEP-SDSU/idep/master/classes/librarySetup.R</a></p>
<div class="sourceCode" id="cb182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb182-1"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-1" aria-hidden="true" tabindex="-1"></a><span class="co"># dplyr complains this required libraries: libudunits2-dev, libmariadb-client-lgpl-dev</span></span>
<span id="cb182-2"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-2" aria-hidden="true" tabindex="-1"></a><span class="co">#demo</span></span>
<span id="cb182-3"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-3" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;plotly&quot;, repos=&quot;http://cran.rstudio.com/&quot;, dependencies=TRUE)</span></span>
<span id="cb182-4"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-4" aria-hidden="true" tabindex="-1"></a><span class="co"># sometimes need to remove all installed packages: https://www.r-bloggers.com/how-to-remove-all-user-installed-packages-in-r/ </span></span>
<span id="cb182-5"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-5" aria-hidden="true" tabindex="-1"></a>list.of.packages <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb182-6"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;shiny&quot;</span>, <span class="st">&quot;shinyAce&quot;</span>, <span class="st">&quot;shinyBS&quot;</span>, <span class="st">&quot;plotly&quot;</span>,</span>
<span id="cb182-7"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;RSQLite&quot;</span>, <span class="st">&quot;gplots&quot;</span>, </span>
<span id="cb182-8"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;ggplot2&quot;</span>, <span class="st">&quot;dplyr&quot;</span>, <span class="co">#&quot;tidyverse&quot;,</span></span>
<span id="cb182-9"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;plotly&quot;</span>,</span>
<span id="cb182-10"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-10" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;e1071&quot;</span>, <span class="st">&quot;reshape2&quot;</span>, <span class="st">&quot;DT&quot;</span>,</span>
<span id="cb182-11"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-11" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;data.table&quot;</span>, <span class="st">&quot;Rcpp&quot;</span>,<span class="st">&quot;WGCNA&quot;</span>,<span class="st">&quot;flashClust&quot;</span>,<span class="st">&quot;statmod&quot;</span>,<span class="st">&quot;biclust&quot;</span>,<span class="st">&quot;igraph&quot;</span>,<span class="st">&quot;Rtsne&quot;</span>,</span>
<span id="cb182-12"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-12" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;visNetwork&quot;</span>, <span class="st">&quot;BiocManager&quot;</span>,<span class="st">&quot;feather&quot;</span>,<span class="st">&quot;shinyjs&quot;</span>,<span class="st">&quot;reactable&quot;</span></span>
<span id="cb182-13"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb182-14"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-15"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-15" aria-hidden="true" tabindex="-1"></a>list.of.bio.packages  <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb182-16"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-16" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;getDEE2&quot;</span>, <span class="st">&quot;limma&quot;</span>, <span class="st">&quot;DESeq2&quot;</span>, <span class="st">&quot;edgeR&quot;</span>, <span class="st">&quot;gage&quot;</span>, <span class="st">&quot;fgsea&quot;</span>, <span class="st">&quot;ReactomePA&quot;</span>, <span class="st">&quot;pathview&quot;</span>, <span class="st">&quot;PREDA&quot;</span>,</span>
<span id="cb182-17"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;impute&quot;</span>, <span class="st">&quot;runibic&quot;</span>,<span class="st">&quot;QUBIC&quot;</span>,<span class="st">&quot;rhdf5&quot;</span>, <span class="st">&quot;STRINGdb&quot;</span>,</span>
<span id="cb182-18"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;PREDAsampledata&quot;</span>, <span class="st">&quot;sfsmisc&quot;</span>, <span class="st">&quot;lokern&quot;</span>, <span class="st">&quot;multtest&quot;</span>, <span class="st">&quot;hgu133plus2.db&quot;</span>, </span>
<span id="cb182-19"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-19" aria-hidden="true" tabindex="-1"></a>   <span class="st">&quot;org.Ag.eg.db&quot;</span>,<span class="st">&quot;org.At.tair.db&quot;</span>,<span class="st">&quot;org.Bt.eg.db&quot;</span>,<span class="st">&quot;org.Ce.eg.db&quot;</span>,<span class="st">&quot;org.Cf.eg.db&quot;</span>,</span>
<span id="cb182-20"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-20" aria-hidden="true" tabindex="-1"></a>   <span class="st">&quot;org.Dm.eg.db&quot;</span>,<span class="st">&quot;org.EcK12.eg.db&quot;</span>,<span class="st">&quot;org.EcSakai.eg.db&quot;</span>,<span class="st">&quot;org.Gg.eg.db&quot;</span>,</span>
<span id="cb182-21"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-21" aria-hidden="true" tabindex="-1"></a>   <span class="st">&quot;org.Hs.eg.db&quot;</span>,<span class="st">&quot;org.Mm.eg.db&quot;</span>,<span class="st">&quot;org.Mmu.eg.db&quot;</span>,<span class="st">&quot;org.Pf.plasmo.db&quot;</span>,</span>
<span id="cb182-22"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-22" aria-hidden="true" tabindex="-1"></a>   <span class="st">&quot;org.Pt.eg.db&quot;</span>,<span class="st">&quot;org.Rn.eg.db&quot;</span>,<span class="st">&quot;org.Sc.sgd.db&quot;</span>,<span class="st">&quot;org.Ss.eg.db&quot;</span>,<span class="st">&quot;org.Xl.eg.db&quot;</span></span>
<span id="cb182-23"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-23" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb182-24"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-25"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-25" aria-hidden="true" tabindex="-1"></a> <span class="cf">if</span>(<span class="dv">1</span>) { <span class="co"># remove all old packages, to solve problem caused by Bioconductor upgrade</span></span>
<span id="cb182-26"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># create a list of all installed packages</span></span>
<span id="cb182-27"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-27" aria-hidden="true" tabindex="-1"></a>     ip <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">installed.packages</span>())</span>
<span id="cb182-28"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># head(ip)</span></span>
<span id="cb182-29"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-29" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if you use MRO, make sure that no packages in this library will be removed</span></span>
<span id="cb182-30"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-30" aria-hidden="true" tabindex="-1"></a>     ip <span class="ot">&lt;-</span> <span class="fu">subset</span>(ip, <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;MRO&quot;</span>, ip<span class="sc">$</span>LibPath))</span>
<span id="cb182-31"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-31" aria-hidden="true" tabindex="-1"></a>    <span class="co"># we don&#39;t want to remove base or recommended packages either\</span></span>
<span id="cb182-32"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-32" aria-hidden="true" tabindex="-1"></a>     ip <span class="ot">&lt;-</span> ip[<span class="sc">!</span>(ip[,<span class="st">&quot;Priority&quot;</span>] <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;base&quot;</span>, <span class="st">&quot;recommended&quot;</span>)),]</span>
<span id="cb182-33"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-33" aria-hidden="true" tabindex="-1"></a>    <span class="co"># determine the library where the packages are installed</span></span>
<span id="cb182-34"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-34" aria-hidden="true" tabindex="-1"></a>     path.lib <span class="ot">&lt;-</span> <span class="fu">unique</span>(ip<span class="sc">$</span>LibPath)</span>
<span id="cb182-35"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-35" aria-hidden="true" tabindex="-1"></a>    <span class="co"># create a vector with all the names of the packages you want to remove</span></span>
<span id="cb182-36"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-36" aria-hidden="true" tabindex="-1"></a>     pkgs.to.remove <span class="ot">&lt;-</span> ip[,<span class="dv">1</span>]</span>
<span id="cb182-37"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-37" aria-hidden="true" tabindex="-1"></a>    <span class="co"># head(pkgs.to.remove)</span></span>
<span id="cb182-38"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-38" aria-hidden="true" tabindex="-1"></a>    <span class="co"># remove the packages</span></span>
<span id="cb182-39"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-39" aria-hidden="true" tabindex="-1"></a>     <span class="fu">sapply</span>(pkgs.to.remove, remove.packages, <span class="at">lib =</span> path.lib)</span>
<span id="cb182-40"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-40" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-41"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-42"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-42" aria-hidden="true" tabindex="-1"></a>new.packages <span class="ot">&lt;-</span> list.of.packages[<span class="sc">!</span>(list.of.packages <span class="sc">%in%</span> <span class="fu">installed.packages</span>()[,<span class="st">&quot;Package&quot;</span>])]</span>
<span id="cb182-43"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-43" aria-hidden="true" tabindex="-1"></a>new.bio.packages <span class="ot">&lt;-</span> list.of.bio.packages[<span class="sc">!</span>(list.of.bio.packages <span class="sc">%in%</span> <span class="fu">installed.packages</span>()[,<span class="st">&quot;Package&quot;</span>])]</span>
<span id="cb182-44"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-44" aria-hidden="true" tabindex="-1"></a>notInstalledPackageCount <span class="ot">=</span> <span class="fu">length</span>(new.packages) <span class="sc">+</span> <span class="fu">length</span>(new.bio.packages)</span>
<span id="cb182-45"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-46"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-46" aria-hidden="true" tabindex="-1"></a><span class="co">#Install Require packages</span></span>
<span id="cb182-47"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-47" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span>(notInstalledPackageCount <span class="sc">!=</span> <span class="dv">0</span>){</span>
<span id="cb182-48"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-49"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-49" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(new.packages)) <span class="fu">install.packages</span>(new.packages, <span class="at">repos=</span><span class="st">&quot;http://cran.rstudio.com/&quot;</span>, <span class="at">dependencies=</span><span class="cn">TRUE</span>, <span class="at">quiet=</span><span class="cn">TRUE</span>)</span>
<span id="cb182-50"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-50" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(new.bio.packages)){</span>
<span id="cb182-51"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-51" aria-hidden="true" tabindex="-1"></a>        BiocManager<span class="sc">::</span><span class="fu">install</span>(new.bio.packages, <span class="at">ask =</span> <span class="cn">FALSE</span>, <span class="at">dependencies=</span><span class="cn">TRUE</span>, <span class="at">quiet=</span><span class="cn">TRUE</span>)</span>
<span id="cb182-52"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-52" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-53"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-54"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-54" aria-hidden="true" tabindex="-1"></a>    new.packages <span class="ot">&lt;-</span> list.of.packages[<span class="sc">!</span>(list.of.packages <span class="sc">%in%</span> <span class="fu">installed.packages</span>()[,<span class="st">&quot;Package&quot;</span>])]</span>
<span id="cb182-55"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-55" aria-hidden="true" tabindex="-1"></a>    new.bio.packages <span class="ot">&lt;-</span> list.of.bio.packages[<span class="sc">!</span>(list.of.bio.packages <span class="sc">%in%</span> <span class="fu">installed.packages</span>()[,<span class="st">&quot;Package&quot;</span>])]</span>
<span id="cb182-56"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-56" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( notInstalledPackageCount <span class="sc">==</span> <span class="fu">length</span>(new.packages) <span class="sc">+</span> <span class="fu">length</span>(new.bio.packages) )</span>
<span id="cb182-57"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-57" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb182-58"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-58" aria-hidden="true" tabindex="-1"></a>        <span class="co">#no new package installed.</span></span>
<span id="cb182-59"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-59" aria-hidden="true" tabindex="-1"></a>        <span class="cf">break</span></span>
<span id="cb182-60"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-60" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-61"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-61" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb182-62"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-62" aria-hidden="true" tabindex="-1"></a>       notInstalledPackageCount <span class="ot">=</span> <span class="fu">length</span>(new.packages) <span class="sc">+</span> <span class="fu">length</span>(new.bio.packages)</span>
<span id="cb182-63"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-63" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-64"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-64" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-65"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-66"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-66" aria-hidden="true" tabindex="-1"></a><span class="co">#PGSEA is deprecated since Bioconductor 3.12. So we have to install manually from source.</span></span>
<span id="cb182-67"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-67" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="fu">c</span>(<span class="st">&quot;GO.db&quot;</span>,<span class="st">&quot;KEGG.db&quot;</span>, <span class="st">&quot;annaffy&quot;</span>)) <span class="co"># required y PGSEA</span></span>
<span id="cb182-68"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-68" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;https://bioconductor.org/packages/3.10/bioc/src/contrib/PGSEA_1.60.0.tar.gz&quot;</span>, </span>
<span id="cb182-69"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-69" aria-hidden="true" tabindex="-1"></a>                 <span class="at">repos=</span><span class="cn">NULL</span>, <span class="at">type=</span><span class="st">&quot;source&quot;</span>)</span>
<span id="cb182-70"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-70" aria-hidden="true" tabindex="-1"></a>list.of.bio.packages <span class="ot">=</span> <span class="fu">c</span>(list.of.bio.packages, <span class="st">&quot;PGSEA&quot;</span>) <span class="co"># add package for testing</span></span>
<span id="cb182-71"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-72"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-72" aria-hidden="true" tabindex="-1"></a><span class="co">#Load Packages</span></span>
<span id="cb182-73"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-73" aria-hidden="true" tabindex="-1"></a>suc <span class="ot">=</span> <span class="fu">unlist</span> ( <span class="fu">lapply</span>(list.of.packages, require, <span class="at">character.only =</span> <span class="cn">TRUE</span>) )</span>
<span id="cb182-74"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-74" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">sum</span>(suc) <span class="sc">&lt;</span> <span class="fu">length</span>(list.of.packages) )</span>
<span id="cb182-75"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-75" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span> (<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">Warnning!!!!!! These R packages cannot be loaded:&quot;</span>, list.of.packages[<span class="sc">!</span>suc] )</span>
<span id="cb182-76"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-77"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-77" aria-hidden="true" tabindex="-1"></a>suc <span class="ot">=</span> <span class="fu">unlist</span> ( <span class="fu">lapply</span>(list.of.bio.packages, require, <span class="at">character.only =</span> <span class="cn">TRUE</span>) )</span>
<span id="cb182-78"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-78" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">sum</span>(suc) <span class="sc">&lt;</span> <span class="fu">length</span>(list.of.bio.packages) )</span>
<span id="cb182-79"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb182-79" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span> (<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">Warnning!!!!!! These Bioconductor packages cannot be loaded:&quot;</span>, list.of.bio.packages[<span class="sc">!</span>suc] )</span></code></pre></div>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-1" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb183-2"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Install packages</span></span>
<span id="cb183-3"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-3" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb183-4"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-5"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-5" aria-hidden="true" tabindex="-1"></a><span class="co"># dplyr complains this required libraries: libudunits2-dev, libmariadb-client-lgpl-dev</span></span>
<span id="cb183-6"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-6" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;plotly&quot;, repos=&quot;http://cran.rstudio.com/&quot;, dependencies=TRUE)</span></span>
<span id="cb183-7"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-7" aria-hidden="true" tabindex="-1"></a><span class="co"># sometimes need to remove all installed packages: https://www.r-bloggers.com/how-to-remove-all-user-installed-packages-in-r/ </span></span>
<span id="cb183-8"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-8" aria-hidden="true" tabindex="-1"></a><span class="co">#cat(&quot;\n Installing lots of R packages, this may take several hours ... \n</span></span>
<span id="cb183-9"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   Each of these packages took years to develop.\n So be a patient thief.\n</span></span>
<span id="cb183-10"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-10" aria-hidden="true" tabindex="-1"></a><span class="co">#   \n Note 1: Sometimes dependencies needs to be installed manually. </span></span>
<span id="cb183-11"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-11" aria-hidden="true" tabindex="-1"></a><span class="co">#   \n Note 2: If you are using an older version of R, and having trouble with package installation,</span></span>
<span id="cb183-12"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-12" aria-hidden="true" tabindex="-1"></a><span class="co">#   \n         sometimes, it is easier to install the new version of R and delete all old packages, and start fresh. </span></span>
<span id="cb183-13"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-13" aria-hidden="true" tabindex="-1"></a><span class="co">#  &quot;)</span></span>
<span id="cb183-14"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-15"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-15" aria-hidden="true" tabindex="-1"></a>list.of.packages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;dendextend&quot;</span>, <span class="st">&quot;htmlwidgets&quot;</span>,<span class="st">&quot;RSQLite&quot;</span>, </span>
<span id="cb183-16"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;gplots&quot;</span>, <span class="st">&quot;ggplot2&quot;</span>, <span class="st">&quot;dplyr&quot;</span>, <span class="st">&quot;tidyverse&quot;</span>,<span class="st">&quot;plotly&quot;</span>,<span class="st">&quot;e1071&quot;</span>, </span>
<span id="cb183-17"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;reshape2&quot;</span>,  <span class="st">&quot;Rcpp&quot;</span>,<span class="st">&quot;flashClust&quot;</span>,<span class="st">&quot;statmod&quot;</span>,</span>
<span id="cb183-18"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;biclust&quot;</span>,<span class="st">&quot;igraph&quot;</span>,<span class="st">&quot;Rtsne&quot;</span>)</span>
<span id="cb183-19"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-20"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-20" aria-hidden="true" tabindex="-1"></a>list.of.bio.packages  <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;STRINGdb&quot;</span>, <span class="st">&quot;limma&quot;</span>, <span class="st">&quot;DESeq2&quot;</span>, <span class="st">&quot;edgeR&quot;</span>, <span class="st">&quot;gage&quot;</span>, </span>
<span id="cb183-21"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;PGSEA&quot;</span>, <span class="st">&quot;fgsea&quot;</span>, <span class="st">&quot;ReactomePA&quot;</span>, <span class="st">&quot;pathview&quot;</span>, <span class="st">&quot;PREDA&quot;</span>, <span class="st">&quot;impute&quot;</span>, <span class="st">&quot;runibic&quot;</span>,</span>
<span id="cb183-22"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-22" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;QUBIC&quot;</span>,<span class="st">&quot;rhdf5&quot;</span>, <span class="st">&quot;PREDAsampledata&quot;</span>, <span class="st">&quot;sfsmisc&quot;</span>, <span class="st">&quot;lokern&quot;</span>, <span class="st">&quot;multtest&quot;</span>, </span>
<span id="cb183-23"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-23" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;GSEABase&quot;</span>,<span class="st">&quot;hgu133plus2.db&quot;</span>)</span>
<span id="cb183-24"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-25"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-25" aria-hidden="true" tabindex="-1"></a><span class="co">#Install Require packages</span></span>
<span id="cb183-26"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-26" aria-hidden="true" tabindex="-1"></a>new.packages <span class="ot">&lt;-</span> list.of.packages[<span class="sc">!</span>(list.of.packages <span class="sc">%in%</span> <span class="fu">installed.packages</span>()[,<span class="st">&quot;Package&quot;</span>])]</span>
<span id="cb183-27"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-27" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">length</span>(new.packages)) <span class="fu">install.packages</span>(new.packages, <span class="at">repos=</span><span class="st">&quot;http://cran.rstudio.com/&quot;</span>, <span class="at">dependencies=</span><span class="cn">TRUE</span>)</span>
<span id="cb183-28"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-29"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-29" aria-hidden="true" tabindex="-1"></a>new.bio.packages <span class="ot">&lt;-</span> list.of.bio.packages[<span class="sc">!</span>(list.of.bio.packages <span class="sc">%in%</span> <span class="fu">installed.packages</span>()[,<span class="st">&quot;Package&quot;</span>])]</span>
<span id="cb183-30"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-30" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">length</span>(new.bio.packages)){</span>
<span id="cb183-31"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-31" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb183-32"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb183-33"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-33" aria-hidden="true" tabindex="-1"></a>    BiocManager<span class="sc">::</span><span class="fu">install</span>(new.bio.packages, <span class="at">suppressUpdates =</span> T)</span>
<span id="cb183-34"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-34" aria-hidden="true" tabindex="-1"></a>}   </span>
<span id="cb183-35"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-36"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-36" aria-hidden="true" tabindex="-1"></a><span class="co"># WGCNA must be installed after AnnotationDbi, impute, GO.db, preprocessCore</span></span>
<span id="cb183-37"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-37" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(c(&quot;WGCNA&quot;))</span></span>
<span id="cb183-38"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-39"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-39" aria-hidden="true" tabindex="-1"></a><span class="co">#Test Load Packages    This cause an error when too many packages are loaded.</span></span>
<span id="cb183-40"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-40" aria-hidden="true" tabindex="-1"></a><span class="co"># suc = unlist ( lapply(list.of.packages, require, character.only = TRUE) )</span></span>
<span id="cb183-41"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-41" aria-hidden="true" tabindex="-1"></a><span class="co"># if(sum(suc) &lt; length(list.of.packages) )</span></span>
<span id="cb183-42"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-42" aria-hidden="true" tabindex="-1"></a><span class="co">#   cat (&quot;\n\nWarnning!!!!!! These R packages cannot be loaded:&quot;, list.of.packages[!suc] )</span></span>
<span id="cb183-43"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-44"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-44" aria-hidden="true" tabindex="-1"></a><span class="co"># suc = unlist ( lapply(list.of.bio.packages, require, character.only = TRUE) )</span></span>
<span id="cb183-45"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-45" aria-hidden="true" tabindex="-1"></a><span class="co"># if(sum(suc) &lt; length(list.of.bio.packages) )</span></span>
<span id="cb183-46"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb183-46" aria-hidden="true" tabindex="-1"></a><span class="co">#   cat (&quot;\n\nWarnning!!!!!! These Bioconductor packages cannot be loaded:&quot;, list.of.bio.packages[!suc] )</span></span></code></pre></div>
</div>
<div id="global-variables" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Global variables</h2>
<div class="sourceCode" id="cb184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb184-1"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-1" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb184-2"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Global variables</span></span>
<span id="cb184-3"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-3" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb184-4"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-5"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RSQLite,<span class="at">verbose=</span><span class="cn">FALSE</span>)  <span class="co"># for database connection</span></span>
<span id="cb184-6"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gplots,<span class="at">verbose=</span><span class="cn">FALSE</span>)       <span class="co"># for hierarchical clustering</span></span>
<span id="cb184-7"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2,<span class="at">verbose=</span><span class="cn">FALSE</span>)  <span class="co"># graphics</span></span>
<span id="cb184-8"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(e1071,<span class="at">verbose=</span><span class="cn">FALSE</span>)        <span class="co"># computing kurtosis</span></span>
<span id="cb184-9"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DT,<span class="at">verbose=</span><span class="cn">FALSE</span>)       <span class="co"># for renderDataTable</span></span>
<span id="cb184-10"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly,<span class="at">verbose=</span><span class="cn">FALSE</span>)   <span class="co"># for interactive heatmap</span></span>
<span id="cb184-11"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2,<span class="at">verbose=</span><span class="cn">FALSE</span>)     <span class="co"># for melt correlation matrix in heatmap</span></span>
<span id="cb184-12"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-12" aria-hidden="true" tabindex="-1"></a>input_goButton <span class="ot">=</span> <span class="dv">0</span>    <span class="co"># if 1, use demo data file</span></span>
<span id="cb184-13"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-13" aria-hidden="true" tabindex="-1"></a>Min_overlap <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb184-14"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-14" aria-hidden="true" tabindex="-1"></a>minSetSize <span class="ot">=</span> <span class="dv">3</span>; </span>
<span id="cb184-15"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-15" aria-hidden="true" tabindex="-1"></a>mappingCoverage <span class="ot">=</span> <span class="fl">0.60</span> <span class="co"># 60% percent genes has to be mapped for confident mapping</span></span>
<span id="cb184-16"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-16" aria-hidden="true" tabindex="-1"></a>mappingEdge <span class="ot">=</span> <span class="fl">0.5</span>  <span class="co"># Top species has 50% more genes mapped</span></span>
<span id="cb184-17"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-17" aria-hidden="true" tabindex="-1"></a>PvalGeneInfo <span class="ot">=</span> <span class="fl">0.05</span>; minGenes <span class="ot">=</span> <span class="dv">10</span> <span class="co"># min number of genes for ploting</span></span>
<span id="cb184-18"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-18" aria-hidden="true" tabindex="-1"></a>kurtosis.log <span class="ot">=</span> <span class="dv">50</span>  <span class="co"># log transform is enforced when kurtosis is big</span></span>
<span id="cb184-19"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-19" aria-hidden="true" tabindex="-1"></a>kurtosis.warning <span class="ot">=</span> <span class="dv">10</span> <span class="co"># log transformation recommnded </span></span>
<span id="cb184-20"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-20" aria-hidden="true" tabindex="-1"></a>minGenesEnrichment <span class="ot">=</span> <span class="dv">2</span> <span class="co"># perform GO or promoter analysis only if more than this many genes</span></span>
<span id="cb184-21"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-21" aria-hidden="true" tabindex="-1"></a>PREDA_Permutations <span class="ot">=</span><span class="dv">1000</span></span>
<span id="cb184-22"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-22" aria-hidden="true" tabindex="-1"></a>maxGeneClustering <span class="ot">=</span> <span class="dv">12000</span>  <span class="co"># max genes for hierarchical clustering and k-Means clustering. Slow if larger</span></span>
<span id="cb184-23"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-23" aria-hidden="true" tabindex="-1"></a>maxGeneWGCNA <span class="ot">=</span> <span class="dv">2000</span> <span class="co"># max genes for co-expression network</span></span>
<span id="cb184-24"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-24" aria-hidden="true" tabindex="-1"></a>maxFactors <span class="ot">=</span><span class="dv">6</span>  <span class="co"># max number of factors in DESeq2 models</span></span>
<span id="cb184-25"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-25" aria-hidden="true" tabindex="-1"></a>redudantGeneSetsRatio <span class="ot">=</span> <span class="fl">0.9</span>  <span class="co"># remove redundant genesets in enrichment analysis</span></span>
<span id="cb184-26"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-26" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">2</span>) <span class="co"># seed for random number generator</span></span>
<span id="cb184-27"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-27" aria-hidden="true" tabindex="-1"></a>mycolors <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">rainbow</span>(<span class="dv">20</span>))[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">20</span>,<span class="dv">10</span>,<span class="dv">11</span>,<span class="dv">2</span>,<span class="dv">19</span>,<span class="dv">3</span>,<span class="dv">12</span>,<span class="dv">4</span>,<span class="dv">13</span>,<span class="dv">5</span>,<span class="dv">14</span>,<span class="dv">6</span>,<span class="dv">15</span>,<span class="dv">7</span>,<span class="dv">16</span>,<span class="dv">8</span>,<span class="dv">17</span>,<span class="dv">9</span>,<span class="dv">18</span>)] <span class="co"># 20 colors for kNN clusters</span></span>
<span id="cb184-28"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-28" aria-hidden="true" tabindex="-1"></a><span class="co">#Each row of this matrix represents a color scheme;</span></span>
<span id="cb184-29"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-30"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-30" aria-hidden="true" tabindex="-1"></a>hmcols <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&quot;#D73027&quot;</span>, <span class="st">&quot;#FC8D59&quot;</span>, <span class="st">&quot;#FEE090&quot;</span>, <span class="st">&quot;#FFFFBF&quot;</span>,</span>
<span id="cb184-31"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-31" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;#E0F3F8&quot;</span>, <span class="st">&quot;#91BFDB&quot;</span>, <span class="st">&quot;#4575B4&quot;</span>)))(<span class="dv">75</span>)</span>
<span id="cb184-32"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-32" aria-hidden="true" tabindex="-1"></a>heatColors <span class="ot">=</span> <span class="fu">rbind</span>(      <span class="fu">greenred</span>(<span class="dv">75</span>),     <span class="fu">bluered</span>(<span class="dv">75</span>),     <span class="fu">colorpanel</span>(<span class="dv">75</span>,<span class="st">&quot;green&quot;</span>,<span class="st">&quot;black&quot;</span>,<span class="st">&quot;magenta&quot;</span>),<span class="fu">colorpanel</span>(<span class="dv">75</span>,<span class="st">&quot;blue&quot;</span>,<span class="st">&quot;yellow&quot;</span>,<span class="st">&quot;red&quot;</span>),hmcols )</span>
<span id="cb184-33"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-33" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heatColors) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Green-Black-Red&quot;</span>,<span class="st">&quot;Blue-White-Red&quot;</span>,<span class="st">&quot;Green-Black-Magenta&quot;</span>,<span class="st">&quot;Blue-Yellow-Red&quot;</span>,<span class="st">&quot;Blue-white-brown&quot;</span>)</span>
<span id="cb184-34"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-34" aria-hidden="true" tabindex="-1"></a>colorChoices <span class="ot">=</span> <span class="fu">setNames</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(heatColors)[<span class="dv">1</span>],<span class="fu">rownames</span>(heatColors)) <span class="co"># for pull down menu</span></span>
<span id="cb184-35"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-36"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-36" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a list for Select Input options</span></span>
<span id="cb184-37"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-37" aria-hidden="true" tabindex="-1"></a>speciesChoice <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb184-38"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-38" aria-hidden="true" tabindex="-1"></a>speciesChoice <span class="ot">&lt;-</span> <span class="fu">append</span>( <span class="fu">setNames</span>( <span class="st">&quot;NEW&quot;</span>,<span class="st">&quot;**NEW SPECIES**&quot;</span>), speciesChoice  )</span>
<span id="cb184-39"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb184-39" aria-hidden="true" tabindex="-1"></a>speciesChoice <span class="ot">&lt;-</span> <span class="fu">append</span>( <span class="fu">setNames</span>( <span class="st">&quot;BestMatch&quot;</span>,<span class="st">&quot;Best matching species&quot;</span>), speciesChoice  )</span></code></pre></div>
</div>
<div id="read-data-and-pre-process" class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> Read data and pre-process</h2>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-1" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb185-2"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Read data and pre-process</span></span>
<span id="cb185-3"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-3" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb185-4"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-5"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-5" aria-hidden="true" tabindex="-1"></a>readData <span class="ot">&lt;-</span> <span class="cf">function</span>(inFile ) {</span>
<span id="cb185-6"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-7"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-7" aria-hidden="true" tabindex="-1"></a>                <span class="co"># these packages moved here to reduce loading time</span></span>
<span id="cb185-8"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-8" aria-hidden="true" tabindex="-1"></a>                <span class="fu">library</span>(edgeR,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># count data D.E.</span></span>
<span id="cb185-9"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-9" aria-hidden="true" tabindex="-1"></a>                <span class="fu">library</span>(DESeq2,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># count data analysis</span></span>
<span id="cb185-10"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-11"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-11" aria-hidden="true" tabindex="-1"></a>                dataTypeWarning <span class="ot">=</span><span class="dv">0</span></span>
<span id="cb185-12"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-12" aria-hidden="true" tabindex="-1"></a>                dataType <span class="ot">=</span><span class="fu">c</span>(<span class="cn">TRUE</span>)</span>
<span id="cb185-13"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-14"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-14" aria-hidden="true" tabindex="-1"></a>                <span class="co">#---------------Read file</span></span>
<span id="cb185-15"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-15" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(inFile)   <span class="co"># try CSV</span></span>
<span id="cb185-16"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-16" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="fu">dim</span>(x)[<span class="dv">2</span>] <span class="sc">&lt;=</span> <span class="dv">2</span> )   <span class="co"># if less than 3 columns, try tab-deliminated</span></span>
<span id="cb185-17"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-17" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">&lt;-</span> <span class="fu">read.table</span>(inFile, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">header=</span><span class="cn">TRUE</span>)   </span>
<span id="cb185-18"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-18" aria-hidden="true" tabindex="-1"></a>                <span class="co">#-------Remove non-numeric columns, except the first column</span></span>
<span id="cb185-19"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-19" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb185-20"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-20" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">dim</span>(x)[<span class="dv">2</span>])</span>
<span id="cb185-21"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-21" aria-hidden="true" tabindex="-1"></a>                    dataType <span class="ot">=</span> <span class="fu">c</span>( dataType, <span class="fu">is.numeric</span>(x[,i]) )</span>
<span id="cb185-22"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-22" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="fu">sum</span>(dataType) <span class="sc">&lt;=</span><span class="dv">2</span>) <span class="fu">return</span> (<span class="cn">NULL</span>)  <span class="co"># only less than 2 columns are numbers</span></span>
<span id="cb185-23"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-23" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">&lt;-</span> x[,dataType]  <span class="co"># only keep numeric columns</span></span>
<span id="cb185-24"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-24" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb185-25"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-25" aria-hidden="true" tabindex="-1"></a>                <span class="co"># rows with all missing values</span></span>
<span id="cb185-26"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-26" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">=</span> <span class="fu">which</span>( <span class="fu">apply</span>(x[,<span class="sc">-</span><span class="dv">1</span>],<span class="dv">1</span>, <span class="cf">function</span>(y) <span class="fu">sum</span>( <span class="fu">is.na</span>(y) ) ) <span class="sc">!=</span> <span class="fu">dim</span>(x)[<span class="dv">2</span>]<span class="sc">-</span><span class="dv">1</span> )</span>
<span id="cb185-27"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-27" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">&lt;-</span> x[ix,]</span>
<span id="cb185-28"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-28" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb185-29"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-29" aria-hidden="true" tabindex="-1"></a>                dataSizeOriginal <span class="ot">=</span> <span class="fu">dim</span>(x); dataSizeOriginal[<span class="dv">2</span>] <span class="ot">=</span> dataSizeOriginal[<span class="dv">2</span>] <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb185-30"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-30" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb185-31"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-31" aria-hidden="true" tabindex="-1"></a>                x[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">toupper</span>(x[,<span class="dv">1</span>])</span>
<span id="cb185-32"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-32" aria-hidden="true" tabindex="-1"></a>                x[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>,<span class="st">&quot;&quot;</span>,x[,<span class="dv">1</span>]) <span class="co"># remove spaces in gene ids</span></span>
<span id="cb185-33"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-33" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">=</span> x[<span class="fu">order</span>(<span class="sc">-</span> <span class="fu">apply</span>(x[,<span class="dv">2</span><span class="sc">:</span><span class="fu">dim</span>(x)[<span class="dv">2</span>]],<span class="dv">1</span>,sd) ),]  <span class="co"># sort by SD</span></span>
<span id="cb185-34"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-34" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">&lt;-</span> x[<span class="sc">!</span><span class="fu">duplicated</span>(x[,<span class="dv">1</span>]) ,]  <span class="co"># remove duplicated genes</span></span>
<span id="cb185-35"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-35" aria-hidden="true" tabindex="-1"></a>                <span class="fu">rownames</span>(x) <span class="ot">&lt;-</span> x[,<span class="dv">1</span>]</span>
<span id="cb185-36"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-36" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(x[,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)])</span>
<span id="cb185-37"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-37" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb185-38"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-38" aria-hidden="true" tabindex="-1"></a>                <span class="co"># remove &quot;-&quot; or &quot;.&quot; from sample names</span></span>
<span id="cb185-39"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-39" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(x) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(x))</span>
<span id="cb185-40"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-40" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(x) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(x))</span>
<span id="cb185-41"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-42"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-42" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb185-43"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-43" aria-hidden="true" tabindex="-1"></a>                <span class="co">#cat(&quot;\nhere&quot;,dim(x))</span></span>
<span id="cb185-44"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-44" aria-hidden="true" tabindex="-1"></a>                <span class="co"># missng value for median value</span></span>
<span id="cb185-45"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-45" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="fu">sum</span>(<span class="fu">is.na</span>(x))<span class="sc">&gt;</span><span class="dv">0</span>) {<span class="co"># if there is missing values</span></span>
<span id="cb185-46"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-46" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>(input_missingValue <span class="sc">==</span><span class="st">&quot;geneMedian&quot;</span>) { </span>
<span id="cb185-47"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-47" aria-hidden="true" tabindex="-1"></a>                        rowMedians <span class="ot">&lt;-</span> <span class="fu">apply</span>(x,<span class="dv">1</span>, <span class="cf">function</span> (y)  <span class="fu">median</span>(y,<span class="at">na.rm=</span>T))</span>
<span id="cb185-48"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-48" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(x)[<span class="dv">2</span>] ) {</span>
<span id="cb185-49"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-49" aria-hidden="true" tabindex="-1"></a>                            ix <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">is.na</span>(x[,i]) )</span>
<span id="cb185-50"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-50" aria-hidden="true" tabindex="-1"></a>                            x[ix,i] <span class="ot">&lt;-</span> rowMedians[ix]                       </span>
<span id="cb185-51"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-51" aria-hidden="true" tabindex="-1"></a>                        }</span>
<span id="cb185-52"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-52" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb185-53"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-53" aria-hidden="true" tabindex="-1"></a>                    } <span class="cf">else</span> <span class="cf">if</span>(input_missingValue <span class="sc">==</span><span class="st">&quot;treatAsZero&quot;</span>) {</span>
<span id="cb185-54"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-54" aria-hidden="true" tabindex="-1"></a>                        x[<span class="fu">is.na</span>(x) ] <span class="ot">&lt;-</span> <span class="dv">0</span>                   </span>
<span id="cb185-55"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-55" aria-hidden="true" tabindex="-1"></a>                    } <span class="cf">else</span> <span class="cf">if</span> (input_missingValue <span class="sc">==</span><span class="st">&quot;geneMedianInGroup&quot;</span>) {</span>
<span id="cb185-56"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-56" aria-hidden="true" tabindex="-1"></a>                        sampleGroups <span class="ot">=</span> <span class="fu">detectGroups</span>( <span class="fu">colnames</span>(x))</span>
<span id="cb185-57"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-57" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">for</span> (group <span class="cf">in</span> <span class="fu">unique</span>( sampleGroups) ){      </span>
<span id="cb185-58"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-58" aria-hidden="true" tabindex="-1"></a>                            samples <span class="ot">=</span> <span class="fu">which</span>( sampleGroups <span class="sc">==</span> group )</span>
<span id="cb185-59"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-59" aria-hidden="true" tabindex="-1"></a>                            rowMedians <span class="ot">&lt;-</span> <span class="fu">apply</span>(x[,samples, <span class="at">drop=</span>F],<span class="dv">1</span>, <span class="cf">function</span> (y)  <span class="fu">median</span>(y,<span class="at">na.rm=</span>T))</span>
<span id="cb185-60"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-60" aria-hidden="true" tabindex="-1"></a>                            <span class="cf">for</span>( i <span class="cf">in</span>  samples ) { </span>
<span id="cb185-61"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-61" aria-hidden="true" tabindex="-1"></a>                                ix <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">is.na</span>(x[ ,i] ) ) </span>
<span id="cb185-62"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-62" aria-hidden="true" tabindex="-1"></a>                                <span class="cf">if</span>(<span class="fu">length</span>(ix) <span class="sc">&gt;</span><span class="dv">0</span> )</span>
<span id="cb185-63"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-63" aria-hidden="true" tabindex="-1"></a>                                    x[ix, i  ]  <span class="ot">&lt;-</span> rowMedians[ix]</span>
<span id="cb185-64"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-64" aria-hidden="true" tabindex="-1"></a>                            }                                       </span>
<span id="cb185-65"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-65" aria-hidden="true" tabindex="-1"></a>                        }</span>
<span id="cb185-66"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-66" aria-hidden="true" tabindex="-1"></a>                        </span>
<span id="cb185-67"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-67" aria-hidden="true" tabindex="-1"></a>                        <span class="co"># missing for entire sample group, use median for all samples</span></span>
<span id="cb185-68"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-68" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>(<span class="fu">sum</span>(<span class="fu">is.na</span>(x) )<span class="sc">&gt;</span><span class="dv">0</span> ) { </span>
<span id="cb185-69"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-69" aria-hidden="true" tabindex="-1"></a>                            rowMedians <span class="ot">&lt;-</span> <span class="fu">apply</span>(x,<span class="dv">1</span>, <span class="cf">function</span> (y)  <span class="fu">median</span>(y,<span class="at">na.rm=</span>T))</span>
<span id="cb185-70"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-70" aria-hidden="true" tabindex="-1"></a>                            <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(x)[<span class="dv">2</span>] ) {</span>
<span id="cb185-71"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-71" aria-hidden="true" tabindex="-1"></a>                                ix <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">is.na</span>(x[,i]) )</span>
<span id="cb185-72"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-72" aria-hidden="true" tabindex="-1"></a>                                x[ix,i] <span class="ot">&lt;-</span> rowMedians[ix]                       </span>
<span id="cb185-73"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-73" aria-hidden="true" tabindex="-1"></a>                            }                       </span>
<span id="cb185-74"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-74" aria-hidden="true" tabindex="-1"></a>                        }</span>
<span id="cb185-75"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-75" aria-hidden="true" tabindex="-1"></a>                    }</span>
<span id="cb185-76"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-76" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb185-77"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-78"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-78" aria-hidden="true" tabindex="-1"></a>                <span class="co"># Compute kurtosis</span></span>
<span id="cb185-79"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-79" aria-hidden="true" tabindex="-1"></a>                mean.kurtosis <span class="ot">=</span> <span class="fu">mean</span>(<span class="fu">apply</span>(x,<span class="dv">2</span>, kurtosis),<span class="at">na.rm=</span>T)</span>
<span id="cb185-80"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-80" aria-hidden="true" tabindex="-1"></a>                rawCounts <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb185-81"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-81" aria-hidden="true" tabindex="-1"></a>                pvals<span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb185-82"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-82" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> (input_dataFileFormat <span class="sc">==</span> <span class="dv">2</span> ) {  <span class="co"># if FPKM, microarray</span></span>
<span id="cb185-83"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-84"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-84" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span> ( <span class="fu">is.integer</span>(x) ) dataTypeWarning <span class="ot">=</span> <span class="dv">1</span>;  <span class="co"># Data appears to be read counts</span></span>
<span id="cb185-85"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-86"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-86" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#-------------filtering</span></span>
<span id="cb185-87"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-87" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#tem &lt;- apply(x,1,max)</span></span>
<span id="cb185-88"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-88" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#x &lt;- x[which(tem &gt; input_lowFilter),]  # max by row is at least </span></span>
<span id="cb185-89"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-89" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">&lt;-</span> x[ <span class="fu">which</span>( <span class="fu">apply</span>( x, <span class="dv">1</span>,  <span class="cf">function</span>(y) <span class="fu">sum</span>(y <span class="sc">&gt;=</span> input_lowFilter)) <span class="sc">&gt;=</span> input_NminSamples2 ) , ] </span>
<span id="cb185-90"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-91"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-91" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb185-92"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-92" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">&lt;-</span> x[<span class="fu">which</span>(<span class="fu">apply</span>(x,<span class="dv">1</span>, <span class="cf">function</span>(y) <span class="fu">max</span>(y)<span class="sc">-</span> <span class="fu">min</span>(y) ) <span class="sc">&gt;</span> <span class="dv">0</span>  ),]  <span class="co"># remove rows with all the same levels</span></span>
<span id="cb185-93"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-94"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-94" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#--------------Log transform</span></span>
<span id="cb185-95"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-95" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># Takes log if log is selected OR kurtosis is big than 100</span></span>
<span id="cb185-96"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-96" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span> ( (input_transform <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">|</span> (mean.kurtosis <span class="sc">&gt;</span> kurtosis.log ) ) </span>
<span id="cb185-97"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-97" aria-hidden="true" tabindex="-1"></a>                        x <span class="ot">=</span> <span class="fu">log</span>(x<span class="sc">+</span><span class="fu">abs</span>( input_logStart),<span class="dv">2</span>)</span>
<span id="cb185-98"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-99"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-99" aria-hidden="true" tabindex="-1"></a>                    tem <span class="ot">&lt;-</span> <span class="fu">apply</span>(x,<span class="dv">1</span>,sd) </span>
<span id="cb185-100"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-100" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">&lt;-</span> x[<span class="fu">order</span>(<span class="sc">-</span>tem),]  <span class="co"># sort by SD</span></span>
<span id="cb185-101"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-102"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-102" aria-hidden="true" tabindex="-1"></a>                } <span class="cf">else</span> </span>
<span id="cb185-103"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-103" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>( input_dataFileFormat <span class="sc">==</span> <span class="dv">1</span>) {  <span class="co"># counts data</span></span>
<span id="cb185-104"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-105"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-106"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-106" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># data not seems to be read counts</span></span>
<span id="cb185-107"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-107" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.integer</span>(x) <span class="sc">&amp;</span> mean.kurtosis <span class="sc">&lt;</span> kurtosis.log ) {</span>
<span id="cb185-108"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-108" aria-hidden="true" tabindex="-1"></a>                        dataTypeWarning <span class="ot">=</span> <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb185-109"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-109" aria-hidden="true" tabindex="-1"></a>                    }</span>
<span id="cb185-110"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-110" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># not used as some counts data like those from CRISPR screen</span></span>
<span id="cb185-111"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-111" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#validate(   # if Kurtosis is less than a threshold, it is not read-count</span></span>
<span id="cb185-112"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-112" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#   need(mean.kurtosis &gt; kurtosis.log, &quot;Data does not seem to be read count based on distribution. Please double check.&quot;)</span></span>
<span id="cb185-113"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-113" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># )</span></span>
<span id="cb185-114"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-114" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">&lt;-</span> <span class="fu">round</span>(x,<span class="dv">0</span>) <span class="co"># enforce the read counts to be integers. Sailfish outputs has decimal points.</span></span>
<span id="cb185-115"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-115" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#x &lt;- x[ which( apply(x,1,max) &gt;= input_minCounts ) , ] # remove all zero counts</span></span>
<span id="cb185-116"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-116" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb185-117"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-118"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-118" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># remove genes if it does not at least have minCounts in at least NminSamples</span></span>
<span id="cb185-119"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-119" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#x &lt;- x[ which( apply(x,1,function(y) sum(y&gt;=input_minCounts)) &gt;= input_NminSamples ) , ]  # filtering on raw counts</span></span>
<span id="cb185-120"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-120" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># using counts per million (CPM) for filtering out genes.</span></span>
<span id="cb185-121"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-121" aria-hidden="true" tabindex="-1"></a>                                             <span class="co"># CPM matrix                  #N samples &gt; minCounts</span></span>
<span id="cb185-122"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-122" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">&lt;-</span> x[ <span class="fu">which</span>( <span class="fu">apply</span>( <span class="fu">cpm</span>(<span class="fu">DGEList</span>(<span class="at">counts =</span> x)), <span class="dv">1</span>,  </span>
<span id="cb185-123"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-123" aria-hidden="true" tabindex="-1"></a>                            <span class="cf">function</span>(y) <span class="fu">sum</span>(y<span class="sc">&gt;=</span>input_minCounts)) <span class="sc">&gt;=</span> input_NminSamples ) , ] </span>
<span id="cb185-124"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-124" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb185-125"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-126"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-127"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-127" aria-hidden="true" tabindex="-1"></a>                    rawCounts <span class="ot">=</span> x; <span class="co"># ??? </span></span>
<span id="cb185-128"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-128" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># construct DESeqExpression Object</span></span>
<span id="cb185-129"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-129" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># colData = cbind(colnames(x), as.character(detectGroups( colnames(x) )) )</span></span>
<span id="cb185-130"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-130" aria-hidden="true" tabindex="-1"></a>                    tem <span class="ot">=</span> <span class="fu">rep</span>(<span class="st">&quot;A&quot;</span>,<span class="fu">dim</span>(x)[<span class="dv">2</span>]); tem[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;B&quot;</span>   <span class="co"># making a fake design matrix to allow process, even when there is no replicates</span></span>
<span id="cb185-131"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-131" aria-hidden="true" tabindex="-1"></a>                    colData <span class="ot">=</span> <span class="fu">cbind</span>(<span class="fu">colnames</span>(x), tem )</span>
<span id="cb185-132"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-132" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">colnames</span>(colData)  <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;groups&quot;</span>)</span>
<span id="cb185-133"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-133" aria-hidden="true" tabindex="-1"></a>                    dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> x, <span class="at">colData =</span> colData, <span class="at">design =</span> <span class="sc">~</span> groups)</span>
<span id="cb185-134"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-134" aria-hidden="true" tabindex="-1"></a>                    dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds) <span class="co"># estimate size factor for use in normalization later for started log method</span></span>
<span id="cb185-135"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-136"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-137"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-137" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># regularized log  or VST transformation</span></span>
<span id="cb185-138"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-138" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>( input_CountsTransform <span class="sc">==</span> <span class="dv">3</span> ) { <span class="co"># rlog is slow, only do it with 10 samples</span></span>
<span id="cb185-139"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-139" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>(<span class="fu">dim</span>(x)[<span class="dv">2</span>]<span class="sc">&lt;=</span><span class="dv">20</span> ) { </span>
<span id="cb185-140"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-140" aria-hidden="true" tabindex="-1"></a>                         x <span class="ot">&lt;-</span> <span class="fu">rlog</span>(dds, <span class="at">blind=</span><span class="cn">TRUE</span>); x <span class="ot">&lt;-</span> <span class="fu">assay</span>(x) } <span class="cf">else</span> </span>
<span id="cb185-141"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-141" aria-hidden="true" tabindex="-1"></a>                         x <span class="ot">&lt;-</span> <span class="fu">log2</span>( <span class="fu">counts</span>(dds, <span class="at">normalized=</span><span class="cn">TRUE</span>) <span class="sc">+</span> input_countsLogStart ) </span>
<span id="cb185-142"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-142" aria-hidden="true" tabindex="-1"></a>                         }  </span>
<span id="cb185-143"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-144"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-144" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">else</span> {</span>
<span id="cb185-145"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-145" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span> ( input_CountsTransform <span class="sc">==</span> <span class="dv">2</span> ) {    <span class="co"># vst is fast but aggressive</span></span>
<span id="cb185-146"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-146" aria-hidden="true" tabindex="-1"></a>                            x <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind=</span><span class="cn">TRUE</span>)</span>
<span id="cb185-147"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-147" aria-hidden="true" tabindex="-1"></a>                            x <span class="ot">&lt;-</span> <span class="fu">assay</span>(x)  </span>
<span id="cb185-148"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-148" aria-hidden="true" tabindex="-1"></a>                        } <span class="cf">else</span>{  <span class="co"># normalized by library sizes and add a constant.</span></span>
<span id="cb185-149"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-149" aria-hidden="true" tabindex="-1"></a>                            x <span class="ot">&lt;-</span> <span class="fu">log2</span>( <span class="fu">counts</span>(dds, <span class="at">normalized=</span><span class="cn">TRUE</span>) <span class="sc">+</span> input_countsLogStart )   <span class="co"># log(x+c) </span></span>
<span id="cb185-150"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-150" aria-hidden="true" tabindex="-1"></a>                            <span class="co"># This is equivalent to below. But the prior counts is more important</span></span>
<span id="cb185-151"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-151" aria-hidden="true" tabindex="-1"></a>                            <span class="co">#x &lt;- cpm(DGEList(counts = x),log=TRUE, prior.count=input_countsLogStart )  #log CPM from edgeR</span></span>
<span id="cb185-152"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-152" aria-hidden="true" tabindex="-1"></a>                            <span class="co">#x &lt;- x-min(x)  # shift values to avoid negative numbers</span></span>
<span id="cb185-153"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-153" aria-hidden="true" tabindex="-1"></a>                        }</span>
<span id="cb185-154"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-154" aria-hidden="true" tabindex="-1"></a>                    }</span>
<span id="cb185-155"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-155" aria-hidden="true" tabindex="-1"></a>                } <span class="cf">else</span> </span>
<span id="cb185-156"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-156" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>( input_dataFileFormat <span class="sc">==</span> <span class="dv">3</span>)  {  <span class="co"># other data type</span></span>
<span id="cb185-157"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-158"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-158" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#neg_lfc neg_fdr pos_lfc pos_fdr </span></span>
<span id="cb185-159"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-159" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#11       1      11       1 </span></span>
<span id="cb185-160"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-160" aria-hidden="true" tabindex="-1"></a>                        </span>
<span id="cb185-161"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-161" aria-hidden="true" tabindex="-1"></a>                        n2 <span class="ot">=</span> ( <span class="fu">dim</span>(x)[<span class="dv">2</span>] <span class="sc">%/%</span> <span class="dv">2</span>) <span class="co"># 5 --&gt; 2</span></span>
<span id="cb185-162"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-162" aria-hidden="true" tabindex="-1"></a>                        <span class="co"># It looks like it contains P values</span></span>
<span id="cb185-163"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-163" aria-hidden="true" tabindex="-1"></a>                        <span class="co"># ranges of columns add 0.2 and round to whole. For P value columns this should be 1</span></span>
<span id="cb185-164"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-164" aria-hidden="true" tabindex="-1"></a>                        tem <span class="ot">=</span> <span class="fu">round</span>( <span class="fu">apply</span>(x, <span class="dv">2</span>, <span class="cf">function</span>( y) <span class="fu">max</span>(y)<span class="sc">-</span> <span class="fu">min</span>(y))  <span class="sc">+</span> .<span class="dv">2</span>)     </span>
<span id="cb185-165"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-165" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>( <span class="fu">sum</span>(tem[(<span class="dv">1</span><span class="sc">:</span>n2)<span class="sc">*</span><span class="dv">2</span>  ] <span class="sc">==</span>  <span class="dv">1</span> ) <span class="sc">==</span> n2 <span class="sc">|</span> </span>
<span id="cb185-166"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-166" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">sum</span>(tem[(<span class="dv">1</span><span class="sc">:</span>n2)<span class="sc">*</span><span class="dv">2-1</span>  ] <span class="sc">==</span>  <span class="dv">1</span> ) <span class="sc">==</span> n2 ) {         </span>
<span id="cb185-167"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-167" aria-hidden="true" tabindex="-1"></a>                            x <span class="ot">=</span> x[,<span class="dv">1</span><span class="sc">:</span>(<span class="dv">2</span><span class="sc">*</span>n2) ,drop<span class="ot">=</span><span class="cn">FALSE</span> ] <span class="co"># if 5, change it to 4            </span></span>
<span id="cb185-168"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-168" aria-hidden="true" tabindex="-1"></a>                            <span class="cf">if</span>(tem[<span class="dv">2</span>] <span class="sc">==</span> <span class="dv">1</span>) { <span class="co"># FDR follows Fold-change</span></span>
<span id="cb185-169"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-169" aria-hidden="true" tabindex="-1"></a>                                pvals <span class="ot">=</span> x [,<span class="dv">2</span><span class="sc">*</span>(<span class="dv">1</span><span class="sc">:</span>n2 ),drop<span class="ot">=</span><span class="cn">FALSE</span> ]  <span class="co"># 2, 4, 6</span></span>
<span id="cb185-170"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-170" aria-hidden="true" tabindex="-1"></a>                                x <span class="ot">=</span> x[, <span class="dv">2</span><span class="sc">*</span>(<span class="dv">1</span><span class="sc">:</span>n2 )<span class="sc">-</span><span class="dv">1</span>,drop<span class="ot">=</span><span class="cn">FALSE</span>]   <span class="co"># 1, 3, 5</span></span>
<span id="cb185-171"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-172"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-172" aria-hidden="true" tabindex="-1"></a>                            } <span class="cf">else</span> {    <span class="co"># FDR follows Fold-change</span></span>
<span id="cb185-173"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-173" aria-hidden="true" tabindex="-1"></a>                                pvals <span class="ot">=</span> x [,<span class="dv">2</span><span class="sc">*</span>(<span class="dv">1</span><span class="sc">:</span>n2 )<span class="sc">-</span><span class="dv">1</span>,drop<span class="ot">=</span><span class="cn">FALSE</span> ]  <span class="co"># 2, 4, 6     </span></span>
<span id="cb185-174"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-174" aria-hidden="true" tabindex="-1"></a>                                x <span class="ot">=</span> x[, <span class="dv">2</span><span class="sc">*</span>(<span class="dv">1</span><span class="sc">:</span> n2 ),drop<span class="ot">=</span><span class="cn">FALSE</span>]   <span class="co"># 1, 3, 5</span></span>
<span id="cb185-175"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-175" aria-hidden="true" tabindex="-1"></a>                            }</span>
<span id="cb185-176"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-176" aria-hidden="true" tabindex="-1"></a>                        }</span>
<span id="cb185-177"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-177" aria-hidden="true" tabindex="-1"></a>                    ix <span class="ot">=</span>  <span class="fu">which</span>(<span class="fu">apply</span>(x,<span class="dv">1</span>, <span class="cf">function</span>(y) <span class="fu">max</span>(y)<span class="sc">-</span> <span class="fu">min</span>(y) ) <span class="sc">&gt;</span> <span class="dv">0</span>  )</span>
<span id="cb185-178"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-178" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">&lt;-</span> x[ix,]  <span class="co"># remove rows with all the same levels</span></span>
<span id="cb185-179"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-179" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(pvals) )</span>
<span id="cb185-180"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-180" aria-hidden="true" tabindex="-1"></a>                        pvals <span class="ot">=</span> pvals[ix,]</span>
<span id="cb185-181"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-181" aria-hidden="true" tabindex="-1"></a>                        </span>
<span id="cb185-182"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-182" aria-hidden="true" tabindex="-1"></a>                    }</span>
<span id="cb185-183"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-183" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb185-184"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-184" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb185-185"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-185" aria-hidden="true" tabindex="-1"></a>                dataSize <span class="ot">=</span> <span class="fu">dim</span>(x);</span>
<span id="cb185-186"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-186" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="sc">!</span>(<span class="fu">dim</span>(x)[<span class="dv">1</span>]<span class="sc">&gt;</span><span class="dv">5</span> <span class="sc">&amp;</span> <span class="fu">dim</span>(x)[<span class="dv">2</span>]<span class="sc">&gt;</span><span class="dv">1</span>)) </span>
<span id="cb185-187"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-187" aria-hidden="true" tabindex="-1"></a>                <span class="fu">stop</span> ( <span class="st">&quot;Data file not recognized. Please double check.&quot;</span>)</span>
<span id="cb185-188"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-189"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-190"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-190" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb185-191"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-191" aria-hidden="true" tabindex="-1"></a>                sampleInfoDemo<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb185-192"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-192" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>( input_goButton <span class="sc">&gt;</span><span class="dv">0</span>)</span>
<span id="cb185-193"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-193" aria-hidden="true" tabindex="-1"></a>                    sampleInfoDemo <span class="ot">&lt;-</span> <span class="fu">t</span>( <span class="fu">read.csv</span>(demoDataFile2,<span class="at">row.names=</span><span class="dv">1</span>,<span class="at">header=</span>T,<span class="at">colClasses=</span><span class="st">&quot;character&quot;</span>) )</span>
<span id="cb185-194"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-195"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-195" aria-hidden="true" tabindex="-1"></a>                    finalResult <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">data =</span> <span class="fu">as.matrix</span>(x), <span class="at">mean.kurtosis =</span> mean.kurtosis, <span class="at">rawCounts =</span> rawCounts, <span class="at">dataTypeWarning=</span>dataTypeWarning, <span class="at">dataSize=</span><span class="fu">c</span>(dataSizeOriginal,dataSize),<span class="at">sampleInfoDemo=</span>sampleInfoDemo, <span class="at">pvals =</span>pvals )</span>
<span id="cb185-196"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-196" aria-hidden="true" tabindex="-1"></a>                <span class="fu">return</span>(finalResult)</span>
<span id="cb185-197"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-198"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-198" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb185-199"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-200"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-200" aria-hidden="true" tabindex="-1"></a>readSampleInfo <span class="ot">&lt;-</span> <span class="cf">function</span>(inFile){</span>
<span id="cb185-201"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-202"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-202" aria-hidden="true" tabindex="-1"></a>                dataTypeWarning <span class="ot">=</span><span class="dv">0</span></span>
<span id="cb185-203"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-203" aria-hidden="true" tabindex="-1"></a>                dataType <span class="ot">=</span><span class="fu">c</span>(<span class="cn">TRUE</span>)</span>
<span id="cb185-204"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-205"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-205" aria-hidden="true" tabindex="-1"></a>                <span class="co">#---------------Read file</span></span>
<span id="cb185-206"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-206" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(inFile,<span class="at">row.names=</span><span class="dv">1</span>,<span class="at">header=</span>T,<span class="at">colClasses=</span><span class="st">&quot;character&quot;</span>)   <span class="co"># try CSV</span></span>
<span id="cb185-207"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-207" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="fu">dim</span>(x)[<span class="dv">2</span>] <span class="sc">&lt;=</span> <span class="dv">2</span> )   <span class="co"># if less than 3 columns, try tab-deliminated</span></span>
<span id="cb185-208"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-208" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">&lt;-</span> <span class="fu">read.table</span>(inFile, <span class="at">row.names=</span><span class="dv">1</span>,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">header=</span><span class="cn">TRUE</span>,<span class="at">colClasses=</span><span class="st">&quot;character&quot;</span>)</span>
<span id="cb185-209"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-209" aria-hidden="true" tabindex="-1"></a>                <span class="co"># remove &quot;-&quot; or &quot;.&quot; from sample names</span></span>
<span id="cb185-210"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-210" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(x) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(x))</span>
<span id="cb185-211"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-211" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(x) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(x))    </span>
<span id="cb185-212"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-213"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-213" aria-hidden="true" tabindex="-1"></a>                <span class="co">#----------------Matching with column names of expression file</span></span>
<span id="cb185-214"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-214" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">=</span> <span class="fu">match</span>(<span class="fu">toupper</span>(<span class="fu">colnames</span>(readData.out<span class="sc">$</span>data)), <span class="fu">toupper</span>(<span class="fu">colnames</span>(x)) ) </span>
<span id="cb185-215"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-215" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">=</span> ix[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(ix))] <span class="co"># remove NA</span></span>
<span id="cb185-216"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-217"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-217" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="sc">!</span> ( <span class="fu">length</span>(<span class="fu">unique</span>(ix) ) <span class="sc">==</span> <span class="fu">dim</span>(readData.out<span class="sc">$</span>data)[<span class="dv">2</span>] </span>
<span id="cb185-218"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-218" aria-hidden="true" tabindex="-1"></a>                       <span class="sc">&amp;</span> <span class="fu">dim</span>(x)[<span class="dv">1</span>]<span class="sc">&gt;=</span><span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="sc">&lt;</span><span class="dv">500</span> ) ) <span class="co"># at least one row, it can not be more than 500 rows</span></span>
<span id="cb185-219"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-219" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">stop</span>( <span class="st">&quot;Error!!! Sample information file not recognized. Sample names must be exactly the same. Each row is a factor. Each column represent a sample.  Please see documentation on format.&quot;</span>)</span>
<span id="cb185-220"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-220" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb185-221"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-221" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb185-222"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-222" aria-hidden="true" tabindex="-1"></a>                <span class="co">#-----------Double check factor levels, change if needed</span></span>
<span id="cb185-223"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-223" aria-hidden="true" tabindex="-1"></a>                <span class="co"># remove &quot;-&quot; or &quot;.&quot; from factor levels</span></span>
<span id="cb185-224"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-224" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(x)[<span class="dv">1</span>]) {</span>
<span id="cb185-225"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-225" aria-hidden="true" tabindex="-1"></a>                   x[i,] <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;&quot;</span>,x[i,])</span>
<span id="cb185-226"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-226" aria-hidden="true" tabindex="-1"></a>                   x[i,] <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>,x[i,])             </span>
<span id="cb185-227"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-227" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb185-228"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-228" aria-hidden="true" tabindex="-1"></a>                <span class="co"># if levels from different factors match</span></span>
<span id="cb185-229"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-229" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>( <span class="fu">length</span>(<span class="fu">unique</span>(ix) ) <span class="sc">==</span> <span class="fu">dim</span>(readData.out<span class="sc">$</span>data)[<span class="dv">2</span>]) { <span class="co"># matches exactly</span></span>
<span id="cb185-230"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-230" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">=</span> x[,ix]</span>
<span id="cb185-231"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-231" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># if the levels of different factors are the same, it may cause problems</span></span>
<span id="cb185-232"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-232" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>( <span class="fu">sum</span>( <span class="fu">apply</span>(x, <span class="dv">1</span>, <span class="cf">function</span>(y) <span class="fu">length</span>(<span class="fu">unique</span>(y)))) <span class="sc">&gt;</span> <span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">unlist</span>(x) ) ) ) {</span>
<span id="cb185-233"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-233" aria-hidden="true" tabindex="-1"></a>                        tem2 <span class="ot">=</span><span class="fu">apply</span>(x,<span class="dv">2</span>, <span class="cf">function</span>(y) <span class="fu">paste0</span>( <span class="fu">names</span>(y),y)) <span class="co"># factor names are added to levels</span></span>
<span id="cb185-234"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-234" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">rownames</span>(tem2) <span class="ot">=</span> <span class="fu">rownames</span>(x)</span>
<span id="cb185-235"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-235" aria-hidden="true" tabindex="-1"></a>                        x <span class="ot">&lt;-</span> tem2               </span>
<span id="cb185-236"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-236" aria-hidden="true" tabindex="-1"></a>                    }</span>
<span id="cb185-237"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-237" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">return</span>(<span class="fu">t</span>( x ) )         </span>
<span id="cb185-238"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-238" aria-hidden="true" tabindex="-1"></a>                } <span class="cf">else</span> <span class="fu">retrun</span>(<span class="cn">NULL</span>)</span>
<span id="cb185-239"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-239" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb185-240"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-240" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb185-241"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-241" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb185-242"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-243"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-243" aria-hidden="true" tabindex="-1"></a>textTransform <span class="ot">&lt;-</span> <span class="cf">function</span> () { </span>
<span id="cb185-244"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-244" aria-hidden="true" tabindex="-1"></a>        k.value <span class="ot">=</span>  readData.out<span class="sc">$</span>mean.kurtosis     </span>
<span id="cb185-245"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-245" aria-hidden="true" tabindex="-1"></a>        tem <span class="ot">=</span> <span class="fu">paste</span>( <span class="st">&quot;Mean Kurtosis =  &quot;</span>, <span class="fu">round</span>(k.value,<span class="dv">2</span>), <span class="st">&quot;.</span><span class="sc">\n</span><span class="st">&quot;</span>,<span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb185-246"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-246" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( k.value <span class="sc">&gt;</span> kurtosis.log) tem <span class="ot">=</span> <span class="fu">paste</span>(tem, <span class="st">&quot; Detected extremely large values. </span><span class="sc">\n</span><span class="st">  When kurtosis &gt;&quot;</span>, kurtosis.log,</span>
<span id="cb185-247"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-247" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;, log transformation is automatically applied.</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>) <span class="cf">else</span> <span class="cf">if</span> (k.value<span class="sc">&gt;</span>kurtosis.warning)</span>
<span id="cb185-248"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-248" aria-hidden="true" tabindex="-1"></a>        {tem <span class="ot">=</span> <span class="fu">paste</span>(tem, <span class="st">&quot; Detected  large numbers with kurtosis &gt;&quot;</span>,</span>
<span id="cb185-249"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-249" aria-hidden="true" tabindex="-1"></a>        kurtosis.warning,<span class="st">&quot;. </span><span class="sc">\n</span><span class="st"> Log transformation is recommended.&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>) }</span>
<span id="cb185-250"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-250" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(readData.out<span class="sc">$</span>dataTypeWarning <span class="sc">==</span> <span class="dv">1</span> ) {</span>
<span id="cb185-251"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-251" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">paste</span> (tem, <span class="st">&quot; ------Warning!!! </span><span class="sc">\n</span><span class="st">Data matrix contains all integers. </span><span class="sc">\n</span><span class="st"> It seems to be read counts!!! </span><span class="sc">\n</span><span class="st"> Please select appropriate data type in the previous page and reload file.&quot;</span>)}</span>
<span id="cb185-252"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-252" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(readData.out<span class="sc">$</span>dataTypeWarning <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span> ) {</span>
<span id="cb185-253"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-253" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">paste</span> (tem, <span class="st">&quot;  ------Warning!!! </span><span class="sc">\n</span><span class="st">Data does not look like read counts data. </span><span class="sc">\n</span><span class="st"> Please select appropriate data type in the previous page and reload file.&quot;</span>)}</span>
<span id="cb185-254"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-254" aria-hidden="true" tabindex="-1"></a>        tem</span>
<span id="cb185-255"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-255" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb185-256"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-257"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-258"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-258" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up gene sets. Remove spaces and other control characters from gene names  </span></span>
<span id="cb185-259"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-259" aria-hidden="true" tabindex="-1"></a>cleanGeneSet <span class="ot">&lt;-</span> <span class="cf">function</span> (x){</span>
<span id="cb185-260"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-260" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove duplicate; upper case; remove special characters</span></span>
<span id="cb185-261"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-261" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">unique</span>( <span class="fu">toupper</span>( <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">| &quot;</span>,<span class="st">&quot;&quot;</span>,x) ) )</span>
<span id="cb185-262"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-262" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> x[<span class="fu">which</span>( <span class="fu">nchar</span>(x)<span class="sc">&gt;</span><span class="dv">1</span>) ]  <span class="co"># genes should have at least two characters</span></span>
<span id="cb185-263"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(x)</span>
<span id="cb185-264"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-264" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb185-265"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-265" aria-hidden="true" tabindex="-1"></a>findSpeciesById <span class="ot">&lt;-</span> <span class="cf">function</span> (speciesID){ <span class="co"># find species name use id</span></span>
<span id="cb185-266"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>( orgInfo[<span class="fu">which</span>(orgInfo<span class="sc">$</span>id <span class="sc">==</span> speciesID),]  )</span>
<span id="cb185-267"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-267" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb185-268"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-269"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-269" aria-hidden="true" tabindex="-1"></a><span class="co"># just return name</span></span>
<span id="cb185-270"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-270" aria-hidden="true" tabindex="-1"></a>findSpeciesByIdName <span class="ot">&lt;-</span> <span class="cf">function</span> (speciesID){ <span class="co"># find species name use id</span></span>
<span id="cb185-271"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-271" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>( orgInfo[<span class="fu">which</span>(orgInfo<span class="sc">$</span>id <span class="sc">==</span> speciesID),<span class="dv">3</span>]  )</span>
<span id="cb185-272"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-272" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb185-273"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-274"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-274" aria-hidden="true" tabindex="-1"></a><span class="co"># convert gene IDs to ensembl gene ids and find species</span></span>
<span id="cb185-275"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-275" aria-hidden="true" tabindex="-1"></a>convertID <span class="ot">&lt;-</span> <span class="cf">function</span> (query,selectOrg, selectGO) {</span>
<span id="cb185-276"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-276" aria-hidden="true" tabindex="-1"></a>    querySet <span class="ot">&lt;-</span> <span class="fu">cleanGeneSet</span>( <span class="fu">unlist</span>( <span class="fu">strsplit</span>( <span class="fu">toupper</span>(query),<span class="st">&#39;</span><span class="sc">\t</span><span class="st">| |</span><span class="sc">\n</span><span class="st">|</span><span class="sc">\\</span><span class="st">,&#39;</span>)))</span>
<span id="cb185-277"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-277" aria-hidden="true" tabindex="-1"></a>    <span class="co"># querySet is ensgene data for example, ENSG00000198888, ENSG00000198763, ENSG00000198804</span></span>
<span id="cb185-278"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-278" aria-hidden="true" tabindex="-1"></a>    querySTMT <span class="ot">&lt;-</span> <span class="fu">paste</span>( <span class="st">&quot;select distinct id,ens,species from mapping where id IN (&#39;&quot;</span>, <span class="fu">paste</span>(querySet,<span class="at">collapse=</span><span class="st">&quot;&#39;, &#39;&quot;</span>),<span class="st">&quot;&#39;)&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb185-279"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-279" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(convert, querySTMT)</span>
<span id="cb185-280"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-280" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>  ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb185-281"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-281" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(selectOrg <span class="sc">==</span> speciesChoice[[<span class="dv">1</span>]]) {</span>
<span id="cb185-282"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-282" aria-hidden="true" tabindex="-1"></a>        comb <span class="ot">=</span> <span class="fu">paste</span>( result<span class="sc">$</span>species,result<span class="sc">$</span>idType)</span>
<span id="cb185-283"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-283" aria-hidden="true" tabindex="-1"></a>        sortedCounts <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">table</span>(comb),<span class="at">decreasing=</span>T)</span>
<span id="cb185-284"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-284" aria-hidden="true" tabindex="-1"></a>        recognized <span class="ot">=</span><span class="fu">names</span>(sortedCounts[<span class="dv">1</span>])</span>
<span id="cb185-285"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-285" aria-hidden="true" tabindex="-1"></a>        result <span class="ot">&lt;-</span> result[<span class="fu">which</span>(comb <span class="sc">==</span> recognized),]</span>
<span id="cb185-286"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-286" aria-hidden="true" tabindex="-1"></a>        speciesMatched<span class="ot">=</span>sortedCounts</span>
<span id="cb185-287"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-287" aria-hidden="true" tabindex="-1"></a>        <span class="fu">names</span>(speciesMatched )<span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot; .*&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">names</span>(sortedCounts) ) ), findSpeciesByIdName  ) </span>
<span id="cb185-288"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-288" aria-hidden="true" tabindex="-1"></a>        speciesMatched <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>( speciesMatched )</span>
<span id="cb185-289"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-289" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">length</span>(sortedCounts) <span class="sc">==</span> <span class="dv">1</span>) { <span class="co"># if only  one species matched</span></span>
<span id="cb185-290"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-290" aria-hidden="true" tabindex="-1"></a>        speciesMatched[<span class="dv">1</span>,<span class="dv">1</span>] <span class="ot">&lt;-</span><span class="fu">paste</span>( <span class="fu">rownames</span>(speciesMatched), <span class="st">&quot;(&quot;</span>,speciesMatched[<span class="dv">1</span>,<span class="dv">1</span>],<span class="st">&quot;)&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb185-291"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-291" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {<span class="co"># if more than one species matched</span></span>
<span id="cb185-292"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-292" aria-hidden="true" tabindex="-1"></a>            speciesMatched[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(speciesMatched[,<span class="dv">1</span>])</span>
<span id="cb185-293"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-293" aria-hidden="true" tabindex="-1"></a>            speciesMatched[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">paste</span>( speciesMatched[,<span class="dv">1</span>],<span class="st">&quot; (&quot;</span>,speciesMatched[,<span class="dv">2</span>], <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>) </span>
<span id="cb185-294"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-294" aria-hidden="true" tabindex="-1"></a>            speciesMatched[<span class="dv">1</span>,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">paste</span>( speciesMatched[<span class="dv">1</span>,<span class="dv">1</span>],<span class="st">&quot;   ***Used in mapping***  To change, select from above and resubmit query.&quot;</span>)     </span>
<span id="cb185-295"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-295" aria-hidden="true" tabindex="-1"></a>            speciesMatched <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(speciesMatched[,<span class="dv">1</span>])</span>
<span id="cb185-296"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-296" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb185-297"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-297" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> { <span class="co"># if species is selected</span></span>
<span id="cb185-298"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-298" aria-hidden="true" tabindex="-1"></a>        result <span class="ot">&lt;-</span> result[<span class="fu">which</span>(result<span class="sc">$</span>species <span class="sc">==</span> selectOrg ) ,]</span>
<span id="cb185-299"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-299" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>  ) <span class="fu">return</span>(<span class="cn">NULL</span>) <span class="co">#stop(&quot;ID not recognized!&quot;)</span></span>
<span id="cb185-300"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-300" aria-hidden="true" tabindex="-1"></a>        speciesMatched <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">paste</span>(<span class="st">&quot;Using selected species &quot;</span>, <span class="fu">findSpeciesByIdName</span>(selectOrg) )  )</span>
<span id="cb185-301"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-301" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb185-302"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-302" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> result[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">duplicated</span>(result[,<span class="dv">2</span>]) ),] <span class="co"># remove duplicates in ensembl_gene_id</span></span>
<span id="cb185-303"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-303" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> result[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">duplicated</span>(result[,<span class="dv">1</span>]) ),] <span class="co"># remove duplicates in user ID</span></span>
<span id="cb185-304"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-304" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(speciesMatched) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Matched Species (genes)&quot;</span> ) </span>
<span id="cb185-305"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-305" aria-hidden="true" tabindex="-1"></a>    conversionTable <span class="ot">&lt;-</span> result[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]; <span class="fu">colnames</span>(conversionTable) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;User_input&quot;</span>,<span class="st">&quot;ensembl_gene_id&quot;</span>)</span>
<span id="cb185-306"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-306" aria-hidden="true" tabindex="-1"></a>    conversionTable<span class="sc">$</span>Species <span class="ot">=</span> <span class="fu">sapply</span>(result[,<span class="dv">3</span>], findSpeciesByIdName )</span>
<span id="cb185-307"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-308"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-308" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">originalIDs =</span> querySet,<span class="at">IDs=</span><span class="fu">unique</span>( result[,<span class="dv">2</span>]), </span>
<span id="cb185-309"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-309" aria-hidden="true" tabindex="-1"></a>                <span class="at">species =</span> <span class="fu">findSpeciesById</span>(result<span class="sc">$</span>species[<span class="dv">1</span>]), </span>
<span id="cb185-310"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-310" aria-hidden="true" tabindex="-1"></a>                <span class="co">#idType = findIDtypeById(result$idType[1] ),</span></span>
<span id="cb185-311"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-311" aria-hidden="true" tabindex="-1"></a>                <span class="at">speciesMatched =</span> speciesMatched,</span>
<span id="cb185-312"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-312" aria-hidden="true" tabindex="-1"></a>                <span class="at">conversionTable =</span> conversionTable</span>
<span id="cb185-313"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-313" aria-hidden="true" tabindex="-1"></a>                ) )</span>
<span id="cb185-314"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-314" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb185-315"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-316"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-317"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-317" aria-hidden="true" tabindex="-1"></a><span class="co"># retrieve detailed info on genes</span></span>
<span id="cb185-318"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-318" aria-hidden="true" tabindex="-1"></a>geneInfo <span class="ot">&lt;-</span> <span class="cf">function</span> (<span class="at">fileName =</span> <span class="cn">NULL</span>){</span>
<span id="cb185-319"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-320"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-320" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(fileName))  <span class="co"># if only one file           #WBGene0000001 some ensembl gene ids in lower case</span></span>
<span id="cb185-321"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-321" aria-hidden="true" tabindex="-1"></a>    { x <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="fu">as.character</span>(fileName) ); x[,<span class="dv">1</span>]<span class="ot">=</span> <span class="fu">toupper</span>(x[,<span class="dv">1</span>]) } <span class="cf">else</span> <span class="co"># read in the chosen file </span></span>
<span id="cb185-322"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-322" aria-hidden="true" tabindex="-1"></a>    { <span class="fu">return</span>(<span class="fu">as.data.frame</span>(<span class="st">&quot;No file.&quot;</span>) )   }</span>
<span id="cb185-323"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-323" aria-hidden="true" tabindex="-1"></a>    Set<span class="ot">=</span><span class="fu">rep</span>(<span class="st">&quot;List&quot;</span>,<span class="fu">nrow</span>(x))</span>
<span id="cb185-324"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-324" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">cbind</span>(x,Set) )</span>
<span id="cb185-325"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-325" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb185-326"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-327"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-327" aria-hidden="true" tabindex="-1"></a>convertedData <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb185-328"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-328" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">is.null</span>(converted.out ) ) <span class="fu">return</span>( readData.out<span class="sc">$</span>data) <span class="co"># if id or species is not recognized use original data.</span></span>
<span id="cb185-329"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-329" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb185-330"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-330" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(input_noIDConversion) <span class="fu">return</span>( readData.out<span class="sc">$</span>data )</span>
<span id="cb185-331"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-331" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb185-332"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-332" aria-hidden="true" tabindex="-1"></a>                mapping <span class="ot">&lt;-</span> converted.out<span class="sc">$</span>conversionTable</span>
<span id="cb185-333"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-333" aria-hidden="true" tabindex="-1"></a>                <span class="co"># cat (paste( &quot;\nData:&quot;,input_selectOrg) )</span></span>
<span id="cb185-334"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-334" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">=</span>readData.out<span class="sc">$</span>data</span>
<span id="cb185-335"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-336"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-336" aria-hidden="true" tabindex="-1"></a>                <span class="fu">rownames</span>(x) <span class="ot">=</span> <span class="fu">toupper</span>(<span class="fu">rownames</span>(x))</span>
<span id="cb185-337"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-337" aria-hidden="true" tabindex="-1"></a>                <span class="co"># any gene not recognized by the database is disregarded</span></span>
<span id="cb185-338"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-338" aria-hidden="true" tabindex="-1"></a>                <span class="co"># x1 = merge(mapping[,1:2],x,  by.y = &#39;row.names&#39;, by.x = &#39;User_input&#39;)</span></span>
<span id="cb185-339"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-339" aria-hidden="true" tabindex="-1"></a>                <span class="co"># the 3 lines keeps the unrecogized genes using original IDs</span></span>
<span id="cb185-340"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-340" aria-hidden="true" tabindex="-1"></a>                x1 <span class="ot">=</span> <span class="fu">merge</span>(mapping[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],x,  <span class="at">by.y =</span> <span class="st">&#39;row.names&#39;</span>, <span class="at">by.x =</span> <span class="st">&#39;User_input&#39;</span>, <span class="at">all.y=</span><span class="cn">TRUE</span>)</span>
<span id="cb185-341"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-342"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-342" aria-hidden="true" tabindex="-1"></a>                <span class="co"># original IDs used if ID is not matched in database</span></span>
<span id="cb185-343"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-343" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">is.na</span>(x1[,<span class="dv">2</span>]) )</span>
<span id="cb185-344"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-344" aria-hidden="true" tabindex="-1"></a>                x1[ix,<span class="dv">2</span>] <span class="ot">=</span> x1[ix,<span class="dv">1</span>] </span>
<span id="cb185-345"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-345" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb185-346"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-346" aria-hidden="true" tabindex="-1"></a>                <span class="co">#multiple matched IDs, use the one with highest SD</span></span>
<span id="cb185-347"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-347" aria-hidden="true" tabindex="-1"></a>                tem <span class="ot">=</span> <span class="fu">apply</span>(x1[,<span class="dv">3</span><span class="sc">:</span>(<span class="fu">dim</span>(x1)[<span class="dv">2</span>])],<span class="dv">1</span>,sd)</span>
<span id="cb185-348"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-348" aria-hidden="true" tabindex="-1"></a>                x1 <span class="ot">=</span> x1[<span class="fu">order</span>(x1[,<span class="dv">2</span>],<span class="sc">-</span>tem),]</span>
<span id="cb185-349"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-349" aria-hidden="true" tabindex="-1"></a>                x1 <span class="ot">=</span> x1[<span class="sc">!</span><span class="fu">duplicated</span>(x1[,<span class="dv">2</span>]) ,]</span>
<span id="cb185-350"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-350" aria-hidden="true" tabindex="-1"></a>                <span class="fu">rownames</span>(x1) <span class="ot">=</span> x1[,<span class="dv">2</span>]</span>
<span id="cb185-351"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-351" aria-hidden="true" tabindex="-1"></a>                x1 <span class="ot">=</span> <span class="fu">as.matrix</span>(x1[,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="sc">-</span><span class="dv">2</span>)])</span>
<span id="cb185-352"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-352" aria-hidden="true" tabindex="-1"></a>                tem <span class="ot">=</span> <span class="fu">apply</span>(x1,<span class="dv">1</span>,sd)</span>
<span id="cb185-353"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-353" aria-hidden="true" tabindex="-1"></a>                x1 <span class="ot">=</span> x1[<span class="fu">order</span>(<span class="sc">-</span>tem),]  <span class="co"># sort again by SD</span></span>
<span id="cb185-354"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-354" aria-hidden="true" tabindex="-1"></a>                <span class="fu">return</span>(x1)</span>
<span id="cb185-355"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-356"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-356" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb185-357"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-358"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-358" aria-hidden="true" tabindex="-1"></a>convertedCounts <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb185-359"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-359" aria-hidden="true" tabindex="-1"></a>              <span class="cf">if</span>( <span class="fu">is.null</span>(converted.out ) ) <span class="fu">return</span>( readData.out<span class="sc">$</span>rawCounts)</span>
<span id="cb185-360"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-360" aria-hidden="true" tabindex="-1"></a>                mapping <span class="ot">&lt;-</span> converted.out<span class="sc">$</span>conversionTable</span>
<span id="cb185-361"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-362"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-362" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">=</span>readData.out<span class="sc">$</span>rawCounts</span>
<span id="cb185-363"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-364"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-364" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">rownames</span>(x) <span class="ot">=</span> <span class="fu">toupper</span>(<span class="fu">rownames</span>(x))</span>
<span id="cb185-365"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-365" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># any gene not recognized by the database is disregarded</span></span>
<span id="cb185-366"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-366" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># x1 = merge(mapping[,1:2],x,  by.y = &#39;row.names&#39;, by.x = &#39;User_input&#39;)</span></span>
<span id="cb185-367"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-367" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># the 3 lines keeps the unrecogized genes using original IDs</span></span>
<span id="cb185-368"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-368" aria-hidden="true" tabindex="-1"></a>                    x1 <span class="ot">=</span> <span class="fu">merge</span>(mapping[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],x,  <span class="at">by.y =</span> <span class="st">&#39;row.names&#39;</span>, <span class="at">by.x =</span> <span class="st">&#39;User_input&#39;</span>, <span class="at">all.y=</span><span class="cn">TRUE</span>)</span>
<span id="cb185-369"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-369" aria-hidden="true" tabindex="-1"></a>                    ix <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">is.na</span>(x1[,<span class="dv">2</span>]) )</span>
<span id="cb185-370"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-370" aria-hidden="true" tabindex="-1"></a>                    x1[ix,<span class="dv">2</span>] <span class="ot">=</span> x1[ix,<span class="dv">1</span>] <span class="co"># original IDs used</span></span>
<span id="cb185-371"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-371" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb185-372"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-372" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#multiple matched IDs, use the one with highest SD</span></span>
<span id="cb185-373"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-373" aria-hidden="true" tabindex="-1"></a>                    tem <span class="ot">=</span> <span class="fu">apply</span>(x1[,<span class="dv">3</span><span class="sc">:</span>(<span class="fu">dim</span>(x1)[<span class="dv">2</span>])],<span class="dv">1</span>,sd)</span>
<span id="cb185-374"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-374" aria-hidden="true" tabindex="-1"></a>                    x1 <span class="ot">=</span> x1[<span class="fu">order</span>(x1[,<span class="dv">2</span>],<span class="sc">-</span>tem),]</span>
<span id="cb185-375"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-375" aria-hidden="true" tabindex="-1"></a>                    x1 <span class="ot">=</span> x1[<span class="sc">!</span><span class="fu">duplicated</span>(x1[,<span class="dv">2</span>]) ,]</span>
<span id="cb185-376"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-376" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">rownames</span>(x1) <span class="ot">=</span> x1[,<span class="dv">2</span>]</span>
<span id="cb185-377"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-377" aria-hidden="true" tabindex="-1"></a>                    x1 <span class="ot">=</span> <span class="fu">as.matrix</span>(x1[,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="sc">-</span><span class="dv">2</span>)])</span>
<span id="cb185-378"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-378" aria-hidden="true" tabindex="-1"></a>                    tem <span class="ot">=</span> <span class="fu">apply</span>(x1,<span class="dv">1</span>,sd)</span>
<span id="cb185-379"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-379" aria-hidden="true" tabindex="-1"></a>                    x1 <span class="ot">=</span> x1[<span class="fu">order</span>(<span class="sc">-</span>tem),]  <span class="co"># sort again by SD</span></span>
<span id="cb185-380"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-381"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-381" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">return</span>(x1)</span>
<span id="cb185-382"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-382" aria-hidden="true" tabindex="-1"></a>} <span class="co"># here</span></span>
<span id="cb185-383"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-384"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-384" aria-hidden="true" tabindex="-1"></a>nGenesFilter <span class="ot">&lt;-</span> <span class="cf">function</span>() { </span>
<span id="cb185-385"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-385" aria-hidden="true" tabindex="-1"></a>        tem <span class="ot">=</span> readData.out<span class="sc">$</span>dataSize</span>
<span id="cb185-386"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-386" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">match</span>( <span class="fu">toupper</span>( <span class="fu">rownames</span>(convertedData.out)), <span class="fu">toupper</span>(converted.out<span class="sc">$</span>conversionTable<span class="sc">$</span>ensembl_gene_id  ) )</span>
<span id="cb185-387"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-387" aria-hidden="true" tabindex="-1"></a>        nMatched <span class="ot">=</span> <span class="fu">sum</span>( <span class="sc">!</span><span class="fu">is.na</span>(ix) )</span>
<span id="cb185-388"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-388" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( input_noIDConversion) </span>
<span id="cb185-389"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-389" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>( <span class="fu">paste</span>(tem[<span class="dv">1</span>], <span class="st">&quot;genes in&quot;</span>, tem[<span class="dv">4</span>], <span class="st">&quot;samples.&quot;</span>, tem[<span class="dv">3</span>], <span class="st">&quot; genes passed filter.</span><span class="sc">\n</span><span class="st"> Original gene IDs used.&quot;</span> ) )  <span class="cf">else</span></span>
<span id="cb185-390"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-390" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>( <span class="fu">paste</span>(tem[<span class="dv">1</span>], <span class="st">&quot;genes in&quot;</span>, tem[<span class="dv">4</span>], <span class="st">&quot;samples. Of the&quot;</span>, tem[<span class="dv">3</span>], <span class="st">&quot; genes passed filter, </span><span class="sc">\n</span><span class="st">&quot;</span>, </span>
<span id="cb185-391"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-391" aria-hidden="true" tabindex="-1"></a>            nMatched, <span class="st">&quot; were converted to Ensembl gene IDs in our database. </span><span class="sc">\n</span></span>
<span id="cb185-392"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-392" aria-hidden="true" tabindex="-1"></a><span class="st">              The remaining &quot;</span>, tem[<span class="dv">3</span>]<span class="sc">-</span>nMatched, <span class="st">&quot;genes were kept in the data using original IDs.&quot;</span> ) )     </span>
<span id="cb185-393"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-393" aria-hidden="true" tabindex="-1"></a>}       </span>
<span id="cb185-394"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-395"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-395" aria-hidden="true" tabindex="-1"></a><span class="co"># Define sample groups based on column names</span></span>
<span id="cb185-396"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-396" aria-hidden="true" tabindex="-1"></a> detectGroups <span class="ot">&lt;-</span> <span class="cf">function</span> (x){  <span class="co"># x are col names</span></span>
<span id="cb185-397"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-397" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;[0-9]*$&quot;</span>,<span class="st">&quot;&quot;</span>,x) <span class="co"># Remove all numbers from end</span></span>
<span id="cb185-398"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-398" aria-hidden="true" tabindex="-1"></a>    <span class="co">#tem = gsub(&quot;_Rep|_rep|_REP&quot;,&quot;&quot;,tem)</span></span>
<span id="cb185-399"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-399" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_$&quot;</span>,<span class="st">&quot;&quot;</span>,tem); <span class="co"># remove &quot;_&quot; from end</span></span>
<span id="cb185-400"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-400" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_Rep$&quot;</span>,<span class="st">&quot;&quot;</span>,tem); <span class="co"># remove &quot;_Rep&quot; from end</span></span>
<span id="cb185-401"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-401" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_rep$&quot;</span>,<span class="st">&quot;&quot;</span>,tem); <span class="co"># remove &quot;_rep&quot; from end</span></span>
<span id="cb185-402"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-402" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_REP$&quot;</span>,<span class="st">&quot;&quot;</span>,tem)  <span class="co"># remove &quot;_REP&quot; from end</span></span>
<span id="cb185-403"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-403" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( tem )</span>
<span id="cb185-404"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-404" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb185-405"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-406"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-406" aria-hidden="true" tabindex="-1"></a>  <span class="co"># EDA plots -------------------------------------</span></span>
<span id="cb185-407"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-407" aria-hidden="true" tabindex="-1"></a> densityPlot <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb185-408"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-408" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">2</span>) <span class="co"># seed for random number generator</span></span>
<span id="cb185-409"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-409" aria-hidden="true" tabindex="-1"></a>mycolors <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">rainbow</span>(<span class="dv">20</span>))[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">20</span>,<span class="dv">10</span>,<span class="dv">11</span>,<span class="dv">2</span>,<span class="dv">19</span>,<span class="dv">3</span>,<span class="dv">12</span>,<span class="dv">4</span>,<span class="dv">13</span>,<span class="dv">5</span>,<span class="dv">14</span>,<span class="dv">6</span>,<span class="dv">15</span>,<span class="dv">7</span>,<span class="dv">16</span>,<span class="dv">8</span>,<span class="dv">17</span>,<span class="dv">9</span>,<span class="dv">18</span>)] <span class="co"># 20 colors for kNN clusters</span></span>
<span id="cb185-410"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-410" aria-hidden="true" tabindex="-1"></a><span class="co">#Each row of this matrix represents a color scheme;</span></span>
<span id="cb185-411"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-411" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb185-412"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-412" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> readData.out<span class="sc">$</span>data</span>
<span id="cb185-413"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-414"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-414" aria-hidden="true" tabindex="-1"></a>    myColors <span class="ot">=</span> mycolors</span>
<span id="cb185-415"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-415" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot</span>(<span class="fu">density</span>(x[,<span class="dv">1</span>]),<span class="at">col =</span> myColors[<span class="dv">1</span>], <span class="at">lwd=</span><span class="dv">2</span>,</span>
<span id="cb185-416"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-416" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab=</span><span class="st">&quot;Expresson values&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;Density&quot;</span>, <span class="at">main=</span> <span class="st">&quot;Distribution of transformed data&quot;</span>,</span>
<span id="cb185-417"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-417" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(<span class="fu">density</span>(x[,<span class="dv">1</span>])<span class="sc">$</span>y)<span class="sc">+</span>.<span class="dv">02</span> ) )</span>
<span id="cb185-418"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-418" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb185-419"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-419" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">dim</span>(x)[<span class="dv">2</span>] )</span>
<span id="cb185-420"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-420" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lines</span>(<span class="fu">density</span>(x[,i]),<span class="at">col=</span>myColors[i], <span class="at">lwd=</span><span class="dv">2</span> )</span>
<span id="cb185-421"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-421" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">dim</span>(x)[<span class="dv">2</span>]<span class="sc">&lt;</span> <span class="dv">31</span> ) <span class="co"># if too many samples do not show legends</span></span>
<span id="cb185-422"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-422" aria-hidden="true" tabindex="-1"></a>        <span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>, <span class="at">cex=</span><span class="dv">1</span>,<span class="fu">colnames</span>(x), <span class="at">lty=</span><span class="fu">rep</span>(<span class="dv">1</span>,<span class="fu">dim</span>(x)[<span class="dv">2</span>]), <span class="at">col=</span>myColors )  </span>
<span id="cb185-423"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-423" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb185-424"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-424" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb185-425"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-426"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-426" aria-hidden="true" tabindex="-1"></a> genePlot <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb185-427"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-427" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> convertedData.out</span>
<span id="cb185-428"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-428" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb185-429"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-429" aria-hidden="true" tabindex="-1"></a>    Symbols <span class="ot">&lt;-</span> <span class="fu">rownames</span>(x)</span>
<span id="cb185-430"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-430" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_selectOrg <span class="sc">!=</span> <span class="st">&quot;NEW&quot;</span>) {</span>
<span id="cb185-431"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-431" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">match</span>( <span class="fu">rownames</span>(x), allGeneInfo.out[,<span class="dv">1</span>])</span>
<span id="cb185-432"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-432" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">sum</span>( <span class="fu">is.na</span>(allGeneInfo.out<span class="sc">$</span>symbol )) <span class="sc">!=</span> <span class="fu">dim</span>(allGeneInfo.out )[<span class="dv">1</span>] ) {  <span class="co"># symbol really exists? </span></span>
<span id="cb185-433"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-433" aria-hidden="true" tabindex="-1"></a>            Symbols <span class="ot">=</span> <span class="fu">as.character</span>( allGeneInfo.out<span class="sc">$</span>symbol[ix] )</span>
<span id="cb185-434"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-434" aria-hidden="true" tabindex="-1"></a>            Symbols[<span class="fu">which</span>( <span class="fu">nchar</span>(Symbols) <span class="sc">&lt;=</span> <span class="dv">2</span> ) ] <span class="ot">&lt;-</span> <span class="fu">rownames</span>(x) [<span class="fu">which</span>( <span class="fu">nchar</span>(Symbols) <span class="sc">&lt;=</span> <span class="dv">2</span> ) ] </span>
<span id="cb185-435"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-435" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb185-436"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-436" aria-hidden="true" tabindex="-1"></a>       }</span>
<span id="cb185-437"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-437" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">=</span> <span class="fu">as.data.frame</span>(x)</span>
<span id="cb185-438"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-438" aria-hidden="true" tabindex="-1"></a>    x<span class="sc">$</span>Genes <span class="ot">=</span> Symbols</span>
<span id="cb185-439"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-439" aria-hidden="true" tabindex="-1"></a>    <span class="co"># matching from the beginning of symbol</span></span>
<span id="cb185-440"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-440" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">regexpr</span>(  <span class="fu">paste</span>(<span class="st">&quot;^&quot;</span> , <span class="fu">toupper</span>(input_geneSearch),<span class="at">sep=</span><span class="st">&quot;&quot;</span>)   ,<span class="fu">toupper</span>(x<span class="sc">$</span>Genes)) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb185-441"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-441" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb185-442"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-442" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">grepl</span>(<span class="st">&quot; &quot;</span>, input_geneSearch)  )  <span class="co"># if there is space character, do exact match</span></span>
<span id="cb185-443"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-443" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">match</span>(<span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">toupper</span>(input_geneSearch)),x<span class="sc">$</span>Genes)</span>
<span id="cb185-444"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-444" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb185-445"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-445" aria-hidden="true" tabindex="-1"></a>    <span class="co"># too few or too many genes found</span></span>
<span id="cb185-446"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-446" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">length</span>(ix) <span class="sc">&gt;</span> <span class="dv">50</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb185-447"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-447" aria-hidden="true" tabindex="-1"></a>      <span class="co"># no genes found</span></span>
<span id="cb185-448"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-448" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb185-449"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-449" aria-hidden="true" tabindex="-1"></a>    mdf <span class="ot">=</span> <span class="fu">melt</span>(x[ix,],<span class="at">id.vars=</span><span class="st">&quot;Genes&quot;</span>, <span class="at">value.name=</span><span class="st">&quot;value&quot;</span>, <span class="at">variable.name=</span><span class="st">&quot;samples&quot;</span>)</span>
<span id="cb185-450"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-451"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-451" aria-hidden="true" tabindex="-1"></a>    p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>mdf, <span class="fu">aes</span>(<span class="at">x=</span>samples, <span class="at">y=</span>value, <span class="at">group =</span> Genes, <span class="at">shape=</span>Genes, <span class="at">colour =</span> Genes)) <span class="sc">+</span></span>
<span id="cb185-452"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-452" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb185-453"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-453" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_point</span>( <span class="at">size=</span><span class="dv">5</span>,  <span class="at">fill=</span><span class="st">&quot;white&quot;</span>)<span class="sc">+</span> <span class="co">#shape=21  circle</span></span>
<span id="cb185-454"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-454" aria-hidden="true" tabindex="-1"></a>        <span class="co">#theme(axis.text.x = element_text(size=16,angle = 45, hjust = 1)) +</span></span>
<span id="cb185-455"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-455" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;Transformed expression level&quot;</span>) <span class="sc">+</span></span>
<span id="cb185-456"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-456" aria-hidden="true" tabindex="-1"></a>        <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(mdf<span class="sc">$</span>value)))</span>
<span id="cb185-457"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-457" aria-hidden="true" tabindex="-1"></a>    p1 <span class="ot">&lt;-</span> p1 <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>,<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="co"># theme(aspect.ratio=1) +</span></span>
<span id="cb185-458"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-458" aria-hidden="true" tabindex="-1"></a>     <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">size =</span> <span class="dv">16</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb185-459"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-459" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.text.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb185-460"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-460" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb185-461"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-461" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>) ) <span class="sc">+</span></span>
<span id="cb185-462"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-462" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">12</span>))    </span>
<span id="cb185-463"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-463" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb185-464"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-464" aria-hidden="true" tabindex="-1"></a>    p1  </span>
<span id="cb185-465"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-465" aria-hidden="true" tabindex="-1"></a>    }   </span>
<span id="cb185-466"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-466" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-467"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-467" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb185-468"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-468" aria-hidden="true" tabindex="-1"></a>geneBarPlotError <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb185-469"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-469" aria-hidden="true" tabindex="-1"></a>   x <span class="ot">&lt;-</span> convertedData.out</span>
<span id="cb185-470"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-470" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb185-471"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-471" aria-hidden="true" tabindex="-1"></a>    Symbols <span class="ot">&lt;-</span> <span class="fu">rownames</span>(x)</span>
<span id="cb185-472"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-472" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_selectOrg <span class="sc">!=</span> <span class="st">&quot;NEW&quot;</span>) {</span>
<span id="cb185-473"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-473" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">match</span>( <span class="fu">rownames</span>(x), allGeneInfo.out[,<span class="dv">1</span>])</span>
<span id="cb185-474"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-474" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">sum</span>( <span class="fu">is.na</span>(allGeneInfo.out<span class="sc">$</span>symbol )) <span class="sc">!=</span> <span class="fu">dim</span>(allGeneInfo.out )[<span class="dv">1</span>] ) {  <span class="co"># symbol really exists? </span></span>
<span id="cb185-475"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-475" aria-hidden="true" tabindex="-1"></a>            Symbols <span class="ot">=</span> <span class="fu">as.character</span>( allGeneInfo.out<span class="sc">$</span>symbol[ix] )</span>
<span id="cb185-476"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-476" aria-hidden="true" tabindex="-1"></a>            Symbols[<span class="fu">which</span>( <span class="fu">nchar</span>(Symbols) <span class="sc">&lt;=</span> <span class="dv">2</span> ) ] <span class="ot">&lt;-</span> <span class="fu">rownames</span>(x) [<span class="fu">which</span>( <span class="fu">nchar</span>(Symbols) <span class="sc">&lt;=</span> <span class="dv">2</span> ) ] </span>
<span id="cb185-477"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-477" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb185-478"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-478" aria-hidden="true" tabindex="-1"></a>       }</span>
<span id="cb185-479"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-479" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">=</span> <span class="fu">as.data.frame</span>(x)</span>
<span id="cb185-480"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-480" aria-hidden="true" tabindex="-1"></a>    x<span class="sc">$</span>Genes <span class="ot">=</span> Symbols</span>
<span id="cb185-481"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-481" aria-hidden="true" tabindex="-1"></a>    <span class="co">#write.csv(x,&quot;tem.csv&quot;)</span></span>
<span id="cb185-482"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-482" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Search for genes</span></span>
<span id="cb185-483"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-483" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ix = grep(&quot;HOXA&quot;,toupper(x$Genes) )</span></span>
<span id="cb185-484"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-484" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ix = grep(toupper(input_geneSearch),toupper(x$Genes))  # sox --&gt; Tsox  </span></span>
<span id="cb185-485"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-485" aria-hidden="true" tabindex="-1"></a>    <span class="co"># matching from the beginning of symbol</span></span>
<span id="cb185-486"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-486" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">regexpr</span>(  <span class="fu">paste</span>(<span class="st">&quot;^&quot;</span> , <span class="fu">toupper</span>(input_geneSearch),<span class="at">sep=</span><span class="st">&quot;&quot;</span>)   ,<span class="fu">toupper</span>(x<span class="sc">$</span>Genes)) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb185-487"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-487" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb185-488"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-488" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">grepl</span>(<span class="st">&quot; &quot;</span>, input_geneSearch)  )  <span class="co"># if there is space character, do exact match</span></span>
<span id="cb185-489"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-489" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">match</span>(<span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">toupper</span>(input_geneSearch)),x<span class="sc">$</span>Genes)</span>
<span id="cb185-490"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-490" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb185-491"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-491" aria-hidden="true" tabindex="-1"></a>    <span class="co"># too few or too many genes found</span></span>
<span id="cb185-492"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-492" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">length</span>(ix) <span class="sc">&gt;</span> <span class="dv">50</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb185-493"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-493" aria-hidden="true" tabindex="-1"></a>      <span class="co"># no genes found</span></span>
<span id="cb185-494"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-494" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb185-495"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-495" aria-hidden="true" tabindex="-1"></a>    mdf <span class="ot">=</span> <span class="fu">melt</span>(x[ix,],<span class="at">id.vars=</span><span class="st">&quot;Genes&quot;</span>, <span class="at">value.name=</span><span class="st">&quot;value&quot;</span>, <span class="at">variable.name=</span><span class="st">&quot;samples&quot;</span>)</span>
<span id="cb185-496"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-496" aria-hidden="true" tabindex="-1"></a>    mdf<span class="sc">$</span>count <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb185-497"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-497" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">=</span> <span class="fu">detectGroups</span>(mdf<span class="sc">$</span>samples)</span>
<span id="cb185-498"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-498" aria-hidden="true" tabindex="-1"></a>    Means <span class="ot">=</span> <span class="fu">aggregate</span>(mdf<span class="sc">$</span>value,<span class="at">by=</span><span class="fu">list</span>( g, mdf<span class="sc">$</span>Genes ), <span class="at">FUN =</span> mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>  )</span>
<span id="cb185-499"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-499" aria-hidden="true" tabindex="-1"></a>    SDs <span class="ot">=</span> <span class="fu">aggregate</span>(mdf<span class="sc">$</span>value,<span class="at">by=</span><span class="fu">list</span>( g, mdf<span class="sc">$</span>Genes ), <span class="at">FUN =</span> sd, <span class="at">na.rm=</span><span class="cn">TRUE</span>  )</span>
<span id="cb185-500"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-500" aria-hidden="true" tabindex="-1"></a>    Ns <span class="ot">=</span> <span class="fu">aggregate</span>(mdf<span class="sc">$</span>count, <span class="at">by=</span> <span class="fu">list</span>(g, mdf<span class="sc">$</span>Genes) , <span class="at">FUN =</span> sum  )</span>
<span id="cb185-501"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-501" aria-hidden="true" tabindex="-1"></a>    summarized <span class="ot">=</span> <span class="fu">cbind</span>(Means,SDs[,<span class="dv">3</span>],Ns[,<span class="dv">3</span>])</span>
<span id="cb185-502"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-502" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(summarized)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Samples&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;Mean&quot;</span>,<span class="st">&quot;SD&quot;</span>,<span class="st">&quot;N&quot;</span>)</span>
<span id="cb185-503"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-503" aria-hidden="true" tabindex="-1"></a>    summarized<span class="sc">$</span>SE <span class="ot">=</span> summarized<span class="sc">$</span>SD <span class="sc">/</span> <span class="fu">sqrt</span>(summarized<span class="sc">$</span>N)  </span>
<span id="cb185-504"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-504" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb185-505"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-505" aria-hidden="true" tabindex="-1"></a>    <span class="co">#http://www.sthda.com/english/wiki/ggplot2-barplots-quick-start-guide-r-software-and-data-visualization</span></span>
<span id="cb185-506"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-506" aria-hidden="true" tabindex="-1"></a>    p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(summarized, <span class="fu">aes</span>(<span class="at">x=</span>Genes, <span class="at">y=</span>Mean,<span class="at">fill=</span>Samples) ) <span class="sc">+</span> <span class="co"># data &amp; aesthetic mapping</span></span>
<span id="cb185-507"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-507" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">position=</span><span class="fu">position_dodge</span>()) <span class="sc">+</span> <span class="co"># bars represent average</span></span>
<span id="cb185-508"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-508" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>Mean<span class="sc">-</span>SE, <span class="at">ymax=</span>Mean<span class="sc">+</span>SE), <span class="at">width=</span><span class="fl">0.2</span>,<span class="at">position=</span><span class="fu">position_dodge</span>(.<span class="dv">9</span>)) <span class="sc">+</span></span>
<span id="cb185-509"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-509" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;Expression Level&quot;</span>) </span>
<span id="cb185-510"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-510" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_useSD <span class="sc">==</span> <span class="dv">1</span>) { </span>
<span id="cb185-511"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-511" aria-hidden="true" tabindex="-1"></a>    p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(summarized, <span class="fu">aes</span>(<span class="at">x=</span>Genes, <span class="at">y=</span>Mean,<span class="at">fill=</span>Samples) ) <span class="sc">+</span> <span class="co"># data &amp; aesthetic mapping</span></span>
<span id="cb185-512"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-512" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">position=</span><span class="fu">position_dodge</span>()) <span class="sc">+</span> <span class="co"># bars represent average</span></span>
<span id="cb185-513"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-513" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>Mean<span class="sc">-</span>SD, <span class="at">ymax=</span>Mean<span class="sc">+</span>SD), <span class="at">width=</span><span class="fl">0.2</span>,<span class="at">position=</span><span class="fu">position_dodge</span>(.<span class="dv">9</span>)) <span class="sc">+</span></span>
<span id="cb185-514"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-514" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;Expression Level&quot;</span>) </span>
<span id="cb185-515"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-515" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb185-516"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-516" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb185-517"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-517" aria-hidden="true" tabindex="-1"></a>    p2 <span class="ot">&lt;-</span> p2 <span class="sc">+</span>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>,<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="co"># theme(aspect.ratio=1) +</span></span>
<span id="cb185-518"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-518" aria-hidden="true" tabindex="-1"></a>     <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">size =</span> <span class="dv">16</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb185-519"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-519" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.text.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb185-520"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-520" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb185-521"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-521" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>) ) <span class="sc">+</span></span>
<span id="cb185-522"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-522" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span>
<span id="cb185-523"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-523" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb185-524"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-524" aria-hidden="true" tabindex="-1"></a>    p2</span>
<span id="cb185-525"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb185-525" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="hierarchical-clustering" class="section level2" number="6.5">
<h2><span class="header-section-number">6.5</span> Hierarchical clustering</h2>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb186-1"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-1" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb186-2"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Hierarchical clustering</span></span>
<span id="cb186-3"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-3" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb186-4"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-5"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Functions for hierarchical clustering</span></span>
<span id="cb186-6"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-6" aria-hidden="true" tabindex="-1"></a>hclust2 <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">method=</span><span class="st">&quot;average&quot;</span>, ...)  <span class="co"># average linkage</span></span>
<span id="cb186-7"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(x, <span class="at">method=</span>method, ...)</span>
<span id="cb186-8"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-8" aria-hidden="true" tabindex="-1"></a>hclust.ward.D <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">method=</span><span class="st">&quot;ward.D&quot;</span>, ...)  <span class="co"># average linkage</span></span>
<span id="cb186-9"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(x, <span class="at">method=</span>method, ...)</span>
<span id="cb186-10"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-10" aria-hidden="true" tabindex="-1"></a>hclust.ward.D2 <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">method=</span><span class="st">&quot;ward.D2&quot;</span>, ...)  <span class="co"># average linkage</span></span>
<span id="cb186-11"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(x, <span class="at">method=</span>method, ...)</span>
<span id="cb186-12"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-12" aria-hidden="true" tabindex="-1"></a>hclust.single <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">method=</span><span class="st">&quot;single&quot;</span>, ...)  <span class="co"># average linkage</span></span>
<span id="cb186-13"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(x, <span class="at">method=</span>method, ...)</span>
<span id="cb186-14"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-14" aria-hidden="true" tabindex="-1"></a>hclust.mcquitty <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">method=</span><span class="st">&quot;mcquitty&quot;</span>, ...)  <span class="co"># average linkage</span></span>
<span id="cb186-15"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(x, <span class="at">method=</span>method, ...)</span>
<span id="cb186-16"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-16" aria-hidden="true" tabindex="-1"></a>hclust.median <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">method=</span><span class="st">&quot;median&quot;</span>, ...)  <span class="co"># average linkage</span></span>
<span id="cb186-17"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(x, <span class="at">method=</span>method, ...)</span>
<span id="cb186-18"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-18" aria-hidden="true" tabindex="-1"></a>hclust.centroid <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">method=</span><span class="st">&quot;centroid&quot;</span>, ...)  <span class="co"># average linkage</span></span>
<span id="cb186-19"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(x, <span class="at">method=</span>method, ...)</span>
<span id="cb186-20"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb186-21"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-21" aria-hidden="true" tabindex="-1"></a>hclustFuns <span class="ot">&lt;-</span> <span class="fu">list</span>( <span class="at">averge =</span> hclust2, <span class="at">complete=</span>hclust, <span class="at">single=</span>hclust.single,</span>
<span id="cb186-22"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">median=</span>hclust.median, <span class="at">centroid=</span>hclust.centroid, <span class="at">mcquitty=</span>hclust.mcquitty)</span>
<span id="cb186-23"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-23" aria-hidden="true" tabindex="-1"></a>hclustChoices <span class="ot">=</span> <span class="fu">setNames</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(hclustFuns),<span class="fu">names</span>(hclustFuns)) <span class="co"># for pull down menu</span></span>
<span id="cb186-24"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-25"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-25" aria-hidden="true" tabindex="-1"></a>dist2 <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ...)   <span class="co"># distance function = 1-PCC (Pearson&#39;s correlation coefficient)</span></span>
<span id="cb186-26"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span><span class="fu">cor</span>(<span class="fu">t</span>(x), <span class="at">method=</span><span class="st">&quot;pearson&quot;</span>))</span>
<span id="cb186-27"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb186-28"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-28" aria-hidden="true" tabindex="-1"></a>dist3 <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ...)   <span class="co"># distance function = 1-abs(PCC) (Pearson&#39;s correlation coefficient)</span></span>
<span id="cb186-29"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span><span class="fu">abs</span>(<span class="fu">cor</span>(<span class="fu">t</span>(x), <span class="at">method=</span><span class="st">&quot;pearson&quot;</span>)))  </span>
<span id="cb186-30"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb186-31"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-31" aria-hidden="true" tabindex="-1"></a> distFuns <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">Correlation=</span>dist2, <span class="at">Euclidean=</span>dist,<span class="at">AbsolutePCC=</span>dist3) </span>
<span id="cb186-32"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># hierarchical clustering tree ----------------------------------------</span></span>
<span id="cb186-33"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-34"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-34" aria-hidden="true" tabindex="-1"></a>hmcols <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&quot;#D73027&quot;</span>, <span class="st">&quot;#FC8D59&quot;</span>, <span class="st">&quot;#FEE090&quot;</span>, <span class="st">&quot;#FFFFBF&quot;</span>,</span>
<span id="cb186-35"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-35" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;#E0F3F8&quot;</span>, <span class="st">&quot;#91BFDB&quot;</span>, <span class="st">&quot;#4575B4&quot;</span>)))(<span class="dv">75</span>)</span>
<span id="cb186-36"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-36" aria-hidden="true" tabindex="-1"></a>heatColors <span class="ot">=</span> <span class="fu">rbind</span>(      <span class="fu">greenred</span>(<span class="dv">75</span>),     <span class="fu">bluered</span>(<span class="dv">75</span>),     <span class="fu">colorpanel</span>(<span class="dv">75</span>,<span class="st">&quot;green&quot;</span>,<span class="st">&quot;black&quot;</span>,<span class="st">&quot;magenta&quot;</span>),<span class="fu">colorpanel</span>(<span class="dv">75</span>,<span class="st">&quot;blue&quot;</span>,<span class="st">&quot;yellow&quot;</span>,<span class="st">&quot;red&quot;</span>),hmcols )</span>
<span id="cb186-37"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-37" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heatColors) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Green-Black-Red&quot;</span>,<span class="st">&quot;Blue-White-Red&quot;</span>,<span class="st">&quot;Green-Black-Magenta&quot;</span>,<span class="st">&quot;Blue-Yellow-Red&quot;</span>,<span class="st">&quot;Blue-white-brown&quot;</span>)</span>
<span id="cb186-38"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-38" aria-hidden="true" tabindex="-1"></a>colorChoices <span class="ot">=</span> <span class="fu">setNames</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(heatColors)[<span class="dv">1</span>],<span class="fu">rownames</span>(heatColors)) <span class="co"># for pull down menu</span></span>
<span id="cb186-39"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-39" aria-hidden="true" tabindex="-1"></a><span class="co"># adding sample legends to heatmap; this is for the main heatmap</span></span>
<span id="cb186-40"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-40" aria-hidden="true" tabindex="-1"></a><span class="co"># https://stackoverflow.com/questions/3932038/plot-a-legend-outside-of-the-plotting-area-in-base-graphics</span></span>
<span id="cb186-41"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-41" aria-hidden="true" tabindex="-1"></a>add_legend <span class="ot">&lt;-</span> <span class="cf">function</span>(...) {</span>
<span id="cb186-42"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-42" aria-hidden="true" tabindex="-1"></a>  opar <span class="ot">&lt;-</span> <span class="fu">par</span>(<span class="at">fig=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>), <span class="at">oma=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), </span>
<span id="cb186-43"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">mar=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">6</span>), <span class="at">new=</span><span class="cn">TRUE</span>)</span>
<span id="cb186-44"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">on.exit</span>(<span class="fu">par</span>(opar))</span>
<span id="cb186-45"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="at">type=</span><span class="st">&#39;n&#39;</span>, <span class="at">bty=</span><span class="st">&#39;n&#39;</span>, <span class="at">xaxt=</span><span class="st">&#39;n&#39;</span>, <span class="at">yaxt=</span><span class="st">&#39;n&#39;</span>)</span>
<span id="cb186-46"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">legend</span>(...)</span>
<span id="cb186-47"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-47" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb186-48"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-49"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-49" aria-hidden="true" tabindex="-1"></a>correlationMatrix <span class="ot">&lt;-</span> <span class="cf">function</span>( ) {</span>
<span id="cb186-50"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-50" aria-hidden="true" tabindex="-1"></a>    <span class="co"># heatmap of correlation matrix</span></span>
<span id="cb186-51"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-51" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> readData.out<span class="sc">$</span>data</span>
<span id="cb186-52"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-52" aria-hidden="true" tabindex="-1"></a>    maxGene <span class="ot">&lt;-</span> <span class="fu">apply</span>(x,<span class="dv">1</span>,max)</span>
<span id="cb186-53"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-53" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> x[<span class="fu">which</span>(maxGene <span class="sc">&gt;</span> <span class="fu">quantile</span>(maxGene)[<span class="dv">1</span>] ) ,] <span class="co"># remove bottom 25% lowly expressed genes, which inflate the PPC</span></span>
<span id="cb186-54"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-54" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb186-55"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-55" aria-hidden="true" tabindex="-1"></a>       melted_cormat <span class="ot">&lt;-</span> <span class="fu">melt</span>(<span class="fu">round</span>(<span class="fu">cor</span>(x),<span class="dv">2</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb186-56"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-56" aria-hidden="true" tabindex="-1"></a>    <span class="co"># melted_cormat &lt;- melted_cormat[which(melted_cormat[,1] != melted_cormat[,2] ) , ]</span></span>
<span id="cb186-57"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-57" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Create a ggheatmap</span></span>
<span id="cb186-58"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-58" aria-hidden="true" tabindex="-1"></a>        ggheatmap <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(melted_cormat, <span class="fu">aes</span>(Var2, Var1, <span class="at">fill =</span> value))<span class="sc">+</span></span>
<span id="cb186-59"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-59" aria-hidden="true" tabindex="-1"></a>         <span class="fu">geom_tile</span>(<span class="at">color =</span> <span class="st">&quot;white&quot;</span>)<span class="sc">+</span></span>
<span id="cb186-60"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-60" aria-hidden="true" tabindex="-1"></a>         <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">&quot;green&quot;</span>, <span class="at">high =</span> <span class="st">&quot;red&quot;</span>,  <span class="at">mid =</span> <span class="st">&quot;white&quot;</span>, </span>
<span id="cb186-61"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-61" aria-hidden="true" tabindex="-1"></a>            <span class="at">space =</span> <span class="st">&quot;Lab&quot;</span>,  <span class="at">limit =</span> <span class="fu">c</span>(<span class="fu">min</span>(melted_cormat[,<span class="dv">3</span>]) ,<span class="fu">max</span>(melted_cormat[,<span class="dv">3</span>])), <span class="at">midpoint =</span> <span class="fu">median</span>(melted_cormat[,<span class="dv">3</span>]),</span>
<span id="cb186-62"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-62" aria-hidden="true" tabindex="-1"></a>            <span class="at">name=</span><span class="st">&quot;Pearson</span><span class="sc">\n</span><span class="st">Correlation&quot;</span></span>
<span id="cb186-63"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-63" aria-hidden="true" tabindex="-1"></a>          ) <span class="sc">+</span></span>
<span id="cb186-64"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-64" aria-hidden="true" tabindex="-1"></a>         <span class="fu">theme_minimal</span>()<span class="sc">+</span> <span class="co"># minimal theme</span></span>
<span id="cb186-65"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-65" aria-hidden="true" tabindex="-1"></a>         <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, </span>
<span id="cb186-66"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-66" aria-hidden="true" tabindex="-1"></a>            <span class="at">size =</span> <span class="dv">15</span>, <span class="at">hjust =</span> <span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb186-67"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-67" aria-hidden="true" tabindex="-1"></a>         <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>( </span>
<span id="cb186-68"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-68" aria-hidden="true" tabindex="-1"></a>            <span class="at">size =</span> <span class="dv">15</span>))<span class="sc">+</span></span>
<span id="cb186-69"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-69" aria-hidden="true" tabindex="-1"></a>         <span class="fu">coord_fixed</span>()<span class="sc">+</span></span>
<span id="cb186-70"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-70" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme</span>(</span>
<span id="cb186-71"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-71" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb186-72"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-72" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb186-73"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-73" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb186-74"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-74" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb186-75"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-75" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb186-76"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-76" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb186-77"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-77" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">0</span>),</span>
<span id="cb186-78"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-78" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="fl">0.7</span>),</span>
<span id="cb186-79"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-79" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.direction =</span> <span class="st">&quot;horizontal&quot;</span>)<span class="sc">+</span></span>
<span id="cb186-80"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-80" aria-hidden="true" tabindex="-1"></a>         <span class="fu">guides</span>(<span class="at">fill =</span> <span class="cn">FALSE</span>) <span class="co"># + ggtitle(&quot;Pearson&#39;s Correlation Coefficient (all genes)&quot;)</span></span>
<span id="cb186-81"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-81" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb186-82"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-82" aria-hidden="true" tabindex="-1"></a>         <span class="cf">if</span>(input_labelPCC <span class="sc">&amp;&amp;</span> <span class="fu">ncol</span>(x)<span class="sc">&lt;</span><span class="dv">30</span>)</span>
<span id="cb186-83"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-83" aria-hidden="true" tabindex="-1"></a>                ggheatmap <span class="ot">&lt;-</span> ggheatmap <span class="sc">+</span>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(Var2, Var1, <span class="at">label =</span> value), <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">4</span>)</span>
<span id="cb186-84"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-85"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-85" aria-hidden="true" tabindex="-1"></a>        ggheatmap</span>
<span id="cb186-86"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-86" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb186-87"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-87" aria-hidden="true" tabindex="-1"></a>}        </span>
<span id="cb186-88"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-89"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-89" aria-hidden="true" tabindex="-1"></a>staticHeatmap <span class="ot">&lt;-</span> <span class="cf">function</span> () { </span>
<span id="cb186-90"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-90" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> readData.out<span class="sc">$</span>data   <span class="co"># x = read.csv(&quot;expression1.csv&quot;)</span></span>
<span id="cb186-91"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-92"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-92" aria-hidden="true" tabindex="-1"></a>    n<span class="ot">=</span>input_nGenes</span>
<span id="cb186-93"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-93" aria-hidden="true" tabindex="-1"></a>    <span class="co">#if(n&gt;6000) n = 6000 # max</span></span>
<span id="cb186-94"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-94" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(n<span class="sc">&gt;</span><span class="fu">dim</span>(x)[<span class="dv">1</span>]) n <span class="ot">=</span> <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="co"># max as data</span></span>
<span id="cb186-95"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-95" aria-hidden="true" tabindex="-1"></a>    <span class="co"># this will cutoff very large values, which could skew the color </span></span>
<span id="cb186-96"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-96" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_geneCentering)</span>
<span id="cb186-97"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-97" aria-hidden="true" tabindex="-1"></a>        x<span class="ot">=</span><span class="fu">as.matrix</span>(x[<span class="dv">1</span><span class="sc">:</span>n,])<span class="sc">-</span><span class="fu">apply</span>(x[<span class="dv">1</span><span class="sc">:</span>n,],<span class="dv">1</span>,mean)</span>
<span id="cb186-98"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-98" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb186-99"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-99" aria-hidden="true" tabindex="-1"></a>    <span class="co"># standardize by gene</span></span>
<span id="cb186-100"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-100" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_geneNormalize) </span>
<span id="cb186-101"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-101" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> x <span class="sc">/</span> <span class="fu">apply</span>(x,<span class="dv">1</span>,sd)</span>
<span id="cb186-102"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-102" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb186-103"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-103" aria-hidden="true" tabindex="-1"></a>    <span class="co"># row centering and normalize</span></span>
<span id="cb186-104"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-104" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">scale</span>(x, <span class="at">center =</span> input_sampleCentering, <span class="at">scale =</span> input_sampleNormalize) </span>
<span id="cb186-105"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-106"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-106" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb186-107"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-107" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb186-108"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-108" aria-hidden="true" tabindex="-1"></a>    cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">+</span> input_heatmapCutoff <span class="sc">*</span> <span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb186-109"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-109" aria-hidden="true" tabindex="-1"></a>    x[x<span class="sc">&gt;</span>cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb186-110"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-110" aria-hidden="true" tabindex="-1"></a>    cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">-</span> input_heatmapCutoff <span class="sc">*</span><span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb186-111"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-111" aria-hidden="true" tabindex="-1"></a>    x[x<span class="sc">&lt;</span> cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb186-112"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-112" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb186-113"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-113" aria-hidden="true" tabindex="-1"></a>    groups <span class="ot">=</span> <span class="fu">detectGroups</span>(<span class="fu">colnames</span>(x) )</span>
<span id="cb186-114"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-114" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if sample info file is uploaded us that info:</span></span>
<span id="cb186-115"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-116"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-116" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(readSampleInfo.out) <span class="sc">&amp;&amp;</span>  <span class="sc">!</span><span class="fu">is.null</span>(input_selectFactorsHeatmap) ) { </span>
<span id="cb186-117"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-117" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(input_selectFactorsHeatmap <span class="sc">==</span> <span class="st">&quot;Sample_Name&quot;</span> )</span>
<span id="cb186-118"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-118" aria-hidden="true" tabindex="-1"></a>            groups <span class="ot">=</span> <span class="fu">detectGroups</span>(<span class="fu">colnames</span>(x) ) <span class="cf">else</span> </span>
<span id="cb186-119"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-119" aria-hidden="true" tabindex="-1"></a>            {   ix <span class="ot">=</span> <span class="fu">match</span>(input_selectFactorsHeatmap, <span class="fu">colnames</span>(readSampleInfo.out ) ) </span>
<span id="cb186-120"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-120" aria-hidden="true" tabindex="-1"></a>                groups <span class="ot">=</span> readSampleInfo.out[,ix]</span>
<span id="cb186-121"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-121" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb186-122"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-122" aria-hidden="true" tabindex="-1"></a>    }   </span>
<span id="cb186-123"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-123" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb186-124"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-124" aria-hidden="true" tabindex="-1"></a>    groups.colors <span class="ot">=</span> <span class="fu">rainbow</span>(<span class="fu">length</span>(<span class="fu">unique</span>(groups) ) )</span>
<span id="cb186-125"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-126"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-126" aria-hidden="true" tabindex="-1"></a>    <span class="co">#http://stackoverflow.com/questions/15351575/moving-color-key-in-r-heatmap-2-function-of-gplots-package</span></span>
<span id="cb186-127"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-127" aria-hidden="true" tabindex="-1"></a>    lmat <span class="ot">=</span> <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">4</span>),<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">2</span>),<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">0</span>))</span>
<span id="cb186-128"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-128" aria-hidden="true" tabindex="-1"></a>    lwid <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">6</span>) <span class="co"># width of gene tree; width of heatmap</span></span>
<span id="cb186-129"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-129" aria-hidden="true" tabindex="-1"></a>    lhei <span class="ot">=</span> <span class="fu">c</span>(<span class="fl">1.5</span>,.<span class="dv">2</span>,<span class="dv">8</span>,<span class="fl">1.1</span>)</span>
<span id="cb186-130"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-130" aria-hidden="true" tabindex="-1"></a>    <span class="co">#layout matrix</span></span>
<span id="cb186-131"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-131" aria-hidden="true" tabindex="-1"></a>    <span class="co">#        [,1] [,2]</span></span>
<span id="cb186-132"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-132" aria-hidden="true" tabindex="-1"></a>    <span class="co">#   [1,]    0    4</span></span>
<span id="cb186-133"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-133" aria-hidden="true" tabindex="-1"></a>    <span class="co">#   [2,]    0    1</span></span>
<span id="cb186-134"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-134" aria-hidden="true" tabindex="-1"></a>    <span class="co">#   [3,]    3    2</span></span>
<span id="cb186-135"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-135" aria-hidden="true" tabindex="-1"></a>    <span class="co">#   [4,]    5    0</span></span>
<span id="cb186-136"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-136" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 4--&gt; column tree; 1--&gt; column color bar; 2--&gt; heatmap; 3-&gt; row tree; 5--&gt; color key.</span></span>
<span id="cb186-137"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-137" aria-hidden="true" tabindex="-1"></a>    <span class="co"># height of 4 rows is specified by lhei; width of columns is given by lwid</span></span>
<span id="cb186-138"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-139"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-140"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-140" aria-hidden="true" tabindex="-1"></a>    <span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">12</span>, <span class="dv">4</span>, <span class="fl">1.4</span>, <span class="fl">0.2</span>))</span>
<span id="cb186-141"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-142"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-143"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-143" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( n<span class="sc">&gt;</span><span class="dv">110</span>) </span>
<span id="cb186-144"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-144" aria-hidden="true" tabindex="-1"></a>    <span class="fu">heatmap.2</span>(x, <span class="at">distfun =</span> distFuns[[<span class="fu">as.integer</span>(input_distFunctions)]]</span>
<span id="cb186-145"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-145" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">hclustfun=</span>hclustFuns[[<span class="fu">as.integer</span>(input_hclustFunctions)]]</span>
<span id="cb186-146"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-146" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">Colv=</span><span class="sc">!</span>input_noSampleClustering</span>
<span id="cb186-147"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-147" aria-hidden="true" tabindex="-1"></a>        <span class="co">#col=colorpanel(75,&quot;green&quot;,&quot;black&quot;,&quot;magenta&quot;)  ,</span></span>
<span id="cb186-148"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-148" aria-hidden="true" tabindex="-1"></a>        <span class="co">#col=bluered(75),</span></span>
<span id="cb186-149"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-149" aria-hidden="true" tabindex="-1"></a>        <span class="co">#col=greenred(75), </span></span>
<span id="cb186-150"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-150" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">col=</span> heatColors[<span class="fu">as.integer</span>(input_heatColors1),]</span>
<span id="cb186-151"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-151" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">density.info=</span><span class="st">&quot;none&quot;</span>, <span class="at">trace=</span><span class="st">&quot;none&quot;</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">keysize=</span>.<span class="dv">5</span></span>
<span id="cb186-152"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-152" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">key=</span>T, <span class="at">symkey=</span>F</span>
<span id="cb186-153"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-153" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">ColSideColors=</span>groups.colors[ <span class="fu">as.factor</span>(groups)]</span>
<span id="cb186-154"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-154" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">labRow=</span><span class="st">&quot;&quot;</span></span>
<span id="cb186-155"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-155" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">margins=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">2</span>)</span>
<span id="cb186-156"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-156" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">srtCol=</span><span class="dv">45</span></span>
<span id="cb186-157"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-157" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">cexCol=</span><span class="dv">2</span>  <span class="co"># size of font for sample names</span></span>
<span id="cb186-158"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-158" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">lmat =</span> lmat, <span class="at">lwid =</span> lwid, <span class="at">lhei =</span> lhei</span>
<span id="cb186-159"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-159" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb186-160"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-161"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-161" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( n<span class="sc">&lt;=</span><span class="dv">110</span>) </span>
<span id="cb186-162"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-162" aria-hidden="true" tabindex="-1"></a>    <span class="fu">heatmap.2</span>(x, <span class="at">distfun =</span>  distFuns[[<span class="fu">as.integer</span>(input_distFunctions)]]</span>
<span id="cb186-163"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-163" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">hclustfun=</span>hclustFuns[[<span class="fu">as.integer</span>(input_hclustFunctions)]]</span>
<span id="cb186-164"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-164" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">Colv=</span><span class="sc">!</span>input_noSampleClustering     </span>
<span id="cb186-165"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-165" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">col=</span> heatColors[<span class="fu">as.integer</span>(input_heatColors1),], <span class="at">density.info=</span><span class="st">&quot;none&quot;</span>, <span class="at">trace=</span><span class="st">&quot;none&quot;</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">keysize=</span>.<span class="dv">5</span></span>
<span id="cb186-166"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-166" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">key=</span>T, <span class="at">symkey=</span>F</span>
<span id="cb186-167"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-167" aria-hidden="true" tabindex="-1"></a>        <span class="co">#,labRow=labRow</span></span>
<span id="cb186-168"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-168" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">ColSideColors=</span>groups.colors[ <span class="fu">as.factor</span>(groups)]</span>
<span id="cb186-169"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-169" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">margins=</span><span class="fu">c</span>(<span class="dv">18</span>,<span class="dv">12</span>)</span>
<span id="cb186-170"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-170" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">cexRow=</span><span class="dv">1</span></span>
<span id="cb186-171"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-171" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">srtCol=</span><span class="dv">45</span></span>
<span id="cb186-172"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-172" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">cexCol=</span><span class="fl">1.8</span>  <span class="co"># size of font for sample names</span></span>
<span id="cb186-173"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-173" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">lmat =</span> lmat, <span class="at">lwid =</span> lwid, <span class="at">lhei =</span> lhei</span>
<span id="cb186-174"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-174" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb186-175"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-175" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb186-176"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-176" aria-hidden="true" tabindex="-1"></a>    <span class="fu">par</span>(<span class="at">lend =</span> <span class="dv">1</span>)           <span class="co"># square line ends for the color legend</span></span>
<span id="cb186-177"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-177" aria-hidden="true" tabindex="-1"></a>    <span class="fu">add_legend</span>(<span class="st">&quot;topleft&quot;</span>,</span>
<span id="cb186-178"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-178" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend =</span> <span class="fu">unique</span>(groups), <span class="co"># category labels</span></span>
<span id="cb186-179"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-179" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> groups.colors[ <span class="fu">unique</span>(<span class="fu">as.factor</span>(groups))],  <span class="co"># color key</span></span>
<span id="cb186-180"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-180" aria-hidden="true" tabindex="-1"></a>        <span class="at">lty=</span> <span class="dv">1</span>,             <span class="co"># line style</span></span>
<span id="cb186-181"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-181" aria-hidden="true" tabindex="-1"></a>        <span class="at">lwd =</span> <span class="dv">10</span>            <span class="co"># line width</span></span>
<span id="cb186-182"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-182" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb186-183"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-183" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb186-184"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-184" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb186-185"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-186"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-187"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-187" aria-hidden="true" tabindex="-1"></a>    <span class="co">#Interactive heatmap via Plotly plot ----------------------------------------------</span></span>
<span id="cb186-188"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-188" aria-hidden="true" tabindex="-1"></a>    <span class="co"># interactive heatmap with plotly</span></span>
<span id="cb186-189"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-189" aria-hidden="true" tabindex="-1"></a>heatmapPlotly <span class="ot">&lt;-</span> <span class="cf">function</span> () {  </span>
<span id="cb186-190"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-190" aria-hidden="true" tabindex="-1"></a>    input_nGenesPlotly<span class="ot">=</span> <span class="dv">50</span></span>
<span id="cb186-191"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-191" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb186-192"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-192" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb186-193"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-194"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-194" aria-hidden="true" tabindex="-1"></a>       x <span class="ot">&lt;-</span> convertedData.out</span>
<span id="cb186-195"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-196"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-196" aria-hidden="true" tabindex="-1"></a>    n<span class="ot">=</span>input_nGenesPlotly</span>
<span id="cb186-197"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-197" aria-hidden="true" tabindex="-1"></a>    <span class="co">#if(n&gt;6000) n = 6000 # max</span></span>
<span id="cb186-198"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-198" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(n<span class="sc">&gt;</span><span class="fu">dim</span>(x)[<span class="dv">1</span>]) n <span class="ot">=</span> <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="co"># max as data</span></span>
<span id="cb186-199"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-200"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-200" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_geneCentering)</span>
<span id="cb186-201"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-201" aria-hidden="true" tabindex="-1"></a>        x<span class="ot">=</span><span class="fu">as.matrix</span>(x[<span class="dv">1</span><span class="sc">:</span>n,])<span class="sc">-</span><span class="fu">apply</span>(x[<span class="dv">1</span><span class="sc">:</span>n,],<span class="dv">1</span>,mean)</span>
<span id="cb186-202"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-202" aria-hidden="true" tabindex="-1"></a>    <span class="co"># standardize by gene</span></span>
<span id="cb186-203"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-203" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_geneNormalize) </span>
<span id="cb186-204"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-204" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> x <span class="sc">/</span> <span class="fu">apply</span>(x,<span class="dv">1</span>,sd)</span>
<span id="cb186-205"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-205" aria-hidden="true" tabindex="-1"></a>    <span class="co"># row centering and normalize</span></span>
<span id="cb186-206"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-206" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">scale</span>(x, <span class="at">center =</span> input_sampleCentering, <span class="at">scale =</span> input_sampleNormalize) </span>
<span id="cb186-207"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-208"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-208" aria-hidden="true" tabindex="-1"></a>    cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">+</span> <span class="dv">3</span><span class="sc">*</span><span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb186-209"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-209" aria-hidden="true" tabindex="-1"></a>    x[x<span class="sc">&gt;</span>cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb186-210"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-210" aria-hidden="true" tabindex="-1"></a>    cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">-</span> <span class="dv">3</span><span class="sc">*</span><span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb186-211"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-211" aria-hidden="true" tabindex="-1"></a>    x[x<span class="sc">&lt;</span> cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb186-212"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-213"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-213" aria-hidden="true" tabindex="-1"></a>    <span class="co"># adding gene symbol</span></span>
<span id="cb186-214"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-214" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">&lt;-</span> <span class="fu">match</span>( <span class="fu">rownames</span>(x), allGeneInfo.out[,<span class="dv">1</span>])</span>
<span id="cb186-215"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-215" aria-hidden="true" tabindex="-1"></a>    geneSymbols <span class="ot">&lt;-</span> <span class="fu">as.character</span>( allGeneInfo.out<span class="sc">$</span>symbol)[ix]</span>
<span id="cb186-216"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-216" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if missing or duplicated, use Ensembl ID</span></span>
<span id="cb186-217"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-217" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">nchar</span>( geneSymbols) <span class="sc">&lt;=</span><span class="dv">1</span> <span class="sc">|</span> <span class="fu">duplicated</span>(geneSymbols ) <span class="sc">|</span> <span class="fu">is.null</span>(geneSymbols)  <span class="sc">|</span> <span class="fu">is.na</span>(geneSymbols)    );   geneSymbols[ ix ] <span class="ot">&lt;-</span> <span class="fu">rownames</span>(x)[ix]</span>
<span id="cb186-218"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-218" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>( x) <span class="ot">=</span> geneSymbols;</span>
<span id="cb186-219"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-220"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-221"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-221" aria-hidden="true" tabindex="-1"></a>    clust <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb186-222"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-222" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dist2</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb186-223"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-223" aria-hidden="true" tabindex="-1"></a>      <span class="fu">hclust2</span>()</span>
<span id="cb186-224"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-225"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-225" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get order</span></span>
<span id="cb186-226"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-226" aria-hidden="true" tabindex="-1"></a>    ord <span class="ot">&lt;-</span> clust<span class="sc">$</span>order</span>
<span id="cb186-227"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-228"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-228" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Re-arrange based on order</span></span>
<span id="cb186-229"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-229" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">t</span>( x[ord,] )<span class="sc">%&gt;%</span></span>
<span id="cb186-230"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-230" aria-hidden="true" tabindex="-1"></a>       <span class="fu">melt</span>()</span>
<span id="cb186-231"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-231" aria-hidden="true" tabindex="-1"></a>       </span>
<span id="cb186-232"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-232" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(df)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;X&quot;</span>,<span class="st">&quot;Y&quot;</span>)</span>
<span id="cb186-233"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-233" aria-hidden="true" tabindex="-1"></a>    colorNames <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(<span class="fu">tolower</span>(<span class="fu">rownames</span>(heatColors)[ <span class="fu">as.integer</span>(input_heatColors1)   ]),<span class="st">&quot;-&quot;</span> ) )</span>
<span id="cb186-234"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-234" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb186-235"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-235" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ggplot</span>(<span class="fu">aes</span>(X, Y, <span class="at">fill =</span> value)) <span class="sc">+</span> </span>
<span id="cb186-236"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-236" aria-hidden="true" tabindex="-1"></a>           <span class="fu">geom_tile</span>()<span class="sc">+</span> <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> colorNames[<span class="dv">1</span>], <span class="at">mid =</span> colorNames[<span class="dv">2</span>],<span class="at">high =</span> colorNames[<span class="dv">3</span>]) <span class="sc">+</span></span>
<span id="cb186-237"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-237" aria-hidden="true" tabindex="-1"></a>           <span class="fu">theme</span>(<span class="at">axis.title.y=</span><span class="fu">element_blank</span>(),   <span class="co"># remove y labels</span></span>
<span id="cb186-238"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-238" aria-hidden="true" tabindex="-1"></a>           <span class="co"># axis.text.y=element_blank.out,  # keep gene names for zooming</span></span>
<span id="cb186-239"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-239" aria-hidden="true" tabindex="-1"></a>            <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb186-240"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-240" aria-hidden="true" tabindex="-1"></a>            <span class="at">axis.title.x=</span><span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb186-241"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-241" aria-hidden="true" tabindex="-1"></a>            <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>,<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb186-242"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-243"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb186-244"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-244" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplotly</span>(p) <span class="sc">%&gt;%</span> </span>
<span id="cb186-245"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-245" aria-hidden="true" tabindex="-1"></a>        <span class="fu">layout</span>(<span class="at">margin =</span> <span class="fu">list</span>(<span class="at">b =</span> <span class="dv">150</span>,<span class="at">l=</span><span class="dv">200</span>))  <span class="co"># prevent cutoff of sample names</span></span>
<span id="cb186-246"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb186-246" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="pca-1" class="section level2" number="6.6">
<h2><span class="header-section-number">6.6</span> PCA</h2>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-1" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb187-2"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-2" aria-hidden="true" tabindex="-1"></a><span class="co"># PCA</span></span>
<span id="cb187-3"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-3" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb187-4"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-5"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-5" aria-hidden="true" tabindex="-1"></a>PCAplot <span class="ot">&lt;-</span> <span class="cf">function</span>() {   <span class="co">#PCA</span></span>
<span id="cb187-6"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-6" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> convertedData.out;</span>
<span id="cb187-7"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-7" aria-hidden="true" tabindex="-1"></a>     pca.object <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(x))</span>
<span id="cb187-8"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-8" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb187-9"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-9" aria-hidden="true" tabindex="-1"></a>    pcaData <span class="ot">=</span> <span class="fu">as.data.frame</span>(pca.object<span class="sc">$</span>x[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]); pcaData <span class="ot">=</span> <span class="fu">cbind</span>(pcaData,<span class="fu">detectGroups</span>(<span class="fu">colnames</span>(x)) )</span>
<span id="cb187-10"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(pcaData) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;PC1&quot;</span>, <span class="st">&quot;PC2&quot;</span>, <span class="st">&quot;Sample_Name&quot;</span>)</span>
<span id="cb187-11"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-11" aria-hidden="true" tabindex="-1"></a>    percentVar<span class="ot">=</span><span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span><span class="fu">summary</span>(pca.object)<span class="sc">$</span>importance[<span class="dv">2</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],<span class="dv">0</span>)</span>
<span id="cb187-12"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(readSampleInfo.out)) { </span>
<span id="cb187-13"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-13" aria-hidden="true" tabindex="-1"></a>        p<span class="ot">=</span><span class="fu">ggplot</span>(pcaData, <span class="fu">aes</span>(PC1, PC2, <span class="at">color=</span>Sample_Name, <span class="at">shape =</span> Sample_Name)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>) </span>
<span id="cb187-14"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-14" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb187-15"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-15" aria-hidden="true" tabindex="-1"></a>        pcaData <span class="ot">=</span> <span class="fu">cbind</span>(pcaData,readSampleInfo.out )</span>
<span id="cb187-16"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-16" aria-hidden="true" tabindex="-1"></a>        p<span class="ot">=</span><span class="fu">ggplot</span>(pcaData, <span class="fu">aes_string</span>(<span class="st">&quot;PC1&quot;</span>, <span class="st">&quot;PC2&quot;</span>, <span class="at">color=</span>input_selectFactors,<span class="at">shape=</span>input_selectFactors2)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>) </span>
<span id="cb187-17"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-17" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb187-18"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-18" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb187-19"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-19" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC1: &quot;</span>,percentVar[<span class="dv">1</span>],<span class="st">&quot;% variance&quot;</span>)) </span>
<span id="cb187-20"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-20" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC2: &quot;</span>,percentVar[<span class="dv">2</span>],<span class="st">&quot;% variance&quot;</span>)) </span>
<span id="cb187-21"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-21" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">ggtitle</span>(<span class="st">&quot;Principal component analysis (PCA)&quot;</span>)<span class="sc">+</span><span class="fu">coord_fixed</span>(<span class="at">ratio=</span><span class="fl">1.0</span>)<span class="sc">+</span> </span>
<span id="cb187-22"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-22" aria-hidden="true" tabindex="-1"></a>     <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>,<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb187-23"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-23" aria-hidden="true" tabindex="-1"></a>     <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb187-24"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-24" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.text.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb187-25"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-25" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.x =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb187-26"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-26" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>) ) <span class="sc">+</span></span>
<span id="cb187-27"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span>
<span id="cb187-28"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-28" aria-hidden="true" tabindex="-1"></a>       <span class="fu">print</span>(p)</span>
<span id="cb187-29"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-29" aria-hidden="true" tabindex="-1"></a>       }</span>
<span id="cb187-30"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-30" aria-hidden="true" tabindex="-1"></a>    <span class="co"># variance chart</span></span>
<span id="cb187-31"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-31" aria-hidden="true" tabindex="-1"></a>    <span class="co"># plot(pca.object,type=&quot;bar&quot;, xlab=&quot;Principal Components&quot;, main =&quot;Variances explained&quot;)</span></span>
<span id="cb187-32"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-32" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb187-33"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-33" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb187-34"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-34" aria-hidden="true" tabindex="-1"></a>MDSplot <span class="ot">&lt;-</span> <span class="cf">function</span>() {  <span class="co"># MDS</span></span>
<span id="cb187-35"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-35" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> convertedData.out;</span>
<span id="cb187-36"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-36" aria-hidden="true" tabindex="-1"></a>     fit <span class="ot">=</span> <span class="fu">cmdscale</span>( <span class="fu">dist2</span>(<span class="fu">t</span>(x) ), <span class="at">eig=</span>T, <span class="at">k=</span><span class="dv">2</span>)</span>
<span id="cb187-37"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-37" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb187-38"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-38" aria-hidden="true" tabindex="-1"></a>    <span class="co"># par(pin=c(5,5))</span></span>
<span id="cb187-39"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-39" aria-hidden="true" tabindex="-1"></a>    pcaData <span class="ot">=</span> <span class="fu">as.data.frame</span>(fit<span class="sc">$</span>points[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]); pcaData <span class="ot">=</span> <span class="fu">cbind</span>(pcaData,<span class="fu">detectGroups</span>(<span class="fu">colnames</span>(x)) )</span>
<span id="cb187-40"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(pcaData) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;x1&quot;</span>, <span class="st">&quot;x2&quot;</span>, <span class="st">&quot;Sample_Name&quot;</span>)</span>
<span id="cb187-41"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-41" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb187-42"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-43"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-43" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(readSampleInfo.out)) { </span>
<span id="cb187-44"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-44" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span><span class="fu">ggplot</span>(pcaData, <span class="fu">aes</span>(x1, x2, <span class="at">color=</span>Sample_Name, <span class="at">shape =</span> Sample_Name)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>) </span>
<span id="cb187-45"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-45" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb187-46"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-46" aria-hidden="true" tabindex="-1"></a>        pcaData <span class="ot">=</span> <span class="fu">cbind</span>(pcaData,readSampleInfo.out )</span>
<span id="cb187-47"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-47" aria-hidden="true" tabindex="-1"></a>        p<span class="ot">=</span><span class="fu">ggplot</span>(pcaData, <span class="fu">aes_string</span>(<span class="st">&quot;x1&quot;</span>, <span class="st">&quot;x2&quot;</span>, <span class="at">color=</span>input_selectFactors,<span class="at">shape=</span>input_selectFactors2)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>) </span>
<span id="cb187-48"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-48" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb187-49"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-49" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">&quot;Dimension 1&quot;</span>) </span>
<span id="cb187-50"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-50" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">&quot;Dimension 2&quot;</span>) </span>
<span id="cb187-51"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-51" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">ggtitle</span>(<span class="st">&quot;Multidimensional scaling (MDS)&quot;</span>)<span class="sc">+</span> <span class="fu">coord_fixed</span>(<span class="at">ratio=</span><span class="fl">1.</span>)<span class="sc">+</span> </span>
<span id="cb187-52"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-52" aria-hidden="true" tabindex="-1"></a>     <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb187-53"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-53" aria-hidden="true" tabindex="-1"></a>         <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb187-54"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-54" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.text.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb187-55"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-55" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.x =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb187-56"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-56" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>) ) <span class="sc">+</span></span>
<span id="cb187-57"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span>
<span id="cb187-58"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-58" aria-hidden="true" tabindex="-1"></a>       <span class="fu">print</span>(p)</span>
<span id="cb187-59"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-59" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb187-60"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-60" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb187-61"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-62"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-62" aria-hidden="true" tabindex="-1"></a>tSNEplot <span class="ot">&lt;-</span> <span class="cf">function</span>() {  <span class="co"># t-SNE</span></span>
<span id="cb187-63"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-63" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> convertedData.out;</span>
<span id="cb187-64"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-64" aria-hidden="true" tabindex="-1"></a>     <span class="fu">library</span>(Rtsne,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb187-65"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-65" aria-hidden="true" tabindex="-1"></a>     <span class="fu">set.seed</span>(input_tsneSeed2)</span>
<span id="cb187-66"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-66" aria-hidden="true" tabindex="-1"></a>     tsne <span class="ot">&lt;-</span> <span class="fu">Rtsne</span>(<span class="fu">t</span>(x), <span class="at">dims =</span> <span class="dv">2</span>, <span class="at">perplexity=</span><span class="dv">1</span>, <span class="at">verbose=</span><span class="cn">FALSE</span>, <span class="at">max_iter =</span> <span class="dv">400</span>)</span>
<span id="cb187-67"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-68"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-68" aria-hidden="true" tabindex="-1"></a>    pcaData <span class="ot">=</span> <span class="fu">as.data.frame</span>(tsne<span class="sc">$</span>Y); pcaData <span class="ot">=</span> <span class="fu">cbind</span>(pcaData,<span class="fu">detectGroups</span>(<span class="fu">colnames</span>(x)) )</span>
<span id="cb187-69"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(pcaData) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;x1&quot;</span>, <span class="st">&quot;x2&quot;</span>, <span class="st">&quot;Sample_Name&quot;</span>)</span>
<span id="cb187-70"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-70" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb187-71"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-72"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-72" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(readSampleInfo.out)) { </span>
<span id="cb187-73"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-73" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span><span class="fu">ggplot</span>(pcaData, <span class="fu">aes</span>(x1, x2, <span class="at">color=</span>Sample_Name, <span class="at">shape =</span> Sample_Name)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>) </span>
<span id="cb187-74"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-74" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb187-75"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-75" aria-hidden="true" tabindex="-1"></a>        pcaData <span class="ot">=</span> <span class="fu">cbind</span>(pcaData,readSampleInfo.out )</span>
<span id="cb187-76"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-76" aria-hidden="true" tabindex="-1"></a>        p<span class="ot">=</span><span class="fu">ggplot</span>(pcaData, <span class="fu">aes_string</span>(<span class="st">&quot;x1&quot;</span>, <span class="st">&quot;x2&quot;</span>, <span class="at">color=</span>input_selectFactors,<span class="at">shape=</span>input_selectFactors2)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>) </span>
<span id="cb187-77"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-77" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb187-78"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-78" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">&quot;Dimension 1&quot;</span>) </span>
<span id="cb187-79"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-79" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">&quot;Dimension 2&quot;</span>) </span>
<span id="cb187-80"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-80" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">ggtitle</span>(<span class="st">&quot;t-SNE plot&quot;</span>)<span class="sc">+</span> <span class="fu">coord_fixed</span>(<span class="at">ratio=</span><span class="fl">1.</span>)<span class="sc">+</span> </span>
<span id="cb187-81"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-81" aria-hidden="true" tabindex="-1"></a>     <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb187-82"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-82" aria-hidden="true" tabindex="-1"></a>         <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb187-83"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-83" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.text.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb187-84"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-84" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.x =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb187-85"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-85" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>) ) <span class="sc">+</span></span>
<span id="cb187-86"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-86" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span>
<span id="cb187-87"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-87" aria-hidden="true" tabindex="-1"></a>       <span class="fu">print</span>(p)</span>
<span id="cb187-88"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-88" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb187-89"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-89" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb187-90"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-91"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-91" aria-hidden="true" tabindex="-1"></a> <span class="co"># read pathway data from local file, not used if using database</span></span>
<span id="cb187-92"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-92" aria-hidden="true" tabindex="-1"></a> <span class="co"># Read gene sets GMT file</span></span>
<span id="cb187-93"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-93" aria-hidden="true" tabindex="-1"></a><span class="co"># This functions cleans and converts to upper case</span></span>
<span id="cb187-94"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-95"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-95" aria-hidden="true" tabindex="-1"></a>readGMTRobust <span class="ot">&lt;-</span> <span class="cf">function</span> (file1) {   <span class="co"># size restriction</span></span>
<span id="cb187-96"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-96" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Read in the first file </span></span>
<span id="cb187-97"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-97" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> <span class="fu">scan</span>(file1, <span class="at">what=</span><span class="st">&quot;&quot;</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb187-98"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-98" aria-hidden="true" tabindex="-1"></a>        <span class="co"># x &lt;- gsub(&quot;\t\t.&quot;,&quot;&quot;,x)     # GMT files saved by Excel has a lot of empty cells &quot;\t\t\t\t&quot;   &quot;\t.&quot; means one or more tab</span></span>
<span id="cb187-99"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-99" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>,<span class="st">&quot;&quot;</span>,x)  <span class="co"># remove white space</span></span>
<span id="cb187-100"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-100" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> <span class="fu">toupper</span>(x)    <span class="co"># convert to upper case</span></span>
<span id="cb187-101"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-102"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-102" aria-hidden="true" tabindex="-1"></a>        <span class="co">#----Process the first file</span></span>
<span id="cb187-103"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-103" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Separate elements by one or more whitespace</span></span>
<span id="cb187-104"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-104" aria-hidden="true" tabindex="-1"></a>        y <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(x, <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb187-105"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-105" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Extract the first vector element and set it as the list element name</span></span>
<span id="cb187-106"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-106" aria-hidden="true" tabindex="-1"></a>        <span class="fu">names</span>(y) <span class="ot">&lt;-</span> <span class="fu">sapply</span>(y, <span class="st">`</span><span class="at">[[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb187-107"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-107" aria-hidden="true" tabindex="-1"></a>        <span class="co">#names(y) &lt;- sapply(y, function(x) x[[1]]) # same as above</span></span>
<span id="cb187-108"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-108" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Remove the first vector element from each list element</span></span>
<span id="cb187-109"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-109" aria-hidden="true" tabindex="-1"></a>        y <span class="ot">&lt;-</span> <span class="fu">lapply</span>(y, <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb187-110"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-110" aria-hidden="true" tabindex="-1"></a>        <span class="co">#y &lt;- lapply(y, function(x) x[-1]) # same as above</span></span>
<span id="cb187-111"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-111" aria-hidden="true" tabindex="-1"></a>        <span class="co"># remove duplicated elements</span></span>
<span id="cb187-112"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-112" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> ( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(y) )  y[[i]] <span class="ot">&lt;-</span> <span class="fu">cleanGeneSet</span>(y[[i]])</span>
<span id="cb187-113"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-113" aria-hidden="true" tabindex="-1"></a>        <span class="co"># check the distribution of the size of gene lists sapply(y, length) hold a vector of sizes</span></span>
<span id="cb187-114"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-114" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">max</span>( <span class="fu">sapply</span>(y,length) ) <span class="sc">&lt;</span><span class="dv">5</span>) <span class="fu">cat</span>(<span class="st">&quot;Warning! Gene sets have very small number of genes!</span><span class="sc">\n</span><span class="st"> Please double check format.&quot;</span>)</span>
<span id="cb187-115"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-115" aria-hidden="true" tabindex="-1"></a>        y <span class="ot">&lt;-</span> y[<span class="fu">which</span>(<span class="fu">sapply</span>(y,length) <span class="sc">&gt;</span> <span class="dv">1</span>)]  <span class="co"># gene sets smaller than 1 is ignored!!!</span></span>
<span id="cb187-116"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-117"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-117" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>(y)</span>
<span id="cb187-118"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-118" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb187-119"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-120"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-121"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-121" aria-hidden="true" tabindex="-1"></a>readGeneSets <span class="ot">&lt;-</span> <span class="cf">function</span> (fileName, convertedData, GO,selectOrg, myrange) {</span>
<span id="cb187-122"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-123"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-123" aria-hidden="true" tabindex="-1"></a>    pathway <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(sqlite,fileName)</span>
<span id="cb187-124"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-124" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb187-125"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-125" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(GO)  ) GO <span class="ot">&lt;-</span> <span class="st">&quot;GOBP&quot;</span>   <span class="co"># initial value not properly set; enforcing  </span></span>
<span id="cb187-126"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-127"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-127" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get Gene sets</span></span>
<span id="cb187-128"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-128" aria-hidden="true" tabindex="-1"></a>    querySet <span class="ot">=</span> <span class="fu">rownames</span>(convertedData)</span>
<span id="cb187-129"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-129" aria-hidden="true" tabindex="-1"></a>    sqlQuery <span class="ot">=</span> <span class="fu">paste</span>( <span class="st">&quot; select distinct gene,pathwayID from pathway where gene IN (&#39;&quot;</span>, <span class="fu">paste</span>(querySet,<span class="at">collapse=</span><span class="st">&quot;&#39;, &#39;&quot;</span>),<span class="st">&quot;&#39;)&quot;</span> ,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb187-130"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-130" aria-hidden="true" tabindex="-1"></a>    <span class="co"># cat(paste0(&quot;\n\nhere:&quot;,GO,&quot;There&quot;))</span></span>
<span id="cb187-131"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-132"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-132" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( GO <span class="sc">!=</span> <span class="st">&quot;All&quot;</span>) sqlQuery <span class="ot">=</span> <span class="fu">paste0</span>(sqlQuery, <span class="st">&quot; AND category =&#39;&quot;</span>,GO,<span class="st">&quot;&#39;&quot;</span>)</span>
<span id="cb187-133"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-133" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>( pathway, sqlQuery  )</span>
<span id="cb187-134"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-134" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">1</span>] <span class="sc">==</span><span class="dv">0</span>) {<span class="fu">return</span>(<span class="fu">list</span>( <span class="at">x=</span><span class="fu">as.data.frame</span>(<span class="st">&quot;No matching species or gene ID file!&quot;</span> )) )}</span>
<span id="cb187-135"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-135" aria-hidden="true" tabindex="-1"></a>    <span class="co"># list pathways and frequency of genes</span></span>
<span id="cb187-136"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-136" aria-hidden="true" tabindex="-1"></a>    pathwayIDs <span class="ot">=</span> <span class="fu">aggregate</span>( result<span class="sc">$</span>pathwayID, <span class="at">by   =</span> <span class="fu">list</span>(<span class="at">unique.values =</span> result<span class="sc">$</span>pathwayID), <span class="at">FUN =</span> length)</span>
<span id="cb187-137"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-137" aria-hidden="true" tabindex="-1"></a>    pathwayIDs <span class="ot">=</span> pathwayIDs[<span class="fu">which</span>(pathwayIDs[,<span class="dv">2</span>]<span class="sc">&gt;=</span> myrange[<span class="dv">1</span>] ),]</span>
<span id="cb187-138"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-138" aria-hidden="true" tabindex="-1"></a>    pathwayIDs <span class="ot">=</span> pathwayIDs[<span class="fu">which</span>( pathwayIDs[,<span class="dv">2</span>] <span class="sc">&lt;=</span> myrange[<span class="dv">2</span>] ),]</span>
<span id="cb187-139"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-139" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">dim</span>(pathwayIDs)[<span class="dv">1</span>] <span class="sc">==</span><span class="dv">0</span> ) geneSets <span class="ot">=</span> <span class="cn">NULL</span>;</span>
<span id="cb187-140"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-140" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb187-141"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-141" aria-hidden="true" tabindex="-1"></a>    <span class="co"># convert pathways into lists like those generated by readGMT</span></span>
<span id="cb187-142"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-142" aria-hidden="true" tabindex="-1"></a>    geneSets <span class="ot">=</span> <span class="fu">lapply</span>(pathwayIDs[,<span class="dv">1</span>], <span class="cf">function</span>(x)  result[<span class="fu">which</span>(result<span class="sc">$</span>pathwayID <span class="sc">==</span> x ),<span class="dv">1</span>]     )</span>
<span id="cb187-143"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-143" aria-hidden="true" tabindex="-1"></a>    pathwayInfo <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>( pathway, <span class="fu">paste</span>( <span class="st">&quot; select distinct id,Description from pathwayInfo where id IN (&#39;&quot;</span>, </span>
<span id="cb187-144"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-144" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">paste</span>(pathwayIDs[,<span class="dv">1</span>],<span class="at">collapse=</span><span class="st">&quot;&#39;, &#39;&quot;</span>),   <span class="st">&quot;&#39;) &quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>) )</span>
<span id="cb187-145"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-145" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span> <span class="fu">match</span>( pathwayIDs[,<span class="dv">1</span>], pathwayInfo[,<span class="dv">1</span>])</span>
<span id="cb187-146"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-146" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(geneSets) <span class="ot">&lt;-</span> pathwayInfo[ix,<span class="dv">2</span>]  </span>
<span id="cb187-147"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-147" aria-hidden="true" tabindex="-1"></a>    <span class="co">#geneSets &lt;- geneSets[ -which(duplicated(names(geneSets) ))] # remove geneSets with the same name</span></span>
<span id="cb187-148"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-148" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dbDisconnect</span>(pathway)</span>
<span id="cb187-149"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-149" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( geneSets )</span>
<span id="cb187-150"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-150" aria-hidden="true" tabindex="-1"></a>} </span>
<span id="cb187-151"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-152"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-153"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-153" aria-hidden="true" tabindex="-1"></a><span class="co"># Runs pathway analysis using PGSEA; this is copied and revised from PGSEA package</span></span>
<span id="cb187-154"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-154" aria-hidden="true" tabindex="-1"></a>myPGSEA  <span class="ot">&lt;-</span> <span class="cf">function</span> (exprs, cl, <span class="at">range =</span> <span class="fu">c</span>(<span class="dv">25</span>, <span class="dv">500</span>), <span class="at">ref =</span> <span class="cn">NULL</span>, <span class="at">center =</span> <span class="cn">TRUE</span>, </span>
<span id="cb187-155"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-155" aria-hidden="true" tabindex="-1"></a>    <span class="at">p.value =</span> <span class="fl">0.005</span>, <span class="at">weighted =</span> <span class="cn">TRUE</span>, <span class="at">nPermutation=</span><span class="dv">100</span>, <span class="at">enforceRange =</span> <span class="cn">TRUE</span>, ...) {</span>
<span id="cb187-156"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-156" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is</span>(exprs, <span class="st">&quot;ExpressionSet&quot;</span>)) </span>
<span id="cb187-157"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-157" aria-hidden="true" tabindex="-1"></a>        exprs <span class="ot">&lt;-</span> <span class="fu">exprs</span>(exprs)</span>
<span id="cb187-158"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-158" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.list</span>(cl)) </span>
<span id="cb187-159"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-159" aria-hidden="true" tabindex="-1"></a>        <span class="fu">stop</span>(<span class="st">&quot;cl need to be a list&quot;</span>)</span>
<span id="cb187-160"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-160" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(ref)) {</span>
<span id="cb187-161"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-161" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.numeric</span>(ref)) </span>
<span id="cb187-162"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-162" aria-hidden="true" tabindex="-1"></a>            <span class="fu">stop</span>(<span class="st">&quot;column index&#39;s required&quot;</span>)</span>
<span id="cb187-163"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-163" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb187-164"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-164" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(ref)) {</span>
<span id="cb187-165"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-165" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (options.out<span class="sc">$</span>verbose) </span>
<span id="cb187-166"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-166" aria-hidden="true" tabindex="-1"></a>            <span class="fu">cat</span>(<span class="st">&quot;Creating ratios...&quot;</span>, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb187-167"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-167" aria-hidden="true" tabindex="-1"></a>        ref_mean <span class="ot">&lt;-</span> <span class="fu">apply</span>(exprs[, ref], <span class="dv">1</span>, mean, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb187-168"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-168" aria-hidden="true" tabindex="-1"></a>        exprs <span class="ot">&lt;-</span> <span class="fu">sweep</span>(exprs, <span class="dv">1</span>, ref_mean, <span class="st">&quot;-&quot;</span>)</span>
<span id="cb187-169"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-169" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb187-170"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-170" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (center) </span>
<span id="cb187-171"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-171" aria-hidden="true" tabindex="-1"></a>        exprs <span class="ot">&lt;-</span> <span class="fu">scale</span>(exprs, <span class="at">scale =</span> <span class="cn">FALSE</span>)         <span class="co"># column centering is done</span></span>
<span id="cb187-172"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-172" aria-hidden="true" tabindex="-1"></a>    results <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="fu">length</span>(cl), <span class="fu">ncol</span>(exprs))</span>
<span id="cb187-173"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-173" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(results) <span class="ot">&lt;-</span> <span class="fu">names</span>(cl)</span>
<span id="cb187-174"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-174" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(results) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(exprs)</span>
<span id="cb187-175"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-175" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mode</span>(results) <span class="ot">&lt;-</span> <span class="st">&quot;numeric&quot;</span></span>
<span id="cb187-176"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-176" aria-hidden="true" tabindex="-1"></a>    Setsize <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">0</span>,<span class="fu">length</span>(cl)))     <span class="co"># gene set size vector</span></span>
<span id="cb187-177"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-177" aria-hidden="true" tabindex="-1"></a>    mean2 <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">0</span>,<span class="fu">length</span>(cl)))     <span class="co"># mean of the range of means </span></span>
<span id="cb187-178"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-178" aria-hidden="true" tabindex="-1"></a>    meanSD <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">0</span>,<span class="fu">length</span>(cl)))     <span class="co"># SD of the range of means    </span></span>
<span id="cb187-179"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-179" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.logical</span>(p.value)) </span>
<span id="cb187-180"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-180" aria-hidden="true" tabindex="-1"></a>        { p.results <span class="ot">&lt;-</span> results; mean.results <span class="ot">&lt;-</span> results;}</span>
<span id="cb187-181"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-181" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(cl)) {              <span class="co"># for each gene list</span></span>
<span id="cb187-182"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-182" aria-hidden="true" tabindex="-1"></a>        <span class="co">#cat(&quot;\nProcessing gene set&quot;,i);</span></span>
<span id="cb187-183"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-183" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">class</span>(cl[[i]]) <span class="sc">==</span> <span class="st">&quot;smc&quot;</span>) {</span>
<span id="cb187-184"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-184" aria-hidden="true" tabindex="-1"></a>            clids <span class="ot">&lt;-</span> cl[[i]]<span class="sc">@</span>ids</span>
<span id="cb187-185"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-185" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb187-186"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-186" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span> <span class="cf">if</span> (<span class="fu">class</span>(cl[[i]]) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;GeneColorSet&quot;</span>, <span class="st">&quot;GeneSet&quot;</span>)) {</span>
<span id="cb187-187"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-187" aria-hidden="true" tabindex="-1"></a>            clids <span class="ot">&lt;-</span> cl[[i]]<span class="sc">@</span>geneIds</span>
<span id="cb187-188"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-188" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb187-189"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-189" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span> {</span>
<span id="cb187-190"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-190" aria-hidden="true" tabindex="-1"></a>            clids <span class="ot">&lt;-</span> cl[[i]]</span>
<span id="cb187-191"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-191" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb187-192"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-192" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">options</span>()<span class="sc">$</span>verbose) </span>
<span id="cb187-193"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-193" aria-hidden="true" tabindex="-1"></a>            <span class="fu">cat</span>(<span class="st">&quot;Testing region &quot;</span>, i, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb187-194"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-194" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">&lt;-</span> <span class="fu">match</span>(clids, <span class="fu">rownames</span>(exprs))</span>
<span id="cb187-195"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-195" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">&lt;-</span> <span class="fu">unique</span>(ix[<span class="sc">!</span><span class="fu">is.na</span>(ix)])</span>
<span id="cb187-196"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-196" aria-hidden="true" tabindex="-1"></a>        present <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(ix))</span>
<span id="cb187-197"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-197" aria-hidden="true" tabindex="-1"></a>        Setsize[i] <span class="ot">&lt;-</span> present </span>
<span id="cb187-198"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-198" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (present <span class="sc">&lt;</span> range[<span class="dv">1</span>]) {</span>
<span id="cb187-199"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-199" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (<span class="fu">options</span>()<span class="sc">$</span>verbose) </span>
<span id="cb187-200"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-200" aria-hidden="true" tabindex="-1"></a>                <span class="fu">cat</span>(<span class="st">&quot;Skipping region &quot;</span>, i, <span class="st">&quot; because too small-&quot;</span>, </span>
<span id="cb187-201"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-201" aria-hidden="true" tabindex="-1"></a>                  present, <span class="st">&quot;,</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb187-202"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-202" aria-hidden="true" tabindex="-1"></a>            <span class="cf">next</span></span>
<span id="cb187-203"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-203" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb187-204"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-204" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (present <span class="sc">&gt;</span> range[<span class="dv">2</span>]) {</span>
<span id="cb187-205"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-205" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (<span class="fu">options</span>()<span class="sc">$</span>verbose) </span>
<span id="cb187-206"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-206" aria-hidden="true" tabindex="-1"></a>                <span class="fu">cat</span>(<span class="st">&quot;Skipping region &quot;</span>, i, <span class="st">&quot; because too large-&quot;</span>, </span>
<span id="cb187-207"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-207" aria-hidden="true" tabindex="-1"></a>                  present, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb187-208"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-208" aria-hidden="true" tabindex="-1"></a>            <span class="cf">next</span></span>
<span id="cb187-209"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-209" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb187-210"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-210" aria-hidden="true" tabindex="-1"></a>        texprs <span class="ot">&lt;-</span> exprs[ix, ]           <span class="co"># expression matrix for genes in gene set</span></span>
<span id="cb187-211"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-211" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">any</span>(<span class="fu">is.na</span>(texprs))) </span>
<span id="cb187-212"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-212" aria-hidden="true" tabindex="-1"></a>            <span class="fu">cat</span>(<span class="st">&quot;Warning - &#39;NA&#39; values within expression data, enrichment scores are estimates only.</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb187-213"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-213" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.matrix</span>(texprs)) </span>
<span id="cb187-214"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-214" aria-hidden="true" tabindex="-1"></a>            texprs <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(texprs)</span>
<span id="cb187-215"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-215" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb187-216"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-216" aria-hidden="true" tabindex="-1"></a>        stat <span class="ot">&lt;-</span> <span class="fu">try</span>(<span class="fu">apply</span>(texprs, <span class="dv">2</span>, t.test, ...))</span>
<span id="cb187-217"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-217" aria-hidden="true" tabindex="-1"></a>        means <span class="ot">&lt;-</span> <span class="fu">try</span>(<span class="fu">apply</span>(texprs, <span class="dv">2</span>, mean,<span class="at">trim=</span><span class="fl">0.1</span>))   <span class="co"># trim mean</span></span>
<span id="cb187-218"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-218" aria-hidden="true" tabindex="-1"></a>        ps <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(stat, <span class="cf">function</span>(x) x<span class="sc">$</span>p.value))</span>
<span id="cb187-219"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-219" aria-hidden="true" tabindex="-1"></a>        stat <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(stat, <span class="cf">function</span>(x) x<span class="sc">$</span>statistic))</span>
<span id="cb187-220"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-220" aria-hidden="true" tabindex="-1"></a>        p.results[i, ] <span class="ot">&lt;-</span> ps</span>
<span id="cb187-221"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-221" aria-hidden="true" tabindex="-1"></a>        mean.results[i,] <span class="ot">&lt;-</span> means</span>
<span id="cb187-222"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-222" aria-hidden="true" tabindex="-1"></a>        results[i, ] <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(stat)</span>
<span id="cb187-223"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-223" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb187-224"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-224" aria-hidden="true" tabindex="-1"></a>        <span class="co"># permutation of gene sets of the same size</span></span>
<span id="cb187-225"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-225" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(nPermutation <span class="sc">&gt;</span> <span class="dv">2</span> )  { <span class="co"># no permutation if &lt;=2</span></span>
<span id="cb187-226"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-226" aria-hidden="true" tabindex="-1"></a>            meansRanges <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">rep</span>(nPermutation))</span>
<span id="cb187-227"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-227" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>( k <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nPermutation ) {</span>
<span id="cb187-228"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-228" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">&lt;-</span> <span class="fu">sample.int</span>( <span class="fu">dim</span>(exprs)[<span class="dv">1</span>], <span class="fu">length</span>(ix) )</span>
<span id="cb187-229"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-229" aria-hidden="true" tabindex="-1"></a>                texprs <span class="ot">&lt;-</span> exprs[ix, ] </span>
<span id="cb187-230"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-230" aria-hidden="true" tabindex="-1"></a>                means <span class="ot">&lt;-</span> <span class="fu">try</span>(<span class="fu">apply</span>(texprs, <span class="dv">2</span>, mean,<span class="at">trim=</span><span class="fl">0.1</span>))   <span class="co"># trim mean</span></span>
<span id="cb187-231"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-231" aria-hidden="true" tabindex="-1"></a>                meansRanges[k] <span class="ot">=</span> <span class="fu">dynamicRange</span>(means)</span>
<span id="cb187-232"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-232" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb187-233"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-233" aria-hidden="true" tabindex="-1"></a>            mean2[i] <span class="ot">=</span> <span class="fu">mean</span>(meansRanges)</span>
<span id="cb187-234"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-234" aria-hidden="true" tabindex="-1"></a>            meanSD[i]<span class="ot">=</span> <span class="fu">sd</span>(meansRanges,<span class="at">na.rm=</span><span class="cn">TRUE</span>)   <span class="co"># NA are removed before calculating standard deviation</span></span>
<span id="cb187-235"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-235" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb187-236"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-236" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb187-237"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-237" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">results =</span> results, <span class="at">p.results =</span> p.results, <span class="at">means =</span> mean.results, <span class="at">size=</span>Setsize, <span class="at">mean2=</span>mean2, <span class="at">meanSD=</span>meanSD))</span>
<span id="cb187-238"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-238" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb187-239"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-240"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-240" aria-hidden="true" tabindex="-1"></a>PCApathway<span class="ot">&lt;-</span> <span class="cf">function</span> () {  <span class="co"># pathway</span></span>
<span id="cb187-241"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-241" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> convertedData.out;</span>
<span id="cb187-242"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-242" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(PGSEA,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb187-243"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-243" aria-hidden="true" tabindex="-1"></a>    pca.object <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(x))</span>
<span id="cb187-244"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-244" aria-hidden="true" tabindex="-1"></a>    pca <span class="ot">=</span> <span class="dv">100</span><span class="sc">*</span>pca.object<span class="sc">$</span>rotation </span>
<span id="cb187-245"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-245" aria-hidden="true" tabindex="-1"></a>    Npca <span class="ot">=</span> <span class="dv">5</span></span>
<span id="cb187-246"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-246" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (Npca <span class="sc">&gt;</span> <span class="fu">dim</span>(pca)[<span class="dv">2</span>]) { Npca <span class="ot">=</span> <span class="fu">dim</span>(pca)[<span class="dv">2</span>] } <span class="cf">else</span> pca <span class="ot">&lt;-</span>  pca[,<span class="dv">1</span><span class="sc">:</span>Npca]</span>
<span id="cb187-247"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-247" aria-hidden="true" tabindex="-1"></a>    <span class="co">#pca = pca[,1:5]</span></span>
<span id="cb187-248"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-248" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(GeneSets.out ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)  <span class="co"># no species recognized</span></span>
<span id="cb187-249"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-249" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(GeneSets.out ) <span class="sc">&lt;=</span> <span class="dv">1</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb187-250"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-250" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(&quot;\n\nGene Sets:&quot;,length( GeneSets.out))</span></span>
<span id="cb187-251"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-251" aria-hidden="true" tabindex="-1"></a>    pg <span class="ot">=</span> <span class="fu">myPGSEA</span> (pca,<span class="at">cl=</span>GeneSets.out,<span class="at">range=</span><span class="fu">c</span>(<span class="dv">15</span>,<span class="dv">2000</span>),<span class="at">p.value=</span><span class="cn">TRUE</span>, <span class="at">weighted=</span><span class="cn">FALSE</span>,<span class="at">nPermutation=</span><span class="dv">1</span>)</span>
<span id="cb187-252"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-253"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-253" aria-hidden="true" tabindex="-1"></a>    <span class="co"># correcting for multiple testing</span></span>
<span id="cb187-254"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-254" aria-hidden="true" tabindex="-1"></a>    p.matrix <span class="ot">=</span> pg<span class="sc">$</span>p.result</span>
<span id="cb187-255"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-255" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> <span class="fu">p.adjust</span>(<span class="fu">as.numeric</span>(p.matrix),<span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb187-256"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-256" aria-hidden="true" tabindex="-1"></a>    p.matrix <span class="ot">=</span> <span class="fu">matrix</span>(tem, <span class="at">nrow=</span><span class="fu">dim</span>(p.matrix)[<span class="dv">1</span>], <span class="at">ncol =</span> <span class="fu">dim</span>(p.matrix)[<span class="dv">2</span>] )</span>
<span id="cb187-257"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-257" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(p.matrix) <span class="ot">=</span> <span class="fu">rownames</span>(pg<span class="sc">$</span>p.result); <span class="fu">colnames</span>(p.matrix) <span class="ot">=</span> <span class="fu">colnames</span>(pg<span class="sc">$</span>p.result)</span>
<span id="cb187-258"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-259"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-260"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-260" aria-hidden="true" tabindex="-1"></a>    selected <span class="ot">=</span><span class="fu">c</span>()</span>
<span id="cb187-261"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-261" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(p.matrix)[<span class="dv">2</span>]) {</span>
<span id="cb187-262"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-262" aria-hidden="true" tabindex="-1"></a>      tem <span class="ot">=</span> <span class="fu">which</span>( <span class="fu">rank</span>(p.matrix[,i],<span class="at">ties.method=</span><span class="st">&#39;first&#39;</span>) <span class="sc">&lt;=</span> <span class="dv">5</span>)  <span class="co"># rank by P value</span></span>
<span id="cb187-263"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-263" aria-hidden="true" tabindex="-1"></a>     <span class="co">#tem = which( rank(pg$result[,i],ties.method=&#39;first&#39;) &gt;= dim(p.matrix)[1]-3.1) # rank by mean</span></span>
<span id="cb187-264"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-264" aria-hidden="true" tabindex="-1"></a>     <span class="fu">names</span>(tem) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;PC&quot;</span>,i,<span class="st">&quot; &quot;</span>, <span class="fu">rownames</span>(p.matrix)[tem], <span class="at">sep=</span><span class="st">&quot;&quot;</span> )</span>
<span id="cb187-265"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-265" aria-hidden="true" tabindex="-1"></a>     selected <span class="ot">=</span> <span class="fu">c</span>(selected, tem)</span>
<span id="cb187-266"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-266" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb187-267"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-267" aria-hidden="true" tabindex="-1"></a>    rowids <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; .*&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">names</span>(selected))</span>
<span id="cb187-268"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-268" aria-hidden="true" tabindex="-1"></a>    rowids <span class="ot">=</span> <span class="fu">as.numeric</span>( <span class="fu">gsub</span>(<span class="st">&quot;PC&quot;</span>,<span class="st">&quot;&quot;</span>,rowids) )</span>
<span id="cb187-269"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-269" aria-hidden="true" tabindex="-1"></a>    pvals <span class="ot">=</span> p.matrix[ <span class="fu">cbind</span>(selected,rowids) ]</span>
<span id="cb187-270"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-270" aria-hidden="true" tabindex="-1"></a>    a<span class="ot">=</span><span class="fu">sprintf</span>(<span class="st">&quot;%-1.0e&quot;</span>,pvals)</span>
<span id="cb187-271"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-271" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> pg<span class="sc">$</span>result[selected,]</span>
<span id="cb187-272"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-272" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(tem) <span class="ot">=</span> <span class="fu">paste</span>(a,<span class="fu">names</span>(selected)); <span class="co">#colnames(tem)= paste(&quot;PC&quot;,colnames(tem),sep=&quot;&quot;)</span></span>
<span id="cb187-273"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-273" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb187-274"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-274" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> tem[<span class="sc">!</span><span class="fu">duplicated</span>(selected),] </span>
<span id="cb187-275"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-276"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-276" aria-hidden="true" tabindex="-1"></a>    <span class="co">#tem = t(tem); tem = t( (tem - apply(tem,1,mean)) ) #/apply(tem,1,sd) )</span></span>
<span id="cb187-277"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-278"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-278" aria-hidden="true" tabindex="-1"></a>    <span class="fu">smcPlot</span>(tem,<span class="at">scale =</span>  <span class="fu">c</span>(<span class="sc">-</span><span class="fu">max</span>(tem), <span class="fu">max</span>(tem)), <span class="at">show.grid =</span> T, <span class="at">margins =</span> <span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">1</span>, <span class="dv">6</span>, <span class="dv">23</span>), <span class="at">col =</span> .rwb,<span class="at">cex.lab=</span><span class="fl">0.5</span>, <span class="at">main=</span><span class="st">&quot;Pathways analysis on PCA&quot;</span>)</span>
<span id="cb187-279"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-279" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb187-280"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-280" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb187-281"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-282"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-282" aria-hidden="true" tabindex="-1"></a><span class="co"># correlation PCA with factors  </span></span>
<span id="cb187-283"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-283" aria-hidden="true" tabindex="-1"></a>PCA2factor <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb187-284"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-285"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-285" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(readSampleInfo.out)) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb187-286"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-286" aria-hidden="true" tabindex="-1"></a>    npc <span class="ot">=</span> <span class="dv">5</span> <span class="co"># number of Principal components</span></span>
<span id="cb187-287"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-287" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> readData.out<span class="sc">$</span>data</span>
<span id="cb187-288"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-288" aria-hidden="true" tabindex="-1"></a>    y <span class="ot">&lt;-</span> readSampleInfo.out</span>
<span id="cb187-289"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-290"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-290" aria-hidden="true" tabindex="-1"></a>    pca.object <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(x))</span>
<span id="cb187-291"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-291" aria-hidden="true" tabindex="-1"></a>    pcaData <span class="ot">=</span> <span class="fu">as.data.frame</span>(pca.object<span class="sc">$</span>x[,<span class="dv">1</span><span class="sc">:</span>npc]); </span>
<span id="cb187-292"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-292" aria-hidden="true" tabindex="-1"></a>    pvals <span class="ot">=</span> <span class="fu">matrix</span>(<span class="dv">1</span>,<span class="at">nrow=</span>npc,<span class="at">ncol=</span><span class="fu">ncol</span>(y))</span>
<span id="cb187-293"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-293" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>npc ){</span>
<span id="cb187-294"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-294" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(y) )</span>
<span id="cb187-295"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-295" aria-hidden="true" tabindex="-1"></a>            pvals[i,j] <span class="ot">=</span><span class="fu">summary</span>( <span class="fu">aov</span>(pcaData[,i] <span class="sc">~</span> <span class="fu">as.factor</span>(y[,j])))[[<span class="dv">1</span>]][[<span class="st">&quot;Pr(&gt;F)&quot;</span>]][<span class="dv">1</span>]</span>
<span id="cb187-296"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-296" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb187-297"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-297" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb187-298"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-298" aria-hidden="true" tabindex="-1"></a>    pvals <span class="ot">=</span> pvals <span class="sc">*</span> npc<span class="sc">*</span> <span class="fu">ncol</span>(y)   <span class="co"># correcting for multiple testing</span></span>
<span id="cb187-299"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-299" aria-hidden="true" tabindex="-1"></a>    pvals[pvals<span class="sc">&gt;</span><span class="dv">1</span>] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb187-300"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-301"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-301" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(pvals) <span class="ot">=</span> <span class="fu">colnames</span>(y)</span>
<span id="cb187-302"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-302" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(pvals) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">&quot;PC&quot;</span>,<span class="dv">1</span><span class="sc">:</span>npc)</span>
<span id="cb187-303"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-303" aria-hidden="true" tabindex="-1"></a>    a<span class="ot">=</span><span class="st">&quot;</span><span class="sc">\n</span><span class="st"> Correlation between Principal Components (PCs) with factors</span><span class="sc">\n</span><span class="st">&quot;</span></span>
<span id="cb187-304"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-304" aria-hidden="true" tabindex="-1"></a>    nchar0 <span class="ot">=</span> <span class="fu">nchar</span>(a)</span>
<span id="cb187-305"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-305" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> ( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>npc){</span>
<span id="cb187-306"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-306" aria-hidden="true" tabindex="-1"></a>        j <span class="ot">=</span> <span class="fu">which.min</span>(pvals[i,])</span>
<span id="cb187-307"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-307" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(pvals[i,j]<span class="sc">&lt;</span> <span class="fl">0.05</span>) a<span class="ot">=</span><span class="fu">paste0</span>(a,<span class="fu">rownames</span>(pvals)[i], </span>
<span id="cb187-308"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-308" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot; is correlated with &quot;</span>, <span class="fu">colnames</span>(pvals)[j],</span>
<span id="cb187-309"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-309" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot; (p=&quot;</span>,   <span class="fu">sprintf</span>(<span class="st">&quot;%-3.2e&quot;</span>,pvals[i,j]),<span class="st">&quot;).</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb187-310"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-310" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb187-311"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-311" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">nchar</span>(a) <span class="sc">==</span> nchar0 ) <span class="fu">return</span>(<span class="cn">NULL</span>) <span class="cf">else</span> </span>
<span id="cb187-312"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-312" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>( a )</span>
<span id="cb187-313"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-313" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb187-314"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-314" aria-hidden="true" tabindex="-1"></a>} </span>
<span id="cb187-315"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-316"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-316" aria-hidden="true" tabindex="-1"></a><span class="co"># detecting sequencing depth bias</span></span>
<span id="cb187-317"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-317" aria-hidden="true" tabindex="-1"></a>readCountsBias <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb187-318"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-319"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-319" aria-hidden="true" tabindex="-1"></a>    totalCounts <span class="ot">=</span> <span class="fu">colSums</span>(readData.out<span class="sc">$</span>rawCounts) </span>
<span id="cb187-320"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-320" aria-hidden="true" tabindex="-1"></a>    groups <span class="ot">=</span> <span class="fu">as.factor</span>( <span class="fu">detectGroups</span>(<span class="fu">colnames</span>(readData.out<span class="sc">$</span>rawCounts ) ) )</span>
<span id="cb187-321"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-321" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb187-322"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-322" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ANOVA of total read counts vs sample groups parsed by sample name</span></span>
<span id="cb187-323"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-323" aria-hidden="true" tabindex="-1"></a>    pval <span class="ot">=</span> <span class="fu">summary</span>( <span class="fu">aov</span>(totalCounts <span class="sc">~</span> groups ))[[<span class="dv">1</span>]][[<span class="st">&quot;Pr(&gt;F)&quot;</span>]][<span class="dv">1</span>]</span>
<span id="cb187-324"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-324" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(pval <span class="sc">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb187-325"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-325" aria-hidden="true" tabindex="-1"></a>      tem <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;Warning! Sequencing depth bias detected. Total read counts are significantly different among sample groups (p=&quot;</span>,</span>
<span id="cb187-326"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-326" aria-hidden="true" tabindex="-1"></a>                <span class="fu">sprintf</span>(<span class="st">&quot;%-3.2e&quot;</span>,pval),<span class="st">&quot;) based on ANOVA.&quot;</span>)</span>
<span id="cb187-327"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-328"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-328" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ANOVA of total read counts vs factors in experiment design</span></span>
<span id="cb187-329"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-329" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(readSampleInfo.out   )  ) {</span>
<span id="cb187-330"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-330" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> readSampleInfo.out</span>
<span id="cb187-331"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-331" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(y) ) { </span>
<span id="cb187-332"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-332" aria-hidden="true" tabindex="-1"></a>        pval <span class="ot">=</span> <span class="fu">summary</span>( <span class="fu">aov</span>(totalCounts <span class="sc">~</span> <span class="fu">as.factor</span>(y[,j])))[[<span class="dv">1</span>]][[<span class="st">&quot;Pr(&gt;F)&quot;</span>]][<span class="dv">1</span>]</span>
<span id="cb187-333"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-334"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-334" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(pval <span class="sc">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb187-335"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-335" aria-hidden="true" tabindex="-1"></a>        tem <span class="ot">=</span> <span class="fu">paste</span>(tem, <span class="st">&quot; Total read counts seem to be correlated with factor&quot;</span>,<span class="fu">colnames</span>(y)[j], </span>
<span id="cb187-336"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-336" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;(p=&quot;</span>,  <span class="fu">sprintf</span>(<span class="st">&quot;%-3.2e&quot;</span>,pval),<span class="st">&quot;).  &quot;</span>)</span>
<span id="cb187-337"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-337" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb187-338"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-338" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb187-339"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-339" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(tem)) <span class="fu">return</span>( <span class="st">&quot;No bias detected&quot;</span>) <span class="cf">else</span></span>
<span id="cb187-340"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-340" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( tem )</span>
<span id="cb187-341"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-341" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb187-342"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb187-342" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="k-means-clustering" class="section level2" number="6.7">
<h2><span class="header-section-number">6.7</span> K-means clustering</h2>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-1" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb188-2"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-2" aria-hidden="true" tabindex="-1"></a><span class="co"># K-means clustering</span></span>
<span id="cb188-3"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-3" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb188-4"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-4" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb188-5"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-5" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Distribution of SDs </span></span>
<span id="cb188-6"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-6" aria-hidden="true" tabindex="-1"></a>distributionSD <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb188-7"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-8"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-8" aria-hidden="true" tabindex="-1"></a>        SDs<span class="ot">=</span><span class="fu">apply</span>(convertedData.out,<span class="dv">1</span>,sd)</span>
<span id="cb188-9"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-9" aria-hidden="true" tabindex="-1"></a>        maxSD <span class="ot">=</span> <span class="fu">mean</span>(SDs)<span class="sc">+</span> <span class="dv">4</span><span class="sc">*</span><span class="fu">sd</span>(SDs)</span>
<span id="cb188-10"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-10" aria-hidden="true" tabindex="-1"></a>        SDs[ SDs <span class="sc">&gt;</span> maxSD] <span class="ot">=</span> maxSD</span>
<span id="cb188-11"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-11" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb188-12"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-12" aria-hidden="true" tabindex="-1"></a>        top <span class="ot">=</span> input_nGenesKNN</span>
<span id="cb188-13"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-13" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(top <span class="sc">&gt;</span> <span class="fu">length</span>(SDs)) top <span class="ot">=</span> <span class="fu">length</span>(SDs)</span>
<span id="cb188-14"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-14" aria-hidden="true" tabindex="-1"></a>        Cutoff<span class="ot">=</span><span class="fu">sort</span>(SDs,<span class="at">decreasing=</span><span class="cn">TRUE</span>)[top] </span>
<span id="cb188-15"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-16"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-16" aria-hidden="true" tabindex="-1"></a>        SDs <span class="ot">=</span> <span class="fu">as.data.frame</span>(SDs)</span>
<span id="cb188-17"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-18"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-18" aria-hidden="true" tabindex="-1"></a>        p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(SDs, <span class="fu">aes</span>(<span class="at">x=</span>SDs)) <span class="sc">+</span> </span>
<span id="cb188-19"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-19" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_density</span>(<span class="at">color=</span><span class="st">&quot;darkblue&quot;</span>, <span class="at">fill=</span><span class="st">&quot;lightblue&quot;</span>) <span class="sc">+</span></span>
<span id="cb188-20"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-20" aria-hidden="true" tabindex="-1"></a>          <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Standard deviations of all genes&quot;</span>, <span class="at">y=</span><span class="st">&quot;Density&quot;</span>)<span class="sc">+</span></span>
<span id="cb188-21"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-21" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept=</span>Cutoff),</span>
<span id="cb188-22"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">color=</span><span class="st">&quot;red&quot;</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb188-23"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-23" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="at">x =</span> Cutoff <span class="sc">+</span> <span class="fl">0.4</span><span class="sc">*</span><span class="fu">sd</span>(SDs[,<span class="dv">1</span>]), <span class="at">y =</span> <span class="dv">1</span>,<span class="at">colour =</span> <span class="st">&quot;red&quot;</span>, <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;Top &quot;</span>, top))</span>
<span id="cb188-24"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-25"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-25" aria-hidden="true" tabindex="-1"></a>        p</span>
<span id="cb188-26"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb188-27"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-28"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-29"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Decide number of clusters</span></span>
<span id="cb188-30"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-30" aria-hidden="true" tabindex="-1"></a>KmeansNclusters <span class="ot">&lt;-</span> <span class="cf">function</span>() { <span class="co"># Kmeans clustering</span></span>
<span id="cb188-31"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-32"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-32" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> convertedData.out</span>
<span id="cb188-33"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-33" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x &lt;- readData.out</span></span>
<span id="cb188-34"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-34" aria-hidden="true" tabindex="-1"></a>    <span class="co">#par(mfrow=c(1,2))</span></span>
<span id="cb188-35"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-35" aria-hidden="true" tabindex="-1"></a>    n<span class="ot">=</span>input_nGenesKNN</span>
<span id="cb188-36"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-36" aria-hidden="true" tabindex="-1"></a>    <span class="co">#if(n&gt;6000) n = 6000 # max</span></span>
<span id="cb188-37"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-37" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(n<span class="sc">&gt;</span><span class="fu">dim</span>(x)[<span class="dv">1</span>]) n <span class="ot">=</span> <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="co"># max as data</span></span>
<span id="cb188-38"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-38" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x1 &lt;- x;</span></span>
<span id="cb188-39"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-39" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x=as.matrix(x[1:n,])-apply(x[1:n,],1,mean)</span></span>
<span id="cb188-40"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-40" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">=</span> <span class="dv">100</span><span class="sc">*</span> x[<span class="dv">1</span><span class="sc">:</span>n,] <span class="sc">/</span> <span class="fu">apply</span>(x[<span class="dv">1</span><span class="sc">:</span>n,],<span class="dv">1</span>,sum)  <span class="co"># this is causing problem??????</span></span>
<span id="cb188-41"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-41" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x = x - apply(x,1,mean)  # this is causing problem??????</span></span>
<span id="cb188-42"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-42" aria-hidden="true" tabindex="-1"></a>    <span class="co">#colnames(x) = gsub(&quot;_.*&quot;,&quot;&quot;,colnames(x))</span></span>
<span id="cb188-43"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set.seed</span>(<span class="dv">2</span>)</span>
<span id="cb188-44"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-44" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb188-45"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-45" aria-hidden="true" tabindex="-1"></a>    k <span class="ot">=</span> <span class="dv">30</span></span>
<span id="cb188-46"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-46" aria-hidden="true" tabindex="-1"></a>    wss <span class="ot">&lt;-</span> (<span class="fu">nrow</span>(x)<span class="sc">-</span><span class="dv">1</span>)<span class="sc">*</span><span class="fu">sum</span>(<span class="fu">apply</span>(x,<span class="dv">2</span>,var))</span>
<span id="cb188-47"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-47" aria-hidden="true" tabindex="-1"></a>      <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span>k) wss[i] <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">kmeans</span>(x,<span class="at">centers=</span>i,<span class="at">iter.max =</span> <span class="dv">30</span>)<span class="sc">$</span>withinss)</span>
<span id="cb188-48"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-48" aria-hidden="true" tabindex="-1"></a>        <span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">4</span>,<span class="dv">4</span>))</span>
<span id="cb188-49"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot</span>(<span class="dv">1</span><span class="sc">:</span>k, wss, <span class="at">type=</span><span class="st">&quot;b&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;Number of Clusters (k)&quot;</span>,</span>
<span id="cb188-50"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-50" aria-hidden="true" tabindex="-1"></a>         <span class="at">ylab=</span><span class="st">&quot;Within groups sum of squares&quot;</span>,</span>
<span id="cb188-51"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-51" aria-hidden="true" tabindex="-1"></a>         <span class="at">cex=</span><span class="dv">2</span>,<span class="at">cex.lab=</span><span class="dv">2</span>, <span class="at">cex.axis=</span><span class="dv">2</span>, <span class="at">cex.main=</span><span class="dv">2</span>, <span class="at">cex.sub=</span><span class="dv">2</span> ,<span class="at">xaxt=</span><span class="st">&quot;n&quot;</span>    )</span>
<span id="cb188-52"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">axis</span>(<span class="dv">1</span>, <span class="at">at =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">30</span>, <span class="at">by =</span> <span class="dv">2</span>),<span class="at">cex.axis=</span><span class="fl">1.5</span>,<span class="at">cex=</span><span class="fl">1.5</span>)</span>
<span id="cb188-53"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-53" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb188-54"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-54" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb188-55"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-55" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb188-56"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-56" aria-hidden="true" tabindex="-1"></a>Kmeans <span class="ot">&lt;-</span> <span class="cf">function</span>() { <span class="co"># Kmeans clustering</span></span>
<span id="cb188-57"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-58"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-58" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> convertedData.out</span>
<span id="cb188-59"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-59" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x &lt;- readData.out</span></span>
<span id="cb188-60"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-60" aria-hidden="true" tabindex="-1"></a>    <span class="co">#par(mfrow=c(1,2))</span></span>
<span id="cb188-61"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-61" aria-hidden="true" tabindex="-1"></a>    n<span class="ot">=</span>input_nGenesKNN</span>
<span id="cb188-62"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-62" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(n<span class="sc">&gt;</span>maxGeneClustering) n <span class="ot">=</span> maxGeneClustering <span class="co"># max</span></span>
<span id="cb188-63"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-63" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(n<span class="sc">&gt;</span><span class="fu">dim</span>(x)[<span class="dv">1</span>]) n <span class="ot">=</span> <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="co"># max as data</span></span>
<span id="cb188-64"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-64" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x1 &lt;- x;</span></span>
<span id="cb188-65"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-65" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x=as.matrix(x[1:n,])-apply(x[1:n,],1,mean)</span></span>
<span id="cb188-66"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-66" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x = 100* x[1:n,] / apply(x[1:n,],1,sum) </span></span>
<span id="cb188-67"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-67" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">=</span> x[<span class="dv">1</span><span class="sc">:</span>n,]</span>
<span id="cb188-68"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-68" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_kmeansNormalization <span class="sc">==</span> <span class="st">&#39;L1Norm&#39;</span>)</span>
<span id="cb188-69"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-69" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">=</span> <span class="dv">100</span><span class="sc">*</span> x <span class="sc">/</span> <span class="fu">apply</span>(x,<span class="dv">1</span>,<span class="cf">function</span>(y) <span class="fu">sum</span>(<span class="fu">abs</span>(y))) <span class="cf">else</span> <span class="co"># L1 norm</span></span>
<span id="cb188-70"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-70" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_kmeansNormalization <span class="sc">==</span> <span class="st">&#39;geneMean&#39;</span>)</span>
<span id="cb188-71"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-71" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">=</span> x <span class="sc">-</span> <span class="fu">apply</span>(x,<span class="dv">1</span>,mean)  <span class="cf">else</span> <span class="co"># this is causing problem??????</span></span>
<span id="cb188-72"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-72" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_kmeansNormalization <span class="sc">==</span> <span class="st">&#39;geneStandardization&#39;</span>) </span>
<span id="cb188-73"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-73" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">=</span> (x <span class="sc">-</span> <span class="fu">apply</span>(x,<span class="dv">1</span>,mean) ) <span class="sc">/</span> <span class="fu">apply</span>(x,<span class="dv">1</span>,sd)</span>
<span id="cb188-74"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-74" aria-hidden="true" tabindex="-1"></a>    <span class="co">#colnames(x) = gsub(&quot;_.*&quot;,&quot;&quot;,colnames(x))</span></span>
<span id="cb188-75"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-75" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set.seed</span>(input_KmeansReRun)</span>
<span id="cb188-76"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-76" aria-hidden="true" tabindex="-1"></a>    k<span class="ot">=</span>input_nClusters</span>
<span id="cb188-77"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-78"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-78" aria-hidden="true" tabindex="-1"></a>    cl <span class="ot">=</span> <span class="fu">kmeans</span>(x,k,<span class="at">iter.max =</span> <span class="dv">50</span>)</span>
<span id="cb188-79"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-79" aria-hidden="true" tabindex="-1"></a>    <span class="co">#myheatmap(cl$centers)  </span></span>
<span id="cb188-80"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-80" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb188-81"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-82"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-82" aria-hidden="true" tabindex="-1"></a>    hc <span class="ot">&lt;-</span> <span class="fu">hclust2</span>(<span class="fu">dist2</span>(cl<span class="sc">$</span>centers<span class="sc">-</span><span class="fu">apply</span>(cl<span class="sc">$</span>centers,<span class="dv">1</span>,mean) )  )<span class="co"># perform cluster for the reordering of samples</span></span>
<span id="cb188-83"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-83" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> <span class="fu">match</span>(cl<span class="sc">$</span>cluster,hc<span class="sc">$</span>order) <span class="co">#  new order </span></span>
<span id="cb188-84"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-84" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">=</span> x[<span class="fu">order</span>(tem),] ;    bar <span class="ot">=</span> <span class="fu">sort</span>(tem)</span>
<span id="cb188-85"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-86"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-86" aria-hidden="true" tabindex="-1"></a>    <span class="co">#myheatmap2(x-apply(x,1,mean), bar,1000)</span></span>
<span id="cb188-87"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-87" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">list</span>( <span class="at">x =</span> x, <span class="at">bar =</span> bar)) </span>
<span id="cb188-88"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-89"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-89" aria-hidden="true" tabindex="-1"></a>  } </span>
<span id="cb188-90"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-91"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-92"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-92" aria-hidden="true" tabindex="-1"></a>myheatmap2 <span class="ot">&lt;-</span> <span class="cf">function</span> (x,<span class="at">bar=</span><span class="cn">NULL</span>,<span class="at">n=</span><span class="sc">-</span><span class="dv">1</span>,<span class="at">mycolor=</span><span class="dv">1</span>,<span class="at">clusterNames=</span><span class="cn">NULL</span>,<span class="at">sideColors=</span><span class="cn">NULL</span> ) {</span>
<span id="cb188-93"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-93" aria-hidden="true" tabindex="-1"></a>    <span class="co"># number of genes to show</span></span>
<span id="cb188-94"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-94" aria-hidden="true" tabindex="-1"></a>    ngenes <span class="ot">=</span> <span class="fu">as.character</span>( <span class="fu">table</span>(bar))</span>
<span id="cb188-95"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-95" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(bar) <span class="sc">&gt;</span>n <span class="sc">&amp;&amp;</span> n <span class="sc">!=</span> <span class="sc">-</span><span class="dv">1</span>) {ix <span class="ot">=</span> <span class="fu">sort</span>( <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(bar),n) ); bar <span class="ot">=</span> bar[ix]; x <span class="ot">=</span> x[ix,]  }</span>
<span id="cb188-96"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-96" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span> <span class="fu">is.null</span>(bar) )</span>
<span id="cb188-97"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-97" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>(sideColors) ) </span>
<span id="cb188-98"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-98" aria-hidden="true" tabindex="-1"></a>            sideColors <span class="ot">=</span> mycolors</span>
<span id="cb188-99"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-100"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-100" aria-hidden="true" tabindex="-1"></a>    <span class="co"># this will cutoff very large values, which could skew the color </span></span>
<span id="cb188-101"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-101" aria-hidden="true" tabindex="-1"></a>    x<span class="ot">=</span><span class="fu">as.matrix</span>(x)<span class="sc">-</span><span class="fu">apply</span>(x,<span class="dv">1</span>,mean)</span>
<span id="cb188-102"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-102" aria-hidden="true" tabindex="-1"></a>    cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">+</span> <span class="dv">3</span><span class="sc">*</span><span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb188-103"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-103" aria-hidden="true" tabindex="-1"></a>    x[x<span class="sc">&gt;</span>cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb188-104"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-104" aria-hidden="true" tabindex="-1"></a>    cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">-</span> <span class="dv">3</span><span class="sc">*</span><span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb188-105"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-105" aria-hidden="true" tabindex="-1"></a>    x[x<span class="sc">&lt;</span> cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb188-106"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-106" aria-hidden="true" tabindex="-1"></a>    <span class="co">#colnames(x)= detectGroups(colnames(x))</span></span>
<span id="cb188-107"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-107" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(bar)) <span class="co"># no side colors</span></span>
<span id="cb188-108"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-108" aria-hidden="true" tabindex="-1"></a>        <span class="fu">heatmap.2</span>(x,  <span class="at">Rowv =</span>F,<span class="at">Colv=</span>F, <span class="at">dendrogram =</span><span class="st">&quot;none&quot;</span>,</span>
<span id="cb188-109"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-109" aria-hidden="true" tabindex="-1"></a>            <span class="at">col=</span>heatColors[<span class="fu">as.integer</span>(mycolor),], <span class="at">density.info=</span><span class="st">&quot;none&quot;</span>, <span class="at">trace=</span><span class="st">&quot;none&quot;</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">keysize=</span>.<span class="dv">3</span></span>
<span id="cb188-110"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-110" aria-hidden="true" tabindex="-1"></a>            ,<span class="at">key=</span>F, <span class="at">labRow =</span> F</span>
<span id="cb188-111"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-111" aria-hidden="true" tabindex="-1"></a>            <span class="co">#,RowSideColors = mycolors[bar]</span></span>
<span id="cb188-112"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-112" aria-hidden="true" tabindex="-1"></a>            ,<span class="at">margins =</span> <span class="fu">c</span>(<span class="dv">8</span>, <span class="dv">24</span>)</span>
<span id="cb188-113"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-113" aria-hidden="true" tabindex="-1"></a>            ,<span class="at">srtCol=</span><span class="dv">45</span></span>
<span id="cb188-114"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-114" aria-hidden="true" tabindex="-1"></a>        ) <span class="cf">else</span></span>
<span id="cb188-115"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-115" aria-hidden="true" tabindex="-1"></a>        <span class="fu">heatmap.2</span>(x,  <span class="at">Rowv =</span>F,<span class="at">Colv=</span>F, <span class="at">dendrogram =</span><span class="st">&quot;none&quot;</span>,</span>
<span id="cb188-116"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-116" aria-hidden="true" tabindex="-1"></a>            <span class="at">col=</span>heatColors[<span class="fu">as.integer</span>(mycolor),], <span class="at">density.info=</span><span class="st">&quot;none&quot;</span>, <span class="at">trace=</span><span class="st">&quot;none&quot;</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">keysize=</span>.<span class="dv">3</span></span>
<span id="cb188-117"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-117" aria-hidden="true" tabindex="-1"></a>            ,<span class="at">key=</span>F, <span class="at">labRow =</span> F</span>
<span id="cb188-118"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-118" aria-hidden="true" tabindex="-1"></a>            ,<span class="at">RowSideColors =</span> sideColors[bar]</span>
<span id="cb188-119"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-119" aria-hidden="true" tabindex="-1"></a>            ,<span class="at">margins =</span> <span class="fu">c</span>(<span class="dv">8</span>, <span class="dv">24</span>)</span>
<span id="cb188-120"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-120" aria-hidden="true" tabindex="-1"></a>            ,<span class="at">srtCol=</span><span class="dv">45</span></span>
<span id="cb188-121"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-121" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb188-122"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-122" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb188-123"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-123" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(bar)) { </span>
<span id="cb188-124"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-125"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-125" aria-hidden="true" tabindex="-1"></a>        legend.text <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;Cluster &quot;</span>, <span class="fu">toupper</span>(letters)[<span class="fu">unique</span>(bar)], <span class="st">&quot; (N=&quot;</span>, ngenes,<span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>) </span>
<span id="cb188-126"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-126" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>( clusterNames ) <span class="sc">&amp;&amp;</span> <span class="fu">length</span>(clusterNames)<span class="sc">&gt;=</span> <span class="fu">length</span>( <span class="fu">unique</span>(bar) ) )  </span>
<span id="cb188-127"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-127" aria-hidden="true" tabindex="-1"></a>            legend.text <span class="ot">=</span> <span class="fu">paste</span>(clusterNames[ <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>( <span class="fu">unique</span>(bar) )  ], <span class="st">&quot; (N=&quot;</span>, ngenes,<span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>) </span>
<span id="cb188-128"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-128" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb188-129"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-129" aria-hidden="true" tabindex="-1"></a>        <span class="fu">par</span>(<span class="at">lend =</span> <span class="dv">1</span>)           <span class="co"># square line ends for the color legend</span></span>
<span id="cb188-130"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-130" aria-hidden="true" tabindex="-1"></a>        <span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>,      <span class="co"># location of the legend on the heatmap plot</span></span>
<span id="cb188-131"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-131" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend =</span> legend.text, <span class="co"># category labels</span></span>
<span id="cb188-132"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-132" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> sideColors,  <span class="co"># color key</span></span>
<span id="cb188-133"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-133" aria-hidden="true" tabindex="-1"></a>        <span class="at">lty=</span> <span class="dv">1</span>,             <span class="co"># line style</span></span>
<span id="cb188-134"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-134" aria-hidden="true" tabindex="-1"></a>        <span class="at">lwd =</span> <span class="dv">10</span> )           <span class="co"># line width</span></span>
<span id="cb188-135"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-135" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb188-136"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-136" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb188-137"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-138"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-138" aria-hidden="true" tabindex="-1"></a>KmeansHeatmap <span class="ot">&lt;-</span> <span class="cf">function</span>() { <span class="co"># Kmeans clustering</span></span>
<span id="cb188-139"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-140"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-140" aria-hidden="true" tabindex="-1"></a>    <span class="fu">myheatmap2</span>(Kmeans.out<span class="sc">$</span>x<span class="sc">-</span><span class="fu">apply</span>(Kmeans.out<span class="sc">$</span>x,<span class="dv">1</span>,mean), Kmeans.out<span class="sc">$</span>bar,<span class="dv">1000</span>,<span class="at">mycolor=</span>input_heatColors1)</span>
<span id="cb188-141"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-141" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb188-142"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-142" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb188-143"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-144"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-144" aria-hidden="true" tabindex="-1"></a>tSNEgenePlot <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb188-145"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-145" aria-hidden="true" tabindex="-1"></a>            Cluster <span class="ot">&lt;-</span> Kmeans.out<span class="sc">$</span>bar</span>
<span id="cb188-146"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-146" aria-hidden="true" tabindex="-1"></a>            train <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>( <span class="fu">cbind</span>(Cluster,Kmeans.out<span class="sc">$</span>x) )</span>
<span id="cb188-147"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-148"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-148" aria-hidden="true" tabindex="-1"></a>            <span class="fu">library</span>(Rtsne,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb188-149"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-150"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-150" aria-hidden="true" tabindex="-1"></a>            train <span class="ot">=</span> <span class="fu">unique</span>(train)</span>
<span id="cb188-151"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-151" aria-hidden="true" tabindex="-1"></a>            Cluster <span class="ot">=</span> train<span class="sc">$</span>Cluster </span>
<span id="cb188-152"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-152" aria-hidden="true" tabindex="-1"></a>            <span class="fu">set.seed</span>(input_seedTSNE)</span>
<span id="cb188-153"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-153" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb188-154"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-154" aria-hidden="true" tabindex="-1"></a>            <span class="do">## Executing the algorithm on curated data</span></span>
<span id="cb188-155"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-155" aria-hidden="true" tabindex="-1"></a>            tsne <span class="ot">&lt;-</span> <span class="fu">Rtsne</span>(train[,<span class="sc">-</span><span class="dv">1</span>], <span class="at">dims =</span> <span class="dv">2</span>, <span class="at">perplexity=</span><span class="dv">30</span>, <span class="at">verbose=</span><span class="cn">FALSE</span>, <span class="at">max_iter =</span> <span class="dv">400</span>)</span>
<span id="cb188-156"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-157"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-158"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-158" aria-hidden="true" tabindex="-1"></a>            nClusters <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">unique</span>(Cluster) )</span>
<span id="cb188-159"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-159" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(input_colorGenes) {          </span>
<span id="cb188-160"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-160" aria-hidden="true" tabindex="-1"></a>                <span class="fu">plot</span>(tsne<span class="sc">$</span>Y[,<span class="dv">1</span>], tsne<span class="sc">$</span>Y[,<span class="dv">2</span>], <span class="at">pch =</span> (<span class="dv">0</span><span class="sc">:</span>(nClusters<span class="dv">-1</span>))[Cluster], <span class="at">cex =</span> <span class="fl">1.</span>,<span class="at">col =</span> mycolors[Cluster], <span class="at">xlab=</span><span class="st">&quot;X&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Y&quot;</span>)</span>
<span id="cb188-161"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-161" aria-hidden="true" tabindex="-1"></a>                <span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>,<span class="fu">toupper</span>(letters)[<span class="dv">1</span><span class="sc">:</span>nClusters], <span class="at">pch =</span> <span class="dv">0</span><span class="sc">:</span>(nClusters<span class="dv">-1</span>), <span class="at">col=</span>mycolors, <span class="at">title=</span><span class="st">&quot;Cluster&quot;</span>  )</span>
<span id="cb188-162"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-162" aria-hidden="true" tabindex="-1"></a>            } <span class="cf">else</span></span>
<span id="cb188-163"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-163" aria-hidden="true" tabindex="-1"></a>                <span class="fu">plot</span>(tsne<span class="sc">$</span>Y[,<span class="dv">1</span>], tsne<span class="sc">$</span>Y[,<span class="dv">2</span>],  <span class="at">cex =</span> <span class="fl">1.</span>, <span class="at">xlab=</span><span class="st">&quot;X&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Y&quot;</span>)</span>
<span id="cb188-164"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-165"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-165" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb188-166"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-167"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-168"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-168" aria-hidden="true" tabindex="-1"></a><span class="co"># Main function. Find a query set of genes enriched with functional category</span></span>
<span id="cb188-169"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-169" aria-hidden="true" tabindex="-1"></a>FindOverlap <span class="ot">&lt;-</span> <span class="cf">function</span> (converted,gInfo, GO,selectOrg,minFDR, <span class="at">reduced =</span> <span class="cn">FALSE</span>) {</span>
<span id="cb188-170"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-170" aria-hidden="true" tabindex="-1"></a>    maxTerms <span class="ot">=</span><span class="dv">15</span> <span class="co"># max number of enriched terms</span></span>
<span id="cb188-171"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-171" aria-hidden="true" tabindex="-1"></a>    idNotRecognized <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;ID not recognized!&quot;</span>)</span>
<span id="cb188-172"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-172" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb188-173"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-173" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(converted) ) <span class="fu">return</span>(idNotRecognized) <span class="co"># no ID </span></span>
<span id="cb188-174"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-174" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb188-175"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-175" aria-hidden="true" tabindex="-1"></a>    <span class="co"># only coding</span></span>
<span id="cb188-176"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-176" aria-hidden="true" tabindex="-1"></a>    gInfo <span class="ot">&lt;-</span> gInfo[<span class="fu">which</span>( gInfo<span class="sc">$</span>gene_biotype <span class="sc">==</span> <span class="st">&quot;protein_coding&quot;</span>),]  </span>
<span id="cb188-177"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-177" aria-hidden="true" tabindex="-1"></a>    querySet <span class="ot">&lt;-</span> <span class="fu">intersect</span>( converted<span class="sc">$</span>IDs, gInfo[,<span class="dv">1</span>]);</span>
<span id="cb188-178"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-178" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb188-179"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-179" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(querySet) <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>(idNotRecognized )</span>
<span id="cb188-180"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-180" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb188-181"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-181" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span> <span class="fu">grep</span>(converted<span class="sc">$</span>species[<span class="dv">1</span>,<span class="dv">1</span>],gmtFiles)</span>
<span id="cb188-182"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-182" aria-hidden="true" tabindex="-1"></a>    totalGenes <span class="ot">&lt;-</span> converted<span class="sc">$</span>species[<span class="dv">1</span>,<span class="dv">7</span>]</span>
<span id="cb188-183"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-183" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb188-184"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-184" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> ) {<span class="fu">return</span>(idNotRecognized )}</span>
<span id="cb188-185"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-185" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb188-186"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-186" aria-hidden="true" tabindex="-1"></a>    <span class="co"># If selected species is not the default &quot;bestMatch&quot;, use that species directly</span></span>
<span id="cb188-187"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-187" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(selectOrg <span class="sc">!=</span> speciesChoice[[<span class="dv">1</span>]]) {  </span>
<span id="cb188-188"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-188" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">grep</span>(<span class="fu">findSpeciesById</span>(selectOrg)[<span class="dv">1</span>,<span class="dv">1</span>], gmtFiles )</span>
<span id="cb188-189"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-189" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> ) {<span class="fu">return</span>(idNotRecognized )}</span>
<span id="cb188-190"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-190" aria-hidden="true" tabindex="-1"></a>        totalGenes <span class="ot">&lt;-</span> orgInfo[<span class="fu">which</span>(orgInfo<span class="sc">$</span>id <span class="sc">==</span> <span class="fu">as.numeric</span>(selectOrg)),<span class="dv">7</span>]</span>
<span id="cb188-191"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-191" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb188-192"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-192" aria-hidden="true" tabindex="-1"></a>    pathway <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(sqlite,gmtFiles[ix],<span class="at">flags=</span>SQLITE_RO)</span>
<span id="cb188-193"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-193" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb188-194"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-194" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb188-195"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-195" aria-hidden="true" tabindex="-1"></a>    sqlQuery <span class="ot">=</span> <span class="fu">paste</span>( <span class="st">&quot; select distinct gene,pathwayID from pathway where gene IN (&#39;&quot;</span>, <span class="fu">paste</span>(querySet,<span class="at">collapse=</span><span class="st">&quot;&#39;, &#39;&quot;</span>),<span class="st">&quot;&#39;)&quot;</span> ,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb188-196"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-196" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb188-197"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-197" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(paste0(&quot;HH&quot;,GO,&quot;HH&quot;) )</span></span>
<span id="cb188-198"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-198" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb188-199"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-199" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( GO <span class="sc">!=</span> <span class="st">&quot;All&quot;</span>) sqlQuery <span class="ot">=</span> <span class="fu">paste0</span>(sqlQuery, <span class="st">&quot; AND category =&#39;&quot;</span>,GO,<span class="st">&quot;&#39;&quot;</span>)</span>
<span id="cb188-200"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-200" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>( pathway, sqlQuery  )</span>
<span id="cb188-201"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-201" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">1</span>] <span class="sc">==</span><span class="dv">0</span>) {<span class="fu">return</span>(<span class="fu">as.data.frame</span>(<span class="st">&quot;No matching species or gene ID file!&quot;</span> )) }</span>
<span id="cb188-202"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-203"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-203" aria-hidden="true" tabindex="-1"></a>    <span class="co"># given a pathway id, it finds the overlapped genes, symbol preferred</span></span>
<span id="cb188-204"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-204" aria-hidden="true" tabindex="-1"></a>    sharedGenesPrefered <span class="ot">&lt;-</span> <span class="cf">function</span>(pathwayID) {</span>
<span id="cb188-205"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-205" aria-hidden="true" tabindex="-1"></a>        tem <span class="ot">&lt;-</span> result[<span class="fu">which</span>(result[,<span class="dv">2</span>]<span class="sc">==</span> pathwayID ),<span class="dv">1</span>]</span>
<span id="cb188-206"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-206" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">match</span>(tem, converted<span class="sc">$</span>conversionTable<span class="sc">$</span>ensembl_gene_id) <span class="co"># convert back to original</span></span>
<span id="cb188-207"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-207" aria-hidden="true" tabindex="-1"></a>        tem2 <span class="ot">&lt;-</span> <span class="fu">unique</span>( converted<span class="sc">$</span>conversionTable<span class="sc">$</span>User_input[ix] )</span>
<span id="cb188-208"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-208" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">length</span>(<span class="fu">unique</span>(gInfo<span class="sc">$</span>symbol) )<span class="sc">/</span><span class="fu">dim</span>(gInfo)[<span class="dv">1</span>] <span class="sc">&gt;</span>.<span class="dv">7</span>  ) <span class="co"># if 70% genes has symbol in geneInfo</span></span>
<span id="cb188-209"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-209" aria-hidden="true" tabindex="-1"></a>        { ix <span class="ot">=</span> <span class="fu">match</span>(tem, gInfo<span class="sc">$</span>ensembl_gene_id); </span>
<span id="cb188-210"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-210" aria-hidden="true" tabindex="-1"></a>        tem2 <span class="ot">&lt;-</span> <span class="fu">unique</span>( gInfo<span class="sc">$</span>symbol[ix] )      }</span>
<span id="cb188-211"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-211" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">paste</span>( tem2 ,<span class="at">collapse=</span><span class="st">&quot; &quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>) )}</span>
<span id="cb188-212"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-212" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb188-213"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-213" aria-hidden="true" tabindex="-1"></a>    x0 <span class="ot">=</span> <span class="fu">table</span>(result<span class="sc">$</span>pathwayID)                    </span>
<span id="cb188-214"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-214" aria-hidden="true" tabindex="-1"></a>    x0 <span class="ot">=</span> <span class="fu">as.data.frame</span>( x0[<span class="fu">which</span>(x0<span class="sc">&gt;=</span>Min_overlap)] )<span class="co"># remove low overlaps</span></span>
<span id="cb188-215"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-215" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">dim</span>(x0)[<span class="dv">1</span>] <span class="sc">&lt;=</span> <span class="dv">5</span> ) <span class="fu">return</span>(idNotRecognized) <span class="co"># no data</span></span>
<span id="cb188-216"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-216" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(x0)<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;pathwayID&quot;</span>,<span class="st">&quot;overlap&quot;</span>)</span>
<span id="cb188-217"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-217" aria-hidden="true" tabindex="-1"></a>    pathwayInfo <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>( pathway, <span class="fu">paste</span>( <span class="st">&quot; select distinct id,n,Description from pathwayInfo where id IN (&#39;&quot;</span>, </span>
<span id="cb188-218"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-218" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">paste</span>(x0<span class="sc">$</span>pathwayID,<span class="at">collapse=</span><span class="st">&quot;&#39;, &#39;&quot;</span>),   <span class="st">&quot;&#39;) &quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>) )</span>
<span id="cb188-219"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-219" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb188-220"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-220" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">=</span> <span class="fu">merge</span>(x0,pathwayInfo, <span class="at">by.x=</span><span class="st">&#39;pathwayID&#39;</span>, <span class="at">by.y=</span><span class="st">&#39;id&#39;</span>)</span>
<span id="cb188-221"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-221" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb188-222"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-222" aria-hidden="true" tabindex="-1"></a>    x<span class="sc">$</span>Pval<span class="ot">=</span><span class="fu">phyper</span>(x<span class="sc">$</span>overlap<span class="dv">-1</span>,<span class="fu">length</span>(querySet),totalGenes <span class="sc">-</span> <span class="fu">length</span>(querySet),<span class="fu">as.numeric</span>(x<span class="sc">$</span>n), <span class="at">lower.tail=</span><span class="cn">FALSE</span> );</span>
<span id="cb188-223"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-223" aria-hidden="true" tabindex="-1"></a>    x<span class="sc">$</span>FDR <span class="ot">=</span> <span class="fu">p.adjust</span>(x<span class="sc">$</span>Pval,<span class="at">method=</span><span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb188-224"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-224" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> x[ <span class="fu">order</span>( x<span class="sc">$</span>FDR)  ,]  <span class="co"># sort according to FDR</span></span>
<span id="cb188-225"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-225" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb188-226"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-226" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="sc">&gt;</span> maxTerms ) x <span class="ot">=</span> x[<span class="dv">1</span><span class="sc">:</span>maxTerms,]    </span>
<span id="cb188-227"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-227" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb188-228"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-228" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">min</span>(x<span class="sc">$</span>FDR) <span class="sc">&gt;</span> minFDR) x<span class="ot">=</span><span class="fu">as.data.frame</span>(<span class="st">&quot;No significant enrichment found!&quot;</span>) <span class="cf">else</span> {</span>
<span id="cb188-229"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-229" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> x[<span class="fu">which</span>(x<span class="sc">$</span>FDR <span class="sc">&lt;</span> minFDR),] </span>
<span id="cb188-230"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-231"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-231" aria-hidden="true" tabindex="-1"></a>        x<span class="ot">=</span> <span class="fu">cbind</span>(x,<span class="fu">sapply</span>( x<span class="sc">$</span>pathwayID, sharedGenesPrefered ) )</span>
<span id="cb188-232"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-232" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(x)[<span class="dv">7</span>]<span class="ot">=</span> <span class="st">&quot;Genes&quot;</span></span>
<span id="cb188-233"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-233" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> <span class="fu">subset</span>(x,<span class="at">select =</span> <span class="fu">c</span>(FDR,overlap,n,description,Genes) )</span>
<span id="cb188-234"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-234" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(x) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Corrected P value (FDR)&quot;</span>, <span class="st">&quot;Genes in list&quot;</span>, <span class="st">&quot;Total genes in category&quot;</span>,<span class="st">&quot;Functional Category&quot;</span>,<span class="st">&quot;Genes&quot;</span>  )</span>
<span id="cb188-235"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-235" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb188-236"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-236" aria-hidden="true" tabindex="-1"></a>        <span class="co"># remove redudant gene sets</span></span>
<span id="cb188-237"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-237" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(reduced <span class="sc">!=</span> <span class="cn">FALSE</span> ){  <span class="co"># reduced=FALSE no filtering,  reduced = 0.9 filter sets overlap with 90%</span></span>
<span id="cb188-238"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-238" aria-hidden="true" tabindex="-1"></a>            n<span class="ot">=</span>  <span class="fu">nrow</span>(x)</span>
<span id="cb188-239"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-239" aria-hidden="true" tabindex="-1"></a>            tem<span class="ot">=</span><span class="fu">rep</span>(<span class="cn">TRUE</span>,n )</span>
<span id="cb188-240"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-240" aria-hidden="true" tabindex="-1"></a>            geneLists <span class="ot">=</span> <span class="fu">lapply</span>(x<span class="sc">$</span>Genes, <span class="cf">function</span>(y) <span class="fu">unlist</span>( <span class="fu">strsplit</span>(<span class="fu">as.character</span>(y),<span class="st">&quot; &quot;</span> )   ) )</span>
<span id="cb188-241"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-241" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span>n)</span>
<span id="cb188-242"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-242" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span>( j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(i<span class="dv">-1</span>) ) { </span>
<span id="cb188-243"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-243" aria-hidden="true" tabindex="-1"></a>                  <span class="cf">if</span>(tem[j]) { <span class="co"># skip if this one is already removed</span></span>
<span id="cb188-244"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-244" aria-hidden="true" tabindex="-1"></a>                      commonGenes <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">intersect</span>(geneLists[i] ,geneLists[j] ) )</span>
<span id="cb188-245"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-245" aria-hidden="true" tabindex="-1"></a>                      <span class="cf">if</span>( commonGenes<span class="sc">/</span> <span class="fu">length</span>(geneLists[j] ) <span class="sc">&gt;</span> reduced )</span>
<span id="cb188-246"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-246" aria-hidden="true" tabindex="-1"></a>                        tem[i] <span class="ot">=</span> <span class="cn">FALSE</span>  </span>
<span id="cb188-247"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-247" aria-hidden="true" tabindex="-1"></a>                  }         </span>
<span id="cb188-248"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-248" aria-hidden="true" tabindex="-1"></a>                }                               </span>
<span id="cb188-249"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-249" aria-hidden="true" tabindex="-1"></a>            x <span class="ot">&lt;-</span> x[<span class="fu">which</span>(tem),]     </span>
<span id="cb188-250"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-250" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb188-251"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-251" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb188-252"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-253"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-253" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb188-254"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-254" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb188-255"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-255" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dbDisconnect</span>(pathway)</span>
<span id="cb188-256"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-256" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(x )</span>
<span id="cb188-257"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-257" aria-hidden="true" tabindex="-1"></a>} </span>
<span id="cb188-258"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-258" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb188-259"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-259" aria-hidden="true" tabindex="-1"></a><span class="co"># Given a gene set, finds significant overlaps with a gene set database  object </span></span>
<span id="cb188-260"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-260" aria-hidden="true" tabindex="-1"></a>findOverlapGMT <span class="ot">&lt;-</span> <span class="cf">function</span> ( query, geneSet, <span class="at">minFDR=</span>.<span class="dv">2</span> ,<span class="at">minSize=</span><span class="dv">2</span>,<span class="at">maxSize=</span><span class="dv">10000</span> ){ </span>
<span id="cb188-261"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-261" aria-hidden="true" tabindex="-1"></a>    <span class="co">#geneSets &lt;- readGMT(&quot;exampleData/MousePath_TF_gmt.gmt&quot;)</span></span>
<span id="cb188-262"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-262" aria-hidden="true" tabindex="-1"></a>    <span class="co">#query &lt;-  geneSets[[&#39;TF_MM_FRIARD_C-REL&#39;]] </span></span>
<span id="cb188-263"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-263" aria-hidden="true" tabindex="-1"></a>    <span class="co">#query &lt;- query[1:60]</span></span>
<span id="cb188-264"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-264" aria-hidden="true" tabindex="-1"></a>    total_elements <span class="ot">=</span> <span class="dv">30000</span></span>
<span id="cb188-265"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-265" aria-hidden="true" tabindex="-1"></a>    Min_overlap <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb188-266"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-266" aria-hidden="true" tabindex="-1"></a>    maxTerms <span class="ot">=</span><span class="dv">10</span> <span class="co"># max number of enriched terms</span></span>
<span id="cb188-267"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-267" aria-hidden="true" tabindex="-1"></a>    noSig <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No significant enrichment found!&quot;</span>)</span>
<span id="cb188-268"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-268" aria-hidden="true" tabindex="-1"></a>    query <span class="ot">&lt;-</span> <span class="fu">cleanGeneSet</span>(query)   <span class="co"># convert to upper case, unique()</span></span>
<span id="cb188-269"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-270"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-270" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(query) <span class="sc">&lt;=</span><span class="dv">2</span>) <span class="fu">return</span>(noSig)</span>
<span id="cb188-271"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-271" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(geneSet) <span class="sc">&lt;</span><span class="dv">1</span>) <span class="fu">return</span>(noSig)</span>
<span id="cb188-272"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-272" aria-hidden="true" tabindex="-1"></a>      geneSet <span class="ot">&lt;-</span> geneSet[<span class="fu">which</span>(<span class="fu">sapply</span>(geneSet,length) <span class="sc">&gt;</span> minSize)]  <span class="co"># gene sets smaller than 1 is ignored!!!</span></span>
<span id="cb188-273"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-273" aria-hidden="true" tabindex="-1"></a>      geneSet <span class="ot">&lt;-</span> geneSet[<span class="fu">which</span>(<span class="fu">sapply</span>(geneSet,length) <span class="sc">&lt;</span> maxSize)]  <span class="co"># gene sets smaller than 1 is ignored!!!</span></span>
<span id="cb188-274"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-274" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">unlist</span>( <span class="fu">lapply</span>(geneSet, <span class="cf">function</span>(x) <span class="fu">length</span>( <span class="fu">intersect</span> (query, x) ) ) )</span>
<span id="cb188-275"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-275" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">unlist</span>( <span class="fu">lapply</span>(geneSet, length) ), result )</span>
<span id="cb188-276"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-276" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> result[ <span class="fu">which</span>(result[,<span class="dv">2</span>]<span class="sc">&gt;</span>Min_overlap), ,drop<span class="ot">=</span>F]</span>
<span id="cb188-277"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-277" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">dim</span>(result)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>( noSig)</span>
<span id="cb188-278"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-278" aria-hidden="true" tabindex="-1"></a>    xx <span class="ot">&lt;-</span> result[,<span class="dv">2</span>]</span>
<span id="cb188-279"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-279" aria-hidden="true" tabindex="-1"></a>    mm <span class="ot">&lt;-</span> <span class="fu">length</span>(query)</span>
<span id="cb188-280"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-280" aria-hidden="true" tabindex="-1"></a>    nn <span class="ot">&lt;-</span> total_elements <span class="sc">-</span> mm</span>
<span id="cb188-281"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-281" aria-hidden="true" tabindex="-1"></a>    kk <span class="ot">&lt;-</span> result[,<span class="dv">1</span>]</span>
<span id="cb188-282"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-282" aria-hidden="true" tabindex="-1"></a>    Pval_enrich<span class="ot">=</span><span class="fu">phyper</span>(xx<span class="dv">-1</span>,mm,nn,kk, <span class="at">lower.tail=</span><span class="cn">FALSE</span> );</span>
<span id="cb188-283"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-283" aria-hidden="true" tabindex="-1"></a>    FDR <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(Pval_enrich,<span class="at">method=</span><span class="st">&quot;fdr&quot;</span>,<span class="at">n=</span><span class="fu">length</span>(geneSet) )</span>
<span id="cb188-284"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-284" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(FDR,result))</span>
<span id="cb188-285"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-285" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> result[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>)]</span>
<span id="cb188-286"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-286" aria-hidden="true" tabindex="-1"></a>    result<span class="sc">$</span>pathway <span class="ot">=</span> <span class="fu">rownames</span>(result)</span>
<span id="cb188-287"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-287" aria-hidden="true" tabindex="-1"></a>    result<span class="sc">$</span>Genes <span class="ot">=</span> <span class="st">&quot;&quot;</span>  <span class="co"># place holder just </span></span>
<span id="cb188-288"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-288" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(result)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Corrected P value (FDR)&quot;</span>, <span class="st">&quot;Genes in list&quot;</span>, <span class="st">&quot;Total genes in category&quot;</span>,<span class="st">&quot;Functional Category&quot;</span>,<span class="st">&quot;Genes&quot;</span>  )</span>
<span id="cb188-289"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-289" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> result[ <span class="fu">which</span>( result[,<span class="dv">1</span>] <span class="sc">&lt;</span> minFDR),,drop<span class="ot">=</span>F]</span>
<span id="cb188-290"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-290" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>( result)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>(noSig) </span>
<span id="cb188-291"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-291" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">min</span>(FDR) <span class="sc">&gt;</span> minFDR) <span class="fu">return</span>(noSig) </span>
<span id="cb188-292"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-292" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> result[<span class="fu">order</span>(result[,<span class="dv">1</span>] ),]</span>
<span id="cb188-293"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-293" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">dim</span>(result)[<span class="dv">1</span>] <span class="sc">&gt;</span> maxTerms ) result <span class="ot">&lt;-</span> result[<span class="dv">1</span><span class="sc">:</span>maxTerms,]</span>
<span id="cb188-294"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-295"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-295" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( result)</span>
<span id="cb188-296"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-296" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb188-297"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-298"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-298" aria-hidden="true" tabindex="-1"></a>KmeansGO <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb188-299"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-299" aria-hidden="true" tabindex="-1"></a>        pp<span class="ot">=</span><span class="dv">0</span></span>
<span id="cb188-300"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-300" aria-hidden="true" tabindex="-1"></a>        minFDR <span class="ot">=</span> <span class="fl">0.01</span></span>
<span id="cb188-301"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-302"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-302" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>input_nClusters) {</span>
<span id="cb188-303"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-303" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-304"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-304" aria-hidden="true" tabindex="-1"></a>            query <span class="ot">=</span> <span class="fu">rownames</span>(Kmeans.out<span class="sc">$</span>x)[<span class="fu">which</span>(Kmeans.out<span class="sc">$</span>bar <span class="sc">==</span> i)]</span>
<span id="cb188-305"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-306"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-306" aria-hidden="true" tabindex="-1"></a>                result <span class="ot">&lt;-</span> <span class="fu">findOverlapGMT</span>( query, GeneSets.out,<span class="dv">1</span>) </span>
<span id="cb188-307"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-308"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-308" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">2</span>] <span class="sc">==</span><span class="dv">1</span>) <span class="cf">next</span>;   <span class="co"># result could be NULL</span></span>
<span id="cb188-309"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-309" aria-hidden="true" tabindex="-1"></a>            result<span class="sc">$</span>Genes <span class="ot">=</span> <span class="fu">toupper</span>(letters)[i] </span>
<span id="cb188-310"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-310" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (pp<span class="sc">==</span><span class="dv">0</span> ) { results <span class="ot">&lt;-</span> result; pp <span class="ot">&lt;-</span> <span class="dv">1</span>;</span>
<span id="cb188-311"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-311" aria-hidden="true" tabindex="-1"></a>            } <span class="cf">else</span> {</span>
<span id="cb188-312"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-312" aria-hidden="true" tabindex="-1"></a>                results <span class="ot">&lt;-</span> <span class="fu">rbind</span>(results,result)</span>
<span id="cb188-313"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-313" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb188-314"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-314" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb188-315"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-316"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-316" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(pp <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>( <span class="fu">as.data.frame</span>(<span class="st">&quot;No enrichment found.&quot;</span>))</span>
<span id="cb188-317"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-317" aria-hidden="true" tabindex="-1"></a>        results<span class="ot">=</span> results[,<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">4</span>)]</span>
<span id="cb188-318"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-318" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(results)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Cluster&quot;</span>,<span class="st">&quot;FDR&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;Pathways&quot;</span>)</span>
<span id="cb188-319"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-319" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">min</span>(results<span class="sc">$</span>FDR) <span class="sc">&gt;</span> minFDR ) results <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No signficant enrichment found.&quot;</span>) <span class="cf">else</span></span>
<span id="cb188-320"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-320" aria-hidden="true" tabindex="-1"></a>        results <span class="ot">=</span> results[<span class="fu">which</span>(results<span class="sc">$</span>FDR <span class="sc">&lt;</span> minFDR),]</span>
<span id="cb188-321"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-321" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb188-322"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-322" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">is.null</span>(results) )  <span class="fu">return</span> ( <span class="fu">as.matrix</span>(<span class="st">&quot;No significant enrichment.&quot;</span>) )  </span>
<span id="cb188-323"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-323" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">class</span>(results) <span class="sc">!=</span> <span class="st">&quot;data.frame&quot;</span>)  <span class="fu">return</span> ( <span class="fu">as.matrix</span>(<span class="st">&quot;No significant enrichment.&quot;</span>) )</span>
<span id="cb188-324"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-324" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>(results)[<span class="dv">2</span>] <span class="sc">==</span><span class="dv">1</span>)  <span class="fu">return</span> ( <span class="fu">as.matrix</span>(<span class="st">&quot;No significant enrichment.&quot;</span>) )</span>
<span id="cb188-325"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-325" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(results)[<span class="dv">2</span>] <span class="ot">=</span> <span class="st">&quot;adj.Pval&quot;</span></span>
<span id="cb188-326"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-326" aria-hidden="true" tabindex="-1"></a>    results<span class="sc">$</span>Genes <span class="ot">&lt;-</span> <span class="fu">as.character</span>(results<span class="sc">$</span>Genes)</span>
<span id="cb188-327"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-327" aria-hidden="true" tabindex="-1"></a>    results<span class="sc">$</span>Cluster[<span class="fu">which</span>( <span class="fu">duplicated</span>(results<span class="sc">$</span>Cluster) ) ] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb188-328"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-328" aria-hidden="true" tabindex="-1"></a>    results</span>
<span id="cb188-329"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb188-329" aria-hidden="true" tabindex="-1"></a>  }</span></code></pre></div>
</div>
<div id="differential-expression-1" class="section level2" number="6.8">
<h2><span class="header-section-number">6.8</span> Differential expression</h2>
<div class="sourceCode" id="cb189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb189-1"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb189-2"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Differential expression</span></span>
<span id="cb189-3"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-3" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb189-4"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-4" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb189-5"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Differential expression using LIMMA </span></span>
<span id="cb189-6"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-6" aria-hidden="true" tabindex="-1"></a>DEG.limma <span class="ot">&lt;-</span> <span class="cf">function</span> (x, <span class="at">maxP_limma=</span>.<span class="dv">1</span>, <span class="at">minFC_limma=</span><span class="dv">2</span>, rawCounts,countsDEGMethods,priorCounts, dataFormat, <span class="at">selectedComparisons=</span><span class="cn">NULL</span>, <span class="at">sampleInfo =</span> <span class="cn">NULL</span>,<span class="at">modelFactors=</span><span class="cn">NULL</span>, <span class="at">blockFactor =</span> <span class="cn">NULL</span>){</span>
<span id="cb189-7"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(limma,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># Differential expression</span></span>
<span id="cb189-8"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(statmod,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb189-9"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-9" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-10"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># many different situations: 1. just use sample names 2. just one factor  3. two factors no interaction</span></span>
<span id="cb189-11"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 4. two factors with interaction   5. block factor </span></span>
<span id="cb189-12"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-13"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-13" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-14"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-14" aria-hidden="true" tabindex="-1"></a>    topGenes <span class="ot">=</span> <span class="fu">list</span>();  limmaTrend <span class="ot">=</span> <span class="cn">FALSE</span></span>
<span id="cb189-15"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-15" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( dataFormat <span class="sc">==</span> <span class="dv">2</span>) {   <span class="co"># if normalized data</span></span>
<span id="cb189-16"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-16" aria-hidden="true" tabindex="-1"></a>        eset <span class="ot">=</span> <span class="fu">new</span>(<span class="st">&quot;ExpressionSet&quot;</span>, <span class="at">exprs=</span><span class="fu">as.matrix</span>(x)) } <span class="cf">else</span> { <span class="co"># counts data</span></span>
<span id="cb189-17"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-17" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (countsDEGMethods <span class="sc">==</span> <span class="dv">1</span> ) { <span class="co"># limma-trend method selected for counts data</span></span>
<span id="cb189-18"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-18" aria-hidden="true" tabindex="-1"></a>                <span class="co">#dge &lt;- DGEList(counts=rawCounts);</span></span>
<span id="cb189-19"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-19" aria-hidden="true" tabindex="-1"></a>                <span class="co">#dge &lt;- calcNormFactors(dge, method = &quot;TMM&quot;)</span></span>
<span id="cb189-20"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-20" aria-hidden="true" tabindex="-1"></a>                <span class="co">#eset &lt;- cpm(dge, log=TRUE, prior.count=priorCounts)</span></span>
<span id="cb189-21"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-21" aria-hidden="true" tabindex="-1"></a>                eset <span class="ot">=</span> <span class="fu">new</span>(<span class="st">&quot;ExpressionSet&quot;</span>, <span class="at">exprs=</span><span class="fu">as.matrix</span>(x))  <span class="co"># use transformed data for limma </span></span>
<span id="cb189-22"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-22" aria-hidden="true" tabindex="-1"></a>                limmaTrend <span class="ot">=</span> <span class="cn">TRUE</span></span>
<span id="cb189-23"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-23" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb189-24"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-24" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-25"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-26"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-26" aria-hidden="true" tabindex="-1"></a>    groups <span class="ot">=</span> <span class="fu">colnames</span>(x)</span>
<span id="cb189-27"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-27" aria-hidden="true" tabindex="-1"></a>    groups <span class="ot">=</span> <span class="fu">detectGroups</span>( groups)</span>
<span id="cb189-28"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-28" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">=</span>  <span class="fu">unique</span>(groups)  </span>
<span id="cb189-29"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-29" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-30"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-30" aria-hidden="true" tabindex="-1"></a>    <span class="co"># check for replicates, removes samples without replicates</span></span>
<span id="cb189-31"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-31" aria-hidden="true" tabindex="-1"></a>    reps <span class="ot">=</span> <span class="fu">as.matrix</span>(<span class="fu">table</span>(groups)) <span class="co"># number of replicates per biological sample</span></span>
<span id="cb189-32"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-32" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> ( <span class="fu">sum</span>( reps[,<span class="dv">1</span>] <span class="sc">&gt;=</span> <span class="dv">2</span>) <span class="sc">&lt;</span><span class="dv">2</span>    ) <span class="co"># if less than 2 samples with replicates</span></span>
<span id="cb189-33"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">results=</span> <span class="cn">NULL</span>, <span class="at">comparisons =</span> <span class="cn">NULL</span>, <span class="at">Exp.type=</span><span class="st">&quot;Failed to parse sample names to define groups. </span></span>
<span id="cb189-34"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-34" aria-hidden="true" tabindex="-1"></a><span class="st">        Cannot perform DEGs and pathway analysis. Please double check column names! Use WT_Rep1, WT_Rep2 etc. &quot;</span>, <span class="at">topGenes=</span><span class="cn">NULL</span>)) </span>
<span id="cb189-35"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-35" aria-hidden="true" tabindex="-1"></a>    <span class="co"># remove samples without replicates</span></span>
<span id="cb189-36"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-36" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">rownames</span>(reps)[<span class="fu">which</span>(reps[,<span class="dv">1</span>] <span class="sc">&gt;</span><span class="dv">1</span>)]</span>
<span id="cb189-37"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-37" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">&lt;-</span> <span class="fu">which</span>( groups <span class="sc">%in%</span> g)  </span>
<span id="cb189-38"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-38" aria-hidden="true" tabindex="-1"></a>    groups <span class="ot">&lt;-</span> groups[ix]   </span>
<span id="cb189-39"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-39" aria-hidden="true" tabindex="-1"></a>    x<span class="ot">&lt;-</span> x[,ix]; rawCounts <span class="ot">&lt;-</span> rawCounts[,ix] </span>
<span id="cb189-40"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-40" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-41"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-41" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(g) <span class="sc">==</span><span class="dv">2</span> ) { </span>
<span id="cb189-42"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-42" aria-hidden="true" tabindex="-1"></a>        g<span class="ot">=</span> <span class="fu">unique</span>(groups)</span>
<span id="cb189-43"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-43" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">&lt;-</span>  <span class="fu">paste</span>(g[<span class="dv">2</span>],<span class="st">&quot;-&quot;</span>,g[<span class="dv">1</span>],<span class="at">sep=</span><span class="st">&quot;&quot;</span>)  <span class="co"># &quot;Mutant-WT&quot;</span></span>
<span id="cb189-44"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-44" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-45"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-45" aria-hidden="true" tabindex="-1"></a>        <span class="co"># no sample file, but user selected comparisons using column names</span></span>
<span id="cb189-46"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-46" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">is.null</span>(modelFactors) <span class="sc">&amp;</span> <span class="fu">length</span>( selectedComparisons) <span class="sc">&gt;</span><span class="dv">0</span>  )  </span>
<span id="cb189-47"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-47" aria-hidden="true" tabindex="-1"></a>            comparisons <span class="ot">=</span> selectedComparisons</span>
<span id="cb189-48"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-48" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-49"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-49" aria-hidden="true" tabindex="-1"></a>        design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span><span class="dv">0</span><span class="sc">+</span>groups)</span>
<span id="cb189-50"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-50" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> g</span>
<span id="cb189-51"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-51" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-52"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-52" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>(rawCounts) <span class="sc">&amp;&amp;</span> countsDEGMethods <span class="sc">==</span> <span class="dv">2</span>) {  <span class="co"># voom</span></span>
<span id="cb189-53"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-53" aria-hidden="true" tabindex="-1"></a>            dge <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts=</span>rawCounts);</span>
<span id="cb189-54"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-54" aria-hidden="true" tabindex="-1"></a>            dge <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(dge, <span class="at">method =</span> <span class="st">&quot;TMM&quot;</span>)  <span class="co"># normalization</span></span>
<span id="cb189-55"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-55" aria-hidden="true" tabindex="-1"></a>            v <span class="ot">&lt;-</span> <span class="fu">voom</span>(dge, design); fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(v, design) } <span class="cf">else</span> </span>
<span id="cb189-56"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-56" aria-hidden="true" tabindex="-1"></a>            fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(eset, design)      <span class="co"># regular limma</span></span>
<span id="cb189-57"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-57" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-58"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-58" aria-hidden="true" tabindex="-1"></a>        cont.matrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons, <span class="at">levels=</span>design)</span>
<span id="cb189-59"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-59" aria-hidden="true" tabindex="-1"></a>        fit2 <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit, cont.matrix)</span>
<span id="cb189-60"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-60" aria-hidden="true" tabindex="-1"></a>        fit2 <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit2, <span class="at">trend=</span>limmaTrend)</span>
<span id="cb189-61"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-62"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-62" aria-hidden="true" tabindex="-1"></a>        <span class="co"># calls differential gene expression 1 for up, -1 for down</span></span>
<span id="cb189-63"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-63" aria-hidden="true" tabindex="-1"></a>        results <span class="ot">&lt;-</span> <span class="fu">decideTests</span>(fit2, <span class="at">p.value=</span>maxP_limma, <span class="at">lfc=</span><span class="fu">log2</span>(minFC_limma) )</span>
<span id="cb189-64"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-64" aria-hidden="true" tabindex="-1"></a>        <span class="co">#vennDiagram(results,circle.col=rainbow(5))</span></span>
<span id="cb189-65"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-65" aria-hidden="true" tabindex="-1"></a>        topGenes1 <span class="ot">=</span><span class="fu">topTable</span>(fit2, <span class="at">number =</span> <span class="fl">1e12</span>,<span class="at">sort.by=</span><span class="st">&quot;M&quot;</span> )</span>
<span id="cb189-66"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-66" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">dim</span>(topGenes1)[<span class="dv">1</span>] <span class="sc">!=</span> <span class="dv">0</span>) {</span>
<span id="cb189-67"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-67" aria-hidden="true" tabindex="-1"></a>        topGenes1 <span class="ot">=</span> topGenes1[,<span class="fu">c</span>(<span class="st">&#39;logFC&#39;</span>,<span class="st">&#39;adj.P.Val&#39;</span>)] </span>
<span id="cb189-68"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-68" aria-hidden="true" tabindex="-1"></a>        <span class="co"># topGenes1[,1] &lt;-  -1* topGenes1[,1] # reverse direction</span></span>
<span id="cb189-69"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-69" aria-hidden="true" tabindex="-1"></a>        topGenes[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> topGenes1 }</span>
<span id="cb189-70"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-70" aria-hidden="true" tabindex="-1"></a>        <span class="co"># log fold change is actually substract of means. So if the data is natral log transformed, it shoudl be natral log.</span></span>
<span id="cb189-71"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-71" aria-hidden="true" tabindex="-1"></a>        Exp.type <span class="ot">=</span> <span class="st">&quot;2 sample groups.&quot;</span></span>
<span id="cb189-72"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-72" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-73"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-73" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-74"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-74" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(g) <span class="sc">&gt;</span> <span class="dv">2</span> ) { <span class="co"># more than two sample groups</span></span>
<span id="cb189-75"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-75" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-76"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-76" aria-hidden="true" tabindex="-1"></a>        design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span><span class="sc">+</span><span class="fu">factor</span>(groups))</span>
<span id="cb189-77"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-77" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;.*)&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(design))</span>
<span id="cb189-78"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-78" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-79"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-79" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>(rawCounts) <span class="sc">&amp;&amp;</span> countsDEGMethods <span class="sc">==</span> <span class="dv">2</span>) {  <span class="co"># voom</span></span>
<span id="cb189-80"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-80" aria-hidden="true" tabindex="-1"></a>            v <span class="ot">&lt;-</span> <span class="fu">voom</span>(rawCounts, design); </span>
<span id="cb189-81"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-81" aria-hidden="true" tabindex="-1"></a>            fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(v, design) </span>
<span id="cb189-82"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-82" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> </span>
<span id="cb189-83"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-83" aria-hidden="true" tabindex="-1"></a>            fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(eset, design)</span>
<span id="cb189-84"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-84" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-85"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-85" aria-hidden="true" tabindex="-1"></a>        fit <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit, <span class="at">trend=</span>limmaTrend)</span>
<span id="cb189-86"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-86" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-87"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-87" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">=</span> <span class="st">&quot;&quot;</span></span>
<span id="cb189-88"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-88" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(g)<span class="sc">-</span><span class="dv">1</span>) )</span>
<span id="cb189-89"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-89" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> (j <span class="cf">in</span> (i<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(g)) </span>
<span id="cb189-90"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-90" aria-hidden="true" tabindex="-1"></a>            comparisons <span class="ot">=</span> <span class="fu">c</span>(comparisons,<span class="fu">paste</span>(g[j],<span class="st">&quot;-&quot;</span>,g[i],<span class="at">sep=</span><span class="st">&quot;&quot;</span> ) )</span>
<span id="cb189-91"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-91" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">&lt;-</span> comparisons[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb189-92"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-93"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-93" aria-hidden="true" tabindex="-1"></a>        <span class="co"># no sample file, but user selected comparisons using column names</span></span>
<span id="cb189-94"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-94" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">is.null</span>(modelFactors) <span class="sc">&amp;</span> <span class="fu">length</span>( selectedComparisons) <span class="sc">&gt;</span><span class="dv">0</span>  )  </span>
<span id="cb189-95"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-95" aria-hidden="true" tabindex="-1"></a>            comparisons <span class="ot">=</span> selectedComparisons</span>
<span id="cb189-96"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-96" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-97"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-97" aria-hidden="true" tabindex="-1"></a>        contrast1 <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons[<span class="dv">1</span>], <span class="at">levels=</span>design)</span>
<span id="cb189-98"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-98" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">length</span>(comparisons)<span class="sc">&gt;</span><span class="dv">1</span> )</span>
<span id="cb189-99"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-99" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>( kk <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(comparisons) )</span>
<span id="cb189-100"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-100" aria-hidden="true" tabindex="-1"></a>                contrast1<span class="ot">&lt;-</span>  <span class="fu">cbind</span>(contrast1,<span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons[kk], <span class="at">levels=</span>design)   )</span>
<span id="cb189-101"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-101" aria-hidden="true" tabindex="-1"></a>        Exp.type <span class="ot">=</span> <span class="fu">paste</span>(<span class="fu">length</span>(g),<span class="st">&quot; sample groups detected.&quot;</span>)   </span>
<span id="cb189-102"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-102" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-103"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-103" aria-hidden="true" tabindex="-1"></a>        <span class="co"># if factorial design 2x2, 2x3, 3x5 etc.</span></span>
<span id="cb189-104"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-104" aria-hidden="true" tabindex="-1"></a>            <span class="co"># all samples must be something like WT_control_rep1</span></span>
<span id="cb189-105"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-105" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> ( <span class="fu">sum</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(g,<span class="st">&quot;_&quot;</span>),length) <span class="sc">==</span> <span class="dv">2</span> ) <span class="sc">==</span> <span class="fu">length</span>(g) ) {</span>
<span id="cb189-106"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-106" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-107"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-107" aria-hidden="true" tabindex="-1"></a>            <span class="co">#comparisons</span></span>
<span id="cb189-108"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-108" aria-hidden="true" tabindex="-1"></a>            comparisons <span class="ot">=</span> <span class="st">&quot;&quot;</span></span>
<span id="cb189-109"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-109" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(g)<span class="sc">-</span><span class="dv">1</span>) )</span>
<span id="cb189-110"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-110" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span> (j <span class="cf">in</span> (i<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(g)) </span>
<span id="cb189-111"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-111" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>( <span class="fu">strsplit</span>(g[i],<span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>] <span class="sc">==</span> <span class="fu">strsplit</span>(g[j],<span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]<span class="sc">|</span> <span class="fu">strsplit</span>(g[i],<span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>] <span class="sc">==</span> <span class="fu">strsplit</span>(g[j],<span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>]) <span class="co"># only compare WT_control vs. WT_treatment</span></span>
<span id="cb189-112"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-112" aria-hidden="true" tabindex="-1"></a>                    comparisons <span class="ot">=</span> <span class="fu">c</span>(comparisons,<span class="fu">paste</span>(g[j],<span class="st">&quot;-&quot;</span>,g[i],<span class="at">sep=</span><span class="st">&quot;&quot;</span> ) )</span>
<span id="cb189-113"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-113" aria-hidden="true" tabindex="-1"></a>            comparisons <span class="ot">&lt;-</span> comparisons[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb189-114"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-115"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-115" aria-hidden="true" tabindex="-1"></a>            <span class="co">#factors genotype treatment levels</span></span>
<span id="cb189-116"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-116" aria-hidden="true" tabindex="-1"></a>            extract_treatment <span class="ot">&lt;-</span> <span class="cf">function</span> (x) <span class="fu">paste</span>( <span class="fu">gsub</span>( <span class="st">&quot;.*_&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">unlist</span>( <span class="fu">strsplit</span>(x,<span class="st">&quot;-&quot;</span>)) ), <span class="at">collapse=</span><span class="st">&quot;-&quot;</span>)</span>
<span id="cb189-117"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-117" aria-hidden="true" tabindex="-1"></a>            extract_genotype <span class="ot">&lt;-</span> <span class="cf">function</span> (x) <span class="fu">gsub</span>( <span class="st">&quot;_.*&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">unlist</span>( <span class="fu">strsplit</span>(x,<span class="st">&quot;-&quot;</span>)) )[<span class="dv">1</span>]</span>
<span id="cb189-118"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-118" aria-hidden="true" tabindex="-1"></a>            extract_treatment_counting <span class="ot">&lt;-</span> <span class="fu">unique</span>( <span class="fu">gsub</span>( <span class="st">&quot;.*_&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">unlist</span>( <span class="fu">strsplit</span>(g,<span class="st">&quot;-&quot;</span>)) ))</span>
<span id="cb189-119"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-119" aria-hidden="true" tabindex="-1"></a>            treatments <span class="ot">=</span> <span class="fu">sapply</span>(comparisons, extract_treatment)</span>
<span id="cb189-120"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-120" aria-hidden="true" tabindex="-1"></a>            genotypes <span class="ot">=</span> <span class="fu">sapply</span>(comparisons, extract_genotype)</span>
<span id="cb189-121"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-121" aria-hidden="true" tabindex="-1"></a>            Exp.type <span class="ot">=</span> <span class="fu">paste</span>( Exp.type, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">Factorial design:&quot;</span>,<span class="fu">length</span>(<span class="fu">unique</span>(genotypes)),<span class="st">&quot;X&quot;</span>, <span class="fu">length</span>( extract_treatment_counting ), <span class="at">sep=</span><span class="st">&quot;&quot;</span> )</span>
<span id="cb189-122"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-123"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-123" aria-hidden="true" tabindex="-1"></a>            <span class="co"># pairwise contrasts</span></span>
<span id="cb189-124"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-124" aria-hidden="true" tabindex="-1"></a>            contrast1 <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons[<span class="dv">1</span>], <span class="at">levels=</span>design)</span>
<span id="cb189-125"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-125" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>( kk <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(comparisons) )</span>
<span id="cb189-126"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-126" aria-hidden="true" tabindex="-1"></a>                contrast1<span class="ot">&lt;-</span>  <span class="fu">cbind</span>(contrast1,<span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons[kk], <span class="at">levels=</span>design)   )</span>
<span id="cb189-127"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-127" aria-hidden="true" tabindex="-1"></a>            contrast.names <span class="ot">=</span> <span class="fu">colnames</span>(contrast1)</span>
<span id="cb189-128"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-129"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-129" aria-hidden="true" tabindex="-1"></a>            <span class="co"># interaction contrasts</span></span>
<span id="cb189-130"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-130" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> ( kk <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(comparisons)<span class="sc">-</span><span class="dv">1</span>) ) {</span>
<span id="cb189-131"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-131" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>( kp <span class="cf">in</span> (kk<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(comparisons)) </span>
<span id="cb189-132"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-132" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>( treatments[kp]<span class="sc">==</span> treatments[kk] ) </span>
<span id="cb189-133"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-133" aria-hidden="true" tabindex="-1"></a>                {  </span>
<span id="cb189-134"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-134" aria-hidden="true" tabindex="-1"></a>                    contrast1 <span class="ot">=</span> <span class="fu">cbind</span>(contrast1, contrast1[,kp]<span class="sc">-</span> contrast1[,kk] )</span>
<span id="cb189-135"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-135" aria-hidden="true" tabindex="-1"></a>                    contrast.names <span class="ot">=</span> <span class="fu">c</span>(contrast.names, <span class="fu">paste</span>(<span class="st">&quot;I:&quot;</span>,  genotypes[kp], <span class="st">&quot;-&quot;</span>, genotypes[kk],<span class="st">&quot;(&quot;</span>,<span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;.vs.&quot;</span>,treatments[kp]),<span class="st">&quot;)&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span> ) )</span>
<span id="cb189-136"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-136" aria-hidden="true" tabindex="-1"></a>                }   </span>
<span id="cb189-137"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-137" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb189-138"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-138" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(contrast1)<span class="ot">=</span>contrast.names</span>
<span id="cb189-139"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-139" aria-hidden="true" tabindex="-1"></a>            comparisons <span class="ot">=</span> contrast.names</span>
<span id="cb189-140"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-141"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-141" aria-hidden="true" tabindex="-1"></a>        } <span class="co"># if interaction terms    </span></span>
<span id="cb189-142"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-142" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-143"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-144"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-144" aria-hidden="true" tabindex="-1"></a>        <span class="co"># if sample information is uploaded and user selected factors and comparisons</span></span>
<span id="cb189-145"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-145" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>(modelFactors) <span class="sc">&amp;</span> <span class="fu">length</span>( selectedComparisons) <span class="sc">&gt;</span><span class="dv">0</span>  ) {</span>
<span id="cb189-146"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-146" aria-hidden="true" tabindex="-1"></a>            Exp.type <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;Model: ~&quot;</span>, <span class="fu">paste</span>(modelFactors,<span class="at">collapse=</span><span class="st">&quot; + &quot;</span>) )</span>
<span id="cb189-147"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-147" aria-hidden="true" tabindex="-1"></a>            interactionTerm <span class="ot">=</span> <span class="cn">FALSE</span> <span class="co"># default value to be re-write if needed</span></span>
<span id="cb189-148"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-148" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-149"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-149" aria-hidden="true" tabindex="-1"></a>            <span class="co"># model factors that does not contain interaction terms</span></span>
<span id="cb189-150"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-150" aria-hidden="true" tabindex="-1"></a>            <span class="co"># modelFactors &quot;genotype&quot;, &quot;condition&quot;, &quot;genotype:condition&quot;</span></span>
<span id="cb189-151"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-151" aria-hidden="true" tabindex="-1"></a>            keyModelFactors <span class="ot">=</span> modelFactors[ <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;:&quot;</span>,modelFactors) ]</span>
<span id="cb189-152"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-152" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-153"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-153" aria-hidden="true" tabindex="-1"></a>            <span class="co"># &quot;genotype: control vs. mutant&quot;</span></span>
<span id="cb189-154"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-154" aria-hidden="true" tabindex="-1"></a>            factorsVector<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;:.*&quot;</span>,<span class="st">&quot;&quot;</span>,selectedComparisons) <span class="co"># corresponding factors for each comparison</span></span>
<span id="cb189-155"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-155" aria-hidden="true" tabindex="-1"></a>            <span class="co"># remove factors not used in comparison, these are batch effects/pairs/blocks, </span></span>
<span id="cb189-156"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-156" aria-hidden="true" tabindex="-1"></a>            <span class="co"># keyModelFactors = keyModelFactors[ !is.na(match(keyModelFactors, factorsVector)) ]    </span></span>
<span id="cb189-157"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-157" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-158"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-158" aria-hidden="true" tabindex="-1"></a>            <span class="co"># if a factor is selected both in block and main factors, then use it as block factor</span></span>
<span id="cb189-159"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-159" aria-hidden="true" tabindex="-1"></a>            keyModelFactors <span class="ot">=</span> keyModelFactors[ <span class="fu">is.na</span>(<span class="fu">match</span>(keyModelFactors, blockFactor)) ] </span>
<span id="cb189-160"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-160" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-161"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-161" aria-hidden="true" tabindex="-1"></a>            <span class="co">#------Design matrix            </span></span>
<span id="cb189-162"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-162" aria-hidden="true" tabindex="-1"></a>            sampleInfo2 <span class="ot">=</span> sampleInfo[,keyModelFactors,drop<span class="ot">=</span>F] <span class="co"># remove factors not used.</span></span>
<span id="cb189-163"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-163" aria-hidden="true" tabindex="-1"></a>            groups <span class="ot">=</span> <span class="fu">apply</span>(sampleInfo2,<span class="dv">1</span>, <span class="cf">function</span>(x) <span class="fu">paste</span>(x,<span class="at">collapse=</span><span class="st">&quot;_&quot;</span>)  )</span>
<span id="cb189-164"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-164" aria-hidden="true" tabindex="-1"></a>            g <span class="ot">=</span>  <span class="fu">unique</span>(groups)  </span>
<span id="cb189-165"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-165" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-166"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-166" aria-hidden="true" tabindex="-1"></a>            design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span><span class="sc">+</span><span class="fu">factor</span>(groups))</span>
<span id="cb189-167"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-167" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;.*)&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(design))</span>
<span id="cb189-168"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-168" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-169"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-169" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>(rawCounts) <span class="sc">&amp;&amp;</span> countsDEGMethods <span class="sc">==</span> <span class="dv">2</span>) {  <span class="co"># voom</span></span>
<span id="cb189-170"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-170" aria-hidden="true" tabindex="-1"></a>                v <span class="ot">&lt;-</span> <span class="fu">voom</span>(rawCounts, design); </span>
<span id="cb189-171"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-171" aria-hidden="true" tabindex="-1"></a>                fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(v, design) </span>
<span id="cb189-172"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-172" aria-hidden="true" tabindex="-1"></a>            } <span class="cf">else</span> </span>
<span id="cb189-173"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-173" aria-hidden="true" tabindex="-1"></a>                fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(eset, design)</span>
<span id="cb189-174"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-174" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-175"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-175" aria-hidden="true" tabindex="-1"></a>            fit <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit, <span class="at">trend=</span>limmaTrend)    </span>
<span id="cb189-176"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-176" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-177"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-177" aria-hidden="true" tabindex="-1"></a>            <span class="co">#-----------Making comaprisons</span></span>
<span id="cb189-178"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-178" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">length</span>(keyModelFactors) <span class="sc">!=</span> <span class="dv">2</span> <span class="sc">|</span> <span class="fu">length</span>(blockFactor) <span class="sc">&gt;</span><span class="dv">1</span> )  { <span class="co"># if only one factor, or more than two then use all pairwise comparisons</span></span>
<span id="cb189-179"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-179" aria-hidden="true" tabindex="-1"></a>                comparisons <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;.*: &quot;</span>,<span class="st">&quot;&quot;</span>,selectedComparisons)</span>
<span id="cb189-180"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-180" aria-hidden="true" tabindex="-1"></a>                comparisons <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; vs</span><span class="sc">\\</span><span class="st">. &quot;</span>,<span class="st">&quot;-&quot;</span>,comparisons)</span>
<span id="cb189-181"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-181" aria-hidden="true" tabindex="-1"></a>            } <span class="cf">else</span> <span class="cf">if</span>( <span class="fu">length</span>(keyModelFactors) <span class="sc">==</span> <span class="dv">2</span> ){ <span class="co"># two key factors</span></span>
<span id="cb189-182"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-183"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-183" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>( <span class="fu">sum</span>(<span class="fu">grepl</span>(<span class="st">&quot;:&quot;</span>,modelFactors) <span class="sc">&gt;</span><span class="dv">0</span>) ) {  <span class="co"># interaction? </span></span>
<span id="cb189-184"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-184" aria-hidden="true" tabindex="-1"></a>                    interactionTerm <span class="ot">=</span><span class="cn">TRUE</span></span>
<span id="cb189-185"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-185" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># all pairwise comparisons</span></span>
<span id="cb189-186"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-186" aria-hidden="true" tabindex="-1"></a>                    comparisons <span class="ot">=</span> <span class="st">&quot;&quot;</span></span>
<span id="cb189-187"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-187" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(g)<span class="sc">-</span><span class="dv">1</span>) )</span>
<span id="cb189-188"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-188" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">for</span> (j <span class="cf">in</span> (i<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(g)) </span>
<span id="cb189-189"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-189" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>( <span class="fu">strsplit</span>(g[i],<span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>] <span class="sc">==</span> <span class="fu">strsplit</span>(g[j],<span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]<span class="sc">|</span> <span class="fu">strsplit</span>(g[i],<span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>] <span class="sc">==</span> <span class="fu">strsplit</span>(g[j],<span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>]) <span class="co"># only compare WT_control vs. WT_treatment</span></span>
<span id="cb189-190"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-190" aria-hidden="true" tabindex="-1"></a>                            comparisons <span class="ot">=</span> <span class="fu">c</span>(comparisons,<span class="fu">paste</span>(g[j],<span class="st">&quot;-&quot;</span>,g[i],<span class="at">sep=</span><span class="st">&quot;&quot;</span> ) )</span>
<span id="cb189-191"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-191" aria-hidden="true" tabindex="-1"></a>                    comparisons <span class="ot">&lt;-</span> comparisons[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb189-192"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-192" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb189-193"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-193" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># pairwise contrasts</span></span>
<span id="cb189-194"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-194" aria-hidden="true" tabindex="-1"></a>                    contrast1 <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons[<span class="dv">1</span>], <span class="at">levels=</span>design)</span>
<span id="cb189-195"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-195" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>(<span class="fu">length</span>(comparisons)<span class="sc">&gt;</span><span class="dv">1</span> )</span>
<span id="cb189-196"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-196" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span>( kk <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(comparisons) )</span>
<span id="cb189-197"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-197" aria-hidden="true" tabindex="-1"></a>                        contrast1<span class="ot">&lt;-</span>  <span class="fu">cbind</span>(contrast1,<span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons[kk], <span class="at">levels=</span>design)   )</span>
<span id="cb189-198"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-198" aria-hidden="true" tabindex="-1"></a>                    contrast.names <span class="ot">=</span> <span class="fu">colnames</span>(contrast1)        </span>
<span id="cb189-199"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-199" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb189-200"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-200" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># all possible interactions</span></span>
<span id="cb189-201"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-201" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># interaction contrasts</span></span>
<span id="cb189-202"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-202" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb189-203"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-203" aria-hidden="true" tabindex="-1"></a>                    contrast2 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb189-204"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-204" aria-hidden="true" tabindex="-1"></a>                    contrast.names <span class="ot">=</span><span class="st">&quot;&quot;</span></span>
<span id="cb189-205"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-205" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span> ( kk <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">dim</span>(contrast1)[<span class="dv">2</span>]<span class="sc">-</span><span class="dv">1</span>) ) {</span>
<span id="cb189-206"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-206" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span>( kp <span class="cf">in</span> (kk<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">dim</span>(contrast1)[<span class="dv">2</span>]) </span>
<span id="cb189-207"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-207" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#if( treatments[kp]== treatments[kk] ) </span></span>
<span id="cb189-208"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-208" aria-hidden="true" tabindex="-1"></a>                        {  <span class="cf">if</span>(<span class="fu">is.null</span>(contrast2))  </span>
<span id="cb189-209"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-209" aria-hidden="true" tabindex="-1"></a>                            contrast2 <span class="ot">=</span> contrast1[,kp]<span class="sc">-</span> contrast1[,kk]  <span class="cf">else</span>                    </span>
<span id="cb189-210"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-210" aria-hidden="true" tabindex="-1"></a>                            contrast2 <span class="ot">=</span> <span class="fu">cbind</span>(contrast2, contrast1[,kp]<span class="sc">-</span> contrast1[,kk] )</span>
<span id="cb189-211"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-211" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb189-212"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-212" aria-hidden="true" tabindex="-1"></a>                            contrast.names <span class="ot">=</span> <span class="fu">c</span>(contrast.names, <span class="fu">paste0</span>(<span class="st">&quot;I:&quot;</span>,  <span class="fu">colnames</span>(contrast1)[kp], <span class="st">&quot;.vs.&quot;</span>, <span class="fu">colnames</span>(contrast1)[kk] ) )</span>
<span id="cb189-213"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-213" aria-hidden="true" tabindex="-1"></a>                        }   </span>
<span id="cb189-214"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-214" aria-hidden="true" tabindex="-1"></a>                    }</span>
<span id="cb189-215"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-215" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">colnames</span>(contrast2)<span class="ot">=</span>contrast.names[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb189-216"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-216" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb189-217"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-217" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># remove nonsense contrasts from interactions</span></span>
<span id="cb189-218"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-218" aria-hidden="true" tabindex="-1"></a>                     contrast2 <span class="ot">=</span> contrast2[,<span class="fu">which</span>(<span class="fu">apply</span>(<span class="fu">abs</span>(contrast2),<span class="dv">2</span>,max)<span class="sc">==</span><span class="dv">1</span>),drop<span class="ot">=</span>F]</span>
<span id="cb189-219"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-219" aria-hidden="true" tabindex="-1"></a>                     contrast2 <span class="ot">=</span> contrast2[,<span class="fu">which</span>(<span class="fu">apply</span>(<span class="fu">abs</span>(contrast2),<span class="dv">2</span>,sum)<span class="sc">==</span><span class="dv">4</span>),drop<span class="ot">=</span>F]   </span>
<span id="cb189-220"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-220" aria-hidden="true" tabindex="-1"></a>                     contrast2 <span class="ot">=</span> <span class="fu">t</span>( <span class="fu">unique</span>(<span class="fu">t</span>(contrast2)) ) <span class="co"># remove duplicate columns       </span></span>
<span id="cb189-221"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-221" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb189-222"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-222" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># remove unwanted contrasts involving more than three levels in either factor</span></span>
<span id="cb189-223"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-223" aria-hidden="true" tabindex="-1"></a>                    keep<span class="ot">=</span> <span class="fu">c</span>()</span>
<span id="cb189-224"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-224" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(contrast2)[<span class="dv">2</span>]) {</span>
<span id="cb189-225"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-225" aria-hidden="true" tabindex="-1"></a>                        tem <span class="ot">=</span> <span class="fu">rownames</span>(contrast2)[ contrast2[ ,i ] <span class="sc">!=</span> <span class="dv">0</span>   ]</span>
<span id="cb189-226"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-226" aria-hidden="true" tabindex="-1"></a>                        tem1 <span class="ot">=</span> <span class="fu">unique</span> (  <span class="fu">unlist</span>(<span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>,<span class="st">&quot;&quot;</span>, tem) ) )</span>
<span id="cb189-227"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-227" aria-hidden="true" tabindex="-1"></a>                        tem2 <span class="ot">=</span> <span class="fu">unique</span> (  <span class="fu">unlist</span>(<span class="fu">gsub</span>(<span class="st">&quot;.*_&quot;</span>,<span class="st">&quot;&quot;</span>, tem) ) )</span>
<span id="cb189-228"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-228" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>( <span class="fu">length</span>(tem1) <span class="sc">==</span> <span class="dv">2</span> <span class="sc">&amp;</span> <span class="fu">length</span>(tem2) <span class="sc">==</span><span class="dv">2</span> )</span>
<span id="cb189-229"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-229" aria-hidden="true" tabindex="-1"></a>                        keep <span class="ot">=</span> <span class="fu">c</span>(keep, <span class="fu">colnames</span>(contrast2) [i] )</span>
<span id="cb189-230"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-230" aria-hidden="true" tabindex="-1"></a>                    }</span>
<span id="cb189-231"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-231" aria-hidden="true" tabindex="-1"></a>                    contrast2 <span class="ot">=</span> contrast2[,keep,drop<span class="ot">=</span>F]</span>
<span id="cb189-232"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-232" aria-hidden="true" tabindex="-1"></a>                    comparisons2 <span class="ot">=</span> <span class="fu">colnames</span>(contrast2)               </span>
<span id="cb189-233"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-233" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb189-234"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-235"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-235" aria-hidden="true" tabindex="-1"></a>                <span class="co"># &quot;stage: MN vs. EN&quot;  --&gt;  c(&quot;MN_AB-EN_AB&quot;, &quot;EN_Nodule-EN_AB&quot;) </span></span>
<span id="cb189-236"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-236" aria-hidden="true" tabindex="-1"></a>                <span class="co">#  comparisons in all levels of the other factor </span></span>
<span id="cb189-237"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-237" aria-hidden="true" tabindex="-1"></a>                transformComparisons <span class="ot">&lt;-</span> <span class="cf">function</span> (comparison1){</span>
<span id="cb189-238"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-238" aria-hidden="true" tabindex="-1"></a>                    tem <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;.*: &quot;</span>,<span class="st">&quot;&quot;</span>,comparison1)</span>
<span id="cb189-239"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-239" aria-hidden="true" tabindex="-1"></a>                    tem <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(tem, <span class="st">&quot; vs</span><span class="sc">\\</span><span class="st">. &quot;</span>) ) <span class="co"># control  mutant                           </span></span>
<span id="cb189-240"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-240" aria-hidden="true" tabindex="-1"></a>                    factor1<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;:.*&quot;</span>,<span class="st">&quot;&quot;</span>,comparison1)</span>
<span id="cb189-241"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-242"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-242" aria-hidden="true" tabindex="-1"></a>                    ix <span class="ot">=</span> <span class="fu">match</span>(factor1, keyModelFactors) <span class="co"># 1: first factor, 2: 2nd factor</span></span>
<span id="cb189-243"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-243" aria-hidden="true" tabindex="-1"></a>                    otherFactor <span class="ot">=</span> keyModelFactors[<span class="dv">3</span><span class="sc">-</span>ix]   <span class="co"># 3-1 = 2; 3-1=1</span></span>
<span id="cb189-244"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-244" aria-hidden="true" tabindex="-1"></a>                    otherFactorLevels <span class="ot">=</span> <span class="fu">unique</span>( sampleInfo2[,otherFactor] )             </span>
<span id="cb189-245"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-245" aria-hidden="true" tabindex="-1"></a>                    comparisons <span class="ot">=</span> <span class="fu">c</span>( )</span>
<span id="cb189-246"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-246" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb189-247"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-247" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span> (factorLevels <span class="cf">in</span> otherFactorLevels) {</span>
<span id="cb189-248"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-248" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>( ix <span class="sc">==</span> <span class="dv">1</span>){</span>
<span id="cb189-249"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-249" aria-hidden="true" tabindex="-1"></a>                            comparisons <span class="ot">=</span> <span class="fu">c</span>( comparisons, <span class="fu">paste</span>(<span class="fu">paste0</span>( tem, <span class="st">&quot;_&quot;</span>,factorLevels),<span class="at">collapse=</span><span class="st">&quot;-&quot;</span>) )</span>
<span id="cb189-250"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-250" aria-hidden="true" tabindex="-1"></a>                        } <span class="cf">else</span> {</span>
<span id="cb189-251"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-251" aria-hidden="true" tabindex="-1"></a>                            comparisons <span class="ot">=</span> <span class="fu">c</span>( comparisons,  <span class="fu">paste</span>(<span class="fu">paste0</span>(factorLevels, <span class="st">&quot;_&quot;</span>, tem),<span class="at">collapse=</span><span class="st">&quot;-&quot;</span>) )</span>
<span id="cb189-252"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-252" aria-hidden="true" tabindex="-1"></a>                        }</span>
<span id="cb189-253"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-253" aria-hidden="true" tabindex="-1"></a>                    }</span>
<span id="cb189-254"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-254" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">return</span>(comparisons)     </span>
<span id="cb189-255"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-255" aria-hidden="true" tabindex="-1"></a>                }   </span>
<span id="cb189-256"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-256" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb189-257"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-257" aria-hidden="true" tabindex="-1"></a>                comparisons <span class="ot">=</span> <span class="fu">unlist</span>( <span class="fu">sapply</span>(selectedComparisons, transformComparisons ))</span>
<span id="cb189-258"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-258" aria-hidden="true" tabindex="-1"></a>                comparisons <span class="ot">=</span> <span class="fu">as.vector</span>(comparisons)</span>
<span id="cb189-259"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-259" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-260"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-260" aria-hidden="true" tabindex="-1"></a>            } <span class="co"># two factors</span></span>
<span id="cb189-261"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-261" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-262"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-262" aria-hidden="true" tabindex="-1"></a>            <span class="co"># make contrasts</span></span>
<span id="cb189-263"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-263" aria-hidden="true" tabindex="-1"></a>            contrast1 <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons[<span class="dv">1</span>], <span class="at">levels=</span>design)</span>
<span id="cb189-264"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-264" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">length</span>(comparisons) <span class="sc">&gt;</span><span class="dv">1</span> )</span>
<span id="cb189-265"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-265" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>( kk <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(comparisons) )</span>
<span id="cb189-266"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-266" aria-hidden="true" tabindex="-1"></a>                contrast1<span class="ot">&lt;-</span>  <span class="fu">cbind</span>(contrast1,<span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons[kk], <span class="at">levels=</span>design)   )</span>
<span id="cb189-267"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-268"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-268" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( interactionTerm ) {  <span class="co"># if interaction terms</span></span>
<span id="cb189-269"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-269" aria-hidden="true" tabindex="-1"></a>                contrast1 <span class="ot">=</span> <span class="fu">cbind</span>(contrast1,contrast2)</span>
<span id="cb189-270"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-270" aria-hidden="true" tabindex="-1"></a>                contrast.names <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">colnames</span>(contrast1), <span class="fu">colnames</span>(contrast2) )</span>
<span id="cb189-271"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-271" aria-hidden="true" tabindex="-1"></a>                comparisons <span class="ot">=</span> <span class="fu">c</span>(comparisons,comparisons2)</span>
<span id="cb189-272"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-272" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb189-273"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-273" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-274"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-274" aria-hidden="true" tabindex="-1"></a>            <span class="co"># block design to remove batch effect or paired samples</span></span>
<span id="cb189-275"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-275" aria-hidden="true" tabindex="-1"></a>            <span class="co"># corfit &lt;- duplicateCorrelation(eset,design,block=targets$Subject)</span></span>
<span id="cb189-276"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-276" aria-hidden="true" tabindex="-1"></a>            <span class="co"># corfit$consensus</span></span>
<span id="cb189-277"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-277" aria-hidden="true" tabindex="-1"></a>            <span class="co">#Then this inter-subject correlation is input into the linear model fit:</span></span>
<span id="cb189-278"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-278" aria-hidden="true" tabindex="-1"></a>            <span class="co"># fit &lt;- lmFit(eset,design,block=targets$Subject,correlation=corfit$consensus)</span></span>
<span id="cb189-279"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-280"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-280" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">length</span>(blockFactor) <span class="sc">&gt;=</span> <span class="dv">1</span> ) { <span class="co"># if a factor is selected as block</span></span>
<span id="cb189-281"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-281" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="fu">length</span>(blockFactor) <span class="sc">&gt;=</span> <span class="dv">1</span> ) </span>
<span id="cb189-282"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-282" aria-hidden="true" tabindex="-1"></a>                    blockFactor <span class="ot">=</span> blockFactor[<span class="dv">1</span>] <span class="co"># if multiple use the first one</span></span>
<span id="cb189-283"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-284"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-284" aria-hidden="true" tabindex="-1"></a>                block <span class="ot">=</span> sampleInfo[, blockFactor]  <span class="co"># the column not used</span></span>
<span id="cb189-285"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-285" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-286"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-286" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>(rawCounts) <span class="sc">&amp;&amp;</span> countsDEGMethods <span class="sc">==</span> <span class="dv">2</span>) {  <span class="co"># voom</span></span>
<span id="cb189-287"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-287" aria-hidden="true" tabindex="-1"></a>                    v <span class="ot">&lt;-</span> <span class="fu">voom</span>(rawCounts, design);</span>
<span id="cb189-288"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-288" aria-hidden="true" tabindex="-1"></a>                    corfit <span class="ot">&lt;-</span> <span class="fu">duplicateCorrelation</span>(v,design,<span class="at">block=</span> block)           </span>
<span id="cb189-289"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-289" aria-hidden="true" tabindex="-1"></a>                    fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(v, design,<span class="at">block=</span>block,<span class="at">correlation=</span>corfit<span class="sc">$</span>consensus) </span>
<span id="cb189-290"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-290" aria-hidden="true" tabindex="-1"></a>                } <span class="cf">else</span> {</span>
<span id="cb189-291"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-291" aria-hidden="true" tabindex="-1"></a>                    corfit <span class="ot">&lt;-</span> <span class="fu">duplicateCorrelation</span>(eset,design,<span class="at">block=</span> block)</span>
<span id="cb189-292"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-292" aria-hidden="true" tabindex="-1"></a>                    fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(eset, design,<span class="at">block=</span>block,<span class="at">correlation=</span>corfit<span class="sc">$</span>consensus)</span>
<span id="cb189-293"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-293" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb189-294"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-294" aria-hidden="true" tabindex="-1"></a>                fit <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit, <span class="at">trend=</span>limmaTrend)            </span>
<span id="cb189-295"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-295" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb189-296"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-296" aria-hidden="true" tabindex="-1"></a>            } <span class="co"># block factors</span></span>
<span id="cb189-297"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-298"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-299"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-299" aria-hidden="true" tabindex="-1"></a>        } <span class="co"># use selected factors</span></span>
<span id="cb189-300"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-301"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-301" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-302"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-302" aria-hidden="true" tabindex="-1"></a>        fit2 <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit, contrast1)</span>
<span id="cb189-303"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-303" aria-hidden="true" tabindex="-1"></a>        fit2 <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit2, <span class="at">trend=</span>limmaTrend)</span>
<span id="cb189-304"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-304" aria-hidden="true" tabindex="-1"></a>        <span class="co">#topTable(fit2, coef=1, adjust=&quot;BH&quot;)</span></span>
<span id="cb189-305"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-305" aria-hidden="true" tabindex="-1"></a>        results <span class="ot">&lt;-</span> <span class="fu">decideTests</span>(fit2, <span class="at">p.value=</span>maxP_limma, <span class="at">lfc=</span> <span class="fu">log2</span>(minFC_limma ))</span>
<span id="cb189-306"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-306" aria-hidden="true" tabindex="-1"></a>        <span class="co">#vennDiagram(results[,1:5],circle.col=rainbow(5))</span></span>
<span id="cb189-307"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-308"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-308" aria-hidden="true" tabindex="-1"></a>        <span class="co">#colnames(results) = unlist(sapply( colnames(results), changeNames  ) )</span></span>
<span id="cb189-309"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-309" aria-hidden="true" tabindex="-1"></a>        <span class="co">#comparisons3 &lt;-  unlist(sapply( comparisons, changeNames  ) ) </span></span>
<span id="cb189-310"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-311"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-311" aria-hidden="true" tabindex="-1"></a>        <span class="co"># extract fold change for each comparison</span></span>
<span id="cb189-312"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-312" aria-hidden="true" tabindex="-1"></a>        <span class="co"># there is issues with direction of foldchange. Sometimes opposite</span></span>
<span id="cb189-313"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-313" aria-hidden="true" tabindex="-1"></a>        top <span class="ot">&lt;-</span> <span class="cf">function</span> (comp) {</span>
<span id="cb189-314"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-314" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit2, <span class="at">number =</span> <span class="fl">1e12</span>,<span class="at">coef=</span>comp,<span class="at">sort.by=</span><span class="st">&quot;M&quot;</span> ) </span>
<span id="cb189-315"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-315" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(tem)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>) { <span class="fu">return</span> (<span class="dv">1</span>) </span>
<span id="cb189-316"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-316" aria-hidden="true" tabindex="-1"></a>            } <span class="cf">else</span>  {           </span>
<span id="cb189-317"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-317" aria-hidden="true" tabindex="-1"></a>                <span class="co"># compute fold change for the first gene (ranked by absolute value)</span></span>
<span id="cb189-318"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-318" aria-hidden="true" tabindex="-1"></a>                tem2 <span class="ot">=</span> <span class="fu">as.numeric</span>( x[ <span class="fu">which</span>(<span class="fu">rownames</span>(x)<span class="sc">==</span> <span class="fu">rownames</span>(tem)[<span class="dv">1</span>]) , ] )</span>
<span id="cb189-319"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-319" aria-hidden="true" tabindex="-1"></a>                <span class="fu">names</span>(tem2) <span class="ot">=</span> <span class="fu">colnames</span>(x) </span>
<span id="cb189-320"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-320" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb189-321"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-321" aria-hidden="true" tabindex="-1"></a>                <span class="fu">return</span>( tem[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">5</span>)]) </span>
<span id="cb189-322"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-322" aria-hidden="true" tabindex="-1"></a>            }  </span>
<span id="cb189-323"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-323" aria-hidden="true" tabindex="-1"></a>                                                    </span>
<span id="cb189-324"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-324" aria-hidden="true" tabindex="-1"></a>        }  <span class="co"># no significant gene returns 1, otherwise a data frame</span></span>
<span id="cb189-325"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-325" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-326"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-326" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-327"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-327" aria-hidden="true" tabindex="-1"></a>        topGenes <span class="ot">&lt;-</span> <span class="fu">lapply</span>(comparisons, top)</span>
<span id="cb189-328"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-328" aria-hidden="true" tabindex="-1"></a>        topGenes <span class="ot">&lt;-</span> <span class="fu">setNames</span>(topGenes, comparisons )</span>
<span id="cb189-329"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-330"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-330" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">&lt;-</span> <span class="fu">which</span>( <span class="fu">unlist</span>( <span class="fu">lapply</span>(topGenes, class) ) <span class="sc">==</span> <span class="st">&quot;numeric&quot;</span>)</span>
<span id="cb189-331"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-331" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">length</span>(ix)<span class="sc">&gt;</span><span class="dv">0</span>) topGenes <span class="ot">&lt;-</span> topGenes[ <span class="sc">-</span> ix ]</span>
<span id="cb189-332"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-332" aria-hidden="true" tabindex="-1"></a>        <span class="co"># if (length(topGenes) == 0) topGenes = NULL;</span></span>
<span id="cb189-333"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-333" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-334"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-334" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-335"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-335" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(&quot;\n&quot;, names(topGenes) )</span></span>
<span id="cb189-336"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-336" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(&quot;\n&quot;, colnames(results))</span></span>
<span id="cb189-337"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-337" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(&quot;\n&quot;, comparisons3)</span></span>
<span id="cb189-338"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-338" aria-hidden="true" tabindex="-1"></a>    <span class="co">#comparisons &lt;- comparisons3</span></span>
<span id="cb189-339"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-339" aria-hidden="true" tabindex="-1"></a>    <span class="co"># it does not make any sense! comparisons can not be changed!</span></span>
<span id="cb189-340"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-340" aria-hidden="true" tabindex="-1"></a>    <span class="co">#comparisons =comparisons3  </span></span>
<span id="cb189-341"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-341" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(&quot;\n&quot;, comparisons)     </span></span>
<span id="cb189-342"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-342" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-343"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-343" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">results=</span> results, <span class="at">comparisons=</span>comparisons, <span class="at">Exp.type=</span>Exp.type, <span class="at">topGenes=</span>topGenes)) </span>
<span id="cb189-344"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-344" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-345"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-346"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-346" aria-hidden="true" tabindex="-1"></a><span class="co"># Differential expression using DESeq2</span></span>
<span id="cb189-347"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-347" aria-hidden="true" tabindex="-1"></a>DEG.DESeq2 <span class="ot">&lt;-</span> <span class="cf">function</span> (  rawCounts,<span class="at">maxP_limma=</span>.<span class="dv">05</span>, <span class="at">minFC_limma=</span><span class="dv">2</span>, <span class="at">selectedComparisons=</span><span class="cn">NULL</span>, <span class="at">sampleInfo =</span> <span class="cn">NULL</span>,<span class="at">modelFactors=</span><span class="cn">NULL</span>, <span class="at">blockFactor =</span> <span class="cn">NULL</span>, <span class="at">referenceLevels=</span><span class="cn">NULL</span>){</span>
<span id="cb189-348"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-348" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(DESeq2,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># count data analysis</span></span>
<span id="cb189-349"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-349" aria-hidden="true" tabindex="-1"></a>    groups <span class="ot">=</span> <span class="fu">as.character</span> ( <span class="fu">detectGroups</span>( <span class="fu">colnames</span>( rawCounts ) ) )</span>
<span id="cb189-350"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-350" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">=</span> <span class="fu">unique</span>(groups)<span class="co"># order is reversed</span></span>
<span id="cb189-351"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-351" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-352"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-352" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-353"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-353" aria-hidden="true" tabindex="-1"></a>    <span class="co"># check for replicates, removes samples without replicates</span></span>
<span id="cb189-354"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-354" aria-hidden="true" tabindex="-1"></a>    reps <span class="ot">=</span> <span class="fu">as.matrix</span>(<span class="fu">table</span>(groups)) <span class="co"># number of replicates per biological sample</span></span>
<span id="cb189-355"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-355" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> ( <span class="fu">sum</span>( reps[,<span class="dv">1</span>] <span class="sc">&gt;=</span> <span class="dv">2</span>) <span class="sc">&lt;</span><span class="dv">2</span> ) <span class="co"># if less than 2 samples with replicates</span></span>
<span id="cb189-356"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-356" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">results=</span> <span class="cn">NULL</span>, <span class="at">comparisons =</span> <span class="cn">NULL</span>, <span class="at">Exp.type=</span><span class="st">&quot;Failed to parse sample names to define groups. </span></span>
<span id="cb189-357"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-357" aria-hidden="true" tabindex="-1"></a><span class="st">        Cannot perform DEGs and pathway analysis. Please double check column names! Use WT_Rep1, WT_Rep2 etc. &quot;</span>, <span class="at">topGenes=</span><span class="cn">NULL</span>)) </span>
<span id="cb189-358"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-358" aria-hidden="true" tabindex="-1"></a>    <span class="co"># remove samples without replicates</span></span>
<span id="cb189-359"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-359" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">rownames</span>(reps)[<span class="fu">which</span>(reps[,<span class="dv">1</span>] <span class="sc">&gt;</span><span class="dv">1</span>)]</span>
<span id="cb189-360"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-360" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">&lt;-</span> <span class="fu">which</span>( groups <span class="sc">%in%</span> g)  </span>
<span id="cb189-361"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-361" aria-hidden="true" tabindex="-1"></a>    groups <span class="ot">&lt;-</span> groups[ix]   </span>
<span id="cb189-362"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-362" aria-hidden="true" tabindex="-1"></a>    rawCounts <span class="ot">&lt;-</span> rawCounts[,ix] </span>
<span id="cb189-363"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-363" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-364"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-364" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-365"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-365" aria-hidden="true" tabindex="-1"></a>    Exp.type <span class="ot">=</span> <span class="fu">paste</span>(<span class="fu">length</span>(g),<span class="st">&quot; sample groups detected.&quot;</span>)</span>
<span id="cb189-366"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-366" aria-hidden="true" tabindex="-1"></a>    comparisons <span class="ot">=</span> <span class="st">&quot;&quot;</span></span>
<span id="cb189-367"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-367" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(g)<span class="sc">-</span><span class="dv">1</span>) )</span>
<span id="cb189-368"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-368" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (j <span class="cf">in</span> (i<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(g)) </span>
<span id="cb189-369"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-369" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">=</span> <span class="fu">c</span>(comparisons,<span class="fu">paste</span>(g[j],<span class="st">&quot;-&quot;</span>,g[i],<span class="at">sep=</span><span class="st">&quot;&quot;</span> ) )</span>
<span id="cb189-370"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-370" aria-hidden="true" tabindex="-1"></a>    comparisons <span class="ot">&lt;-</span> comparisons[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb189-371"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-371" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-372"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-372" aria-hidden="true" tabindex="-1"></a>    colData <span class="ot">=</span> <span class="fu">cbind</span>(<span class="fu">colnames</span>(rawCounts), groups )</span>
<span id="cb189-373"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-374"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-374" aria-hidden="true" tabindex="-1"></a>    <span class="co"># no sample file, but user selected comparisons using column names</span></span>
<span id="cb189-375"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-375" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">is.null</span>(modelFactors) <span class="sc">&amp;</span> <span class="fu">length</span>( selectedComparisons) <span class="sc">&gt;</span><span class="dv">0</span>  )  </span>
<span id="cb189-376"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-376" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">=</span> selectedComparisons</span>
<span id="cb189-377"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-377" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-378"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-378" aria-hidden="true" tabindex="-1"></a>    comparisons2 <span class="ot">=</span> comparisons   <span class="co"># this is for showing comparison names, which might be different from internally   </span></span>
<span id="cb189-379"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-379" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set up the DESeqDataSet Object and run the DESeq pipeline</span></span>
<span id="cb189-380"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-380" aria-hidden="true" tabindex="-1"></a>    dds <span class="ot">=</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData=</span>rawCounts,</span>
<span id="cb189-381"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-381" aria-hidden="true" tabindex="-1"></a>                                <span class="at">colData=</span>colData,</span>
<span id="cb189-382"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-382" aria-hidden="true" tabindex="-1"></a>                                <span class="at">design=</span><span class="sc">~</span>groups)                             </span>
<span id="cb189-383"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-383" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-384"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-384" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">is.null</span>(modelFactors)  ) </span>
<span id="cb189-385"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-385" aria-hidden="true" tabindex="-1"></a>        dds <span class="ot">=</span> <span class="fu">DESeq</span>(dds)    <span class="cf">else</span>  </span>
<span id="cb189-386"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-386" aria-hidden="true" tabindex="-1"></a>    {    <span class="co"># using selected factors and comparisons</span></span>
<span id="cb189-387"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-387" aria-hidden="true" tabindex="-1"></a>        <span class="co"># build model</span></span>
<span id="cb189-388"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-388" aria-hidden="true" tabindex="-1"></a>        modelFactors <span class="ot">=</span> <span class="fu">c</span>(modelFactors,blockFactor) <span class="co"># block factor is just added in. </span></span>
<span id="cb189-389"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-390"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-390" aria-hidden="true" tabindex="-1"></a>        factors <span class="ot">=</span> modelFactors   <span class="co"># selected factors and interactions: c( &quot;strain&quot;, &quot;treatment&quot;,  &quot;strain:treatment&quot;)</span></span>
<span id="cb189-391"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-391" aria-hidden="true" tabindex="-1"></a>        factors <span class="ot">=</span> factors[ <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;:&quot;</span>,factors )]   <span class="co"># non-interaction terms</span></span>
<span id="cb189-392"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-392" aria-hidden="true" tabindex="-1"></a>        <span class="co"># interaction terms like strain:treatment</span></span>
<span id="cb189-393"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-393" aria-hidden="true" tabindex="-1"></a>        Interactions <span class="ot">=</span> modelFactors[ <span class="fu">grepl</span>(<span class="st">&quot;:&quot;</span>,modelFactors )]</span>
<span id="cb189-394"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-394" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-395"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-395" aria-hidden="true" tabindex="-1"></a>        colData <span class="ot">=</span> sampleInfo  </span>
<span id="cb189-396"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-396" aria-hidden="true" tabindex="-1"></a>        factorsCoded <span class="ot">=</span> <span class="fu">toupper</span>(letters )[<span class="dv">1</span><span class="sc">:</span> <span class="fu">dim</span>(colData)[<span class="dv">2</span>] ]   <span class="co"># Factors are encoded as &quot;A&quot;, &quot;B&quot;, &quot;C&quot;; this avoid illigal letters</span></span>
<span id="cb189-397"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-397" aria-hidden="true" tabindex="-1"></a>        <span class="fu">names</span>(factorsCoded) <span class="ot">=</span>  <span class="fu">colnames</span>(colData)  <span class="co"># this is for look up; each column of sampleInfo  </span></span>
<span id="cb189-398"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-398" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(colData) <span class="ot">=</span> factorsCoded <span class="co"># all columns named A B C D </span></span>
<span id="cb189-399"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-400"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-400" aria-hidden="true" tabindex="-1"></a>        colData <span class="ot">=</span> <span class="fu">as.data.frame</span>(colData)</span>
<span id="cb189-401"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-401" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-402"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-402" aria-hidden="true" tabindex="-1"></a>        <span class="co"># set reference levels for factors</span></span>
<span id="cb189-403"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-403" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="sc">!</span> <span class="fu">is.null</span>( referenceLevels) ) {   <span class="co"># c(&quot;genotype:wt&quot;, &quot;treatment:control&quot; )</span></span>
<span id="cb189-404"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-404" aria-hidden="true" tabindex="-1"></a>            <span class="co"># first factor</span></span>
<span id="cb189-405"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-405" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> ( refs <span class="cf">in</span> referenceLevels)</span>
<span id="cb189-406"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-406" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="sc">!</span> <span class="fu">is.null</span>( refs) ) {</span>
<span id="cb189-407"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-407" aria-hidden="true" tabindex="-1"></a>                    ix <span class="ot">=</span> <span class="fu">match</span>(<span class="fu">gsub</span>(<span class="st">&quot;:.*&quot;</span>,<span class="st">&quot;&quot;</span>,refs), <span class="fu">colnames</span>(sampleInfo) ) <span class="co"># corresponding column id for factor</span></span>
<span id="cb189-408"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-408" aria-hidden="true" tabindex="-1"></a>                    colData[,ix] <span class="ot">=</span> <span class="fu">as.factor</span>( colData[,ix] )</span>
<span id="cb189-409"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-409" aria-hidden="true" tabindex="-1"></a>                    colData[,ix] <span class="ot">=</span> <span class="fu">relevel</span>(colData[,ix],<span class="fu">gsub</span>(<span class="st">&quot;.*:&quot;</span>,<span class="st">&quot;&quot;</span>,refs)  )</span>
<span id="cb189-410"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-410" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb189-411"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-411" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb189-412"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-412" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb189-413"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-413" aria-hidden="true" tabindex="-1"></a>        <span class="co"># base model</span></span>
<span id="cb189-414"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-414" aria-hidden="true" tabindex="-1"></a>        DESeq2.Object<span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;dds = DESeqDataSetFromMatrix(countData=rawCounts, colData=colData, design=~ &quot;</span>, </span>
<span id="cb189-415"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-415" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">paste</span>( factorsCoded[factors],<span class="at">collapse=</span><span class="st">&quot;+&quot;</span>)) <span class="co"># only use selected factors        </span></span>
<span id="cb189-416"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-416" aria-hidden="true" tabindex="-1"></a>        Exp.type <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;Model: ~&quot;</span>, <span class="fu">paste</span>(modelFactors,<span class="at">collapse=</span><span class="st">&quot; + &quot;</span>) )</span>
<span id="cb189-417"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-418"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-418" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-419"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-419" aria-hidden="true" tabindex="-1"></a>        <span class="co"># create model</span></span>
<span id="cb189-420"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-420" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">length</span>(Interactions)<span class="sc">&gt;</span><span class="dv">0</span> ) { <span class="co"># if there is interaction</span></span>
<span id="cb189-421"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-421" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>( interactionTerms <span class="cf">in</span> Interactions) {</span>
<span id="cb189-422"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-422" aria-hidden="true" tabindex="-1"></a>                interactingFactors <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(interactionTerms,<span class="st">&quot;:&quot;</span> ) )  <span class="co"># split strain:treatment as &quot;strain&quot; and &quot;mutant&quot;</span></span>
<span id="cb189-423"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-423" aria-hidden="true" tabindex="-1"></a>                tem <span class="ot">=</span> <span class="fu">paste</span>(factorsCoded [ interactingFactors ],<span class="at">collapse=</span><span class="st">&quot;:&quot;</span>)   <span class="co"># convert &quot;strain:mutant&quot; to &quot;A:B&quot;</span></span>
<span id="cb189-424"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-424" aria-hidden="true" tabindex="-1"></a>                DESeq2.Object <span class="ot">=</span> <span class="fu">paste</span>(DESeq2.Object, <span class="st">&quot; + &quot;</span>,tem)</span>
<span id="cb189-425"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-425" aria-hidden="true" tabindex="-1"></a>            }           </span>
<span id="cb189-426"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-426" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb189-427"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-427" aria-hidden="true" tabindex="-1"></a>        DESeq2.Object<span class="ot">=</span> <span class="fu">paste</span>( DESeq2.Object, <span class="st">&quot;)&quot;</span>) <span class="co"># ends the model</span></span>
<span id="cb189-428"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-429"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-429" aria-hidden="true" tabindex="-1"></a>        <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> DESeq2.Object) )</span>
<span id="cb189-430"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-430" aria-hidden="true" tabindex="-1"></a>        dds <span class="ot">=</span> <span class="fu">DESeq</span>(dds)  <span class="co"># main function       </span></span>
<span id="cb189-431"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-431" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-432"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-432" aria-hidden="true" tabindex="-1"></a>        <span class="co"># comparisons </span></span>
<span id="cb189-433"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-433" aria-hidden="true" tabindex="-1"></a>        <span class="co"># &quot;group: control vs. mutant&quot;</span></span>
<span id="cb189-434"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-434" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;.*: &quot;</span>,<span class="st">&quot;&quot;</span>,selectedComparisons)</span>
<span id="cb189-435"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-435" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; vs</span><span class="sc">\\</span><span class="st">. &quot;</span>,<span class="st">&quot;-&quot;</span>,comparisons)</span>
<span id="cb189-436"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-436" aria-hidden="true" tabindex="-1"></a>        factorsVector<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;:.*&quot;</span>,<span class="st">&quot;&quot;</span>,selectedComparisons) <span class="co"># corresponding factors for each comparison</span></span>
<span id="cb189-437"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-437" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-438"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-438" aria-hidden="true" tabindex="-1"></a>        <span class="co"># comparison2 holds names for display with real factor names</span></span>
<span id="cb189-439"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-439" aria-hidden="true" tabindex="-1"></a>        <span class="co"># comparison  is used in calculation it is A, B, C for factors</span></span>
<span id="cb189-440"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-440" aria-hidden="true" tabindex="-1"></a>        comparisons2 <span class="ot">=</span> comparisons    </span>
<span id="cb189-441"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-441" aria-hidden="true" tabindex="-1"></a>        <span class="co">#comparisons2 = gsub(&quot; vs\\. &quot;,&quot;-&quot;,selectedComparisons) </span></span>
<span id="cb189-442"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-442" aria-hidden="true" tabindex="-1"></a>        <span class="co">#comparisons2 = gsub(&quot;:&quot;,&quot;_&quot;,comparisons2)      </span></span>
<span id="cb189-443"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-443" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Note that with interaction terms, not all meaningful comparisons is listed for selection. </span></span>
<span id="cb189-444"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-444" aria-hidden="true" tabindex="-1"></a>        <span class="co"># this is complex. Only under reference level.</span></span>
<span id="cb189-445"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-445" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-446"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-446" aria-hidden="true" tabindex="-1"></a>        <span class="co"># comparisons due to interaction terms</span></span>
<span id="cb189-447"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-447" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">length</span>(Interactions)<span class="sc">&gt;</span><span class="dv">0</span> ) { <span class="co"># if there is interaction</span></span>
<span id="cb189-448"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-448" aria-hidden="true" tabindex="-1"></a>            interactionComparisons <span class="ot">=</span> <span class="fu">resultsNames</span>(dds)</span>
<span id="cb189-449"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-449" aria-hidden="true" tabindex="-1"></a>            interactionComparisons <span class="ot">=</span> interactionComparisons[ <span class="fu">grepl</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,interactionComparisons )   ]</span>
<span id="cb189-450"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-450" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-451"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-451" aria-hidden="true" tabindex="-1"></a>            comparisons <span class="ot">=</span> <span class="fu">c</span>(comparisons,interactionComparisons )</span>
<span id="cb189-452"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-452" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-453"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-453" aria-hidden="true" tabindex="-1"></a>            <span class="co"># translate comparisons generated in interaction terms back to real factor names</span></span>
<span id="cb189-454"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-454" aria-hidden="true" tabindex="-1"></a>            interactionComparisons2 <span class="ot">=</span> interactionComparisons</span>
<span id="cb189-455"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-455" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> ( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(interactionComparisons2 ) ) {</span>
<span id="cb189-456"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-456" aria-hidden="true" tabindex="-1"></a>                tem <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(interactionComparisons2[i],<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span> ) )</span>
<span id="cb189-457"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-457" aria-hidden="true" tabindex="-1"></a>                tem_factors <span class="ot">=</span> <span class="fu">substr</span>(tem,<span class="dv">1</span>,<span class="dv">1</span>) </span>
<span id="cb189-458"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-458" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb189-459"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-459" aria-hidden="true" tabindex="-1"></a>                tem_factors[<span class="dv">1</span>] <span class="ot">=</span> <span class="fu">names</span>(factorsCoded)[factorsCoded <span class="sc">==</span> tem_factors[<span class="dv">1</span>]]  <span class="co"># get the first letter and translate into real factor names</span></span>
<span id="cb189-460"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-460" aria-hidden="true" tabindex="-1"></a>                tem_factors[<span class="dv">2</span>] <span class="ot">=</span> <span class="fu">names</span>(factorsCoded)[factorsCoded <span class="sc">==</span> tem_factors[<span class="dv">2</span>]]  <span class="co"># get the 2nd letters and translate into real factor names</span></span>
<span id="cb189-461"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-461" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-462"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-462" aria-hidden="true" tabindex="-1"></a>                interactionComparisons2[i] <span class="ot">&lt;-</span> <span class="fu">paste0</span>( <span class="st">&quot;I:&quot;</span>,tem_factors[<span class="dv">1</span>], <span class="st">&quot;_&quot;</span>,<span class="fu">substr</span>(tem[<span class="dv">1</span>],<span class="dv">2</span>,<span class="fu">nchar</span>(tem[<span class="dv">1</span>]) ),<span class="st">&quot;.&quot;</span>,</span>
<span id="cb189-463"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-463" aria-hidden="true" tabindex="-1"></a>                                                              tem_factors[<span class="dv">2</span>], <span class="st">&quot;_&quot;</span>,<span class="fu">substr</span>(tem[<span class="dv">2</span>],<span class="dv">2</span>,<span class="fu">nchar</span>(tem[<span class="dv">2</span>]) ) </span>
<span id="cb189-464"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-464" aria-hidden="true" tabindex="-1"></a>                                                    )               </span>
<span id="cb189-465"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-465" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb189-466"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-466" aria-hidden="true" tabindex="-1"></a>            comparisons2 <span class="ot">=</span> <span class="fu">c</span>(comparisons2,interactionComparisons2 )</span>
<span id="cb189-467"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-467" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb189-468"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-468" aria-hidden="true" tabindex="-1"></a>    } <span class="co"># if selected factors </span></span>
<span id="cb189-469"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-469" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-470"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-470" aria-hidden="true" tabindex="-1"></a>    <span class="co"># extract contrasts according to comprisons defined above</span></span>
<span id="cb189-471"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-471" aria-hidden="true" tabindex="-1"></a>    result1 <span class="ot">=</span> <span class="cn">NULL</span>; allCalls <span class="ot">=</span> <span class="cn">NULL</span>;</span>
<span id="cb189-472"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-472" aria-hidden="true" tabindex="-1"></a>    topGenes <span class="ot">=</span> <span class="fu">list</span>(); pk <span class="ot">=</span> <span class="dv">1</span> <span class="co"># counter</span></span>
<span id="cb189-473"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-473" aria-hidden="true" tabindex="-1"></a>    pp<span class="ot">=</span><span class="dv">0</span> <span class="co"># first results?</span></span>
<span id="cb189-474"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-474" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>( kk <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(comparisons) ) {</span>
<span id="cb189-475"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-475" aria-hidden="true" tabindex="-1"></a>        tem <span class="ot">=</span> <span class="fu">unlist</span>( <span class="fu">strsplit</span>(comparisons[kk],<span class="st">&quot;-&quot;</span>) )</span>
<span id="cb189-476"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-476" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-477"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-477" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>(modelFactors)) <span class="co"># if just group comparison using sample names</span></span>
<span id="cb189-478"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-478" aria-hidden="true" tabindex="-1"></a>            selected <span class="ot">=</span> <span class="fu">results</span>(dds, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">&quot;groups&quot;</span>, tem[<span class="dv">1</span>], tem[<span class="dv">2</span>]) )   <span class="cf">else</span> {</span>
<span id="cb189-479"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-479" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>, comparisons[kk] ) )    <span class="co"># if not interaction term: they contain .  interaction term</span></span>
<span id="cb189-480"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-480" aria-hidden="true" tabindex="-1"></a>                selected <span class="ot">=</span> <span class="fu">results</span>(dds, <span class="at">contrast=</span><span class="fu">c</span>( factorsCoded[ factorsVector[kk] ],tem[<span class="dv">1</span>], tem[<span class="dv">2</span>]) ) <span class="cf">else</span> <span class="co"># either A, B, C ...</span></span>
<span id="cb189-481"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-481" aria-hidden="true" tabindex="-1"></a>                selected <span class="ot">=</span> <span class="fu">results</span>(dds, <span class="at">name=</span>comparisons[kk] ) <span class="co"># interaction term</span></span>
<span id="cb189-482"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-482" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb189-483"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-484"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-484" aria-hidden="true" tabindex="-1"></a>        selected<span class="sc">$</span>calls <span class="ot">=</span><span class="dv">0</span>   </span>
<span id="cb189-485"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-485" aria-hidden="true" tabindex="-1"></a>        selected<span class="sc">$</span>calls [<span class="fu">which</span>( selected<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="fu">log2</span>(minFC_limma) <span class="sc">&amp;</span> selected<span class="sc">$</span>padj <span class="sc">&lt;</span> maxP_limma ) ]  <span class="ot">&lt;-</span>  <span class="dv">1</span></span>
<span id="cb189-486"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-486" aria-hidden="true" tabindex="-1"></a>        selected<span class="sc">$</span>calls [ <span class="fu">which</span>( selected<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span>  <span class="sc">-</span><span class="fu">log2</span>(minFC_limma) <span class="sc">&amp;</span> selected<span class="sc">$</span>padj <span class="sc">&lt;</span> maxP_limma ) ] <span class="ot">&lt;-</span>  <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb189-487"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-487" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(selected)<span class="ot">=</span> <span class="fu">paste</span>( <span class="fu">as.character</span>(comparisons2[kk]), <span class="st">&quot;___&quot;</span>,<span class="fu">colnames</span>(selected),<span class="at">sep=</span><span class="st">&quot;&quot;</span> )</span>
<span id="cb189-488"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-488" aria-hidden="true" tabindex="-1"></a>        selected <span class="ot">=</span> <span class="fu">as.data.frame</span>(selected)</span>
<span id="cb189-489"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-489" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (pp<span class="sc">==</span><span class="dv">0</span>){  <span class="co"># if first one with significant genes, collect gene list and Pval+ fold</span></span>
<span id="cb189-490"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-490" aria-hidden="true" tabindex="-1"></a>        result1 <span class="ot">=</span> selected; pp <span class="ot">=</span> <span class="dv">1</span>; </span>
<span id="cb189-491"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-491" aria-hidden="true" tabindex="-1"></a>        <span class="co"># selected[,2] &lt;- -1 * selected[,2] # reverse fold change direction</span></span>
<span id="cb189-492"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-492" aria-hidden="true" tabindex="-1"></a>        topGenes[[<span class="dv">1</span>]] <span class="ot">=</span> selected[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">6</span>)]; </span>
<span id="cb189-493"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-493" aria-hidden="true" tabindex="-1"></a>        <span class="fu">names</span>(topGenes)[<span class="dv">1</span>] <span class="ot">=</span> comparisons2[kk]; } <span class="cf">else</span> </span>
<span id="cb189-494"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-494" aria-hidden="true" tabindex="-1"></a>            { result1 <span class="ot">=</span> <span class="fu">merge</span>(result1,selected,<span class="at">by=</span><span class="st">&quot;row.names&quot;</span>); </span>
<span id="cb189-495"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-495" aria-hidden="true" tabindex="-1"></a>                <span class="fu">rownames</span>(result1) <span class="ot">=</span> result1[,<span class="dv">1</span>]; </span>
<span id="cb189-496"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-496" aria-hidden="true" tabindex="-1"></a>                result1 <span class="ot">&lt;-</span> result1[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb189-497"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-497" aria-hidden="true" tabindex="-1"></a>                pk<span class="ot">=</span> pk<span class="sc">+</span><span class="dv">1</span>; </span>
<span id="cb189-498"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-498" aria-hidden="true" tabindex="-1"></a>                <span class="co"># selected[,2] &lt;- -1 * selected[,2] # reverse fold change direction</span></span>
<span id="cb189-499"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-499" aria-hidden="true" tabindex="-1"></a>                topGenes[[pk]] <span class="ot">=</span> selected[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">6</span>)]; </span>
<span id="cb189-500"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-500" aria-hidden="true" tabindex="-1"></a>                <span class="fu">names</span>(topGenes)[pk] <span class="ot">=</span> comparisons2[kk];  <span class="co"># assign name to comprison</span></span>
<span id="cb189-501"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-501" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb189-502"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-502" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-503"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-504"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-504" aria-hidden="true" tabindex="-1"></a>    Interactions <span class="ot">=</span> <span class="fu">c</span>()</span>
<span id="cb189-505"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-505" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>(modelFactors) )</span>
<span id="cb189-506"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-506" aria-hidden="true" tabindex="-1"></a>        Interactions <span class="ot">=</span> modelFactors[ <span class="fu">grepl</span>(<span class="st">&quot;:&quot;</span>,modelFactors )]</span>
<span id="cb189-507"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-507" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-508"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-508" aria-hidden="true" tabindex="-1"></a><span class="co">#---  add comprisons for non-reference levels. It adds to the results1 object.  </span></span>
<span id="cb189-509"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-509" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">length</span>(Interactions)<span class="sc">&gt;</span><span class="dv">0</span> ) { <span class="co"># if there is interaction</span></span>
<span id="cb189-510"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-510" aria-hidden="true" tabindex="-1"></a>        factorLookup<span class="ot">=</span><span class="fu">c</span>() <span class="co"># a factor whose values are factors and names are factor and level combination conditionTreated, genotypeWT</span></span>
<span id="cb189-511"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-511" aria-hidden="true" tabindex="-1"></a>        levelLookup <span class="ot">=</span> <span class="fu">c</span>()</span>
<span id="cb189-512"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-512" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-513"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-513" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(sampleInfo)[<span class="dv">2</span>]) {</span>
<span id="cb189-514"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-514" aria-hidden="true" tabindex="-1"></a>            sampleInfo2 <span class="ot">=</span> <span class="fu">unique</span>(sampleInfo)</span>
<span id="cb189-515"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-515" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">rep</span>(<span class="fu">toupper</span>(letters)[i],<span class="fu">dim</span>(sampleInfo2)[<span class="dv">1</span>]  )</span>
<span id="cb189-516"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-516" aria-hidden="true" tabindex="-1"></a>            <span class="fu">names</span>(tem) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">toupper</span>(letters)[i],sampleInfo2[,i])</span>
<span id="cb189-517"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-517" aria-hidden="true" tabindex="-1"></a>            factorLookup <span class="ot">=</span> <span class="fu">c</span>(factorLookup,tem)  </span>
<span id="cb189-518"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-518" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-519"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-519" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">as.character</span>( sampleInfo2[,i] )</span>
<span id="cb189-520"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-520" aria-hidden="true" tabindex="-1"></a>            <span class="fu">names</span>(tem) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">toupper</span>(letters)[i],sampleInfo2[,i])</span>
<span id="cb189-521"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-521" aria-hidden="true" tabindex="-1"></a>            levelLookup <span class="ot">=</span> <span class="fu">c</span>(levelLookup, tem)</span>
<span id="cb189-522"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-522" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb189-523"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-523" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-524"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-524" aria-hidden="true" tabindex="-1"></a>        <span class="co"># split  genotypeI.conditionTrt --&gt; c(&quot;genotype&quot;,&quot;I&quot;,&quot;conditoin&quot;,&quot;Trt&quot;)</span></span>
<span id="cb189-525"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-525" aria-hidden="true" tabindex="-1"></a>        splitInteractionTerms <span class="ot">&lt;-</span> <span class="cf">function</span> (term) {</span>
<span id="cb189-526"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-526" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,term) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb189-527"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-527" aria-hidden="true" tabindex="-1"></a>            terms2 <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(term,<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) )</span>
<span id="cb189-528"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-528" aria-hidden="true" tabindex="-1"></a>                     <span class="co"># factor1, level1, factor2, level2</span></span>
<span id="cb189-529"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-529" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>(<span class="fu">c</span>(factorLookup[terms2[<span class="dv">1</span>]], levelLookup[terms2[<span class="dv">1</span>]],factorLookup[terms2[<span class="dv">2</span>]], levelLookup[terms2[<span class="dv">2</span>]]   ) )</span>
<span id="cb189-530"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-530" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb189-531"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-531" aria-hidden="true" tabindex="-1"></a>        <span class="co"># none interaction terms </span></span>
<span id="cb189-532"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-532" aria-hidden="true" tabindex="-1"></a>        NoneInterTerms <span class="ot">=</span> <span class="fu">resultsNames</span>(dds)[ <span class="sc">!</span><span class="fu">grepl</span>( <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>, <span class="fu">resultsNames</span>(dds)) ]</span>
<span id="cb189-533"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-533" aria-hidden="true" tabindex="-1"></a>        NoneInterTerms<span class="ot">=</span>NoneInterTerms[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb189-534"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-534" aria-hidden="true" tabindex="-1"></a>        allInteractionTerms <span class="ot">=</span> <span class="fu">resultsNames</span>(dds)[ <span class="fu">grepl</span>( <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>, <span class="fu">resultsNames</span>(dds)) ]</span>
<span id="cb189-535"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-535" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-536"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-537"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-537" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( kk <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(NoneInterTerms) ) { <span class="co"># for each none interaction term</span></span>
<span id="cb189-538"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-538" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(modelFactors) ) {<span class="co"># if not just group comparison using sample names</span></span>
<span id="cb189-539"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-539" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#current factor</span></span>
<span id="cb189-540"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-540" aria-hidden="true" tabindex="-1"></a>                    cFactor <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>,<span class="st">&quot;&quot;</span>,NoneInterTerms[kk] )</span>
<span id="cb189-541"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-541" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb189-542"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-542" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span>(interactionTerm <span class="cf">in</span> allInteractionTerms ) {</span>
<span id="cb189-543"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-543" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb189-544"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-544" aria-hidden="true" tabindex="-1"></a>                        splited <span class="ot">=</span> <span class="fu">splitInteractionTerms</span> (interactionTerm)  <span class="co"># 4 components</span></span>
<span id="cb189-545"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-545" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span> (cFactor <span class="sc">!=</span> splited[<span class="dv">1</span>] <span class="sc">&amp;</span> cFactor <span class="sc">!=</span> splited[<span class="dv">3</span>]  ) </span>
<span id="cb189-546"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-546" aria-hidden="true" tabindex="-1"></a>                            <span class="cf">next</span>;                       </span>
<span id="cb189-547"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-547" aria-hidden="true" tabindex="-1"></a>                        </span>
<span id="cb189-548"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-548" aria-hidden="true" tabindex="-1"></a>                        selected <span class="ot">=</span> <span class="fu">results</span>(dds, <span class="fu">list</span>(<span class="fu">c</span>( NoneInterTerms[kk],interactionTerm ) ) ) </span>
<span id="cb189-549"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-549" aria-hidden="true" tabindex="-1"></a>                        comparisonName <span class="ot">=</span> <span class="fu">paste0</span>( NoneInterTerms[kk],<span class="st">&quot;__&quot;</span>, <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>,interactionTerm) )</span>
<span id="cb189-550"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-550" aria-hidden="true" tabindex="-1"></a>                        </span>
<span id="cb189-551"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-551" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>( cFactor <span class="sc">==</span> splited[<span class="dv">1</span>] )</span>
<span id="cb189-552"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-552" aria-hidden="true" tabindex="-1"></a>                            otherLevel <span class="ot">=</span> splited[<span class="dv">4</span>] <span class="cf">else</span> otherLevel <span class="ot">=</span> splited[<span class="dv">2</span>]</span>
<span id="cb189-553"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-553" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb189-554"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-554" aria-hidden="true" tabindex="-1"></a>                        comparisonName <span class="ot">=</span> <span class="fu">paste0</span>(<span class="co">#names(factorsCoded)[which(factorsCoded==cFactor)], # real factor name</span></span>
<span id="cb189-555"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-555" aria-hidden="true" tabindex="-1"></a>                                                <span class="fu">gsub</span>(<span class="st">&quot;_vs_&quot;</span>,<span class="st">&quot;-&quot;</span>, <span class="fu">substr</span>(NoneInterTerms[kk], <span class="dv">3</span>, <span class="fu">nchar</span>(NoneInterTerms[kk]  )  )), <span class="co"># the comparison</span></span>
<span id="cb189-556"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-556" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">&quot;_for_&quot;</span>,otherLevel)</span>
<span id="cb189-557"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-557" aria-hidden="true" tabindex="-1"></a>                        comparisons2 <span class="ot">=</span> <span class="fu">c</span>(comparisons2, comparisonName)</span>
<span id="cb189-558"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-558" aria-hidden="true" tabindex="-1"></a>                        selected<span class="sc">$</span>calls <span class="ot">=</span><span class="dv">0</span>   </span>
<span id="cb189-559"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-559" aria-hidden="true" tabindex="-1"></a>                        selected<span class="sc">$</span>calls [<span class="fu">which</span>( selected<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="fu">log2</span>(minFC_limma) <span class="sc">&amp;</span> selected<span class="sc">$</span>padj <span class="sc">&lt;</span> maxP_limma ) ]  <span class="ot">&lt;-</span>  <span class="dv">1</span></span>
<span id="cb189-560"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-560" aria-hidden="true" tabindex="-1"></a>                        selected<span class="sc">$</span>calls [ <span class="fu">which</span>( selected<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span>  <span class="sc">-</span><span class="fu">log2</span>(minFC_limma) <span class="sc">&amp;</span> selected<span class="sc">$</span>padj <span class="sc">&lt;</span> maxP_limma ) ] <span class="ot">&lt;-</span>  <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb189-561"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-561" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">colnames</span>(selected)<span class="ot">=</span> <span class="fu">paste</span>( comparisonName, <span class="st">&quot;___&quot;</span>,<span class="fu">colnames</span>(selected),<span class="at">sep=</span><span class="st">&quot;&quot;</span> )</span>
<span id="cb189-562"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-562" aria-hidden="true" tabindex="-1"></a>                        selected <span class="ot">=</span> <span class="fu">as.data.frame</span>(selected)</span>
<span id="cb189-563"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-563" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span> (pp<span class="sc">==</span><span class="dv">0</span>){  <span class="co"># if first one with significant genes, collect gene list and Pval+ fold</span></span>
<span id="cb189-564"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-564" aria-hidden="true" tabindex="-1"></a>                            result1 <span class="ot">=</span> selected; pp <span class="ot">=</span> <span class="dv">1</span>; </span>
<span id="cb189-565"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-565" aria-hidden="true" tabindex="-1"></a>                            <span class="co"># selected[,2] &lt;- -1 * selected[,2] # reverse fold change direction</span></span>
<span id="cb189-566"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-566" aria-hidden="true" tabindex="-1"></a>                            topGenes[[<span class="dv">1</span>]] <span class="ot">=</span> selected[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">6</span>)]; </span>
<span id="cb189-567"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-567" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">names</span>(topGenes)[<span class="dv">1</span>] <span class="ot">=</span> comparisonName; } <span class="cf">else</span> </span>
<span id="cb189-568"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-568" aria-hidden="true" tabindex="-1"></a>                            { result1 <span class="ot">=</span> <span class="fu">merge</span>(result1,selected,<span class="at">by=</span><span class="st">&quot;row.names&quot;</span>); </span>
<span id="cb189-569"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-569" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">rownames</span>(result1) <span class="ot">=</span> result1[,<span class="dv">1</span>]; </span>
<span id="cb189-570"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-570" aria-hidden="true" tabindex="-1"></a>                                result1 <span class="ot">&lt;-</span> result1[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb189-571"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-571" aria-hidden="true" tabindex="-1"></a>                                pk<span class="ot">=</span> pk<span class="sc">+</span><span class="dv">1</span>; </span>
<span id="cb189-572"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-572" aria-hidden="true" tabindex="-1"></a>                                <span class="co"># selected[,2] &lt;- -1 * selected[,2] # reverse fold change direction</span></span>
<span id="cb189-573"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-573" aria-hidden="true" tabindex="-1"></a>                                topGenes[[pk]] <span class="ot">=</span> selected[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">6</span>)]; </span>
<span id="cb189-574"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-574" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">names</span>(topGenes)[pk] <span class="ot">=</span> comparisonName;  <span class="co"># assign name to comprison</span></span>
<span id="cb189-575"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-575" aria-hidden="true" tabindex="-1"></a>                            }</span>
<span id="cb189-576"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-576" aria-hidden="true" tabindex="-1"></a>                    } <span class="co">#for  </span></span>
<span id="cb189-577"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-577" aria-hidden="true" tabindex="-1"></a>                        </span>
<span id="cb189-578"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-578" aria-hidden="true" tabindex="-1"></a>            } <span class="co">#if</span></span>
<span id="cb189-579"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-579" aria-hidden="true" tabindex="-1"></a>        } <span class="co">#for</span></span>
<span id="cb189-580"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-580" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-581"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-581" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-582"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-582" aria-hidden="true" tabindex="-1"></a>    } <span class="co">#if</span></span>
<span id="cb189-583"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-583" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-584"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-585"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-585" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-586"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-587"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-587" aria-hidden="true" tabindex="-1"></a><span class="co">#---</span></span>
<span id="cb189-588"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-588" aria-hidden="true" tabindex="-1"></a>    <span class="co">#if( length(comparisons) == 1) topGenes &lt;- topGenes[[1]] # if only one comparison, topGenes is not a list, just a data frame itself.</span></span>
<span id="cb189-589"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-589" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span> <span class="fu">is.null</span>(result1)) { </span>
<span id="cb189-590"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-590" aria-hidden="true" tabindex="-1"></a>    <span class="co"># note that when you only select 1 column from a data frame it automatically converts to a vector. drop =FALSE prevents that.</span></span>
<span id="cb189-591"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-591" aria-hidden="true" tabindex="-1"></a>    allCalls <span class="ot">=</span> <span class="fu">as.matrix</span>( result1[,<span class="fu">grep</span>(<span class="st">&quot;calls&quot;</span>,<span class="fu">colnames</span>(result1)), <span class="at">drop =</span> <span class="cn">FALSE</span>  ] )</span>
<span id="cb189-592"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-592" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(allCalls)<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;___.*&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(allCalls))</span>
<span id="cb189-593"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-593" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(allCalls)<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;-&quot;</span>, <span class="fu">colnames</span>(allCalls)) <span class="co"># note that samples names should have no &quot;.&quot;</span></span>
<span id="cb189-594"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-594" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(allCalls)<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;^I-&quot;</span>,<span class="st">&quot;I:&quot;</span>, <span class="fu">colnames</span>(allCalls))</span>
<span id="cb189-595"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-595" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-596"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-596" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">results=</span> allCalls, <span class="at">comparisons =</span> comparisons2, <span class="at">Exp.type=</span>Exp.type, <span class="at">topGenes=</span>topGenes)) </span>
<span id="cb189-597"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-597" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-598"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-598" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-599"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-599" aria-hidden="true" tabindex="-1"></a><span class="co"># main function</span></span>
<span id="cb189-600"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-600" aria-hidden="true" tabindex="-1"></a>limma <span class="ot">&lt;-</span> <span class="cf">function</span>() {  </span>
<span id="cb189-601"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-601" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_dataFileFormat <span class="sc">==</span> <span class="dv">1</span> ) {  <span class="co"># if count data</span></span>
<span id="cb189-602"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-602" aria-hidden="true" tabindex="-1"></a>         <span class="cf">if</span>(input_CountsDEGMethod <span class="sc">==</span> <span class="dv">3</span> ) {    <span class="co"># if DESeq2 method</span></span>
<span id="cb189-603"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-603" aria-hidden="true" tabindex="-1"></a>                <span class="co"># rawCounts = read.csv(&quot;exampleData/airway_GSE52778.csv&quot;, row.names=1)</span></span>
<span id="cb189-604"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-604" aria-hidden="true" tabindex="-1"></a>                <span class="co"># res =DEG.DESeq2(rawCounts, .05, 2) </span></span>
<span id="cb189-605"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-605" aria-hidden="true" tabindex="-1"></a>                <span class="co"># res1 =DEG.limma(rawCounts, .1, 1.5,rawCounts, 2,3) </span></span>
<span id="cb189-606"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-606" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb189-607"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-607" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>(   <span class="fu">DEG.DESeq2</span>( convertedCounts.out,input_limmaPval, input_limmaFC,</span>
<span id="cb189-608"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-608" aria-hidden="true" tabindex="-1"></a>                                    input_selectModelComprions, readSampleInfo.out,</span>
<span id="cb189-609"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-609" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">c</span>(input_selectFactorsModel,input_selectInteractions), </span>
<span id="cb189-610"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-610" aria-hidden="true" tabindex="-1"></a>                                    input_selectBlockFactorsModel, factorReferenceLevels.out )  )</span>
<span id="cb189-611"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-611" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb189-612"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-612" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(input_CountsDEGMethod <span class="sc">&lt;</span> <span class="dv">3</span> )    <span class="co"># voom or limma-trend</span></span>
<span id="cb189-613"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-613" aria-hidden="true" tabindex="-1"></a>                <span class="fu">return</span>( <span class="fu">DEG.limma</span>(convertedData.out, input_limmaPval, input_limmaFC,</span>
<span id="cb189-614"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-614" aria-hidden="true" tabindex="-1"></a>                                    convertedCounts.out, input_CountsDEGMethod,</span>
<span id="cb189-615"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-615" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">priorCounts=</span>input_countsLogStart,input_dataFileFormat,</span>
<span id="cb189-616"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-616" aria-hidden="true" tabindex="-1"></a>                                    input_selectModelComprions, readSampleInfo.out,</span>
<span id="cb189-617"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-617" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">c</span>(input_selectFactorsModel,input_selectInteractions),</span>
<span id="cb189-618"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-618" aria-hidden="true" tabindex="-1"></a>                                    input_selectBlockFactorsModel) )</span>
<span id="cb189-619"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-619" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (input_dataFileFormat <span class="sc">==</span> <span class="dv">2</span> ){ <span class="co"># normalized data</span></span>
<span id="cb189-620"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-620" aria-hidden="true" tabindex="-1"></a>     <span class="fu">return</span>( <span class="fu">DEG.limma</span>(convertedData.out, input_limmaPval, input_limmaFC,</span>
<span id="cb189-621"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-621" aria-hidden="true" tabindex="-1"></a>                        convertedCounts.out, input_CountsDEGMethod,</span>
<span id="cb189-622"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-622" aria-hidden="true" tabindex="-1"></a>                        <span class="at">priorCounts=</span>input_countsLogStart,input_dataFileFormat,</span>
<span id="cb189-623"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-623" aria-hidden="true" tabindex="-1"></a>                        input_selectModelComprions, readSampleInfo.out,</span>
<span id="cb189-624"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-624" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">c</span>(input_selectFactorsModel,input_selectInteractions),</span>
<span id="cb189-625"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-625" aria-hidden="true" tabindex="-1"></a>                        input_selectBlockFactorsModel) )</span>
<span id="cb189-626"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-626" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {   <span class="co"># dataFileFormat == 3 user just uploaded fold change matrix</span></span>
<span id="cb189-627"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-627" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-628"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-628" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">=</span> convertedData.out</span>
<span id="cb189-629"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-629" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-630"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-630" aria-hidden="true" tabindex="-1"></a>        pvals <span class="ot">=</span> convertedPvals.out</span>
<span id="cb189-631"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-631" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(pvals) ) {</span>
<span id="cb189-632"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-632" aria-hidden="true" tabindex="-1"></a>          ix <span class="ot">=</span> <span class="fu">match</span>(<span class="fu">rownames</span>(x), <span class="fu">rownames</span>(pvals))</span>
<span id="cb189-633"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-633" aria-hidden="true" tabindex="-1"></a>          pvals <span class="ot">=</span> pvals[ix,]</span>
<span id="cb189-634"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-634" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb189-635"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-635" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-636"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-636" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-637"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-637" aria-hidden="true" tabindex="-1"></a>        <span class="co"># looks like ratio data, take log2</span></span>
<span id="cb189-638"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-638" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">sum</span>(<span class="fu">round</span>(<span class="fu">apply</span>(x,<span class="dv">2</span>, median) <span class="sc">+</span> .<span class="dv">2</span>) <span class="sc">==</span> <span class="dv">1</span> ) <span class="sc">==</span> <span class="fu">dim</span>(x)[<span class="dv">2</span>] <span class="sc">&amp;</span> <span class="fu">min</span>(x) <span class="sc">&gt;</span> <span class="dv">0</span>) </span>
<span id="cb189-639"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-639" aria-hidden="true" tabindex="-1"></a>            x <span class="ot">=</span> <span class="fu">log2</span>(x)</span>
<span id="cb189-640"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-640" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-641"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-641" aria-hidden="true" tabindex="-1"></a>        Exp.type <span class="ot">=</span> <span class="st">&quot;None standard data without replicates.&quot;</span></span>
<span id="cb189-642"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-642" aria-hidden="true" tabindex="-1"></a>        all.Calls <span class="ot">=</span> x <span class="co"># fake calls</span></span>
<span id="cb189-643"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-643" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span> <span class="fu">dim</span>(all.Calls)[<span class="dv">2</span>]) { </span>
<span id="cb189-644"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-644" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">&lt;-</span> all.Calls[,i]</span>
<span id="cb189-645"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-645" aria-hidden="true" tabindex="-1"></a>            all.Calls[<span class="fu">which</span>( tem <span class="sc">&lt;=</span> <span class="fu">log2</span>(input_limmaFC) <span class="sc">&amp;</span> tem <span class="sc">&gt;=</span>  <span class="sc">-</span><span class="fu">log2</span>(input_limmaFC) ) ,i] <span class="ot">=</span> <span class="dv">0</span>            </span>
<span id="cb189-646"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-646" aria-hidden="true" tabindex="-1"></a>            all.Calls[<span class="fu">which</span>( tem <span class="sc">&gt;</span>  <span class="fu">log2</span>(input_limmaFC) ) ,i] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb189-647"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-647" aria-hidden="true" tabindex="-1"></a>            all.Calls[<span class="fu">which</span>( tem <span class="sc">&lt;</span> <span class="sc">-</span><span class="fu">log2</span>(input_limmaFC) ) ,i] <span class="ot">=</span> <span class="sc">-</span><span class="dv">1</span>      </span>
<span id="cb189-648"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-648" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(pvals) ) </span>
<span id="cb189-649"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-649" aria-hidden="true" tabindex="-1"></a>                all.Calls[ <span class="fu">which</span>( pvals[,i] <span class="sc">&gt;</span> input_limmaPval),i] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb189-650"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-650" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb189-651"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-651" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">=</span> <span class="fu">colnames</span>(all.Calls)</span>
<span id="cb189-652"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-652" aria-hidden="true" tabindex="-1"></a>        extractColumn <span class="ot">&lt;-</span> <span class="cf">function</span> (i) {</span>
<span id="cb189-653"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-653" aria-hidden="true" tabindex="-1"></a>            topGenes <span class="ot">=</span> <span class="fu">as.data.frame</span>( convertedData.out[,i,<span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb189-654"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-654" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">is.null</span>(pvals) ) topGenes<span class="sc">$</span>FDR <span class="ot">=</span> <span class="dv">0</span> <span class="cf">else</span> </span>
<span id="cb189-655"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-655" aria-hidden="true" tabindex="-1"></a>                topGenes<span class="sc">$</span>FDR <span class="ot">=</span> pvals[,i]<span class="co"># fake fdr</span></span>
<span id="cb189-656"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-656" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb189-657"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-657" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(topGenes) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb189-658"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-658" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>(topGenes)    </span>
<span id="cb189-659"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-659" aria-hidden="true" tabindex="-1"></a>        } </span>
<span id="cb189-660"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-660" aria-hidden="true" tabindex="-1"></a>        topGenes <span class="ot">=</span> <span class="fu">lapply</span>( <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>( x )[<span class="dv">2</span>], extractColumn )</span>
<span id="cb189-661"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-661" aria-hidden="true" tabindex="-1"></a>        topGenes <span class="ot">&lt;-</span> <span class="fu">setNames</span>(topGenes, <span class="fu">colnames</span>(x ) )</span>
<span id="cb189-662"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-662" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-663"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-663" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">results=</span> all.Calls, <span class="at">comparisons =</span> <span class="fu">colnames</span>(x ), <span class="at">Exp.type=</span>Exp.type, <span class="at">topGenes=</span>topGenes) )</span>
<span id="cb189-664"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-664" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-665"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-665" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-666"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-666" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-667"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-667" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-668"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-668" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-669"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-669" aria-hidden="true" tabindex="-1"></a>DEG.data <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb189-670"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-670" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-671"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-671" aria-hidden="true" tabindex="-1"></a>          genes <span class="ot">=</span> limma.out<span class="sc">$</span>results</span>
<span id="cb189-672"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-672" aria-hidden="true" tabindex="-1"></a>          genes <span class="ot">=</span> <span class="fu">as.data.frame</span>( genes[<span class="fu">which</span>( <span class="fu">rowSums</span>( <span class="fu">abs</span> (genes) ) <span class="sc">!=</span> <span class="dv">0</span> ),] )</span>
<span id="cb189-673"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-673" aria-hidden="true" tabindex="-1"></a>          <span class="fu">colnames</span>(genes) <span class="ot">=</span> <span class="fu">colnames</span>( limma.out<span class="sc">$</span>results )</span>
<span id="cb189-674"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-674" aria-hidden="true" tabindex="-1"></a>          genes <span class="ot">=</span> <span class="fu">merge</span>(genes,convertedData.out, <span class="at">by=</span><span class="st">&#39;row.names&#39;</span>)</span>
<span id="cb189-675"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-675" aria-hidden="true" tabindex="-1"></a>          <span class="fu">colnames</span>(genes)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">&quot;1: upregulation, -1: downregulation&quot;</span></span>
<span id="cb189-676"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-676" aria-hidden="true" tabindex="-1"></a>            <span class="co"># add gene symbol</span></span>
<span id="cb189-677"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-677" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">match</span>( genes[,<span class="dv">1</span>], allGeneInfo.out[,<span class="dv">1</span>])</span>
<span id="cb189-678"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-678" aria-hidden="true" tabindex="-1"></a>        genes <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">as.character</span>( allGeneInfo.out<span class="sc">$</span>symbol)[ix],genes) </span>
<span id="cb189-679"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-679" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(genes)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">&quot;Symbol&quot;</span></span>
<span id="cb189-680"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-680" aria-hidden="true" tabindex="-1"></a>        genes <span class="ot">&lt;-</span> genes[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">3</span><span class="sc">:</span><span class="fu">dim</span>(genes)[<span class="dv">2</span>]) ]</span>
<span id="cb189-681"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-681" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>(genes)</span>
<span id="cb189-682"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-682" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb189-683"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-683" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb189-684"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-684" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-685"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-685" aria-hidden="true" tabindex="-1"></a>vennPlot <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb189-686"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-686" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-687"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-687" aria-hidden="true" tabindex="-1"></a>            results <span class="ot">=</span> limma.out<span class="sc">$</span>results</span>
<span id="cb189-688"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-688" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-689"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-689" aria-hidden="true" tabindex="-1"></a>            <span class="co"># split by up or down regulation</span></span>
<span id="cb189-690"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-690" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(input_UpDownRegulated) {     </span>
<span id="cb189-691"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-691" aria-hidden="true" tabindex="-1"></a>                resultUp <span class="ot">=</span> results; </span>
<span id="cb189-692"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-692" aria-hidden="true" tabindex="-1"></a>                resultUp[resultUp <span class="sc">&lt;</span> <span class="dv">0</span> ] <span class="ot">&lt;-</span> <span class="dv">0</span>;</span>
<span id="cb189-693"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-693" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(resultUp) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">&quot;Up_&quot;</span>, <span class="fu">colnames</span>(resultUp))</span>
<span id="cb189-694"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-694" aria-hidden="true" tabindex="-1"></a>                resultDown <span class="ot">=</span> results; </span>
<span id="cb189-695"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-695" aria-hidden="true" tabindex="-1"></a>                resultDown[resultDown <span class="sc">&gt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="dv">0</span>;</span>
<span id="cb189-696"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-696" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(resultDown) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">&quot;Down_&quot;</span>, <span class="fu">colnames</span>(resultDown))                </span>
<span id="cb189-697"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-697" aria-hidden="true" tabindex="-1"></a>                results <span class="ot">&lt;-</span> <span class="fu">cbind</span>(resultUp, resultDown)</span>
<span id="cb189-698"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-698" aria-hidden="true" tabindex="-1"></a>            }           </span>
<span id="cb189-699"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-699" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-700"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-700" aria-hidden="true" tabindex="-1"></a>            ixa <span class="ot">=</span> <span class="fu">c</span>()</span>
<span id="cb189-701"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-701" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-702"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-702" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-703"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-703" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> (comps <span class="cf">in</span>  input_selectComparisonsVenn) { </span>
<span id="cb189-704"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-704" aria-hidden="true" tabindex="-1"></a>                 <span class="cf">if</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;^I:|^I-|^Up_I:|^Up_I-|^Down_I:|^Down_I-&quot;</span>, comps) ) {  <span class="co"># if not interaction term</span></span>
<span id="cb189-705"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-705" aria-hidden="true" tabindex="-1"></a>                    ix <span class="ot">=</span> <span class="fu">match</span>(comps, <span class="fu">colnames</span>(results)) </span>
<span id="cb189-706"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-706" aria-hidden="true" tabindex="-1"></a>                  } <span class="cf">else</span> {</span>
<span id="cb189-707"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-707" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#mismatch in comparison names for interaction terms for DESeq2</span></span>
<span id="cb189-708"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-708" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#I:water_Wet.genetic_Hy      in the selected Contrast</span></span>
<span id="cb189-709"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-709" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#Diff-water_Wet-genetic_Hy   in column names</span></span>
<span id="cb189-710"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-710" aria-hidden="true" tabindex="-1"></a>                        tem <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;^I-&quot;</span>,<span class="st">&quot;I:&quot;</span> ,<span class="fu">colnames</span>(results))</span>
<span id="cb189-711"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-711" aria-hidden="true" tabindex="-1"></a>                        tem <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,tem)</span>
<span id="cb189-712"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-712" aria-hidden="true" tabindex="-1"></a>                        ix <span class="ot">=</span> <span class="fu">match</span>(comps, tem) </span>
<span id="cb189-713"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-713" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-714"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-714" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>(<span class="fu">is.na</span>(ix) ) <span class="co"># this is for limma package</span></span>
<span id="cb189-715"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-715" aria-hidden="true" tabindex="-1"></a>                            ix <span class="ot">=</span> <span class="fu">match</span>(comps, <span class="fu">colnames</span>(results))                        </span>
<span id="cb189-716"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-716" aria-hidden="true" tabindex="-1"></a>                      }</span>
<span id="cb189-717"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-717" aria-hidden="true" tabindex="-1"></a>                    ixa <span class="ot">=</span> <span class="fu">c</span>(ixa,ix)</span>
<span id="cb189-718"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-718" aria-hidden="true" tabindex="-1"></a>                  }</span>
<span id="cb189-719"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-719" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb189-720"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-720" aria-hidden="true" tabindex="-1"></a>            results <span class="ot">=</span> results[,ixa,drop<span class="ot">=</span><span class="cn">FALSE</span>] <span class="co"># only use selected comparisons</span></span>
<span id="cb189-721"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-721" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(results)[<span class="dv">2</span>] <span class="sc">&gt;</span><span class="dv">5</span>) results <span class="ot">&lt;-</span> results[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb189-722"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-722" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(results) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;^I-&quot;</span>,<span class="st">&quot;I:&quot;</span> ,<span class="fu">colnames</span>(results)) </span>
<span id="cb189-723"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-723" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-724"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-724" aria-hidden="true" tabindex="-1"></a>            <span class="fu">vennDiagram</span>(results,<span class="at">circle.col=</span><span class="fu">rainbow</span>(<span class="dv">5</span>), <span class="at">cex=</span><span class="fu">c</span>(<span class="fl">1.</span>,<span class="dv">1</span>, <span class="fl">0.7</span>) ) <span class="co"># part of limma package</span></span>
<span id="cb189-725"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-725" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-726"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-726" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-727"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-727" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb189-728"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-728" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-729"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-729" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-730"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-730" aria-hidden="true" tabindex="-1"></a>sigGeneStats <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb189-731"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-731" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-732"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-732" aria-hidden="true" tabindex="-1"></a>        results <span class="ot">=</span> limma.out<span class="sc">$</span>results</span>
<span id="cb189-733"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-733" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-734"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-734" aria-hidden="true" tabindex="-1"></a>         <span class="fu">library</span>(reshape2)</span>
<span id="cb189-735"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-735" aria-hidden="true" tabindex="-1"></a>         Up <span class="ot">=</span>  <span class="fu">apply</span>(results, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">sum</span>(x <span class="sc">==</span> <span class="dv">1</span>) )</span>
<span id="cb189-736"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-736" aria-hidden="true" tabindex="-1"></a>         Down <span class="ot">=</span> <span class="fu">apply</span>(results, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">sum</span>(x <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>) ) </span>
<span id="cb189-737"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-737" aria-hidden="true" tabindex="-1"></a>         stats <span class="ot">=</span> <span class="fu">rbind</span>(Up, Down)</span>
<span id="cb189-738"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-738" aria-hidden="true" tabindex="-1"></a>                 </span>
<span id="cb189-739"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-739" aria-hidden="true" tabindex="-1"></a>         gg <span class="ot">&lt;-</span> <span class="fu">melt</span>(stats)</span>
<span id="cb189-740"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-740" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-741"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-741" aria-hidden="true" tabindex="-1"></a>         <span class="fu">colnames</span>(gg) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Regulation&quot;</span>,<span class="st">&quot;Comparisons&quot;</span>,<span class="st">&quot;Genes&quot;</span>)</span>
<span id="cb189-742"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-742" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb189-743"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-743" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb189-744"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-744" aria-hidden="true" tabindex="-1"></a>         p<span class="ot">=</span> <span class="fu">ggplot</span>(gg, <span class="fu">aes</span>(<span class="at">x=</span>Comparisons, <span class="at">y=</span>Genes, <span class="at">fill=</span>  Regulation )  )<span class="sc">+</span></span>
<span id="cb189-745"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-745" aria-hidden="true" tabindex="-1"></a>             <span class="fu">geom_bar</span>(<span class="at">position=</span><span class="st">&quot;dodge&quot;</span>, <span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb189-746"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-746" aria-hidden="true" tabindex="-1"></a>             <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;top&quot;</span>) <span class="sc">+</span> </span>
<span id="cb189-747"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-747" aria-hidden="true" tabindex="-1"></a>             <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)) <span class="sc">+</span></span>
<span id="cb189-748"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-748" aria-hidden="true" tabindex="-1"></a>             <span class="fu">ylab</span>(<span class="st">&quot;Number of differntially expressed genes&quot;</span>) <span class="sc">+</span></span>
<span id="cb189-749"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-749" aria-hidden="true" tabindex="-1"></a>             <span class="fu">theme</span>(<span class="at">axis.title.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb189-750"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-750" aria-hidden="true" tabindex="-1"></a>                <span class="at">axis.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">14</span>)) </span>
<span id="cb189-751"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-751" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-752"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-752" aria-hidden="true" tabindex="-1"></a>        p</span>
<span id="cb189-753"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-753" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-754"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-754" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-755"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-755" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-756"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-756" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-757"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-757" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-758"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-758" aria-hidden="true" tabindex="-1"></a>sigGeneStatsTable <span class="ot">&lt;-</span> <span class="cf">function</span>( ) {</span>
<span id="cb189-759"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-759" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-760"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-760" aria-hidden="true" tabindex="-1"></a>        results <span class="ot">=</span> limma.out<span class="sc">$</span>results</span>
<span id="cb189-761"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-761" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-762"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-762" aria-hidden="true" tabindex="-1"></a>         Up <span class="ot">=</span>  <span class="fu">apply</span>(results, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">sum</span>(x <span class="sc">==</span> <span class="dv">1</span>) )</span>
<span id="cb189-763"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-763" aria-hidden="true" tabindex="-1"></a>         Down <span class="ot">=</span> <span class="fu">apply</span>(results, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">sum</span>(x <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>) ) </span>
<span id="cb189-764"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-764" aria-hidden="true" tabindex="-1"></a>         stats <span class="ot">=</span> <span class="fu">rbind</span>(Up, Down)</span>
<span id="cb189-765"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-765" aria-hidden="true" tabindex="-1"></a>         stats <span class="ot">=</span> <span class="fu">t</span>(stats)</span>
<span id="cb189-766"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-766" aria-hidden="true" tabindex="-1"></a>         stats<span class="ot">=</span><span class="fu">cbind</span>(<span class="fu">rownames</span>(stats), stats)</span>
<span id="cb189-767"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-767" aria-hidden="true" tabindex="-1"></a>         <span class="fu">colnames</span>(stats)[<span class="dv">1</span>]<span class="ot">=</span><span class="st">&quot;Comparisons&quot;</span></span>
<span id="cb189-768"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-768" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-769"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-769" aria-hidden="true" tabindex="-1"></a>         <span class="fu">return</span>(<span class="fu">as.data.frame</span>(stats))</span>
<span id="cb189-770"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-770" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-771"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-771" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-772"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-772" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-773"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-773" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-774"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-774" aria-hidden="true" tabindex="-1"></a>selectedHeatmap.data <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb189-775"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-775" aria-hidden="true" tabindex="-1"></a>          genes <span class="ot">&lt;-</span> limma.out<span class="sc">$</span>results</span>
<span id="cb189-776"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-776" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>( <span class="fu">is.null</span>(genes) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb189-777"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-777" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;I:&quot;</span>, input_selectContrast) ) {  <span class="co"># if not interaction term</span></span>
<span id="cb189-778"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-778" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, <span class="fu">colnames</span>(genes)) </span>
<span id="cb189-779"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-779" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> {</span>
<span id="cb189-780"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-780" aria-hidden="true" tabindex="-1"></a>            <span class="co">#mismatch in comparison names for interaction terms for DESeq2</span></span>
<span id="cb189-781"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-781" aria-hidden="true" tabindex="-1"></a>            <span class="co">#I:water_Wet.genetic_Hy      in the selected Contrast</span></span>
<span id="cb189-782"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-782" aria-hidden="true" tabindex="-1"></a>            <span class="co">#Diff-water_Wet-genetic_Hy   in column names</span></span>
<span id="cb189-783"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-783" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;I-&quot;</span>,<span class="st">&quot;I:&quot;</span> ,<span class="fu">colnames</span>(genes))</span>
<span id="cb189-784"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-784" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,tem)</span>
<span id="cb189-785"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-785" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, tem) </span>
<span id="cb189-786"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-786" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-787"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-787" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">is.na</span>(ix) ) <span class="co"># this is for limma package</span></span>
<span id="cb189-788"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-788" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, <span class="fu">colnames</span>(genes))           </span>
<span id="cb189-789"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-789" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-790"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-790" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb189-791"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-791" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb189-792"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-792" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">is.null</span>(ix)) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb189-793"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-793" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">is.na</span>(ix)) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb189-794"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-794" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>( <span class="fu">sum</span>(<span class="fu">abs</span>(genes[,ix] )  ) <span class="sc">&lt;=</span> <span class="dv">1</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>) <span class="co"># no significant genes for this comparison</span></span>
<span id="cb189-795"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-795" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">dim</span>(genes)[<span class="dv">2</span>] <span class="sc">&lt;</span> ix ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb189-796"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-796" aria-hidden="true" tabindex="-1"></a>          query <span class="ot">=</span> <span class="fu">rownames</span>(genes)[<span class="fu">which</span>(genes[,ix] <span class="sc">!=</span> <span class="dv">0</span>)]</span>
<span id="cb189-797"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-797" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">length</span>(query) <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb189-798"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-798" aria-hidden="true" tabindex="-1"></a>          iy <span class="ot">=</span> <span class="fu">match</span>(query, <span class="fu">rownames</span>(convertedData.out  ) )</span>
<span id="cb189-799"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-799" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb189-800"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-800" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-801"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-801" aria-hidden="true" tabindex="-1"></a>         iz <span class="ot">=</span> <span class="fu">findContrastSamples</span>(  input_selectContrast, </span>
<span id="cb189-802"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-802" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">colnames</span>(convertedData.out),</span>
<span id="cb189-803"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-803" aria-hidden="true" tabindex="-1"></a>                                    readSampleInfo.out,</span>
<span id="cb189-804"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-804" aria-hidden="true" tabindex="-1"></a>                                    input_selectFactorsModel,</span>
<span id="cb189-805"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-805" aria-hidden="true" tabindex="-1"></a>                                    input_selectModelComprions, </span>
<span id="cb189-806"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-806" aria-hidden="true" tabindex="-1"></a>                                    factorReferenceLevels.out,</span>
<span id="cb189-807"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-807" aria-hidden="true" tabindex="-1"></a>                                    input_CountsDEGMethod,</span>
<span id="cb189-808"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-808" aria-hidden="true" tabindex="-1"></a>                                    input_dataFileFormat</span>
<span id="cb189-809"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-809" aria-hidden="true" tabindex="-1"></a>                                )</span>
<span id="cb189-810"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-810" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-811"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-811" aria-hidden="true" tabindex="-1"></a>        <span class="co"># color bar</span></span>
<span id="cb189-812"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-812" aria-hidden="true" tabindex="-1"></a>         bar <span class="ot">=</span> genes[,ix]</span>
<span id="cb189-813"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-813" aria-hidden="true" tabindex="-1"></a>         bar <span class="ot">=</span> bar[bar<span class="sc">!=</span><span class="dv">0</span>]</span>
<span id="cb189-814"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-814" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-815"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-815" aria-hidden="true" tabindex="-1"></a>         <span class="co"># retreive related data         </span></span>
<span id="cb189-816"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-816" aria-hidden="true" tabindex="-1"></a>         genes <span class="ot">=</span> convertedData.out[iy,iz,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb189-817"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-817" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb189-818"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-818" aria-hidden="true" tabindex="-1"></a>         genes <span class="ot">=</span> genes[<span class="fu">order</span>(bar),,drop<span class="ot">=</span><span class="cn">FALSE</span>] <span class="co"># needs to be sorted because myheatmap2 does not reorder genes</span></span>
<span id="cb189-819"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-819" aria-hidden="true" tabindex="-1"></a>         bar <span class="ot">=</span> <span class="fu">sort</span>(bar)</span>
<span id="cb189-820"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-820" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-821"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-821" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-822"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-822" aria-hidden="true" tabindex="-1"></a>         <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">genes=</span>genes, <span class="at">bar=</span>bar ))</span>
<span id="cb189-823"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-823" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-824"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-824" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-825"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-825" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-826"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-826" aria-hidden="true" tabindex="-1"></a><span class="co"># find sample index for selected comparisons</span></span>
<span id="cb189-827"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-827" aria-hidden="true" tabindex="-1"></a>findContrastSamples <span class="ot">&lt;-</span> <span class="cf">function</span>(selectContrast, allSampleNames,<span class="at">sampleInfo=</span><span class="cn">NULL</span>, <span class="at">selectFactorsModel=</span><span class="cn">NULL</span>,<span class="at">selectModelComprions =</span><span class="cn">NULL</span> , <span class="at">referenceLevels=</span><span class="cn">NULL</span>, <span class="at">countsDEGMethod=</span><span class="cn">NULL</span>, <span class="at">dataFileFormat=</span><span class="cn">NULL</span> ){</span>
<span id="cb189-828"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-828" aria-hidden="true" tabindex="-1"></a>    iz<span class="ot">=</span> <span class="fu">match</span>( <span class="fu">detectGroups</span>(allSampleNames), <span class="fu">unlist</span>(<span class="fu">strsplit</span>( selectContrast, <span class="st">&quot;-&quot;</span>))   )</span>
<span id="cb189-829"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-829" aria-hidden="true" tabindex="-1"></a>    iz <span class="ot">=</span> <span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(iz))       </span>
<span id="cb189-830"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-830" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> ( <span class="sc">!</span><span class="fu">is.null</span>(sampleInfo) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(selectFactorsModel) <span class="sc">&amp;</span> <span class="fu">length</span>(selectModelComprions)<span class="sc">&gt;</span><span class="dv">0</span> ) {</span>
<span id="cb189-831"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-831" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-832"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-832" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;.*: &quot;</span>,<span class="st">&quot;&quot;</span>,selectModelComprions)   <span class="co"># strings like: &quot;groups: mutant vs. control&quot;</span></span>
<span id="cb189-833"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-833" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; vs</span><span class="sc">\\</span><span class="st">. &quot;</span>,<span class="st">&quot;-&quot;</span>,comparisons)       </span>
<span id="cb189-834"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-834" aria-hidden="true" tabindex="-1"></a>        factorsVector<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;:.*&quot;</span>,<span class="st">&quot;&quot;</span>,selectModelComprions) <span class="co"># corresponding factors</span></span>
<span id="cb189-835"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-835" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-836"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-836" aria-hidden="true" tabindex="-1"></a>          <span class="co"># if read counts data and DESeq2</span></span>
<span id="cb189-837"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-837" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(dataFileFormat<span class="sc">==</span><span class="dv">1</span> <span class="sc">&amp;</span> countsDEGMethod <span class="sc">==</span> <span class="dv">3</span>) { <span class="co"># if DESeq2</span></span>
<span id="cb189-838"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-838" aria-hidden="true" tabindex="-1"></a>            contrast <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;_for_.*&quot;</span>,<span class="st">&quot;&quot;</span>,selectContrast) <span class="co"># could be &quot;wt-mu&quot;   or &quot;wt-mu_for_conditionB&quot;</span></span>
<span id="cb189-839"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-839" aria-hidden="true" tabindex="-1"></a>            ik <span class="ot">=</span> <span class="fu">match</span>( contrast, comparisons )   <span class="co"># selected contrast lookes like: &quot;mutant-control&quot;</span></span>
<span id="cb189-840"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-840" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-841"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-841" aria-hidden="true" tabindex="-1"></a>            otherFactorLevel <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;.*_for_&quot;</span>,<span class="st">&quot;&quot;</span>,selectContrast)</span>
<span id="cb189-842"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-842" aria-hidden="true" tabindex="-1"></a>            <span class="co"># find the corresponding factor for the other factor</span></span>
<span id="cb189-843"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-843" aria-hidden="true" tabindex="-1"></a>            otherFactor<span class="ot">=</span><span class="st">&quot; &quot;</span></span>
<span id="cb189-844"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-844" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">nchar</span>( otherFactorLevel ) <span class="sc">&gt;</span><span class="dv">0</span>){</span>
<span id="cb189-845"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-845" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span>( eachFactor <span class="cf">in</span> <span class="fu">colnames</span>(sampleInfo) )</span>
<span id="cb189-846"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-846" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span> ( otherFactorLevel <span class="sc">%in%</span>  sampleInfo[,eachFactor ] )</span>
<span id="cb189-847"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-847" aria-hidden="true" tabindex="-1"></a>                        otherFactor <span class="ot">=</span> eachFactor        </span>
<span id="cb189-848"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-848" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb189-849"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-849" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-850"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-850" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (<span class="fu">is.na</span>(ik)) iz<span class="ot">=</span><span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(allSampleNames))  <span class="cf">else</span> {  <span class="co"># interaction term, use all samples       </span></span>
<span id="cb189-851"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-851" aria-hidden="true" tabindex="-1"></a>                selectedfactor<span class="ot">=</span> factorsVector[ ik ] <span class="co"># corresponding factors</span></span>
<span id="cb189-852"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-852" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-853"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-853" aria-hidden="true" tabindex="-1"></a>                iz <span class="ot">=</span> <span class="fu">which</span>(sampleInfo[,selectedfactor] <span class="sc">%in%</span>  <span class="fu">unlist</span>(<span class="fu">strsplit</span>( contrast, <span class="st">&quot;-&quot;</span>)) )</span>
<span id="cb189-854"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-854" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-855"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-855" aria-hidden="true" tabindex="-1"></a>                <span class="co">#filter by other factors: reference level</span></span>
<span id="cb189-856"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-856" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="sc">!</span> <span class="fu">is.null</span>( referenceLevels) ) {   <span class="co"># c(&quot;genotype:wt&quot;, &quot;treatment:control&quot; )</span></span>
<span id="cb189-857"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-857" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span> ( refs <span class="cf">in</span> referenceLevels)</span>
<span id="cb189-858"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-858" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>(<span class="sc">!</span> <span class="fu">is.null</span>( refs) <span class="sc">&amp;</span> <span class="fu">gsub</span>(<span class="st">&quot;:.*&quot;</span>,<span class="st">&quot;&quot;</span>,refs) <span class="sc">!=</span> selectedfactor ) {</span>
<span id="cb189-859"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-859" aria-hidden="true" tabindex="-1"></a>                            currentFactor <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;:.*&quot;</span>,<span class="st">&quot;&quot;</span>,refs)</span>
<span id="cb189-860"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-860" aria-hidden="true" tabindex="-1"></a>                            <span class="cf">if</span>(<span class="fu">nchar</span>( otherFactorLevel ) <span class="sc">&gt;</span><span class="dv">0</span> <span class="sc">&amp;</span> currentFactor <span class="sc">==</span> otherFactor ) { <span class="co"># if not reference level</span></span>
<span id="cb189-861"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-861" aria-hidden="true" tabindex="-1"></a>                                iz <span class="ot">=</span> <span class="fu">intersect</span>( iz, <span class="fu">which</span>(sampleInfo[,currentFactor] <span class="sc">==</span> otherFactorLevel  ) )</span>
<span id="cb189-862"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-862" aria-hidden="true" tabindex="-1"></a>                            } <span class="cf">else</span></span>
<span id="cb189-863"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-863" aria-hidden="true" tabindex="-1"></a>                                iz <span class="ot">=</span> <span class="fu">intersect</span>( iz, <span class="fu">which</span>(sampleInfo[,currentFactor] <span class="sc">==</span> <span class="fu">gsub</span>(<span class="st">&quot;.*:&quot;</span>,<span class="st">&quot;&quot;</span>,refs)  ) )</span>
<span id="cb189-864"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-864" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb189-865"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-865" aria-hidden="true" tabindex="-1"></a>                        }</span>
<span id="cb189-866"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-866" aria-hidden="true" tabindex="-1"></a>                }           </span>
<span id="cb189-867"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-867" aria-hidden="true" tabindex="-1"></a>                iz <span class="ot">=</span> iz[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(iz))]</span>
<span id="cb189-868"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-868" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb189-869"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-869" aria-hidden="true" tabindex="-1"></a>            <span class="co"># switching from limma to DESeq2 causes problem, as reference level is not defined.</span></span>
<span id="cb189-870"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-870" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-871"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-871" aria-hidden="true" tabindex="-1"></a>            } </span>
<span id="cb189-872"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-872" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-873"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-873" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {  <span class="co"># not DESeq2</span></span>
<span id="cb189-874"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-874" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-875"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-875" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># given level find corresponding sample ids</span></span>
<span id="cb189-876"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-876" aria-hidden="true" tabindex="-1"></a>                    findIDsFromLevel <span class="ot">&lt;-</span> <span class="cf">function</span> (aLevel){</span>
<span id="cb189-877"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-877" aria-hidden="true" tabindex="-1"></a>                        <span class="co"># find factor</span></span>
<span id="cb189-878"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-878" aria-hidden="true" tabindex="-1"></a>                        currentFactor<span class="ot">=</span><span class="st">&quot;&quot;</span></span>
<span id="cb189-879"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-879" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">for</span>( eachFactor <span class="cf">in</span> <span class="fu">colnames</span>(sampleInfo) )</span>
<span id="cb189-880"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-880" aria-hidden="true" tabindex="-1"></a>                            <span class="cf">if</span> ( aLevel <span class="sc">%in%</span>  sampleInfo[,eachFactor ] )</span>
<span id="cb189-881"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-881" aria-hidden="true" tabindex="-1"></a>                                currentFactor <span class="ot">=</span> eachFactor          </span>
<span id="cb189-882"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-882" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>(<span class="fu">nchar</span>(currentFactor) <span class="sc">&gt;</span><span class="dv">0</span> ) </span>
<span id="cb189-883"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-883" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">return</span>( <span class="fu">which</span>(sampleInfo[,currentFactor ] <span class="sc">%in%</span> aLevel   )  ) <span class="cf">else</span> <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb189-884"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-884" aria-hidden="true" tabindex="-1"></a>                    }       </span>
<span id="cb189-885"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-885" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb189-886"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-886" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>( <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;.*_.*-.*_.*&quot;</span>,selectContrast )) iz <span class="ot">=</span> c.out</span>
<span id="cb189-887"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-887" aria-hidden="true" tabindex="-1"></a>                    levels4 <span class="ot">=</span> <span class="fu">unlist</span>( <span class="fu">strsplit</span>( <span class="fu">unlist</span>( <span class="fu">strsplit</span>(selectContrast,<span class="st">&quot;-&quot;</span>) ), <span class="st">&quot;_&quot;</span>) ) <span class="co">#double split!</span></span>
<span id="cb189-888"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-888" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>(<span class="fu">length</span>(levels4)<span class="sc">!=</span><span class="dv">4</span>) { </span>
<span id="cb189-889"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-889" aria-hidden="true" tabindex="-1"></a>                        iz <span class="ot">=</span> c.out </span>
<span id="cb189-890"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-890" aria-hidden="true" tabindex="-1"></a>                    } <span class="cf">else</span> {</span>
<span id="cb189-891"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-891" aria-hidden="true" tabindex="-1"></a>                        iz <span class="ot">=</span> <span class="fu">intersect</span>( <span class="fu">findIDsFromLevel</span>(levels4[<span class="dv">1</span>]),  <span class="fu">findIDsFromLevel</span>(levels4[<span class="dv">2</span>])  ) <span class="co"># first sample</span></span>
<span id="cb189-892"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-892" aria-hidden="true" tabindex="-1"></a>                        iz <span class="ot">=</span> <span class="fu">c</span>(iz, <span class="fu">intersect</span>( <span class="fu">findIDsFromLevel</span>(levels4[<span class="dv">3</span>]),  <span class="fu">findIDsFromLevel</span>(levels4[<span class="dv">4</span>])  )   ) <span class="co"># 2nd sample</span></span>
<span id="cb189-893"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-893" aria-hidden="true" tabindex="-1"></a>                        }</span>
<span id="cb189-894"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-894" aria-hidden="true" tabindex="-1"></a>                } <span class="co">#else</span></span>
<span id="cb189-895"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-895" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-896"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-896" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-897"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-897" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-898"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-898" aria-hidden="true" tabindex="-1"></a>        }   </span>
<span id="cb189-899"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-899" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb189-900"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-900" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span> (<span class="fu">grepl</span>(<span class="st">&quot;I:&quot;</span>,selectContrast)) iz<span class="ot">=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(allSampleNames) <span class="co"># if it is factor design use all samples</span></span>
<span id="cb189-901"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-901" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>( <span class="fu">is.na</span>(iz)[<span class="dv">1</span>] <span class="sc">|</span> <span class="fu">length</span>(iz)<span class="sc">&lt;=</span><span class="dv">1</span> )    iz<span class="ot">=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(allSampleNames)</span>
<span id="cb189-902"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-902" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-903"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-903" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(iz)</span>
<span id="cb189-904"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-904" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-905"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-905" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-906"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-906" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-907"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-907" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-908"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-908" aria-hidden="true" tabindex="-1"></a>selectedHeatmap <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb189-909"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-909" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-910"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-910" aria-hidden="true" tabindex="-1"></a>         bar <span class="ot">=</span> selectedHeatmap.data.out<span class="sc">$</span>bar <span class="sc">+</span><span class="dv">2</span>;</span>
<span id="cb189-911"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-911" aria-hidden="true" tabindex="-1"></a>         bar[bar<span class="sc">==</span><span class="dv">3</span>] <span class="ot">=</span><span class="dv">2</span></span>
<span id="cb189-912"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-912" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-913"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-913" aria-hidden="true" tabindex="-1"></a>         <span class="fu">myheatmap2</span>( selectedHeatmap.data.out<span class="sc">$</span>genes,bar,<span class="dv">200</span>,<span class="at">mycolor=</span>input_heatColors1,<span class="fu">c</span>(<span class="st">&quot;Down&quot;</span>,<span class="st">&quot;Up&quot;</span>) )</span>
<span id="cb189-914"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-914" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb189-915"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-915" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-916"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-916" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-917"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-917" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-918"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-918" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-919"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-919" aria-hidden="true" tabindex="-1"></a>selectedHeatmap.data <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb189-920"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-920" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-921"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-921" aria-hidden="true" tabindex="-1"></a>          genes <span class="ot">&lt;-</span> limma.out<span class="sc">$</span>results</span>
<span id="cb189-922"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-922" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>( <span class="fu">is.null</span>(genes) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb189-923"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-923" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;I:&quot;</span>, input_selectContrast) ) {  <span class="co"># if not interaction term</span></span>
<span id="cb189-924"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-924" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, <span class="fu">colnames</span>(genes)) </span>
<span id="cb189-925"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-925" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> {</span>
<span id="cb189-926"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-926" aria-hidden="true" tabindex="-1"></a>            <span class="co">#mismatch in comparison names for interaction terms for DESeq2</span></span>
<span id="cb189-927"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-927" aria-hidden="true" tabindex="-1"></a>            <span class="co">#I:water_Wet.genetic_Hy      in the selected Contrast</span></span>
<span id="cb189-928"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-928" aria-hidden="true" tabindex="-1"></a>            <span class="co">#Diff-water_Wet-genetic_Hy   in column names</span></span>
<span id="cb189-929"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-929" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;I-&quot;</span>,<span class="st">&quot;I:&quot;</span> ,<span class="fu">colnames</span>(genes))</span>
<span id="cb189-930"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-930" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,tem)</span>
<span id="cb189-931"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-931" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, tem) </span>
<span id="cb189-932"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-932" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-933"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-933" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">is.na</span>(ix) ) <span class="co"># this is for limma package</span></span>
<span id="cb189-934"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-934" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, <span class="fu">colnames</span>(genes))           </span>
<span id="cb189-935"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-935" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-936"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-936" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb189-937"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-937" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb189-938"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-938" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">is.null</span>(ix)) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb189-939"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-939" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">is.na</span>(ix)) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb189-940"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-940" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>( <span class="fu">sum</span>(<span class="fu">abs</span>(genes[,ix] )  ) <span class="sc">&lt;=</span> <span class="dv">1</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>) <span class="co"># no significant genes for this comparison</span></span>
<span id="cb189-941"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-941" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">dim</span>(genes)[<span class="dv">2</span>] <span class="sc">&lt;</span> ix ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb189-942"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-942" aria-hidden="true" tabindex="-1"></a>          query <span class="ot">=</span> <span class="fu">rownames</span>(genes)[<span class="fu">which</span>(genes[,ix] <span class="sc">!=</span> <span class="dv">0</span>)]</span>
<span id="cb189-943"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-943" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">length</span>(query) <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb189-944"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-944" aria-hidden="true" tabindex="-1"></a>          iy <span class="ot">=</span> <span class="fu">match</span>(query, <span class="fu">rownames</span>(convertedData.out  ) )</span>
<span id="cb189-945"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-945" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb189-946"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-946" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-947"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-947" aria-hidden="true" tabindex="-1"></a>         iz <span class="ot">=</span> <span class="fu">findContrastSamples</span>(  input_selectContrast, </span>
<span id="cb189-948"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-948" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">colnames</span>(convertedData.out),</span>
<span id="cb189-949"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-949" aria-hidden="true" tabindex="-1"></a>                                    readSampleInfo.out,</span>
<span id="cb189-950"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-950" aria-hidden="true" tabindex="-1"></a>                                    input_selectFactorsModel,</span>
<span id="cb189-951"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-951" aria-hidden="true" tabindex="-1"></a>                                    input_selectModelComprions, </span>
<span id="cb189-952"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-952" aria-hidden="true" tabindex="-1"></a>                                    factorReferenceLevels.out,</span>
<span id="cb189-953"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-953" aria-hidden="true" tabindex="-1"></a>                                    input_CountsDEGMethod,</span>
<span id="cb189-954"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-954" aria-hidden="true" tabindex="-1"></a>                                    input_dataFileFormat</span>
<span id="cb189-955"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-955" aria-hidden="true" tabindex="-1"></a>                                )</span>
<span id="cb189-956"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-956" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-957"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-957" aria-hidden="true" tabindex="-1"></a>        <span class="co"># color bar</span></span>
<span id="cb189-958"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-958" aria-hidden="true" tabindex="-1"></a>         bar <span class="ot">=</span> genes[,ix]</span>
<span id="cb189-959"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-959" aria-hidden="true" tabindex="-1"></a>         bar <span class="ot">=</span> bar[bar<span class="sc">!=</span><span class="dv">0</span>]</span>
<span id="cb189-960"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-960" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-961"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-961" aria-hidden="true" tabindex="-1"></a>         <span class="co"># retreive related data         </span></span>
<span id="cb189-962"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-962" aria-hidden="true" tabindex="-1"></a>         genes <span class="ot">=</span> <span class="fu">convertedData</span>()[iy,iz,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb189-963"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-963" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb189-964"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-964" aria-hidden="true" tabindex="-1"></a>         genes <span class="ot">=</span> genes[<span class="fu">order</span>(bar),,drop<span class="ot">=</span><span class="cn">FALSE</span>] <span class="co"># needs to be sorted because myheatmap2 does not reorder genes</span></span>
<span id="cb189-965"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-965" aria-hidden="true" tabindex="-1"></a>         bar <span class="ot">=</span> <span class="fu">sort</span>(bar)</span>
<span id="cb189-966"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-966" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-967"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-967" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-968"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-968" aria-hidden="true" tabindex="-1"></a>         <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">genes=</span>genes, <span class="at">bar=</span>bar ))</span>
<span id="cb189-969"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-969" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-970"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-970" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-971"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-971" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-972"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-972" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-973"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-973" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-974"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-974" aria-hidden="true" tabindex="-1"></a>DEG.data <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb189-975"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-975" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-976"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-976" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">=</span> limma.out<span class="sc">$</span>results</span>
<span id="cb189-977"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-977" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">=</span> <span class="fu">as.data.frame</span>( genes[<span class="fu">which</span>( <span class="fu">rowSums</span>(genes) <span class="sc">!=</span> <span class="dv">0</span> ),] )</span>
<span id="cb189-978"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-978" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(genes) <span class="ot">=</span> <span class="fu">colnames</span>( limma.out<span class="sc">$</span>results )</span>
<span id="cb189-979"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-979" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">=</span> <span class="fu">merge</span>(genes,convertedData.out, <span class="at">by=</span><span class="st">&#39;row.names&#39;</span>)</span>
<span id="cb189-980"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-980" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(genes)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">&quot;1: upregulation, -1: downregulation&quot;</span></span>
<span id="cb189-981"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-981" aria-hidden="true" tabindex="-1"></a>    <span class="co"># add gene symbol</span></span>
<span id="cb189-982"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-982" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span> <span class="fu">match</span>( genes[,<span class="dv">1</span>], allGeneInfo.out[,<span class="dv">1</span>])</span>
<span id="cb189-983"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-983" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">as.character</span>( allGeneInfo.out<span class="sc">$</span>symbol)[ix],genes) </span>
<span id="cb189-984"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-984" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(genes)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">&quot;Symbol&quot;</span></span>
<span id="cb189-985"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-985" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">&lt;-</span> genes[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">3</span><span class="sc">:</span><span class="fu">dim</span>(genes)[<span class="dv">2</span>]) ]</span>
<span id="cb189-986"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-986" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(genes)</span>
<span id="cb189-987"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-987" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-988"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-988" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-989"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-989" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-990"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-990" aria-hidden="true" tabindex="-1"></a>AllGeneListsGMT <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb189-991"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-991" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-992"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-992" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-993"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-993" aria-hidden="true" tabindex="-1"></a>            results <span class="ot">=</span> limma.out<span class="sc">$</span>results</span>
<span id="cb189-994"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-994" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-995"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-995" aria-hidden="true" tabindex="-1"></a>            results2 <span class="ot">=</span> <span class="fu">cbind</span>(results, results)</span>
<span id="cb189-996"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-996" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(results2)<span class="ot">=</span> <span class="fu">c</span>( <span class="fu">paste0</span>(<span class="st">&quot;UP_&quot;</span>, <span class="fu">colnames</span>(results)),<span class="fu">paste0</span>(<span class="st">&quot;Down_&quot;</span>, <span class="fu">colnames</span>(results)) )</span>
<span id="cb189-997"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-997" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(results)[<span class="dv">2</span>] ) {</span>
<span id="cb189-998"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-998" aria-hidden="true" tabindex="-1"></a>                results2[,i<span class="sc">*</span><span class="dv">2-1</span>] <span class="ot">=</span> results[,i]</span>
<span id="cb189-999"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-999" aria-hidden="true" tabindex="-1"></a>                results2[ <span class="fu">which</span>(results2[,i<span class="sc">*</span><span class="dv">2-1</span>] <span class="sc">&lt;</span> <span class="dv">0</span> ) , i<span class="sc">*</span><span class="dv">2-1</span>] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb189-1000"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1000" aria-hidden="true" tabindex="-1"></a>                results2[,i<span class="sc">*</span><span class="dv">2</span>] <span class="ot">=</span> results[,i]    </span>
<span id="cb189-1001"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1001" aria-hidden="true" tabindex="-1"></a>                results2[ <span class="fu">which</span>(results2[,i<span class="sc">*</span><span class="dv">2</span>] <span class="sc">&gt;</span> <span class="dv">0</span> ) , i<span class="sc">*</span><span class="dv">2</span>] <span class="ot">=</span> <span class="dv">0</span> </span>
<span id="cb189-1002"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1002" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb189-1003"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1003" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1004"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1004" aria-hidden="true" tabindex="-1"></a>            geneList1 <span class="ot">&lt;-</span> <span class="cf">function</span> (i) {</span>
<span id="cb189-1005"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1005" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">=</span> <span class="fu">which</span>(results2[,i] <span class="sc">!=</span><span class="dv">0</span> )</span>
<span id="cb189-1006"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1006" aria-hidden="true" tabindex="-1"></a>                <span class="fu">return</span>(<span class="fu">paste0</span>(<span class="fu">colnames</span>(results2)[i],<span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="fu">length</span>(ix),<span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb189-1007"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1007" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">paste</span>(<span class="fu">rownames</span>(results2 )[ix], <span class="at">collapse=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span> ) ) )          </span>
<span id="cb189-1008"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1008" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb189-1009"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1009" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(results2)[<span class="dv">2</span>],geneList1 )</span>
<span id="cb189-1010"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1010" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb189-1011"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1011" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>( <span class="fu">paste</span>(tem, <span class="at">collapse=</span><span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>) )</span>
<span id="cb189-1012"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1012" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1013"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1013" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1014"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1014" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1015"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1015" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1016"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1016" aria-hidden="true" tabindex="-1"></a>geneListData <span class="ot">&lt;-</span> <span class="cf">function</span>( ) {</span>
<span id="cb189-1017"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1017" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-1018"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1018" aria-hidden="true" tabindex="-1"></a>        noSig <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No significant genes find!&quot;</span>)</span>
<span id="cb189-1019"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1019" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">is.null</span>(input_selectContrast) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb189-1020"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1020" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">is.null</span>( limma.out<span class="sc">$</span>comparisons ) ) <span class="fu">return</span>(<span class="cn">NULL</span>) <span class="co"># if no significant genes found</span></span>
<span id="cb189-1021"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1021" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">length</span>(limma.out<span class="sc">$</span>topGenes) <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(noSig)</span>
<span id="cb189-1022"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1022" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb189-1023"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1023" aria-hidden="true" tabindex="-1"></a>        { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb189-1024"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1024" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb189-1025"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1025" aria-hidden="true" tabindex="-1"></a>          top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb189-1026"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1026" aria-hidden="true" tabindex="-1"></a>          ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, <span class="fu">names</span>(top))</span>
<span id="cb189-1027"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1027" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (noSig)</span>
<span id="cb189-1028"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1028" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb189-1029"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1029" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb189-1030"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1030" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (noSig)</span>
<span id="cb189-1031"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1031" aria-hidden="true" tabindex="-1"></a>          <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb189-1032"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1032" aria-hidden="true" tabindex="-1"></a>          <span class="co">#top1 = merge(top1,convertedData(), by=&#39;row.names&#39;)</span></span>
<span id="cb189-1033"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1033" aria-hidden="true" tabindex="-1"></a>          <span class="co">#colnames(top1)[1] = &quot;Genes&quot;</span></span>
<span id="cb189-1034"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1034" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">=</span> top1[<span class="fu">order</span>(<span class="sc">-</span><span class="fu">abs</span>(top1<span class="sc">$</span>Fold)) ,]</span>
<span id="cb189-1035"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1035" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span> ( <span class="fu">length</span>( <span class="fu">which</span>( top1<span class="sc">$</span>FDR <span class="sc">&lt;=</span>  input_limmaPval  <span class="sc">&amp;</span>  <span class="fu">abs</span>(top1<span class="sc">$</span>Fold)  <span class="sc">&gt;=</span> <span class="fu">log2</span>(input_limmaFC) ) ) <span class="sc">==</span> <span class="dv">0</span> )</span>
<span id="cb189-1036"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1036" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>( noSig)</span>
<span id="cb189-1037"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1037" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top1[<span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;=</span>  input_limmaPval ) ,]</span>
<span id="cb189-1038"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1038" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top1[<span class="fu">which</span>(<span class="fu">abs</span>(top1<span class="sc">$</span>Fold)  <span class="sc">&gt;=</span> <span class="fu">log2</span>( input_limmaFC)) ,]</span>
<span id="cb189-1039"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1039" aria-hidden="true" tabindex="-1"></a>          top1<span class="sc">$</span>Top_Genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top1)</span>
<span id="cb189-1040"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1040" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top1[,<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">2</span>)]</span>
<span id="cb189-1041"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1041" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb189-1042"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1042" aria-hidden="true" tabindex="-1"></a>          <span class="co"># if new species</span></span>
<span id="cb189-1043"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1043" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>( input_selectGO2 <span class="sc">==</span> <span class="st">&quot;ID not recognized!&quot;</span> <span class="sc">|</span> input_selectOrg <span class="sc">==</span> <span class="st">&quot;NEW&quot;</span>) <span class="fu">return</span> (top1); </span>
<span id="cb189-1044"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1044" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb189-1045"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1045" aria-hidden="true" tabindex="-1"></a>          <span class="co">#convertedID = convertID(top1[,1],input_selectOrg, &quot;GOBP&quot; );#&quot;gmax_eg_gene&quot;</span></span>
<span id="cb189-1046"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1046" aria-hidden="true" tabindex="-1"></a>          <span class="co"># tem &lt;- geneInfo(convertedID,input_selectOrg) #input_selectOrg ) ;</span></span>
<span id="cb189-1047"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1047" aria-hidden="true" tabindex="-1"></a>         <span class="co">#  tem &lt;- geneInfo(converted(),input_selectOrg)</span></span>
<span id="cb189-1048"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1048" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> <span class="fu">merge</span>(top1, allGeneInfo.out, <span class="at">by.x =</span><span class="st">&quot;Top_Genes&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="at">all.x=</span>T )</span>
<span id="cb189-1049"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1049" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb189-1050"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1050" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span> ( <span class="fu">sum</span>( <span class="fu">is.na</span>(top1<span class="sc">$</span>band)) <span class="sc">==</span> <span class="fu">dim</span>(top1)[<span class="dv">1</span>] ) top1<span class="sc">$</span>chr <span class="ot">=</span> top1<span class="sc">$</span>chromosome_name <span class="cf">else</span></span>
<span id="cb189-1051"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1051" aria-hidden="true" tabindex="-1"></a>            top1<span class="sc">$</span>chr <span class="ot">=</span> <span class="fu">paste</span>( top1<span class="sc">$</span>chromosome_name, top1<span class="sc">$</span>band,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb189-1052"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1052" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb189-1053"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1053" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top1[,<span class="fu">c</span>(<span class="st">&#39;Top_Genes&#39;</span>,<span class="st">&#39;Fold&#39;</span>,<span class="st">&#39;FDR&#39;</span>,<span class="st">&#39;symbol&#39;</span>,<span class="st">&#39;chr&#39;</span>,<span class="st">&#39;gene_biotype&#39;</span>)]</span>
<span id="cb189-1054"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1054" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb189-1055"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1055" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb189-1056"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1056" aria-hidden="true" tabindex="-1"></a>        <span class="co">#  ix = match(top1[,1], tem$ensembl_gene_id)</span></span>
<span id="cb189-1057"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1057" aria-hidden="true" tabindex="-1"></a>         <span class="co"># if( sum(is.na( tem$Symbol[ix]) ) != length(ix) ) </span></span>
<span id="cb189-1058"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1058" aria-hidden="true" tabindex="-1"></a>          <span class="co"># { top1 &lt;- cbind(top1, tem$Symbol[ix]); colnames(top1)[4]= &quot;Symbol&quot; }</span></span>
<span id="cb189-1059"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1059" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">=</span> top1[<span class="fu">order</span>(<span class="sc">-</span><span class="fu">abs</span>(<span class="fu">as.numeric</span>( top1<span class="sc">$</span>Fold))) ,]</span>
<span id="cb189-1060"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1060" aria-hidden="true" tabindex="-1"></a>          top1<span class="sc">$</span>FDR <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;%-3.2e&quot;</span>,top1<span class="sc">$</span>FDR )</span>
<span id="cb189-1061"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1061" aria-hidden="true" tabindex="-1"></a>          <span class="fu">colnames</span>(top1) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Ensembl ID&quot;</span>, <span class="st">&quot;log2 Fold Change&quot;</span>, <span class="st">&quot;Adj.Pval&quot;</span>, <span class="st">&quot;Symbol&quot;</span>,<span class="st">&quot;Chr&quot;</span>,<span class="st">&quot;Type&quot;</span>)</span>
<span id="cb189-1062"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1062" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span> ( <span class="fu">sum</span>( <span class="fu">is.na</span>(top1<span class="sc">$</span>Symbol)) <span class="sc">==</span> <span class="fu">dim</span>(top1)[<span class="dv">1</span>] ) top1 <span class="ot">&lt;-</span> top1[,<span class="sc">-</span><span class="dv">4</span>] </span>
<span id="cb189-1063"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1063" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1064"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1064" aria-hidden="true" tabindex="-1"></a>          <span class="fu">return</span>(top1)</span>
<span id="cb189-1065"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1065" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-1066"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1066" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb189-1067"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1067" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb189-1068"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1068" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1069"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1069" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb189-1070"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1070" aria-hidden="true" tabindex="-1"></a>volcanoPlot <span class="ot">&lt;-</span> <span class="cf">function</span>( ) {</span>
<span id="cb189-1071"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1071" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb189-1072"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1072" aria-hidden="true" tabindex="-1"></a>    { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb189-1073"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1073" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb189-1074"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1074" aria-hidden="true" tabindex="-1"></a>      top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb189-1075"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1075" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, <span class="fu">names</span>(top))</span>
<span id="cb189-1076"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1076" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb189-1077"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1077" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb189-1078"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1078" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb189-1079"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1079" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1080"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1080" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb189-1081"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1081" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb189-1082"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1082" aria-hidden="true" tabindex="-1"></a>     top1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(top1) <span class="co"># convert to data frame</span></span>
<span id="cb189-1083"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1083" aria-hidden="true" tabindex="-1"></a>     top1 <span class="ot">&lt;-</span> top1[<span class="fu">which</span>(<span class="sc">!</span>(<span class="fu">is.na</span>(top1<span class="sc">$</span>Fold)<span class="sc">|</span><span class="fu">is.na</span>(top1<span class="sc">$</span>FDR)    )),] <span class="co"># remove NA&#39;s </span></span>
<span id="cb189-1084"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1084" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb189-1085"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1085" aria-hidden="true" tabindex="-1"></a>     <span class="co">#write.csv(top1,&quot;tem.csv&quot;)</span></span>
<span id="cb189-1086"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1086" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown[ <span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;=</span>  input_limmaPval<span class="sc">&amp;</span> top1<span class="sc">$</span>Fold  <span class="sc">&gt;=</span> <span class="fu">log2</span>( input_limmaFC)) ]  <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb189-1087"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1087" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown[ <span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;=</span>  input_limmaPval <span class="sc">&amp;</span> top1<span class="sc">$</span>Fold  <span class="sc">&lt;=</span> <span class="sc">-</span><span class="fu">log2</span>( input_limmaFC)) ]  <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb189-1088"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1088" aria-hidden="true" tabindex="-1"></a>     <span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb189-1089"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1089" aria-hidden="true" tabindex="-1"></a>     <span class="fu">plot</span>(top1<span class="sc">$</span>Fold,<span class="sc">-</span><span class="fu">log10</span>(top1<span class="sc">$</span>FDR),<span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;grey30&quot;</span>, <span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>)[top1<span class="sc">$</span>upOrDown],</span>
<span id="cb189-1090"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1090" aria-hidden="true" tabindex="-1"></a>     <span class="at">pch =</span><span class="dv">16</span>, <span class="at">cex =</span> .<span class="dv">3</span>, <span class="at">xlab=</span> <span class="st">&quot;log2 fold change&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;- log10 (FDR)&quot;</span>,</span>
<span id="cb189-1091"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1091" aria-hidden="true" tabindex="-1"></a>     <span class="at">cex.lab=</span><span class="dv">2</span>, <span class="at">cex.axis=</span><span class="dv">2</span>, <span class="at">cex.main=</span><span class="dv">2</span>, <span class="at">cex.sub=</span><span class="dv">2</span>   )    </span>
<span id="cb189-1092"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1092" aria-hidden="true" tabindex="-1"></a>     <span class="fu">legend</span>(<span class="st">&quot;bottomright&quot;</span>,<span class="fu">c</span>(<span class="st">&quot;Upregulated&quot;</span>,<span class="st">&quot;Downregulated&quot;</span>),<span class="at">fill =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>),<span class="at">cex=</span><span class="fl">1.1</span> )</span>
<span id="cb189-1093"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1093" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb189-1094"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1094" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1095"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1095" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1096"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1096" aria-hidden="true" tabindex="-1"></a>scatterPlot <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb189-1097"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1097" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb189-1098"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1098" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb189-1099"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1099" aria-hidden="true" tabindex="-1"></a>    { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb189-1100"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1100" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb189-1101"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1101" aria-hidden="true" tabindex="-1"></a>      top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb189-1102"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1102" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, <span class="fu">names</span>(top))</span>
<span id="cb189-1103"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1103" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb189-1104"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1104" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb189-1105"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1105" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb189-1106"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1106" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb189-1107"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1107" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb189-1108"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1108" aria-hidden="true" tabindex="-1"></a>     top1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(top1) <span class="co"># convert to data frame</span></span>
<span id="cb189-1109"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1109" aria-hidden="true" tabindex="-1"></a>     top1 <span class="ot">&lt;-</span> top1[<span class="fu">which</span>(<span class="sc">!</span>(<span class="fu">is.na</span>(top1<span class="sc">$</span>Fold)<span class="sc">|</span><span class="fu">is.na</span>(top1<span class="sc">$</span>FDR)    )),] <span class="co"># remove NA&#39;s </span></span>
<span id="cb189-1110"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1110" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb189-1111"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1111" aria-hidden="true" tabindex="-1"></a>     <span class="co">#write.csv(top1,&quot;tem.csv&quot;)</span></span>
<span id="cb189-1112"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1112" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown[ <span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;=</span>  input_limmaPval<span class="sc">&amp;</span> top1<span class="sc">$</span>Fold  <span class="sc">&gt;=</span> <span class="fu">log2</span>( input_limmaFC)) ]  <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb189-1113"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1113" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown[ <span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;=</span>  input_limmaPval <span class="sc">&amp;</span> top1<span class="sc">$</span>Fold  <span class="sc">&lt;=</span> <span class="sc">-</span><span class="fu">log2</span>( input_limmaFC)) ]  <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb189-1114"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1115"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1116"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1116" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb189-1117"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1117" aria-hidden="true" tabindex="-1"></a>         iz <span class="ot">=</span> <span class="fu">findContrastSamples</span>(  input_selectContrast, </span>
<span id="cb189-1118"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1118" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">colnames</span>(convertedData.out),</span>
<span id="cb189-1119"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1119" aria-hidden="true" tabindex="-1"></a>                                    readSampleInfo.out,</span>
<span id="cb189-1120"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1120" aria-hidden="true" tabindex="-1"></a>                                    input_selectFactorsModel,</span>
<span id="cb189-1121"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1121" aria-hidden="true" tabindex="-1"></a>                                    input_selectModelComprions, </span>
<span id="cb189-1122"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1122" aria-hidden="true" tabindex="-1"></a>                                    factorReferenceLevels.out,</span>
<span id="cb189-1123"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1123" aria-hidden="true" tabindex="-1"></a>                                    input_CountsDEGMethod,</span>
<span id="cb189-1124"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1124" aria-hidden="true" tabindex="-1"></a>                                    input_dataFileFormat</span>
<span id="cb189-1125"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1125" aria-hidden="true" tabindex="-1"></a>                                )</span>
<span id="cb189-1126"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1126" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb189-1127"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1127" aria-hidden="true" tabindex="-1"></a>     genes <span class="ot">&lt;-</span> convertedData.out[,iz]</span>
<span id="cb189-1128"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1128" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb189-1129"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1129" aria-hidden="true" tabindex="-1"></a>     g <span class="ot">=</span> <span class="fu">detectGroups</span>(<span class="fu">colnames</span>(genes))</span>
<span id="cb189-1130"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1130" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb189-1131"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1131" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>(<span class="fu">length</span>(<span class="fu">unique</span>(g))  <span class="sc">&gt;</span> <span class="dv">2</span>) { <span class="fu">plot.new</span>(); <span class="fu">text</span>(<span class="fl">0.5</span>,<span class="fl">0.5</span>, <span class="st">&quot;Not available.&quot;</span>) } <span class="cf">else</span>{</span>
<span id="cb189-1132"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1132" aria-hidden="true" tabindex="-1"></a>        average1 <span class="ot">&lt;-</span> <span class="fu">apply</span>( genes[, <span class="fu">which</span>( g <span class="sc">==</span> <span class="fu">unique</span>(g)[<span class="dv">1</span>] ) ],<span class="dv">1</span>,mean)</span>
<span id="cb189-1133"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1134"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1134" aria-hidden="true" tabindex="-1"></a>        average2 <span class="ot">&lt;-</span> <span class="fu">apply</span>(  genes[, <span class="fu">which</span>( g <span class="sc">==</span> <span class="fu">unique</span>(g)[<span class="dv">2</span>] ) ],<span class="dv">1</span>,mean)</span>
<span id="cb189-1135"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1136"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1136" aria-hidden="true" tabindex="-1"></a>        genes2 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(average1,average2)</span>
<span id="cb189-1137"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1137" aria-hidden="true" tabindex="-1"></a>        <span class="fu">rownames</span>(genes2) <span class="ot">=</span> <span class="fu">rownames</span>(genes)</span>
<span id="cb189-1138"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1138" aria-hidden="true" tabindex="-1"></a>        genes2 <span class="ot">&lt;-</span>  <span class="fu">merge</span>(genes2,top1,<span class="at">by=</span><span class="st">&quot;row.names&quot;</span>)</span>
<span id="cb189-1139"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1140"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1140" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-1141"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1141" aria-hidden="true" tabindex="-1"></a>        <span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb189-1142"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1142" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot</span>(genes2<span class="sc">$</span>average2,genes2<span class="sc">$</span>average1,<span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;grey45&quot;</span>,<span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>)[genes2<span class="sc">$</span>upOrDown],</span>
<span id="cb189-1143"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1143" aria-hidden="true" tabindex="-1"></a>        <span class="at">pch =</span><span class="dv">16</span>, <span class="at">cex =</span> .<span class="dv">3</span>, <span class="at">xlab=</span> <span class="fu">paste</span>(<span class="st">&quot;Average expression in&quot;</span>, <span class="fu">unique</span>(g)[<span class="dv">2</span>] ), </span>
<span id="cb189-1144"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1144" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab =</span> <span class="fu">paste</span>(<span class="st">&quot;Average expression in&quot;</span>, <span class="fu">unique</span>(g)[<span class="dv">1</span>] ),</span>
<span id="cb189-1145"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1145" aria-hidden="true" tabindex="-1"></a>        <span class="at">cex.lab=</span><span class="dv">2</span>, <span class="at">cex.axis=</span><span class="dv">2</span>, <span class="at">cex.main=</span><span class="dv">2</span>, <span class="at">cex.sub=</span><span class="dv">2</span>    )    </span>
<span id="cb189-1146"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1146" aria-hidden="true" tabindex="-1"></a>        <span class="fu">legend</span>(<span class="st">&quot;bottomright&quot;</span>,<span class="fu">c</span>(<span class="st">&quot;Upregulated&quot;</span>,<span class="st">&quot;Downregulated&quot;</span>),<span class="at">fill =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>),<span class="at">cex=</span><span class="fl">1.3</span> )</span>
<span id="cb189-1147"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1148"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1149"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1149" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb189-1150"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1150" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb189-1151"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1151" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb189-1152"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1152" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1153"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1154"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1155"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1155" aria-hidden="true" tabindex="-1"></a>MAplot <span class="ot">&lt;-</span> <span class="cf">function</span> ( ) {</span>
<span id="cb189-1156"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1156" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb189-1157"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1157" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb189-1158"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1158" aria-hidden="true" tabindex="-1"></a>    { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb189-1159"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1159" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb189-1160"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1160" aria-hidden="true" tabindex="-1"></a>      top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb189-1161"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1161" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, <span class="fu">names</span>(top))</span>
<span id="cb189-1162"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1162" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb189-1163"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1163" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb189-1164"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1164" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb189-1165"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1165" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb189-1166"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1166" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb189-1167"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1167" aria-hidden="true" tabindex="-1"></a>     top1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(top1) <span class="co"># convert to data frame</span></span>
<span id="cb189-1168"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1168" aria-hidden="true" tabindex="-1"></a>     top1 <span class="ot">&lt;-</span> top1[<span class="fu">which</span>(<span class="sc">!</span>(<span class="fu">is.na</span>(top1<span class="sc">$</span>Fold)<span class="sc">|</span><span class="fu">is.na</span>(top1<span class="sc">$</span>FDR)    )),] <span class="co"># remove NA&#39;s </span></span>
<span id="cb189-1169"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1169" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb189-1170"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1170" aria-hidden="true" tabindex="-1"></a>     <span class="co">#write.csv(top1,&quot;tem.csv&quot;)</span></span>
<span id="cb189-1171"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1171" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown[ <span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;=</span>  input_limmaPval<span class="sc">&amp;</span> top1<span class="sc">$</span>Fold  <span class="sc">&gt;=</span> <span class="fu">log2</span>( input_limmaFC)) ]  <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb189-1172"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1172" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown[ <span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;=</span>  input_limmaPval <span class="sc">&amp;</span> top1<span class="sc">$</span>Fold  <span class="sc">&lt;=</span> <span class="sc">-</span><span class="fu">log2</span>( input_limmaFC)) ]  <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb189-1173"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1173" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb189-1174"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1174" aria-hidden="true" tabindex="-1"></a>         iz <span class="ot">=</span> <span class="fu">findContrastSamples</span>(  input_selectContrast, </span>
<span id="cb189-1175"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1175" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">colnames</span>(convertedData.out),</span>
<span id="cb189-1176"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1176" aria-hidden="true" tabindex="-1"></a>                                    readSampleInfo.out,</span>
<span id="cb189-1177"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1177" aria-hidden="true" tabindex="-1"></a>                                    input_selectFactorsModel,</span>
<span id="cb189-1178"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1178" aria-hidden="true" tabindex="-1"></a>                                    input_selectModelComprions, </span>
<span id="cb189-1179"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1179" aria-hidden="true" tabindex="-1"></a>                                    factorReferenceLevels.out,</span>
<span id="cb189-1180"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1180" aria-hidden="true" tabindex="-1"></a>                                    input_CountsDEGMethod,</span>
<span id="cb189-1181"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1181" aria-hidden="true" tabindex="-1"></a>                                    input_dataFileFormat</span>
<span id="cb189-1182"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1182" aria-hidden="true" tabindex="-1"></a>                                )</span>
<span id="cb189-1183"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1183" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb189-1184"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1185"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1186"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1186" aria-hidden="true" tabindex="-1"></a>        average1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>( <span class="fu">apply</span>( <span class="fu">convertedData</span>()[,iz],<span class="dv">1</span>,mean) )</span>
<span id="cb189-1187"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1187" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(average1) <span class="ot">=</span> <span class="st">&quot;Average&quot;</span></span>
<span id="cb189-1188"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1188" aria-hidden="true" tabindex="-1"></a>        <span class="fu">rownames</span>(average1) <span class="ot">=</span> <span class="fu">rownames</span>(convertedData.out)</span>
<span id="cb189-1189"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1189" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-1190"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1190" aria-hidden="true" tabindex="-1"></a>        genes2 <span class="ot">&lt;-</span>  <span class="fu">merge</span>(average1,top1,<span class="at">by=</span><span class="st">&quot;row.names&quot;</span>)</span>
<span id="cb189-1191"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1192"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1192" aria-hidden="true" tabindex="-1"></a>        <span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb189-1193"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1193" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot</span>(genes2<span class="sc">$</span>Average,genes2<span class="sc">$</span>Fold,<span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;grey45&quot;</span>,<span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>)[genes2<span class="sc">$</span>upOrDown],</span>
<span id="cb189-1194"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1194" aria-hidden="true" tabindex="-1"></a>        <span class="at">pch =</span><span class="dv">16</span>, <span class="at">cex =</span> .<span class="dv">3</span>, <span class="at">xlab=</span> <span class="st">&quot;Average expression&quot;</span>, </span>
<span id="cb189-1195"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1195" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab =</span> <span class="st">&quot;Log2 fold change&quot;</span>,</span>
<span id="cb189-1196"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1196" aria-hidden="true" tabindex="-1"></a>        <span class="at">cex.lab=</span><span class="dv">2</span>, <span class="at">cex.axis=</span><span class="dv">2</span>, <span class="at">cex.main=</span><span class="dv">2</span>, <span class="at">cex.sub=</span><span class="dv">2</span>    )   </span>
<span id="cb189-1197"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1197" aria-hidden="true" tabindex="-1"></a>            <span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>)</span>
<span id="cb189-1198"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1198" aria-hidden="true" tabindex="-1"></a>        <span class="fu">legend</span>(<span class="st">&quot;bottomright&quot;</span>,<span class="fu">c</span>(<span class="st">&quot;Upregulated&quot;</span>,<span class="st">&quot;Downregulated&quot;</span>),<span class="at">fill =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>),<span class="at">cex=</span><span class="fl">1.3</span> )</span>
<span id="cb189-1199"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1199" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb189-1200"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1200" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1201"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1202"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1203"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1203" aria-hidden="true" tabindex="-1"></a>geneListGOTable <span class="ot">&lt;-</span> <span class="cf">function</span>() {     </span>
<span id="cb189-1204"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1204" aria-hidden="true" tabindex="-1"></a>        NoSig<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb189-1205"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1205" aria-hidden="true" tabindex="-1"></a>        <span class="co"># using expression data</span></span>
<span id="cb189-1206"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1206" aria-hidden="true" tabindex="-1"></a>        genes <span class="ot">&lt;-</span> selectedHeatmap.data.out<span class="sc">$</span>genes</span>
<span id="cb189-1207"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1207" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>(genes) ) <span class="fu">return</span>(<span class="cn">NULL</span>) </span>
<span id="cb189-1208"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1208" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">dim</span>(genes)[<span class="dv">1</span>] <span class="sc">&lt;=</span> minGenesEnrichment ) <span class="fu">return</span>(NoSig) <span class="co"># if has only few genes</span></span>
<span id="cb189-1209"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1209" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-1210"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1210" aria-hidden="true" tabindex="-1"></a>        fc <span class="ot">=</span> selectedHeatmap.data.out<span class="sc">$</span>bar       </span>
<span id="cb189-1211"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1211" aria-hidden="true" tabindex="-1"></a>        <span class="co"># GO</span></span>
<span id="cb189-1212"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1212" aria-hidden="true" tabindex="-1"></a>        results1 <span class="ot">&lt;-</span> <span class="cn">NULL</span>; result <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-1213"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1213" aria-hidden="true" tabindex="-1"></a>        pp <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb189-1214"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1214" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( i <span class="cf">in</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="sc">-</span><span class="dv">1</span>) ) {</span>
<span id="cb189-1215"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1216"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1216" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-1217"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1217" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">length</span>(<span class="fu">which</span>(fc<span class="sc">*</span>i<span class="sc">&lt;</span><span class="dv">0</span>)) <span class="sc">&lt;=</span> minGenesEnrichment) <span class="cf">next</span>; </span>
<span id="cb189-1218"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1218" aria-hidden="true" tabindex="-1"></a>            query <span class="ot">=</span> <span class="fu">rownames</span>(genes)[<span class="fu">which</span>(fc<span class="sc">*</span>i<span class="sc">&lt;</span><span class="dv">0</span>)]</span>
<span id="cb189-1219"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1219" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">length</span>(query) <span class="sc">&lt;=</span> minGenesEnrichment) <span class="cf">next</span>;  </span>
<span id="cb189-1220"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1220" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-1221"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1222"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1222" aria-hidden="true" tabindex="-1"></a>            result <span class="ot">&lt;-</span> <span class="fu">findOverlapGMT</span>( query, GeneSets.out,<span class="dv">1</span>) </span>
<span id="cb189-1223"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1224"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1225"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1225" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">2</span>] <span class="sc">==</span><span class="dv">1</span>) <span class="cf">next</span>;   <span class="co"># result could be NULL</span></span>
<span id="cb189-1226"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1226" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(i <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>) result<span class="sc">$</span>direction <span class="ot">=</span> <span class="st">&quot;Up regulated&quot;</span>  <span class="cf">else</span> result<span class="sc">$</span>direction <span class="ot">=</span> <span class="st">&quot;Down regulated&quot;</span> <span class="co"># changed</span></span>
<span id="cb189-1227"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1227" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (pp<span class="sc">==</span><span class="dv">0</span> ) { results1 <span class="ot">&lt;-</span> result; pp <span class="ot">=</span> <span class="dv">1</span>;} <span class="cf">else</span>  results1 <span class="ot">=</span> <span class="fu">rbind</span>(results1,result)</span>
<span id="cb189-1228"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1228" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb189-1229"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1230"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1230" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> ( pp <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (NoSig)</span>
<span id="cb189-1231"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1231" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> ( <span class="fu">is.null</span>( results1) ) <span class="fu">return</span> (NoSig)</span>
<span id="cb189-1232"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1232" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">==</span> <span class="dv">1</span> ) <span class="fu">return</span>(NoSig)  <span class="co"># Returns a data frame: &quot;No significant results found!&quot;</span></span>
<span id="cb189-1233"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1233" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-1234"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1234" aria-hidden="true" tabindex="-1"></a>        results1<span class="ot">=</span> results1[,<span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">5</span>)] <span class="co"># changed</span></span>
<span id="cb189-1235"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1235" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(results1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;List&quot;</span>,<span class="st">&quot;FDR&quot;</span>,<span class="st">&quot;nGenes&quot;</span>,<span class="st">&quot;GO terms or pathways&quot;</span>,<span class="st">&quot;Genes&quot;</span>) <span class="co"># changed</span></span>
<span id="cb189-1236"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1236" aria-hidden="true" tabindex="-1"></a>        minFDR <span class="ot">=</span> <span class="fl">0.01</span></span>
<span id="cb189-1237"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1237" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">min</span>(results1<span class="sc">$</span>FDR) <span class="sc">&gt;</span> minFDR ) results1 <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No signficant enrichment found.&quot;</span>) <span class="cf">else</span></span>
<span id="cb189-1238"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1238" aria-hidden="true" tabindex="-1"></a>        results1 <span class="ot">=</span> results1[<span class="fu">which</span>(results1<span class="sc">$</span>FDR <span class="sc">&lt;</span> minFDR),]</span>
<span id="cb189-1239"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1239" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-1240"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1240" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">!=</span> <span class="dv">5</span>) <span class="fu">return</span>(NoSig)  <span class="co"># changed</span></span>
<span id="cb189-1241"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1241" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(results1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Direction&quot;</span>,<span class="st">&quot;adj.Pval&quot;</span>,<span class="st">&quot;nGenes&quot;</span>,<span class="st">&quot;Pathways&quot;</span>, <span class="st">&quot;Genes&quot;</span>) <span class="co"># changed</span></span>
<span id="cb189-1242"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1242" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-1243"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1243" aria-hidden="true" tabindex="-1"></a>        results1</span>
<span id="cb189-1244"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1244" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-1245"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1245" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb189-1246"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1247"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1248"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1248" aria-hidden="true" tabindex="-1"></a>geneListGO <span class="ot">&lt;-</span> <span class="cf">function</span>() {  </span>
<span id="cb189-1249"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1249" aria-hidden="true" tabindex="-1"></a>  results1 <span class="ot">=</span> geneListGOTable.out</span>
<span id="cb189-1250"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1250" aria-hidden="true" tabindex="-1"></a>  tem <span class="ot">=</span> input_removeRedudantSets</span>
<span id="cb189-1251"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1251" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">==</span><span class="dv">1</span>) <span class="fu">return</span>(results1) <span class="cf">else</span> { </span>
<span id="cb189-1252"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1252" aria-hidden="true" tabindex="-1"></a>    results1<span class="sc">$</span>adj.Pval <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;%-2.1e&quot;</span>,<span class="fu">as.numeric</span>(results1<span class="sc">$</span>adj.Pval) )</span>
<span id="cb189-1253"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1253" aria-hidden="true" tabindex="-1"></a>    results1[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(results1[,<span class="dv">1</span>])</span>
<span id="cb189-1254"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1254" aria-hidden="true" tabindex="-1"></a>    results1[ <span class="fu">duplicated</span> (results1[,<span class="dv">1</span>] ),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span>  </span>
<span id="cb189-1255"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1255" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-1256"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1256" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( results1[,<span class="sc">-</span><span class="dv">5</span>] )</span>
<span id="cb189-1257"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1257" aria-hidden="true" tabindex="-1"></a>    }   </span>
<span id="cb189-1258"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1258" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb189-1259"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1260"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1261"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1261" aria-hidden="true" tabindex="-1"></a><span class="co"># a program for ploting enrichment results by highlighting the similarities among terms</span></span>
<span id="cb189-1262"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1262" aria-hidden="true" tabindex="-1"></a><span class="co"># must have columns: Direction, adj.Pval   Pathways Genes</span></span>
<span id="cb189-1263"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1263" aria-hidden="true" tabindex="-1"></a><span class="co">#  Direction    adj.Pval    nGenes  Pathways        Genes</span></span>
<span id="cb189-1264"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1264" aria-hidden="true" tabindex="-1"></a><span class="co">#Down regulated 3.58E-59    131 Ribonucleoprotein complex biogenesis    36  Nsun5 Nhp2 Rrp15 </span></span>
<span id="cb189-1265"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1265" aria-hidden="true" tabindex="-1"></a><span class="co">#Down regulated 2.55E-57    135 NcRNA metabolic process 23  Nsun5 Nhp2 Rrp15 Emg1 Ddx56 Rsl1d1 enrichmentPlot &lt;- function( enrichedTerms){</span></span>
<span id="cb189-1266"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1266" aria-hidden="true" tabindex="-1"></a><span class="co"># Up or down regulation is color-coded</span></span>
<span id="cb189-1267"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1267" aria-hidden="true" tabindex="-1"></a><span class="co"># gene set size if represented by the size of marker</span></span>
<span id="cb189-1268"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1268" aria-hidden="true" tabindex="-1"></a>enrichmentPlot <span class="ot">&lt;-</span> <span class="cf">function</span>( enrichedTerms, <span class="at">rightMargin=</span><span class="dv">33</span>) {</span>
<span id="cb189-1269"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1269" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">class</span>(enrichedTerms) <span class="sc">!=</span> <span class="st">&quot;data.frame&quot;</span>) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb189-1270"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1270" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(dendextend) <span class="co"># customizing tree</span></span>
<span id="cb189-1271"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1271" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb189-1272"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1272" aria-hidden="true" tabindex="-1"></a>  geneLists <span class="ot">=</span> <span class="fu">lapply</span>(enrichedTerms<span class="sc">$</span>Genes, <span class="cf">function</span>(x) <span class="fu">unlist</span>( <span class="fu">strsplit</span>(<span class="fu">as.character</span>(x),<span class="st">&quot; &quot;</span> )   ) )</span>
<span id="cb189-1273"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1273" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(geneLists)<span class="ot">=</span> enrichedTerms<span class="sc">$</span>Pathways</span>
<span id="cb189-1274"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1275"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1275" aria-hidden="true" tabindex="-1"></a>  <span class="co"># compute overlaps percentage--------------------</span></span>
<span id="cb189-1276"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1277"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1277" aria-hidden="true" tabindex="-1"></a>  n <span class="ot">=</span> <span class="fu">length</span>(geneLists)</span>
<span id="cb189-1278"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1278" aria-hidden="true" tabindex="-1"></a>  w <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="at">nrow =</span> n, <span class="at">ncol =</span> n)</span>
<span id="cb189-1279"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1279" aria-hidden="true" tabindex="-1"></a><span class="co"># compute overlaps among all gene lists</span></span>
<span id="cb189-1280"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1280" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n) {</span>
<span id="cb189-1281"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1281" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (j <span class="cf">in</span> i<span class="sc">:</span>n) {</span>
<span id="cb189-1282"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1282" aria-hidden="true" tabindex="-1"></a>            u <span class="ot">&lt;-</span> <span class="fu">unlist</span>(geneLists[i])</span>
<span id="cb189-1283"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1283" aria-hidden="true" tabindex="-1"></a>            v <span class="ot">&lt;-</span> <span class="fu">unlist</span>(geneLists[j])</span>
<span id="cb189-1284"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1284" aria-hidden="true" tabindex="-1"></a>            w[i, j] <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">intersect</span>(u, v))<span class="sc">/</span><span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">c</span>(u,v)))</span>
<span id="cb189-1285"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1285" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb189-1286"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1286" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-1287"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1287" aria-hidden="true" tabindex="-1"></a><span class="co"># the lower half of the matrix filled in based on symmetry</span></span>
<span id="cb189-1288"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1288" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n) </span>
<span id="cb189-1289"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1289" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(i<span class="dv">-1</span>)) </span>
<span id="cb189-1290"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1290" aria-hidden="true" tabindex="-1"></a>            w[i, j] <span class="ot">=</span> w[j,i] </span>
<span id="cb189-1291"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1292"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1292" aria-hidden="true" tabindex="-1"></a>  Terms <span class="ot">=</span> <span class="fu">paste</span>( <span class="fu">sprintf</span>(<span class="st">&quot;%-1.0e&quot;</span>,<span class="fu">as.numeric</span>(enrichedTerms<span class="sc">$</span>adj.Pval)), </span>
<span id="cb189-1293"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1293" aria-hidden="true" tabindex="-1"></a>                <span class="fu">names</span>(geneLists))</span>
<span id="cb189-1294"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1294" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(w) <span class="ot">=</span> Terms</span>
<span id="cb189-1295"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1295" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(w) <span class="ot">=</span> Terms</span>
<span id="cb189-1296"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1296" aria-hidden="true" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,rightMargin)) <span class="co"># a large margin for showing </span></span>
<span id="cb189-1297"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1298"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1298" aria-hidden="true" tabindex="-1"></a>  dend <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>w) <span class="sc">%&gt;%</span></span>
<span id="cb189-1299"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1299" aria-hidden="true" tabindex="-1"></a>    <span class="fu">hclust</span> (<span class="at">method=</span><span class="st">&quot;average&quot;</span>) </span>
<span id="cb189-1300"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1300" aria-hidden="true" tabindex="-1"></a>  ix <span class="ot">=</span> dend<span class="sc">$</span>order <span class="co"># permutated order of leaves</span></span>
<span id="cb189-1301"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1302"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1302" aria-hidden="true" tabindex="-1"></a>  leafType<span class="ot">=</span> <span class="fu">as.factor</span>( <span class="fu">gsub</span>(<span class="st">&quot; .*&quot;</span>,<span class="st">&quot;&quot;</span>, enrichedTerms<span class="sc">$</span>Direction[ix] ) )</span>
<span id="cb189-1303"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1303" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">length</span>(<span class="fu">unique</span>(enrichedTerms<span class="sc">$</span>Direction)  ) <span class="sc">==</span><span class="dv">2</span> )</span>
<span id="cb189-1304"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1304" aria-hidden="true" tabindex="-1"></a>    leafColors <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>,<span class="st">&quot;red&quot;</span>)  <span class="cf">else</span>  <span class="co"># mycolors</span></span>
<span id="cb189-1305"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1305" aria-hidden="true" tabindex="-1"></a>    leafColors <span class="ot">=</span> mycolors </span>
<span id="cb189-1306"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1306" aria-hidden="true" tabindex="-1"></a>  <span class="co">#leafSize = unlist( lapply(geneLists,length) ) # leaf size represent number of genes</span></span>
<span id="cb189-1307"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1307" aria-hidden="true" tabindex="-1"></a>  <span class="co">#leafSize = sqrt( leafSize[ix] )  </span></span>
<span id="cb189-1308"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1308" aria-hidden="true" tabindex="-1"></a>  leafSize <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">as.numeric</span>( enrichedTerms<span class="sc">$</span>adj.Pval[ix] ) ) <span class="co"># leaf size represent P values</span></span>
<span id="cb189-1309"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1309" aria-hidden="true" tabindex="-1"></a>  leafSize <span class="ot">=</span> <span class="fl">1.5</span><span class="sc">*</span>leafSize<span class="sc">/</span><span class="fu">max</span>( leafSize ) <span class="sc">+</span> .<span class="dv">2</span></span>
<span id="cb189-1310"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1310" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb189-1311"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1311" aria-hidden="true" tabindex="-1"></a>    dend <span class="sc">%&gt;%</span> </span>
<span id="cb189-1312"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1312" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.dendrogram</span>(<span class="at">hang=</span><span class="sc">-</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb189-1313"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1313" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set</span>(<span class="st">&quot;leaves_pch&quot;</span>, <span class="dv">19</span>) <span class="sc">%&gt;%</span>   <span class="co"># type of marker</span></span>
<span id="cb189-1314"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1314" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set</span>(<span class="st">&quot;leaves_cex&quot;</span>, leafSize) <span class="sc">%&gt;%</span> <span class="co">#Size</span></span>
<span id="cb189-1315"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1315" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set</span>(<span class="st">&quot;leaves_col&quot;</span>, leafColors[leafType]) <span class="sc">%&gt;%</span> <span class="co"># up or down genes</span></span>
<span id="cb189-1316"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1316" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot</span>(<span class="at">horiz=</span><span class="cn">TRUE</span>)</span>
<span id="cb189-1317"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1317" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-1318"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1318" aria-hidden="true" tabindex="-1"></a>  <span class="co">#legend(&quot;top&quot;,pch=19, col=leafColors[1:2],legend=levels(leafType),bty = &quot;n&quot;,horiz =T  )</span></span>
<span id="cb189-1319"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1319" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add legend using a second layer</span></span>
<span id="cb189-1320"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1320" aria-hidden="true" tabindex="-1"></a>    <span class="fu">par</span>(<span class="at">lend =</span> <span class="dv">1</span>)           <span class="co"># square line ends for the color legend</span></span>
<span id="cb189-1321"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1321" aria-hidden="true" tabindex="-1"></a>    <span class="fu">add_legend</span>(<span class="st">&quot;top&quot;</span>,<span class="at">pch=</span><span class="dv">19</span>, <span class="at">col=</span>leafColors[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],<span class="at">legend=</span><span class="fu">levels</span>(leafType),<span class="at">bty =</span> <span class="st">&quot;n&quot;</span>,<span class="at">horiz =</span>T </span>
<span id="cb189-1322"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1323"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1323" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb189-1324"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1324" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb189-1325"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1325" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1326"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1327"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1328"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1328" aria-hidden="true" tabindex="-1"></a><span class="co"># numChar=100 maximum number of characters</span></span>
<span id="cb189-1329"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1329" aria-hidden="true" tabindex="-1"></a><span class="co"># n=200  maximum number of nodes</span></span>
<span id="cb189-1330"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1330" aria-hidden="true" tabindex="-1"></a><span class="co"># degree.cutoff = 0    Remove node if less connected</span></span>
<span id="cb189-1331"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1331" aria-hidden="true" tabindex="-1"></a><span class="co">#from PPInfer</span></span>
<span id="cb189-1332"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1332" aria-hidden="true" tabindex="-1"></a>enrich.net2 <span class="ot">&lt;-</span>  <span class="cf">function</span> (x, gene.set, node.id, <span class="at">node.name =</span> node.id, pvalue, </span>
<span id="cb189-1333"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1333" aria-hidden="true" tabindex="-1"></a>    <span class="at">n =</span> <span class="dv">50</span>, <span class="at">numChar =</span> <span class="cn">NULL</span>, <span class="at">pvalue.cutoff =</span> <span class="fl">0.05</span>, <span class="at">edge.cutoff =</span> <span class="fl">0.05</span>, </span>
<span id="cb189-1334"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1334" aria-hidden="true" tabindex="-1"></a>    <span class="at">degree.cutoff =</span> <span class="dv">0</span>, <span class="at">edge.width =</span> <span class="cf">function</span>(x) {</span>
<span id="cb189-1335"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1335" aria-hidden="true" tabindex="-1"></a>        <span class="dv">5</span> <span class="sc">*</span> x<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb189-1336"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1336" aria-hidden="true" tabindex="-1"></a>    }, <span class="at">node.size =</span> <span class="cf">function</span>(x) {</span>
<span id="cb189-1337"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1337" aria-hidden="true" tabindex="-1"></a>        <span class="fl">2.5</span> <span class="sc">*</span> <span class="fu">log10</span>(x)</span>
<span id="cb189-1338"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1338" aria-hidden="true" tabindex="-1"></a>    }, <span class="at">group =</span> <span class="cn">FALSE</span>, <span class="at">group.color =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>,<span class="st">&quot;red&quot;</span> ), <span class="at">group.shape =</span> <span class="fu">c</span>(<span class="st">&quot;circle&quot;</span>, </span>
<span id="cb189-1339"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1339" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;square&quot;</span>), <span class="at">legend.parameter =</span> <span class="fu">list</span>(<span class="st">&quot;topright&quot;</span>), <span class="at">show.legend =</span> <span class="cn">TRUE</span>, <span class="at">plotting=</span><span class="cn">TRUE</span>,</span>
<span id="cb189-1340"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1340" aria-hidden="true" tabindex="-1"></a>    ...) </span>
<span id="cb189-1341"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1341" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb189-1342"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1342" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(igraph)</span>
<span id="cb189-1343"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1343" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-1344"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1344" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(x, group)</span>
<span id="cb189-1345"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1345" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(x)[<span class="fu">length</span>(<span class="fu">colnames</span>(x))] <span class="ot">&lt;-</span> <span class="st">&quot;Group&quot;</span></span>
<span id="cb189-1346"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1346" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> x[<span class="fu">as.numeric</span>( x[, pvalue]) <span class="sc">&lt;</span> pvalue.cutoff, ]</span>
<span id="cb189-1347"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1347" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> x[<span class="fu">order</span>(x[, pvalue]), ]</span>
<span id="cb189-1348"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1348" aria-hidden="true" tabindex="-1"></a>    n <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">nrow</span>(x), n)</span>
<span id="cb189-1349"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1349" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (n <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb189-1350"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1350" aria-hidden="true" tabindex="-1"></a>        <span class="fu">stop</span>(<span class="st">&quot;no enriched term found...&quot;</span>)</span>
<span id="cb189-1351"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1351" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-1352"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1352" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> x[<span class="dv">1</span><span class="sc">:</span>n, ]</span>
<span id="cb189-1353"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1353" aria-hidden="true" tabindex="-1"></a>    index <span class="ot">&lt;-</span> <span class="fu">match</span>(x[, node.id], <span class="fu">names</span>(gene.set))</span>
<span id="cb189-1354"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1354" aria-hidden="true" tabindex="-1"></a>    geneSets <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb189-1355"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1355" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n) {</span>
<span id="cb189-1356"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1356" aria-hidden="true" tabindex="-1"></a>        geneSets[[i]] <span class="ot">&lt;-</span> gene.set[[index[i]]]</span>
<span id="cb189-1357"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1357" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-1358"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1358" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(geneSets) <span class="ot">&lt;-</span> x[, node.name]</span>
<span id="cb189-1359"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1359" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(numChar)) {</span>
<span id="cb189-1360"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1360" aria-hidden="true" tabindex="-1"></a>        numChar <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">nchar</span>(<span class="fu">as.character</span>(x[, node.name])))</span>
<span id="cb189-1361"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1361" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-1362"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1362" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb189-1363"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1363" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">substr</span>(x[, node.name], <span class="dv">1</span>, numChar))) <span class="sc">&lt;</span> </span>
<span id="cb189-1364"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1364" aria-hidden="true" tabindex="-1"></a>            <span class="fu">nrow</span>(x)) {</span>
<span id="cb189-1365"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1365" aria-hidden="true" tabindex="-1"></a>            numChar <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">nchar</span>(<span class="fu">as.character</span>(x[, node.name])))</span>
<span id="cb189-1366"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1366" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Note : numChar is too small.&quot;</span>, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb189-1367"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1367" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb189-1368"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1368" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-1369"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1369" aria-hidden="true" tabindex="-1"></a>    x[, node.name] <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">substr</span>(x[, node.name], <span class="dv">1</span>, numChar), </span>
<span id="cb189-1370"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1370" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ifelse</span>(<span class="fu">nchar</span>(<span class="fu">as.character</span>(x[, node.name])) <span class="sc">&gt;</span> numChar, </span>
<span id="cb189-1371"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1371" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;...&quot;</span>, <span class="st">&quot;&quot;</span>), <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb189-1372"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1372" aria-hidden="true" tabindex="-1"></a>    w <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="at">nrow =</span> n, <span class="at">ncol =</span> n)</span>
<span id="cb189-1373"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1374"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1374" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n) {</span>
<span id="cb189-1375"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1375" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (j <span class="cf">in</span> i<span class="sc">:</span>n) {</span>
<span id="cb189-1376"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1376" aria-hidden="true" tabindex="-1"></a>            u <span class="ot">&lt;-</span> <span class="fu">unlist</span>(geneSets[i])</span>
<span id="cb189-1377"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1377" aria-hidden="true" tabindex="-1"></a>            v <span class="ot">&lt;-</span> <span class="fu">unlist</span>(geneSets[j])</span>
<span id="cb189-1378"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1378" aria-hidden="true" tabindex="-1"></a>            w[i, j] <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">intersect</span>(u, v))<span class="sc">/</span><span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">c</span>(u, </span>
<span id="cb189-1379"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1379" aria-hidden="true" tabindex="-1"></a>                v)))</span>
<span id="cb189-1380"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1380" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb189-1381"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1381" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-1382"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1382" aria-hidden="true" tabindex="-1"></a>    list.edges <span class="ot">&lt;-</span> <span class="fu">stack</span>(<span class="fu">data.frame</span>(w))</span>
<span id="cb189-1383"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1383" aria-hidden="true" tabindex="-1"></a>    list.edges <span class="ot">&lt;-</span> <span class="fu">cbind</span>(list.edges[, <span class="dv">1</span>], <span class="fu">rep</span>(x[, node.name], </span>
<span id="cb189-1384"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1384" aria-hidden="true" tabindex="-1"></a>        n), <span class="fu">rep</span>(x[, node.name], <span class="at">each =</span> n))</span>
<span id="cb189-1385"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1385" aria-hidden="true" tabindex="-1"></a>    list.edges <span class="ot">&lt;-</span> list.edges[list.edges[, <span class="dv">2</span>] <span class="sc">!=</span> list.edges[,<span class="dv">3</span>], ]</span>
<span id="cb189-1386"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1386" aria-hidden="true" tabindex="-1"></a>    list.edges <span class="ot">&lt;-</span> list.edges[<span class="sc">!</span><span class="fu">is.na</span>(list.edges[, <span class="dv">1</span>]), ]</span>
<span id="cb189-1387"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1387" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">graph.data.frame</span>(list.edges[, <span class="sc">-</span><span class="dv">1</span>], <span class="at">directed =</span> <span class="cn">FALSE</span>)</span>
<span id="cb189-1388"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1388" aria-hidden="true" tabindex="-1"></a>    <span class="fu">E</span>(g)<span class="sc">$</span>width <span class="ot">=</span> <span class="fu">edge.width</span>(<span class="fu">as.numeric</span>(list.edges[, <span class="dv">1</span>]))</span>
<span id="cb189-1389"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1389" aria-hidden="true" tabindex="-1"></a>    <span class="fu">V</span>(g)<span class="sc">$</span>size <span class="ot">&lt;-</span> <span class="fu">node.size</span>(<span class="fu">lengths</span>(geneSets))</span>
<span id="cb189-1390"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1390" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">delete.edges</span>(g, <span class="fu">E</span>(g)[<span class="fu">as.numeric</span>(list.edges[, <span class="dv">1</span>]) <span class="sc">&lt;</span> edge.cutoff])</span>
<span id="cb189-1391"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1391" aria-hidden="true" tabindex="-1"></a>    index.deg <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">degree</span>(g) <span class="sc">&gt;=</span> degree.cutoff</span>
<span id="cb189-1392"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1392" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">delete.vertices</span>(g, <span class="fu">V</span>(g)[<span class="sc">!</span>index.deg])</span>
<span id="cb189-1393"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1393" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> x[index.deg, ]</span>
<span id="cb189-1394"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1394" aria-hidden="true" tabindex="-1"></a>    index <span class="ot">&lt;-</span> index[index.deg]</span>
<span id="cb189-1395"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1395" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(<span class="fu">V</span>(g)) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb189-1396"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1396" aria-hidden="true" tabindex="-1"></a>        <span class="fu">stop</span>(<span class="st">&quot;no categories greater than degree.cutoff...&quot;</span>)</span>
<span id="cb189-1397"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1397" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-1398"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1398" aria-hidden="true" tabindex="-1"></a>    n <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">nrow</span>(x), n)</span>
<span id="cb189-1399"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1399" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> x[<span class="dv">1</span><span class="sc">:</span>n, ]</span>
<span id="cb189-1400"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1400" aria-hidden="true" tabindex="-1"></a>    group.level <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">unique</span>(group))</span>
<span id="cb189-1401"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1401" aria-hidden="true" tabindex="-1"></a>    pvalues <span class="ot">&lt;-</span> x[, pvalue]</span>
<span id="cb189-1402"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1402" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(group.level)) {</span>
<span id="cb189-1403"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1403" aria-hidden="true" tabindex="-1"></a>        index <span class="ot">&lt;-</span> x[, <span class="st">&quot;Group&quot;</span>] <span class="sc">==</span> group.level[i]</span>
<span id="cb189-1404"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1404" aria-hidden="true" tabindex="-1"></a>        <span class="fu">V</span>(g)<span class="sc">$</span>shape[index] <span class="ot">&lt;-</span> group.shape[i]</span>
<span id="cb189-1405"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1405" aria-hidden="true" tabindex="-1"></a>        group.pvalues <span class="ot">&lt;-</span> pvalues[index]</span>
<span id="cb189-1406"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1406" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">length</span>(group.pvalues) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb189-1407"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1407" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (<span class="fu">max</span>(group.pvalues) <span class="sc">==</span> <span class="fu">min</span>(group.pvalues)) {</span>
<span id="cb189-1408"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1408" aria-hidden="true" tabindex="-1"></a>                <span class="fu">V</span>(g)<span class="sc">$</span>color[index] <span class="ot">&lt;-</span> <span class="fu">adjustcolor</span>(group.color[i], </span>
<span id="cb189-1409"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1409" aria-hidden="true" tabindex="-1"></a>                  <span class="at">alpha.f =</span> <span class="fl">0.5</span>)</span>
<span id="cb189-1410"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1410" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb189-1411"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1411" aria-hidden="true" tabindex="-1"></a>            <span class="cf">else</span> {</span>
<span id="cb189-1412"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1412" aria-hidden="true" tabindex="-1"></a>                <span class="fu">V</span>(g)<span class="sc">$</span>color[index] <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span> <span class="sc">-</span> (group.pvalues <span class="sc">-</span> </span>
<span id="cb189-1413"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1413" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">min</span>(group.pvalues))<span class="sc">/</span>(<span class="fu">max</span>(group.pvalues) <span class="sc">-</span> <span class="fu">min</span>(group.pvalues)), </span>
<span id="cb189-1414"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1414" aria-hidden="true" tabindex="-1"></a>                  <span class="cf">function</span>(x) {</span>
<span id="cb189-1415"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1415" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">adjustcolor</span>(group.color[i], <span class="at">alpha.f =</span> x)</span>
<span id="cb189-1416"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1416" aria-hidden="true" tabindex="-1"></a>                  })</span>
<span id="cb189-1417"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1417" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb189-1418"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1418" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb189-1419"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1419" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-1420"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1420" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(plotting) { </span>
<span id="cb189-1421"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1421" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot</span>(g,, <span class="at">vertex.label.dist=</span><span class="dv">1</span>, ...)</span>
<span id="cb189-1422"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1422" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (show.legend) {</span>
<span id="cb189-1423"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1423" aria-hidden="true" tabindex="-1"></a>            legend.parameter<span class="sc">$</span>legend <span class="ot">&lt;-</span> group.level</span>
<span id="cb189-1424"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1424" aria-hidden="true" tabindex="-1"></a>            legend.parameter<span class="sc">$</span>text.col <span class="ot">&lt;-</span> group.color</span>
<span id="cb189-1425"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1425" aria-hidden="true" tabindex="-1"></a>            legend.parameter<span class="sc">$</span>bty <span class="ot">&lt;-</span> <span class="st">&quot;n&quot;</span> </span>
<span id="cb189-1426"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1426" aria-hidden="true" tabindex="-1"></a>            <span class="fu">do.call</span>(legend, legend.parameter)</span>
<span id="cb189-1427"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1427" aria-hidden="true" tabindex="-1"></a>        }}</span>
<span id="cb189-1428"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1428" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(g)</span>
<span id="cb189-1429"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1429" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1430"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1431"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1432"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1432" aria-hidden="true" tabindex="-1"></a>enrichmentNetwork <span class="ot">&lt;-</span> <span class="cf">function</span>(enrichedTerms){</span>
<span id="cb189-1433"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1433" aria-hidden="true" tabindex="-1"></a>    geneLists <span class="ot">=</span> <span class="fu">lapply</span>(enrichedTerms<span class="sc">$</span>Genes, <span class="cf">function</span>(x) <span class="fu">unlist</span>( <span class="fu">strsplit</span>(<span class="fu">as.character</span>(x),<span class="st">&quot; &quot;</span> )   ) )</span>
<span id="cb189-1434"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1434" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(geneLists)<span class="ot">=</span> enrichedTerms<span class="sc">$</span>Pathways</span>
<span id="cb189-1435"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1435" aria-hidden="true" tabindex="-1"></a>    enrichedTerms<span class="sc">$</span>Direction <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; .*&quot;</span>,<span class="st">&quot;&quot;</span>,enrichedTerms<span class="sc">$</span>Direction )</span>
<span id="cb189-1436"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1436" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1437"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1437" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">enrich.net2</span>(enrichedTerms, geneLists, <span class="at">node.id =</span> <span class="st">&quot;Pathways&quot;</span>, <span class="at">numChar =</span> <span class="dv">100</span>, </span>
<span id="cb189-1438"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1438" aria-hidden="true" tabindex="-1"></a>       <span class="at">pvalue =</span> <span class="st">&quot;adj.Pval&quot;</span>, <span class="at">edge.cutoff =</span> <span class="fl">0.2</span>, <span class="at">pvalue.cutoff =</span> <span class="dv">1</span>, <span class="at">degree.cutoff =</span> <span class="dv">0</span>,</span>
<span id="cb189-1439"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1439" aria-hidden="true" tabindex="-1"></a>       <span class="at">n =</span> <span class="dv">200</span>, <span class="at">group =</span> enrichedTerms<span class="sc">$</span>Direction, <span class="at">vertex.label.cex =</span> <span class="dv">1</span>, <span class="at">vertex.label.color =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb189-1440"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1441"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1441" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1442"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1442" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb189-1443"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1443" aria-hidden="true" tabindex="-1"></a>enrichmentNetworkPlotly <span class="ot">&lt;-</span> <span class="cf">function</span>(enrichedTerms){</span>
<span id="cb189-1444"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1444" aria-hidden="true" tabindex="-1"></a>    geneLists <span class="ot">=</span> <span class="fu">lapply</span>(enrichedTerms<span class="sc">$</span>Genes, <span class="cf">function</span>(x) <span class="fu">unlist</span>( <span class="fu">strsplit</span>(<span class="fu">as.character</span>(x),<span class="st">&quot; &quot;</span> )   ) )</span>
<span id="cb189-1445"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1445" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(geneLists)<span class="ot">=</span> enrichedTerms<span class="sc">$</span>Pathways</span>
<span id="cb189-1446"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1447"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1447" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">enrich.net2</span>(enrichedTerms, geneLists, <span class="at">node.id =</span> <span class="st">&quot;Pathways&quot;</span>, <span class="at">numChar =</span> <span class="dv">100</span>, </span>
<span id="cb189-1448"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1448" aria-hidden="true" tabindex="-1"></a>       <span class="at">pvalue =</span> <span class="st">&quot;adj.Pval&quot;</span>, <span class="at">edge.cutoff =</span> <span class="fl">0.2</span>, <span class="at">pvalue.cutoff =</span> <span class="dv">1</span>, <span class="at">degree.cutoff =</span> <span class="dv">0</span>,</span>
<span id="cb189-1449"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1449" aria-hidden="true" tabindex="-1"></a>       <span class="at">n =</span> <span class="dv">200</span>, <span class="at">group =</span> enrichedTerms<span class="sc">$</span>Direction, <span class="at">vertex.label.cex =</span> <span class="fl">0.8</span>, <span class="at">vertex.label.color =</span> <span class="st">&quot;black&quot;</span></span>
<span id="cb189-1450"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1450" aria-hidden="true" tabindex="-1"></a>       ,<span class="at">plotting=</span><span class="cn">TRUE</span>)</span>
<span id="cb189-1451"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1451" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1452"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1452" aria-hidden="true" tabindex="-1"></a>    vs <span class="ot">&lt;-</span> <span class="fu">V</span>(g)</span>
<span id="cb189-1453"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1453" aria-hidden="true" tabindex="-1"></a>    es <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">get.edgelist</span>(g))</span>
<span id="cb189-1454"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1454" aria-hidden="true" tabindex="-1"></a>    Nv <span class="ot">&lt;-</span> <span class="fu">length</span>(vs)</span>
<span id="cb189-1455"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1455" aria-hidden="true" tabindex="-1"></a>    Ne <span class="ot">&lt;-</span> <span class="fu">length</span>(es[<span class="dv">1</span>]<span class="sc">$</span>V1)</span>
<span id="cb189-1456"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1456" aria-hidden="true" tabindex="-1"></a>    <span class="co"># create nodes</span></span>
<span id="cb189-1457"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1457" aria-hidden="true" tabindex="-1"></a>    L <span class="ot">&lt;-</span> <span class="fu">layout.kamada.kawai</span>(g)</span>
<span id="cb189-1458"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1458" aria-hidden="true" tabindex="-1"></a>    Xn <span class="ot">&lt;-</span> L[,<span class="dv">1</span>]</span>
<span id="cb189-1459"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1459" aria-hidden="true" tabindex="-1"></a>    Yn <span class="ot">&lt;-</span> L[,<span class="dv">2</span>]</span>
<span id="cb189-1460"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1460" aria-hidden="true" tabindex="-1"></a>    inc <span class="ot">=</span> (<span class="fu">max</span>(Yn)<span class="sc">-</span><span class="fu">min</span>(Yn))<span class="sc">/</span><span class="dv">50</span>  <span class="co"># for shifting</span></span>
<span id="cb189-1461"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1461" aria-hidden="true" tabindex="-1"></a>    <span class="co">#group &lt;- ifelse(V(g)$shape == &quot;circle&quot;, &quot;GO&quot;, &quot;KEGG&quot;)</span></span>
<span id="cb189-1462"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1462" aria-hidden="true" tabindex="-1"></a>     group <span class="ot">&lt;-</span> <span class="fu">as.character</span>(enrichedTerms<span class="sc">$</span>Direction)</span>
<span id="cb189-1463"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1463" aria-hidden="true" tabindex="-1"></a>    network <span class="ot">&lt;-</span> <span class="fu">plot_ly</span>(<span class="at">x =</span> <span class="sc">~</span>Xn, <span class="at">y =</span> <span class="sc">~</span>Yn, <span class="at">type =</span> <span class="st">&quot;scatter&quot;</span>, <span class="at">mode =</span> <span class="st">&quot;markers&quot;</span>,</span>
<span id="cb189-1464"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1464" aria-hidden="true" tabindex="-1"></a>       <span class="at">marker =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="fu">V</span>(g)<span class="sc">$</span>color, <span class="at">size =</span> <span class="fu">V</span>(g)<span class="sc">$</span>size<span class="sc">*</span><span class="dv">2</span>,</span>
<span id="cb189-1465"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1465" aria-hidden="true" tabindex="-1"></a>          <span class="at">symbol=</span> <span class="sc">~</span><span class="fu">V</span>(g)<span class="sc">$</span>shape, <span class="at">line =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;gray&quot;</span>, <span class="at">width =</span> <span class="dv">2</span>)),</span>
<span id="cb189-1466"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1466" aria-hidden="true" tabindex="-1"></a>       <span class="at">hoverinfo =</span> <span class="st">&quot;text&quot;</span>, <span class="at">text =</span> <span class="sc">~</span><span class="fu">paste</span>(<span class="st">&quot;&lt;/br&gt;&quot;</span>, group, <span class="st">&quot;&lt;/br&gt;&quot;</span>, <span class="fu">names</span>(vs))) <span class="sc">%&gt;%</span></span>
<span id="cb189-1467"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1467" aria-hidden="true" tabindex="-1"></a>       <span class="fu">add_annotations</span>( <span class="at">x =</span> <span class="sc">~</span>Xn, <span class="at">y =</span> <span class="sc">~</span>Yn<span class="sc">+</span>inc, <span class="at">text =</span> <span class="fu">names</span>(vs), <span class="at">showarrow =</span> <span class="cn">FALSE</span>,</span>
<span id="cb189-1468"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1468" aria-hidden="true" tabindex="-1"></a>          <span class="at">font =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;#030303&quot;</span>, <span class="at">size =</span> <span class="dv">12</span>))</span>
<span id="cb189-1469"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1469" aria-hidden="true" tabindex="-1"></a>    <span class="co"># create edges</span></span>
<span id="cb189-1470"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1470" aria-hidden="true" tabindex="-1"></a>    edge_shapes <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb189-1471"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1471" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>Ne)</span>
<span id="cb189-1472"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1472" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb189-1473"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1473" aria-hidden="true" tabindex="-1"></a>       v0 <span class="ot">&lt;-</span> es[i,]<span class="sc">$</span>V1</span>
<span id="cb189-1474"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1474" aria-hidden="true" tabindex="-1"></a>       v1 <span class="ot">&lt;-</span> es[i,]<span class="sc">$</span>V2</span>
<span id="cb189-1475"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1475" aria-hidden="true" tabindex="-1"></a>       index0 <span class="ot">&lt;-</span> <span class="fu">match</span>(v0, <span class="fu">names</span>(<span class="fu">V</span>(g)))</span>
<span id="cb189-1476"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1476" aria-hidden="true" tabindex="-1"></a>       index1 <span class="ot">&lt;-</span> <span class="fu">match</span>(v1, <span class="fu">names</span>(<span class="fu">V</span>(g)))</span>
<span id="cb189-1477"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1477" aria-hidden="true" tabindex="-1"></a>       edge_shape <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">type =</span> <span class="st">&quot;line&quot;</span>, <span class="at">line =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;gray&quot;</span> ,</span>
<span id="cb189-1478"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1478" aria-hidden="true" tabindex="-1"></a>          <span class="at">width =</span> <span class="fu">E</span>(g)<span class="sc">$</span>width[i]<span class="sc">/</span><span class="dv">2</span>), <span class="at">x0 =</span> Xn[index0], <span class="at">y0 =</span> Yn[index0],</span>
<span id="cb189-1479"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1479" aria-hidden="true" tabindex="-1"></a>          <span class="at">x1 =</span> Xn[index1], <span class="at">y1 =</span> Yn[index1])</span>
<span id="cb189-1480"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1480" aria-hidden="true" tabindex="-1"></a>       edge_shapes[[i]] <span class="ot">&lt;-</span> edge_shape</span>
<span id="cb189-1481"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1481" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb189-1482"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1482" aria-hidden="true" tabindex="-1"></a>    <span class="co"># create network</span></span>
<span id="cb189-1483"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1483" aria-hidden="true" tabindex="-1"></a>    axis <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">title =</span> <span class="st">&quot;&quot;</span>, <span class="at">showgrid =</span> <span class="cn">FALSE</span>, <span class="at">showticklabels =</span> <span class="cn">FALSE</span>, <span class="at">zeroline =</span> <span class="cn">FALSE</span>)</span>
<span id="cb189-1484"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1484" aria-hidden="true" tabindex="-1"></a>    h <span class="ot">&lt;-</span> <span class="fu">layout</span>(network, <span class="at">title =</span> <span class="st">&quot;Enrichment Network&quot;</span>, <span class="at">shapes =</span> edge_shapes,</span>
<span id="cb189-1485"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1485" aria-hidden="true" tabindex="-1"></a>       <span class="at">xaxis =</span> axis, <span class="at">yaxis =</span> axis)</span>
<span id="cb189-1486"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1486" aria-hidden="true" tabindex="-1"></a>    <span class="fu">config</span>(h, <span class="at">showLink =</span> <span class="cn">TRUE</span>)</span>
<span id="cb189-1487"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1487" aria-hidden="true" tabindex="-1"></a>       </span>
<span id="cb189-1488"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1488" aria-hidden="true" tabindex="-1"></a>       </span>
<span id="cb189-1489"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1489" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1490"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1490" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb189-1491"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1491" aria-hidden="true" tabindex="-1"></a> <span class="co">#Homo sapies --&gt; hsapiens</span></span>
<span id="cb189-1492"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1492" aria-hidden="true" tabindex="-1"></a>shortSpeciesNames <span class="ot">&lt;-</span> <span class="cf">function</span>(tem){</span>
<span id="cb189-1493"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1493" aria-hidden="true" tabindex="-1"></a>     tem2 <span class="ot">=</span> <span class="fu">strsplit</span>(<span class="fu">as.character</span>(tem),<span class="st">&quot; &quot;</span> )       </span>
<span id="cb189-1494"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1494" aria-hidden="true" tabindex="-1"></a>     <span class="fu">return</span>( <span class="fu">tolower</span>( <span class="fu">paste0</span>(<span class="fu">substr</span>(tem2[[<span class="dv">1</span>]][<span class="dv">1</span>],<span class="dv">1</span>,<span class="dv">1</span>), tem2[[<span class="dv">1</span>]][<span class="dv">2</span>]  ) ) )</span>
<span id="cb189-1495"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1495" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1496"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1497"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1497" aria-hidden="true" tabindex="-1"></a>findTaxonomyID <span class="ot">&lt;-</span> <span class="cf">function</span>( <span class="at">speciesName =</span> <span class="st">&quot;mmusculus&quot;</span>) {</span>
<span id="cb189-1498"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1498" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb189-1499"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1499" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(input_speciesName) ) { <span class="co"># if species name is entered</span></span>
<span id="cb189-1500"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1500" aria-hidden="true" tabindex="-1"></a>       ix <span class="ot">=</span> <span class="fu">match</span>(input_speciesName, STRING10_species<span class="sc">$</span>official_name)</span>
<span id="cb189-1501"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1501" aria-hidden="true" tabindex="-1"></a>       } <span class="cf">else</span> <span class="cf">if</span>( input_selectGO2 <span class="sc">!=</span> <span class="st">&quot;ID not recognized!&quot;</span> )</span>
<span id="cb189-1502"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1502" aria-hidden="true" tabindex="-1"></a>       { <span class="co"># if no species is entered, try to resolve species using existing info     </span></span>
<span id="cb189-1503"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1503" aria-hidden="true" tabindex="-1"></a>            codedNames <span class="ot">=</span> <span class="fu">sapply</span>(STRING10_species<span class="sc">$</span>compact_name,shortSpeciesNames )</span>
<span id="cb189-1504"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1504" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">match</span>(speciesName , codedNames)</span>
<span id="cb189-1505"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1505" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(input_selectOrg <span class="sc">!=</span> speciesChoice[[<span class="dv">1</span>]]) {  <span class="co"># if species is entered</span></span>
<span id="cb189-1506"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1506" aria-hidden="true" tabindex="-1"></a>                selectedSpecies <span class="ot">=</span> <span class="fu">findSpeciesById</span>(input_selectOrg)[<span class="dv">1</span>,<span class="dv">1</span>]</span>
<span id="cb189-1507"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1507" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">=</span> <span class="fu">match</span>( <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>,<span class="st">&quot;&quot;</span>, selectedSpecies ), codedNames)               </span>
<span id="cb189-1508"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1508" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb189-1509"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1509" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1510"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1510" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> <span class="fu">return</span>(<span class="cn">NULL</span>) </span>
<span id="cb189-1511"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1511" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb189-1512"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1512" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">is.na</span>(ix) ) <span class="fu">return</span>(<span class="cn">NULL</span>) </span>
<span id="cb189-1513"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1513" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(STRING10_species<span class="sc">$</span>species_id[ix])</span>
<span id="cb189-1514"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1514" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1515"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1515" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1516"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1516" aria-hidden="true" tabindex="-1"></a>STRINGdb_geneList <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb189-1517"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1517" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(STRINGdb,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb189-1518"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1518" aria-hidden="true" tabindex="-1"></a>    taxonomyID <span class="ot">=</span> findTaxonomyID.out</span>
<span id="cb189-1519"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1519" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1520"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1520" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>( taxonomyID ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb189-1521"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1521" aria-hidden="true" tabindex="-1"></a>        <span class="co">#Intialization</span></span>
<span id="cb189-1522"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1522" aria-hidden="true" tabindex="-1"></a>        string_db <span class="ot">&lt;-</span> STRINGdb<span class="sc">$</span><span class="fu">new</span>( <span class="at">version=</span><span class="st">&quot;10&quot;</span>, <span class="at">species=</span>taxonomyID,</span>
<span id="cb189-1523"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1523" aria-hidden="true" tabindex="-1"></a>                               <span class="at">score_threshold=</span><span class="dv">0</span>, <span class="at">input_directory=</span><span class="st">&quot;&quot;</span> )</span>
<span id="cb189-1524"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1524" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb189-1525"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1525" aria-hidden="true" tabindex="-1"></a>        <span class="co"># using expression data</span></span>
<span id="cb189-1526"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1526" aria-hidden="true" tabindex="-1"></a>        genes <span class="ot">&lt;-</span> geneListData.out</span>
<span id="cb189-1527"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1527" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(genes)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;lfc&quot;</span>)</span>
<span id="cb189-1528"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1528" aria-hidden="true" tabindex="-1"></a>        mapped <span class="ot">&lt;-</span> string_db<span class="sc">$</span><span class="fu">map</span>(genes,<span class="st">&quot;gene&quot;</span>, <span class="at">removeUnmappedRows =</span> <span class="cn">TRUE</span> )</span>
<span id="cb189-1529"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1529" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1530"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1530" aria-hidden="true" tabindex="-1"></a>        up<span class="ot">=</span> <span class="fu">subset</span>(mapped, lfc<span class="sc">&gt;</span><span class="dv">0</span>, <span class="at">select=</span><span class="st">&quot;STRING_id&quot;</span>, <span class="at">drop=</span><span class="cn">TRUE</span> )</span>
<span id="cb189-1531"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1532"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1532" aria-hidden="true" tabindex="-1"></a>        down<span class="ot">=</span> <span class="fu">subset</span>(mapped, lfc<span class="sc">&lt;</span><span class="dv">0</span>, <span class="at">select=</span><span class="st">&quot;STRING_id&quot;</span>, <span class="at">drop=</span><span class="cn">TRUE</span> )     </span>
<span id="cb189-1533"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1533" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-1534"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1534" aria-hidden="true" tabindex="-1"></a>        mappingRatio <span class="ot">=</span> <span class="fu">nrow</span>(mapped)<span class="sc">/</span> <span class="fu">nrow</span>(genes)</span>
<span id="cb189-1535"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1535" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">nrow</span>(mapped) <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>(<span class="cn">NULL</span>) <span class="cf">else</span></span>
<span id="cb189-1536"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1536" aria-hidden="true" tabindex="-1"></a>         <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">up=</span>up, <span class="at">down=</span>down, <span class="at">ratio=</span>mappingRatio, <span class="at">geneTable=</span>mapped ) )</span>
<span id="cb189-1537"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1537" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1538"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1538" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1539"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1540"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1540" aria-hidden="true" tabindex="-1"></a>STRINGDB_mapping_stat <span class="ot">&lt;-</span> <span class="cf">function</span>( ) {</span>
<span id="cb189-1541"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1541" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1542"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1542" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">is.null</span>(STRINGdb_geneList.out ) ) <span class="fu">return</span>(<span class="st">&quot;No genes mapped by STRINGdb. Please enter or double-check species name above.&quot;</span>)</span>
<span id="cb189-1543"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1543" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="sc">!</span> <span class="fu">is.null</span>(STRINGdb_geneList.out ) ) { </span>
<span id="cb189-1544"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1544" aria-hidden="true" tabindex="-1"></a>            tem<span class="ot">=</span><span class="fu">paste0</span>( <span class="dv">100</span><span class="sc">*</span><span class="fu">round</span>(<span class="fu">STRINGdb_geneList</span>()<span class="sc">$</span>ratio,<span class="dv">3</span>), <span class="st">&quot;% genes mapped by STRING web server.&quot;</span>)</span>
<span id="cb189-1545"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1545" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">STRINGdb_geneList</span>()<span class="sc">$</span>ratio <span class="sc">&lt;</span><span class="fl">0.3</span> ) tem <span class="ot">=</span> <span class="fu">paste</span>(tem, <span class="st">&quot;Warning!!! Very few gene mapped. Double check if the correct species is selected.&quot;</span>)</span>
<span id="cb189-1546"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1546" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>( tem  )</span>
<span id="cb189-1547"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1547" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb189-1548"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1548" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1549"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1549" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1550"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1551"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1551" aria-hidden="true" tabindex="-1"></a>stringDB_GO_enrichmentData <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb189-1552"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1552" aria-hidden="true" tabindex="-1"></a>        <span class="fu">library</span>(STRINGdb,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb189-1553"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1553" aria-hidden="true" tabindex="-1"></a>                           </span>
<span id="cb189-1554"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1554" aria-hidden="true" tabindex="-1"></a>        tem <span class="ot">=</span> input_STRINGdbGO</span>
<span id="cb189-1555"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1555" aria-hidden="true" tabindex="-1"></a>        taxonomyID <span class="ot">=</span> findTaxonomyID.out</span>
<span id="cb189-1556"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1556" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>( taxonomyID ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)     </span>
<span id="cb189-1557"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1557" aria-hidden="true" tabindex="-1"></a>        NoSig <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No significant enrichment found.&quot;</span>)</span>
<span id="cb189-1558"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1558" aria-hidden="true" tabindex="-1"></a>        <span class="do">####################################</span></span>
<span id="cb189-1559"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1559" aria-hidden="true" tabindex="-1"></a>        <span class="co">#Intialization</span></span>
<span id="cb189-1560"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1560" aria-hidden="true" tabindex="-1"></a>        string_db <span class="ot">&lt;-</span> STRINGdb<span class="sc">$</span><span class="fu">new</span>( <span class="at">version=</span><span class="st">&quot;10&quot;</span>, <span class="at">species=</span>taxonomyID,</span>
<span id="cb189-1561"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1561" aria-hidden="true" tabindex="-1"></a>                               <span class="at">score_threshold=</span><span class="dv">0</span>, <span class="at">input_directory=</span><span class="st">&quot;&quot;</span> )</span>
<span id="cb189-1562"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1562" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb189-1563"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1563" aria-hidden="true" tabindex="-1"></a>        <span class="co"># using expression data</span></span>
<span id="cb189-1564"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1564" aria-hidden="true" tabindex="-1"></a>        genes <span class="ot">&lt;-</span> selectedHeatmap.data.out<span class="sc">$</span>genes</span>
<span id="cb189-1565"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1565" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>(genes) ) <span class="fu">return</span>(<span class="cn">NULL</span>) </span>
<span id="cb189-1566"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1566" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">dim</span>(genes)[<span class="dv">1</span>] <span class="sc">&lt;=</span> minGenesEnrichment ) <span class="fu">return</span>(NoSig) <span class="co"># if has only few genes</span></span>
<span id="cb189-1567"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1567" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-1568"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1568" aria-hidden="true" tabindex="-1"></a>        fc <span class="ot">=</span> selectedHeatmap.data.out<span class="sc">$</span>bar       </span>
<span id="cb189-1569"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1569" aria-hidden="true" tabindex="-1"></a>        <span class="co"># GO</span></span>
<span id="cb189-1570"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1570" aria-hidden="true" tabindex="-1"></a>        results1 <span class="ot">&lt;-</span> <span class="cn">NULL</span>; result <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb189-1571"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1571" aria-hidden="true" tabindex="-1"></a>        pp <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb189-1572"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1572" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( i <span class="cf">in</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>) ) {</span>
<span id="cb189-1573"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1573" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-1574"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1574" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">length</span>(<span class="fu">which</span>(fc<span class="sc">*</span>i<span class="sc">&lt;</span><span class="dv">0</span>)) <span class="sc">&lt;=</span> minGenesEnrichment) <span class="cf">next</span>; </span>
<span id="cb189-1575"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1575" aria-hidden="true" tabindex="-1"></a>            query <span class="ot">=</span> <span class="fu">rownames</span>(genes)[<span class="fu">which</span>(fc<span class="sc">*</span>i<span class="sc">&lt;</span><span class="dv">0</span>)]</span>
<span id="cb189-1576"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1576" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-1577"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1577" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">length</span>(query) <span class="sc">&lt;=</span> minGenesEnrichment) <span class="cf">next</span>;  </span>
<span id="cb189-1578"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1578" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-1579"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1579" aria-hidden="true" tabindex="-1"></a>            ids <span class="ot">=</span> STRINGdb_geneList.out[[i]]</span>
<span id="cb189-1580"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1580" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-1581"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1581" aria-hidden="true" tabindex="-1"></a>            result <span class="ot">&lt;-</span> string_db<span class="sc">$</span><span class="fu">get_enrichment</span>( ids, <span class="at">category =</span> input_STRINGdbGO, <span class="at">methodMT =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">iea =</span> <span class="cn">TRUE</span> )</span>
<span id="cb189-1582"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1582" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">nrow</span>(result) <span class="sc">==</span> <span class="dv">0</span> ) <span class="cf">next</span>; </span>
<span id="cb189-1583"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1583" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">nrow</span>(result) <span class="sc">&gt;</span> <span class="dv">15</span>)  result <span class="ot">&lt;-</span> result[<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>,]</span>
<span id="cb189-1584"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1585"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1585" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">2</span>] <span class="sc">==</span><span class="dv">1</span>) <span class="cf">next</span>;   <span class="co"># result could be NULL</span></span>
<span id="cb189-1586"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1586" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(i <span class="sc">==</span> <span class="dv">1</span>) result<span class="sc">$</span>direction <span class="ot">=</span> <span class="st">&quot;Up regulated&quot;</span>  <span class="cf">else</span> result<span class="sc">$</span>direction <span class="ot">=</span> <span class="st">&quot;Down regulated&quot;</span></span>
<span id="cb189-1587"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1587" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (pp<span class="sc">==</span><span class="dv">0</span> ) { results1 <span class="ot">&lt;-</span> result; pp <span class="ot">=</span> <span class="dv">1</span>;} <span class="cf">else</span>  results1 <span class="ot">=</span> <span class="fu">rbind</span>(results1,result)</span>
<span id="cb189-1588"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1588" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb189-1589"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1589" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1590"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1590" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> ( pp <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (NoSig)</span>
<span id="cb189-1591"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1591" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> ( <span class="fu">is.null</span>( results1) ) <span class="fu">return</span> (NoSig)</span>
<span id="cb189-1592"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1592" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">==</span> <span class="dv">1</span> ) <span class="fu">return</span>(NoSig)  <span class="co"># Returns a data frame: &quot;No significant results found!&quot;</span></span>
<span id="cb189-1593"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1593" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-1594"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1594" aria-hidden="true" tabindex="-1"></a>        results1<span class="ot">=</span> results1[,<span class="fu">c</span>(<span class="dv">7</span>,<span class="dv">5</span>,<span class="dv">3</span>, <span class="dv">6</span>)]</span>
<span id="cb189-1595"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1595" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(results1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;List&quot;</span>,<span class="st">&quot;FDR&quot;</span>,<span class="st">&quot;nGenes&quot;</span>,<span class="st">&quot;GO terms or pathways&quot;</span>)</span>
<span id="cb189-1596"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1596" aria-hidden="true" tabindex="-1"></a>        minFDR <span class="ot">=</span> <span class="fl">0.01</span></span>
<span id="cb189-1597"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1597" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">min</span>(results1<span class="sc">$</span>FDR) <span class="sc">&gt;</span> minFDR ) results1 <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No signficant enrichment found.&quot;</span>) <span class="cf">else</span></span>
<span id="cb189-1598"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1598" aria-hidden="true" tabindex="-1"></a>        results1 <span class="ot">=</span> results1[<span class="fu">which</span>(results1<span class="sc">$</span>FDR <span class="sc">&lt;</span> minFDR),]</span>
<span id="cb189-1599"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1599" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-1600"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1600" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">!=</span> <span class="dv">4</span>) <span class="fu">return</span>(NoSig)</span>
<span id="cb189-1601"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1601" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(results1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Direction&quot;</span>,<span class="st">&quot;adj.Pval&quot;</span>,<span class="st">&quot;nGenes&quot;</span>,<span class="st">&quot;Pathways&quot;</span>)</span>
<span id="cb189-1602"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1602" aria-hidden="true" tabindex="-1"></a>        results1<span class="sc">$</span>adj.Pval <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;%-2.1e&quot;</span>,<span class="fu">as.numeric</span>(results1<span class="sc">$</span>adj.Pval) )   </span>
<span id="cb189-1603"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1603" aria-hidden="true" tabindex="-1"></a>        <span class="fu">rownames</span>(results1)<span class="ot">=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(results1)</span>
<span id="cb189-1604"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1604" aria-hidden="true" tabindex="-1"></a>        results1[ <span class="fu">duplicated</span> (results1[,<span class="dv">1</span>] ),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span>  </span>
<span id="cb189-1605"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1605" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-1606"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1606" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>( results1 )</span>
<span id="cb189-1607"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1607" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1608"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1608" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1609"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1609" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1610"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1610" aria-hidden="true" tabindex="-1"></a>stringDB_network1 <span class="ot">&lt;-</span> <span class="cf">function</span>( <span class="at">geneLists =</span> <span class="dv">1</span> ) {  <span class="co"># geneLists =2 for both up and down</span></span>
<span id="cb189-1611"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1611" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(STRINGdb)</span>
<span id="cb189-1612"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1612" aria-hidden="true" tabindex="-1"></a>                           </span>
<span id="cb189-1613"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1613" aria-hidden="true" tabindex="-1"></a>        tem <span class="ot">=</span> input_nGenesPPI</span>
<span id="cb189-1614"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1614" aria-hidden="true" tabindex="-1"></a>        taxonomyID <span class="ot">=</span> findTaxonomyID.out</span>
<span id="cb189-1615"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1615" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>( taxonomyID ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)     </span>
<span id="cb189-1616"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1616" aria-hidden="true" tabindex="-1"></a>        <span class="do">####################################</span></span>
<span id="cb189-1617"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1617" aria-hidden="true" tabindex="-1"></a>        NoSig <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No significant enrichment found.&quot;</span>)</span>
<span id="cb189-1618"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1618" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>(STRINGdb_geneList.out ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb189-1619"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1619" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-1620"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1621"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1621" aria-hidden="true" tabindex="-1"></a>        <span class="co">#Intialization</span></span>
<span id="cb189-1622"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1622" aria-hidden="true" tabindex="-1"></a>        string_db <span class="ot">&lt;-</span> STRINGdb<span class="sc">$</span><span class="fu">new</span>( <span class="at">version=</span><span class="st">&quot;10&quot;</span>, <span class="at">species=</span>taxonomyID,</span>
<span id="cb189-1623"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1623" aria-hidden="true" tabindex="-1"></a>                               <span class="at">score_threshold=</span><span class="dv">0</span>, <span class="at">input_directory=</span><span class="st">&quot;&quot;</span> )</span>
<span id="cb189-1624"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1624" aria-hidden="true" tabindex="-1"></a>        <span class="co"># only up regulated is ploted       </span></span>
<span id="cb189-1625"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1625" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( i <span class="cf">in</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span>geneLists) ) {</span>
<span id="cb189-1626"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1626" aria-hidden="true" tabindex="-1"></a>            ids <span class="ot">=</span> STRINGdb_geneList.out[[i]]</span>
<span id="cb189-1627"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1627" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">length</span>(ids)<span class="sc">&gt;</span> input_nGenesPPI )  <span class="co"># n of genes cannot be more than 400</span></span>
<span id="cb189-1628"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1628" aria-hidden="true" tabindex="-1"></a>                ids <span class="ot">&lt;-</span> ids[<span class="dv">1</span><span class="sc">:</span>input_nGenesPPI]</span>
<span id="cb189-1629"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1629" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1630"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1630" aria-hidden="true" tabindex="-1"></a>            string_db<span class="sc">$</span><span class="fu">plot_network</span>( ids,<span class="at">add_link=</span><span class="cn">FALSE</span>)</span>
<span id="cb189-1631"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1631" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1632"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1632" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb189-1633"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1633" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb189-1634"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1634" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1635"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1635" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1636"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1636" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1637"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1637" aria-hidden="true" tabindex="-1"></a><span class="co"># generates a html file containing results and also links to STRING-db</span></span>
<span id="cb189-1638"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1638" aria-hidden="true" tabindex="-1"></a>stringDB_network_link <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb189-1639"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1639" aria-hidden="true" tabindex="-1"></a>        <span class="fu">library</span>(STRINGdb,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb189-1640"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1640" aria-hidden="true" tabindex="-1"></a>        taxonomyID <span class="ot">=</span> findTaxonomyID.out</span>
<span id="cb189-1641"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1641" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>( taxonomyID ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)     </span>
<span id="cb189-1642"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1642" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb189-1643"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1643" aria-hidden="true" tabindex="-1"></a>        NoSig <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No significant enrichment found.&quot;</span>)</span>
<span id="cb189-1644"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1644" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>(STRINGdb_geneList.out ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb189-1645"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1645" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1646"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1646" aria-hidden="true" tabindex="-1"></a>        <span class="co">#Intialization</span></span>
<span id="cb189-1647"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1647" aria-hidden="true" tabindex="-1"></a>        string_db <span class="ot">&lt;-</span> STRINGdb<span class="sc">$</span><span class="fu">new</span>( <span class="at">version=</span><span class="st">&quot;10&quot;</span>, <span class="at">species=</span>taxonomyID,</span>
<span id="cb189-1648"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1648" aria-hidden="true" tabindex="-1"></a>                               <span class="at">score_threshold=</span><span class="dv">0</span>, <span class="at">input_directory=</span><span class="st">&quot;&quot;</span> )</span>
<span id="cb189-1649"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1649" aria-hidden="true" tabindex="-1"></a>            <span class="co"># upregulated</span></span>
<span id="cb189-1650"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1650" aria-hidden="true" tabindex="-1"></a>           ids <span class="ot">=</span> STRINGdb_geneList.out[[<span class="dv">1</span>]]</span>
<span id="cb189-1651"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1651" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">length</span>(ids)<span class="sc">&gt;</span> input_nGenesPPI )  <span class="co"># n of genes cannot be more than 400</span></span>
<span id="cb189-1652"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1652" aria-hidden="true" tabindex="-1"></a>                ids <span class="ot">&lt;-</span> ids[<span class="dv">1</span><span class="sc">:</span>input_nGenesPPI]</span>
<span id="cb189-1653"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1653" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1654"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1654" aria-hidden="true" tabindex="-1"></a>            link1 <span class="ot">=</span> string_db<span class="sc">$</span><span class="fu">get_link</span>( ids)</span>
<span id="cb189-1655"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1655" aria-hidden="true" tabindex="-1"></a>            Pval1 <span class="ot">=</span> string_db<span class="sc">$</span><span class="fu">get_ppi_enrichment</span>( ids)</span>
<span id="cb189-1656"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1656" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="st">&quot;&lt;h5&gt; Interactive and annotated PPI networks among DEGs: &lt;br/&gt;  &quot;</span></span>
<span id="cb189-1657"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1657" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">paste</span>(tem, <span class="st">&quot;&lt;a href=</span><span class="sc">\&quot;</span><span class="st">&quot;</span>, link1, <span class="st">&quot;</span><span class="sc">\&quot;</span><span class="st"> target=</span><span class="sc">\&quot;</span><span class="st">_blank</span><span class="sc">\&quot;</span><span class="st">&gt; Up-regulated; &lt;/a&gt;&quot;</span>  )</span>
<span id="cb189-1658"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1658" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1659"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1659" aria-hidden="true" tabindex="-1"></a>            <span class="co"># downregulated</span></span>
<span id="cb189-1660"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1660" aria-hidden="true" tabindex="-1"></a>            ids <span class="ot">=</span> <span class="fu">STRINGdb_geneList</span>()[[<span class="dv">2</span>]]</span>
<span id="cb189-1661"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1661" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">length</span>(ids)<span class="sc">&gt;</span> input_nGenesPPI )</span>
<span id="cb189-1662"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1662" aria-hidden="true" tabindex="-1"></a>                ids <span class="ot">&lt;-</span> ids[<span class="dv">1</span><span class="sc">:</span>input_nGenesPPI]</span>
<span id="cb189-1663"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1663" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1664"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1664" aria-hidden="true" tabindex="-1"></a>            link2 <span class="ot">=</span> string_db<span class="sc">$</span><span class="fu">get_link</span>( ids)</span>
<span id="cb189-1665"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1665" aria-hidden="true" tabindex="-1"></a>            Pval2 <span class="ot">=</span> string_db<span class="sc">$</span><span class="fu">get_ppi_enrichment</span>( ids)              </span>
<span id="cb189-1666"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1666" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">paste</span>(tem, <span class="st">&quot; &amp;nbsp  &lt;a href=</span><span class="sc">\&quot;</span><span class="st">&quot;</span>, link2, <span class="st">&quot;</span><span class="sc">\&quot;</span><span class="st">target=</span><span class="sc">\&quot;</span><span class="st">_blank</span><span class="sc">\&quot;</span><span class="st">&gt; Down-regulated; &lt;/a&gt;&quot;</span>  )</span>
<span id="cb189-1667"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1667" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb189-1668"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1668" aria-hidden="true" tabindex="-1"></a>            <span class="co"># both up and down with color code</span></span>
<span id="cb189-1669"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1669" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1670"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1670" aria-hidden="true" tabindex="-1"></a>            geneTable <span class="ot">=</span> STRINGdb_geneList.out<span class="sc">$</span>geneTable</span>
<span id="cb189-1671"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1671" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">nrow</span>(geneTable)<span class="sc">&gt;</span> input_nGenesPPI ) </span>
<span id="cb189-1672"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1672" aria-hidden="true" tabindex="-1"></a>                geneTable <span class="ot">&lt;-</span> geneTable[<span class="dv">1</span><span class="sc">:</span>input_nGenesPPI,]</span>
<span id="cb189-1673"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1673" aria-hidden="true" tabindex="-1"></a>            geneTable <span class="ot">=</span>  string_db<span class="sc">$</span><span class="fu">add_diff_exp_color</span>( geneTable, <span class="at">logFcColStr=</span><span class="st">&quot;lfc&quot;</span> ) </span>
<span id="cb189-1674"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1674" aria-hidden="true" tabindex="-1"></a>            payload_id <span class="ot">&lt;-</span> string_db<span class="sc">$</span><span class="fu">post_payload</span>( geneTable<span class="sc">$</span>STRING_id,<span class="at">colors=</span>geneTable<span class="sc">$</span>color )</span>
<span id="cb189-1675"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1675" aria-hidden="true" tabindex="-1"></a>            link3 <span class="ot">=</span> string_db<span class="sc">$</span><span class="fu">get_link</span>(geneTable<span class="sc">$</span>STRING_id, <span class="at">payload_id =</span> payload_id)            </span>
<span id="cb189-1676"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1676" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">paste</span>(tem, <span class="st">&quot; &amp;nbsp  &lt;a href=</span><span class="sc">\&quot;</span><span class="st">&quot;</span>, link3, <span class="st">&quot;</span><span class="sc">\&quot;</span><span class="st">target=</span><span class="sc">\&quot;</span><span class="st">_blank</span><span class="sc">\&quot;</span><span class="st">&gt; Both with fold-changes color coded.&lt;/a&gt;&lt;/h5&gt;&quot;</span>  )</span>
<span id="cb189-1677"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1677" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-1678"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1678" aria-hidden="true" tabindex="-1"></a>            tem2 <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;&lt;h5&gt; PPI enrichment P values: &quot;</span>)  </span>
<span id="cb189-1679"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1679" aria-hidden="true" tabindex="-1"></a>            tem2 <span class="ot">=</span> <span class="fu">paste0</span>(tem2,<span class="st">&quot;Up-regulated: &quot;</span>, <span class="fu">sprintf</span>(<span class="st">&quot;%-3.2e&quot;</span>,Pval1[<span class="dv">1</span>]), <span class="st">&quot; &amp;nbsp Down-regulated: &quot;</span>, <span class="fu">sprintf</span>(<span class="st">&quot;%-3.2e&quot;</span>,Pval2[<span class="dv">1</span>]),<span class="st">&quot;.&quot;</span>)</span>
<span id="cb189-1680"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1680" aria-hidden="true" tabindex="-1"></a>            tem2 <span class="ot">=</span> <span class="fu">paste</span>(tem2, <span class="st">&quot; Small P values indicate more PPIs among DEGs than background. &lt;/h5&gt;&quot;</span> )</span>
<span id="cb189-1681"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1681" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">paste</span>(tem2,tem )</span>
<span id="cb189-1682"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1682" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>(tem) </span>
<span id="cb189-1683"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb189-1683" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="pathway-analysis" class="section level2" number="6.9">
<h2><span class="header-section-number">6.9</span> Pathway analysis</h2>
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-1" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb190-2"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Pathway analysis</span></span>
<span id="cb190-3"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-3" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb190-4"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-5"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-5" aria-hidden="true" tabindex="-1"></a>gagePathwayData <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb190-6"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(gage,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># pathway analysis  </span></span>
<span id="cb190-7"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-8"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-9"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-10"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-10" aria-hidden="true" tabindex="-1"></a>    <span class="do">####################################</span></span>
<span id="cb190-11"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-11" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-12"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-12" aria-hidden="true" tabindex="-1"></a>        myrange <span class="ot">=</span> <span class="fu">c</span>(input_minSetSize, input_maxSetSize)</span>
<span id="cb190-13"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-13" aria-hidden="true" tabindex="-1"></a>        noSig <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No significant pathway found.&quot;</span>)</span>
<span id="cb190-14"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-14" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">length</span>(limma.out<span class="sc">$</span>topGenes) <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(noSig)</span>
<span id="cb190-15"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-15" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb190-16"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-16" aria-hidden="true" tabindex="-1"></a>        { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb190-17"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-17" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb190-18"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-18" aria-hidden="true" tabindex="-1"></a>          top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb190-19"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-19" aria-hidden="true" tabindex="-1"></a>          ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast1, <span class="fu">names</span>(top))</span>
<span id="cb190-20"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-20" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (noSig)</span>
<span id="cb190-21"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-21" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb190-22"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-22" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb190-23"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-23" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (noSig)</span>
<span id="cb190-24"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-24" aria-hidden="true" tabindex="-1"></a>          <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb190-25"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-26"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-26" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">=</span> top1[<span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;</span>input_GenePvalCutoff) , ]</span>
<span id="cb190-27"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-27" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb190-28"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-29"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-29" aria-hidden="true" tabindex="-1"></a>          gmt <span class="ot">=</span> GeneSets.out </span>
<span id="cb190-30"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-30" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">length</span>( GeneSets.out )  <span class="sc">==</span> <span class="dv">0</span>)  { <span class="fu">return</span>(<span class="fu">as.data.frame</span>(<span class="st">&quot;No gene set found!&quot;</span>))}</span>
<span id="cb190-31"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-31" aria-hidden="true" tabindex="-1"></a>          <span class="co">#converted = convertID(rownames(top1),input_selectOrg)</span></span>
<span id="cb190-32"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-32" aria-hidden="true" tabindex="-1"></a>          <span class="co">#</span></span>
<span id="cb190-33"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-33" aria-hidden="true" tabindex="-1"></a>           <span class="co">#gmt = readGeneSets(converted, top1, input_selectGO, input_selectOrg, myrange )</span></span>
<span id="cb190-34"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-34" aria-hidden="true" tabindex="-1"></a>         <span class="co"># cat(&quot;Sets&quot;,length(gmt))</span></span>
<span id="cb190-35"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-36"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-36" aria-hidden="true" tabindex="-1"></a>         fold <span class="ot">=</span> top1[,<span class="dv">1</span>]; <span class="fu">names</span>(fold) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top1)</span>
<span id="cb190-37"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-37" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb190-38"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-38" aria-hidden="true" tabindex="-1"></a>         <span class="cf">if</span>(input_absoluteFold) fold <span class="ot">&lt;-</span> <span class="fu">abs</span>(fold)</span>
<span id="cb190-39"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-39" aria-hidden="true" tabindex="-1"></a>         paths <span class="ot">&lt;-</span> <span class="fu">gage</span>(fold, <span class="at">gsets =</span> gmt, <span class="at">ref =</span> <span class="cn">NULL</span>, <span class="at">samp =</span> <span class="cn">NULL</span>)</span>
<span id="cb190-40"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-41"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-41" aria-hidden="true" tabindex="-1"></a>          paths <span class="ot">&lt;-</span>  <span class="fu">rbind</span>(paths<span class="sc">$</span>greater,paths<span class="sc">$</span>less)</span>
<span id="cb190-42"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-42" aria-hidden="true" tabindex="-1"></a>         <span class="co"># write.csv(paths,&quot;tem.csv&quot;)     </span></span>
<span id="cb190-43"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-43" aria-hidden="true" tabindex="-1"></a>         <span class="co"># cat( dim(paths) )</span></span>
<span id="cb190-44"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-44" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">dim</span>(paths)[<span class="dv">1</span>] <span class="sc">&lt;</span> <span class="dv">1</span> <span class="sc">|</span> <span class="fu">dim</span>(paths)[<span class="dv">2</span>]<span class="sc">&lt;</span> <span class="dv">6</span> ) <span class="fu">return</span>( noSig )</span>
<span id="cb190-45"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-45" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> paths[,<span class="fu">c</span>(<span class="st">&#39;stat.mean&#39;</span>,<span class="st">&#39;set.size&#39;</span>,<span class="st">&#39;q.val&#39;</span>)]</span>
<span id="cb190-46"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-46" aria-hidden="true" tabindex="-1"></a>          <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;statistic&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;adj.Pval&quot;</span>)</span>
<span id="cb190-47"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-47" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top1[<span class="fu">order</span>(top1[,<span class="dv">3</span>]) ,]  </span>
<span id="cb190-48"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-48" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span> ( <span class="fu">length</span>( <span class="fu">which</span>( top1[,<span class="dv">3</span>] <span class="sc">&lt;=</span>  input_pathwayPvalCutoff   ) ) <span class="sc">==</span> <span class="dv">0</span> )</span>
<span id="cb190-49"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-49" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>( noSig)</span>
<span id="cb190-50"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-50" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top1[<span class="fu">which</span>(top1[,<span class="dv">3</span>] <span class="sc">&lt;=</span>  input_pathwayPvalCutoff ) ,,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb190-51"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-51" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">&gt;</span> input_nPathwayShow ) </span>
<span id="cb190-52"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-52" aria-hidden="true" tabindex="-1"></a>             top1 <span class="ot">&lt;-</span> top1[<span class="dv">1</span><span class="sc">:</span>input_nPathwayShow, ,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb190-53"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-53" aria-hidden="true" tabindex="-1"></a>             </span>
<span id="cb190-54"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-54" aria-hidden="true" tabindex="-1"></a>            top1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(top1)</span>
<span id="cb190-55"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-55" aria-hidden="true" tabindex="-1"></a>            top1 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">rep</span>( input_selectContrast1, <span class="fu">dim</span>(top1)[<span class="dv">1</span>]),<span class="fu">row.names</span>(top1), top1); </span>
<span id="cb190-56"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-56" aria-hidden="true" tabindex="-1"></a>            top1<span class="sc">$</span>statistic <span class="ot">&lt;-</span> <span class="fu">as.character</span>( <span class="fu">round</span>(<span class="fu">as.numeric</span>(top1<span class="sc">$</span>statistic),<span class="dv">4</span>)); </span>
<span id="cb190-57"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-57" aria-hidden="true" tabindex="-1"></a>            top1<span class="sc">$</span>adj.Pval <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;%-2.1e&quot;</span>,<span class="fu">as.numeric</span>(top1<span class="sc">$</span>adj.Pval) )</span>
<span id="cb190-58"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-58" aria-hidden="true" tabindex="-1"></a>            top1[,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(top1[,<span class="dv">2</span>]);top1[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(top1[,<span class="dv">1</span>])</span>
<span id="cb190-59"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-59" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(top1)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Direction&quot;</span></span>
<span id="cb190-60"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-60" aria-hidden="true" tabindex="-1"></a>            p.m <span class="ot">&lt;-</span> <span class="st">&quot;GAGE&quot;</span></span>
<span id="cb190-61"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-62"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-62" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(top1)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">paste</span>(p.m,<span class="st">&quot; analysis:&quot;</span>, <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot; vs &quot;</span>,input_selectContrast1 ) )</span>
<span id="cb190-63"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-63" aria-hidden="true" tabindex="-1"></a>            top1[ <span class="fu">which</span>( top1[,<span class="dv">3</span>] <span class="sc">&gt;</span><span class="dv">0</span>),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;Up&quot;</span> <span class="co">#gsub(&quot;-&quot;,&quot; &gt; &quot;,input_selectContrast1 )</span></span>
<span id="cb190-64"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-64" aria-hidden="true" tabindex="-1"></a>            top1[ <span class="fu">which</span>( top1[,<span class="dv">3</span>] <span class="sc">&lt;</span><span class="dv">0</span>),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;Down&quot;</span> <span class="co"># gsub(&quot;-&quot;,&quot; &lt; &quot;,input_selectContrast1 )</span></span>
<span id="cb190-65"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-65" aria-hidden="true" tabindex="-1"></a>            top1 <span class="ot">&lt;-</span> top1[<span class="fu">order</span>( top1[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">abs</span>(<span class="fu">as.numeric</span>( top1[,<span class="dv">3</span>]) ) ) ,]</span>
<span id="cb190-66"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-66" aria-hidden="true" tabindex="-1"></a>            top1[ <span class="fu">duplicated</span> (top1[,<span class="dv">1</span>] ),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb190-67"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-68"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-68" aria-hidden="true" tabindex="-1"></a>          <span class="fu">return</span>( top1)</span>
<span id="cb190-69"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-69" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb190-70"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-71"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-71" aria-hidden="true" tabindex="-1"></a>fgseaPathwayData <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb190-72"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-72" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(fgsea,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># fast GSEA</span></span>
<span id="cb190-73"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-74"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-74" aria-hidden="true" tabindex="-1"></a>    noSig <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No significant pathway found.&quot;</span>)</span>
<span id="cb190-75"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-75" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">length</span>(limma.out<span class="sc">$</span>topGenes) <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(noSig)</span>
<span id="cb190-76"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-76" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb190-77"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-77" aria-hidden="true" tabindex="-1"></a>    { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb190-78"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-78" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb190-79"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-79" aria-hidden="true" tabindex="-1"></a>      top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb190-80"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-80" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast1, <span class="fu">names</span>(top))</span>
<span id="cb190-81"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-81" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (noSig)</span>
<span id="cb190-82"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-82" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb190-83"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-83" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb190-84"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-84" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (noSig)</span>
<span id="cb190-85"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-85" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb190-86"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-86" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb190-87"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-87" aria-hidden="true" tabindex="-1"></a>      <span class="co"># remove some genes</span></span>
<span id="cb190-88"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-88" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">=</span> top1[<span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;</span>input_GenePvalCutoff) , ]</span>
<span id="cb190-89"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-89" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb190-90"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-91"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-91" aria-hidden="true" tabindex="-1"></a>      gmt <span class="ot">=</span> GeneSets.out </span>
<span id="cb190-92"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-92" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>(<span class="fu">length</span>( GeneSets.out )  <span class="sc">==</span> <span class="dv">0</span>)  { <span class="fu">return</span>(<span class="fu">as.data.frame</span>(<span class="st">&quot;No gene set found!&quot;</span>))}</span>
<span id="cb190-93"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-94"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-94" aria-hidden="true" tabindex="-1"></a>      <span class="co">#converted = convertID(rownames(top1),input_selectOrg)</span></span>
<span id="cb190-95"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-95" aria-hidden="true" tabindex="-1"></a>      <span class="co">#</span></span>
<span id="cb190-96"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-96" aria-hidden="true" tabindex="-1"></a>       <span class="co">#gmt = readGeneSets(converted, top1, input_selectGO, input_selectOrg, myrange )</span></span>
<span id="cb190-97"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-97" aria-hidden="true" tabindex="-1"></a>     <span class="co"># cat(&quot;Sets&quot;,length(gmt))</span></span>
<span id="cb190-98"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-99"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-99" aria-hidden="true" tabindex="-1"></a>     fold <span class="ot">=</span> top1[,<span class="dv">1</span>]; <span class="fu">names</span>(fold) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top1)</span>
<span id="cb190-100"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-100" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>(input_absoluteFold) fold <span class="ot">&lt;-</span> <span class="fu">abs</span>(fold) <span class="co"># use absolute value of fold change, disregard direction</span></span>
<span id="cb190-101"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-101" aria-hidden="true" tabindex="-1"></a>     paths <span class="ot">&lt;-</span> <span class="fu">fgsea</span>(<span class="at">pathways =</span> gmt, </span>
<span id="cb190-102"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-102" aria-hidden="true" tabindex="-1"></a>                  <span class="at">stats =</span> fold,</span>
<span id="cb190-103"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-103" aria-hidden="true" tabindex="-1"></a>                  <span class="at">minSize=</span>input_minSetSize,</span>
<span id="cb190-104"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-104" aria-hidden="true" tabindex="-1"></a>                  <span class="at">maxSize=</span>input_maxSetSize,</span>
<span id="cb190-105"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-105" aria-hidden="true" tabindex="-1"></a>                  <span class="at">nperm=</span><span class="dv">10000</span>)</span>
<span id="cb190-106"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-106" aria-hidden="true" tabindex="-1"></a>     <span class="co"># paths &lt;-  rbind(paths$greater,paths$less)</span></span>
<span id="cb190-107"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-107" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(paths)[<span class="dv">1</span>] <span class="sc">&lt;</span> <span class="dv">1</span>  ) <span class="fu">return</span>( noSig )</span>
<span id="cb190-108"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-108" aria-hidden="true" tabindex="-1"></a>           paths <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(paths)</span>
<span id="cb190-109"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-109" aria-hidden="true" tabindex="-1"></a>       paths <span class="ot">&lt;-</span> paths[<span class="fu">order</span>(<span class="sc">-</span><span class="fu">abs</span>( paths[,<span class="dv">5</span>])) ,]  <span class="co"># sort by NES</span></span>
<span id="cb190-110"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-110" aria-hidden="true" tabindex="-1"></a>      <span class="co"># paths &lt;- paths[order( paths[,3]) ,]  # sort by FDR</span></span>
<span id="cb190-111"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-111" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> paths[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">7</span>,<span class="dv">3</span>)]</span>
<span id="cb190-112"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-112" aria-hidden="true" tabindex="-1"></a>      <span class="co"># rownames(top1) &lt;- paths[,1] #paste(1:dim(paths)[1],&quot;: &quot;,paths[,1],sep=&quot;&quot; )</span></span>
<span id="cb190-113"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-113" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Pathway&quot;</span>, <span class="st">&quot;NES&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;adj.Pval&quot;</span>)</span>
<span id="cb190-114"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-114" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb190-115"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-115" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> ( <span class="fu">length</span>( <span class="fu">which</span>( top1[,<span class="dv">4</span>] <span class="sc">&lt;=</span>  input_pathwayPvalCutoff   ) ) <span class="sc">==</span> <span class="dv">0</span> )</span>
<span id="cb190-116"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-116" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>( noSig)</span>
<span id="cb190-117"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-117" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top1[<span class="fu">which</span>(top1[,<span class="dv">4</span>] <span class="sc">&lt;=</span>  input_pathwayPvalCutoff ) , ,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb190-118"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-118" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">&gt;</span> input_nPathwayShow ) </span>
<span id="cb190-119"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-119" aria-hidden="true" tabindex="-1"></a>         top1 <span class="ot">&lt;-</span> top1[<span class="dv">1</span><span class="sc">:</span>input_nPathwayShow,,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb190-120"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-120" aria-hidden="true" tabindex="-1"></a>     <span class="co">#top1 &lt;- cbind(row.names(top1), top1); colnames(top1)[1] &lt;-input_selectContrast1   </span></span>
<span id="cb190-121"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-121" aria-hidden="true" tabindex="-1"></a>        top1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(top1)</span>
<span id="cb190-122"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-122" aria-hidden="true" tabindex="-1"></a>        top1 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">rep</span>( input_selectContrast1, <span class="fu">dim</span>(top1)[<span class="dv">1</span>]), top1); </span>
<span id="cb190-123"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-123" aria-hidden="true" tabindex="-1"></a>        top1[,<span class="dv">4</span>] <span class="ot">=</span> <span class="fu">as.character</span>( <span class="fu">round</span>(<span class="fu">as.numeric</span>(top1[,<span class="dv">4</span>]),<span class="dv">4</span>)); </span>
<span id="cb190-124"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-124" aria-hidden="true" tabindex="-1"></a>        top1<span class="sc">$</span>adj.Pval <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;%-2.1e&quot;</span>,<span class="fu">as.numeric</span>(top1<span class="sc">$</span>adj.Pval) )</span>
<span id="cb190-125"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-125" aria-hidden="true" tabindex="-1"></a>        top1[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(top1[,<span class="dv">1</span>])</span>
<span id="cb190-126"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-126" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(top1)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Direction&quot;</span></span>
<span id="cb190-127"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-128"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-128" aria-hidden="true" tabindex="-1"></a>        p.m <span class="ot">&lt;-</span> <span class="st">&quot;GSEA&quot;</span></span>
<span id="cb190-129"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-130"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-130" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(top1)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">paste</span>(p.m,<span class="st">&quot; analysis:&quot;</span>, <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot; vs &quot;</span>,input_selectContrast1 ) )</span>
<span id="cb190-131"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-131" aria-hidden="true" tabindex="-1"></a>        top1[ <span class="fu">which</span>( <span class="fu">as.numeric</span>( top1[,<span class="dv">3</span>]) <span class="sc">&gt;</span><span class="dv">0</span>),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;Up&quot;</span> <span class="co">#gsub(&quot;-&quot;,&quot; &gt; &quot;,input_selectContrast1 )</span></span>
<span id="cb190-132"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-132" aria-hidden="true" tabindex="-1"></a>        top1[ <span class="fu">which</span>( <span class="fu">as.numeric</span>( top1[,<span class="dv">3</span>]) <span class="sc">&lt;</span><span class="dv">0</span>),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;Down&quot;</span> <span class="co">#gsub(&quot;-&quot;,&quot; &lt; &quot;,input_selectContrast1 )</span></span>
<span id="cb190-133"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-133" aria-hidden="true" tabindex="-1"></a>        top1 <span class="ot">&lt;-</span> top1[<span class="fu">order</span>( top1[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">abs</span>(<span class="fu">as.numeric</span>( top1[,<span class="dv">3</span>]) ) ) ,]</span>
<span id="cb190-134"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-134" aria-hidden="true" tabindex="-1"></a>        top1[ <span class="fu">duplicated</span> (top1[,<span class="dv">1</span>] ),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span>   </span>
<span id="cb190-135"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-135" aria-hidden="true" tabindex="-1"></a>        top1[,<span class="dv">3</span>] <span class="ot">=</span> <span class="fu">as.character</span>( <span class="fu">round</span>(<span class="fu">as.numeric</span>(top1[,<span class="dv">3</span>]),<span class="dv">4</span>));</span>
<span id="cb190-136"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-137"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-137" aria-hidden="true" tabindex="-1"></a>     <span class="fu">return</span>( top1)</span>
<span id="cb190-138"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-138" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb190-139"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-140"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-140" aria-hidden="true" tabindex="-1"></a>extract1 <span class="ot">&lt;-</span> <span class="cf">function</span> (x) {</span>
<span id="cb190-141"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-141" aria-hidden="true" tabindex="-1"></a>  words <span class="ot">&lt;-</span> <span class="fu">unlist</span> ( <span class="fu">strsplit</span>(x,<span class="st">&quot;_&quot;</span>))</span>
<span id="cb190-142"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-142" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">length</span>( words )  <span class="sc">&lt;=</span><span class="dv">4</span> ) <span class="fu">return</span>(<span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>,<span class="st">&quot; &quot;</span>,x)) <span class="cf">else</span> {</span>
<span id="cb190-143"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-143" aria-hidden="true" tabindex="-1"></a>  words <span class="ot">&lt;-</span> words[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)]</span>
<span id="cb190-144"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-144" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>( <span class="fu">proper</span>(<span class="fu">paste</span>(words,<span class="at">collapse =</span> <span class="st">&quot; &quot;</span>) ) )}</span>
<span id="cb190-145"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-145" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb190-146"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-147"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-147" aria-hidden="true" tabindex="-1"></a>PGSEApathway <span class="ot">&lt;-</span> <span class="cf">function</span> (converted,convertedData, selectOrg,GO,gmt, myrange,Pval_pathway,top){</span>
<span id="cb190-148"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-148" aria-hidden="true" tabindex="-1"></a>    subtype <span class="ot">=</span> <span class="fu">detectGroups</span>(<span class="fu">colnames</span>(convertedData))</span>
<span id="cb190-149"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-149" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(PGSEA,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb190-150"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-150" aria-hidden="true" tabindex="-1"></a>    Pvalue <span class="ot">=</span> <span class="fl">0.01</span>  <span class="co"># cut off to report in PGSEA. Otherwise NA</span></span>
<span id="cb190-151"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-151" aria-hidden="true" tabindex="-1"></a>    <span class="co">#Pval_pathway = 0.2   # cut off for P value of ANOVA test  to writ to file </span></span>
<span id="cb190-152"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-152" aria-hidden="true" tabindex="-1"></a>    <span class="co"># top = 30   # number of pathways to show</span></span>
<span id="cb190-153"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-153" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(gmt) <span class="sc">==</span><span class="dv">0</span> ) <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">pg3 =</span> <span class="cn">NULL</span>, <span class="at">best =</span> <span class="dv">1</span> ) )</span>
<span id="cb190-154"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-154" aria-hidden="true" tabindex="-1"></a>    <span class="co"># centering by mean</span></span>
<span id="cb190-155"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-155" aria-hidden="true" tabindex="-1"></a>    <span class="co">#pg = myPGSEA (convertedData - rowMeans(convertedData),</span></span>
<span id="cb190-156"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-156" aria-hidden="true" tabindex="-1"></a>     <span class="co">#            cl=gmt,range=myrange,p.value=TRUE, weighted=FALSE,nPermutation=100)</span></span>
<span id="cb190-157"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-158"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-158" aria-hidden="true" tabindex="-1"></a>    <span class="co">#if( class(convertedData) != &quot;data.frame&quot; | class(convertedData) != &quot;matarix&quot;) return( list(pg3 = NULL, best = 1 ) )</span></span>
<span id="cb190-159"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-159" aria-hidden="true" tabindex="-1"></a>    <span class="co">#if( dim(convertedData)[2] &lt;2 ) return( list(pg3 = NULL, best = 1 ) )</span></span>
<span id="cb190-160"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-160" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-161"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-161" aria-hidden="true" tabindex="-1"></a>     pg <span class="ot">=</span> <span class="fu">PGSEA</span> (convertedData <span class="sc">-</span> <span class="fu">rowMeans</span>(convertedData),<span class="at">cl=</span>gmt,<span class="at">range=</span>myrange,<span class="at">p.value=</span><span class="cn">TRUE</span>, <span class="at">weighted=</span><span class="cn">FALSE</span>)</span>
<span id="cb190-162"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-162" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-163"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-163" aria-hidden="true" tabindex="-1"></a>    pg2 <span class="ot">=</span> pg<span class="sc">$</span>results;</span>
<span id="cb190-164"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-164" aria-hidden="true" tabindex="-1"></a>    pg2 <span class="ot">=</span> pg2[<span class="fu">rowSums</span>(<span class="fu">is.na</span>(pg2))<span class="sc">&lt;</span><span class="fu">ncol</span>(pg2) ,]  <span class="co"># remove se/wrts with all missing(non-signficant)</span></span>
<span id="cb190-165"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-165" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">dim</span>(pg2)[<span class="dv">1</span>] <span class="sc">&lt;</span> <span class="dv">2</span> ) return.out</span>
<span id="cb190-166"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-166" aria-hidden="true" tabindex="-1"></a>    best <span class="ot">=</span> <span class="fu">max</span>(<span class="fu">abs</span>(pg2))</span>
<span id="cb190-167"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-167" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-168"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-168" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(subtype) <span class="sc">&lt;</span> <span class="dv">4</span> <span class="sc">||</span> <span class="fu">length</span>(<span class="fu">unique</span>(subtype)) <span class="sc">&lt;</span><span class="dv">2</span> <span class="sc">||</span><span class="fu">length</span>(<span class="fu">unique</span>(subtype)) <span class="sc">==</span> <span class="fu">dim</span>(convertedData)[<span class="dv">2</span>] ) { </span>
<span id="cb190-169"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-169" aria-hidden="true" tabindex="-1"></a>     pg2 <span class="ot">=</span> pg2[<span class="fu">order</span>(<span class="sc">-</span><span class="fu">apply</span>(pg2,<span class="dv">1</span>,sd)     )   ,]</span>
<span id="cb190-170"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-170" aria-hidden="true" tabindex="-1"></a>     <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">pg3 =</span> pg2[<span class="dv">1</span><span class="sc">:</span>top,], <span class="at">best =</span> best ) )</span>
<span id="cb190-171"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-171" aria-hidden="true" tabindex="-1"></a>    } </span>
<span id="cb190-172"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-172" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-173"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-173" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">Computing P values using ANOVA</span><span class="sc">\n</span><span class="st">&quot;</span>);</span>
<span id="cb190-174"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-174" aria-hidden="true" tabindex="-1"></a>    pathPvalue <span class="ot">&lt;-</span> <span class="cf">function</span> ( k){</span>
<span id="cb190-175"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-175" aria-hidden="true" tabindex="-1"></a>     <span class="fu">return</span>( <span class="fu">summary</span>(<span class="fu">aov</span>(pg2[k,]<span class="sc">~</span>subtype) )[[<span class="dv">1</span>]][[<span class="st">&quot;Pr(&gt;F)&quot;</span>]][<span class="dv">1</span>] )</span>
<span id="cb190-176"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-176" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb190-177"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-177" aria-hidden="true" tabindex="-1"></a>    Pvalues <span class="ot">=</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(pg2)[<span class="dv">1</span>], pathPvalue)</span>
<span id="cb190-178"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-178" aria-hidden="true" tabindex="-1"></a>    Pvalues <span class="ot">=</span> <span class="fu">p.adjust</span>(Pvalues, <span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb190-179"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-179" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-180"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-180" aria-hidden="true" tabindex="-1"></a>    <span class="co">#if(min(Pvalues) &gt; Pval_pathway ) return( list(pg3 = NULL, best = best ) )  else {  </span></span>
<span id="cb190-181"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-181" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">sort</span>(Pvalues)[<span class="dv">2</span>] <span class="sc">&gt;</span> Pval_pathway ) <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">pg3 =</span> <span class="cn">NULL</span>, <span class="at">best =</span> best ) )  <span class="cf">else</span> {  </span>
<span id="cb190-182"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-183"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-183" aria-hidden="true" tabindex="-1"></a>    NsigT <span class="ot">=</span> <span class="fu">rowSums</span>(pg<span class="sc">$</span>p.results<span class="sc">&lt;</span>Pvalue)</span>
<span id="cb190-184"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-184" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-185"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-185" aria-hidden="true" tabindex="-1"></a>    result<span class="ot">=</span><span class="fu">cbind</span>( <span class="fu">as.matrix</span>(Pvalues),NsigT,pg2); </span>
<span id="cb190-186"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-186" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">=</span> result[ <span class="fu">order</span>(result[,<span class="dv">1</span>])   ,] </span>
<span id="cb190-187"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-187" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-188"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-188" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">=</span> result[<span class="fu">which</span>(result[,<span class="dv">1</span>] <span class="sc">&lt;</span> Pval_pathway),,drop<span class="ot">=</span>F]</span>
<span id="cb190-189"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-189" aria-hidden="true" tabindex="-1"></a>    <span class="co">#result = result[which(result[,2] &gt;2)    ,]</span></span>
<span id="cb190-190"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-190" aria-hidden="true" tabindex="-1"></a>    pg2 <span class="ot">=</span> result[,<span class="sc">-</span><span class="dv">2</span>]</span>
<span id="cb190-191"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-192"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-192" aria-hidden="true" tabindex="-1"></a>    <span class="co"># when there is only 1 left in the matrix pg2 becomes a vector</span></span>
<span id="cb190-193"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-193" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">sum</span>( Pvalues<span class="sc">&lt;</span>Pval_pathway) <span class="sc">==</span> <span class="dv">1</span>) { pg3 <span class="ot">=</span> <span class="fu">t</span>( <span class="fu">as.matrix</span>(pg2));pg3 <span class="ot">=</span> <span class="fu">rbind</span>(pg3,pg3);} <span class="cf">else</span></span>
<span id="cb190-194"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-194" aria-hidden="true" tabindex="-1"></a>    { <span class="cf">if</span>(<span class="fu">dim</span>(pg2)[<span class="dv">1</span>] <span class="sc">&gt;</span> top ) {  pg3 <span class="ot">=</span> pg2[<span class="dv">1</span><span class="sc">:</span>top,]; } <span class="cf">else</span> {  pg3 <span class="ot">=</span> pg2;  } }</span>
<span id="cb190-195"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-196"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-196" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(pg3) <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">rownames</span>(pg3) , extract1)</span>
<span id="cb190-197"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-197" aria-hidden="true" tabindex="-1"></a>    a<span class="ot">=</span><span class="fu">sprintf</span>(<span class="st">&quot;%-3.2e&quot;</span>,pg3[,<span class="dv">1</span>])</span>
<span id="cb190-198"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-198" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(pg3) <span class="ot">=</span> <span class="fu">paste</span>(a,<span class="fu">rownames</span>(pg3),<span class="at">sep=</span><span class="st">&quot; &quot;</span>)</span>
<span id="cb190-199"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-199" aria-hidden="true" tabindex="-1"></a>    pg3 <span class="ot">=</span>pg3[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb190-200"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-200" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-201"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-201" aria-hidden="true" tabindex="-1"></a>    pg3 <span class="ot">&lt;-</span> pg3[<span class="fu">order</span>( <span class="sc">-</span><span class="fu">apply</span>(pg3,<span class="dv">1</span>,sd)    ),] <span class="co"># sort by SD</span></span>
<span id="cb190-202"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-202" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb190-203"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-203" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">pg3 =</span> pg3, <span class="at">best =</span> best ) )</span>
<span id="cb190-204"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-204" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb190-205"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-205" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb190-206"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-207"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-207" aria-hidden="true" tabindex="-1"></a>PGSEAplot <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb190-208"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-208" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb190-209"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-209" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(PGSEA,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb190-210"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-211"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-211" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_selectGO <span class="sc">==</span> <span class="st">&quot;ID not recognized!&quot;</span> ) <span class="fu">return</span>( <span class="cn">NULL</span>)</span>
<span id="cb190-212"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-213"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-213" aria-hidden="true" tabindex="-1"></a>    myrange <span class="ot">=</span> <span class="fu">c</span>(input_minSetSize, input_maxSetSize)</span>
<span id="cb190-214"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-214" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">=</span> convertedData.out</span>
<span id="cb190-215"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-215" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">is.null</span>(input_selectContrast1 ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb190-216"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-217"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-217" aria-hidden="true" tabindex="-1"></a>    gmt <span class="ot">=</span> GeneSets.out</span>
<span id="cb190-218"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-219"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-220"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-220" aria-hidden="true" tabindex="-1"></a>    <span class="co"># find related samples</span></span>
<span id="cb190-221"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-221" aria-hidden="true" tabindex="-1"></a>    iz <span class="ot">=</span> <span class="fu">findContrastSamples</span>(input_selectContrast1, <span class="fu">colnames</span>(convertedData.out),readSampleInfo.out,</span>
<span id="cb190-222"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-222" aria-hidden="true" tabindex="-1"></a>                                        input_selectFactorsModel,input_selectModelComprions, </span>
<span id="cb190-223"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-223" aria-hidden="true" tabindex="-1"></a>                                        factorReferenceLevels.out,input_CountsDEGMethod,</span>
<span id="cb190-224"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-224" aria-hidden="true" tabindex="-1"></a>                                        input_dataFileFormat  )</span>
<span id="cb190-225"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-226"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-226" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-227"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-227" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-228"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-228" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">=</span> genes[,iz]  </span>
<span id="cb190-229"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-229" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-230"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-230" aria-hidden="true" tabindex="-1"></a>    subtype <span class="ot">=</span> <span class="fu">detectGroups</span>(<span class="fu">colnames</span>(genes )) </span>
<span id="cb190-231"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-231" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( GeneSets.out )  <span class="sc">==</span> <span class="dv">0</span>)  { <span class="fu">plot.new</span>(); <span class="fu">text</span>(<span class="fl">0.5</span>,<span class="fl">0.5</span>, <span class="st">&quot;No gene sets!&quot;</span>)} <span class="cf">else</span> {</span>
<span id="cb190-232"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-232" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">=</span> <span class="fu">PGSEApathway</span>(converted.out,genes, input_selectOrg,input_selectGO,</span>
<span id="cb190-233"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-233" aria-hidden="true" tabindex="-1"></a>                 GeneSets.out,  myrange, input_pathwayPvalCutoff, input_nPathwayShow    )</span>
<span id="cb190-234"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-234" aria-hidden="true" tabindex="-1"></a>                     </span>
<span id="cb190-235"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-235" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">is.null</span>(result<span class="sc">$</span>pg3) ) { <span class="fu">plot.new</span>(); <span class="fu">text</span>(<span class="fl">0.5</span>,<span class="dv">1</span>, <span class="st">&quot;No significant pathway found!&quot;</span>)} <span class="cf">else</span> </span>
<span id="cb190-236"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-236" aria-hidden="true" tabindex="-1"></a>    <span class="fu">smcPlot</span>(result<span class="sc">$</span>pg3,<span class="fu">factor</span>(subtype),<span class="at">scale =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fu">max</span>(result<span class="sc">$</span>pg3), <span class="fu">max</span>(result<span class="sc">$</span>pg3)), </span>
<span id="cb190-237"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-237" aria-hidden="true" tabindex="-1"></a>    <span class="at">show.grid =</span> T, <span class="at">margins =</span> <span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">1</span>, <span class="dv">13</span>,<span class="dv">28</span>), <span class="at">col =</span> .rwb,<span class="at">cex.lab=</span><span class="fl">0.5</span>, <span class="at">main=</span><span class="st">&quot;Pathway Analysis:PGSEA&quot;</span>)</span>
<span id="cb190-238"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-238" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb190-239"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-239" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-240"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-240" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb190-241"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-242"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-242" aria-hidden="true" tabindex="-1"></a>convertEnsembl2Entrez <span class="ot">&lt;-</span> <span class="cf">function</span> (query,Species) { </span>
<span id="cb190-243"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-243" aria-hidden="true" tabindex="-1"></a>    querySet <span class="ot">&lt;-</span> <span class="fu">cleanGeneSet</span>( <span class="fu">unlist</span>( <span class="fu">strsplit</span>( <span class="fu">toupper</span>(<span class="fu">names</span>( query)),<span class="st">&#39;</span><span class="sc">\t</span><span class="st">| |</span><span class="sc">\n</span><span class="st">|</span><span class="sc">\\</span><span class="st">,&#39;</span> )  ) )</span>
<span id="cb190-244"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-244" aria-hidden="true" tabindex="-1"></a>    speciesID <span class="ot">&lt;-</span> orgInfo<span class="sc">$</span>id[ <span class="fu">which</span>(orgInfo<span class="sc">$</span>ensembl_dataset <span class="sc">==</span> Species)]  <span class="co"># note uses species Identifying</span></span>
<span id="cb190-245"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-245" aria-hidden="true" tabindex="-1"></a>    <span class="co"># idType 6 for entrez gene ID</span></span>
<span id="cb190-246"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-246" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>( convert,</span>
<span id="cb190-247"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-247" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">paste</span>( <span class="st">&quot; select  id,ens,species from mapping where ens IN (&#39;&quot;</span>, <span class="fu">paste</span>(querySet,<span class="at">collapse=</span><span class="st">&quot;&#39;, &#39;&quot;</span>),</span>
<span id="cb190-248"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-248" aria-hidden="true" tabindex="-1"></a>                                <span class="st">&quot;&#39;) AND  idType =&#39;&quot;</span>,idType_Entrez,<span class="st">&quot;&#39;&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>) ) <span class="co"># slow</span></span>
<span id="cb190-249"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-249" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb190-250"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-250" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>  ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb190-251"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-251" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">subset</span>(result, species<span class="sc">==</span>speciesID, <span class="at">select =</span> <span class="sc">-</span>species)</span>
<span id="cb190-252"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-253"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-253" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span> <span class="fu">match</span>(result<span class="sc">$</span>ens,<span class="fu">names</span>(query)  )</span>
<span id="cb190-254"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-255"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-255" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">&lt;-</span> query[ix];  <span class="fu">names</span>(tem) <span class="ot">=</span> result<span class="sc">$</span>id</span>
<span id="cb190-256"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-256" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(tem)</span>
<span id="cb190-257"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-257" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb190-258"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-258" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb190-259"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-260"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-260" aria-hidden="true" tabindex="-1"></a>ReactomePAPathwayData <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb190-261"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-261" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(ReactomePA,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># pathway analysis</span></span>
<span id="cb190-262"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-262" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb190-263"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-263" aria-hidden="true" tabindex="-1"></a>    ensemblSpecies <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;hsapiens_gene_ensembl&quot;</span>,<span class="st">&quot;rnorvegicus_gene_ensembl&quot;</span>, <span class="st">&quot;mmusculus_gene_ensembl&quot;</span>,</span>
<span id="cb190-264"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-264" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&quot;celegans_gene_ensembl&quot;</span>,<span class="st">&quot;scerevisiae_gene_ensembl&quot;</span>, <span class="st">&quot;drerio_gene_ensembl&quot;</span>, <span class="st">&quot;dmelanogaster_gene_ensembl&quot;</span>)</span>
<span id="cb190-265"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-265" aria-hidden="true" tabindex="-1"></a>    ReactomePASpecies<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;human&quot;</span>, <span class="st">&quot;rat&quot;</span>, <span class="st">&quot;mouse&quot;</span>, <span class="st">&quot;celegans&quot;</span>, <span class="st">&quot;yeast&quot;</span>, <span class="st">&quot;zebrafish&quot;</span>, <span class="st">&quot;fly&quot;</span> )</span>
<span id="cb190-266"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-266" aria-hidden="true" tabindex="-1"></a>    <span class="co"># cbind(ensemblSpecies,ReactomePASpecies)  # double check mapping</span></span>
<span id="cb190-267"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-267" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-268"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-268" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-269"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-269" aria-hidden="true" tabindex="-1"></a>    myrange <span class="ot">=</span> <span class="fu">c</span>(input_minSetSize, input_maxSetSize)</span>
<span id="cb190-270"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-270" aria-hidden="true" tabindex="-1"></a>    noSig <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No significant pathway found.&quot;</span>)</span>
<span id="cb190-271"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-271" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">length</span>(limma.out<span class="sc">$</span>topGenes) <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(noSig)</span>
<span id="cb190-272"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-272" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb190-273"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-273" aria-hidden="true" tabindex="-1"></a>    { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb190-274"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-274" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb190-275"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-275" aria-hidden="true" tabindex="-1"></a>      top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb190-276"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-276" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast1, <span class="fu">names</span>(top))</span>
<span id="cb190-277"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-277" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (noSig)</span>
<span id="cb190-278"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-278" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb190-279"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-279" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb190-280"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-280" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (noSig)</span>
<span id="cb190-281"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-281" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb190-282"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-283"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-283" aria-hidden="true" tabindex="-1"></a>      <span class="co"># remove some genes</span></span>
<span id="cb190-284"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-284" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">=</span> top1[<span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;</span>input_GenePvalCutoff) , ]</span>
<span id="cb190-285"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-285" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb190-286"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-286" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb190-287"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-287" aria-hidden="true" tabindex="-1"></a>     fold <span class="ot">=</span> top1[,<span class="dv">1</span>]; <span class="fu">names</span>(fold) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top1)</span>
<span id="cb190-288"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-288" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>(input_absoluteFold) fold <span class="ot">&lt;-</span> <span class="fu">abs</span>(fold) <span class="co"># use absolute value of fold change, disregard direction</span></span>
<span id="cb190-289"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-290"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-290" aria-hidden="true" tabindex="-1"></a>       Species <span class="ot">&lt;-</span> converted.out<span class="sc">$</span>species</span>
<span id="cb190-291"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-291" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">&lt;-</span> <span class="fu">match</span>( Species,ensemblSpecies ) </span>
<span id="cb190-292"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-292" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.na</span>(ix) ) <span class="fu">return</span>(<span class="fu">as.data.frame</span>(<span class="st">&quot;Species not coverted by ReactomePA package!&quot;</span>))</span>
<span id="cb190-293"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-293" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb190-294"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-294" aria-hidden="true" tabindex="-1"></a>      fold <span class="ot">&lt;-</span> <span class="fu">convertEnsembl2Entrez</span> (fold, Species)  </span>
<span id="cb190-295"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-295" aria-hidden="true" tabindex="-1"></a>         fold <span class="ot">&lt;-</span> <span class="fu">sort</span>(fold,<span class="at">decreasing =</span>T)</span>
<span id="cb190-296"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-296" aria-hidden="true" tabindex="-1"></a>      paths <span class="ot">&lt;-</span> <span class="fu">gsePathway</span>(fold, <span class="at">nPerm=</span><span class="dv">5000</span>, <span class="at">organism =</span> ReactomePASpecies[ix],</span>
<span id="cb190-297"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-297" aria-hidden="true" tabindex="-1"></a>                <span class="at">minGSSize=</span> input_minSetSize, </span>
<span id="cb190-298"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-298" aria-hidden="true" tabindex="-1"></a>                <span class="at">maxGSSize=</span> input_maxSetSize,</span>
<span id="cb190-299"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-299" aria-hidden="true" tabindex="-1"></a>                <span class="at">pvalueCutoff=</span><span class="fl">0.5</span>,</span>
<span id="cb190-300"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-300" aria-hidden="true" tabindex="-1"></a>                <span class="at">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb190-301"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-301" aria-hidden="true" tabindex="-1"></a>      paths <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(paths)</span>
<span id="cb190-302"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-302" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb190-303"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-303" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.null</span>(paths) ) <span class="fu">return</span>( noSig)</span>
<span id="cb190-304"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-304" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(paths)[<span class="dv">1</span>] <span class="sc">==</span><span class="dv">0</span> ) <span class="fu">return</span>( noSig)</span>
<span id="cb190-305"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-306"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-306" aria-hidden="true" tabindex="-1"></a>     <span class="co"># paths &lt;-  rbind(paths$greater,paths$less)</span></span>
<span id="cb190-307"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-307" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(paths)[<span class="dv">1</span>] <span class="sc">&lt;</span> <span class="dv">1</span>  ) <span class="fu">return</span>( noSig )</span>
<span id="cb190-308"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-308" aria-hidden="true" tabindex="-1"></a>           paths <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(paths)</span>
<span id="cb190-309"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-309" aria-hidden="true" tabindex="-1"></a>       paths <span class="ot">&lt;-</span> paths[<span class="fu">order</span>(<span class="sc">-</span><span class="fu">abs</span>( paths[,<span class="dv">5</span>])) ,]  <span class="co"># sort by NES</span></span>
<span id="cb190-310"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-310" aria-hidden="true" tabindex="-1"></a>      <span class="co"># paths &lt;- paths[order( paths[,3]) ,]  # sort by FDR</span></span>
<span id="cb190-311"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-311" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> paths[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">3</span>,<span class="dv">7</span>)]</span>
<span id="cb190-312"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-312" aria-hidden="true" tabindex="-1"></a>      <span class="co"># rownames(top1) &lt;- paths[,1] #paste(1:dim(paths)[1],&quot;: &quot;,paths[,1],sep=&quot;&quot; )</span></span>
<span id="cb190-313"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-313" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Pathway&quot;</span>, <span class="st">&quot;NES&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;adj.Pval&quot;</span>)</span>
<span id="cb190-314"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-314" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb190-315"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-315" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> ( <span class="fu">length</span>( <span class="fu">which</span>( top1[,<span class="dv">4</span>] <span class="sc">&lt;=</span>  input_pathwayPvalCutoff   ) ) <span class="sc">==</span> <span class="dv">0</span> )</span>
<span id="cb190-316"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-316" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>( noSig)</span>
<span id="cb190-317"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-317" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top1[<span class="fu">which</span>(top1[,<span class="dv">4</span>] <span class="sc">&lt;=</span>  input_pathwayPvalCutoff ) ,,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb190-318"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-318" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">&gt;</span> input_nPathwayShow ) </span>
<span id="cb190-319"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-319" aria-hidden="true" tabindex="-1"></a>         top1 <span class="ot">&lt;-</span> top1[<span class="dv">1</span><span class="sc">:</span>input_nPathwayShow,,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb190-320"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-320" aria-hidden="true" tabindex="-1"></a>     <span class="co">#top1 &lt;- cbind(row.names(top1), top1); colnames(top1)[1] &lt;-input_selectContrast1   </span></span>
<span id="cb190-321"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-321" aria-hidden="true" tabindex="-1"></a>        top1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(top1)</span>
<span id="cb190-322"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-322" aria-hidden="true" tabindex="-1"></a>        top1 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">rep</span>( input_selectContrast1, <span class="fu">dim</span>(top1)[<span class="dv">1</span>]), top1); </span>
<span id="cb190-323"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-323" aria-hidden="true" tabindex="-1"></a>        top1[,<span class="dv">4</span>] <span class="ot">=</span> <span class="fu">as.character</span>( <span class="fu">round</span>(<span class="fu">as.numeric</span>(top1[,<span class="dv">4</span>]),<span class="dv">4</span>)); </span>
<span id="cb190-324"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-324" aria-hidden="true" tabindex="-1"></a>        top1<span class="sc">$</span>adj.Pval <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;%-2.1e&quot;</span>,<span class="fu">as.numeric</span>(top1<span class="sc">$</span>adj.Pval) )</span>
<span id="cb190-325"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-325" aria-hidden="true" tabindex="-1"></a>        top1[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(top1[,<span class="dv">1</span>])</span>
<span id="cb190-326"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-326" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(top1)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Direction&quot;</span></span>
<span id="cb190-327"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-327" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(input_pathwayMethod <span class="sc">==</span> <span class="dv">1</span> ) p.m <span class="ot">&lt;-</span> <span class="st">&quot;GAGE&quot;</span></span>
<span id="cb190-328"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-328" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span> <span class="cf">if</span>(input_pathwayMethod <span class="sc">==</span> <span class="dv">2</span> ) p.m <span class="ot">&lt;-</span> <span class="st">&quot;PGSEA&quot;</span></span>
<span id="cb190-329"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-329" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span> <span class="cf">if</span>(input_pathwayMethod <span class="sc">==</span> <span class="dv">3</span> ) p.m <span class="ot">&lt;-</span> <span class="st">&quot;GSEA&quot;</span></span>
<span id="cb190-330"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-330" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span> <span class="cf">if</span>(input_pathwayMethod <span class="sc">==</span> <span class="dv">4</span> ) p.m <span class="ot">&lt;-</span> <span class="st">&quot;PGSEA_All&quot;</span></span>
<span id="cb190-331"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-331" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span> <span class="cf">if</span>(input_pathwayMethod <span class="sc">==</span> <span class="dv">5</span> ) p.m <span class="ot">&lt;-</span> <span class="st">&quot;ReactomePA&quot;</span></span>
<span id="cb190-332"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-332" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(top1)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">paste</span>(p.m,<span class="st">&quot; analysis:&quot;</span>, <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot; vs &quot;</span>,input_selectContrast1 ) )</span>
<span id="cb190-333"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-333" aria-hidden="true" tabindex="-1"></a>        top1[ <span class="fu">which</span>( <span class="fu">as.numeric</span>( top1[,<span class="dv">3</span>]) <span class="sc">&gt;</span><span class="dv">0</span>),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;Up&quot;</span> <span class="co">#gsub(&quot;-&quot;,&quot; &gt; &quot;,input_selectContrast1 )</span></span>
<span id="cb190-334"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-334" aria-hidden="true" tabindex="-1"></a>        top1[ <span class="fu">which</span>( <span class="fu">as.numeric</span>( top1[,<span class="dv">3</span>]) <span class="sc">&lt;</span><span class="dv">0</span>),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;Down&quot;</span> <span class="co">#gsub(&quot;-&quot;,&quot; &lt; &quot;,input_selectContrast1 )</span></span>
<span id="cb190-335"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-335" aria-hidden="true" tabindex="-1"></a>        top1 <span class="ot">&lt;-</span> top1[<span class="fu">order</span>( top1[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">abs</span>(<span class="fu">as.numeric</span>( top1[,<span class="dv">3</span>]) ) ) ,]</span>
<span id="cb190-336"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-336" aria-hidden="true" tabindex="-1"></a>        top1[ <span class="fu">duplicated</span> (top1[,<span class="dv">1</span>] ),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span>   </span>
<span id="cb190-337"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-337" aria-hidden="true" tabindex="-1"></a>      top1[,<span class="dv">3</span>] <span class="ot">=</span> <span class="fu">as.character</span>( <span class="fu">round</span>(<span class="fu">as.numeric</span>(top1[,<span class="dv">3</span>]),<span class="dv">4</span>));</span>
<span id="cb190-338"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-338" aria-hidden="true" tabindex="-1"></a>     <span class="fu">return</span>( top1)</span>
<span id="cb190-339"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-339" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb190-340"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-341"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-341" aria-hidden="true" tabindex="-1"></a>selectedPathwayData <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb190-342"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-342" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb190-343"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-343" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_sigPathways <span class="sc">==</span> <span class="st">&quot;All&quot;</span>) <span class="fu">return</span> (<span class="cn">NULL</span>) </span>
<span id="cb190-344"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-344" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">names</span>(GeneSets.out ) <span class="sc">==</span> input_sigPathways   ) <span class="co"># find the gene set</span></span>
<span id="cb190-345"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-345" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb190-346"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-346" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">&lt;-</span> GeneSets.out[[ix]] <span class="co"># retrieve genes</span></span>
<span id="cb190-347"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-348"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-348" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-349"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-349" aria-hidden="true" tabindex="-1"></a>    <span class="co"># find related samples  </span></span>
<span id="cb190-350"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-350" aria-hidden="true" tabindex="-1"></a>    iz <span class="ot">=</span> <span class="fu">findContrastSamples</span>(input_selectContrast1, <span class="fu">colnames</span>(convertedData.out),readSampleInfo.out,</span>
<span id="cb190-351"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-351" aria-hidden="true" tabindex="-1"></a>                                        input_selectFactorsModel,input_selectModelComprions, </span>
<span id="cb190-352"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-352" aria-hidden="true" tabindex="-1"></a>                                        factorReferenceLevels.out,input_CountsDEGMethod ,</span>
<span id="cb190-353"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-353" aria-hidden="true" tabindex="-1"></a>                                        input_dataFileFormat )</span>
<span id="cb190-354"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-354" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span>  convertedData.out[<span class="fu">which</span>(<span class="fu">rownames</span>(convertedData.out) <span class="sc">%in%</span> genes), iz ]</span>
<span id="cb190-355"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-355" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_selectOrg <span class="sc">!=</span> <span class="st">&quot;NEW&quot;</span>) {</span>
<span id="cb190-356"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-356" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">match</span>( <span class="fu">rownames</span>(x), allGeneInfo.out[,<span class="dv">1</span>])</span>
<span id="cb190-357"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-357" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">sum</span>( <span class="fu">is.na</span>(allGeneInfo.out<span class="sc">$</span>symbol )) <span class="sc">!=</span> <span class="fu">dim</span>(allGeneInfo.out )[<span class="dv">1</span>] )  <span class="co"># symbol really exists? </span></span>
<span id="cb190-358"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-358" aria-hidden="true" tabindex="-1"></a>           <span class="fu">rownames</span>(x) <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">rownames</span>(x),<span class="st">&quot;:&quot;</span>, <span class="fu">as.character</span>( allGeneInfo.out<span class="sc">$</span>symbol)[ix])</span>
<span id="cb190-359"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-359" aria-hidden="true" tabindex="-1"></a>   }</span>
<span id="cb190-360"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-360" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-361"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-361" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( x )</span>
<span id="cb190-362"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-362" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb190-363"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-364"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-364" aria-hidden="true" tabindex="-1"></a>selectedPathwayHeatmap <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb190-365"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-366"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-366" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">=</span> selectedPathwayData.out</span>
<span id="cb190-367"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-367" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">dim</span>(x)[<span class="dv">1</span>]<span class="sc">&lt;=</span><span class="dv">2</span> <span class="sc">|</span> <span class="fu">dim</span>(x)[<span class="dv">2</span>]<span class="sc">&lt;=</span><span class="dv">2</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb190-368"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-368" aria-hidden="true" tabindex="-1"></a>    groups <span class="ot">=</span> <span class="fu">detectGroups</span>(<span class="fu">colnames</span>(x))</span>
<span id="cb190-369"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-370"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-370" aria-hidden="true" tabindex="-1"></a>    <span class="co"># this will cutoff very large values, which could skew the color </span></span>
<span id="cb190-371"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-371" aria-hidden="true" tabindex="-1"></a>    x<span class="ot">=</span><span class="fu">as.matrix</span>(x)<span class="sc">-</span><span class="fu">apply</span>(<span class="fu">as.matrix</span>(x),<span class="dv">1</span>,mean)</span>
<span id="cb190-372"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-372" aria-hidden="true" tabindex="-1"></a>    cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">+</span> <span class="dv">3</span><span class="sc">*</span><span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb190-373"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-373" aria-hidden="true" tabindex="-1"></a>    x[x<span class="sc">&gt;</span>cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb190-374"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-374" aria-hidden="true" tabindex="-1"></a>    cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">-</span> <span class="dv">3</span><span class="sc">*</span><span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb190-375"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-375" aria-hidden="true" tabindex="-1"></a>    x[x<span class="sc">&lt;</span> cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb190-376"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-376" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-377"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-377" aria-hidden="true" tabindex="-1"></a>    <span class="co"># sometimes, a gene can be all zero in selected samples.</span></span>
<span id="cb190-378"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-378" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> x[<span class="fu">which</span>(<span class="fu">apply</span>(x,<span class="dv">1</span>,sd)<span class="sc">&gt;</span><span class="dv">0</span>) ,]</span>
<span id="cb190-379"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-379" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-380"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-380" aria-hidden="true" tabindex="-1"></a>    lmat <span class="ot">=</span> <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">4</span>),<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">2</span>))</span>
<span id="cb190-381"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-381" aria-hidden="true" tabindex="-1"></a>    lwid <span class="ot">=</span> <span class="fu">c</span>(<span class="fl">1.5</span>,<span class="dv">6</span>)</span>
<span id="cb190-382"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-382" aria-hidden="true" tabindex="-1"></a>    lhei <span class="ot">=</span> <span class="fu">c</span>(.<span class="dv">5</span>,.<span class="dv">2</span>,<span class="dv">8</span>)</span>
<span id="cb190-383"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-383" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-384"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-384" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-385"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-385" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>(x)[<span class="dv">1</span>]<span class="sc">&gt;</span><span class="dv">200</span>) </span>
<span id="cb190-386"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-386" aria-hidden="true" tabindex="-1"></a>    <span class="fu">heatmap.2</span>(x, <span class="at">distfun =</span> dist2,<span class="at">hclustfun=</span>hclust2,</span>
<span id="cb190-387"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-387" aria-hidden="true" tabindex="-1"></a>     <span class="at">col=</span>heatColors[<span class="fu">as.integer</span>(input_heatColors1),], <span class="at">density.info=</span><span class="st">&quot;none&quot;</span>, <span class="at">trace=</span><span class="st">&quot;none&quot;</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">keysize=</span>.<span class="dv">5</span></span>
<span id="cb190-388"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-388" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">key=</span>T, <span class="at">symkey=</span>F</span>
<span id="cb190-389"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-389" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">dendrogram =</span> <span class="st">&quot;row&quot;</span></span>
<span id="cb190-390"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-390" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">ColSideColors=</span>mycolors[ groups]</span>
<span id="cb190-391"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-391" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">labRow=</span><span class="st">&quot;&quot;</span></span>
<span id="cb190-392"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-392" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">margins=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">15</span>)</span>
<span id="cb190-393"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-393" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">srtCol=</span><span class="dv">45</span></span>
<span id="cb190-394"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-394" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">lmat =</span> lmat, <span class="at">lwid =</span> lwid, <span class="at">lhei =</span> lhei</span>
<span id="cb190-395"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-395" aria-hidden="true" tabindex="-1"></a>    <span class="co">#,main =&quot;Title&quot;</span></span>
<span id="cb190-396"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-396" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb190-397"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-398"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-398" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>(x)[<span class="dv">1</span>]<span class="sc">&lt;=</span><span class="dv">200</span>) </span>
<span id="cb190-399"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-399" aria-hidden="true" tabindex="-1"></a>    <span class="fu">heatmap.2</span>(x, <span class="at">distfun =</span> dist2,<span class="at">hclustfun=</span>hclust2,</span>
<span id="cb190-400"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-400" aria-hidden="true" tabindex="-1"></a>     <span class="at">col=</span>heatColors[<span class="fu">as.integer</span>(input_heatColors1),], <span class="at">density.info=</span><span class="st">&quot;none&quot;</span>, <span class="at">trace=</span><span class="st">&quot;none&quot;</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">keysize=</span>.<span class="dv">5</span></span>
<span id="cb190-401"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-401" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">key=</span>T, <span class="at">symkey=</span>F, </span>
<span id="cb190-402"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-402" aria-hidden="true" tabindex="-1"></a>    <span class="at">dendrogram =</span> <span class="st">&quot;row&quot;</span>,</span>
<span id="cb190-403"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-403" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">labRow=</span><span class="fu">gsub</span>(<span class="st">&quot;.*:&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">rownames</span>(x))</span>
<span id="cb190-404"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-404" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">ColSideColors=</span>mycolors[ groups]</span>
<span id="cb190-405"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-405" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">margins=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">15</span>)</span>
<span id="cb190-406"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-406" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">cexRow=</span><span class="fl">1.2</span></span>
<span id="cb190-407"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-407" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">srtCol=</span><span class="dv">45</span></span>
<span id="cb190-408"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-408" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">lmat =</span> lmat, <span class="at">lwid =</span> lwid, <span class="at">lhei =</span> lhei</span>
<span id="cb190-409"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-409" aria-hidden="true" tabindex="-1"></a>    <span class="co">#,main =&quot;Title&quot;</span></span>
<span id="cb190-410"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-410" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb190-411"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-411" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb190-412"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-413"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-413" aria-hidden="true" tabindex="-1"></a>convertEnsembl2KEGG <span class="ot">&lt;-</span> <span class="cf">function</span> (query,Species) {  <span class="co"># not working</span></span>
<span id="cb190-414"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-414" aria-hidden="true" tabindex="-1"></a>    querySet <span class="ot">&lt;-</span> <span class="fu">cleanGeneSet</span>( <span class="fu">unlist</span>( <span class="fu">strsplit</span>( <span class="fu">toupper</span>(<span class="fu">names</span>( query)),<span class="st">&#39;</span><span class="sc">\t</span><span class="st">| |</span><span class="sc">\n</span><span class="st">|</span><span class="sc">\\</span><span class="st">,&#39;</span> )  ) )</span>
<span id="cb190-415"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-415" aria-hidden="true" tabindex="-1"></a>    speciesID <span class="ot">&lt;-</span> orgInfo<span class="sc">$</span>id[ <span class="fu">which</span>(orgInfo<span class="sc">$</span>ensembl_dataset <span class="sc">==</span> Species)]  <span class="co"># note uses species Identifying</span></span>
<span id="cb190-416"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-416" aria-hidden="true" tabindex="-1"></a>    <span class="co"># idType 6 for entrez gene ID</span></span>
<span id="cb190-417"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-417" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>( convert,</span>
<span id="cb190-418"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-418" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">paste</span>( <span class="st">&quot; select  id,ens,species from mapping where ens IN (&#39;&quot;</span>, <span class="fu">paste</span>(querySet,<span class="at">collapse=</span><span class="st">&quot;&#39;, &#39;&quot;</span>),</span>
<span id="cb190-419"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-419" aria-hidden="true" tabindex="-1"></a>                                <span class="st">&quot;&#39;) AND  idType =&#39;&quot;</span>,idType_KEGG,<span class="st">&quot;&#39;&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>) )   <span class="co"># slow</span></span>
<span id="cb190-420"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-420" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb190-421"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-421" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>  ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb190-422"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-422" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">subset</span>(result, species<span class="sc">==</span>speciesID, <span class="at">select =</span> <span class="sc">-</span>species)</span>
<span id="cb190-423"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-424"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-424" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span> <span class="fu">match</span>(result<span class="sc">$</span>ens,<span class="fu">names</span>(query)  )</span>
<span id="cb190-425"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-426"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-426" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">&lt;-</span> query[ix];  <span class="fu">names</span>(tem) <span class="ot">=</span> result<span class="sc">$</span>id</span>
<span id="cb190-427"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-427" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(tem)  </span>
<span id="cb190-428"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-428" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb190-429"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-430"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-430" aria-hidden="true" tabindex="-1"></a>keggPathwayID <span class="ot">&lt;-</span> <span class="cf">function</span> (pathwayDescription, Species, GO,selectOrg) {</span>
<span id="cb190-431"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-431" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span> <span class="fu">grep</span>(Species,gmtFiles)</span>
<span id="cb190-432"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-432" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-433"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-433" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> ) {<span class="fu">return</span>(<span class="cn">NULL</span>)}</span>
<span id="cb190-434"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-434" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-435"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-435" aria-hidden="true" tabindex="-1"></a>    <span class="co"># If selected species is not the default &quot;bestMatch&quot;, use that species directly</span></span>
<span id="cb190-436"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-436" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(selectOrg <span class="sc">!=</span> speciesChoice[[<span class="dv">1</span>]]) {  </span>
<span id="cb190-437"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-437" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">grep</span>(<span class="fu">findSpeciesById</span>(selectOrg)[<span class="dv">1</span>,<span class="dv">1</span>], gmtFiles )</span>
<span id="cb190-438"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-438" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> ) {<span class="fu">return</span>(<span class="cn">NULL</span> )}</span>
<span id="cb190-439"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-439" aria-hidden="true" tabindex="-1"></a>        totalGenes <span class="ot">&lt;-</span> orgInfo[<span class="fu">which</span>(orgInfo<span class="sc">$</span>id <span class="sc">==</span> <span class="fu">as.numeric</span>(selectOrg)),<span class="dv">7</span>]</span>
<span id="cb190-440"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-440" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb190-441"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-441" aria-hidden="true" tabindex="-1"></a>    pathway <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(sqlite,gmtFiles[ix],<span class="at">flags=</span>SQLITE_RO)</span>
<span id="cb190-442"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-442" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-443"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-443" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change Parkinson&#39;s disease to Parkinson\&#39;s disease    otherwise SQL </span></span>
<span id="cb190-444"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-444" aria-hidden="true" tabindex="-1"></a>    pathwayDescription <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\&#39;</span><span class="st">&quot;</span>,<span class="st">&quot;</span><span class="sc">\&#39;\&#39;</span><span class="st">&quot;</span>,pathwayDescription)</span>
<span id="cb190-445"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-445" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb190-446"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-446" aria-hidden="true" tabindex="-1"></a>    pathwayInfo <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>( pathway, <span class="fu">paste</span>( <span class="st">&quot; select * from pathwayInfo where description =  &#39;&quot;</span>, </span>
<span id="cb190-447"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-447" aria-hidden="true" tabindex="-1"></a>                            pathwayDescription,   <span class="st">&quot;&#39; AND name LIKE &#39;&quot;</span>,GO,<span class="st">&quot;%&#39;&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>) )     </span>
<span id="cb190-448"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-448" aria-hidden="true" tabindex="-1"></a>    pathwayInfo <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>( pathway, <span class="fu">paste</span>( <span class="st">&quot; select * from pathwayInfo where description =  &#39;&quot;</span>, </span>
<span id="cb190-449"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-449" aria-hidden="true" tabindex="-1"></a>                            pathwayDescription,   <span class="st">&quot;&#39;&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>) ) </span>
<span id="cb190-450"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-450" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb190-451"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-451" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dbDisconnect</span>(pathway);</span>
<span id="cb190-452"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-452" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">dim</span>(pathwayInfo)[<span class="dv">1</span>] <span class="sc">!=</span> <span class="dv">1</span> ) {<span class="fu">return</span>(<span class="cn">NULL</span>) }</span>
<span id="cb190-453"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-453" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;.*:&quot;</span>,<span class="st">&quot;&quot;</span>,pathwayInfo[<span class="dv">1</span>,<span class="dv">2</span>])  </span>
<span id="cb190-454"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-454" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>,<span class="st">&quot;&quot;</span>,tem) )</span>
<span id="cb190-455"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-455" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb190-456"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-457"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-457" aria-hidden="true" tabindex="-1"></a>KeggImage <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb190-458"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-459"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-459" aria-hidden="true" tabindex="-1"></a>   <span class="co"># First generate a blank image. Otherwise return(NULL) gives us errors.</span></span>
<span id="cb190-460"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-460" aria-hidden="true" tabindex="-1"></a>    outfile <span class="ot">&lt;-</span> <span class="fu">tempfile</span>(<span class="at">fileext=</span><span class="st">&#39;.png&#39;</span>)</span>
<span id="cb190-461"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-461" aria-hidden="true" tabindex="-1"></a>    <span class="fu">png</span>(outfile, <span class="at">width=</span><span class="dv">400</span>, <span class="at">height=</span><span class="dv">300</span>)</span>
<span id="cb190-462"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-463"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-463" aria-hidden="true" tabindex="-1"></a>    <span class="fu">frame</span>()</span>
<span id="cb190-464"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-464" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dev.off</span>()</span>
<span id="cb190-465"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-465" aria-hidden="true" tabindex="-1"></a>    blank <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">src =</span> outfile,</span>
<span id="cb190-466"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-466" aria-hidden="true" tabindex="-1"></a>         <span class="at">contentType =</span> <span class="st">&#39;image/png&#39;</span>,</span>
<span id="cb190-467"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-467" aria-hidden="true" tabindex="-1"></a>         <span class="at">width =</span> <span class="dv">400</span>,</span>
<span id="cb190-468"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-468" aria-hidden="true" tabindex="-1"></a>         <span class="at">height =</span> <span class="dv">300</span>,</span>
<span id="cb190-469"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-469" aria-hidden="true" tabindex="-1"></a>         <span class="at">alt =</span> <span class="st">&quot; &quot;</span>) </span>
<span id="cb190-470"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-471"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-471" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>( input_selectGO ) ) <span class="fu">return</span>(outfile)</span>
<span id="cb190-472"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-472" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_selectGO <span class="sc">!=</span> <span class="st">&quot;KEGG&quot;</span>) <span class="fu">return</span>(outfile)</span>
<span id="cb190-473"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-473" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(gagePathwayData.out ) ) <span class="fu">return</span>(outfile)</span>
<span id="cb190-474"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-474" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>( input_sigPathways))  <span class="fu">return</span> (outfile) </span>
<span id="cb190-475"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-475" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if( is.null(selectedPathwayData()) ) return(blank)</span></span>
<span id="cb190-476"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-477"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-477" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(pathview)</span>
<span id="cb190-478"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-478" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-479"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-479" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-480"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-480" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(input_selectContrast1 ) ) <span class="fu">return</span>(outfile)</span>
<span id="cb190-481"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-481" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-482"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-482" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_sigPathways <span class="sc">==</span> <span class="st">&quot;All&quot;</span>) <span class="fu">return</span> (outfile) </span>
<span id="cb190-483"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-483" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-484"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-484" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">length</span>(limma.out<span class="sc">$</span>topGenes) <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(outfile)</span>
<span id="cb190-485"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-486"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-486" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get fold change</span></span>
<span id="cb190-487"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-487" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb190-488"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-488" aria-hidden="true" tabindex="-1"></a>    { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb190-489"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-489" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb190-490"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-490" aria-hidden="true" tabindex="-1"></a>      top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb190-491"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-491" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast1, <span class="fu">names</span>(top))</span>
<span id="cb190-492"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-492" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (outfile)</span>
<span id="cb190-493"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-493" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb190-494"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-494" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb190-495"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-495" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (outfile)</span>
<span id="cb190-496"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-496" aria-hidden="true" tabindex="-1"></a>     <span class="co"># cat(&quot;here5&quot;)</span></span>
<span id="cb190-497"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-497" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb190-498"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-498" aria-hidden="true" tabindex="-1"></a>      Species <span class="ot">&lt;-</span> converted.out<span class="sc">$</span>species</span>
<span id="cb190-499"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-499" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb190-500"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-500" aria-hidden="true" tabindex="-1"></a>     fold <span class="ot">=</span> top1[,<span class="dv">1</span>]; <span class="fu">names</span>(fold) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top1)</span>
<span id="cb190-501"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-501" aria-hidden="true" tabindex="-1"></a>     fold <span class="ot">&lt;-</span> <span class="fu">convertEnsembl2Entrez</span>(fold,Species)</span>
<span id="cb190-502"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-502" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb190-503"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-503" aria-hidden="true" tabindex="-1"></a>     keggSpecies <span class="ot">&lt;-</span> <span class="fu">as.character</span>( keggSpeciesID[<span class="fu">which</span>(keggSpeciesID[,<span class="dv">1</span>] <span class="sc">==</span> Species),<span class="dv">3</span>] )</span>
<span id="cb190-504"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-504" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb190-505"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-505" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>(<span class="fu">nchar</span>( keggSpecies) <span class="sc">&lt;=</span><span class="dv">2</span> ) <span class="fu">return</span>(outfile) <span class="co"># not in KEGG</span></span>
<span id="cb190-506"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-507"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-507" aria-hidden="true" tabindex="-1"></a>     <span class="co"># kegg pathway id</span></span>
<span id="cb190-508"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-508" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-509"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-509" aria-hidden="true" tabindex="-1"></a>    pathID <span class="ot">=</span> <span class="fu">keggPathwayID</span>(input_sigPathways, Species, <span class="st">&quot;KEGG&quot;</span>,input_selectOrg)</span>
<span id="cb190-510"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-510" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-511"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-511" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(&quot;\nhere5  &quot;,keggSpecies, &quot; &quot;,Species,&quot; &quot;,input_sigPathways, &quot;pathID:&quot;,pathID,&quot;End&quot;, fold[1:5],names(fold)[1:5],&quot;\n&quot;)</span></span>
<span id="cb190-512"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-512" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(&quot;\npathway:&quot;,is.na(input_sigPathways))</span></span>
<span id="cb190-513"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-513" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(&quot;\n&quot;,fold[1:5],&quot;\n&quot;,keggSpecies,&quot;\n&quot;,pathID)</span></span>
<span id="cb190-514"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-514" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(pathID) ) <span class="fu">return</span>(blank) <span class="co"># kegg pathway id not found.</span></span>
<span id="cb190-515"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-515" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">nchar</span>(pathID)<span class="sc">&lt;</span><span class="dv">3</span> ) <span class="fu">return</span>(blank)</span>
<span id="cb190-516"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-516" aria-hidden="true" tabindex="-1"></a>    randomString <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;.*file&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">tempfile</span>()) </span>
<span id="cb190-517"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-517" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-518"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-518" aria-hidden="true" tabindex="-1"></a>    outfile <span class="ot">&lt;-</span> <span class="fu">paste</span>(pathID,<span class="st">&quot;.&quot;</span>,randomString,<span class="st">&quot;.png&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb190-519"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-519" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-520"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-520" aria-hidden="true" tabindex="-1"></a>        <span class="fu">try</span>( </span>
<span id="cb190-521"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-521" aria-hidden="true" tabindex="-1"></a>         pv.out <span class="ot">&lt;-</span> <span class="fu">pathview</span>(<span class="at">gene.data =</span> fold, <span class="at">pathway.id =</span> pathID, <span class="at">out.suffix =</span> randomString, <span class="at">species =</span> keggSpecies, <span class="at">kegg.native=</span><span class="cn">TRUE</span>)</span>
<span id="cb190-522"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-522" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb190-523"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-523" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb190-524"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-524" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-525"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-525" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( outfile )</span>
<span id="cb190-526"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-526" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb190-527"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-528"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-528" aria-hidden="true" tabindex="-1"></a><span class="co"># list of pathways with details</span></span>
<span id="cb190-529"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-529" aria-hidden="true" tabindex="-1"></a>pathwayListData  <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb190-530"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-530" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-531"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-531" aria-hidden="true" tabindex="-1"></a>    pathways <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb190-532"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-532" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_pathwayMethod <span class="sc">==</span> <span class="dv">1</span>)  </span>
<span id="cb190-533"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-533" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(gagePathwayData.out)) </span>
<span id="cb190-534"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-534" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(gagePathwayData.out)[<span class="dv">2</span>] <span class="sc">&gt;</span><span class="dv">1</span>) { </span>
<span id="cb190-535"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-535" aria-hidden="true" tabindex="-1"></a>                pathways <span class="ot">&lt;-</span> gagePathwayData.out</span>
<span id="cb190-536"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-536" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(pathways)[<span class="dv">2</span>] <span class="ot">=</span><span class="st">&quot;Pathways&quot;</span>;  </span>
<span id="cb190-537"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-537" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(pathways)[<span class="dv">4</span>] <span class="ot">=</span><span class="st">&quot;nGenes&quot;</span>; </span>
<span id="cb190-538"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-538" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb190-539"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-539" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_pathwayMethod <span class="sc">==</span> <span class="dv">3</span>) </span>
<span id="cb190-540"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-540" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(fgseaPathwayData.out)) </span>
<span id="cb190-541"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-541" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(fgseaPathwayData.out)[<span class="dv">2</span>] <span class="sc">&gt;</span><span class="dv">1</span>) {</span>
<span id="cb190-542"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-542" aria-hidden="true" tabindex="-1"></a>                pathways <span class="ot">&lt;-</span> fgseaPathwayData.out</span>
<span id="cb190-543"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-543" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(pathways)[<span class="dv">2</span>] <span class="ot">=</span><span class="st">&quot;Pathways&quot;</span>;  </span>
<span id="cb190-544"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-544" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(pathways)[<span class="dv">4</span>] <span class="ot">=</span><span class="st">&quot;nGenes&quot;</span>; </span>
<span id="cb190-545"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-545" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb190-546"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-546" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-547"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-547" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_pathwayMethod <span class="sc">==</span> <span class="dv">2</span>) </span>
<span id="cb190-548"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-548" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(PGSEAplot.data.out)) </span>
<span id="cb190-549"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-549" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(PGSEAplot.data.out)[<span class="dv">2</span>] <span class="sc">&gt;</span><span class="dv">1</span>) {</span>
<span id="cb190-550"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-550" aria-hidden="true" tabindex="-1"></a>                pathways <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>( PGSEAplot.data.out)</span>
<span id="cb190-551"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-551" aria-hidden="true" tabindex="-1"></a>                pathways<span class="sc">$</span>Pathways <span class="ot">=</span> <span class="fu">substr</span>(<span class="fu">rownames</span>(pathways),<span class="dv">10</span>, <span class="fu">nchar</span>( <span class="fu">rownames</span>(pathways)) )</span>
<span id="cb190-552"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-552" aria-hidden="true" tabindex="-1"></a>                pathways<span class="sc">$</span>adj.Pval <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; .*&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(pathways))</span>
<span id="cb190-553"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-553" aria-hidden="true" tabindex="-1"></a>                pathways<span class="sc">$</span>Direction <span class="ot">=</span><span class="st">&quot;Diff&quot;</span></span>
<span id="cb190-554"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-554" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb190-555"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-555" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb190-556"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-556" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_pathwayMethod <span class="sc">==</span> <span class="dv">4</span>) </span>
<span id="cb190-557"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-557" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(PGSEAplotAllSamples.data.out)) </span>
<span id="cb190-558"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-558" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(PGSEAplotAllSamples.data.out)[<span class="dv">2</span>] <span class="sc">&gt;</span><span class="dv">1</span>) {</span>
<span id="cb190-559"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-559" aria-hidden="true" tabindex="-1"></a>                pathways <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>( PGSEAplotAllSamples.data.out)</span>
<span id="cb190-560"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-560" aria-hidden="true" tabindex="-1"></a>                pathways<span class="sc">$</span>Pathways <span class="ot">=</span> <span class="fu">substr</span>(<span class="fu">rownames</span>(pathways),<span class="dv">10</span>, <span class="fu">nchar</span>( <span class="fu">rownames</span>(pathways)) )</span>
<span id="cb190-561"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-561" aria-hidden="true" tabindex="-1"></a>                pathways<span class="sc">$</span>adj.Pval <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; .*&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(pathways))</span>
<span id="cb190-562"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-562" aria-hidden="true" tabindex="-1"></a>                pathways<span class="sc">$</span>Direction <span class="ot">=</span><span class="st">&quot;Diff&quot;</span></span>
<span id="cb190-563"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-563" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb190-564"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-564" aria-hidden="true" tabindex="-1"></a>                }           </span>
<span id="cb190-565"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-565" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">is.null</span>( pathways) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb190-566"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-566" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-567"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-567" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if no gene set data, return pathway list</span></span>
<span id="cb190-568"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-568" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(GeneSets.out ) ) <span class="fu">return</span>(pathways) </span>
<span id="cb190-569"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-569" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-570"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-570" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-571"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-571" aria-hidden="true" tabindex="-1"></a>    pathways<span class="sc">$</span>adj.Pval <span class="ot">=</span> <span class="fu">as.numeric</span>(pathways<span class="sc">$</span>adj.Pval)</span>
<span id="cb190-572"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-572" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-573"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-573" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">nrow</span>(pathways)<span class="sc">&gt;</span><span class="dv">1</span>)</span>
<span id="cb190-574"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-574" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">nrow</span>(pathways) )</span>
<span id="cb190-575"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-575" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">nchar</span>(pathways<span class="sc">$</span>Direction[i]) <span class="sc">&lt;=</span><span class="dv">1</span>)</span>
<span id="cb190-576"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-576" aria-hidden="true" tabindex="-1"></a>            pathways<span class="sc">$</span>Direction[i] <span class="ot">=</span> pathways<span class="sc">$</span>Direction[i<span class="dv">-1</span>]</span>
<span id="cb190-577"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-577" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb190-578"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-578" aria-hidden="true" tabindex="-1"></a>    <span class="co"># gene symbol matching symbols </span></span>
<span id="cb190-579"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-579" aria-hidden="true" tabindex="-1"></a>    probeToGene <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb190-580"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-580" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_selectGO <span class="sc">!=</span> <span class="st">&quot;ID not recognized!&quot;</span> <span class="sc">&amp;</span> input_selectOrg <span class="sc">!=</span> <span class="st">&quot;NEW&quot;</span>)</span>
<span id="cb190-581"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-581" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">sum</span>(<span class="fu">is.na</span>( allGeneInfo.out<span class="sc">$</span>symbol ) )<span class="sc">/</span> <span class="fu">dim</span>( allGeneInfo.out )[<span class="dv">1</span>] <span class="sc">&lt;</span>.<span class="dv">5</span> ) { <span class="co"># if more than 50% genes has symbol</span></span>
<span id="cb190-582"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-582" aria-hidden="true" tabindex="-1"></a>        probeToGene <span class="ot">=</span> allGeneInfo.out[,<span class="fu">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="st">&quot;symbol&quot;</span>)]</span>
<span id="cb190-583"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-583" aria-hidden="true" tabindex="-1"></a>        probeToGene<span class="sc">$</span>symbol <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>,<span class="st">&quot;&quot;</span>,probeToGene<span class="sc">$</span>symbol)</span>
<span id="cb190-584"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-585"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-585" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">which</span>( <span class="fu">is.na</span>(probeToGene<span class="sc">$</span>symbol) <span class="sc">|</span></span>
<span id="cb190-586"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-586" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">nchar</span>(probeToGene<span class="sc">$</span>symbol)<span class="sc">&lt;</span><span class="dv">2</span> <span class="sc">|</span> </span>
<span id="cb190-587"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-587" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">toupper</span>(probeToGene<span class="sc">$</span>symbol)<span class="sc">==</span><span class="st">&quot;NA&quot;</span> <span class="sc">|</span>  </span>
<span id="cb190-588"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-588" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">toupper</span>(probeToGene<span class="sc">$</span>symbol)<span class="sc">==</span><span class="st">&quot;0&quot;</span>  )             </span>
<span id="cb190-589"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-589" aria-hidden="true" tabindex="-1"></a>        probeToGene[ix,<span class="dv">2</span>] <span class="ot">=</span> probeToGene[ix,<span class="dv">1</span>]  <span class="co"># use gene ID</span></span>
<span id="cb190-590"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-590" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-591"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-591" aria-hidden="true" tabindex="-1"></a>    }       </span>
<span id="cb190-592"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-592" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-593"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-593" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-594"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-594" aria-hidden="true" tabindex="-1"></a>    pathways<span class="sc">$</span>Genes <span class="ot">=</span><span class="st">&quot;&quot;</span></span>
<span id="cb190-595"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-595" aria-hidden="true" tabindex="-1"></a>    <span class="co"># looking up genes for each pathway</span></span>
<span id="cb190-596"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-596" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(pathways) ){ </span>
<span id="cb190-597"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-597" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">names</span>(GeneSets.out ) <span class="sc">==</span> pathways<span class="sc">$</span>Pathways[i]   ) <span class="co"># find the gene set</span></span>
<span id="cb190-598"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-598" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">length</span>(ix) <span class="sc">!=</span> <span class="dv">0</span> ) { </span>
<span id="cb190-599"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-599" aria-hidden="true" tabindex="-1"></a>            genes <span class="ot">&lt;-</span> GeneSets.out[[ix]] <span class="co"># retrieve genes</span></span>
<span id="cb190-600"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-600" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb190-601"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-601" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(probeToGene) ) { </span>
<span id="cb190-602"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-602" aria-hidden="true" tabindex="-1"></a>                iy <span class="ot">=</span> <span class="fu">match</span>(genes,probeToGene[,<span class="dv">1</span>])</span>
<span id="cb190-603"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-603" aria-hidden="true" tabindex="-1"></a>                genes <span class="ot">=</span> probeToGene[iy,<span class="dv">2</span>]</span>
<span id="cb190-604"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-604" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb190-605"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-605" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb190-606"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-606" aria-hidden="true" tabindex="-1"></a>            pathways<span class="sc">$</span>Genes[i] <span class="ot">=</span> <span class="fu">paste</span>(genes, <span class="at">collapse=</span><span class="st">&quot; &quot;</span>)</span>
<span id="cb190-607"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-607" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb190-608"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-608" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb190-609"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-609" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(pathways)</span>
<span id="cb190-610"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-611"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-611" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-612"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb190-612" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="genome-wide-view" class="section level2" number="6.10">
<h2><span class="header-section-number">6.10</span> Genome-wide view</h2>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-1" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb191-2"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Genome-wide view</span></span>
<span id="cb191-3"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-3" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb191-4"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-4" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb191-5"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-5" aria-hidden="true" tabindex="-1"></a>genomePlotly <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb191-6"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-6" aria-hidden="true" tabindex="-1"></a>        <span class="co"># default plot</span></span>
<span id="cb191-7"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-7" aria-hidden="true" tabindex="-1"></a>        fake <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">a=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="at">b=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)</span>
<span id="cb191-8"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-8" aria-hidden="true" tabindex="-1"></a>        p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(fake, <span class="fu">aes</span>(<span class="at">x =</span> a, <span class="at">y =</span> b)) <span class="sc">+</span></span>
<span id="cb191-9"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-9" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">geom_blank</span>() <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;No genes with position info.&quot;</span>) <span class="sc">+</span></span>
<span id="cb191-10"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-10" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">theme</span>(<span class="at">axis.title.x=</span><span class="fu">element_blank</span>(),<span class="at">axis.title.y=</span><span class="fu">element_blank</span>())</span>
<span id="cb191-11"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-11" aria-hidden="true" tabindex="-1"></a>                             </span>
<span id="cb191-12"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-12" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  {</span>
<span id="cb191-13"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-13" aria-hidden="true" tabindex="-1"></a>            top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb191-14"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-14" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb191-15"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-15" aria-hidden="true" tabindex="-1"></a>          top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb191-16"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-16" aria-hidden="true" tabindex="-1"></a>          ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast2, <span class="fu">names</span>(top))</span>
<span id="cb191-17"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-17" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (<span class="fu">ggplotly</span>(p))</span>
<span id="cb191-18"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-18" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb191-19"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-19" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb191-20"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-20" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (<span class="fu">ggplotly</span>(p))</span>
<span id="cb191-21"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-21" aria-hidden="true" tabindex="-1"></a>          <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb191-22"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-22" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb191-23"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-23" aria-hidden="true" tabindex="-1"></a>         <span class="co"># write.csv(merge(top1,allGeneInfo(), by.x=&quot;row.names&quot;,by.y=&quot;ensembl_gene_id&quot;  ),&quot;tem.csv&quot;  )</span></span>
<span id="cb191-24"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-24" aria-hidden="true" tabindex="-1"></a>         x <span class="ot">&lt;-</span> <span class="fu">merge</span>(top1,allGeneInfo.out, <span class="at">by.x=</span><span class="st">&quot;row.names&quot;</span>,<span class="at">by.y=</span><span class="st">&quot;ensembl_gene_id&quot;</span>  )</span>
<span id="cb191-25"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-26"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-26" aria-hidden="true" tabindex="-1"></a>         <span class="co"># if no chromosomes found. For example if user do not convert gene IDs.</span></span>
<span id="cb191-27"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-27" aria-hidden="true" tabindex="-1"></a>         <span class="cf">if</span>( <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="sc">&gt;</span><span class="dv">5</span>  ) { </span>
<span id="cb191-28"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-28" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb191-29"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-29" aria-hidden="true" tabindex="-1"></a>             x <span class="ot">&lt;-</span> x[<span class="fu">order</span>(x<span class="sc">$</span>chromosome_name,x<span class="sc">$</span>start_position),]</span>
<span id="cb191-30"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-30" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb191-31"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-31" aria-hidden="true" tabindex="-1"></a>             tem <span class="ot">=</span> <span class="fu">sort</span>( <span class="fu">table</span>( x<span class="sc">$</span>chromosome_name), <span class="at">decreasing=</span>T)</span>
<span id="cb191-32"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-33"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-33" aria-hidden="true" tabindex="-1"></a>             chromosomes <span class="ot">&lt;-</span> <span class="fu">names</span>( tem[tem <span class="sc">&gt;=</span> <span class="dv">5</span> ] )  <span class="co"># chromosomes with less than 100 genes are excluded</span></span>
<span id="cb191-34"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-34" aria-hidden="true" tabindex="-1"></a>             <span class="cf">if</span>(<span class="fu">length</span>(chromosomes) <span class="sc">&gt;</span> <span class="dv">50</span>) chromosomes <span class="ot">&lt;-</span> chromosomes[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]  <span class="co"># at most 50 chromosomes</span></span>
<span id="cb191-35"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-35" aria-hidden="true" tabindex="-1"></a>             chromosomes <span class="ot">&lt;-</span> chromosomes[ <span class="fu">nchar</span>(chromosomes)<span class="sc">&lt;=</span><span class="dv">12</span>] <span class="co"># chr. name less than 10 characters</span></span>
<span id="cb191-36"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-36" aria-hidden="true" tabindex="-1"></a>             chromosomes <span class="ot">=</span> chromosomes[<span class="fu">order</span>(<span class="fu">as.numeric</span>(chromosomes) ) ]</span>
<span id="cb191-37"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-37" aria-hidden="true" tabindex="-1"></a>             <span class="co"># chromosomes = chromosomes[!is.na(as.numeric(chromosomes) ) ]</span></span>
<span id="cb191-38"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-38" aria-hidden="true" tabindex="-1"></a>             chromosomesNumbers <span class="ot">=</span> <span class="fu">as.numeric</span>(chromosomes)</span>
<span id="cb191-39"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-40"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-40" aria-hidden="true" tabindex="-1"></a>             <span class="co"># convert chr.x to numbers      </span></span>
<span id="cb191-41"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-41" aria-hidden="true" tabindex="-1"></a>              j <span class="ot">=</span> <span class="fu">max</span>( chromosomesNumbers,<span class="at">na.rm=</span>T) </span>
<span id="cb191-42"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-42" aria-hidden="true" tabindex="-1"></a>              <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>( chromosomes)) {</span>
<span id="cb191-43"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-43" aria-hidden="true" tabindex="-1"></a>               <span class="cf">if</span> ( <span class="fu">is.na</span>(chromosomesNumbers[i]) ) </span>
<span id="cb191-44"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-44" aria-hidden="true" tabindex="-1"></a>               { chromosomesNumbers[i] <span class="ot">&lt;-</span> j<span class="sc">+</span><span class="dv">1</span>; j <span class="ot">&lt;-</span> j<span class="sc">+</span><span class="dv">1</span>; }</span>
<span id="cb191-45"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-45" aria-hidden="true" tabindex="-1"></a>             }</span>
<span id="cb191-46"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-46" aria-hidden="true" tabindex="-1"></a>              </span>
<span id="cb191-47"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-47" aria-hidden="true" tabindex="-1"></a>             x <span class="ot">&lt;-</span> x[<span class="fu">which</span>(x<span class="sc">$</span>chromosome_name <span class="sc">%in%</span> chromosomes   ),]</span>
<span id="cb191-48"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-48" aria-hidden="true" tabindex="-1"></a>             x <span class="ot">&lt;-</span> <span class="fu">droplevels</span>(x)</span>
<span id="cb191-49"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-49" aria-hidden="true" tabindex="-1"></a>             </span>
<span id="cb191-50"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-50" aria-hidden="true" tabindex="-1"></a>            <span class="co"># find the number coding for chromosome </span></span>
<span id="cb191-51"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-51" aria-hidden="true" tabindex="-1"></a>             getChrNumber <span class="ot">&lt;-</span> <span class="cf">function</span> (chrName){</span>
<span id="cb191-52"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-52" aria-hidden="true" tabindex="-1"></a>             <span class="fu">return</span>( chromosomesNumbers[ <span class="fu">which</span>( chromosomes <span class="sc">==</span> chrName)] )</span>
<span id="cb191-53"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-53" aria-hidden="true" tabindex="-1"></a>             }</span>
<span id="cb191-54"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-54" aria-hidden="true" tabindex="-1"></a>              x<span class="sc">$</span>chrNum <span class="ot">=</span> <span class="dv">1</span> <span class="co"># numeric coding</span></span>
<span id="cb191-55"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-55" aria-hidden="true" tabindex="-1"></a>              x<span class="sc">$</span>chrNum <span class="ot">&lt;-</span> <span class="fu">unlist</span>( <span class="fu">lapply</span>( x<span class="sc">$</span>chromosome_name, getChrNumber) )</span>
<span id="cb191-56"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-56" aria-hidden="true" tabindex="-1"></a>             </span>
<span id="cb191-57"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-57" aria-hidden="true" tabindex="-1"></a>             x<span class="sc">$</span>Row.names <span class="ot">&lt;-</span> <span class="fu">as.character</span>( x<span class="sc">$</span>Row.names)</span>
<span id="cb191-58"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-58" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb191-59"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-59" aria-hidden="true" tabindex="-1"></a>             <span class="co"># if symbol is missing use Ensembl id</span></span>
<span id="cb191-60"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-60" aria-hidden="true" tabindex="-1"></a>             x<span class="sc">$</span>symbol <span class="ot">=</span> <span class="fu">as.character</span>(x<span class="sc">$</span>symbol)   </span>
<span id="cb191-61"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-61" aria-hidden="true" tabindex="-1"></a>             ix <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">is.na</span>(x<span class="sc">$</span>symbol))</span>
<span id="cb191-62"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-62" aria-hidden="true" tabindex="-1"></a>             ix2 <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">nchar</span>(<span class="fu">as.character</span>(x<span class="sc">$</span>symbol))<span class="sc">&lt;=</span> <span class="dv">2</span> )</span>
<span id="cb191-63"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-63" aria-hidden="true" tabindex="-1"></a>             ix3 <span class="ot">=</span> <span class="fu">which</span>( <span class="fu">duplicated</span>(x<span class="sc">$</span>symbol))</span>
<span id="cb191-64"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-64" aria-hidden="true" tabindex="-1"></a>             ix <span class="ot">=</span> <span class="fu">unique</span>( <span class="fu">c</span>(ix,ix2,ix3))</span>
<span id="cb191-65"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-65" aria-hidden="true" tabindex="-1"></a>             x<span class="sc">$</span>symbol[ix] <span class="ot">&lt;-</span> x<span class="sc">$</span>Row.names[ix] </span>
<span id="cb191-66"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-66" aria-hidden="true" tabindex="-1"></a>                         </span>
<span id="cb191-67"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-67" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb191-68"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-68" aria-hidden="true" tabindex="-1"></a>             <span class="do">##################################</span></span>
<span id="cb191-69"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-69" aria-hidden="true" tabindex="-1"></a>             <span class="co"># plotting</span></span>
<span id="cb191-70"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-70" aria-hidden="true" tabindex="-1"></a>            <span class="co"># x = read.csv(&quot;tem_genome.csv&quot;)</span></span>
<span id="cb191-71"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-71" aria-hidden="true" tabindex="-1"></a>            x <span class="ot">=</span> x[<span class="sc">!</span><span class="fu">is.na</span>(x<span class="sc">$</span>chromosome_name),]</span>
<span id="cb191-72"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-72" aria-hidden="true" tabindex="-1"></a>            x <span class="ot">=</span> x[<span class="sc">!</span><span class="fu">is.na</span>(x<span class="sc">$</span>start_position),]    </span>
<span id="cb191-73"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-73" aria-hidden="true" tabindex="-1"></a>            <span class="co"># only keep significant genes</span></span>
<span id="cb191-74"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-74" aria-hidden="true" tabindex="-1"></a>            <span class="co">#x = x[which(x$FDR&lt;input_limmaPval),]</span></span>
<span id="cb191-75"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-75" aria-hidden="true" tabindex="-1"></a>            <span class="co"># x = x[which(abs(x$Fold) &gt; log2( input_limmaFC)),]</span></span>
<span id="cb191-76"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-76" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb191-77"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-77" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">which</span>( (x<span class="sc">$</span>FDR<span class="sc">&lt;</span> <span class="fu">as.numeric</span>(input_limmaPvalViz)) <span class="sc">&amp;</span> (<span class="fu">abs</span>(x<span class="sc">$</span>Fold) <span class="sc">&gt;</span> <span class="fu">as.numeric</span>(input_limmaFCViz) ) )</span>
<span id="cb191-78"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-78" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb191-79"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-79" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (<span class="fu">length</span>(ix) <span class="sc">&gt;</span> <span class="dv">5</span>) { </span>
<span id="cb191-80"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-81"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-81" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">=</span> x[ix,]      </span>
<span id="cb191-82"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-82" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb191-83"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-83" aria-hidden="true" tabindex="-1"></a>                x<span class="sc">$</span>start_position <span class="ot">=</span> x<span class="sc">$</span>start_position<span class="sc">/</span><span class="dv">1000000</span> <span class="co"># Mbp</span></span>
<span id="cb191-84"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-84" aria-hidden="true" tabindex="-1"></a>                chrD <span class="ot">=</span> <span class="dv">20</span> <span class="co"># distance between chrs.</span></span>
<span id="cb191-85"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-85" aria-hidden="true" tabindex="-1"></a>                foldCutoff <span class="ot">=</span> <span class="dv">3</span>   <span class="co"># max log2 fold </span></span>
<span id="cb191-86"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-86" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb191-87"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-87" aria-hidden="true" tabindex="-1"></a>                x<span class="sc">$</span>Fold <span class="ot">=</span> x<span class="sc">$</span>Fold <span class="sc">/</span> <span class="fu">sd</span>(x<span class="sc">$</span>Fold)  <span class="co"># standardize fold change</span></span>
<span id="cb191-88"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-88" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb191-89"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-89" aria-hidden="true" tabindex="-1"></a>                x<span class="sc">$</span>Fold[<span class="fu">which</span>(x<span class="sc">$</span>Fold <span class="sc">&gt;</span> foldCutoff )] <span class="ot">=</span> foldCutoff   <span class="co"># log2fold within -5 to 5</span></span>
<span id="cb191-90"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-90" aria-hidden="true" tabindex="-1"></a>                x<span class="sc">$</span>Fold[<span class="fu">which</span>(x<span class="sc">$</span>Fold <span class="sc">&lt;</span>   <span class="sc">-</span><span class="dv">1</span><span class="sc">*</span>foldCutoff )] <span class="ot">=</span> <span class="sc">-</span><span class="dv">1</span><span class="sc">*</span>foldCutoff </span>
<span id="cb191-91"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-91" aria-hidden="true" tabindex="-1"></a>                x<span class="sc">$</span>Fold <span class="ot">=</span> <span class="dv">4</span><span class="sc">*</span> x<span class="sc">$</span>Fold</span>
<span id="cb191-92"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-92" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb191-93"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-94"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-94" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb191-95"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-95" aria-hidden="true" tabindex="-1"></a>                x<span class="sc">$</span>y <span class="ot">=</span> x<span class="sc">$</span>chrNum<span class="sc">*</span>chrD <span class="sc">+</span> x<span class="sc">$</span>Fold</span>
<span id="cb191-96"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-96" aria-hidden="true" tabindex="-1"></a>                chrLengthTable <span class="ot">=</span> <span class="fu">aggregate</span>(start_position<span class="sc">~</span>chrNum, <span class="at">data=</span>x,max )</span>
<span id="cb191-97"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-97" aria-hidden="true" tabindex="-1"></a>                chrTotal <span class="ot">=</span> <span class="fu">dim</span>(chrLengthTable)[<span class="dv">1</span>]   </span>
<span id="cb191-98"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-98" aria-hidden="true" tabindex="-1"></a>                x<span class="sc">$</span>R <span class="ot">=</span> <span class="fu">as.factor</span>(<span class="fu">sign</span>(x<span class="sc">$</span>Fold))</span>
<span id="cb191-99"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-99" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb191-100"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-100" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(x)[ <span class="fu">which</span>(<span class="fu">colnames</span>(x) <span class="sc">==</span> <span class="st">&quot;start_position&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;x&quot;</span></span>
<span id="cb191-101"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-102"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-102" aria-hidden="true" tabindex="-1"></a>                p<span class="ot">=</span> <span class="fu">ggplot</span>(x, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">colour =</span> R, <span class="at">text =</span> symbol ) ) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">20</span>, <span class="at">size =</span> .<span class="dv">2</span>)</span>
<span id="cb191-103"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-103" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb191-104"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-104" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#label y with chr names</span></span>
<span id="cb191-105"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-105" aria-hidden="true" tabindex="-1"></a>                p <span class="ot">&lt;-</span> p <span class="sc">+</span>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> <span class="fu">paste</span>(<span class="st">&quot;chr&quot;</span>,chromosomes[chrLengthTable<span class="sc">$</span>chrNum],<span class="at">sep=</span><span class="st">&quot;&quot;</span>), <span class="at">breaks =</span> chrD<span class="sc">*</span> (<span class="dv">1</span><span class="sc">:</span>chrTotal), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, </span>
<span id="cb191-106"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-106" aria-hidden="true" tabindex="-1"></a>                    chrD<span class="sc">*</span>chrTotal <span class="sc">+</span> <span class="dv">5</span>) )</span>
<span id="cb191-107"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-107" aria-hidden="true" tabindex="-1"></a>                <span class="co"># draw horizontal lines for each chr.</span></span>
<span id="cb191-108"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-108" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(chrLengthTable)[<span class="dv">1</span>] )</span>
<span id="cb191-109"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-109" aria-hidden="true" tabindex="-1"></a>                    p <span class="ot">=</span> p<span class="sc">+</span> <span class="fu">annotate</span>( <span class="st">&quot;segment&quot;</span>,<span class="at">x =</span> <span class="dv">0</span>, <span class="at">xend =</span> chrLengthTable<span class="sc">$</span>start_position[i],</span>
<span id="cb191-110"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-110" aria-hidden="true" tabindex="-1"></a>                        <span class="at">y =</span> chrLengthTable<span class="sc">$</span>chrNum[i]<span class="sc">*</span>chrD, <span class="at">yend =</span> chrLengthTable<span class="sc">$</span>chrNum[i]<span class="sc">*</span>chrD)</span>
<span id="cb191-111"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-111" aria-hidden="true" tabindex="-1"></a>                <span class="co"># change legend     http://ggplot2.tidyverse.org/reference/scale_manual.html</span></span>
<span id="cb191-112"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-112" aria-hidden="true" tabindex="-1"></a>                p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">scale_colour_manual</span>(<span class="at">name=</span><span class="st">&quot;&quot;</span>,   <span class="co"># customize legend text</span></span>
<span id="cb191-113"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-113" aria-hidden="true" tabindex="-1"></a>                    <span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb191-114"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-114" aria-hidden="true" tabindex="-1"></a>                    <span class="at">breaks=</span><span class="fu">c</span>(<span class="st">&quot;1&quot;</span>,<span class="st">&quot;-1&quot;</span>),</span>
<span id="cb191-115"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-115" aria-hidden="true" tabindex="-1"></a>                    <span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;Up&quot;</span>, <span class="st">&quot;Dn&quot;</span>))   </span>
<span id="cb191-116"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-116" aria-hidden="true" tabindex="-1"></a>                p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">&quot;Position on chrs. (Mbp)&quot;</span>) <span class="sc">+</span>    <span class="fu">theme</span>(<span class="at">axis.title.y=</span><span class="fu">element_blank</span>())                     </span>
<span id="cb191-117"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-117" aria-hidden="true" tabindex="-1"></a>                p<span class="ot">=</span> p <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb191-118"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-118" aria-hidden="true" tabindex="-1"></a>                <span class="co"># p &lt;- ggplot(mtcars, aes(x=hp, y=mpg)) + geom_point(shape=20, size=17)</span></span>
<span id="cb191-119"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-119" aria-hidden="true" tabindex="-1"></a>               <span class="co"># p=p+ geom_smooth(method = &quot;lm&quot;,se=FALSE)</span></span>
<span id="cb191-120"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-120" aria-hidden="true" tabindex="-1"></a>                <span class="co">#p+ geom_line(aes(y=rollmean(y,7, na.pad=TRUE)))</span></span>
<span id="cb191-121"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-121" aria-hidden="true" tabindex="-1"></a>              </span>
<span id="cb191-122"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-122" aria-hidden="true" tabindex="-1"></a>               <span class="co"># Customize hover text https://cran.r-project.org/web/packages/plotly/plotly.pdf</span></span>
<span id="cb191-123"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-123" aria-hidden="true" tabindex="-1"></a>               <span class="co"># style(ggplotly(p),hoverinfo=&quot;text&quot;)  # not working</span></span>
<span id="cb191-124"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-124" aria-hidden="true" tabindex="-1"></a>              }  <span class="co"># if no genes else</span></span>
<span id="cb191-125"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-125" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb191-126"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-126" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb191-127"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-127" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ggplotly</span>(p)</span>
<span id="cb191-128"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-128" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb191-129"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-130"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-130" aria-hidden="true" tabindex="-1"></a><span class="co"># results from PREDA</span></span>
<span id="cb191-131"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-131" aria-hidden="true" tabindex="-1"></a>genomePlotData <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb191-132"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-132" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(input_selectContrast2 ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb191-133"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-133" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb191-134"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-134" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">length</span>(limma.out<span class="sc">$</span>topGenes) <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb191-135"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-135" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb191-136"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-136" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb191-137"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-137" aria-hidden="true" tabindex="-1"></a>    { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb191-138"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-138" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb191-139"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-139" aria-hidden="true" tabindex="-1"></a>      top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb191-140"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-140" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast2, <span class="fu">names</span>(top))</span>
<span id="cb191-141"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-141" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb191-142"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-142" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb191-143"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-143" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb191-144"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-144" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb191-145"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-145" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb191-146"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-146" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb191-147"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-147" aria-hidden="true" tabindex="-1"></a>     <span class="co"># write.csv(merge(top1,allGeneInfo(), by.x=&quot;row.names&quot;,by.y=&quot;ensembl_gene_id&quot;  ),&quot;tem.csv&quot;  )</span></span>
<span id="cb191-148"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-148" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> <span class="fu">merge</span>(top1,allGeneInfo.out, <span class="at">by.x=</span><span class="st">&quot;row.names&quot;</span>,<span class="at">by.y=</span><span class="st">&quot;ensembl_gene_id&quot;</span>  )</span>
<span id="cb191-149"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-149" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb191-150"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-150" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> x[<span class="fu">order</span>(x<span class="sc">$</span>chromosome_name,x<span class="sc">$</span>start_position),]</span>
<span id="cb191-151"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-151" aria-hidden="true" tabindex="-1"></a>     tem <span class="ot">=</span> <span class="fu">sort</span>( <span class="fu">table</span>( x<span class="sc">$</span>chromosome_name), <span class="at">decreasing=</span>T) </span>
<span id="cb191-152"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-152" aria-hidden="true" tabindex="-1"></a>     chromosomes <span class="ot">&lt;-</span> <span class="fu">names</span>( tem[tem <span class="sc">&gt;</span> <span class="dv">100</span> ] )  <span class="co"># chromosomes with less than 100 genes are excluded</span></span>
<span id="cb191-153"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-153" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>(<span class="fu">length</span>(chromosomes) <span class="sc">&gt;</span> <span class="dv">50</span>) chromosomes <span class="ot">&lt;-</span> chromosomes[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]  <span class="co"># at most 50 chromosomes</span></span>
<span id="cb191-154"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-155"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-155" aria-hidden="true" tabindex="-1"></a>     chromosomes <span class="ot">=</span> chromosomes[<span class="fu">order</span>(<span class="fu">as.numeric</span>(chromosomes) ) ]</span>
<span id="cb191-156"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-156" aria-hidden="true" tabindex="-1"></a>     <span class="co"># chromosomes = chromosomes[!is.na(as.numeric(chromosomes) ) ]</span></span>
<span id="cb191-157"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-157" aria-hidden="true" tabindex="-1"></a>     chromosomesNumbers <span class="ot">=</span> <span class="fu">as.numeric</span>(chromosomes)</span>
<span id="cb191-158"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-158" aria-hidden="true" tabindex="-1"></a>     <span class="co"># convert chr.x to numbers</span></span>
<span id="cb191-159"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-159" aria-hidden="true" tabindex="-1"></a>      j <span class="ot">=</span> <span class="fu">max</span>( chromosomesNumbers,<span class="at">na.rm=</span>T) </span>
<span id="cb191-160"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-160" aria-hidden="true" tabindex="-1"></a>      <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>( chromosomes)) {</span>
<span id="cb191-161"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-161" aria-hidden="true" tabindex="-1"></a>       <span class="cf">if</span> ( <span class="fu">is.na</span>(chromosomesNumbers[i]) ) </span>
<span id="cb191-162"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-162" aria-hidden="true" tabindex="-1"></a>       { chromosomesNumbers[i] <span class="ot">&lt;-</span> j<span class="sc">+</span><span class="dv">1</span>; j <span class="ot">&lt;-</span> j<span class="sc">+</span><span class="dv">1</span>; }</span>
<span id="cb191-163"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-163" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb191-164"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-164" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb191-165"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-165" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> x[<span class="fu">which</span>(x<span class="sc">$</span>chromosome_name <span class="sc">%in%</span> chromosomes   ),]</span>
<span id="cb191-166"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-166" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> <span class="fu">droplevels</span>(x)</span>
<span id="cb191-167"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-167" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb191-168"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-168" aria-hidden="true" tabindex="-1"></a>    <span class="co"># find the number coding for chromosome </span></span>
<span id="cb191-169"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-169" aria-hidden="true" tabindex="-1"></a>     getChrNumber <span class="ot">&lt;-</span> <span class="cf">function</span> (chrName){</span>
<span id="cb191-170"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-170" aria-hidden="true" tabindex="-1"></a>     <span class="fu">return</span>( chromosomesNumbers[ <span class="fu">which</span>( chromosomes <span class="sc">==</span> chrName)] )</span>
<span id="cb191-171"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-171" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb191-172"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-172" aria-hidden="true" tabindex="-1"></a>      x<span class="sc">$</span>chrNum <span class="ot">=</span> <span class="dv">1</span> <span class="co"># numeric coding</span></span>
<span id="cb191-173"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-173" aria-hidden="true" tabindex="-1"></a>      x<span class="sc">$</span>chrNum <span class="ot">&lt;-</span> <span class="fu">unlist</span>( <span class="fu">lapply</span>( x<span class="sc">$</span>chromosome_name, getChrNumber) )</span>
<span id="cb191-174"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-174" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb191-175"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-175" aria-hidden="true" tabindex="-1"></a>     x<span class="sc">$</span>Row.names <span class="ot">&lt;-</span> <span class="fu">as.character</span>( x<span class="sc">$</span>Row.names)</span>
<span id="cb191-176"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-176" aria-hidden="true" tabindex="-1"></a>     fold <span class="ot">=</span> x<span class="sc">$</span>Fold; <span class="fu">names</span>(fold) <span class="ot">=</span> x<span class="sc">$</span>Row.names</span>
<span id="cb191-177"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-177" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb191-178"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-178" aria-hidden="true" tabindex="-1"></a>    <span class="co"># write.csv(x,&quot;tem.csv&quot;)</span></span>
<span id="cb191-179"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-179" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> x[<span class="sc">!</span><span class="fu">duplicated</span>(x<span class="sc">$</span>Row.names),]</span>
<span id="cb191-180"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-180" aria-hidden="true" tabindex="-1"></a>     <span class="co">#rownames(x) = x$Row.names</span></span>
<span id="cb191-181"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-181" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb191-182"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-183"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-183" aria-hidden="true" tabindex="-1"></a>    GEanalysisResults <span class="ot">&lt;-</span> genomePlotDataPre.out; </span>
<span id="cb191-184"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-184" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb191-185"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-185" aria-hidden="true" tabindex="-1"></a>    genomic_regions_UP<span class="ot">&lt;-</span><span class="fu">PREDAResults2GenomicRegions</span>(GEanalysisResults, <span class="at">qval.threshold=</span>input_RegionsPvalCutoff, <span class="at">smoothStatistic.tail=</span><span class="st">&quot;upper&quot;</span>, <span class="at">smoothStatistic.threshold=</span> input_StatisticCutoff)</span>
<span id="cb191-186"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-186" aria-hidden="true" tabindex="-1"></a>    genomic_regions_DOWN<span class="ot">&lt;-</span><span class="fu">PREDAResults2GenomicRegions</span>(GEanalysisResults, <span class="at">qval.threshold=</span>input_RegionsPvalCutoff, <span class="at">smoothStatistic.tail=</span><span class="st">&quot;lower&quot;</span>, <span class="at">smoothStatistic.threshold=</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">*</span>input_StatisticCutoff)</span>
<span id="cb191-187"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-188"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-188" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">is.null</span>(genomic_regions_UP<span class="sc">$</span>Test) <span class="sc">&amp;&amp;</span>  <span class="fu">is.null</span>(genomic_regions_UP<span class="sc">$</span>Test)  ) <span class="fu">return</span>(<span class="sc">-</span><span class="dv">1</span>) <span class="co"># no significant regions</span></span>
<span id="cb191-189"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-189" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb191-190"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-190" aria-hidden="true" tabindex="-1"></a>    regions <span class="ot">&lt;-</span> <span class="dv">0</span> </span>
<span id="cb191-191"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-191" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>(genomic_regions_UP<span class="sc">$</span>Test)) { </span>
<span id="cb191-192"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-192" aria-hidden="true" tabindex="-1"></a>    dataframe_UPregions<span class="ot">&lt;-</span><span class="fu">GenomicRegions2dataframe</span>(genomic_regions_UP[[<span class="dv">1</span>]])</span>
<span id="cb191-193"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-193" aria-hidden="true" tabindex="-1"></a>    dataframe_UPregions<span class="sc">$</span>Regulation <span class="ot">&lt;-</span> <span class="st">&quot;Up&quot;</span></span>
<span id="cb191-194"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-194" aria-hidden="true" tabindex="-1"></a>    regions <span class="ot">=</span> dataframe_UPregions</span>
<span id="cb191-195"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-195" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb191-196"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-197"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-197" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>(genomic_regions_DOWN<span class="sc">$</span>Test) ){ </span>
<span id="cb191-198"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-198" aria-hidden="true" tabindex="-1"></a>    dataframe_DOWNregions<span class="ot">&lt;-</span><span class="fu">GenomicRegions2dataframe</span>(genomic_regions_DOWN[[<span class="dv">1</span>]])</span>
<span id="cb191-199"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-199" aria-hidden="true" tabindex="-1"></a>    dataframe_DOWNregions<span class="sc">$</span>Regulation <span class="ot">&lt;-</span> <span class="st">&quot;Down&quot;</span></span>
<span id="cb191-200"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-200" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> ( <span class="fu">class</span>(regions) <span class="sc">!=</span> <span class="st">&quot;data.frame&quot;</span> ) regions <span class="ot">&lt;-</span> dataframe_DOWNregions <span class="cf">else</span> <span class="co"># if UP regions is NULL</span></span>
<span id="cb191-201"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-201" aria-hidden="true" tabindex="-1"></a>     regions <span class="ot">=</span> <span class="fu">rbind</span>(regions,dataframe_DOWNregions  )</span>
<span id="cb191-202"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-202" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb191-203"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-204"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-204" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">class</span>(regions) <span class="sc">!=</span> <span class="st">&quot;data.frame&quot;</span>) <span class="fu">return</span>(<span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb191-205"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-206"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-206" aria-hidden="true" tabindex="-1"></a>    Regions <span class="ot">&lt;-</span> regions[,<span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)] </span>
<span id="cb191-207"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-207" aria-hidden="true" tabindex="-1"></a>    Regions <span class="ot">&lt;-</span> Regions[<span class="fu">which</span>(Regions<span class="sc">$</span>end <span class="sc">!=</span> Regions<span class="sc">$</span>start),]</span>
<span id="cb191-208"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-208" aria-hidden="true" tabindex="-1"></a>    Regions<span class="sc">$</span>size <span class="ot">=</span> <span class="fu">round</span>((Regions<span class="sc">$</span>end <span class="sc">-</span> Regions<span class="sc">$</span>start )<span class="sc">/</span><span class="dv">1000000</span>,<span class="dv">3</span>)</span>
<span id="cb191-209"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-209" aria-hidden="true" tabindex="-1"></a>    Regions<span class="sc">$</span>chr <span class="ot">&lt;-</span> chromosomes[ Regions<span class="sc">$</span>chr ] <span class="co"># convert from chr. number to names</span></span>
<span id="cb191-210"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-211"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-211" aria-hidden="true" tabindex="-1"></a>    <span class="co"># find the gene indices in the up or down regulated regions</span></span>
<span id="cb191-212"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-212" aria-hidden="true" tabindex="-1"></a>    regulatedGenes <span class="ot">&lt;-</span> <span class="cf">function</span> (i) {</span>
<span id="cb191-213"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-213" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span>  <span class="fu">which</span>( Regions<span class="sc">$</span>chr <span class="sc">==</span> x<span class="sc">$</span>chromosome_name[i] <span class="sc">&amp;</span></span>
<span id="cb191-214"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-214" aria-hidden="true" tabindex="-1"></a>         Regions<span class="sc">$</span>start <span class="sc">&lt;</span> x<span class="sc">$</span>start_position[i] <span class="co">#&amp;(Regions$end &gt; x$start_position[i] + x$genomeSpan[i])</span></span>
<span id="cb191-215"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-215" aria-hidden="true" tabindex="-1"></a>         <span class="sc">&amp;</span>(Regions<span class="sc">$</span>end <span class="sc">&gt;</span> x<span class="sc">$</span>start_position[i] )) <span class="co"># if the start position is within the region</span></span>
<span id="cb191-216"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-216" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">length</span>(ix) <span class="sc">&gt;</span><span class="dv">1</span>  ) <span class="fu">return</span>(<span class="cn">NA</span>) <span class="cf">else</span> <span class="fu">return</span>( ix )</span>
<span id="cb191-217"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-217" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb191-218"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-219"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-219" aria-hidden="true" tabindex="-1"></a>    regionID <span class="ot">=</span> <span class="fu">unlist</span>( <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(x)[<span class="dv">1</span>], regulatedGenes) )</span>
<span id="cb191-220"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-220" aria-hidden="true" tabindex="-1"></a>    x1 <span class="ot">=</span> x[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(regionID)),]</span>
<span id="cb191-221"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-221" aria-hidden="true" tabindex="-1"></a>    regionID <span class="ot">=</span> regionID[<span class="sc">!</span><span class="fu">is.na</span>(regionID)]</span>
<span id="cb191-222"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-222" aria-hidden="true" tabindex="-1"></a>    x1 <span class="ot">=</span> <span class="fu">cbind</span>(regionID,Regions[regionID, ], x1[,<span class="fu">c</span>(<span class="st">&#39;symbol&#39;</span>, <span class="st">&#39;Row.names&#39;</span>, <span class="st">&#39;Fold&#39;</span>,<span class="st">&#39;FDR&#39;</span>, <span class="st">&#39;band&#39;</span>,<span class="st">&#39;start_position&#39;</span>)])</span>
<span id="cb191-223"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-223" aria-hidden="true" tabindex="-1"></a>    x1 <span class="ot">=</span> x1[<span class="fu">order</span>(x1<span class="sc">$</span>regionID,x1<span class="sc">$</span>start_position ), ]    </span>
<span id="cb191-224"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-224" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(x1)[<span class="dv">8</span>] <span class="ot">=</span> <span class="st">&quot;Ensembl&quot;</span>; <span class="fu">colnames</span>(x1)[<span class="dv">4</span>] <span class="ot">=</span> <span class="st">&quot;Region.Start&quot;</span>;<span class="fu">colnames</span>(x1)[<span class="dv">5</span>] <span class="ot">=</span> <span class="st">&quot;Region.End&quot;</span>; <span class="fu">colnames</span>(x1)[<span class="dv">12</span>] <span class="ot">=</span> <span class="st">&quot;Gene.Start&quot;</span></span>
<span id="cb191-225"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-226"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-226" aria-hidden="true" tabindex="-1"></a>    <span class="co"># number of genes</span></span>
<span id="cb191-227"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-227" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> <span class="fu">table</span>(x1<span class="sc">$</span>regionID)</span>
<span id="cb191-228"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-228" aria-hidden="true" tabindex="-1"></a>    Regions<span class="sc">$</span>Ngenes <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb191-229"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-229" aria-hidden="true" tabindex="-1"></a>    Regions<span class="sc">$</span>Ngenes[<span class="fu">as.integer</span>(<span class="fu">names</span>(tem) ) ] <span class="ot">&lt;-</span> tem</span>
<span id="cb191-230"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-231"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-231" aria-hidden="true" tabindex="-1"></a>    <span class="co"># cytoband per region</span></span>
<span id="cb191-232"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-232" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> <span class="fu">unique</span>(x1[,<span class="fu">c</span>(<span class="st">&#39;regionID&#39;</span>,<span class="st">&#39;band&#39;</span>)])</span>
<span id="cb191-233"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-233" aria-hidden="true" tabindex="-1"></a>    tem<span class="sc">$</span>band <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">..*&quot;</span>,<span class="st">&quot;&quot;</span>,tem<span class="sc">$</span>band)</span>
<span id="cb191-234"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-234" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> <span class="fu">unique</span>(tem)</span>
<span id="cb191-235"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-235" aria-hidden="true" tabindex="-1"></a>    Regions<span class="sc">$</span>band <span class="ot">=</span><span class="st">&quot;&quot;</span></span>
<span id="cb191-236"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-236" aria-hidden="true" tabindex="-1"></a>    Regions<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(Regions)[<span class="dv">1</span>]</span>
<span id="cb191-237"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-237" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> ( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(Regions)[<span class="dv">1</span>] )</span>
<span id="cb191-238"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-238" aria-hidden="true" tabindex="-1"></a>      Regions<span class="sc">$</span>band[i] <span class="ot">&lt;-</span> <span class="fu">paste</span>( tem[ <span class="fu">which</span>( tem[,<span class="dv">1</span>] <span class="sc">==</span> Regions<span class="sc">$</span>ID[i]),<span class="dv">2</span>],  <span class="at">collapse=</span><span class="st">&quot;;&quot;</span> )</span>
<span id="cb191-239"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-240"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-240" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Genes</span></span>
<span id="cb191-241"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-241" aria-hidden="true" tabindex="-1"></a>    Regions<span class="sc">$</span>Genes <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb191-242"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-242" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> ( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(Regions)[<span class="dv">1</span>] )</span>
<span id="cb191-243"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-243" aria-hidden="true" tabindex="-1"></a>      Regions<span class="sc">$</span>Genes[i] <span class="ot">&lt;-</span> <span class="fu">paste</span>( x1<span class="sc">$</span>symbol[ <span class="fu">which</span>( x1[,<span class="dv">1</span>] <span class="sc">==</span> Regions<span class="sc">$</span>ID[i])],  <span class="at">collapse=</span><span class="st">&quot; &quot;</span> )</span>
<span id="cb191-244"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-244" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">mainResult =</span> GEanalysisResults, <span class="at">Regions =</span> Regions, <span class="at">Genes =</span> x1,<span class="at">legend.x=</span><span class="fu">max</span>(x<span class="sc">$</span>start_position)<span class="sc">*</span>.<span class="dv">6</span>, <span class="at">legend.y=</span><span class="fu">max</span>(chromosomesNumbers)<span class="sc">-</span><span class="dv">3</span>  ) )</span>
<span id="cb191-245"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-245" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb191-246"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-247"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-247" aria-hidden="true" tabindex="-1"></a>genomePlot <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb191-248"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-248" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(PREDA,<span class="at">verbose=</span><span class="cn">FALSE</span>)  <span class="co"># showing expression on genome</span></span>
<span id="cb191-249"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-249" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(PREDAsampledata,<span class="at">verbose=</span><span class="cn">FALSE</span>) </span>
<span id="cb191-250"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-250" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(hgu133plus2.db,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb191-251"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-251" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb191-252"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-252" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>( <span class="fu">class</span>(genomePlotData.out) <span class="sc">!=</span> <span class="st">&quot;list&quot;</span> ) { <span class="fu">plot.new</span>(); <span class="fu">text</span>(<span class="fl">0.2</span>,<span class="dv">1</span>, <span class="st">&quot;No significant regions found!&quot;</span>)} <span class="cf">else</span> {</span>
<span id="cb191-253"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-253" aria-hidden="true" tabindex="-1"></a>   GEanalysisResults <span class="ot">&lt;-</span> genomePlotData.out<span class="sc">$</span>mainResult;</span>
<span id="cb191-254"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-254" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>( <span class="fu">is.null</span>( GEanalysisResults ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)    </span>
<span id="cb191-255"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-255" aria-hidden="true" tabindex="-1"></a>    genomic_regions_UP<span class="ot">&lt;-</span><span class="fu">PREDAResults2GenomicRegions</span>(GEanalysisResults, <span class="at">qval.threshold=</span>input_RegionsPvalCutoff, <span class="at">smoothStatistic.tail=</span><span class="st">&quot;upper&quot;</span>, <span class="at">smoothStatistic.threshold=</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">*</span>input_StatisticCutoff)</span>
<span id="cb191-256"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-256" aria-hidden="true" tabindex="-1"></a>    genomic_regions_DOWN<span class="ot">&lt;-</span><span class="fu">PREDAResults2GenomicRegions</span>(GEanalysisResults, <span class="at">qval.threshold=</span>input_RegionsPvalCutoff, <span class="at">smoothStatistic.tail=</span><span class="st">&quot;lower&quot;</span>, <span class="at">smoothStatistic.threshold=</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">*</span>input_StatisticCutoff)</span>
<span id="cb191-257"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-257" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb191-258"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-258" aria-hidden="true" tabindex="-1"></a>    checkplot<span class="ot">&lt;-</span><span class="fu">genomePlot</span>(GEanalysisResults, <span class="at">genomicRegions=</span><span class="fu">c</span>(genomic_regions_UP, genomic_regions_DOWN), <span class="at">grouping=</span><span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>), <span class="at">scale.positions=</span><span class="st">&quot;Mb&quot;</span>, <span class="at">region.colors=</span><span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>))</span>
<span id="cb191-259"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-259" aria-hidden="true" tabindex="-1"></a>    <span class="fu">legend</span>(<span class="at">x=</span>genomePlotData.out<span class="sc">$</span>legend.x, <span class="at">y=</span>genomePlotData.out<span class="sc">$</span>legend.y , <span class="at">legend=</span><span class="fu">c</span>(<span class="st">&quot;UP&quot;</span>, <span class="st">&quot;DOWN&quot;</span>), <span class="at">fill=</span><span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>))</span>
<span id="cb191-260"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-260" aria-hidden="true" tabindex="-1"></a>    } <span class="co">#if  else</span></span>
<span id="cb191-261"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-261" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb191-262"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-263"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-263" aria-hidden="true" tabindex="-1"></a><span class="co"># pre-calculating PREDA, so that changing FDR cutoffs does not trigger entire calculation</span></span>
<span id="cb191-264"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-264" aria-hidden="true" tabindex="-1"></a>genomePlotDataPre <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb191-265"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-265" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(PREDA,<span class="at">verbose=</span><span class="cn">FALSE</span>)  <span class="co"># showing expression on genome</span></span>
<span id="cb191-266"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-266" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(PREDAsampledata,<span class="at">verbose=</span><span class="cn">FALSE</span>) </span>
<span id="cb191-267"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-267" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(hgu133plus2.db,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb191-268"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-268" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(input_selectContrast2 ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb191-269"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-269" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb191-270"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-270" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">length</span>(limma.out<span class="sc">$</span>topGenes) <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb191-271"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-271" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb191-272"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-272" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb191-273"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-273" aria-hidden="true" tabindex="-1"></a>    { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb191-274"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-274" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb191-275"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-275" aria-hidden="true" tabindex="-1"></a>      top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb191-276"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-276" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast2, <span class="fu">names</span>(top))</span>
<span id="cb191-277"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-277" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb191-278"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-278" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb191-279"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-279" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb191-280"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-280" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb191-281"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-281" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb191-282"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-282" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb191-283"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-283" aria-hidden="true" tabindex="-1"></a>     <span class="co"># write.csv(merge(top1,allGeneInfo(), by.x=&quot;row.names&quot;,by.y=&quot;ensembl_gene_id&quot;  ),&quot;tem.csv&quot;  )</span></span>
<span id="cb191-284"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-284" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> <span class="fu">merge</span>(top1,allGeneInfo.out, <span class="at">by.x=</span><span class="st">&quot;row.names&quot;</span>,<span class="at">by.y=</span><span class="st">&quot;ensembl_gene_id&quot;</span>  )</span>
<span id="cb191-285"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-285" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb191-286"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-286" aria-hidden="true" tabindex="-1"></a>     <span class="co">#hist(top1[,1])</span></span>
<span id="cb191-287"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-287" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb191-288"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-288" aria-hidden="true" tabindex="-1"></a>     <span class="do">########## PREDA</span></span>
<span id="cb191-289"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-289" aria-hidden="true" tabindex="-1"></a>    infofile <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;sampledata&quot;</span>, <span class="st">&quot;GeneExpression&quot;</span>, <span class="st">&quot;sampleinfoGE_PREDA.txt&quot;</span>, <span class="at">package =</span> <span class="st">&quot;PREDAsampledata&quot;</span>)</span>
<span id="cb191-290"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-290" aria-hidden="true" tabindex="-1"></a>    sampleinfo<span class="ot">&lt;-</span><span class="fu">read.table</span>(infofile, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb191-291"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-291" aria-hidden="true" tabindex="-1"></a>    <span class="fu">head</span>(sampleinfo)</span>
<span id="cb191-292"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-293"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-293" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data</span>(ExpressionSetRCC)</span>
<span id="cb191-294"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-294" aria-hidden="true" tabindex="-1"></a>    GEstatisticsForPREDA<span class="ot">&lt;-</span><span class="fu">statisticsForPREDAfromEset</span>(ExpressionSetRCC, <span class="at">statisticType=</span><span class="st">&quot;tstatistic&quot;</span>, <span class="at">referenceGroupLabel=</span><span class="st">&quot;normal&quot;</span>, <span class="at">classVector=</span>sampleinfo[,<span class="st">&quot;Class&quot;</span>])</span>
<span id="cb191-295"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-295" aria-hidden="true" tabindex="-1"></a>    <span class="fu">analysesNames</span>(GEstatisticsForPREDA)</span>
<span id="cb191-296"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-296" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb191-297"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-297" aria-hidden="true" tabindex="-1"></a>    GEGenomicAnnotations<span class="ot">&lt;-</span><span class="fu">eset2GenomicAnnotations</span>(ExpressionSetRCC, <span class="at">retain.chrs=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>) <span class="co"># needs hgu133plus2.db package</span></span>
<span id="cb191-298"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-298" aria-hidden="true" tabindex="-1"></a>    GEGenomicAnnotationsForPREDA<span class="ot">&lt;-</span><span class="fu">GenomicAnnotations2GenomicAnnotationsForPREDA</span>(GEGenomicAnnotations, <span class="at">reference_position_type=</span><span class="st">&quot;median&quot;</span>)</span>
<span id="cb191-299"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-299" aria-hidden="true" tabindex="-1"></a>    GEDataForPREDA<span class="ot">&lt;-</span><span class="fu">MergeStatisticAnnotations2DataForPREDA</span>(GEstatisticsForPREDA, GEGenomicAnnotationsForPREDA, <span class="at">sortAndCleanNA=</span><span class="cn">TRUE</span>)</span>
<span id="cb191-300"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-301"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-301" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x = read.csv(&quot;PREDA_test.csv&quot;)</span></span>
<span id="cb191-302"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-303"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-303" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x &lt;- x[,-1] # remove the row index, not needed in Shiny</span></span>
<span id="cb191-304"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-305"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-305" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> x[<span class="fu">order</span>(x<span class="sc">$</span>chromosome_name,x<span class="sc">$</span>start_position),]</span>
<span id="cb191-306"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-306" aria-hidden="true" tabindex="-1"></a>     tem <span class="ot">=</span> <span class="fu">sort</span>( <span class="fu">table</span>( x<span class="sc">$</span>chromosome_name), <span class="at">decreasing=</span>T) </span>
<span id="cb191-307"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-307" aria-hidden="true" tabindex="-1"></a>     chromosomes <span class="ot">&lt;-</span> <span class="fu">names</span>( tem[tem <span class="sc">&gt;</span> <span class="dv">100</span> ] )  <span class="co"># chromosomes with less than 100 genes are excluded</span></span>
<span id="cb191-308"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-308" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>(<span class="fu">length</span>(chromosomes) <span class="sc">&gt;</span> <span class="dv">50</span>) chromosomes <span class="ot">&lt;-</span> chromosomes[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]  <span class="co"># at most 50 chromosomes</span></span>
<span id="cb191-309"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-310"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-310" aria-hidden="true" tabindex="-1"></a>     chromosomes <span class="ot">=</span> chromosomes[<span class="fu">order</span>(<span class="fu">as.numeric</span>(chromosomes) ) ]</span>
<span id="cb191-311"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-311" aria-hidden="true" tabindex="-1"></a>     <span class="co"># chromosomes = chromosomes[!is.na(as.numeric(chromosomes) ) ]</span></span>
<span id="cb191-312"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-312" aria-hidden="true" tabindex="-1"></a>     chromosomesNumbers <span class="ot">=</span> <span class="fu">as.numeric</span>(chromosomes)</span>
<span id="cb191-313"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-313" aria-hidden="true" tabindex="-1"></a>     <span class="co"># convert chr.x to numbers</span></span>
<span id="cb191-314"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-314" aria-hidden="true" tabindex="-1"></a>      j <span class="ot">=</span> <span class="fu">max</span>( chromosomesNumbers,<span class="at">na.rm=</span>T) </span>
<span id="cb191-315"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-315" aria-hidden="true" tabindex="-1"></a>      <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>( chromosomes)) {</span>
<span id="cb191-316"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-316" aria-hidden="true" tabindex="-1"></a>       <span class="cf">if</span> ( <span class="fu">is.na</span>(chromosomesNumbers[i]) ) </span>
<span id="cb191-317"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-317" aria-hidden="true" tabindex="-1"></a>       { chromosomesNumbers[i] <span class="ot">&lt;-</span> j<span class="sc">+</span><span class="dv">1</span>; j <span class="ot">&lt;-</span> j<span class="sc">+</span><span class="dv">1</span>; }</span>
<span id="cb191-318"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-318" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb191-319"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-319" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb191-320"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-320" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> x[<span class="fu">which</span>(x<span class="sc">$</span>chromosome_name <span class="sc">%in%</span> chromosomes   ),]</span>
<span id="cb191-321"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-321" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> <span class="fu">droplevels</span>(x)</span>
<span id="cb191-322"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-322" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb191-323"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-323" aria-hidden="true" tabindex="-1"></a>    <span class="co"># find the number coding for chromosome </span></span>
<span id="cb191-324"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-324" aria-hidden="true" tabindex="-1"></a>     getChrNumber <span class="ot">&lt;-</span> <span class="cf">function</span> (chrName){</span>
<span id="cb191-325"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-325" aria-hidden="true" tabindex="-1"></a>     <span class="fu">return</span>( chromosomesNumbers[ <span class="fu">which</span>( chromosomes <span class="sc">==</span> chrName)] )</span>
<span id="cb191-326"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-326" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb191-327"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-327" aria-hidden="true" tabindex="-1"></a>      x<span class="sc">$</span>chrNum <span class="ot">=</span> <span class="dv">1</span> <span class="co"># numeric coding</span></span>
<span id="cb191-328"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-328" aria-hidden="true" tabindex="-1"></a>      x<span class="sc">$</span>chrNum <span class="ot">&lt;-</span> <span class="fu">unlist</span>( <span class="fu">lapply</span>( x<span class="sc">$</span>chromosome_name, getChrNumber) )</span>
<span id="cb191-329"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-329" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb191-330"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-330" aria-hidden="true" tabindex="-1"></a>     x<span class="sc">$</span>Row.names <span class="ot">&lt;-</span> <span class="fu">as.character</span>( x<span class="sc">$</span>Row.names)</span>
<span id="cb191-331"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-331" aria-hidden="true" tabindex="-1"></a>     fold <span class="ot">=</span> x<span class="sc">$</span>Fold; <span class="fu">names</span>(fold) <span class="ot">=</span> x<span class="sc">$</span>Row.names</span>
<span id="cb191-332"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-332" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb191-333"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-333" aria-hidden="true" tabindex="-1"></a>    <span class="co"># write.csv(x,&quot;tem.csv&quot;)</span></span>
<span id="cb191-334"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-334" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> x[<span class="sc">!</span><span class="fu">duplicated</span>(x<span class="sc">$</span>Row.names),]</span>
<span id="cb191-335"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-335" aria-hidden="true" tabindex="-1"></a>     <span class="co">#rownames(x) = x$Row.names</span></span>
<span id="cb191-336"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-336" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb191-337"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-337" aria-hidden="true" tabindex="-1"></a>     myData <span class="ot">&lt;-</span> GEDataForPREDA</span>
<span id="cb191-338"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-338" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb191-339"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-339" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>position <span class="ot">&lt;-</span> <span class="fu">as.integer</span>( x<span class="sc">$</span>start_position<span class="sc">+</span> x<span class="sc">$</span>genomeSpan<span class="sc">/</span><span class="dv">2</span> )</span>
<span id="cb191-340"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-340" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>ids <span class="ot">&lt;-</span> <span class="fu">as.character</span>( x<span class="sc">$</span>Row.names)</span>
<span id="cb191-341"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-341" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>chr <span class="ot">&lt;-</span> <span class="fu">as.integer</span>( x<span class="sc">$</span>chrNum )</span>
<span id="cb191-342"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-342" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>start <span class="ot">&lt;-</span> x<span class="sc">$</span>start_position</span>
<span id="cb191-343"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-343" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>end <span class="ot">&lt;-</span> x<span class="sc">$</span>start_position <span class="sc">+</span> x<span class="sc">$</span>genomeSpan</span>
<span id="cb191-344"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-344" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>strand <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">1</span>,<span class="fu">dim</span>(x)[<span class="dv">1</span>])</span>
<span id="cb191-345"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-345" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>chromosomesLabels <span class="ot">&lt;-</span> chromosomes</span>
<span id="cb191-346"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-346" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>chromosomesNumbers <span class="ot">&lt;-</span> <span class="fu">as.integer</span>( chromosomesNumbers )</span>
<span id="cb191-347"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-347" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>statistic <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>( x[,<span class="dv">2</span>,<span class="at">drop=</span><span class="cn">FALSE</span>] )</span>
<span id="cb191-348"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-348" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>analysesNames <span class="ot">&lt;-</span> <span class="st">&quot;Test&quot;</span></span>
<span id="cb191-349"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-349" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>testedTail <span class="ot">&lt;-</span> <span class="st">&quot;both&quot;</span></span>
<span id="cb191-350"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-350" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>optionalAnnotations <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>( x[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>,<span class="at">drop=</span><span class="cn">FALSE</span>] )</span>
<span id="cb191-351"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-351" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>optionalAnnotationsHeaders <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;a&quot;</span>,<span class="st">&quot;b&quot;</span>)</span>
<span id="cb191-352"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-352" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb191-353"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-353" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set.seed</span>(<span class="dv">2</span>)</span>
<span id="cb191-354"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-354" aria-hidden="true" tabindex="-1"></a>    GEanalysisResults<span class="ot">&lt;-</span><span class="fu">PREDA_main</span>(myData,<span class="at">nperms =</span> PREDA_Permutations)</span>
<span id="cb191-355"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-356"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-356" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( GEanalysisResults )</span>
<span id="cb191-357"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb191-357" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="biclustering" class="section level2" number="6.11">
<h2><span class="header-section-number">6.11</span> Biclustering</h2>
<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-1" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb192-2"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Biclustering</span></span>
<span id="cb192-3"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-3" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb192-4"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-4" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb192-5"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-5" aria-hidden="true" tabindex="-1"></a>biclustering <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb192-6"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-6" aria-hidden="true" tabindex="-1"></a>            <span class="fu">library</span>(biclust,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb192-7"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-8"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-8" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(input_biclustMethod <span class="sc">==</span> <span class="st">&quot;BCQU()&quot;</span> )</span>
<span id="cb192-9"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-9" aria-hidden="true" tabindex="-1"></a>                <span class="fu">library</span>(QUBIC,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># have trouble installing on Linux</span></span>
<span id="cb192-10"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-10" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(input_biclustMethod <span class="sc">==</span> <span class="st">&quot;BCUnibic()&quot;</span> )</span>
<span id="cb192-11"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-11" aria-hidden="true" tabindex="-1"></a>                <span class="fu">library</span>(runibic,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># </span></span>
<span id="cb192-12"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-12" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb192-13"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-13" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb192-14"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-14" aria-hidden="true" tabindex="-1"></a>            x <span class="ot">&lt;-</span> convertedData.out</span>
<span id="cb192-15"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-15" aria-hidden="true" tabindex="-1"></a>            n<span class="ot">=</span>input_nGenesBiclust   </span>
<span id="cb192-16"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-16" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(n<span class="sc">&gt;</span><span class="fu">dim</span>(x)[<span class="dv">1</span>]) n <span class="ot">=</span> <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="co"># max as data</span></span>
<span id="cb192-17"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-17" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(n<span class="sc">&lt;</span><span class="dv">10</span>) n <span class="ot">=</span> <span class="dv">10</span> </span>
<span id="cb192-18"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-18" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(n<span class="sc">&gt;</span> <span class="dv">2000</span> ) n <span class="ot">=</span> <span class="dv">2000</span>           </span>
<span id="cb192-19"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-19" aria-hidden="true" tabindex="-1"></a>            x<span class="ot">=</span><span class="fu">as.matrix</span>(x[<span class="dv">1</span><span class="sc">:</span>n,])<span class="sc">-</span><span class="fu">apply</span>(x[<span class="dv">1</span><span class="sc">:</span>n,],<span class="dv">1</span>,mean)</span>
<span id="cb192-20"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-20" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb192-21"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-21" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( input_biclustMethod <span class="sc">==</span> <span class="st">&quot;BCXmotifs()&quot;</span>  )</span>
<span id="cb192-22"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-22" aria-hidden="true" tabindex="-1"></a>                x<span class="ot">&lt;-</span><span class="fu">discretize</span>(x)</span>
<span id="cb192-23"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-24"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-24" aria-hidden="true" tabindex="-1"></a>            <span class="co">#res &lt;- biclust::biclust(as.matrix( x), method = BCQU()) </span></span>
<span id="cb192-25"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-25" aria-hidden="true" tabindex="-1"></a>            runR <span class="ot">=</span> <span class="fu">paste</span>( <span class="st">&quot;res &lt;- biclust::biclust(as.matrix( x), method =&quot;</span>, input_biclustMethod ,<span class="st">&quot;)&quot;</span> )</span>
<span id="cb192-26"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-26" aria-hidden="true" tabindex="-1"></a>            <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> runR ) )</span>
<span id="cb192-27"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-27" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb192-28"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-29"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-29" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>(<span class="fu">list</span>( <span class="at">x=</span>x, <span class="at">res=</span>res)  )    </span>
<span id="cb192-30"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb192-31"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-32"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-33"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-33" aria-hidden="true" tabindex="-1"></a>biclustHeatmap <span class="ot">&lt;-</span> <span class="cf">function</span> ( ){</span>
<span id="cb192-34"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-34" aria-hidden="true" tabindex="-1"></a>            res <span class="ot">=</span> <span class="fu">biclustering</span>()<span class="sc">$</span>res</span>
<span id="cb192-35"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-35" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( res<span class="sc">@</span>Number <span class="sc">==</span> <span class="dv">0</span> ) { <span class="fu">plot.new</span>(); <span class="fu">text</span>(<span class="fl">0.5</span>,<span class="fl">0.5</span>, <span class="st">&quot;No cluster found!&quot;</span>)} <span class="cf">else</span> {</span>
<span id="cb192-36"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-36" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb192-37"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-37" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">=</span> biclust<span class="sc">::</span><span class="fu">bicluster</span>(biclustering.out<span class="sc">$</span>x, res, <span class="fu">as.numeric</span>( input_selectBicluster)  )[[<span class="dv">1</span>]]</span>
<span id="cb192-38"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-39"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-39" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">4</span>, <span class="fl">1.4</span>, <span class="fl">0.2</span>))</span>
<span id="cb192-40"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-41"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-41" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>( <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="sc">&lt;=</span><span class="dv">30</span> )</span>
<span id="cb192-42"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-42" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">heatmap.2</span>(x,  <span class="at">Rowv =</span>T,<span class="at">Colv=</span>F, <span class="at">dendrogram =</span><span class="st">&quot;none&quot;</span>,</span>
<span id="cb192-43"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-43" aria-hidden="true" tabindex="-1"></a>                    <span class="at">col=</span>heatColors[<span class="fu">as.integer</span>(input_heatColors1),], <span class="at">density.info=</span><span class="st">&quot;none&quot;</span>, <span class="at">trace=</span><span class="st">&quot;none&quot;</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">keysize=</span>.<span class="dv">2</span></span>
<span id="cb192-44"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-44" aria-hidden="true" tabindex="-1"></a>                    ,<span class="at">key=</span>F, <span class="co">#labRow = T,</span></span>
<span id="cb192-45"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-45" aria-hidden="true" tabindex="-1"></a>                    ,<span class="at">margins =</span> <span class="fu">c</span>(<span class="dv">8</span>, <span class="dv">24</span>)</span>
<span id="cb192-46"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-46" aria-hidden="true" tabindex="-1"></a>                    ,<span class="at">cexRow=</span><span class="dv">1</span></span>
<span id="cb192-47"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-47" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#,srtCol=45</span></span>
<span id="cb192-48"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-48" aria-hidden="true" tabindex="-1"></a>                    ,<span class="at">cexCol=</span><span class="fl">1.</span>  <span class="co"># size of font for sample names</span></span>
<span id="cb192-49"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-49" aria-hidden="true" tabindex="-1"></a>                    ) <span class="cf">else</span>      </span>
<span id="cb192-50"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-50" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">heatmap.2</span>(x,  <span class="at">Rowv =</span>T,<span class="at">Colv=</span>F, <span class="at">dendrogram =</span><span class="st">&quot;none&quot;</span>,</span>
<span id="cb192-51"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-51" aria-hidden="true" tabindex="-1"></a>                    <span class="at">col=</span>heatColors[<span class="fu">as.integer</span>(input_heatColors1),], <span class="at">density.info=</span><span class="st">&quot;none&quot;</span>, <span class="at">trace=</span><span class="st">&quot;none&quot;</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">keysize=</span>.<span class="dv">2</span></span>
<span id="cb192-52"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-52" aria-hidden="true" tabindex="-1"></a>                    ,<span class="at">key=</span>F, <span class="at">labRow =</span> F,</span>
<span id="cb192-53"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-53" aria-hidden="true" tabindex="-1"></a>                    ,<span class="at">margins =</span> <span class="fu">c</span>(<span class="dv">8</span>, <span class="dv">4</span>)</span>
<span id="cb192-54"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-54" aria-hidden="true" tabindex="-1"></a>                    ,<span class="at">cexRow=</span><span class="dv">1</span></span>
<span id="cb192-55"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-55" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#,srtCol=45</span></span>
<span id="cb192-56"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-56" aria-hidden="true" tabindex="-1"></a>                    ,<span class="at">cexCol=</span><span class="fl">1.</span>  <span class="co"># size of font for sample names</span></span>
<span id="cb192-57"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-58"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-58" aria-hidden="true" tabindex="-1"></a>                    )               </span>
<span id="cb192-59"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-59" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb192-60"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-60" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb192-61"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-62"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-63"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-63" aria-hidden="true" tabindex="-1"></a>geneListBclustGO <span class="ot">&lt;-</span> <span class="cf">function</span>( ){        </span>
<span id="cb192-64"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-64" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb192-65"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-65" aria-hidden="true" tabindex="-1"></a>            res <span class="ot">=</span> biclustering.out<span class="sc">$</span>res</span>
<span id="cb192-66"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-66" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( res<span class="sc">@</span>Number <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(<span class="fu">as.data.frame</span>(<span class="st">&quot;No clusters found!&quot;</span>) ) </span>
<span id="cb192-67"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-67" aria-hidden="true" tabindex="-1"></a>            x <span class="ot">=</span> biclust<span class="sc">::</span><span class="fu">bicluster</span>(biclustering.out<span class="sc">$</span>x, res, <span class="fu">as.numeric</span>( input_selectBicluster)  )[[<span class="dv">1</span>]]</span>
<span id="cb192-68"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-68" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb192-69"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-69" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">dim</span>(x)[<span class="dv">1</span>]  <span class="sc">&lt;=</span> minGenesEnrichment) <span class="fu">return</span>(NoSig) </span>
<span id="cb192-70"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-70" aria-hidden="true" tabindex="-1"></a>            query <span class="ot">=</span> <span class="fu">rownames</span>(x)</span>
<span id="cb192-71"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-71" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb192-72"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-73"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-73" aria-hidden="true" tabindex="-1"></a>            result <span class="ot">&lt;-</span> <span class="fu">findOverlapGMT</span>( query, GeneSets.out,<span class="dv">1</span>) </span>
<span id="cb192-74"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-74" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb192-75"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-75" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb192-76"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-76" aria-hidden="true" tabindex="-1"></a>            result<span class="sc">$</span>Genes <span class="ot">=</span> <span class="st">&quot;Up regulated&quot;</span></span>
<span id="cb192-77"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-77" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb192-78"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-79"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-79" aria-hidden="true" tabindex="-1"></a>            results1 <span class="ot">=</span> result</span>
<span id="cb192-80"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-80" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> ( <span class="fu">is.null</span>( results1) ) <span class="fu">return</span> (NoSig)</span>
<span id="cb192-81"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-81" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">&lt;</span><span class="dv">5</span> ) <span class="fu">return</span>(NoSig)  <span class="co"># Returns a data frame: &quot;No significant results found!&quot;</span></span>
<span id="cb192-82"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-83"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-83" aria-hidden="true" tabindex="-1"></a>            results1<span class="ot">=</span> results1[,<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">4</span>)]</span>
<span id="cb192-84"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-84" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(results1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;List&quot;</span>,<span class="st">&quot;FDR&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;GO terms or pathways&quot;</span>)</span>
<span id="cb192-85"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-85" aria-hidden="true" tabindex="-1"></a>            minFDR <span class="ot">=</span> <span class="fl">0.01</span></span>
<span id="cb192-86"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-86" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">min</span>(results1<span class="sc">$</span>FDR) <span class="sc">&gt;</span> minFDR ) results1 <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No signficant enrichment found.&quot;</span>) <span class="cf">else</span></span>
<span id="cb192-87"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-87" aria-hidden="true" tabindex="-1"></a>            results1 <span class="ot">=</span> results1[<span class="fu">which</span>(results1<span class="sc">$</span>FDR <span class="sc">&lt;</span> minFDR),]</span>
<span id="cb192-88"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-89"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-89" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb192-90"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-90" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">!=</span> <span class="dv">4</span>) <span class="fu">return</span>(NoSig)</span>
<span id="cb192-91"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-91" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(results1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Direction&quot;</span>,<span class="st">&quot;adj.Pval&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;Pathways&quot;</span>)</span>
<span id="cb192-92"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-92" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb192-93"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-93" aria-hidden="true" tabindex="-1"></a>            results1<span class="sc">$</span>adj.Pval <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;%-2.1e&quot;</span>,<span class="fu">as.numeric</span>(results1<span class="sc">$</span>adj.Pval) )</span>
<span id="cb192-94"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-94" aria-hidden="true" tabindex="-1"></a>            results1[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(results1[,<span class="dv">1</span>])</span>
<span id="cb192-95"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-95" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">&lt;-</span> results1[,<span class="dv">1</span>]</span>
<span id="cb192-96"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-97"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-97" aria-hidden="true" tabindex="-1"></a>            results1[ <span class="fu">duplicated</span> (results1[,<span class="dv">1</span>] ),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb192-98"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-98" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb192-99"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-99" aria-hidden="true" tabindex="-1"></a>            results1[,<span class="sc">-</span><span class="dv">1</span>]           </span>
<span id="cb192-100"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-101"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb192-101" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="co-expression-network" class="section level2" number="6.12">
<h2><span class="header-section-number">6.12</span> Co-expression network</h2>
<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-1" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb193-2"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Co-expression network</span></span>
<span id="cb193-3"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-3" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb193-4"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-5"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-5" aria-hidden="true" tabindex="-1"></a>wgcna <span class="ot">&lt;-</span> <span class="cf">function</span> ( ){</span>
<span id="cb193-6"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-6" aria-hidden="true" tabindex="-1"></a>            <span class="co">#http://pklab.med.harvard.edu/scw2014/WGCNA.html</span></span>
<span id="cb193-7"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-7" aria-hidden="true" tabindex="-1"></a>            <span class="fu">library</span>(WGCNA)</span>
<span id="cb193-8"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-9"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-9" aria-hidden="true" tabindex="-1"></a>            x <span class="ot">&lt;-</span> convertedData.out</span>
<span id="cb193-10"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-10" aria-hidden="true" tabindex="-1"></a>            n<span class="ot">=</span>input_nGenesNetwork   </span>
<span id="cb193-11"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-11" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(n<span class="sc">&gt;</span><span class="fu">dim</span>(x)[<span class="dv">1</span>]) n <span class="ot">=</span> <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="co"># max as data</span></span>
<span id="cb193-12"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-12" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(n<span class="sc">&lt;</span><span class="dv">50</span>) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb193-13"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-13" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(x)[<span class="dv">2</span>] <span class="sc">&lt;</span><span class="dv">4</span>) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb193-14"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-14" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(n<span class="sc">&gt;</span> maxGeneWGCNA ) n <span class="ot">=</span> maxGeneWGCNA           </span>
<span id="cb193-15"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-15" aria-hidden="true" tabindex="-1"></a>            <span class="co">#x=as.matrix(x[1:n,])-apply(x[1:n,],1,mean)</span></span>
<span id="cb193-16"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-16" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb193-17"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-17" aria-hidden="true" tabindex="-1"></a>            datExpr<span class="ot">=</span><span class="fu">t</span>(x[<span class="dv">1</span><span class="sc">:</span>n,])</span>
<span id="cb193-18"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-19"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-19" aria-hidden="true" tabindex="-1"></a>            subGeneNames<span class="ot">=</span><span class="fu">colnames</span>(datExpr)</span>
<span id="cb193-20"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-20" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb193-21"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-21" aria-hidden="true" tabindex="-1"></a>            <span class="co">#Choosing a soft-threshold to fit a scale-free topology to the network</span></span>
<span id="cb193-22"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-22" aria-hidden="true" tabindex="-1"></a>            powers <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>), <span class="fu">seq</span>(<span class="at">from =</span> <span class="dv">12</span>, <span class="at">to=</span><span class="dv">20</span>, <span class="at">by=</span><span class="dv">2</span>));</span>
<span id="cb193-23"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-23" aria-hidden="true" tabindex="-1"></a>            sft<span class="ot">=</span><span class="fu">pickSoftThreshold</span>(datExpr,<span class="at">dataIsExpr =</span> <span class="cn">TRUE</span>,<span class="at">powerVector =</span> powers,<span class="at">corFnc =</span> cor,<span class="at">corOptions =</span> <span class="fu">list</span>(<span class="at">use =</span> <span class="st">&#39;p&#39;</span>),<span class="at">networkType =</span> <span class="st">&quot;unsigned&quot;</span>)</span>
<span id="cb193-24"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-25"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-26"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-26" aria-hidden="true" tabindex="-1"></a>            softPower <span class="ot">=</span> input_mySoftPower;</span>
<span id="cb193-27"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-28"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-28" aria-hidden="true" tabindex="-1"></a>            <span class="co">#calclute the adjacency matrix</span></span>
<span id="cb193-29"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-29" aria-hidden="true" tabindex="-1"></a>            adj<span class="ot">=</span> <span class="fu">adjacency</span>(datExpr,<span class="at">type =</span> <span class="st">&quot;unsigned&quot;</span>, <span class="at">power =</span> softPower);</span>
<span id="cb193-30"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-31"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-31" aria-hidden="true" tabindex="-1"></a>            <span class="co">#turn adjacency matrix into topological overlap to minimize the effects of noise and spurious associations</span></span>
<span id="cb193-32"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-32" aria-hidden="true" tabindex="-1"></a>            TOM<span class="ot">=</span><span class="fu">TOMsimilarityFromExpr</span>(datExpr,<span class="at">networkType =</span> <span class="st">&quot;unsigned&quot;</span>, <span class="at">TOMType =</span> <span class="st">&quot;unsigned&quot;</span>, <span class="at">power =</span> softPower);</span>
<span id="cb193-33"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-33" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(TOM) <span class="ot">=</span>subGeneNames</span>
<span id="cb193-34"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-34" aria-hidden="true" tabindex="-1"></a>            <span class="fu">rownames</span>(TOM) <span class="ot">=</span>subGeneNames</span>
<span id="cb193-35"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-35" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb193-36"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-36" aria-hidden="true" tabindex="-1"></a>            <span class="do">##########################################</span></span>
<span id="cb193-37"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-37" aria-hidden="true" tabindex="-1"></a>            <span class="co"># module detection</span></span>
<span id="cb193-38"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-39"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-39" aria-hidden="true" tabindex="-1"></a>            <span class="fu">library</span>(flashClust,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb193-40"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-41"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-41" aria-hidden="true" tabindex="-1"></a>            geneTree <span class="ot">=</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>TOM),<span class="at">method=</span><span class="st">&quot;average&quot;</span>);</span>
<span id="cb193-42"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-43"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-43" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Set the minimum module size</span></span>
<span id="cb193-44"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-44" aria-hidden="true" tabindex="-1"></a>            minModuleSize <span class="ot">=</span> input_minModuleSize;</span>
<span id="cb193-45"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-46"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-46" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Module identification using dynamic tree cut</span></span>
<span id="cb193-47"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-47" aria-hidden="true" tabindex="-1"></a>            dynamicMods <span class="ot">=</span> <span class="fu">cutreeDynamic</span>(<span class="at">dendro =</span> geneTree,  <span class="at">method=</span><span class="st">&quot;tree&quot;</span>, <span class="at">minClusterSize =</span> minModuleSize);</span>
<span id="cb193-48"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-48" aria-hidden="true" tabindex="-1"></a>            <span class="co">#dynamicMods = cutreeDynamic(dendro = geneTree, distM = dissTOM, method=&quot;hybrid&quot;, deepSplit = 2, pamRespectsDendro = FALSE, minClusterSize = minModuleSize);</span></span>
<span id="cb193-49"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-49" aria-hidden="true" tabindex="-1"></a>            <span class="co"># table(dynamicMods)</span></span>
<span id="cb193-50"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-50" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb193-51"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-51" aria-hidden="true" tabindex="-1"></a>            dynamicColors <span class="ot">=</span> <span class="fu">labels2colors</span>(dynamicMods)</span>
<span id="cb193-52"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-52" aria-hidden="true" tabindex="-1"></a>            <span class="co">#table(dynamicColors)</span></span>
<span id="cb193-53"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-54"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-54" aria-hidden="true" tabindex="-1"></a>            <span class="co">#discard the unassigned genes, and focus on the rest  # this causes problems</span></span>
<span id="cb193-55"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-55" aria-hidden="true" tabindex="-1"></a>            <span class="co">#restGenes= (dynamicColors != &quot;grey&quot;)</span></span>
<span id="cb193-56"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-56" aria-hidden="true" tabindex="-1"></a>            <span class="co">#diss1=1-TOMsimilarityFromExpr(datExpr[,restGenes], power = softPower)</span></span>
<span id="cb193-57"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-57" aria-hidden="true" tabindex="-1"></a>            <span class="co">#colnames(diss1) =rownames(diss1) =subGeneNames[restGenes]</span></span>
<span id="cb193-58"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-58" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb193-59"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-59" aria-hidden="true" tabindex="-1"></a>            <span class="co">#colorCode = as.character(dynamicColors[restGenes])</span></span>
<span id="cb193-60"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-61"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-61" aria-hidden="true" tabindex="-1"></a>            moduleInfo <span class="ot">=</span> <span class="fu">cbind</span>( subGeneNames, dynamicColors, dynamicMods)</span>
<span id="cb193-62"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-62" aria-hidden="true" tabindex="-1"></a>            moduleInfo <span class="ot">=</span> moduleInfo[<span class="fu">which</span>(moduleInfo[,<span class="dv">2</span>] <span class="sc">!=</span> <span class="st">&quot;grey&quot;</span>) ,] <span class="co"># remove genes not in any modules</span></span>
<span id="cb193-63"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-63" aria-hidden="true" tabindex="-1"></a>            moduleInfo <span class="ot">=</span> moduleInfo[<span class="fu">order</span>(moduleInfo[,<span class="dv">3</span>]),] <span class="co"># sort</span></span>
<span id="cb193-64"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-64" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb193-65"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-65" aria-hidden="true" tabindex="-1"></a>            n.modules <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">unique</span>(dynamicColors) ) <span class="sc">-</span><span class="dv">1</span> ; nGenes <span class="ot">=</span> <span class="fu">dim</span>(moduleInfo)[<span class="dv">1</span>] </span>
<span id="cb193-66"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-66" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb193-67"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-67" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">x =</span> <span class="fu">t</span>(datExpr),<span class="at">powers=</span>powers,<span class="at">sft=</span>sft, <span class="at">TOM =</span> TOM, <span class="at">dynamicColors =</span> dynamicColors, <span class="at">moduleInfo =</span> moduleInfo,<span class="at">n.modules=</span>n.modules, <span class="at">nGenes =</span>nGenes) )</span>
<span id="cb193-68"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-68" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb193-69"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-70"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-71"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-71" aria-hidden="true" tabindex="-1"></a>softPower <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb193-72"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-73"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-73" aria-hidden="true" tabindex="-1"></a>        <span class="do">####################################  </span></span>
<span id="cb193-74"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-74" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb193-75"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-75" aria-hidden="true" tabindex="-1"></a>        sft <span class="ot">=</span> wgcna.out<span class="sc">$</span>sft;</span>
<span id="cb193-76"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-76" aria-hidden="true" tabindex="-1"></a>        powers<span class="ot">=</span>wgcna.out<span class="sc">$</span>powers;</span>
<span id="cb193-77"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-77" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Plot the results</span></span>
<span id="cb193-78"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-78" aria-hidden="true" tabindex="-1"></a>        <span class="co">#sizeGrWindow(9, 5)</span></span>
<span id="cb193-79"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-79" aria-hidden="true" tabindex="-1"></a>        <span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>));</span>
<span id="cb193-80"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-80" aria-hidden="true" tabindex="-1"></a>        cex1 <span class="ot">=</span> <span class="fl">0.9</span>;</span>
<span id="cb193-81"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-82"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-82" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Scale-free topology fit index as a function of the soft-thresholding power</span></span>
<span id="cb193-83"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-83" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">3</span>])<span class="sc">*</span>sft<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],<span class="at">xlab=</span><span class="st">&quot;Soft Threshold (power)&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Scale Free Topology Model Fit, signed R^2&quot;</span>,<span class="at">type=</span><span class="st">&quot;n&quot;</span>, <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Scale independence&quot;</span>));</span>
<span id="cb193-84"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-84" aria-hidden="true" tabindex="-1"></a>        <span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">3</span>])<span class="sc">*</span>sft<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],<span class="at">labels=</span>powers,<span class="at">cex=</span>cex1,<span class="at">col=</span><span class="st">&quot;red&quot;</span>);</span>
<span id="cb193-85"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-86"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-86" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Red line corresponds to using an R^2 cut-off</span></span>
<span id="cb193-87"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-87" aria-hidden="true" tabindex="-1"></a>        <span class="fu">abline</span>(<span class="at">h=</span><span class="fl">0.80</span>,<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span>
<span id="cb193-88"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-89"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-89" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Mean connectivity as a function of the soft-thresholding power</span></span>
<span id="cb193-90"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-90" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], sft<span class="sc">$</span>fitIndices[,<span class="dv">5</span>],<span class="at">xlab=</span><span class="st">&quot;Soft Threshold (power)&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Mean Connectivity&quot;</span>, <span class="at">type=</span><span class="st">&quot;n&quot;</span>,<span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Mean connectivity&quot;</span>))</span>
<span id="cb193-91"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-91" aria-hidden="true" tabindex="-1"></a>        <span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], sft<span class="sc">$</span>fitIndices[,<span class="dv">5</span>], <span class="at">labels=</span>powers, <span class="at">cex=</span>cex1,<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span>
<span id="cb193-92"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-92" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb193-93"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-94"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-95"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-95" aria-hidden="true" tabindex="-1"></a>modulePlot <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb193-96"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-96" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb193-97"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-97" aria-hidden="true" tabindex="-1"></a>        diss1 <span class="ot">=</span> <span class="dv">1</span><span class="sc">-</span>wgcna.out<span class="sc">$</span>TOM;</span>
<span id="cb193-98"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-98" aria-hidden="true" tabindex="-1"></a>        dynamicColors <span class="ot">=</span> wgcna.out<span class="sc">$</span>dynamicColors</span>
<span id="cb193-99"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-99" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb193-100"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-100" aria-hidden="true" tabindex="-1"></a>        hier1<span class="ot">=</span><span class="fu">flashClust</span>(<span class="fu">as.dist</span>(diss1), <span class="at">method=</span><span class="st">&quot;average&quot;</span> )</span>
<span id="cb193-101"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-101" aria-hidden="true" tabindex="-1"></a>        <span class="co">#set the diagonal of the dissimilarity to NA </span></span>
<span id="cb193-102"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-102" aria-hidden="true" tabindex="-1"></a>        <span class="fu">diag</span>(diss1) <span class="ot">=</span> <span class="cn">NA</span>;</span>
<span id="cb193-103"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-103" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plotDendroAndColors</span>(hier1, dynamicColors, <span class="st">&quot;Dynamic Tree Cut&quot;</span>, <span class="at">dendroLabels =</span> <span class="cn">FALSE</span>, <span class="at">hang =</span> <span class="fl">0.03</span>, <span class="at">addGuide =</span> <span class="cn">TRUE</span>, <span class="at">guideHang =</span> <span class="fl">0.05</span>, <span class="at">main =</span> <span class="st">&quot;Gene dendrogram and module colors&quot;</span>)</span>
<span id="cb193-104"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-105"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-105" aria-hidden="true" tabindex="-1"></a>        <span class="co">#Visualize the Tom plot. Raise the dissimilarity matrix to the power of 4 to bring out the module structure</span></span>
<span id="cb193-106"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-106" aria-hidden="true" tabindex="-1"></a>        <span class="co">#TOMplot(diss1, hier1, colorCode)           </span></span>
<span id="cb193-107"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-107" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb193-108"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-109"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-110"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-110" aria-hidden="true" tabindex="-1"></a>listWGCNA.Modules <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb193-111"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-112"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-112" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">is.null</span>(wgcna.out ) ){ <span class="co"># if sample info is uploaded and correctly parsed.</span></span>
<span id="cb193-113"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-113" aria-hidden="true" tabindex="-1"></a>           <span class="fu">return</span>(<span class="cn">NULL</span>)    </span>
<span id="cb193-114"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-114" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> { </span>
<span id="cb193-115"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-115" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">dim</span>(wgcna.out<span class="sc">$</span>moduleInfo)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>  ) {  <span class="co"># if no module</span></span>
<span id="cb193-116"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-116" aria-hidden="true" tabindex="-1"></a>                <span class="fu">return</span>(<span class="cn">NULL</span>) </span>
<span id="cb193-117"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-117" aria-hidden="true" tabindex="-1"></a>            }   <span class="cf">else</span>  { </span>
<span id="cb193-118"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-118" aria-hidden="true" tabindex="-1"></a>                    modules <span class="ot">=</span> <span class="fu">unique</span>(wgcna.out<span class="sc">$</span>moduleInfo[, <span class="fu">c</span>(<span class="st">&quot;dynamicMods&quot;</span>,<span class="st">&quot;dynamicColors&quot;</span>)] )</span>
<span id="cb193-119"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-119" aria-hidden="true" tabindex="-1"></a>                    moduleList <span class="ot">=</span> <span class="fu">apply</span>(modules,<span class="dv">1</span>,paste,<span class="at">collapse=</span><span class="st">&quot;. &quot;</span>)</span>
<span id="cb193-120"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-120" aria-hidden="true" tabindex="-1"></a>                    moduleList  <span class="ot">=</span> <span class="fu">paste0</span>( moduleList, <span class="st">&quot; (&quot;</span>, <span class="fu">table</span>(wgcna.out<span class="sc">$</span>moduleInfo[,<span class="st">&quot;dynamicMods&quot;</span>] ),<span class="st">&quot; genes)&quot;</span>  )</span>
<span id="cb193-121"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-121" aria-hidden="true" tabindex="-1"></a>                    moduleList <span class="ot">=</span> <span class="fu">c</span>(moduleList,<span class="st">&quot;Entire network&quot;</span>)</span>
<span id="cb193-122"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-122" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">return</span>(moduleList)</span>
<span id="cb193-123"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-124"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-125"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-125" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb193-126"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-127"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-127" aria-hidden="true" tabindex="-1"></a>        }           </span>
<span id="cb193-128"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-128" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb193-129"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-130"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-131"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-131" aria-hidden="true" tabindex="-1"></a>moduleNetwork <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb193-132"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-133"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-133" aria-hidden="true" tabindex="-1"></a>        outfile <span class="ot">&lt;-</span> <span class="fu">tempfile</span>(<span class="at">fileext=</span><span class="st">&#39;.txt&#39;</span>)</span>
<span id="cb193-134"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-134" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb193-135"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-135" aria-hidden="true" tabindex="-1"></a>        <span class="co">#module = gsub(&quot;.* &quot;,&quot;&quot;,input_selectWGCNA.Module)</span></span>
<span id="cb193-136"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-136" aria-hidden="true" tabindex="-1"></a>        module <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(input_selectWGCNA.Module,<span class="st">&quot; &quot;</span> ) )[<span class="dv">2</span>]</span>
<span id="cb193-137"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-137" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb193-138"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-138" aria-hidden="true" tabindex="-1"></a>        moduleColors <span class="ot">=</span> wgcna.out<span class="sc">$</span>dynamicColors </span>
<span id="cb193-139"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-139" aria-hidden="true" tabindex="-1"></a>        inModule <span class="ot">=</span> (moduleColors<span class="sc">==</span>module);</span>
<span id="cb193-140"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-140" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb193-141"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-141" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( input_selectWGCNA.Module <span class="sc">==</span> <span class="st">&quot;Entire network&quot;</span>)</span>
<span id="cb193-142"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-142" aria-hidden="true" tabindex="-1"></a>          inModule <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">TRUE</span>, <span class="fu">length</span>(inModule))</span>
<span id="cb193-143"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-143" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb193-144"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-144" aria-hidden="true" tabindex="-1"></a>        datExpr <span class="ot">=</span> <span class="fu">t</span>(wgcna.out<span class="sc">$</span>x )</span>
<span id="cb193-145"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-145" aria-hidden="true" tabindex="-1"></a>        probes <span class="ot">=</span> <span class="fu">colnames</span>(datExpr)</span>
<span id="cb193-146"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-146" aria-hidden="true" tabindex="-1"></a>        modProbes <span class="ot">=</span> probes[inModule];</span>
<span id="cb193-147"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-147" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb193-148"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-148" aria-hidden="true" tabindex="-1"></a>        modTOM <span class="ot">=</span> wgcna.out<span class="sc">$</span>TOM[inModule, inModule];</span>
<span id="cb193-149"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-149" aria-hidden="true" tabindex="-1"></a>        <span class="fu">dimnames</span>(modTOM) <span class="ot">=</span> <span class="fu">list</span>(modProbes, modProbes)</span>
<span id="cb193-150"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-151"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-151" aria-hidden="true" tabindex="-1"></a>        nTop <span class="ot">=</span> input_topGenesNetwork;</span>
<span id="cb193-152"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-152" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb193-153"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-153" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( nTop <span class="sc">&gt;</span> <span class="dv">1000</span>) nTop <span class="ot">=</span> <span class="dv">1000</span>; </span>
<span id="cb193-154"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-154" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb193-155"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-155" aria-hidden="true" tabindex="-1"></a>        IMConn <span class="ot">=</span> <span class="fu">softConnectivity</span>(datExpr[, modProbes]);</span>
<span id="cb193-156"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-156" aria-hidden="true" tabindex="-1"></a>        top <span class="ot">=</span> (<span class="fu">rank</span>(<span class="sc">-</span>IMConn) <span class="sc">&lt;=</span> nTop)</span>
<span id="cb193-157"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-157" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb193-158"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-158" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adding symbols </span></span>
<span id="cb193-159"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-159" aria-hidden="true" tabindex="-1"></a>        probeToGene <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb193-160"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-160" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( input_selectGO5 <span class="sc">!=</span> <span class="st">&quot;ID not recognized!&quot;</span> <span class="sc">&amp;</span> input_selectOrg <span class="sc">!=</span> <span class="st">&quot;NEW&quot;</span>)</span>
<span id="cb193-161"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-161" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">sum</span>(<span class="fu">is.na</span>( allGeneInfo.out<span class="sc">$</span>symbol ) )<span class="sc">/</span> <span class="fu">dim</span>( allGeneInfo.out )[<span class="dv">1</span>] <span class="sc">&lt;</span>.<span class="dv">5</span> ) { <span class="co"># if more than 50% genes has symbol</span></span>
<span id="cb193-162"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-162" aria-hidden="true" tabindex="-1"></a>            probeToGene <span class="ot">=</span> allGeneInfo.out[,<span class="fu">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="st">&quot;symbol&quot;</span>)]</span>
<span id="cb193-163"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-163" aria-hidden="true" tabindex="-1"></a>            probeToGene<span class="sc">$</span>symbol <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>,<span class="st">&quot;&quot;</span>,probeToGene<span class="sc">$</span>symbol)</span>
<span id="cb193-164"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-165"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-165" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">which</span>( <span class="fu">is.na</span>(probeToGene<span class="sc">$</span>symbol) <span class="sc">|</span></span>
<span id="cb193-166"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-166" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">nchar</span>(probeToGene<span class="sc">$</span>symbol)<span class="sc">&lt;</span><span class="dv">2</span> <span class="sc">|</span> </span>
<span id="cb193-167"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-167" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">toupper</span>(probeToGene<span class="sc">$</span>symbol)<span class="sc">==</span><span class="st">&quot;NA&quot;</span> <span class="sc">|</span>  </span>
<span id="cb193-168"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-168" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">toupper</span>(probeToGene<span class="sc">$</span>symbol)<span class="sc">==</span><span class="st">&quot;0&quot;</span>  )             </span>
<span id="cb193-169"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-169" aria-hidden="true" tabindex="-1"></a>            probeToGene[ix,<span class="dv">2</span>] <span class="ot">=</span> probeToGene[ix,<span class="dv">1</span>]  <span class="co"># use gene ID</span></span>
<span id="cb193-170"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-171"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-171" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb193-172"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-172" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb193-173"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-173" aria-hidden="true" tabindex="-1"></a>        net <span class="ot">&lt;-</span> modTOM[top,top] <span class="sc">&gt;</span>input_edgeThreshold </span>
<span id="cb193-174"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-175"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-175" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb193-176"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-176" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(net)[<span class="dv">1</span>])  <span class="co"># remove self connection</span></span>
<span id="cb193-177"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-177" aria-hidden="true" tabindex="-1"></a>            net[i,i] <span class="ot">=</span> <span class="cn">FALSE</span></span>
<span id="cb193-178"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-178" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(probeToGene) <span class="sc">&amp;</span> <span class="sc">!</span>input_noIDConversion) { <span class="co"># if gene symbol exist</span></span>
<span id="cb193-179"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-179" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">match</span>( <span class="fu">colnames</span>(net), probeToGene[,<span class="dv">1</span>])     </span>
<span id="cb193-180"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-180" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(net) <span class="ot">=</span> probeToGene[ix,<span class="dv">2</span>]</span>
<span id="cb193-181"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-181" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">match</span>( <span class="fu">rownames</span>(net), probeToGene[,<span class="dv">1</span>])     </span>
<span id="cb193-182"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-182" aria-hidden="true" tabindex="-1"></a>            <span class="fu">rownames</span>(net) <span class="ot">=</span> probeToGene[ix,<span class="dv">2</span>]       </span>
<span id="cb193-183"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-183" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb193-184"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-184" aria-hidden="true" tabindex="-1"></a>        <span class="fu">library</span>(igraph,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb193-185"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-185" aria-hidden="true" tabindex="-1"></a>        <span class="co">#plot(graph_from_data_frame(d=data.frame(1:10,ncol=2)  ,directed=F) )</span></span>
<span id="cb193-186"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-186" aria-hidden="true" tabindex="-1"></a>        <span class="co"># http://www.kateto.net/wp-content/uploads/2016/01/NetSciX_2016_Workshop.pdf</span></span>
<span id="cb193-187"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-187" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot</span>( <span class="fu">graph_from_adjacency_matrix</span>( net, <span class="at">mod =</span><span class="st">&quot;undirected&quot;</span> ), </span>
<span id="cb193-188"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-188" aria-hidden="true" tabindex="-1"></a>                <span class="at">vertex.label.color=</span><span class="st">&quot;black&quot;</span>, <span class="at">vertex.label.dist=</span><span class="dv">3</span>,<span class="at">vertex.size=</span><span class="dv">7</span>)</span>
<span id="cb193-189"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-189" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb193-190"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-190" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb193-191"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-192"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-193"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-193" aria-hidden="true" tabindex="-1"></a>networkModuleGO <span class="ot">&lt;-</span> <span class="cf">function</span>(){      </span>
<span id="cb193-194"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-195"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-195" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb193-196"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-196" aria-hidden="true" tabindex="-1"></a>            <span class="co">#module = gsub(&quot;.* &quot;,&quot;&quot;,input_selectWGCNA.Module)</span></span>
<span id="cb193-197"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-197" aria-hidden="true" tabindex="-1"></a>            module <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(input_selectWGCNA.Module,<span class="st">&quot; &quot;</span> ) )[<span class="dv">2</span>]    </span>
<span id="cb193-198"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-198" aria-hidden="true" tabindex="-1"></a>            moduleColors <span class="ot">=</span> wgcna.out<span class="sc">$</span>dynamicColors </span>
<span id="cb193-199"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-199" aria-hidden="true" tabindex="-1"></a>            inModule <span class="ot">=</span> (moduleColors<span class="sc">==</span>module);</span>
<span id="cb193-200"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-200" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb193-201"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-201" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( input_selectWGCNA.Module <span class="sc">==</span> <span class="st">&quot;Entire network&quot;</span>)</span>
<span id="cb193-202"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-202" aria-hidden="true" tabindex="-1"></a>              inModule <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">TRUE</span>, <span class="fu">length</span>(inModule))</span>
<span id="cb193-203"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-204"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-204" aria-hidden="true" tabindex="-1"></a>            probes <span class="ot">=</span> <span class="fu">rownames</span>(wgcna.out<span class="sc">$</span>x   )</span>
<span id="cb193-205"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-205" aria-hidden="true" tabindex="-1"></a>            query  <span class="ot">=</span> probes[inModule];</span>
<span id="cb193-206"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-206" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb193-207"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-207" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">length</span>(query)  <span class="sc">&lt;=</span> minGenesEnrichment) <span class="fu">return</span>(NoSig) </span>
<span id="cb193-208"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-209"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-209" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb193-210"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-211"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-211" aria-hidden="true" tabindex="-1"></a>            result <span class="ot">&lt;-</span> <span class="fu">findOverlapGMT</span>( query, GeneSets.out,<span class="dv">1</span>) </span>
<span id="cb193-212"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-212" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb193-213"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-213" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb193-214"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-214" aria-hidden="true" tabindex="-1"></a>            result<span class="sc">$</span>Genes <span class="ot">=</span> <span class="st">&quot;Up regulated&quot;</span></span>
<span id="cb193-215"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-215" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb193-216"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-217"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-217" aria-hidden="true" tabindex="-1"></a>            results1 <span class="ot">=</span> result</span>
<span id="cb193-218"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-218" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> ( <span class="fu">is.null</span>( results1) ) <span class="fu">return</span> (NoSig)</span>
<span id="cb193-219"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-219" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">&lt;</span><span class="dv">5</span> ) <span class="fu">return</span>(NoSig)  <span class="co"># Returns a data frame: &quot;No significant results found!&quot;</span></span>
<span id="cb193-220"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-221"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-221" aria-hidden="true" tabindex="-1"></a>            results1<span class="ot">=</span> results1[,<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">4</span>)]</span>
<span id="cb193-222"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-222" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(results1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;List&quot;</span>,<span class="st">&quot;FDR&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;GO terms or pathways&quot;</span>)</span>
<span id="cb193-223"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-223" aria-hidden="true" tabindex="-1"></a>            minFDR <span class="ot">=</span> <span class="fl">0.01</span></span>
<span id="cb193-224"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-224" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">min</span>(results1<span class="sc">$</span>FDR) <span class="sc">&gt;</span> minFDR ) results1 <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No signficant enrichment found.&quot;</span>) <span class="cf">else</span></span>
<span id="cb193-225"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-225" aria-hidden="true" tabindex="-1"></a>            results1 <span class="ot">=</span> results1[<span class="fu">which</span>(results1<span class="sc">$</span>FDR <span class="sc">&lt;</span> minFDR),]</span>
<span id="cb193-226"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-226" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb193-227"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-227" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb193-228"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-228" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb193-229"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-229" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">!=</span> <span class="dv">4</span>) <span class="fu">return</span>(NoSig)</span>
<span id="cb193-230"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-230" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(results1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Direction&quot;</span>,<span class="st">&quot;adj.Pval&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;Pathways&quot;</span>)</span>
<span id="cb193-231"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-231" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb193-232"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-232" aria-hidden="true" tabindex="-1"></a>            results1<span class="sc">$</span>adj.Pval <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;%-2.1e&quot;</span>,<span class="fu">as.numeric</span>(results1<span class="sc">$</span>adj.Pval) )</span>
<span id="cb193-233"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-233" aria-hidden="true" tabindex="-1"></a>            results1[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(results1[,<span class="dv">1</span>])</span>
<span id="cb193-234"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-234" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">&lt;-</span> results1[,<span class="dv">1</span>]</span>
<span id="cb193-235"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-236"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-236" aria-hidden="true" tabindex="-1"></a>            results1[ <span class="fu">duplicated</span> (results1[,<span class="dv">1</span>] ),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb193-237"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-237" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb193-238"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-238" aria-hidden="true" tabindex="-1"></a>            results1[,<span class="sc">-</span><span class="dv">1</span>]           </span>
<span id="cb193-239"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#cb193-239" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="rna-seq-mkdata.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="wgcna-rna-seq-blog.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Bioinformatics_RNA-seq.pdf", "Bioinformatics_RNA-seq.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
