<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 R code modules to be used for stand-alone analysis. | Bioinformatics_RNA-seq</title>
  <meta name="description" content="RNA-seq notes and trials" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 R code modules to be used for stand-alone analysis. | Bioinformatics_RNA-seq" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="RNA-seq notes and trials" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 R code modules to be used for stand-alone analysis. | Bioinformatics_RNA-seq" />
  
  <meta name="twitter:description" content="RNA-seq notes and trials" />
  

<meta name="author" content="J. Marcelo Rosales R." />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="rna-seq-mkdata.html"/>
<link rel="next" href="wgcna-rna-seq-blog.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to RNA sequencing (RNA_seq)</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#downloading-and-setting-up-conda-environments."><i class="fa fa-check"></i><b>1.1</b> Downloading and setting up conda environments.</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#install-miniconda-or-anaconda."><i class="fa fa-check"></i><b>1.2</b> Install Miniconda, or Anaconda.</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#conda-terminal-commands."><i class="fa fa-check"></i><b>1.2.1</b> Conda terminal commands.</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#anacondaminiconda-install-summary-of-steps."><i class="fa fa-check"></i><b>1.2.2</b> Anaconda/Miniconda install Summary of Steps.</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#conda-activate-error"><i class="fa fa-check"></i><b>1.2.3</b> Conda activate error</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#alignment-procedures-and-files."><i class="fa fa-check"></i><b>1.3</b> Alignment Procedures and files.</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#using-tophat-early-version"><i class="fa fa-check"></i><b>1.3.1</b> Using Tophat (early version)</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#using-tophat"><i class="fa fa-check"></i><b>1.3.2</b> Using Tophat</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#using-star"><i class="fa fa-check"></i><b>1.3.3</b> Using STAR</a></li>
<li class="chapter" data-level="1.3.4" data-path="index.html"><a href="index.html#Mus-musculus-NCBI-build37-2"><i class="fa fa-check"></i><b>1.3.4</b> Custom/personal way to Download the Mus_musculus_NCBI_build37.2.tar.gz file.</a></li>
<li class="chapter" data-level="1.3.5" data-path="index.html"><a href="index.html#get-the-sample-files."><i class="fa fa-check"></i><b>1.3.5</b> Get the Sample files.</a></li>
<li class="chapter" data-level="1.3.6" data-path="index.html"><a href="index.html#starting-to-analize-fastq.gz-files"><i class="fa fa-check"></i><b>1.3.6</b> Starting to analize fastq.gz files</a></li>
<li class="chapter" data-level="1.3.7" data-path="index.html"><a href="index.html#trimming-with-trimmomatic"><i class="fa fa-check"></i><b>1.3.7</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="1.3.8" data-path="index.html"><a href="index.html#using-star-to-map-genes"><i class="fa fa-check"></i><b>1.3.8</b> Using STAR to map genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html"><i class="fa fa-check"></i><b>2</b> File Download (or SRA), QC, and Trimming</a>
<ul>
<li class="chapter" data-level="2.1" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#trimmomatic"><i class="fa fa-check"></i><b>2.1</b> Trimmomatic</a></li>
<li class="chapter" data-level="2.2" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#indexing-with-star"><i class="fa fa-check"></i><b>2.2</b> Indexing with STAR</a></li>
<li class="chapter" data-level="2.3" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#microbial-contamination"><i class="fa fa-check"></i><b>2.3</b> Microbial contamination</a></li>
<li class="chapter" data-level="2.4" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#feature-counts."><i class="fa fa-check"></i><b>2.4</b> Feature counts.</a></li>
<li class="chapter" data-level="2.5" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#setting-r-environment."><i class="fa fa-check"></i><b>2.5</b> Setting R environment.</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>3</b> Methods</a></li>
<li class="chapter" data-level="4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><i class="fa fa-check"></i><b>4</b> RNA Seq Differentially Expressed, GO Enrichment, and Pathway analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#background"><i class="fa fa-check"></i><b>4.1</b> <strong>Background</strong></a></li>
<li class="chapter" data-level="4.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#cleaning-data"><i class="fa fa-check"></i><b>4.2</b> <strong>Cleaning Data</strong></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#programs-used"><i class="fa fa-check"></i><b>4.2.1</b> Programs used?</a></li>
<li class="chapter" data-level="4.2.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#acquiring-sequences"><i class="fa fa-check"></i><b>4.2.2</b> Acquiring sequences</a></li>
<li class="chapter" data-level="4.2.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#fastqc"><i class="fa fa-check"></i><b>4.2.3</b> FastQC</a></li>
<li class="chapter" data-level="4.2.4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#trimming-with-trimmomatic-1"><i class="fa fa-check"></i><b>4.2.4</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="4.2.5" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#alignment"><i class="fa fa-check"></i><b>4.2.5</b> Alignment</a></li>
<li class="chapter" data-level="4.2.6" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#assembling-transcripts-with-cufflinks"><i class="fa fa-check"></i><b>4.2.6</b> Assembling transcripts with Cufflinks</a></li>
<li class="chapter" data-level="4.2.7" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#checking-for-contamination"><i class="fa fa-check"></i><b>4.2.7</b> Checking for Contamination</a></li>
<li class="chapter" data-level="4.2.8" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#counting-transcripts"><i class="fa fa-check"></i><b>4.2.8</b> Counting Transcripts</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#differentially-expressed-sequence-identification"><i class="fa fa-check"></i><b>4.3</b> <strong>Differentially Expressed Sequence Identification</strong></a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#analyzing-reads-counts"><i class="fa fa-check"></i><b>4.3.1</b> Analyzing Reads Counts</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e.g.-p53-control-and-ir-and-p53---control-and-ir"><i class="fa fa-check"></i><b>4.4</b> Interactions cause a difference between the lfc betwen pooled data, e.g. p53+/+ (control and IR) and p53-/- (control and IR)</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>4.4.1</b> Exploratory Data Analysis</a></li>
<li class="chapter" data-level="4.4.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#identification-of-differentially-expressed-genes"><i class="fa fa-check"></i><b>4.4.2</b> Identification of Differentially Expressed Genes</a></li>
<li class="chapter" data-level="4.4.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#gene-annotations"><i class="fa fa-check"></i><b>4.4.3</b> Gene Annotations</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#go-enrichment-analysis-using-gostats"><i class="fa fa-check"></i><b>4.5</b> 7. GO Enrichment analysis using GOstats</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#go-enrichment-analysis-of-downregulated-genes"><i class="fa fa-check"></i><b>4.5.1</b> GO Enrichment analysis of downregulated genes</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-analysis-using-expression-data"><i class="fa fa-check"></i><b>4.6</b> 8. Pathway analysis using expression data</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#prepare-data"><i class="fa fa-check"></i><b>4.6.1</b> Prepare data</a></li>
<li class="chapter" data-level="4.6.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#kegg-pathways"><i class="fa fa-check"></i><b>4.6.2</b> KEGG pathways</a></li>
<li class="chapter" data-level="4.6.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-oxidative-phosphorylation."><i class="fa fa-check"></i><b>4.6.3</b> Pathway and regulation of genes for Oxidative phosphorylation.</a></li>
<li class="chapter" data-level="4.6.4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-glycosylphosphatidylinositolgpi-anchor-biosynthesis."><i class="fa fa-check"></i><b>4.6.4</b> Pathway and regulation of genes for Glycosylphosphatidylinositol(GPI)-anchor biosynthesis.</a></li>
<li class="chapter" data-level="4.6.5" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-rna-polymerase."><i class="fa fa-check"></i><b>4.6.5</b> Pathway and regulation of genes for RNA polymerase.</a></li>
<li class="chapter" data-level="4.6.6" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-nucleotide-excision-repair."><i class="fa fa-check"></i><b>4.6.6</b> Pathway and regulation of genes for Nucleotide excision repair.</a></li>
<li class="chapter" data-level="4.6.7" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-non-homologous-end-joining."><i class="fa fa-check"></i><b>4.6.7</b> Pathway and regulation of genes for Non-homologous end-joining.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html"><i class="fa fa-check"></i><b>5</b> RNA-Seq-MKdata</a>
<ul>
<li class="chapter" data-level="5.1" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#pca"><i class="fa fa-check"></i><b>5.1</b> PCA</a></li>
<li class="chapter" data-level="5.2" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#differential-expression"><i class="fa fa-check"></i><b>5.2</b> Differential expression</a></li>
<li class="chapter" data-level="5.3" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#rna-seq-scienceparksg"><i class="fa fa-check"></i><b>5.3</b> RNA-seq (ScienceparkSG)</a></li>
<li class="chapter" data-level="5.4" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#false-discovery-rates-fdr-benjamini-hocheberg-method"><i class="fa fa-check"></i><b>5.4</b> False discovery rates FDR / Benjamini-Hocheberg method</a></li>
<li class="chapter" data-level="5.5" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#volcano-plot"><i class="fa fa-check"></i><b>5.5</b> Volcano plot</a></li>
<li class="chapter" data-level="5.6" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#heatmap-normalization"><i class="fa fa-check"></i><b>5.6</b> Heatmap &amp; normalization</a></li>
<li class="chapter" data-level="5.7" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#ma-plots"><i class="fa fa-check"></i><b>5.7</b> MA plots</a></li>
<li class="chapter" data-level="5.8" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#eastern-new-mexico-university-enmu-method"><i class="fa fa-check"></i><b>5.8</b> Eastern New Mexico University ENMU method</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="r-code-modules-to-be-used-for-stand-alone-analysis-.html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html"><i class="fa fa-check"></i><b>6</b> R code modules to be used for stand-alone analysis.</a></li>
<li class="chapter" data-level="7" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html"><i class="fa fa-check"></i><b>7</b> WGCNA RNA-seq Blog</a>
<ul>
<li class="chapter" data-level="7.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#WGCNA"><i class="fa fa-check"></i><b>7.1</b> Introduction to the WGCNA Rpackage</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-rpackage-installation-procedure."><i class="fa fa-check"></i><b>7.1.1</b> WGCNA Rpackage Installation procedure.</a></li>
<li class="chapter" data-level="7.1.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#final-installation-workflow"><i class="fa fa-check"></i><b>7.1.2</b> Final installation workflow:</a></li>
<li class="chapter" data-level="7.1.3" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-tutorial."><i class="fa fa-check"></i><b>7.1.3</b> WGCNA tutorial.</a></li>
<li class="chapter" data-level="7.1.4" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#gene-expression-analysis"><i class="fa fa-check"></i><b>7.1.4</b> 3.4 Gene expression analysis</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#rna-seq-analisys-with-kakus-data-sample."><i class="fa fa-check"></i><b>7.2</b> RNA-seq Analisys with Kaku’s Data sample.</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#data-analysis."><i class="fa fa-check"></i><b>7.2.1</b> Data Analysis.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="youtube-references-.html"><a href="youtube-references-.html"><i class="fa fa-check"></i><b>8</b> Youtube references.</a>
<ul>
<li class="chapter" data-level="8.1" data-path="youtube-references-.html"><a href="youtube-references-.html#anaconda-set-up-robust-data-science-environment-with-miniconda-and-conda-forge."><i class="fa fa-check"></i><b>8.1</b> Anaconda set up, Robust Data Science Environment with Miniconda and Conda-Forge.</a></li>
<li class="chapter" data-level="8.2" data-path="youtube-references-.html"><a href="youtube-references-.html#quality-control-preprocessing-of-raw-reads"><i class="fa fa-check"></i><b>8.2</b> Quality control &amp; preprocessing of raw reads</a></li>
<li class="chapter" data-level="8.3" data-path="youtube-references-.html"><a href="youtube-references-.html#trimmomatic."><i class="fa fa-check"></i><b>8.3</b> Trimmomatic.</a></li>
<li class="chapter" data-level="8.4" data-path="youtube-references-.html"><a href="youtube-references-.html#building-genome-index-and-aligning-with-star"><i class="fa fa-check"></i><b>8.4</b> Building Genome Index and Aligning with STAR</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="external-ref.html"><a href="external-ref.html"><i class="fa fa-check"></i><b>9</b> External Ref</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics_RNA-seq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="r-code-modules-to-be-used-for-stand-alone-analysis." class="section level1" number="6">
<h1><span class="header-section-number">Chapter 6</span> R code modules to be used for stand-alone analysis.</h1>
<p>Base on the South Dakota State University analysis protocol</p>
<div class="sourceCode" id="cb182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb182-1"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-2"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Install packages</span></span>
<span id="cb182-3"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-4"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-5"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-5" aria-hidden="true" tabindex="-1"></a><span class="co"># dplyr complains this required libraries: libudunits2-dev, libmariadb-client-lgpl-dev</span></span>
<span id="cb182-6"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-6" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;plotly&quot;, repos=&quot;http://cran.rstudio.com/&quot;, dependencies=TRUE)</span></span>
<span id="cb182-7"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-7" aria-hidden="true" tabindex="-1"></a><span class="co"># sometimes need to remove all installed packages: https://www.r-bloggers.com/how-to-remove-all-user-installed-packages-in-r/ </span></span>
<span id="cb182-8"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-8" aria-hidden="true" tabindex="-1"></a><span class="co">#cat(&quot;\n Installing lots of R packages, this may take several hours ... \n</span></span>
<span id="cb182-9"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   Each of these packages took years to develop.\n So be a patient thief.\n</span></span>
<span id="cb182-10"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-10" aria-hidden="true" tabindex="-1"></a><span class="co">#   \n Note 1: Sometimes dependencies needs to be installed manually. </span></span>
<span id="cb182-11"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-11" aria-hidden="true" tabindex="-1"></a><span class="co">#   \n Note 2: If you are using an older version of R, and having trouble with package installation,</span></span>
<span id="cb182-12"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-12" aria-hidden="true" tabindex="-1"></a><span class="co">#   \n         sometimes, it is easier to install the new version of R and delete all old packages, and start fresh. </span></span>
<span id="cb182-13"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-13" aria-hidden="true" tabindex="-1"></a><span class="co">#  &quot;)</span></span>
<span id="cb182-14"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-15"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-15" aria-hidden="true" tabindex="-1"></a>list.of.packages <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;dendextend&quot;</span>, <span class="st">&quot;htmlwidgets&quot;</span>,<span class="st">&quot;RSQLite&quot;</span>, </span>
<span id="cb182-16"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;gplots&quot;</span>, <span class="st">&quot;ggplot2&quot;</span>, <span class="st">&quot;dplyr&quot;</span>, <span class="st">&quot;tidyverse&quot;</span>,<span class="st">&quot;plotly&quot;</span>,<span class="st">&quot;e1071&quot;</span>, </span>
<span id="cb182-17"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;reshape2&quot;</span>,  <span class="st">&quot;Rcpp&quot;</span>,<span class="st">&quot;flashClust&quot;</span>,<span class="st">&quot;statmod&quot;</span>,</span>
<span id="cb182-18"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;biclust&quot;</span>,<span class="st">&quot;igraph&quot;</span>,<span class="st">&quot;Rtsne&quot;</span>)</span>
<span id="cb182-19"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-20"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-20" aria-hidden="true" tabindex="-1"></a>list.of.bio.packages  <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;STRINGdb&quot;</span>, <span class="st">&quot;limma&quot;</span>, <span class="st">&quot;DESeq2&quot;</span>, <span class="st">&quot;edgeR&quot;</span>, <span class="st">&quot;gage&quot;</span>, </span>
<span id="cb182-21"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;PGSEA&quot;</span>, <span class="st">&quot;fgsea&quot;</span>, <span class="st">&quot;ReactomePA&quot;</span>, <span class="st">&quot;pathview&quot;</span>, <span class="st">&quot;PREDA&quot;</span>, <span class="st">&quot;impute&quot;</span>, <span class="st">&quot;runibic&quot;</span>,</span>
<span id="cb182-22"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-22" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;QUBIC&quot;</span>,<span class="st">&quot;rhdf5&quot;</span>, <span class="st">&quot;PREDAsampledata&quot;</span>, <span class="st">&quot;sfsmisc&quot;</span>, <span class="st">&quot;lokern&quot;</span>, <span class="st">&quot;multtest&quot;</span>, </span>
<span id="cb182-23"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-23" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;GSEABase&quot;</span>,<span class="st">&quot;hgu133plus2.db&quot;</span>)</span>
<span id="cb182-24"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-25"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-25" aria-hidden="true" tabindex="-1"></a><span class="co">#Install Require packages</span></span>
<span id="cb182-26"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-26" aria-hidden="true" tabindex="-1"></a>new.packages <span class="ot">&lt;-</span> list.of.packages[<span class="sc">!</span>(list.of.packages <span class="sc">%in%</span> <span class="fu">installed.packages</span>()[,<span class="st">&quot;Package&quot;</span>])]</span>
<span id="cb182-27"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-27" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">length</span>(new.packages)) <span class="fu">install.packages</span>(new.packages, <span class="at">repos=</span><span class="st">&quot;http://cran.rstudio.com/&quot;</span>, <span class="at">dependencies=</span><span class="cn">TRUE</span>)</span>
<span id="cb182-28"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-29"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-29" aria-hidden="true" tabindex="-1"></a>new.bio.packages <span class="ot">&lt;-</span> list.of.bio.packages[<span class="sc">!</span>(list.of.bio.packages <span class="sc">%in%</span> <span class="fu">installed.packages</span>()[,<span class="st">&quot;Package&quot;</span>])]</span>
<span id="cb182-30"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-30" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">length</span>(new.bio.packages)){</span>
<span id="cb182-31"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-31" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb182-32"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb182-33"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-33" aria-hidden="true" tabindex="-1"></a>    BiocManager<span class="sc">::</span><span class="fu">install</span>(new.bio.packages, <span class="at">suppressUpdates =</span> T)</span>
<span id="cb182-34"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-34" aria-hidden="true" tabindex="-1"></a>}   </span>
<span id="cb182-35"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-36"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-36" aria-hidden="true" tabindex="-1"></a><span class="co"># WGCNA must be installed after AnnotationDbi, impute, GO.db, preprocessCore</span></span>
<span id="cb182-37"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-37" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(c(&quot;WGCNA&quot;))</span></span>
<span id="cb182-38"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-39"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-39" aria-hidden="true" tabindex="-1"></a><span class="co">#Test Load Packages    This cause an error when too many packages are loaded.</span></span>
<span id="cb182-40"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-40" aria-hidden="true" tabindex="-1"></a><span class="co"># suc = unlist ( lapply(list.of.packages, require, character.only = TRUE) )</span></span>
<span id="cb182-41"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-41" aria-hidden="true" tabindex="-1"></a><span class="co"># if(sum(suc) &lt; length(list.of.packages) )</span></span>
<span id="cb182-42"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-42" aria-hidden="true" tabindex="-1"></a><span class="co">#   cat (&quot;\n\nWarnning!!!!!! These R packages cannot be loaded:&quot;, list.of.packages[!suc] )</span></span>
<span id="cb182-43"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-44"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-44" aria-hidden="true" tabindex="-1"></a><span class="co"># suc = unlist ( lapply(list.of.bio.packages, require, character.only = TRUE) )</span></span>
<span id="cb182-45"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-45" aria-hidden="true" tabindex="-1"></a><span class="co"># if(sum(suc) &lt; length(list.of.bio.packages) )</span></span>
<span id="cb182-46"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-46" aria-hidden="true" tabindex="-1"></a><span class="co">#   cat (&quot;\n\nWarnning!!!!!! These Bioconductor packages cannot be loaded:&quot;, list.of.bio.packages[!suc] )</span></span>
<span id="cb182-47"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-48"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-49"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-49" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-50"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-50" aria-hidden="true" tabindex="-1"></a><span class="co"># Global variables</span></span>
<span id="cb182-51"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-51" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-52"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-53"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-53" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RSQLite,<span class="at">verbose=</span><span class="cn">FALSE</span>)  <span class="co"># for database connection</span></span>
<span id="cb182-54"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-54" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gplots,<span class="at">verbose=</span><span class="cn">FALSE</span>)       <span class="co"># for hierarchical clustering</span></span>
<span id="cb182-55"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-55" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2,<span class="at">verbose=</span><span class="cn">FALSE</span>)  <span class="co"># graphics</span></span>
<span id="cb182-56"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-56" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(e1071,<span class="at">verbose=</span><span class="cn">FALSE</span>)        <span class="co"># computing kurtosis</span></span>
<span id="cb182-57"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-57" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DT,<span class="at">verbose=</span><span class="cn">FALSE</span>)       <span class="co"># for renderDataTable</span></span>
<span id="cb182-58"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-58" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly,<span class="at">verbose=</span><span class="cn">FALSE</span>)   <span class="co"># for interactive heatmap</span></span>
<span id="cb182-59"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-59" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2,<span class="at">verbose=</span><span class="cn">FALSE</span>)     <span class="co"># for melt correlation matrix in heatmap</span></span>
<span id="cb182-60"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-60" aria-hidden="true" tabindex="-1"></a>input_goButton <span class="ot">=</span> <span class="dv">0</span>    <span class="co"># if 1, use demo data file</span></span>
<span id="cb182-61"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-61" aria-hidden="true" tabindex="-1"></a>Min_overlap <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb182-62"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-62" aria-hidden="true" tabindex="-1"></a>minSetSize <span class="ot">=</span> <span class="dv">3</span>; </span>
<span id="cb182-63"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-63" aria-hidden="true" tabindex="-1"></a>mappingCoverage <span class="ot">=</span> <span class="fl">0.60</span> <span class="co"># 60% percent genes has to be mapped for confident mapping</span></span>
<span id="cb182-64"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-64" aria-hidden="true" tabindex="-1"></a>mappingEdge <span class="ot">=</span> <span class="fl">0.5</span>  <span class="co"># Top species has 50% more genes mapped</span></span>
<span id="cb182-65"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-65" aria-hidden="true" tabindex="-1"></a>PvalGeneInfo <span class="ot">=</span> <span class="fl">0.05</span>; minGenes <span class="ot">=</span> <span class="dv">10</span> <span class="co"># min number of genes for ploting</span></span>
<span id="cb182-66"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-66" aria-hidden="true" tabindex="-1"></a>kurtosis.log <span class="ot">=</span> <span class="dv">50</span>  <span class="co"># log transform is enforced when kurtosis is big</span></span>
<span id="cb182-67"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-67" aria-hidden="true" tabindex="-1"></a>kurtosis.warning <span class="ot">=</span> <span class="dv">10</span> <span class="co"># log transformation recommnded </span></span>
<span id="cb182-68"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-68" aria-hidden="true" tabindex="-1"></a>minGenesEnrichment <span class="ot">=</span> <span class="dv">2</span> <span class="co"># perform GO or promoter analysis only if more than this many genes</span></span>
<span id="cb182-69"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-69" aria-hidden="true" tabindex="-1"></a>PREDA_Permutations <span class="ot">=</span><span class="dv">1000</span></span>
<span id="cb182-70"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-70" aria-hidden="true" tabindex="-1"></a>maxGeneClustering <span class="ot">=</span> <span class="dv">12000</span>  <span class="co"># max genes for hierarchical clustering and k-Means clustering. Slow if larger</span></span>
<span id="cb182-71"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-71" aria-hidden="true" tabindex="-1"></a>maxGeneWGCNA <span class="ot">=</span> <span class="dv">2000</span> <span class="co"># max genes for co-expression network</span></span>
<span id="cb182-72"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-72" aria-hidden="true" tabindex="-1"></a>maxFactors <span class="ot">=</span><span class="dv">6</span>  <span class="co"># max number of factors in DESeq2 models</span></span>
<span id="cb182-73"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-73" aria-hidden="true" tabindex="-1"></a>redudantGeneSetsRatio <span class="ot">=</span> <span class="fl">0.9</span>  <span class="co"># remove redundant genesets in enrichment analysis</span></span>
<span id="cb182-74"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-74" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">2</span>) <span class="co"># seed for random number generator</span></span>
<span id="cb182-75"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-75" aria-hidden="true" tabindex="-1"></a>mycolors <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">rainbow</span>(<span class="dv">20</span>))[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">20</span>,<span class="dv">10</span>,<span class="dv">11</span>,<span class="dv">2</span>,<span class="dv">19</span>,<span class="dv">3</span>,<span class="dv">12</span>,<span class="dv">4</span>,<span class="dv">13</span>,<span class="dv">5</span>,<span class="dv">14</span>,<span class="dv">6</span>,<span class="dv">15</span>,<span class="dv">7</span>,<span class="dv">16</span>,<span class="dv">8</span>,<span class="dv">17</span>,<span class="dv">9</span>,<span class="dv">18</span>)] <span class="co"># 20 colors for kNN clusters</span></span>
<span id="cb182-76"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-76" aria-hidden="true" tabindex="-1"></a><span class="co">#Each row of this matrix represents a color scheme;</span></span>
<span id="cb182-77"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-78"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-78" aria-hidden="true" tabindex="-1"></a>hmcols <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&quot;#D73027&quot;</span>, <span class="st">&quot;#FC8D59&quot;</span>, <span class="st">&quot;#FEE090&quot;</span>, <span class="st">&quot;#FFFFBF&quot;</span>,</span>
<span id="cb182-79"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-79" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;#E0F3F8&quot;</span>, <span class="st">&quot;#91BFDB&quot;</span>, <span class="st">&quot;#4575B4&quot;</span>)))(<span class="dv">75</span>)</span>
<span id="cb182-80"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-80" aria-hidden="true" tabindex="-1"></a>heatColors <span class="ot">=</span> <span class="fu">rbind</span>(      <span class="fu">greenred</span>(<span class="dv">75</span>),     <span class="fu">bluered</span>(<span class="dv">75</span>),     <span class="fu">colorpanel</span>(<span class="dv">75</span>,<span class="st">&quot;green&quot;</span>,<span class="st">&quot;black&quot;</span>,<span class="st">&quot;magenta&quot;</span>),<span class="fu">colorpanel</span>(<span class="dv">75</span>,<span class="st">&quot;blue&quot;</span>,<span class="st">&quot;yellow&quot;</span>,<span class="st">&quot;red&quot;</span>),hmcols )</span>
<span id="cb182-81"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-81" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heatColors) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Green-Black-Red&quot;</span>,<span class="st">&quot;Blue-White-Red&quot;</span>,<span class="st">&quot;Green-Black-Magenta&quot;</span>,<span class="st">&quot;Blue-Yellow-Red&quot;</span>,<span class="st">&quot;Blue-white-brown&quot;</span>)</span>
<span id="cb182-82"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-82" aria-hidden="true" tabindex="-1"></a>colorChoices <span class="ot">=</span> <span class="fu">setNames</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(heatColors)[<span class="dv">1</span>],<span class="fu">rownames</span>(heatColors)) <span class="co"># for pull down menu</span></span>
<span id="cb182-83"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-84"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-84" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a list for Select Input options</span></span>
<span id="cb182-85"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-85" aria-hidden="true" tabindex="-1"></a>speciesChoice <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb182-86"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-86" aria-hidden="true" tabindex="-1"></a>speciesChoice <span class="ot">&lt;-</span> <span class="fu">append</span>( <span class="fu">setNames</span>( <span class="st">&quot;NEW&quot;</span>,<span class="st">&quot;**NEW SPECIES**&quot;</span>), speciesChoice  )</span>
<span id="cb182-87"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-87" aria-hidden="true" tabindex="-1"></a>speciesChoice <span class="ot">&lt;-</span> <span class="fu">append</span>( <span class="fu">setNames</span>( <span class="st">&quot;BestMatch&quot;</span>,<span class="st">&quot;Best matching species&quot;</span>), speciesChoice  )</span>
<span id="cb182-88"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-89"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-89" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-90"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-90" aria-hidden="true" tabindex="-1"></a><span class="co"># Read data and pre-process</span></span>
<span id="cb182-91"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-91" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-92"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-93"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-93" aria-hidden="true" tabindex="-1"></a>readData <span class="ot">&lt;-</span> <span class="cf">function</span>(inFile ) {</span>
<span id="cb182-94"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-95"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-95" aria-hidden="true" tabindex="-1"></a>                <span class="co"># these packages moved here to reduce loading time</span></span>
<span id="cb182-96"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-96" aria-hidden="true" tabindex="-1"></a>                <span class="fu">library</span>(edgeR,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># count data D.E.</span></span>
<span id="cb182-97"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-97" aria-hidden="true" tabindex="-1"></a>                <span class="fu">library</span>(DESeq2,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># count data analysis</span></span>
<span id="cb182-98"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-99"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-99" aria-hidden="true" tabindex="-1"></a>                dataTypeWarning <span class="ot">=</span><span class="dv">0</span></span>
<span id="cb182-100"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-100" aria-hidden="true" tabindex="-1"></a>                dataType <span class="ot">=</span><span class="fu">c</span>(<span class="cn">TRUE</span>)</span>
<span id="cb182-101"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-102"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-102" aria-hidden="true" tabindex="-1"></a>                <span class="co">#---------------Read file</span></span>
<span id="cb182-103"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-103" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(inFile)   <span class="co"># try CSV</span></span>
<span id="cb182-104"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-104" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="fu">dim</span>(x)[<span class="dv">2</span>] <span class="sc">&lt;=</span> <span class="dv">2</span> )   <span class="co"># if less than 3 columns, try tab-deliminated</span></span>
<span id="cb182-105"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-105" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">&lt;-</span> <span class="fu">read.table</span>(inFile, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">header=</span><span class="cn">TRUE</span>)   </span>
<span id="cb182-106"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-106" aria-hidden="true" tabindex="-1"></a>                <span class="co">#-------Remove non-numeric columns, except the first column</span></span>
<span id="cb182-107"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-107" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-108"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-108" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">dim</span>(x)[<span class="dv">2</span>])</span>
<span id="cb182-109"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-109" aria-hidden="true" tabindex="-1"></a>                    dataType <span class="ot">=</span> <span class="fu">c</span>( dataType, <span class="fu">is.numeric</span>(x[,i]) )</span>
<span id="cb182-110"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-110" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="fu">sum</span>(dataType) <span class="sc">&lt;=</span><span class="dv">2</span>) <span class="fu">return</span> (<span class="cn">NULL</span>)  <span class="co"># only less than 2 columns are numbers</span></span>
<span id="cb182-111"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-111" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">&lt;-</span> x[,dataType]  <span class="co"># only keep numeric columns</span></span>
<span id="cb182-112"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-112" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-113"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-113" aria-hidden="true" tabindex="-1"></a>                <span class="co"># rows with all missing values</span></span>
<span id="cb182-114"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-114" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">=</span> <span class="fu">which</span>( <span class="fu">apply</span>(x[,<span class="sc">-</span><span class="dv">1</span>],<span class="dv">1</span>, <span class="cf">function</span>(y) <span class="fu">sum</span>( <span class="fu">is.na</span>(y) ) ) <span class="sc">!=</span> <span class="fu">dim</span>(x)[<span class="dv">2</span>]<span class="sc">-</span><span class="dv">1</span> )</span>
<span id="cb182-115"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-115" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">&lt;-</span> x[ix,]</span>
<span id="cb182-116"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-116" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-117"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-117" aria-hidden="true" tabindex="-1"></a>                dataSizeOriginal <span class="ot">=</span> <span class="fu">dim</span>(x); dataSizeOriginal[<span class="dv">2</span>] <span class="ot">=</span> dataSizeOriginal[<span class="dv">2</span>] <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb182-118"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-118" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-119"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-119" aria-hidden="true" tabindex="-1"></a>                x[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">toupper</span>(x[,<span class="dv">1</span>])</span>
<span id="cb182-120"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-120" aria-hidden="true" tabindex="-1"></a>                x[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>,<span class="st">&quot;&quot;</span>,x[,<span class="dv">1</span>]) <span class="co"># remove spaces in gene ids</span></span>
<span id="cb182-121"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-121" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">=</span> x[<span class="fu">order</span>(<span class="sc">-</span> <span class="fu">apply</span>(x[,<span class="dv">2</span><span class="sc">:</span><span class="fu">dim</span>(x)[<span class="dv">2</span>]],<span class="dv">1</span>,sd) ),]  <span class="co"># sort by SD</span></span>
<span id="cb182-122"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-122" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">&lt;-</span> x[<span class="sc">!</span><span class="fu">duplicated</span>(x[,<span class="dv">1</span>]) ,]  <span class="co"># remove duplicated genes</span></span>
<span id="cb182-123"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-123" aria-hidden="true" tabindex="-1"></a>                <span class="fu">rownames</span>(x) <span class="ot">&lt;-</span> x[,<span class="dv">1</span>]</span>
<span id="cb182-124"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-124" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(x[,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>)])</span>
<span id="cb182-125"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-125" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-126"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-126" aria-hidden="true" tabindex="-1"></a>                <span class="co"># remove &quot;-&quot; or &quot;.&quot; from sample names</span></span>
<span id="cb182-127"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-127" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(x) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(x))</span>
<span id="cb182-128"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-128" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(x) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(x))</span>
<span id="cb182-129"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-130"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-130" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-131"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-131" aria-hidden="true" tabindex="-1"></a>                <span class="co">#cat(&quot;\nhere&quot;,dim(x))</span></span>
<span id="cb182-132"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-132" aria-hidden="true" tabindex="-1"></a>                <span class="co"># missng value for median value</span></span>
<span id="cb182-133"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-133" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="fu">sum</span>(<span class="fu">is.na</span>(x))<span class="sc">&gt;</span><span class="dv">0</span>) {<span class="co"># if there is missing values</span></span>
<span id="cb182-134"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-134" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>(input_missingValue <span class="sc">==</span><span class="st">&quot;geneMedian&quot;</span>) { </span>
<span id="cb182-135"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-135" aria-hidden="true" tabindex="-1"></a>                        rowMedians <span class="ot">&lt;-</span> <span class="fu">apply</span>(x,<span class="dv">1</span>, <span class="cf">function</span> (y)  <span class="fu">median</span>(y,<span class="at">na.rm=</span>T))</span>
<span id="cb182-136"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-136" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(x)[<span class="dv">2</span>] ) {</span>
<span id="cb182-137"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-137" aria-hidden="true" tabindex="-1"></a>                            ix <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">is.na</span>(x[,i]) )</span>
<span id="cb182-138"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-138" aria-hidden="true" tabindex="-1"></a>                            x[ix,i] <span class="ot">&lt;-</span> rowMedians[ix]                       </span>
<span id="cb182-139"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-139" aria-hidden="true" tabindex="-1"></a>                        }</span>
<span id="cb182-140"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-140" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb182-141"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-141" aria-hidden="true" tabindex="-1"></a>                    } <span class="cf">else</span> <span class="cf">if</span>(input_missingValue <span class="sc">==</span><span class="st">&quot;treatAsZero&quot;</span>) {</span>
<span id="cb182-142"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-142" aria-hidden="true" tabindex="-1"></a>                        x[<span class="fu">is.na</span>(x) ] <span class="ot">&lt;-</span> <span class="dv">0</span>                   </span>
<span id="cb182-143"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-143" aria-hidden="true" tabindex="-1"></a>                    } <span class="cf">else</span> <span class="cf">if</span> (input_missingValue <span class="sc">==</span><span class="st">&quot;geneMedianInGroup&quot;</span>) {</span>
<span id="cb182-144"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-144" aria-hidden="true" tabindex="-1"></a>                        sampleGroups <span class="ot">=</span> <span class="fu">detectGroups</span>( <span class="fu">colnames</span>(x))</span>
<span id="cb182-145"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-145" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">for</span> (group <span class="cf">in</span> <span class="fu">unique</span>( sampleGroups) ){      </span>
<span id="cb182-146"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-146" aria-hidden="true" tabindex="-1"></a>                            samples <span class="ot">=</span> <span class="fu">which</span>( sampleGroups <span class="sc">==</span> group )</span>
<span id="cb182-147"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-147" aria-hidden="true" tabindex="-1"></a>                            rowMedians <span class="ot">&lt;-</span> <span class="fu">apply</span>(x[,samples, <span class="at">drop=</span>F],<span class="dv">1</span>, <span class="cf">function</span> (y)  <span class="fu">median</span>(y,<span class="at">na.rm=</span>T))</span>
<span id="cb182-148"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-148" aria-hidden="true" tabindex="-1"></a>                            <span class="cf">for</span>( i <span class="cf">in</span>  samples ) { </span>
<span id="cb182-149"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-149" aria-hidden="true" tabindex="-1"></a>                                ix <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">is.na</span>(x[ ,i] ) ) </span>
<span id="cb182-150"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-150" aria-hidden="true" tabindex="-1"></a>                                <span class="cf">if</span>(<span class="fu">length</span>(ix) <span class="sc">&gt;</span><span class="dv">0</span> )</span>
<span id="cb182-151"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-151" aria-hidden="true" tabindex="-1"></a>                                    x[ix, i  ]  <span class="ot">&lt;-</span> rowMedians[ix]</span>
<span id="cb182-152"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-152" aria-hidden="true" tabindex="-1"></a>                            }                                       </span>
<span id="cb182-153"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-153" aria-hidden="true" tabindex="-1"></a>                        }</span>
<span id="cb182-154"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-154" aria-hidden="true" tabindex="-1"></a>                        </span>
<span id="cb182-155"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-155" aria-hidden="true" tabindex="-1"></a>                        <span class="co"># missing for entire sample group, use median for all samples</span></span>
<span id="cb182-156"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-156" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>(<span class="fu">sum</span>(<span class="fu">is.na</span>(x) )<span class="sc">&gt;</span><span class="dv">0</span> ) { </span>
<span id="cb182-157"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-157" aria-hidden="true" tabindex="-1"></a>                            rowMedians <span class="ot">&lt;-</span> <span class="fu">apply</span>(x,<span class="dv">1</span>, <span class="cf">function</span> (y)  <span class="fu">median</span>(y,<span class="at">na.rm=</span>T))</span>
<span id="cb182-158"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-158" aria-hidden="true" tabindex="-1"></a>                            <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(x)[<span class="dv">2</span>] ) {</span>
<span id="cb182-159"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-159" aria-hidden="true" tabindex="-1"></a>                                ix <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">is.na</span>(x[,i]) )</span>
<span id="cb182-160"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-160" aria-hidden="true" tabindex="-1"></a>                                x[ix,i] <span class="ot">&lt;-</span> rowMedians[ix]                       </span>
<span id="cb182-161"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-161" aria-hidden="true" tabindex="-1"></a>                            }                       </span>
<span id="cb182-162"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-162" aria-hidden="true" tabindex="-1"></a>                        }</span>
<span id="cb182-163"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-163" aria-hidden="true" tabindex="-1"></a>                    }</span>
<span id="cb182-164"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-164" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb182-165"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-166"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-166" aria-hidden="true" tabindex="-1"></a>                <span class="co"># Compute kurtosis</span></span>
<span id="cb182-167"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-167" aria-hidden="true" tabindex="-1"></a>                mean.kurtosis <span class="ot">=</span> <span class="fu">mean</span>(<span class="fu">apply</span>(x,<span class="dv">2</span>, kurtosis),<span class="at">na.rm=</span>T)</span>
<span id="cb182-168"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-168" aria-hidden="true" tabindex="-1"></a>                rawCounts <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb182-169"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-169" aria-hidden="true" tabindex="-1"></a>                pvals<span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb182-170"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-170" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> (input_dataFileFormat <span class="sc">==</span> <span class="dv">2</span> ) {  <span class="co"># if FPKM, microarray</span></span>
<span id="cb182-171"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-172"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-172" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span> ( <span class="fu">is.integer</span>(x) ) dataTypeWarning <span class="ot">=</span> <span class="dv">1</span>;  <span class="co"># Data appears to be read counts</span></span>
<span id="cb182-173"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-174"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-174" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#-------------filtering</span></span>
<span id="cb182-175"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-175" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#tem &lt;- apply(x,1,max)</span></span>
<span id="cb182-176"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-176" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#x &lt;- x[which(tem &gt; input_lowFilter),]  # max by row is at least </span></span>
<span id="cb182-177"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-177" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">&lt;-</span> x[ <span class="fu">which</span>( <span class="fu">apply</span>( x, <span class="dv">1</span>,  <span class="cf">function</span>(y) <span class="fu">sum</span>(y <span class="sc">&gt;=</span> input_lowFilter)) <span class="sc">&gt;=</span> input_NminSamples2 ) , ] </span>
<span id="cb182-178"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-179"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-179" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb182-180"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-180" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">&lt;-</span> x[<span class="fu">which</span>(<span class="fu">apply</span>(x,<span class="dv">1</span>, <span class="cf">function</span>(y) <span class="fu">max</span>(y)<span class="sc">-</span> <span class="fu">min</span>(y) ) <span class="sc">&gt;</span> <span class="dv">0</span>  ),]  <span class="co"># remove rows with all the same levels</span></span>
<span id="cb182-181"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-182"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-182" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#--------------Log transform</span></span>
<span id="cb182-183"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-183" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># Takes log if log is selected OR kurtosis is big than 100</span></span>
<span id="cb182-184"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-184" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span> ( (input_transform <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">|</span> (mean.kurtosis <span class="sc">&gt;</span> kurtosis.log ) ) </span>
<span id="cb182-185"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-185" aria-hidden="true" tabindex="-1"></a>                        x <span class="ot">=</span> <span class="fu">log</span>(x<span class="sc">+</span><span class="fu">abs</span>( input_logStart),<span class="dv">2</span>)</span>
<span id="cb182-186"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-187"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-187" aria-hidden="true" tabindex="-1"></a>                    tem <span class="ot">&lt;-</span> <span class="fu">apply</span>(x,<span class="dv">1</span>,sd) </span>
<span id="cb182-188"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-188" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">&lt;-</span> x[<span class="fu">order</span>(<span class="sc">-</span>tem),]  <span class="co"># sort by SD</span></span>
<span id="cb182-189"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-190"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-190" aria-hidden="true" tabindex="-1"></a>                } <span class="cf">else</span> </span>
<span id="cb182-191"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-191" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>( input_dataFileFormat <span class="sc">==</span> <span class="dv">1</span>) {  <span class="co"># counts data</span></span>
<span id="cb182-192"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-193"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-194"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-194" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># data not seems to be read counts</span></span>
<span id="cb182-195"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-195" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.integer</span>(x) <span class="sc">&amp;</span> mean.kurtosis <span class="sc">&lt;</span> kurtosis.log ) {</span>
<span id="cb182-196"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-196" aria-hidden="true" tabindex="-1"></a>                        dataTypeWarning <span class="ot">=</span> <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb182-197"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-197" aria-hidden="true" tabindex="-1"></a>                    }</span>
<span id="cb182-198"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-198" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># not used as some counts data like those from CRISPR screen</span></span>
<span id="cb182-199"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-199" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#validate(   # if Kurtosis is less than a threshold, it is not read-count</span></span>
<span id="cb182-200"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-200" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#   need(mean.kurtosis &gt; kurtosis.log, &quot;Data does not seem to be read count based on distribution. Please double check.&quot;)</span></span>
<span id="cb182-201"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-201" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># )</span></span>
<span id="cb182-202"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-202" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">&lt;-</span> <span class="fu">round</span>(x,<span class="dv">0</span>) <span class="co"># enforce the read counts to be integers. Sailfish outputs has decimal points.</span></span>
<span id="cb182-203"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-203" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#x &lt;- x[ which( apply(x,1,max) &gt;= input_minCounts ) , ] # remove all zero counts</span></span>
<span id="cb182-204"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-204" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb182-205"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-206"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-206" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># remove genes if it does not at least have minCounts in at least NminSamples</span></span>
<span id="cb182-207"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-207" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#x &lt;- x[ which( apply(x,1,function(y) sum(y&gt;=input_minCounts)) &gt;= input_NminSamples ) , ]  # filtering on raw counts</span></span>
<span id="cb182-208"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-208" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># using counts per million (CPM) for filtering out genes.</span></span>
<span id="cb182-209"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-209" aria-hidden="true" tabindex="-1"></a>                                             <span class="co"># CPM matrix                  #N samples &gt; minCounts</span></span>
<span id="cb182-210"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-210" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">&lt;-</span> x[ <span class="fu">which</span>( <span class="fu">apply</span>( <span class="fu">cpm</span>(<span class="fu">DGEList</span>(<span class="at">counts =</span> x)), <span class="dv">1</span>,  </span>
<span id="cb182-211"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-211" aria-hidden="true" tabindex="-1"></a>                            <span class="cf">function</span>(y) <span class="fu">sum</span>(y<span class="sc">&gt;=</span>input_minCounts)) <span class="sc">&gt;=</span> input_NminSamples ) , ] </span>
<span id="cb182-212"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-212" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb182-213"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-214"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-215"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-215" aria-hidden="true" tabindex="-1"></a>                    rawCounts <span class="ot">=</span> x; <span class="co"># ??? </span></span>
<span id="cb182-216"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-216" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># construct DESeqExpression Object</span></span>
<span id="cb182-217"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-217" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># colData = cbind(colnames(x), as.character(detectGroups( colnames(x) )) )</span></span>
<span id="cb182-218"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-218" aria-hidden="true" tabindex="-1"></a>                    tem <span class="ot">=</span> <span class="fu">rep</span>(<span class="st">&quot;A&quot;</span>,<span class="fu">dim</span>(x)[<span class="dv">2</span>]); tem[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;B&quot;</span>   <span class="co"># making a fake design matrix to allow process, even when there is no replicates</span></span>
<span id="cb182-219"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-219" aria-hidden="true" tabindex="-1"></a>                    colData <span class="ot">=</span> <span class="fu">cbind</span>(<span class="fu">colnames</span>(x), tem )</span>
<span id="cb182-220"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-220" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">colnames</span>(colData)  <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;groups&quot;</span>)</span>
<span id="cb182-221"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-221" aria-hidden="true" tabindex="-1"></a>                    dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> x, <span class="at">colData =</span> colData, <span class="at">design =</span> <span class="sc">~</span> groups)</span>
<span id="cb182-222"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-222" aria-hidden="true" tabindex="-1"></a>                    dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds) <span class="co"># estimate size factor for use in normalization later for started log method</span></span>
<span id="cb182-223"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-224"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-225"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-225" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># regularized log  or VST transformation</span></span>
<span id="cb182-226"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-226" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>( input_CountsTransform <span class="sc">==</span> <span class="dv">3</span> ) { <span class="co"># rlog is slow, only do it with 10 samples</span></span>
<span id="cb182-227"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-227" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>(<span class="fu">dim</span>(x)[<span class="dv">2</span>]<span class="sc">&lt;=</span><span class="dv">20</span> ) { </span>
<span id="cb182-228"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-228" aria-hidden="true" tabindex="-1"></a>                         x <span class="ot">&lt;-</span> <span class="fu">rlog</span>(dds, <span class="at">blind=</span><span class="cn">TRUE</span>); x <span class="ot">&lt;-</span> <span class="fu">assay</span>(x) } <span class="cf">else</span> </span>
<span id="cb182-229"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-229" aria-hidden="true" tabindex="-1"></a>                         x <span class="ot">&lt;-</span> <span class="fu">log2</span>( <span class="fu">counts</span>(dds, <span class="at">normalized=</span><span class="cn">TRUE</span>) <span class="sc">+</span> input_countsLogStart ) </span>
<span id="cb182-230"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-230" aria-hidden="true" tabindex="-1"></a>                         }  </span>
<span id="cb182-231"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-232"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-232" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">else</span> {</span>
<span id="cb182-233"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-233" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span> ( input_CountsTransform <span class="sc">==</span> <span class="dv">2</span> ) {    <span class="co"># vst is fast but aggressive</span></span>
<span id="cb182-234"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-234" aria-hidden="true" tabindex="-1"></a>                            x <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind=</span><span class="cn">TRUE</span>)</span>
<span id="cb182-235"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-235" aria-hidden="true" tabindex="-1"></a>                            x <span class="ot">&lt;-</span> <span class="fu">assay</span>(x)  </span>
<span id="cb182-236"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-236" aria-hidden="true" tabindex="-1"></a>                        } <span class="cf">else</span>{  <span class="co"># normalized by library sizes and add a constant.</span></span>
<span id="cb182-237"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-237" aria-hidden="true" tabindex="-1"></a>                            x <span class="ot">&lt;-</span> <span class="fu">log2</span>( <span class="fu">counts</span>(dds, <span class="at">normalized=</span><span class="cn">TRUE</span>) <span class="sc">+</span> input_countsLogStart )   <span class="co"># log(x+c) </span></span>
<span id="cb182-238"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-238" aria-hidden="true" tabindex="-1"></a>                            <span class="co"># This is equivalent to below. But the prior counts is more important</span></span>
<span id="cb182-239"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-239" aria-hidden="true" tabindex="-1"></a>                            <span class="co">#x &lt;- cpm(DGEList(counts = x),log=TRUE, prior.count=input_countsLogStart )  #log CPM from edgeR</span></span>
<span id="cb182-240"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-240" aria-hidden="true" tabindex="-1"></a>                            <span class="co">#x &lt;- x-min(x)  # shift values to avoid negative numbers</span></span>
<span id="cb182-241"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-241" aria-hidden="true" tabindex="-1"></a>                        }</span>
<span id="cb182-242"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-242" aria-hidden="true" tabindex="-1"></a>                    }</span>
<span id="cb182-243"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-243" aria-hidden="true" tabindex="-1"></a>                } <span class="cf">else</span> </span>
<span id="cb182-244"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-244" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>( input_dataFileFormat <span class="sc">==</span> <span class="dv">3</span>)  {  <span class="co"># other data type</span></span>
<span id="cb182-245"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-246"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-246" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#neg_lfc neg_fdr pos_lfc pos_fdr </span></span>
<span id="cb182-247"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-247" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#11       1      11       1 </span></span>
<span id="cb182-248"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-248" aria-hidden="true" tabindex="-1"></a>                        </span>
<span id="cb182-249"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-249" aria-hidden="true" tabindex="-1"></a>                        n2 <span class="ot">=</span> ( <span class="fu">dim</span>(x)[<span class="dv">2</span>] <span class="sc">%/%</span> <span class="dv">2</span>) <span class="co"># 5 --&gt; 2</span></span>
<span id="cb182-250"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-250" aria-hidden="true" tabindex="-1"></a>                        <span class="co"># It looks like it contains P values</span></span>
<span id="cb182-251"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-251" aria-hidden="true" tabindex="-1"></a>                        <span class="co"># ranges of columns add 0.2 and round to whole. For P value columns this should be 1</span></span>
<span id="cb182-252"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-252" aria-hidden="true" tabindex="-1"></a>                        tem <span class="ot">=</span> <span class="fu">round</span>( <span class="fu">apply</span>(x, <span class="dv">2</span>, <span class="cf">function</span>( y) <span class="fu">max</span>(y)<span class="sc">-</span> <span class="fu">min</span>(y))  <span class="sc">+</span> .<span class="dv">2</span>)     </span>
<span id="cb182-253"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-253" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>( <span class="fu">sum</span>(tem[(<span class="dv">1</span><span class="sc">:</span>n2)<span class="sc">*</span><span class="dv">2</span>  ] <span class="sc">==</span>  <span class="dv">1</span> ) <span class="sc">==</span> n2 <span class="sc">|</span> </span>
<span id="cb182-254"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-254" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">sum</span>(tem[(<span class="dv">1</span><span class="sc">:</span>n2)<span class="sc">*</span><span class="dv">2-1</span>  ] <span class="sc">==</span>  <span class="dv">1</span> ) <span class="sc">==</span> n2 ) {         </span>
<span id="cb182-255"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-255" aria-hidden="true" tabindex="-1"></a>                            x <span class="ot">=</span> x[,<span class="dv">1</span><span class="sc">:</span>(<span class="dv">2</span><span class="sc">*</span>n2) ,drop<span class="ot">=</span><span class="cn">FALSE</span> ] <span class="co"># if 5, change it to 4            </span></span>
<span id="cb182-256"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-256" aria-hidden="true" tabindex="-1"></a>                            <span class="cf">if</span>(tem[<span class="dv">2</span>] <span class="sc">==</span> <span class="dv">1</span>) { <span class="co"># FDR follows Fold-change</span></span>
<span id="cb182-257"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-257" aria-hidden="true" tabindex="-1"></a>                                pvals <span class="ot">=</span> x [,<span class="dv">2</span><span class="sc">*</span>(<span class="dv">1</span><span class="sc">:</span>n2 ),drop<span class="ot">=</span><span class="cn">FALSE</span> ]  <span class="co"># 2, 4, 6</span></span>
<span id="cb182-258"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-258" aria-hidden="true" tabindex="-1"></a>                                x <span class="ot">=</span> x[, <span class="dv">2</span><span class="sc">*</span>(<span class="dv">1</span><span class="sc">:</span>n2 )<span class="sc">-</span><span class="dv">1</span>,drop<span class="ot">=</span><span class="cn">FALSE</span>]   <span class="co"># 1, 3, 5</span></span>
<span id="cb182-259"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-260"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-260" aria-hidden="true" tabindex="-1"></a>                            } <span class="cf">else</span> {    <span class="co"># FDR follows Fold-change</span></span>
<span id="cb182-261"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-261" aria-hidden="true" tabindex="-1"></a>                                pvals <span class="ot">=</span> x [,<span class="dv">2</span><span class="sc">*</span>(<span class="dv">1</span><span class="sc">:</span>n2 )<span class="sc">-</span><span class="dv">1</span>,drop<span class="ot">=</span><span class="cn">FALSE</span> ]  <span class="co"># 2, 4, 6     </span></span>
<span id="cb182-262"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-262" aria-hidden="true" tabindex="-1"></a>                                x <span class="ot">=</span> x[, <span class="dv">2</span><span class="sc">*</span>(<span class="dv">1</span><span class="sc">:</span> n2 ),drop<span class="ot">=</span><span class="cn">FALSE</span>]   <span class="co"># 1, 3, 5</span></span>
<span id="cb182-263"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-263" aria-hidden="true" tabindex="-1"></a>                            }</span>
<span id="cb182-264"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-264" aria-hidden="true" tabindex="-1"></a>                        }</span>
<span id="cb182-265"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-265" aria-hidden="true" tabindex="-1"></a>                    ix <span class="ot">=</span>  <span class="fu">which</span>(<span class="fu">apply</span>(x,<span class="dv">1</span>, <span class="cf">function</span>(y) <span class="fu">max</span>(y)<span class="sc">-</span> <span class="fu">min</span>(y) ) <span class="sc">&gt;</span> <span class="dv">0</span>  )</span>
<span id="cb182-266"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-266" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">&lt;-</span> x[ix,]  <span class="co"># remove rows with all the same levels</span></span>
<span id="cb182-267"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-267" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(pvals) )</span>
<span id="cb182-268"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-268" aria-hidden="true" tabindex="-1"></a>                        pvals <span class="ot">=</span> pvals[ix,]</span>
<span id="cb182-269"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-269" aria-hidden="true" tabindex="-1"></a>                        </span>
<span id="cb182-270"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-270" aria-hidden="true" tabindex="-1"></a>                    }</span>
<span id="cb182-271"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-271" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb182-272"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-272" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb182-273"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-273" aria-hidden="true" tabindex="-1"></a>                dataSize <span class="ot">=</span> <span class="fu">dim</span>(x);</span>
<span id="cb182-274"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-274" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="sc">!</span>(<span class="fu">dim</span>(x)[<span class="dv">1</span>]<span class="sc">&gt;</span><span class="dv">5</span> <span class="sc">&amp;</span> <span class="fu">dim</span>(x)[<span class="dv">2</span>]<span class="sc">&gt;</span><span class="dv">1</span>)) </span>
<span id="cb182-275"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-275" aria-hidden="true" tabindex="-1"></a>                <span class="fu">stop</span> ( <span class="st">&quot;Data file not recognized. Please double check.&quot;</span>)</span>
<span id="cb182-276"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-277"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-278"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-278" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-279"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-279" aria-hidden="true" tabindex="-1"></a>                sampleInfoDemo<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb182-280"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-280" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>( input_goButton <span class="sc">&gt;</span><span class="dv">0</span>)</span>
<span id="cb182-281"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-281" aria-hidden="true" tabindex="-1"></a>                    sampleInfoDemo <span class="ot">&lt;-</span> <span class="fu">t</span>( <span class="fu">read.csv</span>(demoDataFile2,<span class="at">row.names=</span><span class="dv">1</span>,<span class="at">header=</span>T,<span class="at">colClasses=</span><span class="st">&quot;character&quot;</span>) )</span>
<span id="cb182-282"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-283"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-283" aria-hidden="true" tabindex="-1"></a>                    finalResult <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">data =</span> <span class="fu">as.matrix</span>(x), <span class="at">mean.kurtosis =</span> mean.kurtosis, <span class="at">rawCounts =</span> rawCounts, <span class="at">dataTypeWarning=</span>dataTypeWarning, <span class="at">dataSize=</span><span class="fu">c</span>(dataSizeOriginal,dataSize),<span class="at">sampleInfoDemo=</span>sampleInfoDemo, <span class="at">pvals =</span>pvals )</span>
<span id="cb182-284"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-284" aria-hidden="true" tabindex="-1"></a>                <span class="fu">return</span>(finalResult)</span>
<span id="cb182-285"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-286"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-286" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-287"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-288"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-288" aria-hidden="true" tabindex="-1"></a>readSampleInfo <span class="ot">&lt;-</span> <span class="cf">function</span>(inFile){</span>
<span id="cb182-289"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-290"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-290" aria-hidden="true" tabindex="-1"></a>                dataTypeWarning <span class="ot">=</span><span class="dv">0</span></span>
<span id="cb182-291"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-291" aria-hidden="true" tabindex="-1"></a>                dataType <span class="ot">=</span><span class="fu">c</span>(<span class="cn">TRUE</span>)</span>
<span id="cb182-292"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-293"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-293" aria-hidden="true" tabindex="-1"></a>                <span class="co">#---------------Read file</span></span>
<span id="cb182-294"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-294" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(inFile,<span class="at">row.names=</span><span class="dv">1</span>,<span class="at">header=</span>T,<span class="at">colClasses=</span><span class="st">&quot;character&quot;</span>)   <span class="co"># try CSV</span></span>
<span id="cb182-295"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-295" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="fu">dim</span>(x)[<span class="dv">2</span>] <span class="sc">&lt;=</span> <span class="dv">2</span> )   <span class="co"># if less than 3 columns, try tab-deliminated</span></span>
<span id="cb182-296"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-296" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">&lt;-</span> <span class="fu">read.table</span>(inFile, <span class="at">row.names=</span><span class="dv">1</span>,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">header=</span><span class="cn">TRUE</span>,<span class="at">colClasses=</span><span class="st">&quot;character&quot;</span>)</span>
<span id="cb182-297"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-297" aria-hidden="true" tabindex="-1"></a>                <span class="co"># remove &quot;-&quot; or &quot;.&quot; from sample names</span></span>
<span id="cb182-298"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-298" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(x) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(x))</span>
<span id="cb182-299"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-299" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(x) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(x))    </span>
<span id="cb182-300"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-301"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-301" aria-hidden="true" tabindex="-1"></a>                <span class="co">#----------------Matching with column names of expression file</span></span>
<span id="cb182-302"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-302" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">=</span> <span class="fu">match</span>(<span class="fu">toupper</span>(<span class="fu">colnames</span>(readData.out<span class="sc">$</span>data)), <span class="fu">toupper</span>(<span class="fu">colnames</span>(x)) ) </span>
<span id="cb182-303"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-303" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">=</span> ix[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(ix))] <span class="co"># remove NA</span></span>
<span id="cb182-304"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-305"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-305" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="sc">!</span> ( <span class="fu">length</span>(<span class="fu">unique</span>(ix) ) <span class="sc">==</span> <span class="fu">dim</span>(readData.out<span class="sc">$</span>data)[<span class="dv">2</span>] </span>
<span id="cb182-306"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-306" aria-hidden="true" tabindex="-1"></a>                       <span class="sc">&amp;</span> <span class="fu">dim</span>(x)[<span class="dv">1</span>]<span class="sc">&gt;=</span><span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="sc">&lt;</span><span class="dv">500</span> ) ) <span class="co"># at least one row, it can not be more than 500 rows</span></span>
<span id="cb182-307"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-307" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">stop</span>( <span class="st">&quot;Error!!! Sample information file not recognized. Sample names must be exactly the same. Each row is a factor. Each column represent a sample.  Please see documentation on format.&quot;</span>)</span>
<span id="cb182-308"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-308" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-309"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-309" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-310"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-310" aria-hidden="true" tabindex="-1"></a>                <span class="co">#-----------Double check factor levels, change if needed</span></span>
<span id="cb182-311"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-311" aria-hidden="true" tabindex="-1"></a>                <span class="co"># remove &quot;-&quot; or &quot;.&quot; from factor levels</span></span>
<span id="cb182-312"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-312" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(x)[<span class="dv">1</span>]) {</span>
<span id="cb182-313"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-313" aria-hidden="true" tabindex="-1"></a>                   x[i,] <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;&quot;</span>,x[i,])</span>
<span id="cb182-314"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-314" aria-hidden="true" tabindex="-1"></a>                   x[i,] <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>,x[i,])             </span>
<span id="cb182-315"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-315" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb182-316"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-316" aria-hidden="true" tabindex="-1"></a>                <span class="co"># if levels from different factors match</span></span>
<span id="cb182-317"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-317" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>( <span class="fu">length</span>(<span class="fu">unique</span>(ix) ) <span class="sc">==</span> <span class="fu">dim</span>(readData.out<span class="sc">$</span>data)[<span class="dv">2</span>]) { <span class="co"># matches exactly</span></span>
<span id="cb182-318"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-318" aria-hidden="true" tabindex="-1"></a>                    x <span class="ot">=</span> x[,ix]</span>
<span id="cb182-319"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-319" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># if the levels of different factors are the same, it may cause problems</span></span>
<span id="cb182-320"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-320" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>( <span class="fu">sum</span>( <span class="fu">apply</span>(x, <span class="dv">1</span>, <span class="cf">function</span>(y) <span class="fu">length</span>(<span class="fu">unique</span>(y)))) <span class="sc">&gt;</span> <span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">unlist</span>(x) ) ) ) {</span>
<span id="cb182-321"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-321" aria-hidden="true" tabindex="-1"></a>                        tem2 <span class="ot">=</span><span class="fu">apply</span>(x,<span class="dv">2</span>, <span class="cf">function</span>(y) <span class="fu">paste0</span>( <span class="fu">names</span>(y),y)) <span class="co"># factor names are added to levels</span></span>
<span id="cb182-322"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-322" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">rownames</span>(tem2) <span class="ot">=</span> <span class="fu">rownames</span>(x)</span>
<span id="cb182-323"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-323" aria-hidden="true" tabindex="-1"></a>                        x <span class="ot">&lt;-</span> tem2               </span>
<span id="cb182-324"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-324" aria-hidden="true" tabindex="-1"></a>                    }</span>
<span id="cb182-325"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-325" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">return</span>(<span class="fu">t</span>( x ) )         </span>
<span id="cb182-326"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-326" aria-hidden="true" tabindex="-1"></a>                } <span class="cf">else</span> <span class="fu">retrun</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-327"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-327" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb182-328"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-328" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-329"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-329" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-330"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-331"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-331" aria-hidden="true" tabindex="-1"></a>textTransform <span class="ot">&lt;-</span> <span class="cf">function</span> () { </span>
<span id="cb182-332"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-332" aria-hidden="true" tabindex="-1"></a>        k.value <span class="ot">=</span>  readData.out<span class="sc">$</span>mean.kurtosis     </span>
<span id="cb182-333"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-333" aria-hidden="true" tabindex="-1"></a>        tem <span class="ot">=</span> <span class="fu">paste</span>( <span class="st">&quot;Mean Kurtosis =  &quot;</span>, <span class="fu">round</span>(k.value,<span class="dv">2</span>), <span class="st">&quot;.</span><span class="sc">\n</span><span class="st">&quot;</span>,<span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb182-334"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-334" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( k.value <span class="sc">&gt;</span> kurtosis.log) tem <span class="ot">=</span> <span class="fu">paste</span>(tem, <span class="st">&quot; Detected extremely large values. </span><span class="sc">\n</span><span class="st">  When kurtosis &gt;&quot;</span>, kurtosis.log,</span>
<span id="cb182-335"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-335" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;, log transformation is automatically applied.</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>) <span class="cf">else</span> <span class="cf">if</span> (k.value<span class="sc">&gt;</span>kurtosis.warning)</span>
<span id="cb182-336"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-336" aria-hidden="true" tabindex="-1"></a>        {tem <span class="ot">=</span> <span class="fu">paste</span>(tem, <span class="st">&quot; Detected  large numbers with kurtosis &gt;&quot;</span>,</span>
<span id="cb182-337"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-337" aria-hidden="true" tabindex="-1"></a>        kurtosis.warning,<span class="st">&quot;. </span><span class="sc">\n</span><span class="st"> Log transformation is recommended.&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>) }</span>
<span id="cb182-338"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-338" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(readData.out<span class="sc">$</span>dataTypeWarning <span class="sc">==</span> <span class="dv">1</span> ) {</span>
<span id="cb182-339"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-339" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">paste</span> (tem, <span class="st">&quot; ------Warning!!! </span><span class="sc">\n</span><span class="st">Data matrix contains all integers. </span><span class="sc">\n</span><span class="st"> It seems to be read counts!!! </span><span class="sc">\n</span><span class="st"> Please select appropriate data type in the previous page and reload file.&quot;</span>)}</span>
<span id="cb182-340"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-340" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(readData.out<span class="sc">$</span>dataTypeWarning <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span> ) {</span>
<span id="cb182-341"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-341" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">paste</span> (tem, <span class="st">&quot;  ------Warning!!! </span><span class="sc">\n</span><span class="st">Data does not look like read counts data. </span><span class="sc">\n</span><span class="st"> Please select appropriate data type in the previous page and reload file.&quot;</span>)}</span>
<span id="cb182-342"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-342" aria-hidden="true" tabindex="-1"></a>        tem</span>
<span id="cb182-343"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-343" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-344"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-345"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-346"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-346" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up gene sets. Remove spaces and other control characters from gene names  </span></span>
<span id="cb182-347"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-347" aria-hidden="true" tabindex="-1"></a>cleanGeneSet <span class="ot">&lt;-</span> <span class="cf">function</span> (x){</span>
<span id="cb182-348"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-348" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove duplicate; upper case; remove special characters</span></span>
<span id="cb182-349"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-349" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">unique</span>( <span class="fu">toupper</span>( <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">| &quot;</span>,<span class="st">&quot;&quot;</span>,x) ) )</span>
<span id="cb182-350"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-350" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> x[<span class="fu">which</span>( <span class="fu">nchar</span>(x)<span class="sc">&gt;</span><span class="dv">1</span>) ]  <span class="co"># genes should have at least two characters</span></span>
<span id="cb182-351"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-351" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(x)</span>
<span id="cb182-352"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-352" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-353"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-353" aria-hidden="true" tabindex="-1"></a>findSpeciesById <span class="ot">&lt;-</span> <span class="cf">function</span> (speciesID){ <span class="co"># find species name use id</span></span>
<span id="cb182-354"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-354" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>( orgInfo[<span class="fu">which</span>(orgInfo<span class="sc">$</span>id <span class="sc">==</span> speciesID),]  )</span>
<span id="cb182-355"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-355" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-356"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-357"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-357" aria-hidden="true" tabindex="-1"></a><span class="co"># just return name</span></span>
<span id="cb182-358"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-358" aria-hidden="true" tabindex="-1"></a>findSpeciesByIdName <span class="ot">&lt;-</span> <span class="cf">function</span> (speciesID){ <span class="co"># find species name use id</span></span>
<span id="cb182-359"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-359" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>( orgInfo[<span class="fu">which</span>(orgInfo<span class="sc">$</span>id <span class="sc">==</span> speciesID),<span class="dv">3</span>]  )</span>
<span id="cb182-360"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-360" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-361"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-362"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-362" aria-hidden="true" tabindex="-1"></a><span class="co"># convert gene IDs to ensembl gene ids and find species</span></span>
<span id="cb182-363"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-363" aria-hidden="true" tabindex="-1"></a>convertID <span class="ot">&lt;-</span> <span class="cf">function</span> (query,selectOrg, selectGO) {</span>
<span id="cb182-364"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-364" aria-hidden="true" tabindex="-1"></a>    querySet <span class="ot">&lt;-</span> <span class="fu">cleanGeneSet</span>( <span class="fu">unlist</span>( <span class="fu">strsplit</span>( <span class="fu">toupper</span>(query),<span class="st">&#39;</span><span class="sc">\t</span><span class="st">| |</span><span class="sc">\n</span><span class="st">|</span><span class="sc">\\</span><span class="st">,&#39;</span>)))</span>
<span id="cb182-365"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-365" aria-hidden="true" tabindex="-1"></a>    <span class="co"># querySet is ensgene data for example, ENSG00000198888, ENSG00000198763, ENSG00000198804</span></span>
<span id="cb182-366"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-366" aria-hidden="true" tabindex="-1"></a>    querySTMT <span class="ot">&lt;-</span> <span class="fu">paste</span>( <span class="st">&quot;select distinct id,ens,species from mapping where id IN (&#39;&quot;</span>, <span class="fu">paste</span>(querySet,<span class="at">collapse=</span><span class="st">&quot;&#39;, &#39;&quot;</span>),<span class="st">&quot;&#39;)&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb182-367"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-367" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>(convert, querySTMT)</span>
<span id="cb182-368"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-368" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>  ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-369"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-369" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(selectOrg <span class="sc">==</span> speciesChoice[[<span class="dv">1</span>]]) {</span>
<span id="cb182-370"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-370" aria-hidden="true" tabindex="-1"></a>        comb <span class="ot">=</span> <span class="fu">paste</span>( result<span class="sc">$</span>species,result<span class="sc">$</span>idType)</span>
<span id="cb182-371"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-371" aria-hidden="true" tabindex="-1"></a>        sortedCounts <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">table</span>(comb),<span class="at">decreasing=</span>T)</span>
<span id="cb182-372"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-372" aria-hidden="true" tabindex="-1"></a>        recognized <span class="ot">=</span><span class="fu">names</span>(sortedCounts[<span class="dv">1</span>])</span>
<span id="cb182-373"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-373" aria-hidden="true" tabindex="-1"></a>        result <span class="ot">&lt;-</span> result[<span class="fu">which</span>(comb <span class="sc">==</span> recognized),]</span>
<span id="cb182-374"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-374" aria-hidden="true" tabindex="-1"></a>        speciesMatched<span class="ot">=</span>sortedCounts</span>
<span id="cb182-375"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-375" aria-hidden="true" tabindex="-1"></a>        <span class="fu">names</span>(speciesMatched )<span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot; .*&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">names</span>(sortedCounts) ) ), findSpeciesByIdName  ) </span>
<span id="cb182-376"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-376" aria-hidden="true" tabindex="-1"></a>        speciesMatched <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>( speciesMatched )</span>
<span id="cb182-377"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-377" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">length</span>(sortedCounts) <span class="sc">==</span> <span class="dv">1</span>) { <span class="co"># if only  one species matched</span></span>
<span id="cb182-378"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-378" aria-hidden="true" tabindex="-1"></a>        speciesMatched[<span class="dv">1</span>,<span class="dv">1</span>] <span class="ot">&lt;-</span><span class="fu">paste</span>( <span class="fu">rownames</span>(speciesMatched), <span class="st">&quot;(&quot;</span>,speciesMatched[<span class="dv">1</span>,<span class="dv">1</span>],<span class="st">&quot;)&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb182-379"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-379" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {<span class="co"># if more than one species matched</span></span>
<span id="cb182-380"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-380" aria-hidden="true" tabindex="-1"></a>            speciesMatched[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(speciesMatched[,<span class="dv">1</span>])</span>
<span id="cb182-381"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-381" aria-hidden="true" tabindex="-1"></a>            speciesMatched[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">paste</span>( speciesMatched[,<span class="dv">1</span>],<span class="st">&quot; (&quot;</span>,speciesMatched[,<span class="dv">2</span>], <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>) </span>
<span id="cb182-382"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-382" aria-hidden="true" tabindex="-1"></a>            speciesMatched[<span class="dv">1</span>,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">paste</span>( speciesMatched[<span class="dv">1</span>,<span class="dv">1</span>],<span class="st">&quot;   ***Used in mapping***  To change, select from above and resubmit query.&quot;</span>)     </span>
<span id="cb182-383"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-383" aria-hidden="true" tabindex="-1"></a>            speciesMatched <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(speciesMatched[,<span class="dv">1</span>])</span>
<span id="cb182-384"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-384" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-385"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-385" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> { <span class="co"># if species is selected</span></span>
<span id="cb182-386"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-386" aria-hidden="true" tabindex="-1"></a>        result <span class="ot">&lt;-</span> result[<span class="fu">which</span>(result<span class="sc">$</span>species <span class="sc">==</span> selectOrg ) ,]</span>
<span id="cb182-387"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-387" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>  ) <span class="fu">return</span>(<span class="cn">NULL</span>) <span class="co">#stop(&quot;ID not recognized!&quot;)</span></span>
<span id="cb182-388"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-388" aria-hidden="true" tabindex="-1"></a>        speciesMatched <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">paste</span>(<span class="st">&quot;Using selected species &quot;</span>, <span class="fu">findSpeciesByIdName</span>(selectOrg) )  )</span>
<span id="cb182-389"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-389" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-390"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-390" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> result[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">duplicated</span>(result[,<span class="dv">2</span>]) ),] <span class="co"># remove duplicates in ensembl_gene_id</span></span>
<span id="cb182-391"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-391" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> result[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">duplicated</span>(result[,<span class="dv">1</span>]) ),] <span class="co"># remove duplicates in user ID</span></span>
<span id="cb182-392"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-392" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(speciesMatched) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Matched Species (genes)&quot;</span> ) </span>
<span id="cb182-393"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-393" aria-hidden="true" tabindex="-1"></a>    conversionTable <span class="ot">&lt;-</span> result[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]; <span class="fu">colnames</span>(conversionTable) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;User_input&quot;</span>,<span class="st">&quot;ensembl_gene_id&quot;</span>)</span>
<span id="cb182-394"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-394" aria-hidden="true" tabindex="-1"></a>    conversionTable<span class="sc">$</span>Species <span class="ot">=</span> <span class="fu">sapply</span>(result[,<span class="dv">3</span>], findSpeciesByIdName )</span>
<span id="cb182-395"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-395" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-396"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-396" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">originalIDs =</span> querySet,<span class="at">IDs=</span><span class="fu">unique</span>( result[,<span class="dv">2</span>]), </span>
<span id="cb182-397"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-397" aria-hidden="true" tabindex="-1"></a>                <span class="at">species =</span> <span class="fu">findSpeciesById</span>(result<span class="sc">$</span>species[<span class="dv">1</span>]), </span>
<span id="cb182-398"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-398" aria-hidden="true" tabindex="-1"></a>                <span class="co">#idType = findIDtypeById(result$idType[1] ),</span></span>
<span id="cb182-399"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-399" aria-hidden="true" tabindex="-1"></a>                <span class="at">speciesMatched =</span> speciesMatched,</span>
<span id="cb182-400"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-400" aria-hidden="true" tabindex="-1"></a>                <span class="at">conversionTable =</span> conversionTable</span>
<span id="cb182-401"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-401" aria-hidden="true" tabindex="-1"></a>                ) )</span>
<span id="cb182-402"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-402" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-403"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-404"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-404" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-405"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-405" aria-hidden="true" tabindex="-1"></a><span class="co"># retrieve detailed info on genes</span></span>
<span id="cb182-406"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-406" aria-hidden="true" tabindex="-1"></a>geneInfo <span class="ot">&lt;-</span> <span class="cf">function</span> (<span class="at">fileName =</span> <span class="cn">NULL</span>){</span>
<span id="cb182-407"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-408"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-408" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(fileName))  <span class="co"># if only one file           #WBGene0000001 some ensembl gene ids in lower case</span></span>
<span id="cb182-409"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-409" aria-hidden="true" tabindex="-1"></a>    { x <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="fu">as.character</span>(fileName) ); x[,<span class="dv">1</span>]<span class="ot">=</span> <span class="fu">toupper</span>(x[,<span class="dv">1</span>]) } <span class="cf">else</span> <span class="co"># read in the chosen file </span></span>
<span id="cb182-410"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-410" aria-hidden="true" tabindex="-1"></a>    { <span class="fu">return</span>(<span class="fu">as.data.frame</span>(<span class="st">&quot;No file.&quot;</span>) )   }</span>
<span id="cb182-411"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-411" aria-hidden="true" tabindex="-1"></a>    Set<span class="ot">=</span><span class="fu">rep</span>(<span class="st">&quot;List&quot;</span>,<span class="fu">nrow</span>(x))</span>
<span id="cb182-412"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-412" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">cbind</span>(x,Set) )</span>
<span id="cb182-413"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-413" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb182-414"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-415"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-415" aria-hidden="true" tabindex="-1"></a>convertedData <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb182-416"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-416" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">is.null</span>(converted.out ) ) <span class="fu">return</span>( readData.out<span class="sc">$</span>data) <span class="co"># if id or species is not recognized use original data.</span></span>
<span id="cb182-417"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-417" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-418"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-418" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(input_noIDConversion) <span class="fu">return</span>( readData.out<span class="sc">$</span>data )</span>
<span id="cb182-419"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-419" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-420"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-420" aria-hidden="true" tabindex="-1"></a>                mapping <span class="ot">&lt;-</span> converted.out<span class="sc">$</span>conversionTable</span>
<span id="cb182-421"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-421" aria-hidden="true" tabindex="-1"></a>                <span class="co"># cat (paste( &quot;\nData:&quot;,input_selectOrg) )</span></span>
<span id="cb182-422"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-422" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">=</span>readData.out<span class="sc">$</span>data</span>
<span id="cb182-423"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-424"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-424" aria-hidden="true" tabindex="-1"></a>                <span class="fu">rownames</span>(x) <span class="ot">=</span> <span class="fu">toupper</span>(<span class="fu">rownames</span>(x))</span>
<span id="cb182-425"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-425" aria-hidden="true" tabindex="-1"></a>                <span class="co"># any gene not recognized by the database is disregarded</span></span>
<span id="cb182-426"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-426" aria-hidden="true" tabindex="-1"></a>                <span class="co"># x1 = merge(mapping[,1:2],x,  by.y = &#39;row.names&#39;, by.x = &#39;User_input&#39;)</span></span>
<span id="cb182-427"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-427" aria-hidden="true" tabindex="-1"></a>                <span class="co"># the 3 lines keeps the unrecogized genes using original IDs</span></span>
<span id="cb182-428"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-428" aria-hidden="true" tabindex="-1"></a>                x1 <span class="ot">=</span> <span class="fu">merge</span>(mapping[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],x,  <span class="at">by.y =</span> <span class="st">&#39;row.names&#39;</span>, <span class="at">by.x =</span> <span class="st">&#39;User_input&#39;</span>, <span class="at">all.y=</span><span class="cn">TRUE</span>)</span>
<span id="cb182-429"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-430"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-430" aria-hidden="true" tabindex="-1"></a>                <span class="co"># original IDs used if ID is not matched in database</span></span>
<span id="cb182-431"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-431" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">is.na</span>(x1[,<span class="dv">2</span>]) )</span>
<span id="cb182-432"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-432" aria-hidden="true" tabindex="-1"></a>                x1[ix,<span class="dv">2</span>] <span class="ot">=</span> x1[ix,<span class="dv">1</span>] </span>
<span id="cb182-433"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-433" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-434"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-434" aria-hidden="true" tabindex="-1"></a>                <span class="co">#multiple matched IDs, use the one with highest SD</span></span>
<span id="cb182-435"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-435" aria-hidden="true" tabindex="-1"></a>                tem <span class="ot">=</span> <span class="fu">apply</span>(x1[,<span class="dv">3</span><span class="sc">:</span>(<span class="fu">dim</span>(x1)[<span class="dv">2</span>])],<span class="dv">1</span>,sd)</span>
<span id="cb182-436"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-436" aria-hidden="true" tabindex="-1"></a>                x1 <span class="ot">=</span> x1[<span class="fu">order</span>(x1[,<span class="dv">2</span>],<span class="sc">-</span>tem),]</span>
<span id="cb182-437"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-437" aria-hidden="true" tabindex="-1"></a>                x1 <span class="ot">=</span> x1[<span class="sc">!</span><span class="fu">duplicated</span>(x1[,<span class="dv">2</span>]) ,]</span>
<span id="cb182-438"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-438" aria-hidden="true" tabindex="-1"></a>                <span class="fu">rownames</span>(x1) <span class="ot">=</span> x1[,<span class="dv">2</span>]</span>
<span id="cb182-439"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-439" aria-hidden="true" tabindex="-1"></a>                x1 <span class="ot">=</span> <span class="fu">as.matrix</span>(x1[,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="sc">-</span><span class="dv">2</span>)])</span>
<span id="cb182-440"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-440" aria-hidden="true" tabindex="-1"></a>                tem <span class="ot">=</span> <span class="fu">apply</span>(x1,<span class="dv">1</span>,sd)</span>
<span id="cb182-441"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-441" aria-hidden="true" tabindex="-1"></a>                x1 <span class="ot">=</span> x1[<span class="fu">order</span>(<span class="sc">-</span>tem),]  <span class="co"># sort again by SD</span></span>
<span id="cb182-442"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-442" aria-hidden="true" tabindex="-1"></a>                <span class="fu">return</span>(x1)</span>
<span id="cb182-443"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-444"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-444" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-445"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-446"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-446" aria-hidden="true" tabindex="-1"></a>convertedCounts <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb182-447"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-447" aria-hidden="true" tabindex="-1"></a>              <span class="cf">if</span>( <span class="fu">is.null</span>(converted.out ) ) <span class="fu">return</span>( readData.out<span class="sc">$</span>rawCounts)</span>
<span id="cb182-448"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-448" aria-hidden="true" tabindex="-1"></a>                mapping <span class="ot">&lt;-</span> converted.out<span class="sc">$</span>conversionTable</span>
<span id="cb182-449"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-450"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-450" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">=</span>readData.out<span class="sc">$</span>rawCounts</span>
<span id="cb182-451"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-451" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-452"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-452" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">rownames</span>(x) <span class="ot">=</span> <span class="fu">toupper</span>(<span class="fu">rownames</span>(x))</span>
<span id="cb182-453"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-453" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># any gene not recognized by the database is disregarded</span></span>
<span id="cb182-454"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-454" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># x1 = merge(mapping[,1:2],x,  by.y = &#39;row.names&#39;, by.x = &#39;User_input&#39;)</span></span>
<span id="cb182-455"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-455" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># the 3 lines keeps the unrecogized genes using original IDs</span></span>
<span id="cb182-456"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-456" aria-hidden="true" tabindex="-1"></a>                    x1 <span class="ot">=</span> <span class="fu">merge</span>(mapping[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],x,  <span class="at">by.y =</span> <span class="st">&#39;row.names&#39;</span>, <span class="at">by.x =</span> <span class="st">&#39;User_input&#39;</span>, <span class="at">all.y=</span><span class="cn">TRUE</span>)</span>
<span id="cb182-457"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-457" aria-hidden="true" tabindex="-1"></a>                    ix <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">is.na</span>(x1[,<span class="dv">2</span>]) )</span>
<span id="cb182-458"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-458" aria-hidden="true" tabindex="-1"></a>                    x1[ix,<span class="dv">2</span>] <span class="ot">=</span> x1[ix,<span class="dv">1</span>] <span class="co"># original IDs used</span></span>
<span id="cb182-459"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-459" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb182-460"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-460" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#multiple matched IDs, use the one with highest SD</span></span>
<span id="cb182-461"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-461" aria-hidden="true" tabindex="-1"></a>                    tem <span class="ot">=</span> <span class="fu">apply</span>(x1[,<span class="dv">3</span><span class="sc">:</span>(<span class="fu">dim</span>(x1)[<span class="dv">2</span>])],<span class="dv">1</span>,sd)</span>
<span id="cb182-462"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-462" aria-hidden="true" tabindex="-1"></a>                    x1 <span class="ot">=</span> x1[<span class="fu">order</span>(x1[,<span class="dv">2</span>],<span class="sc">-</span>tem),]</span>
<span id="cb182-463"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-463" aria-hidden="true" tabindex="-1"></a>                    x1 <span class="ot">=</span> x1[<span class="sc">!</span><span class="fu">duplicated</span>(x1[,<span class="dv">2</span>]) ,]</span>
<span id="cb182-464"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-464" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">rownames</span>(x1) <span class="ot">=</span> x1[,<span class="dv">2</span>]</span>
<span id="cb182-465"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-465" aria-hidden="true" tabindex="-1"></a>                    x1 <span class="ot">=</span> <span class="fu">as.matrix</span>(x1[,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="sc">-</span><span class="dv">2</span>)])</span>
<span id="cb182-466"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-466" aria-hidden="true" tabindex="-1"></a>                    tem <span class="ot">=</span> <span class="fu">apply</span>(x1,<span class="dv">1</span>,sd)</span>
<span id="cb182-467"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-467" aria-hidden="true" tabindex="-1"></a>                    x1 <span class="ot">=</span> x1[<span class="fu">order</span>(<span class="sc">-</span>tem),]  <span class="co"># sort again by SD</span></span>
<span id="cb182-468"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-468" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-469"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-469" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">return</span>(x1)</span>
<span id="cb182-470"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-470" aria-hidden="true" tabindex="-1"></a>} <span class="co"># here</span></span>
<span id="cb182-471"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-472"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-472" aria-hidden="true" tabindex="-1"></a>nGenesFilter <span class="ot">&lt;-</span> <span class="cf">function</span>() { </span>
<span id="cb182-473"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-473" aria-hidden="true" tabindex="-1"></a>        tem <span class="ot">=</span> readData.out<span class="sc">$</span>dataSize</span>
<span id="cb182-474"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-474" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">match</span>( <span class="fu">toupper</span>( <span class="fu">rownames</span>(convertedData.out)), <span class="fu">toupper</span>(converted.out<span class="sc">$</span>conversionTable<span class="sc">$</span>ensembl_gene_id  ) )</span>
<span id="cb182-475"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-475" aria-hidden="true" tabindex="-1"></a>        nMatched <span class="ot">=</span> <span class="fu">sum</span>( <span class="sc">!</span><span class="fu">is.na</span>(ix) )</span>
<span id="cb182-476"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-476" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( input_noIDConversion) </span>
<span id="cb182-477"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-477" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>( <span class="fu">paste</span>(tem[<span class="dv">1</span>], <span class="st">&quot;genes in&quot;</span>, tem[<span class="dv">4</span>], <span class="st">&quot;samples.&quot;</span>, tem[<span class="dv">3</span>], <span class="st">&quot; genes passed filter.</span><span class="sc">\n</span><span class="st"> Original gene IDs used.&quot;</span> ) )  <span class="cf">else</span></span>
<span id="cb182-478"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-478" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>( <span class="fu">paste</span>(tem[<span class="dv">1</span>], <span class="st">&quot;genes in&quot;</span>, tem[<span class="dv">4</span>], <span class="st">&quot;samples. Of the&quot;</span>, tem[<span class="dv">3</span>], <span class="st">&quot; genes passed filter, </span><span class="sc">\n</span><span class="st">&quot;</span>, </span>
<span id="cb182-479"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-479" aria-hidden="true" tabindex="-1"></a>            nMatched, <span class="st">&quot; were converted to Ensembl gene IDs in our database. </span><span class="sc">\n</span></span>
<span id="cb182-480"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-480" aria-hidden="true" tabindex="-1"></a><span class="st">              The remaining &quot;</span>, tem[<span class="dv">3</span>]<span class="sc">-</span>nMatched, <span class="st">&quot;genes were kept in the data using original IDs.&quot;</span> ) )     </span>
<span id="cb182-481"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-481" aria-hidden="true" tabindex="-1"></a>}       </span>
<span id="cb182-482"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-482" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-483"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-483" aria-hidden="true" tabindex="-1"></a><span class="co"># Define sample groups based on column names</span></span>
<span id="cb182-484"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-484" aria-hidden="true" tabindex="-1"></a> detectGroups <span class="ot">&lt;-</span> <span class="cf">function</span> (x){  <span class="co"># x are col names</span></span>
<span id="cb182-485"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-485" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;[0-9]*$&quot;</span>,<span class="st">&quot;&quot;</span>,x) <span class="co"># Remove all numbers from end</span></span>
<span id="cb182-486"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-486" aria-hidden="true" tabindex="-1"></a>    <span class="co">#tem = gsub(&quot;_Rep|_rep|_REP&quot;,&quot;&quot;,tem)</span></span>
<span id="cb182-487"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-487" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_$&quot;</span>,<span class="st">&quot;&quot;</span>,tem); <span class="co"># remove &quot;_&quot; from end</span></span>
<span id="cb182-488"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-488" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_Rep$&quot;</span>,<span class="st">&quot;&quot;</span>,tem); <span class="co"># remove &quot;_Rep&quot; from end</span></span>
<span id="cb182-489"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-489" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_rep$&quot;</span>,<span class="st">&quot;&quot;</span>,tem); <span class="co"># remove &quot;_rep&quot; from end</span></span>
<span id="cb182-490"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-490" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_REP$&quot;</span>,<span class="st">&quot;&quot;</span>,tem)  <span class="co"># remove &quot;_REP&quot; from end</span></span>
<span id="cb182-491"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-491" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( tem )</span>
<span id="cb182-492"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-492" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb182-493"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-493" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-494"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-494" aria-hidden="true" tabindex="-1"></a>  <span class="co"># EDA plots -------------------------------------</span></span>
<span id="cb182-495"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-495" aria-hidden="true" tabindex="-1"></a> densityPlot <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb182-496"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-496" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">2</span>) <span class="co"># seed for random number generator</span></span>
<span id="cb182-497"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-497" aria-hidden="true" tabindex="-1"></a>mycolors <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">rainbow</span>(<span class="dv">20</span>))[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">20</span>,<span class="dv">10</span>,<span class="dv">11</span>,<span class="dv">2</span>,<span class="dv">19</span>,<span class="dv">3</span>,<span class="dv">12</span>,<span class="dv">4</span>,<span class="dv">13</span>,<span class="dv">5</span>,<span class="dv">14</span>,<span class="dv">6</span>,<span class="dv">15</span>,<span class="dv">7</span>,<span class="dv">16</span>,<span class="dv">8</span>,<span class="dv">17</span>,<span class="dv">9</span>,<span class="dv">18</span>)] <span class="co"># 20 colors for kNN clusters</span></span>
<span id="cb182-498"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-498" aria-hidden="true" tabindex="-1"></a><span class="co">#Each row of this matrix represents a color scheme;</span></span>
<span id="cb182-499"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-499" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-500"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-500" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> readData.out<span class="sc">$</span>data</span>
<span id="cb182-501"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-502"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-502" aria-hidden="true" tabindex="-1"></a>    myColors <span class="ot">=</span> mycolors</span>
<span id="cb182-503"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-503" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot</span>(<span class="fu">density</span>(x[,<span class="dv">1</span>]),<span class="at">col =</span> myColors[<span class="dv">1</span>], <span class="at">lwd=</span><span class="dv">2</span>,</span>
<span id="cb182-504"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-504" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab=</span><span class="st">&quot;Expresson values&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;Density&quot;</span>, <span class="at">main=</span> <span class="st">&quot;Distribution of transformed data&quot;</span>,</span>
<span id="cb182-505"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-505" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(<span class="fu">density</span>(x[,<span class="dv">1</span>])<span class="sc">$</span>y)<span class="sc">+</span>.<span class="dv">02</span> ) )</span>
<span id="cb182-506"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-506" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb182-507"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-507" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">dim</span>(x)[<span class="dv">2</span>] )</span>
<span id="cb182-508"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-508" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lines</span>(<span class="fu">density</span>(x[,i]),<span class="at">col=</span>myColors[i], <span class="at">lwd=</span><span class="dv">2</span> )</span>
<span id="cb182-509"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-509" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">dim</span>(x)[<span class="dv">2</span>]<span class="sc">&lt;</span> <span class="dv">31</span> ) <span class="co"># if too many samples do not show legends</span></span>
<span id="cb182-510"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-510" aria-hidden="true" tabindex="-1"></a>        <span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>, <span class="at">cex=</span><span class="dv">1</span>,<span class="fu">colnames</span>(x), <span class="at">lty=</span><span class="fu">rep</span>(<span class="dv">1</span>,<span class="fu">dim</span>(x)[<span class="dv">2</span>]), <span class="at">col=</span>myColors )  </span>
<span id="cb182-511"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-511" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb182-512"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-512" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-513"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-513" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-514"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-514" aria-hidden="true" tabindex="-1"></a> genePlot <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb182-515"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-515" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> convertedData.out</span>
<span id="cb182-516"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-516" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-517"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-517" aria-hidden="true" tabindex="-1"></a>    Symbols <span class="ot">&lt;-</span> <span class="fu">rownames</span>(x)</span>
<span id="cb182-518"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-518" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_selectOrg <span class="sc">!=</span> <span class="st">&quot;NEW&quot;</span>) {</span>
<span id="cb182-519"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-519" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">match</span>( <span class="fu">rownames</span>(x), allGeneInfo.out[,<span class="dv">1</span>])</span>
<span id="cb182-520"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-520" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">sum</span>( <span class="fu">is.na</span>(allGeneInfo.out<span class="sc">$</span>symbol )) <span class="sc">!=</span> <span class="fu">dim</span>(allGeneInfo.out )[<span class="dv">1</span>] ) {  <span class="co"># symbol really exists? </span></span>
<span id="cb182-521"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-521" aria-hidden="true" tabindex="-1"></a>            Symbols <span class="ot">=</span> <span class="fu">as.character</span>( allGeneInfo.out<span class="sc">$</span>symbol[ix] )</span>
<span id="cb182-522"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-522" aria-hidden="true" tabindex="-1"></a>            Symbols[<span class="fu">which</span>( <span class="fu">nchar</span>(Symbols) <span class="sc">&lt;=</span> <span class="dv">2</span> ) ] <span class="ot">&lt;-</span> <span class="fu">rownames</span>(x) [<span class="fu">which</span>( <span class="fu">nchar</span>(Symbols) <span class="sc">&lt;=</span> <span class="dv">2</span> ) ] </span>
<span id="cb182-523"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-523" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-524"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-524" aria-hidden="true" tabindex="-1"></a>       }</span>
<span id="cb182-525"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-525" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">=</span> <span class="fu">as.data.frame</span>(x)</span>
<span id="cb182-526"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-526" aria-hidden="true" tabindex="-1"></a>    x<span class="sc">$</span>Genes <span class="ot">=</span> Symbols</span>
<span id="cb182-527"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-527" aria-hidden="true" tabindex="-1"></a>    <span class="co"># matching from the beginning of symbol</span></span>
<span id="cb182-528"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-528" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">regexpr</span>(  <span class="fu">paste</span>(<span class="st">&quot;^&quot;</span> , <span class="fu">toupper</span>(input_geneSearch),<span class="at">sep=</span><span class="st">&quot;&quot;</span>)   ,<span class="fu">toupper</span>(x<span class="sc">$</span>Genes)) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb182-529"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-529" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-530"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-530" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">grepl</span>(<span class="st">&quot; &quot;</span>, input_geneSearch)  )  <span class="co"># if there is space character, do exact match</span></span>
<span id="cb182-531"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-531" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">match</span>(<span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">toupper</span>(input_geneSearch)),x<span class="sc">$</span>Genes)</span>
<span id="cb182-532"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-532" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-533"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-533" aria-hidden="true" tabindex="-1"></a>    <span class="co"># too few or too many genes found</span></span>
<span id="cb182-534"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-534" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">length</span>(ix) <span class="sc">&gt;</span> <span class="dv">50</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-535"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-535" aria-hidden="true" tabindex="-1"></a>      <span class="co"># no genes found</span></span>
<span id="cb182-536"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-536" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb182-537"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-537" aria-hidden="true" tabindex="-1"></a>    mdf <span class="ot">=</span> <span class="fu">melt</span>(x[ix,],<span class="at">id.vars=</span><span class="st">&quot;Genes&quot;</span>, <span class="at">value.name=</span><span class="st">&quot;value&quot;</span>, <span class="at">variable.name=</span><span class="st">&quot;samples&quot;</span>)</span>
<span id="cb182-538"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-538" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-539"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-539" aria-hidden="true" tabindex="-1"></a>    p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>mdf, <span class="fu">aes</span>(<span class="at">x=</span>samples, <span class="at">y=</span>value, <span class="at">group =</span> Genes, <span class="at">shape=</span>Genes, <span class="at">colour =</span> Genes)) <span class="sc">+</span></span>
<span id="cb182-540"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-540" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb182-541"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-541" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_point</span>( <span class="at">size=</span><span class="dv">5</span>,  <span class="at">fill=</span><span class="st">&quot;white&quot;</span>)<span class="sc">+</span> <span class="co">#shape=21  circle</span></span>
<span id="cb182-542"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-542" aria-hidden="true" tabindex="-1"></a>        <span class="co">#theme(axis.text.x = element_text(size=16,angle = 45, hjust = 1)) +</span></span>
<span id="cb182-543"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-543" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;Transformed expression level&quot;</span>) <span class="sc">+</span></span>
<span id="cb182-544"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-544" aria-hidden="true" tabindex="-1"></a>        <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(mdf<span class="sc">$</span>value)))</span>
<span id="cb182-545"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-545" aria-hidden="true" tabindex="-1"></a>    p1 <span class="ot">&lt;-</span> p1 <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>,<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="co"># theme(aspect.ratio=1) +</span></span>
<span id="cb182-546"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-546" aria-hidden="true" tabindex="-1"></a>     <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">size =</span> <span class="dv">16</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb182-547"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-547" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.text.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb182-548"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-548" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb182-549"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-549" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>) ) <span class="sc">+</span></span>
<span id="cb182-550"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-550" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">12</span>))    </span>
<span id="cb182-551"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-551" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-552"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-552" aria-hidden="true" tabindex="-1"></a>    p1  </span>
<span id="cb182-553"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-553" aria-hidden="true" tabindex="-1"></a>    }   </span>
<span id="cb182-554"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-554" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-555"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-555" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-556"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-556" aria-hidden="true" tabindex="-1"></a>geneBarPlotError <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb182-557"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-557" aria-hidden="true" tabindex="-1"></a>   x <span class="ot">&lt;-</span> convertedData.out</span>
<span id="cb182-558"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-558" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-559"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-559" aria-hidden="true" tabindex="-1"></a>    Symbols <span class="ot">&lt;-</span> <span class="fu">rownames</span>(x)</span>
<span id="cb182-560"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-560" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_selectOrg <span class="sc">!=</span> <span class="st">&quot;NEW&quot;</span>) {</span>
<span id="cb182-561"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-561" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">match</span>( <span class="fu">rownames</span>(x), allGeneInfo.out[,<span class="dv">1</span>])</span>
<span id="cb182-562"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-562" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">sum</span>( <span class="fu">is.na</span>(allGeneInfo.out<span class="sc">$</span>symbol )) <span class="sc">!=</span> <span class="fu">dim</span>(allGeneInfo.out )[<span class="dv">1</span>] ) {  <span class="co"># symbol really exists? </span></span>
<span id="cb182-563"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-563" aria-hidden="true" tabindex="-1"></a>            Symbols <span class="ot">=</span> <span class="fu">as.character</span>( allGeneInfo.out<span class="sc">$</span>symbol[ix] )</span>
<span id="cb182-564"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-564" aria-hidden="true" tabindex="-1"></a>            Symbols[<span class="fu">which</span>( <span class="fu">nchar</span>(Symbols) <span class="sc">&lt;=</span> <span class="dv">2</span> ) ] <span class="ot">&lt;-</span> <span class="fu">rownames</span>(x) [<span class="fu">which</span>( <span class="fu">nchar</span>(Symbols) <span class="sc">&lt;=</span> <span class="dv">2</span> ) ] </span>
<span id="cb182-565"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-565" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-566"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-566" aria-hidden="true" tabindex="-1"></a>       }</span>
<span id="cb182-567"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-567" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">=</span> <span class="fu">as.data.frame</span>(x)</span>
<span id="cb182-568"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-568" aria-hidden="true" tabindex="-1"></a>    x<span class="sc">$</span>Genes <span class="ot">=</span> Symbols</span>
<span id="cb182-569"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-569" aria-hidden="true" tabindex="-1"></a>    <span class="co">#write.csv(x,&quot;tem.csv&quot;)</span></span>
<span id="cb182-570"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-570" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Search for genes</span></span>
<span id="cb182-571"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-571" aria-hidden="true" tabindex="-1"></a>    <span class="co">#ix = grep(&quot;HOXA&quot;,toupper(x$Genes) )</span></span>
<span id="cb182-572"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-572" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ix = grep(toupper(input_geneSearch),toupper(x$Genes))  # sox --&gt; Tsox  </span></span>
<span id="cb182-573"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-573" aria-hidden="true" tabindex="-1"></a>    <span class="co"># matching from the beginning of symbol</span></span>
<span id="cb182-574"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-574" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">regexpr</span>(  <span class="fu">paste</span>(<span class="st">&quot;^&quot;</span> , <span class="fu">toupper</span>(input_geneSearch),<span class="at">sep=</span><span class="st">&quot;&quot;</span>)   ,<span class="fu">toupper</span>(x<span class="sc">$</span>Genes)) <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb182-575"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-575" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-576"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-576" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">grepl</span>(<span class="st">&quot; &quot;</span>, input_geneSearch)  )  <span class="co"># if there is space character, do exact match</span></span>
<span id="cb182-577"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-577" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">match</span>(<span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">toupper</span>(input_geneSearch)),x<span class="sc">$</span>Genes)</span>
<span id="cb182-578"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-578" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-579"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-579" aria-hidden="true" tabindex="-1"></a>    <span class="co"># too few or too many genes found</span></span>
<span id="cb182-580"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-580" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">length</span>(ix) <span class="sc">&gt;</span> <span class="dv">50</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-581"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-581" aria-hidden="true" tabindex="-1"></a>      <span class="co"># no genes found</span></span>
<span id="cb182-582"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-582" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb182-583"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-583" aria-hidden="true" tabindex="-1"></a>    mdf <span class="ot">=</span> <span class="fu">melt</span>(x[ix,],<span class="at">id.vars=</span><span class="st">&quot;Genes&quot;</span>, <span class="at">value.name=</span><span class="st">&quot;value&quot;</span>, <span class="at">variable.name=</span><span class="st">&quot;samples&quot;</span>)</span>
<span id="cb182-584"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-584" aria-hidden="true" tabindex="-1"></a>    mdf<span class="sc">$</span>count <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb182-585"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-585" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">=</span> <span class="fu">detectGroups</span>(mdf<span class="sc">$</span>samples)</span>
<span id="cb182-586"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-586" aria-hidden="true" tabindex="-1"></a>    Means <span class="ot">=</span> <span class="fu">aggregate</span>(mdf<span class="sc">$</span>value,<span class="at">by=</span><span class="fu">list</span>( g, mdf<span class="sc">$</span>Genes ), <span class="at">FUN =</span> mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>  )</span>
<span id="cb182-587"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-587" aria-hidden="true" tabindex="-1"></a>    SDs <span class="ot">=</span> <span class="fu">aggregate</span>(mdf<span class="sc">$</span>value,<span class="at">by=</span><span class="fu">list</span>( g, mdf<span class="sc">$</span>Genes ), <span class="at">FUN =</span> sd, <span class="at">na.rm=</span><span class="cn">TRUE</span>  )</span>
<span id="cb182-588"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-588" aria-hidden="true" tabindex="-1"></a>    Ns <span class="ot">=</span> <span class="fu">aggregate</span>(mdf<span class="sc">$</span>count, <span class="at">by=</span> <span class="fu">list</span>(g, mdf<span class="sc">$</span>Genes) , <span class="at">FUN =</span> sum  )</span>
<span id="cb182-589"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-589" aria-hidden="true" tabindex="-1"></a>    summarized <span class="ot">=</span> <span class="fu">cbind</span>(Means,SDs[,<span class="dv">3</span>],Ns[,<span class="dv">3</span>])</span>
<span id="cb182-590"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-590" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(summarized)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Samples&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;Mean&quot;</span>,<span class="st">&quot;SD&quot;</span>,<span class="st">&quot;N&quot;</span>)</span>
<span id="cb182-591"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-591" aria-hidden="true" tabindex="-1"></a>    summarized<span class="sc">$</span>SE <span class="ot">=</span> summarized<span class="sc">$</span>SD <span class="sc">/</span> <span class="fu">sqrt</span>(summarized<span class="sc">$</span>N)  </span>
<span id="cb182-592"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-592" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-593"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-593" aria-hidden="true" tabindex="-1"></a>    <span class="co">#http://www.sthda.com/english/wiki/ggplot2-barplots-quick-start-guide-r-software-and-data-visualization</span></span>
<span id="cb182-594"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-594" aria-hidden="true" tabindex="-1"></a>    p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(summarized, <span class="fu">aes</span>(<span class="at">x=</span>Genes, <span class="at">y=</span>Mean,<span class="at">fill=</span>Samples) ) <span class="sc">+</span> <span class="co"># data &amp; aesthetic mapping</span></span>
<span id="cb182-595"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-595" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">position=</span><span class="fu">position_dodge</span>()) <span class="sc">+</span> <span class="co"># bars represent average</span></span>
<span id="cb182-596"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-596" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>Mean<span class="sc">-</span>SE, <span class="at">ymax=</span>Mean<span class="sc">+</span>SE), <span class="at">width=</span><span class="fl">0.2</span>,<span class="at">position=</span><span class="fu">position_dodge</span>(.<span class="dv">9</span>)) <span class="sc">+</span></span>
<span id="cb182-597"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-597" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;Expression Level&quot;</span>) </span>
<span id="cb182-598"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-598" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_useSD <span class="sc">==</span> <span class="dv">1</span>) { </span>
<span id="cb182-599"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-599" aria-hidden="true" tabindex="-1"></a>    p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(summarized, <span class="fu">aes</span>(<span class="at">x=</span>Genes, <span class="at">y=</span>Mean,<span class="at">fill=</span>Samples) ) <span class="sc">+</span> <span class="co"># data &amp; aesthetic mapping</span></span>
<span id="cb182-600"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-600" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">position=</span><span class="fu">position_dodge</span>()) <span class="sc">+</span> <span class="co"># bars represent average</span></span>
<span id="cb182-601"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-601" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>Mean<span class="sc">-</span>SD, <span class="at">ymax=</span>Mean<span class="sc">+</span>SD), <span class="at">width=</span><span class="fl">0.2</span>,<span class="at">position=</span><span class="fu">position_dodge</span>(.<span class="dv">9</span>)) <span class="sc">+</span></span>
<span id="cb182-602"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-602" aria-hidden="true" tabindex="-1"></a>        <span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;Expression Level&quot;</span>) </span>
<span id="cb182-603"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-603" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-604"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-604" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-605"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-605" aria-hidden="true" tabindex="-1"></a>    p2 <span class="ot">&lt;-</span> p2 <span class="sc">+</span>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>,<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="co"># theme(aspect.ratio=1) +</span></span>
<span id="cb182-606"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-606" aria-hidden="true" tabindex="-1"></a>     <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">size =</span> <span class="dv">16</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb182-607"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-607" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.text.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb182-608"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-608" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb182-609"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-609" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>) ) <span class="sc">+</span></span>
<span id="cb182-610"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-610" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span>
<span id="cb182-611"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-611" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-612"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-612" aria-hidden="true" tabindex="-1"></a>    p2</span>
<span id="cb182-613"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-613" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-614"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-614" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-615"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-615" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-616"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-616" aria-hidden="true" tabindex="-1"></a><span class="co"># Hierarchical clustering</span></span>
<span id="cb182-617"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-617" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-618"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-618" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-619"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-619" aria-hidden="true" tabindex="-1"></a><span class="co"># Functions for hierarchical clustering</span></span>
<span id="cb182-620"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-620" aria-hidden="true" tabindex="-1"></a>hclust2 <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">method=</span><span class="st">&quot;average&quot;</span>, ...)  <span class="co"># average linkage</span></span>
<span id="cb182-621"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-621" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(x, <span class="at">method=</span>method, ...)</span>
<span id="cb182-622"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-622" aria-hidden="true" tabindex="-1"></a>hclust.ward.D <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">method=</span><span class="st">&quot;ward.D&quot;</span>, ...)  <span class="co"># average linkage</span></span>
<span id="cb182-623"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-623" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(x, <span class="at">method=</span>method, ...)</span>
<span id="cb182-624"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-624" aria-hidden="true" tabindex="-1"></a>hclust.ward.D2 <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">method=</span><span class="st">&quot;ward.D2&quot;</span>, ...)  <span class="co"># average linkage</span></span>
<span id="cb182-625"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-625" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(x, <span class="at">method=</span>method, ...)</span>
<span id="cb182-626"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-626" aria-hidden="true" tabindex="-1"></a>hclust.single <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">method=</span><span class="st">&quot;single&quot;</span>, ...)  <span class="co"># average linkage</span></span>
<span id="cb182-627"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-627" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(x, <span class="at">method=</span>method, ...)</span>
<span id="cb182-628"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-628" aria-hidden="true" tabindex="-1"></a>hclust.mcquitty <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">method=</span><span class="st">&quot;mcquitty&quot;</span>, ...)  <span class="co"># average linkage</span></span>
<span id="cb182-629"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-629" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(x, <span class="at">method=</span>method, ...)</span>
<span id="cb182-630"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-630" aria-hidden="true" tabindex="-1"></a>hclust.median <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">method=</span><span class="st">&quot;median&quot;</span>, ...)  <span class="co"># average linkage</span></span>
<span id="cb182-631"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-631" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(x, <span class="at">method=</span>method, ...)</span>
<span id="cb182-632"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-632" aria-hidden="true" tabindex="-1"></a>hclust.centroid <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">method=</span><span class="st">&quot;centroid&quot;</span>, ...)  <span class="co"># average linkage</span></span>
<span id="cb182-633"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-633" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(x, <span class="at">method=</span>method, ...)</span>
<span id="cb182-634"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-634" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb182-635"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-635" aria-hidden="true" tabindex="-1"></a>hclustFuns <span class="ot">&lt;-</span> <span class="fu">list</span>( <span class="at">averge =</span> hclust2, <span class="at">complete=</span>hclust, <span class="at">single=</span>hclust.single,</span>
<span id="cb182-636"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-636" aria-hidden="true" tabindex="-1"></a>                    <span class="at">median=</span>hclust.median, <span class="at">centroid=</span>hclust.centroid, <span class="at">mcquitty=</span>hclust.mcquitty)</span>
<span id="cb182-637"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-637" aria-hidden="true" tabindex="-1"></a>hclustChoices <span class="ot">=</span> <span class="fu">setNames</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(hclustFuns),<span class="fu">names</span>(hclustFuns)) <span class="co"># for pull down menu</span></span>
<span id="cb182-638"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-638" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-639"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-639" aria-hidden="true" tabindex="-1"></a>dist2 <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ...)   <span class="co"># distance function = 1-PCC (Pearson&#39;s correlation coefficient)</span></span>
<span id="cb182-640"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-640" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span><span class="fu">cor</span>(<span class="fu">t</span>(x), <span class="at">method=</span><span class="st">&quot;pearson&quot;</span>))</span>
<span id="cb182-641"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-641" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb182-642"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-642" aria-hidden="true" tabindex="-1"></a>dist3 <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ...)   <span class="co"># distance function = 1-abs(PCC) (Pearson&#39;s correlation coefficient)</span></span>
<span id="cb182-643"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-643" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span><span class="fu">abs</span>(<span class="fu">cor</span>(<span class="fu">t</span>(x), <span class="at">method=</span><span class="st">&quot;pearson&quot;</span>)))  </span>
<span id="cb182-644"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-644" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb182-645"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-645" aria-hidden="true" tabindex="-1"></a> distFuns <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">Correlation=</span>dist2, <span class="at">Euclidean=</span>dist,<span class="at">AbsolutePCC=</span>dist3) </span>
<span id="cb182-646"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-646" aria-hidden="true" tabindex="-1"></a>  <span class="co"># hierarchical clustering tree ----------------------------------------</span></span>
<span id="cb182-647"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-647" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-648"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-648" aria-hidden="true" tabindex="-1"></a>hmcols <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&quot;#D73027&quot;</span>, <span class="st">&quot;#FC8D59&quot;</span>, <span class="st">&quot;#FEE090&quot;</span>, <span class="st">&quot;#FFFFBF&quot;</span>,</span>
<span id="cb182-649"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-649" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;#E0F3F8&quot;</span>, <span class="st">&quot;#91BFDB&quot;</span>, <span class="st">&quot;#4575B4&quot;</span>)))(<span class="dv">75</span>)</span>
<span id="cb182-650"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-650" aria-hidden="true" tabindex="-1"></a>heatColors <span class="ot">=</span> <span class="fu">rbind</span>(      <span class="fu">greenred</span>(<span class="dv">75</span>),     <span class="fu">bluered</span>(<span class="dv">75</span>),     <span class="fu">colorpanel</span>(<span class="dv">75</span>,<span class="st">&quot;green&quot;</span>,<span class="st">&quot;black&quot;</span>,<span class="st">&quot;magenta&quot;</span>),<span class="fu">colorpanel</span>(<span class="dv">75</span>,<span class="st">&quot;blue&quot;</span>,<span class="st">&quot;yellow&quot;</span>,<span class="st">&quot;red&quot;</span>),hmcols )</span>
<span id="cb182-651"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-651" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heatColors) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Green-Black-Red&quot;</span>,<span class="st">&quot;Blue-White-Red&quot;</span>,<span class="st">&quot;Green-Black-Magenta&quot;</span>,<span class="st">&quot;Blue-Yellow-Red&quot;</span>,<span class="st">&quot;Blue-white-brown&quot;</span>)</span>
<span id="cb182-652"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-652" aria-hidden="true" tabindex="-1"></a>colorChoices <span class="ot">=</span> <span class="fu">setNames</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(heatColors)[<span class="dv">1</span>],<span class="fu">rownames</span>(heatColors)) <span class="co"># for pull down menu</span></span>
<span id="cb182-653"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-653" aria-hidden="true" tabindex="-1"></a><span class="co"># adding sample legends to heatmap; this is for the main heatmap</span></span>
<span id="cb182-654"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-654" aria-hidden="true" tabindex="-1"></a><span class="co"># https://stackoverflow.com/questions/3932038/plot-a-legend-outside-of-the-plotting-area-in-base-graphics</span></span>
<span id="cb182-655"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-655" aria-hidden="true" tabindex="-1"></a>add_legend <span class="ot">&lt;-</span> <span class="cf">function</span>(...) {</span>
<span id="cb182-656"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-656" aria-hidden="true" tabindex="-1"></a>  opar <span class="ot">&lt;-</span> <span class="fu">par</span>(<span class="at">fig=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>), <span class="at">oma=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), </span>
<span id="cb182-657"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-657" aria-hidden="true" tabindex="-1"></a>    <span class="at">mar=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">6</span>), <span class="at">new=</span><span class="cn">TRUE</span>)</span>
<span id="cb182-658"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-658" aria-hidden="true" tabindex="-1"></a>  <span class="fu">on.exit</span>(<span class="fu">par</span>(opar))</span>
<span id="cb182-659"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-659" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="at">type=</span><span class="st">&#39;n&#39;</span>, <span class="at">bty=</span><span class="st">&#39;n&#39;</span>, <span class="at">xaxt=</span><span class="st">&#39;n&#39;</span>, <span class="at">yaxt=</span><span class="st">&#39;n&#39;</span>)</span>
<span id="cb182-660"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-660" aria-hidden="true" tabindex="-1"></a>  <span class="fu">legend</span>(...)</span>
<span id="cb182-661"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-661" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-662"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-662" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-663"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-663" aria-hidden="true" tabindex="-1"></a>correlationMatrix <span class="ot">&lt;-</span> <span class="cf">function</span>( ) {</span>
<span id="cb182-664"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-664" aria-hidden="true" tabindex="-1"></a>    <span class="co"># heatmap of correlation matrix</span></span>
<span id="cb182-665"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-665" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> readData.out<span class="sc">$</span>data</span>
<span id="cb182-666"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-666" aria-hidden="true" tabindex="-1"></a>    maxGene <span class="ot">&lt;-</span> <span class="fu">apply</span>(x,<span class="dv">1</span>,max)</span>
<span id="cb182-667"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-667" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> x[<span class="fu">which</span>(maxGene <span class="sc">&gt;</span> <span class="fu">quantile</span>(maxGene)[<span class="dv">1</span>] ) ,] <span class="co"># remove bottom 25% lowly expressed genes, which inflate the PPC</span></span>
<span id="cb182-668"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-668" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-669"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-669" aria-hidden="true" tabindex="-1"></a>       melted_cormat <span class="ot">&lt;-</span> <span class="fu">melt</span>(<span class="fu">round</span>(<span class="fu">cor</span>(x),<span class="dv">2</span>), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb182-670"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-670" aria-hidden="true" tabindex="-1"></a>    <span class="co"># melted_cormat &lt;- melted_cormat[which(melted_cormat[,1] != melted_cormat[,2] ) , ]</span></span>
<span id="cb182-671"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-671" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Create a ggheatmap</span></span>
<span id="cb182-672"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-672" aria-hidden="true" tabindex="-1"></a>        ggheatmap <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(melted_cormat, <span class="fu">aes</span>(Var2, Var1, <span class="at">fill =</span> value))<span class="sc">+</span></span>
<span id="cb182-673"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-673" aria-hidden="true" tabindex="-1"></a>         <span class="fu">geom_tile</span>(<span class="at">color =</span> <span class="st">&quot;white&quot;</span>)<span class="sc">+</span></span>
<span id="cb182-674"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-674" aria-hidden="true" tabindex="-1"></a>         <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">&quot;green&quot;</span>, <span class="at">high =</span> <span class="st">&quot;red&quot;</span>,  <span class="at">mid =</span> <span class="st">&quot;white&quot;</span>, </span>
<span id="cb182-675"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-675" aria-hidden="true" tabindex="-1"></a>            <span class="at">space =</span> <span class="st">&quot;Lab&quot;</span>,  <span class="at">limit =</span> <span class="fu">c</span>(<span class="fu">min</span>(melted_cormat[,<span class="dv">3</span>]) ,<span class="fu">max</span>(melted_cormat[,<span class="dv">3</span>])), <span class="at">midpoint =</span> <span class="fu">median</span>(melted_cormat[,<span class="dv">3</span>]),</span>
<span id="cb182-676"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-676" aria-hidden="true" tabindex="-1"></a>            <span class="at">name=</span><span class="st">&quot;Pearson</span><span class="sc">\n</span><span class="st">Correlation&quot;</span></span>
<span id="cb182-677"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-677" aria-hidden="true" tabindex="-1"></a>          ) <span class="sc">+</span></span>
<span id="cb182-678"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-678" aria-hidden="true" tabindex="-1"></a>         <span class="fu">theme_minimal</span>()<span class="sc">+</span> <span class="co"># minimal theme</span></span>
<span id="cb182-679"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-679" aria-hidden="true" tabindex="-1"></a>         <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, </span>
<span id="cb182-680"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-680" aria-hidden="true" tabindex="-1"></a>            <span class="at">size =</span> <span class="dv">15</span>, <span class="at">hjust =</span> <span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb182-681"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-681" aria-hidden="true" tabindex="-1"></a>         <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>( </span>
<span id="cb182-682"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-682" aria-hidden="true" tabindex="-1"></a>            <span class="at">size =</span> <span class="dv">15</span>))<span class="sc">+</span></span>
<span id="cb182-683"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-683" aria-hidden="true" tabindex="-1"></a>         <span class="fu">coord_fixed</span>()<span class="sc">+</span></span>
<span id="cb182-684"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-684" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme</span>(</span>
<span id="cb182-685"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-685" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb182-686"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-686" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb182-687"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-687" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb182-688"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-688" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb182-689"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-689" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb182-690"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-690" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb182-691"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-691" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">0</span>),</span>
<span id="cb182-692"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-692" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="fl">0.7</span>),</span>
<span id="cb182-693"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-693" aria-hidden="true" tabindex="-1"></a>         <span class="at">legend.direction =</span> <span class="st">&quot;horizontal&quot;</span>)<span class="sc">+</span></span>
<span id="cb182-694"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-694" aria-hidden="true" tabindex="-1"></a>         <span class="fu">guides</span>(<span class="at">fill =</span> <span class="cn">FALSE</span>) <span class="co"># + ggtitle(&quot;Pearson&#39;s Correlation Coefficient (all genes)&quot;)</span></span>
<span id="cb182-695"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-695" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb182-696"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-696" aria-hidden="true" tabindex="-1"></a>         <span class="cf">if</span>(input_labelPCC <span class="sc">&amp;&amp;</span> <span class="fu">ncol</span>(x)<span class="sc">&lt;</span><span class="dv">30</span>)</span>
<span id="cb182-697"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-697" aria-hidden="true" tabindex="-1"></a>                ggheatmap <span class="ot">&lt;-</span> ggheatmap <span class="sc">+</span>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(Var2, Var1, <span class="at">label =</span> value), <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">4</span>)</span>
<span id="cb182-698"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-698" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-699"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-699" aria-hidden="true" tabindex="-1"></a>        ggheatmap</span>
<span id="cb182-700"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-700" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb182-701"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-701" aria-hidden="true" tabindex="-1"></a>}        </span>
<span id="cb182-702"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-702" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-703"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-703" aria-hidden="true" tabindex="-1"></a>staticHeatmap <span class="ot">&lt;-</span> <span class="cf">function</span> () { </span>
<span id="cb182-704"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-704" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> readData.out<span class="sc">$</span>data   <span class="co"># x = read.csv(&quot;expression1.csv&quot;)</span></span>
<span id="cb182-705"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-705" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-706"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-706" aria-hidden="true" tabindex="-1"></a>    n<span class="ot">=</span>input_nGenes</span>
<span id="cb182-707"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-707" aria-hidden="true" tabindex="-1"></a>    <span class="co">#if(n&gt;6000) n = 6000 # max</span></span>
<span id="cb182-708"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-708" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(n<span class="sc">&gt;</span><span class="fu">dim</span>(x)[<span class="dv">1</span>]) n <span class="ot">=</span> <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="co"># max as data</span></span>
<span id="cb182-709"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-709" aria-hidden="true" tabindex="-1"></a>    <span class="co"># this will cutoff very large values, which could skew the color </span></span>
<span id="cb182-710"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-710" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_geneCentering)</span>
<span id="cb182-711"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-711" aria-hidden="true" tabindex="-1"></a>        x<span class="ot">=</span><span class="fu">as.matrix</span>(x[<span class="dv">1</span><span class="sc">:</span>n,])<span class="sc">-</span><span class="fu">apply</span>(x[<span class="dv">1</span><span class="sc">:</span>n,],<span class="dv">1</span>,mean)</span>
<span id="cb182-712"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-712" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-713"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-713" aria-hidden="true" tabindex="-1"></a>    <span class="co"># standardize by gene</span></span>
<span id="cb182-714"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-714" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_geneNormalize) </span>
<span id="cb182-715"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-715" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> x <span class="sc">/</span> <span class="fu">apply</span>(x,<span class="dv">1</span>,sd)</span>
<span id="cb182-716"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-716" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-717"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-717" aria-hidden="true" tabindex="-1"></a>    <span class="co"># row centering and normalize</span></span>
<span id="cb182-718"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-718" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">scale</span>(x, <span class="at">center =</span> input_sampleCentering, <span class="at">scale =</span> input_sampleNormalize) </span>
<span id="cb182-719"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-719" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-720"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-720" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-721"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-721" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-722"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-722" aria-hidden="true" tabindex="-1"></a>    cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">+</span> input_heatmapCutoff <span class="sc">*</span> <span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb182-723"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-723" aria-hidden="true" tabindex="-1"></a>    x[x<span class="sc">&gt;</span>cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb182-724"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-724" aria-hidden="true" tabindex="-1"></a>    cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">-</span> input_heatmapCutoff <span class="sc">*</span><span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb182-725"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-725" aria-hidden="true" tabindex="-1"></a>    x[x<span class="sc">&lt;</span> cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb182-726"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-726" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-727"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-727" aria-hidden="true" tabindex="-1"></a>    groups <span class="ot">=</span> <span class="fu">detectGroups</span>(<span class="fu">colnames</span>(x) )</span>
<span id="cb182-728"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-728" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if sample info file is uploaded us that info:</span></span>
<span id="cb182-729"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-729" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-730"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-730" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(readSampleInfo.out) <span class="sc">&amp;&amp;</span>  <span class="sc">!</span><span class="fu">is.null</span>(input_selectFactorsHeatmap) ) { </span>
<span id="cb182-731"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-731" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(input_selectFactorsHeatmap <span class="sc">==</span> <span class="st">&quot;Sample_Name&quot;</span> )</span>
<span id="cb182-732"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-732" aria-hidden="true" tabindex="-1"></a>            groups <span class="ot">=</span> <span class="fu">detectGroups</span>(<span class="fu">colnames</span>(x) ) <span class="cf">else</span> </span>
<span id="cb182-733"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-733" aria-hidden="true" tabindex="-1"></a>            {   ix <span class="ot">=</span> <span class="fu">match</span>(input_selectFactorsHeatmap, <span class="fu">colnames</span>(readSampleInfo.out ) ) </span>
<span id="cb182-734"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-734" aria-hidden="true" tabindex="-1"></a>                groups <span class="ot">=</span> readSampleInfo.out[,ix]</span>
<span id="cb182-735"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-735" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-736"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-736" aria-hidden="true" tabindex="-1"></a>    }   </span>
<span id="cb182-737"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-737" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-738"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-738" aria-hidden="true" tabindex="-1"></a>    groups.colors <span class="ot">=</span> <span class="fu">rainbow</span>(<span class="fu">length</span>(<span class="fu">unique</span>(groups) ) )</span>
<span id="cb182-739"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-739" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-740"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-740" aria-hidden="true" tabindex="-1"></a>    <span class="co">#http://stackoverflow.com/questions/15351575/moving-color-key-in-r-heatmap-2-function-of-gplots-package</span></span>
<span id="cb182-741"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-741" aria-hidden="true" tabindex="-1"></a>    lmat <span class="ot">=</span> <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">4</span>),<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">2</span>),<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">0</span>))</span>
<span id="cb182-742"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-742" aria-hidden="true" tabindex="-1"></a>    lwid <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">6</span>) <span class="co"># width of gene tree; width of heatmap</span></span>
<span id="cb182-743"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-743" aria-hidden="true" tabindex="-1"></a>    lhei <span class="ot">=</span> <span class="fu">c</span>(<span class="fl">1.5</span>,.<span class="dv">2</span>,<span class="dv">8</span>,<span class="fl">1.1</span>)</span>
<span id="cb182-744"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-744" aria-hidden="true" tabindex="-1"></a>    <span class="co">#layout matrix</span></span>
<span id="cb182-745"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-745" aria-hidden="true" tabindex="-1"></a>    <span class="co">#        [,1] [,2]</span></span>
<span id="cb182-746"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-746" aria-hidden="true" tabindex="-1"></a>    <span class="co">#   [1,]    0    4</span></span>
<span id="cb182-747"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-747" aria-hidden="true" tabindex="-1"></a>    <span class="co">#   [2,]    0    1</span></span>
<span id="cb182-748"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-748" aria-hidden="true" tabindex="-1"></a>    <span class="co">#   [3,]    3    2</span></span>
<span id="cb182-749"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-749" aria-hidden="true" tabindex="-1"></a>    <span class="co">#   [4,]    5    0</span></span>
<span id="cb182-750"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-750" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 4--&gt; column tree; 1--&gt; column color bar; 2--&gt; heatmap; 3-&gt; row tree; 5--&gt; color key.</span></span>
<span id="cb182-751"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-751" aria-hidden="true" tabindex="-1"></a>    <span class="co"># height of 4 rows is specified by lhei; width of columns is given by lwid</span></span>
<span id="cb182-752"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-752" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-753"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-753" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-754"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-754" aria-hidden="true" tabindex="-1"></a>    <span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">12</span>, <span class="dv">4</span>, <span class="fl">1.4</span>, <span class="fl">0.2</span>))</span>
<span id="cb182-755"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-755" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-756"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-756" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-757"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-757" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( n<span class="sc">&gt;</span><span class="dv">110</span>) </span>
<span id="cb182-758"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-758" aria-hidden="true" tabindex="-1"></a>    <span class="fu">heatmap.2</span>(x, <span class="at">distfun =</span> distFuns[[<span class="fu">as.integer</span>(input_distFunctions)]]</span>
<span id="cb182-759"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-759" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">hclustfun=</span>hclustFuns[[<span class="fu">as.integer</span>(input_hclustFunctions)]]</span>
<span id="cb182-760"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-760" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">Colv=</span><span class="sc">!</span>input_noSampleClustering</span>
<span id="cb182-761"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-761" aria-hidden="true" tabindex="-1"></a>        <span class="co">#col=colorpanel(75,&quot;green&quot;,&quot;black&quot;,&quot;magenta&quot;)  ,</span></span>
<span id="cb182-762"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-762" aria-hidden="true" tabindex="-1"></a>        <span class="co">#col=bluered(75),</span></span>
<span id="cb182-763"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-763" aria-hidden="true" tabindex="-1"></a>        <span class="co">#col=greenred(75), </span></span>
<span id="cb182-764"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-764" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">col=</span> heatColors[<span class="fu">as.integer</span>(input_heatColors1),]</span>
<span id="cb182-765"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-765" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">density.info=</span><span class="st">&quot;none&quot;</span>, <span class="at">trace=</span><span class="st">&quot;none&quot;</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">keysize=</span>.<span class="dv">5</span></span>
<span id="cb182-766"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-766" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">key=</span>T, <span class="at">symkey=</span>F</span>
<span id="cb182-767"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-767" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">ColSideColors=</span>groups.colors[ <span class="fu">as.factor</span>(groups)]</span>
<span id="cb182-768"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-768" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">labRow=</span><span class="st">&quot;&quot;</span></span>
<span id="cb182-769"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-769" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">margins=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">2</span>)</span>
<span id="cb182-770"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-770" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">srtCol=</span><span class="dv">45</span></span>
<span id="cb182-771"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-771" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">cexCol=</span><span class="dv">2</span>  <span class="co"># size of font for sample names</span></span>
<span id="cb182-772"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-772" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">lmat =</span> lmat, <span class="at">lwid =</span> lwid, <span class="at">lhei =</span> lhei</span>
<span id="cb182-773"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-773" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb182-774"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-774" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-775"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-775" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( n<span class="sc">&lt;=</span><span class="dv">110</span>) </span>
<span id="cb182-776"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-776" aria-hidden="true" tabindex="-1"></a>    <span class="fu">heatmap.2</span>(x, <span class="at">distfun =</span>  distFuns[[<span class="fu">as.integer</span>(input_distFunctions)]]</span>
<span id="cb182-777"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-777" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">hclustfun=</span>hclustFuns[[<span class="fu">as.integer</span>(input_hclustFunctions)]]</span>
<span id="cb182-778"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-778" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">Colv=</span><span class="sc">!</span>input_noSampleClustering     </span>
<span id="cb182-779"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-779" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">col=</span> heatColors[<span class="fu">as.integer</span>(input_heatColors1),], <span class="at">density.info=</span><span class="st">&quot;none&quot;</span>, <span class="at">trace=</span><span class="st">&quot;none&quot;</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">keysize=</span>.<span class="dv">5</span></span>
<span id="cb182-780"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-780" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">key=</span>T, <span class="at">symkey=</span>F</span>
<span id="cb182-781"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-781" aria-hidden="true" tabindex="-1"></a>        <span class="co">#,labRow=labRow</span></span>
<span id="cb182-782"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-782" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">ColSideColors=</span>groups.colors[ <span class="fu">as.factor</span>(groups)]</span>
<span id="cb182-783"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-783" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">margins=</span><span class="fu">c</span>(<span class="dv">18</span>,<span class="dv">12</span>)</span>
<span id="cb182-784"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-784" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">cexRow=</span><span class="dv">1</span></span>
<span id="cb182-785"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-785" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">srtCol=</span><span class="dv">45</span></span>
<span id="cb182-786"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-786" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">cexCol=</span><span class="fl">1.8</span>  <span class="co"># size of font for sample names</span></span>
<span id="cb182-787"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-787" aria-hidden="true" tabindex="-1"></a>        ,<span class="at">lmat =</span> lmat, <span class="at">lwid =</span> lwid, <span class="at">lhei =</span> lhei</span>
<span id="cb182-788"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-788" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb182-789"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-789" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-790"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-790" aria-hidden="true" tabindex="-1"></a>    <span class="fu">par</span>(<span class="at">lend =</span> <span class="dv">1</span>)           <span class="co"># square line ends for the color legend</span></span>
<span id="cb182-791"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-791" aria-hidden="true" tabindex="-1"></a>    <span class="fu">add_legend</span>(<span class="st">&quot;topleft&quot;</span>,</span>
<span id="cb182-792"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-792" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend =</span> <span class="fu">unique</span>(groups), <span class="co"># category labels</span></span>
<span id="cb182-793"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-793" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> groups.colors[ <span class="fu">unique</span>(<span class="fu">as.factor</span>(groups))],  <span class="co"># color key</span></span>
<span id="cb182-794"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-794" aria-hidden="true" tabindex="-1"></a>        <span class="at">lty=</span> <span class="dv">1</span>,             <span class="co"># line style</span></span>
<span id="cb182-795"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-795" aria-hidden="true" tabindex="-1"></a>        <span class="at">lwd =</span> <span class="dv">10</span>            <span class="co"># line width</span></span>
<span id="cb182-796"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-796" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb182-797"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-797" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-798"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-798" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-799"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-799" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-800"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-800" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-801"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-801" aria-hidden="true" tabindex="-1"></a>    <span class="co">#Interactive heatmap via Plotly plot ----------------------------------------------</span></span>
<span id="cb182-802"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-802" aria-hidden="true" tabindex="-1"></a>    <span class="co"># interactive heatmap with plotly</span></span>
<span id="cb182-803"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-803" aria-hidden="true" tabindex="-1"></a>heatmapPlotly <span class="ot">&lt;-</span> <span class="cf">function</span> () {  </span>
<span id="cb182-804"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-804" aria-hidden="true" tabindex="-1"></a>    input_nGenesPlotly<span class="ot">=</span> <span class="dv">50</span></span>
<span id="cb182-805"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-805" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-806"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-806" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-807"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-807" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-808"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-808" aria-hidden="true" tabindex="-1"></a>       x <span class="ot">&lt;-</span> convertedData.out</span>
<span id="cb182-809"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-809" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-810"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-810" aria-hidden="true" tabindex="-1"></a>    n<span class="ot">=</span>input_nGenesPlotly</span>
<span id="cb182-811"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-811" aria-hidden="true" tabindex="-1"></a>    <span class="co">#if(n&gt;6000) n = 6000 # max</span></span>
<span id="cb182-812"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-812" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(n<span class="sc">&gt;</span><span class="fu">dim</span>(x)[<span class="dv">1</span>]) n <span class="ot">=</span> <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="co"># max as data</span></span>
<span id="cb182-813"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-813" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-814"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-814" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_geneCentering)</span>
<span id="cb182-815"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-815" aria-hidden="true" tabindex="-1"></a>        x<span class="ot">=</span><span class="fu">as.matrix</span>(x[<span class="dv">1</span><span class="sc">:</span>n,])<span class="sc">-</span><span class="fu">apply</span>(x[<span class="dv">1</span><span class="sc">:</span>n,],<span class="dv">1</span>,mean)</span>
<span id="cb182-816"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-816" aria-hidden="true" tabindex="-1"></a>    <span class="co"># standardize by gene</span></span>
<span id="cb182-817"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-817" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_geneNormalize) </span>
<span id="cb182-818"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-818" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> x <span class="sc">/</span> <span class="fu">apply</span>(x,<span class="dv">1</span>,sd)</span>
<span id="cb182-819"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-819" aria-hidden="true" tabindex="-1"></a>    <span class="co"># row centering and normalize</span></span>
<span id="cb182-820"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-820" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">scale</span>(x, <span class="at">center =</span> input_sampleCentering, <span class="at">scale =</span> input_sampleNormalize) </span>
<span id="cb182-821"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-821" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-822"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-822" aria-hidden="true" tabindex="-1"></a>    cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">+</span> <span class="dv">3</span><span class="sc">*</span><span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb182-823"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-823" aria-hidden="true" tabindex="-1"></a>    x[x<span class="sc">&gt;</span>cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb182-824"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-824" aria-hidden="true" tabindex="-1"></a>    cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">-</span> <span class="dv">3</span><span class="sc">*</span><span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb182-825"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-825" aria-hidden="true" tabindex="-1"></a>    x[x<span class="sc">&lt;</span> cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb182-826"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-826" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-827"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-827" aria-hidden="true" tabindex="-1"></a>    <span class="co"># adding gene symbol</span></span>
<span id="cb182-828"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-828" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">&lt;-</span> <span class="fu">match</span>( <span class="fu">rownames</span>(x), allGeneInfo.out[,<span class="dv">1</span>])</span>
<span id="cb182-829"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-829" aria-hidden="true" tabindex="-1"></a>    geneSymbols <span class="ot">&lt;-</span> <span class="fu">as.character</span>( allGeneInfo.out<span class="sc">$</span>symbol)[ix]</span>
<span id="cb182-830"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-830" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if missing or duplicated, use Ensembl ID</span></span>
<span id="cb182-831"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-831" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">nchar</span>( geneSymbols) <span class="sc">&lt;=</span><span class="dv">1</span> <span class="sc">|</span> <span class="fu">duplicated</span>(geneSymbols ) <span class="sc">|</span> <span class="fu">is.null</span>(geneSymbols)  <span class="sc">|</span> <span class="fu">is.na</span>(geneSymbols)    );   geneSymbols[ ix ] <span class="ot">&lt;-</span> <span class="fu">rownames</span>(x)[ix]</span>
<span id="cb182-832"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-832" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>( x) <span class="ot">=</span> geneSymbols;</span>
<span id="cb182-833"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-833" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-834"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-834" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-835"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-835" aria-hidden="true" tabindex="-1"></a>    clust <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb182-836"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-836" aria-hidden="true" tabindex="-1"></a>      <span class="fu">dist2</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb182-837"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-837" aria-hidden="true" tabindex="-1"></a>      <span class="fu">hclust2</span>()</span>
<span id="cb182-838"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-838" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-839"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-839" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get order</span></span>
<span id="cb182-840"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-840" aria-hidden="true" tabindex="-1"></a>    ord <span class="ot">&lt;-</span> clust<span class="sc">$</span>order</span>
<span id="cb182-841"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-841" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-842"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-842" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Re-arrange based on order</span></span>
<span id="cb182-843"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-843" aria-hidden="true" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">t</span>( x[ord,] )<span class="sc">%&gt;%</span></span>
<span id="cb182-844"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-844" aria-hidden="true" tabindex="-1"></a>       <span class="fu">melt</span>()</span>
<span id="cb182-845"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-845" aria-hidden="true" tabindex="-1"></a>       </span>
<span id="cb182-846"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-846" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(df)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;X&quot;</span>,<span class="st">&quot;Y&quot;</span>)</span>
<span id="cb182-847"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-847" aria-hidden="true" tabindex="-1"></a>    colorNames <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(<span class="fu">tolower</span>(<span class="fu">rownames</span>(heatColors)[ <span class="fu">as.integer</span>(input_heatColors1)   ]),<span class="st">&quot;-&quot;</span> ) )</span>
<span id="cb182-848"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-848" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb182-849"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-849" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ggplot</span>(<span class="fu">aes</span>(X, Y, <span class="at">fill =</span> value)) <span class="sc">+</span> </span>
<span id="cb182-850"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-850" aria-hidden="true" tabindex="-1"></a>           <span class="fu">geom_tile</span>()<span class="sc">+</span> <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> colorNames[<span class="dv">1</span>], <span class="at">mid =</span> colorNames[<span class="dv">2</span>],<span class="at">high =</span> colorNames[<span class="dv">3</span>]) <span class="sc">+</span></span>
<span id="cb182-851"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-851" aria-hidden="true" tabindex="-1"></a>           <span class="fu">theme</span>(<span class="at">axis.title.y=</span><span class="fu">element_blank</span>(),   <span class="co"># remove y labels</span></span>
<span id="cb182-852"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-852" aria-hidden="true" tabindex="-1"></a>           <span class="co"># axis.text.y=element_blank.out,  # keep gene names for zooming</span></span>
<span id="cb182-853"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-853" aria-hidden="true" tabindex="-1"></a>            <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb182-854"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-854" aria-hidden="true" tabindex="-1"></a>            <span class="at">axis.title.x=</span><span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb182-855"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-855" aria-hidden="true" tabindex="-1"></a>            <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>,<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb182-856"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-856" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-857"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-857" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-858"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-858" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplotly</span>(p) <span class="sc">%&gt;%</span> </span>
<span id="cb182-859"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-859" aria-hidden="true" tabindex="-1"></a>        <span class="fu">layout</span>(<span class="at">margin =</span> <span class="fu">list</span>(<span class="at">b =</span> <span class="dv">150</span>,<span class="at">l=</span><span class="dv">200</span>))  <span class="co"># prevent cutoff of sample names</span></span>
<span id="cb182-860"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-860" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-861"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-861" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-862"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-862" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-863"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-863" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-864"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-864" aria-hidden="true" tabindex="-1"></a><span class="co"># PCA</span></span>
<span id="cb182-865"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-865" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-866"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-866" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-867"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-867" aria-hidden="true" tabindex="-1"></a>PCAplot <span class="ot">&lt;-</span> <span class="cf">function</span>() {   <span class="co">#PCA</span></span>
<span id="cb182-868"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-868" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> convertedData.out;</span>
<span id="cb182-869"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-869" aria-hidden="true" tabindex="-1"></a>     pca.object <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(x))</span>
<span id="cb182-870"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-870" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-871"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-871" aria-hidden="true" tabindex="-1"></a>    pcaData <span class="ot">=</span> <span class="fu">as.data.frame</span>(pca.object<span class="sc">$</span>x[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]); pcaData <span class="ot">=</span> <span class="fu">cbind</span>(pcaData,<span class="fu">detectGroups</span>(<span class="fu">colnames</span>(x)) )</span>
<span id="cb182-872"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-872" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(pcaData) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;PC1&quot;</span>, <span class="st">&quot;PC2&quot;</span>, <span class="st">&quot;Sample_Name&quot;</span>)</span>
<span id="cb182-873"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-873" aria-hidden="true" tabindex="-1"></a>    percentVar<span class="ot">=</span><span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span><span class="fu">summary</span>(pca.object)<span class="sc">$</span>importance[<span class="dv">2</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],<span class="dv">0</span>)</span>
<span id="cb182-874"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-874" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(readSampleInfo.out)) { </span>
<span id="cb182-875"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-875" aria-hidden="true" tabindex="-1"></a>        p<span class="ot">=</span><span class="fu">ggplot</span>(pcaData, <span class="fu">aes</span>(PC1, PC2, <span class="at">color=</span>Sample_Name, <span class="at">shape =</span> Sample_Name)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>) </span>
<span id="cb182-876"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-876" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb182-877"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-877" aria-hidden="true" tabindex="-1"></a>        pcaData <span class="ot">=</span> <span class="fu">cbind</span>(pcaData,readSampleInfo.out )</span>
<span id="cb182-878"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-878" aria-hidden="true" tabindex="-1"></a>        p<span class="ot">=</span><span class="fu">ggplot</span>(pcaData, <span class="fu">aes_string</span>(<span class="st">&quot;PC1&quot;</span>, <span class="st">&quot;PC2&quot;</span>, <span class="at">color=</span>input_selectFactors,<span class="at">shape=</span>input_selectFactors2)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>) </span>
<span id="cb182-879"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-879" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-880"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-880" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-881"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-881" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC1: &quot;</span>,percentVar[<span class="dv">1</span>],<span class="st">&quot;% variance&quot;</span>)) </span>
<span id="cb182-882"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-882" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC2: &quot;</span>,percentVar[<span class="dv">2</span>],<span class="st">&quot;% variance&quot;</span>)) </span>
<span id="cb182-883"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-883" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">ggtitle</span>(<span class="st">&quot;Principal component analysis (PCA)&quot;</span>)<span class="sc">+</span><span class="fu">coord_fixed</span>(<span class="at">ratio=</span><span class="fl">1.0</span>)<span class="sc">+</span> </span>
<span id="cb182-884"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-884" aria-hidden="true" tabindex="-1"></a>     <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>,<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb182-885"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-885" aria-hidden="true" tabindex="-1"></a>     <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb182-886"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-886" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.text.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb182-887"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-887" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.x =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb182-888"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-888" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>) ) <span class="sc">+</span></span>
<span id="cb182-889"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-889" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span>
<span id="cb182-890"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-890" aria-hidden="true" tabindex="-1"></a>       <span class="fu">print</span>(p)</span>
<span id="cb182-891"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-891" aria-hidden="true" tabindex="-1"></a>       }</span>
<span id="cb182-892"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-892" aria-hidden="true" tabindex="-1"></a>    <span class="co"># variance chart</span></span>
<span id="cb182-893"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-893" aria-hidden="true" tabindex="-1"></a>    <span class="co"># plot(pca.object,type=&quot;bar&quot;, xlab=&quot;Principal Components&quot;, main =&quot;Variances explained&quot;)</span></span>
<span id="cb182-894"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-894" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-895"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-895" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-896"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-896" aria-hidden="true" tabindex="-1"></a>MDSplot <span class="ot">&lt;-</span> <span class="cf">function</span>() {  <span class="co"># MDS</span></span>
<span id="cb182-897"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-897" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> convertedData.out;</span>
<span id="cb182-898"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-898" aria-hidden="true" tabindex="-1"></a>     fit <span class="ot">=</span> <span class="fu">cmdscale</span>( <span class="fu">dist2</span>(<span class="fu">t</span>(x) ), <span class="at">eig=</span>T, <span class="at">k=</span><span class="dv">2</span>)</span>
<span id="cb182-899"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-899" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-900"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-900" aria-hidden="true" tabindex="-1"></a>    <span class="co"># par(pin=c(5,5))</span></span>
<span id="cb182-901"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-901" aria-hidden="true" tabindex="-1"></a>    pcaData <span class="ot">=</span> <span class="fu">as.data.frame</span>(fit<span class="sc">$</span>points[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]); pcaData <span class="ot">=</span> <span class="fu">cbind</span>(pcaData,<span class="fu">detectGroups</span>(<span class="fu">colnames</span>(x)) )</span>
<span id="cb182-902"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-902" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(pcaData) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;x1&quot;</span>, <span class="st">&quot;x2&quot;</span>, <span class="st">&quot;Sample_Name&quot;</span>)</span>
<span id="cb182-903"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-903" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-904"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-904" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-905"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-905" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(readSampleInfo.out)) { </span>
<span id="cb182-906"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-906" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span><span class="fu">ggplot</span>(pcaData, <span class="fu">aes</span>(x1, x2, <span class="at">color=</span>Sample_Name, <span class="at">shape =</span> Sample_Name)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>) </span>
<span id="cb182-907"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-907" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb182-908"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-908" aria-hidden="true" tabindex="-1"></a>        pcaData <span class="ot">=</span> <span class="fu">cbind</span>(pcaData,readSampleInfo.out )</span>
<span id="cb182-909"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-909" aria-hidden="true" tabindex="-1"></a>        p<span class="ot">=</span><span class="fu">ggplot</span>(pcaData, <span class="fu">aes_string</span>(<span class="st">&quot;x1&quot;</span>, <span class="st">&quot;x2&quot;</span>, <span class="at">color=</span>input_selectFactors,<span class="at">shape=</span>input_selectFactors2)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>) </span>
<span id="cb182-910"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-910" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-911"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-911" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">&quot;Dimension 1&quot;</span>) </span>
<span id="cb182-912"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-912" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">&quot;Dimension 2&quot;</span>) </span>
<span id="cb182-913"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-913" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">ggtitle</span>(<span class="st">&quot;Multidimensional scaling (MDS)&quot;</span>)<span class="sc">+</span> <span class="fu">coord_fixed</span>(<span class="at">ratio=</span><span class="fl">1.</span>)<span class="sc">+</span> </span>
<span id="cb182-914"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-914" aria-hidden="true" tabindex="-1"></a>     <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb182-915"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-915" aria-hidden="true" tabindex="-1"></a>         <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb182-916"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-916" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.text.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb182-917"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-917" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.x =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb182-918"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-918" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>) ) <span class="sc">+</span></span>
<span id="cb182-919"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-919" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span>
<span id="cb182-920"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-920" aria-hidden="true" tabindex="-1"></a>       <span class="fu">print</span>(p)</span>
<span id="cb182-921"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-921" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-922"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-922" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb182-923"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-923" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-924"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-924" aria-hidden="true" tabindex="-1"></a>tSNEplot <span class="ot">&lt;-</span> <span class="cf">function</span>() {  <span class="co"># t-SNE</span></span>
<span id="cb182-925"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-925" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> convertedData.out;</span>
<span id="cb182-926"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-926" aria-hidden="true" tabindex="-1"></a>     <span class="fu">library</span>(Rtsne,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb182-927"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-927" aria-hidden="true" tabindex="-1"></a>     <span class="fu">set.seed</span>(input_tsneSeed2)</span>
<span id="cb182-928"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-928" aria-hidden="true" tabindex="-1"></a>     tsne <span class="ot">&lt;-</span> <span class="fu">Rtsne</span>(<span class="fu">t</span>(x), <span class="at">dims =</span> <span class="dv">2</span>, <span class="at">perplexity=</span><span class="dv">1</span>, <span class="at">verbose=</span><span class="cn">FALSE</span>, <span class="at">max_iter =</span> <span class="dv">400</span>)</span>
<span id="cb182-929"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-929" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-930"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-930" aria-hidden="true" tabindex="-1"></a>    pcaData <span class="ot">=</span> <span class="fu">as.data.frame</span>(tsne<span class="sc">$</span>Y); pcaData <span class="ot">=</span> <span class="fu">cbind</span>(pcaData,<span class="fu">detectGroups</span>(<span class="fu">colnames</span>(x)) )</span>
<span id="cb182-931"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-931" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(pcaData) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;x1&quot;</span>, <span class="st">&quot;x2&quot;</span>, <span class="st">&quot;Sample_Name&quot;</span>)</span>
<span id="cb182-932"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-932" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-933"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-933" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-934"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-934" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(readSampleInfo.out)) { </span>
<span id="cb182-935"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-935" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span><span class="fu">ggplot</span>(pcaData, <span class="fu">aes</span>(x1, x2, <span class="at">color=</span>Sample_Name, <span class="at">shape =</span> Sample_Name)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>) </span>
<span id="cb182-936"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-936" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb182-937"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-937" aria-hidden="true" tabindex="-1"></a>        pcaData <span class="ot">=</span> <span class="fu">cbind</span>(pcaData,readSampleInfo.out )</span>
<span id="cb182-938"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-938" aria-hidden="true" tabindex="-1"></a>        p<span class="ot">=</span><span class="fu">ggplot</span>(pcaData, <span class="fu">aes_string</span>(<span class="st">&quot;x1&quot;</span>, <span class="st">&quot;x2&quot;</span>, <span class="at">color=</span>input_selectFactors,<span class="at">shape=</span>input_selectFactors2)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>) </span>
<span id="cb182-939"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-939" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-940"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-940" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">&quot;Dimension 1&quot;</span>) </span>
<span id="cb182-941"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-941" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">&quot;Dimension 2&quot;</span>) </span>
<span id="cb182-942"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-942" aria-hidden="true" tabindex="-1"></a>    p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">ggtitle</span>(<span class="st">&quot;t-SNE plot&quot;</span>)<span class="sc">+</span> <span class="fu">coord_fixed</span>(<span class="at">ratio=</span><span class="fl">1.</span>)<span class="sc">+</span> </span>
<span id="cb182-943"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-943" aria-hidden="true" tabindex="-1"></a>     <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb182-944"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-944" aria-hidden="true" tabindex="-1"></a>         <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb182-945"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-945" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.text.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb182-946"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-946" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.x =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb182-947"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-947" aria-hidden="true" tabindex="-1"></a>           <span class="at">axis.title.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>) ) <span class="sc">+</span></span>
<span id="cb182-948"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-948" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span>
<span id="cb182-949"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-949" aria-hidden="true" tabindex="-1"></a>       <span class="fu">print</span>(p)</span>
<span id="cb182-950"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-950" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-951"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-951" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb182-952"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-952" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-953"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-953" aria-hidden="true" tabindex="-1"></a> <span class="co"># read pathway data from local file, not used if using database</span></span>
<span id="cb182-954"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-954" aria-hidden="true" tabindex="-1"></a> <span class="co"># Read gene sets GMT file</span></span>
<span id="cb182-955"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-955" aria-hidden="true" tabindex="-1"></a><span class="co"># This functions cleans and converts to upper case</span></span>
<span id="cb182-956"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-956" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-957"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-957" aria-hidden="true" tabindex="-1"></a>readGMTRobust <span class="ot">&lt;-</span> <span class="cf">function</span> (file1) {   <span class="co"># size restriction</span></span>
<span id="cb182-958"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-958" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Read in the first file </span></span>
<span id="cb182-959"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-959" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> <span class="fu">scan</span>(file1, <span class="at">what=</span><span class="st">&quot;&quot;</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb182-960"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-960" aria-hidden="true" tabindex="-1"></a>        <span class="co"># x &lt;- gsub(&quot;\t\t.&quot;,&quot;&quot;,x)     # GMT files saved by Excel has a lot of empty cells &quot;\t\t\t\t&quot;   &quot;\t.&quot; means one or more tab</span></span>
<span id="cb182-961"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-961" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>,<span class="st">&quot;&quot;</span>,x)  <span class="co"># remove white space</span></span>
<span id="cb182-962"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-962" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> <span class="fu">toupper</span>(x)    <span class="co"># convert to upper case</span></span>
<span id="cb182-963"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-963" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-964"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-964" aria-hidden="true" tabindex="-1"></a>        <span class="co">#----Process the first file</span></span>
<span id="cb182-965"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-965" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Separate elements by one or more whitespace</span></span>
<span id="cb182-966"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-966" aria-hidden="true" tabindex="-1"></a>        y <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(x, <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb182-967"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-967" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Extract the first vector element and set it as the list element name</span></span>
<span id="cb182-968"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-968" aria-hidden="true" tabindex="-1"></a>        <span class="fu">names</span>(y) <span class="ot">&lt;-</span> <span class="fu">sapply</span>(y, <span class="st">`</span><span class="at">[[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb182-969"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-969" aria-hidden="true" tabindex="-1"></a>        <span class="co">#names(y) &lt;- sapply(y, function(x) x[[1]]) # same as above</span></span>
<span id="cb182-970"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-970" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Remove the first vector element from each list element</span></span>
<span id="cb182-971"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-971" aria-hidden="true" tabindex="-1"></a>        y <span class="ot">&lt;-</span> <span class="fu">lapply</span>(y, <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb182-972"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-972" aria-hidden="true" tabindex="-1"></a>        <span class="co">#y &lt;- lapply(y, function(x) x[-1]) # same as above</span></span>
<span id="cb182-973"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-973" aria-hidden="true" tabindex="-1"></a>        <span class="co"># remove duplicated elements</span></span>
<span id="cb182-974"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-974" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> ( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(y) )  y[[i]] <span class="ot">&lt;-</span> <span class="fu">cleanGeneSet</span>(y[[i]])</span>
<span id="cb182-975"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-975" aria-hidden="true" tabindex="-1"></a>        <span class="co"># check the distribution of the size of gene lists sapply(y, length) hold a vector of sizes</span></span>
<span id="cb182-976"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-976" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">max</span>( <span class="fu">sapply</span>(y,length) ) <span class="sc">&lt;</span><span class="dv">5</span>) <span class="fu">cat</span>(<span class="st">&quot;Warning! Gene sets have very small number of genes!</span><span class="sc">\n</span><span class="st"> Please double check format.&quot;</span>)</span>
<span id="cb182-977"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-977" aria-hidden="true" tabindex="-1"></a>        y <span class="ot">&lt;-</span> y[<span class="fu">which</span>(<span class="fu">sapply</span>(y,length) <span class="sc">&gt;</span> <span class="dv">1</span>)]  <span class="co"># gene sets smaller than 1 is ignored!!!</span></span>
<span id="cb182-978"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-978" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-979"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-979" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>(y)</span>
<span id="cb182-980"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-980" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-981"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-981" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-982"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-982" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-983"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-983" aria-hidden="true" tabindex="-1"></a>readGeneSets <span class="ot">&lt;-</span> <span class="cf">function</span> (fileName, convertedData, GO,selectOrg, myrange) {</span>
<span id="cb182-984"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-984" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-985"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-985" aria-hidden="true" tabindex="-1"></a>    pathway <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(sqlite,fileName)</span>
<span id="cb182-986"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-986" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-987"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-987" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(GO)  ) GO <span class="ot">&lt;-</span> <span class="st">&quot;GOBP&quot;</span>   <span class="co"># initial value not properly set; enforcing  </span></span>
<span id="cb182-988"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-988" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-989"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-989" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get Gene sets</span></span>
<span id="cb182-990"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-990" aria-hidden="true" tabindex="-1"></a>    querySet <span class="ot">=</span> <span class="fu">rownames</span>(convertedData)</span>
<span id="cb182-991"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-991" aria-hidden="true" tabindex="-1"></a>    sqlQuery <span class="ot">=</span> <span class="fu">paste</span>( <span class="st">&quot; select distinct gene,pathwayID from pathway where gene IN (&#39;&quot;</span>, <span class="fu">paste</span>(querySet,<span class="at">collapse=</span><span class="st">&quot;&#39;, &#39;&quot;</span>),<span class="st">&quot;&#39;)&quot;</span> ,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb182-992"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-992" aria-hidden="true" tabindex="-1"></a>    <span class="co"># cat(paste0(&quot;\n\nhere:&quot;,GO,&quot;There&quot;))</span></span>
<span id="cb182-993"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-993" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-994"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-994" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( GO <span class="sc">!=</span> <span class="st">&quot;All&quot;</span>) sqlQuery <span class="ot">=</span> <span class="fu">paste0</span>(sqlQuery, <span class="st">&quot; AND category =&#39;&quot;</span>,GO,<span class="st">&quot;&#39;&quot;</span>)</span>
<span id="cb182-995"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-995" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>( pathway, sqlQuery  )</span>
<span id="cb182-996"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-996" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">1</span>] <span class="sc">==</span><span class="dv">0</span>) {<span class="fu">return</span>(<span class="fu">list</span>( <span class="at">x=</span><span class="fu">as.data.frame</span>(<span class="st">&quot;No matching species or gene ID file!&quot;</span> )) )}</span>
<span id="cb182-997"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-997" aria-hidden="true" tabindex="-1"></a>    <span class="co"># list pathways and frequency of genes</span></span>
<span id="cb182-998"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-998" aria-hidden="true" tabindex="-1"></a>    pathwayIDs <span class="ot">=</span> <span class="fu">aggregate</span>( result<span class="sc">$</span>pathwayID, <span class="at">by   =</span> <span class="fu">list</span>(<span class="at">unique.values =</span> result<span class="sc">$</span>pathwayID), <span class="at">FUN =</span> length)</span>
<span id="cb182-999"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-999" aria-hidden="true" tabindex="-1"></a>    pathwayIDs <span class="ot">=</span> pathwayIDs[<span class="fu">which</span>(pathwayIDs[,<span class="dv">2</span>]<span class="sc">&gt;=</span> myrange[<span class="dv">1</span>] ),]</span>
<span id="cb182-1000"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1000" aria-hidden="true" tabindex="-1"></a>    pathwayIDs <span class="ot">=</span> pathwayIDs[<span class="fu">which</span>( pathwayIDs[,<span class="dv">2</span>] <span class="sc">&lt;=</span> myrange[<span class="dv">2</span>] ),]</span>
<span id="cb182-1001"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1001" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">dim</span>(pathwayIDs)[<span class="dv">1</span>] <span class="sc">==</span><span class="dv">0</span> ) geneSets <span class="ot">=</span> <span class="cn">NULL</span>;</span>
<span id="cb182-1002"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1002" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1003"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1003" aria-hidden="true" tabindex="-1"></a>    <span class="co"># convert pathways into lists like those generated by readGMT</span></span>
<span id="cb182-1004"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1004" aria-hidden="true" tabindex="-1"></a>    geneSets <span class="ot">=</span> <span class="fu">lapply</span>(pathwayIDs[,<span class="dv">1</span>], <span class="cf">function</span>(x)  result[<span class="fu">which</span>(result<span class="sc">$</span>pathwayID <span class="sc">==</span> x ),<span class="dv">1</span>]     )</span>
<span id="cb182-1005"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1005" aria-hidden="true" tabindex="-1"></a>    pathwayInfo <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>( pathway, <span class="fu">paste</span>( <span class="st">&quot; select distinct id,Description from pathwayInfo where id IN (&#39;&quot;</span>, </span>
<span id="cb182-1006"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1006" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">paste</span>(pathwayIDs[,<span class="dv">1</span>],<span class="at">collapse=</span><span class="st">&quot;&#39;, &#39;&quot;</span>),   <span class="st">&quot;&#39;) &quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>) )</span>
<span id="cb182-1007"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1007" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span> <span class="fu">match</span>( pathwayIDs[,<span class="dv">1</span>], pathwayInfo[,<span class="dv">1</span>])</span>
<span id="cb182-1008"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1008" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(geneSets) <span class="ot">&lt;-</span> pathwayInfo[ix,<span class="dv">2</span>]  </span>
<span id="cb182-1009"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1009" aria-hidden="true" tabindex="-1"></a>    <span class="co">#geneSets &lt;- geneSets[ -which(duplicated(names(geneSets) ))] # remove geneSets with the same name</span></span>
<span id="cb182-1010"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1010" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dbDisconnect</span>(pathway)</span>
<span id="cb182-1011"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1011" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( geneSets )</span>
<span id="cb182-1012"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1012" aria-hidden="true" tabindex="-1"></a>} </span>
<span id="cb182-1013"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1013" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1014"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1014" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1015"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1015" aria-hidden="true" tabindex="-1"></a><span class="co"># Runs pathway analysis using PGSEA; this is copied and revised from PGSEA package</span></span>
<span id="cb182-1016"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1016" aria-hidden="true" tabindex="-1"></a>myPGSEA  <span class="ot">&lt;-</span> <span class="cf">function</span> (exprs, cl, <span class="at">range =</span> <span class="fu">c</span>(<span class="dv">25</span>, <span class="dv">500</span>), <span class="at">ref =</span> <span class="cn">NULL</span>, <span class="at">center =</span> <span class="cn">TRUE</span>, </span>
<span id="cb182-1017"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1017" aria-hidden="true" tabindex="-1"></a>    <span class="at">p.value =</span> <span class="fl">0.005</span>, <span class="at">weighted =</span> <span class="cn">TRUE</span>, <span class="at">nPermutation=</span><span class="dv">100</span>, <span class="at">enforceRange =</span> <span class="cn">TRUE</span>, ...) {</span>
<span id="cb182-1018"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1018" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is</span>(exprs, <span class="st">&quot;ExpressionSet&quot;</span>)) </span>
<span id="cb182-1019"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1019" aria-hidden="true" tabindex="-1"></a>        exprs <span class="ot">&lt;-</span> <span class="fu">exprs</span>(exprs)</span>
<span id="cb182-1020"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1020" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.list</span>(cl)) </span>
<span id="cb182-1021"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1021" aria-hidden="true" tabindex="-1"></a>        <span class="fu">stop</span>(<span class="st">&quot;cl need to be a list&quot;</span>)</span>
<span id="cb182-1022"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1022" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(ref)) {</span>
<span id="cb182-1023"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1023" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.numeric</span>(ref)) </span>
<span id="cb182-1024"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1024" aria-hidden="true" tabindex="-1"></a>            <span class="fu">stop</span>(<span class="st">&quot;column index&#39;s required&quot;</span>)</span>
<span id="cb182-1025"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1025" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-1026"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1026" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(ref)) {</span>
<span id="cb182-1027"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1027" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (options.out<span class="sc">$</span>verbose) </span>
<span id="cb182-1028"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1028" aria-hidden="true" tabindex="-1"></a>            <span class="fu">cat</span>(<span class="st">&quot;Creating ratios...&quot;</span>, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb182-1029"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1029" aria-hidden="true" tabindex="-1"></a>        ref_mean <span class="ot">&lt;-</span> <span class="fu">apply</span>(exprs[, ref], <span class="dv">1</span>, mean, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb182-1030"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1030" aria-hidden="true" tabindex="-1"></a>        exprs <span class="ot">&lt;-</span> <span class="fu">sweep</span>(exprs, <span class="dv">1</span>, ref_mean, <span class="st">&quot;-&quot;</span>)</span>
<span id="cb182-1031"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1031" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-1032"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1032" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (center) </span>
<span id="cb182-1033"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1033" aria-hidden="true" tabindex="-1"></a>        exprs <span class="ot">&lt;-</span> <span class="fu">scale</span>(exprs, <span class="at">scale =</span> <span class="cn">FALSE</span>)         <span class="co"># column centering is done</span></span>
<span id="cb182-1034"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1034" aria-hidden="true" tabindex="-1"></a>    results <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="fu">length</span>(cl), <span class="fu">ncol</span>(exprs))</span>
<span id="cb182-1035"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1035" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(results) <span class="ot">&lt;-</span> <span class="fu">names</span>(cl)</span>
<span id="cb182-1036"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1036" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(results) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(exprs)</span>
<span id="cb182-1037"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1037" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mode</span>(results) <span class="ot">&lt;-</span> <span class="st">&quot;numeric&quot;</span></span>
<span id="cb182-1038"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1038" aria-hidden="true" tabindex="-1"></a>    Setsize <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">0</span>,<span class="fu">length</span>(cl)))     <span class="co"># gene set size vector</span></span>
<span id="cb182-1039"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1039" aria-hidden="true" tabindex="-1"></a>    mean2 <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">0</span>,<span class="fu">length</span>(cl)))     <span class="co"># mean of the range of means </span></span>
<span id="cb182-1040"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1040" aria-hidden="true" tabindex="-1"></a>    meanSD <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">0</span>,<span class="fu">length</span>(cl)))     <span class="co"># SD of the range of means    </span></span>
<span id="cb182-1041"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1041" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.logical</span>(p.value)) </span>
<span id="cb182-1042"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1042" aria-hidden="true" tabindex="-1"></a>        { p.results <span class="ot">&lt;-</span> results; mean.results <span class="ot">&lt;-</span> results;}</span>
<span id="cb182-1043"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1043" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(cl)) {              <span class="co"># for each gene list</span></span>
<span id="cb182-1044"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1044" aria-hidden="true" tabindex="-1"></a>        <span class="co">#cat(&quot;\nProcessing gene set&quot;,i);</span></span>
<span id="cb182-1045"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1045" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">class</span>(cl[[i]]) <span class="sc">==</span> <span class="st">&quot;smc&quot;</span>) {</span>
<span id="cb182-1046"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1046" aria-hidden="true" tabindex="-1"></a>            clids <span class="ot">&lt;-</span> cl[[i]]<span class="sc">@</span>ids</span>
<span id="cb182-1047"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1047" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-1048"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1048" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span> <span class="cf">if</span> (<span class="fu">class</span>(cl[[i]]) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;GeneColorSet&quot;</span>, <span class="st">&quot;GeneSet&quot;</span>)) {</span>
<span id="cb182-1049"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1049" aria-hidden="true" tabindex="-1"></a>            clids <span class="ot">&lt;-</span> cl[[i]]<span class="sc">@</span>geneIds</span>
<span id="cb182-1050"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1050" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-1051"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1051" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span> {</span>
<span id="cb182-1052"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1052" aria-hidden="true" tabindex="-1"></a>            clids <span class="ot">&lt;-</span> cl[[i]]</span>
<span id="cb182-1053"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1053" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-1054"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1054" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">options</span>()<span class="sc">$</span>verbose) </span>
<span id="cb182-1055"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1055" aria-hidden="true" tabindex="-1"></a>            <span class="fu">cat</span>(<span class="st">&quot;Testing region &quot;</span>, i, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb182-1056"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1056" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">&lt;-</span> <span class="fu">match</span>(clids, <span class="fu">rownames</span>(exprs))</span>
<span id="cb182-1057"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1057" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">&lt;-</span> <span class="fu">unique</span>(ix[<span class="sc">!</span><span class="fu">is.na</span>(ix)])</span>
<span id="cb182-1058"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1058" aria-hidden="true" tabindex="-1"></a>        present <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(ix))</span>
<span id="cb182-1059"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1059" aria-hidden="true" tabindex="-1"></a>        Setsize[i] <span class="ot">&lt;-</span> present </span>
<span id="cb182-1060"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1060" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (present <span class="sc">&lt;</span> range[<span class="dv">1</span>]) {</span>
<span id="cb182-1061"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1061" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (<span class="fu">options</span>()<span class="sc">$</span>verbose) </span>
<span id="cb182-1062"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1062" aria-hidden="true" tabindex="-1"></a>                <span class="fu">cat</span>(<span class="st">&quot;Skipping region &quot;</span>, i, <span class="st">&quot; because too small-&quot;</span>, </span>
<span id="cb182-1063"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1063" aria-hidden="true" tabindex="-1"></a>                  present, <span class="st">&quot;,</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb182-1064"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1064" aria-hidden="true" tabindex="-1"></a>            <span class="cf">next</span></span>
<span id="cb182-1065"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1065" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-1066"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1066" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (present <span class="sc">&gt;</span> range[<span class="dv">2</span>]) {</span>
<span id="cb182-1067"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1067" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (<span class="fu">options</span>()<span class="sc">$</span>verbose) </span>
<span id="cb182-1068"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1068" aria-hidden="true" tabindex="-1"></a>                <span class="fu">cat</span>(<span class="st">&quot;Skipping region &quot;</span>, i, <span class="st">&quot; because too large-&quot;</span>, </span>
<span id="cb182-1069"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1069" aria-hidden="true" tabindex="-1"></a>                  present, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb182-1070"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1070" aria-hidden="true" tabindex="-1"></a>            <span class="cf">next</span></span>
<span id="cb182-1071"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1071" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-1072"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1072" aria-hidden="true" tabindex="-1"></a>        texprs <span class="ot">&lt;-</span> exprs[ix, ]           <span class="co"># expression matrix for genes in gene set</span></span>
<span id="cb182-1073"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1073" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">any</span>(<span class="fu">is.na</span>(texprs))) </span>
<span id="cb182-1074"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1074" aria-hidden="true" tabindex="-1"></a>            <span class="fu">cat</span>(<span class="st">&quot;Warning - &#39;NA&#39; values within expression data, enrichment scores are estimates only.</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb182-1075"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1075" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.matrix</span>(texprs)) </span>
<span id="cb182-1076"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1076" aria-hidden="true" tabindex="-1"></a>            texprs <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(texprs)</span>
<span id="cb182-1077"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1077" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb182-1078"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1078" aria-hidden="true" tabindex="-1"></a>        stat <span class="ot">&lt;-</span> <span class="fu">try</span>(<span class="fu">apply</span>(texprs, <span class="dv">2</span>, t.test, ...))</span>
<span id="cb182-1079"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1079" aria-hidden="true" tabindex="-1"></a>        means <span class="ot">&lt;-</span> <span class="fu">try</span>(<span class="fu">apply</span>(texprs, <span class="dv">2</span>, mean,<span class="at">trim=</span><span class="fl">0.1</span>))   <span class="co"># trim mean</span></span>
<span id="cb182-1080"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1080" aria-hidden="true" tabindex="-1"></a>        ps <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(stat, <span class="cf">function</span>(x) x<span class="sc">$</span>p.value))</span>
<span id="cb182-1081"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1081" aria-hidden="true" tabindex="-1"></a>        stat <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(stat, <span class="cf">function</span>(x) x<span class="sc">$</span>statistic))</span>
<span id="cb182-1082"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1082" aria-hidden="true" tabindex="-1"></a>        p.results[i, ] <span class="ot">&lt;-</span> ps</span>
<span id="cb182-1083"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1083" aria-hidden="true" tabindex="-1"></a>        mean.results[i,] <span class="ot">&lt;-</span> means</span>
<span id="cb182-1084"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1084" aria-hidden="true" tabindex="-1"></a>        results[i, ] <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(stat)</span>
<span id="cb182-1085"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1085" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1086"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1086" aria-hidden="true" tabindex="-1"></a>        <span class="co"># permutation of gene sets of the same size</span></span>
<span id="cb182-1087"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1087" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(nPermutation <span class="sc">&gt;</span> <span class="dv">2</span> )  { <span class="co"># no permutation if &lt;=2</span></span>
<span id="cb182-1088"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1088" aria-hidden="true" tabindex="-1"></a>            meansRanges <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">rep</span>(nPermutation))</span>
<span id="cb182-1089"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1089" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>( k <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nPermutation ) {</span>
<span id="cb182-1090"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1090" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">&lt;-</span> <span class="fu">sample.int</span>( <span class="fu">dim</span>(exprs)[<span class="dv">1</span>], <span class="fu">length</span>(ix) )</span>
<span id="cb182-1091"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1091" aria-hidden="true" tabindex="-1"></a>                texprs <span class="ot">&lt;-</span> exprs[ix, ] </span>
<span id="cb182-1092"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1092" aria-hidden="true" tabindex="-1"></a>                means <span class="ot">&lt;-</span> <span class="fu">try</span>(<span class="fu">apply</span>(texprs, <span class="dv">2</span>, mean,<span class="at">trim=</span><span class="fl">0.1</span>))   <span class="co"># trim mean</span></span>
<span id="cb182-1093"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1093" aria-hidden="true" tabindex="-1"></a>                meansRanges[k] <span class="ot">=</span> <span class="fu">dynamicRange</span>(means)</span>
<span id="cb182-1094"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1094" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-1095"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1095" aria-hidden="true" tabindex="-1"></a>            mean2[i] <span class="ot">=</span> <span class="fu">mean</span>(meansRanges)</span>
<span id="cb182-1096"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1096" aria-hidden="true" tabindex="-1"></a>            meanSD[i]<span class="ot">=</span> <span class="fu">sd</span>(meansRanges,<span class="at">na.rm=</span><span class="cn">TRUE</span>)   <span class="co"># NA are removed before calculating standard deviation</span></span>
<span id="cb182-1097"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1097" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-1098"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1098" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-1099"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1099" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">results =</span> results, <span class="at">p.results =</span> p.results, <span class="at">means =</span> mean.results, <span class="at">size=</span>Setsize, <span class="at">mean2=</span>mean2, <span class="at">meanSD=</span>meanSD))</span>
<span id="cb182-1100"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1100" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-1101"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1102"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1102" aria-hidden="true" tabindex="-1"></a>PCApathway<span class="ot">&lt;-</span> <span class="cf">function</span> () {  <span class="co"># pathway</span></span>
<span id="cb182-1103"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1103" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> convertedData.out;</span>
<span id="cb182-1104"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1104" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(PGSEA,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb182-1105"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1105" aria-hidden="true" tabindex="-1"></a>    pca.object <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(x))</span>
<span id="cb182-1106"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1106" aria-hidden="true" tabindex="-1"></a>    pca <span class="ot">=</span> <span class="dv">100</span><span class="sc">*</span>pca.object<span class="sc">$</span>rotation </span>
<span id="cb182-1107"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1107" aria-hidden="true" tabindex="-1"></a>    Npca <span class="ot">=</span> <span class="dv">5</span></span>
<span id="cb182-1108"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1108" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (Npca <span class="sc">&gt;</span> <span class="fu">dim</span>(pca)[<span class="dv">2</span>]) { Npca <span class="ot">=</span> <span class="fu">dim</span>(pca)[<span class="dv">2</span>] } <span class="cf">else</span> pca <span class="ot">&lt;-</span>  pca[,<span class="dv">1</span><span class="sc">:</span>Npca]</span>
<span id="cb182-1109"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1109" aria-hidden="true" tabindex="-1"></a>    <span class="co">#pca = pca[,1:5]</span></span>
<span id="cb182-1110"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1110" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(GeneSets.out ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)  <span class="co"># no species recognized</span></span>
<span id="cb182-1111"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1111" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(GeneSets.out ) <span class="sc">&lt;=</span> <span class="dv">1</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-1112"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1112" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(&quot;\n\nGene Sets:&quot;,length( GeneSets.out))</span></span>
<span id="cb182-1113"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1113" aria-hidden="true" tabindex="-1"></a>    pg <span class="ot">=</span> <span class="fu">myPGSEA</span> (pca,<span class="at">cl=</span>GeneSets.out,<span class="at">range=</span><span class="fu">c</span>(<span class="dv">15</span>,<span class="dv">2000</span>),<span class="at">p.value=</span><span class="cn">TRUE</span>, <span class="at">weighted=</span><span class="cn">FALSE</span>,<span class="at">nPermutation=</span><span class="dv">1</span>)</span>
<span id="cb182-1114"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1115"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1115" aria-hidden="true" tabindex="-1"></a>    <span class="co"># correcting for multiple testing</span></span>
<span id="cb182-1116"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1116" aria-hidden="true" tabindex="-1"></a>    p.matrix <span class="ot">=</span> pg<span class="sc">$</span>p.result</span>
<span id="cb182-1117"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1117" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> <span class="fu">p.adjust</span>(<span class="fu">as.numeric</span>(p.matrix),<span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb182-1118"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1118" aria-hidden="true" tabindex="-1"></a>    p.matrix <span class="ot">=</span> <span class="fu">matrix</span>(tem, <span class="at">nrow=</span><span class="fu">dim</span>(p.matrix)[<span class="dv">1</span>], <span class="at">ncol =</span> <span class="fu">dim</span>(p.matrix)[<span class="dv">2</span>] )</span>
<span id="cb182-1119"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1119" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(p.matrix) <span class="ot">=</span> <span class="fu">rownames</span>(pg<span class="sc">$</span>p.result); <span class="fu">colnames</span>(p.matrix) <span class="ot">=</span> <span class="fu">colnames</span>(pg<span class="sc">$</span>p.result)</span>
<span id="cb182-1120"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1121"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1122"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1122" aria-hidden="true" tabindex="-1"></a>    selected <span class="ot">=</span><span class="fu">c</span>()</span>
<span id="cb182-1123"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1123" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(p.matrix)[<span class="dv">2</span>]) {</span>
<span id="cb182-1124"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1124" aria-hidden="true" tabindex="-1"></a>      tem <span class="ot">=</span> <span class="fu">which</span>( <span class="fu">rank</span>(p.matrix[,i],<span class="at">ties.method=</span><span class="st">&#39;first&#39;</span>) <span class="sc">&lt;=</span> <span class="dv">5</span>)  <span class="co"># rank by P value</span></span>
<span id="cb182-1125"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1125" aria-hidden="true" tabindex="-1"></a>     <span class="co">#tem = which( rank(pg$result[,i],ties.method=&#39;first&#39;) &gt;= dim(p.matrix)[1]-3.1) # rank by mean</span></span>
<span id="cb182-1126"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1126" aria-hidden="true" tabindex="-1"></a>     <span class="fu">names</span>(tem) <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;PC&quot;</span>,i,<span class="st">&quot; &quot;</span>, <span class="fu">rownames</span>(p.matrix)[tem], <span class="at">sep=</span><span class="st">&quot;&quot;</span> )</span>
<span id="cb182-1127"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1127" aria-hidden="true" tabindex="-1"></a>     selected <span class="ot">=</span> <span class="fu">c</span>(selected, tem)</span>
<span id="cb182-1128"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1128" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-1129"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1129" aria-hidden="true" tabindex="-1"></a>    rowids <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; .*&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">names</span>(selected))</span>
<span id="cb182-1130"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1130" aria-hidden="true" tabindex="-1"></a>    rowids <span class="ot">=</span> <span class="fu">as.numeric</span>( <span class="fu">gsub</span>(<span class="st">&quot;PC&quot;</span>,<span class="st">&quot;&quot;</span>,rowids) )</span>
<span id="cb182-1131"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1131" aria-hidden="true" tabindex="-1"></a>    pvals <span class="ot">=</span> p.matrix[ <span class="fu">cbind</span>(selected,rowids) ]</span>
<span id="cb182-1132"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1132" aria-hidden="true" tabindex="-1"></a>    a<span class="ot">=</span><span class="fu">sprintf</span>(<span class="st">&quot;%-1.0e&quot;</span>,pvals)</span>
<span id="cb182-1133"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1133" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> pg<span class="sc">$</span>result[selected,]</span>
<span id="cb182-1134"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1134" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(tem) <span class="ot">=</span> <span class="fu">paste</span>(a,<span class="fu">names</span>(selected)); <span class="co">#colnames(tem)= paste(&quot;PC&quot;,colnames(tem),sep=&quot;&quot;)</span></span>
<span id="cb182-1135"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1135" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1136"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1136" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> tem[<span class="sc">!</span><span class="fu">duplicated</span>(selected),] </span>
<span id="cb182-1137"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1138"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1138" aria-hidden="true" tabindex="-1"></a>    <span class="co">#tem = t(tem); tem = t( (tem - apply(tem,1,mean)) ) #/apply(tem,1,sd) )</span></span>
<span id="cb182-1139"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1140"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1140" aria-hidden="true" tabindex="-1"></a>    <span class="fu">smcPlot</span>(tem,<span class="at">scale =</span>  <span class="fu">c</span>(<span class="sc">-</span><span class="fu">max</span>(tem), <span class="fu">max</span>(tem)), <span class="at">show.grid =</span> T, <span class="at">margins =</span> <span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">1</span>, <span class="dv">6</span>, <span class="dv">23</span>), <span class="at">col =</span> .rwb,<span class="at">cex.lab=</span><span class="fl">0.5</span>, <span class="at">main=</span><span class="st">&quot;Pathways analysis on PCA&quot;</span>)</span>
<span id="cb182-1141"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1141" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1142"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1142" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb182-1143"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1144"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1144" aria-hidden="true" tabindex="-1"></a><span class="co"># correlation PCA with factors  </span></span>
<span id="cb182-1145"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1145" aria-hidden="true" tabindex="-1"></a>PCA2factor <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb182-1146"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1147"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1147" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(readSampleInfo.out)) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-1148"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1148" aria-hidden="true" tabindex="-1"></a>    npc <span class="ot">=</span> <span class="dv">5</span> <span class="co"># number of Principal components</span></span>
<span id="cb182-1149"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1149" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> readData.out<span class="sc">$</span>data</span>
<span id="cb182-1150"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1150" aria-hidden="true" tabindex="-1"></a>    y <span class="ot">&lt;-</span> readSampleInfo.out</span>
<span id="cb182-1151"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1152"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1152" aria-hidden="true" tabindex="-1"></a>    pca.object <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(x))</span>
<span id="cb182-1153"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1153" aria-hidden="true" tabindex="-1"></a>    pcaData <span class="ot">=</span> <span class="fu">as.data.frame</span>(pca.object<span class="sc">$</span>x[,<span class="dv">1</span><span class="sc">:</span>npc]); </span>
<span id="cb182-1154"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1154" aria-hidden="true" tabindex="-1"></a>    pvals <span class="ot">=</span> <span class="fu">matrix</span>(<span class="dv">1</span>,<span class="at">nrow=</span>npc,<span class="at">ncol=</span><span class="fu">ncol</span>(y))</span>
<span id="cb182-1155"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1155" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>npc ){</span>
<span id="cb182-1156"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1156" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(y) )</span>
<span id="cb182-1157"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1157" aria-hidden="true" tabindex="-1"></a>            pvals[i,j] <span class="ot">=</span><span class="fu">summary</span>( <span class="fu">aov</span>(pcaData[,i] <span class="sc">~</span> <span class="fu">as.factor</span>(y[,j])))[[<span class="dv">1</span>]][[<span class="st">&quot;Pr(&gt;F)&quot;</span>]][<span class="dv">1</span>]</span>
<span id="cb182-1158"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1158" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-1159"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1159" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1160"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1160" aria-hidden="true" tabindex="-1"></a>    pvals <span class="ot">=</span> pvals <span class="sc">*</span> npc<span class="sc">*</span> <span class="fu">ncol</span>(y)   <span class="co"># correcting for multiple testing</span></span>
<span id="cb182-1161"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1161" aria-hidden="true" tabindex="-1"></a>    pvals[pvals<span class="sc">&gt;</span><span class="dv">1</span>] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb182-1162"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1163"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1163" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(pvals) <span class="ot">=</span> <span class="fu">colnames</span>(y)</span>
<span id="cb182-1164"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1164" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(pvals) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">&quot;PC&quot;</span>,<span class="dv">1</span><span class="sc">:</span>npc)</span>
<span id="cb182-1165"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1165" aria-hidden="true" tabindex="-1"></a>    a<span class="ot">=</span><span class="st">&quot;</span><span class="sc">\n</span><span class="st"> Correlation between Principal Components (PCs) with factors</span><span class="sc">\n</span><span class="st">&quot;</span></span>
<span id="cb182-1166"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1166" aria-hidden="true" tabindex="-1"></a>    nchar0 <span class="ot">=</span> <span class="fu">nchar</span>(a)</span>
<span id="cb182-1167"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1167" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> ( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>npc){</span>
<span id="cb182-1168"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1168" aria-hidden="true" tabindex="-1"></a>        j <span class="ot">=</span> <span class="fu">which.min</span>(pvals[i,])</span>
<span id="cb182-1169"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1169" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(pvals[i,j]<span class="sc">&lt;</span> <span class="fl">0.05</span>) a<span class="ot">=</span><span class="fu">paste0</span>(a,<span class="fu">rownames</span>(pvals)[i], </span>
<span id="cb182-1170"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1170" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot; is correlated with &quot;</span>, <span class="fu">colnames</span>(pvals)[j],</span>
<span id="cb182-1171"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1171" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot; (p=&quot;</span>,   <span class="fu">sprintf</span>(<span class="st">&quot;%-3.2e&quot;</span>,pvals[i,j]),<span class="st">&quot;).</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb182-1172"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1172" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-1173"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1173" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">nchar</span>(a) <span class="sc">==</span> nchar0 ) <span class="fu">return</span>(<span class="cn">NULL</span>) <span class="cf">else</span> </span>
<span id="cb182-1174"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1174" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>( a )</span>
<span id="cb182-1175"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1175" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb182-1176"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1176" aria-hidden="true" tabindex="-1"></a>} </span>
<span id="cb182-1177"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1178"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1178" aria-hidden="true" tabindex="-1"></a><span class="co"># detecting sequencing depth bias</span></span>
<span id="cb182-1179"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1179" aria-hidden="true" tabindex="-1"></a>readCountsBias <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb182-1180"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1181"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1181" aria-hidden="true" tabindex="-1"></a>    totalCounts <span class="ot">=</span> <span class="fu">colSums</span>(readData.out<span class="sc">$</span>rawCounts) </span>
<span id="cb182-1182"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1182" aria-hidden="true" tabindex="-1"></a>    groups <span class="ot">=</span> <span class="fu">as.factor</span>( <span class="fu">detectGroups</span>(<span class="fu">colnames</span>(readData.out<span class="sc">$</span>rawCounts ) ) )</span>
<span id="cb182-1183"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1183" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb182-1184"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1184" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ANOVA of total read counts vs sample groups parsed by sample name</span></span>
<span id="cb182-1185"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1185" aria-hidden="true" tabindex="-1"></a>    pval <span class="ot">=</span> <span class="fu">summary</span>( <span class="fu">aov</span>(totalCounts <span class="sc">~</span> groups ))[[<span class="dv">1</span>]][[<span class="st">&quot;Pr(&gt;F)&quot;</span>]][<span class="dv">1</span>]</span>
<span id="cb182-1186"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1186" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(pval <span class="sc">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb182-1187"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1187" aria-hidden="true" tabindex="-1"></a>      tem <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;Warning! Sequencing depth bias detected. Total read counts are significantly different among sample groups (p=&quot;</span>,</span>
<span id="cb182-1188"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1188" aria-hidden="true" tabindex="-1"></a>                <span class="fu">sprintf</span>(<span class="st">&quot;%-3.2e&quot;</span>,pval),<span class="st">&quot;) based on ANOVA.&quot;</span>)</span>
<span id="cb182-1189"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1190"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1190" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ANOVA of total read counts vs factors in experiment design</span></span>
<span id="cb182-1191"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1191" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(readSampleInfo.out   )  ) {</span>
<span id="cb182-1192"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1192" aria-hidden="true" tabindex="-1"></a>      y <span class="ot">&lt;-</span> readSampleInfo.out</span>
<span id="cb182-1193"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1193" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(y) ) { </span>
<span id="cb182-1194"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1194" aria-hidden="true" tabindex="-1"></a>        pval <span class="ot">=</span> <span class="fu">summary</span>( <span class="fu">aov</span>(totalCounts <span class="sc">~</span> <span class="fu">as.factor</span>(y[,j])))[[<span class="dv">1</span>]][[<span class="st">&quot;Pr(&gt;F)&quot;</span>]][<span class="dv">1</span>]</span>
<span id="cb182-1195"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1196"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1196" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(pval <span class="sc">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb182-1197"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1197" aria-hidden="true" tabindex="-1"></a>        tem <span class="ot">=</span> <span class="fu">paste</span>(tem, <span class="st">&quot; Total read counts seem to be correlated with factor&quot;</span>,<span class="fu">colnames</span>(y)[j], </span>
<span id="cb182-1198"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1198" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;(p=&quot;</span>,  <span class="fu">sprintf</span>(<span class="st">&quot;%-3.2e&quot;</span>,pval),<span class="st">&quot;).  &quot;</span>)</span>
<span id="cb182-1199"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1199" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb182-1200"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1200" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb182-1201"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1201" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(tem)) <span class="fu">return</span>( <span class="st">&quot;No bias detected&quot;</span>) <span class="cf">else</span></span>
<span id="cb182-1202"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1202" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( tem )</span>
<span id="cb182-1203"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1203" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb182-1204"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1204" aria-hidden="true" tabindex="-1"></a>} </span>
<span id="cb182-1205"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1206"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1207"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1207" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-1208"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1208" aria-hidden="true" tabindex="-1"></a><span class="co"># K-means clustering</span></span>
<span id="cb182-1209"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1209" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-1210"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1210" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-1211"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1211" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Distribution of SDs </span></span>
<span id="cb182-1212"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1212" aria-hidden="true" tabindex="-1"></a>distributionSD <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb182-1213"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1214"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1214" aria-hidden="true" tabindex="-1"></a>        SDs<span class="ot">=</span><span class="fu">apply</span>(convertedData.out,<span class="dv">1</span>,sd)</span>
<span id="cb182-1215"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1215" aria-hidden="true" tabindex="-1"></a>        maxSD <span class="ot">=</span> <span class="fu">mean</span>(SDs)<span class="sc">+</span> <span class="dv">4</span><span class="sc">*</span><span class="fu">sd</span>(SDs)</span>
<span id="cb182-1216"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1216" aria-hidden="true" tabindex="-1"></a>        SDs[ SDs <span class="sc">&gt;</span> maxSD] <span class="ot">=</span> maxSD</span>
<span id="cb182-1217"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1217" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1218"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1218" aria-hidden="true" tabindex="-1"></a>        top <span class="ot">=</span> input_nGenesKNN</span>
<span id="cb182-1219"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1219" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(top <span class="sc">&gt;</span> <span class="fu">length</span>(SDs)) top <span class="ot">=</span> <span class="fu">length</span>(SDs)</span>
<span id="cb182-1220"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1220" aria-hidden="true" tabindex="-1"></a>        Cutoff<span class="ot">=</span><span class="fu">sort</span>(SDs,<span class="at">decreasing=</span><span class="cn">TRUE</span>)[top] </span>
<span id="cb182-1221"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1222"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1222" aria-hidden="true" tabindex="-1"></a>        SDs <span class="ot">=</span> <span class="fu">as.data.frame</span>(SDs)</span>
<span id="cb182-1223"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1224"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1224" aria-hidden="true" tabindex="-1"></a>        p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(SDs, <span class="fu">aes</span>(<span class="at">x=</span>SDs)) <span class="sc">+</span> </span>
<span id="cb182-1225"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1225" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_density</span>(<span class="at">color=</span><span class="st">&quot;darkblue&quot;</span>, <span class="at">fill=</span><span class="st">&quot;lightblue&quot;</span>) <span class="sc">+</span></span>
<span id="cb182-1226"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1226" aria-hidden="true" tabindex="-1"></a>          <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Standard deviations of all genes&quot;</span>, <span class="at">y=</span><span class="st">&quot;Density&quot;</span>)<span class="sc">+</span></span>
<span id="cb182-1227"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1227" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_vline</span>(<span class="fu">aes</span>(<span class="at">xintercept=</span>Cutoff),</span>
<span id="cb182-1228"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1228" aria-hidden="true" tabindex="-1"></a>                    <span class="at">color=</span><span class="st">&quot;red&quot;</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb182-1229"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1229" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="at">x =</span> Cutoff <span class="sc">+</span> <span class="fl">0.4</span><span class="sc">*</span><span class="fu">sd</span>(SDs[,<span class="dv">1</span>]), <span class="at">y =</span> <span class="dv">1</span>,<span class="at">colour =</span> <span class="st">&quot;red&quot;</span>, <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;Top &quot;</span>, top))</span>
<span id="cb182-1230"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1231"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1231" aria-hidden="true" tabindex="-1"></a>        p</span>
<span id="cb182-1232"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1232" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-1233"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1234"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1235"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1235" aria-hidden="true" tabindex="-1"></a><span class="co"># Decide number of clusters</span></span>
<span id="cb182-1236"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1236" aria-hidden="true" tabindex="-1"></a>KmeansNclusters <span class="ot">&lt;-</span> <span class="cf">function</span>() { <span class="co"># Kmeans clustering</span></span>
<span id="cb182-1237"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1238"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1238" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> convertedData.out</span>
<span id="cb182-1239"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1239" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x &lt;- readData.out</span></span>
<span id="cb182-1240"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1240" aria-hidden="true" tabindex="-1"></a>    <span class="co">#par(mfrow=c(1,2))</span></span>
<span id="cb182-1241"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1241" aria-hidden="true" tabindex="-1"></a>    n<span class="ot">=</span>input_nGenesKNN</span>
<span id="cb182-1242"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1242" aria-hidden="true" tabindex="-1"></a>    <span class="co">#if(n&gt;6000) n = 6000 # max</span></span>
<span id="cb182-1243"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1243" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(n<span class="sc">&gt;</span><span class="fu">dim</span>(x)[<span class="dv">1</span>]) n <span class="ot">=</span> <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="co"># max as data</span></span>
<span id="cb182-1244"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1244" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x1 &lt;- x;</span></span>
<span id="cb182-1245"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1245" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x=as.matrix(x[1:n,])-apply(x[1:n,],1,mean)</span></span>
<span id="cb182-1246"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1246" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">=</span> <span class="dv">100</span><span class="sc">*</span> x[<span class="dv">1</span><span class="sc">:</span>n,] <span class="sc">/</span> <span class="fu">apply</span>(x[<span class="dv">1</span><span class="sc">:</span>n,],<span class="dv">1</span>,sum)  <span class="co"># this is causing problem??????</span></span>
<span id="cb182-1247"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1247" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x = x - apply(x,1,mean)  # this is causing problem??????</span></span>
<span id="cb182-1248"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1248" aria-hidden="true" tabindex="-1"></a>    <span class="co">#colnames(x) = gsub(&quot;_.*&quot;,&quot;&quot;,colnames(x))</span></span>
<span id="cb182-1249"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1249" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set.seed</span>(<span class="dv">2</span>)</span>
<span id="cb182-1250"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1250" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1251"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1251" aria-hidden="true" tabindex="-1"></a>    k <span class="ot">=</span> <span class="dv">30</span></span>
<span id="cb182-1252"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1252" aria-hidden="true" tabindex="-1"></a>    wss <span class="ot">&lt;-</span> (<span class="fu">nrow</span>(x)<span class="sc">-</span><span class="dv">1</span>)<span class="sc">*</span><span class="fu">sum</span>(<span class="fu">apply</span>(x,<span class="dv">2</span>,var))</span>
<span id="cb182-1253"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1253" aria-hidden="true" tabindex="-1"></a>      <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span>k) wss[i] <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">kmeans</span>(x,<span class="at">centers=</span>i,<span class="at">iter.max =</span> <span class="dv">30</span>)<span class="sc">$</span>withinss)</span>
<span id="cb182-1254"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1254" aria-hidden="true" tabindex="-1"></a>        <span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">4</span>,<span class="dv">4</span>))</span>
<span id="cb182-1255"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1255" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot</span>(<span class="dv">1</span><span class="sc">:</span>k, wss, <span class="at">type=</span><span class="st">&quot;b&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;Number of Clusters (k)&quot;</span>,</span>
<span id="cb182-1256"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1256" aria-hidden="true" tabindex="-1"></a>         <span class="at">ylab=</span><span class="st">&quot;Within groups sum of squares&quot;</span>,</span>
<span id="cb182-1257"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1257" aria-hidden="true" tabindex="-1"></a>         <span class="at">cex=</span><span class="dv">2</span>,<span class="at">cex.lab=</span><span class="dv">2</span>, <span class="at">cex.axis=</span><span class="dv">2</span>, <span class="at">cex.main=</span><span class="dv">2</span>, <span class="at">cex.sub=</span><span class="dv">2</span> ,<span class="at">xaxt=</span><span class="st">&quot;n&quot;</span>    )</span>
<span id="cb182-1258"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1258" aria-hidden="true" tabindex="-1"></a>    <span class="fu">axis</span>(<span class="dv">1</span>, <span class="at">at =</span> <span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">30</span>, <span class="at">by =</span> <span class="dv">2</span>),<span class="at">cex.axis=</span><span class="fl">1.5</span>,<span class="at">cex=</span><span class="fl">1.5</span>)</span>
<span id="cb182-1259"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1259" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1260"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1260" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-1261"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1261" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb182-1262"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1262" aria-hidden="true" tabindex="-1"></a>Kmeans <span class="ot">&lt;-</span> <span class="cf">function</span>() { <span class="co"># Kmeans clustering</span></span>
<span id="cb182-1263"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1264"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1264" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> convertedData.out</span>
<span id="cb182-1265"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1265" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x &lt;- readData.out</span></span>
<span id="cb182-1266"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1266" aria-hidden="true" tabindex="-1"></a>    <span class="co">#par(mfrow=c(1,2))</span></span>
<span id="cb182-1267"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1267" aria-hidden="true" tabindex="-1"></a>    n<span class="ot">=</span>input_nGenesKNN</span>
<span id="cb182-1268"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1268" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(n<span class="sc">&gt;</span>maxGeneClustering) n <span class="ot">=</span> maxGeneClustering <span class="co"># max</span></span>
<span id="cb182-1269"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1269" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(n<span class="sc">&gt;</span><span class="fu">dim</span>(x)[<span class="dv">1</span>]) n <span class="ot">=</span> <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="co"># max as data</span></span>
<span id="cb182-1270"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1270" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x1 &lt;- x;</span></span>
<span id="cb182-1271"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1271" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x=as.matrix(x[1:n,])-apply(x[1:n,],1,mean)</span></span>
<span id="cb182-1272"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1272" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x = 100* x[1:n,] / apply(x[1:n,],1,sum) </span></span>
<span id="cb182-1273"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1273" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">=</span> x[<span class="dv">1</span><span class="sc">:</span>n,]</span>
<span id="cb182-1274"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1274" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_kmeansNormalization <span class="sc">==</span> <span class="st">&#39;L1Norm&#39;</span>)</span>
<span id="cb182-1275"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1275" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">=</span> <span class="dv">100</span><span class="sc">*</span> x <span class="sc">/</span> <span class="fu">apply</span>(x,<span class="dv">1</span>,<span class="cf">function</span>(y) <span class="fu">sum</span>(<span class="fu">abs</span>(y))) <span class="cf">else</span> <span class="co"># L1 norm</span></span>
<span id="cb182-1276"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1276" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_kmeansNormalization <span class="sc">==</span> <span class="st">&#39;geneMean&#39;</span>)</span>
<span id="cb182-1277"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1277" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">=</span> x <span class="sc">-</span> <span class="fu">apply</span>(x,<span class="dv">1</span>,mean)  <span class="cf">else</span> <span class="co"># this is causing problem??????</span></span>
<span id="cb182-1278"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1278" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_kmeansNormalization <span class="sc">==</span> <span class="st">&#39;geneStandardization&#39;</span>) </span>
<span id="cb182-1279"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1279" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">=</span> (x <span class="sc">-</span> <span class="fu">apply</span>(x,<span class="dv">1</span>,mean) ) <span class="sc">/</span> <span class="fu">apply</span>(x,<span class="dv">1</span>,sd)</span>
<span id="cb182-1280"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1280" aria-hidden="true" tabindex="-1"></a>    <span class="co">#colnames(x) = gsub(&quot;_.*&quot;,&quot;&quot;,colnames(x))</span></span>
<span id="cb182-1281"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1281" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set.seed</span>(input_KmeansReRun)</span>
<span id="cb182-1282"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1282" aria-hidden="true" tabindex="-1"></a>    k<span class="ot">=</span>input_nClusters</span>
<span id="cb182-1283"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1284"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1284" aria-hidden="true" tabindex="-1"></a>    cl <span class="ot">=</span> <span class="fu">kmeans</span>(x,k,<span class="at">iter.max =</span> <span class="dv">50</span>)</span>
<span id="cb182-1285"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1285" aria-hidden="true" tabindex="-1"></a>    <span class="co">#myheatmap(cl$centers)  </span></span>
<span id="cb182-1286"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1286" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb182-1287"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1288"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1288" aria-hidden="true" tabindex="-1"></a>    hc <span class="ot">&lt;-</span> <span class="fu">hclust2</span>(<span class="fu">dist2</span>(cl<span class="sc">$</span>centers<span class="sc">-</span><span class="fu">apply</span>(cl<span class="sc">$</span>centers,<span class="dv">1</span>,mean) )  )<span class="co"># perform cluster for the reordering of samples</span></span>
<span id="cb182-1289"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1289" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> <span class="fu">match</span>(cl<span class="sc">$</span>cluster,hc<span class="sc">$</span>order) <span class="co">#  new order </span></span>
<span id="cb182-1290"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1290" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">=</span> x[<span class="fu">order</span>(tem),] ;    bar <span class="ot">=</span> <span class="fu">sort</span>(tem)</span>
<span id="cb182-1291"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1292"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1292" aria-hidden="true" tabindex="-1"></a>    <span class="co">#myheatmap2(x-apply(x,1,mean), bar,1000)</span></span>
<span id="cb182-1293"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1293" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">list</span>( <span class="at">x =</span> x, <span class="at">bar =</span> bar)) </span>
<span id="cb182-1294"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1295"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1295" aria-hidden="true" tabindex="-1"></a>  } </span>
<span id="cb182-1296"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1297"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1298"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1298" aria-hidden="true" tabindex="-1"></a>myheatmap2 <span class="ot">&lt;-</span> <span class="cf">function</span> (x,<span class="at">bar=</span><span class="cn">NULL</span>,<span class="at">n=</span><span class="sc">-</span><span class="dv">1</span>,<span class="at">mycolor=</span><span class="dv">1</span>,<span class="at">clusterNames=</span><span class="cn">NULL</span>,<span class="at">sideColors=</span><span class="cn">NULL</span> ) {</span>
<span id="cb182-1299"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1299" aria-hidden="true" tabindex="-1"></a>    <span class="co"># number of genes to show</span></span>
<span id="cb182-1300"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1300" aria-hidden="true" tabindex="-1"></a>    ngenes <span class="ot">=</span> <span class="fu">as.character</span>( <span class="fu">table</span>(bar))</span>
<span id="cb182-1301"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1301" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(bar) <span class="sc">&gt;</span>n <span class="sc">&amp;&amp;</span> n <span class="sc">!=</span> <span class="sc">-</span><span class="dv">1</span>) {ix <span class="ot">=</span> <span class="fu">sort</span>( <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(bar),n) ); bar <span class="ot">=</span> bar[ix]; x <span class="ot">=</span> x[ix,]  }</span>
<span id="cb182-1302"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1302" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span> <span class="fu">is.null</span>(bar) )</span>
<span id="cb182-1303"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1303" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>(sideColors) ) </span>
<span id="cb182-1304"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1304" aria-hidden="true" tabindex="-1"></a>            sideColors <span class="ot">=</span> mycolors</span>
<span id="cb182-1305"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1306"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1306" aria-hidden="true" tabindex="-1"></a>    <span class="co"># this will cutoff very large values, which could skew the color </span></span>
<span id="cb182-1307"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1307" aria-hidden="true" tabindex="-1"></a>    x<span class="ot">=</span><span class="fu">as.matrix</span>(x)<span class="sc">-</span><span class="fu">apply</span>(x,<span class="dv">1</span>,mean)</span>
<span id="cb182-1308"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1308" aria-hidden="true" tabindex="-1"></a>    cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">+</span> <span class="dv">3</span><span class="sc">*</span><span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb182-1309"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1309" aria-hidden="true" tabindex="-1"></a>    x[x<span class="sc">&gt;</span>cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb182-1310"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1310" aria-hidden="true" tabindex="-1"></a>    cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">-</span> <span class="dv">3</span><span class="sc">*</span><span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb182-1311"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1311" aria-hidden="true" tabindex="-1"></a>    x[x<span class="sc">&lt;</span> cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb182-1312"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1312" aria-hidden="true" tabindex="-1"></a>    <span class="co">#colnames(x)= detectGroups(colnames(x))</span></span>
<span id="cb182-1313"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1313" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(bar)) <span class="co"># no side colors</span></span>
<span id="cb182-1314"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1314" aria-hidden="true" tabindex="-1"></a>        <span class="fu">heatmap.2</span>(x,  <span class="at">Rowv =</span>F,<span class="at">Colv=</span>F, <span class="at">dendrogram =</span><span class="st">&quot;none&quot;</span>,</span>
<span id="cb182-1315"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1315" aria-hidden="true" tabindex="-1"></a>            <span class="at">col=</span>heatColors[<span class="fu">as.integer</span>(mycolor),], <span class="at">density.info=</span><span class="st">&quot;none&quot;</span>, <span class="at">trace=</span><span class="st">&quot;none&quot;</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">keysize=</span>.<span class="dv">3</span></span>
<span id="cb182-1316"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1316" aria-hidden="true" tabindex="-1"></a>            ,<span class="at">key=</span>F, <span class="at">labRow =</span> F</span>
<span id="cb182-1317"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1317" aria-hidden="true" tabindex="-1"></a>            <span class="co">#,RowSideColors = mycolors[bar]</span></span>
<span id="cb182-1318"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1318" aria-hidden="true" tabindex="-1"></a>            ,<span class="at">margins =</span> <span class="fu">c</span>(<span class="dv">8</span>, <span class="dv">24</span>)</span>
<span id="cb182-1319"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1319" aria-hidden="true" tabindex="-1"></a>            ,<span class="at">srtCol=</span><span class="dv">45</span></span>
<span id="cb182-1320"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1320" aria-hidden="true" tabindex="-1"></a>        ) <span class="cf">else</span></span>
<span id="cb182-1321"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1321" aria-hidden="true" tabindex="-1"></a>        <span class="fu">heatmap.2</span>(x,  <span class="at">Rowv =</span>F,<span class="at">Colv=</span>F, <span class="at">dendrogram =</span><span class="st">&quot;none&quot;</span>,</span>
<span id="cb182-1322"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1322" aria-hidden="true" tabindex="-1"></a>            <span class="at">col=</span>heatColors[<span class="fu">as.integer</span>(mycolor),], <span class="at">density.info=</span><span class="st">&quot;none&quot;</span>, <span class="at">trace=</span><span class="st">&quot;none&quot;</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">keysize=</span>.<span class="dv">3</span></span>
<span id="cb182-1323"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1323" aria-hidden="true" tabindex="-1"></a>            ,<span class="at">key=</span>F, <span class="at">labRow =</span> F</span>
<span id="cb182-1324"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1324" aria-hidden="true" tabindex="-1"></a>            ,<span class="at">RowSideColors =</span> sideColors[bar]</span>
<span id="cb182-1325"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1325" aria-hidden="true" tabindex="-1"></a>            ,<span class="at">margins =</span> <span class="fu">c</span>(<span class="dv">8</span>, <span class="dv">24</span>)</span>
<span id="cb182-1326"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1326" aria-hidden="true" tabindex="-1"></a>            ,<span class="at">srtCol=</span><span class="dv">45</span></span>
<span id="cb182-1327"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1327" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb182-1328"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1328" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1329"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1329" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(bar)) { </span>
<span id="cb182-1330"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1331"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1331" aria-hidden="true" tabindex="-1"></a>        legend.text <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;Cluster &quot;</span>, <span class="fu">toupper</span>(letters)[<span class="fu">unique</span>(bar)], <span class="st">&quot; (N=&quot;</span>, ngenes,<span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>) </span>
<span id="cb182-1332"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1332" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>( clusterNames ) <span class="sc">&amp;&amp;</span> <span class="fu">length</span>(clusterNames)<span class="sc">&gt;=</span> <span class="fu">length</span>( <span class="fu">unique</span>(bar) ) )  </span>
<span id="cb182-1333"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1333" aria-hidden="true" tabindex="-1"></a>            legend.text <span class="ot">=</span> <span class="fu">paste</span>(clusterNames[ <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>( <span class="fu">unique</span>(bar) )  ], <span class="st">&quot; (N=&quot;</span>, ngenes,<span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>) </span>
<span id="cb182-1334"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1334" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1335"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1335" aria-hidden="true" tabindex="-1"></a>        <span class="fu">par</span>(<span class="at">lend =</span> <span class="dv">1</span>)           <span class="co"># square line ends for the color legend</span></span>
<span id="cb182-1336"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1336" aria-hidden="true" tabindex="-1"></a>        <span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>,      <span class="co"># location of the legend on the heatmap plot</span></span>
<span id="cb182-1337"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1337" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend =</span> legend.text, <span class="co"># category labels</span></span>
<span id="cb182-1338"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1338" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> sideColors,  <span class="co"># color key</span></span>
<span id="cb182-1339"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1339" aria-hidden="true" tabindex="-1"></a>        <span class="at">lty=</span> <span class="dv">1</span>,             <span class="co"># line style</span></span>
<span id="cb182-1340"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1340" aria-hidden="true" tabindex="-1"></a>        <span class="at">lwd =</span> <span class="dv">10</span> )           <span class="co"># line width</span></span>
<span id="cb182-1341"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1341" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-1342"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1342" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-1343"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1344"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1344" aria-hidden="true" tabindex="-1"></a>KmeansHeatmap <span class="ot">&lt;-</span> <span class="cf">function</span>() { <span class="co"># Kmeans clustering</span></span>
<span id="cb182-1345"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1346"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1346" aria-hidden="true" tabindex="-1"></a>    <span class="fu">myheatmap2</span>(Kmeans.out<span class="sc">$</span>x<span class="sc">-</span><span class="fu">apply</span>(Kmeans.out<span class="sc">$</span>x,<span class="dv">1</span>,mean), Kmeans.out<span class="sc">$</span>bar,<span class="dv">1000</span>,<span class="at">mycolor=</span>input_heatColors1)</span>
<span id="cb182-1347"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1347" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-1348"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1348" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb182-1349"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1350"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1350" aria-hidden="true" tabindex="-1"></a>tSNEgenePlot <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb182-1351"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1351" aria-hidden="true" tabindex="-1"></a>            Cluster <span class="ot">&lt;-</span> Kmeans.out<span class="sc">$</span>bar</span>
<span id="cb182-1352"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1352" aria-hidden="true" tabindex="-1"></a>            train <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>( <span class="fu">cbind</span>(Cluster,Kmeans.out<span class="sc">$</span>x) )</span>
<span id="cb182-1353"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1354"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1354" aria-hidden="true" tabindex="-1"></a>            <span class="fu">library</span>(Rtsne,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb182-1355"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1356"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1356" aria-hidden="true" tabindex="-1"></a>            train <span class="ot">=</span> <span class="fu">unique</span>(train)</span>
<span id="cb182-1357"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1357" aria-hidden="true" tabindex="-1"></a>            Cluster <span class="ot">=</span> train<span class="sc">$</span>Cluster </span>
<span id="cb182-1358"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1358" aria-hidden="true" tabindex="-1"></a>            <span class="fu">set.seed</span>(input_seedTSNE)</span>
<span id="cb182-1359"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1359" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-1360"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1360" aria-hidden="true" tabindex="-1"></a>            <span class="do">## Executing the algorithm on curated data</span></span>
<span id="cb182-1361"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1361" aria-hidden="true" tabindex="-1"></a>            tsne <span class="ot">&lt;-</span> <span class="fu">Rtsne</span>(train[,<span class="sc">-</span><span class="dv">1</span>], <span class="at">dims =</span> <span class="dv">2</span>, <span class="at">perplexity=</span><span class="dv">30</span>, <span class="at">verbose=</span><span class="cn">FALSE</span>, <span class="at">max_iter =</span> <span class="dv">400</span>)</span>
<span id="cb182-1362"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1363"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1364"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1364" aria-hidden="true" tabindex="-1"></a>            nClusters <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">unique</span>(Cluster) )</span>
<span id="cb182-1365"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1365" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(input_colorGenes) {          </span>
<span id="cb182-1366"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1366" aria-hidden="true" tabindex="-1"></a>                <span class="fu">plot</span>(tsne<span class="sc">$</span>Y[,<span class="dv">1</span>], tsne<span class="sc">$</span>Y[,<span class="dv">2</span>], <span class="at">pch =</span> (<span class="dv">0</span><span class="sc">:</span>(nClusters<span class="dv">-1</span>))[Cluster], <span class="at">cex =</span> <span class="fl">1.</span>,<span class="at">col =</span> mycolors[Cluster], <span class="at">xlab=</span><span class="st">&quot;X&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Y&quot;</span>)</span>
<span id="cb182-1367"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1367" aria-hidden="true" tabindex="-1"></a>                <span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>,<span class="fu">toupper</span>(letters)[<span class="dv">1</span><span class="sc">:</span>nClusters], <span class="at">pch =</span> <span class="dv">0</span><span class="sc">:</span>(nClusters<span class="dv">-1</span>), <span class="at">col=</span>mycolors, <span class="at">title=</span><span class="st">&quot;Cluster&quot;</span>  )</span>
<span id="cb182-1368"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1368" aria-hidden="true" tabindex="-1"></a>            } <span class="cf">else</span></span>
<span id="cb182-1369"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1369" aria-hidden="true" tabindex="-1"></a>                <span class="fu">plot</span>(tsne<span class="sc">$</span>Y[,<span class="dv">1</span>], tsne<span class="sc">$</span>Y[,<span class="dv">2</span>],  <span class="at">cex =</span> <span class="fl">1.</span>, <span class="at">xlab=</span><span class="st">&quot;X&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Y&quot;</span>)</span>
<span id="cb182-1370"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1371"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1371" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-1372"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1373"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1374"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1374" aria-hidden="true" tabindex="-1"></a><span class="co"># Main function. Find a query set of genes enriched with functional category</span></span>
<span id="cb182-1375"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1375" aria-hidden="true" tabindex="-1"></a>FindOverlap <span class="ot">&lt;-</span> <span class="cf">function</span> (converted,gInfo, GO,selectOrg,minFDR, <span class="at">reduced =</span> <span class="cn">FALSE</span>) {</span>
<span id="cb182-1376"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1376" aria-hidden="true" tabindex="-1"></a>    maxTerms <span class="ot">=</span><span class="dv">15</span> <span class="co"># max number of enriched terms</span></span>
<span id="cb182-1377"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1377" aria-hidden="true" tabindex="-1"></a>    idNotRecognized <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;ID not recognized!&quot;</span>)</span>
<span id="cb182-1378"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1378" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1379"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1379" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(converted) ) <span class="fu">return</span>(idNotRecognized) <span class="co"># no ID </span></span>
<span id="cb182-1380"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1380" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1381"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1381" aria-hidden="true" tabindex="-1"></a>    <span class="co"># only coding</span></span>
<span id="cb182-1382"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1382" aria-hidden="true" tabindex="-1"></a>    gInfo <span class="ot">&lt;-</span> gInfo[<span class="fu">which</span>( gInfo<span class="sc">$</span>gene_biotype <span class="sc">==</span> <span class="st">&quot;protein_coding&quot;</span>),]  </span>
<span id="cb182-1383"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1383" aria-hidden="true" tabindex="-1"></a>    querySet <span class="ot">&lt;-</span> <span class="fu">intersect</span>( converted<span class="sc">$</span>IDs, gInfo[,<span class="dv">1</span>]);</span>
<span id="cb182-1384"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1384" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1385"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1385" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(querySet) <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>(idNotRecognized )</span>
<span id="cb182-1386"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1386" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1387"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1387" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span> <span class="fu">grep</span>(converted<span class="sc">$</span>species[<span class="dv">1</span>,<span class="dv">1</span>],gmtFiles)</span>
<span id="cb182-1388"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1388" aria-hidden="true" tabindex="-1"></a>    totalGenes <span class="ot">&lt;-</span> converted<span class="sc">$</span>species[<span class="dv">1</span>,<span class="dv">7</span>]</span>
<span id="cb182-1389"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1389" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1390"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1390" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> ) {<span class="fu">return</span>(idNotRecognized )}</span>
<span id="cb182-1391"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1391" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1392"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1392" aria-hidden="true" tabindex="-1"></a>    <span class="co"># If selected species is not the default &quot;bestMatch&quot;, use that species directly</span></span>
<span id="cb182-1393"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1393" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(selectOrg <span class="sc">!=</span> speciesChoice[[<span class="dv">1</span>]]) {  </span>
<span id="cb182-1394"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1394" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">grep</span>(<span class="fu">findSpeciesById</span>(selectOrg)[<span class="dv">1</span>,<span class="dv">1</span>], gmtFiles )</span>
<span id="cb182-1395"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1395" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> ) {<span class="fu">return</span>(idNotRecognized )}</span>
<span id="cb182-1396"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1396" aria-hidden="true" tabindex="-1"></a>        totalGenes <span class="ot">&lt;-</span> orgInfo[<span class="fu">which</span>(orgInfo<span class="sc">$</span>id <span class="sc">==</span> <span class="fu">as.numeric</span>(selectOrg)),<span class="dv">7</span>]</span>
<span id="cb182-1397"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1397" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-1398"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1398" aria-hidden="true" tabindex="-1"></a>    pathway <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(sqlite,gmtFiles[ix],<span class="at">flags=</span>SQLITE_RO)</span>
<span id="cb182-1399"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1399" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1400"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1400" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1401"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1401" aria-hidden="true" tabindex="-1"></a>    sqlQuery <span class="ot">=</span> <span class="fu">paste</span>( <span class="st">&quot; select distinct gene,pathwayID from pathway where gene IN (&#39;&quot;</span>, <span class="fu">paste</span>(querySet,<span class="at">collapse=</span><span class="st">&quot;&#39;, &#39;&quot;</span>),<span class="st">&quot;&#39;)&quot;</span> ,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb182-1402"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1402" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1403"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1403" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(paste0(&quot;HH&quot;,GO,&quot;HH&quot;) )</span></span>
<span id="cb182-1404"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1404" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1405"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1405" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( GO <span class="sc">!=</span> <span class="st">&quot;All&quot;</span>) sqlQuery <span class="ot">=</span> <span class="fu">paste0</span>(sqlQuery, <span class="st">&quot; AND category =&#39;&quot;</span>,GO,<span class="st">&quot;&#39;&quot;</span>)</span>
<span id="cb182-1406"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1406" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>( pathway, sqlQuery  )</span>
<span id="cb182-1407"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1407" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">1</span>] <span class="sc">==</span><span class="dv">0</span>) {<span class="fu">return</span>(<span class="fu">as.data.frame</span>(<span class="st">&quot;No matching species or gene ID file!&quot;</span> )) }</span>
<span id="cb182-1408"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1408" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1409"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1409" aria-hidden="true" tabindex="-1"></a>    <span class="co"># given a pathway id, it finds the overlapped genes, symbol preferred</span></span>
<span id="cb182-1410"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1410" aria-hidden="true" tabindex="-1"></a>    sharedGenesPrefered <span class="ot">&lt;-</span> <span class="cf">function</span>(pathwayID) {</span>
<span id="cb182-1411"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1411" aria-hidden="true" tabindex="-1"></a>        tem <span class="ot">&lt;-</span> result[<span class="fu">which</span>(result[,<span class="dv">2</span>]<span class="sc">==</span> pathwayID ),<span class="dv">1</span>]</span>
<span id="cb182-1412"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1412" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">match</span>(tem, converted<span class="sc">$</span>conversionTable<span class="sc">$</span>ensembl_gene_id) <span class="co"># convert back to original</span></span>
<span id="cb182-1413"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1413" aria-hidden="true" tabindex="-1"></a>        tem2 <span class="ot">&lt;-</span> <span class="fu">unique</span>( converted<span class="sc">$</span>conversionTable<span class="sc">$</span>User_input[ix] )</span>
<span id="cb182-1414"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1414" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">length</span>(<span class="fu">unique</span>(gInfo<span class="sc">$</span>symbol) )<span class="sc">/</span><span class="fu">dim</span>(gInfo)[<span class="dv">1</span>] <span class="sc">&gt;</span>.<span class="dv">7</span>  ) <span class="co"># if 70% genes has symbol in geneInfo</span></span>
<span id="cb182-1415"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1415" aria-hidden="true" tabindex="-1"></a>        { ix <span class="ot">=</span> <span class="fu">match</span>(tem, gInfo<span class="sc">$</span>ensembl_gene_id); </span>
<span id="cb182-1416"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1416" aria-hidden="true" tabindex="-1"></a>        tem2 <span class="ot">&lt;-</span> <span class="fu">unique</span>( gInfo<span class="sc">$</span>symbol[ix] )      }</span>
<span id="cb182-1417"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1417" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">paste</span>( tem2 ,<span class="at">collapse=</span><span class="st">&quot; &quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>) )}</span>
<span id="cb182-1418"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1418" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1419"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1419" aria-hidden="true" tabindex="-1"></a>    x0 <span class="ot">=</span> <span class="fu">table</span>(result<span class="sc">$</span>pathwayID)                    </span>
<span id="cb182-1420"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1420" aria-hidden="true" tabindex="-1"></a>    x0 <span class="ot">=</span> <span class="fu">as.data.frame</span>( x0[<span class="fu">which</span>(x0<span class="sc">&gt;=</span>Min_overlap)] )<span class="co"># remove low overlaps</span></span>
<span id="cb182-1421"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1421" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">dim</span>(x0)[<span class="dv">1</span>] <span class="sc">&lt;=</span> <span class="dv">5</span> ) <span class="fu">return</span>(idNotRecognized) <span class="co"># no data</span></span>
<span id="cb182-1422"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1422" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(x0)<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;pathwayID&quot;</span>,<span class="st">&quot;overlap&quot;</span>)</span>
<span id="cb182-1423"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1423" aria-hidden="true" tabindex="-1"></a>    pathwayInfo <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>( pathway, <span class="fu">paste</span>( <span class="st">&quot; select distinct id,n,Description from pathwayInfo where id IN (&#39;&quot;</span>, </span>
<span id="cb182-1424"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1424" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">paste</span>(x0<span class="sc">$</span>pathwayID,<span class="at">collapse=</span><span class="st">&quot;&#39;, &#39;&quot;</span>),   <span class="st">&quot;&#39;) &quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>) )</span>
<span id="cb182-1425"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1425" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1426"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1426" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">=</span> <span class="fu">merge</span>(x0,pathwayInfo, <span class="at">by.x=</span><span class="st">&#39;pathwayID&#39;</span>, <span class="at">by.y=</span><span class="st">&#39;id&#39;</span>)</span>
<span id="cb182-1427"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1427" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1428"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1428" aria-hidden="true" tabindex="-1"></a>    x<span class="sc">$</span>Pval<span class="ot">=</span><span class="fu">phyper</span>(x<span class="sc">$</span>overlap<span class="dv">-1</span>,<span class="fu">length</span>(querySet),totalGenes <span class="sc">-</span> <span class="fu">length</span>(querySet),<span class="fu">as.numeric</span>(x<span class="sc">$</span>n), <span class="at">lower.tail=</span><span class="cn">FALSE</span> );</span>
<span id="cb182-1429"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1429" aria-hidden="true" tabindex="-1"></a>    x<span class="sc">$</span>FDR <span class="ot">=</span> <span class="fu">p.adjust</span>(x<span class="sc">$</span>Pval,<span class="at">method=</span><span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb182-1430"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1430" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> x[ <span class="fu">order</span>( x<span class="sc">$</span>FDR)  ,]  <span class="co"># sort according to FDR</span></span>
<span id="cb182-1431"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1431" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1432"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1432" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="sc">&gt;</span> maxTerms ) x <span class="ot">=</span> x[<span class="dv">1</span><span class="sc">:</span>maxTerms,]    </span>
<span id="cb182-1433"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1433" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1434"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1434" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">min</span>(x<span class="sc">$</span>FDR) <span class="sc">&gt;</span> minFDR) x<span class="ot">=</span><span class="fu">as.data.frame</span>(<span class="st">&quot;No significant enrichment found!&quot;</span>) <span class="cf">else</span> {</span>
<span id="cb182-1435"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1435" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> x[<span class="fu">which</span>(x<span class="sc">$</span>FDR <span class="sc">&lt;</span> minFDR),] </span>
<span id="cb182-1436"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1436" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1437"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1437" aria-hidden="true" tabindex="-1"></a>        x<span class="ot">=</span> <span class="fu">cbind</span>(x,<span class="fu">sapply</span>( x<span class="sc">$</span>pathwayID, sharedGenesPrefered ) )</span>
<span id="cb182-1438"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1438" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(x)[<span class="dv">7</span>]<span class="ot">=</span> <span class="st">&quot;Genes&quot;</span></span>
<span id="cb182-1439"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1439" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">&lt;-</span> <span class="fu">subset</span>(x,<span class="at">select =</span> <span class="fu">c</span>(FDR,overlap,n,description,Genes) )</span>
<span id="cb182-1440"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1440" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(x) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Corrected P value (FDR)&quot;</span>, <span class="st">&quot;Genes in list&quot;</span>, <span class="st">&quot;Total genes in category&quot;</span>,<span class="st">&quot;Functional Category&quot;</span>,<span class="st">&quot;Genes&quot;</span>  )</span>
<span id="cb182-1441"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1441" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1442"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1442" aria-hidden="true" tabindex="-1"></a>        <span class="co"># remove redudant gene sets</span></span>
<span id="cb182-1443"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1443" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(reduced <span class="sc">!=</span> <span class="cn">FALSE</span> ){  <span class="co"># reduced=FALSE no filtering,  reduced = 0.9 filter sets overlap with 90%</span></span>
<span id="cb182-1444"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1444" aria-hidden="true" tabindex="-1"></a>            n<span class="ot">=</span>  <span class="fu">nrow</span>(x)</span>
<span id="cb182-1445"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1445" aria-hidden="true" tabindex="-1"></a>            tem<span class="ot">=</span><span class="fu">rep</span>(<span class="cn">TRUE</span>,n )</span>
<span id="cb182-1446"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1446" aria-hidden="true" tabindex="-1"></a>            geneLists <span class="ot">=</span> <span class="fu">lapply</span>(x<span class="sc">$</span>Genes, <span class="cf">function</span>(y) <span class="fu">unlist</span>( <span class="fu">strsplit</span>(<span class="fu">as.character</span>(y),<span class="st">&quot; &quot;</span> )   ) )</span>
<span id="cb182-1447"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1447" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span>n)</span>
<span id="cb182-1448"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1448" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span>( j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(i<span class="dv">-1</span>) ) { </span>
<span id="cb182-1449"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1449" aria-hidden="true" tabindex="-1"></a>                  <span class="cf">if</span>(tem[j]) { <span class="co"># skip if this one is already removed</span></span>
<span id="cb182-1450"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1450" aria-hidden="true" tabindex="-1"></a>                      commonGenes <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">intersect</span>(geneLists[i] ,geneLists[j] ) )</span>
<span id="cb182-1451"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1451" aria-hidden="true" tabindex="-1"></a>                      <span class="cf">if</span>( commonGenes<span class="sc">/</span> <span class="fu">length</span>(geneLists[j] ) <span class="sc">&gt;</span> reduced )</span>
<span id="cb182-1452"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1452" aria-hidden="true" tabindex="-1"></a>                        tem[i] <span class="ot">=</span> <span class="cn">FALSE</span>  </span>
<span id="cb182-1453"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1453" aria-hidden="true" tabindex="-1"></a>                  }         </span>
<span id="cb182-1454"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1454" aria-hidden="true" tabindex="-1"></a>                }                               </span>
<span id="cb182-1455"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1455" aria-hidden="true" tabindex="-1"></a>            x <span class="ot">&lt;-</span> x[<span class="fu">which</span>(tem),]     </span>
<span id="cb182-1456"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1456" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-1457"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1457" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1458"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1459"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1459" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-1460"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1460" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-1461"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1461" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dbDisconnect</span>(pathway)</span>
<span id="cb182-1462"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1462" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(x )</span>
<span id="cb182-1463"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1463" aria-hidden="true" tabindex="-1"></a>} </span>
<span id="cb182-1464"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1464" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb182-1465"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1465" aria-hidden="true" tabindex="-1"></a><span class="co"># Given a gene set, finds significant overlaps with a gene set database  object </span></span>
<span id="cb182-1466"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1466" aria-hidden="true" tabindex="-1"></a>findOverlapGMT <span class="ot">&lt;-</span> <span class="cf">function</span> ( query, geneSet, <span class="at">minFDR=</span>.<span class="dv">2</span> ,<span class="at">minSize=</span><span class="dv">2</span>,<span class="at">maxSize=</span><span class="dv">10000</span> ){ </span>
<span id="cb182-1467"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1467" aria-hidden="true" tabindex="-1"></a>    <span class="co">#geneSets &lt;- readGMT(&quot;exampleData/MousePath_TF_gmt.gmt&quot;)</span></span>
<span id="cb182-1468"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1468" aria-hidden="true" tabindex="-1"></a>    <span class="co">#query &lt;-  geneSets[[&#39;TF_MM_FRIARD_C-REL&#39;]] </span></span>
<span id="cb182-1469"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1469" aria-hidden="true" tabindex="-1"></a>    <span class="co">#query &lt;- query[1:60]</span></span>
<span id="cb182-1470"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1470" aria-hidden="true" tabindex="-1"></a>    total_elements <span class="ot">=</span> <span class="dv">30000</span></span>
<span id="cb182-1471"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1471" aria-hidden="true" tabindex="-1"></a>    Min_overlap <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb182-1472"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1472" aria-hidden="true" tabindex="-1"></a>    maxTerms <span class="ot">=</span><span class="dv">10</span> <span class="co"># max number of enriched terms</span></span>
<span id="cb182-1473"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1473" aria-hidden="true" tabindex="-1"></a>    noSig <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No significant enrichment found!&quot;</span>)</span>
<span id="cb182-1474"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1474" aria-hidden="true" tabindex="-1"></a>    query <span class="ot">&lt;-</span> <span class="fu">cleanGeneSet</span>(query)   <span class="co"># convert to upper case, unique()</span></span>
<span id="cb182-1475"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1476"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1476" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(query) <span class="sc">&lt;=</span><span class="dv">2</span>) <span class="fu">return</span>(noSig)</span>
<span id="cb182-1477"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1477" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(geneSet) <span class="sc">&lt;</span><span class="dv">1</span>) <span class="fu">return</span>(noSig)</span>
<span id="cb182-1478"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1478" aria-hidden="true" tabindex="-1"></a>      geneSet <span class="ot">&lt;-</span> geneSet[<span class="fu">which</span>(<span class="fu">sapply</span>(geneSet,length) <span class="sc">&gt;</span> minSize)]  <span class="co"># gene sets smaller than 1 is ignored!!!</span></span>
<span id="cb182-1479"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1479" aria-hidden="true" tabindex="-1"></a>      geneSet <span class="ot">&lt;-</span> geneSet[<span class="fu">which</span>(<span class="fu">sapply</span>(geneSet,length) <span class="sc">&lt;</span> maxSize)]  <span class="co"># gene sets smaller than 1 is ignored!!!</span></span>
<span id="cb182-1480"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1480" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">unlist</span>( <span class="fu">lapply</span>(geneSet, <span class="cf">function</span>(x) <span class="fu">length</span>( <span class="fu">intersect</span> (query, x) ) ) )</span>
<span id="cb182-1481"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1481" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">unlist</span>( <span class="fu">lapply</span>(geneSet, length) ), result )</span>
<span id="cb182-1482"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1482" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> result[ <span class="fu">which</span>(result[,<span class="dv">2</span>]<span class="sc">&gt;</span>Min_overlap), ,drop<span class="ot">=</span>F]</span>
<span id="cb182-1483"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1483" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">dim</span>(result)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>( noSig)</span>
<span id="cb182-1484"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1484" aria-hidden="true" tabindex="-1"></a>    xx <span class="ot">&lt;-</span> result[,<span class="dv">2</span>]</span>
<span id="cb182-1485"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1485" aria-hidden="true" tabindex="-1"></a>    mm <span class="ot">&lt;-</span> <span class="fu">length</span>(query)</span>
<span id="cb182-1486"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1486" aria-hidden="true" tabindex="-1"></a>    nn <span class="ot">&lt;-</span> total_elements <span class="sc">-</span> mm</span>
<span id="cb182-1487"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1487" aria-hidden="true" tabindex="-1"></a>    kk <span class="ot">&lt;-</span> result[,<span class="dv">1</span>]</span>
<span id="cb182-1488"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1488" aria-hidden="true" tabindex="-1"></a>    Pval_enrich<span class="ot">=</span><span class="fu">phyper</span>(xx<span class="dv">-1</span>,mm,nn,kk, <span class="at">lower.tail=</span><span class="cn">FALSE</span> );</span>
<span id="cb182-1489"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1489" aria-hidden="true" tabindex="-1"></a>    FDR <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(Pval_enrich,<span class="at">method=</span><span class="st">&quot;fdr&quot;</span>,<span class="at">n=</span><span class="fu">length</span>(geneSet) )</span>
<span id="cb182-1490"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1490" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(FDR,result))</span>
<span id="cb182-1491"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1491" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> result[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>)]</span>
<span id="cb182-1492"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1492" aria-hidden="true" tabindex="-1"></a>    result<span class="sc">$</span>pathway <span class="ot">=</span> <span class="fu">rownames</span>(result)</span>
<span id="cb182-1493"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1493" aria-hidden="true" tabindex="-1"></a>    result<span class="sc">$</span>Genes <span class="ot">=</span> <span class="st">&quot;&quot;</span>  <span class="co"># place holder just </span></span>
<span id="cb182-1494"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1494" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(result)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Corrected P value (FDR)&quot;</span>, <span class="st">&quot;Genes in list&quot;</span>, <span class="st">&quot;Total genes in category&quot;</span>,<span class="st">&quot;Functional Category&quot;</span>,<span class="st">&quot;Genes&quot;</span>  )</span>
<span id="cb182-1495"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1495" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> result[ <span class="fu">which</span>( result[,<span class="dv">1</span>] <span class="sc">&lt;</span> minFDR),,drop<span class="ot">=</span>F]</span>
<span id="cb182-1496"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1496" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>( result)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>(noSig) </span>
<span id="cb182-1497"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1497" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">min</span>(FDR) <span class="sc">&gt;</span> minFDR) <span class="fu">return</span>(noSig) </span>
<span id="cb182-1498"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1498" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> result[<span class="fu">order</span>(result[,<span class="dv">1</span>] ),]</span>
<span id="cb182-1499"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1499" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">dim</span>(result)[<span class="dv">1</span>] <span class="sc">&gt;</span> maxTerms ) result <span class="ot">&lt;-</span> result[<span class="dv">1</span><span class="sc">:</span>maxTerms,]</span>
<span id="cb182-1500"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1500" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1501"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1501" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( result)</span>
<span id="cb182-1502"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1502" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-1503"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1504"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1504" aria-hidden="true" tabindex="-1"></a>KmeansGO <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb182-1505"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1505" aria-hidden="true" tabindex="-1"></a>        pp<span class="ot">=</span><span class="dv">0</span></span>
<span id="cb182-1506"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1506" aria-hidden="true" tabindex="-1"></a>        minFDR <span class="ot">=</span> <span class="fl">0.01</span></span>
<span id="cb182-1507"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1508"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1508" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>input_nClusters) {</span>
<span id="cb182-1509"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1509" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1510"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1510" aria-hidden="true" tabindex="-1"></a>            query <span class="ot">=</span> <span class="fu">rownames</span>(Kmeans.out<span class="sc">$</span>x)[<span class="fu">which</span>(Kmeans.out<span class="sc">$</span>bar <span class="sc">==</span> i)]</span>
<span id="cb182-1511"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1512"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1512" aria-hidden="true" tabindex="-1"></a>                result <span class="ot">&lt;-</span> <span class="fu">findOverlapGMT</span>( query, GeneSets.out,<span class="dv">1</span>) </span>
<span id="cb182-1513"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1513" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1514"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1514" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">2</span>] <span class="sc">==</span><span class="dv">1</span>) <span class="cf">next</span>;   <span class="co"># result could be NULL</span></span>
<span id="cb182-1515"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1515" aria-hidden="true" tabindex="-1"></a>            result<span class="sc">$</span>Genes <span class="ot">=</span> <span class="fu">toupper</span>(letters)[i] </span>
<span id="cb182-1516"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1516" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (pp<span class="sc">==</span><span class="dv">0</span> ) { results <span class="ot">&lt;-</span> result; pp <span class="ot">&lt;-</span> <span class="dv">1</span>;</span>
<span id="cb182-1517"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1517" aria-hidden="true" tabindex="-1"></a>            } <span class="cf">else</span> {</span>
<span id="cb182-1518"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1518" aria-hidden="true" tabindex="-1"></a>                results <span class="ot">&lt;-</span> <span class="fu">rbind</span>(results,result)</span>
<span id="cb182-1519"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1519" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-1520"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1520" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-1521"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1521" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1522"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1522" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(pp <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>( <span class="fu">as.data.frame</span>(<span class="st">&quot;No enrichment found.&quot;</span>))</span>
<span id="cb182-1523"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1523" aria-hidden="true" tabindex="-1"></a>        results<span class="ot">=</span> results[,<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">4</span>)]</span>
<span id="cb182-1524"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1524" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(results)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Cluster&quot;</span>,<span class="st">&quot;FDR&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;Pathways&quot;</span>)</span>
<span id="cb182-1525"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1525" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">min</span>(results<span class="sc">$</span>FDR) <span class="sc">&gt;</span> minFDR ) results <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No signficant enrichment found.&quot;</span>) <span class="cf">else</span></span>
<span id="cb182-1526"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1526" aria-hidden="true" tabindex="-1"></a>        results <span class="ot">=</span> results[<span class="fu">which</span>(results<span class="sc">$</span>FDR <span class="sc">&lt;</span> minFDR),]</span>
<span id="cb182-1527"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1527" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1528"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1528" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">is.null</span>(results) )  <span class="fu">return</span> ( <span class="fu">as.matrix</span>(<span class="st">&quot;No significant enrichment.&quot;</span>) )  </span>
<span id="cb182-1529"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1529" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">class</span>(results) <span class="sc">!=</span> <span class="st">&quot;data.frame&quot;</span>)  <span class="fu">return</span> ( <span class="fu">as.matrix</span>(<span class="st">&quot;No significant enrichment.&quot;</span>) )</span>
<span id="cb182-1530"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1530" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>(results)[<span class="dv">2</span>] <span class="sc">==</span><span class="dv">1</span>)  <span class="fu">return</span> ( <span class="fu">as.matrix</span>(<span class="st">&quot;No significant enrichment.&quot;</span>) )</span>
<span id="cb182-1531"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1531" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(results)[<span class="dv">2</span>] <span class="ot">=</span> <span class="st">&quot;adj.Pval&quot;</span></span>
<span id="cb182-1532"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1532" aria-hidden="true" tabindex="-1"></a>    results<span class="sc">$</span>Genes <span class="ot">&lt;-</span> <span class="fu">as.character</span>(results<span class="sc">$</span>Genes)</span>
<span id="cb182-1533"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1533" aria-hidden="true" tabindex="-1"></a>    results<span class="sc">$</span>Cluster[<span class="fu">which</span>( <span class="fu">duplicated</span>(results<span class="sc">$</span>Cluster) ) ] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb182-1534"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1534" aria-hidden="true" tabindex="-1"></a>    results</span>
<span id="cb182-1535"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1535" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb182-1536"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1537"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1538"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1538" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-1539"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1539" aria-hidden="true" tabindex="-1"></a><span class="co"># Differential expression</span></span>
<span id="cb182-1540"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1540" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-1541"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1541" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb182-1542"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1542" aria-hidden="true" tabindex="-1"></a><span class="co"># Differential expression using LIMMA </span></span>
<span id="cb182-1543"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1543" aria-hidden="true" tabindex="-1"></a>DEG.limma <span class="ot">&lt;-</span> <span class="cf">function</span> (x, <span class="at">maxP_limma=</span>.<span class="dv">1</span>, <span class="at">minFC_limma=</span><span class="dv">2</span>, rawCounts,countsDEGMethods,priorCounts, dataFormat, <span class="at">selectedComparisons=</span><span class="cn">NULL</span>, <span class="at">sampleInfo =</span> <span class="cn">NULL</span>,<span class="at">modelFactors=</span><span class="cn">NULL</span>, <span class="at">blockFactor =</span> <span class="cn">NULL</span>){</span>
<span id="cb182-1544"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1544" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(limma,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># Differential expression</span></span>
<span id="cb182-1545"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1545" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(statmod,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb182-1546"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1546" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1547"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1547" aria-hidden="true" tabindex="-1"></a>    <span class="co"># many different situations: 1. just use sample names 2. just one factor  3. two factors no interaction</span></span>
<span id="cb182-1548"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1548" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 4. two factors with interaction   5. block factor </span></span>
<span id="cb182-1549"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1549" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1550"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1550" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1551"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1551" aria-hidden="true" tabindex="-1"></a>    topGenes <span class="ot">=</span> <span class="fu">list</span>();  limmaTrend <span class="ot">=</span> <span class="cn">FALSE</span></span>
<span id="cb182-1552"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1552" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( dataFormat <span class="sc">==</span> <span class="dv">2</span>) {   <span class="co"># if normalized data</span></span>
<span id="cb182-1553"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1553" aria-hidden="true" tabindex="-1"></a>        eset <span class="ot">=</span> <span class="fu">new</span>(<span class="st">&quot;ExpressionSet&quot;</span>, <span class="at">exprs=</span><span class="fu">as.matrix</span>(x)) } <span class="cf">else</span> { <span class="co"># counts data</span></span>
<span id="cb182-1554"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1554" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (countsDEGMethods <span class="sc">==</span> <span class="dv">1</span> ) { <span class="co"># limma-trend method selected for counts data</span></span>
<span id="cb182-1555"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1555" aria-hidden="true" tabindex="-1"></a>                <span class="co">#dge &lt;- DGEList(counts=rawCounts);</span></span>
<span id="cb182-1556"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1556" aria-hidden="true" tabindex="-1"></a>                <span class="co">#dge &lt;- calcNormFactors(dge, method = &quot;TMM&quot;)</span></span>
<span id="cb182-1557"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1557" aria-hidden="true" tabindex="-1"></a>                <span class="co">#eset &lt;- cpm(dge, log=TRUE, prior.count=priorCounts)</span></span>
<span id="cb182-1558"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1558" aria-hidden="true" tabindex="-1"></a>                eset <span class="ot">=</span> <span class="fu">new</span>(<span class="st">&quot;ExpressionSet&quot;</span>, <span class="at">exprs=</span><span class="fu">as.matrix</span>(x))  <span class="co"># use transformed data for limma </span></span>
<span id="cb182-1559"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1559" aria-hidden="true" tabindex="-1"></a>                limmaTrend <span class="ot">=</span> <span class="cn">TRUE</span></span>
<span id="cb182-1560"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1560" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-1561"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1561" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-1562"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1562" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1563"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1563" aria-hidden="true" tabindex="-1"></a>    groups <span class="ot">=</span> <span class="fu">colnames</span>(x)</span>
<span id="cb182-1564"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1564" aria-hidden="true" tabindex="-1"></a>    groups <span class="ot">=</span> <span class="fu">detectGroups</span>( groups)</span>
<span id="cb182-1565"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1565" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">=</span>  <span class="fu">unique</span>(groups)  </span>
<span id="cb182-1566"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1566" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1567"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1567" aria-hidden="true" tabindex="-1"></a>    <span class="co"># check for replicates, removes samples without replicates</span></span>
<span id="cb182-1568"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1568" aria-hidden="true" tabindex="-1"></a>    reps <span class="ot">=</span> <span class="fu">as.matrix</span>(<span class="fu">table</span>(groups)) <span class="co"># number of replicates per biological sample</span></span>
<span id="cb182-1569"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1569" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> ( <span class="fu">sum</span>( reps[,<span class="dv">1</span>] <span class="sc">&gt;=</span> <span class="dv">2</span>) <span class="sc">&lt;</span><span class="dv">2</span>    ) <span class="co"># if less than 2 samples with replicates</span></span>
<span id="cb182-1570"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1570" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">results=</span> <span class="cn">NULL</span>, <span class="at">comparisons =</span> <span class="cn">NULL</span>, <span class="at">Exp.type=</span><span class="st">&quot;Failed to parse sample names to define groups. </span></span>
<span id="cb182-1571"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1571" aria-hidden="true" tabindex="-1"></a><span class="st">        Cannot perform DEGs and pathway analysis. Please double check column names! Use WT_Rep1, WT_Rep2 etc. &quot;</span>, <span class="at">topGenes=</span><span class="cn">NULL</span>)) </span>
<span id="cb182-1572"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1572" aria-hidden="true" tabindex="-1"></a>    <span class="co"># remove samples without replicates</span></span>
<span id="cb182-1573"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1573" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">rownames</span>(reps)[<span class="fu">which</span>(reps[,<span class="dv">1</span>] <span class="sc">&gt;</span><span class="dv">1</span>)]</span>
<span id="cb182-1574"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1574" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">&lt;-</span> <span class="fu">which</span>( groups <span class="sc">%in%</span> g)  </span>
<span id="cb182-1575"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1575" aria-hidden="true" tabindex="-1"></a>    groups <span class="ot">&lt;-</span> groups[ix]   </span>
<span id="cb182-1576"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1576" aria-hidden="true" tabindex="-1"></a>    x<span class="ot">&lt;-</span> x[,ix]; rawCounts <span class="ot">&lt;-</span> rawCounts[,ix] </span>
<span id="cb182-1577"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1577" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1578"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1578" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(g) <span class="sc">==</span><span class="dv">2</span> ) { </span>
<span id="cb182-1579"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1579" aria-hidden="true" tabindex="-1"></a>        g<span class="ot">=</span> <span class="fu">unique</span>(groups)</span>
<span id="cb182-1580"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1580" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">&lt;-</span>  <span class="fu">paste</span>(g[<span class="dv">2</span>],<span class="st">&quot;-&quot;</span>,g[<span class="dv">1</span>],<span class="at">sep=</span><span class="st">&quot;&quot;</span>)  <span class="co"># &quot;Mutant-WT&quot;</span></span>
<span id="cb182-1581"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1581" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1582"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1582" aria-hidden="true" tabindex="-1"></a>        <span class="co"># no sample file, but user selected comparisons using column names</span></span>
<span id="cb182-1583"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1583" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">is.null</span>(modelFactors) <span class="sc">&amp;</span> <span class="fu">length</span>( selectedComparisons) <span class="sc">&gt;</span><span class="dv">0</span>  )  </span>
<span id="cb182-1584"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1584" aria-hidden="true" tabindex="-1"></a>            comparisons <span class="ot">=</span> selectedComparisons</span>
<span id="cb182-1585"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1585" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1586"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1586" aria-hidden="true" tabindex="-1"></a>        design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span><span class="dv">0</span><span class="sc">+</span>groups)</span>
<span id="cb182-1587"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1587" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> g</span>
<span id="cb182-1588"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1588" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1589"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1589" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>(rawCounts) <span class="sc">&amp;&amp;</span> countsDEGMethods <span class="sc">==</span> <span class="dv">2</span>) {  <span class="co"># voom</span></span>
<span id="cb182-1590"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1590" aria-hidden="true" tabindex="-1"></a>            dge <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts=</span>rawCounts);</span>
<span id="cb182-1591"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1591" aria-hidden="true" tabindex="-1"></a>            dge <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(dge, <span class="at">method =</span> <span class="st">&quot;TMM&quot;</span>)  <span class="co"># normalization</span></span>
<span id="cb182-1592"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1592" aria-hidden="true" tabindex="-1"></a>            v <span class="ot">&lt;-</span> <span class="fu">voom</span>(dge, design); fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(v, design) } <span class="cf">else</span> </span>
<span id="cb182-1593"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1593" aria-hidden="true" tabindex="-1"></a>            fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(eset, design)      <span class="co"># regular limma</span></span>
<span id="cb182-1594"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1594" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-1595"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1595" aria-hidden="true" tabindex="-1"></a>        cont.matrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons, <span class="at">levels=</span>design)</span>
<span id="cb182-1596"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1596" aria-hidden="true" tabindex="-1"></a>        fit2 <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit, cont.matrix)</span>
<span id="cb182-1597"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1597" aria-hidden="true" tabindex="-1"></a>        fit2 <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit2, <span class="at">trend=</span>limmaTrend)</span>
<span id="cb182-1598"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1598" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1599"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1599" aria-hidden="true" tabindex="-1"></a>        <span class="co"># calls differential gene expression 1 for up, -1 for down</span></span>
<span id="cb182-1600"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1600" aria-hidden="true" tabindex="-1"></a>        results <span class="ot">&lt;-</span> <span class="fu">decideTests</span>(fit2, <span class="at">p.value=</span>maxP_limma, <span class="at">lfc=</span><span class="fu">log2</span>(minFC_limma) )</span>
<span id="cb182-1601"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1601" aria-hidden="true" tabindex="-1"></a>        <span class="co">#vennDiagram(results,circle.col=rainbow(5))</span></span>
<span id="cb182-1602"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1602" aria-hidden="true" tabindex="-1"></a>        topGenes1 <span class="ot">=</span><span class="fu">topTable</span>(fit2, <span class="at">number =</span> <span class="fl">1e12</span>,<span class="at">sort.by=</span><span class="st">&quot;M&quot;</span> )</span>
<span id="cb182-1603"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1603" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">dim</span>(topGenes1)[<span class="dv">1</span>] <span class="sc">!=</span> <span class="dv">0</span>) {</span>
<span id="cb182-1604"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1604" aria-hidden="true" tabindex="-1"></a>        topGenes1 <span class="ot">=</span> topGenes1[,<span class="fu">c</span>(<span class="st">&#39;logFC&#39;</span>,<span class="st">&#39;adj.P.Val&#39;</span>)] </span>
<span id="cb182-1605"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1605" aria-hidden="true" tabindex="-1"></a>        <span class="co"># topGenes1[,1] &lt;-  -1* topGenes1[,1] # reverse direction</span></span>
<span id="cb182-1606"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1606" aria-hidden="true" tabindex="-1"></a>        topGenes[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> topGenes1 }</span>
<span id="cb182-1607"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1607" aria-hidden="true" tabindex="-1"></a>        <span class="co"># log fold change is actually substract of means. So if the data is natral log transformed, it shoudl be natral log.</span></span>
<span id="cb182-1608"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1608" aria-hidden="true" tabindex="-1"></a>        Exp.type <span class="ot">=</span> <span class="st">&quot;2 sample groups.&quot;</span></span>
<span id="cb182-1609"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1609" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-1610"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1610" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1611"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1611" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(g) <span class="sc">&gt;</span> <span class="dv">2</span> ) { <span class="co"># more than two sample groups</span></span>
<span id="cb182-1612"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1612" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1613"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1613" aria-hidden="true" tabindex="-1"></a>        design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span><span class="sc">+</span><span class="fu">factor</span>(groups))</span>
<span id="cb182-1614"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1614" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;.*)&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(design))</span>
<span id="cb182-1615"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1615" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1616"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1616" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>(rawCounts) <span class="sc">&amp;&amp;</span> countsDEGMethods <span class="sc">==</span> <span class="dv">2</span>) {  <span class="co"># voom</span></span>
<span id="cb182-1617"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1617" aria-hidden="true" tabindex="-1"></a>            v <span class="ot">&lt;-</span> <span class="fu">voom</span>(rawCounts, design); </span>
<span id="cb182-1618"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1618" aria-hidden="true" tabindex="-1"></a>            fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(v, design) </span>
<span id="cb182-1619"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1619" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> </span>
<span id="cb182-1620"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1620" aria-hidden="true" tabindex="-1"></a>            fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(eset, design)</span>
<span id="cb182-1621"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1621" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1622"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1622" aria-hidden="true" tabindex="-1"></a>        fit <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit, <span class="at">trend=</span>limmaTrend)</span>
<span id="cb182-1623"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1623" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1624"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1624" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">=</span> <span class="st">&quot;&quot;</span></span>
<span id="cb182-1625"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1625" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(g)<span class="sc">-</span><span class="dv">1</span>) )</span>
<span id="cb182-1626"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1626" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> (j <span class="cf">in</span> (i<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(g)) </span>
<span id="cb182-1627"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1627" aria-hidden="true" tabindex="-1"></a>            comparisons <span class="ot">=</span> <span class="fu">c</span>(comparisons,<span class="fu">paste</span>(g[j],<span class="st">&quot;-&quot;</span>,g[i],<span class="at">sep=</span><span class="st">&quot;&quot;</span> ) )</span>
<span id="cb182-1628"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1628" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">&lt;-</span> comparisons[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb182-1629"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1629" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1630"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1630" aria-hidden="true" tabindex="-1"></a>        <span class="co"># no sample file, but user selected comparisons using column names</span></span>
<span id="cb182-1631"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1631" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">is.null</span>(modelFactors) <span class="sc">&amp;</span> <span class="fu">length</span>( selectedComparisons) <span class="sc">&gt;</span><span class="dv">0</span>  )  </span>
<span id="cb182-1632"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1632" aria-hidden="true" tabindex="-1"></a>            comparisons <span class="ot">=</span> selectedComparisons</span>
<span id="cb182-1633"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1633" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1634"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1634" aria-hidden="true" tabindex="-1"></a>        contrast1 <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons[<span class="dv">1</span>], <span class="at">levels=</span>design)</span>
<span id="cb182-1635"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1635" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">length</span>(comparisons)<span class="sc">&gt;</span><span class="dv">1</span> )</span>
<span id="cb182-1636"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1636" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>( kk <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(comparisons) )</span>
<span id="cb182-1637"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1637" aria-hidden="true" tabindex="-1"></a>                contrast1<span class="ot">&lt;-</span>  <span class="fu">cbind</span>(contrast1,<span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons[kk], <span class="at">levels=</span>design)   )</span>
<span id="cb182-1638"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1638" aria-hidden="true" tabindex="-1"></a>        Exp.type <span class="ot">=</span> <span class="fu">paste</span>(<span class="fu">length</span>(g),<span class="st">&quot; sample groups detected.&quot;</span>)   </span>
<span id="cb182-1639"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1639" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1640"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1640" aria-hidden="true" tabindex="-1"></a>        <span class="co"># if factorial design 2x2, 2x3, 3x5 etc.</span></span>
<span id="cb182-1641"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1641" aria-hidden="true" tabindex="-1"></a>            <span class="co"># all samples must be something like WT_control_rep1</span></span>
<span id="cb182-1642"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1642" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> ( <span class="fu">sum</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(g,<span class="st">&quot;_&quot;</span>),length) <span class="sc">==</span> <span class="dv">2</span> ) <span class="sc">==</span> <span class="fu">length</span>(g) ) {</span>
<span id="cb182-1643"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1643" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1644"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1644" aria-hidden="true" tabindex="-1"></a>            <span class="co">#comparisons</span></span>
<span id="cb182-1645"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1645" aria-hidden="true" tabindex="-1"></a>            comparisons <span class="ot">=</span> <span class="st">&quot;&quot;</span></span>
<span id="cb182-1646"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1646" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(g)<span class="sc">-</span><span class="dv">1</span>) )</span>
<span id="cb182-1647"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1647" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span> (j <span class="cf">in</span> (i<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(g)) </span>
<span id="cb182-1648"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1648" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>( <span class="fu">strsplit</span>(g[i],<span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>] <span class="sc">==</span> <span class="fu">strsplit</span>(g[j],<span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]<span class="sc">|</span> <span class="fu">strsplit</span>(g[i],<span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>] <span class="sc">==</span> <span class="fu">strsplit</span>(g[j],<span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>]) <span class="co"># only compare WT_control vs. WT_treatment</span></span>
<span id="cb182-1649"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1649" aria-hidden="true" tabindex="-1"></a>                    comparisons <span class="ot">=</span> <span class="fu">c</span>(comparisons,<span class="fu">paste</span>(g[j],<span class="st">&quot;-&quot;</span>,g[i],<span class="at">sep=</span><span class="st">&quot;&quot;</span> ) )</span>
<span id="cb182-1650"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1650" aria-hidden="true" tabindex="-1"></a>            comparisons <span class="ot">&lt;-</span> comparisons[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb182-1651"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1651" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1652"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1652" aria-hidden="true" tabindex="-1"></a>            <span class="co">#factors genotype treatment levels</span></span>
<span id="cb182-1653"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1653" aria-hidden="true" tabindex="-1"></a>            extract_treatment <span class="ot">&lt;-</span> <span class="cf">function</span> (x) <span class="fu">paste</span>( <span class="fu">gsub</span>( <span class="st">&quot;.*_&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">unlist</span>( <span class="fu">strsplit</span>(x,<span class="st">&quot;-&quot;</span>)) ), <span class="at">collapse=</span><span class="st">&quot;-&quot;</span>)</span>
<span id="cb182-1654"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1654" aria-hidden="true" tabindex="-1"></a>            extract_genotype <span class="ot">&lt;-</span> <span class="cf">function</span> (x) <span class="fu">gsub</span>( <span class="st">&quot;_.*&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">unlist</span>( <span class="fu">strsplit</span>(x,<span class="st">&quot;-&quot;</span>)) )[<span class="dv">1</span>]</span>
<span id="cb182-1655"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1655" aria-hidden="true" tabindex="-1"></a>            extract_treatment_counting <span class="ot">&lt;-</span> <span class="fu">unique</span>( <span class="fu">gsub</span>( <span class="st">&quot;.*_&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">unlist</span>( <span class="fu">strsplit</span>(g,<span class="st">&quot;-&quot;</span>)) ))</span>
<span id="cb182-1656"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1656" aria-hidden="true" tabindex="-1"></a>            treatments <span class="ot">=</span> <span class="fu">sapply</span>(comparisons, extract_treatment)</span>
<span id="cb182-1657"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1657" aria-hidden="true" tabindex="-1"></a>            genotypes <span class="ot">=</span> <span class="fu">sapply</span>(comparisons, extract_genotype)</span>
<span id="cb182-1658"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1658" aria-hidden="true" tabindex="-1"></a>            Exp.type <span class="ot">=</span> <span class="fu">paste</span>( Exp.type, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">Factorial design:&quot;</span>,<span class="fu">length</span>(<span class="fu">unique</span>(genotypes)),<span class="st">&quot;X&quot;</span>, <span class="fu">length</span>( extract_treatment_counting ), <span class="at">sep=</span><span class="st">&quot;&quot;</span> )</span>
<span id="cb182-1659"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1659" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1660"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1660" aria-hidden="true" tabindex="-1"></a>            <span class="co"># pairwise contrasts</span></span>
<span id="cb182-1661"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1661" aria-hidden="true" tabindex="-1"></a>            contrast1 <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons[<span class="dv">1</span>], <span class="at">levels=</span>design)</span>
<span id="cb182-1662"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1662" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>( kk <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(comparisons) )</span>
<span id="cb182-1663"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1663" aria-hidden="true" tabindex="-1"></a>                contrast1<span class="ot">&lt;-</span>  <span class="fu">cbind</span>(contrast1,<span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons[kk], <span class="at">levels=</span>design)   )</span>
<span id="cb182-1664"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1664" aria-hidden="true" tabindex="-1"></a>            contrast.names <span class="ot">=</span> <span class="fu">colnames</span>(contrast1)</span>
<span id="cb182-1665"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1665" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1666"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1666" aria-hidden="true" tabindex="-1"></a>            <span class="co"># interaction contrasts</span></span>
<span id="cb182-1667"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1667" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> ( kk <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(comparisons)<span class="sc">-</span><span class="dv">1</span>) ) {</span>
<span id="cb182-1668"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1668" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>( kp <span class="cf">in</span> (kk<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(comparisons)) </span>
<span id="cb182-1669"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1669" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>( treatments[kp]<span class="sc">==</span> treatments[kk] ) </span>
<span id="cb182-1670"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1670" aria-hidden="true" tabindex="-1"></a>                {  </span>
<span id="cb182-1671"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1671" aria-hidden="true" tabindex="-1"></a>                    contrast1 <span class="ot">=</span> <span class="fu">cbind</span>(contrast1, contrast1[,kp]<span class="sc">-</span> contrast1[,kk] )</span>
<span id="cb182-1672"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1672" aria-hidden="true" tabindex="-1"></a>                    contrast.names <span class="ot">=</span> <span class="fu">c</span>(contrast.names, <span class="fu">paste</span>(<span class="st">&quot;I:&quot;</span>,  genotypes[kp], <span class="st">&quot;-&quot;</span>, genotypes[kk],<span class="st">&quot;(&quot;</span>,<span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;.vs.&quot;</span>,treatments[kp]),<span class="st">&quot;)&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span> ) )</span>
<span id="cb182-1673"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1673" aria-hidden="true" tabindex="-1"></a>                }   </span>
<span id="cb182-1674"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1674" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-1675"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1675" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(contrast1)<span class="ot">=</span>contrast.names</span>
<span id="cb182-1676"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1676" aria-hidden="true" tabindex="-1"></a>            comparisons <span class="ot">=</span> contrast.names</span>
<span id="cb182-1677"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1677" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1678"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1678" aria-hidden="true" tabindex="-1"></a>        } <span class="co"># if interaction terms    </span></span>
<span id="cb182-1679"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1679" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1680"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1680" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1681"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1681" aria-hidden="true" tabindex="-1"></a>        <span class="co"># if sample information is uploaded and user selected factors and comparisons</span></span>
<span id="cb182-1682"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1682" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>(modelFactors) <span class="sc">&amp;</span> <span class="fu">length</span>( selectedComparisons) <span class="sc">&gt;</span><span class="dv">0</span>  ) {</span>
<span id="cb182-1683"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1683" aria-hidden="true" tabindex="-1"></a>            Exp.type <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;Model: ~&quot;</span>, <span class="fu">paste</span>(modelFactors,<span class="at">collapse=</span><span class="st">&quot; + &quot;</span>) )</span>
<span id="cb182-1684"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1684" aria-hidden="true" tabindex="-1"></a>            interactionTerm <span class="ot">=</span> <span class="cn">FALSE</span> <span class="co"># default value to be re-write if needed</span></span>
<span id="cb182-1685"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1685" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-1686"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1686" aria-hidden="true" tabindex="-1"></a>            <span class="co"># model factors that does not contain interaction terms</span></span>
<span id="cb182-1687"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1687" aria-hidden="true" tabindex="-1"></a>            <span class="co"># modelFactors &quot;genotype&quot;, &quot;condition&quot;, &quot;genotype:condition&quot;</span></span>
<span id="cb182-1688"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1688" aria-hidden="true" tabindex="-1"></a>            keyModelFactors <span class="ot">=</span> modelFactors[ <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;:&quot;</span>,modelFactors) ]</span>
<span id="cb182-1689"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1689" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-1690"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1690" aria-hidden="true" tabindex="-1"></a>            <span class="co"># &quot;genotype: control vs. mutant&quot;</span></span>
<span id="cb182-1691"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1691" aria-hidden="true" tabindex="-1"></a>            factorsVector<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;:.*&quot;</span>,<span class="st">&quot;&quot;</span>,selectedComparisons) <span class="co"># corresponding factors for each comparison</span></span>
<span id="cb182-1692"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1692" aria-hidden="true" tabindex="-1"></a>            <span class="co"># remove factors not used in comparison, these are batch effects/pairs/blocks, </span></span>
<span id="cb182-1693"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1693" aria-hidden="true" tabindex="-1"></a>            <span class="co"># keyModelFactors = keyModelFactors[ !is.na(match(keyModelFactors, factorsVector)) ]    </span></span>
<span id="cb182-1694"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1694" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-1695"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1695" aria-hidden="true" tabindex="-1"></a>            <span class="co"># if a factor is selected both in block and main factors, then use it as block factor</span></span>
<span id="cb182-1696"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1696" aria-hidden="true" tabindex="-1"></a>            keyModelFactors <span class="ot">=</span> keyModelFactors[ <span class="fu">is.na</span>(<span class="fu">match</span>(keyModelFactors, blockFactor)) ] </span>
<span id="cb182-1697"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1697" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1698"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1698" aria-hidden="true" tabindex="-1"></a>            <span class="co">#------Design matrix            </span></span>
<span id="cb182-1699"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1699" aria-hidden="true" tabindex="-1"></a>            sampleInfo2 <span class="ot">=</span> sampleInfo[,keyModelFactors,drop<span class="ot">=</span>F] <span class="co"># remove factors not used.</span></span>
<span id="cb182-1700"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1700" aria-hidden="true" tabindex="-1"></a>            groups <span class="ot">=</span> <span class="fu">apply</span>(sampleInfo2,<span class="dv">1</span>, <span class="cf">function</span>(x) <span class="fu">paste</span>(x,<span class="at">collapse=</span><span class="st">&quot;_&quot;</span>)  )</span>
<span id="cb182-1701"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1701" aria-hidden="true" tabindex="-1"></a>            g <span class="ot">=</span>  <span class="fu">unique</span>(groups)  </span>
<span id="cb182-1702"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1702" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-1703"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1703" aria-hidden="true" tabindex="-1"></a>            design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span><span class="sc">+</span><span class="fu">factor</span>(groups))</span>
<span id="cb182-1704"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1704" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;.*)&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(design))</span>
<span id="cb182-1705"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1705" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1706"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1706" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>(rawCounts) <span class="sc">&amp;&amp;</span> countsDEGMethods <span class="sc">==</span> <span class="dv">2</span>) {  <span class="co"># voom</span></span>
<span id="cb182-1707"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1707" aria-hidden="true" tabindex="-1"></a>                v <span class="ot">&lt;-</span> <span class="fu">voom</span>(rawCounts, design); </span>
<span id="cb182-1708"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1708" aria-hidden="true" tabindex="-1"></a>                fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(v, design) </span>
<span id="cb182-1709"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1709" aria-hidden="true" tabindex="-1"></a>            } <span class="cf">else</span> </span>
<span id="cb182-1710"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1710" aria-hidden="true" tabindex="-1"></a>                fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(eset, design)</span>
<span id="cb182-1711"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1711" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1712"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1712" aria-hidden="true" tabindex="-1"></a>            fit <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit, <span class="at">trend=</span>limmaTrend)    </span>
<span id="cb182-1713"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1713" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-1714"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1714" aria-hidden="true" tabindex="-1"></a>            <span class="co">#-----------Making comaprisons</span></span>
<span id="cb182-1715"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1715" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">length</span>(keyModelFactors) <span class="sc">!=</span> <span class="dv">2</span> <span class="sc">|</span> <span class="fu">length</span>(blockFactor) <span class="sc">&gt;</span><span class="dv">1</span> )  { <span class="co"># if only one factor, or more than two then use all pairwise comparisons</span></span>
<span id="cb182-1716"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1716" aria-hidden="true" tabindex="-1"></a>                comparisons <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;.*: &quot;</span>,<span class="st">&quot;&quot;</span>,selectedComparisons)</span>
<span id="cb182-1717"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1717" aria-hidden="true" tabindex="-1"></a>                comparisons <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; vs</span><span class="sc">\\</span><span class="st">. &quot;</span>,<span class="st">&quot;-&quot;</span>,comparisons)</span>
<span id="cb182-1718"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1718" aria-hidden="true" tabindex="-1"></a>            } <span class="cf">else</span> <span class="cf">if</span>( <span class="fu">length</span>(keyModelFactors) <span class="sc">==</span> <span class="dv">2</span> ){ <span class="co"># two key factors</span></span>
<span id="cb182-1719"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1719" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1720"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1720" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>( <span class="fu">sum</span>(<span class="fu">grepl</span>(<span class="st">&quot;:&quot;</span>,modelFactors) <span class="sc">&gt;</span><span class="dv">0</span>) ) {  <span class="co"># interaction? </span></span>
<span id="cb182-1721"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1721" aria-hidden="true" tabindex="-1"></a>                    interactionTerm <span class="ot">=</span><span class="cn">TRUE</span></span>
<span id="cb182-1722"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1722" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># all pairwise comparisons</span></span>
<span id="cb182-1723"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1723" aria-hidden="true" tabindex="-1"></a>                    comparisons <span class="ot">=</span> <span class="st">&quot;&quot;</span></span>
<span id="cb182-1724"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1724" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(g)<span class="sc">-</span><span class="dv">1</span>) )</span>
<span id="cb182-1725"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1725" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">for</span> (j <span class="cf">in</span> (i<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(g)) </span>
<span id="cb182-1726"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1726" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>( <span class="fu">strsplit</span>(g[i],<span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>] <span class="sc">==</span> <span class="fu">strsplit</span>(g[j],<span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]<span class="sc">|</span> <span class="fu">strsplit</span>(g[i],<span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>] <span class="sc">==</span> <span class="fu">strsplit</span>(g[j],<span class="st">&quot;_&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>]) <span class="co"># only compare WT_control vs. WT_treatment</span></span>
<span id="cb182-1727"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1727" aria-hidden="true" tabindex="-1"></a>                            comparisons <span class="ot">=</span> <span class="fu">c</span>(comparisons,<span class="fu">paste</span>(g[j],<span class="st">&quot;-&quot;</span>,g[i],<span class="at">sep=</span><span class="st">&quot;&quot;</span> ) )</span>
<span id="cb182-1728"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1728" aria-hidden="true" tabindex="-1"></a>                    comparisons <span class="ot">&lt;-</span> comparisons[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb182-1729"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1729" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb182-1730"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1730" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># pairwise contrasts</span></span>
<span id="cb182-1731"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1731" aria-hidden="true" tabindex="-1"></a>                    contrast1 <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons[<span class="dv">1</span>], <span class="at">levels=</span>design)</span>
<span id="cb182-1732"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1732" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>(<span class="fu">length</span>(comparisons)<span class="sc">&gt;</span><span class="dv">1</span> )</span>
<span id="cb182-1733"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1733" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span>( kk <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(comparisons) )</span>
<span id="cb182-1734"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1734" aria-hidden="true" tabindex="-1"></a>                        contrast1<span class="ot">&lt;-</span>  <span class="fu">cbind</span>(contrast1,<span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons[kk], <span class="at">levels=</span>design)   )</span>
<span id="cb182-1735"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1735" aria-hidden="true" tabindex="-1"></a>                    contrast.names <span class="ot">=</span> <span class="fu">colnames</span>(contrast1)        </span>
<span id="cb182-1736"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1736" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-1737"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1737" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># all possible interactions</span></span>
<span id="cb182-1738"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1738" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># interaction contrasts</span></span>
<span id="cb182-1739"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1739" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb182-1740"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1740" aria-hidden="true" tabindex="-1"></a>                    contrast2 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb182-1741"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1741" aria-hidden="true" tabindex="-1"></a>                    contrast.names <span class="ot">=</span><span class="st">&quot;&quot;</span></span>
<span id="cb182-1742"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1742" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span> ( kk <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">dim</span>(contrast1)[<span class="dv">2</span>]<span class="sc">-</span><span class="dv">1</span>) ) {</span>
<span id="cb182-1743"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1743" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span>( kp <span class="cf">in</span> (kk<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">dim</span>(contrast1)[<span class="dv">2</span>]) </span>
<span id="cb182-1744"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1744" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#if( treatments[kp]== treatments[kk] ) </span></span>
<span id="cb182-1745"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1745" aria-hidden="true" tabindex="-1"></a>                        {  <span class="cf">if</span>(<span class="fu">is.null</span>(contrast2))  </span>
<span id="cb182-1746"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1746" aria-hidden="true" tabindex="-1"></a>                            contrast2 <span class="ot">=</span> contrast1[,kp]<span class="sc">-</span> contrast1[,kk]  <span class="cf">else</span>                    </span>
<span id="cb182-1747"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1747" aria-hidden="true" tabindex="-1"></a>                            contrast2 <span class="ot">=</span> <span class="fu">cbind</span>(contrast2, contrast1[,kp]<span class="sc">-</span> contrast1[,kk] )</span>
<span id="cb182-1748"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1748" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb182-1749"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1749" aria-hidden="true" tabindex="-1"></a>                            contrast.names <span class="ot">=</span> <span class="fu">c</span>(contrast.names, <span class="fu">paste0</span>(<span class="st">&quot;I:&quot;</span>,  <span class="fu">colnames</span>(contrast1)[kp], <span class="st">&quot;.vs.&quot;</span>, <span class="fu">colnames</span>(contrast1)[kk] ) )</span>
<span id="cb182-1750"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1750" aria-hidden="true" tabindex="-1"></a>                        }   </span>
<span id="cb182-1751"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1751" aria-hidden="true" tabindex="-1"></a>                    }</span>
<span id="cb182-1752"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1752" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">colnames</span>(contrast2)<span class="ot">=</span>contrast.names[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb182-1753"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1753" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb182-1754"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1754" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># remove nonsense contrasts from interactions</span></span>
<span id="cb182-1755"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1755" aria-hidden="true" tabindex="-1"></a>                     contrast2 <span class="ot">=</span> contrast2[,<span class="fu">which</span>(<span class="fu">apply</span>(<span class="fu">abs</span>(contrast2),<span class="dv">2</span>,max)<span class="sc">==</span><span class="dv">1</span>),drop<span class="ot">=</span>F]</span>
<span id="cb182-1756"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1756" aria-hidden="true" tabindex="-1"></a>                     contrast2 <span class="ot">=</span> contrast2[,<span class="fu">which</span>(<span class="fu">apply</span>(<span class="fu">abs</span>(contrast2),<span class="dv">2</span>,sum)<span class="sc">==</span><span class="dv">4</span>),drop<span class="ot">=</span>F]   </span>
<span id="cb182-1757"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1757" aria-hidden="true" tabindex="-1"></a>                     contrast2 <span class="ot">=</span> <span class="fu">t</span>( <span class="fu">unique</span>(<span class="fu">t</span>(contrast2)) ) <span class="co"># remove duplicate columns       </span></span>
<span id="cb182-1758"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1758" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb182-1759"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1759" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># remove unwanted contrasts involving more than three levels in either factor</span></span>
<span id="cb182-1760"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1760" aria-hidden="true" tabindex="-1"></a>                    keep<span class="ot">=</span> <span class="fu">c</span>()</span>
<span id="cb182-1761"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1761" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(contrast2)[<span class="dv">2</span>]) {</span>
<span id="cb182-1762"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1762" aria-hidden="true" tabindex="-1"></a>                        tem <span class="ot">=</span> <span class="fu">rownames</span>(contrast2)[ contrast2[ ,i ] <span class="sc">!=</span> <span class="dv">0</span>   ]</span>
<span id="cb182-1763"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1763" aria-hidden="true" tabindex="-1"></a>                        tem1 <span class="ot">=</span> <span class="fu">unique</span> (  <span class="fu">unlist</span>(<span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>,<span class="st">&quot;&quot;</span>, tem) ) )</span>
<span id="cb182-1764"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1764" aria-hidden="true" tabindex="-1"></a>                        tem2 <span class="ot">=</span> <span class="fu">unique</span> (  <span class="fu">unlist</span>(<span class="fu">gsub</span>(<span class="st">&quot;.*_&quot;</span>,<span class="st">&quot;&quot;</span>, tem) ) )</span>
<span id="cb182-1765"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1765" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>( <span class="fu">length</span>(tem1) <span class="sc">==</span> <span class="dv">2</span> <span class="sc">&amp;</span> <span class="fu">length</span>(tem2) <span class="sc">==</span><span class="dv">2</span> )</span>
<span id="cb182-1766"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1766" aria-hidden="true" tabindex="-1"></a>                        keep <span class="ot">=</span> <span class="fu">c</span>(keep, <span class="fu">colnames</span>(contrast2) [i] )</span>
<span id="cb182-1767"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1767" aria-hidden="true" tabindex="-1"></a>                    }</span>
<span id="cb182-1768"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1768" aria-hidden="true" tabindex="-1"></a>                    contrast2 <span class="ot">=</span> contrast2[,keep,drop<span class="ot">=</span>F]</span>
<span id="cb182-1769"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1769" aria-hidden="true" tabindex="-1"></a>                    comparisons2 <span class="ot">=</span> <span class="fu">colnames</span>(contrast2)               </span>
<span id="cb182-1770"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1770" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb182-1771"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1771" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1772"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1772" aria-hidden="true" tabindex="-1"></a>                <span class="co"># &quot;stage: MN vs. EN&quot;  --&gt;  c(&quot;MN_AB-EN_AB&quot;, &quot;EN_Nodule-EN_AB&quot;) </span></span>
<span id="cb182-1773"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1773" aria-hidden="true" tabindex="-1"></a>                <span class="co">#  comparisons in all levels of the other factor </span></span>
<span id="cb182-1774"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1774" aria-hidden="true" tabindex="-1"></a>                transformComparisons <span class="ot">&lt;-</span> <span class="cf">function</span> (comparison1){</span>
<span id="cb182-1775"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1775" aria-hidden="true" tabindex="-1"></a>                    tem <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;.*: &quot;</span>,<span class="st">&quot;&quot;</span>,comparison1)</span>
<span id="cb182-1776"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1776" aria-hidden="true" tabindex="-1"></a>                    tem <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(tem, <span class="st">&quot; vs</span><span class="sc">\\</span><span class="st">. &quot;</span>) ) <span class="co"># control  mutant                           </span></span>
<span id="cb182-1777"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1777" aria-hidden="true" tabindex="-1"></a>                    factor1<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;:.*&quot;</span>,<span class="st">&quot;&quot;</span>,comparison1)</span>
<span id="cb182-1778"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1778" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1779"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1779" aria-hidden="true" tabindex="-1"></a>                    ix <span class="ot">=</span> <span class="fu">match</span>(factor1, keyModelFactors) <span class="co"># 1: first factor, 2: 2nd factor</span></span>
<span id="cb182-1780"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1780" aria-hidden="true" tabindex="-1"></a>                    otherFactor <span class="ot">=</span> keyModelFactors[<span class="dv">3</span><span class="sc">-</span>ix]   <span class="co"># 3-1 = 2; 3-1=1</span></span>
<span id="cb182-1781"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1781" aria-hidden="true" tabindex="-1"></a>                    otherFactorLevels <span class="ot">=</span> <span class="fu">unique</span>( sampleInfo2[,otherFactor] )             </span>
<span id="cb182-1782"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1782" aria-hidden="true" tabindex="-1"></a>                    comparisons <span class="ot">=</span> <span class="fu">c</span>( )</span>
<span id="cb182-1783"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1783" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb182-1784"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1784" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span> (factorLevels <span class="cf">in</span> otherFactorLevels) {</span>
<span id="cb182-1785"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1785" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>( ix <span class="sc">==</span> <span class="dv">1</span>){</span>
<span id="cb182-1786"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1786" aria-hidden="true" tabindex="-1"></a>                            comparisons <span class="ot">=</span> <span class="fu">c</span>( comparisons, <span class="fu">paste</span>(<span class="fu">paste0</span>( tem, <span class="st">&quot;_&quot;</span>,factorLevels),<span class="at">collapse=</span><span class="st">&quot;-&quot;</span>) )</span>
<span id="cb182-1787"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1787" aria-hidden="true" tabindex="-1"></a>                        } <span class="cf">else</span> {</span>
<span id="cb182-1788"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1788" aria-hidden="true" tabindex="-1"></a>                            comparisons <span class="ot">=</span> <span class="fu">c</span>( comparisons,  <span class="fu">paste</span>(<span class="fu">paste0</span>(factorLevels, <span class="st">&quot;_&quot;</span>, tem),<span class="at">collapse=</span><span class="st">&quot;-&quot;</span>) )</span>
<span id="cb182-1789"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1789" aria-hidden="true" tabindex="-1"></a>                        }</span>
<span id="cb182-1790"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1790" aria-hidden="true" tabindex="-1"></a>                    }</span>
<span id="cb182-1791"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1791" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">return</span>(comparisons)     </span>
<span id="cb182-1792"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1792" aria-hidden="true" tabindex="-1"></a>                }   </span>
<span id="cb182-1793"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1793" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-1794"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1794" aria-hidden="true" tabindex="-1"></a>                comparisons <span class="ot">=</span> <span class="fu">unlist</span>( <span class="fu">sapply</span>(selectedComparisons, transformComparisons ))</span>
<span id="cb182-1795"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1795" aria-hidden="true" tabindex="-1"></a>                comparisons <span class="ot">=</span> <span class="fu">as.vector</span>(comparisons)</span>
<span id="cb182-1796"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1796" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-1797"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1797" aria-hidden="true" tabindex="-1"></a>            } <span class="co"># two factors</span></span>
<span id="cb182-1798"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1798" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-1799"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1799" aria-hidden="true" tabindex="-1"></a>            <span class="co"># make contrasts</span></span>
<span id="cb182-1800"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1800" aria-hidden="true" tabindex="-1"></a>            contrast1 <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons[<span class="dv">1</span>], <span class="at">levels=</span>design)</span>
<span id="cb182-1801"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1801" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">length</span>(comparisons) <span class="sc">&gt;</span><span class="dv">1</span> )</span>
<span id="cb182-1802"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1802" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>( kk <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(comparisons) )</span>
<span id="cb182-1803"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1803" aria-hidden="true" tabindex="-1"></a>                contrast1<span class="ot">&lt;-</span>  <span class="fu">cbind</span>(contrast1,<span class="fu">makeContrasts</span>(<span class="at">contrasts=</span>comparisons[kk], <span class="at">levels=</span>design)   )</span>
<span id="cb182-1804"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1804" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1805"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1805" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( interactionTerm ) {  <span class="co"># if interaction terms</span></span>
<span id="cb182-1806"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1806" aria-hidden="true" tabindex="-1"></a>                contrast1 <span class="ot">=</span> <span class="fu">cbind</span>(contrast1,contrast2)</span>
<span id="cb182-1807"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1807" aria-hidden="true" tabindex="-1"></a>                contrast.names <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">colnames</span>(contrast1), <span class="fu">colnames</span>(contrast2) )</span>
<span id="cb182-1808"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1808" aria-hidden="true" tabindex="-1"></a>                comparisons <span class="ot">=</span> <span class="fu">c</span>(comparisons,comparisons2)</span>
<span id="cb182-1809"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1809" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-1810"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1810" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-1811"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1811" aria-hidden="true" tabindex="-1"></a>            <span class="co"># block design to remove batch effect or paired samples</span></span>
<span id="cb182-1812"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1812" aria-hidden="true" tabindex="-1"></a>            <span class="co"># corfit &lt;- duplicateCorrelation(eset,design,block=targets$Subject)</span></span>
<span id="cb182-1813"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1813" aria-hidden="true" tabindex="-1"></a>            <span class="co"># corfit$consensus</span></span>
<span id="cb182-1814"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1814" aria-hidden="true" tabindex="-1"></a>            <span class="co">#Then this inter-subject correlation is input into the linear model fit:</span></span>
<span id="cb182-1815"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1815" aria-hidden="true" tabindex="-1"></a>            <span class="co"># fit &lt;- lmFit(eset,design,block=targets$Subject,correlation=corfit$consensus)</span></span>
<span id="cb182-1816"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1816" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1817"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1817" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">length</span>(blockFactor) <span class="sc">&gt;=</span> <span class="dv">1</span> ) { <span class="co"># if a factor is selected as block</span></span>
<span id="cb182-1818"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1818" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="fu">length</span>(blockFactor) <span class="sc">&gt;=</span> <span class="dv">1</span> ) </span>
<span id="cb182-1819"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1819" aria-hidden="true" tabindex="-1"></a>                    blockFactor <span class="ot">=</span> blockFactor[<span class="dv">1</span>] <span class="co"># if multiple use the first one</span></span>
<span id="cb182-1820"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1820" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1821"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1821" aria-hidden="true" tabindex="-1"></a>                block <span class="ot">=</span> sampleInfo[, blockFactor]  <span class="co"># the column not used</span></span>
<span id="cb182-1822"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1822" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-1823"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1823" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>(rawCounts) <span class="sc">&amp;&amp;</span> countsDEGMethods <span class="sc">==</span> <span class="dv">2</span>) {  <span class="co"># voom</span></span>
<span id="cb182-1824"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1824" aria-hidden="true" tabindex="-1"></a>                    v <span class="ot">&lt;-</span> <span class="fu">voom</span>(rawCounts, design);</span>
<span id="cb182-1825"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1825" aria-hidden="true" tabindex="-1"></a>                    corfit <span class="ot">&lt;-</span> <span class="fu">duplicateCorrelation</span>(v,design,<span class="at">block=</span> block)           </span>
<span id="cb182-1826"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1826" aria-hidden="true" tabindex="-1"></a>                    fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(v, design,<span class="at">block=</span>block,<span class="at">correlation=</span>corfit<span class="sc">$</span>consensus) </span>
<span id="cb182-1827"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1827" aria-hidden="true" tabindex="-1"></a>                } <span class="cf">else</span> {</span>
<span id="cb182-1828"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1828" aria-hidden="true" tabindex="-1"></a>                    corfit <span class="ot">&lt;-</span> <span class="fu">duplicateCorrelation</span>(eset,design,<span class="at">block=</span> block)</span>
<span id="cb182-1829"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1829" aria-hidden="true" tabindex="-1"></a>                    fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(eset, design,<span class="at">block=</span>block,<span class="at">correlation=</span>corfit<span class="sc">$</span>consensus)</span>
<span id="cb182-1830"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1830" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb182-1831"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1831" aria-hidden="true" tabindex="-1"></a>                fit <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit, <span class="at">trend=</span>limmaTrend)            </span>
<span id="cb182-1832"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1832" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-1833"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1833" aria-hidden="true" tabindex="-1"></a>            } <span class="co"># block factors</span></span>
<span id="cb182-1834"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1834" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1835"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1835" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1836"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1836" aria-hidden="true" tabindex="-1"></a>        } <span class="co"># use selected factors</span></span>
<span id="cb182-1837"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1837" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1838"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1838" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1839"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1839" aria-hidden="true" tabindex="-1"></a>        fit2 <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit, contrast1)</span>
<span id="cb182-1840"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1840" aria-hidden="true" tabindex="-1"></a>        fit2 <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit2, <span class="at">trend=</span>limmaTrend)</span>
<span id="cb182-1841"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1841" aria-hidden="true" tabindex="-1"></a>        <span class="co">#topTable(fit2, coef=1, adjust=&quot;BH&quot;)</span></span>
<span id="cb182-1842"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1842" aria-hidden="true" tabindex="-1"></a>        results <span class="ot">&lt;-</span> <span class="fu">decideTests</span>(fit2, <span class="at">p.value=</span>maxP_limma, <span class="at">lfc=</span> <span class="fu">log2</span>(minFC_limma ))</span>
<span id="cb182-1843"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1843" aria-hidden="true" tabindex="-1"></a>        <span class="co">#vennDiagram(results[,1:5],circle.col=rainbow(5))</span></span>
<span id="cb182-1844"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1844" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1845"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1845" aria-hidden="true" tabindex="-1"></a>        <span class="co">#colnames(results) = unlist(sapply( colnames(results), changeNames  ) )</span></span>
<span id="cb182-1846"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1846" aria-hidden="true" tabindex="-1"></a>        <span class="co">#comparisons3 &lt;-  unlist(sapply( comparisons, changeNames  ) ) </span></span>
<span id="cb182-1847"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1847" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1848"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1848" aria-hidden="true" tabindex="-1"></a>        <span class="co"># extract fold change for each comparison</span></span>
<span id="cb182-1849"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1849" aria-hidden="true" tabindex="-1"></a>        <span class="co"># there is issues with direction of foldchange. Sometimes opposite</span></span>
<span id="cb182-1850"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1850" aria-hidden="true" tabindex="-1"></a>        top <span class="ot">&lt;-</span> <span class="cf">function</span> (comp) {</span>
<span id="cb182-1851"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1851" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit2, <span class="at">number =</span> <span class="fl">1e12</span>,<span class="at">coef=</span>comp,<span class="at">sort.by=</span><span class="st">&quot;M&quot;</span> ) </span>
<span id="cb182-1852"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1852" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(tem)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>) { <span class="fu">return</span> (<span class="dv">1</span>) </span>
<span id="cb182-1853"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1853" aria-hidden="true" tabindex="-1"></a>            } <span class="cf">else</span>  {           </span>
<span id="cb182-1854"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1854" aria-hidden="true" tabindex="-1"></a>                <span class="co"># compute fold change for the first gene (ranked by absolute value)</span></span>
<span id="cb182-1855"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1855" aria-hidden="true" tabindex="-1"></a>                tem2 <span class="ot">=</span> <span class="fu">as.numeric</span>( x[ <span class="fu">which</span>(<span class="fu">rownames</span>(x)<span class="sc">==</span> <span class="fu">rownames</span>(tem)[<span class="dv">1</span>]) , ] )</span>
<span id="cb182-1856"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1856" aria-hidden="true" tabindex="-1"></a>                <span class="fu">names</span>(tem2) <span class="ot">=</span> <span class="fu">colnames</span>(x) </span>
<span id="cb182-1857"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1857" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb182-1858"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1858" aria-hidden="true" tabindex="-1"></a>                <span class="fu">return</span>( tem[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">5</span>)]) </span>
<span id="cb182-1859"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1859" aria-hidden="true" tabindex="-1"></a>            }  </span>
<span id="cb182-1860"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1860" aria-hidden="true" tabindex="-1"></a>                                                    </span>
<span id="cb182-1861"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1861" aria-hidden="true" tabindex="-1"></a>        }  <span class="co"># no significant gene returns 1, otherwise a data frame</span></span>
<span id="cb182-1862"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1862" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1863"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1863" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1864"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1864" aria-hidden="true" tabindex="-1"></a>        topGenes <span class="ot">&lt;-</span> <span class="fu">lapply</span>(comparisons, top)</span>
<span id="cb182-1865"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1865" aria-hidden="true" tabindex="-1"></a>        topGenes <span class="ot">&lt;-</span> <span class="fu">setNames</span>(topGenes, comparisons )</span>
<span id="cb182-1866"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1866" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1867"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1867" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">&lt;-</span> <span class="fu">which</span>( <span class="fu">unlist</span>( <span class="fu">lapply</span>(topGenes, class) ) <span class="sc">==</span> <span class="st">&quot;numeric&quot;</span>)</span>
<span id="cb182-1868"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1868" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">length</span>(ix)<span class="sc">&gt;</span><span class="dv">0</span>) topGenes <span class="ot">&lt;-</span> topGenes[ <span class="sc">-</span> ix ]</span>
<span id="cb182-1869"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1869" aria-hidden="true" tabindex="-1"></a>        <span class="co"># if (length(topGenes) == 0) topGenes = NULL;</span></span>
<span id="cb182-1870"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1870" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-1871"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1871" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1872"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1872" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(&quot;\n&quot;, names(topGenes) )</span></span>
<span id="cb182-1873"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1873" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(&quot;\n&quot;, colnames(results))</span></span>
<span id="cb182-1874"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1874" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(&quot;\n&quot;, comparisons3)</span></span>
<span id="cb182-1875"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1875" aria-hidden="true" tabindex="-1"></a>    <span class="co">#comparisons &lt;- comparisons3</span></span>
<span id="cb182-1876"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1876" aria-hidden="true" tabindex="-1"></a>    <span class="co"># it does not make any sense! comparisons can not be changed!</span></span>
<span id="cb182-1877"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1877" aria-hidden="true" tabindex="-1"></a>    <span class="co">#comparisons =comparisons3  </span></span>
<span id="cb182-1878"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1878" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(&quot;\n&quot;, comparisons)     </span></span>
<span id="cb182-1879"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1879" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1880"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1880" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">results=</span> results, <span class="at">comparisons=</span>comparisons, <span class="at">Exp.type=</span>Exp.type, <span class="at">topGenes=</span>topGenes)) </span>
<span id="cb182-1881"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1881" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-1882"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1882" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1883"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1883" aria-hidden="true" tabindex="-1"></a><span class="co"># Differential expression using DESeq2</span></span>
<span id="cb182-1884"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1884" aria-hidden="true" tabindex="-1"></a>DEG.DESeq2 <span class="ot">&lt;-</span> <span class="cf">function</span> (  rawCounts,<span class="at">maxP_limma=</span>.<span class="dv">05</span>, <span class="at">minFC_limma=</span><span class="dv">2</span>, <span class="at">selectedComparisons=</span><span class="cn">NULL</span>, <span class="at">sampleInfo =</span> <span class="cn">NULL</span>,<span class="at">modelFactors=</span><span class="cn">NULL</span>, <span class="at">blockFactor =</span> <span class="cn">NULL</span>, <span class="at">referenceLevels=</span><span class="cn">NULL</span>){</span>
<span id="cb182-1885"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1885" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(DESeq2,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># count data analysis</span></span>
<span id="cb182-1886"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1886" aria-hidden="true" tabindex="-1"></a>    groups <span class="ot">=</span> <span class="fu">as.character</span> ( <span class="fu">detectGroups</span>( <span class="fu">colnames</span>( rawCounts ) ) )</span>
<span id="cb182-1887"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1887" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">=</span> <span class="fu">unique</span>(groups)<span class="co"># order is reversed</span></span>
<span id="cb182-1888"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1888" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1889"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1889" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1890"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1890" aria-hidden="true" tabindex="-1"></a>    <span class="co"># check for replicates, removes samples without replicates</span></span>
<span id="cb182-1891"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1891" aria-hidden="true" tabindex="-1"></a>    reps <span class="ot">=</span> <span class="fu">as.matrix</span>(<span class="fu">table</span>(groups)) <span class="co"># number of replicates per biological sample</span></span>
<span id="cb182-1892"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1892" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> ( <span class="fu">sum</span>( reps[,<span class="dv">1</span>] <span class="sc">&gt;=</span> <span class="dv">2</span>) <span class="sc">&lt;</span><span class="dv">2</span> ) <span class="co"># if less than 2 samples with replicates</span></span>
<span id="cb182-1893"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1893" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">results=</span> <span class="cn">NULL</span>, <span class="at">comparisons =</span> <span class="cn">NULL</span>, <span class="at">Exp.type=</span><span class="st">&quot;Failed to parse sample names to define groups. </span></span>
<span id="cb182-1894"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1894" aria-hidden="true" tabindex="-1"></a><span class="st">        Cannot perform DEGs and pathway analysis. Please double check column names! Use WT_Rep1, WT_Rep2 etc. &quot;</span>, <span class="at">topGenes=</span><span class="cn">NULL</span>)) </span>
<span id="cb182-1895"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1895" aria-hidden="true" tabindex="-1"></a>    <span class="co"># remove samples without replicates</span></span>
<span id="cb182-1896"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1896" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">rownames</span>(reps)[<span class="fu">which</span>(reps[,<span class="dv">1</span>] <span class="sc">&gt;</span><span class="dv">1</span>)]</span>
<span id="cb182-1897"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1897" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">&lt;-</span> <span class="fu">which</span>( groups <span class="sc">%in%</span> g)  </span>
<span id="cb182-1898"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1898" aria-hidden="true" tabindex="-1"></a>    groups <span class="ot">&lt;-</span> groups[ix]   </span>
<span id="cb182-1899"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1899" aria-hidden="true" tabindex="-1"></a>    rawCounts <span class="ot">&lt;-</span> rawCounts[,ix] </span>
<span id="cb182-1900"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1900" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1901"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1901" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1902"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1902" aria-hidden="true" tabindex="-1"></a>    Exp.type <span class="ot">=</span> <span class="fu">paste</span>(<span class="fu">length</span>(g),<span class="st">&quot; sample groups detected.&quot;</span>)</span>
<span id="cb182-1903"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1903" aria-hidden="true" tabindex="-1"></a>    comparisons <span class="ot">=</span> <span class="st">&quot;&quot;</span></span>
<span id="cb182-1904"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1904" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(g)<span class="sc">-</span><span class="dv">1</span>) )</span>
<span id="cb182-1905"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1905" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (j <span class="cf">in</span> (i<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(g)) </span>
<span id="cb182-1906"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1906" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">=</span> <span class="fu">c</span>(comparisons,<span class="fu">paste</span>(g[j],<span class="st">&quot;-&quot;</span>,g[i],<span class="at">sep=</span><span class="st">&quot;&quot;</span> ) )</span>
<span id="cb182-1907"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1907" aria-hidden="true" tabindex="-1"></a>    comparisons <span class="ot">&lt;-</span> comparisons[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb182-1908"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1908" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1909"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1909" aria-hidden="true" tabindex="-1"></a>    colData <span class="ot">=</span> <span class="fu">cbind</span>(<span class="fu">colnames</span>(rawCounts), groups )</span>
<span id="cb182-1910"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1910" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1911"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1911" aria-hidden="true" tabindex="-1"></a>    <span class="co"># no sample file, but user selected comparisons using column names</span></span>
<span id="cb182-1912"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1912" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">is.null</span>(modelFactors) <span class="sc">&amp;</span> <span class="fu">length</span>( selectedComparisons) <span class="sc">&gt;</span><span class="dv">0</span>  )  </span>
<span id="cb182-1913"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1913" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">=</span> selectedComparisons</span>
<span id="cb182-1914"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1914" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1915"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1915" aria-hidden="true" tabindex="-1"></a>    comparisons2 <span class="ot">=</span> comparisons   <span class="co"># this is for showing comparison names, which might be different from internally   </span></span>
<span id="cb182-1916"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1916" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Set up the DESeqDataSet Object and run the DESeq pipeline</span></span>
<span id="cb182-1917"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1917" aria-hidden="true" tabindex="-1"></a>    dds <span class="ot">=</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData=</span>rawCounts,</span>
<span id="cb182-1918"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1918" aria-hidden="true" tabindex="-1"></a>                                <span class="at">colData=</span>colData,</span>
<span id="cb182-1919"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1919" aria-hidden="true" tabindex="-1"></a>                                <span class="at">design=</span><span class="sc">~</span>groups)                             </span>
<span id="cb182-1920"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1920" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1921"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1921" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">is.null</span>(modelFactors)  ) </span>
<span id="cb182-1922"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1922" aria-hidden="true" tabindex="-1"></a>        dds <span class="ot">=</span> <span class="fu">DESeq</span>(dds)    <span class="cf">else</span>  </span>
<span id="cb182-1923"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1923" aria-hidden="true" tabindex="-1"></a>    {    <span class="co"># using selected factors and comparisons</span></span>
<span id="cb182-1924"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1924" aria-hidden="true" tabindex="-1"></a>        <span class="co"># build model</span></span>
<span id="cb182-1925"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1925" aria-hidden="true" tabindex="-1"></a>        modelFactors <span class="ot">=</span> <span class="fu">c</span>(modelFactors,blockFactor) <span class="co"># block factor is just added in. </span></span>
<span id="cb182-1926"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1926" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1927"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1927" aria-hidden="true" tabindex="-1"></a>        factors <span class="ot">=</span> modelFactors   <span class="co"># selected factors and interactions: c( &quot;strain&quot;, &quot;treatment&quot;,  &quot;strain:treatment&quot;)</span></span>
<span id="cb182-1928"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1928" aria-hidden="true" tabindex="-1"></a>        factors <span class="ot">=</span> factors[ <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;:&quot;</span>,factors )]   <span class="co"># non-interaction terms</span></span>
<span id="cb182-1929"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1929" aria-hidden="true" tabindex="-1"></a>        <span class="co"># interaction terms like strain:treatment</span></span>
<span id="cb182-1930"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1930" aria-hidden="true" tabindex="-1"></a>        Interactions <span class="ot">=</span> modelFactors[ <span class="fu">grepl</span>(<span class="st">&quot;:&quot;</span>,modelFactors )]</span>
<span id="cb182-1931"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1931" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1932"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1932" aria-hidden="true" tabindex="-1"></a>        colData <span class="ot">=</span> sampleInfo  </span>
<span id="cb182-1933"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1933" aria-hidden="true" tabindex="-1"></a>        factorsCoded <span class="ot">=</span> <span class="fu">toupper</span>(letters )[<span class="dv">1</span><span class="sc">:</span> <span class="fu">dim</span>(colData)[<span class="dv">2</span>] ]   <span class="co"># Factors are encoded as &quot;A&quot;, &quot;B&quot;, &quot;C&quot;; this avoid illigal letters</span></span>
<span id="cb182-1934"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1934" aria-hidden="true" tabindex="-1"></a>        <span class="fu">names</span>(factorsCoded) <span class="ot">=</span>  <span class="fu">colnames</span>(colData)  <span class="co"># this is for look up; each column of sampleInfo  </span></span>
<span id="cb182-1935"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1935" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(colData) <span class="ot">=</span> factorsCoded <span class="co"># all columns named A B C D </span></span>
<span id="cb182-1936"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1936" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1937"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1937" aria-hidden="true" tabindex="-1"></a>        colData <span class="ot">=</span> <span class="fu">as.data.frame</span>(colData)</span>
<span id="cb182-1938"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1938" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1939"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1939" aria-hidden="true" tabindex="-1"></a>        <span class="co"># set reference levels for factors</span></span>
<span id="cb182-1940"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1940" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="sc">!</span> <span class="fu">is.null</span>( referenceLevels) ) {   <span class="co"># c(&quot;genotype:wt&quot;, &quot;treatment:control&quot; )</span></span>
<span id="cb182-1941"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1941" aria-hidden="true" tabindex="-1"></a>            <span class="co"># first factor</span></span>
<span id="cb182-1942"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1942" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> ( refs <span class="cf">in</span> referenceLevels)</span>
<span id="cb182-1943"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1943" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="sc">!</span> <span class="fu">is.null</span>( refs) ) {</span>
<span id="cb182-1944"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1944" aria-hidden="true" tabindex="-1"></a>                    ix <span class="ot">=</span> <span class="fu">match</span>(<span class="fu">gsub</span>(<span class="st">&quot;:.*&quot;</span>,<span class="st">&quot;&quot;</span>,refs), <span class="fu">colnames</span>(sampleInfo) ) <span class="co"># corresponding column id for factor</span></span>
<span id="cb182-1945"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1945" aria-hidden="true" tabindex="-1"></a>                    colData[,ix] <span class="ot">=</span> <span class="fu">as.factor</span>( colData[,ix] )</span>
<span id="cb182-1946"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1946" aria-hidden="true" tabindex="-1"></a>                    colData[,ix] <span class="ot">=</span> <span class="fu">relevel</span>(colData[,ix],<span class="fu">gsub</span>(<span class="st">&quot;.*:&quot;</span>,<span class="st">&quot;&quot;</span>,refs)  )</span>
<span id="cb182-1947"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1947" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb182-1948"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1948" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-1949"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1949" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-1950"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1950" aria-hidden="true" tabindex="-1"></a>        <span class="co"># base model</span></span>
<span id="cb182-1951"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1951" aria-hidden="true" tabindex="-1"></a>        DESeq2.Object<span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;dds = DESeqDataSetFromMatrix(countData=rawCounts, colData=colData, design=~ &quot;</span>, </span>
<span id="cb182-1952"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1952" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">paste</span>( factorsCoded[factors],<span class="at">collapse=</span><span class="st">&quot;+&quot;</span>)) <span class="co"># only use selected factors        </span></span>
<span id="cb182-1953"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1953" aria-hidden="true" tabindex="-1"></a>        Exp.type <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;Model: ~&quot;</span>, <span class="fu">paste</span>(modelFactors,<span class="at">collapse=</span><span class="st">&quot; + &quot;</span>) )</span>
<span id="cb182-1954"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1954" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1955"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1955" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1956"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1956" aria-hidden="true" tabindex="-1"></a>        <span class="co"># create model</span></span>
<span id="cb182-1957"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1957" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">length</span>(Interactions)<span class="sc">&gt;</span><span class="dv">0</span> ) { <span class="co"># if there is interaction</span></span>
<span id="cb182-1958"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1958" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>( interactionTerms <span class="cf">in</span> Interactions) {</span>
<span id="cb182-1959"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1959" aria-hidden="true" tabindex="-1"></a>                interactingFactors <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(interactionTerms,<span class="st">&quot;:&quot;</span> ) )  <span class="co"># split strain:treatment as &quot;strain&quot; and &quot;mutant&quot;</span></span>
<span id="cb182-1960"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1960" aria-hidden="true" tabindex="-1"></a>                tem <span class="ot">=</span> <span class="fu">paste</span>(factorsCoded [ interactingFactors ],<span class="at">collapse=</span><span class="st">&quot;:&quot;</span>)   <span class="co"># convert &quot;strain:mutant&quot; to &quot;A:B&quot;</span></span>
<span id="cb182-1961"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1961" aria-hidden="true" tabindex="-1"></a>                DESeq2.Object <span class="ot">=</span> <span class="fu">paste</span>(DESeq2.Object, <span class="st">&quot; + &quot;</span>,tem)</span>
<span id="cb182-1962"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1962" aria-hidden="true" tabindex="-1"></a>            }           </span>
<span id="cb182-1963"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1963" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-1964"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1964" aria-hidden="true" tabindex="-1"></a>        DESeq2.Object<span class="ot">=</span> <span class="fu">paste</span>( DESeq2.Object, <span class="st">&quot;)&quot;</span>) <span class="co"># ends the model</span></span>
<span id="cb182-1965"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1965" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1966"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1966" aria-hidden="true" tabindex="-1"></a>        <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> DESeq2.Object) )</span>
<span id="cb182-1967"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1967" aria-hidden="true" tabindex="-1"></a>        dds <span class="ot">=</span> <span class="fu">DESeq</span>(dds)  <span class="co"># main function       </span></span>
<span id="cb182-1968"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1968" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1969"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1969" aria-hidden="true" tabindex="-1"></a>        <span class="co"># comparisons </span></span>
<span id="cb182-1970"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1970" aria-hidden="true" tabindex="-1"></a>        <span class="co"># &quot;group: control vs. mutant&quot;</span></span>
<span id="cb182-1971"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1971" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;.*: &quot;</span>,<span class="st">&quot;&quot;</span>,selectedComparisons)</span>
<span id="cb182-1972"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1972" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; vs</span><span class="sc">\\</span><span class="st">. &quot;</span>,<span class="st">&quot;-&quot;</span>,comparisons)</span>
<span id="cb182-1973"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1973" aria-hidden="true" tabindex="-1"></a>        factorsVector<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;:.*&quot;</span>,<span class="st">&quot;&quot;</span>,selectedComparisons) <span class="co"># corresponding factors for each comparison</span></span>
<span id="cb182-1974"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1974" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1975"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1975" aria-hidden="true" tabindex="-1"></a>        <span class="co"># comparison2 holds names for display with real factor names</span></span>
<span id="cb182-1976"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1976" aria-hidden="true" tabindex="-1"></a>        <span class="co"># comparison  is used in calculation it is A, B, C for factors</span></span>
<span id="cb182-1977"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1977" aria-hidden="true" tabindex="-1"></a>        comparisons2 <span class="ot">=</span> comparisons    </span>
<span id="cb182-1978"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1978" aria-hidden="true" tabindex="-1"></a>        <span class="co">#comparisons2 = gsub(&quot; vs\\. &quot;,&quot;-&quot;,selectedComparisons) </span></span>
<span id="cb182-1979"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1979" aria-hidden="true" tabindex="-1"></a>        <span class="co">#comparisons2 = gsub(&quot;:&quot;,&quot;_&quot;,comparisons2)      </span></span>
<span id="cb182-1980"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1980" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Note that with interaction terms, not all meaningful comparisons is listed for selection. </span></span>
<span id="cb182-1981"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1981" aria-hidden="true" tabindex="-1"></a>        <span class="co"># this is complex. Only under reference level.</span></span>
<span id="cb182-1982"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1982" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-1983"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1983" aria-hidden="true" tabindex="-1"></a>        <span class="co"># comparisons due to interaction terms</span></span>
<span id="cb182-1984"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1984" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">length</span>(Interactions)<span class="sc">&gt;</span><span class="dv">0</span> ) { <span class="co"># if there is interaction</span></span>
<span id="cb182-1985"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1985" aria-hidden="true" tabindex="-1"></a>            interactionComparisons <span class="ot">=</span> <span class="fu">resultsNames</span>(dds)</span>
<span id="cb182-1986"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1986" aria-hidden="true" tabindex="-1"></a>            interactionComparisons <span class="ot">=</span> interactionComparisons[ <span class="fu">grepl</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,interactionComparisons )   ]</span>
<span id="cb182-1987"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1987" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-1988"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1988" aria-hidden="true" tabindex="-1"></a>            comparisons <span class="ot">=</span> <span class="fu">c</span>(comparisons,interactionComparisons )</span>
<span id="cb182-1989"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1989" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-1990"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1990" aria-hidden="true" tabindex="-1"></a>            <span class="co"># translate comparisons generated in interaction terms back to real factor names</span></span>
<span id="cb182-1991"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1991" aria-hidden="true" tabindex="-1"></a>            interactionComparisons2 <span class="ot">=</span> interactionComparisons</span>
<span id="cb182-1992"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1992" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> ( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(interactionComparisons2 ) ) {</span>
<span id="cb182-1993"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1993" aria-hidden="true" tabindex="-1"></a>                tem <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(interactionComparisons2[i],<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span> ) )</span>
<span id="cb182-1994"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1994" aria-hidden="true" tabindex="-1"></a>                tem_factors <span class="ot">=</span> <span class="fu">substr</span>(tem,<span class="dv">1</span>,<span class="dv">1</span>) </span>
<span id="cb182-1995"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1995" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-1996"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1996" aria-hidden="true" tabindex="-1"></a>                tem_factors[<span class="dv">1</span>] <span class="ot">=</span> <span class="fu">names</span>(factorsCoded)[factorsCoded <span class="sc">==</span> tem_factors[<span class="dv">1</span>]]  <span class="co"># get the first letter and translate into real factor names</span></span>
<span id="cb182-1997"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1997" aria-hidden="true" tabindex="-1"></a>                tem_factors[<span class="dv">2</span>] <span class="ot">=</span> <span class="fu">names</span>(factorsCoded)[factorsCoded <span class="sc">==</span> tem_factors[<span class="dv">2</span>]]  <span class="co"># get the 2nd letters and translate into real factor names</span></span>
<span id="cb182-1998"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1998" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-1999"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-1999" aria-hidden="true" tabindex="-1"></a>                interactionComparisons2[i] <span class="ot">&lt;-</span> <span class="fu">paste0</span>( <span class="st">&quot;I:&quot;</span>,tem_factors[<span class="dv">1</span>], <span class="st">&quot;_&quot;</span>,<span class="fu">substr</span>(tem[<span class="dv">1</span>],<span class="dv">2</span>,<span class="fu">nchar</span>(tem[<span class="dv">1</span>]) ),<span class="st">&quot;.&quot;</span>,</span>
<span id="cb182-2000"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2000" aria-hidden="true" tabindex="-1"></a>                                                              tem_factors[<span class="dv">2</span>], <span class="st">&quot;_&quot;</span>,<span class="fu">substr</span>(tem[<span class="dv">2</span>],<span class="dv">2</span>,<span class="fu">nchar</span>(tem[<span class="dv">2</span>]) ) </span>
<span id="cb182-2001"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2001" aria-hidden="true" tabindex="-1"></a>                                                    )               </span>
<span id="cb182-2002"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2002" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-2003"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2003" aria-hidden="true" tabindex="-1"></a>            comparisons2 <span class="ot">=</span> <span class="fu">c</span>(comparisons2,interactionComparisons2 )</span>
<span id="cb182-2004"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2004" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-2005"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2005" aria-hidden="true" tabindex="-1"></a>    } <span class="co"># if selected factors </span></span>
<span id="cb182-2006"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2006" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-2007"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2007" aria-hidden="true" tabindex="-1"></a>    <span class="co"># extract contrasts according to comprisons defined above</span></span>
<span id="cb182-2008"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2008" aria-hidden="true" tabindex="-1"></a>    result1 <span class="ot">=</span> <span class="cn">NULL</span>; allCalls <span class="ot">=</span> <span class="cn">NULL</span>;</span>
<span id="cb182-2009"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2009" aria-hidden="true" tabindex="-1"></a>    topGenes <span class="ot">=</span> <span class="fu">list</span>(); pk <span class="ot">=</span> <span class="dv">1</span> <span class="co"># counter</span></span>
<span id="cb182-2010"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2010" aria-hidden="true" tabindex="-1"></a>    pp<span class="ot">=</span><span class="dv">0</span> <span class="co"># first results?</span></span>
<span id="cb182-2011"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2011" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>( kk <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(comparisons) ) {</span>
<span id="cb182-2012"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2012" aria-hidden="true" tabindex="-1"></a>        tem <span class="ot">=</span> <span class="fu">unlist</span>( <span class="fu">strsplit</span>(comparisons[kk],<span class="st">&quot;-&quot;</span>) )</span>
<span id="cb182-2013"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2013" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-2014"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2014" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>(modelFactors)) <span class="co"># if just group comparison using sample names</span></span>
<span id="cb182-2015"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2015" aria-hidden="true" tabindex="-1"></a>            selected <span class="ot">=</span> <span class="fu">results</span>(dds, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">&quot;groups&quot;</span>, tem[<span class="dv">1</span>], tem[<span class="dv">2</span>]) )   <span class="cf">else</span> {</span>
<span id="cb182-2016"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2016" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>, comparisons[kk] ) )    <span class="co"># if not interaction term: they contain .  interaction term</span></span>
<span id="cb182-2017"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2017" aria-hidden="true" tabindex="-1"></a>                selected <span class="ot">=</span> <span class="fu">results</span>(dds, <span class="at">contrast=</span><span class="fu">c</span>( factorsCoded[ factorsVector[kk] ],tem[<span class="dv">1</span>], tem[<span class="dv">2</span>]) ) <span class="cf">else</span> <span class="co"># either A, B, C ...</span></span>
<span id="cb182-2018"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2018" aria-hidden="true" tabindex="-1"></a>                selected <span class="ot">=</span> <span class="fu">results</span>(dds, <span class="at">name=</span>comparisons[kk] ) <span class="co"># interaction term</span></span>
<span id="cb182-2019"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2019" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-2020"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2020" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2021"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2021" aria-hidden="true" tabindex="-1"></a>        selected<span class="sc">$</span>calls <span class="ot">=</span><span class="dv">0</span>   </span>
<span id="cb182-2022"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2022" aria-hidden="true" tabindex="-1"></a>        selected<span class="sc">$</span>calls [<span class="fu">which</span>( selected<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="fu">log2</span>(minFC_limma) <span class="sc">&amp;</span> selected<span class="sc">$</span>padj <span class="sc">&lt;</span> maxP_limma ) ]  <span class="ot">&lt;-</span>  <span class="dv">1</span></span>
<span id="cb182-2023"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2023" aria-hidden="true" tabindex="-1"></a>        selected<span class="sc">$</span>calls [ <span class="fu">which</span>( selected<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span>  <span class="sc">-</span><span class="fu">log2</span>(minFC_limma) <span class="sc">&amp;</span> selected<span class="sc">$</span>padj <span class="sc">&lt;</span> maxP_limma ) ] <span class="ot">&lt;-</span>  <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb182-2024"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2024" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(selected)<span class="ot">=</span> <span class="fu">paste</span>( <span class="fu">as.character</span>(comparisons2[kk]), <span class="st">&quot;___&quot;</span>,<span class="fu">colnames</span>(selected),<span class="at">sep=</span><span class="st">&quot;&quot;</span> )</span>
<span id="cb182-2025"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2025" aria-hidden="true" tabindex="-1"></a>        selected <span class="ot">=</span> <span class="fu">as.data.frame</span>(selected)</span>
<span id="cb182-2026"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2026" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (pp<span class="sc">==</span><span class="dv">0</span>){  <span class="co"># if first one with significant genes, collect gene list and Pval+ fold</span></span>
<span id="cb182-2027"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2027" aria-hidden="true" tabindex="-1"></a>        result1 <span class="ot">=</span> selected; pp <span class="ot">=</span> <span class="dv">1</span>; </span>
<span id="cb182-2028"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2028" aria-hidden="true" tabindex="-1"></a>        <span class="co"># selected[,2] &lt;- -1 * selected[,2] # reverse fold change direction</span></span>
<span id="cb182-2029"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2029" aria-hidden="true" tabindex="-1"></a>        topGenes[[<span class="dv">1</span>]] <span class="ot">=</span> selected[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">6</span>)]; </span>
<span id="cb182-2030"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2030" aria-hidden="true" tabindex="-1"></a>        <span class="fu">names</span>(topGenes)[<span class="dv">1</span>] <span class="ot">=</span> comparisons2[kk]; } <span class="cf">else</span> </span>
<span id="cb182-2031"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2031" aria-hidden="true" tabindex="-1"></a>            { result1 <span class="ot">=</span> <span class="fu">merge</span>(result1,selected,<span class="at">by=</span><span class="st">&quot;row.names&quot;</span>); </span>
<span id="cb182-2032"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2032" aria-hidden="true" tabindex="-1"></a>                <span class="fu">rownames</span>(result1) <span class="ot">=</span> result1[,<span class="dv">1</span>]; </span>
<span id="cb182-2033"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2033" aria-hidden="true" tabindex="-1"></a>                result1 <span class="ot">&lt;-</span> result1[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb182-2034"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2034" aria-hidden="true" tabindex="-1"></a>                pk<span class="ot">=</span> pk<span class="sc">+</span><span class="dv">1</span>; </span>
<span id="cb182-2035"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2035" aria-hidden="true" tabindex="-1"></a>                <span class="co"># selected[,2] &lt;- -1 * selected[,2] # reverse fold change direction</span></span>
<span id="cb182-2036"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2036" aria-hidden="true" tabindex="-1"></a>                topGenes[[pk]] <span class="ot">=</span> selected[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">6</span>)]; </span>
<span id="cb182-2037"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2037" aria-hidden="true" tabindex="-1"></a>                <span class="fu">names</span>(topGenes)[pk] <span class="ot">=</span> comparisons2[kk];  <span class="co"># assign name to comprison</span></span>
<span id="cb182-2038"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2038" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-2039"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2039" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-2040"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2040" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2041"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2041" aria-hidden="true" tabindex="-1"></a>    Interactions <span class="ot">=</span> <span class="fu">c</span>()</span>
<span id="cb182-2042"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2042" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>(modelFactors) )</span>
<span id="cb182-2043"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2043" aria-hidden="true" tabindex="-1"></a>        Interactions <span class="ot">=</span> modelFactors[ <span class="fu">grepl</span>(<span class="st">&quot;:&quot;</span>,modelFactors )]</span>
<span id="cb182-2044"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2044" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-2045"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2045" aria-hidden="true" tabindex="-1"></a><span class="co">#---  add comprisons for non-reference levels. It adds to the results1 object.  </span></span>
<span id="cb182-2046"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2046" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">length</span>(Interactions)<span class="sc">&gt;</span><span class="dv">0</span> ) { <span class="co"># if there is interaction</span></span>
<span id="cb182-2047"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2047" aria-hidden="true" tabindex="-1"></a>        factorLookup<span class="ot">=</span><span class="fu">c</span>() <span class="co"># a factor whose values are factors and names are factor and level combination conditionTreated, genotypeWT</span></span>
<span id="cb182-2048"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2048" aria-hidden="true" tabindex="-1"></a>        levelLookup <span class="ot">=</span> <span class="fu">c</span>()</span>
<span id="cb182-2049"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2049" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-2050"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2050" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(sampleInfo)[<span class="dv">2</span>]) {</span>
<span id="cb182-2051"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2051" aria-hidden="true" tabindex="-1"></a>            sampleInfo2 <span class="ot">=</span> <span class="fu">unique</span>(sampleInfo)</span>
<span id="cb182-2052"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2052" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">rep</span>(<span class="fu">toupper</span>(letters)[i],<span class="fu">dim</span>(sampleInfo2)[<span class="dv">1</span>]  )</span>
<span id="cb182-2053"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2053" aria-hidden="true" tabindex="-1"></a>            <span class="fu">names</span>(tem) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">toupper</span>(letters)[i],sampleInfo2[,i])</span>
<span id="cb182-2054"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2054" aria-hidden="true" tabindex="-1"></a>            factorLookup <span class="ot">=</span> <span class="fu">c</span>(factorLookup,tem)  </span>
<span id="cb182-2055"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2055" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2056"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2056" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">as.character</span>( sampleInfo2[,i] )</span>
<span id="cb182-2057"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2057" aria-hidden="true" tabindex="-1"></a>            <span class="fu">names</span>(tem) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">toupper</span>(letters)[i],sampleInfo2[,i])</span>
<span id="cb182-2058"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2058" aria-hidden="true" tabindex="-1"></a>            levelLookup <span class="ot">=</span> <span class="fu">c</span>(levelLookup, tem)</span>
<span id="cb182-2059"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2059" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-2060"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2060" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-2061"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2061" aria-hidden="true" tabindex="-1"></a>        <span class="co"># split  genotypeI.conditionTrt --&gt; c(&quot;genotype&quot;,&quot;I&quot;,&quot;conditoin&quot;,&quot;Trt&quot;)</span></span>
<span id="cb182-2062"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2062" aria-hidden="true" tabindex="-1"></a>        splitInteractionTerms <span class="ot">&lt;-</span> <span class="cf">function</span> (term) {</span>
<span id="cb182-2063"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2063" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,term) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-2064"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2064" aria-hidden="true" tabindex="-1"></a>            terms2 <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(term,<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>) )</span>
<span id="cb182-2065"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2065" aria-hidden="true" tabindex="-1"></a>                     <span class="co"># factor1, level1, factor2, level2</span></span>
<span id="cb182-2066"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2066" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>(<span class="fu">c</span>(factorLookup[terms2[<span class="dv">1</span>]], levelLookup[terms2[<span class="dv">1</span>]],factorLookup[terms2[<span class="dv">2</span>]], levelLookup[terms2[<span class="dv">2</span>]]   ) )</span>
<span id="cb182-2067"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2067" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-2068"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2068" aria-hidden="true" tabindex="-1"></a>        <span class="co"># none interaction terms </span></span>
<span id="cb182-2069"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2069" aria-hidden="true" tabindex="-1"></a>        NoneInterTerms <span class="ot">=</span> <span class="fu">resultsNames</span>(dds)[ <span class="sc">!</span><span class="fu">grepl</span>( <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>, <span class="fu">resultsNames</span>(dds)) ]</span>
<span id="cb182-2070"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2070" aria-hidden="true" tabindex="-1"></a>        NoneInterTerms<span class="ot">=</span>NoneInterTerms[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb182-2071"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2071" aria-hidden="true" tabindex="-1"></a>        allInteractionTerms <span class="ot">=</span> <span class="fu">resultsNames</span>(dds)[ <span class="fu">grepl</span>( <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>, <span class="fu">resultsNames</span>(dds)) ]</span>
<span id="cb182-2072"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2072" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2073"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2073" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2074"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2074" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( kk <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(NoneInterTerms) ) { <span class="co"># for each none interaction term</span></span>
<span id="cb182-2075"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2075" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(modelFactors) ) {<span class="co"># if not just group comparison using sample names</span></span>
<span id="cb182-2076"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2076" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#current factor</span></span>
<span id="cb182-2077"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2077" aria-hidden="true" tabindex="-1"></a>                    cFactor <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>,<span class="st">&quot;&quot;</span>,NoneInterTerms[kk] )</span>
<span id="cb182-2078"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2078" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-2079"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2079" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span>(interactionTerm <span class="cf">in</span> allInteractionTerms ) {</span>
<span id="cb182-2080"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2080" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb182-2081"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2081" aria-hidden="true" tabindex="-1"></a>                        splited <span class="ot">=</span> <span class="fu">splitInteractionTerms</span> (interactionTerm)  <span class="co"># 4 components</span></span>
<span id="cb182-2082"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2082" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span> (cFactor <span class="sc">!=</span> splited[<span class="dv">1</span>] <span class="sc">&amp;</span> cFactor <span class="sc">!=</span> splited[<span class="dv">3</span>]  ) </span>
<span id="cb182-2083"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2083" aria-hidden="true" tabindex="-1"></a>                            <span class="cf">next</span>;                       </span>
<span id="cb182-2084"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2084" aria-hidden="true" tabindex="-1"></a>                        </span>
<span id="cb182-2085"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2085" aria-hidden="true" tabindex="-1"></a>                        selected <span class="ot">=</span> <span class="fu">results</span>(dds, <span class="fu">list</span>(<span class="fu">c</span>( NoneInterTerms[kk],interactionTerm ) ) ) </span>
<span id="cb182-2086"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2086" aria-hidden="true" tabindex="-1"></a>                        comparisonName <span class="ot">=</span> <span class="fu">paste0</span>( NoneInterTerms[kk],<span class="st">&quot;__&quot;</span>, <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;&quot;</span>,interactionTerm) )</span>
<span id="cb182-2087"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2087" aria-hidden="true" tabindex="-1"></a>                        </span>
<span id="cb182-2088"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2088" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>( cFactor <span class="sc">==</span> splited[<span class="dv">1</span>] )</span>
<span id="cb182-2089"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2089" aria-hidden="true" tabindex="-1"></a>                            otherLevel <span class="ot">=</span> splited[<span class="dv">4</span>] <span class="cf">else</span> otherLevel <span class="ot">=</span> splited[<span class="dv">2</span>]</span>
<span id="cb182-2090"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2090" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb182-2091"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2091" aria-hidden="true" tabindex="-1"></a>                        comparisonName <span class="ot">=</span> <span class="fu">paste0</span>(<span class="co">#names(factorsCoded)[which(factorsCoded==cFactor)], # real factor name</span></span>
<span id="cb182-2092"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2092" aria-hidden="true" tabindex="-1"></a>                                                <span class="fu">gsub</span>(<span class="st">&quot;_vs_&quot;</span>,<span class="st">&quot;-&quot;</span>, <span class="fu">substr</span>(NoneInterTerms[kk], <span class="dv">3</span>, <span class="fu">nchar</span>(NoneInterTerms[kk]  )  )), <span class="co"># the comparison</span></span>
<span id="cb182-2093"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2093" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">&quot;_for_&quot;</span>,otherLevel)</span>
<span id="cb182-2094"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2094" aria-hidden="true" tabindex="-1"></a>                        comparisons2 <span class="ot">=</span> <span class="fu">c</span>(comparisons2, comparisonName)</span>
<span id="cb182-2095"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2095" aria-hidden="true" tabindex="-1"></a>                        selected<span class="sc">$</span>calls <span class="ot">=</span><span class="dv">0</span>   </span>
<span id="cb182-2096"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2096" aria-hidden="true" tabindex="-1"></a>                        selected<span class="sc">$</span>calls [<span class="fu">which</span>( selected<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="fu">log2</span>(minFC_limma) <span class="sc">&amp;</span> selected<span class="sc">$</span>padj <span class="sc">&lt;</span> maxP_limma ) ]  <span class="ot">&lt;-</span>  <span class="dv">1</span></span>
<span id="cb182-2097"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2097" aria-hidden="true" tabindex="-1"></a>                        selected<span class="sc">$</span>calls [ <span class="fu">which</span>( selected<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span>  <span class="sc">-</span><span class="fu">log2</span>(minFC_limma) <span class="sc">&amp;</span> selected<span class="sc">$</span>padj <span class="sc">&lt;</span> maxP_limma ) ] <span class="ot">&lt;-</span>  <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb182-2098"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2098" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">colnames</span>(selected)<span class="ot">=</span> <span class="fu">paste</span>( comparisonName, <span class="st">&quot;___&quot;</span>,<span class="fu">colnames</span>(selected),<span class="at">sep=</span><span class="st">&quot;&quot;</span> )</span>
<span id="cb182-2099"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2099" aria-hidden="true" tabindex="-1"></a>                        selected <span class="ot">=</span> <span class="fu">as.data.frame</span>(selected)</span>
<span id="cb182-2100"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2100" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span> (pp<span class="sc">==</span><span class="dv">0</span>){  <span class="co"># if first one with significant genes, collect gene list and Pval+ fold</span></span>
<span id="cb182-2101"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2101" aria-hidden="true" tabindex="-1"></a>                            result1 <span class="ot">=</span> selected; pp <span class="ot">=</span> <span class="dv">1</span>; </span>
<span id="cb182-2102"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2102" aria-hidden="true" tabindex="-1"></a>                            <span class="co"># selected[,2] &lt;- -1 * selected[,2] # reverse fold change direction</span></span>
<span id="cb182-2103"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2103" aria-hidden="true" tabindex="-1"></a>                            topGenes[[<span class="dv">1</span>]] <span class="ot">=</span> selected[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">6</span>)]; </span>
<span id="cb182-2104"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2104" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">names</span>(topGenes)[<span class="dv">1</span>] <span class="ot">=</span> comparisonName; } <span class="cf">else</span> </span>
<span id="cb182-2105"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2105" aria-hidden="true" tabindex="-1"></a>                            { result1 <span class="ot">=</span> <span class="fu">merge</span>(result1,selected,<span class="at">by=</span><span class="st">&quot;row.names&quot;</span>); </span>
<span id="cb182-2106"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2106" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">rownames</span>(result1) <span class="ot">=</span> result1[,<span class="dv">1</span>]; </span>
<span id="cb182-2107"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2107" aria-hidden="true" tabindex="-1"></a>                                result1 <span class="ot">&lt;-</span> result1[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb182-2108"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2108" aria-hidden="true" tabindex="-1"></a>                                pk<span class="ot">=</span> pk<span class="sc">+</span><span class="dv">1</span>; </span>
<span id="cb182-2109"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2109" aria-hidden="true" tabindex="-1"></a>                                <span class="co"># selected[,2] &lt;- -1 * selected[,2] # reverse fold change direction</span></span>
<span id="cb182-2110"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2110" aria-hidden="true" tabindex="-1"></a>                                topGenes[[pk]] <span class="ot">=</span> selected[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">6</span>)]; </span>
<span id="cb182-2111"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2111" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">names</span>(topGenes)[pk] <span class="ot">=</span> comparisonName;  <span class="co"># assign name to comprison</span></span>
<span id="cb182-2112"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2112" aria-hidden="true" tabindex="-1"></a>                            }</span>
<span id="cb182-2113"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2113" aria-hidden="true" tabindex="-1"></a>                    } <span class="co">#for  </span></span>
<span id="cb182-2114"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2114" aria-hidden="true" tabindex="-1"></a>                        </span>
<span id="cb182-2115"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2115" aria-hidden="true" tabindex="-1"></a>            } <span class="co">#if</span></span>
<span id="cb182-2116"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2116" aria-hidden="true" tabindex="-1"></a>        } <span class="co">#for</span></span>
<span id="cb182-2117"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2117" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-2118"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2118" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-2119"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2119" aria-hidden="true" tabindex="-1"></a>    } <span class="co">#if</span></span>
<span id="cb182-2120"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2121"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2122"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2123"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2124"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2124" aria-hidden="true" tabindex="-1"></a><span class="co">#---</span></span>
<span id="cb182-2125"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2125" aria-hidden="true" tabindex="-1"></a>    <span class="co">#if( length(comparisons) == 1) topGenes &lt;- topGenes[[1]] # if only one comparison, topGenes is not a list, just a data frame itself.</span></span>
<span id="cb182-2126"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2126" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span> <span class="fu">is.null</span>(result1)) { </span>
<span id="cb182-2127"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2127" aria-hidden="true" tabindex="-1"></a>    <span class="co"># note that when you only select 1 column from a data frame it automatically converts to a vector. drop =FALSE prevents that.</span></span>
<span id="cb182-2128"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2128" aria-hidden="true" tabindex="-1"></a>    allCalls <span class="ot">=</span> <span class="fu">as.matrix</span>( result1[,<span class="fu">grep</span>(<span class="st">&quot;calls&quot;</span>,<span class="fu">colnames</span>(result1)), <span class="at">drop =</span> <span class="cn">FALSE</span>  ] )</span>
<span id="cb182-2129"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2129" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(allCalls)<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;___.*&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(allCalls))</span>
<span id="cb182-2130"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2130" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(allCalls)<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;-&quot;</span>, <span class="fu">colnames</span>(allCalls)) <span class="co"># note that samples names should have no &quot;.&quot;</span></span>
<span id="cb182-2131"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2131" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(allCalls)<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;^I-&quot;</span>,<span class="st">&quot;I:&quot;</span>, <span class="fu">colnames</span>(allCalls))</span>
<span id="cb182-2132"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2132" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-2133"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2133" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">results=</span> allCalls, <span class="at">comparisons =</span> comparisons2, <span class="at">Exp.type=</span>Exp.type, <span class="at">topGenes=</span>topGenes)) </span>
<span id="cb182-2134"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2134" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-2135"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2136"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2136" aria-hidden="true" tabindex="-1"></a><span class="co"># main function</span></span>
<span id="cb182-2137"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2137" aria-hidden="true" tabindex="-1"></a>limma <span class="ot">&lt;-</span> <span class="cf">function</span>() {  </span>
<span id="cb182-2138"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2138" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_dataFileFormat <span class="sc">==</span> <span class="dv">1</span> ) {  <span class="co"># if count data</span></span>
<span id="cb182-2139"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2139" aria-hidden="true" tabindex="-1"></a>         <span class="cf">if</span>(input_CountsDEGMethod <span class="sc">==</span> <span class="dv">3</span> ) {    <span class="co"># if DESeq2 method</span></span>
<span id="cb182-2140"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2140" aria-hidden="true" tabindex="-1"></a>                <span class="co"># rawCounts = read.csv(&quot;exampleData/airway_GSE52778.csv&quot;, row.names=1)</span></span>
<span id="cb182-2141"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2141" aria-hidden="true" tabindex="-1"></a>                <span class="co"># res =DEG.DESeq2(rawCounts, .05, 2) </span></span>
<span id="cb182-2142"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2142" aria-hidden="true" tabindex="-1"></a>                <span class="co"># res1 =DEG.limma(rawCounts, .1, 1.5,rawCounts, 2,3) </span></span>
<span id="cb182-2143"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2143" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb182-2144"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2144" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>(   <span class="fu">DEG.DESeq2</span>( convertedCounts.out,input_limmaPval, input_limmaFC,</span>
<span id="cb182-2145"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2145" aria-hidden="true" tabindex="-1"></a>                                    input_selectModelComprions, readSampleInfo.out,</span>
<span id="cb182-2146"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2146" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">c</span>(input_selectFactorsModel,input_selectInteractions), </span>
<span id="cb182-2147"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2147" aria-hidden="true" tabindex="-1"></a>                                    input_selectBlockFactorsModel, factorReferenceLevels.out )  )</span>
<span id="cb182-2148"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2148" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-2149"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2149" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(input_CountsDEGMethod <span class="sc">&lt;</span> <span class="dv">3</span> )    <span class="co"># voom or limma-trend</span></span>
<span id="cb182-2150"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2150" aria-hidden="true" tabindex="-1"></a>                <span class="fu">return</span>( <span class="fu">DEG.limma</span>(convertedData.out, input_limmaPval, input_limmaFC,</span>
<span id="cb182-2151"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2151" aria-hidden="true" tabindex="-1"></a>                                    convertedCounts.out, input_CountsDEGMethod,</span>
<span id="cb182-2152"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2152" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">priorCounts=</span>input_countsLogStart,input_dataFileFormat,</span>
<span id="cb182-2153"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2153" aria-hidden="true" tabindex="-1"></a>                                    input_selectModelComprions, readSampleInfo.out,</span>
<span id="cb182-2154"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2154" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">c</span>(input_selectFactorsModel,input_selectInteractions),</span>
<span id="cb182-2155"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2155" aria-hidden="true" tabindex="-1"></a>                                    input_selectBlockFactorsModel) )</span>
<span id="cb182-2156"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2156" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span> (input_dataFileFormat <span class="sc">==</span> <span class="dv">2</span> ){ <span class="co"># normalized data</span></span>
<span id="cb182-2157"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2157" aria-hidden="true" tabindex="-1"></a>     <span class="fu">return</span>( <span class="fu">DEG.limma</span>(convertedData.out, input_limmaPval, input_limmaFC,</span>
<span id="cb182-2158"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2158" aria-hidden="true" tabindex="-1"></a>                        convertedCounts.out, input_CountsDEGMethod,</span>
<span id="cb182-2159"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2159" aria-hidden="true" tabindex="-1"></a>                        <span class="at">priorCounts=</span>input_countsLogStart,input_dataFileFormat,</span>
<span id="cb182-2160"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2160" aria-hidden="true" tabindex="-1"></a>                        input_selectModelComprions, readSampleInfo.out,</span>
<span id="cb182-2161"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2161" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">c</span>(input_selectFactorsModel,input_selectInteractions),</span>
<span id="cb182-2162"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2162" aria-hidden="true" tabindex="-1"></a>                        input_selectBlockFactorsModel) )</span>
<span id="cb182-2163"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2163" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {   <span class="co"># dataFileFormat == 3 user just uploaded fold change matrix</span></span>
<span id="cb182-2164"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2164" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-2165"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2165" aria-hidden="true" tabindex="-1"></a>        x <span class="ot">=</span> convertedData.out</span>
<span id="cb182-2166"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2166" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-2167"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2167" aria-hidden="true" tabindex="-1"></a>        pvals <span class="ot">=</span> convertedPvals.out</span>
<span id="cb182-2168"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2168" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(pvals) ) {</span>
<span id="cb182-2169"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2169" aria-hidden="true" tabindex="-1"></a>          ix <span class="ot">=</span> <span class="fu">match</span>(<span class="fu">rownames</span>(x), <span class="fu">rownames</span>(pvals))</span>
<span id="cb182-2170"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2170" aria-hidden="true" tabindex="-1"></a>          pvals <span class="ot">=</span> pvals[ix,]</span>
<span id="cb182-2171"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2171" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-2172"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2173"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2174"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2174" aria-hidden="true" tabindex="-1"></a>        <span class="co"># looks like ratio data, take log2</span></span>
<span id="cb182-2175"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2175" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">sum</span>(<span class="fu">round</span>(<span class="fu">apply</span>(x,<span class="dv">2</span>, median) <span class="sc">+</span> .<span class="dv">2</span>) <span class="sc">==</span> <span class="dv">1</span> ) <span class="sc">==</span> <span class="fu">dim</span>(x)[<span class="dv">2</span>] <span class="sc">&amp;</span> <span class="fu">min</span>(x) <span class="sc">&gt;</span> <span class="dv">0</span>) </span>
<span id="cb182-2176"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2176" aria-hidden="true" tabindex="-1"></a>            x <span class="ot">=</span> <span class="fu">log2</span>(x)</span>
<span id="cb182-2177"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2177" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-2178"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2178" aria-hidden="true" tabindex="-1"></a>        Exp.type <span class="ot">=</span> <span class="st">&quot;None standard data without replicates.&quot;</span></span>
<span id="cb182-2179"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2179" aria-hidden="true" tabindex="-1"></a>        all.Calls <span class="ot">=</span> x <span class="co"># fake calls</span></span>
<span id="cb182-2180"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2180" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span> <span class="fu">dim</span>(all.Calls)[<span class="dv">2</span>]) { </span>
<span id="cb182-2181"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2181" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">&lt;-</span> all.Calls[,i]</span>
<span id="cb182-2182"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2182" aria-hidden="true" tabindex="-1"></a>            all.Calls[<span class="fu">which</span>( tem <span class="sc">&lt;=</span> <span class="fu">log2</span>(input_limmaFC) <span class="sc">&amp;</span> tem <span class="sc">&gt;=</span>  <span class="sc">-</span><span class="fu">log2</span>(input_limmaFC) ) ,i] <span class="ot">=</span> <span class="dv">0</span>            </span>
<span id="cb182-2183"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2183" aria-hidden="true" tabindex="-1"></a>            all.Calls[<span class="fu">which</span>( tem <span class="sc">&gt;</span>  <span class="fu">log2</span>(input_limmaFC) ) ,i] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb182-2184"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2184" aria-hidden="true" tabindex="-1"></a>            all.Calls[<span class="fu">which</span>( tem <span class="sc">&lt;</span> <span class="sc">-</span><span class="fu">log2</span>(input_limmaFC) ) ,i] <span class="ot">=</span> <span class="sc">-</span><span class="dv">1</span>      </span>
<span id="cb182-2185"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2185" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(pvals) ) </span>
<span id="cb182-2186"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2186" aria-hidden="true" tabindex="-1"></a>                all.Calls[ <span class="fu">which</span>( pvals[,i] <span class="sc">&gt;</span> input_limmaPval),i] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb182-2187"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2187" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-2188"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2188" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">=</span> <span class="fu">colnames</span>(all.Calls)</span>
<span id="cb182-2189"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2189" aria-hidden="true" tabindex="-1"></a>        extractColumn <span class="ot">&lt;-</span> <span class="cf">function</span> (i) {</span>
<span id="cb182-2190"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2190" aria-hidden="true" tabindex="-1"></a>            topGenes <span class="ot">=</span> <span class="fu">as.data.frame</span>( convertedData.out[,i,<span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb182-2191"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2191" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">is.null</span>(pvals) ) topGenes<span class="sc">$</span>FDR <span class="ot">=</span> <span class="dv">0</span> <span class="cf">else</span> </span>
<span id="cb182-2192"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2192" aria-hidden="true" tabindex="-1"></a>                topGenes<span class="sc">$</span>FDR <span class="ot">=</span> pvals[,i]<span class="co"># fake fdr</span></span>
<span id="cb182-2193"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2193" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-2194"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2194" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(topGenes) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb182-2195"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2195" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>(topGenes)    </span>
<span id="cb182-2196"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2196" aria-hidden="true" tabindex="-1"></a>        } </span>
<span id="cb182-2197"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2197" aria-hidden="true" tabindex="-1"></a>        topGenes <span class="ot">=</span> <span class="fu">lapply</span>( <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>( x )[<span class="dv">2</span>], extractColumn )</span>
<span id="cb182-2198"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2198" aria-hidden="true" tabindex="-1"></a>        topGenes <span class="ot">&lt;-</span> <span class="fu">setNames</span>(topGenes, <span class="fu">colnames</span>(x ) )</span>
<span id="cb182-2199"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2199" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-2200"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2200" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">results=</span> all.Calls, <span class="at">comparisons =</span> <span class="fu">colnames</span>(x ), <span class="at">Exp.type=</span>Exp.type, <span class="at">topGenes=</span>topGenes) )</span>
<span id="cb182-2201"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2201" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-2202"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2202" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-2203"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2203" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-2204"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2205"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2205" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-2206"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2206" aria-hidden="true" tabindex="-1"></a>DEG.data <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb182-2207"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2208"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2208" aria-hidden="true" tabindex="-1"></a>          genes <span class="ot">=</span> limma.out<span class="sc">$</span>results</span>
<span id="cb182-2209"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2209" aria-hidden="true" tabindex="-1"></a>          genes <span class="ot">=</span> <span class="fu">as.data.frame</span>( genes[<span class="fu">which</span>( <span class="fu">rowSums</span>( <span class="fu">abs</span> (genes) ) <span class="sc">!=</span> <span class="dv">0</span> ),] )</span>
<span id="cb182-2210"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2210" aria-hidden="true" tabindex="-1"></a>          <span class="fu">colnames</span>(genes) <span class="ot">=</span> <span class="fu">colnames</span>( limma.out<span class="sc">$</span>results )</span>
<span id="cb182-2211"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2211" aria-hidden="true" tabindex="-1"></a>          genes <span class="ot">=</span> <span class="fu">merge</span>(genes,convertedData.out, <span class="at">by=</span><span class="st">&#39;row.names&#39;</span>)</span>
<span id="cb182-2212"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2212" aria-hidden="true" tabindex="-1"></a>          <span class="fu">colnames</span>(genes)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">&quot;1: upregulation, -1: downregulation&quot;</span></span>
<span id="cb182-2213"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2213" aria-hidden="true" tabindex="-1"></a>            <span class="co"># add gene symbol</span></span>
<span id="cb182-2214"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2214" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">match</span>( genes[,<span class="dv">1</span>], allGeneInfo.out[,<span class="dv">1</span>])</span>
<span id="cb182-2215"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2215" aria-hidden="true" tabindex="-1"></a>        genes <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">as.character</span>( allGeneInfo.out<span class="sc">$</span>symbol)[ix],genes) </span>
<span id="cb182-2216"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2216" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(genes)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">&quot;Symbol&quot;</span></span>
<span id="cb182-2217"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2217" aria-hidden="true" tabindex="-1"></a>        genes <span class="ot">&lt;-</span> genes[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">3</span><span class="sc">:</span><span class="fu">dim</span>(genes)[<span class="dv">2</span>]) ]</span>
<span id="cb182-2218"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2218" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>(genes)</span>
<span id="cb182-2219"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2219" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-2220"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2220" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb182-2221"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2222"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2222" aria-hidden="true" tabindex="-1"></a>vennPlot <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb182-2223"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2223" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-2224"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2224" aria-hidden="true" tabindex="-1"></a>            results <span class="ot">=</span> limma.out<span class="sc">$</span>results</span>
<span id="cb182-2225"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2226"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2226" aria-hidden="true" tabindex="-1"></a>            <span class="co"># split by up or down regulation</span></span>
<span id="cb182-2227"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2227" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(input_UpDownRegulated) {     </span>
<span id="cb182-2228"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2228" aria-hidden="true" tabindex="-1"></a>                resultUp <span class="ot">=</span> results; </span>
<span id="cb182-2229"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2229" aria-hidden="true" tabindex="-1"></a>                resultUp[resultUp <span class="sc">&lt;</span> <span class="dv">0</span> ] <span class="ot">&lt;-</span> <span class="dv">0</span>;</span>
<span id="cb182-2230"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2230" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(resultUp) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">&quot;Up_&quot;</span>, <span class="fu">colnames</span>(resultUp))</span>
<span id="cb182-2231"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2231" aria-hidden="true" tabindex="-1"></a>                resultDown <span class="ot">=</span> results; </span>
<span id="cb182-2232"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2232" aria-hidden="true" tabindex="-1"></a>                resultDown[resultDown <span class="sc">&gt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="dv">0</span>;</span>
<span id="cb182-2233"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2233" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(resultDown) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">&quot;Down_&quot;</span>, <span class="fu">colnames</span>(resultDown))                </span>
<span id="cb182-2234"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2234" aria-hidden="true" tabindex="-1"></a>                results <span class="ot">&lt;-</span> <span class="fu">cbind</span>(resultUp, resultDown)</span>
<span id="cb182-2235"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2235" aria-hidden="true" tabindex="-1"></a>            }           </span>
<span id="cb182-2236"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2236" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2237"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2237" aria-hidden="true" tabindex="-1"></a>            ixa <span class="ot">=</span> <span class="fu">c</span>()</span>
<span id="cb182-2238"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2238" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2239"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2239" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2240"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2240" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> (comps <span class="cf">in</span>  input_selectComparisonsVenn) { </span>
<span id="cb182-2241"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2241" aria-hidden="true" tabindex="-1"></a>                 <span class="cf">if</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;^I:|^I-|^Up_I:|^Up_I-|^Down_I:|^Down_I-&quot;</span>, comps) ) {  <span class="co"># if not interaction term</span></span>
<span id="cb182-2242"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2242" aria-hidden="true" tabindex="-1"></a>                    ix <span class="ot">=</span> <span class="fu">match</span>(comps, <span class="fu">colnames</span>(results)) </span>
<span id="cb182-2243"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2243" aria-hidden="true" tabindex="-1"></a>                  } <span class="cf">else</span> {</span>
<span id="cb182-2244"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2244" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#mismatch in comparison names for interaction terms for DESeq2</span></span>
<span id="cb182-2245"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2245" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#I:water_Wet.genetic_Hy      in the selected Contrast</span></span>
<span id="cb182-2246"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2246" aria-hidden="true" tabindex="-1"></a>                        <span class="co">#Diff-water_Wet-genetic_Hy   in column names</span></span>
<span id="cb182-2247"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2247" aria-hidden="true" tabindex="-1"></a>                        tem <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;^I-&quot;</span>,<span class="st">&quot;I:&quot;</span> ,<span class="fu">colnames</span>(results))</span>
<span id="cb182-2248"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2248" aria-hidden="true" tabindex="-1"></a>                        tem <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,tem)</span>
<span id="cb182-2249"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2249" aria-hidden="true" tabindex="-1"></a>                        ix <span class="ot">=</span> <span class="fu">match</span>(comps, tem) </span>
<span id="cb182-2250"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2251"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2251" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>(<span class="fu">is.na</span>(ix) ) <span class="co"># this is for limma package</span></span>
<span id="cb182-2252"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2252" aria-hidden="true" tabindex="-1"></a>                            ix <span class="ot">=</span> <span class="fu">match</span>(comps, <span class="fu">colnames</span>(results))                        </span>
<span id="cb182-2253"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2253" aria-hidden="true" tabindex="-1"></a>                      }</span>
<span id="cb182-2254"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2254" aria-hidden="true" tabindex="-1"></a>                    ixa <span class="ot">=</span> <span class="fu">c</span>(ixa,ix)</span>
<span id="cb182-2255"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2255" aria-hidden="true" tabindex="-1"></a>                  }</span>
<span id="cb182-2256"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2256" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb182-2257"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2257" aria-hidden="true" tabindex="-1"></a>            results <span class="ot">=</span> results[,ixa,drop<span class="ot">=</span><span class="cn">FALSE</span>] <span class="co"># only use selected comparisons</span></span>
<span id="cb182-2258"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2258" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(results)[<span class="dv">2</span>] <span class="sc">&gt;</span><span class="dv">5</span>) results <span class="ot">&lt;-</span> results[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb182-2259"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2259" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(results) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;^I-&quot;</span>,<span class="st">&quot;I:&quot;</span> ,<span class="fu">colnames</span>(results)) </span>
<span id="cb182-2260"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2260" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2261"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2261" aria-hidden="true" tabindex="-1"></a>            <span class="fu">vennDiagram</span>(results,<span class="at">circle.col=</span><span class="fu">rainbow</span>(<span class="dv">5</span>), <span class="at">cex=</span><span class="fu">c</span>(<span class="fl">1.</span>,<span class="dv">1</span>, <span class="fl">0.7</span>) ) <span class="co"># part of limma package</span></span>
<span id="cb182-2262"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2263"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2264"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2264" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb182-2265"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2266"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2267"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2267" aria-hidden="true" tabindex="-1"></a>sigGeneStats <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb182-2268"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2268" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-2269"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2269" aria-hidden="true" tabindex="-1"></a>        results <span class="ot">=</span> limma.out<span class="sc">$</span>results</span>
<span id="cb182-2270"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2271"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2271" aria-hidden="true" tabindex="-1"></a>         <span class="fu">library</span>(reshape2)</span>
<span id="cb182-2272"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2272" aria-hidden="true" tabindex="-1"></a>         Up <span class="ot">=</span>  <span class="fu">apply</span>(results, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">sum</span>(x <span class="sc">==</span> <span class="dv">1</span>) )</span>
<span id="cb182-2273"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2273" aria-hidden="true" tabindex="-1"></a>         Down <span class="ot">=</span> <span class="fu">apply</span>(results, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">sum</span>(x <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>) ) </span>
<span id="cb182-2274"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2274" aria-hidden="true" tabindex="-1"></a>         stats <span class="ot">=</span> <span class="fu">rbind</span>(Up, Down)</span>
<span id="cb182-2275"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2275" aria-hidden="true" tabindex="-1"></a>                 </span>
<span id="cb182-2276"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2276" aria-hidden="true" tabindex="-1"></a>         gg <span class="ot">&lt;-</span> <span class="fu">melt</span>(stats)</span>
<span id="cb182-2277"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2278"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2278" aria-hidden="true" tabindex="-1"></a>         <span class="fu">colnames</span>(gg) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Regulation&quot;</span>,<span class="st">&quot;Comparisons&quot;</span>,<span class="st">&quot;Genes&quot;</span>)</span>
<span id="cb182-2279"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2279" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb182-2280"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2280" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb182-2281"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2281" aria-hidden="true" tabindex="-1"></a>         p<span class="ot">=</span> <span class="fu">ggplot</span>(gg, <span class="fu">aes</span>(<span class="at">x=</span>Comparisons, <span class="at">y=</span>Genes, <span class="at">fill=</span>  Regulation )  )<span class="sc">+</span></span>
<span id="cb182-2282"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2282" aria-hidden="true" tabindex="-1"></a>             <span class="fu">geom_bar</span>(<span class="at">position=</span><span class="st">&quot;dodge&quot;</span>, <span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb182-2283"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2283" aria-hidden="true" tabindex="-1"></a>             <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;top&quot;</span>) <span class="sc">+</span> </span>
<span id="cb182-2284"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2284" aria-hidden="true" tabindex="-1"></a>             <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>)) <span class="sc">+</span></span>
<span id="cb182-2285"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2285" aria-hidden="true" tabindex="-1"></a>             <span class="fu">ylab</span>(<span class="st">&quot;Number of differntially expressed genes&quot;</span>) <span class="sc">+</span></span>
<span id="cb182-2286"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2286" aria-hidden="true" tabindex="-1"></a>             <span class="fu">theme</span>(<span class="at">axis.title.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb182-2287"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2287" aria-hidden="true" tabindex="-1"></a>                <span class="at">axis.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">14</span>)) </span>
<span id="cb182-2288"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2288" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2289"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2289" aria-hidden="true" tabindex="-1"></a>        p</span>
<span id="cb182-2290"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2290" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2291"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2292"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2292" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-2293"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2294"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2295"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2295" aria-hidden="true" tabindex="-1"></a>sigGeneStatsTable <span class="ot">&lt;-</span> <span class="cf">function</span>( ) {</span>
<span id="cb182-2296"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2297"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2297" aria-hidden="true" tabindex="-1"></a>        results <span class="ot">=</span> limma.out<span class="sc">$</span>results</span>
<span id="cb182-2298"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2299"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2299" aria-hidden="true" tabindex="-1"></a>         Up <span class="ot">=</span>  <span class="fu">apply</span>(results, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">sum</span>(x <span class="sc">==</span> <span class="dv">1</span>) )</span>
<span id="cb182-2300"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2300" aria-hidden="true" tabindex="-1"></a>         Down <span class="ot">=</span> <span class="fu">apply</span>(results, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">sum</span>(x <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>) ) </span>
<span id="cb182-2301"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2301" aria-hidden="true" tabindex="-1"></a>         stats <span class="ot">=</span> <span class="fu">rbind</span>(Up, Down)</span>
<span id="cb182-2302"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2302" aria-hidden="true" tabindex="-1"></a>         stats <span class="ot">=</span> <span class="fu">t</span>(stats)</span>
<span id="cb182-2303"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2303" aria-hidden="true" tabindex="-1"></a>         stats<span class="ot">=</span><span class="fu">cbind</span>(<span class="fu">rownames</span>(stats), stats)</span>
<span id="cb182-2304"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2304" aria-hidden="true" tabindex="-1"></a>         <span class="fu">colnames</span>(stats)[<span class="dv">1</span>]<span class="ot">=</span><span class="st">&quot;Comparisons&quot;</span></span>
<span id="cb182-2305"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2306"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2306" aria-hidden="true" tabindex="-1"></a>         <span class="fu">return</span>(<span class="fu">as.data.frame</span>(stats))</span>
<span id="cb182-2307"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2308"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2308" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-2309"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2310"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2311"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2311" aria-hidden="true" tabindex="-1"></a>selectedHeatmap.data <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb182-2312"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2312" aria-hidden="true" tabindex="-1"></a>          genes <span class="ot">&lt;-</span> limma.out<span class="sc">$</span>results</span>
<span id="cb182-2313"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2313" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>( <span class="fu">is.null</span>(genes) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-2314"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2314" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;I:&quot;</span>, input_selectContrast) ) {  <span class="co"># if not interaction term</span></span>
<span id="cb182-2315"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2315" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, <span class="fu">colnames</span>(genes)) </span>
<span id="cb182-2316"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2316" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> {</span>
<span id="cb182-2317"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2317" aria-hidden="true" tabindex="-1"></a>            <span class="co">#mismatch in comparison names for interaction terms for DESeq2</span></span>
<span id="cb182-2318"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2318" aria-hidden="true" tabindex="-1"></a>            <span class="co">#I:water_Wet.genetic_Hy      in the selected Contrast</span></span>
<span id="cb182-2319"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2319" aria-hidden="true" tabindex="-1"></a>            <span class="co">#Diff-water_Wet-genetic_Hy   in column names</span></span>
<span id="cb182-2320"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2320" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;I-&quot;</span>,<span class="st">&quot;I:&quot;</span> ,<span class="fu">colnames</span>(genes))</span>
<span id="cb182-2321"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2321" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,tem)</span>
<span id="cb182-2322"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2322" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, tem) </span>
<span id="cb182-2323"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2323" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2324"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2324" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">is.na</span>(ix) ) <span class="co"># this is for limma package</span></span>
<span id="cb182-2325"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2325" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, <span class="fu">colnames</span>(genes))           </span>
<span id="cb182-2326"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2326" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2327"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2327" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb182-2328"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2328" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb182-2329"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2329" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">is.null</span>(ix)) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-2330"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2330" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">is.na</span>(ix)) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-2331"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2331" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>( <span class="fu">sum</span>(<span class="fu">abs</span>(genes[,ix] )  ) <span class="sc">&lt;=</span> <span class="dv">1</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>) <span class="co"># no significant genes for this comparison</span></span>
<span id="cb182-2332"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2332" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">dim</span>(genes)[<span class="dv">2</span>] <span class="sc">&lt;</span> ix ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-2333"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2333" aria-hidden="true" tabindex="-1"></a>          query <span class="ot">=</span> <span class="fu">rownames</span>(genes)[<span class="fu">which</span>(genes[,ix] <span class="sc">!=</span> <span class="dv">0</span>)]</span>
<span id="cb182-2334"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2334" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">length</span>(query) <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-2335"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2335" aria-hidden="true" tabindex="-1"></a>          iy <span class="ot">=</span> <span class="fu">match</span>(query, <span class="fu">rownames</span>(convertedData.out  ) )</span>
<span id="cb182-2336"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2336" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb182-2337"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2338"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2338" aria-hidden="true" tabindex="-1"></a>         iz <span class="ot">=</span> <span class="fu">findContrastSamples</span>(  input_selectContrast, </span>
<span id="cb182-2339"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2339" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">colnames</span>(convertedData.out),</span>
<span id="cb182-2340"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2340" aria-hidden="true" tabindex="-1"></a>                                    readSampleInfo.out,</span>
<span id="cb182-2341"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2341" aria-hidden="true" tabindex="-1"></a>                                    input_selectFactorsModel,</span>
<span id="cb182-2342"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2342" aria-hidden="true" tabindex="-1"></a>                                    input_selectModelComprions, </span>
<span id="cb182-2343"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2343" aria-hidden="true" tabindex="-1"></a>                                    factorReferenceLevels.out,</span>
<span id="cb182-2344"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2344" aria-hidden="true" tabindex="-1"></a>                                    input_CountsDEGMethod,</span>
<span id="cb182-2345"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2345" aria-hidden="true" tabindex="-1"></a>                                    input_dataFileFormat</span>
<span id="cb182-2346"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2346" aria-hidden="true" tabindex="-1"></a>                                )</span>
<span id="cb182-2347"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2347" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-2348"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2348" aria-hidden="true" tabindex="-1"></a>        <span class="co"># color bar</span></span>
<span id="cb182-2349"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2349" aria-hidden="true" tabindex="-1"></a>         bar <span class="ot">=</span> genes[,ix]</span>
<span id="cb182-2350"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2350" aria-hidden="true" tabindex="-1"></a>         bar <span class="ot">=</span> bar[bar<span class="sc">!=</span><span class="dv">0</span>]</span>
<span id="cb182-2351"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2352"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2352" aria-hidden="true" tabindex="-1"></a>         <span class="co"># retreive related data         </span></span>
<span id="cb182-2353"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2353" aria-hidden="true" tabindex="-1"></a>         genes <span class="ot">=</span> convertedData.out[iy,iz,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb182-2354"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2354" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb182-2355"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2355" aria-hidden="true" tabindex="-1"></a>         genes <span class="ot">=</span> genes[<span class="fu">order</span>(bar),,drop<span class="ot">=</span><span class="cn">FALSE</span>] <span class="co"># needs to be sorted because myheatmap2 does not reorder genes</span></span>
<span id="cb182-2356"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2356" aria-hidden="true" tabindex="-1"></a>         bar <span class="ot">=</span> <span class="fu">sort</span>(bar)</span>
<span id="cb182-2357"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2358"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2359"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2359" aria-hidden="true" tabindex="-1"></a>         <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">genes=</span>genes, <span class="at">bar=</span>bar ))</span>
<span id="cb182-2360"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2361"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2361" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-2362"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2362" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-2363"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2363" aria-hidden="true" tabindex="-1"></a><span class="co"># find sample index for selected comparisons</span></span>
<span id="cb182-2364"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2364" aria-hidden="true" tabindex="-1"></a>findContrastSamples <span class="ot">&lt;-</span> <span class="cf">function</span>(selectContrast, allSampleNames,<span class="at">sampleInfo=</span><span class="cn">NULL</span>, <span class="at">selectFactorsModel=</span><span class="cn">NULL</span>,<span class="at">selectModelComprions =</span><span class="cn">NULL</span> , <span class="at">referenceLevels=</span><span class="cn">NULL</span>, <span class="at">countsDEGMethod=</span><span class="cn">NULL</span>, <span class="at">dataFileFormat=</span><span class="cn">NULL</span> ){</span>
<span id="cb182-2365"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2365" aria-hidden="true" tabindex="-1"></a>    iz<span class="ot">=</span> <span class="fu">match</span>( <span class="fu">detectGroups</span>(allSampleNames), <span class="fu">unlist</span>(<span class="fu">strsplit</span>( selectContrast, <span class="st">&quot;-&quot;</span>))   )</span>
<span id="cb182-2366"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2366" aria-hidden="true" tabindex="-1"></a>    iz <span class="ot">=</span> <span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(iz))       </span>
<span id="cb182-2367"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2367" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> ( <span class="sc">!</span><span class="fu">is.null</span>(sampleInfo) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(selectFactorsModel) <span class="sc">&amp;</span> <span class="fu">length</span>(selectModelComprions)<span class="sc">&gt;</span><span class="dv">0</span> ) {</span>
<span id="cb182-2368"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2369"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2369" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;.*: &quot;</span>,<span class="st">&quot;&quot;</span>,selectModelComprions)   <span class="co"># strings like: &quot;groups: mutant vs. control&quot;</span></span>
<span id="cb182-2370"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2370" aria-hidden="true" tabindex="-1"></a>        comparisons <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; vs</span><span class="sc">\\</span><span class="st">. &quot;</span>,<span class="st">&quot;-&quot;</span>,comparisons)       </span>
<span id="cb182-2371"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2371" aria-hidden="true" tabindex="-1"></a>        factorsVector<span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;:.*&quot;</span>,<span class="st">&quot;&quot;</span>,selectModelComprions) <span class="co"># corresponding factors</span></span>
<span id="cb182-2372"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2373"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2373" aria-hidden="true" tabindex="-1"></a>          <span class="co"># if read counts data and DESeq2</span></span>
<span id="cb182-2374"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2374" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(dataFileFormat<span class="sc">==</span><span class="dv">1</span> <span class="sc">&amp;</span> countsDEGMethod <span class="sc">==</span> <span class="dv">3</span>) { <span class="co"># if DESeq2</span></span>
<span id="cb182-2375"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2375" aria-hidden="true" tabindex="-1"></a>            contrast <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;_for_.*&quot;</span>,<span class="st">&quot;&quot;</span>,selectContrast) <span class="co"># could be &quot;wt-mu&quot;   or &quot;wt-mu_for_conditionB&quot;</span></span>
<span id="cb182-2376"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2376" aria-hidden="true" tabindex="-1"></a>            ik <span class="ot">=</span> <span class="fu">match</span>( contrast, comparisons )   <span class="co"># selected contrast lookes like: &quot;mutant-control&quot;</span></span>
<span id="cb182-2377"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2378"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2378" aria-hidden="true" tabindex="-1"></a>            otherFactorLevel <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;.*_for_&quot;</span>,<span class="st">&quot;&quot;</span>,selectContrast)</span>
<span id="cb182-2379"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2379" aria-hidden="true" tabindex="-1"></a>            <span class="co"># find the corresponding factor for the other factor</span></span>
<span id="cb182-2380"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2380" aria-hidden="true" tabindex="-1"></a>            otherFactor<span class="ot">=</span><span class="st">&quot; &quot;</span></span>
<span id="cb182-2381"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2381" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">nchar</span>( otherFactorLevel ) <span class="sc">&gt;</span><span class="dv">0</span>){</span>
<span id="cb182-2382"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2382" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span>( eachFactor <span class="cf">in</span> <span class="fu">colnames</span>(sampleInfo) )</span>
<span id="cb182-2383"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2383" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span> ( otherFactorLevel <span class="sc">%in%</span>  sampleInfo[,eachFactor ] )</span>
<span id="cb182-2384"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2384" aria-hidden="true" tabindex="-1"></a>                        otherFactor <span class="ot">=</span> eachFactor        </span>
<span id="cb182-2385"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2385" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-2386"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2386" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2387"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2387" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (<span class="fu">is.na</span>(ik)) iz<span class="ot">=</span><span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(allSampleNames))  <span class="cf">else</span> {  <span class="co"># interaction term, use all samples       </span></span>
<span id="cb182-2388"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2388" aria-hidden="true" tabindex="-1"></a>                selectedfactor<span class="ot">=</span> factorsVector[ ik ] <span class="co"># corresponding factors</span></span>
<span id="cb182-2389"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2390"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2390" aria-hidden="true" tabindex="-1"></a>                iz <span class="ot">=</span> <span class="fu">which</span>(sampleInfo[,selectedfactor] <span class="sc">%in%</span>  <span class="fu">unlist</span>(<span class="fu">strsplit</span>( contrast, <span class="st">&quot;-&quot;</span>)) )</span>
<span id="cb182-2391"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2392"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2392" aria-hidden="true" tabindex="-1"></a>                <span class="co">#filter by other factors: reference level</span></span>
<span id="cb182-2393"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2393" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(<span class="sc">!</span> <span class="fu">is.null</span>( referenceLevels) ) {   <span class="co"># c(&quot;genotype:wt&quot;, &quot;treatment:control&quot; )</span></span>
<span id="cb182-2394"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2394" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">for</span> ( refs <span class="cf">in</span> referenceLevels)</span>
<span id="cb182-2395"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2395" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>(<span class="sc">!</span> <span class="fu">is.null</span>( refs) <span class="sc">&amp;</span> <span class="fu">gsub</span>(<span class="st">&quot;:.*&quot;</span>,<span class="st">&quot;&quot;</span>,refs) <span class="sc">!=</span> selectedfactor ) {</span>
<span id="cb182-2396"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2396" aria-hidden="true" tabindex="-1"></a>                            currentFactor <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;:.*&quot;</span>,<span class="st">&quot;&quot;</span>,refs)</span>
<span id="cb182-2397"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2397" aria-hidden="true" tabindex="-1"></a>                            <span class="cf">if</span>(<span class="fu">nchar</span>( otherFactorLevel ) <span class="sc">&gt;</span><span class="dv">0</span> <span class="sc">&amp;</span> currentFactor <span class="sc">==</span> otherFactor ) { <span class="co"># if not reference level</span></span>
<span id="cb182-2398"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2398" aria-hidden="true" tabindex="-1"></a>                                iz <span class="ot">=</span> <span class="fu">intersect</span>( iz, <span class="fu">which</span>(sampleInfo[,currentFactor] <span class="sc">==</span> otherFactorLevel  ) )</span>
<span id="cb182-2399"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2399" aria-hidden="true" tabindex="-1"></a>                            } <span class="cf">else</span></span>
<span id="cb182-2400"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2400" aria-hidden="true" tabindex="-1"></a>                                iz <span class="ot">=</span> <span class="fu">intersect</span>( iz, <span class="fu">which</span>(sampleInfo[,currentFactor] <span class="sc">==</span> <span class="fu">gsub</span>(<span class="st">&quot;.*:&quot;</span>,<span class="st">&quot;&quot;</span>,refs)  ) )</span>
<span id="cb182-2401"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2401" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb182-2402"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2402" aria-hidden="true" tabindex="-1"></a>                        }</span>
<span id="cb182-2403"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2403" aria-hidden="true" tabindex="-1"></a>                }           </span>
<span id="cb182-2404"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2404" aria-hidden="true" tabindex="-1"></a>                iz <span class="ot">=</span> iz[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(iz))]</span>
<span id="cb182-2405"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2405" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-2406"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2406" aria-hidden="true" tabindex="-1"></a>            <span class="co"># switching from limma to DESeq2 causes problem, as reference level is not defined.</span></span>
<span id="cb182-2407"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2408"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2408" aria-hidden="true" tabindex="-1"></a>            } </span>
<span id="cb182-2409"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2409" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2410"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2410" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {  <span class="co"># not DESeq2</span></span>
<span id="cb182-2411"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2411" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2412"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2412" aria-hidden="true" tabindex="-1"></a>                    <span class="co"># given level find corresponding sample ids</span></span>
<span id="cb182-2413"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2413" aria-hidden="true" tabindex="-1"></a>                    findIDsFromLevel <span class="ot">&lt;-</span> <span class="cf">function</span> (aLevel){</span>
<span id="cb182-2414"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2414" aria-hidden="true" tabindex="-1"></a>                        <span class="co"># find factor</span></span>
<span id="cb182-2415"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2415" aria-hidden="true" tabindex="-1"></a>                        currentFactor<span class="ot">=</span><span class="st">&quot;&quot;</span></span>
<span id="cb182-2416"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2416" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">for</span>( eachFactor <span class="cf">in</span> <span class="fu">colnames</span>(sampleInfo) )</span>
<span id="cb182-2417"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2417" aria-hidden="true" tabindex="-1"></a>                            <span class="cf">if</span> ( aLevel <span class="sc">%in%</span>  sampleInfo[,eachFactor ] )</span>
<span id="cb182-2418"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2418" aria-hidden="true" tabindex="-1"></a>                                currentFactor <span class="ot">=</span> eachFactor          </span>
<span id="cb182-2419"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2419" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span>(<span class="fu">nchar</span>(currentFactor) <span class="sc">&gt;</span><span class="dv">0</span> ) </span>
<span id="cb182-2420"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2420" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">return</span>( <span class="fu">which</span>(sampleInfo[,currentFactor ] <span class="sc">%in%</span> aLevel   )  ) <span class="cf">else</span> <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-2421"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2421" aria-hidden="true" tabindex="-1"></a>                    }       </span>
<span id="cb182-2422"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2422" aria-hidden="true" tabindex="-1"></a>                    </span>
<span id="cb182-2423"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2423" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>( <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;.*_.*-.*_.*&quot;</span>,selectContrast )) iz <span class="ot">=</span> c.out</span>
<span id="cb182-2424"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2424" aria-hidden="true" tabindex="-1"></a>                    levels4 <span class="ot">=</span> <span class="fu">unlist</span>( <span class="fu">strsplit</span>( <span class="fu">unlist</span>( <span class="fu">strsplit</span>(selectContrast,<span class="st">&quot;-&quot;</span>) ), <span class="st">&quot;_&quot;</span>) ) <span class="co">#double split!</span></span>
<span id="cb182-2425"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2425" aria-hidden="true" tabindex="-1"></a>                    <span class="cf">if</span>(<span class="fu">length</span>(levels4)<span class="sc">!=</span><span class="dv">4</span>) { </span>
<span id="cb182-2426"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2426" aria-hidden="true" tabindex="-1"></a>                        iz <span class="ot">=</span> c.out </span>
<span id="cb182-2427"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2427" aria-hidden="true" tabindex="-1"></a>                    } <span class="cf">else</span> {</span>
<span id="cb182-2428"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2428" aria-hidden="true" tabindex="-1"></a>                        iz <span class="ot">=</span> <span class="fu">intersect</span>( <span class="fu">findIDsFromLevel</span>(levels4[<span class="dv">1</span>]),  <span class="fu">findIDsFromLevel</span>(levels4[<span class="dv">2</span>])  ) <span class="co"># first sample</span></span>
<span id="cb182-2429"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2429" aria-hidden="true" tabindex="-1"></a>                        iz <span class="ot">=</span> <span class="fu">c</span>(iz, <span class="fu">intersect</span>( <span class="fu">findIDsFromLevel</span>(levels4[<span class="dv">3</span>]),  <span class="fu">findIDsFromLevel</span>(levels4[<span class="dv">4</span>])  )   ) <span class="co"># 2nd sample</span></span>
<span id="cb182-2430"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2430" aria-hidden="true" tabindex="-1"></a>                        }</span>
<span id="cb182-2431"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2431" aria-hidden="true" tabindex="-1"></a>                } <span class="co">#else</span></span>
<span id="cb182-2432"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2432" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2433"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2433" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2434"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2434" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2435"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2435" aria-hidden="true" tabindex="-1"></a>        }   </span>
<span id="cb182-2436"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2436" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-2437"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2437" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span> (<span class="fu">grepl</span>(<span class="st">&quot;I:&quot;</span>,selectContrast)) iz<span class="ot">=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(allSampleNames) <span class="co"># if it is factor design use all samples</span></span>
<span id="cb182-2438"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2438" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>( <span class="fu">is.na</span>(iz)[<span class="dv">1</span>] <span class="sc">|</span> <span class="fu">length</span>(iz)<span class="sc">&lt;=</span><span class="dv">1</span> )    iz<span class="ot">=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(allSampleNames)</span>
<span id="cb182-2439"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2440"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2440" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(iz)</span>
<span id="cb182-2441"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2441" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-2442"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2442" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2443"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2444"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2444" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2445"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2445" aria-hidden="true" tabindex="-1"></a>selectedHeatmap <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb182-2446"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2447"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2447" aria-hidden="true" tabindex="-1"></a>         bar <span class="ot">=</span> selectedHeatmap.data.out<span class="sc">$</span>bar <span class="sc">+</span><span class="dv">2</span>;</span>
<span id="cb182-2448"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2448" aria-hidden="true" tabindex="-1"></a>         bar[bar<span class="sc">==</span><span class="dv">3</span>] <span class="ot">=</span><span class="dv">2</span></span>
<span id="cb182-2449"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2450"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2450" aria-hidden="true" tabindex="-1"></a>         <span class="fu">myheatmap2</span>( selectedHeatmap.data.out<span class="sc">$</span>genes,bar,<span class="dv">200</span>,<span class="at">mycolor=</span>input_heatColors1,<span class="fu">c</span>(<span class="st">&quot;Down&quot;</span>,<span class="st">&quot;Up&quot;</span>) )</span>
<span id="cb182-2451"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2451" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-2452"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2452" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-2453"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2454"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2455"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2456"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2456" aria-hidden="true" tabindex="-1"></a>selectedHeatmap.data <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb182-2457"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2458"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2458" aria-hidden="true" tabindex="-1"></a>          genes <span class="ot">&lt;-</span> limma.out<span class="sc">$</span>results</span>
<span id="cb182-2459"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2459" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>( <span class="fu">is.null</span>(genes) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-2460"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2460" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;I:&quot;</span>, input_selectContrast) ) {  <span class="co"># if not interaction term</span></span>
<span id="cb182-2461"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2461" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, <span class="fu">colnames</span>(genes)) </span>
<span id="cb182-2462"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2462" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> {</span>
<span id="cb182-2463"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2463" aria-hidden="true" tabindex="-1"></a>            <span class="co">#mismatch in comparison names for interaction terms for DESeq2</span></span>
<span id="cb182-2464"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2464" aria-hidden="true" tabindex="-1"></a>            <span class="co">#I:water_Wet.genetic_Hy      in the selected Contrast</span></span>
<span id="cb182-2465"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2465" aria-hidden="true" tabindex="-1"></a>            <span class="co">#Diff-water_Wet-genetic_Hy   in column names</span></span>
<span id="cb182-2466"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2466" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;I-&quot;</span>,<span class="st">&quot;I:&quot;</span> ,<span class="fu">colnames</span>(genes))</span>
<span id="cb182-2467"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2467" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,tem)</span>
<span id="cb182-2468"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2468" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, tem) </span>
<span id="cb182-2469"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2469" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2470"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2470" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">is.na</span>(ix) ) <span class="co"># this is for limma package</span></span>
<span id="cb182-2471"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2471" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, <span class="fu">colnames</span>(genes))           </span>
<span id="cb182-2472"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2472" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2473"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2473" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb182-2474"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2474" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb182-2475"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2475" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">is.null</span>(ix)) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-2476"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2476" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">is.na</span>(ix)) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-2477"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2477" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>( <span class="fu">sum</span>(<span class="fu">abs</span>(genes[,ix] )  ) <span class="sc">&lt;=</span> <span class="dv">1</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>) <span class="co"># no significant genes for this comparison</span></span>
<span id="cb182-2478"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2478" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">dim</span>(genes)[<span class="dv">2</span>] <span class="sc">&lt;</span> ix ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-2479"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2479" aria-hidden="true" tabindex="-1"></a>          query <span class="ot">=</span> <span class="fu">rownames</span>(genes)[<span class="fu">which</span>(genes[,ix] <span class="sc">!=</span> <span class="dv">0</span>)]</span>
<span id="cb182-2480"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2480" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">length</span>(query) <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-2481"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2481" aria-hidden="true" tabindex="-1"></a>          iy <span class="ot">=</span> <span class="fu">match</span>(query, <span class="fu">rownames</span>(convertedData.out  ) )</span>
<span id="cb182-2482"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2482" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb182-2483"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2484"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2484" aria-hidden="true" tabindex="-1"></a>         iz <span class="ot">=</span> <span class="fu">findContrastSamples</span>(  input_selectContrast, </span>
<span id="cb182-2485"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2485" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">colnames</span>(convertedData.out),</span>
<span id="cb182-2486"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2486" aria-hidden="true" tabindex="-1"></a>                                    readSampleInfo.out,</span>
<span id="cb182-2487"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2487" aria-hidden="true" tabindex="-1"></a>                                    input_selectFactorsModel,</span>
<span id="cb182-2488"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2488" aria-hidden="true" tabindex="-1"></a>                                    input_selectModelComprions, </span>
<span id="cb182-2489"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2489" aria-hidden="true" tabindex="-1"></a>                                    factorReferenceLevels.out,</span>
<span id="cb182-2490"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2490" aria-hidden="true" tabindex="-1"></a>                                    input_CountsDEGMethod,</span>
<span id="cb182-2491"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2491" aria-hidden="true" tabindex="-1"></a>                                    input_dataFileFormat</span>
<span id="cb182-2492"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2492" aria-hidden="true" tabindex="-1"></a>                                )</span>
<span id="cb182-2493"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2493" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-2494"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2494" aria-hidden="true" tabindex="-1"></a>        <span class="co"># color bar</span></span>
<span id="cb182-2495"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2495" aria-hidden="true" tabindex="-1"></a>         bar <span class="ot">=</span> genes[,ix]</span>
<span id="cb182-2496"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2496" aria-hidden="true" tabindex="-1"></a>         bar <span class="ot">=</span> bar[bar<span class="sc">!=</span><span class="dv">0</span>]</span>
<span id="cb182-2497"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2498"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2498" aria-hidden="true" tabindex="-1"></a>         <span class="co"># retreive related data         </span></span>
<span id="cb182-2499"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2499" aria-hidden="true" tabindex="-1"></a>         genes <span class="ot">=</span> <span class="fu">convertedData</span>()[iy,iz,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb182-2500"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2500" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb182-2501"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2501" aria-hidden="true" tabindex="-1"></a>         genes <span class="ot">=</span> genes[<span class="fu">order</span>(bar),,drop<span class="ot">=</span><span class="cn">FALSE</span>] <span class="co"># needs to be sorted because myheatmap2 does not reorder genes</span></span>
<span id="cb182-2502"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2502" aria-hidden="true" tabindex="-1"></a>         bar <span class="ot">=</span> <span class="fu">sort</span>(bar)</span>
<span id="cb182-2503"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2504"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2505"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2505" aria-hidden="true" tabindex="-1"></a>         <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">genes=</span>genes, <span class="at">bar=</span>bar ))</span>
<span id="cb182-2506"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2507"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2507" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-2508"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2508" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2509"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2509" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2510"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2510" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2511"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2511" aria-hidden="true" tabindex="-1"></a>DEG.data <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb182-2512"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2512" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2513"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2513" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">=</span> limma.out<span class="sc">$</span>results</span>
<span id="cb182-2514"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2514" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">=</span> <span class="fu">as.data.frame</span>( genes[<span class="fu">which</span>( <span class="fu">rowSums</span>(genes) <span class="sc">!=</span> <span class="dv">0</span> ),] )</span>
<span id="cb182-2515"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2515" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(genes) <span class="ot">=</span> <span class="fu">colnames</span>( limma.out<span class="sc">$</span>results )</span>
<span id="cb182-2516"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2516" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">=</span> <span class="fu">merge</span>(genes,convertedData.out, <span class="at">by=</span><span class="st">&#39;row.names&#39;</span>)</span>
<span id="cb182-2517"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2517" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(genes)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">&quot;1: upregulation, -1: downregulation&quot;</span></span>
<span id="cb182-2518"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2518" aria-hidden="true" tabindex="-1"></a>    <span class="co"># add gene symbol</span></span>
<span id="cb182-2519"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2519" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span> <span class="fu">match</span>( genes[,<span class="dv">1</span>], allGeneInfo.out[,<span class="dv">1</span>])</span>
<span id="cb182-2520"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2520" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">as.character</span>( allGeneInfo.out<span class="sc">$</span>symbol)[ix],genes) </span>
<span id="cb182-2521"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2521" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(genes)[<span class="dv">1</span>] <span class="ot">=</span> <span class="st">&quot;Symbol&quot;</span></span>
<span id="cb182-2522"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2522" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">&lt;-</span> genes[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">3</span><span class="sc">:</span><span class="fu">dim</span>(genes)[<span class="dv">2</span>]) ]</span>
<span id="cb182-2523"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2523" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(genes)</span>
<span id="cb182-2524"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2524" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-2525"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2525" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2526"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2527"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2527" aria-hidden="true" tabindex="-1"></a>AllGeneListsGMT <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb182-2528"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2528" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2529"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2529" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-2530"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2530" aria-hidden="true" tabindex="-1"></a>            results <span class="ot">=</span> limma.out<span class="sc">$</span>results</span>
<span id="cb182-2531"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2532"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2532" aria-hidden="true" tabindex="-1"></a>            results2 <span class="ot">=</span> <span class="fu">cbind</span>(results, results)</span>
<span id="cb182-2533"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2533" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(results2)<span class="ot">=</span> <span class="fu">c</span>( <span class="fu">paste0</span>(<span class="st">&quot;UP_&quot;</span>, <span class="fu">colnames</span>(results)),<span class="fu">paste0</span>(<span class="st">&quot;Down_&quot;</span>, <span class="fu">colnames</span>(results)) )</span>
<span id="cb182-2534"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2534" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(results)[<span class="dv">2</span>] ) {</span>
<span id="cb182-2535"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2535" aria-hidden="true" tabindex="-1"></a>                results2[,i<span class="sc">*</span><span class="dv">2-1</span>] <span class="ot">=</span> results[,i]</span>
<span id="cb182-2536"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2536" aria-hidden="true" tabindex="-1"></a>                results2[ <span class="fu">which</span>(results2[,i<span class="sc">*</span><span class="dv">2-1</span>] <span class="sc">&lt;</span> <span class="dv">0</span> ) , i<span class="sc">*</span><span class="dv">2-1</span>] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb182-2537"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2537" aria-hidden="true" tabindex="-1"></a>                results2[,i<span class="sc">*</span><span class="dv">2</span>] <span class="ot">=</span> results[,i]    </span>
<span id="cb182-2538"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2538" aria-hidden="true" tabindex="-1"></a>                results2[ <span class="fu">which</span>(results2[,i<span class="sc">*</span><span class="dv">2</span>] <span class="sc">&gt;</span> <span class="dv">0</span> ) , i<span class="sc">*</span><span class="dv">2</span>] <span class="ot">=</span> <span class="dv">0</span> </span>
<span id="cb182-2539"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2539" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-2540"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2540" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2541"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2541" aria-hidden="true" tabindex="-1"></a>            geneList1 <span class="ot">&lt;-</span> <span class="cf">function</span> (i) {</span>
<span id="cb182-2542"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2542" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">=</span> <span class="fu">which</span>(results2[,i] <span class="sc">!=</span><span class="dv">0</span> )</span>
<span id="cb182-2543"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2543" aria-hidden="true" tabindex="-1"></a>                <span class="fu">return</span>(<span class="fu">paste0</span>(<span class="fu">colnames</span>(results2)[i],<span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="fu">length</span>(ix),<span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb182-2544"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2544" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">paste</span>(<span class="fu">rownames</span>(results2 )[ix], <span class="at">collapse=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span> ) ) )          </span>
<span id="cb182-2545"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2545" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-2546"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2546" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(results2)[<span class="dv">2</span>],geneList1 )</span>
<span id="cb182-2547"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2547" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-2548"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2548" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>( <span class="fu">paste</span>(tem, <span class="at">collapse=</span><span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>) )</span>
<span id="cb182-2549"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2549" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-2550"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2551"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2551" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2552"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2552" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2553"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2553" aria-hidden="true" tabindex="-1"></a>geneListData <span class="ot">&lt;-</span> <span class="cf">function</span>( ) {</span>
<span id="cb182-2554"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2554" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-2555"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2555" aria-hidden="true" tabindex="-1"></a>        noSig <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No significant genes find!&quot;</span>)</span>
<span id="cb182-2556"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2556" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">is.null</span>(input_selectContrast) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-2557"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2557" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">is.null</span>( limma.out<span class="sc">$</span>comparisons ) ) <span class="fu">return</span>(<span class="cn">NULL</span>) <span class="co"># if no significant genes found</span></span>
<span id="cb182-2558"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2558" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">length</span>(limma.out<span class="sc">$</span>topGenes) <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(noSig)</span>
<span id="cb182-2559"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2559" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb182-2560"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2560" aria-hidden="true" tabindex="-1"></a>        { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb182-2561"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2561" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb182-2562"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2562" aria-hidden="true" tabindex="-1"></a>          top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb182-2563"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2563" aria-hidden="true" tabindex="-1"></a>          ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, <span class="fu">names</span>(top))</span>
<span id="cb182-2564"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2564" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (noSig)</span>
<span id="cb182-2565"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2565" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb182-2566"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2566" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb182-2567"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2567" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (noSig)</span>
<span id="cb182-2568"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2568" aria-hidden="true" tabindex="-1"></a>          <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb182-2569"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2569" aria-hidden="true" tabindex="-1"></a>          <span class="co">#top1 = merge(top1,convertedData(), by=&#39;row.names&#39;)</span></span>
<span id="cb182-2570"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2570" aria-hidden="true" tabindex="-1"></a>          <span class="co">#colnames(top1)[1] = &quot;Genes&quot;</span></span>
<span id="cb182-2571"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2571" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">=</span> top1[<span class="fu">order</span>(<span class="sc">-</span><span class="fu">abs</span>(top1<span class="sc">$</span>Fold)) ,]</span>
<span id="cb182-2572"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2572" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span> ( <span class="fu">length</span>( <span class="fu">which</span>( top1<span class="sc">$</span>FDR <span class="sc">&lt;=</span>  input_limmaPval  <span class="sc">&amp;</span>  <span class="fu">abs</span>(top1<span class="sc">$</span>Fold)  <span class="sc">&gt;=</span> <span class="fu">log2</span>(input_limmaFC) ) ) <span class="sc">==</span> <span class="dv">0</span> )</span>
<span id="cb182-2573"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2573" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>( noSig)</span>
<span id="cb182-2574"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2574" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top1[<span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;=</span>  input_limmaPval ) ,]</span>
<span id="cb182-2575"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2575" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top1[<span class="fu">which</span>(<span class="fu">abs</span>(top1<span class="sc">$</span>Fold)  <span class="sc">&gt;=</span> <span class="fu">log2</span>( input_limmaFC)) ,]</span>
<span id="cb182-2576"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2576" aria-hidden="true" tabindex="-1"></a>          top1<span class="sc">$</span>Top_Genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top1)</span>
<span id="cb182-2577"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2577" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top1[,<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">2</span>)]</span>
<span id="cb182-2578"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2578" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb182-2579"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2579" aria-hidden="true" tabindex="-1"></a>          <span class="co"># if new species</span></span>
<span id="cb182-2580"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2580" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>( input_selectGO2 <span class="sc">==</span> <span class="st">&quot;ID not recognized!&quot;</span> <span class="sc">|</span> input_selectOrg <span class="sc">==</span> <span class="st">&quot;NEW&quot;</span>) <span class="fu">return</span> (top1); </span>
<span id="cb182-2581"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2581" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb182-2582"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2582" aria-hidden="true" tabindex="-1"></a>          <span class="co">#convertedID = convertID(top1[,1],input_selectOrg, &quot;GOBP&quot; );#&quot;gmax_eg_gene&quot;</span></span>
<span id="cb182-2583"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2583" aria-hidden="true" tabindex="-1"></a>          <span class="co"># tem &lt;- geneInfo(convertedID,input_selectOrg) #input_selectOrg ) ;</span></span>
<span id="cb182-2584"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2584" aria-hidden="true" tabindex="-1"></a>         <span class="co">#  tem &lt;- geneInfo(converted(),input_selectOrg)</span></span>
<span id="cb182-2585"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2585" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> <span class="fu">merge</span>(top1, allGeneInfo.out, <span class="at">by.x =</span><span class="st">&quot;Top_Genes&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="at">all.x=</span>T )</span>
<span id="cb182-2586"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2586" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb182-2587"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2587" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span> ( <span class="fu">sum</span>( <span class="fu">is.na</span>(top1<span class="sc">$</span>band)) <span class="sc">==</span> <span class="fu">dim</span>(top1)[<span class="dv">1</span>] ) top1<span class="sc">$</span>chr <span class="ot">=</span> top1<span class="sc">$</span>chromosome_name <span class="cf">else</span></span>
<span id="cb182-2588"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2588" aria-hidden="true" tabindex="-1"></a>            top1<span class="sc">$</span>chr <span class="ot">=</span> <span class="fu">paste</span>( top1<span class="sc">$</span>chromosome_name, top1<span class="sc">$</span>band,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb182-2589"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2589" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb182-2590"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2590" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top1[,<span class="fu">c</span>(<span class="st">&#39;Top_Genes&#39;</span>,<span class="st">&#39;Fold&#39;</span>,<span class="st">&#39;FDR&#39;</span>,<span class="st">&#39;symbol&#39;</span>,<span class="st">&#39;chr&#39;</span>,<span class="st">&#39;gene_biotype&#39;</span>)]</span>
<span id="cb182-2591"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2591" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb182-2592"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2592" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb182-2593"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2593" aria-hidden="true" tabindex="-1"></a>        <span class="co">#  ix = match(top1[,1], tem$ensembl_gene_id)</span></span>
<span id="cb182-2594"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2594" aria-hidden="true" tabindex="-1"></a>         <span class="co"># if( sum(is.na( tem$Symbol[ix]) ) != length(ix) ) </span></span>
<span id="cb182-2595"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2595" aria-hidden="true" tabindex="-1"></a>          <span class="co"># { top1 &lt;- cbind(top1, tem$Symbol[ix]); colnames(top1)[4]= &quot;Symbol&quot; }</span></span>
<span id="cb182-2596"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2596" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">=</span> top1[<span class="fu">order</span>(<span class="sc">-</span><span class="fu">abs</span>(<span class="fu">as.numeric</span>( top1<span class="sc">$</span>Fold))) ,]</span>
<span id="cb182-2597"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2597" aria-hidden="true" tabindex="-1"></a>          top1<span class="sc">$</span>FDR <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;%-3.2e&quot;</span>,top1<span class="sc">$</span>FDR )</span>
<span id="cb182-2598"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2598" aria-hidden="true" tabindex="-1"></a>          <span class="fu">colnames</span>(top1) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Ensembl ID&quot;</span>, <span class="st">&quot;log2 Fold Change&quot;</span>, <span class="st">&quot;Adj.Pval&quot;</span>, <span class="st">&quot;Symbol&quot;</span>,<span class="st">&quot;Chr&quot;</span>,<span class="st">&quot;Type&quot;</span>)</span>
<span id="cb182-2599"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2599" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span> ( <span class="fu">sum</span>( <span class="fu">is.na</span>(top1<span class="sc">$</span>Symbol)) <span class="sc">==</span> <span class="fu">dim</span>(top1)[<span class="dv">1</span>] ) top1 <span class="ot">&lt;-</span> top1[,<span class="sc">-</span><span class="dv">4</span>] </span>
<span id="cb182-2600"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2600" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2601"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2601" aria-hidden="true" tabindex="-1"></a>          <span class="fu">return</span>(top1)</span>
<span id="cb182-2602"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2602" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-2603"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2603" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb182-2604"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2604" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb182-2605"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2605" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2606"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2606" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb182-2607"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2607" aria-hidden="true" tabindex="-1"></a>volcanoPlot <span class="ot">&lt;-</span> <span class="cf">function</span>( ) {</span>
<span id="cb182-2608"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2608" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb182-2609"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2609" aria-hidden="true" tabindex="-1"></a>    { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb182-2610"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2610" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb182-2611"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2611" aria-hidden="true" tabindex="-1"></a>      top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb182-2612"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2612" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, <span class="fu">names</span>(top))</span>
<span id="cb182-2613"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2613" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb182-2614"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2614" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb182-2615"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2615" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb182-2616"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2617"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2617" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb182-2618"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2618" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb182-2619"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2619" aria-hidden="true" tabindex="-1"></a>     top1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(top1) <span class="co"># convert to data frame</span></span>
<span id="cb182-2620"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2620" aria-hidden="true" tabindex="-1"></a>     top1 <span class="ot">&lt;-</span> top1[<span class="fu">which</span>(<span class="sc">!</span>(<span class="fu">is.na</span>(top1<span class="sc">$</span>Fold)<span class="sc">|</span><span class="fu">is.na</span>(top1<span class="sc">$</span>FDR)    )),] <span class="co"># remove NA&#39;s </span></span>
<span id="cb182-2621"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2621" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb182-2622"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2622" aria-hidden="true" tabindex="-1"></a>     <span class="co">#write.csv(top1,&quot;tem.csv&quot;)</span></span>
<span id="cb182-2623"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2623" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown[ <span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;=</span>  input_limmaPval<span class="sc">&amp;</span> top1<span class="sc">$</span>Fold  <span class="sc">&gt;=</span> <span class="fu">log2</span>( input_limmaFC)) ]  <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb182-2624"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2624" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown[ <span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;=</span>  input_limmaPval <span class="sc">&amp;</span> top1<span class="sc">$</span>Fold  <span class="sc">&lt;=</span> <span class="sc">-</span><span class="fu">log2</span>( input_limmaFC)) ]  <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb182-2625"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2625" aria-hidden="true" tabindex="-1"></a>     <span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb182-2626"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2626" aria-hidden="true" tabindex="-1"></a>     <span class="fu">plot</span>(top1<span class="sc">$</span>Fold,<span class="sc">-</span><span class="fu">log10</span>(top1<span class="sc">$</span>FDR),<span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;grey30&quot;</span>, <span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>)[top1<span class="sc">$</span>upOrDown],</span>
<span id="cb182-2627"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2627" aria-hidden="true" tabindex="-1"></a>     <span class="at">pch =</span><span class="dv">16</span>, <span class="at">cex =</span> .<span class="dv">3</span>, <span class="at">xlab=</span> <span class="st">&quot;log2 fold change&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;- log10 (FDR)&quot;</span>,</span>
<span id="cb182-2628"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2628" aria-hidden="true" tabindex="-1"></a>     <span class="at">cex.lab=</span><span class="dv">2</span>, <span class="at">cex.axis=</span><span class="dv">2</span>, <span class="at">cex.main=</span><span class="dv">2</span>, <span class="at">cex.sub=</span><span class="dv">2</span>   )    </span>
<span id="cb182-2629"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2629" aria-hidden="true" tabindex="-1"></a>     <span class="fu">legend</span>(<span class="st">&quot;bottomright&quot;</span>,<span class="fu">c</span>(<span class="st">&quot;Upregulated&quot;</span>,<span class="st">&quot;Downregulated&quot;</span>),<span class="at">fill =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>),<span class="at">cex=</span><span class="fl">1.1</span> )</span>
<span id="cb182-2630"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2630" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-2631"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2631" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-2632"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2632" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2633"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2633" aria-hidden="true" tabindex="-1"></a>scatterPlot <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb182-2634"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2634" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb182-2635"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2635" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb182-2636"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2636" aria-hidden="true" tabindex="-1"></a>    { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb182-2637"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2637" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb182-2638"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2638" aria-hidden="true" tabindex="-1"></a>      top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb182-2639"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2639" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, <span class="fu">names</span>(top))</span>
<span id="cb182-2640"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2640" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb182-2641"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2641" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb182-2642"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2642" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb182-2643"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2643" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb182-2644"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2644" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb182-2645"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2645" aria-hidden="true" tabindex="-1"></a>     top1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(top1) <span class="co"># convert to data frame</span></span>
<span id="cb182-2646"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2646" aria-hidden="true" tabindex="-1"></a>     top1 <span class="ot">&lt;-</span> top1[<span class="fu">which</span>(<span class="sc">!</span>(<span class="fu">is.na</span>(top1<span class="sc">$</span>Fold)<span class="sc">|</span><span class="fu">is.na</span>(top1<span class="sc">$</span>FDR)    )),] <span class="co"># remove NA&#39;s </span></span>
<span id="cb182-2647"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2647" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb182-2648"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2648" aria-hidden="true" tabindex="-1"></a>     <span class="co">#write.csv(top1,&quot;tem.csv&quot;)</span></span>
<span id="cb182-2649"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2649" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown[ <span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;=</span>  input_limmaPval<span class="sc">&amp;</span> top1<span class="sc">$</span>Fold  <span class="sc">&gt;=</span> <span class="fu">log2</span>( input_limmaFC)) ]  <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb182-2650"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2650" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown[ <span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;=</span>  input_limmaPval <span class="sc">&amp;</span> top1<span class="sc">$</span>Fold  <span class="sc">&lt;=</span> <span class="sc">-</span><span class="fu">log2</span>( input_limmaFC)) ]  <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb182-2651"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2651" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2652"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2652" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2653"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2653" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-2654"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2654" aria-hidden="true" tabindex="-1"></a>         iz <span class="ot">=</span> <span class="fu">findContrastSamples</span>(  input_selectContrast, </span>
<span id="cb182-2655"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2655" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">colnames</span>(convertedData.out),</span>
<span id="cb182-2656"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2656" aria-hidden="true" tabindex="-1"></a>                                    readSampleInfo.out,</span>
<span id="cb182-2657"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2657" aria-hidden="true" tabindex="-1"></a>                                    input_selectFactorsModel,</span>
<span id="cb182-2658"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2658" aria-hidden="true" tabindex="-1"></a>                                    input_selectModelComprions, </span>
<span id="cb182-2659"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2659" aria-hidden="true" tabindex="-1"></a>                                    factorReferenceLevels.out,</span>
<span id="cb182-2660"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2660" aria-hidden="true" tabindex="-1"></a>                                    input_CountsDEGMethod,</span>
<span id="cb182-2661"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2661" aria-hidden="true" tabindex="-1"></a>                                    input_dataFileFormat</span>
<span id="cb182-2662"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2662" aria-hidden="true" tabindex="-1"></a>                                )</span>
<span id="cb182-2663"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2663" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-2664"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2664" aria-hidden="true" tabindex="-1"></a>     genes <span class="ot">&lt;-</span> convertedData.out[,iz]</span>
<span id="cb182-2665"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2665" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-2666"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2666" aria-hidden="true" tabindex="-1"></a>     g <span class="ot">=</span> <span class="fu">detectGroups</span>(<span class="fu">colnames</span>(genes))</span>
<span id="cb182-2667"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2667" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-2668"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2668" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>(<span class="fu">length</span>(<span class="fu">unique</span>(g))  <span class="sc">&gt;</span> <span class="dv">2</span>) { <span class="fu">plot.new</span>(); <span class="fu">text</span>(<span class="fl">0.5</span>,<span class="fl">0.5</span>, <span class="st">&quot;Not available.&quot;</span>) } <span class="cf">else</span>{</span>
<span id="cb182-2669"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2669" aria-hidden="true" tabindex="-1"></a>        average1 <span class="ot">&lt;-</span> <span class="fu">apply</span>( genes[, <span class="fu">which</span>( g <span class="sc">==</span> <span class="fu">unique</span>(g)[<span class="dv">1</span>] ) ],<span class="dv">1</span>,mean)</span>
<span id="cb182-2670"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2670" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2671"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2671" aria-hidden="true" tabindex="-1"></a>        average2 <span class="ot">&lt;-</span> <span class="fu">apply</span>(  genes[, <span class="fu">which</span>( g <span class="sc">==</span> <span class="fu">unique</span>(g)[<span class="dv">2</span>] ) ],<span class="dv">1</span>,mean)</span>
<span id="cb182-2672"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2672" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2673"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2673" aria-hidden="true" tabindex="-1"></a>        genes2 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(average1,average2)</span>
<span id="cb182-2674"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2674" aria-hidden="true" tabindex="-1"></a>        <span class="fu">rownames</span>(genes2) <span class="ot">=</span> <span class="fu">rownames</span>(genes)</span>
<span id="cb182-2675"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2675" aria-hidden="true" tabindex="-1"></a>        genes2 <span class="ot">&lt;-</span>  <span class="fu">merge</span>(genes2,top1,<span class="at">by=</span><span class="st">&quot;row.names&quot;</span>)</span>
<span id="cb182-2676"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2676" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2677"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2677" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-2678"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2678" aria-hidden="true" tabindex="-1"></a>        <span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb182-2679"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2679" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot</span>(genes2<span class="sc">$</span>average2,genes2<span class="sc">$</span>average1,<span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;grey45&quot;</span>,<span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>)[genes2<span class="sc">$</span>upOrDown],</span>
<span id="cb182-2680"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2680" aria-hidden="true" tabindex="-1"></a>        <span class="at">pch =</span><span class="dv">16</span>, <span class="at">cex =</span> .<span class="dv">3</span>, <span class="at">xlab=</span> <span class="fu">paste</span>(<span class="st">&quot;Average expression in&quot;</span>, <span class="fu">unique</span>(g)[<span class="dv">2</span>] ), </span>
<span id="cb182-2681"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2681" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab =</span> <span class="fu">paste</span>(<span class="st">&quot;Average expression in&quot;</span>, <span class="fu">unique</span>(g)[<span class="dv">1</span>] ),</span>
<span id="cb182-2682"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2682" aria-hidden="true" tabindex="-1"></a>        <span class="at">cex.lab=</span><span class="dv">2</span>, <span class="at">cex.axis=</span><span class="dv">2</span>, <span class="at">cex.main=</span><span class="dv">2</span>, <span class="at">cex.sub=</span><span class="dv">2</span>    )    </span>
<span id="cb182-2683"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2683" aria-hidden="true" tabindex="-1"></a>        <span class="fu">legend</span>(<span class="st">&quot;bottomright&quot;</span>,<span class="fu">c</span>(<span class="st">&quot;Upregulated&quot;</span>,<span class="st">&quot;Downregulated&quot;</span>),<span class="at">fill =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>),<span class="at">cex=</span><span class="fl">1.3</span> )</span>
<span id="cb182-2684"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2684" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2685"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2685" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2686"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2686" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-2687"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2687" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb182-2688"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2688" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-2689"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2689" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-2690"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2690" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2691"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2691" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2692"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2692" aria-hidden="true" tabindex="-1"></a>MAplot <span class="ot">&lt;-</span> <span class="cf">function</span> ( ) {</span>
<span id="cb182-2693"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2693" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb182-2694"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2694" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb182-2695"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2695" aria-hidden="true" tabindex="-1"></a>    { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb182-2696"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2696" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb182-2697"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2697" aria-hidden="true" tabindex="-1"></a>      top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb182-2698"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2698" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast, <span class="fu">names</span>(top))</span>
<span id="cb182-2699"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2699" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb182-2700"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2700" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb182-2701"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2701" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb182-2702"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2702" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb182-2703"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2703" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb182-2704"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2704" aria-hidden="true" tabindex="-1"></a>     top1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(top1) <span class="co"># convert to data frame</span></span>
<span id="cb182-2705"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2705" aria-hidden="true" tabindex="-1"></a>     top1 <span class="ot">&lt;-</span> top1[<span class="fu">which</span>(<span class="sc">!</span>(<span class="fu">is.na</span>(top1<span class="sc">$</span>Fold)<span class="sc">|</span><span class="fu">is.na</span>(top1<span class="sc">$</span>FDR)    )),] <span class="co"># remove NA&#39;s </span></span>
<span id="cb182-2706"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2706" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb182-2707"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2707" aria-hidden="true" tabindex="-1"></a>     <span class="co">#write.csv(top1,&quot;tem.csv&quot;)</span></span>
<span id="cb182-2708"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2708" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown[ <span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;=</span>  input_limmaPval<span class="sc">&amp;</span> top1<span class="sc">$</span>Fold  <span class="sc">&gt;=</span> <span class="fu">log2</span>( input_limmaFC)) ]  <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb182-2709"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2709" aria-hidden="true" tabindex="-1"></a>     top1<span class="sc">$</span>upOrDown[ <span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;=</span>  input_limmaPval <span class="sc">&amp;</span> top1<span class="sc">$</span>Fold  <span class="sc">&lt;=</span> <span class="sc">-</span><span class="fu">log2</span>( input_limmaFC)) ]  <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb182-2710"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2710" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-2711"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2711" aria-hidden="true" tabindex="-1"></a>         iz <span class="ot">=</span> <span class="fu">findContrastSamples</span>(  input_selectContrast, </span>
<span id="cb182-2712"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2712" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">colnames</span>(convertedData.out),</span>
<span id="cb182-2713"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2713" aria-hidden="true" tabindex="-1"></a>                                    readSampleInfo.out,</span>
<span id="cb182-2714"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2714" aria-hidden="true" tabindex="-1"></a>                                    input_selectFactorsModel,</span>
<span id="cb182-2715"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2715" aria-hidden="true" tabindex="-1"></a>                                    input_selectModelComprions, </span>
<span id="cb182-2716"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2716" aria-hidden="true" tabindex="-1"></a>                                    factorReferenceLevels.out,</span>
<span id="cb182-2717"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2717" aria-hidden="true" tabindex="-1"></a>                                    input_CountsDEGMethod,</span>
<span id="cb182-2718"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2718" aria-hidden="true" tabindex="-1"></a>                                    input_dataFileFormat</span>
<span id="cb182-2719"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2719" aria-hidden="true" tabindex="-1"></a>                                )</span>
<span id="cb182-2720"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2720" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-2721"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2721" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2722"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2722" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2723"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2723" aria-hidden="true" tabindex="-1"></a>        average1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>( <span class="fu">apply</span>( <span class="fu">convertedData</span>()[,iz],<span class="dv">1</span>,mean) )</span>
<span id="cb182-2724"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2724" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(average1) <span class="ot">=</span> <span class="st">&quot;Average&quot;</span></span>
<span id="cb182-2725"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2725" aria-hidden="true" tabindex="-1"></a>        <span class="fu">rownames</span>(average1) <span class="ot">=</span> <span class="fu">rownames</span>(convertedData.out)</span>
<span id="cb182-2726"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2726" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-2727"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2727" aria-hidden="true" tabindex="-1"></a>        genes2 <span class="ot">&lt;-</span>  <span class="fu">merge</span>(average1,top1,<span class="at">by=</span><span class="st">&quot;row.names&quot;</span>)</span>
<span id="cb182-2728"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2728" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2729"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2729" aria-hidden="true" tabindex="-1"></a>        <span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">5</span>,<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb182-2730"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2730" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot</span>(genes2<span class="sc">$</span>Average,genes2<span class="sc">$</span>Fold,<span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;grey45&quot;</span>,<span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>)[genes2<span class="sc">$</span>upOrDown],</span>
<span id="cb182-2731"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2731" aria-hidden="true" tabindex="-1"></a>        <span class="at">pch =</span><span class="dv">16</span>, <span class="at">cex =</span> .<span class="dv">3</span>, <span class="at">xlab=</span> <span class="st">&quot;Average expression&quot;</span>, </span>
<span id="cb182-2732"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2732" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab =</span> <span class="st">&quot;Log2 fold change&quot;</span>,</span>
<span id="cb182-2733"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2733" aria-hidden="true" tabindex="-1"></a>        <span class="at">cex.lab=</span><span class="dv">2</span>, <span class="at">cex.axis=</span><span class="dv">2</span>, <span class="at">cex.main=</span><span class="dv">2</span>, <span class="at">cex.sub=</span><span class="dv">2</span>    )   </span>
<span id="cb182-2734"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2734" aria-hidden="true" tabindex="-1"></a>            <span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>)</span>
<span id="cb182-2735"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2735" aria-hidden="true" tabindex="-1"></a>        <span class="fu">legend</span>(<span class="st">&quot;bottomright&quot;</span>,<span class="fu">c</span>(<span class="st">&quot;Upregulated&quot;</span>,<span class="st">&quot;Downregulated&quot;</span>),<span class="at">fill =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>),<span class="at">cex=</span><span class="fl">1.3</span> )</span>
<span id="cb182-2736"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2736" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb182-2737"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2737" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-2738"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2738" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2739"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2739" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2740"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2740" aria-hidden="true" tabindex="-1"></a>geneListGOTable <span class="ot">&lt;-</span> <span class="cf">function</span>() {     </span>
<span id="cb182-2741"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2741" aria-hidden="true" tabindex="-1"></a>        NoSig<span class="ot">=</span><span class="cn">NULL</span></span>
<span id="cb182-2742"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2742" aria-hidden="true" tabindex="-1"></a>        <span class="co"># using expression data</span></span>
<span id="cb182-2743"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2743" aria-hidden="true" tabindex="-1"></a>        genes <span class="ot">&lt;-</span> selectedHeatmap.data.out<span class="sc">$</span>genes</span>
<span id="cb182-2744"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2744" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>(genes) ) <span class="fu">return</span>(<span class="cn">NULL</span>) </span>
<span id="cb182-2745"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2745" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">dim</span>(genes)[<span class="dv">1</span>] <span class="sc">&lt;=</span> minGenesEnrichment ) <span class="fu">return</span>(NoSig) <span class="co"># if has only few genes</span></span>
<span id="cb182-2746"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2746" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-2747"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2747" aria-hidden="true" tabindex="-1"></a>        fc <span class="ot">=</span> selectedHeatmap.data.out<span class="sc">$</span>bar       </span>
<span id="cb182-2748"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2748" aria-hidden="true" tabindex="-1"></a>        <span class="co"># GO</span></span>
<span id="cb182-2749"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2749" aria-hidden="true" tabindex="-1"></a>        results1 <span class="ot">&lt;-</span> <span class="cn">NULL</span>; result <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb182-2750"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2750" aria-hidden="true" tabindex="-1"></a>        pp <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb182-2751"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2751" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( i <span class="cf">in</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="sc">-</span><span class="dv">1</span>) ) {</span>
<span id="cb182-2752"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2752" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2753"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2753" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2754"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2754" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">length</span>(<span class="fu">which</span>(fc<span class="sc">*</span>i<span class="sc">&lt;</span><span class="dv">0</span>)) <span class="sc">&lt;=</span> minGenesEnrichment) <span class="cf">next</span>; </span>
<span id="cb182-2755"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2755" aria-hidden="true" tabindex="-1"></a>            query <span class="ot">=</span> <span class="fu">rownames</span>(genes)[<span class="fu">which</span>(fc<span class="sc">*</span>i<span class="sc">&lt;</span><span class="dv">0</span>)]</span>
<span id="cb182-2756"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2756" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">length</span>(query) <span class="sc">&lt;=</span> minGenesEnrichment) <span class="cf">next</span>;  </span>
<span id="cb182-2757"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2757" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-2758"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2758" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2759"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2759" aria-hidden="true" tabindex="-1"></a>            result <span class="ot">&lt;-</span> <span class="fu">findOverlapGMT</span>( query, GeneSets.out,<span class="dv">1</span>) </span>
<span id="cb182-2760"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2760" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2761"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2761" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2762"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2762" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">2</span>] <span class="sc">==</span><span class="dv">1</span>) <span class="cf">next</span>;   <span class="co"># result could be NULL</span></span>
<span id="cb182-2763"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2763" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(i <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>) result<span class="sc">$</span>direction <span class="ot">=</span> <span class="st">&quot;Up regulated&quot;</span>  <span class="cf">else</span> result<span class="sc">$</span>direction <span class="ot">=</span> <span class="st">&quot;Down regulated&quot;</span> <span class="co"># changed</span></span>
<span id="cb182-2764"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2764" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (pp<span class="sc">==</span><span class="dv">0</span> ) { results1 <span class="ot">&lt;-</span> result; pp <span class="ot">=</span> <span class="dv">1</span>;} <span class="cf">else</span>  results1 <span class="ot">=</span> <span class="fu">rbind</span>(results1,result)</span>
<span id="cb182-2765"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2765" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-2766"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2766" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2767"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2767" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> ( pp <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (NoSig)</span>
<span id="cb182-2768"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2768" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> ( <span class="fu">is.null</span>( results1) ) <span class="fu">return</span> (NoSig)</span>
<span id="cb182-2769"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2769" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">==</span> <span class="dv">1</span> ) <span class="fu">return</span>(NoSig)  <span class="co"># Returns a data frame: &quot;No significant results found!&quot;</span></span>
<span id="cb182-2770"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2770" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-2771"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2771" aria-hidden="true" tabindex="-1"></a>        results1<span class="ot">=</span> results1[,<span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">5</span>)] <span class="co"># changed</span></span>
<span id="cb182-2772"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2772" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(results1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;List&quot;</span>,<span class="st">&quot;FDR&quot;</span>,<span class="st">&quot;nGenes&quot;</span>,<span class="st">&quot;GO terms or pathways&quot;</span>,<span class="st">&quot;Genes&quot;</span>) <span class="co"># changed</span></span>
<span id="cb182-2773"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2773" aria-hidden="true" tabindex="-1"></a>        minFDR <span class="ot">=</span> <span class="fl">0.01</span></span>
<span id="cb182-2774"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2774" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">min</span>(results1<span class="sc">$</span>FDR) <span class="sc">&gt;</span> minFDR ) results1 <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No signficant enrichment found.&quot;</span>) <span class="cf">else</span></span>
<span id="cb182-2775"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2775" aria-hidden="true" tabindex="-1"></a>        results1 <span class="ot">=</span> results1[<span class="fu">which</span>(results1<span class="sc">$</span>FDR <span class="sc">&lt;</span> minFDR),]</span>
<span id="cb182-2776"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2776" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-2777"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2777" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">!=</span> <span class="dv">5</span>) <span class="fu">return</span>(NoSig)  <span class="co"># changed</span></span>
<span id="cb182-2778"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2778" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(results1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Direction&quot;</span>,<span class="st">&quot;adj.Pval&quot;</span>,<span class="st">&quot;nGenes&quot;</span>,<span class="st">&quot;Pathways&quot;</span>, <span class="st">&quot;Genes&quot;</span>) <span class="co"># changed</span></span>
<span id="cb182-2779"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2779" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-2780"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2780" aria-hidden="true" tabindex="-1"></a>        results1</span>
<span id="cb182-2781"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2781" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-2782"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2782" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb182-2783"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2783" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2784"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2784" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2785"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2785" aria-hidden="true" tabindex="-1"></a>geneListGO <span class="ot">&lt;-</span> <span class="cf">function</span>() {  </span>
<span id="cb182-2786"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2786" aria-hidden="true" tabindex="-1"></a>  results1 <span class="ot">=</span> geneListGOTable.out</span>
<span id="cb182-2787"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2787" aria-hidden="true" tabindex="-1"></a>  tem <span class="ot">=</span> input_removeRedudantSets</span>
<span id="cb182-2788"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2788" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">==</span><span class="dv">1</span>) <span class="fu">return</span>(results1) <span class="cf">else</span> { </span>
<span id="cb182-2789"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2789" aria-hidden="true" tabindex="-1"></a>    results1<span class="sc">$</span>adj.Pval <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;%-2.1e&quot;</span>,<span class="fu">as.numeric</span>(results1<span class="sc">$</span>adj.Pval) )</span>
<span id="cb182-2790"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2790" aria-hidden="true" tabindex="-1"></a>    results1[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(results1[,<span class="dv">1</span>])</span>
<span id="cb182-2791"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2791" aria-hidden="true" tabindex="-1"></a>    results1[ <span class="fu">duplicated</span> (results1[,<span class="dv">1</span>] ),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span>  </span>
<span id="cb182-2792"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2792" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-2793"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2793" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( results1[,<span class="sc">-</span><span class="dv">5</span>] )</span>
<span id="cb182-2794"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2794" aria-hidden="true" tabindex="-1"></a>    }   </span>
<span id="cb182-2795"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2795" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb182-2796"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2796" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2797"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2797" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2798"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2798" aria-hidden="true" tabindex="-1"></a><span class="co"># a program for ploting enrichment results by highlighting the similarities among terms</span></span>
<span id="cb182-2799"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2799" aria-hidden="true" tabindex="-1"></a><span class="co"># must have columns: Direction, adj.Pval   Pathways Genes</span></span>
<span id="cb182-2800"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2800" aria-hidden="true" tabindex="-1"></a><span class="co">#  Direction    adj.Pval    nGenes  Pathways        Genes</span></span>
<span id="cb182-2801"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2801" aria-hidden="true" tabindex="-1"></a><span class="co">#Down regulated 3.58E-59    131 Ribonucleoprotein complex biogenesis    36  Nsun5 Nhp2 Rrp15 </span></span>
<span id="cb182-2802"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2802" aria-hidden="true" tabindex="-1"></a><span class="co">#Down regulated 2.55E-57    135 NcRNA metabolic process 23  Nsun5 Nhp2 Rrp15 Emg1 Ddx56 Rsl1d1 enrichmentPlot &lt;- function( enrichedTerms){</span></span>
<span id="cb182-2803"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2803" aria-hidden="true" tabindex="-1"></a><span class="co"># Up or down regulation is color-coded</span></span>
<span id="cb182-2804"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2804" aria-hidden="true" tabindex="-1"></a><span class="co"># gene set size if represented by the size of marker</span></span>
<span id="cb182-2805"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2805" aria-hidden="true" tabindex="-1"></a>enrichmentPlot <span class="ot">&lt;-</span> <span class="cf">function</span>( enrichedTerms, <span class="at">rightMargin=</span><span class="dv">33</span>) {</span>
<span id="cb182-2806"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2806" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">class</span>(enrichedTerms) <span class="sc">!=</span> <span class="st">&quot;data.frame&quot;</span>) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-2807"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2807" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(dendextend) <span class="co"># customizing tree</span></span>
<span id="cb182-2808"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2808" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb182-2809"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2809" aria-hidden="true" tabindex="-1"></a>  geneLists <span class="ot">=</span> <span class="fu">lapply</span>(enrichedTerms<span class="sc">$</span>Genes, <span class="cf">function</span>(x) <span class="fu">unlist</span>( <span class="fu">strsplit</span>(<span class="fu">as.character</span>(x),<span class="st">&quot; &quot;</span> )   ) )</span>
<span id="cb182-2810"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2810" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(geneLists)<span class="ot">=</span> enrichedTerms<span class="sc">$</span>Pathways</span>
<span id="cb182-2811"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2811" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2812"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2812" aria-hidden="true" tabindex="-1"></a>  <span class="co"># compute overlaps percentage--------------------</span></span>
<span id="cb182-2813"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2813" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2814"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2814" aria-hidden="true" tabindex="-1"></a>  n <span class="ot">=</span> <span class="fu">length</span>(geneLists)</span>
<span id="cb182-2815"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2815" aria-hidden="true" tabindex="-1"></a>  w <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="at">nrow =</span> n, <span class="at">ncol =</span> n)</span>
<span id="cb182-2816"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2816" aria-hidden="true" tabindex="-1"></a><span class="co"># compute overlaps among all gene lists</span></span>
<span id="cb182-2817"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2817" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n) {</span>
<span id="cb182-2818"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2818" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (j <span class="cf">in</span> i<span class="sc">:</span>n) {</span>
<span id="cb182-2819"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2819" aria-hidden="true" tabindex="-1"></a>            u <span class="ot">&lt;-</span> <span class="fu">unlist</span>(geneLists[i])</span>
<span id="cb182-2820"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2820" aria-hidden="true" tabindex="-1"></a>            v <span class="ot">&lt;-</span> <span class="fu">unlist</span>(geneLists[j])</span>
<span id="cb182-2821"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2821" aria-hidden="true" tabindex="-1"></a>            w[i, j] <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">intersect</span>(u, v))<span class="sc">/</span><span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">c</span>(u,v)))</span>
<span id="cb182-2822"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2822" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-2823"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2823" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-2824"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2824" aria-hidden="true" tabindex="-1"></a><span class="co"># the lower half of the matrix filled in based on symmetry</span></span>
<span id="cb182-2825"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2825" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n) </span>
<span id="cb182-2826"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2826" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(i<span class="dv">-1</span>)) </span>
<span id="cb182-2827"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2827" aria-hidden="true" tabindex="-1"></a>            w[i, j] <span class="ot">=</span> w[j,i] </span>
<span id="cb182-2828"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2828" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2829"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2829" aria-hidden="true" tabindex="-1"></a>  Terms <span class="ot">=</span> <span class="fu">paste</span>( <span class="fu">sprintf</span>(<span class="st">&quot;%-1.0e&quot;</span>,<span class="fu">as.numeric</span>(enrichedTerms<span class="sc">$</span>adj.Pval)), </span>
<span id="cb182-2830"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2830" aria-hidden="true" tabindex="-1"></a>                <span class="fu">names</span>(geneLists))</span>
<span id="cb182-2831"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2831" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(w) <span class="ot">=</span> Terms</span>
<span id="cb182-2832"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2832" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(w) <span class="ot">=</span> Terms</span>
<span id="cb182-2833"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2833" aria-hidden="true" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,rightMargin)) <span class="co"># a large margin for showing </span></span>
<span id="cb182-2834"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2834" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2835"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2835" aria-hidden="true" tabindex="-1"></a>  dend <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>w) <span class="sc">%&gt;%</span></span>
<span id="cb182-2836"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2836" aria-hidden="true" tabindex="-1"></a>    <span class="fu">hclust</span> (<span class="at">method=</span><span class="st">&quot;average&quot;</span>) </span>
<span id="cb182-2837"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2837" aria-hidden="true" tabindex="-1"></a>  ix <span class="ot">=</span> dend<span class="sc">$</span>order <span class="co"># permutated order of leaves</span></span>
<span id="cb182-2838"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2838" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2839"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2839" aria-hidden="true" tabindex="-1"></a>  leafType<span class="ot">=</span> <span class="fu">as.factor</span>( <span class="fu">gsub</span>(<span class="st">&quot; .*&quot;</span>,<span class="st">&quot;&quot;</span>, enrichedTerms<span class="sc">$</span>Direction[ix] ) )</span>
<span id="cb182-2840"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2840" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">length</span>(<span class="fu">unique</span>(enrichedTerms<span class="sc">$</span>Direction)  ) <span class="sc">==</span><span class="dv">2</span> )</span>
<span id="cb182-2841"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2841" aria-hidden="true" tabindex="-1"></a>    leafColors <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>,<span class="st">&quot;red&quot;</span>)  <span class="cf">else</span>  <span class="co"># mycolors</span></span>
<span id="cb182-2842"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2842" aria-hidden="true" tabindex="-1"></a>    leafColors <span class="ot">=</span> mycolors </span>
<span id="cb182-2843"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2843" aria-hidden="true" tabindex="-1"></a>  <span class="co">#leafSize = unlist( lapply(geneLists,length) ) # leaf size represent number of genes</span></span>
<span id="cb182-2844"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2844" aria-hidden="true" tabindex="-1"></a>  <span class="co">#leafSize = sqrt( leafSize[ix] )  </span></span>
<span id="cb182-2845"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2845" aria-hidden="true" tabindex="-1"></a>  leafSize <span class="ot">=</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">as.numeric</span>( enrichedTerms<span class="sc">$</span>adj.Pval[ix] ) ) <span class="co"># leaf size represent P values</span></span>
<span id="cb182-2846"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2846" aria-hidden="true" tabindex="-1"></a>  leafSize <span class="ot">=</span> <span class="fl">1.5</span><span class="sc">*</span>leafSize<span class="sc">/</span><span class="fu">max</span>( leafSize ) <span class="sc">+</span> .<span class="dv">2</span></span>
<span id="cb182-2847"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2847" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb182-2848"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2848" aria-hidden="true" tabindex="-1"></a>    dend <span class="sc">%&gt;%</span> </span>
<span id="cb182-2849"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2849" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.dendrogram</span>(<span class="at">hang=</span><span class="sc">-</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb182-2850"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2850" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set</span>(<span class="st">&quot;leaves_pch&quot;</span>, <span class="dv">19</span>) <span class="sc">%&gt;%</span>   <span class="co"># type of marker</span></span>
<span id="cb182-2851"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2851" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set</span>(<span class="st">&quot;leaves_cex&quot;</span>, leafSize) <span class="sc">%&gt;%</span> <span class="co">#Size</span></span>
<span id="cb182-2852"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2852" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set</span>(<span class="st">&quot;leaves_col&quot;</span>, leafColors[leafType]) <span class="sc">%&gt;%</span> <span class="co"># up or down genes</span></span>
<span id="cb182-2853"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2853" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plot</span>(<span class="at">horiz=</span><span class="cn">TRUE</span>)</span>
<span id="cb182-2854"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2854" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-2855"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2855" aria-hidden="true" tabindex="-1"></a>  <span class="co">#legend(&quot;top&quot;,pch=19, col=leafColors[1:2],legend=levels(leafType),bty = &quot;n&quot;,horiz =T  )</span></span>
<span id="cb182-2856"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2856" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add legend using a second layer</span></span>
<span id="cb182-2857"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2857" aria-hidden="true" tabindex="-1"></a>    <span class="fu">par</span>(<span class="at">lend =</span> <span class="dv">1</span>)           <span class="co"># square line ends for the color legend</span></span>
<span id="cb182-2858"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2858" aria-hidden="true" tabindex="-1"></a>    <span class="fu">add_legend</span>(<span class="st">&quot;top&quot;</span>,<span class="at">pch=</span><span class="dv">19</span>, <span class="at">col=</span>leafColors[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],<span class="at">legend=</span><span class="fu">levels</span>(leafType),<span class="at">bty =</span> <span class="st">&quot;n&quot;</span>,<span class="at">horiz =</span>T </span>
<span id="cb182-2859"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2859" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2860"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2860" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb182-2861"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2861" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb182-2862"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2862" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-2863"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2863" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2864"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2864" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2865"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2865" aria-hidden="true" tabindex="-1"></a><span class="co"># numChar=100 maximum number of characters</span></span>
<span id="cb182-2866"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2866" aria-hidden="true" tabindex="-1"></a><span class="co"># n=200  maximum number of nodes</span></span>
<span id="cb182-2867"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2867" aria-hidden="true" tabindex="-1"></a><span class="co"># degree.cutoff = 0    Remove node if less connected</span></span>
<span id="cb182-2868"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2868" aria-hidden="true" tabindex="-1"></a><span class="co">#from PPInfer</span></span>
<span id="cb182-2869"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2869" aria-hidden="true" tabindex="-1"></a>enrich.net2 <span class="ot">&lt;-</span>  <span class="cf">function</span> (x, gene.set, node.id, <span class="at">node.name =</span> node.id, pvalue, </span>
<span id="cb182-2870"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2870" aria-hidden="true" tabindex="-1"></a>    <span class="at">n =</span> <span class="dv">50</span>, <span class="at">numChar =</span> <span class="cn">NULL</span>, <span class="at">pvalue.cutoff =</span> <span class="fl">0.05</span>, <span class="at">edge.cutoff =</span> <span class="fl">0.05</span>, </span>
<span id="cb182-2871"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2871" aria-hidden="true" tabindex="-1"></a>    <span class="at">degree.cutoff =</span> <span class="dv">0</span>, <span class="at">edge.width =</span> <span class="cf">function</span>(x) {</span>
<span id="cb182-2872"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2872" aria-hidden="true" tabindex="-1"></a>        <span class="dv">5</span> <span class="sc">*</span> x<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb182-2873"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2873" aria-hidden="true" tabindex="-1"></a>    }, <span class="at">node.size =</span> <span class="cf">function</span>(x) {</span>
<span id="cb182-2874"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2874" aria-hidden="true" tabindex="-1"></a>        <span class="fl">2.5</span> <span class="sc">*</span> <span class="fu">log10</span>(x)</span>
<span id="cb182-2875"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2875" aria-hidden="true" tabindex="-1"></a>    }, <span class="at">group =</span> <span class="cn">FALSE</span>, <span class="at">group.color =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>,<span class="st">&quot;red&quot;</span> ), <span class="at">group.shape =</span> <span class="fu">c</span>(<span class="st">&quot;circle&quot;</span>, </span>
<span id="cb182-2876"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2876" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;square&quot;</span>), <span class="at">legend.parameter =</span> <span class="fu">list</span>(<span class="st">&quot;topright&quot;</span>), <span class="at">show.legend =</span> <span class="cn">TRUE</span>, <span class="at">plotting=</span><span class="cn">TRUE</span>,</span>
<span id="cb182-2877"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2877" aria-hidden="true" tabindex="-1"></a>    ...) </span>
<span id="cb182-2878"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2878" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb182-2879"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2879" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(igraph)</span>
<span id="cb182-2880"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2880" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-2881"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2881" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(x, group)</span>
<span id="cb182-2882"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2882" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(x)[<span class="fu">length</span>(<span class="fu">colnames</span>(x))] <span class="ot">&lt;-</span> <span class="st">&quot;Group&quot;</span></span>
<span id="cb182-2883"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2883" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> x[<span class="fu">as.numeric</span>( x[, pvalue]) <span class="sc">&lt;</span> pvalue.cutoff, ]</span>
<span id="cb182-2884"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2884" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> x[<span class="fu">order</span>(x[, pvalue]), ]</span>
<span id="cb182-2885"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2885" aria-hidden="true" tabindex="-1"></a>    n <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">nrow</span>(x), n)</span>
<span id="cb182-2886"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2886" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (n <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb182-2887"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2887" aria-hidden="true" tabindex="-1"></a>        <span class="fu">stop</span>(<span class="st">&quot;no enriched term found...&quot;</span>)</span>
<span id="cb182-2888"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2888" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-2889"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2889" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> x[<span class="dv">1</span><span class="sc">:</span>n, ]</span>
<span id="cb182-2890"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2890" aria-hidden="true" tabindex="-1"></a>    index <span class="ot">&lt;-</span> <span class="fu">match</span>(x[, node.id], <span class="fu">names</span>(gene.set))</span>
<span id="cb182-2891"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2891" aria-hidden="true" tabindex="-1"></a>    geneSets <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb182-2892"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2892" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n) {</span>
<span id="cb182-2893"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2893" aria-hidden="true" tabindex="-1"></a>        geneSets[[i]] <span class="ot">&lt;-</span> gene.set[[index[i]]]</span>
<span id="cb182-2894"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2894" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-2895"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2895" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(geneSets) <span class="ot">&lt;-</span> x[, node.name]</span>
<span id="cb182-2896"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2896" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(numChar)) {</span>
<span id="cb182-2897"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2897" aria-hidden="true" tabindex="-1"></a>        numChar <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">nchar</span>(<span class="fu">as.character</span>(x[, node.name])))</span>
<span id="cb182-2898"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2898" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-2899"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2899" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb182-2900"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2900" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">substr</span>(x[, node.name], <span class="dv">1</span>, numChar))) <span class="sc">&lt;</span> </span>
<span id="cb182-2901"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2901" aria-hidden="true" tabindex="-1"></a>            <span class="fu">nrow</span>(x)) {</span>
<span id="cb182-2902"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2902" aria-hidden="true" tabindex="-1"></a>            numChar <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">nchar</span>(<span class="fu">as.character</span>(x[, node.name])))</span>
<span id="cb182-2903"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2903" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Note : numChar is too small.&quot;</span>, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb182-2904"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2904" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-2905"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2905" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-2906"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2906" aria-hidden="true" tabindex="-1"></a>    x[, node.name] <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">substr</span>(x[, node.name], <span class="dv">1</span>, numChar), </span>
<span id="cb182-2907"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2907" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ifelse</span>(<span class="fu">nchar</span>(<span class="fu">as.character</span>(x[, node.name])) <span class="sc">&gt;</span> numChar, </span>
<span id="cb182-2908"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2908" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;...&quot;</span>, <span class="st">&quot;&quot;</span>), <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb182-2909"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2909" aria-hidden="true" tabindex="-1"></a>    w <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="at">nrow =</span> n, <span class="at">ncol =</span> n)</span>
<span id="cb182-2910"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2910" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2911"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2911" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n) {</span>
<span id="cb182-2912"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2912" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (j <span class="cf">in</span> i<span class="sc">:</span>n) {</span>
<span id="cb182-2913"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2913" aria-hidden="true" tabindex="-1"></a>            u <span class="ot">&lt;-</span> <span class="fu">unlist</span>(geneSets[i])</span>
<span id="cb182-2914"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2914" aria-hidden="true" tabindex="-1"></a>            v <span class="ot">&lt;-</span> <span class="fu">unlist</span>(geneSets[j])</span>
<span id="cb182-2915"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2915" aria-hidden="true" tabindex="-1"></a>            w[i, j] <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">intersect</span>(u, v))<span class="sc">/</span><span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">c</span>(u, </span>
<span id="cb182-2916"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2916" aria-hidden="true" tabindex="-1"></a>                v)))</span>
<span id="cb182-2917"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2917" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-2918"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2918" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-2919"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2919" aria-hidden="true" tabindex="-1"></a>    list.edges <span class="ot">&lt;-</span> <span class="fu">stack</span>(<span class="fu">data.frame</span>(w))</span>
<span id="cb182-2920"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2920" aria-hidden="true" tabindex="-1"></a>    list.edges <span class="ot">&lt;-</span> <span class="fu">cbind</span>(list.edges[, <span class="dv">1</span>], <span class="fu">rep</span>(x[, node.name], </span>
<span id="cb182-2921"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2921" aria-hidden="true" tabindex="-1"></a>        n), <span class="fu">rep</span>(x[, node.name], <span class="at">each =</span> n))</span>
<span id="cb182-2922"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2922" aria-hidden="true" tabindex="-1"></a>    list.edges <span class="ot">&lt;-</span> list.edges[list.edges[, <span class="dv">2</span>] <span class="sc">!=</span> list.edges[,<span class="dv">3</span>], ]</span>
<span id="cb182-2923"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2923" aria-hidden="true" tabindex="-1"></a>    list.edges <span class="ot">&lt;-</span> list.edges[<span class="sc">!</span><span class="fu">is.na</span>(list.edges[, <span class="dv">1</span>]), ]</span>
<span id="cb182-2924"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2924" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">graph.data.frame</span>(list.edges[, <span class="sc">-</span><span class="dv">1</span>], <span class="at">directed =</span> <span class="cn">FALSE</span>)</span>
<span id="cb182-2925"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2925" aria-hidden="true" tabindex="-1"></a>    <span class="fu">E</span>(g)<span class="sc">$</span>width <span class="ot">=</span> <span class="fu">edge.width</span>(<span class="fu">as.numeric</span>(list.edges[, <span class="dv">1</span>]))</span>
<span id="cb182-2926"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2926" aria-hidden="true" tabindex="-1"></a>    <span class="fu">V</span>(g)<span class="sc">$</span>size <span class="ot">&lt;-</span> <span class="fu">node.size</span>(<span class="fu">lengths</span>(geneSets))</span>
<span id="cb182-2927"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2927" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">delete.edges</span>(g, <span class="fu">E</span>(g)[<span class="fu">as.numeric</span>(list.edges[, <span class="dv">1</span>]) <span class="sc">&lt;</span> edge.cutoff])</span>
<span id="cb182-2928"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2928" aria-hidden="true" tabindex="-1"></a>    index.deg <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">degree</span>(g) <span class="sc">&gt;=</span> degree.cutoff</span>
<span id="cb182-2929"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2929" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">delete.vertices</span>(g, <span class="fu">V</span>(g)[<span class="sc">!</span>index.deg])</span>
<span id="cb182-2930"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2930" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> x[index.deg, ]</span>
<span id="cb182-2931"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2931" aria-hidden="true" tabindex="-1"></a>    index <span class="ot">&lt;-</span> index[index.deg]</span>
<span id="cb182-2932"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2932" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(<span class="fu">V</span>(g)) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb182-2933"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2933" aria-hidden="true" tabindex="-1"></a>        <span class="fu">stop</span>(<span class="st">&quot;no categories greater than degree.cutoff...&quot;</span>)</span>
<span id="cb182-2934"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2934" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-2935"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2935" aria-hidden="true" tabindex="-1"></a>    n <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">nrow</span>(x), n)</span>
<span id="cb182-2936"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2936" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> x[<span class="dv">1</span><span class="sc">:</span>n, ]</span>
<span id="cb182-2937"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2937" aria-hidden="true" tabindex="-1"></a>    group.level <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">unique</span>(group))</span>
<span id="cb182-2938"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2938" aria-hidden="true" tabindex="-1"></a>    pvalues <span class="ot">&lt;-</span> x[, pvalue]</span>
<span id="cb182-2939"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2939" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(group.level)) {</span>
<span id="cb182-2940"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2940" aria-hidden="true" tabindex="-1"></a>        index <span class="ot">&lt;-</span> x[, <span class="st">&quot;Group&quot;</span>] <span class="sc">==</span> group.level[i]</span>
<span id="cb182-2941"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2941" aria-hidden="true" tabindex="-1"></a>        <span class="fu">V</span>(g)<span class="sc">$</span>shape[index] <span class="ot">&lt;-</span> group.shape[i]</span>
<span id="cb182-2942"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2942" aria-hidden="true" tabindex="-1"></a>        group.pvalues <span class="ot">&lt;-</span> pvalues[index]</span>
<span id="cb182-2943"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2943" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">length</span>(group.pvalues) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb182-2944"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2944" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (<span class="fu">max</span>(group.pvalues) <span class="sc">==</span> <span class="fu">min</span>(group.pvalues)) {</span>
<span id="cb182-2945"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2945" aria-hidden="true" tabindex="-1"></a>                <span class="fu">V</span>(g)<span class="sc">$</span>color[index] <span class="ot">&lt;-</span> <span class="fu">adjustcolor</span>(group.color[i], </span>
<span id="cb182-2946"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2946" aria-hidden="true" tabindex="-1"></a>                  <span class="at">alpha.f =</span> <span class="fl">0.5</span>)</span>
<span id="cb182-2947"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2947" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-2948"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2948" aria-hidden="true" tabindex="-1"></a>            <span class="cf">else</span> {</span>
<span id="cb182-2949"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2949" aria-hidden="true" tabindex="-1"></a>                <span class="fu">V</span>(g)<span class="sc">$</span>color[index] <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="dv">1</span> <span class="sc">-</span> (group.pvalues <span class="sc">-</span> </span>
<span id="cb182-2950"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2950" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">min</span>(group.pvalues))<span class="sc">/</span>(<span class="fu">max</span>(group.pvalues) <span class="sc">-</span> <span class="fu">min</span>(group.pvalues)), </span>
<span id="cb182-2951"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2951" aria-hidden="true" tabindex="-1"></a>                  <span class="cf">function</span>(x) {</span>
<span id="cb182-2952"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2952" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">adjustcolor</span>(group.color[i], <span class="at">alpha.f =</span> x)</span>
<span id="cb182-2953"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2953" aria-hidden="true" tabindex="-1"></a>                  })</span>
<span id="cb182-2954"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2954" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-2955"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2955" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-2956"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2956" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-2957"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2957" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(plotting) { </span>
<span id="cb182-2958"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2958" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot</span>(g,, <span class="at">vertex.label.dist=</span><span class="dv">1</span>, ...)</span>
<span id="cb182-2959"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2959" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (show.legend) {</span>
<span id="cb182-2960"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2960" aria-hidden="true" tabindex="-1"></a>            legend.parameter<span class="sc">$</span>legend <span class="ot">&lt;-</span> group.level</span>
<span id="cb182-2961"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2961" aria-hidden="true" tabindex="-1"></a>            legend.parameter<span class="sc">$</span>text.col <span class="ot">&lt;-</span> group.color</span>
<span id="cb182-2962"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2962" aria-hidden="true" tabindex="-1"></a>            legend.parameter<span class="sc">$</span>bty <span class="ot">&lt;-</span> <span class="st">&quot;n&quot;</span> </span>
<span id="cb182-2963"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2963" aria-hidden="true" tabindex="-1"></a>            <span class="fu">do.call</span>(legend, legend.parameter)</span>
<span id="cb182-2964"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2964" aria-hidden="true" tabindex="-1"></a>        }}</span>
<span id="cb182-2965"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2965" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(g)</span>
<span id="cb182-2966"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2966" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-2967"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2967" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2968"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2968" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2969"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2969" aria-hidden="true" tabindex="-1"></a>enrichmentNetwork <span class="ot">&lt;-</span> <span class="cf">function</span>(enrichedTerms){</span>
<span id="cb182-2970"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2970" aria-hidden="true" tabindex="-1"></a>    geneLists <span class="ot">=</span> <span class="fu">lapply</span>(enrichedTerms<span class="sc">$</span>Genes, <span class="cf">function</span>(x) <span class="fu">unlist</span>( <span class="fu">strsplit</span>(<span class="fu">as.character</span>(x),<span class="st">&quot; &quot;</span> )   ) )</span>
<span id="cb182-2971"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2971" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(geneLists)<span class="ot">=</span> enrichedTerms<span class="sc">$</span>Pathways</span>
<span id="cb182-2972"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2972" aria-hidden="true" tabindex="-1"></a>    enrichedTerms<span class="sc">$</span>Direction <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; .*&quot;</span>,<span class="st">&quot;&quot;</span>,enrichedTerms<span class="sc">$</span>Direction )</span>
<span id="cb182-2973"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2973" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2974"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2974" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">enrich.net2</span>(enrichedTerms, geneLists, <span class="at">node.id =</span> <span class="st">&quot;Pathways&quot;</span>, <span class="at">numChar =</span> <span class="dv">100</span>, </span>
<span id="cb182-2975"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2975" aria-hidden="true" tabindex="-1"></a>       <span class="at">pvalue =</span> <span class="st">&quot;adj.Pval&quot;</span>, <span class="at">edge.cutoff =</span> <span class="fl">0.2</span>, <span class="at">pvalue.cutoff =</span> <span class="dv">1</span>, <span class="at">degree.cutoff =</span> <span class="dv">0</span>,</span>
<span id="cb182-2976"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2976" aria-hidden="true" tabindex="-1"></a>       <span class="at">n =</span> <span class="dv">200</span>, <span class="at">group =</span> enrichedTerms<span class="sc">$</span>Direction, <span class="at">vertex.label.cex =</span> <span class="dv">1</span>, <span class="at">vertex.label.color =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb182-2977"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2977" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2978"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2978" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-2979"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2979" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb182-2980"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2980" aria-hidden="true" tabindex="-1"></a>enrichmentNetworkPlotly <span class="ot">&lt;-</span> <span class="cf">function</span>(enrichedTerms){</span>
<span id="cb182-2981"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2981" aria-hidden="true" tabindex="-1"></a>    geneLists <span class="ot">=</span> <span class="fu">lapply</span>(enrichedTerms<span class="sc">$</span>Genes, <span class="cf">function</span>(x) <span class="fu">unlist</span>( <span class="fu">strsplit</span>(<span class="fu">as.character</span>(x),<span class="st">&quot; &quot;</span> )   ) )</span>
<span id="cb182-2982"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2982" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(geneLists)<span class="ot">=</span> enrichedTerms<span class="sc">$</span>Pathways</span>
<span id="cb182-2983"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2983" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2984"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2984" aria-hidden="true" tabindex="-1"></a>    g <span class="ot">&lt;-</span> <span class="fu">enrich.net2</span>(enrichedTerms, geneLists, <span class="at">node.id =</span> <span class="st">&quot;Pathways&quot;</span>, <span class="at">numChar =</span> <span class="dv">100</span>, </span>
<span id="cb182-2985"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2985" aria-hidden="true" tabindex="-1"></a>       <span class="at">pvalue =</span> <span class="st">&quot;adj.Pval&quot;</span>, <span class="at">edge.cutoff =</span> <span class="fl">0.2</span>, <span class="at">pvalue.cutoff =</span> <span class="dv">1</span>, <span class="at">degree.cutoff =</span> <span class="dv">0</span>,</span>
<span id="cb182-2986"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2986" aria-hidden="true" tabindex="-1"></a>       <span class="at">n =</span> <span class="dv">200</span>, <span class="at">group =</span> enrichedTerms<span class="sc">$</span>Direction, <span class="at">vertex.label.cex =</span> <span class="fl">0.8</span>, <span class="at">vertex.label.color =</span> <span class="st">&quot;black&quot;</span></span>
<span id="cb182-2987"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2987" aria-hidden="true" tabindex="-1"></a>       ,<span class="at">plotting=</span><span class="cn">TRUE</span>)</span>
<span id="cb182-2988"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2988" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-2989"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2989" aria-hidden="true" tabindex="-1"></a>    vs <span class="ot">&lt;-</span> <span class="fu">V</span>(g)</span>
<span id="cb182-2990"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2990" aria-hidden="true" tabindex="-1"></a>    es <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">get.edgelist</span>(g))</span>
<span id="cb182-2991"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2991" aria-hidden="true" tabindex="-1"></a>    Nv <span class="ot">&lt;-</span> <span class="fu">length</span>(vs)</span>
<span id="cb182-2992"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2992" aria-hidden="true" tabindex="-1"></a>    Ne <span class="ot">&lt;-</span> <span class="fu">length</span>(es[<span class="dv">1</span>]<span class="sc">$</span>V1)</span>
<span id="cb182-2993"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2993" aria-hidden="true" tabindex="-1"></a>    <span class="co"># create nodes</span></span>
<span id="cb182-2994"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2994" aria-hidden="true" tabindex="-1"></a>    L <span class="ot">&lt;-</span> <span class="fu">layout.kamada.kawai</span>(g)</span>
<span id="cb182-2995"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2995" aria-hidden="true" tabindex="-1"></a>    Xn <span class="ot">&lt;-</span> L[,<span class="dv">1</span>]</span>
<span id="cb182-2996"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2996" aria-hidden="true" tabindex="-1"></a>    Yn <span class="ot">&lt;-</span> L[,<span class="dv">2</span>]</span>
<span id="cb182-2997"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2997" aria-hidden="true" tabindex="-1"></a>    inc <span class="ot">=</span> (<span class="fu">max</span>(Yn)<span class="sc">-</span><span class="fu">min</span>(Yn))<span class="sc">/</span><span class="dv">50</span>  <span class="co"># for shifting</span></span>
<span id="cb182-2998"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2998" aria-hidden="true" tabindex="-1"></a>    <span class="co">#group &lt;- ifelse(V(g)$shape == &quot;circle&quot;, &quot;GO&quot;, &quot;KEGG&quot;)</span></span>
<span id="cb182-2999"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-2999" aria-hidden="true" tabindex="-1"></a>     group <span class="ot">&lt;-</span> <span class="fu">as.character</span>(enrichedTerms<span class="sc">$</span>Direction)</span>
<span id="cb182-3000"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3000" aria-hidden="true" tabindex="-1"></a>    network <span class="ot">&lt;-</span> <span class="fu">plot_ly</span>(<span class="at">x =</span> <span class="sc">~</span>Xn, <span class="at">y =</span> <span class="sc">~</span>Yn, <span class="at">type =</span> <span class="st">&quot;scatter&quot;</span>, <span class="at">mode =</span> <span class="st">&quot;markers&quot;</span>,</span>
<span id="cb182-3001"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3001" aria-hidden="true" tabindex="-1"></a>       <span class="at">marker =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="fu">V</span>(g)<span class="sc">$</span>color, <span class="at">size =</span> <span class="fu">V</span>(g)<span class="sc">$</span>size<span class="sc">*</span><span class="dv">2</span>,</span>
<span id="cb182-3002"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3002" aria-hidden="true" tabindex="-1"></a>          <span class="at">symbol=</span> <span class="sc">~</span><span class="fu">V</span>(g)<span class="sc">$</span>shape, <span class="at">line =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;gray&quot;</span>, <span class="at">width =</span> <span class="dv">2</span>)),</span>
<span id="cb182-3003"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3003" aria-hidden="true" tabindex="-1"></a>       <span class="at">hoverinfo =</span> <span class="st">&quot;text&quot;</span>, <span class="at">text =</span> <span class="sc">~</span><span class="fu">paste</span>(<span class="st">&quot;&lt;/br&gt;&quot;</span>, group, <span class="st">&quot;&lt;/br&gt;&quot;</span>, <span class="fu">names</span>(vs))) <span class="sc">%&gt;%</span></span>
<span id="cb182-3004"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3004" aria-hidden="true" tabindex="-1"></a>       <span class="fu">add_annotations</span>( <span class="at">x =</span> <span class="sc">~</span>Xn, <span class="at">y =</span> <span class="sc">~</span>Yn<span class="sc">+</span>inc, <span class="at">text =</span> <span class="fu">names</span>(vs), <span class="at">showarrow =</span> <span class="cn">FALSE</span>,</span>
<span id="cb182-3005"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3005" aria-hidden="true" tabindex="-1"></a>          <span class="at">font =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;#030303&quot;</span>, <span class="at">size =</span> <span class="dv">12</span>))</span>
<span id="cb182-3006"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3006" aria-hidden="true" tabindex="-1"></a>    <span class="co"># create edges</span></span>
<span id="cb182-3007"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3007" aria-hidden="true" tabindex="-1"></a>    edge_shapes <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb182-3008"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3008" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>Ne)</span>
<span id="cb182-3009"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3009" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb182-3010"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3010" aria-hidden="true" tabindex="-1"></a>       v0 <span class="ot">&lt;-</span> es[i,]<span class="sc">$</span>V1</span>
<span id="cb182-3011"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3011" aria-hidden="true" tabindex="-1"></a>       v1 <span class="ot">&lt;-</span> es[i,]<span class="sc">$</span>V2</span>
<span id="cb182-3012"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3012" aria-hidden="true" tabindex="-1"></a>       index0 <span class="ot">&lt;-</span> <span class="fu">match</span>(v0, <span class="fu">names</span>(<span class="fu">V</span>(g)))</span>
<span id="cb182-3013"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3013" aria-hidden="true" tabindex="-1"></a>       index1 <span class="ot">&lt;-</span> <span class="fu">match</span>(v1, <span class="fu">names</span>(<span class="fu">V</span>(g)))</span>
<span id="cb182-3014"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3014" aria-hidden="true" tabindex="-1"></a>       edge_shape <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">type =</span> <span class="st">&quot;line&quot;</span>, <span class="at">line =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;gray&quot;</span> ,</span>
<span id="cb182-3015"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3015" aria-hidden="true" tabindex="-1"></a>          <span class="at">width =</span> <span class="fu">E</span>(g)<span class="sc">$</span>width[i]<span class="sc">/</span><span class="dv">2</span>), <span class="at">x0 =</span> Xn[index0], <span class="at">y0 =</span> Yn[index0],</span>
<span id="cb182-3016"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3016" aria-hidden="true" tabindex="-1"></a>          <span class="at">x1 =</span> Xn[index1], <span class="at">y1 =</span> Yn[index1])</span>
<span id="cb182-3017"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3017" aria-hidden="true" tabindex="-1"></a>       edge_shapes[[i]] <span class="ot">&lt;-</span> edge_shape</span>
<span id="cb182-3018"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3018" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-3019"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3019" aria-hidden="true" tabindex="-1"></a>    <span class="co"># create network</span></span>
<span id="cb182-3020"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3020" aria-hidden="true" tabindex="-1"></a>    axis <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">title =</span> <span class="st">&quot;&quot;</span>, <span class="at">showgrid =</span> <span class="cn">FALSE</span>, <span class="at">showticklabels =</span> <span class="cn">FALSE</span>, <span class="at">zeroline =</span> <span class="cn">FALSE</span>)</span>
<span id="cb182-3021"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3021" aria-hidden="true" tabindex="-1"></a>    h <span class="ot">&lt;-</span> <span class="fu">layout</span>(network, <span class="at">title =</span> <span class="st">&quot;Enrichment Network&quot;</span>, <span class="at">shapes =</span> edge_shapes,</span>
<span id="cb182-3022"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3022" aria-hidden="true" tabindex="-1"></a>       <span class="at">xaxis =</span> axis, <span class="at">yaxis =</span> axis)</span>
<span id="cb182-3023"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3023" aria-hidden="true" tabindex="-1"></a>    <span class="fu">config</span>(h, <span class="at">showLink =</span> <span class="cn">TRUE</span>)</span>
<span id="cb182-3024"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3024" aria-hidden="true" tabindex="-1"></a>       </span>
<span id="cb182-3025"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3025" aria-hidden="true" tabindex="-1"></a>       </span>
<span id="cb182-3026"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3026" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3027"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3027" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb182-3028"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3028" aria-hidden="true" tabindex="-1"></a> <span class="co">#Homo sapies --&gt; hsapiens</span></span>
<span id="cb182-3029"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3029" aria-hidden="true" tabindex="-1"></a>shortSpeciesNames <span class="ot">&lt;-</span> <span class="cf">function</span>(tem){</span>
<span id="cb182-3030"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3030" aria-hidden="true" tabindex="-1"></a>     tem2 <span class="ot">=</span> <span class="fu">strsplit</span>(<span class="fu">as.character</span>(tem),<span class="st">&quot; &quot;</span> )       </span>
<span id="cb182-3031"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3031" aria-hidden="true" tabindex="-1"></a>     <span class="fu">return</span>( <span class="fu">tolower</span>( <span class="fu">paste0</span>(<span class="fu">substr</span>(tem2[[<span class="dv">1</span>]][<span class="dv">1</span>],<span class="dv">1</span>,<span class="dv">1</span>), tem2[[<span class="dv">1</span>]][<span class="dv">2</span>]  ) ) )</span>
<span id="cb182-3032"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3032" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3033"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3033" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3034"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3034" aria-hidden="true" tabindex="-1"></a>findTaxonomyID <span class="ot">&lt;-</span> <span class="cf">function</span>( <span class="at">speciesName =</span> <span class="st">&quot;mmusculus&quot;</span>) {</span>
<span id="cb182-3035"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3035" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3036"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3036" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(input_speciesName) ) { <span class="co"># if species name is entered</span></span>
<span id="cb182-3037"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3037" aria-hidden="true" tabindex="-1"></a>       ix <span class="ot">=</span> <span class="fu">match</span>(input_speciesName, STRING10_species<span class="sc">$</span>official_name)</span>
<span id="cb182-3038"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3038" aria-hidden="true" tabindex="-1"></a>       } <span class="cf">else</span> <span class="cf">if</span>( input_selectGO2 <span class="sc">!=</span> <span class="st">&quot;ID not recognized!&quot;</span> )</span>
<span id="cb182-3039"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3039" aria-hidden="true" tabindex="-1"></a>       { <span class="co"># if no species is entered, try to resolve species using existing info     </span></span>
<span id="cb182-3040"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3040" aria-hidden="true" tabindex="-1"></a>            codedNames <span class="ot">=</span> <span class="fu">sapply</span>(STRING10_species<span class="sc">$</span>compact_name,shortSpeciesNames )</span>
<span id="cb182-3041"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3041" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">match</span>(speciesName , codedNames)</span>
<span id="cb182-3042"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3042" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(input_selectOrg <span class="sc">!=</span> speciesChoice[[<span class="dv">1</span>]]) {  <span class="co"># if species is entered</span></span>
<span id="cb182-3043"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3043" aria-hidden="true" tabindex="-1"></a>                selectedSpecies <span class="ot">=</span> <span class="fu">findSpeciesById</span>(input_selectOrg)[<span class="dv">1</span>,<span class="dv">1</span>]</span>
<span id="cb182-3044"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3044" aria-hidden="true" tabindex="-1"></a>                ix <span class="ot">=</span> <span class="fu">match</span>( <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>,<span class="st">&quot;&quot;</span>, selectedSpecies ), codedNames)               </span>
<span id="cb182-3045"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3045" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-3046"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3046" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3047"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3047" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> <span class="fu">return</span>(<span class="cn">NULL</span>) </span>
<span id="cb182-3048"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3048" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb182-3049"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3049" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">is.na</span>(ix) ) <span class="fu">return</span>(<span class="cn">NULL</span>) </span>
<span id="cb182-3050"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3050" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(STRING10_species<span class="sc">$</span>species_id[ix])</span>
<span id="cb182-3051"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3051" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3052"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3052" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3053"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3053" aria-hidden="true" tabindex="-1"></a>STRINGdb_geneList <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb182-3054"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3054" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(STRINGdb,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb182-3055"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3055" aria-hidden="true" tabindex="-1"></a>    taxonomyID <span class="ot">=</span> findTaxonomyID.out</span>
<span id="cb182-3056"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3056" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3057"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3057" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>( taxonomyID ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-3058"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3058" aria-hidden="true" tabindex="-1"></a>        <span class="co">#Intialization</span></span>
<span id="cb182-3059"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3059" aria-hidden="true" tabindex="-1"></a>        string_db <span class="ot">&lt;-</span> STRINGdb<span class="sc">$</span><span class="fu">new</span>( <span class="at">version=</span><span class="st">&quot;10&quot;</span>, <span class="at">species=</span>taxonomyID,</span>
<span id="cb182-3060"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3060" aria-hidden="true" tabindex="-1"></a>                               <span class="at">score_threshold=</span><span class="dv">0</span>, <span class="at">input_directory=</span><span class="st">&quot;&quot;</span> )</span>
<span id="cb182-3061"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3061" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-3062"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3062" aria-hidden="true" tabindex="-1"></a>        <span class="co"># using expression data</span></span>
<span id="cb182-3063"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3063" aria-hidden="true" tabindex="-1"></a>        genes <span class="ot">&lt;-</span> geneListData.out</span>
<span id="cb182-3064"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3064" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(genes)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;lfc&quot;</span>)</span>
<span id="cb182-3065"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3065" aria-hidden="true" tabindex="-1"></a>        mapped <span class="ot">&lt;-</span> string_db<span class="sc">$</span><span class="fu">map</span>(genes,<span class="st">&quot;gene&quot;</span>, <span class="at">removeUnmappedRows =</span> <span class="cn">TRUE</span> )</span>
<span id="cb182-3066"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3066" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3067"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3067" aria-hidden="true" tabindex="-1"></a>        up<span class="ot">=</span> <span class="fu">subset</span>(mapped, lfc<span class="sc">&gt;</span><span class="dv">0</span>, <span class="at">select=</span><span class="st">&quot;STRING_id&quot;</span>, <span class="at">drop=</span><span class="cn">TRUE</span> )</span>
<span id="cb182-3068"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3068" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3069"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3069" aria-hidden="true" tabindex="-1"></a>        down<span class="ot">=</span> <span class="fu">subset</span>(mapped, lfc<span class="sc">&lt;</span><span class="dv">0</span>, <span class="at">select=</span><span class="st">&quot;STRING_id&quot;</span>, <span class="at">drop=</span><span class="cn">TRUE</span> )     </span>
<span id="cb182-3070"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3070" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-3071"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3071" aria-hidden="true" tabindex="-1"></a>        mappingRatio <span class="ot">=</span> <span class="fu">nrow</span>(mapped)<span class="sc">/</span> <span class="fu">nrow</span>(genes)</span>
<span id="cb182-3072"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3072" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">nrow</span>(mapped) <span class="sc">==</span> <span class="dv">0</span>) <span class="fu">return</span>(<span class="cn">NULL</span>) <span class="cf">else</span></span>
<span id="cb182-3073"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3073" aria-hidden="true" tabindex="-1"></a>         <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">up=</span>up, <span class="at">down=</span>down, <span class="at">ratio=</span>mappingRatio, <span class="at">geneTable=</span>mapped ) )</span>
<span id="cb182-3074"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3074" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3075"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3075" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3076"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3076" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3077"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3077" aria-hidden="true" tabindex="-1"></a>STRINGDB_mapping_stat <span class="ot">&lt;-</span> <span class="cf">function</span>( ) {</span>
<span id="cb182-3078"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3078" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3079"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3079" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">is.null</span>(STRINGdb_geneList.out ) ) <span class="fu">return</span>(<span class="st">&quot;No genes mapped by STRINGdb. Please enter or double-check species name above.&quot;</span>)</span>
<span id="cb182-3080"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3080" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="sc">!</span> <span class="fu">is.null</span>(STRINGdb_geneList.out ) ) { </span>
<span id="cb182-3081"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3081" aria-hidden="true" tabindex="-1"></a>            tem<span class="ot">=</span><span class="fu">paste0</span>( <span class="dv">100</span><span class="sc">*</span><span class="fu">round</span>(<span class="fu">STRINGdb_geneList</span>()<span class="sc">$</span>ratio,<span class="dv">3</span>), <span class="st">&quot;% genes mapped by STRING web server.&quot;</span>)</span>
<span id="cb182-3082"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3082" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">STRINGdb_geneList</span>()<span class="sc">$</span>ratio <span class="sc">&lt;</span><span class="fl">0.3</span> ) tem <span class="ot">=</span> <span class="fu">paste</span>(tem, <span class="st">&quot;Warning!!! Very few gene mapped. Double check if the correct species is selected.&quot;</span>)</span>
<span id="cb182-3083"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3083" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>( tem  )</span>
<span id="cb182-3084"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3084" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-3085"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3085" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3086"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3086" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3087"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3087" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3088"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3088" aria-hidden="true" tabindex="-1"></a>stringDB_GO_enrichmentData <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb182-3089"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3089" aria-hidden="true" tabindex="-1"></a>        <span class="fu">library</span>(STRINGdb,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb182-3090"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3090" aria-hidden="true" tabindex="-1"></a>                           </span>
<span id="cb182-3091"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3091" aria-hidden="true" tabindex="-1"></a>        tem <span class="ot">=</span> input_STRINGdbGO</span>
<span id="cb182-3092"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3092" aria-hidden="true" tabindex="-1"></a>        taxonomyID <span class="ot">=</span> findTaxonomyID.out</span>
<span id="cb182-3093"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3093" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>( taxonomyID ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)     </span>
<span id="cb182-3094"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3094" aria-hidden="true" tabindex="-1"></a>        NoSig <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No significant enrichment found.&quot;</span>)</span>
<span id="cb182-3095"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3095" aria-hidden="true" tabindex="-1"></a>        <span class="do">####################################</span></span>
<span id="cb182-3096"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3096" aria-hidden="true" tabindex="-1"></a>        <span class="co">#Intialization</span></span>
<span id="cb182-3097"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3097" aria-hidden="true" tabindex="-1"></a>        string_db <span class="ot">&lt;-</span> STRINGdb<span class="sc">$</span><span class="fu">new</span>( <span class="at">version=</span><span class="st">&quot;10&quot;</span>, <span class="at">species=</span>taxonomyID,</span>
<span id="cb182-3098"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3098" aria-hidden="true" tabindex="-1"></a>                               <span class="at">score_threshold=</span><span class="dv">0</span>, <span class="at">input_directory=</span><span class="st">&quot;&quot;</span> )</span>
<span id="cb182-3099"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3099" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-3100"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3100" aria-hidden="true" tabindex="-1"></a>        <span class="co"># using expression data</span></span>
<span id="cb182-3101"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3101" aria-hidden="true" tabindex="-1"></a>        genes <span class="ot">&lt;-</span> selectedHeatmap.data.out<span class="sc">$</span>genes</span>
<span id="cb182-3102"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3102" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>(genes) ) <span class="fu">return</span>(<span class="cn">NULL</span>) </span>
<span id="cb182-3103"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3103" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">dim</span>(genes)[<span class="dv">1</span>] <span class="sc">&lt;=</span> minGenesEnrichment ) <span class="fu">return</span>(NoSig) <span class="co"># if has only few genes</span></span>
<span id="cb182-3104"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3104" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-3105"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3105" aria-hidden="true" tabindex="-1"></a>        fc <span class="ot">=</span> selectedHeatmap.data.out<span class="sc">$</span>bar       </span>
<span id="cb182-3106"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3106" aria-hidden="true" tabindex="-1"></a>        <span class="co"># GO</span></span>
<span id="cb182-3107"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3107" aria-hidden="true" tabindex="-1"></a>        results1 <span class="ot">&lt;-</span> <span class="cn">NULL</span>; result <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb182-3108"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3108" aria-hidden="true" tabindex="-1"></a>        pp <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb182-3109"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3109" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( i <span class="cf">in</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>) ) {</span>
<span id="cb182-3110"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3110" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-3111"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3111" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">length</span>(<span class="fu">which</span>(fc<span class="sc">*</span>i<span class="sc">&lt;</span><span class="dv">0</span>)) <span class="sc">&lt;=</span> minGenesEnrichment) <span class="cf">next</span>; </span>
<span id="cb182-3112"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3112" aria-hidden="true" tabindex="-1"></a>            query <span class="ot">=</span> <span class="fu">rownames</span>(genes)[<span class="fu">which</span>(fc<span class="sc">*</span>i<span class="sc">&lt;</span><span class="dv">0</span>)]</span>
<span id="cb182-3113"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3113" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-3114"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3114" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">length</span>(query) <span class="sc">&lt;=</span> minGenesEnrichment) <span class="cf">next</span>;  </span>
<span id="cb182-3115"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3115" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-3116"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3116" aria-hidden="true" tabindex="-1"></a>            ids <span class="ot">=</span> STRINGdb_geneList.out[[i]]</span>
<span id="cb182-3117"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3117" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-3118"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3118" aria-hidden="true" tabindex="-1"></a>            result <span class="ot">&lt;-</span> string_db<span class="sc">$</span><span class="fu">get_enrichment</span>( ids, <span class="at">category =</span> input_STRINGdbGO, <span class="at">methodMT =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">iea =</span> <span class="cn">TRUE</span> )</span>
<span id="cb182-3119"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3119" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">nrow</span>(result) <span class="sc">==</span> <span class="dv">0</span> ) <span class="cf">next</span>; </span>
<span id="cb182-3120"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3120" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">nrow</span>(result) <span class="sc">&gt;</span> <span class="dv">15</span>)  result <span class="ot">&lt;-</span> result[<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>,]</span>
<span id="cb182-3121"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3122"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3122" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">2</span>] <span class="sc">==</span><span class="dv">1</span>) <span class="cf">next</span>;   <span class="co"># result could be NULL</span></span>
<span id="cb182-3123"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3123" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(i <span class="sc">==</span> <span class="dv">1</span>) result<span class="sc">$</span>direction <span class="ot">=</span> <span class="st">&quot;Up regulated&quot;</span>  <span class="cf">else</span> result<span class="sc">$</span>direction <span class="ot">=</span> <span class="st">&quot;Down regulated&quot;</span></span>
<span id="cb182-3124"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3124" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (pp<span class="sc">==</span><span class="dv">0</span> ) { results1 <span class="ot">&lt;-</span> result; pp <span class="ot">=</span> <span class="dv">1</span>;} <span class="cf">else</span>  results1 <span class="ot">=</span> <span class="fu">rbind</span>(results1,result)</span>
<span id="cb182-3125"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3125" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-3126"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3127"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3127" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> ( pp <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (NoSig)</span>
<span id="cb182-3128"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3128" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> ( <span class="fu">is.null</span>( results1) ) <span class="fu">return</span> (NoSig)</span>
<span id="cb182-3129"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3129" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">==</span> <span class="dv">1</span> ) <span class="fu">return</span>(NoSig)  <span class="co"># Returns a data frame: &quot;No significant results found!&quot;</span></span>
<span id="cb182-3130"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3130" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-3131"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3131" aria-hidden="true" tabindex="-1"></a>        results1<span class="ot">=</span> results1[,<span class="fu">c</span>(<span class="dv">7</span>,<span class="dv">5</span>,<span class="dv">3</span>, <span class="dv">6</span>)]</span>
<span id="cb182-3132"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3132" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(results1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;List&quot;</span>,<span class="st">&quot;FDR&quot;</span>,<span class="st">&quot;nGenes&quot;</span>,<span class="st">&quot;GO terms or pathways&quot;</span>)</span>
<span id="cb182-3133"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3133" aria-hidden="true" tabindex="-1"></a>        minFDR <span class="ot">=</span> <span class="fl">0.01</span></span>
<span id="cb182-3134"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3134" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">min</span>(results1<span class="sc">$</span>FDR) <span class="sc">&gt;</span> minFDR ) results1 <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No signficant enrichment found.&quot;</span>) <span class="cf">else</span></span>
<span id="cb182-3135"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3135" aria-hidden="true" tabindex="-1"></a>        results1 <span class="ot">=</span> results1[<span class="fu">which</span>(results1<span class="sc">$</span>FDR <span class="sc">&lt;</span> minFDR),]</span>
<span id="cb182-3136"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3136" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-3137"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3137" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">!=</span> <span class="dv">4</span>) <span class="fu">return</span>(NoSig)</span>
<span id="cb182-3138"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3138" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(results1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Direction&quot;</span>,<span class="st">&quot;adj.Pval&quot;</span>,<span class="st">&quot;nGenes&quot;</span>,<span class="st">&quot;Pathways&quot;</span>)</span>
<span id="cb182-3139"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3139" aria-hidden="true" tabindex="-1"></a>        results1<span class="sc">$</span>adj.Pval <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;%-2.1e&quot;</span>,<span class="fu">as.numeric</span>(results1<span class="sc">$</span>adj.Pval) )   </span>
<span id="cb182-3140"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3140" aria-hidden="true" tabindex="-1"></a>        <span class="fu">rownames</span>(results1)<span class="ot">=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(results1)</span>
<span id="cb182-3141"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3141" aria-hidden="true" tabindex="-1"></a>        results1[ <span class="fu">duplicated</span> (results1[,<span class="dv">1</span>] ),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span>  </span>
<span id="cb182-3142"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3142" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-3143"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3143" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>( results1 )</span>
<span id="cb182-3144"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3144" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3145"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3146"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3147"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3147" aria-hidden="true" tabindex="-1"></a>stringDB_network1 <span class="ot">&lt;-</span> <span class="cf">function</span>( <span class="at">geneLists =</span> <span class="dv">1</span> ) {  <span class="co"># geneLists =2 for both up and down</span></span>
<span id="cb182-3148"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3148" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(STRINGdb)</span>
<span id="cb182-3149"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3149" aria-hidden="true" tabindex="-1"></a>                           </span>
<span id="cb182-3150"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3150" aria-hidden="true" tabindex="-1"></a>        tem <span class="ot">=</span> input_nGenesPPI</span>
<span id="cb182-3151"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3151" aria-hidden="true" tabindex="-1"></a>        taxonomyID <span class="ot">=</span> findTaxonomyID.out</span>
<span id="cb182-3152"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3152" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>( taxonomyID ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)     </span>
<span id="cb182-3153"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3153" aria-hidden="true" tabindex="-1"></a>        <span class="do">####################################</span></span>
<span id="cb182-3154"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3154" aria-hidden="true" tabindex="-1"></a>        NoSig <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No significant enrichment found.&quot;</span>)</span>
<span id="cb182-3155"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3155" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>(STRINGdb_geneList.out ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-3156"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3156" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-3157"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3158"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3158" aria-hidden="true" tabindex="-1"></a>        <span class="co">#Intialization</span></span>
<span id="cb182-3159"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3159" aria-hidden="true" tabindex="-1"></a>        string_db <span class="ot">&lt;-</span> STRINGdb<span class="sc">$</span><span class="fu">new</span>( <span class="at">version=</span><span class="st">&quot;10&quot;</span>, <span class="at">species=</span>taxonomyID,</span>
<span id="cb182-3160"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3160" aria-hidden="true" tabindex="-1"></a>                               <span class="at">score_threshold=</span><span class="dv">0</span>, <span class="at">input_directory=</span><span class="st">&quot;&quot;</span> )</span>
<span id="cb182-3161"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3161" aria-hidden="true" tabindex="-1"></a>        <span class="co"># only up regulated is ploted       </span></span>
<span id="cb182-3162"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3162" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( i <span class="cf">in</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span>geneLists) ) {</span>
<span id="cb182-3163"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3163" aria-hidden="true" tabindex="-1"></a>            ids <span class="ot">=</span> STRINGdb_geneList.out[[i]]</span>
<span id="cb182-3164"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3164" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">length</span>(ids)<span class="sc">&gt;</span> input_nGenesPPI )  <span class="co"># n of genes cannot be more than 400</span></span>
<span id="cb182-3165"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3165" aria-hidden="true" tabindex="-1"></a>                ids <span class="ot">&lt;-</span> ids[<span class="dv">1</span><span class="sc">:</span>input_nGenesPPI]</span>
<span id="cb182-3166"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3167"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3167" aria-hidden="true" tabindex="-1"></a>            string_db<span class="sc">$</span><span class="fu">plot_network</span>( ids,<span class="at">add_link=</span><span class="cn">FALSE</span>)</span>
<span id="cb182-3168"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3169"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3169" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-3170"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3170" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3171"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3172"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3173"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3174"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3174" aria-hidden="true" tabindex="-1"></a><span class="co"># generates a html file containing results and also links to STRING-db</span></span>
<span id="cb182-3175"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3175" aria-hidden="true" tabindex="-1"></a>stringDB_network_link <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb182-3176"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3176" aria-hidden="true" tabindex="-1"></a>        <span class="fu">library</span>(STRINGdb,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb182-3177"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3177" aria-hidden="true" tabindex="-1"></a>        taxonomyID <span class="ot">=</span> findTaxonomyID.out</span>
<span id="cb182-3178"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3178" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>( taxonomyID ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)     </span>
<span id="cb182-3179"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3179" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-3180"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3180" aria-hidden="true" tabindex="-1"></a>        NoSig <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No significant enrichment found.&quot;</span>)</span>
<span id="cb182-3181"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3181" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">is.null</span>(STRINGdb_geneList.out ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-3182"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3183"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3183" aria-hidden="true" tabindex="-1"></a>        <span class="co">#Intialization</span></span>
<span id="cb182-3184"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3184" aria-hidden="true" tabindex="-1"></a>        string_db <span class="ot">&lt;-</span> STRINGdb<span class="sc">$</span><span class="fu">new</span>( <span class="at">version=</span><span class="st">&quot;10&quot;</span>, <span class="at">species=</span>taxonomyID,</span>
<span id="cb182-3185"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3185" aria-hidden="true" tabindex="-1"></a>                               <span class="at">score_threshold=</span><span class="dv">0</span>, <span class="at">input_directory=</span><span class="st">&quot;&quot;</span> )</span>
<span id="cb182-3186"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3186" aria-hidden="true" tabindex="-1"></a>            <span class="co"># upregulated</span></span>
<span id="cb182-3187"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3187" aria-hidden="true" tabindex="-1"></a>           ids <span class="ot">=</span> STRINGdb_geneList.out[[<span class="dv">1</span>]]</span>
<span id="cb182-3188"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3188" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">length</span>(ids)<span class="sc">&gt;</span> input_nGenesPPI )  <span class="co"># n of genes cannot be more than 400</span></span>
<span id="cb182-3189"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3189" aria-hidden="true" tabindex="-1"></a>                ids <span class="ot">&lt;-</span> ids[<span class="dv">1</span><span class="sc">:</span>input_nGenesPPI]</span>
<span id="cb182-3190"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3191"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3191" aria-hidden="true" tabindex="-1"></a>            link1 <span class="ot">=</span> string_db<span class="sc">$</span><span class="fu">get_link</span>( ids)</span>
<span id="cb182-3192"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3192" aria-hidden="true" tabindex="-1"></a>            Pval1 <span class="ot">=</span> string_db<span class="sc">$</span><span class="fu">get_ppi_enrichment</span>( ids)</span>
<span id="cb182-3193"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3193" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="st">&quot;&lt;h5&gt; Interactive and annotated PPI networks among DEGs: &lt;br/&gt;  &quot;</span></span>
<span id="cb182-3194"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3194" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">paste</span>(tem, <span class="st">&quot;&lt;a href=</span><span class="sc">\&quot;</span><span class="st">&quot;</span>, link1, <span class="st">&quot;</span><span class="sc">\&quot;</span><span class="st"> target=</span><span class="sc">\&quot;</span><span class="st">_blank</span><span class="sc">\&quot;</span><span class="st">&gt; Up-regulated; &lt;/a&gt;&quot;</span>  )</span>
<span id="cb182-3195"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3196"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3196" aria-hidden="true" tabindex="-1"></a>            <span class="co"># downregulated</span></span>
<span id="cb182-3197"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3197" aria-hidden="true" tabindex="-1"></a>            ids <span class="ot">=</span> <span class="fu">STRINGdb_geneList</span>()[[<span class="dv">2</span>]]</span>
<span id="cb182-3198"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3198" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">length</span>(ids)<span class="sc">&gt;</span> input_nGenesPPI )</span>
<span id="cb182-3199"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3199" aria-hidden="true" tabindex="-1"></a>                ids <span class="ot">&lt;-</span> ids[<span class="dv">1</span><span class="sc">:</span>input_nGenesPPI]</span>
<span id="cb182-3200"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3201"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3201" aria-hidden="true" tabindex="-1"></a>            link2 <span class="ot">=</span> string_db<span class="sc">$</span><span class="fu">get_link</span>( ids)</span>
<span id="cb182-3202"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3202" aria-hidden="true" tabindex="-1"></a>            Pval2 <span class="ot">=</span> string_db<span class="sc">$</span><span class="fu">get_ppi_enrichment</span>( ids)              </span>
<span id="cb182-3203"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3203" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">paste</span>(tem, <span class="st">&quot; &amp;nbsp  &lt;a href=</span><span class="sc">\&quot;</span><span class="st">&quot;</span>, link2, <span class="st">&quot;</span><span class="sc">\&quot;</span><span class="st">target=</span><span class="sc">\&quot;</span><span class="st">_blank</span><span class="sc">\&quot;</span><span class="st">&gt; Down-regulated; &lt;/a&gt;&quot;</span>  )</span>
<span id="cb182-3204"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3204" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-3205"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3205" aria-hidden="true" tabindex="-1"></a>            <span class="co"># both up and down with color code</span></span>
<span id="cb182-3206"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3207"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3207" aria-hidden="true" tabindex="-1"></a>            geneTable <span class="ot">=</span> STRINGdb_geneList.out<span class="sc">$</span>geneTable</span>
<span id="cb182-3208"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3208" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">nrow</span>(geneTable)<span class="sc">&gt;</span> input_nGenesPPI ) </span>
<span id="cb182-3209"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3209" aria-hidden="true" tabindex="-1"></a>                geneTable <span class="ot">&lt;-</span> geneTable[<span class="dv">1</span><span class="sc">:</span>input_nGenesPPI,]</span>
<span id="cb182-3210"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3210" aria-hidden="true" tabindex="-1"></a>            geneTable <span class="ot">=</span>  string_db<span class="sc">$</span><span class="fu">add_diff_exp_color</span>( geneTable, <span class="at">logFcColStr=</span><span class="st">&quot;lfc&quot;</span> ) </span>
<span id="cb182-3211"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3211" aria-hidden="true" tabindex="-1"></a>            payload_id <span class="ot">&lt;-</span> string_db<span class="sc">$</span><span class="fu">post_payload</span>( geneTable<span class="sc">$</span>STRING_id,<span class="at">colors=</span>geneTable<span class="sc">$</span>color )</span>
<span id="cb182-3212"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3212" aria-hidden="true" tabindex="-1"></a>            link3 <span class="ot">=</span> string_db<span class="sc">$</span><span class="fu">get_link</span>(geneTable<span class="sc">$</span>STRING_id, <span class="at">payload_id =</span> payload_id)            </span>
<span id="cb182-3213"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3213" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">paste</span>(tem, <span class="st">&quot; &amp;nbsp  &lt;a href=</span><span class="sc">\&quot;</span><span class="st">&quot;</span>, link3, <span class="st">&quot;</span><span class="sc">\&quot;</span><span class="st">target=</span><span class="sc">\&quot;</span><span class="st">_blank</span><span class="sc">\&quot;</span><span class="st">&gt; Both with fold-changes color coded.&lt;/a&gt;&lt;/h5&gt;&quot;</span>  )</span>
<span id="cb182-3214"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3215"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3215" aria-hidden="true" tabindex="-1"></a>            tem2 <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;&lt;h5&gt; PPI enrichment P values: &quot;</span>)  </span>
<span id="cb182-3216"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3216" aria-hidden="true" tabindex="-1"></a>            tem2 <span class="ot">=</span> <span class="fu">paste0</span>(tem2,<span class="st">&quot;Up-regulated: &quot;</span>, <span class="fu">sprintf</span>(<span class="st">&quot;%-3.2e&quot;</span>,Pval1[<span class="dv">1</span>]), <span class="st">&quot; &amp;nbsp Down-regulated: &quot;</span>, <span class="fu">sprintf</span>(<span class="st">&quot;%-3.2e&quot;</span>,Pval2[<span class="dv">1</span>]),<span class="st">&quot;.&quot;</span>)</span>
<span id="cb182-3217"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3217" aria-hidden="true" tabindex="-1"></a>            tem2 <span class="ot">=</span> <span class="fu">paste</span>(tem2, <span class="st">&quot; Small P values indicate more PPIs among DEGs than background. &lt;/h5&gt;&quot;</span> )</span>
<span id="cb182-3218"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3218" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">=</span> <span class="fu">paste</span>(tem2,tem )</span>
<span id="cb182-3219"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3219" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>(tem) </span>
<span id="cb182-3220"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3220" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3221"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3222"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3223"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3223" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-3224"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3224" aria-hidden="true" tabindex="-1"></a><span class="co"># Pathway analysis</span></span>
<span id="cb182-3225"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3225" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-3226"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3227"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3227" aria-hidden="true" tabindex="-1"></a>gagePathwayData <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb182-3228"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3228" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(gage,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># pathway analysis  </span></span>
<span id="cb182-3229"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3230"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3231"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3232"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3232" aria-hidden="true" tabindex="-1"></a>    <span class="do">####################################</span></span>
<span id="cb182-3233"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3233" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3234"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3234" aria-hidden="true" tabindex="-1"></a>        myrange <span class="ot">=</span> <span class="fu">c</span>(input_minSetSize, input_maxSetSize)</span>
<span id="cb182-3235"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3235" aria-hidden="true" tabindex="-1"></a>        noSig <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No significant pathway found.&quot;</span>)</span>
<span id="cb182-3236"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3236" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">length</span>(limma.out<span class="sc">$</span>topGenes) <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(noSig)</span>
<span id="cb182-3237"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3237" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb182-3238"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3238" aria-hidden="true" tabindex="-1"></a>        { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb182-3239"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3239" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb182-3240"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3240" aria-hidden="true" tabindex="-1"></a>          top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb182-3241"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3241" aria-hidden="true" tabindex="-1"></a>          ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast1, <span class="fu">names</span>(top))</span>
<span id="cb182-3242"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3242" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (noSig)</span>
<span id="cb182-3243"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3243" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb182-3244"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3244" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb182-3245"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3245" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (noSig)</span>
<span id="cb182-3246"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3246" aria-hidden="true" tabindex="-1"></a>          <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb182-3247"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3248"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3248" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">=</span> top1[<span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;</span>input_GenePvalCutoff) , ]</span>
<span id="cb182-3249"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3249" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb182-3250"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3251"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3251" aria-hidden="true" tabindex="-1"></a>          gmt <span class="ot">=</span> GeneSets.out </span>
<span id="cb182-3252"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3252" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">length</span>( GeneSets.out )  <span class="sc">==</span> <span class="dv">0</span>)  { <span class="fu">return</span>(<span class="fu">as.data.frame</span>(<span class="st">&quot;No gene set found!&quot;</span>))}</span>
<span id="cb182-3253"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3253" aria-hidden="true" tabindex="-1"></a>          <span class="co">#converted = convertID(rownames(top1),input_selectOrg)</span></span>
<span id="cb182-3254"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3254" aria-hidden="true" tabindex="-1"></a>          <span class="co">#</span></span>
<span id="cb182-3255"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3255" aria-hidden="true" tabindex="-1"></a>           <span class="co">#gmt = readGeneSets(converted, top1, input_selectGO, input_selectOrg, myrange )</span></span>
<span id="cb182-3256"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3256" aria-hidden="true" tabindex="-1"></a>         <span class="co"># cat(&quot;Sets&quot;,length(gmt))</span></span>
<span id="cb182-3257"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3258"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3258" aria-hidden="true" tabindex="-1"></a>         fold <span class="ot">=</span> top1[,<span class="dv">1</span>]; <span class="fu">names</span>(fold) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top1)</span>
<span id="cb182-3259"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3259" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb182-3260"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3260" aria-hidden="true" tabindex="-1"></a>         <span class="cf">if</span>(input_absoluteFold) fold <span class="ot">&lt;-</span> <span class="fu">abs</span>(fold)</span>
<span id="cb182-3261"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3261" aria-hidden="true" tabindex="-1"></a>         paths <span class="ot">&lt;-</span> <span class="fu">gage</span>(fold, <span class="at">gsets =</span> gmt, <span class="at">ref =</span> <span class="cn">NULL</span>, <span class="at">samp =</span> <span class="cn">NULL</span>)</span>
<span id="cb182-3262"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3263"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3263" aria-hidden="true" tabindex="-1"></a>          paths <span class="ot">&lt;-</span>  <span class="fu">rbind</span>(paths<span class="sc">$</span>greater,paths<span class="sc">$</span>less)</span>
<span id="cb182-3264"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3264" aria-hidden="true" tabindex="-1"></a>         <span class="co"># write.csv(paths,&quot;tem.csv&quot;)     </span></span>
<span id="cb182-3265"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3265" aria-hidden="true" tabindex="-1"></a>         <span class="co"># cat( dim(paths) )</span></span>
<span id="cb182-3266"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3266" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">dim</span>(paths)[<span class="dv">1</span>] <span class="sc">&lt;</span> <span class="dv">1</span> <span class="sc">|</span> <span class="fu">dim</span>(paths)[<span class="dv">2</span>]<span class="sc">&lt;</span> <span class="dv">6</span> ) <span class="fu">return</span>( noSig )</span>
<span id="cb182-3267"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3267" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> paths[,<span class="fu">c</span>(<span class="st">&#39;stat.mean&#39;</span>,<span class="st">&#39;set.size&#39;</span>,<span class="st">&#39;q.val&#39;</span>)]</span>
<span id="cb182-3268"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3268" aria-hidden="true" tabindex="-1"></a>          <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;statistic&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;adj.Pval&quot;</span>)</span>
<span id="cb182-3269"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3269" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top1[<span class="fu">order</span>(top1[,<span class="dv">3</span>]) ,]  </span>
<span id="cb182-3270"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3270" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span> ( <span class="fu">length</span>( <span class="fu">which</span>( top1[,<span class="dv">3</span>] <span class="sc">&lt;=</span>  input_pathwayPvalCutoff   ) ) <span class="sc">==</span> <span class="dv">0</span> )</span>
<span id="cb182-3271"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3271" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>( noSig)</span>
<span id="cb182-3272"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3272" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top1[<span class="fu">which</span>(top1[,<span class="dv">3</span>] <span class="sc">&lt;=</span>  input_pathwayPvalCutoff ) ,,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb182-3273"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3273" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">&gt;</span> input_nPathwayShow ) </span>
<span id="cb182-3274"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3274" aria-hidden="true" tabindex="-1"></a>             top1 <span class="ot">&lt;-</span> top1[<span class="dv">1</span><span class="sc">:</span>input_nPathwayShow, ,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb182-3275"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3275" aria-hidden="true" tabindex="-1"></a>             </span>
<span id="cb182-3276"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3276" aria-hidden="true" tabindex="-1"></a>            top1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(top1)</span>
<span id="cb182-3277"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3277" aria-hidden="true" tabindex="-1"></a>            top1 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">rep</span>( input_selectContrast1, <span class="fu">dim</span>(top1)[<span class="dv">1</span>]),<span class="fu">row.names</span>(top1), top1); </span>
<span id="cb182-3278"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3278" aria-hidden="true" tabindex="-1"></a>            top1<span class="sc">$</span>statistic <span class="ot">&lt;-</span> <span class="fu">as.character</span>( <span class="fu">round</span>(<span class="fu">as.numeric</span>(top1<span class="sc">$</span>statistic),<span class="dv">4</span>)); </span>
<span id="cb182-3279"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3279" aria-hidden="true" tabindex="-1"></a>            top1<span class="sc">$</span>adj.Pval <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;%-2.1e&quot;</span>,<span class="fu">as.numeric</span>(top1<span class="sc">$</span>adj.Pval) )</span>
<span id="cb182-3280"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3280" aria-hidden="true" tabindex="-1"></a>            top1[,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(top1[,<span class="dv">2</span>]);top1[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(top1[,<span class="dv">1</span>])</span>
<span id="cb182-3281"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3281" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(top1)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Direction&quot;</span></span>
<span id="cb182-3282"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3282" aria-hidden="true" tabindex="-1"></a>            p.m <span class="ot">&lt;-</span> <span class="st">&quot;GAGE&quot;</span></span>
<span id="cb182-3283"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3284"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3284" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(top1)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">paste</span>(p.m,<span class="st">&quot; analysis:&quot;</span>, <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot; vs &quot;</span>,input_selectContrast1 ) )</span>
<span id="cb182-3285"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3285" aria-hidden="true" tabindex="-1"></a>            top1[ <span class="fu">which</span>( top1[,<span class="dv">3</span>] <span class="sc">&gt;</span><span class="dv">0</span>),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;Up&quot;</span> <span class="co">#gsub(&quot;-&quot;,&quot; &gt; &quot;,input_selectContrast1 )</span></span>
<span id="cb182-3286"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3286" aria-hidden="true" tabindex="-1"></a>            top1[ <span class="fu">which</span>( top1[,<span class="dv">3</span>] <span class="sc">&lt;</span><span class="dv">0</span>),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;Down&quot;</span> <span class="co"># gsub(&quot;-&quot;,&quot; &lt; &quot;,input_selectContrast1 )</span></span>
<span id="cb182-3287"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3287" aria-hidden="true" tabindex="-1"></a>            top1 <span class="ot">&lt;-</span> top1[<span class="fu">order</span>( top1[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">abs</span>(<span class="fu">as.numeric</span>( top1[,<span class="dv">3</span>]) ) ) ,]</span>
<span id="cb182-3288"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3288" aria-hidden="true" tabindex="-1"></a>            top1[ <span class="fu">duplicated</span> (top1[,<span class="dv">1</span>] ),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb182-3289"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3290"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3290" aria-hidden="true" tabindex="-1"></a>          <span class="fu">return</span>( top1)</span>
<span id="cb182-3291"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3291" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3292"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3293"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3293" aria-hidden="true" tabindex="-1"></a>fgseaPathwayData <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb182-3294"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3294" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(fgsea,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># fast GSEA</span></span>
<span id="cb182-3295"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3296"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3296" aria-hidden="true" tabindex="-1"></a>    noSig <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No significant pathway found.&quot;</span>)</span>
<span id="cb182-3297"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3297" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">length</span>(limma.out<span class="sc">$</span>topGenes) <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(noSig)</span>
<span id="cb182-3298"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3298" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb182-3299"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3299" aria-hidden="true" tabindex="-1"></a>    { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb182-3300"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3300" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb182-3301"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3301" aria-hidden="true" tabindex="-1"></a>      top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb182-3302"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3302" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast1, <span class="fu">names</span>(top))</span>
<span id="cb182-3303"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3303" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (noSig)</span>
<span id="cb182-3304"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3304" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb182-3305"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3305" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb182-3306"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3306" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (noSig)</span>
<span id="cb182-3307"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3307" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb182-3308"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3308" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb182-3309"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3309" aria-hidden="true" tabindex="-1"></a>      <span class="co"># remove some genes</span></span>
<span id="cb182-3310"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3310" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">=</span> top1[<span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;</span>input_GenePvalCutoff) , ]</span>
<span id="cb182-3311"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3311" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb182-3312"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3313"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3313" aria-hidden="true" tabindex="-1"></a>      gmt <span class="ot">=</span> GeneSets.out </span>
<span id="cb182-3314"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3314" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>(<span class="fu">length</span>( GeneSets.out )  <span class="sc">==</span> <span class="dv">0</span>)  { <span class="fu">return</span>(<span class="fu">as.data.frame</span>(<span class="st">&quot;No gene set found!&quot;</span>))}</span>
<span id="cb182-3315"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3316"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3316" aria-hidden="true" tabindex="-1"></a>      <span class="co">#converted = convertID(rownames(top1),input_selectOrg)</span></span>
<span id="cb182-3317"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3317" aria-hidden="true" tabindex="-1"></a>      <span class="co">#</span></span>
<span id="cb182-3318"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3318" aria-hidden="true" tabindex="-1"></a>       <span class="co">#gmt = readGeneSets(converted, top1, input_selectGO, input_selectOrg, myrange )</span></span>
<span id="cb182-3319"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3319" aria-hidden="true" tabindex="-1"></a>     <span class="co"># cat(&quot;Sets&quot;,length(gmt))</span></span>
<span id="cb182-3320"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3321"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3321" aria-hidden="true" tabindex="-1"></a>     fold <span class="ot">=</span> top1[,<span class="dv">1</span>]; <span class="fu">names</span>(fold) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top1)</span>
<span id="cb182-3322"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3322" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>(input_absoluteFold) fold <span class="ot">&lt;-</span> <span class="fu">abs</span>(fold) <span class="co"># use absolute value of fold change, disregard direction</span></span>
<span id="cb182-3323"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3323" aria-hidden="true" tabindex="-1"></a>     paths <span class="ot">&lt;-</span> <span class="fu">fgsea</span>(<span class="at">pathways =</span> gmt, </span>
<span id="cb182-3324"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3324" aria-hidden="true" tabindex="-1"></a>                  <span class="at">stats =</span> fold,</span>
<span id="cb182-3325"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3325" aria-hidden="true" tabindex="-1"></a>                  <span class="at">minSize=</span>input_minSetSize,</span>
<span id="cb182-3326"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3326" aria-hidden="true" tabindex="-1"></a>                  <span class="at">maxSize=</span>input_maxSetSize,</span>
<span id="cb182-3327"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3327" aria-hidden="true" tabindex="-1"></a>                  <span class="at">nperm=</span><span class="dv">10000</span>)</span>
<span id="cb182-3328"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3328" aria-hidden="true" tabindex="-1"></a>     <span class="co"># paths &lt;-  rbind(paths$greater,paths$less)</span></span>
<span id="cb182-3329"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3329" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(paths)[<span class="dv">1</span>] <span class="sc">&lt;</span> <span class="dv">1</span>  ) <span class="fu">return</span>( noSig )</span>
<span id="cb182-3330"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3330" aria-hidden="true" tabindex="-1"></a>           paths <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(paths)</span>
<span id="cb182-3331"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3331" aria-hidden="true" tabindex="-1"></a>       paths <span class="ot">&lt;-</span> paths[<span class="fu">order</span>(<span class="sc">-</span><span class="fu">abs</span>( paths[,<span class="dv">5</span>])) ,]  <span class="co"># sort by NES</span></span>
<span id="cb182-3332"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3332" aria-hidden="true" tabindex="-1"></a>      <span class="co"># paths &lt;- paths[order( paths[,3]) ,]  # sort by FDR</span></span>
<span id="cb182-3333"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3333" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> paths[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">7</span>,<span class="dv">3</span>)]</span>
<span id="cb182-3334"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3334" aria-hidden="true" tabindex="-1"></a>      <span class="co"># rownames(top1) &lt;- paths[,1] #paste(1:dim(paths)[1],&quot;: &quot;,paths[,1],sep=&quot;&quot; )</span></span>
<span id="cb182-3335"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3335" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Pathway&quot;</span>, <span class="st">&quot;NES&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;adj.Pval&quot;</span>)</span>
<span id="cb182-3336"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3336" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb182-3337"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3337" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> ( <span class="fu">length</span>( <span class="fu">which</span>( top1[,<span class="dv">4</span>] <span class="sc">&lt;=</span>  input_pathwayPvalCutoff   ) ) <span class="sc">==</span> <span class="dv">0</span> )</span>
<span id="cb182-3338"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3338" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>( noSig)</span>
<span id="cb182-3339"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3339" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top1[<span class="fu">which</span>(top1[,<span class="dv">4</span>] <span class="sc">&lt;=</span>  input_pathwayPvalCutoff ) , ,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb182-3340"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3340" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">&gt;</span> input_nPathwayShow ) </span>
<span id="cb182-3341"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3341" aria-hidden="true" tabindex="-1"></a>         top1 <span class="ot">&lt;-</span> top1[<span class="dv">1</span><span class="sc">:</span>input_nPathwayShow,,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb182-3342"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3342" aria-hidden="true" tabindex="-1"></a>     <span class="co">#top1 &lt;- cbind(row.names(top1), top1); colnames(top1)[1] &lt;-input_selectContrast1   </span></span>
<span id="cb182-3343"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3343" aria-hidden="true" tabindex="-1"></a>        top1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(top1)</span>
<span id="cb182-3344"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3344" aria-hidden="true" tabindex="-1"></a>        top1 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">rep</span>( input_selectContrast1, <span class="fu">dim</span>(top1)[<span class="dv">1</span>]), top1); </span>
<span id="cb182-3345"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3345" aria-hidden="true" tabindex="-1"></a>        top1[,<span class="dv">4</span>] <span class="ot">=</span> <span class="fu">as.character</span>( <span class="fu">round</span>(<span class="fu">as.numeric</span>(top1[,<span class="dv">4</span>]),<span class="dv">4</span>)); </span>
<span id="cb182-3346"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3346" aria-hidden="true" tabindex="-1"></a>        top1<span class="sc">$</span>adj.Pval <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;%-2.1e&quot;</span>,<span class="fu">as.numeric</span>(top1<span class="sc">$</span>adj.Pval) )</span>
<span id="cb182-3347"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3347" aria-hidden="true" tabindex="-1"></a>        top1[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(top1[,<span class="dv">1</span>])</span>
<span id="cb182-3348"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3348" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(top1)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Direction&quot;</span></span>
<span id="cb182-3349"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3350"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3350" aria-hidden="true" tabindex="-1"></a>        p.m <span class="ot">&lt;-</span> <span class="st">&quot;GSEA&quot;</span></span>
<span id="cb182-3351"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3352"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3352" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(top1)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">paste</span>(p.m,<span class="st">&quot; analysis:&quot;</span>, <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot; vs &quot;</span>,input_selectContrast1 ) )</span>
<span id="cb182-3353"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3353" aria-hidden="true" tabindex="-1"></a>        top1[ <span class="fu">which</span>( <span class="fu">as.numeric</span>( top1[,<span class="dv">3</span>]) <span class="sc">&gt;</span><span class="dv">0</span>),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;Up&quot;</span> <span class="co">#gsub(&quot;-&quot;,&quot; &gt; &quot;,input_selectContrast1 )</span></span>
<span id="cb182-3354"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3354" aria-hidden="true" tabindex="-1"></a>        top1[ <span class="fu">which</span>( <span class="fu">as.numeric</span>( top1[,<span class="dv">3</span>]) <span class="sc">&lt;</span><span class="dv">0</span>),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;Down&quot;</span> <span class="co">#gsub(&quot;-&quot;,&quot; &lt; &quot;,input_selectContrast1 )</span></span>
<span id="cb182-3355"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3355" aria-hidden="true" tabindex="-1"></a>        top1 <span class="ot">&lt;-</span> top1[<span class="fu">order</span>( top1[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">abs</span>(<span class="fu">as.numeric</span>( top1[,<span class="dv">3</span>]) ) ) ,]</span>
<span id="cb182-3356"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3356" aria-hidden="true" tabindex="-1"></a>        top1[ <span class="fu">duplicated</span> (top1[,<span class="dv">1</span>] ),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span>   </span>
<span id="cb182-3357"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3357" aria-hidden="true" tabindex="-1"></a>        top1[,<span class="dv">3</span>] <span class="ot">=</span> <span class="fu">as.character</span>( <span class="fu">round</span>(<span class="fu">as.numeric</span>(top1[,<span class="dv">3</span>]),<span class="dv">4</span>));</span>
<span id="cb182-3358"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3359"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3359" aria-hidden="true" tabindex="-1"></a>     <span class="fu">return</span>( top1)</span>
<span id="cb182-3360"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3360" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3361"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3362"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3362" aria-hidden="true" tabindex="-1"></a>extract1 <span class="ot">&lt;-</span> <span class="cf">function</span> (x) {</span>
<span id="cb182-3363"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3363" aria-hidden="true" tabindex="-1"></a>  words <span class="ot">&lt;-</span> <span class="fu">unlist</span> ( <span class="fu">strsplit</span>(x,<span class="st">&quot;_&quot;</span>))</span>
<span id="cb182-3364"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3364" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">length</span>( words )  <span class="sc">&lt;=</span><span class="dv">4</span> ) <span class="fu">return</span>(<span class="fu">gsub</span>(<span class="st">&quot;_&quot;</span>,<span class="st">&quot; &quot;</span>,x)) <span class="cf">else</span> {</span>
<span id="cb182-3365"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3365" aria-hidden="true" tabindex="-1"></a>  words <span class="ot">&lt;-</span> words[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)]</span>
<span id="cb182-3366"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3366" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>( <span class="fu">proper</span>(<span class="fu">paste</span>(words,<span class="at">collapse =</span> <span class="st">&quot; &quot;</span>) ) )}</span>
<span id="cb182-3367"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3367" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3368"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3369"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3369" aria-hidden="true" tabindex="-1"></a>PGSEApathway <span class="ot">&lt;-</span> <span class="cf">function</span> (converted,convertedData, selectOrg,GO,gmt, myrange,Pval_pathway,top){</span>
<span id="cb182-3370"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3370" aria-hidden="true" tabindex="-1"></a>    subtype <span class="ot">=</span> <span class="fu">detectGroups</span>(<span class="fu">colnames</span>(convertedData))</span>
<span id="cb182-3371"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3371" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(PGSEA,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb182-3372"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3372" aria-hidden="true" tabindex="-1"></a>    Pvalue <span class="ot">=</span> <span class="fl">0.01</span>  <span class="co"># cut off to report in PGSEA. Otherwise NA</span></span>
<span id="cb182-3373"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3373" aria-hidden="true" tabindex="-1"></a>    <span class="co">#Pval_pathway = 0.2   # cut off for P value of ANOVA test  to writ to file </span></span>
<span id="cb182-3374"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3374" aria-hidden="true" tabindex="-1"></a>    <span class="co"># top = 30   # number of pathways to show</span></span>
<span id="cb182-3375"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3375" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(gmt) <span class="sc">==</span><span class="dv">0</span> ) <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">pg3 =</span> <span class="cn">NULL</span>, <span class="at">best =</span> <span class="dv">1</span> ) )</span>
<span id="cb182-3376"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3376" aria-hidden="true" tabindex="-1"></a>    <span class="co"># centering by mean</span></span>
<span id="cb182-3377"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3377" aria-hidden="true" tabindex="-1"></a>    <span class="co">#pg = myPGSEA (convertedData - rowMeans(convertedData),</span></span>
<span id="cb182-3378"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3378" aria-hidden="true" tabindex="-1"></a>     <span class="co">#            cl=gmt,range=myrange,p.value=TRUE, weighted=FALSE,nPermutation=100)</span></span>
<span id="cb182-3379"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3380"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3380" aria-hidden="true" tabindex="-1"></a>    <span class="co">#if( class(convertedData) != &quot;data.frame&quot; | class(convertedData) != &quot;matarix&quot;) return( list(pg3 = NULL, best = 1 ) )</span></span>
<span id="cb182-3381"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3381" aria-hidden="true" tabindex="-1"></a>    <span class="co">#if( dim(convertedData)[2] &lt;2 ) return( list(pg3 = NULL, best = 1 ) )</span></span>
<span id="cb182-3382"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3382" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3383"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3383" aria-hidden="true" tabindex="-1"></a>     pg <span class="ot">=</span> <span class="fu">PGSEA</span> (convertedData <span class="sc">-</span> <span class="fu">rowMeans</span>(convertedData),<span class="at">cl=</span>gmt,<span class="at">range=</span>myrange,<span class="at">p.value=</span><span class="cn">TRUE</span>, <span class="at">weighted=</span><span class="cn">FALSE</span>)</span>
<span id="cb182-3384"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3384" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3385"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3385" aria-hidden="true" tabindex="-1"></a>    pg2 <span class="ot">=</span> pg<span class="sc">$</span>results;</span>
<span id="cb182-3386"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3386" aria-hidden="true" tabindex="-1"></a>    pg2 <span class="ot">=</span> pg2[<span class="fu">rowSums</span>(<span class="fu">is.na</span>(pg2))<span class="sc">&lt;</span><span class="fu">ncol</span>(pg2) ,]  <span class="co"># remove se/wrts with all missing(non-signficant)</span></span>
<span id="cb182-3387"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3387" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">dim</span>(pg2)[<span class="dv">1</span>] <span class="sc">&lt;</span> <span class="dv">2</span> ) return.out</span>
<span id="cb182-3388"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3388" aria-hidden="true" tabindex="-1"></a>    best <span class="ot">=</span> <span class="fu">max</span>(<span class="fu">abs</span>(pg2))</span>
<span id="cb182-3389"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3389" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3390"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3390" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(subtype) <span class="sc">&lt;</span> <span class="dv">4</span> <span class="sc">||</span> <span class="fu">length</span>(<span class="fu">unique</span>(subtype)) <span class="sc">&lt;</span><span class="dv">2</span> <span class="sc">||</span><span class="fu">length</span>(<span class="fu">unique</span>(subtype)) <span class="sc">==</span> <span class="fu">dim</span>(convertedData)[<span class="dv">2</span>] ) { </span>
<span id="cb182-3391"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3391" aria-hidden="true" tabindex="-1"></a>     pg2 <span class="ot">=</span> pg2[<span class="fu">order</span>(<span class="sc">-</span><span class="fu">apply</span>(pg2,<span class="dv">1</span>,sd)     )   ,]</span>
<span id="cb182-3392"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3392" aria-hidden="true" tabindex="-1"></a>     <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">pg3 =</span> pg2[<span class="dv">1</span><span class="sc">:</span>top,], <span class="at">best =</span> best ) )</span>
<span id="cb182-3393"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3393" aria-hidden="true" tabindex="-1"></a>    } </span>
<span id="cb182-3394"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3394" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3395"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3395" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">Computing P values using ANOVA</span><span class="sc">\n</span><span class="st">&quot;</span>);</span>
<span id="cb182-3396"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3396" aria-hidden="true" tabindex="-1"></a>    pathPvalue <span class="ot">&lt;-</span> <span class="cf">function</span> ( k){</span>
<span id="cb182-3397"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3397" aria-hidden="true" tabindex="-1"></a>     <span class="fu">return</span>( <span class="fu">summary</span>(<span class="fu">aov</span>(pg2[k,]<span class="sc">~</span>subtype) )[[<span class="dv">1</span>]][[<span class="st">&quot;Pr(&gt;F)&quot;</span>]][<span class="dv">1</span>] )</span>
<span id="cb182-3398"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3398" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-3399"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3399" aria-hidden="true" tabindex="-1"></a>    Pvalues <span class="ot">=</span> <span class="fu">sapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(pg2)[<span class="dv">1</span>], pathPvalue)</span>
<span id="cb182-3400"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3400" aria-hidden="true" tabindex="-1"></a>    Pvalues <span class="ot">=</span> <span class="fu">p.adjust</span>(Pvalues, <span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb182-3401"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3401" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3402"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3402" aria-hidden="true" tabindex="-1"></a>    <span class="co">#if(min(Pvalues) &gt; Pval_pathway ) return( list(pg3 = NULL, best = best ) )  else {  </span></span>
<span id="cb182-3403"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3403" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">sort</span>(Pvalues)[<span class="dv">2</span>] <span class="sc">&gt;</span> Pval_pathway ) <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">pg3 =</span> <span class="cn">NULL</span>, <span class="at">best =</span> best ) )  <span class="cf">else</span> {  </span>
<span id="cb182-3404"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3404" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3405"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3405" aria-hidden="true" tabindex="-1"></a>    NsigT <span class="ot">=</span> <span class="fu">rowSums</span>(pg<span class="sc">$</span>p.results<span class="sc">&lt;</span>Pvalue)</span>
<span id="cb182-3406"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3406" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3407"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3407" aria-hidden="true" tabindex="-1"></a>    result<span class="ot">=</span><span class="fu">cbind</span>( <span class="fu">as.matrix</span>(Pvalues),NsigT,pg2); </span>
<span id="cb182-3408"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3408" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">=</span> result[ <span class="fu">order</span>(result[,<span class="dv">1</span>])   ,] </span>
<span id="cb182-3409"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3409" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3410"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3410" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">=</span> result[<span class="fu">which</span>(result[,<span class="dv">1</span>] <span class="sc">&lt;</span> Pval_pathway),,drop<span class="ot">=</span>F]</span>
<span id="cb182-3411"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3411" aria-hidden="true" tabindex="-1"></a>    <span class="co">#result = result[which(result[,2] &gt;2)    ,]</span></span>
<span id="cb182-3412"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3412" aria-hidden="true" tabindex="-1"></a>    pg2 <span class="ot">=</span> result[,<span class="sc">-</span><span class="dv">2</span>]</span>
<span id="cb182-3413"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3414"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3414" aria-hidden="true" tabindex="-1"></a>    <span class="co"># when there is only 1 left in the matrix pg2 becomes a vector</span></span>
<span id="cb182-3415"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3415" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">sum</span>( Pvalues<span class="sc">&lt;</span>Pval_pathway) <span class="sc">==</span> <span class="dv">1</span>) { pg3 <span class="ot">=</span> <span class="fu">t</span>( <span class="fu">as.matrix</span>(pg2));pg3 <span class="ot">=</span> <span class="fu">rbind</span>(pg3,pg3);} <span class="cf">else</span></span>
<span id="cb182-3416"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3416" aria-hidden="true" tabindex="-1"></a>    { <span class="cf">if</span>(<span class="fu">dim</span>(pg2)[<span class="dv">1</span>] <span class="sc">&gt;</span> top ) {  pg3 <span class="ot">=</span> pg2[<span class="dv">1</span><span class="sc">:</span>top,]; } <span class="cf">else</span> {  pg3 <span class="ot">=</span> pg2;  } }</span>
<span id="cb182-3417"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3418"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3418" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(pg3) <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">rownames</span>(pg3) , extract1)</span>
<span id="cb182-3419"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3419" aria-hidden="true" tabindex="-1"></a>    a<span class="ot">=</span><span class="fu">sprintf</span>(<span class="st">&quot;%-3.2e&quot;</span>,pg3[,<span class="dv">1</span>])</span>
<span id="cb182-3420"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3420" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(pg3) <span class="ot">=</span> <span class="fu">paste</span>(a,<span class="fu">rownames</span>(pg3),<span class="at">sep=</span><span class="st">&quot; &quot;</span>)</span>
<span id="cb182-3421"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3421" aria-hidden="true" tabindex="-1"></a>    pg3 <span class="ot">=</span>pg3[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb182-3422"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3422" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3423"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3423" aria-hidden="true" tabindex="-1"></a>    pg3 <span class="ot">&lt;-</span> pg3[<span class="fu">order</span>( <span class="sc">-</span><span class="fu">apply</span>(pg3,<span class="dv">1</span>,sd)    ),] <span class="co"># sort by SD</span></span>
<span id="cb182-3424"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3424" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb182-3425"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3425" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">pg3 =</span> pg3, <span class="at">best =</span> best ) )</span>
<span id="cb182-3426"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3426" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-3427"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3427" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb182-3428"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3429"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3429" aria-hidden="true" tabindex="-1"></a>PGSEAplot <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb182-3430"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3430" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb182-3431"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3431" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(PGSEA,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb182-3432"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3432" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3433"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3433" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_selectGO <span class="sc">==</span> <span class="st">&quot;ID not recognized!&quot;</span> ) <span class="fu">return</span>( <span class="cn">NULL</span>)</span>
<span id="cb182-3434"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3434" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3435"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3435" aria-hidden="true" tabindex="-1"></a>    myrange <span class="ot">=</span> <span class="fu">c</span>(input_minSetSize, input_maxSetSize)</span>
<span id="cb182-3436"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3436" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">=</span> convertedData.out</span>
<span id="cb182-3437"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3437" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">is.null</span>(input_selectContrast1 ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-3438"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3439"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3439" aria-hidden="true" tabindex="-1"></a>    gmt <span class="ot">=</span> GeneSets.out</span>
<span id="cb182-3440"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3441"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3442"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3442" aria-hidden="true" tabindex="-1"></a>    <span class="co"># find related samples</span></span>
<span id="cb182-3443"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3443" aria-hidden="true" tabindex="-1"></a>    iz <span class="ot">=</span> <span class="fu">findContrastSamples</span>(input_selectContrast1, <span class="fu">colnames</span>(convertedData.out),readSampleInfo.out,</span>
<span id="cb182-3444"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3444" aria-hidden="true" tabindex="-1"></a>                                        input_selectFactorsModel,input_selectModelComprions, </span>
<span id="cb182-3445"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3445" aria-hidden="true" tabindex="-1"></a>                                        factorReferenceLevels.out,input_CountsDEGMethod,</span>
<span id="cb182-3446"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3446" aria-hidden="true" tabindex="-1"></a>                                        input_dataFileFormat  )</span>
<span id="cb182-3447"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3447" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3448"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3448" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3449"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3449" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3450"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3450" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">=</span> genes[,iz]  </span>
<span id="cb182-3451"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3451" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3452"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3452" aria-hidden="true" tabindex="-1"></a>    subtype <span class="ot">=</span> <span class="fu">detectGroups</span>(<span class="fu">colnames</span>(genes )) </span>
<span id="cb182-3453"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3453" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( GeneSets.out )  <span class="sc">==</span> <span class="dv">0</span>)  { <span class="fu">plot.new</span>(); <span class="fu">text</span>(<span class="fl">0.5</span>,<span class="fl">0.5</span>, <span class="st">&quot;No gene sets!&quot;</span>)} <span class="cf">else</span> {</span>
<span id="cb182-3454"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3454" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">=</span> <span class="fu">PGSEApathway</span>(converted.out,genes, input_selectOrg,input_selectGO,</span>
<span id="cb182-3455"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3455" aria-hidden="true" tabindex="-1"></a>                 GeneSets.out,  myrange, input_pathwayPvalCutoff, input_nPathwayShow    )</span>
<span id="cb182-3456"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3456" aria-hidden="true" tabindex="-1"></a>                     </span>
<span id="cb182-3457"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3457" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">is.null</span>(result<span class="sc">$</span>pg3) ) { <span class="fu">plot.new</span>(); <span class="fu">text</span>(<span class="fl">0.5</span>,<span class="dv">1</span>, <span class="st">&quot;No significant pathway found!&quot;</span>)} <span class="cf">else</span> </span>
<span id="cb182-3458"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3458" aria-hidden="true" tabindex="-1"></a>    <span class="fu">smcPlot</span>(result<span class="sc">$</span>pg3,<span class="fu">factor</span>(subtype),<span class="at">scale =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fu">max</span>(result<span class="sc">$</span>pg3), <span class="fu">max</span>(result<span class="sc">$</span>pg3)), </span>
<span id="cb182-3459"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3459" aria-hidden="true" tabindex="-1"></a>    <span class="at">show.grid =</span> T, <span class="at">margins =</span> <span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">1</span>, <span class="dv">13</span>,<span class="dv">28</span>), <span class="at">col =</span> .rwb,<span class="at">cex.lab=</span><span class="fl">0.5</span>, <span class="at">main=</span><span class="st">&quot;Pathway Analysis:PGSEA&quot;</span>)</span>
<span id="cb182-3460"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3460" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-3461"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3461" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3462"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3462" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3463"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3463" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3464"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3464" aria-hidden="true" tabindex="-1"></a>convertEnsembl2Entrez <span class="ot">&lt;-</span> <span class="cf">function</span> (query,Species) { </span>
<span id="cb182-3465"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3465" aria-hidden="true" tabindex="-1"></a>    querySet <span class="ot">&lt;-</span> <span class="fu">cleanGeneSet</span>( <span class="fu">unlist</span>( <span class="fu">strsplit</span>( <span class="fu">toupper</span>(<span class="fu">names</span>( query)),<span class="st">&#39;</span><span class="sc">\t</span><span class="st">| |</span><span class="sc">\n</span><span class="st">|</span><span class="sc">\\</span><span class="st">,&#39;</span> )  ) )</span>
<span id="cb182-3466"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3466" aria-hidden="true" tabindex="-1"></a>    speciesID <span class="ot">&lt;-</span> orgInfo<span class="sc">$</span>id[ <span class="fu">which</span>(orgInfo<span class="sc">$</span>ensembl_dataset <span class="sc">==</span> Species)]  <span class="co"># note uses species Identifying</span></span>
<span id="cb182-3467"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3467" aria-hidden="true" tabindex="-1"></a>    <span class="co"># idType 6 for entrez gene ID</span></span>
<span id="cb182-3468"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3468" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>( convert,</span>
<span id="cb182-3469"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3469" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">paste</span>( <span class="st">&quot; select  id,ens,species from mapping where ens IN (&#39;&quot;</span>, <span class="fu">paste</span>(querySet,<span class="at">collapse=</span><span class="st">&quot;&#39;, &#39;&quot;</span>),</span>
<span id="cb182-3470"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3470" aria-hidden="true" tabindex="-1"></a>                                <span class="st">&quot;&#39;) AND  idType =&#39;&quot;</span>,idType_Entrez,<span class="st">&quot;&#39;&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>) ) <span class="co"># slow</span></span>
<span id="cb182-3471"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3471" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb182-3472"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3472" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>  ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-3473"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3473" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">subset</span>(result, species<span class="sc">==</span>speciesID, <span class="at">select =</span> <span class="sc">-</span>species)</span>
<span id="cb182-3474"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3475"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3475" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span> <span class="fu">match</span>(result<span class="sc">$</span>ens,<span class="fu">names</span>(query)  )</span>
<span id="cb182-3476"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3477"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3477" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">&lt;-</span> query[ix];  <span class="fu">names</span>(tem) <span class="ot">=</span> result<span class="sc">$</span>id</span>
<span id="cb182-3478"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3478" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(tem)</span>
<span id="cb182-3479"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3479" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb182-3480"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3480" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3481"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3481" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3482"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3482" aria-hidden="true" tabindex="-1"></a>ReactomePAPathwayData <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb182-3483"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3483" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(ReactomePA,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># pathway analysis</span></span>
<span id="cb182-3484"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3484" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb182-3485"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3485" aria-hidden="true" tabindex="-1"></a>    ensemblSpecies <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;hsapiens_gene_ensembl&quot;</span>,<span class="st">&quot;rnorvegicus_gene_ensembl&quot;</span>, <span class="st">&quot;mmusculus_gene_ensembl&quot;</span>,</span>
<span id="cb182-3486"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3486" aria-hidden="true" tabindex="-1"></a>                       <span class="st">&quot;celegans_gene_ensembl&quot;</span>,<span class="st">&quot;scerevisiae_gene_ensembl&quot;</span>, <span class="st">&quot;drerio_gene_ensembl&quot;</span>, <span class="st">&quot;dmelanogaster_gene_ensembl&quot;</span>)</span>
<span id="cb182-3487"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3487" aria-hidden="true" tabindex="-1"></a>    ReactomePASpecies<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;human&quot;</span>, <span class="st">&quot;rat&quot;</span>, <span class="st">&quot;mouse&quot;</span>, <span class="st">&quot;celegans&quot;</span>, <span class="st">&quot;yeast&quot;</span>, <span class="st">&quot;zebrafish&quot;</span>, <span class="st">&quot;fly&quot;</span> )</span>
<span id="cb182-3488"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3488" aria-hidden="true" tabindex="-1"></a>    <span class="co"># cbind(ensemblSpecies,ReactomePASpecies)  # double check mapping</span></span>
<span id="cb182-3489"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3489" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3490"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3490" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3491"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3491" aria-hidden="true" tabindex="-1"></a>    myrange <span class="ot">=</span> <span class="fu">c</span>(input_minSetSize, input_maxSetSize)</span>
<span id="cb182-3492"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3492" aria-hidden="true" tabindex="-1"></a>    noSig <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No significant pathway found.&quot;</span>)</span>
<span id="cb182-3493"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3493" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">length</span>(limma.out<span class="sc">$</span>topGenes) <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(noSig)</span>
<span id="cb182-3494"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3494" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb182-3495"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3495" aria-hidden="true" tabindex="-1"></a>    { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb182-3496"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3496" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb182-3497"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3497" aria-hidden="true" tabindex="-1"></a>      top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb182-3498"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3498" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast1, <span class="fu">names</span>(top))</span>
<span id="cb182-3499"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3499" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (noSig)</span>
<span id="cb182-3500"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3500" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb182-3501"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3501" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb182-3502"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3502" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (noSig)</span>
<span id="cb182-3503"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3503" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb182-3504"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3505"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3505" aria-hidden="true" tabindex="-1"></a>      <span class="co"># remove some genes</span></span>
<span id="cb182-3506"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3506" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">=</span> top1[<span class="fu">which</span>(top1<span class="sc">$</span>FDR <span class="sc">&lt;</span>input_GenePvalCutoff) , ]</span>
<span id="cb182-3507"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3507" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb182-3508"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3508" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-3509"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3509" aria-hidden="true" tabindex="-1"></a>     fold <span class="ot">=</span> top1[,<span class="dv">1</span>]; <span class="fu">names</span>(fold) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top1)</span>
<span id="cb182-3510"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3510" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>(input_absoluteFold) fold <span class="ot">&lt;-</span> <span class="fu">abs</span>(fold) <span class="co"># use absolute value of fold change, disregard direction</span></span>
<span id="cb182-3511"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3512"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3512" aria-hidden="true" tabindex="-1"></a>       Species <span class="ot">&lt;-</span> converted.out<span class="sc">$</span>species</span>
<span id="cb182-3513"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3513" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">&lt;-</span> <span class="fu">match</span>( Species,ensemblSpecies ) </span>
<span id="cb182-3514"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3514" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.na</span>(ix) ) <span class="fu">return</span>(<span class="fu">as.data.frame</span>(<span class="st">&quot;Species not coverted by ReactomePA package!&quot;</span>))</span>
<span id="cb182-3515"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3515" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb182-3516"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3516" aria-hidden="true" tabindex="-1"></a>      fold <span class="ot">&lt;-</span> <span class="fu">convertEnsembl2Entrez</span> (fold, Species)  </span>
<span id="cb182-3517"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3517" aria-hidden="true" tabindex="-1"></a>         fold <span class="ot">&lt;-</span> <span class="fu">sort</span>(fold,<span class="at">decreasing =</span>T)</span>
<span id="cb182-3518"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3518" aria-hidden="true" tabindex="-1"></a>      paths <span class="ot">&lt;-</span> <span class="fu">gsePathway</span>(fold, <span class="at">nPerm=</span><span class="dv">5000</span>, <span class="at">organism =</span> ReactomePASpecies[ix],</span>
<span id="cb182-3519"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3519" aria-hidden="true" tabindex="-1"></a>                <span class="at">minGSSize=</span> input_minSetSize, </span>
<span id="cb182-3520"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3520" aria-hidden="true" tabindex="-1"></a>                <span class="at">maxGSSize=</span> input_maxSetSize,</span>
<span id="cb182-3521"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3521" aria-hidden="true" tabindex="-1"></a>                <span class="at">pvalueCutoff=</span><span class="fl">0.5</span>,</span>
<span id="cb182-3522"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3522" aria-hidden="true" tabindex="-1"></a>                <span class="at">pAdjustMethod=</span><span class="st">&quot;BH&quot;</span>, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb182-3523"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3523" aria-hidden="true" tabindex="-1"></a>      paths <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(paths)</span>
<span id="cb182-3524"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3524" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb182-3525"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3525" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.null</span>(paths) ) <span class="fu">return</span>( noSig)</span>
<span id="cb182-3526"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3526" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(paths)[<span class="dv">1</span>] <span class="sc">==</span><span class="dv">0</span> ) <span class="fu">return</span>( noSig)</span>
<span id="cb182-3527"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3528"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3528" aria-hidden="true" tabindex="-1"></a>     <span class="co"># paths &lt;-  rbind(paths$greater,paths$less)</span></span>
<span id="cb182-3529"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3529" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(paths)[<span class="dv">1</span>] <span class="sc">&lt;</span> <span class="dv">1</span>  ) <span class="fu">return</span>( noSig )</span>
<span id="cb182-3530"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3530" aria-hidden="true" tabindex="-1"></a>           paths <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(paths)</span>
<span id="cb182-3531"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3531" aria-hidden="true" tabindex="-1"></a>       paths <span class="ot">&lt;-</span> paths[<span class="fu">order</span>(<span class="sc">-</span><span class="fu">abs</span>( paths[,<span class="dv">5</span>])) ,]  <span class="co"># sort by NES</span></span>
<span id="cb182-3532"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3532" aria-hidden="true" tabindex="-1"></a>      <span class="co"># paths &lt;- paths[order( paths[,3]) ,]  # sort by FDR</span></span>
<span id="cb182-3533"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3533" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> paths[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">3</span>,<span class="dv">7</span>)]</span>
<span id="cb182-3534"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3534" aria-hidden="true" tabindex="-1"></a>      <span class="co"># rownames(top1) &lt;- paths[,1] #paste(1:dim(paths)[1],&quot;: &quot;,paths[,1],sep=&quot;&quot; )</span></span>
<span id="cb182-3535"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3535" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Pathway&quot;</span>, <span class="st">&quot;NES&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;adj.Pval&quot;</span>)</span>
<span id="cb182-3536"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3536" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb182-3537"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3537" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> ( <span class="fu">length</span>( <span class="fu">which</span>( top1[,<span class="dv">4</span>] <span class="sc">&lt;=</span>  input_pathwayPvalCutoff   ) ) <span class="sc">==</span> <span class="dv">0</span> )</span>
<span id="cb182-3538"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3538" aria-hidden="true" tabindex="-1"></a>        <span class="fu">return</span>( noSig)</span>
<span id="cb182-3539"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3539" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top1[<span class="fu">which</span>(top1[,<span class="dv">4</span>] <span class="sc">&lt;=</span>  input_pathwayPvalCutoff ) ,,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb182-3540"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3540" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">&gt;</span> input_nPathwayShow ) </span>
<span id="cb182-3541"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3541" aria-hidden="true" tabindex="-1"></a>         top1 <span class="ot">&lt;-</span> top1[<span class="dv">1</span><span class="sc">:</span>input_nPathwayShow,,drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb182-3542"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3542" aria-hidden="true" tabindex="-1"></a>     <span class="co">#top1 &lt;- cbind(row.names(top1), top1); colnames(top1)[1] &lt;-input_selectContrast1   </span></span>
<span id="cb182-3543"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3543" aria-hidden="true" tabindex="-1"></a>        top1 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(top1)</span>
<span id="cb182-3544"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3544" aria-hidden="true" tabindex="-1"></a>        top1 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">rep</span>( input_selectContrast1, <span class="fu">dim</span>(top1)[<span class="dv">1</span>]), top1); </span>
<span id="cb182-3545"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3545" aria-hidden="true" tabindex="-1"></a>        top1[,<span class="dv">4</span>] <span class="ot">=</span> <span class="fu">as.character</span>( <span class="fu">round</span>(<span class="fu">as.numeric</span>(top1[,<span class="dv">4</span>]),<span class="dv">4</span>)); </span>
<span id="cb182-3546"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3546" aria-hidden="true" tabindex="-1"></a>        top1<span class="sc">$</span>adj.Pval <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;%-2.1e&quot;</span>,<span class="fu">as.numeric</span>(top1<span class="sc">$</span>adj.Pval) )</span>
<span id="cb182-3547"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3547" aria-hidden="true" tabindex="-1"></a>        top1[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(top1[,<span class="dv">1</span>])</span>
<span id="cb182-3548"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3548" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(top1)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Direction&quot;</span></span>
<span id="cb182-3549"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3549" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(input_pathwayMethod <span class="sc">==</span> <span class="dv">1</span> ) p.m <span class="ot">&lt;-</span> <span class="st">&quot;GAGE&quot;</span></span>
<span id="cb182-3550"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3550" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span> <span class="cf">if</span>(input_pathwayMethod <span class="sc">==</span> <span class="dv">2</span> ) p.m <span class="ot">&lt;-</span> <span class="st">&quot;PGSEA&quot;</span></span>
<span id="cb182-3551"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3551" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span> <span class="cf">if</span>(input_pathwayMethod <span class="sc">==</span> <span class="dv">3</span> ) p.m <span class="ot">&lt;-</span> <span class="st">&quot;GSEA&quot;</span></span>
<span id="cb182-3552"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3552" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span> <span class="cf">if</span>(input_pathwayMethod <span class="sc">==</span> <span class="dv">4</span> ) p.m <span class="ot">&lt;-</span> <span class="st">&quot;PGSEA_All&quot;</span></span>
<span id="cb182-3553"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3553" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span> <span class="cf">if</span>(input_pathwayMethod <span class="sc">==</span> <span class="dv">5</span> ) p.m <span class="ot">&lt;-</span> <span class="st">&quot;ReactomePA&quot;</span></span>
<span id="cb182-3554"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3554" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(top1)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">paste</span>(p.m,<span class="st">&quot; analysis:&quot;</span>, <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>,<span class="st">&quot; vs &quot;</span>,input_selectContrast1 ) )</span>
<span id="cb182-3555"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3555" aria-hidden="true" tabindex="-1"></a>        top1[ <span class="fu">which</span>( <span class="fu">as.numeric</span>( top1[,<span class="dv">3</span>]) <span class="sc">&gt;</span><span class="dv">0</span>),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;Up&quot;</span> <span class="co">#gsub(&quot;-&quot;,&quot; &gt; &quot;,input_selectContrast1 )</span></span>
<span id="cb182-3556"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3556" aria-hidden="true" tabindex="-1"></a>        top1[ <span class="fu">which</span>( <span class="fu">as.numeric</span>( top1[,<span class="dv">3</span>]) <span class="sc">&lt;</span><span class="dv">0</span>),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;Down&quot;</span> <span class="co">#gsub(&quot;-&quot;,&quot; &lt; &quot;,input_selectContrast1 )</span></span>
<span id="cb182-3557"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3557" aria-hidden="true" tabindex="-1"></a>        top1 <span class="ot">&lt;-</span> top1[<span class="fu">order</span>( top1[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">abs</span>(<span class="fu">as.numeric</span>( top1[,<span class="dv">3</span>]) ) ) ,]</span>
<span id="cb182-3558"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3558" aria-hidden="true" tabindex="-1"></a>        top1[ <span class="fu">duplicated</span> (top1[,<span class="dv">1</span>] ),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span>   </span>
<span id="cb182-3559"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3559" aria-hidden="true" tabindex="-1"></a>      top1[,<span class="dv">3</span>] <span class="ot">=</span> <span class="fu">as.character</span>( <span class="fu">round</span>(<span class="fu">as.numeric</span>(top1[,<span class="dv">3</span>]),<span class="dv">4</span>));</span>
<span id="cb182-3560"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3560" aria-hidden="true" tabindex="-1"></a>     <span class="fu">return</span>( top1)</span>
<span id="cb182-3561"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3561" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3562"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3562" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3563"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3563" aria-hidden="true" tabindex="-1"></a>selectedPathwayData <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb182-3564"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3564" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb182-3565"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3565" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_sigPathways <span class="sc">==</span> <span class="st">&quot;All&quot;</span>) <span class="fu">return</span> (<span class="cn">NULL</span>) </span>
<span id="cb182-3566"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3566" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">names</span>(GeneSets.out ) <span class="sc">==</span> input_sigPathways   ) <span class="co"># find the gene set</span></span>
<span id="cb182-3567"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3567" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-3568"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3568" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">&lt;-</span> GeneSets.out[[ix]] <span class="co"># retrieve genes</span></span>
<span id="cb182-3569"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3569" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3570"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3570" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3571"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3571" aria-hidden="true" tabindex="-1"></a>    <span class="co"># find related samples  </span></span>
<span id="cb182-3572"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3572" aria-hidden="true" tabindex="-1"></a>    iz <span class="ot">=</span> <span class="fu">findContrastSamples</span>(input_selectContrast1, <span class="fu">colnames</span>(convertedData.out),readSampleInfo.out,</span>
<span id="cb182-3573"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3573" aria-hidden="true" tabindex="-1"></a>                                        input_selectFactorsModel,input_selectModelComprions, </span>
<span id="cb182-3574"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3574" aria-hidden="true" tabindex="-1"></a>                                        factorReferenceLevels.out,input_CountsDEGMethod ,</span>
<span id="cb182-3575"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3575" aria-hidden="true" tabindex="-1"></a>                                        input_dataFileFormat )</span>
<span id="cb182-3576"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3576" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span>  convertedData.out[<span class="fu">which</span>(<span class="fu">rownames</span>(convertedData.out) <span class="sc">%in%</span> genes), iz ]</span>
<span id="cb182-3577"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3577" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_selectOrg <span class="sc">!=</span> <span class="st">&quot;NEW&quot;</span>) {</span>
<span id="cb182-3578"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3578" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">match</span>( <span class="fu">rownames</span>(x), allGeneInfo.out[,<span class="dv">1</span>])</span>
<span id="cb182-3579"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3579" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( <span class="fu">sum</span>( <span class="fu">is.na</span>(allGeneInfo.out<span class="sc">$</span>symbol )) <span class="sc">!=</span> <span class="fu">dim</span>(allGeneInfo.out )[<span class="dv">1</span>] )  <span class="co"># symbol really exists? </span></span>
<span id="cb182-3580"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3580" aria-hidden="true" tabindex="-1"></a>           <span class="fu">rownames</span>(x) <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">rownames</span>(x),<span class="st">&quot;:&quot;</span>, <span class="fu">as.character</span>( allGeneInfo.out<span class="sc">$</span>symbol)[ix])</span>
<span id="cb182-3581"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3581" aria-hidden="true" tabindex="-1"></a>   }</span>
<span id="cb182-3582"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3582" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3583"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3583" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( x )</span>
<span id="cb182-3584"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3584" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3585"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3585" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3586"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3586" aria-hidden="true" tabindex="-1"></a>selectedPathwayHeatmap <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb182-3587"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3587" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3588"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3588" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">=</span> selectedPathwayData.out</span>
<span id="cb182-3589"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3589" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">dim</span>(x)[<span class="dv">1</span>]<span class="sc">&lt;=</span><span class="dv">2</span> <span class="sc">|</span> <span class="fu">dim</span>(x)[<span class="dv">2</span>]<span class="sc">&lt;=</span><span class="dv">2</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-3590"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3590" aria-hidden="true" tabindex="-1"></a>    groups <span class="ot">=</span> <span class="fu">detectGroups</span>(<span class="fu">colnames</span>(x))</span>
<span id="cb182-3591"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3591" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3592"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3592" aria-hidden="true" tabindex="-1"></a>    <span class="co"># this will cutoff very large values, which could skew the color </span></span>
<span id="cb182-3593"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3593" aria-hidden="true" tabindex="-1"></a>    x<span class="ot">=</span><span class="fu">as.matrix</span>(x)<span class="sc">-</span><span class="fu">apply</span>(<span class="fu">as.matrix</span>(x),<span class="dv">1</span>,mean)</span>
<span id="cb182-3594"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3594" aria-hidden="true" tabindex="-1"></a>    cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">+</span> <span class="dv">3</span><span class="sc">*</span><span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb182-3595"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3595" aria-hidden="true" tabindex="-1"></a>    x[x<span class="sc">&gt;</span>cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb182-3596"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3596" aria-hidden="true" tabindex="-1"></a>    cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">-</span> <span class="dv">3</span><span class="sc">*</span><span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb182-3597"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3597" aria-hidden="true" tabindex="-1"></a>    x[x<span class="sc">&lt;</span> cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb182-3598"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3598" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3599"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3599" aria-hidden="true" tabindex="-1"></a>    <span class="co"># sometimes, a gene can be all zero in selected samples.</span></span>
<span id="cb182-3600"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3600" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> x[<span class="fu">which</span>(<span class="fu">apply</span>(x,<span class="dv">1</span>,sd)<span class="sc">&gt;</span><span class="dv">0</span>) ,]</span>
<span id="cb182-3601"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3601" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3602"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3602" aria-hidden="true" tabindex="-1"></a>    lmat <span class="ot">=</span> <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">4</span>),<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">2</span>))</span>
<span id="cb182-3603"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3603" aria-hidden="true" tabindex="-1"></a>    lwid <span class="ot">=</span> <span class="fu">c</span>(<span class="fl">1.5</span>,<span class="dv">6</span>)</span>
<span id="cb182-3604"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3604" aria-hidden="true" tabindex="-1"></a>    lhei <span class="ot">=</span> <span class="fu">c</span>(.<span class="dv">5</span>,.<span class="dv">2</span>,<span class="dv">8</span>)</span>
<span id="cb182-3605"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3605" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3606"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3606" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3607"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3607" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>(x)[<span class="dv">1</span>]<span class="sc">&gt;</span><span class="dv">200</span>) </span>
<span id="cb182-3608"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3608" aria-hidden="true" tabindex="-1"></a>    <span class="fu">heatmap.2</span>(x, <span class="at">distfun =</span> dist2,<span class="at">hclustfun=</span>hclust2,</span>
<span id="cb182-3609"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3609" aria-hidden="true" tabindex="-1"></a>     <span class="at">col=</span>heatColors[<span class="fu">as.integer</span>(input_heatColors1),], <span class="at">density.info=</span><span class="st">&quot;none&quot;</span>, <span class="at">trace=</span><span class="st">&quot;none&quot;</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">keysize=</span>.<span class="dv">5</span></span>
<span id="cb182-3610"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3610" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">key=</span>T, <span class="at">symkey=</span>F</span>
<span id="cb182-3611"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3611" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">dendrogram =</span> <span class="st">&quot;row&quot;</span></span>
<span id="cb182-3612"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3612" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">ColSideColors=</span>mycolors[ groups]</span>
<span id="cb182-3613"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3613" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">labRow=</span><span class="st">&quot;&quot;</span></span>
<span id="cb182-3614"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3614" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">margins=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">15</span>)</span>
<span id="cb182-3615"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3615" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">srtCol=</span><span class="dv">45</span></span>
<span id="cb182-3616"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3616" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">lmat =</span> lmat, <span class="at">lwid =</span> lwid, <span class="at">lhei =</span> lhei</span>
<span id="cb182-3617"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3617" aria-hidden="true" tabindex="-1"></a>    <span class="co">#,main =&quot;Title&quot;</span></span>
<span id="cb182-3618"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3618" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb182-3619"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3619" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3620"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3620" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>(x)[<span class="dv">1</span>]<span class="sc">&lt;=</span><span class="dv">200</span>) </span>
<span id="cb182-3621"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3621" aria-hidden="true" tabindex="-1"></a>    <span class="fu">heatmap.2</span>(x, <span class="at">distfun =</span> dist2,<span class="at">hclustfun=</span>hclust2,</span>
<span id="cb182-3622"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3622" aria-hidden="true" tabindex="-1"></a>     <span class="at">col=</span>heatColors[<span class="fu">as.integer</span>(input_heatColors1),], <span class="at">density.info=</span><span class="st">&quot;none&quot;</span>, <span class="at">trace=</span><span class="st">&quot;none&quot;</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">keysize=</span>.<span class="dv">5</span></span>
<span id="cb182-3623"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3623" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">key=</span>T, <span class="at">symkey=</span>F, </span>
<span id="cb182-3624"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3624" aria-hidden="true" tabindex="-1"></a>    <span class="at">dendrogram =</span> <span class="st">&quot;row&quot;</span>,</span>
<span id="cb182-3625"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3625" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">labRow=</span><span class="fu">gsub</span>(<span class="st">&quot;.*:&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">rownames</span>(x))</span>
<span id="cb182-3626"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3626" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">ColSideColors=</span>mycolors[ groups]</span>
<span id="cb182-3627"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3627" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">margins=</span><span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">15</span>)</span>
<span id="cb182-3628"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3628" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">cexRow=</span><span class="fl">1.2</span></span>
<span id="cb182-3629"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3629" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">srtCol=</span><span class="dv">45</span></span>
<span id="cb182-3630"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3630" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">lmat =</span> lmat, <span class="at">lwid =</span> lwid, <span class="at">lhei =</span> lhei</span>
<span id="cb182-3631"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3631" aria-hidden="true" tabindex="-1"></a>    <span class="co">#,main =&quot;Title&quot;</span></span>
<span id="cb182-3632"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3632" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb182-3633"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3633" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3634"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3634" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3635"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3635" aria-hidden="true" tabindex="-1"></a>convertEnsembl2KEGG <span class="ot">&lt;-</span> <span class="cf">function</span> (query,Species) {  <span class="co"># not working</span></span>
<span id="cb182-3636"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3636" aria-hidden="true" tabindex="-1"></a>    querySet <span class="ot">&lt;-</span> <span class="fu">cleanGeneSet</span>( <span class="fu">unlist</span>( <span class="fu">strsplit</span>( <span class="fu">toupper</span>(<span class="fu">names</span>( query)),<span class="st">&#39;</span><span class="sc">\t</span><span class="st">| |</span><span class="sc">\n</span><span class="st">|</span><span class="sc">\\</span><span class="st">,&#39;</span> )  ) )</span>
<span id="cb182-3637"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3637" aria-hidden="true" tabindex="-1"></a>    speciesID <span class="ot">&lt;-</span> orgInfo<span class="sc">$</span>id[ <span class="fu">which</span>(orgInfo<span class="sc">$</span>ensembl_dataset <span class="sc">==</span> Species)]  <span class="co"># note uses species Identifying</span></span>
<span id="cb182-3638"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3638" aria-hidden="true" tabindex="-1"></a>    <span class="co"># idType 6 for entrez gene ID</span></span>
<span id="cb182-3639"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3639" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>( convert,</span>
<span id="cb182-3640"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3640" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">paste</span>( <span class="st">&quot; select  id,ens,species from mapping where ens IN (&#39;&quot;</span>, <span class="fu">paste</span>(querySet,<span class="at">collapse=</span><span class="st">&quot;&#39;, &#39;&quot;</span>),</span>
<span id="cb182-3641"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3641" aria-hidden="true" tabindex="-1"></a>                                <span class="st">&quot;&#39;) AND  idType =&#39;&quot;</span>,idType_KEGG,<span class="st">&quot;&#39;&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>) )   <span class="co"># slow</span></span>
<span id="cb182-3642"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3642" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb182-3643"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3643" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">dim</span>(result)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>  ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-3644"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3644" aria-hidden="true" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">subset</span>(result, species<span class="sc">==</span>speciesID, <span class="at">select =</span> <span class="sc">-</span>species)</span>
<span id="cb182-3645"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3645" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3646"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3646" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span> <span class="fu">match</span>(result<span class="sc">$</span>ens,<span class="fu">names</span>(query)  )</span>
<span id="cb182-3647"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3647" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3648"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3648" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">&lt;-</span> query[ix];  <span class="fu">names</span>(tem) <span class="ot">=</span> result<span class="sc">$</span>id</span>
<span id="cb182-3649"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3649" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(tem)  </span>
<span id="cb182-3650"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3650" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3651"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3651" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3652"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3652" aria-hidden="true" tabindex="-1"></a>keggPathwayID <span class="ot">&lt;-</span> <span class="cf">function</span> (pathwayDescription, Species, GO,selectOrg) {</span>
<span id="cb182-3653"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3653" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span> <span class="fu">grep</span>(Species,gmtFiles)</span>
<span id="cb182-3654"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3654" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3655"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3655" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> ) {<span class="fu">return</span>(<span class="cn">NULL</span>)}</span>
<span id="cb182-3656"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3656" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3657"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3657" aria-hidden="true" tabindex="-1"></a>    <span class="co"># If selected species is not the default &quot;bestMatch&quot;, use that species directly</span></span>
<span id="cb182-3658"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3658" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(selectOrg <span class="sc">!=</span> speciesChoice[[<span class="dv">1</span>]]) {  </span>
<span id="cb182-3659"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3659" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">grep</span>(<span class="fu">findSpeciesById</span>(selectOrg)[<span class="dv">1</span>,<span class="dv">1</span>], gmtFiles )</span>
<span id="cb182-3660"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3660" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> ) {<span class="fu">return</span>(<span class="cn">NULL</span> )}</span>
<span id="cb182-3661"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3661" aria-hidden="true" tabindex="-1"></a>        totalGenes <span class="ot">&lt;-</span> orgInfo[<span class="fu">which</span>(orgInfo<span class="sc">$</span>id <span class="sc">==</span> <span class="fu">as.numeric</span>(selectOrg)),<span class="dv">7</span>]</span>
<span id="cb182-3662"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3662" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-3663"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3663" aria-hidden="true" tabindex="-1"></a>    pathway <span class="ot">&lt;-</span> <span class="fu">dbConnect</span>(sqlite,gmtFiles[ix],<span class="at">flags=</span>SQLITE_RO)</span>
<span id="cb182-3664"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3664" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3665"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3665" aria-hidden="true" tabindex="-1"></a>    <span class="co"># change Parkinson&#39;s disease to Parkinson\&#39;s disease    otherwise SQL </span></span>
<span id="cb182-3666"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3666" aria-hidden="true" tabindex="-1"></a>    pathwayDescription <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\&#39;</span><span class="st">&quot;</span>,<span class="st">&quot;</span><span class="sc">\&#39;\&#39;</span><span class="st">&quot;</span>,pathwayDescription)</span>
<span id="cb182-3667"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3667" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb182-3668"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3668" aria-hidden="true" tabindex="-1"></a>    pathwayInfo <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>( pathway, <span class="fu">paste</span>( <span class="st">&quot; select * from pathwayInfo where description =  &#39;&quot;</span>, </span>
<span id="cb182-3669"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3669" aria-hidden="true" tabindex="-1"></a>                            pathwayDescription,   <span class="st">&quot;&#39; AND name LIKE &#39;&quot;</span>,GO,<span class="st">&quot;%&#39;&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>) )     </span>
<span id="cb182-3670"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3670" aria-hidden="true" tabindex="-1"></a>    pathwayInfo <span class="ot">&lt;-</span> <span class="fu">dbGetQuery</span>( pathway, <span class="fu">paste</span>( <span class="st">&quot; select * from pathwayInfo where description =  &#39;&quot;</span>, </span>
<span id="cb182-3671"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3671" aria-hidden="true" tabindex="-1"></a>                            pathwayDescription,   <span class="st">&quot;&#39;&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>) ) </span>
<span id="cb182-3672"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3672" aria-hidden="true" tabindex="-1"></a>                            </span>
<span id="cb182-3673"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3673" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dbDisconnect</span>(pathway);</span>
<span id="cb182-3674"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3674" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">dim</span>(pathwayInfo)[<span class="dv">1</span>] <span class="sc">!=</span> <span class="dv">1</span> ) {<span class="fu">return</span>(<span class="cn">NULL</span>) }</span>
<span id="cb182-3675"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3675" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;.*:&quot;</span>,<span class="st">&quot;&quot;</span>,pathwayInfo[<span class="dv">1</span>,<span class="dv">2</span>])  </span>
<span id="cb182-3676"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3676" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>,<span class="st">&quot;&quot;</span>,tem) )</span>
<span id="cb182-3677"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3677" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3678"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3678" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3679"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3679" aria-hidden="true" tabindex="-1"></a>KeggImage <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb182-3680"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3680" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3681"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3681" aria-hidden="true" tabindex="-1"></a>   <span class="co"># First generate a blank image. Otherwise return(NULL) gives us errors.</span></span>
<span id="cb182-3682"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3682" aria-hidden="true" tabindex="-1"></a>    outfile <span class="ot">&lt;-</span> <span class="fu">tempfile</span>(<span class="at">fileext=</span><span class="st">&#39;.png&#39;</span>)</span>
<span id="cb182-3683"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3683" aria-hidden="true" tabindex="-1"></a>    <span class="fu">png</span>(outfile, <span class="at">width=</span><span class="dv">400</span>, <span class="at">height=</span><span class="dv">300</span>)</span>
<span id="cb182-3684"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3684" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3685"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3685" aria-hidden="true" tabindex="-1"></a>    <span class="fu">frame</span>()</span>
<span id="cb182-3686"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3686" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dev.off</span>()</span>
<span id="cb182-3687"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3687" aria-hidden="true" tabindex="-1"></a>    blank <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">src =</span> outfile,</span>
<span id="cb182-3688"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3688" aria-hidden="true" tabindex="-1"></a>         <span class="at">contentType =</span> <span class="st">&#39;image/png&#39;</span>,</span>
<span id="cb182-3689"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3689" aria-hidden="true" tabindex="-1"></a>         <span class="at">width =</span> <span class="dv">400</span>,</span>
<span id="cb182-3690"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3690" aria-hidden="true" tabindex="-1"></a>         <span class="at">height =</span> <span class="dv">300</span>,</span>
<span id="cb182-3691"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3691" aria-hidden="true" tabindex="-1"></a>         <span class="at">alt =</span> <span class="st">&quot; &quot;</span>) </span>
<span id="cb182-3692"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3692" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3693"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3693" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>( input_selectGO ) ) <span class="fu">return</span>(outfile)</span>
<span id="cb182-3694"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3694" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_selectGO <span class="sc">!=</span> <span class="st">&quot;KEGG&quot;</span>) <span class="fu">return</span>(outfile)</span>
<span id="cb182-3695"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3695" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(gagePathwayData.out ) ) <span class="fu">return</span>(outfile)</span>
<span id="cb182-3696"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3696" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>( input_sigPathways))  <span class="fu">return</span> (outfile) </span>
<span id="cb182-3697"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3697" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if( is.null(selectedPathwayData()) ) return(blank)</span></span>
<span id="cb182-3698"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3698" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3699"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3699" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(pathview)</span>
<span id="cb182-3700"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3700" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3701"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3701" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3702"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3702" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(input_selectContrast1 ) ) <span class="fu">return</span>(outfile)</span>
<span id="cb182-3703"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3703" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3704"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3704" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(input_sigPathways <span class="sc">==</span> <span class="st">&quot;All&quot;</span>) <span class="fu">return</span> (outfile) </span>
<span id="cb182-3705"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3705" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3706"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3706" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">length</span>(limma.out<span class="sc">$</span>topGenes) <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(outfile)</span>
<span id="cb182-3707"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3707" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3708"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3708" aria-hidden="true" tabindex="-1"></a>    <span class="co"># get fold change</span></span>
<span id="cb182-3709"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3709" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb182-3710"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3710" aria-hidden="true" tabindex="-1"></a>    { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb182-3711"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3711" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb182-3712"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3712" aria-hidden="true" tabindex="-1"></a>      top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb182-3713"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3713" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast1, <span class="fu">names</span>(top))</span>
<span id="cb182-3714"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3714" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (outfile)</span>
<span id="cb182-3715"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3715" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb182-3716"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3716" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb182-3717"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3717" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (outfile)</span>
<span id="cb182-3718"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3718" aria-hidden="true" tabindex="-1"></a>     <span class="co"># cat(&quot;here5&quot;)</span></span>
<span id="cb182-3719"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3719" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb182-3720"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3720" aria-hidden="true" tabindex="-1"></a>      Species <span class="ot">&lt;-</span> converted.out<span class="sc">$</span>species</span>
<span id="cb182-3721"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3721" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb182-3722"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3722" aria-hidden="true" tabindex="-1"></a>     fold <span class="ot">=</span> top1[,<span class="dv">1</span>]; <span class="fu">names</span>(fold) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(top1)</span>
<span id="cb182-3723"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3723" aria-hidden="true" tabindex="-1"></a>     fold <span class="ot">&lt;-</span> <span class="fu">convertEnsembl2Entrez</span>(fold,Species)</span>
<span id="cb182-3724"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3724" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-3725"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3725" aria-hidden="true" tabindex="-1"></a>     keggSpecies <span class="ot">&lt;-</span> <span class="fu">as.character</span>( keggSpeciesID[<span class="fu">which</span>(keggSpeciesID[,<span class="dv">1</span>] <span class="sc">==</span> Species),<span class="dv">3</span>] )</span>
<span id="cb182-3726"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3726" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-3727"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3727" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>(<span class="fu">nchar</span>( keggSpecies) <span class="sc">&lt;=</span><span class="dv">2</span> ) <span class="fu">return</span>(outfile) <span class="co"># not in KEGG</span></span>
<span id="cb182-3728"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3728" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3729"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3729" aria-hidden="true" tabindex="-1"></a>     <span class="co"># kegg pathway id</span></span>
<span id="cb182-3730"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3730" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3731"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3731" aria-hidden="true" tabindex="-1"></a>    pathID <span class="ot">=</span> <span class="fu">keggPathwayID</span>(input_sigPathways, Species, <span class="st">&quot;KEGG&quot;</span>,input_selectOrg)</span>
<span id="cb182-3732"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3732" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3733"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3733" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(&quot;\nhere5  &quot;,keggSpecies, &quot; &quot;,Species,&quot; &quot;,input_sigPathways, &quot;pathID:&quot;,pathID,&quot;End&quot;, fold[1:5],names(fold)[1:5],&quot;\n&quot;)</span></span>
<span id="cb182-3734"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3734" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(&quot;\npathway:&quot;,is.na(input_sigPathways))</span></span>
<span id="cb182-3735"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3735" aria-hidden="true" tabindex="-1"></a>    <span class="co">#cat(&quot;\n&quot;,fold[1:5],&quot;\n&quot;,keggSpecies,&quot;\n&quot;,pathID)</span></span>
<span id="cb182-3736"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3736" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(pathID) ) <span class="fu">return</span>(blank) <span class="co"># kegg pathway id not found.</span></span>
<span id="cb182-3737"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3737" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">nchar</span>(pathID)<span class="sc">&lt;</span><span class="dv">3</span> ) <span class="fu">return</span>(blank)</span>
<span id="cb182-3738"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3738" aria-hidden="true" tabindex="-1"></a>    randomString <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;.*file&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">tempfile</span>()) </span>
<span id="cb182-3739"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3739" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3740"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3740" aria-hidden="true" tabindex="-1"></a>    outfile <span class="ot">&lt;-</span> <span class="fu">paste</span>(pathID,<span class="st">&quot;.&quot;</span>,randomString,<span class="st">&quot;.png&quot;</span>,<span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb182-3741"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3741" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3742"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3742" aria-hidden="true" tabindex="-1"></a>        <span class="fu">try</span>( </span>
<span id="cb182-3743"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3743" aria-hidden="true" tabindex="-1"></a>         pv.out <span class="ot">&lt;-</span> <span class="fu">pathview</span>(<span class="at">gene.data =</span> fold, <span class="at">pathway.id =</span> pathID, <span class="at">out.suffix =</span> randomString, <span class="at">species =</span> keggSpecies, <span class="at">kegg.native=</span><span class="cn">TRUE</span>)</span>
<span id="cb182-3744"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3744" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb182-3745"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3745" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-3746"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3746" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3747"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3747" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( outfile )</span>
<span id="cb182-3748"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3748" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3749"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3749" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3750"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3750" aria-hidden="true" tabindex="-1"></a><span class="co"># list of pathways with details</span></span>
<span id="cb182-3751"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3751" aria-hidden="true" tabindex="-1"></a>pathwayListData  <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb182-3752"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3752" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3753"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3753" aria-hidden="true" tabindex="-1"></a>    pathways <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb182-3754"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3754" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_pathwayMethod <span class="sc">==</span> <span class="dv">1</span>)  </span>
<span id="cb182-3755"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3755" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(gagePathwayData.out)) </span>
<span id="cb182-3756"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3756" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(gagePathwayData.out)[<span class="dv">2</span>] <span class="sc">&gt;</span><span class="dv">1</span>) { </span>
<span id="cb182-3757"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3757" aria-hidden="true" tabindex="-1"></a>                pathways <span class="ot">&lt;-</span> gagePathwayData.out</span>
<span id="cb182-3758"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3758" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(pathways)[<span class="dv">2</span>] <span class="ot">=</span><span class="st">&quot;Pathways&quot;</span>;  </span>
<span id="cb182-3759"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3759" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(pathways)[<span class="dv">4</span>] <span class="ot">=</span><span class="st">&quot;nGenes&quot;</span>; </span>
<span id="cb182-3760"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3760" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-3761"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3761" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_pathwayMethod <span class="sc">==</span> <span class="dv">3</span>) </span>
<span id="cb182-3762"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3762" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(fgseaPathwayData.out)) </span>
<span id="cb182-3763"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3763" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(fgseaPathwayData.out)[<span class="dv">2</span>] <span class="sc">&gt;</span><span class="dv">1</span>) {</span>
<span id="cb182-3764"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3764" aria-hidden="true" tabindex="-1"></a>                pathways <span class="ot">&lt;-</span> fgseaPathwayData.out</span>
<span id="cb182-3765"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3765" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(pathways)[<span class="dv">2</span>] <span class="ot">=</span><span class="st">&quot;Pathways&quot;</span>;  </span>
<span id="cb182-3766"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3766" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(pathways)[<span class="dv">4</span>] <span class="ot">=</span><span class="st">&quot;nGenes&quot;</span>; </span>
<span id="cb182-3767"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3767" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-3768"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3768" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3769"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3769" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_pathwayMethod <span class="sc">==</span> <span class="dv">2</span>) </span>
<span id="cb182-3770"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3770" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(PGSEAplot.data.out)) </span>
<span id="cb182-3771"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3771" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(PGSEAplot.data.out)[<span class="dv">2</span>] <span class="sc">&gt;</span><span class="dv">1</span>) {</span>
<span id="cb182-3772"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3772" aria-hidden="true" tabindex="-1"></a>                pathways <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>( PGSEAplot.data.out)</span>
<span id="cb182-3773"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3773" aria-hidden="true" tabindex="-1"></a>                pathways<span class="sc">$</span>Pathways <span class="ot">=</span> <span class="fu">substr</span>(<span class="fu">rownames</span>(pathways),<span class="dv">10</span>, <span class="fu">nchar</span>( <span class="fu">rownames</span>(pathways)) )</span>
<span id="cb182-3774"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3774" aria-hidden="true" tabindex="-1"></a>                pathways<span class="sc">$</span>adj.Pval <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; .*&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(pathways))</span>
<span id="cb182-3775"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3775" aria-hidden="true" tabindex="-1"></a>                pathways<span class="sc">$</span>Direction <span class="ot">=</span><span class="st">&quot;Diff&quot;</span></span>
<span id="cb182-3776"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3776" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-3777"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3777" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb182-3778"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3778" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_pathwayMethod <span class="sc">==</span> <span class="dv">4</span>) </span>
<span id="cb182-3779"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3779" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(PGSEAplotAllSamples.data.out)) </span>
<span id="cb182-3780"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3780" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(PGSEAplotAllSamples.data.out)[<span class="dv">2</span>] <span class="sc">&gt;</span><span class="dv">1</span>) {</span>
<span id="cb182-3781"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3781" aria-hidden="true" tabindex="-1"></a>                pathways <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>( PGSEAplotAllSamples.data.out)</span>
<span id="cb182-3782"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3782" aria-hidden="true" tabindex="-1"></a>                pathways<span class="sc">$</span>Pathways <span class="ot">=</span> <span class="fu">substr</span>(<span class="fu">rownames</span>(pathways),<span class="dv">10</span>, <span class="fu">nchar</span>( <span class="fu">rownames</span>(pathways)) )</span>
<span id="cb182-3783"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3783" aria-hidden="true" tabindex="-1"></a>                pathways<span class="sc">$</span>adj.Pval <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; .*&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(pathways))</span>
<span id="cb182-3784"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3784" aria-hidden="true" tabindex="-1"></a>                pathways<span class="sc">$</span>Direction <span class="ot">=</span><span class="st">&quot;Diff&quot;</span></span>
<span id="cb182-3785"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3785" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-3786"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3786" aria-hidden="true" tabindex="-1"></a>                }           </span>
<span id="cb182-3787"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3787" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">is.null</span>( pathways) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-3788"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3788" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3789"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3789" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if no gene set data, return pathway list</span></span>
<span id="cb182-3790"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3790" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.null</span>(GeneSets.out ) ) <span class="fu">return</span>(pathways) </span>
<span id="cb182-3791"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3791" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3792"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3792" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3793"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3793" aria-hidden="true" tabindex="-1"></a>    pathways<span class="sc">$</span>adj.Pval <span class="ot">=</span> <span class="fu">as.numeric</span>(pathways<span class="sc">$</span>adj.Pval)</span>
<span id="cb182-3794"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3794" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3795"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3795" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">nrow</span>(pathways)<span class="sc">&gt;</span><span class="dv">1</span>)</span>
<span id="cb182-3796"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3796" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">nrow</span>(pathways) )</span>
<span id="cb182-3797"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3797" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">nchar</span>(pathways<span class="sc">$</span>Direction[i]) <span class="sc">&lt;=</span><span class="dv">1</span>)</span>
<span id="cb182-3798"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3798" aria-hidden="true" tabindex="-1"></a>            pathways<span class="sc">$</span>Direction[i] <span class="ot">=</span> pathways<span class="sc">$</span>Direction[i<span class="dv">-1</span>]</span>
<span id="cb182-3799"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3799" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-3800"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3800" aria-hidden="true" tabindex="-1"></a>    <span class="co"># gene symbol matching symbols </span></span>
<span id="cb182-3801"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3801" aria-hidden="true" tabindex="-1"></a>    probeToGene <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb182-3802"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3802" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( input_selectGO <span class="sc">!=</span> <span class="st">&quot;ID not recognized!&quot;</span> <span class="sc">&amp;</span> input_selectOrg <span class="sc">!=</span> <span class="st">&quot;NEW&quot;</span>)</span>
<span id="cb182-3803"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3803" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">sum</span>(<span class="fu">is.na</span>( allGeneInfo.out<span class="sc">$</span>symbol ) )<span class="sc">/</span> <span class="fu">dim</span>( allGeneInfo.out )[<span class="dv">1</span>] <span class="sc">&lt;</span>.<span class="dv">5</span> ) { <span class="co"># if more than 50% genes has symbol</span></span>
<span id="cb182-3804"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3804" aria-hidden="true" tabindex="-1"></a>        probeToGene <span class="ot">=</span> allGeneInfo.out[,<span class="fu">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="st">&quot;symbol&quot;</span>)]</span>
<span id="cb182-3805"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3805" aria-hidden="true" tabindex="-1"></a>        probeToGene<span class="sc">$</span>symbol <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>,<span class="st">&quot;&quot;</span>,probeToGene<span class="sc">$</span>symbol)</span>
<span id="cb182-3806"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3806" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3807"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3807" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">=</span> <span class="fu">which</span>( <span class="fu">is.na</span>(probeToGene<span class="sc">$</span>symbol) <span class="sc">|</span></span>
<span id="cb182-3808"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3808" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">nchar</span>(probeToGene<span class="sc">$</span>symbol)<span class="sc">&lt;</span><span class="dv">2</span> <span class="sc">|</span> </span>
<span id="cb182-3809"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3809" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">toupper</span>(probeToGene<span class="sc">$</span>symbol)<span class="sc">==</span><span class="st">&quot;NA&quot;</span> <span class="sc">|</span>  </span>
<span id="cb182-3810"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3810" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">toupper</span>(probeToGene<span class="sc">$</span>symbol)<span class="sc">==</span><span class="st">&quot;0&quot;</span>  )             </span>
<span id="cb182-3811"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3811" aria-hidden="true" tabindex="-1"></a>        probeToGene[ix,<span class="dv">2</span>] <span class="ot">=</span> probeToGene[ix,<span class="dv">1</span>]  <span class="co"># use gene ID</span></span>
<span id="cb182-3812"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3812" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3813"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3813" aria-hidden="true" tabindex="-1"></a>    }       </span>
<span id="cb182-3814"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3814" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3815"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3815" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3816"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3816" aria-hidden="true" tabindex="-1"></a>    pathways<span class="sc">$</span>Genes <span class="ot">=</span><span class="st">&quot;&quot;</span></span>
<span id="cb182-3817"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3817" aria-hidden="true" tabindex="-1"></a>    <span class="co"># looking up genes for each pathway</span></span>
<span id="cb182-3818"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3818" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(pathways) ){ </span>
<span id="cb182-3819"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3819" aria-hidden="true" tabindex="-1"></a>        ix <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">names</span>(GeneSets.out ) <span class="sc">==</span> pathways<span class="sc">$</span>Pathways[i]   ) <span class="co"># find the gene set</span></span>
<span id="cb182-3820"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3820" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">length</span>(ix) <span class="sc">!=</span> <span class="dv">0</span> ) { </span>
<span id="cb182-3821"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3821" aria-hidden="true" tabindex="-1"></a>            genes <span class="ot">&lt;-</span> GeneSets.out[[ix]] <span class="co"># retrieve genes</span></span>
<span id="cb182-3822"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3822" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-3823"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3823" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(probeToGene) ) { </span>
<span id="cb182-3824"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3824" aria-hidden="true" tabindex="-1"></a>                iy <span class="ot">=</span> <span class="fu">match</span>(genes,probeToGene[,<span class="dv">1</span>])</span>
<span id="cb182-3825"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3825" aria-hidden="true" tabindex="-1"></a>                genes <span class="ot">=</span> probeToGene[iy,<span class="dv">2</span>]</span>
<span id="cb182-3826"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3826" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-3827"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3827" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-3828"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3828" aria-hidden="true" tabindex="-1"></a>            pathways<span class="sc">$</span>Genes[i] <span class="ot">=</span> <span class="fu">paste</span>(genes, <span class="at">collapse=</span><span class="st">&quot; &quot;</span>)</span>
<span id="cb182-3829"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3829" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-3830"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3830" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-3831"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3831" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(pathways)</span>
<span id="cb182-3832"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3832" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3833"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3833" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3834"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3834" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3835"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3835" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3836"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3836" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-3837"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3837" aria-hidden="true" tabindex="-1"></a><span class="co"># Genome-wide view</span></span>
<span id="cb182-3838"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3838" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-3839"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3839" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb182-3840"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3840" aria-hidden="true" tabindex="-1"></a>genomePlotly <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb182-3841"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3841" aria-hidden="true" tabindex="-1"></a>        <span class="co"># default plot</span></span>
<span id="cb182-3842"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3842" aria-hidden="true" tabindex="-1"></a>        fake <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">a=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="at">b=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)</span>
<span id="cb182-3843"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3843" aria-hidden="true" tabindex="-1"></a>        p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(fake, <span class="fu">aes</span>(<span class="at">x =</span> a, <span class="at">y =</span> b)) <span class="sc">+</span></span>
<span id="cb182-3844"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3844" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">geom_blank</span>() <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;No genes with position info.&quot;</span>) <span class="sc">+</span></span>
<span id="cb182-3845"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3845" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">theme</span>(<span class="at">axis.title.x=</span><span class="fu">element_blank</span>(),<span class="at">axis.title.y=</span><span class="fu">element_blank</span>())</span>
<span id="cb182-3846"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3846" aria-hidden="true" tabindex="-1"></a>                             </span>
<span id="cb182-3847"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3847" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  {</span>
<span id="cb182-3848"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3848" aria-hidden="true" tabindex="-1"></a>            top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb182-3849"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3849" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb182-3850"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3850" aria-hidden="true" tabindex="-1"></a>          top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb182-3851"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3851" aria-hidden="true" tabindex="-1"></a>          ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast2, <span class="fu">names</span>(top))</span>
<span id="cb182-3852"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3852" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (<span class="fu">ggplotly</span>(p))</span>
<span id="cb182-3853"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3853" aria-hidden="true" tabindex="-1"></a>          top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb182-3854"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3854" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-3855"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3855" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (<span class="fu">ggplotly</span>(p))</span>
<span id="cb182-3856"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3856" aria-hidden="true" tabindex="-1"></a>          <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb182-3857"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3857" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb182-3858"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3858" aria-hidden="true" tabindex="-1"></a>         <span class="co"># write.csv(merge(top1,allGeneInfo(), by.x=&quot;row.names&quot;,by.y=&quot;ensembl_gene_id&quot;  ),&quot;tem.csv&quot;  )</span></span>
<span id="cb182-3859"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3859" aria-hidden="true" tabindex="-1"></a>         x <span class="ot">&lt;-</span> <span class="fu">merge</span>(top1,allGeneInfo.out, <span class="at">by.x=</span><span class="st">&quot;row.names&quot;</span>,<span class="at">by.y=</span><span class="st">&quot;ensembl_gene_id&quot;</span>  )</span>
<span id="cb182-3860"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3860" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3861"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3861" aria-hidden="true" tabindex="-1"></a>         <span class="co"># if no chromosomes found. For example if user do not convert gene IDs.</span></span>
<span id="cb182-3862"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3862" aria-hidden="true" tabindex="-1"></a>         <span class="cf">if</span>( <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="sc">&gt;</span><span class="dv">5</span>  ) { </span>
<span id="cb182-3863"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3863" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-3864"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3864" aria-hidden="true" tabindex="-1"></a>             x <span class="ot">&lt;-</span> x[<span class="fu">order</span>(x<span class="sc">$</span>chromosome_name,x<span class="sc">$</span>start_position),]</span>
<span id="cb182-3865"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3865" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-3866"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3866" aria-hidden="true" tabindex="-1"></a>             tem <span class="ot">=</span> <span class="fu">sort</span>( <span class="fu">table</span>( x<span class="sc">$</span>chromosome_name), <span class="at">decreasing=</span>T)</span>
<span id="cb182-3867"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3867" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3868"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3868" aria-hidden="true" tabindex="-1"></a>             chromosomes <span class="ot">&lt;-</span> <span class="fu">names</span>( tem[tem <span class="sc">&gt;=</span> <span class="dv">5</span> ] )  <span class="co"># chromosomes with less than 100 genes are excluded</span></span>
<span id="cb182-3869"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3869" aria-hidden="true" tabindex="-1"></a>             <span class="cf">if</span>(<span class="fu">length</span>(chromosomes) <span class="sc">&gt;</span> <span class="dv">50</span>) chromosomes <span class="ot">&lt;-</span> chromosomes[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]  <span class="co"># at most 50 chromosomes</span></span>
<span id="cb182-3870"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3870" aria-hidden="true" tabindex="-1"></a>             chromosomes <span class="ot">&lt;-</span> chromosomes[ <span class="fu">nchar</span>(chromosomes)<span class="sc">&lt;=</span><span class="dv">12</span>] <span class="co"># chr. name less than 10 characters</span></span>
<span id="cb182-3871"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3871" aria-hidden="true" tabindex="-1"></a>             chromosomes <span class="ot">=</span> chromosomes[<span class="fu">order</span>(<span class="fu">as.numeric</span>(chromosomes) ) ]</span>
<span id="cb182-3872"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3872" aria-hidden="true" tabindex="-1"></a>             <span class="co"># chromosomes = chromosomes[!is.na(as.numeric(chromosomes) ) ]</span></span>
<span id="cb182-3873"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3873" aria-hidden="true" tabindex="-1"></a>             chromosomesNumbers <span class="ot">=</span> <span class="fu">as.numeric</span>(chromosomes)</span>
<span id="cb182-3874"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3874" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3875"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3875" aria-hidden="true" tabindex="-1"></a>             <span class="co"># convert chr.x to numbers      </span></span>
<span id="cb182-3876"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3876" aria-hidden="true" tabindex="-1"></a>              j <span class="ot">=</span> <span class="fu">max</span>( chromosomesNumbers,<span class="at">na.rm=</span>T) </span>
<span id="cb182-3877"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3877" aria-hidden="true" tabindex="-1"></a>              <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>( chromosomes)) {</span>
<span id="cb182-3878"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3878" aria-hidden="true" tabindex="-1"></a>               <span class="cf">if</span> ( <span class="fu">is.na</span>(chromosomesNumbers[i]) ) </span>
<span id="cb182-3879"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3879" aria-hidden="true" tabindex="-1"></a>               { chromosomesNumbers[i] <span class="ot">&lt;-</span> j<span class="sc">+</span><span class="dv">1</span>; j <span class="ot">&lt;-</span> j<span class="sc">+</span><span class="dv">1</span>; }</span>
<span id="cb182-3880"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3880" aria-hidden="true" tabindex="-1"></a>             }</span>
<span id="cb182-3881"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3881" aria-hidden="true" tabindex="-1"></a>              </span>
<span id="cb182-3882"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3882" aria-hidden="true" tabindex="-1"></a>             x <span class="ot">&lt;-</span> x[<span class="fu">which</span>(x<span class="sc">$</span>chromosome_name <span class="sc">%in%</span> chromosomes   ),]</span>
<span id="cb182-3883"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3883" aria-hidden="true" tabindex="-1"></a>             x <span class="ot">&lt;-</span> <span class="fu">droplevels</span>(x)</span>
<span id="cb182-3884"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3884" aria-hidden="true" tabindex="-1"></a>             </span>
<span id="cb182-3885"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3885" aria-hidden="true" tabindex="-1"></a>            <span class="co"># find the number coding for chromosome </span></span>
<span id="cb182-3886"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3886" aria-hidden="true" tabindex="-1"></a>             getChrNumber <span class="ot">&lt;-</span> <span class="cf">function</span> (chrName){</span>
<span id="cb182-3887"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3887" aria-hidden="true" tabindex="-1"></a>             <span class="fu">return</span>( chromosomesNumbers[ <span class="fu">which</span>( chromosomes <span class="sc">==</span> chrName)] )</span>
<span id="cb182-3888"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3888" aria-hidden="true" tabindex="-1"></a>             }</span>
<span id="cb182-3889"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3889" aria-hidden="true" tabindex="-1"></a>              x<span class="sc">$</span>chrNum <span class="ot">=</span> <span class="dv">1</span> <span class="co"># numeric coding</span></span>
<span id="cb182-3890"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3890" aria-hidden="true" tabindex="-1"></a>              x<span class="sc">$</span>chrNum <span class="ot">&lt;-</span> <span class="fu">unlist</span>( <span class="fu">lapply</span>( x<span class="sc">$</span>chromosome_name, getChrNumber) )</span>
<span id="cb182-3891"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3891" aria-hidden="true" tabindex="-1"></a>             </span>
<span id="cb182-3892"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3892" aria-hidden="true" tabindex="-1"></a>             x<span class="sc">$</span>Row.names <span class="ot">&lt;-</span> <span class="fu">as.character</span>( x<span class="sc">$</span>Row.names)</span>
<span id="cb182-3893"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3893" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb182-3894"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3894" aria-hidden="true" tabindex="-1"></a>             <span class="co"># if symbol is missing use Ensembl id</span></span>
<span id="cb182-3895"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3895" aria-hidden="true" tabindex="-1"></a>             x<span class="sc">$</span>symbol <span class="ot">=</span> <span class="fu">as.character</span>(x<span class="sc">$</span>symbol)   </span>
<span id="cb182-3896"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3896" aria-hidden="true" tabindex="-1"></a>             ix <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">is.na</span>(x<span class="sc">$</span>symbol))</span>
<span id="cb182-3897"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3897" aria-hidden="true" tabindex="-1"></a>             ix2 <span class="ot">=</span> <span class="fu">which</span>(<span class="fu">nchar</span>(<span class="fu">as.character</span>(x<span class="sc">$</span>symbol))<span class="sc">&lt;=</span> <span class="dv">2</span> )</span>
<span id="cb182-3898"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3898" aria-hidden="true" tabindex="-1"></a>             ix3 <span class="ot">=</span> <span class="fu">which</span>( <span class="fu">duplicated</span>(x<span class="sc">$</span>symbol))</span>
<span id="cb182-3899"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3899" aria-hidden="true" tabindex="-1"></a>             ix <span class="ot">=</span> <span class="fu">unique</span>( <span class="fu">c</span>(ix,ix2,ix3))</span>
<span id="cb182-3900"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3900" aria-hidden="true" tabindex="-1"></a>             x<span class="sc">$</span>symbol[ix] <span class="ot">&lt;-</span> x<span class="sc">$</span>Row.names[ix] </span>
<span id="cb182-3901"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3901" aria-hidden="true" tabindex="-1"></a>                         </span>
<span id="cb182-3902"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3902" aria-hidden="true" tabindex="-1"></a>         </span>
<span id="cb182-3903"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3903" aria-hidden="true" tabindex="-1"></a>             <span class="do">##################################</span></span>
<span id="cb182-3904"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3904" aria-hidden="true" tabindex="-1"></a>             <span class="co"># plotting</span></span>
<span id="cb182-3905"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3905" aria-hidden="true" tabindex="-1"></a>            <span class="co"># x = read.csv(&quot;tem_genome.csv&quot;)</span></span>
<span id="cb182-3906"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3906" aria-hidden="true" tabindex="-1"></a>            x <span class="ot">=</span> x[<span class="sc">!</span><span class="fu">is.na</span>(x<span class="sc">$</span>chromosome_name),]</span>
<span id="cb182-3907"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3907" aria-hidden="true" tabindex="-1"></a>            x <span class="ot">=</span> x[<span class="sc">!</span><span class="fu">is.na</span>(x<span class="sc">$</span>start_position),]    </span>
<span id="cb182-3908"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3908" aria-hidden="true" tabindex="-1"></a>            <span class="co"># only keep significant genes</span></span>
<span id="cb182-3909"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3909" aria-hidden="true" tabindex="-1"></a>            <span class="co">#x = x[which(x$FDR&lt;input_limmaPval),]</span></span>
<span id="cb182-3910"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3910" aria-hidden="true" tabindex="-1"></a>            <span class="co"># x = x[which(abs(x$Fold) &gt; log2( input_limmaFC)),]</span></span>
<span id="cb182-3911"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3911" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-3912"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3912" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">which</span>( (x<span class="sc">$</span>FDR<span class="sc">&lt;</span> <span class="fu">as.numeric</span>(input_limmaPvalViz)) <span class="sc">&amp;</span> (<span class="fu">abs</span>(x<span class="sc">$</span>Fold) <span class="sc">&gt;</span> <span class="fu">as.numeric</span>(input_limmaFCViz) ) )</span>
<span id="cb182-3913"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3913" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-3914"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3914" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (<span class="fu">length</span>(ix) <span class="sc">&gt;</span> <span class="dv">5</span>) { </span>
<span id="cb182-3915"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3915" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3916"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3916" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">=</span> x[ix,]      </span>
<span id="cb182-3917"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3917" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-3918"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3918" aria-hidden="true" tabindex="-1"></a>                x<span class="sc">$</span>start_position <span class="ot">=</span> x<span class="sc">$</span>start_position<span class="sc">/</span><span class="dv">1000000</span> <span class="co"># Mbp</span></span>
<span id="cb182-3919"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3919" aria-hidden="true" tabindex="-1"></a>                chrD <span class="ot">=</span> <span class="dv">20</span> <span class="co"># distance between chrs.</span></span>
<span id="cb182-3920"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3920" aria-hidden="true" tabindex="-1"></a>                foldCutoff <span class="ot">=</span> <span class="dv">3</span>   <span class="co"># max log2 fold </span></span>
<span id="cb182-3921"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3921" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-3922"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3922" aria-hidden="true" tabindex="-1"></a>                x<span class="sc">$</span>Fold <span class="ot">=</span> x<span class="sc">$</span>Fold <span class="sc">/</span> <span class="fu">sd</span>(x<span class="sc">$</span>Fold)  <span class="co"># standardize fold change</span></span>
<span id="cb182-3923"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3923" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-3924"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3924" aria-hidden="true" tabindex="-1"></a>                x<span class="sc">$</span>Fold[<span class="fu">which</span>(x<span class="sc">$</span>Fold <span class="sc">&gt;</span> foldCutoff )] <span class="ot">=</span> foldCutoff   <span class="co"># log2fold within -5 to 5</span></span>
<span id="cb182-3925"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3925" aria-hidden="true" tabindex="-1"></a>                x<span class="sc">$</span>Fold[<span class="fu">which</span>(x<span class="sc">$</span>Fold <span class="sc">&lt;</span>   <span class="sc">-</span><span class="dv">1</span><span class="sc">*</span>foldCutoff )] <span class="ot">=</span> <span class="sc">-</span><span class="dv">1</span><span class="sc">*</span>foldCutoff </span>
<span id="cb182-3926"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3926" aria-hidden="true" tabindex="-1"></a>                x<span class="sc">$</span>Fold <span class="ot">=</span> <span class="dv">4</span><span class="sc">*</span> x<span class="sc">$</span>Fold</span>
<span id="cb182-3927"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3927" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-3928"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3928" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3929"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3929" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-3930"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3930" aria-hidden="true" tabindex="-1"></a>                x<span class="sc">$</span>y <span class="ot">=</span> x<span class="sc">$</span>chrNum<span class="sc">*</span>chrD <span class="sc">+</span> x<span class="sc">$</span>Fold</span>
<span id="cb182-3931"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3931" aria-hidden="true" tabindex="-1"></a>                chrLengthTable <span class="ot">=</span> <span class="fu">aggregate</span>(start_position<span class="sc">~</span>chrNum, <span class="at">data=</span>x,max )</span>
<span id="cb182-3932"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3932" aria-hidden="true" tabindex="-1"></a>                chrTotal <span class="ot">=</span> <span class="fu">dim</span>(chrLengthTable)[<span class="dv">1</span>]   </span>
<span id="cb182-3933"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3933" aria-hidden="true" tabindex="-1"></a>                x<span class="sc">$</span>R <span class="ot">=</span> <span class="fu">as.factor</span>(<span class="fu">sign</span>(x<span class="sc">$</span>Fold))</span>
<span id="cb182-3934"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3934" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-3935"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3935" aria-hidden="true" tabindex="-1"></a>                <span class="fu">colnames</span>(x)[ <span class="fu">which</span>(<span class="fu">colnames</span>(x) <span class="sc">==</span> <span class="st">&quot;start_position&quot;</span>)] <span class="ot">=</span> <span class="st">&quot;x&quot;</span></span>
<span id="cb182-3936"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3936" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3937"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3937" aria-hidden="true" tabindex="-1"></a>                p<span class="ot">=</span> <span class="fu">ggplot</span>(x, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">colour =</span> R, <span class="at">text =</span> symbol ) ) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">20</span>, <span class="at">size =</span> .<span class="dv">2</span>)</span>
<span id="cb182-3938"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3938" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-3939"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3939" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#label y with chr names</span></span>
<span id="cb182-3940"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3940" aria-hidden="true" tabindex="-1"></a>                p <span class="ot">&lt;-</span> p <span class="sc">+</span>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> <span class="fu">paste</span>(<span class="st">&quot;chr&quot;</span>,chromosomes[chrLengthTable<span class="sc">$</span>chrNum],<span class="at">sep=</span><span class="st">&quot;&quot;</span>), <span class="at">breaks =</span> chrD<span class="sc">*</span> (<span class="dv">1</span><span class="sc">:</span>chrTotal), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, </span>
<span id="cb182-3941"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3941" aria-hidden="true" tabindex="-1"></a>                    chrD<span class="sc">*</span>chrTotal <span class="sc">+</span> <span class="dv">5</span>) )</span>
<span id="cb182-3942"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3942" aria-hidden="true" tabindex="-1"></a>                <span class="co"># draw horizontal lines for each chr.</span></span>
<span id="cb182-3943"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3943" aria-hidden="true" tabindex="-1"></a>                <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(chrLengthTable)[<span class="dv">1</span>] )</span>
<span id="cb182-3944"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3944" aria-hidden="true" tabindex="-1"></a>                    p <span class="ot">=</span> p<span class="sc">+</span> <span class="fu">annotate</span>( <span class="st">&quot;segment&quot;</span>,<span class="at">x =</span> <span class="dv">0</span>, <span class="at">xend =</span> chrLengthTable<span class="sc">$</span>start_position[i],</span>
<span id="cb182-3945"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3945" aria-hidden="true" tabindex="-1"></a>                        <span class="at">y =</span> chrLengthTable<span class="sc">$</span>chrNum[i]<span class="sc">*</span>chrD, <span class="at">yend =</span> chrLengthTable<span class="sc">$</span>chrNum[i]<span class="sc">*</span>chrD)</span>
<span id="cb182-3946"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3946" aria-hidden="true" tabindex="-1"></a>                <span class="co"># change legend     http://ggplot2.tidyverse.org/reference/scale_manual.html</span></span>
<span id="cb182-3947"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3947" aria-hidden="true" tabindex="-1"></a>                p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">scale_colour_manual</span>(<span class="at">name=</span><span class="st">&quot;&quot;</span>,   <span class="co"># customize legend text</span></span>
<span id="cb182-3948"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3948" aria-hidden="true" tabindex="-1"></a>                    <span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb182-3949"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3949" aria-hidden="true" tabindex="-1"></a>                    <span class="at">breaks=</span><span class="fu">c</span>(<span class="st">&quot;1&quot;</span>,<span class="st">&quot;-1&quot;</span>),</span>
<span id="cb182-3950"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3950" aria-hidden="true" tabindex="-1"></a>                    <span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;Up&quot;</span>, <span class="st">&quot;Dn&quot;</span>))   </span>
<span id="cb182-3951"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3951" aria-hidden="true" tabindex="-1"></a>                p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">&quot;Position on chrs. (Mbp)&quot;</span>) <span class="sc">+</span>    <span class="fu">theme</span>(<span class="at">axis.title.y=</span><span class="fu">element_blank</span>())                     </span>
<span id="cb182-3952"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3952" aria-hidden="true" tabindex="-1"></a>                p<span class="ot">=</span> p <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb182-3953"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3953" aria-hidden="true" tabindex="-1"></a>                <span class="co"># p &lt;- ggplot(mtcars, aes(x=hp, y=mpg)) + geom_point(shape=20, size=17)</span></span>
<span id="cb182-3954"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3954" aria-hidden="true" tabindex="-1"></a>               <span class="co"># p=p+ geom_smooth(method = &quot;lm&quot;,se=FALSE)</span></span>
<span id="cb182-3955"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3955" aria-hidden="true" tabindex="-1"></a>                <span class="co">#p+ geom_line(aes(y=rollmean(y,7, na.pad=TRUE)))</span></span>
<span id="cb182-3956"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3956" aria-hidden="true" tabindex="-1"></a>              </span>
<span id="cb182-3957"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3957" aria-hidden="true" tabindex="-1"></a>               <span class="co"># Customize hover text https://cran.r-project.org/web/packages/plotly/plotly.pdf</span></span>
<span id="cb182-3958"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3958" aria-hidden="true" tabindex="-1"></a>               <span class="co"># style(ggplotly(p),hoverinfo=&quot;text&quot;)  # not working</span></span>
<span id="cb182-3959"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3959" aria-hidden="true" tabindex="-1"></a>              }  <span class="co"># if no genes else</span></span>
<span id="cb182-3960"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3960" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-3961"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3961" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-3962"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3962" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ggplotly</span>(p)</span>
<span id="cb182-3963"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3963" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-3964"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3964" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3965"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3965" aria-hidden="true" tabindex="-1"></a><span class="co"># results from PREDA</span></span>
<span id="cb182-3966"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3966" aria-hidden="true" tabindex="-1"></a>genomePlotData <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb182-3967"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3967" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(input_selectContrast2 ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-3968"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3968" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3969"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3969" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">length</span>(limma.out<span class="sc">$</span>topGenes) <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-3970"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3970" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-3971"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3971" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb182-3972"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3972" aria-hidden="true" tabindex="-1"></a>    { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb182-3973"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3973" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb182-3974"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3974" aria-hidden="true" tabindex="-1"></a>      top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb182-3975"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3975" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast2, <span class="fu">names</span>(top))</span>
<span id="cb182-3976"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3976" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb182-3977"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3977" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb182-3978"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3978" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb182-3979"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3979" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb182-3980"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3980" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb182-3981"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3981" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb182-3982"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3982" aria-hidden="true" tabindex="-1"></a>     <span class="co"># write.csv(merge(top1,allGeneInfo(), by.x=&quot;row.names&quot;,by.y=&quot;ensembl_gene_id&quot;  ),&quot;tem.csv&quot;  )</span></span>
<span id="cb182-3983"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3983" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> <span class="fu">merge</span>(top1,allGeneInfo.out, <span class="at">by.x=</span><span class="st">&quot;row.names&quot;</span>,<span class="at">by.y=</span><span class="st">&quot;ensembl_gene_id&quot;</span>  )</span>
<span id="cb182-3984"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3984" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-3985"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3985" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> x[<span class="fu">order</span>(x<span class="sc">$</span>chromosome_name,x<span class="sc">$</span>start_position),]</span>
<span id="cb182-3986"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3986" aria-hidden="true" tabindex="-1"></a>     tem <span class="ot">=</span> <span class="fu">sort</span>( <span class="fu">table</span>( x<span class="sc">$</span>chromosome_name), <span class="at">decreasing=</span>T) </span>
<span id="cb182-3987"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3987" aria-hidden="true" tabindex="-1"></a>     chromosomes <span class="ot">&lt;-</span> <span class="fu">names</span>( tem[tem <span class="sc">&gt;</span> <span class="dv">100</span> ] )  <span class="co"># chromosomes with less than 100 genes are excluded</span></span>
<span id="cb182-3988"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3988" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>(<span class="fu">length</span>(chromosomes) <span class="sc">&gt;</span> <span class="dv">50</span>) chromosomes <span class="ot">&lt;-</span> chromosomes[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]  <span class="co"># at most 50 chromosomes</span></span>
<span id="cb182-3989"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3989" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-3990"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3990" aria-hidden="true" tabindex="-1"></a>     chromosomes <span class="ot">=</span> chromosomes[<span class="fu">order</span>(<span class="fu">as.numeric</span>(chromosomes) ) ]</span>
<span id="cb182-3991"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3991" aria-hidden="true" tabindex="-1"></a>     <span class="co"># chromosomes = chromosomes[!is.na(as.numeric(chromosomes) ) ]</span></span>
<span id="cb182-3992"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3992" aria-hidden="true" tabindex="-1"></a>     chromosomesNumbers <span class="ot">=</span> <span class="fu">as.numeric</span>(chromosomes)</span>
<span id="cb182-3993"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3993" aria-hidden="true" tabindex="-1"></a>     <span class="co"># convert chr.x to numbers</span></span>
<span id="cb182-3994"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3994" aria-hidden="true" tabindex="-1"></a>      j <span class="ot">=</span> <span class="fu">max</span>( chromosomesNumbers,<span class="at">na.rm=</span>T) </span>
<span id="cb182-3995"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3995" aria-hidden="true" tabindex="-1"></a>      <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>( chromosomes)) {</span>
<span id="cb182-3996"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3996" aria-hidden="true" tabindex="-1"></a>       <span class="cf">if</span> ( <span class="fu">is.na</span>(chromosomesNumbers[i]) ) </span>
<span id="cb182-3997"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3997" aria-hidden="true" tabindex="-1"></a>       { chromosomesNumbers[i] <span class="ot">&lt;-</span> j<span class="sc">+</span><span class="dv">1</span>; j <span class="ot">&lt;-</span> j<span class="sc">+</span><span class="dv">1</span>; }</span>
<span id="cb182-3998"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3998" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb182-3999"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-3999" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb182-4000"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4000" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> x[<span class="fu">which</span>(x<span class="sc">$</span>chromosome_name <span class="sc">%in%</span> chromosomes   ),]</span>
<span id="cb182-4001"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4001" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> <span class="fu">droplevels</span>(x)</span>
<span id="cb182-4002"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4002" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-4003"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4003" aria-hidden="true" tabindex="-1"></a>    <span class="co"># find the number coding for chromosome </span></span>
<span id="cb182-4004"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4004" aria-hidden="true" tabindex="-1"></a>     getChrNumber <span class="ot">&lt;-</span> <span class="cf">function</span> (chrName){</span>
<span id="cb182-4005"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4005" aria-hidden="true" tabindex="-1"></a>     <span class="fu">return</span>( chromosomesNumbers[ <span class="fu">which</span>( chromosomes <span class="sc">==</span> chrName)] )</span>
<span id="cb182-4006"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4006" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb182-4007"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4007" aria-hidden="true" tabindex="-1"></a>      x<span class="sc">$</span>chrNum <span class="ot">=</span> <span class="dv">1</span> <span class="co"># numeric coding</span></span>
<span id="cb182-4008"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4008" aria-hidden="true" tabindex="-1"></a>      x<span class="sc">$</span>chrNum <span class="ot">&lt;-</span> <span class="fu">unlist</span>( <span class="fu">lapply</span>( x<span class="sc">$</span>chromosome_name, getChrNumber) )</span>
<span id="cb182-4009"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4009" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-4010"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4010" aria-hidden="true" tabindex="-1"></a>     x<span class="sc">$</span>Row.names <span class="ot">&lt;-</span> <span class="fu">as.character</span>( x<span class="sc">$</span>Row.names)</span>
<span id="cb182-4011"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4011" aria-hidden="true" tabindex="-1"></a>     fold <span class="ot">=</span> x<span class="sc">$</span>Fold; <span class="fu">names</span>(fold) <span class="ot">=</span> x<span class="sc">$</span>Row.names</span>
<span id="cb182-4012"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4012" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-4013"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4013" aria-hidden="true" tabindex="-1"></a>    <span class="co"># write.csv(x,&quot;tem.csv&quot;)</span></span>
<span id="cb182-4014"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4014" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> x[<span class="sc">!</span><span class="fu">duplicated</span>(x<span class="sc">$</span>Row.names),]</span>
<span id="cb182-4015"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4015" aria-hidden="true" tabindex="-1"></a>     <span class="co">#rownames(x) = x$Row.names</span></span>
<span id="cb182-4016"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4016" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-4017"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4017" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4018"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4018" aria-hidden="true" tabindex="-1"></a>    GEanalysisResults <span class="ot">&lt;-</span> genomePlotDataPre.out; </span>
<span id="cb182-4019"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4019" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-4020"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4020" aria-hidden="true" tabindex="-1"></a>    genomic_regions_UP<span class="ot">&lt;-</span><span class="fu">PREDAResults2GenomicRegions</span>(GEanalysisResults, <span class="at">qval.threshold=</span>input_RegionsPvalCutoff, <span class="at">smoothStatistic.tail=</span><span class="st">&quot;upper&quot;</span>, <span class="at">smoothStatistic.threshold=</span> input_StatisticCutoff)</span>
<span id="cb182-4021"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4021" aria-hidden="true" tabindex="-1"></a>    genomic_regions_DOWN<span class="ot">&lt;-</span><span class="fu">PREDAResults2GenomicRegions</span>(GEanalysisResults, <span class="at">qval.threshold=</span>input_RegionsPvalCutoff, <span class="at">smoothStatistic.tail=</span><span class="st">&quot;lower&quot;</span>, <span class="at">smoothStatistic.threshold=</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">*</span>input_StatisticCutoff)</span>
<span id="cb182-4022"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4022" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4023"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4023" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">is.null</span>(genomic_regions_UP<span class="sc">$</span>Test) <span class="sc">&amp;&amp;</span>  <span class="fu">is.null</span>(genomic_regions_UP<span class="sc">$</span>Test)  ) <span class="fu">return</span>(<span class="sc">-</span><span class="dv">1</span>) <span class="co"># no significant regions</span></span>
<span id="cb182-4024"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4024" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-4025"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4025" aria-hidden="true" tabindex="-1"></a>    regions <span class="ot">&lt;-</span> <span class="dv">0</span> </span>
<span id="cb182-4026"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4026" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>(genomic_regions_UP<span class="sc">$</span>Test)) { </span>
<span id="cb182-4027"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4027" aria-hidden="true" tabindex="-1"></a>    dataframe_UPregions<span class="ot">&lt;-</span><span class="fu">GenomicRegions2dataframe</span>(genomic_regions_UP[[<span class="dv">1</span>]])</span>
<span id="cb182-4028"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4028" aria-hidden="true" tabindex="-1"></a>    dataframe_UPregions<span class="sc">$</span>Regulation <span class="ot">&lt;-</span> <span class="st">&quot;Up&quot;</span></span>
<span id="cb182-4029"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4029" aria-hidden="true" tabindex="-1"></a>    regions <span class="ot">=</span> dataframe_UPregions</span>
<span id="cb182-4030"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4030" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-4031"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4031" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4032"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4032" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="sc">!</span><span class="fu">is.null</span>(genomic_regions_DOWN<span class="sc">$</span>Test) ){ </span>
<span id="cb182-4033"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4033" aria-hidden="true" tabindex="-1"></a>    dataframe_DOWNregions<span class="ot">&lt;-</span><span class="fu">GenomicRegions2dataframe</span>(genomic_regions_DOWN[[<span class="dv">1</span>]])</span>
<span id="cb182-4034"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4034" aria-hidden="true" tabindex="-1"></a>    dataframe_DOWNregions<span class="sc">$</span>Regulation <span class="ot">&lt;-</span> <span class="st">&quot;Down&quot;</span></span>
<span id="cb182-4035"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4035" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> ( <span class="fu">class</span>(regions) <span class="sc">!=</span> <span class="st">&quot;data.frame&quot;</span> ) regions <span class="ot">&lt;-</span> dataframe_DOWNregions <span class="cf">else</span> <span class="co"># if UP regions is NULL</span></span>
<span id="cb182-4036"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4036" aria-hidden="true" tabindex="-1"></a>     regions <span class="ot">=</span> <span class="fu">rbind</span>(regions,dataframe_DOWNregions  )</span>
<span id="cb182-4037"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4037" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-4038"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4038" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4039"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4039" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">class</span>(regions) <span class="sc">!=</span> <span class="st">&quot;data.frame&quot;</span>) <span class="fu">return</span>(<span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb182-4040"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4040" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4041"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4041" aria-hidden="true" tabindex="-1"></a>    Regions <span class="ot">&lt;-</span> regions[,<span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)] </span>
<span id="cb182-4042"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4042" aria-hidden="true" tabindex="-1"></a>    Regions <span class="ot">&lt;-</span> Regions[<span class="fu">which</span>(Regions<span class="sc">$</span>end <span class="sc">!=</span> Regions<span class="sc">$</span>start),]</span>
<span id="cb182-4043"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4043" aria-hidden="true" tabindex="-1"></a>    Regions<span class="sc">$</span>size <span class="ot">=</span> <span class="fu">round</span>((Regions<span class="sc">$</span>end <span class="sc">-</span> Regions<span class="sc">$</span>start )<span class="sc">/</span><span class="dv">1000000</span>,<span class="dv">3</span>)</span>
<span id="cb182-4044"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4044" aria-hidden="true" tabindex="-1"></a>    Regions<span class="sc">$</span>chr <span class="ot">&lt;-</span> chromosomes[ Regions<span class="sc">$</span>chr ] <span class="co"># convert from chr. number to names</span></span>
<span id="cb182-4045"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4045" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4046"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4046" aria-hidden="true" tabindex="-1"></a>    <span class="co"># find the gene indices in the up or down regulated regions</span></span>
<span id="cb182-4047"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4047" aria-hidden="true" tabindex="-1"></a>    regulatedGenes <span class="ot">&lt;-</span> <span class="cf">function</span> (i) {</span>
<span id="cb182-4048"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4048" aria-hidden="true" tabindex="-1"></a>    ix <span class="ot">=</span>  <span class="fu">which</span>( Regions<span class="sc">$</span>chr <span class="sc">==</span> x<span class="sc">$</span>chromosome_name[i] <span class="sc">&amp;</span></span>
<span id="cb182-4049"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4049" aria-hidden="true" tabindex="-1"></a>         Regions<span class="sc">$</span>start <span class="sc">&lt;</span> x<span class="sc">$</span>start_position[i] <span class="co">#&amp;(Regions$end &gt; x$start_position[i] + x$genomeSpan[i])</span></span>
<span id="cb182-4050"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4050" aria-hidden="true" tabindex="-1"></a>         <span class="sc">&amp;</span>(Regions<span class="sc">$</span>end <span class="sc">&gt;</span> x<span class="sc">$</span>start_position[i] )) <span class="co"># if the start position is within the region</span></span>
<span id="cb182-4051"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4051" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">length</span>(ix) <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">length</span>(ix) <span class="sc">&gt;</span><span class="dv">1</span>  ) <span class="fu">return</span>(<span class="cn">NA</span>) <span class="cf">else</span> <span class="fu">return</span>( ix )</span>
<span id="cb182-4052"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4052" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb182-4053"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4053" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4054"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4054" aria-hidden="true" tabindex="-1"></a>    regionID <span class="ot">=</span> <span class="fu">unlist</span>( <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(x)[<span class="dv">1</span>], regulatedGenes) )</span>
<span id="cb182-4055"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4055" aria-hidden="true" tabindex="-1"></a>    x1 <span class="ot">=</span> x[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(regionID)),]</span>
<span id="cb182-4056"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4056" aria-hidden="true" tabindex="-1"></a>    regionID <span class="ot">=</span> regionID[<span class="sc">!</span><span class="fu">is.na</span>(regionID)]</span>
<span id="cb182-4057"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4057" aria-hidden="true" tabindex="-1"></a>    x1 <span class="ot">=</span> <span class="fu">cbind</span>(regionID,Regions[regionID, ], x1[,<span class="fu">c</span>(<span class="st">&#39;symbol&#39;</span>, <span class="st">&#39;Row.names&#39;</span>, <span class="st">&#39;Fold&#39;</span>,<span class="st">&#39;FDR&#39;</span>, <span class="st">&#39;band&#39;</span>,<span class="st">&#39;start_position&#39;</span>)])</span>
<span id="cb182-4058"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4058" aria-hidden="true" tabindex="-1"></a>    x1 <span class="ot">=</span> x1[<span class="fu">order</span>(x1<span class="sc">$</span>regionID,x1<span class="sc">$</span>start_position ), ]    </span>
<span id="cb182-4059"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4059" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(x1)[<span class="dv">8</span>] <span class="ot">=</span> <span class="st">&quot;Ensembl&quot;</span>; <span class="fu">colnames</span>(x1)[<span class="dv">4</span>] <span class="ot">=</span> <span class="st">&quot;Region.Start&quot;</span>;<span class="fu">colnames</span>(x1)[<span class="dv">5</span>] <span class="ot">=</span> <span class="st">&quot;Region.End&quot;</span>; <span class="fu">colnames</span>(x1)[<span class="dv">12</span>] <span class="ot">=</span> <span class="st">&quot;Gene.Start&quot;</span></span>
<span id="cb182-4060"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4060" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4061"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4061" aria-hidden="true" tabindex="-1"></a>    <span class="co"># number of genes</span></span>
<span id="cb182-4062"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4062" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> <span class="fu">table</span>(x1<span class="sc">$</span>regionID)</span>
<span id="cb182-4063"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4063" aria-hidden="true" tabindex="-1"></a>    Regions<span class="sc">$</span>Ngenes <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb182-4064"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4064" aria-hidden="true" tabindex="-1"></a>    Regions<span class="sc">$</span>Ngenes[<span class="fu">as.integer</span>(<span class="fu">names</span>(tem) ) ] <span class="ot">&lt;-</span> tem</span>
<span id="cb182-4065"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4065" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4066"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4066" aria-hidden="true" tabindex="-1"></a>    <span class="co"># cytoband per region</span></span>
<span id="cb182-4067"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4067" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> <span class="fu">unique</span>(x1[,<span class="fu">c</span>(<span class="st">&#39;regionID&#39;</span>,<span class="st">&#39;band&#39;</span>)])</span>
<span id="cb182-4068"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4068" aria-hidden="true" tabindex="-1"></a>    tem<span class="sc">$</span>band <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">..*&quot;</span>,<span class="st">&quot;&quot;</span>,tem<span class="sc">$</span>band)</span>
<span id="cb182-4069"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4069" aria-hidden="true" tabindex="-1"></a>    tem <span class="ot">=</span> <span class="fu">unique</span>(tem)</span>
<span id="cb182-4070"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4070" aria-hidden="true" tabindex="-1"></a>    Regions<span class="sc">$</span>band <span class="ot">=</span><span class="st">&quot;&quot;</span></span>
<span id="cb182-4071"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4071" aria-hidden="true" tabindex="-1"></a>    Regions<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(Regions)[<span class="dv">1</span>]</span>
<span id="cb182-4072"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4072" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> ( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(Regions)[<span class="dv">1</span>] )</span>
<span id="cb182-4073"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4073" aria-hidden="true" tabindex="-1"></a>      Regions<span class="sc">$</span>band[i] <span class="ot">&lt;-</span> <span class="fu">paste</span>( tem[ <span class="fu">which</span>( tem[,<span class="dv">1</span>] <span class="sc">==</span> Regions<span class="sc">$</span>ID[i]),<span class="dv">2</span>],  <span class="at">collapse=</span><span class="st">&quot;;&quot;</span> )</span>
<span id="cb182-4074"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4074" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4075"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4075" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Genes</span></span>
<span id="cb182-4076"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4076" aria-hidden="true" tabindex="-1"></a>    Regions<span class="sc">$</span>Genes <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb182-4077"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4077" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> ( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(Regions)[<span class="dv">1</span>] )</span>
<span id="cb182-4078"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4078" aria-hidden="true" tabindex="-1"></a>      Regions<span class="sc">$</span>Genes[i] <span class="ot">&lt;-</span> <span class="fu">paste</span>( x1<span class="sc">$</span>symbol[ <span class="fu">which</span>( x1[,<span class="dv">1</span>] <span class="sc">==</span> Regions<span class="sc">$</span>ID[i])],  <span class="at">collapse=</span><span class="st">&quot; &quot;</span> )</span>
<span id="cb182-4079"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4079" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( <span class="fu">list</span>(<span class="at">mainResult =</span> GEanalysisResults, <span class="at">Regions =</span> Regions, <span class="at">Genes =</span> x1,<span class="at">legend.x=</span><span class="fu">max</span>(x<span class="sc">$</span>start_position)<span class="sc">*</span>.<span class="dv">6</span>, <span class="at">legend.y=</span><span class="fu">max</span>(chromosomesNumbers)<span class="sc">-</span><span class="dv">3</span>  ) )</span>
<span id="cb182-4080"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4080" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-4081"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4081" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4082"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4082" aria-hidden="true" tabindex="-1"></a>genomePlot <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb182-4083"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4083" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(PREDA,<span class="at">verbose=</span><span class="cn">FALSE</span>)  <span class="co"># showing expression on genome</span></span>
<span id="cb182-4084"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4084" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(PREDAsampledata,<span class="at">verbose=</span><span class="cn">FALSE</span>) </span>
<span id="cb182-4085"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4085" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(hgu133plus2.db,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb182-4086"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4086" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-4087"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4087" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>( <span class="fu">class</span>(genomePlotData.out) <span class="sc">!=</span> <span class="st">&quot;list&quot;</span> ) { <span class="fu">plot.new</span>(); <span class="fu">text</span>(<span class="fl">0.2</span>,<span class="dv">1</span>, <span class="st">&quot;No significant regions found!&quot;</span>)} <span class="cf">else</span> {</span>
<span id="cb182-4088"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4088" aria-hidden="true" tabindex="-1"></a>   GEanalysisResults <span class="ot">&lt;-</span> genomePlotData.out<span class="sc">$</span>mainResult;</span>
<span id="cb182-4089"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4089" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>( <span class="fu">is.null</span>( GEanalysisResults ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)    </span>
<span id="cb182-4090"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4090" aria-hidden="true" tabindex="-1"></a>    genomic_regions_UP<span class="ot">&lt;-</span><span class="fu">PREDAResults2GenomicRegions</span>(GEanalysisResults, <span class="at">qval.threshold=</span>input_RegionsPvalCutoff, <span class="at">smoothStatistic.tail=</span><span class="st">&quot;upper&quot;</span>, <span class="at">smoothStatistic.threshold=</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">*</span>input_StatisticCutoff)</span>
<span id="cb182-4091"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4091" aria-hidden="true" tabindex="-1"></a>    genomic_regions_DOWN<span class="ot">&lt;-</span><span class="fu">PREDAResults2GenomicRegions</span>(GEanalysisResults, <span class="at">qval.threshold=</span>input_RegionsPvalCutoff, <span class="at">smoothStatistic.tail=</span><span class="st">&quot;lower&quot;</span>, <span class="at">smoothStatistic.threshold=</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">*</span>input_StatisticCutoff)</span>
<span id="cb182-4092"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4092" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-4093"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4093" aria-hidden="true" tabindex="-1"></a>    checkplot<span class="ot">&lt;-</span><span class="fu">genomePlot</span>(GEanalysisResults, <span class="at">genomicRegions=</span><span class="fu">c</span>(genomic_regions_UP, genomic_regions_DOWN), <span class="at">grouping=</span><span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>), <span class="at">scale.positions=</span><span class="st">&quot;Mb&quot;</span>, <span class="at">region.colors=</span><span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>))</span>
<span id="cb182-4094"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4094" aria-hidden="true" tabindex="-1"></a>    <span class="fu">legend</span>(<span class="at">x=</span>genomePlotData.out<span class="sc">$</span>legend.x, <span class="at">y=</span>genomePlotData.out<span class="sc">$</span>legend.y , <span class="at">legend=</span><span class="fu">c</span>(<span class="st">&quot;UP&quot;</span>, <span class="st">&quot;DOWN&quot;</span>), <span class="at">fill=</span><span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;blue&quot;</span>))</span>
<span id="cb182-4095"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4095" aria-hidden="true" tabindex="-1"></a>    } <span class="co">#if  else</span></span>
<span id="cb182-4096"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4096" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-4097"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4097" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4098"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4098" aria-hidden="true" tabindex="-1"></a><span class="co"># pre-calculating PREDA, so that changing FDR cutoffs does not trigger entire calculation</span></span>
<span id="cb182-4099"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4099" aria-hidden="true" tabindex="-1"></a>genomePlotDataPre <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb182-4100"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4100" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(PREDA,<span class="at">verbose=</span><span class="cn">FALSE</span>)  <span class="co"># showing expression on genome</span></span>
<span id="cb182-4101"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4101" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(PREDAsampledata,<span class="at">verbose=</span><span class="cn">FALSE</span>) </span>
<span id="cb182-4102"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4102" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(hgu133plus2.db,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb182-4103"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4103" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(input_selectContrast2 ) ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-4104"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4104" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-4105"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4105" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>( <span class="fu">length</span>(limma.out<span class="sc">$</span>topGenes) <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-4106"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4106" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-4107"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4107" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">length</span>( limma.out<span class="sc">$</span>comparisons)  <span class="sc">==</span><span class="dv">1</span> )  </span>
<span id="cb182-4108"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4108" aria-hidden="true" tabindex="-1"></a>    { top1<span class="ot">=</span>limma.out<span class="sc">$</span>topGenes[[<span class="dv">1</span>]]  </span>
<span id="cb182-4109"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4109" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb182-4110"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4110" aria-hidden="true" tabindex="-1"></a>      top <span class="ot">=</span> limma.out<span class="sc">$</span>topGenes</span>
<span id="cb182-4111"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4111" aria-hidden="true" tabindex="-1"></a>      ix <span class="ot">=</span> <span class="fu">match</span>(input_selectContrast2, <span class="fu">names</span>(top))</span>
<span id="cb182-4112"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4112" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>( <span class="fu">is.na</span>(ix)) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb182-4113"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4113" aria-hidden="true" tabindex="-1"></a>      top1 <span class="ot">&lt;-</span> top[[ix]]; </span>
<span id="cb182-4114"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4114" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb182-4115"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4115" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">dim</span>(top1)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span> (<span class="cn">NULL</span>)</span>
<span id="cb182-4116"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4116" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(top1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Fold&quot;</span>,<span class="st">&quot;FDR&quot;</span>)</span>
<span id="cb182-4117"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4117" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb182-4118"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4118" aria-hidden="true" tabindex="-1"></a>     <span class="co"># write.csv(merge(top1,allGeneInfo(), by.x=&quot;row.names&quot;,by.y=&quot;ensembl_gene_id&quot;  ),&quot;tem.csv&quot;  )</span></span>
<span id="cb182-4119"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4119" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> <span class="fu">merge</span>(top1,allGeneInfo.out, <span class="at">by.x=</span><span class="st">&quot;row.names&quot;</span>,<span class="at">by.y=</span><span class="st">&quot;ensembl_gene_id&quot;</span>  )</span>
<span id="cb182-4120"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4120" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-4121"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4121" aria-hidden="true" tabindex="-1"></a>     <span class="co">#hist(top1[,1])</span></span>
<span id="cb182-4122"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4122" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-4123"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4123" aria-hidden="true" tabindex="-1"></a>     <span class="do">########## PREDA</span></span>
<span id="cb182-4124"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4124" aria-hidden="true" tabindex="-1"></a>    infofile <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;sampledata&quot;</span>, <span class="st">&quot;GeneExpression&quot;</span>, <span class="st">&quot;sampleinfoGE_PREDA.txt&quot;</span>, <span class="at">package =</span> <span class="st">&quot;PREDAsampledata&quot;</span>)</span>
<span id="cb182-4125"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4125" aria-hidden="true" tabindex="-1"></a>    sampleinfo<span class="ot">&lt;-</span><span class="fu">read.table</span>(infofile, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb182-4126"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4126" aria-hidden="true" tabindex="-1"></a>    <span class="fu">head</span>(sampleinfo)</span>
<span id="cb182-4127"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4128"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4128" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data</span>(ExpressionSetRCC)</span>
<span id="cb182-4129"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4129" aria-hidden="true" tabindex="-1"></a>    GEstatisticsForPREDA<span class="ot">&lt;-</span><span class="fu">statisticsForPREDAfromEset</span>(ExpressionSetRCC, <span class="at">statisticType=</span><span class="st">&quot;tstatistic&quot;</span>, <span class="at">referenceGroupLabel=</span><span class="st">&quot;normal&quot;</span>, <span class="at">classVector=</span>sampleinfo[,<span class="st">&quot;Class&quot;</span>])</span>
<span id="cb182-4130"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4130" aria-hidden="true" tabindex="-1"></a>    <span class="fu">analysesNames</span>(GEstatisticsForPREDA)</span>
<span id="cb182-4131"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4131" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-4132"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4132" aria-hidden="true" tabindex="-1"></a>    GEGenomicAnnotations<span class="ot">&lt;-</span><span class="fu">eset2GenomicAnnotations</span>(ExpressionSetRCC, <span class="at">retain.chrs=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>) <span class="co"># needs hgu133plus2.db package</span></span>
<span id="cb182-4133"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4133" aria-hidden="true" tabindex="-1"></a>    GEGenomicAnnotationsForPREDA<span class="ot">&lt;-</span><span class="fu">GenomicAnnotations2GenomicAnnotationsForPREDA</span>(GEGenomicAnnotations, <span class="at">reference_position_type=</span><span class="st">&quot;median&quot;</span>)</span>
<span id="cb182-4134"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4134" aria-hidden="true" tabindex="-1"></a>    GEDataForPREDA<span class="ot">&lt;-</span><span class="fu">MergeStatisticAnnotations2DataForPREDA</span>(GEstatisticsForPREDA, GEGenomicAnnotationsForPREDA, <span class="at">sortAndCleanNA=</span><span class="cn">TRUE</span>)</span>
<span id="cb182-4135"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4136"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4136" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x = read.csv(&quot;PREDA_test.csv&quot;)</span></span>
<span id="cb182-4137"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4138"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4138" aria-hidden="true" tabindex="-1"></a>    <span class="co">#x &lt;- x[,-1] # remove the row index, not needed in Shiny</span></span>
<span id="cb182-4139"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4140"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4140" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> x[<span class="fu">order</span>(x<span class="sc">$</span>chromosome_name,x<span class="sc">$</span>start_position),]</span>
<span id="cb182-4141"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4141" aria-hidden="true" tabindex="-1"></a>     tem <span class="ot">=</span> <span class="fu">sort</span>( <span class="fu">table</span>( x<span class="sc">$</span>chromosome_name), <span class="at">decreasing=</span>T) </span>
<span id="cb182-4142"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4142" aria-hidden="true" tabindex="-1"></a>     chromosomes <span class="ot">&lt;-</span> <span class="fu">names</span>( tem[tem <span class="sc">&gt;</span> <span class="dv">100</span> ] )  <span class="co"># chromosomes with less than 100 genes are excluded</span></span>
<span id="cb182-4143"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4143" aria-hidden="true" tabindex="-1"></a>     <span class="cf">if</span>(<span class="fu">length</span>(chromosomes) <span class="sc">&gt;</span> <span class="dv">50</span>) chromosomes <span class="ot">&lt;-</span> chromosomes[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]  <span class="co"># at most 50 chromosomes</span></span>
<span id="cb182-4144"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4145"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4145" aria-hidden="true" tabindex="-1"></a>     chromosomes <span class="ot">=</span> chromosomes[<span class="fu">order</span>(<span class="fu">as.numeric</span>(chromosomes) ) ]</span>
<span id="cb182-4146"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4146" aria-hidden="true" tabindex="-1"></a>     <span class="co"># chromosomes = chromosomes[!is.na(as.numeric(chromosomes) ) ]</span></span>
<span id="cb182-4147"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4147" aria-hidden="true" tabindex="-1"></a>     chromosomesNumbers <span class="ot">=</span> <span class="fu">as.numeric</span>(chromosomes)</span>
<span id="cb182-4148"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4148" aria-hidden="true" tabindex="-1"></a>     <span class="co"># convert chr.x to numbers</span></span>
<span id="cb182-4149"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4149" aria-hidden="true" tabindex="-1"></a>      j <span class="ot">=</span> <span class="fu">max</span>( chromosomesNumbers,<span class="at">na.rm=</span>T) </span>
<span id="cb182-4150"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4150" aria-hidden="true" tabindex="-1"></a>      <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>( chromosomes)) {</span>
<span id="cb182-4151"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4151" aria-hidden="true" tabindex="-1"></a>       <span class="cf">if</span> ( <span class="fu">is.na</span>(chromosomesNumbers[i]) ) </span>
<span id="cb182-4152"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4152" aria-hidden="true" tabindex="-1"></a>       { chromosomesNumbers[i] <span class="ot">&lt;-</span> j<span class="sc">+</span><span class="dv">1</span>; j <span class="ot">&lt;-</span> j<span class="sc">+</span><span class="dv">1</span>; }</span>
<span id="cb182-4153"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4153" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb182-4154"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4154" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb182-4155"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4155" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> x[<span class="fu">which</span>(x<span class="sc">$</span>chromosome_name <span class="sc">%in%</span> chromosomes   ),]</span>
<span id="cb182-4156"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4156" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> <span class="fu">droplevels</span>(x)</span>
<span id="cb182-4157"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4157" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-4158"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4158" aria-hidden="true" tabindex="-1"></a>    <span class="co"># find the number coding for chromosome </span></span>
<span id="cb182-4159"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4159" aria-hidden="true" tabindex="-1"></a>     getChrNumber <span class="ot">&lt;-</span> <span class="cf">function</span> (chrName){</span>
<span id="cb182-4160"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4160" aria-hidden="true" tabindex="-1"></a>     <span class="fu">return</span>( chromosomesNumbers[ <span class="fu">which</span>( chromosomes <span class="sc">==</span> chrName)] )</span>
<span id="cb182-4161"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4161" aria-hidden="true" tabindex="-1"></a>     }</span>
<span id="cb182-4162"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4162" aria-hidden="true" tabindex="-1"></a>      x<span class="sc">$</span>chrNum <span class="ot">=</span> <span class="dv">1</span> <span class="co"># numeric coding</span></span>
<span id="cb182-4163"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4163" aria-hidden="true" tabindex="-1"></a>      x<span class="sc">$</span>chrNum <span class="ot">&lt;-</span> <span class="fu">unlist</span>( <span class="fu">lapply</span>( x<span class="sc">$</span>chromosome_name, getChrNumber) )</span>
<span id="cb182-4164"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4164" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-4165"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4165" aria-hidden="true" tabindex="-1"></a>     x<span class="sc">$</span>Row.names <span class="ot">&lt;-</span> <span class="fu">as.character</span>( x<span class="sc">$</span>Row.names)</span>
<span id="cb182-4166"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4166" aria-hidden="true" tabindex="-1"></a>     fold <span class="ot">=</span> x<span class="sc">$</span>Fold; <span class="fu">names</span>(fold) <span class="ot">=</span> x<span class="sc">$</span>Row.names</span>
<span id="cb182-4167"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4167" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-4168"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4168" aria-hidden="true" tabindex="-1"></a>    <span class="co"># write.csv(x,&quot;tem.csv&quot;)</span></span>
<span id="cb182-4169"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4169" aria-hidden="true" tabindex="-1"></a>     x <span class="ot">&lt;-</span> x[<span class="sc">!</span><span class="fu">duplicated</span>(x<span class="sc">$</span>Row.names),]</span>
<span id="cb182-4170"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4170" aria-hidden="true" tabindex="-1"></a>     <span class="co">#rownames(x) = x$Row.names</span></span>
<span id="cb182-4171"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4171" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-4172"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4172" aria-hidden="true" tabindex="-1"></a>     myData <span class="ot">&lt;-</span> GEDataForPREDA</span>
<span id="cb182-4173"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4173" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb182-4174"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4174" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>position <span class="ot">&lt;-</span> <span class="fu">as.integer</span>( x<span class="sc">$</span>start_position<span class="sc">+</span> x<span class="sc">$</span>genomeSpan<span class="sc">/</span><span class="dv">2</span> )</span>
<span id="cb182-4175"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4175" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>ids <span class="ot">&lt;-</span> <span class="fu">as.character</span>( x<span class="sc">$</span>Row.names)</span>
<span id="cb182-4176"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4176" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>chr <span class="ot">&lt;-</span> <span class="fu">as.integer</span>( x<span class="sc">$</span>chrNum )</span>
<span id="cb182-4177"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4177" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>start <span class="ot">&lt;-</span> x<span class="sc">$</span>start_position</span>
<span id="cb182-4178"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4178" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>end <span class="ot">&lt;-</span> x<span class="sc">$</span>start_position <span class="sc">+</span> x<span class="sc">$</span>genomeSpan</span>
<span id="cb182-4179"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4179" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>strand <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">1</span>,<span class="fu">dim</span>(x)[<span class="dv">1</span>])</span>
<span id="cb182-4180"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4180" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>chromosomesLabels <span class="ot">&lt;-</span> chromosomes</span>
<span id="cb182-4181"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4181" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>chromosomesNumbers <span class="ot">&lt;-</span> <span class="fu">as.integer</span>( chromosomesNumbers )</span>
<span id="cb182-4182"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4182" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>statistic <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>( x[,<span class="dv">2</span>,<span class="at">drop=</span><span class="cn">FALSE</span>] )</span>
<span id="cb182-4183"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4183" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>analysesNames <span class="ot">&lt;-</span> <span class="st">&quot;Test&quot;</span></span>
<span id="cb182-4184"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4184" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>testedTail <span class="ot">&lt;-</span> <span class="st">&quot;both&quot;</span></span>
<span id="cb182-4185"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4185" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>optionalAnnotations <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>( x[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>,<span class="at">drop=</span><span class="cn">FALSE</span>] )</span>
<span id="cb182-4186"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4186" aria-hidden="true" tabindex="-1"></a>     myData<span class="sc">@</span>optionalAnnotationsHeaders <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;a&quot;</span>,<span class="st">&quot;b&quot;</span>)</span>
<span id="cb182-4187"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4187" aria-hidden="true" tabindex="-1"></a>     </span>
<span id="cb182-4188"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4188" aria-hidden="true" tabindex="-1"></a>    <span class="fu">set.seed</span>(<span class="dv">2</span>)</span>
<span id="cb182-4189"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4189" aria-hidden="true" tabindex="-1"></a>    GEanalysisResults<span class="ot">&lt;-</span><span class="fu">PREDA_main</span>(myData,<span class="at">nperms =</span> PREDA_Permutations)</span>
<span id="cb182-4190"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4191"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4191" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>( GEanalysisResults )</span>
<span id="cb182-4192"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4192" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-4193"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4194"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4194" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-4195"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4195" aria-hidden="true" tabindex="-1"></a><span class="co"># Biclustering</span></span>
<span id="cb182-4196"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4196" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-4197"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4197" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb182-4198"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4198" aria-hidden="true" tabindex="-1"></a>biclustering <span class="ot">&lt;-</span> <span class="cf">function</span>( ){</span>
<span id="cb182-4199"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4199" aria-hidden="true" tabindex="-1"></a>            <span class="fu">library</span>(biclust,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb182-4200"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4201"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4201" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(input_biclustMethod <span class="sc">==</span> <span class="st">&quot;BCQU()&quot;</span> )</span>
<span id="cb182-4202"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4202" aria-hidden="true" tabindex="-1"></a>                <span class="fu">library</span>(QUBIC,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># have trouble installing on Linux</span></span>
<span id="cb182-4203"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4203" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(input_biclustMethod <span class="sc">==</span> <span class="st">&quot;BCUnibic()&quot;</span> )</span>
<span id="cb182-4204"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4204" aria-hidden="true" tabindex="-1"></a>                <span class="fu">library</span>(runibic,<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="co"># </span></span>
<span id="cb182-4205"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4205" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4206"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4206" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-4207"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4207" aria-hidden="true" tabindex="-1"></a>            x <span class="ot">&lt;-</span> convertedData.out</span>
<span id="cb182-4208"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4208" aria-hidden="true" tabindex="-1"></a>            n<span class="ot">=</span>input_nGenesBiclust   </span>
<span id="cb182-4209"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4209" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(n<span class="sc">&gt;</span><span class="fu">dim</span>(x)[<span class="dv">1</span>]) n <span class="ot">=</span> <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="co"># max as data</span></span>
<span id="cb182-4210"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4210" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(n<span class="sc">&lt;</span><span class="dv">10</span>) n <span class="ot">=</span> <span class="dv">10</span> </span>
<span id="cb182-4211"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4211" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(n<span class="sc">&gt;</span> <span class="dv">2000</span> ) n <span class="ot">=</span> <span class="dv">2000</span>           </span>
<span id="cb182-4212"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4212" aria-hidden="true" tabindex="-1"></a>            x<span class="ot">=</span><span class="fu">as.matrix</span>(x[<span class="dv">1</span><span class="sc">:</span>n,])<span class="sc">-</span><span class="fu">apply</span>(x[<span class="dv">1</span><span class="sc">:</span>n,],<span class="dv">1</span>,mean)</span>
<span id="cb182-4213"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4213" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4214"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4214" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( input_biclustMethod <span class="sc">==</span> <span class="st">&quot;BCXmotifs()&quot;</span>  )</span>
<span id="cb182-4215"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4215" aria-hidden="true" tabindex="-1"></a>                x<span class="ot">&lt;-</span><span class="fu">discretize</span>(x)</span>
<span id="cb182-4216"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4217"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4217" aria-hidden="true" tabindex="-1"></a>            <span class="co">#res &lt;- biclust::biclust(as.matrix( x), method = BCQU()) </span></span>
<span id="cb182-4218"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4218" aria-hidden="true" tabindex="-1"></a>            runR <span class="ot">=</span> <span class="fu">paste</span>( <span class="st">&quot;res &lt;- biclust::biclust(as.matrix( x), method =&quot;</span>, input_biclustMethod ,<span class="st">&quot;)&quot;</span> )</span>
<span id="cb182-4219"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4219" aria-hidden="true" tabindex="-1"></a>            <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> runR ) )</span>
<span id="cb182-4220"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4220" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4221"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4222"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4222" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>(<span class="fu">list</span>( <span class="at">x=</span>x, <span class="at">res=</span>res)  )    </span>
<span id="cb182-4223"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4223" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-4224"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4225"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4226"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4226" aria-hidden="true" tabindex="-1"></a>biclustHeatmap <span class="ot">&lt;-</span> <span class="cf">function</span> ( ){</span>
<span id="cb182-4227"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4227" aria-hidden="true" tabindex="-1"></a>            res <span class="ot">=</span> <span class="fu">biclustering</span>()<span class="sc">$</span>res</span>
<span id="cb182-4228"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4228" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( res<span class="sc">@</span>Number <span class="sc">==</span> <span class="dv">0</span> ) { <span class="fu">plot.new</span>(); <span class="fu">text</span>(<span class="fl">0.5</span>,<span class="fl">0.5</span>, <span class="st">&quot;No cluster found!&quot;</span>)} <span class="cf">else</span> {</span>
<span id="cb182-4229"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4229" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-4230"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4230" aria-hidden="true" tabindex="-1"></a>                x <span class="ot">=</span> biclust<span class="sc">::</span><span class="fu">bicluster</span>(biclustering.out<span class="sc">$</span>x, res, <span class="fu">as.numeric</span>( input_selectBicluster)  )[[<span class="dv">1</span>]]</span>
<span id="cb182-4231"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4232"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4232" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">4</span>, <span class="fl">1.4</span>, <span class="fl">0.2</span>))</span>
<span id="cb182-4233"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4234"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4234" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>( <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="sc">&lt;=</span><span class="dv">30</span> )</span>
<span id="cb182-4235"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4235" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">heatmap.2</span>(x,  <span class="at">Rowv =</span>T,<span class="at">Colv=</span>F, <span class="at">dendrogram =</span><span class="st">&quot;none&quot;</span>,</span>
<span id="cb182-4236"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4236" aria-hidden="true" tabindex="-1"></a>                    <span class="at">col=</span>heatColors[<span class="fu">as.integer</span>(input_heatColors1),], <span class="at">density.info=</span><span class="st">&quot;none&quot;</span>, <span class="at">trace=</span><span class="st">&quot;none&quot;</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">keysize=</span>.<span class="dv">2</span></span>
<span id="cb182-4237"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4237" aria-hidden="true" tabindex="-1"></a>                    ,<span class="at">key=</span>F, <span class="co">#labRow = T,</span></span>
<span id="cb182-4238"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4238" aria-hidden="true" tabindex="-1"></a>                    ,<span class="at">margins =</span> <span class="fu">c</span>(<span class="dv">8</span>, <span class="dv">24</span>)</span>
<span id="cb182-4239"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4239" aria-hidden="true" tabindex="-1"></a>                    ,<span class="at">cexRow=</span><span class="dv">1</span></span>
<span id="cb182-4240"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4240" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#,srtCol=45</span></span>
<span id="cb182-4241"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4241" aria-hidden="true" tabindex="-1"></a>                    ,<span class="at">cexCol=</span><span class="fl">1.</span>  <span class="co"># size of font for sample names</span></span>
<span id="cb182-4242"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4242" aria-hidden="true" tabindex="-1"></a>                    ) <span class="cf">else</span>      </span>
<span id="cb182-4243"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4243" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">heatmap.2</span>(x,  <span class="at">Rowv =</span>T,<span class="at">Colv=</span>F, <span class="at">dendrogram =</span><span class="st">&quot;none&quot;</span>,</span>
<span id="cb182-4244"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4244" aria-hidden="true" tabindex="-1"></a>                    <span class="at">col=</span>heatColors[<span class="fu">as.integer</span>(input_heatColors1),], <span class="at">density.info=</span><span class="st">&quot;none&quot;</span>, <span class="at">trace=</span><span class="st">&quot;none&quot;</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">keysize=</span>.<span class="dv">2</span></span>
<span id="cb182-4245"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4245" aria-hidden="true" tabindex="-1"></a>                    ,<span class="at">key=</span>F, <span class="at">labRow =</span> F,</span>
<span id="cb182-4246"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4246" aria-hidden="true" tabindex="-1"></a>                    ,<span class="at">margins =</span> <span class="fu">c</span>(<span class="dv">8</span>, <span class="dv">4</span>)</span>
<span id="cb182-4247"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4247" aria-hidden="true" tabindex="-1"></a>                    ,<span class="at">cexRow=</span><span class="dv">1</span></span>
<span id="cb182-4248"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4248" aria-hidden="true" tabindex="-1"></a>                    <span class="co">#,srtCol=45</span></span>
<span id="cb182-4249"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4249" aria-hidden="true" tabindex="-1"></a>                    ,<span class="at">cexCol=</span><span class="fl">1.</span>  <span class="co"># size of font for sample names</span></span>
<span id="cb182-4250"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4251"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4251" aria-hidden="true" tabindex="-1"></a>                    )               </span>
<span id="cb182-4252"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4252" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb182-4253"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4253" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-4254"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4255"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4256"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4256" aria-hidden="true" tabindex="-1"></a>geneListBclustGO <span class="ot">&lt;-</span> <span class="cf">function</span>( ){        </span>
<span id="cb182-4257"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4257" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-4258"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4258" aria-hidden="true" tabindex="-1"></a>            res <span class="ot">=</span> biclustering.out<span class="sc">$</span>res</span>
<span id="cb182-4259"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4259" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( res<span class="sc">@</span>Number <span class="sc">==</span> <span class="dv">0</span> ) <span class="fu">return</span>(<span class="fu">as.data.frame</span>(<span class="st">&quot;No clusters found!&quot;</span>) ) </span>
<span id="cb182-4260"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4260" aria-hidden="true" tabindex="-1"></a>            x <span class="ot">=</span> biclust<span class="sc">::</span><span class="fu">bicluster</span>(biclustering.out<span class="sc">$</span>x, res, <span class="fu">as.numeric</span>( input_selectBicluster)  )[[<span class="dv">1</span>]]</span>
<span id="cb182-4261"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4261" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4262"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4262" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">dim</span>(x)[<span class="dv">1</span>]  <span class="sc">&lt;=</span> minGenesEnrichment) <span class="fu">return</span>(NoSig) </span>
<span id="cb182-4263"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4263" aria-hidden="true" tabindex="-1"></a>            query <span class="ot">=</span> <span class="fu">rownames</span>(x)</span>
<span id="cb182-4264"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4264" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4265"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4266"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4266" aria-hidden="true" tabindex="-1"></a>            result <span class="ot">&lt;-</span> <span class="fu">findOverlapGMT</span>( query, GeneSets.out,<span class="dv">1</span>) </span>
<span id="cb182-4267"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4267" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4268"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4268" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-4269"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4269" aria-hidden="true" tabindex="-1"></a>            result<span class="sc">$</span>Genes <span class="ot">=</span> <span class="st">&quot;Up regulated&quot;</span></span>
<span id="cb182-4270"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4270" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4271"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4272"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4272" aria-hidden="true" tabindex="-1"></a>            results1 <span class="ot">=</span> result</span>
<span id="cb182-4273"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4273" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> ( <span class="fu">is.null</span>( results1) ) <span class="fu">return</span> (NoSig)</span>
<span id="cb182-4274"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4274" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">&lt;</span><span class="dv">5</span> ) <span class="fu">return</span>(NoSig)  <span class="co"># Returns a data frame: &quot;No significant results found!&quot;</span></span>
<span id="cb182-4275"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4276"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4276" aria-hidden="true" tabindex="-1"></a>            results1<span class="ot">=</span> results1[,<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">4</span>)]</span>
<span id="cb182-4277"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4277" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(results1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;List&quot;</span>,<span class="st">&quot;FDR&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;GO terms or pathways&quot;</span>)</span>
<span id="cb182-4278"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4278" aria-hidden="true" tabindex="-1"></a>            minFDR <span class="ot">=</span> <span class="fl">0.01</span></span>
<span id="cb182-4279"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4279" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">min</span>(results1<span class="sc">$</span>FDR) <span class="sc">&gt;</span> minFDR ) results1 <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No signficant enrichment found.&quot;</span>) <span class="cf">else</span></span>
<span id="cb182-4280"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4280" aria-hidden="true" tabindex="-1"></a>            results1 <span class="ot">=</span> results1[<span class="fu">which</span>(results1<span class="sc">$</span>FDR <span class="sc">&lt;</span> minFDR),]</span>
<span id="cb182-4281"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4282"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4282" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4283"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4283" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">!=</span> <span class="dv">4</span>) <span class="fu">return</span>(NoSig)</span>
<span id="cb182-4284"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4284" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(results1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Direction&quot;</span>,<span class="st">&quot;adj.Pval&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;Pathways&quot;</span>)</span>
<span id="cb182-4285"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4285" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4286"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4286" aria-hidden="true" tabindex="-1"></a>            results1<span class="sc">$</span>adj.Pval <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;%-2.1e&quot;</span>,<span class="fu">as.numeric</span>(results1<span class="sc">$</span>adj.Pval) )</span>
<span id="cb182-4287"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4287" aria-hidden="true" tabindex="-1"></a>            results1[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(results1[,<span class="dv">1</span>])</span>
<span id="cb182-4288"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4288" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">&lt;-</span> results1[,<span class="dv">1</span>]</span>
<span id="cb182-4289"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4290"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4290" aria-hidden="true" tabindex="-1"></a>            results1[ <span class="fu">duplicated</span> (results1[,<span class="dv">1</span>] ),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb182-4291"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4291" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4292"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4292" aria-hidden="true" tabindex="-1"></a>            results1[,<span class="sc">-</span><span class="dv">1</span>]           </span>
<span id="cb182-4293"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4294"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4294" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-4295"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4296"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4296" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-4297"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4297" aria-hidden="true" tabindex="-1"></a><span class="co"># Co-expression network</span></span>
<span id="cb182-4298"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4298" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################</span></span>
<span id="cb182-4299"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4300"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4300" aria-hidden="true" tabindex="-1"></a>wgcna <span class="ot">&lt;-</span> <span class="cf">function</span> ( ){</span>
<span id="cb182-4301"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4301" aria-hidden="true" tabindex="-1"></a>            <span class="co">#http://pklab.med.harvard.edu/scw2014/WGCNA.html</span></span>
<span id="cb182-4302"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4302" aria-hidden="true" tabindex="-1"></a>            <span class="fu">library</span>(WGCNA)</span>
<span id="cb182-4303"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4303" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4304"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4304" aria-hidden="true" tabindex="-1"></a>            x <span class="ot">&lt;-</span> convertedData.out</span>
<span id="cb182-4305"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4305" aria-hidden="true" tabindex="-1"></a>            n<span class="ot">=</span>input_nGenesNetwork   </span>
<span id="cb182-4306"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4306" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(n<span class="sc">&gt;</span><span class="fu">dim</span>(x)[<span class="dv">1</span>]) n <span class="ot">=</span> <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="co"># max as data</span></span>
<span id="cb182-4307"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4307" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(n<span class="sc">&lt;</span><span class="dv">50</span>) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-4308"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4308" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(x)[<span class="dv">2</span>] <span class="sc">&lt;</span><span class="dv">4</span>) <span class="fu">return</span>(<span class="cn">NULL</span>)</span>
<span id="cb182-4309"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4309" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(n<span class="sc">&gt;</span> maxGeneWGCNA ) n <span class="ot">=</span> maxGeneWGCNA           </span>
<span id="cb182-4310"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4310" aria-hidden="true" tabindex="-1"></a>            <span class="co">#x=as.matrix(x[1:n,])-apply(x[1:n,],1,mean)</span></span>
<span id="cb182-4311"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4311" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4312"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4312" aria-hidden="true" tabindex="-1"></a>            datExpr<span class="ot">=</span><span class="fu">t</span>(x[<span class="dv">1</span><span class="sc">:</span>n,])</span>
<span id="cb182-4313"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4314"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4314" aria-hidden="true" tabindex="-1"></a>            subGeneNames<span class="ot">=</span><span class="fu">colnames</span>(datExpr)</span>
<span id="cb182-4315"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4315" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4316"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4316" aria-hidden="true" tabindex="-1"></a>            <span class="co">#Choosing a soft-threshold to fit a scale-free topology to the network</span></span>
<span id="cb182-4317"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4317" aria-hidden="true" tabindex="-1"></a>            powers <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>), <span class="fu">seq</span>(<span class="at">from =</span> <span class="dv">12</span>, <span class="at">to=</span><span class="dv">20</span>, <span class="at">by=</span><span class="dv">2</span>));</span>
<span id="cb182-4318"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4318" aria-hidden="true" tabindex="-1"></a>            sft<span class="ot">=</span><span class="fu">pickSoftThreshold</span>(datExpr,<span class="at">dataIsExpr =</span> <span class="cn">TRUE</span>,<span class="at">powerVector =</span> powers,<span class="at">corFnc =</span> cor,<span class="at">corOptions =</span> <span class="fu">list</span>(<span class="at">use =</span> <span class="st">&#39;p&#39;</span>),<span class="at">networkType =</span> <span class="st">&quot;unsigned&quot;</span>)</span>
<span id="cb182-4319"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4320"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4321"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4321" aria-hidden="true" tabindex="-1"></a>            softPower <span class="ot">=</span> input_mySoftPower;</span>
<span id="cb182-4322"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4323"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4323" aria-hidden="true" tabindex="-1"></a>            <span class="co">#calclute the adjacency matrix</span></span>
<span id="cb182-4324"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4324" aria-hidden="true" tabindex="-1"></a>            adj<span class="ot">=</span> <span class="fu">adjacency</span>(datExpr,<span class="at">type =</span> <span class="st">&quot;unsigned&quot;</span>, <span class="at">power =</span> softPower);</span>
<span id="cb182-4325"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4326"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4326" aria-hidden="true" tabindex="-1"></a>            <span class="co">#turn adjacency matrix into topological overlap to minimize the effects of noise and spurious associations</span></span>
<span id="cb182-4327"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4327" aria-hidden="true" tabindex="-1"></a>            TOM<span class="ot">=</span><span class="fu">TOMsimilarityFromExpr</span>(datExpr,<span class="at">networkType =</span> <span class="st">&quot;unsigned&quot;</span>, <span class="at">TOMType =</span> <span class="st">&quot;unsigned&quot;</span>, <span class="at">power =</span> softPower);</span>
<span id="cb182-4328"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4328" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(TOM) <span class="ot">=</span>subGeneNames</span>
<span id="cb182-4329"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4329" aria-hidden="true" tabindex="-1"></a>            <span class="fu">rownames</span>(TOM) <span class="ot">=</span>subGeneNames</span>
<span id="cb182-4330"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4330" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4331"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4331" aria-hidden="true" tabindex="-1"></a>            <span class="do">##########################################</span></span>
<span id="cb182-4332"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4332" aria-hidden="true" tabindex="-1"></a>            <span class="co"># module detection</span></span>
<span id="cb182-4333"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4334"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4334" aria-hidden="true" tabindex="-1"></a>            <span class="fu">library</span>(flashClust,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb182-4335"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4336"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4336" aria-hidden="true" tabindex="-1"></a>            geneTree <span class="ot">=</span> <span class="fu">flashClust</span>(<span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>TOM),<span class="at">method=</span><span class="st">&quot;average&quot;</span>);</span>
<span id="cb182-4337"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4338"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4338" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Set the minimum module size</span></span>
<span id="cb182-4339"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4339" aria-hidden="true" tabindex="-1"></a>            minModuleSize <span class="ot">=</span> input_minModuleSize;</span>
<span id="cb182-4340"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4341"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4341" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Module identification using dynamic tree cut</span></span>
<span id="cb182-4342"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4342" aria-hidden="true" tabindex="-1"></a>            dynamicMods <span class="ot">=</span> <span class="fu">cutreeDynamic</span>(<span class="at">dendro =</span> geneTree,  <span class="at">method=</span><span class="st">&quot;tree&quot;</span>, <span class="at">minClusterSize =</span> minModuleSize);</span>
<span id="cb182-4343"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4343" aria-hidden="true" tabindex="-1"></a>            <span class="co">#dynamicMods = cutreeDynamic(dendro = geneTree, distM = dissTOM, method=&quot;hybrid&quot;, deepSplit = 2, pamRespectsDendro = FALSE, minClusterSize = minModuleSize);</span></span>
<span id="cb182-4344"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4344" aria-hidden="true" tabindex="-1"></a>            <span class="co"># table(dynamicMods)</span></span>
<span id="cb182-4345"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4345" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4346"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4346" aria-hidden="true" tabindex="-1"></a>            dynamicColors <span class="ot">=</span> <span class="fu">labels2colors</span>(dynamicMods)</span>
<span id="cb182-4347"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4347" aria-hidden="true" tabindex="-1"></a>            <span class="co">#table(dynamicColors)</span></span>
<span id="cb182-4348"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4349"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4349" aria-hidden="true" tabindex="-1"></a>            <span class="co">#discard the unassigned genes, and focus on the rest  # this causes problems</span></span>
<span id="cb182-4350"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4350" aria-hidden="true" tabindex="-1"></a>            <span class="co">#restGenes= (dynamicColors != &quot;grey&quot;)</span></span>
<span id="cb182-4351"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4351" aria-hidden="true" tabindex="-1"></a>            <span class="co">#diss1=1-TOMsimilarityFromExpr(datExpr[,restGenes], power = softPower)</span></span>
<span id="cb182-4352"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4352" aria-hidden="true" tabindex="-1"></a>            <span class="co">#colnames(diss1) =rownames(diss1) =subGeneNames[restGenes]</span></span>
<span id="cb182-4353"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4353" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4354"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4354" aria-hidden="true" tabindex="-1"></a>            <span class="co">#colorCode = as.character(dynamicColors[restGenes])</span></span>
<span id="cb182-4355"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4356"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4356" aria-hidden="true" tabindex="-1"></a>            moduleInfo <span class="ot">=</span> <span class="fu">cbind</span>( subGeneNames, dynamicColors, dynamicMods)</span>
<span id="cb182-4357"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4357" aria-hidden="true" tabindex="-1"></a>            moduleInfo <span class="ot">=</span> moduleInfo[<span class="fu">which</span>(moduleInfo[,<span class="dv">2</span>] <span class="sc">!=</span> <span class="st">&quot;grey&quot;</span>) ,] <span class="co"># remove genes not in any modules</span></span>
<span id="cb182-4358"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4358" aria-hidden="true" tabindex="-1"></a>            moduleInfo <span class="ot">=</span> moduleInfo[<span class="fu">order</span>(moduleInfo[,<span class="dv">3</span>]),] <span class="co"># sort</span></span>
<span id="cb182-4359"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4359" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4360"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4360" aria-hidden="true" tabindex="-1"></a>            n.modules <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">unique</span>(dynamicColors) ) <span class="sc">-</span><span class="dv">1</span> ; nGenes <span class="ot">=</span> <span class="fu">dim</span>(moduleInfo)[<span class="dv">1</span>] </span>
<span id="cb182-4361"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4361" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4362"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4362" aria-hidden="true" tabindex="-1"></a>            <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">x =</span> <span class="fu">t</span>(datExpr),<span class="at">powers=</span>powers,<span class="at">sft=</span>sft, <span class="at">TOM =</span> TOM, <span class="at">dynamicColors =</span> dynamicColors, <span class="at">moduleInfo =</span> moduleInfo,<span class="at">n.modules=</span>n.modules, <span class="at">nGenes =</span>nGenes) )</span>
<span id="cb182-4363"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4363" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-4364"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4365"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4366"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4366" aria-hidden="true" tabindex="-1"></a>softPower <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb182-4367"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4368"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4368" aria-hidden="true" tabindex="-1"></a>        <span class="do">####################################  </span></span>
<span id="cb182-4369"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4369" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-4370"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4370" aria-hidden="true" tabindex="-1"></a>        sft <span class="ot">=</span> wgcna.out<span class="sc">$</span>sft;</span>
<span id="cb182-4371"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4371" aria-hidden="true" tabindex="-1"></a>        powers<span class="ot">=</span>wgcna.out<span class="sc">$</span>powers;</span>
<span id="cb182-4372"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4372" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Plot the results</span></span>
<span id="cb182-4373"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4373" aria-hidden="true" tabindex="-1"></a>        <span class="co">#sizeGrWindow(9, 5)</span></span>
<span id="cb182-4374"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4374" aria-hidden="true" tabindex="-1"></a>        <span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>));</span>
<span id="cb182-4375"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4375" aria-hidden="true" tabindex="-1"></a>        cex1 <span class="ot">=</span> <span class="fl">0.9</span>;</span>
<span id="cb182-4376"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4377"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4377" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Scale-free topology fit index as a function of the soft-thresholding power</span></span>
<span id="cb182-4378"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4378" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">3</span>])<span class="sc">*</span>sft<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],<span class="at">xlab=</span><span class="st">&quot;Soft Threshold (power)&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Scale Free Topology Model Fit, signed R^2&quot;</span>,<span class="at">type=</span><span class="st">&quot;n&quot;</span>, <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Scale independence&quot;</span>));</span>
<span id="cb182-4379"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4379" aria-hidden="true" tabindex="-1"></a>        <span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">3</span>])<span class="sc">*</span>sft<span class="sc">$</span>fitIndices[,<span class="dv">2</span>],<span class="at">labels=</span>powers,<span class="at">cex=</span>cex1,<span class="at">col=</span><span class="st">&quot;red&quot;</span>);</span>
<span id="cb182-4380"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4381"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4381" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Red line corresponds to using an R^2 cut-off</span></span>
<span id="cb182-4382"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4382" aria-hidden="true" tabindex="-1"></a>        <span class="fu">abline</span>(<span class="at">h=</span><span class="fl">0.80</span>,<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span>
<span id="cb182-4383"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4384"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4384" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Mean connectivity as a function of the soft-thresholding power</span></span>
<span id="cb182-4385"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4385" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], sft<span class="sc">$</span>fitIndices[,<span class="dv">5</span>],<span class="at">xlab=</span><span class="st">&quot;Soft Threshold (power)&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Mean Connectivity&quot;</span>, <span class="at">type=</span><span class="st">&quot;n&quot;</span>,<span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Mean connectivity&quot;</span>))</span>
<span id="cb182-4386"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4386" aria-hidden="true" tabindex="-1"></a>        <span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[,<span class="dv">1</span>], sft<span class="sc">$</span>fitIndices[,<span class="dv">5</span>], <span class="at">labels=</span>powers, <span class="at">cex=</span>cex1,<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span>
<span id="cb182-4387"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4387" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-4388"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4389"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4390"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4390" aria-hidden="true" tabindex="-1"></a>modulePlot <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb182-4391"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4391" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-4392"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4392" aria-hidden="true" tabindex="-1"></a>        diss1 <span class="ot">=</span> <span class="dv">1</span><span class="sc">-</span>wgcna.out<span class="sc">$</span>TOM;</span>
<span id="cb182-4393"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4393" aria-hidden="true" tabindex="-1"></a>        dynamicColors <span class="ot">=</span> wgcna.out<span class="sc">$</span>dynamicColors</span>
<span id="cb182-4394"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4394" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-4395"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4395" aria-hidden="true" tabindex="-1"></a>        hier1<span class="ot">=</span><span class="fu">flashClust</span>(<span class="fu">as.dist</span>(diss1), <span class="at">method=</span><span class="st">&quot;average&quot;</span> )</span>
<span id="cb182-4396"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4396" aria-hidden="true" tabindex="-1"></a>        <span class="co">#set the diagonal of the dissimilarity to NA </span></span>
<span id="cb182-4397"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4397" aria-hidden="true" tabindex="-1"></a>        <span class="fu">diag</span>(diss1) <span class="ot">=</span> <span class="cn">NA</span>;</span>
<span id="cb182-4398"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4398" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plotDendroAndColors</span>(hier1, dynamicColors, <span class="st">&quot;Dynamic Tree Cut&quot;</span>, <span class="at">dendroLabels =</span> <span class="cn">FALSE</span>, <span class="at">hang =</span> <span class="fl">0.03</span>, <span class="at">addGuide =</span> <span class="cn">TRUE</span>, <span class="at">guideHang =</span> <span class="fl">0.05</span>, <span class="at">main =</span> <span class="st">&quot;Gene dendrogram and module colors&quot;</span>)</span>
<span id="cb182-4399"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4400"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4400" aria-hidden="true" tabindex="-1"></a>        <span class="co">#Visualize the Tom plot. Raise the dissimilarity matrix to the power of 4 to bring out the module structure</span></span>
<span id="cb182-4401"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4401" aria-hidden="true" tabindex="-1"></a>        <span class="co">#TOMplot(diss1, hier1, colorCode)           </span></span>
<span id="cb182-4402"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4402" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-4403"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4404"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4404" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4405"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4405" aria-hidden="true" tabindex="-1"></a>listWGCNA.Modules <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb182-4406"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4407"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4407" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">is.null</span>(wgcna.out ) ){ <span class="co"># if sample info is uploaded and correctly parsed.</span></span>
<span id="cb182-4408"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4408" aria-hidden="true" tabindex="-1"></a>           <span class="fu">return</span>(<span class="cn">NULL</span>)    </span>
<span id="cb182-4409"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4409" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> { </span>
<span id="cb182-4410"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4410" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">dim</span>(wgcna.out<span class="sc">$</span>moduleInfo)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>  ) {  <span class="co"># if no module</span></span>
<span id="cb182-4411"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4411" aria-hidden="true" tabindex="-1"></a>                <span class="fu">return</span>(<span class="cn">NULL</span>) </span>
<span id="cb182-4412"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4412" aria-hidden="true" tabindex="-1"></a>            }   <span class="cf">else</span>  { </span>
<span id="cb182-4413"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4413" aria-hidden="true" tabindex="-1"></a>                    modules <span class="ot">=</span> <span class="fu">unique</span>(wgcna.out<span class="sc">$</span>moduleInfo[, <span class="fu">c</span>(<span class="st">&quot;dynamicMods&quot;</span>,<span class="st">&quot;dynamicColors&quot;</span>)] )</span>
<span id="cb182-4414"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4414" aria-hidden="true" tabindex="-1"></a>                    moduleList <span class="ot">=</span> <span class="fu">apply</span>(modules,<span class="dv">1</span>,paste,<span class="at">collapse=</span><span class="st">&quot;. &quot;</span>)</span>
<span id="cb182-4415"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4415" aria-hidden="true" tabindex="-1"></a>                    moduleList  <span class="ot">=</span> <span class="fu">paste0</span>( moduleList, <span class="st">&quot; (&quot;</span>, <span class="fu">table</span>(wgcna.out<span class="sc">$</span>moduleInfo[,<span class="st">&quot;dynamicMods&quot;</span>] ),<span class="st">&quot; genes)&quot;</span>  )</span>
<span id="cb182-4416"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4416" aria-hidden="true" tabindex="-1"></a>                    moduleList <span class="ot">=</span> <span class="fu">c</span>(moduleList,<span class="st">&quot;Entire network&quot;</span>)</span>
<span id="cb182-4417"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4417" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">return</span>(moduleList)</span>
<span id="cb182-4418"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4419"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4420"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4420" aria-hidden="true" tabindex="-1"></a>                }</span>
<span id="cb182-4421"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4421" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4422"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4422" aria-hidden="true" tabindex="-1"></a>        }           </span>
<span id="cb182-4423"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4423" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-4424"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4425"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4426"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4426" aria-hidden="true" tabindex="-1"></a>moduleNetwork <span class="ot">&lt;-</span> <span class="cf">function</span>(){</span>
<span id="cb182-4427"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4428"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4428" aria-hidden="true" tabindex="-1"></a>        outfile <span class="ot">&lt;-</span> <span class="fu">tempfile</span>(<span class="at">fileext=</span><span class="st">&#39;.txt&#39;</span>)</span>
<span id="cb182-4429"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4429" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-4430"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4430" aria-hidden="true" tabindex="-1"></a>        <span class="co">#module = gsub(&quot;.* &quot;,&quot;&quot;,input_selectWGCNA.Module)</span></span>
<span id="cb182-4431"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4431" aria-hidden="true" tabindex="-1"></a>        module <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(input_selectWGCNA.Module,<span class="st">&quot; &quot;</span> ) )[<span class="dv">2</span>]</span>
<span id="cb182-4432"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4432" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-4433"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4433" aria-hidden="true" tabindex="-1"></a>        moduleColors <span class="ot">=</span> wgcna.out<span class="sc">$</span>dynamicColors </span>
<span id="cb182-4434"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4434" aria-hidden="true" tabindex="-1"></a>        inModule <span class="ot">=</span> (moduleColors<span class="sc">==</span>module);</span>
<span id="cb182-4435"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4435" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-4436"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4436" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( input_selectWGCNA.Module <span class="sc">==</span> <span class="st">&quot;Entire network&quot;</span>)</span>
<span id="cb182-4437"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4437" aria-hidden="true" tabindex="-1"></a>          inModule <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">TRUE</span>, <span class="fu">length</span>(inModule))</span>
<span id="cb182-4438"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4438" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-4439"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4439" aria-hidden="true" tabindex="-1"></a>        datExpr <span class="ot">=</span> <span class="fu">t</span>(wgcna.out<span class="sc">$</span>x )</span>
<span id="cb182-4440"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4440" aria-hidden="true" tabindex="-1"></a>        probes <span class="ot">=</span> <span class="fu">colnames</span>(datExpr)</span>
<span id="cb182-4441"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4441" aria-hidden="true" tabindex="-1"></a>        modProbes <span class="ot">=</span> probes[inModule];</span>
<span id="cb182-4442"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4442" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-4443"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4443" aria-hidden="true" tabindex="-1"></a>        modTOM <span class="ot">=</span> wgcna.out<span class="sc">$</span>TOM[inModule, inModule];</span>
<span id="cb182-4444"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4444" aria-hidden="true" tabindex="-1"></a>        <span class="fu">dimnames</span>(modTOM) <span class="ot">=</span> <span class="fu">list</span>(modProbes, modProbes)</span>
<span id="cb182-4445"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4446"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4446" aria-hidden="true" tabindex="-1"></a>        nTop <span class="ot">=</span> input_topGenesNetwork;</span>
<span id="cb182-4447"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4447" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-4448"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4448" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( nTop <span class="sc">&gt;</span> <span class="dv">1000</span>) nTop <span class="ot">=</span> <span class="dv">1000</span>; </span>
<span id="cb182-4449"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4449" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-4450"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4450" aria-hidden="true" tabindex="-1"></a>        IMConn <span class="ot">=</span> <span class="fu">softConnectivity</span>(datExpr[, modProbes]);</span>
<span id="cb182-4451"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4451" aria-hidden="true" tabindex="-1"></a>        top <span class="ot">=</span> (<span class="fu">rank</span>(<span class="sc">-</span>IMConn) <span class="sc">&lt;=</span> nTop)</span>
<span id="cb182-4452"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4452" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-4453"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4453" aria-hidden="true" tabindex="-1"></a>        <span class="co"># adding symbols </span></span>
<span id="cb182-4454"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4454" aria-hidden="true" tabindex="-1"></a>        probeToGene <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb182-4455"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4455" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>( input_selectGO5 <span class="sc">!=</span> <span class="st">&quot;ID not recognized!&quot;</span> <span class="sc">&amp;</span> input_selectOrg <span class="sc">!=</span> <span class="st">&quot;NEW&quot;</span>)</span>
<span id="cb182-4456"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4456" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="fu">sum</span>(<span class="fu">is.na</span>( allGeneInfo.out<span class="sc">$</span>symbol ) )<span class="sc">/</span> <span class="fu">dim</span>( allGeneInfo.out )[<span class="dv">1</span>] <span class="sc">&lt;</span>.<span class="dv">5</span> ) { <span class="co"># if more than 50% genes has symbol</span></span>
<span id="cb182-4457"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4457" aria-hidden="true" tabindex="-1"></a>            probeToGene <span class="ot">=</span> allGeneInfo.out[,<span class="fu">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="st">&quot;symbol&quot;</span>)]</span>
<span id="cb182-4458"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4458" aria-hidden="true" tabindex="-1"></a>            probeToGene<span class="sc">$</span>symbol <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>,<span class="st">&quot;&quot;</span>,probeToGene<span class="sc">$</span>symbol)</span>
<span id="cb182-4459"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4459" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4460"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4460" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">which</span>( <span class="fu">is.na</span>(probeToGene<span class="sc">$</span>symbol) <span class="sc">|</span></span>
<span id="cb182-4461"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4461" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">nchar</span>(probeToGene<span class="sc">$</span>symbol)<span class="sc">&lt;</span><span class="dv">2</span> <span class="sc">|</span> </span>
<span id="cb182-4462"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4462" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">toupper</span>(probeToGene<span class="sc">$</span>symbol)<span class="sc">==</span><span class="st">&quot;NA&quot;</span> <span class="sc">|</span>  </span>
<span id="cb182-4463"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4463" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">toupper</span>(probeToGene<span class="sc">$</span>symbol)<span class="sc">==</span><span class="st">&quot;0&quot;</span>  )             </span>
<span id="cb182-4464"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4464" aria-hidden="true" tabindex="-1"></a>            probeToGene[ix,<span class="dv">2</span>] <span class="ot">=</span> probeToGene[ix,<span class="dv">1</span>]  <span class="co"># use gene ID</span></span>
<span id="cb182-4465"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4466"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4466" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-4467"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4467" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-4468"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4468" aria-hidden="true" tabindex="-1"></a>        net <span class="ot">&lt;-</span> modTOM[top,top] <span class="sc">&gt;</span>input_edgeThreshold </span>
<span id="cb182-4469"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4470"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4470" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-4471"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4471" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(net)[<span class="dv">1</span>])  <span class="co"># remove self connection</span></span>
<span id="cb182-4472"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4472" aria-hidden="true" tabindex="-1"></a>            net[i,i] <span class="ot">=</span> <span class="cn">FALSE</span></span>
<span id="cb182-4473"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4473" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(probeToGene) <span class="sc">&amp;</span> <span class="sc">!</span>input_noIDConversion) { <span class="co"># if gene symbol exist</span></span>
<span id="cb182-4474"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4474" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">match</span>( <span class="fu">colnames</span>(net), probeToGene[,<span class="dv">1</span>])     </span>
<span id="cb182-4475"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4475" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(net) <span class="ot">=</span> probeToGene[ix,<span class="dv">2</span>]</span>
<span id="cb182-4476"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4476" aria-hidden="true" tabindex="-1"></a>            ix <span class="ot">=</span> <span class="fu">match</span>( <span class="fu">rownames</span>(net), probeToGene[,<span class="dv">1</span>])     </span>
<span id="cb182-4477"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4477" aria-hidden="true" tabindex="-1"></a>            <span class="fu">rownames</span>(net) <span class="ot">=</span> probeToGene[ix,<span class="dv">2</span>]       </span>
<span id="cb182-4478"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4478" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb182-4479"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4479" aria-hidden="true" tabindex="-1"></a>        <span class="fu">library</span>(igraph,<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb182-4480"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4480" aria-hidden="true" tabindex="-1"></a>        <span class="co">#plot(graph_from_data_frame(d=data.frame(1:10,ncol=2)  ,directed=F) )</span></span>
<span id="cb182-4481"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4481" aria-hidden="true" tabindex="-1"></a>        <span class="co"># http://www.kateto.net/wp-content/uploads/2016/01/NetSciX_2016_Workshop.pdf</span></span>
<span id="cb182-4482"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4482" aria-hidden="true" tabindex="-1"></a>        <span class="fu">plot</span>( <span class="fu">graph_from_adjacency_matrix</span>( net, <span class="at">mod =</span><span class="st">&quot;undirected&quot;</span> ), </span>
<span id="cb182-4483"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4483" aria-hidden="true" tabindex="-1"></a>                <span class="at">vertex.label.color=</span><span class="st">&quot;black&quot;</span>, <span class="at">vertex.label.dist=</span><span class="dv">3</span>,<span class="at">vertex.size=</span><span class="dv">7</span>)</span>
<span id="cb182-4484"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4484" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-4485"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4485" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb182-4486"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4486" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4487"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4488"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4488" aria-hidden="true" tabindex="-1"></a>networkModuleGO <span class="ot">&lt;-</span> <span class="cf">function</span>(){      </span>
<span id="cb182-4489"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4490"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4490" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb182-4491"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4491" aria-hidden="true" tabindex="-1"></a>            <span class="co">#module = gsub(&quot;.* &quot;,&quot;&quot;,input_selectWGCNA.Module)</span></span>
<span id="cb182-4492"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4492" aria-hidden="true" tabindex="-1"></a>            module <span class="ot">=</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(input_selectWGCNA.Module,<span class="st">&quot; &quot;</span> ) )[<span class="dv">2</span>]    </span>
<span id="cb182-4493"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4493" aria-hidden="true" tabindex="-1"></a>            moduleColors <span class="ot">=</span> wgcna.out<span class="sc">$</span>dynamicColors </span>
<span id="cb182-4494"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4494" aria-hidden="true" tabindex="-1"></a>            inModule <span class="ot">=</span> (moduleColors<span class="sc">==</span>module);</span>
<span id="cb182-4495"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4495" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4496"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4496" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( input_selectWGCNA.Module <span class="sc">==</span> <span class="st">&quot;Entire network&quot;</span>)</span>
<span id="cb182-4497"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4497" aria-hidden="true" tabindex="-1"></a>              inModule <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">TRUE</span>, <span class="fu">length</span>(inModule))</span>
<span id="cb182-4498"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4499"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4499" aria-hidden="true" tabindex="-1"></a>            probes <span class="ot">=</span> <span class="fu">rownames</span>(wgcna.out<span class="sc">$</span>x   )</span>
<span id="cb182-4500"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4500" aria-hidden="true" tabindex="-1"></a>            query  <span class="ot">=</span> probes[inModule];</span>
<span id="cb182-4501"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4501" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-4502"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4502" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">length</span>(query)  <span class="sc">&lt;=</span> minGenesEnrichment) <span class="fu">return</span>(NoSig) </span>
<span id="cb182-4503"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4504"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4504" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4505"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4506"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4506" aria-hidden="true" tabindex="-1"></a>            result <span class="ot">&lt;-</span> <span class="fu">findOverlapGMT</span>( query, GeneSets.out,<span class="dv">1</span>) </span>
<span id="cb182-4507"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4507" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb182-4508"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4508" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb182-4509"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4509" aria-hidden="true" tabindex="-1"></a>            result<span class="sc">$</span>Genes <span class="ot">=</span> <span class="st">&quot;Up regulated&quot;</span></span>
<span id="cb182-4510"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4510" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4511"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4512"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4512" aria-hidden="true" tabindex="-1"></a>            results1 <span class="ot">=</span> result</span>
<span id="cb182-4513"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4513" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> ( <span class="fu">is.null</span>( results1) ) <span class="fu">return</span> (NoSig)</span>
<span id="cb182-4514"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4514" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>( <span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">&lt;</span><span class="dv">5</span> ) <span class="fu">return</span>(NoSig)  <span class="co"># Returns a data frame: &quot;No significant results found!&quot;</span></span>
<span id="cb182-4515"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4515" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4516"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4516" aria-hidden="true" tabindex="-1"></a>            results1<span class="ot">=</span> results1[,<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">4</span>)]</span>
<span id="cb182-4517"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4517" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(results1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;List&quot;</span>,<span class="st">&quot;FDR&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;GO terms or pathways&quot;</span>)</span>
<span id="cb182-4518"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4518" aria-hidden="true" tabindex="-1"></a>            minFDR <span class="ot">=</span> <span class="fl">0.01</span></span>
<span id="cb182-4519"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4519" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">min</span>(results1<span class="sc">$</span>FDR) <span class="sc">&gt;</span> minFDR ) results1 <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="st">&quot;No signficant enrichment found.&quot;</span>) <span class="cf">else</span></span>
<span id="cb182-4520"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4520" aria-hidden="true" tabindex="-1"></a>            results1 <span class="ot">=</span> results1[<span class="fu">which</span>(results1<span class="sc">$</span>FDR <span class="sc">&lt;</span> minFDR),]</span>
<span id="cb182-4521"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4521" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4522"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4522" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb182-4523"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4523" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4524"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4524" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span>(<span class="fu">dim</span>(results1)[<span class="dv">2</span>] <span class="sc">!=</span> <span class="dv">4</span>) <span class="fu">return</span>(NoSig)</span>
<span id="cb182-4525"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4525" aria-hidden="true" tabindex="-1"></a>            <span class="fu">colnames</span>(results1)<span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Direction&quot;</span>,<span class="st">&quot;adj.Pval&quot;</span>,<span class="st">&quot;Genes&quot;</span>,<span class="st">&quot;Pathways&quot;</span>)</span>
<span id="cb182-4526"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4526" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4527"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4527" aria-hidden="true" tabindex="-1"></a>            results1<span class="sc">$</span>adj.Pval <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(<span class="st">&quot;%-2.1e&quot;</span>,<span class="fu">as.numeric</span>(results1<span class="sc">$</span>adj.Pval) )</span>
<span id="cb182-4528"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4528" aria-hidden="true" tabindex="-1"></a>            results1[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">as.character</span>(results1[,<span class="dv">1</span>])</span>
<span id="cb182-4529"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4529" aria-hidden="true" tabindex="-1"></a>            tem <span class="ot">&lt;-</span> results1[,<span class="dv">1</span>]</span>
<span id="cb182-4530"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4531"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4531" aria-hidden="true" tabindex="-1"></a>            results1[ <span class="fu">duplicated</span> (results1[,<span class="dv">1</span>] ),<span class="dv">1</span> ] <span class="ot">&lt;-</span> <span class="st">&quot;&quot;</span></span>
<span id="cb182-4532"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4532" aria-hidden="true" tabindex="-1"></a>            </span>
<span id="cb182-4533"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4533" aria-hidden="true" tabindex="-1"></a>            results1[,<span class="sc">-</span><span class="dv">1</span>]           </span>
<span id="cb182-4534"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html#cb182-4534" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="rna-seq-mkdata.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="wgcna-rna-seq-blog.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Bioinformatics_RNA-seq.pdf", "Bioinformatics_RNA-seq.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
