<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 RA Download, QC, and Trimming | Bioinformatics_RNA-seq</title>
  <meta name="description" content="RNA-seq notes and trials" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 RA Download, QC, and Trimming | Bioinformatics_RNA-seq" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="RNA-seq notes and trials" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 RA Download, QC, and Trimming | Bioinformatics_RNA-seq" />
  
  <meta name="twitter:description" content="RNA-seq notes and trials" />
  

<meta name="author" content="J. Marcelo Rosales R." />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="rna-seq-blog.html"/>
<link rel="next" href="methods.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>




<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to RNA sequencing (RNA_seq)</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#downloading-and-setting-up-conda-environments."><i class="fa fa-check"></i><b>1.1</b> Downloading and setting up conda environments.</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#install-miniconda-or-anaconda."><i class="fa fa-check"></i><b>1.2</b> Install Miniconda, or Anaconda.</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#anacondaminiconda-install-summary-of-steps."><i class="fa fa-check"></i><b>1.2.1</b> Anaconda/Miniconda install Summary of Steps.</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#conda-activate-error"><i class="fa fa-check"></i><b>1.2.2</b> Conda activate error</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#alignment-procedures-and-files."><i class="fa fa-check"></i><b>1.3</b> Alignment Procedures and files.</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#using-tophat-early-version"><i class="fa fa-check"></i><b>1.3.1</b> Using Tophat (early version)</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#using-tophat"><i class="fa fa-check"></i><b>1.3.2</b> Using Tophat</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#using-star"><i class="fa fa-check"></i><b>1.3.3</b> Using STAR</a></li>
<li class="chapter" data-level="1.3.4" data-path="index.html"><a href="index.html#custompersonal-way-to-download-the-mus_musculus_ncbi_build37.2.tar.gz-file."><i class="fa fa-check"></i><b>1.3.4</b> Custom/personal way to Download the Mus_musculus_NCBI_build37.2.tar.gz file.</a></li>
<li class="chapter" data-level="1.3.5" data-path="index.html"><a href="index.html#get-the-sample-files."><i class="fa fa-check"></i><b>1.3.5</b> Get the Sample files.</a></li>
<li class="chapter" data-level="1.3.6" data-path="index.html"><a href="index.html#starting-to-analize-fastq.gz-files"><i class="fa fa-check"></i><b>1.3.6</b> Starting to analize fastq.gz files</a></li>
<li class="chapter" data-level="1.3.7" data-path="index.html"><a href="index.html#trimming-with-trimmomatic"><i class="fa fa-check"></i><b>1.3.7</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="1.3.8" data-path="index.html"><a href="index.html#using-star-to-map-genes"><i class="fa fa-check"></i><b>1.3.8</b> Using STAR to map genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="rna-seq-blog.html"><a href="rna-seq-blog.html"><i class="fa fa-check"></i><b>2</b> RNA-seq Blog</a>
<ul>
<li class="chapter" data-level="2.1" data-path="rna-seq-blog.html"><a href="rna-seq-blog.html#WGCNA"><i class="fa fa-check"></i><b>2.1</b> Introduction to the WGCNA Rpackage</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="rna-seq-blog.html"><a href="rna-seq-blog.html#wgcna-rpackage-installation-procedure."><i class="fa fa-check"></i><b>2.1.1</b> WGCNA Rpackage Installation procedure.</a></li>
<li class="chapter" data-level="2.1.2" data-path="rna-seq-blog.html"><a href="rna-seq-blog.html#final-installation-workflow"><i class="fa fa-check"></i><b>2.1.2</b> Final installation workflow:</a></li>
<li class="chapter" data-level="2.1.3" data-path="rna-seq-blog.html"><a href="rna-seq-blog.html#wgcna-tutorial."><i class="fa fa-check"></i><b>2.1.3</b> WGCNA tutorial.</a></li>
<li class="chapter" data-level="2.1.4" data-path="rna-seq-blog.html"><a href="rna-seq-blog.html#gene-expression-analysis"><i class="fa fa-check"></i><b>2.1.4</b> 3.4 Gene expression analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="rna-seq-blog.html"><a href="rna-seq-blog.html#rna-seq-analisys-with-kakus-data-sample."><i class="fa fa-check"></i><b>2.2</b> RNA-seq Analisys with Kaku’s Data sample.</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="rna-seq-blog.html"><a href="rna-seq-blog.html#experimental-workflow."><i class="fa fa-check"></i><b>2.2.1</b> Experimental Workflow.</a></li>
<li class="chapter" data-level="2.2.2" data-path="rna-seq-blog.html"><a href="rna-seq-blog.html#data-analysis."><i class="fa fa-check"></i><b>2.2.2</b> Data Analysis.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="ra-download-qc-and-trimming.html"><a href="ra-download-qc-and-trimming.html"><i class="fa fa-check"></i><b>3</b> RA Download, QC, and Trimming</a></li>
<li class="chapter" data-level="4" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>4</b> Methods</a></li>
<li class="chapter" data-level="5" data-path="applications.html"><a href="applications.html"><i class="fa fa-check"></i><b>5</b> Applications</a>
<ul>
<li class="chapter" data-level="5.1" data-path="applications.html"><a href="applications.html#example-one"><i class="fa fa-check"></i><b>5.1</b> Example one</a></li>
<li class="chapter" data-level="5.2" data-path="applications.html"><a href="applications.html#example-two"><i class="fa fa-check"></i><b>5.2</b> Example two</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>6</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics_RNA-seq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="ra-download-qc-and-trimming" class="section level1" number="3">
<h1><span class="header-section-number">Chapter 3</span> RA Download, QC, and Trimming</h1>
<p><a href="https://www.youtube.com/watch?v=UkrUTgAKFiE&amp;list=PL-0fKymgD8L8qW7SQDaEf3lrSyUkEkR1k&amp;index=2">Bioinformatics - SRA Download, QC, and Trimming</a>.</p>
<ol style="list-style-type: decimal">
<li>Create environment.<br />
</li>
<li>Init environment<br />
</li>
<li><code>conda install sra-tools</code> This will allow us to pull directly from the sra page. (where the sample ex. Files)<br />
(ア) Go to <a href="https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP061386&amp;o=acc_s%3Aa">page</a>,<br />
(イ) Select only files needed.<br />
(ウ) Then in the Selected &gt; click on Accession List &gt;<br />
(エ) Save<br />
(オ) Drag onto the file structure, so its now in system<br />
(カ) <code>ls</code> to see the files (SRR_Acc_List.txt)<br />
(キ) Check ‘head SRR_Acc_List.txt’ command<br />
(ク) Also you can use the <code>vi SRR_Acc_List.txt</code> command to see the list in the file. Ex.
① SRR2121770
② SRR2121771
③ SRR2121774
④ SRR2121778
⑤ Etc.</li>
<li>Now that we have the list of files, we have to individually download each one of them.<br />
(ア) One way to do that is with the <code>fastq-dump --gzi --split-file SRR2121770 &amp;</code> command.<br />
① This command will take the SRR2121770 file, and save it as a .gzi file (to save space),
② it will split the file in forward (1) and Reverse (2), then
③ the “&amp;” sign means that it will run this command in the background.<br />
(イ) This will take some time since the file is big. A way to see if the process is finished or not is with the command <code>jobs</code>, which gives a list of the jobs in process or nothing if there are not jobs running.<br />
(ウ) Another way to do it is with the pipe command, so we don’t have to do this process for each file. Use the command <code>cat SRR_Acc_List.txt | parallel fastq-dump --gzi --split-file {}</code>.<br />
① This will take the SRR_Acc_List.txt and catalog the list.<br />
② Then it will take each of the element in the list “{}” slipt them, and save them as gzi.<br />
③ In the rawReads folder???.. you will have a list of .fastq.gz_1 and .fastq.gz_2 for each element of the list. <code>…rawReads$ ls</code>.<br />
④ Each time something is changed in the files, create a new folder and save there so the originals are not modified. Ex. Trimmed.<br />
</li>
<li>Once all files are downloaded, install multiqc. <code>conda install multiqc</code>. (if not installed)<br />
</li>
<li>Before running fastqc, create an output directory where to save the processed files. <code>mkdir rawFastQC</code> (make sure you are in the right directory before making this folder).<br />
</li>
<li>Then run fastqc: ` fastqc -t 64 rawReads/*fastqc.gz -o rawFastQC’<br />
(ア) This means, process files with fastqc using 64 treads (-t 64) windows/linux virtual machine;<br />
(イ) by taking all fastqc.gz files in the folder rawReads;<br />
(ウ) and save in the output directory (-o) called rawFastQC.<br />
(エ) This process will take a long time depending on the number and size of the files.<br />
</li>
<li>Once finished <code>cd rawFastQC</code> and check the files <code>ls</code><br />
</li>
<li>Then inside the forlder, run multiqc in <code>multiqc</code>.<br />
</li>
<li>Once this finished, inside the folder a new “multiqc_report.html” will be created.<br />
</li>
<li>Open the files and compare.</li>
</ol>
<p>22:29
Once this quality control is finished we can pass to the trimming of the samples.</p>
<ol style="list-style-type: decimal">
<li>Install trimmomatic, <code>conda install trimmomatic</code> (if not installed).<br />
</li>
<li>If we are going to trim the files, we need to make a new folder, so exit rawFastQC folder. <code>cd..</code> and <code>mkdir trimmedReads</code>, we are going to point trimmomatic the this folder.<br />
</li>
<li>We are going to pull our true adapter sequences to our working directory. Then it will be a lot easier to point trimmomatic to that vs havint to type in the really long path. <code>cp ~/miniconda3/envs/name_of_environment/share/trimmomatic-0.39-1/adapters/TrueSeq3-PE-2.fa .</code> (the last point is important, it means that the command will the done in this folder).<br />
</li>
<li>Now run the trimmomatic.</li>
</ol>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="rna-seq-blog.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="methods.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Bioinformatics_RNA-seq.pdf", "Bioinformatics_RNA-seq.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
