<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>References | Bioinformatics_RNA-seq</title>
  <meta name="description" content="RNA-seq notes and trials" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="References | Bioinformatics_RNA-seq" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="RNA-seq notes and trials" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="References | Bioinformatics_RNA-seq" />
  
  <meta name="twitter:description" content="RNA-seq notes and trials" />
  

<meta name="author" content="J. Marcelo Rosales R." />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="external-ref.html"/>

<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to RNA sequencing (RNA_seq)</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#downloading-and-setting-up-conda-environments."><i class="fa fa-check"></i><b>1.1</b> Downloading and setting up conda environments.</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#install-miniconda-or-anaconda."><i class="fa fa-check"></i><b>1.2</b> Install Miniconda, or Anaconda.</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#conda-terminal-commands."><i class="fa fa-check"></i><b>1.2.1</b> Conda terminal commands.</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#anacondaminiconda-install-summary-of-steps."><i class="fa fa-check"></i><b>1.2.2</b> Anaconda/Miniconda install Summary of Steps.</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#conda-activate-error"><i class="fa fa-check"></i><b>1.2.3</b> Conda activate error</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#alignment-procedures-and-files."><i class="fa fa-check"></i><b>1.3</b> Alignment Procedures and files.</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#using-tophat-early-version"><i class="fa fa-check"></i><b>1.3.1</b> Using Tophat (early version)</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#using-tophat"><i class="fa fa-check"></i><b>1.3.2</b> Using Tophat</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#using-star"><i class="fa fa-check"></i><b>1.3.3</b> Using STAR</a></li>
<li class="chapter" data-level="1.3.4" data-path="index.html"><a href="index.html#Mus-musculus-NCBI-build37-2"><i class="fa fa-check"></i><b>1.3.4</b> Custom/personal way to Download the Mus_musculus_NCBI_build37.2.tar.gz file.</a></li>
<li class="chapter" data-level="1.3.5" data-path="index.html"><a href="index.html#get-the-sample-files."><i class="fa fa-check"></i><b>1.3.5</b> Get the Sample files.</a></li>
<li class="chapter" data-level="1.3.6" data-path="index.html"><a href="index.html#starting-to-analize-fastq.gz-files"><i class="fa fa-check"></i><b>1.3.6</b> Starting to analize fastq.gz files</a></li>
<li class="chapter" data-level="1.3.7" data-path="index.html"><a href="index.html#trimming-with-trimmomatic"><i class="fa fa-check"></i><b>1.3.7</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="1.3.8" data-path="index.html"><a href="index.html#using-star-to-map-genes"><i class="fa fa-check"></i><b>1.3.8</b> Using STAR to map genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html"><i class="fa fa-check"></i><b>2</b> File Download (or SRA), QC, and Trimming</a>
<ul>
<li class="chapter" data-level="2.1" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#trimmomatic"><i class="fa fa-check"></i><b>2.1</b> Trimmomatic</a></li>
<li class="chapter" data-level="2.2" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#indexing-with-star"><i class="fa fa-check"></i><b>2.2</b> Indexing with STAR</a></li>
<li class="chapter" data-level="2.3" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#microbial-contamination"><i class="fa fa-check"></i><b>2.3</b> Microbial contamination</a></li>
<li class="chapter" data-level="2.4" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#feature-counts."><i class="fa fa-check"></i><b>2.4</b> Feature counts.</a></li>
<li class="chapter" data-level="2.5" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#setting-r-environment."><i class="fa fa-check"></i><b>2.5</b> Setting R environment.</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>3</b> Methods</a></li>
<li class="chapter" data-level="4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><i class="fa fa-check"></i><b>4</b> RNA Seq Differentially Expressed, GO Enrichment, and Pathway analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#background"><i class="fa fa-check"></i><b>4.1</b> <strong>Background</strong></a></li>
<li class="chapter" data-level="4.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#cleaning-data"><i class="fa fa-check"></i><b>4.2</b> <strong>Cleaning Data</strong></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#programs-used"><i class="fa fa-check"></i><b>4.2.1</b> Programs used?</a></li>
<li class="chapter" data-level="4.2.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#acquiring-sequences"><i class="fa fa-check"></i><b>4.2.2</b> Acquiring sequences</a></li>
<li class="chapter" data-level="4.2.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#fastqc"><i class="fa fa-check"></i><b>4.2.3</b> FastQC</a></li>
<li class="chapter" data-level="4.2.4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#trimming-with-trimmomatic-1"><i class="fa fa-check"></i><b>4.2.4</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="4.2.5" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#alignment"><i class="fa fa-check"></i><b>4.2.5</b> Alignment</a></li>
<li class="chapter" data-level="4.2.6" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#assembling-transcripts-with-cufflinks"><i class="fa fa-check"></i><b>4.2.6</b> Assembling transcripts with Cufflinks</a></li>
<li class="chapter" data-level="4.2.7" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#checking-for-contamination"><i class="fa fa-check"></i><b>4.2.7</b> Checking for Contamination</a></li>
<li class="chapter" data-level="4.2.8" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#counting-transcripts"><i class="fa fa-check"></i><b>4.2.8</b> Counting Transcripts</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#differentially-expressed-sequence-identification"><i class="fa fa-check"></i><b>4.3</b> <strong>Differentially Expressed Sequence Identification</strong></a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#analyzing-reads-counts"><i class="fa fa-check"></i><b>4.3.1</b> Analyzing Reads Counts</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e.g.-p53-control-and-ir-and-p53---control-and-ir"><i class="fa fa-check"></i><b>4.4</b> Interactions cause a difference between the lfc betwen pooled data, e.g. p53+/+ (control and IR) and p53-/- (control and IR)</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>4.4.1</b> Exploratory Data Analysis</a></li>
<li class="chapter" data-level="4.4.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#identification-of-differentially-expressed-genes"><i class="fa fa-check"></i><b>4.4.2</b> Identification of Differentially Expressed Genes</a></li>
<li class="chapter" data-level="4.4.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#gene-annotations"><i class="fa fa-check"></i><b>4.4.3</b> Gene Annotations</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#go-enrichment-analysis-using-gostats"><i class="fa fa-check"></i><b>4.5</b> 7. GO Enrichment analysis using GOstats</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#go-enrichment-analysis-of-downregulated-genes"><i class="fa fa-check"></i><b>4.5.1</b> GO Enrichment analysis of downregulated genes</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-analysis-using-expression-data"><i class="fa fa-check"></i><b>4.6</b> 8. Pathway analysis using expression data</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#prepare-data"><i class="fa fa-check"></i><b>4.6.1</b> Prepare data</a></li>
<li class="chapter" data-level="4.6.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#kegg-pathways"><i class="fa fa-check"></i><b>4.6.2</b> KEGG pathways</a></li>
<li class="chapter" data-level="4.6.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-oxidative-phosphorylation."><i class="fa fa-check"></i><b>4.6.3</b> Pathway and regulation of genes for Oxidative phosphorylation.</a></li>
<li class="chapter" data-level="4.6.4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-glycosylphosphatidylinositolgpi-anchor-biosynthesis."><i class="fa fa-check"></i><b>4.6.4</b> Pathway and regulation of genes for Glycosylphosphatidylinositol(GPI)-anchor biosynthesis.</a></li>
<li class="chapter" data-level="4.6.5" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-rna-polymerase."><i class="fa fa-check"></i><b>4.6.5</b> Pathway and regulation of genes for RNA polymerase.</a></li>
<li class="chapter" data-level="4.6.6" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-nucleotide-excision-repair."><i class="fa fa-check"></i><b>4.6.6</b> Pathway and regulation of genes for Nucleotide excision repair.</a></li>
<li class="chapter" data-level="4.6.7" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-non-homologous-end-joining."><i class="fa fa-check"></i><b>4.6.7</b> Pathway and regulation of genes for Non-homologous end-joining.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html"><i class="fa fa-check"></i><b>5</b> RNA-Seq-MKdata</a>
<ul>
<li class="chapter" data-level="5.1" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#pca"><i class="fa fa-check"></i><b>5.1</b> PCA</a></li>
<li class="chapter" data-level="5.2" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#differential-expression"><i class="fa fa-check"></i><b>5.2</b> Differential expression</a></li>
<li class="chapter" data-level="5.3" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#rna-seq-scienceparksg"><i class="fa fa-check"></i><b>5.3</b> RNA-seq (ScienceparkSG)</a></li>
<li class="chapter" data-level="5.4" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#false-discovery-rates-fdr-benjamini-hocheberg-method"><i class="fa fa-check"></i><b>5.4</b> False discovery rates FDR / Benjamini-Hocheberg method</a></li>
<li class="chapter" data-level="5.5" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#volcano-plot"><i class="fa fa-check"></i><b>5.5</b> Volcano plot</a></li>
<li class="chapter" data-level="5.6" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#heatmap-normalization"><i class="fa fa-check"></i><b>5.6</b> Heatmap &amp; normalization</a></li>
<li class="chapter" data-level="5.7" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#ma-plots"><i class="fa fa-check"></i><b>5.7</b> MA plots</a></li>
<li class="chapter" data-level="5.8" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#eastern-new-mexico-university-enmu-method"><i class="fa fa-check"></i><b>5.8</b> Eastern New Mexico University ENMU method</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html"><i class="fa fa-check"></i><b>6</b> R code modules to be used for stand-alone analysis.</a>
<ul>
<li class="chapter" data-level="6.1" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#local-installation"><i class="fa fa-check"></i><b>6.1</b> Local installation</a></li>
<li class="chapter" data-level="6.2" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#install-packages"><i class="fa fa-check"></i><b>6.2</b> Install packages</a></li>
<li class="chapter" data-level="6.3" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#global-variables"><i class="fa fa-check"></i><b>6.3</b> Global variables</a></li>
<li class="chapter" data-level="6.4" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#read-data-and-pre-process"><i class="fa fa-check"></i><b>6.4</b> Read data and pre-process</a></li>
<li class="chapter" data-level="6.5" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#hierarchical-clustering"><i class="fa fa-check"></i><b>6.5</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="6.6" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#pca-1"><i class="fa fa-check"></i><b>6.6</b> PCA</a></li>
<li class="chapter" data-level="6.7" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#k-means-clustering"><i class="fa fa-check"></i><b>6.7</b> K-means clustering</a></li>
<li class="chapter" data-level="6.8" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#differential-expression-1"><i class="fa fa-check"></i><b>6.8</b> Differential expression</a></li>
<li class="chapter" data-level="6.9" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#pathway-analysis"><i class="fa fa-check"></i><b>6.9</b> Pathway analysis</a></li>
<li class="chapter" data-level="6.10" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#genome-wide-view"><i class="fa fa-check"></i><b>6.10</b> Genome-wide view</a></li>
<li class="chapter" data-level="6.11" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#biclustering"><i class="fa fa-check"></i><b>6.11</b> Biclustering</a></li>
<li class="chapter" data-level="6.12" data-path="r-code-modules-to-be-used-for-stand-alone-analysis..html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis..html#co-expression-network"><i class="fa fa-check"></i><b>6.12</b> Co-expression network</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html"><i class="fa fa-check"></i><b>7</b> WGCNA RNA-seq Blog</a>
<ul>
<li class="chapter" data-level="7.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#WGCNA"><i class="fa fa-check"></i><b>7.1</b> Introduction to the WGCNA Rpackage</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-rpackage-installation-procedure."><i class="fa fa-check"></i><b>7.1.1</b> WGCNA Rpackage Installation procedure.</a></li>
<li class="chapter" data-level="7.1.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#final-installation-workflow"><i class="fa fa-check"></i><b>7.1.2</b> Final installation workflow:</a></li>
<li class="chapter" data-level="7.1.3" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-tutorial."><i class="fa fa-check"></i><b>7.1.3</b> WGCNA tutorial.</a></li>
<li class="chapter" data-level="7.1.4" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#gene-expression-analysis"><i class="fa fa-check"></i><b>7.1.4</b> 3.4 Gene expression analysis</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#rna-seq-analisys-with-kakus-data-sample."><i class="fa fa-check"></i><b>7.2</b> RNA-seq Analisys with Kaku’s Data sample.</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#data-analysis."><i class="fa fa-check"></i><b>7.2.1</b> Data Analysis.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="youtube-references..html"><a href="youtube-references..html"><i class="fa fa-check"></i><b>8</b> Youtube references.</a>
<ul>
<li class="chapter" data-level="8.1" data-path="youtube-references..html"><a href="youtube-references..html#anaconda-set-up-robust-data-science-environment-with-miniconda-and-conda-forge."><i class="fa fa-check"></i><b>8.1</b> Anaconda set up, Robust Data Science Environment with Miniconda and Conda-Forge.</a></li>
<li class="chapter" data-level="8.2" data-path="youtube-references..html"><a href="youtube-references..html#quality-control-preprocessing-of-raw-reads"><i class="fa fa-check"></i><b>8.2</b> Quality control &amp; preprocessing of raw reads</a></li>
<li class="chapter" data-level="8.3" data-path="youtube-references..html"><a href="youtube-references..html#trimmomatic."><i class="fa fa-check"></i><b>8.3</b> Trimmomatic.</a></li>
<li class="chapter" data-level="8.4" data-path="youtube-references..html"><a href="youtube-references..html#building-genome-index-and-aligning-with-star"><i class="fa fa-check"></i><b>8.4</b> Building Genome Index and Aligning with STAR</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="external-ref.html"><a href="external-ref.html"><i class="fa fa-check"></i><b>9</b> External Ref</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics_RNA-seq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1 unnumbered">
<h1>References</h1>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="external-ref.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Bioinformatics_RNA-seq.pdf", "Bioinformatics_RNA-seq.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
