<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 RNA Seq for STAT736 | Bioinformatics_RNA-seq</title>
  <meta name="description" content="RNA-seq notes and trials" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 RNA Seq for STAT736 | Bioinformatics_RNA-seq" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="RNA-seq notes and trials" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 RNA Seq for STAT736 | Bioinformatics_RNA-seq" />
  
  <meta name="twitter:description" content="RNA-seq notes and trials" />
  

<meta name="author" content="J. Marcelo Rosales R." />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="final-words.html"/>
<link rel="next" href="references.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to RNA sequencing (RNA_seq)</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#downloading-and-setting-up-conda-environments."><i class="fa fa-check"></i><b>1.1</b> Downloading and setting up conda environments.</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#install-miniconda-or-anaconda."><i class="fa fa-check"></i><b>1.2</b> Install Miniconda, or Anaconda.</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#conda-terminal-commands."><i class="fa fa-check"></i><b>1.2.1</b> Conda terminal commands.</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#anacondaminiconda-install-summary-of-steps."><i class="fa fa-check"></i><b>1.2.2</b> Anaconda/Miniconda install Summary of Steps.</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#conda-activate-error"><i class="fa fa-check"></i><b>1.2.3</b> Conda activate error</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#alignment-procedures-and-files."><i class="fa fa-check"></i><b>1.3</b> Alignment Procedures and files.</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#using-tophat-early-version"><i class="fa fa-check"></i><b>1.3.1</b> Using Tophat (early version)</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#using-tophat"><i class="fa fa-check"></i><b>1.3.2</b> Using Tophat</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#using-star"><i class="fa fa-check"></i><b>1.3.3</b> Using STAR</a></li>
<li class="chapter" data-level="1.3.4" data-path="index.html"><a href="index.html#Mus-musculus-NCBI-build37-2"><i class="fa fa-check"></i><b>1.3.4</b> Custom/personal way to Download the Mus_musculus_NCBI_build37.2.tar.gz file.</a></li>
<li class="chapter" data-level="1.3.5" data-path="index.html"><a href="index.html#get-the-sample-files."><i class="fa fa-check"></i><b>1.3.5</b> Get the Sample files.</a></li>
<li class="chapter" data-level="1.3.6" data-path="index.html"><a href="index.html#starting-to-analize-fastq.gz-files"><i class="fa fa-check"></i><b>1.3.6</b> Starting to analize fastq.gz files</a></li>
<li class="chapter" data-level="1.3.7" data-path="index.html"><a href="index.html#trimming-with-trimmomatic"><i class="fa fa-check"></i><b>1.3.7</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="1.3.8" data-path="index.html"><a href="index.html#using-star-to-map-genes"><i class="fa fa-check"></i><b>1.3.8</b> Using STAR to map genes</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#using-star-1"><i class="fa fa-check"></i><b>1.4</b> Using Star</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html"><i class="fa fa-check"></i><b>2</b> WGCNA RNA-seq Blog</a>
<ul>
<li class="chapter" data-level="2.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#WGCNA"><i class="fa fa-check"></i><b>2.1</b> Introduction to the WGCNA Rpackage</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-rpackage-installation-procedure."><i class="fa fa-check"></i><b>2.1.1</b> WGCNA Rpackage Installation procedure.</a></li>
<li class="chapter" data-level="2.1.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#final-installation-workflow"><i class="fa fa-check"></i><b>2.1.2</b> Final installation workflow:</a></li>
<li class="chapter" data-level="2.1.3" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-tutorial."><i class="fa fa-check"></i><b>2.1.3</b> WGCNA tutorial.</a></li>
<li class="chapter" data-level="2.1.4" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#gene-expression-analysis"><i class="fa fa-check"></i><b>2.1.4</b> 3.4 Gene expression analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#rna-seq-analisys-with-kakus-data-sample."><i class="fa fa-check"></i><b>2.2</b> RNA-seq Analisys with Kaku’s Data sample.</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#data-analysis."><i class="fa fa-check"></i><b>2.2.1</b> Data Analysis.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="sra-download-qc-and-trimming.html"><a href="sra-download-qc-and-trimming.html"><i class="fa fa-check"></i><b>3</b> SRA Download, QC, and Trimming</a>
<ul>
<li class="chapter" data-level="3.1" data-path="sra-download-qc-and-trimming.html"><a href="sra-download-qc-and-trimming.html#trimmomatic"><i class="fa fa-check"></i><b>3.1</b> Trimmomatic</a></li>
<li class="chapter" data-level="3.2" data-path="sra-download-qc-and-trimming.html"><a href="sra-download-qc-and-trimming.html#indexing-with-star"><i class="fa fa-check"></i><b>3.2</b> Indexing with STAR</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>4</b> Methods</a></li>
<li class="chapter" data-level="5" data-path="youtube-references-.html"><a href="youtube-references-.html"><i class="fa fa-check"></i><b>5</b> Youtube references.</a>
<ul>
<li class="chapter" data-level="5.1" data-path="youtube-references-.html"><a href="youtube-references-.html#anaconda-set-up-robust-data-science-environment-with-miniconda-and-conda-forge."><i class="fa fa-check"></i><b>5.1</b> Anaconda set up, Robust Data Science Environment with Miniconda and Conda-Forge.</a></li>
<li class="chapter" data-level="5.2" data-path="youtube-references-.html"><a href="youtube-references-.html#quality-control-preprocessing-of-raw-reads"><i class="fa fa-check"></i><b>5.2</b> Quality control &amp; preprocessing of raw reads</a></li>
<li class="chapter" data-level="5.3" data-path="youtube-references-.html"><a href="youtube-references-.html#trimmomatic."><i class="fa fa-check"></i><b>5.3</b> Trimmomatic.</a></li>
<li class="chapter" data-level="5.4" data-path="youtube-references-.html"><a href="youtube-references-.html#building-genome-index-and-aligning-with-star"><i class="fa fa-check"></i><b>5.4</b> Building Genome Index and Aligning with STAR</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>6</b> Final Words</a></li>
<li class="chapter" data-level="7" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html"><i class="fa fa-check"></i><b>7</b> RNA Seq for STAT736</a>
<ul>
<li class="chapter" data-level="7.1" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#background"><i class="fa fa-check"></i><b>7.1</b> <strong>Background</strong></a></li>
<li class="chapter" data-level="7.2" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#cleaning-data"><i class="fa fa-check"></i><b>7.2</b> <strong>Cleaning Data</strong></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#programs-used"><i class="fa fa-check"></i><b>7.2.1</b> Programs used?</a></li>
<li class="chapter" data-level="7.2.2" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#acquiring-sequences"><i class="fa fa-check"></i><b>7.2.2</b> Acquiring sequences</a></li>
<li class="chapter" data-level="7.2.3" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#fastqc"><i class="fa fa-check"></i><b>7.2.3</b> FastQC</a></li>
<li class="chapter" data-level="7.2.4" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#trimming-with-trimmomatic-1"><i class="fa fa-check"></i><b>7.2.4</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="7.2.5" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#alignment"><i class="fa fa-check"></i><b>7.2.5</b> Alignment</a></li>
<li class="chapter" data-level="7.2.6" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#assembling-transcripts-with-cufflinks"><i class="fa fa-check"></i><b>7.2.6</b> Assembling transcripts with Cufflinks</a></li>
<li class="chapter" data-level="7.2.7" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#checking-for-contamination"><i class="fa fa-check"></i><b>7.2.7</b> Checking for Contamination</a></li>
<li class="chapter" data-level="7.2.8" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#counting-transcripts"><i class="fa fa-check"></i><b>7.2.8</b> Counting Transcripts</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#differentially-expressed-sequence-identification"><i class="fa fa-check"></i><b>7.3</b> <strong>Differentially Expressed Sequence Identification</strong></a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#analyzing-reads-counts"><i class="fa fa-check"></i><b>7.3.1</b> Analyzing Reads Counts</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e.g.-p53-control-and-ir-and-p53---control-and-ir"><i class="fa fa-check"></i><b>7.4</b> Interactions cause a difference between the lfc betwen pooled data, e.g. p53+/+ (control and IR) and p53-/- (control and IR)</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>7.4.1</b> Exploratory Data Analysis</a></li>
<li class="chapter" data-level="7.4.2" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#identification-of-differentially-expressed-genes"><i class="fa fa-check"></i><b>7.4.2</b> Identification of Differentially Expressed Genes</a></li>
<li class="chapter" data-level="7.4.3" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#gene-annotations"><i class="fa fa-check"></i><b>7.4.3</b> Gene Annotations</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#go-enrichment-analysis-using-gostats"><i class="fa fa-check"></i><b>7.5</b> 7. GO Enrichment analysis using GOstats</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#go-enrichment-analysis-of-downregulated-genes"><i class="fa fa-check"></i><b>7.5.1</b> GO Enrichment analysis of downregulated genes</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#pathway-analysis-using-expression-data"><i class="fa fa-check"></i><b>7.6</b> 8. Pathway analysis using expression data</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#prepare-data"><i class="fa fa-check"></i><b>7.6.1</b> Prepare data</a></li>
<li class="chapter" data-level="7.6.2" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#kegg-pathways"><i class="fa fa-check"></i><b>7.6.2</b> KEGG pathways</a></li>
<li class="chapter" data-level="7.6.3" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#pathway-and-regulation-of-genes-for-oxidative-phosphorylation."><i class="fa fa-check"></i><b>7.6.3</b> Pathway and regulation of genes for Oxidative phosphorylation.</a></li>
<li class="chapter" data-level="7.6.4" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#pathway-and-regulation-of-genes-for-glycosylphosphatidylinositolgpi-anchor-biosynthesis."><i class="fa fa-check"></i><b>7.6.4</b> Pathway and regulation of genes for Glycosylphosphatidylinositol(GPI)-anchor biosynthesis.</a></li>
<li class="chapter" data-level="7.6.5" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#pathway-and-regulation-of-genes-for-rna-polymerase."><i class="fa fa-check"></i><b>7.6.5</b> Pathway and regulation of genes for RNA polymerase.</a></li>
<li class="chapter" data-level="7.6.6" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#pathway-and-regulation-of-genes-for-nucleotide-excision-repair."><i class="fa fa-check"></i><b>7.6.6</b> Pathway and regulation of genes for Nucleotide excision repair.</a></li>
<li class="chapter" data-level="7.6.7" data-path="rna-seq-for-stat736.html"><a href="rna-seq-for-stat736.html#pathway-and-regulation-of-genes-for-non-homologous-end-joining."><i class="fa fa-check"></i><b>7.6.7</b> Pathway and regulation of genes for Non-homologous end-joining.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics_RNA-seq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="rna-seq-for-stat736" class="section level1" number="7">
<h1><span class="header-section-number">Chapter 7</span> RNA Seq for STAT736</h1>
<p>Alex Soupir Tutorial</p>
<p><a href="https://github.com/ACSoupir/Bioinformatics_RNASeq/blob/master/Mouse_RNA_Seq_p53_genotoxic.Rmd">ACSoupir/Bioinformatics_RNASeq/Mouse_RNA_Seq_p53_genotoxic.Rmd</a> (Raw md file.)</p>
<p>See here for full view:<br />
<a href="https://github.com/ACSoupir/Bioinformatics_RNASeq/blob/master/Mouse_RNA_Seq_p53_genotoxic.md">Mouse_RNA_Seq_p53_genotoxic.md.</a></p>
<div id="background" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> <strong>Background</strong></h2>
<p>For STAT736-Fall-2019, we are analyzing the RNA-Seq from the publication <a href="https://www.ncbi.nlm.nih.gov/pubmed/26372730?dopt=Abstract">Genome-wide analysis of p53 transcriptional programs in B cells upon exposure to genotoxic stress in vivo.</a> We are only using the <a href="https://trace.ncbi.nlm.nih.gov/Traces/study/?acc=SRP061386">sequences</a> <em>B cells from spleen</em> and not the <em>non-B cells from spleen</em> from the SRA Run Selector on NCBI.</p>
<p>The mice were exposed to whole-body ionizing radiation and sequences were extracted from both Bcells and non-B cells from the spleens of the mice. Two genotypes of mice were used: mice with p53 knocked out and the wild-type C57/Bl6. There were 4 different group combinations including the 2 different genotypes; each genotype was subjected to the ionizing radiation as well as control/mock.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="rna-seq-for-stat736.html#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb71-2"><a href="rna-seq-for-stat736.html#cb71-2" aria-hidden="true" tabindex="-1"></a>experimental_design <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="st">&#39;Genotype&#39;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;p53&#39;</span>, <span class="st">&#39;C57/Bl6&#39;</span>, <span class="st">&#39;p53&#39;</span>, <span class="st">&#39;C57/Bl6&#39;</span>), <span class="st">&#39;Treatment&#39;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;Mock&#39;</span>, <span class="st">&#39;Mock&#39;</span>, <span class="st">&#39;IR&#39;</span>, <span class="st">&#39;IR&#39;</span>))</span>
<span id="cb71-3"><a href="rna-seq-for-stat736.html#cb71-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(experimental_design) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;Group 1&#39;</span>, <span class="st">&#39;Group 2&#39;</span>, <span class="st">&#39;Group 3&#39;</span>, <span class="st">&#39;Group 4&#39;</span>)</span>
<span id="cb71-4"><a href="rna-seq-for-stat736.html#cb71-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb71-5"><a href="rna-seq-for-stat736.html#cb71-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reticulate)</span>
<span id="cb71-6"><a href="rna-seq-for-stat736.html#cb71-6" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(experimental_design[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb71-7"><a href="rna-seq-for-stat736.html#cb71-7" aria-hidden="true" tabindex="-1"></a>      <span class="co">#, &quot;latex&quot;</span></span>
<span id="cb71-8"><a href="rna-seq-for-stat736.html#cb71-8" aria-hidden="true" tabindex="-1"></a>      , <span class="at">caption =</span> <span class="st">&#39;Treatment groups of the mice that were either controls or treated with ionizing radiation to determine reaction of p53.&#39;</span></span>
<span id="cb71-9"><a href="rna-seq-for-stat736.html#cb71-9" aria-hidden="true" tabindex="-1"></a>      <span class="co">#,booktabs = T</span></span>
<span id="cb71-10"><a href="rna-seq-for-stat736.html#cb71-10" aria-hidden="true" tabindex="-1"></a>      ) <span class="sc">%&gt;%</span></span>
<span id="cb71-11"><a href="rna-seq-for-stat736.html#cb71-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">latex_options =</span> <span class="fu">c</span>(<span class="st">&quot;striped&quot;</span>, <span class="st">&quot;repeat_header&quot;</span>))</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
(#tab:experimental_design)Treatment groups of the mice that were either controls or treated with ionizing radiation to determine reaction of p53.
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Genotype
</th>
<th style="text-align:left;">
Treatment
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Group 1
</td>
<td style="text-align:left;">
p53
</td>
<td style="text-align:left;">
Mock
</td>
</tr>
<tr>
<td style="text-align:left;">
Group 2
</td>
<td style="text-align:left;">
C57/Bl6
</td>
<td style="text-align:left;">
Mock
</td>
</tr>
<tr>
<td style="text-align:left;">
Group 3
</td>
<td style="text-align:left;">
p53
</td>
<td style="text-align:left;">
IR
</td>
</tr>
<tr>
<td style="text-align:left;">
Group 4
</td>
<td style="text-align:left;">
C57/Bl6
</td>
<td style="text-align:left;">
IR
</td>
</tr>
</tbody>
</table>
<p>This document will contain 2 different pipelines: The first one is going to be using the genome to map the reads too, and the secod is going to be <em>de novo</em>.</p>
<p><a href="#genome-mapping">Genome</a>
<a href="#denovo-assembly"><em>De novo</em></a></p>
<p>##Genome Mapping {#genome-mapping}</p>
<p>The pipeline used in this analysis used <strong>conda</strong> on South Dakota State University’s High Performance Computing cluster to run the programs <strong>FastQC</strong>, <strong>Trimmomatic</strong>, and <strong>Tophat</strong>.</p>
<p>This is different than previous RNA-Seq analyses where I used my workstation pc with <strong>Ubuntu 18.04</strong> to run <strong>FastQC</strong>, <strong>Trimmomatic</strong>, <strong>HiSat2</strong>, <strong>HTSeq</strong>, and <strong>DESeq2</strong> locally. Also, the previous RNA-Seq alayses were of Soybean with treatment combinations of mycorrhizae and rhizobia inoculation.</p>
</div>
<div id="cleaning-data" class="section level2" number="7.2">
<h2><span class="header-section-number">7.2</span> <strong>Cleaning Data</strong></h2>
<div id="programs-used" class="section level3" number="7.2.1">
<h3><span class="header-section-number">7.2.1</span> Programs used?</h3>
<ul>
<li>FastQC</li>
<li>Trimmomatic 0.39</li>
<li>Bowtie 2.2.5.0</li>
<li>Tophat 2.1.1</li>
<li>STAR</li>
<li>Cufflinks</li>
<li>Kraken2</li>
<li>MultiQC</li>
<li>featureCount</li>
</ul>
<div id="picking-the-right-node" class="section level4" number="7.2.1.1">
<h4><span class="header-section-number">7.2.1.1</span> Picking the right node</h4>
<p>To find a node that we can use on our own, we need to see which nodes are already allocated to jobs and which ones are idle. To do this, we can run <strong>sinfo</strong>. We want to pick one of the nodes that are marked ‘idle’ so we get the whole thing and we aren’t interrupting someone elses job. For the sake of this exercise, lets work on <strong>big-mem</strong>.</p>
<p>Once a node that is idle has been found, you can <em>ssh</em> into it by typing <strong>ssh -X big-mem00#</strong> where # is the node number.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb72-1"><a href="rna-seq-for-stat736.html#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ssh</span> <span class="at">-X</span> big-mem005</span></code></pre></div>
<p>Once on the node, the modules will have to be pulled from the shared folder again, otherwise we will be left with very basic ones. <strong>NOTE:</strong> if running programs that are in a personal folder such as miniconda (these examples), it is not necessary to add the other modules.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb73-1"><a href="rna-seq-for-stat736.html#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> use /cm/shared/modulefiles_local/</span></code></pre></div>
<p>After loading the modules you can use it just as you would any other command line.</p>
</div>
<div id="creating-slurm-scripts" class="section level4" number="7.2.1.2">
<h4><span class="header-section-number">7.2.1.2</span> Creating slurm scripts</h4>
<p>When running on a cluster, it can sometime be difficult to find open nodes with the resources needed to run the jobs that we have. Making a slurm script is really easy. Fist we make a new file with the <em>touch</em> command.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb74-1"><a href="rna-seq-for-stat736.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">touch</span> commands.slurm</span></code></pre></div>
<p>Now in our directory we have the file <strong>commands.slurm</strong> which we can edit to hold our code in. We can edit it with the <em>vi</em> command.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb75-1"><a href="rna-seq-for-stat736.html#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="ex">vi</span> commands.slurm</span></code></pre></div>
<p>We have a few things that we need to put in the file header so slurm knows what to do with our commands.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb76-1"><a href="rna-seq-for-stat736.html#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/bin/bash</span></span>
<span id="cb76-2"><a href="rna-seq-for-stat736.html#cb76-2" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --job-name=example</span></span>
<span id="cb76-3"><a href="rna-seq-for-stat736.html#cb76-3" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --nodes=1</span></span>
<span id="cb76-4"><a href="rna-seq-for-stat736.html#cb76-4" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --ntasks-per-node=10</span></span>
<span id="cb76-5"><a href="rna-seq-for-stat736.html#cb76-5" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --output=job-%j-%N.log</span></span>
<span id="cb76-6"><a href="rna-seq-for-stat736.html#cb76-6" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --partition=bigmem</span></span>
<span id="cb76-7"><a href="rna-seq-for-stat736.html#cb76-7" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --time=10:00:00</span></span></code></pre></div>
<p>When we break this down, we see <em>–job-name</em> which is what we will see when we look at whats running later, <em>–nodes</em> is the number of nodes we have, <em>–ntasks-per-node</em> is the number of cores that we are requesting to have allocated, <em>–output</em> is the output log file of the job (here it names the output file with the job number and the node that we used), <em>–partition</em> here is requesting a big-mem node but <strong>compute</strong> can also be used, and finally <em>–time</em> is how long we are requesting the allocation for.</p>
<p>If the time runs out before the job is done I believe that it just kills the job even if not finished so we need to think a little about how much time to set. If the time is set too low, the job is killed and if the time is set too long, we may face issues with getting the node allocated to us.</p>
<p>To submit a job we can use <em>sbatch commands.slurm</em> and then we have the job ID. To check the status of our submission we use <em>sbatch</em> and then it shows all of the submitted jobs and how long they have been running along with the name that we set in the script.</p>
</div>
</div>
<div id="acquiring-sequences" class="section level3" number="7.2.2">
<h3><span class="header-section-number">7.2.2</span> Acquiring sequences</h3>
<p>To download the sequences from the sequence read archive (SRA), the SRA Toolkit was used. The downloading of the files took a very long time, so this was left to run over night. The <strong>–gzip</strong> was used to keep the files a relatively small, although this can be left out to download uncompressed files, and <strong>–split-files</strong> was used to split the forward read from the reverse read for paired end read trimming through <strong>Trimmomatic</strong>.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb77-1"><a href="rna-seq-for-stat736.html#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/tools/sratoolkit.2.9.6-1-centos_linux64/bin/fastq-dump</span> <span class="at">--gzip</span> <span class="at">--split-files</span> SRR2121770</span></code></pre></div>
<p>This is an example of the single file, but the above code needed to be ran for all of the following SRA numbers:</p>
<ul>
<li>SRR2121770</li>
<li>SRR2121771</li>
<li>SRR2121774</li>
<li>SRR2121775</li>
<li>SRR2121778</li>
<li>SRR2121779</li>
<li>SRR2121780</li>
<li>SRR2121781</li>
<li>SRR2121786</li>
<li>SRR2121787</li>
<li>SRR2121788</li>
<li>SRR2121789</li>
</ul>
<p>The results from downloading with <strong>–split-files</strong> gives 2 files per SRR, as mentioned before, one forward and one reverse. The suffix of the split files is one with <strong>_1.fastq.gz</strong> and another with <strong>_2.fastq.gz</strong>.</p>
</div>
<div id="fastqc" class="section level3" number="7.2.3">
<h3><span class="header-section-number">7.2.3</span> FastQC</h3>
<p>FastQC can be run on all of the read files by using the wild card (*) as in <strong>*.fastq.gz</strong>. This prevents the need to hard code each individual read file into a FastQC command, which saves a lot of time since there are 24 read files in total for these 12 samples.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb78-1"><a href="rna-seq-for-stat736.html#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/fastqc</span> <span class="pp">*</span>.fastq.gz</span></code></pre></div>
<p>The output from running FastQC is a zipped folder and an HTML file for each of the <strong>.gz</strong> files in the folder. The HTML document looks something like this:</p>
<pre><code>![FastQC of Raw SRR2121770_1.fastq.gz Read](./TopFastQCRaw.PNG)</code></pre>
<p>This is just the top of the file, and every category under the <strong>Summary</strong> heading has a graph that shows how the read quality looks for that particular metric. These reports can give insight into whether the reads are of decent quality or if the quality is poor.</p>
<p>The raw reads we have here all passed for adapter content and sequence length distribution and everything failed per base sequence content. SRR2121770, SRR2121771, SRR2121774, SRR2121775, SRR2121788, SRR2121781-2, and SRR2121789-1 were fairly decent quality reads. SRR2121778, SRR2121779, SRR2121780, SRR2121786, SRR2121787, SRR2121781-1, and SRR2121789-2 were of fairly lower quality (failing 3 or more in both reads. All of them failed both per base sequence quality and per tile sequence quality.</p>
<div id="multiqc" class="section level4" number="7.2.3.1">
<h4><span class="header-section-number">7.2.3.1</span> MultiQC</h4>
<p>First lets install <strong>multiqc</strong> with conda. The command for this is <strong><em>conda install -c bioconda multiqc</em></strong>.</p>
<p>When that is finished, we can run MultiQC in the folder with the QC files (they should be moved into a folder alone so things don’t get cluttered later on in the analyses).</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb80-1"><a href="rna-seq-for-stat736.html#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda/bin/multiqc</span> .</span></code></pre></div>
<p>When MultiQC is finished running, there will be a new folder called <strong>multiqc_data</strong> where the summaries are stored. Now lets go back up a level where our raw data folder and fastqc folder is and make a new folder for all of our MultiQC data. We will copy the FastQC output from MultiQC to this new folder.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb81-1"><a href="rna-seq-for-stat736.html#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> MultiQC_All</span>
<span id="cb81-2"><a href="rna-seq-for-stat736.html#cb81-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> RawQC/multiqc_data/multiqc_fastqc.txt MultiQC_All/</span></code></pre></div>
</div>
</div>
<div id="trimming-with-trimmomatic-1" class="section level3" number="7.2.4">
<h3><span class="header-section-number">7.2.4</span> Trimming with Trimmomatic</h3>
<p>Conda was used again to run Trimmomatic. This isn’t as easy as using the wildcard like with FastQC because each output has to be personalized for the read files that are input into Trimmomatic. Also, we have to make sure that the adapter sequences are in the same folder that we are running so we can refer to them easily when calling the Trimmomatic program. In this case, we are using the TruSeq3-PE-2.fa adapter sequences For example:</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb82-1"><a href="rna-seq-for-stat736.html#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/trimmomatic</span> PE SRR2121770_1.fastq.gz SRR2121770_2.fastq.gz 770_fp.fq.gz 770_fu.fq.gz 770_rp.fq.gz 770_ru.fq.gz ILLUMINACLIP:TruSeq3-PE-2.fa:2:30:10:2keepBothReads LEADING:3 TRALING:3 MINLEN:36 <span class="kw">&amp;</span></span></code></pre></div>
<p>This would be repeated for each of the pairs (12 in total). We are trimming paired-end reads with the TruSeq3-PE-2 adapters. We are chopping off the first and last 3 bases and if we end up with a sequence less than 36 bases, we get rid of it. We want to make sure that there are enough bases in a read to work with. These parameters can be tweaked for possibly better end results with less being discarded. When Trimmomatic is finished running, it will out put the total number of reads, the total number from both the forward and reverse reads that are kept, the number of only forward reads kept, the number of only reverse reads kept, and the number of discarded reads.</p>
<p>The highest number of reads dropped was from trimming SRR2121786, where 20.55% dropped. Most reads were between 5% and 10% dropped. SRR2121786, SRR2121787, and SRR2121779 had sequence drops greater than 15%.</p>
<p>The trimmed reads can be analyzed again with FastQC to see how well the trimming worked to make the file better quality. After running FastQC on the trimmed files we see that the quality of those that were really bad quality were improved. There were a few different metrics throughout all of the files that bounced from a warning before the failing, or from passing before to a warning, and so forth, overall creating better quality read files.</p>
</div>
<div id="alignment" class="section level3" number="7.2.5">
<h3><span class="header-section-number">7.2.5</span> Alignment</h3>
<div id="using-tophat-1" class="section level4" number="7.2.5.1">
<h4><span class="header-section-number">7.2.5.1</span> Using Tophat</h4>
<p>Tophat can be installed using the same <strong>conda install</strong> (<strong><em>conda install -c bioconda tophat</em></strong>). When this is finished installing, then we will need to get the mouse genome from the <a href="http://ccb.jhu.edu/software/tophat/igenomes.shtml">Johns Hopkins Univeristy Center for computational BIology</a>. The version of the mouse genome that I am using here is the <a href="ftp://igenome:G3nom3s4u@ussd-ftp.illumina.com/Mus_musculus/NCBI/build37.2/Mus_musculus_NCBI_build37.2.tar.gz">NCBI build37.2</a>. Instead of downloading this from the website and having to move it to the cluser, I will just download it using wget into the folder that has the raw reads, trimmed reads, and the FastQC files.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb83-1"><a href="rna-seq-for-stat736.html#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> ftp://igenome:G3nom3s4u@ussd-ftp.illumina.com/Mus_musculus/NCBI/build37.2/Mus_musculus_NCBI_build37.2.tar.gz</span></code></pre></div>
<p>This will take a long time to download because the file is a little less than 16GB zipped.</p>
<p>We notice here that we have a zipped tar file. To make this file easier to use, lets unzip it.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb84-1"><a href="rna-seq-for-stat736.html#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tar</span> zxvf Mus_muculus_NCBI_build37.2.tar.gz</span></code></pre></div>
<p>Since Tophat is requiring <strong>*.bt21</strong> files (large index) and the files downloaded for the genome above are only small index files, we have to create a large index using <strong>bowtie2-build</strong>. For this, lets navigate to the WholeGenomeFasta folder within the extracted folder and then run <strong>bowtie2-build</strong>.</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb85-1"><a href="rna-seq-for-stat736.html#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/bowtie2-build</span> <span class="at">--large-index</span> genome.fa genome</span></code></pre></div>
<p>This process took about 26 minutes to run. Now lets copy the index files to a folder close to our reads so we can access them easier, rather than having to refer to the longer path where we build them. After they are copied to a new folder closer to our working directory, I went ahead and unzipped the trimmed read files to try and make the Tophat faster but it turned out not to work. The multicore call with <strong>-p</strong> didn’t use more cores than 1 until <strong>bowtie2-align-s</strong>, then 20 cores were used.</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb86-1"><a href="rna-seq-for-stat736.html#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/tophat</span> <span class="at">--no-converage-search</span> <span class="at">-p</span> 20 <span class="at">-G</span> Mus_musculus/NCBI/build37.2/Annotation/Archives/archive-2015-07-17-14-32-40/Genes/genes.gtf <span class="at">-0</span> 770_thout ./Index/genome 770_fp.fq.gz 770_rp.fq.gz 770_fu.fq 770_ru.fq</span></code></pre></div>
<p>This run took almost 3 hours to complete.. Running with 80 cores rather than 20 cores took just 4 minutes less, so the whole process must be limited by a single core and the core clock speed. The process does use close to 8,000% at its peak so there is a benefit to multicore, just isn’t very scalable.</p>
</div>
<div id="using-star-2" class="section level4" number="7.2.5.2">
<h4><span class="header-section-number">7.2.5.2</span> Using STAR</h4>
<p>STAR can be installed the same way as the previous programs with <strong>conda install</strong> (<strong><em>conda install -c bioconda star</em></strong>). In order to run STAR, we need to creaate indices just like with tophat, but STAR has this built in. I’m going to be using the same genome and GTF file as previously downloaded, but Dr. Ge uses a different zipped genome from the <em>gencode</em> database.</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb87-1"><a href="rna-seq-for-stat736.html#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/STAR</span> <span class="dt">\</span></span>
<span id="cb87-2"><a href="rna-seq-for-stat736.html#cb87-2" aria-hidden="true" tabindex="-1"></a>--runThreadN 80 <span class="dt">\</span></span>
<span id="cb87-3"><a href="rna-seq-for-stat736.html#cb87-3" aria-hidden="true" tabindex="-1"></a>--runMode genomeGenerate <span class="dt">\</span></span>
<span id="cb87-4"><a href="rna-seq-for-stat736.html#cb87-4" aria-hidden="true" tabindex="-1"></a>--genomeDir starIndex <span class="dt">\</span></span>
<span id="cb87-5"><a href="rna-seq-for-stat736.html#cb87-5" aria-hidden="true" tabindex="-1"></a>--genomeFastaFiles Index/genome.fa <span class="dt">\ </span>#same when we made the bowtie indices </span>
<span id="cb87-6"><a href="rna-seq-for-stat736.html#cb87-6" aria-hidden="true" tabindex="-1"></a><span class="ex">--sjdbGTFfile</span> Mus_musculus/NCBI/build37.2/Annotation/Archives/archive-2015-07-17-14-32-40/Genes/genes.gtf</span></code></pre></div>
<p>With the index files made, we can start aligning with STAR. It’s important here than we only pick the paired end reads and not use all of the reads. Tophat is able to use all 4 reads but STAR doesn’t allow that, so we need to make sure that we feed in the large files from trimming.</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb88-1"><a href="rna-seq-for-stat736.html#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/STAR</span> <span class="at">--runThreadN</span> 80 <span class="at">--genomeDir</span> starIndex <span class="at">--readFilesIn</span> 770_fp.fq 770_rp.fq <span class="at">--outFilterIntronMotifs</span> RemoveNoncanonical <span class="at">--outFileNamePrefix</span> 2121770 <span class="at">--outSAMtype</span> BAM SortedByCoordinate</span></code></pre></div>
</div>
</div>
<div id="assembling-transcripts-with-cufflinks" class="section level3" number="7.2.6">
<h3><span class="header-section-number">7.2.6</span> Assembling transcripts with Cufflinks</h3>
<p>Once STAR is done running, we can assemble the transcripts with Cufflinks. This can also be installed with <strong>conda install</strong> (<strong><em>conda install -c bioconda cufflinks</em></strong>).</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb89-1"><a href="rna-seq-for-stat736.html#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/cufflinks</span> <span class="at">-p</span> 20 <span class="at">-o</span> SRR2121771_clout <span class="at">--library-type</span> fr-firststrand 2121770Aligned.sortedByCoord.out.bam</span></code></pre></div>
</div>
<div id="checking-for-contamination" class="section level3" number="7.2.7">
<h3><span class="header-section-number">7.2.7</span> Checking for Contamination</h3>
<div id="phix-contamination" class="section level4" number="7.2.7.1">
<h4><span class="header-section-number">7.2.7.1</span> PhiX contamination</h4>
<p>Now we will look at what kind of contamination we are looking at. When samples are sequenced with Illumina, a PhiX control is run along side them. This control is for cluster generation, sequencing, alignment, and calibration for cross-talk matrix generation. We will use Bowtie to create a file to determine the PhiX contamination level.</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb90-1"><a href="rna-seq-for-stat736.html#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/bowtie2</span> <span class="at">-p</span> 20 <span class="at">-x</span> PhiX/Illumina/RTA/Sequence/Bowtie2Index/genome <span class="dt">\</span></span>
<span id="cb90-2"><a href="rna-seq-for-stat736.html#cb90-2" aria-hidden="true" tabindex="-1"></a>-1 TrimmedReads/770_fp.fq  <span class="at">-2</span> TrimmedReads/770_rp.fq <span class="at">-S</span> phix.sam <span class="op">&amp;&gt;</span> PhiXout/SRR2121770_phix.out</span></code></pre></div>
<p>When the job is done running, the output file will show how much PhiX contamination we have. For example, lookin at the <strong>SRR2121770_phix.out</strong> created above, we see that 0.11% of the reads aligned with PhiX. The lower this value the better.</p>
</div>
<div id="rrna-sequences" class="section level4" number="7.2.7.2">
<h4><span class="header-section-number">7.2.7.2</span> rRNA Sequences</h4>
<p>To retreive the rRNA sequences for mouse, we need to search the taxonomy database on NCBI for <em>Mus musculus</em>. Click on <em>Mus musculus</em> on the next page, and then the top <em>Mus musculus</em> at the head of the list. Now, select the top subtree link in the <strong>Nucleotide</strong> database. Select rRNA sequences on the left side of the page and download full list just downloading with Send &gt; Complete Record &gt; File &gt; FASTA &gt; Create File. Drag the file using WinSCP to the raw folder on the cluster and rename it to rRNA.fa.</p>
<p>We are going to need to install <strong>bwa</strong> with conda in order to get the alignments to work. This can be done with <strong><em>conda install -c bioconda bwa</em></strong>. Following this, we will need to make indixes for the rRNA that we downloaded. To make this more clean, lets make a directory for the rRNA sequences that we downloaded and the indices that we make.</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb91-1"><a href="rna-seq-for-stat736.html#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> rRNA</span></code></pre></div>
<p>Then we move the <strong>rRNA.fa</strong> to the new <strong>rRNA</strong> folder with WinSCP and then we can run the bwa.</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb92-1"><a href="rna-seq-for-stat736.html#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="bu">time</span> ~/miniconda2/bin/bwa mem <span class="at">-t</span> 20 rRNA/rRNA.fa TrimmedReads/770_fp.fq TrimmedReads/770_rp.fq <span class="op">&gt;</span> rnaAlign/770_rna.sam</span></code></pre></div>
<p>When we are done creating the new <em>*.sam</em> files for all of the forward/reverse read combinations, we can use samtools to convert the <em>*.sam</em> file to <em>*.bam</em> files which are essentially the same file just that <em>sam</em> is easier for us to look at while <em>bam</em> is binary. Samtools can be installed with <strong><em>conda install -c bioconda samtools</em></strong>.</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb93-1"><a href="rna-seq-for-stat736.html#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/samtools</span> view <span class="at">-@</span> 10 <span class="at">-bS</span> <span class="at">-o</span> rnaAlign/770_rna.bam rnaAlign/770.sam</span></code></pre></div>
<p>Now in the rnaAlign folder we have our sam and bam file for each of the libraries. Lets create an output file with <em>flagstat</em>.</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb94-1"><a href="rna-seq-for-stat736.html#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/samtools</span> flagstat <span class="at">-@</span> 10 rnaAlign/770_rna.out</span></code></pre></div>
<p>Wihtin this file we will be able to see the summary of our alignments to the rRNA file that we downloaded from NCBI.</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb95-1"><a href="rna-seq-for-stat736.html#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="co">#From 770_rna.out</span></span>
<span id="cb95-2"><a href="rna-seq-for-stat736.html#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="ex">205559289</span> + 0 in total <span class="er">(</span><span class="ex">QC-passed</span> reads + QC-failed reads<span class="kw">)</span></span>
<span id="cb95-3"><a href="rna-seq-for-stat736.html#cb95-3" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> + 0 secondary</span>
<span id="cb95-4"><a href="rna-seq-for-stat736.html#cb95-4" aria-hidden="true" tabindex="-1"></a><span class="ex">85</span> + 0 supplementary</span>
<span id="cb95-5"><a href="rna-seq-for-stat736.html#cb95-5" aria-hidden="true" tabindex="-1"></a><span class="ex">0</span> + 0 duplicates</span>
<span id="cb95-6"><a href="rna-seq-for-stat736.html#cb95-6" aria-hidden="true" tabindex="-1"></a><span class="ex">4265179</span> + 0 mapped <span class="er">(</span><span class="ex">2.07%</span> : N/A<span class="kw">)</span></span>
<span id="cb95-7"><a href="rna-seq-for-stat736.html#cb95-7" aria-hidden="true" tabindex="-1"></a><span class="ex">205559204</span> + 0 paired in sequencing</span>
<span id="cb95-8"><a href="rna-seq-for-stat736.html#cb95-8" aria-hidden="true" tabindex="-1"></a><span class="ex">102779602</span> + 0 read1</span>
<span id="cb95-9"><a href="rna-seq-for-stat736.html#cb95-9" aria-hidden="true" tabindex="-1"></a><span class="ex">102779602</span> + 0 read2</span>
<span id="cb95-10"><a href="rna-seq-for-stat736.html#cb95-10" aria-hidden="true" tabindex="-1"></a><span class="ex">4151684</span> + 0 properly paired <span class="er">(</span><span class="ex">2.02%</span> : N/A<span class="kw">)</span></span>
<span id="cb95-11"><a href="rna-seq-for-stat736.html#cb95-11" aria-hidden="true" tabindex="-1"></a><span class="ex">4187608</span> + 0 with itself and mate mapped</span>
<span id="cb95-12"><a href="rna-seq-for-stat736.html#cb95-12" aria-hidden="true" tabindex="-1"></a><span class="ex">77486</span> + 0 singletons <span class="er">(</span><span class="ex">0.04%</span> : N/A<span class="kw">)</span></span>
<span id="cb95-13"><a href="rna-seq-for-stat736.html#cb95-13" aria-hidden="true" tabindex="-1"></a><span class="ex">4222</span> + 0 with mate mapped to a different chr</span>
<span id="cb95-14"><a href="rna-seq-for-stat736.html#cb95-14" aria-hidden="true" tabindex="-1"></a><span class="ex">1026</span> + 0 with mate mapped to a different chr <span class="er">(</span><span class="ex">mapQ</span><span class="op">&gt;</span>=5<span class="kw">)</span></span></code></pre></div>
</div>
<div id="bacterial-contamination" class="section level4" number="7.2.7.3">
<h4><span class="header-section-number">7.2.7.3</span> Bacterial contamination</h4>
<p>In order to find out the contamination, we need to install Kraken2 with <strong><em>conda install -c bioconda kraken2</em></strong> and download a pre-built database containing bacteria, archaea, and viral sequences. The database we are going to download only contains about 5% of k-mers from the original database (but directions are sort of lacking to build an entirely new database). More information can be found at <a href="https://ccb.jhu.edu/software/kraken/" class="uri">https://ccb.jhu.edu/software/kraken/</a> for the pre-built databases.</p>
<p>Using the code in the next chunk will download the 8GB database and then extract the files so we can use them with the <strong>Kraken2</strong> program. Lets do this in the main project folder.</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb96-1"><a href="rna-seq-for-stat736.html#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> ftp://ftp.ccb.jhu.edu/pub/data/kraken2_dbs/minikraken2_v2_8GB_201904_UPDATE.tgz</span>
<span id="cb96-2"><a href="rna-seq-for-stat736.html#cb96-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tar</span> xzf minikraken2_v2_8GB_201904_UPDATE.tgz</span></code></pre></div>
<p>Now lets make a directory for the output.</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb97-1"><a href="rna-seq-for-stat736.html#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> krakenOut</span></code></pre></div>
<p>We can call <strong>Kraken</strong> with the extracted database folder and point it to the location of out paired end reads from trimming and to the output folder that we just created for the outputs.</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb98-1"><a href="rna-seq-for-stat736.html#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/kraken2</span> <span class="at">--db</span> minikraken2_v2_8GB_201904_UPDATE/ <span class="at">--output</span> krakenOut/770.out <span class="at">--threads</span> 10 <span class="at">--paired</span> TrimmedReads/770_fp.fq TrimmedReads/770_rp.fq</span></code></pre></div>
<p>When Kraken is done running, it will print out the number (and percentage) of reads that were classified. In this case, we have used 102779602 sequences, of which 19142843 sequences were classified (18.63%) and 83636759 sequences were unclassified (81.37%). My interpretation of this is that 18.63% of the reads are possibly from microbial cell contamination.</p>
</div>
</div>
<div id="counting-transcripts" class="section level3" number="7.2.8">
<h3><span class="header-section-number">7.2.8</span> Counting Transcripts</h3>
<p>Since we have the <strong>bam</strong> files from the alignments of the different samples, we can count the features for each and get the transcipt counts using featureCounts form <strong><em>conda install -c bioconda/label/cf201901 subread</em></strong>. The genome and annotations that we previously downloaded were from genome <strong>mm9</strong> so we have to specify to <em>featureCounts</em> what we want to actually count. FeatureCounts defaults to using <strong>gene_id</strong> which our output bam files don’t have described correctly for <em>featureCounts</em> to read them. This is a single line of code because we can use a wildcard to run through all of the <strong>bam</strong> files.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb99-1"><a href="rna-seq-for-stat736.html#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Move to the Star Alignment output folder for a working directory</span></span>
<span id="cb99-2"><a href="rna-seq-for-stat736.html#cb99-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> StarOut</span>
<span id="cb99-3"><a href="rna-seq-for-stat736.html#cb99-3" aria-hidden="true" tabindex="-1"></a><span class="ex">~/miniconda2/bin/featureCounts</span> <span class="at">-a</span> /gpfs/scratch/alex.soupir/Mus/raw/Mus_musculus/NCBI/build37.2/Annotation/Archives/archive-2015-07-17-14-32-40/Genes/genes.gtf <span class="at">-g</span> <span class="st">&#39;transcript_id&#39;</span> <span class="at">-o</span> readCounds.txt <span class="pp">*</span>bam</span></code></pre></div>
<p>With the files that we are working with, this will take between 3.5 minutes to 5 minutes per <strong>bam</strong> file. The output will be a file that can be imported into excel and saved as csv which we then can work with in R.</p>
<div id="final-qc-of-cleaning-the-data" class="section level4" number="7.2.8.1">
<h4><span class="header-section-number">7.2.8.1</span> Final QC of cleaning the data</h4>
<p>Lets look at the data that we have collected from all of the MultiQC runs that we had with initial FastQC, Trimmomatic, STAR alignment, PhiX contamination, rRNA contamination, and the final feature counts.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="rna-seq-for-stat736.html#cb100-1" aria-hidden="true" tabindex="-1"></a>qc <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="st">&#39;Whole Data QC.csv&#39;</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">na.strings=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb100-2"><a href="rna-seq-for-stat736.html#cb100-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(qc) <span class="sc">%&gt;%</span></span>
<span id="cb100-3"><a href="rna-seq-for-stat736.html#cb100-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb100-4"><a href="rna-seq-for-stat736.html#cb100-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;800px&quot;</span>)</span></code></pre></div>
</div>
</div>
</div>
<div id="differentially-expressed-sequence-identification" class="section level2" number="7.3">
<h2><span class="header-section-number">7.3</span> <strong>Differentially Expressed Sequence Identification</strong></h2>
<p>Programs Used</p>
<ul>
<li>R</li>
<li>RStudio</li>
</ul>
<p>Packages</p>
<ul>
<li>readr (<strong><em>install.packages(‘readr’)</em></strong>)</li>
<li>limma (<strong><em>BiocManager::install(‘limma’)</em></strong>)</li>
<li>DESeq2 (<strong><em>BiocManager::install(‘DESeq2’)</em></strong>)</li>
<li>dplyr (<strong><em>install.packages(“dplyr”)</em></strong>)</li>
<li>ggplot2 (<strong><em>install.packages(“ggplot2”)</em></strong>)</li>
<li>gplots (<strong><em>install.packages(“gplots”)</em></strong>)</li>
<li>Annotations (<strong><em>BiocManager::install(‘AnnotationDbi’)</em></strong>)</li>
<li>org.Hs.eg.db (<strong><em>BiocManager::install(‘org.Hs.eg.db’)</em></strong>)
<ul>
<li>This is for Human</li>
</ul></li>
<li>org.Mm.eg.db (<strong><em>BiocManager::install(‘org.Mm.eg.db’)</em></strong>)
<ul>
<li>This is for Mouse</li>
</ul></li>
<li>ggrepel (<strong><em>install.packages(“ggrepel”)</em></strong>)</li>
<li>ReportingTools (<strong><em>BiocManager::install(‘ReportingTools’)</em></strong>)</li>
<li>GO.db (<strong><em>BiocManager::install(‘GO.db’)</em></strong>)</li>
<li>GOstats (<strong><em>BiocManager::install(‘GOstats’)</em></strong>)</li>
<li>pathview (<strong><em>BiocManager::install(‘pathview’)</em></strong>)</li>
<li>gage (<strong><em>BiocManager::install(‘gage’)</em></strong>)</li>
<li>gageData (<strong><em>BiocManager::install(‘gageData’)</em></strong>)</li>
<li>select (<strong><em>BiocManager::install(‘Select’)</em></strong>)</li>
</ul>
<p>With these, you most certainly will have to step through each and install extra things when you start calling the packages. Take it step by step to ensure that each dependency is installed.</p>
<div id="analyzing-reads-counts" class="section level3" number="7.3.1">
<h3><span class="header-section-number">7.3.1</span> Analyzing Reads Counts</h3>
<p>When the count file is completed, we can import it into R and start working with it to determine differentially expressed genes. First we will import it into R</p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="rna-seq-for-stat736.html#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb101-2"><a href="rna-seq-for-stat736.html#cb101-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb101-3"><a href="rna-seq-for-stat736.html#cb101-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb101-4"><a href="rna-seq-for-stat736.html#cb101-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb101-5"><a href="rna-seq-for-stat736.html#cb101-5" aria-hidden="true" tabindex="-1"></a>countData <span class="ot">=</span> <span class="fu">read_csv</span>(<span class="st">&quot;readCounts.csv&quot;</span>, <span class="at">skip =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>This gives us our dataframe from out featureCounts program, but if we look at the data we see that featureCounts added some extra information that characterizes each gene_id.</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="rna-seq-for-stat736.html#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(countData)) <span class="sc">%&gt;%</span></span>
<span id="cb102-2"><a href="rna-seq-for-stat736.html#cb102-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb102-3"><a href="rna-seq-for-stat736.html#cb102-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;320px&quot;</span>)</span></code></pre></div>
<p>We also need to set out row names to the gene_id. We will do some data frame manipulation and then look at the data again.</p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="rna-seq-for-stat736.html#cb103-1" aria-hidden="true" tabindex="-1"></a>countData <span class="ot">=</span> <span class="fu">as.data.frame</span>(countData)</span>
<span id="cb103-2"><a href="rna-seq-for-stat736.html#cb103-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(countData) <span class="ot">=</span> countData<span class="sc">$</span>Geneid</span>
<span id="cb103-3"><a href="rna-seq-for-stat736.html#cb103-3" aria-hidden="true" tabindex="-1"></a>countData <span class="ot">=</span> countData[,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>)]</span>
<span id="cb103-4"><a href="rna-seq-for-stat736.html#cb103-4" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(countData)) <span class="sc">%&gt;%</span></span>
<span id="cb103-5"><a href="rna-seq-for-stat736.html#cb103-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb103-6"><a href="rna-seq-for-stat736.html#cb103-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span></code></pre></div>
<div id="quick-data-exploration" class="section level4" number="7.3.1.1">
<h4><span class="header-section-number">7.3.1.1</span> Quick Data Exploration</h4>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="rna-seq-for-stat736.html#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(countData)</span></code></pre></div>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="rna-seq-for-stat736.html#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">summary</span>(countData)) <span class="sc">%&gt;%</span></span>
<span id="cb105-2"><a href="rna-seq-for-stat736.html#cb105-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb105-3"><a href="rna-seq-for-stat736.html#cb105-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span></code></pre></div>
<p>Let’s also go ahead and change the names to describe out data a little better.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="rna-seq-for-stat736.html#cb106-1" aria-hidden="true" tabindex="-1"></a>columns <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;Trp53m_mock_1&#39;</span>, <span class="st">&#39;Trp53m_mock_2&#39;</span>, <span class="st">&#39;Trp53m_4h7Gy_1&#39;</span>, <span class="st">&#39;Trp53m_4h7Gy_2&#39;</span>, <span class="st">&#39;Trp53p_mock_1&#39;</span>, <span class="st">&#39;Trp53p_mock_2&#39;</span>, <span class="st">&#39;Trp53p_mock_3&#39;</span>, <span class="st">&#39;Trp53p_mock_4&#39;</span>, <span class="st">&#39;Trp53p_4h7Gy_1&#39;</span>, <span class="st">&#39;Trp53p_4h7Gy_2&#39;</span>, <span class="st">&#39;Trp53p_4h7Gy_3&#39;</span>, <span class="st">&#39;Trp53p_4h7Gy_4&#39;</span>)</span>
<span id="cb106-2"><a href="rna-seq-for-stat736.html#cb106-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(countData) <span class="ot">=</span> columns</span></code></pre></div>
<p>Here we have to make sure that we convert the <em>+/+</em> and <em>-/-</em> to characters. These characters</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="rna-seq-for-stat736.html#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">8</span>,<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">1</span>)<span class="sc">+</span><span class="fl">0.1</span>)</span>
<span id="cb107-2"><a href="rna-seq-for-stat736.html#cb107-2" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>( <span class="fu">colSums</span>(countData)<span class="sc">/</span><span class="fl">1e6</span>, <span class="at">col=</span><span class="st">&quot;green&quot;</span>,<span class="at">las=</span><span class="dv">3</span>,<span class="at">main=</span><span class="st">&quot;Total read counts (millions)&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;Total read counts in millions&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="rna-seq-for-stat736.html#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(countData[,<span class="dv">1</span>], <span class="at">br=</span><span class="dv">200</span>, <span class="at">xlab=</span><span class="st">&quot;Number of Reads Counts per Feature&quot;</span>, <span class="at">main=</span><span class="st">&quot;Histogram of Read Counts for Trp53-/- Mock&quot;</span>)</span></code></pre></div>
<p>We can see that our count data is highly skewed to the right. This is a great case for using <strong>log</strong> transformation!</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="rna-seq-for-stat736.html#cb109-1" aria-hidden="true" tabindex="-1"></a>logCountData <span class="ot">=</span> <span class="fu">log2</span>(<span class="dv">1</span><span class="sc">+</span>countData)</span>
<span id="cb109-2"><a href="rna-seq-for-stat736.html#cb109-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="at">mar=</span><span class="fu">c</span>(<span class="dv">8</span>,<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">1</span>))  <span class="co"># two columns</span></span>
<span id="cb109-3"><a href="rna-seq-for-stat736.html#cb109-3" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(logCountData[,<span class="dv">1</span>], <span class="at">main=</span><span class="st">&quot;Histogram of Log Read Counts&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;Log transformed counts&quot;</span>)</span>
<span id="cb109-4"><a href="rna-seq-for-stat736.html#cb109-4" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(logCountData,<span class="at">las=</span><span class="dv">3</span>, <span class="at">main=</span><span class="st">&quot;Boxplot of Log Read Counts&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="rna-seq-for-stat736.html#cb110-1" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> logCountData</span>
<span id="cb110-2"><a href="rna-seq-for-stat736.html#cb110-2" aria-hidden="true" tabindex="-1"></a>myColors <span class="ot">=</span> <span class="fu">rainbow</span>(<span class="fu">dim</span>(x)[<span class="dv">2</span>])</span>
<span id="cb110-3"><a href="rna-seq-for-stat736.html#cb110-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">density</span>(x[,<span class="dv">1</span>]),<span class="at">col =</span> myColors[<span class="dv">1</span>], <span class="at">lwd=</span><span class="dv">2</span>,</span>
<span id="cb110-4"><a href="rna-seq-for-stat736.html#cb110-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">&quot;Expresson values&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;Density&quot;</span>, <span class="at">main=</span> <span class="st">&quot;Distribution of transformed data&quot;</span>,</span>
<span id="cb110-5"><a href="rna-seq-for-stat736.html#cb110-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(<span class="fu">density</span>(x[,<span class="dv">1</span>])<span class="sc">$</span>y)<span class="sc">+</span>.<span class="dv">02</span> ) )</span>
<span id="cb110-6"><a href="rna-seq-for-stat736.html#cb110-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb110-7"><a href="rna-seq-for-stat736.html#cb110-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>( i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">dim</span>(x)[<span class="dv">2</span>] )</span>
<span id="cb110-8"><a href="rna-seq-for-stat736.html#cb110-8" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">density</span>(x[,i]),<span class="at">col=</span>myColors[i], <span class="at">lwd=</span><span class="dv">2</span>)</span>
<span id="cb110-9"><a href="rna-seq-for-stat736.html#cb110-9" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>, <span class="at">cex=</span><span class="fl">1.1</span>,<span class="fu">colnames</span>(x), <span class="at">lty=</span><span class="fu">rep</span>(<span class="dv">1</span>,<span class="fu">dim</span>(x)[<span class="dv">2</span>]), <span class="at">col=</span>myColors )    </span></code></pre></div>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="rna-seq-for-stat736.html#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(logCountData[,<span class="dv">1</span>],logCountData[,<span class="dv">2</span>], <span class="at">xlab=</span><span class="st">&quot;Trp53-/- mock replication 1&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;Trp53-/- mock replication 2&quot;</span>)</span></code></pre></div>
</div>
<div id="filtering-normalization-and-trasformation-using-deseq2" class="section level4" number="7.3.1.2">
<h4><span class="header-section-number">7.3.1.2</span> Filtering, Normalization, and Trasformation using DESeq2</h4>
<p>We have to make the experiment design into a small dataframe so we can tell DESeq how we want to analyze the data. Here will will make a small table that has the rep names that we changed the column names to previously, and then a column for which columns are Trp53+/+ or Trp53-/-, and which columns were control mice and which columns were treated with ionizing radiation.</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="rna-seq-for-stat736.html#cb112-1" aria-hidden="true" tabindex="-1"></a>detectGroups <span class="ot">&lt;-</span> <span class="cf">function</span> (x){  <span class="co"># x are col names</span></span>
<span id="cb112-2"><a href="rna-seq-for-stat736.html#cb112-2" aria-hidden="true" tabindex="-1"></a>  tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;[0-9]*$&quot;</span>,<span class="st">&quot;&quot;</span>,x) <span class="co"># Remove all numbers from end</span></span>
<span id="cb112-3"><a href="rna-seq-for-stat736.html#cb112-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#tem = gsub(&quot;_Rep|_rep|_REP&quot;,&quot;&quot;,tem)</span></span>
<span id="cb112-4"><a href="rna-seq-for-stat736.html#cb112-4" aria-hidden="true" tabindex="-1"></a>  tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_$&quot;</span>,<span class="st">&quot;&quot;</span>,tem); <span class="co"># remove &quot;_&quot; from end</span></span>
<span id="cb112-5"><a href="rna-seq-for-stat736.html#cb112-5" aria-hidden="true" tabindex="-1"></a>  tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_Rep$&quot;</span>,<span class="st">&quot;&quot;</span>,tem); <span class="co"># remove &quot;_Rep&quot; from end</span></span>
<span id="cb112-6"><a href="rna-seq-for-stat736.html#cb112-6" aria-hidden="true" tabindex="-1"></a>  tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_rep$&quot;</span>,<span class="st">&quot;&quot;</span>,tem); <span class="co"># remove &quot;_rep&quot; from end</span></span>
<span id="cb112-7"><a href="rna-seq-for-stat736.html#cb112-7" aria-hidden="true" tabindex="-1"></a>  tem <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_REP$&quot;</span>,<span class="st">&quot;&quot;</span>,tem)  <span class="co"># remove &quot;_REP&quot; from end</span></span>
<span id="cb112-8"><a href="rna-seq-for-stat736.html#cb112-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>( tem )</span>
<span id="cb112-9"><a href="rna-seq-for-stat736.html#cb112-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb112-10"><a href="rna-seq-for-stat736.html#cb112-10" aria-hidden="true" tabindex="-1"></a>groups <span class="ot">=</span> <span class="fu">as.character</span> ( <span class="fu">detectGroups</span>( <span class="fu">colnames</span>( countData ) ) )</span>
<span id="cb112-11"><a href="rna-seq-for-stat736.html#cb112-11" aria-hidden="true" tabindex="-1"></a>groups</span>
<span id="cb112-12"><a href="rna-seq-for-stat736.html#cb112-12" aria-hidden="true" tabindex="-1"></a>p53 <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;m&quot;</span>, <span class="st">&quot;m&quot;</span>, <span class="st">&quot;m&quot;</span>, <span class="st">&quot;m&quot;</span>,</span>
<span id="cb112-13"><a href="rna-seq-for-stat736.html#cb112-13" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;p&quot;</span>, <span class="st">&quot;p&quot;</span>, <span class="st">&quot;p&quot;</span>, <span class="st">&quot;p&quot;</span>, <span class="st">&quot;p&quot;</span>, <span class="st">&quot;p&quot;</span>, <span class="st">&quot;p&quot;</span>, <span class="st">&quot;p&quot;</span>)</span>
<span id="cb112-14"><a href="rna-seq-for-stat736.html#cb112-14" aria-hidden="true" tabindex="-1"></a>treatment <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;control&quot;</span>, <span class="st">&quot;control&quot;</span>, <span class="st">&quot;IR&quot;</span>, <span class="st">&quot;IR&quot;</span>, <span class="st">&quot;control&quot;</span>, <span class="st">&quot;control&quot;</span>, <span class="st">&quot;control&quot;</span>, <span class="st">&quot;control&quot;</span>,</span>
<span id="cb112-15"><a href="rna-seq-for-stat736.html#cb112-15" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;IR&quot;</span>, <span class="st">&quot;IR&quot;</span>, <span class="st">&quot;IR&quot;</span>, <span class="st">&quot;IR&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="rna-seq-for-stat736.html#cb113-1" aria-hidden="true" tabindex="-1"></a>colData <span class="ot">=</span> <span class="fu">cbind</span>(<span class="fu">colnames</span>(countData), p53 )</span>
<span id="cb113-2"><a href="rna-seq-for-stat736.html#cb113-2" aria-hidden="true" tabindex="-1"></a>colData</span></code></pre></div>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="rna-seq-for-stat736.html#cb114-1" aria-hidden="true" tabindex="-1"></a>colData <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="fu">colnames</span>(countData), p53, treatment))</span>
<span id="cb114-2"><a href="rna-seq-for-stat736.html#cb114-2" aria-hidden="true" tabindex="-1"></a>colData</span>
<span id="cb114-3"><a href="rna-seq-for-stat736.html#cb114-3" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(colData)</span></code></pre></div>
<p>Creating a DESeq Dataset</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="rna-seq-for-stat736.html#cb115-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">=</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData=</span>countData,</span>
<span id="cb115-2"><a href="rna-seq-for-stat736.html#cb115-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">colData=</span>colData,</span>
<span id="cb115-3"><a href="rna-seq-for-stat736.html#cb115-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">design=</span> <span class="sc">~</span> p53<span class="sc">+</span>treatment<span class="sc">+</span>p53<span class="sc">*</span>treatment)   <span class="co"># note that the study design is changed.</span></span>
<span id="cb115-4"><a href="rna-seq-for-stat736.html#cb115-4" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">=</span> <span class="fu">DESeq</span>(dds)  <span class="co"># main function</span></span>
<span id="cb115-5"><a href="rna-seq-for-stat736.html#cb115-5" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(dds)</span></code></pre></div>
<p>Filtering: we will only keep rows that have a sum count between all samples greater than 5. This will remove most of the genes that mostly have “0” counts.</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="rna-seq-for-stat736.html#cb116-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[ <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds)) <span class="sc">&gt;</span> <span class="dv">5</span>, ]</span>
<span id="cb116-2"><a href="rna-seq-for-stat736.html#cb116-2" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(dds)</span></code></pre></div>
<p>Regularized log transformation - used for clustering</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="rna-seq-for-stat736.html#cb117-1" aria-hidden="true" tabindex="-1"></a>rld <span class="ot">&lt;-</span> <span class="fu">rlog</span>(dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb117-2"><a href="rna-seq-for-stat736.html#cb117-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">assay</span>(rld), <span class="dv">6</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb117-3"><a href="rna-seq-for-stat736.html#cb117-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb117-4"><a href="rna-seq-for-stat736.html#cb117-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span></code></pre></div>
<p>Variance Stabilizing Transformation</p>
</div>
</div>
</div>
<div id="interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e.g.-p53-control-and-ir-and-p53---control-and-ir" class="section level2" number="7.4">
<h2><span class="header-section-number">7.4</span> Interactions cause a difference between the lfc betwen pooled data, e.g. p53+/+ (control and IR) and p53-/- (control and IR)</h2>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="rna-seq-for-stat736.html#cb118-1" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb118-2"><a href="rna-seq-for-stat736.html#cb118-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">assay</span>(vsd), <span class="dv">6</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb118-3"><a href="rna-seq-for-stat736.html#cb118-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb118-4"><a href="rna-seq-for-stat736.html#cb118-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span></code></pre></div>
<p>For the log2 approach, we need to first estimate size factors to account for sequencing depth, and then specify normalized=TRUE. Sequencing depth correction is done automatically for the rlog and the vst.</p>
<p>Size Factor</p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="rna-seq-for-stat736.html#cb119-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds)</span>
<span id="cb119-2"><a href="rna-seq-for-stat736.html#cb119-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">sizeFactors</span>(dds)) <span class="sc">%&gt;%</span></span>
<span id="cb119-3"><a href="rna-seq-for-stat736.html#cb119-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb119-4"><a href="rna-seq-for-stat736.html#cb119-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;300px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;520px&quot;</span>)</span></code></pre></div>
<p>We will first look at the log transformed data</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="rna-seq-for-stat736.html#cb120-1" aria-hidden="true" tabindex="-1"></a>slog <span class="ot">&lt;-</span> <span class="fu">log2</span>(<span class="fu">counts</span>(dds, <span class="at">normalized=</span><span class="cn">TRUE</span>)<span class="sc">+</span><span class="dv">1</span>)</span>
<span id="cb120-2"><a href="rna-seq-for-stat736.html#cb120-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(slog)) <span class="sc">%&gt;%</span></span>
<span id="cb120-3"><a href="rna-seq-for-stat736.html#cb120-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb120-4"><a href="rna-seq-for-stat736.html#cb120-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="rna-seq-for-stat736.html#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>))  <span class="co"># 3 columns</span></span>
<span id="cb121-2"><a href="rna-seq-for-stat736.html#cb121-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(slog[,<span class="dv">1</span>],slog[,<span class="dv">2</span>])</span>
<span id="cb121-3"><a href="rna-seq-for-stat736.html#cb121-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">assay</span>(rld)[,<span class="dv">1</span>],<span class="fu">assay</span>(rld)[,<span class="dv">2</span>])</span>
<span id="cb121-4"><a href="rna-seq-for-stat736.html#cb121-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">assay</span>(vsd)[,<span class="dv">1</span>],<span class="fu">assay</span>(vsd)[,<span class="dv">2</span>])</span></code></pre></div>
<p>As the log transformation constant increases, the information of the data is lost.</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="rna-seq-for-stat736.html#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>))  <span class="co"># 3 columns</span></span>
<span id="cb122-2"><a href="rna-seq-for-stat736.html#cb122-2" aria-hidden="true" tabindex="-1"></a>slog <span class="ot">&lt;-</span> <span class="fu">log2</span>(<span class="fu">counts</span>(dds, <span class="at">normalized=</span><span class="cn">TRUE</span>)<span class="sc">+</span><span class="dv">1</span>)</span>
<span id="cb122-3"><a href="rna-seq-for-stat736.html#cb122-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(slog[,<span class="dv">1</span>],slog[,<span class="dv">2</span>])</span>
<span id="cb122-4"><a href="rna-seq-for-stat736.html#cb122-4" aria-hidden="true" tabindex="-1"></a>slog <span class="ot">&lt;-</span> <span class="fu">log2</span>(<span class="fu">counts</span>(dds, <span class="at">normalized=</span><span class="cn">TRUE</span>)<span class="sc">+</span><span class="dv">4</span>)</span>
<span id="cb122-5"><a href="rna-seq-for-stat736.html#cb122-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(slog[,<span class="dv">1</span>],slog[,<span class="dv">2</span>], <span class="at">xlim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">20</span>))</span>
<span id="cb122-6"><a href="rna-seq-for-stat736.html#cb122-6" aria-hidden="true" tabindex="-1"></a>slog <span class="ot">&lt;-</span> <span class="fu">log2</span>(<span class="fu">counts</span>(dds, <span class="at">normalized=</span><span class="cn">TRUE</span>)<span class="sc">+</span><span class="dv">20</span>)</span>
<span id="cb122-7"><a href="rna-seq-for-stat736.html#cb122-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(slog[,<span class="dv">1</span>],slog[,<span class="dv">2</span>], <span class="at">xlim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">20</span>))</span></code></pre></div>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="rna-seq-for-stat736.html#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;dplyr&quot;</span>)</span>
<span id="cb123-2"><a href="rna-seq-for-stat736.html#cb123-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb123-3"><a href="rna-seq-for-stat736.html#cb123-3" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(</span>
<span id="cb123-4"><a href="rna-seq-for-stat736.html#cb123-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_data_frame</span>(slog[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb123-5"><a href="rna-seq-for-stat736.html#cb123-5" aria-hidden="true" tabindex="-1"></a>         <span class="fu">mutate</span>(<span class="at">transformation =</span> <span class="st">&quot;log2(x + 1)&quot;</span>),</span>
<span id="cb123-6"><a href="rna-seq-for-stat736.html#cb123-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_data_frame</span>(<span class="fu">assay</span>(rld)[, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">transformation =</span> <span class="st">&quot;rlog&quot;</span>),</span>
<span id="cb123-7"><a href="rna-seq-for-stat736.html#cb123-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_data_frame</span>(<span class="fu">assay</span>(vsd)[, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">transformation =</span> <span class="st">&quot;vst&quot;</span>))</span>
<span id="cb123-8"><a href="rna-seq-for-stat736.html#cb123-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb123-9"><a href="rna-seq-for-stat736.html#cb123-9" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;x&quot;</span>, <span class="st">&quot;y&quot;</span>)  </span>
<span id="cb123-10"><a href="rna-seq-for-stat736.html#cb123-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y)) <span class="sc">+</span> <span class="fu">geom_hex</span>(<span class="at">bins =</span> <span class="dv">80</span>) <span class="sc">+</span></span>
<span id="cb123-11"><a href="rna-seq-for-stat736.html#cb123-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>() <span class="sc">+</span> <span class="fu">facet_grid</span>( . <span class="sc">~</span> transformation)</span></code></pre></div>
<div id="exploratory-data-analysis" class="section level3" number="7.4.1">
<h3><span class="header-section-number">7.4.1</span> Exploratory Data Analysis</h3>
<p>PCA plot</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="rna-seq-for-stat736.html#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(rld, <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">&quot;p53&quot;</span>, <span class="st">&quot;treatment&quot;</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>)</span></code></pre></div>
<p>A prettier PCA plot created with GGPlot2</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="rna-seq-for-stat736.html#cb125-1" aria-hidden="true" tabindex="-1"></a>pca.object <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(<span class="fu">assay</span>(rld))) <span class="co"># PCA </span></span>
<span id="cb125-2"><a href="rna-seq-for-stat736.html#cb125-2" aria-hidden="true" tabindex="-1"></a>pcaData <span class="ot">=</span> <span class="fu">as.data.frame</span>(pca.object<span class="sc">$</span>x[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]); </span>
<span id="cb125-3"><a href="rna-seq-for-stat736.html#cb125-3" aria-hidden="true" tabindex="-1"></a>pcaData <span class="ot">=</span> <span class="fu">cbind</span>(pcaData,<span class="fu">detectGroups</span>(<span class="fu">colnames</span>(<span class="fu">assay</span>(rld)) ))</span>
<span id="cb125-4"><a href="rna-seq-for-stat736.html#cb125-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(pcaData) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;PC1&quot;</span>, <span class="st">&quot;PC2&quot;</span>, <span class="st">&quot;Type&quot;</span>)</span>
<span id="cb125-5"><a href="rna-seq-for-stat736.html#cb125-5" aria-hidden="true" tabindex="-1"></a>percentVar<span class="ot">=</span><span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span><span class="fu">summary</span>(pca.object)<span class="sc">$</span>importance[<span class="dv">2</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],<span class="dv">0</span>)</span>
<span id="cb125-6"><a href="rna-seq-for-stat736.html#cb125-6" aria-hidden="true" tabindex="-1"></a><span class="co">#plot</span></span>
<span id="cb125-7"><a href="rna-seq-for-stat736.html#cb125-7" aria-hidden="true" tabindex="-1"></a>p<span class="ot">=</span><span class="fu">ggplot</span>(pcaData, <span class="fu">aes</span>(PC1, PC2, <span class="at">color=</span>Type, <span class="at">shape =</span> Type)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>) </span>
<span id="cb125-8"><a href="rna-seq-for-stat736.html#cb125-8" aria-hidden="true" tabindex="-1"></a>p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC1: &quot;</span>,percentVar[<span class="dv">1</span>],<span class="st">&quot;% variance&quot;</span>)) </span>
<span id="cb125-9"><a href="rna-seq-for-stat736.html#cb125-9" aria-hidden="true" tabindex="-1"></a>p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC2: &quot;</span>,percentVar[<span class="dv">2</span>],<span class="st">&quot;% variance&quot;</span>)) </span>
<span id="cb125-10"><a href="rna-seq-for-stat736.html#cb125-10" aria-hidden="true" tabindex="-1"></a>p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">ggtitle</span>(<span class="st">&quot;Principal component analysis (PCA)&quot;</span>)<span class="sc">+</span><span class="fu">coord_fixed</span>(<span class="at">ratio=</span><span class="fl">1.0</span>)<span class="sc">+</span> </span>
<span id="cb125-11"><a href="rna-seq-for-stat736.html#cb125-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>,<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb125-12"><a href="rna-seq-for-stat736.html#cb125-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb125-13"><a href="rna-seq-for-stat736.html#cb125-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb125-14"><a href="rna-seq-for-stat736.html#cb125-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb125-15"><a href="rna-seq-for-stat736.html#cb125-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>) ) <span class="sc">+</span></span>
<span id="cb125-16"><a href="rna-seq-for-stat736.html#cb125-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span>
<span id="cb125-17"><a href="rna-seq-for-stat736.html#cb125-17" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p)</span></code></pre></div>
<p>Multidimensional Scaling Plot</p>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="rna-seq-for-stat736.html#cb126-1" aria-hidden="true" tabindex="-1"></a>dist2 <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ...)   <span class="co"># distance function = 1-PCC (Pearson&#39;s correlation coefficient)</span></span>
<span id="cb126-2"><a href="rna-seq-for-stat736.html#cb126-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span><span class="fu">cor</span>(<span class="fu">t</span>(x), <span class="at">method=</span><span class="st">&quot;pearson&quot;</span>))</span>
<span id="cb126-3"><a href="rna-seq-for-stat736.html#cb126-3" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">=</span> <span class="fu">cmdscale</span>( <span class="fu">dist2</span>(<span class="fu">t</span>(<span class="fu">assay</span>(rld))) , <span class="at">eig=</span>T, <span class="at">k=</span><span class="dv">2</span>)</span>
<span id="cb126-4"><a href="rna-seq-for-stat736.html#cb126-4" aria-hidden="true" tabindex="-1"></a>mdsData <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(fit<span class="sc">$</span>points[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]); </span>
<span id="cb126-5"><a href="rna-seq-for-stat736.html#cb126-5" aria-hidden="true" tabindex="-1"></a>mdsData <span class="ot">&lt;-</span> <span class="fu">cbind</span>(mdsData,<span class="fu">detectGroups</span>(<span class="fu">colnames</span>(<span class="fu">assay</span>(rld))) )</span>
<span id="cb126-6"><a href="rna-seq-for-stat736.html#cb126-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mdsData) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;x1&quot;</span>, <span class="st">&quot;x2&quot;</span>, <span class="st">&quot;Type&quot;</span>)</span>
<span id="cb126-7"><a href="rna-seq-for-stat736.html#cb126-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb126-8"><a href="rna-seq-for-stat736.html#cb126-8" aria-hidden="true" tabindex="-1"></a>p<span class="ot">&lt;-</span><span class="fu">ggplot</span>(mdsData, <span class="fu">aes</span>(x1, x2, <span class="at">color=</span>Type, <span class="at">shape =</span> Type)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">5</span>) </span>
<span id="cb126-9"><a href="rna-seq-for-stat736.html#cb126-9" aria-hidden="true" tabindex="-1"></a>p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">xlab</span>(<span class="st">&quot;Dimension 1&quot;</span>) </span>
<span id="cb126-10"><a href="rna-seq-for-stat736.html#cb126-10" aria-hidden="true" tabindex="-1"></a>p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">ylab</span>(<span class="st">&quot;Dimension 2&quot;</span>) </span>
<span id="cb126-11"><a href="rna-seq-for-stat736.html#cb126-11" aria-hidden="true" tabindex="-1"></a>p<span class="ot">=</span>p<span class="sc">+</span><span class="fu">ggtitle</span>(<span class="st">&quot;Multidimensional scaling (MDS)&quot;</span>)<span class="sc">+</span> <span class="fu">coord_fixed</span>(<span class="at">ratio=</span><span class="fl">1.</span>)<span class="sc">+</span> </span>
<span id="cb126-12"><a href="rna-seq-for-stat736.html#cb126-12" aria-hidden="true" tabindex="-1"></a>     <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb126-13"><a href="rna-seq-for-stat736.html#cb126-13" aria-hidden="true" tabindex="-1"></a>         <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb126-14"><a href="rna-seq-for-stat736.html#cb126-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb126-15"><a href="rna-seq-for-stat736.html#cb126-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb126-16"><a href="rna-seq-for-stat736.html#cb126-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>( <span class="at">size =</span> <span class="dv">16</span>) ) <span class="sc">+</span></span>
<span id="cb126-17"><a href="rna-seq-for-stat736.html#cb126-17" aria-hidden="true" tabindex="-1"></a>       <span class="fu">theme</span>(<span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span>
<span id="cb126-18"><a href="rna-seq-for-stat736.html#cb126-18" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p)</span></code></pre></div>
<p>Creating a heatmap</p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="rna-seq-for-stat736.html#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gplots)</span>
<span id="cb127-2"><a href="rna-seq-for-stat736.html#cb127-2" aria-hidden="true" tabindex="-1"></a>hclust2 <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">method=</span><span class="st">&quot;average&quot;</span>, ...)  <span class="co"># average linkage in hierarchical clustering</span></span>
<span id="cb127-3"><a href="rna-seq-for-stat736.html#cb127-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hclust</span>(x, <span class="at">method=</span>method, ...)</span>
<span id="cb127-4"><a href="rna-seq-for-stat736.html#cb127-4" aria-hidden="true" tabindex="-1"></a>n<span class="ot">=</span><span class="dv">100</span> <span class="co"># number of top genes by standard deviation</span></span>
<span id="cb127-5"><a href="rna-seq-for-stat736.html#cb127-5" aria-hidden="true" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">assay</span>(rld)</span>
<span id="cb127-6"><a href="rna-seq-for-stat736.html#cb127-6" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(n<span class="sc">&gt;</span><span class="fu">dim</span>(x)[<span class="dv">1</span>]) n <span class="ot">=</span> <span class="fu">dim</span>(x)[<span class="dv">1</span>] <span class="co"># max as data</span></span>
<span id="cb127-7"><a href="rna-seq-for-stat736.html#cb127-7" aria-hidden="true" tabindex="-1"></a>x <span class="ot">=</span> x[<span class="fu">order</span>(<span class="fu">apply</span>(x,<span class="dv">1</span>,sd),<span class="at">decreasing=</span><span class="cn">TRUE</span>),]  <span class="co"># sort genes by standard deviation</span></span>
<span id="cb127-8"><a href="rna-seq-for-stat736.html#cb127-8" aria-hidden="true" tabindex="-1"></a>x <span class="ot">=</span> x[<span class="dv">1</span><span class="sc">:</span>n,]   <span class="co"># only keep the n genes</span></span>
<span id="cb127-9"><a href="rna-seq-for-stat736.html#cb127-9" aria-hidden="true" tabindex="-1"></a><span class="co"># this will cutoff very large values, which could skew the color </span></span>
<span id="cb127-10"><a href="rna-seq-for-stat736.html#cb127-10" aria-hidden="true" tabindex="-1"></a>x<span class="ot">=</span><span class="fu">as.matrix</span>(x[<span class="dv">1</span><span class="sc">:</span>n,])<span class="sc">-</span><span class="fu">apply</span>(x[<span class="dv">1</span><span class="sc">:</span>n,],<span class="dv">1</span>,mean)</span>
<span id="cb127-11"><a href="rna-seq-for-stat736.html#cb127-11" aria-hidden="true" tabindex="-1"></a>cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">+</span> <span class="dv">4</span><span class="sc">*</span><span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb127-12"><a href="rna-seq-for-stat736.html#cb127-12" aria-hidden="true" tabindex="-1"></a>x[x<span class="sc">&gt;</span>cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb127-13"><a href="rna-seq-for-stat736.html#cb127-13" aria-hidden="true" tabindex="-1"></a>cutoff <span class="ot">=</span> <span class="fu">median</span>(<span class="fu">unlist</span>(x)) <span class="sc">-</span> <span class="dv">4</span><span class="sc">*</span><span class="fu">sd</span> (<span class="fu">unlist</span>(x)) </span>
<span id="cb127-14"><a href="rna-seq-for-stat736.html#cb127-14" aria-hidden="true" tabindex="-1"></a>x[x<span class="sc">&lt;</span> cutoff] <span class="ot">&lt;-</span> cutoff</span>
<span id="cb127-15"><a href="rna-seq-for-stat736.html#cb127-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb127-16"><a href="rna-seq-for-stat736.html#cb127-16" aria-hidden="true" tabindex="-1"></a>groups <span class="ot">=</span> <span class="fu">detectGroups</span>(<span class="fu">colnames</span>(x) )</span>
<span id="cb127-17"><a href="rna-seq-for-stat736.html#cb127-17" aria-hidden="true" tabindex="-1"></a>groups.colors <span class="ot">=</span> <span class="fu">rainbow</span>(<span class="fu">length</span>(<span class="fu">unique</span>(groups) ) )</span>
<span id="cb127-18"><a href="rna-seq-for-stat736.html#cb127-18" aria-hidden="true" tabindex="-1"></a>    lmat <span class="ot">=</span> <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="dv">5</span>,<span class="dv">4</span>),<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">2</span>))</span>
<span id="cb127-19"><a href="rna-seq-for-stat736.html#cb127-19" aria-hidden="true" tabindex="-1"></a>    lwid <span class="ot">=</span> <span class="fu">c</span>(<span class="fl">1.5</span>,<span class="dv">4</span>)</span>
<span id="cb127-20"><a href="rna-seq-for-stat736.html#cb127-20" aria-hidden="true" tabindex="-1"></a>    lhei <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">1</span>,.<span class="dv">2</span>,<span class="dv">4</span>)</span>
<span id="cb127-21"><a href="rna-seq-for-stat736.html#cb127-21" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap.2</span>(x, <span class="at">distfun =</span> dist2,<span class="at">hclustfun=</span>hclust2,</span>
<span id="cb127-22"><a href="rna-seq-for-stat736.html#cb127-22" aria-hidden="true" tabindex="-1"></a>     <span class="at">col=</span><span class="fu">greenred</span>(<span class="dv">75</span>), <span class="at">density.info=</span><span class="st">&quot;none&quot;</span>, <span class="at">trace=</span><span class="st">&quot;none&quot;</span>, <span class="at">scale=</span><span class="st">&quot;none&quot;</span>, <span class="at">keysize=</span>.<span class="dv">5</span></span>
<span id="cb127-23"><a href="rna-seq-for-stat736.html#cb127-23" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">key=</span>T, <span class="at">symkey=</span>F</span>
<span id="cb127-24"><a href="rna-seq-for-stat736.html#cb127-24" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">ColSideColors=</span>groups.colors[ <span class="fu">as.factor</span>(groups)]</span>
<span id="cb127-25"><a href="rna-seq-for-stat736.html#cb127-25" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">margins=</span><span class="fu">c</span>(<span class="dv">8</span>,<span class="dv">12</span>)</span>
<span id="cb127-26"><a href="rna-seq-for-stat736.html#cb127-26" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">cexRow=</span><span class="dv">1</span></span>
<span id="cb127-27"><a href="rna-seq-for-stat736.html#cb127-27" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">srtCol=</span><span class="dv">45</span></span>
<span id="cb127-28"><a href="rna-seq-for-stat736.html#cb127-28" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">cexCol=</span><span class="fl">1.</span>  <span class="co"># size of font for sample names</span></span>
<span id="cb127-29"><a href="rna-seq-for-stat736.html#cb127-29" aria-hidden="true" tabindex="-1"></a>    ,<span class="at">lmat =</span> lmat, <span class="at">lwid =</span> lwid, <span class="at">lhei =</span> lhei</span>
<span id="cb127-30"><a href="rna-seq-for-stat736.html#cb127-30" aria-hidden="true" tabindex="-1"></a>    )</span></code></pre></div>
</div>
<div id="identification-of-differentially-expressed-genes" class="section level3" number="7.4.2">
<h3><span class="header-section-number">7.4.2</span> Identification of Differentially Expressed Genes</h3>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="rna-seq-for-stat736.html#cb128-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb128-2"><a href="rna-seq-for-stat736.html#cb128-2" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds)</span>
<span id="cb128-3"><a href="rna-seq-for-stat736.html#cb128-3" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(res)) <span class="sc">%&gt;%</span></span>
<span id="cb128-4"><a href="rna-seq-for-stat736.html#cb128-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb128-5"><a href="rna-seq-for-stat736.html#cb128-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span></code></pre></div>
<p>DESeq2 uses the Benjamini-Hochberg (BH) adjustment (Benjamini and Hochberg 1995) as implemented in the base R p.adjust function</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="rna-seq-for-stat736.html#cb129-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">lfcThreshold=</span><span class="fl">0.01</span>)</span>
<span id="cb129-2"><a href="rna-seq-for-stat736.html#cb129-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res)</span></code></pre></div>
<p>Now lets sort genes by fold change</p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="rna-seq-for-stat736.html#cb130-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> res[<span class="fu">order</span>(<span class="fu">abs</span>( res<span class="sc">$</span>log2FoldChange), <span class="at">decreasing=</span><span class="cn">TRUE</span>),]</span>
<span id="cb130-2"><a href="rna-seq-for-stat736.html#cb130-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(res)) <span class="sc">%&gt;%</span></span>
<span id="cb130-3"><a href="rna-seq-for-stat736.html#cb130-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb130-4"><a href="rna-seq-for-stat736.html#cb130-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span></code></pre></div>
<p>MA Plot</p>
<p>Show the significant genes. The lower the average read counts for all samples and the higher the variation between the samples, the less significant those genes are.</p>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="rna-seq-for-stat736.html#cb131-1" aria-hidden="true" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotMA</span>(res,  <span class="at">ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">5</span>))</span></code></pre></div>
<p>Volcano plot</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="rna-seq-for-stat736.html#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb132-2"><a href="rna-seq-for-stat736.html#cb132-2" aria-hidden="true" tabindex="-1"></a>res1 <span class="ot">=</span> <span class="fu">as.data.frame</span>(res)</span>
<span id="cb132-3"><a href="rna-seq-for-stat736.html#cb132-3" aria-hidden="true" tabindex="-1"></a><span class="co"># add a new column using the mutate function in dplyr</span></span>
<span id="cb132-4"><a href="rna-seq-for-stat736.html#cb132-4" aria-hidden="true" tabindex="-1"></a>res1 <span class="ot">=</span> <span class="fu">mutate</span>(res1, <span class="at">sig=</span><span class="fu">ifelse</span>(res1<span class="sc">$</span>padj<span class="sc">&lt;</span><span class="fl">0.05</span>, <span class="st">&quot;FDR&lt;0.05&quot;</span>, <span class="st">&quot;Not Sig&quot;</span>))</span>
<span id="cb132-5"><a href="rna-seq-for-stat736.html#cb132-5" aria-hidden="true" tabindex="-1"></a>res1[<span class="fu">which</span>(<span class="fu">abs</span>(res1<span class="sc">$</span>log2FoldChange)<span class="sc">&lt;</span><span class="fl">0.5</span>),<span class="st">&#39;sig&#39;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Not Sig&quot;</span></span>
<span id="cb132-6"><a href="rna-seq-for-stat736.html#cb132-6" aria-hidden="true" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">ggplot</span>(res1, <span class="fu">aes</span>(log2FoldChange, <span class="sc">-</span><span class="fu">log10</span>(padj))) <span class="sc">+</span></span>
<span id="cb132-7"><a href="rna-seq-for-stat736.html#cb132-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">col=</span>sig)) <span class="sc">+</span></span>
<span id="cb132-8"><a href="rna-seq-for-stat736.html#cb132-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;black&quot;</span>))</span>
<span id="cb132-9"><a href="rna-seq-for-stat736.html#cb132-9" aria-hidden="true" tabindex="-1"></a>p</span></code></pre></div>
</div>
<div id="gene-annotations" class="section level3" number="7.4.3">
<h3><span class="header-section-number">7.4.3</span> Gene Annotations</h3>
<p>Plot counts of top gene</p>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="rna-seq-for-stat736.html#cb133-1" aria-hidden="true" tabindex="-1"></a>topGene <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res)[<span class="dv">1</span>]</span>
<span id="cb133-2"><a href="rna-seq-for-stat736.html#cb133-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotCounts</span>(dds, <span class="at">gene =</span> topGene, <span class="at">intgroup=</span><span class="fu">c</span>(<span class="st">&quot;p53&quot;</span>, <span class="st">&quot;treatment&quot;</span>))</span></code></pre></div>
<p>Here we see an interesting point or our normalized counts under the Trp53p_4h7Gy group that seems to be extremely high, while the other 3 replicates are around 0.5. I cannot get this portion to work, however. I get an error stating that None of the keys entered are valid keys for ‘SYMBOL.’</p>
<p>Let’s look at the keys that we have to work with for our <strong><em>res</em></strong> file from DESeq2</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="rna-seq-for-stat736.html#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">row.names</span>(res))</span></code></pre></div>
<p>Now we need to find the same key in the Mm database.</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="rna-seq-for-stat736.html#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationDbi)</span>
<span id="cb135-2"><a href="rna-seq-for-stat736.html#cb135-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Mm.eg.db)</span>
<span id="cb135-3"><a href="rna-seq-for-stat736.html#cb135-3" aria-hidden="true" tabindex="-1"></a><span class="fu">columns</span>(org.Mm.eg.db)</span>
<span id="cb135-4"><a href="rna-seq-for-stat736.html#cb135-4" aria-hidden="true" tabindex="-1"></a><span class="co">#key = gsub(&quot;\\..*&quot;,&quot;&quot;, row.names(res))</span></span>
<span id="cb135-5"><a href="rna-seq-for-stat736.html#cb135-5" aria-hidden="true" tabindex="-1"></a>res<span class="sc">$</span>symbol <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">..*&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">row.names</span>(res))</span>
<span id="cb135-6"><a href="rna-seq-for-stat736.html#cb135-6" aria-hidden="true" tabindex="-1"></a><span class="co">#res$symbol &lt;- gsub(&quot; &quot;,&quot;&quot;,row.names(res)) </span></span></code></pre></div>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="rna-seq-for-stat736.html#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Ensembl IDs&quot;</span>)</span>
<span id="cb136-2"><a href="rna-seq-for-stat736.html#cb136-2" aria-hidden="true" tabindex="-1"></a>key.en <span class="ot">=</span> <span class="fu">keys</span>(org.Mm.eg.db, <span class="at">keytype=</span><span class="st">&quot;ENSEMBL&quot;</span>)</span>
<span id="cb136-3"><a href="rna-seq-for-stat736.html#cb136-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(key.en)</span>
<span id="cb136-4"><a href="rna-seq-for-stat736.html#cb136-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb136-5"><a href="rna-seq-for-stat736.html#cb136-5" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;SYMBOL names&quot;</span>)</span>
<span id="cb136-6"><a href="rna-seq-for-stat736.html#cb136-6" aria-hidden="true" tabindex="-1"></a>key.sy <span class="ot">=</span> <span class="fu">keys</span>(org.Mm.eg.db, <span class="at">keytype=</span><span class="st">&quot;SYMBOL&quot;</span>)</span>
<span id="cb136-7"><a href="rna-seq-for-stat736.html#cb136-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(key.sy)</span></code></pre></div>
<p>These are ENSEMBL symbols, so we need to designate that when looking for the genes that we have.</p>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="rna-seq-for-stat736.html#cb137-1" aria-hidden="true" tabindex="-1"></a>res<span class="sc">$</span>ensembl <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">..*&quot;</span>,<span class="st">&quot;&quot;</span>, <span class="fu">row.names</span>(res))</span>
<span id="cb137-2"><a href="rna-seq-for-stat736.html#cb137-2" aria-hidden="true" tabindex="-1"></a>res<span class="sc">$</span>entrez <span class="ot">&lt;-</span> <span class="fu">mapIds</span>(org.Mm.eg.db,</span>
<span id="cb137-3"><a href="rna-seq-for-stat736.html#cb137-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">keys=</span> res<span class="sc">$</span>ensembl,</span>
<span id="cb137-4"><a href="rna-seq-for-stat736.html#cb137-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">column=</span><span class="st">&quot;ENTREZID&quot;</span>,</span>
<span id="cb137-5"><a href="rna-seq-for-stat736.html#cb137-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">keytype=</span><span class="st">&quot;ENSEMBL&quot;</span>, <span class="co">#Out ID is ENSMBL</span></span>
<span id="cb137-6"><a href="rna-seq-for-stat736.html#cb137-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">multiVals=</span><span class="st">&quot;first&quot;</span>)</span>
<span id="cb137-7"><a href="rna-seq-for-stat736.html#cb137-7" aria-hidden="true" tabindex="-1"></a>res<span class="sc">$</span>symbol <span class="ot">&lt;-</span> <span class="fu">mapIds</span>(org.Mm.eg.db,</span>
<span id="cb137-8"><a href="rna-seq-for-stat736.html#cb137-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">keys=</span> res<span class="sc">$</span>ensembl,</span>
<span id="cb137-9"><a href="rna-seq-for-stat736.html#cb137-9" aria-hidden="true" tabindex="-1"></a>                     <span class="at">column=</span><span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb137-10"><a href="rna-seq-for-stat736.html#cb137-10" aria-hidden="true" tabindex="-1"></a>                     <span class="at">keytype=</span><span class="st">&quot;ENSEMBL&quot;</span>, <span class="co">#Out ID is ENSMBL</span></span>
<span id="cb137-11"><a href="rna-seq-for-stat736.html#cb137-11" aria-hidden="true" tabindex="-1"></a>                     <span class="at">multiVals=</span><span class="st">&quot;first&quot;</span>)</span>
<span id="cb137-12"><a href="rna-seq-for-stat736.html#cb137-12" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(res, <span class="at">file =</span> <span class="st">&quot;results.csv&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="rna-seq-for-stat736.html#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(res)) <span class="sc">%&gt;%</span></span>
<span id="cb138-2"><a href="rna-seq-for-stat736.html#cb138-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb138-3"><a href="rna-seq-for-stat736.html#cb138-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span></code></pre></div>
<p>Let’s make a file with just the genes with an adjusted p-value &lt; 0.5</p>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="rna-seq-for-stat736.html#cb139-1" aria-hidden="true" tabindex="-1"></a>resSig <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">subset</span>(res,padj<span class="sc">&lt;</span><span class="fl">0.5</span>) )</span>
<span id="cb139-2"><a href="rna-seq-for-stat736.html#cb139-2" aria-hidden="true" tabindex="-1"></a>resSig <span class="ot">=</span> resSig[<span class="fu">order</span>(resSig<span class="sc">$</span>log2FoldChange,<span class="at">decreasing=</span><span class="cn">TRUE</span>),]</span>
<span id="cb139-3"><a href="rna-seq-for-stat736.html#cb139-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(resSig)</span>
<span id="cb139-4"><a href="rna-seq-for-stat736.html#cb139-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(resSig,<span class="st">&quot;SigGenes.csv&quot;</span>)</span></code></pre></div>
<p>Here is a volcano plot that shows the symbol that we created at each point.</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="rna-seq-for-stat736.html#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb140-2"><a href="rna-seq-for-stat736.html#cb140-2" aria-hidden="true" tabindex="-1"></a>res1 <span class="ot">=</span> <span class="fu">as.data.frame</span>(res)</span>
<span id="cb140-3"><a href="rna-seq-for-stat736.html#cb140-3" aria-hidden="true" tabindex="-1"></a><span class="co"># add a new column using the mutate function in dplyr</span></span>
<span id="cb140-4"><a href="rna-seq-for-stat736.html#cb140-4" aria-hidden="true" tabindex="-1"></a>res1 <span class="ot">=</span> <span class="fu">mutate</span>(res1, <span class="at">sig=</span><span class="fu">ifelse</span>(res1<span class="sc">$</span>padj<span class="sc">&lt;</span><span class="fl">0.5</span>, <span class="st">&quot;FDR&lt;0.05&quot;</span>, <span class="st">&quot;Not Sig&quot;</span>))</span>
<span id="cb140-5"><a href="rna-seq-for-stat736.html#cb140-5" aria-hidden="true" tabindex="-1"></a>res1[<span class="fu">which</span>(<span class="fu">abs</span>(res1<span class="sc">$</span>log2FoldChange)<span class="sc">&lt;</span><span class="dv">1</span>),<span class="st">&#39;sig&#39;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Not Sig&quot;</span></span>
<span id="cb140-6"><a href="rna-seq-for-stat736.html#cb140-6" aria-hidden="true" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">ggplot</span>(res1, <span class="fu">aes</span>(log2FoldChange, <span class="sc">-</span><span class="fu">log10</span>(pvalue))) <span class="sc">+</span></span>
<span id="cb140-7"><a href="rna-seq-for-stat736.html#cb140-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">col=</span>sig)) <span class="sc">+</span></span>
<span id="cb140-8"><a href="rna-seq-for-stat736.html#cb140-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;black&quot;</span>))</span>
<span id="cb140-9"><a href="rna-seq-for-stat736.html#cb140-9" aria-hidden="true" tabindex="-1"></a>p<span class="sc">+</span><span class="fu">geom_text</span>(<span class="at">data=</span><span class="fu">filter</span>(res1, padj<span class="sc">&lt;</span><span class="fl">1e-50</span>), <span class="fu">aes</span>(<span class="at">label=</span>symbol))</span></code></pre></div>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="rna-seq-for-stat736.html#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb141-2"><a href="rna-seq-for-stat736.html#cb141-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Install ggrepel package if needed</span></span>
<span id="cb141-3"><a href="rna-seq-for-stat736.html#cb141-3" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;devtools&quot;)</span></span>
<span id="cb141-4"><a href="rna-seq-for-stat736.html#cb141-4" aria-hidden="true" tabindex="-1"></a><span class="co"># devtools::install_github(&quot;slowkow/ggrepel&quot;)</span></span>
<span id="cb141-5"><a href="rna-seq-for-stat736.html#cb141-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel) <span class="co"># &quot;repels&quot; overlapping text</span></span>
<span id="cb141-6"><a href="rna-seq-for-stat736.html#cb141-6" aria-hidden="true" tabindex="-1"></a>p<span class="sc">+</span><span class="fu">geom_text_repel</span>(<span class="at">data=</span><span class="fu">filter</span>(res1,  <span class="fu">abs</span>(log2FoldChange)<span class="sc">&gt;</span><span class="dv">10</span> <span class="sc">|</span> padj <span class="sc">&lt;</span> <span class="fl">1e-50</span> ), <span class="fu">aes</span>(<span class="at">label=</span>symbol))</span></code></pre></div>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="rna-seq-for-stat736.html#cb142-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ReportingTools&quot;</span>)</span>
<span id="cb142-2"><a href="rna-seq-for-stat736.html#cb142-2" aria-hidden="true" tabindex="-1"></a>htmlRep <span class="ot">&lt;-</span> <span class="fu">HTMLReport</span>(<span class="at">shortName=</span><span class="st">&quot;report&quot;</span>, <span class="at">title=</span><span class="st">&quot;My report&quot;</span>,</span>
<span id="cb142-3"><a href="rna-seq-for-stat736.html#cb142-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">reportDirectory=</span><span class="st">&quot;./report&quot;</span>)</span>
<span id="cb142-4"><a href="rna-seq-for-stat736.html#cb142-4" aria-hidden="true" tabindex="-1"></a><span class="fu">publish</span>(resSig, htmlRep)</span>
<span id="cb142-5"><a href="rna-seq-for-stat736.html#cb142-5" aria-hidden="true" tabindex="-1"></a>url <span class="ot">&lt;-</span> <span class="fu">finish</span>(htmlRep)</span>
<span id="cb142-6"><a href="rna-seq-for-stat736.html#cb142-6" aria-hidden="true" tabindex="-1"></a><span class="co">#browseURL(url)</span></span></code></pre></div>
</div>
</div>
<div id="go-enrichment-analysis-using-gostats" class="section level2" number="7.5">
<h2><span class="header-section-number">7.5</span> 7. GO Enrichment analysis using GOstats</h2>
<p>Here we will do a GO Enrichment analysis for genes that have a decreased fold-change of 5 or more</p>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="rna-seq-for-stat736.html#cb143-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GO.db)</span>
<span id="cb143-2"><a href="rna-seq-for-stat736.html#cb143-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GOstats)</span>
<span id="cb143-3"><a href="rna-seq-for-stat736.html#cb143-3" aria-hidden="true" tabindex="-1"></a>selectedGenes <span class="ot">=</span> <span class="fu">unique</span>(resSig[resSig<span class="sc">$</span>log2FoldChange<span class="sc">&gt;</span><span class="dv">5</span>,<span class="st">&#39;entrez&#39;</span>])  <span class="co"># upregulated genes</span></span>
<span id="cb143-4"><a href="rna-seq-for-stat736.html#cb143-4" aria-hidden="true" tabindex="-1"></a>universeGenes <span class="ot">=</span>  <span class="fu">unique</span>( <span class="fu">mapIds</span>(org.Mm.eg.db,</span>
<span id="cb143-5"><a href="rna-seq-for-stat736.html#cb143-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">keys=</span> res<span class="sc">$</span>ensembl,</span>
<span id="cb143-6"><a href="rna-seq-for-stat736.html#cb143-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">column=</span><span class="st">&quot;ENTREZID&quot;</span>,</span>
<span id="cb143-7"><a href="rna-seq-for-stat736.html#cb143-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">keytype=</span><span class="st">&quot;ENSEMBL&quot;</span>, <span class="co">#Out ID is ENSMBL</span></span>
<span id="cb143-8"><a href="rna-seq-for-stat736.html#cb143-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">multiVals=</span><span class="st">&quot;first&quot;</span>)</span>
<span id="cb143-9"><a href="rna-seq-for-stat736.html#cb143-9" aria-hidden="true" tabindex="-1"></a>                    )</span>
<span id="cb143-10"><a href="rna-seq-for-stat736.html#cb143-10" aria-hidden="true" tabindex="-1"></a> hgCutoff <span class="ot">&lt;-</span> <span class="fl">0.001</span></span>
<span id="cb143-11"><a href="rna-seq-for-stat736.html#cb143-11" aria-hidden="true" tabindex="-1"></a> params <span class="ot">&lt;-</span> <span class="fu">new</span>(<span class="st">&quot;GOHyperGParams&quot;</span>,</span>
<span id="cb143-12"><a href="rna-seq-for-stat736.html#cb143-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">geneIds=</span>selectedGenes,</span>
<span id="cb143-13"><a href="rna-seq-for-stat736.html#cb143-13" aria-hidden="true" tabindex="-1"></a>     <span class="at">universeGeneIds=</span>universeGenes,</span>
<span id="cb143-14"><a href="rna-seq-for-stat736.html#cb143-14" aria-hidden="true" tabindex="-1"></a>     <span class="at">annotation=</span><span class="st">&quot;org.Mm.eg.db&quot;</span>,</span>
<span id="cb143-15"><a href="rna-seq-for-stat736.html#cb143-15" aria-hidden="true" tabindex="-1"></a>     <span class="at">ontology=</span><span class="st">&quot;BP&quot;</span>,</span>
<span id="cb143-16"><a href="rna-seq-for-stat736.html#cb143-16" aria-hidden="true" tabindex="-1"></a>     <span class="at">pvalueCutoff=</span>hgCutoff,</span>
<span id="cb143-17"><a href="rna-seq-for-stat736.html#cb143-17" aria-hidden="true" tabindex="-1"></a>     <span class="at">conditional=</span><span class="cn">FALSE</span>,</span>
<span id="cb143-18"><a href="rna-seq-for-stat736.html#cb143-18" aria-hidden="true" tabindex="-1"></a>     <span class="at">testDirection=</span><span class="st">&quot;over&quot;</span>)</span>
<span id="cb143-19"><a href="rna-seq-for-stat736.html#cb143-19" aria-hidden="true" tabindex="-1"></a>hgOver <span class="ot">&lt;-</span> <span class="fu">hyperGTest</span>(params)</span>
<span id="cb143-20"><a href="rna-seq-for-stat736.html#cb143-20" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(hgOver)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,]</span></code></pre></div>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="rna-seq-for-stat736.html#cb144-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(hgOver)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,<span class="fu">c</span>(<span class="st">&quot;GOBPID&quot;</span>,<span class="st">&quot;Pvalue&quot;</span>,<span class="st">&quot;Term&quot;</span>)]</span></code></pre></div>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="rna-seq-for-stat736.html#cb145-1" aria-hidden="true" tabindex="-1"></a>params1 <span class="ot">&lt;-</span> params</span>
<span id="cb145-2"><a href="rna-seq-for-stat736.html#cb145-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ontology</span>(params1) <span class="ot">&lt;-</span> <span class="st">&quot;CC&quot;</span></span>
<span id="cb145-3"><a href="rna-seq-for-stat736.html#cb145-3" aria-hidden="true" tabindex="-1"></a>hgOver <span class="ot">&lt;-</span> <span class="fu">hyperGTest</span>(params1)</span>
<span id="cb145-4"><a href="rna-seq-for-stat736.html#cb145-4" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(hgOver)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,<span class="fu">c</span>(<span class="st">&quot;GOCCID&quot;</span>,<span class="st">&quot;Pvalue&quot;</span>,<span class="st">&quot;Term&quot;</span>)]</span></code></pre></div>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="rna-seq-for-stat736.html#cb146-1" aria-hidden="true" tabindex="-1"></a>params1 <span class="ot">&lt;-</span> params</span>
<span id="cb146-2"><a href="rna-seq-for-stat736.html#cb146-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ontology</span>(params1) <span class="ot">&lt;-</span> <span class="st">&quot;MF&quot;</span></span>
<span id="cb146-3"><a href="rna-seq-for-stat736.html#cb146-3" aria-hidden="true" tabindex="-1"></a>hgOver <span class="ot">&lt;-</span> <span class="fu">hyperGTest</span>(params1)</span>
<span id="cb146-4"><a href="rna-seq-for-stat736.html#cb146-4" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(hgOver)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,<span class="fu">c</span>(<span class="st">&quot;GOMFID&quot;</span>,<span class="st">&quot;Pvalue&quot;</span>,<span class="st">&quot;Term&quot;</span>)]</span></code></pre></div>
<div id="go-enrichment-analysis-of-downregulated-genes" class="section level3" number="7.5.1">
<h3><span class="header-section-number">7.5.1</span> GO Enrichment analysis of downregulated genes</h3>
<p>Next we will have a look at the genes that are upregulated by a fold-change of 5 or greater.</p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="rna-seq-for-stat736.html#cb147-1" aria-hidden="true" tabindex="-1"></a>selectedGenes <span class="ot">=</span> <span class="fu">unique</span>(resSig[resSig<span class="sc">$</span>log2FoldChange<span class="sc">&lt;</span><span class="dv">5</span>,<span class="st">&#39;entrez&#39;</span>])  <span class="co"># upregulated genes</span></span>
<span id="cb147-2"><a href="rna-seq-for-stat736.html#cb147-2" aria-hidden="true" tabindex="-1"></a> params <span class="ot">&lt;-</span> <span class="fu">new</span>(<span class="st">&quot;GOHyperGParams&quot;</span>,</span>
<span id="cb147-3"><a href="rna-seq-for-stat736.html#cb147-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">geneIds=</span>selectedGenes,</span>
<span id="cb147-4"><a href="rna-seq-for-stat736.html#cb147-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">universeGeneIds=</span>universeGenes,</span>
<span id="cb147-5"><a href="rna-seq-for-stat736.html#cb147-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">annotation=</span><span class="st">&quot;org.Mm.eg.db&quot;</span>,</span>
<span id="cb147-6"><a href="rna-seq-for-stat736.html#cb147-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">ontology=</span><span class="st">&quot;BP&quot;</span>,</span>
<span id="cb147-7"><a href="rna-seq-for-stat736.html#cb147-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">pvalueCutoff=</span>hgCutoff,</span>
<span id="cb147-8"><a href="rna-seq-for-stat736.html#cb147-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">conditional=</span><span class="cn">FALSE</span>,</span>
<span id="cb147-9"><a href="rna-seq-for-stat736.html#cb147-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">testDirection=</span><span class="st">&quot;over&quot;</span>)</span>
<span id="cb147-10"><a href="rna-seq-for-stat736.html#cb147-10" aria-hidden="true" tabindex="-1"></a>hgOver <span class="ot">&lt;-</span> <span class="fu">hyperGTest</span>(params)</span>
<span id="cb147-11"><a href="rna-seq-for-stat736.html#cb147-11" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(hgOver)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,<span class="fu">c</span>(<span class="st">&quot;GOBPID&quot;</span>,<span class="st">&quot;Pvalue&quot;</span>,<span class="st">&quot;Term&quot;</span>)]</span></code></pre></div>
</div>
</div>
<div id="pathway-analysis-using-expression-data" class="section level2" number="7.6">
<h2><span class="header-section-number">7.6</span> 8. Pathway analysis using expression data</h2>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="rna-seq-for-stat736.html#cb148-1" aria-hidden="true" tabindex="-1"></a><span class="co"># bioconductor packages</span></span>
<span id="cb148-2"><a href="rna-seq-for-stat736.html#cb148-2" aria-hidden="true" tabindex="-1"></a><span class="co"># source(&quot;https://bioconductor.org/biocLite.R&quot;);</span></span>
<span id="cb148-3"><a href="rna-seq-for-stat736.html#cb148-3" aria-hidden="true" tabindex="-1"></a><span class="co"># biocLite(c(&quot;pathview&quot;,&quot;gage&quot;,&quot;gageData&quot;))</span></span>
<span id="cb148-4"><a href="rna-seq-for-stat736.html#cb148-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pathview) </span>
<span id="cb148-5"><a href="rna-seq-for-stat736.html#cb148-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gage) </span></code></pre></div>
<div id="prepare-data" class="section level3" number="7.6.1">
<h3><span class="header-section-number">7.6.1</span> Prepare data</h3>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="rna-seq-for-stat736.html#cb149-1" aria-hidden="true" tabindex="-1"></a>foldchanges <span class="ot">=</span> res<span class="sc">$</span>log2FoldChange</span>
<span id="cb149-2"><a href="rna-seq-for-stat736.html#cb149-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(foldchanges) <span class="ot">=</span> res<span class="sc">$</span>entrez</span>
<span id="cb149-3"><a href="rna-seq-for-stat736.html#cb149-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(foldchanges)</span></code></pre></div>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="rna-seq-for-stat736.html#cb150-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gageData)</span>
<span id="cb150-2"><a href="rna-seq-for-stat736.html#cb150-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(go.sets.mm)</span>
<span id="cb150-3"><a href="rna-seq-for-stat736.html#cb150-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(go.subs.mm)</span>
<span id="cb150-4"><a href="rna-seq-for-stat736.html#cb150-4" aria-hidden="true" tabindex="-1"></a>gobpsets <span class="ot">=</span> go.sets.mm[go.subs.mm<span class="sc">$</span>BP]</span>
<span id="cb150-5"><a href="rna-seq-for-stat736.html#cb150-5" aria-hidden="true" tabindex="-1"></a>gobpres <span class="ot">=</span> <span class="fu">gage</span>(foldchanges, <span class="at">gsets=</span>gobpsets, <span class="at">same.dir=</span><span class="cn">TRUE</span>)</span>
<span id="cb150-6"><a href="rna-seq-for-stat736.html#cb150-6" aria-hidden="true" tabindex="-1"></a><span class="co">#lapply(gobpres, head)</span></span>
<span id="cb150-7"><a href="rna-seq-for-stat736.html#cb150-7" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Greater&quot;</span>)</span>
<span id="cb150-8"><a href="rna-seq-for-stat736.html#cb150-8" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(gobpres<span class="sc">$</span>greater)) <span class="sc">%&gt;%</span></span>
<span id="cb150-9"><a href="rna-seq-for-stat736.html#cb150-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb150-10"><a href="rna-seq-for-stat736.html#cb150-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span>
<span id="cb150-11"><a href="rna-seq-for-stat736.html#cb150-11" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Less&quot;</span>)</span>
<span id="cb150-12"><a href="rna-seq-for-stat736.html#cb150-12" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(gobpres<span class="sc">$</span>less)) <span class="sc">%&gt;%</span></span>
<span id="cb150-13"><a href="rna-seq-for-stat736.html#cb150-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb150-14"><a href="rna-seq-for-stat736.html#cb150-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span>
<span id="cb150-15"><a href="rna-seq-for-stat736.html#cb150-15" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Stats&quot;</span>)</span>
<span id="cb150-16"><a href="rna-seq-for-stat736.html#cb150-16" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(gobpres<span class="sc">$</span>stats)) <span class="sc">%&gt;%</span></span>
<span id="cb150-17"><a href="rna-seq-for-stat736.html#cb150-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb150-18"><a href="rna-seq-for-stat736.html#cb150-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span></code></pre></div>
</div>
<div id="kegg-pathways" class="section level3" number="7.6.2">
<h3><span class="header-section-number">7.6.2</span> KEGG pathways</h3>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="rna-seq-for-stat736.html#cb151-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gageData)</span>
<span id="cb151-2"><a href="rna-seq-for-stat736.html#cb151-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(kegg.sets.mm)</span>
<span id="cb151-3"><a href="rna-seq-for-stat736.html#cb151-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(sigmet.idx.mm)</span>
<span id="cb151-4"><a href="rna-seq-for-stat736.html#cb151-4" aria-hidden="true" tabindex="-1"></a>kegg.sets.mm <span class="ot">=</span> kegg.sets.mm[sigmet.idx.mm]</span>
<span id="cb151-5"><a href="rna-seq-for-stat736.html#cb151-5" aria-hidden="true" tabindex="-1"></a><span class="co">#head(kegg.sets.mm, 3)</span></span>
<span id="cb151-6"><a href="rna-seq-for-stat736.html#cb151-6" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Greater&quot;</span>)</span>
<span id="cb151-7"><a href="rna-seq-for-stat736.html#cb151-7" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(kegg.sets.mm<span class="sc">$</span>greater)) <span class="sc">%&gt;%</span></span>
<span id="cb151-8"><a href="rna-seq-for-stat736.html#cb151-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb151-9"><a href="rna-seq-for-stat736.html#cb151-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span>
<span id="cb151-10"><a href="rna-seq-for-stat736.html#cb151-10" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Less&quot;</span>)</span>
<span id="cb151-11"><a href="rna-seq-for-stat736.html#cb151-11" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(kegg.sets.mm<span class="sc">$</span>less)) <span class="sc">%&gt;%</span></span>
<span id="cb151-12"><a href="rna-seq-for-stat736.html#cb151-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb151-13"><a href="rna-seq-for-stat736.html#cb151-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span>
<span id="cb151-14"><a href="rna-seq-for-stat736.html#cb151-14" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Stats&quot;</span>)</span>
<span id="cb151-15"><a href="rna-seq-for-stat736.html#cb151-15" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(kegg.sets.mm<span class="sc">$</span>stats)) <span class="sc">%&gt;%</span></span>
<span id="cb151-16"><a href="rna-seq-for-stat736.html#cb151-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb151-17"><a href="rna-seq-for-stat736.html#cb151-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span>
<span id="cb151-18"><a href="rna-seq-for-stat736.html#cb151-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the results</span></span>
<span id="cb151-19"><a href="rna-seq-for-stat736.html#cb151-19" aria-hidden="true" tabindex="-1"></a>keggres <span class="ot">=</span> <span class="fu">gage</span>(foldchanges, <span class="at">gsets=</span>kegg.sets.mm, <span class="at">same.dir=</span><span class="cn">TRUE</span>)</span>
<span id="cb151-20"><a href="rna-seq-for-stat736.html#cb151-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Look at both up (greater), down (less), and statatistics.</span></span>
<span id="cb151-21"><a href="rna-seq-for-stat736.html#cb151-21" aria-hidden="true" tabindex="-1"></a><span class="co">#lapply(keggres, head, n=10)</span></span>
<span id="cb151-22"><a href="rna-seq-for-stat736.html#cb151-22" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Greater&quot;</span>)</span>
<span id="cb151-23"><a href="rna-seq-for-stat736.html#cb151-23" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(keggres<span class="sc">$</span>greater)) <span class="sc">%&gt;%</span></span>
<span id="cb151-24"><a href="rna-seq-for-stat736.html#cb151-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb151-25"><a href="rna-seq-for-stat736.html#cb151-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span>
<span id="cb151-26"><a href="rna-seq-for-stat736.html#cb151-26" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Less&quot;</span>)</span>
<span id="cb151-27"><a href="rna-seq-for-stat736.html#cb151-27" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(keggres<span class="sc">$</span>less)) <span class="sc">%&gt;%</span></span>
<span id="cb151-28"><a href="rna-seq-for-stat736.html#cb151-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb151-29"><a href="rna-seq-for-stat736.html#cb151-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span>
<span id="cb151-30"><a href="rna-seq-for-stat736.html#cb151-30" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Stats&quot;</span>)</span>
<span id="cb151-31"><a href="rna-seq-for-stat736.html#cb151-31" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(keggres<span class="sc">$</span>stats)) <span class="sc">%&gt;%</span></span>
<span id="cb151-32"><a href="rna-seq-for-stat736.html#cb151-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>() <span class="sc">%&gt;%</span></span>
<span id="cb151-33"><a href="rna-seq-for-stat736.html#cb151-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scroll_box</span>(<span class="at">width =</span> <span class="st">&quot;1000px&quot;</span>, <span class="at">height =</span> <span class="st">&quot;300px&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="rna-seq-for-stat736.html#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the pathways</span></span>
<span id="cb152-2"><a href="rna-seq-for-stat736.html#cb152-2" aria-hidden="true" tabindex="-1"></a>keggrespathways <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">id=</span><span class="fu">rownames</span>(keggres<span class="sc">$</span>less), keggres<span class="sc">$</span>less) <span class="sc">%&gt;%</span> </span>
<span id="cb152-3"><a href="rna-seq-for-stat736.html#cb152-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tbl_df</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb152-4"><a href="rna-seq-for-stat736.html#cb152-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">row_number</span>()<span class="sc">&lt;=</span><span class="dv">5</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb152-5"><a href="rna-seq-for-stat736.html#cb152-5" aria-hidden="true" tabindex="-1"></a>  .<span class="sc">$</span>id <span class="sc">%&gt;%</span> </span>
<span id="cb152-6"><a href="rna-seq-for-stat736.html#cb152-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.character</span>()</span>
<span id="cb152-7"><a href="rna-seq-for-stat736.html#cb152-7" aria-hidden="true" tabindex="-1"></a>keggrespathways</span></code></pre></div>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="rna-seq-for-stat736.html#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the IDs.</span></span>
<span id="cb153-2"><a href="rna-seq-for-stat736.html#cb153-2" aria-hidden="true" tabindex="-1"></a>keggresids <span class="ot">=</span> <span class="fu">substr</span>(keggrespathways, <span class="at">start=</span><span class="dv">1</span>, <span class="at">stop=</span><span class="dv">8</span>)</span>
<span id="cb153-3"><a href="rna-seq-for-stat736.html#cb153-3" aria-hidden="true" tabindex="-1"></a>keggresids</span></code></pre></div>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="rna-seq-for-stat736.html#cb154-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define plotting function for applying later</span></span>
<span id="cb154-2"><a href="rna-seq-for-stat736.html#cb154-2" aria-hidden="true" tabindex="-1"></a>plot_pathway <span class="ot">=</span> <span class="cf">function</span>(pid) <span class="fu">pathview</span>(<span class="at">gene.data=</span>foldchanges, <span class="at">pathway.id=</span>pid, <span class="at">species=</span><span class="st">&quot;mmu&quot;</span>, <span class="at">new.signature=</span><span class="cn">FALSE</span>)</span>
<span id="cb154-3"><a href="rna-seq-for-stat736.html#cb154-3" aria-hidden="true" tabindex="-1"></a><span class="co"># plot multiple pathways (plots saved to disk and returns a throwaway list object)</span></span>
<span id="cb154-4"><a href="rna-seq-for-stat736.html#cb154-4" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">=</span> <span class="fu">sapply</span>(keggresids, <span class="cf">function</span>(pid) <span class="fu">pathview</span>(<span class="at">gene.data=</span>foldchanges, <span class="at">pathway.id=</span>pid, <span class="at">species=</span><span class="st">&quot;mmu&quot;</span>))</span></code></pre></div>
</div>
<div id="pathway-and-regulation-of-genes-for-oxidative-phosphorylation." class="section level3" number="7.6.3">
<h3><span class="header-section-number">7.6.3</span> Pathway and regulation of genes for Oxidative phosphorylation.</h3>
<p><code>![Oxidative Phosphorylation](./Mouse_RNA_Seq_p53_genotoxic_files/figure-html/mmu00190.pathview.png)</code></p>
</div>
<div id="pathway-and-regulation-of-genes-for-glycosylphosphatidylinositolgpi-anchor-biosynthesis." class="section level3" number="7.6.4">
<h3><span class="header-section-number">7.6.4</span> Pathway and regulation of genes for Glycosylphosphatidylinositol(GPI)-anchor biosynthesis.</h3>
<p><code>![Glycosylphosphatidylinositol(GPI)-anchor biosynthesis](./Mouse_RNA_Seq_p53_genotoxic_files/figure-html/mmu00563.pathview.png)</code></p>
</div>
<div id="pathway-and-regulation-of-genes-for-rna-polymerase." class="section level3" number="7.6.5">
<h3><span class="header-section-number">7.6.5</span> Pathway and regulation of genes for RNA polymerase.</h3>
<p><code>![RNA polymerase](./Mouse_RNA_Seq_p53_genotoxic_files/figure-html/mmu03020.pathview.png)</code></p>
</div>
<div id="pathway-and-regulation-of-genes-for-nucleotide-excision-repair." class="section level3" number="7.6.6">
<h3><span class="header-section-number">7.6.6</span> Pathway and regulation of genes for Nucleotide excision repair.</h3>
<p><code>![Nucleotide excision repair](./Mouse_RNA_Seq_p53_genotoxic_files/figure-html/mmu03420.pathview.png)</code></p>
</div>
<div id="pathway-and-regulation-of-genes-for-non-homologous-end-joining." class="section level3" number="7.6.7">
<h3><span class="header-section-number">7.6.7</span> Pathway and regulation of genes for Non-homologous end-joining.</h3>
<p><code>![Non-homologous end-joining](./Mouse_RNA_Seq_p53_genotoxic_files/figure-html/mmu03450.pathview.png)</code></p>
<p>##De novo Assembly {#denovo-assembly}</p>
<p><em>AT THIS TIME, MINICONDA3 WAS INSTALLED SO FURTHER TOOLS ARE ALL INSTALLED UNDER THIS</em></p>
<p>Having the genome for RNA-sequencing analysis is very useful, but sometimes it is not available so <em>de novo</em> assembly is used. <em>de novo</em> assembly is using the reads from sequencing to create longer seqeunces called contigs (contiguous sequences). These sequences are then compared to a protein database to get an idea of what proteins are possibly present. The raw sequences are also mapped back to the contigs, treating the contigs as the “genome,” much like we do when we have the genome from the host available.</p>
<p>Lets jump right into it. We are going to be using <a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki">Trinity</a> for the <em>de novo</em> assembly and analysis because there are many tools built into the Trinity tool that allow for the building of contigs, counting of sequences, and even DeSeq/edgeR analyses.</p>
<p>To install Trinity, all I did was ran <strong>conda install -c bioconda trinity</strong>. This installs all of the previously mentioned tools in one go. In addition to Trinity, BLAST (<strong>conda install -c bioconda blast</strong>) and RSEM (<strong>conda install -c bioconda rsem</strong>) were installed.</p>
<div class="sourceCode" id="cb155"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb155-1"><a href="rna-seq-for-stat736.html#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="ex">Trinity</span> <span class="at">--seqType</span> fq <span class="dt">\</span></span>
<span id="cb155-2"><a href="rna-seq-for-stat736.html#cb155-2" aria-hidden="true" tabindex="-1"></a>--left trimmedReads/SRR2121770_Trimmed_1P.fq.gz,trimmedReads/SRR2121771_Trimmed_1P.fq.gz,trimmedReads/SRR2121774_Trimmed_1P.fq.gz,trimmedReads/SRR2121775_Trimmed_1P.fq.gz,trimmedReads/SRR2121778_Trimmed_1P.fq.gz,trimmedReads/SRR2121779_Trimmed_1P.fq.gz,trimmedReads/SRR2121780_Trimmed_1P.fq.gz,trimmedReads/SRR2121781_Trimmed_1P.fq.gz,trimmedReads/SRR2121786_Trimmed_1P.fq.gz,trimmedReads/SRR2121787_Trimmed_1P.fq.gz,trimmedReads/SRR2121788_Trimmed_1P.fq.gz,trimmedReads/SRR2121789_Trimmed_1P.fq.gz <span class="dt">\</span></span>
<span id="cb155-3"><a href="rna-seq-for-stat736.html#cb155-3" aria-hidden="true" tabindex="-1"></a>--right trimmedReads/SRR2121770_Trimmed_2P.fq.gz,trimmedReads/SRR2121771_Trimmed_2P.fq.gz,trimmedReads/SRR2121774_Trimmed_2P.fq.gz,trimmedReads/SRR2121775_Trimmed_2P.fq.gz,trimmedReads/SRR2121778_Trimmed_2P.fq.gz,trimmedReads/SRR2121779_Trimmed_2P.fq.gz,trimmedReads/SRR2121780_Trimmed_2P.fq.gz,trimmedReads/SRR2121781_Trimmed_2P.fq.gz,trimmedReads/SRR2121786_Trimmed_2P.fq.gz,trimmedReads/SRR2121787_Trimmed_2P.fq.gz,trimmedReads/SRR2121788_Trimmed_2P.fq.gz,trimmedReads/SRR2121789_Trimmed_2P.fq.gz <span class="dt">\</span></span>
<span id="cb155-4"><a href="rna-seq-for-stat736.html#cb155-4" aria-hidden="true" tabindex="-1"></a>--CPU 80 <span class="at">--max_memory</span> 2000G <span class="at">--min_contig_length</span> 150</span></code></pre></div>
<div class="sourceCode" id="cb156"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb156-1"><a href="rna-seq-for-stat736.html#cb156-1" aria-hidden="true" tabindex="-1"></a><span class="ex">TrinityStats.pl</span> trinity_out_dir/Trinity.fasta</span>
<span id="cb156-2"><a href="rna-seq-for-stat736.html#cb156-2" aria-hidden="true" tabindex="-1"></a><span class="co">################################</span></span>
<span id="cb156-3"><a href="rna-seq-for-stat736.html#cb156-3" aria-hidden="true" tabindex="-1"></a><span class="co">## Counts of transcripts, etc.</span></span>
<span id="cb156-4"><a href="rna-seq-for-stat736.html#cb156-4" aria-hidden="true" tabindex="-1"></a><span class="co">################################</span></span>
<span id="cb156-5"><a href="rna-seq-for-stat736.html#cb156-5" aria-hidden="true" tabindex="-1"></a><span class="ex">Total</span> trinity <span class="st">&#39;genes&#39;</span>:  602870</span>
<span id="cb156-6"><a href="rna-seq-for-stat736.html#cb156-6" aria-hidden="true" tabindex="-1"></a><span class="ex">Total</span> trinity transcripts:      730297</span>
<span id="cb156-7"><a href="rna-seq-for-stat736.html#cb156-7" aria-hidden="true" tabindex="-1"></a><span class="ex">Percent</span> GC: 45.89</span>
<span id="cb156-8"><a href="rna-seq-for-stat736.html#cb156-8" aria-hidden="true" tabindex="-1"></a><span class="co">########################################</span></span>
<span id="cb156-9"><a href="rna-seq-for-stat736.html#cb156-9" aria-hidden="true" tabindex="-1"></a><span class="ex">Stats</span> based on ALL transcript contigs:</span>
<span id="cb156-10"><a href="rna-seq-for-stat736.html#cb156-10" aria-hidden="true" tabindex="-1"></a><span class="co">########################################</span></span>
<span id="cb156-11"><a href="rna-seq-for-stat736.html#cb156-11" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N10: 8589</span>
<span id="cb156-12"><a href="rna-seq-for-stat736.html#cb156-12" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N20: 5955</span>
<span id="cb156-13"><a href="rna-seq-for-stat736.html#cb156-13" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N30: 4400</span>
<span id="cb156-14"><a href="rna-seq-for-stat736.html#cb156-14" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N40: 3276</span>
<span id="cb156-15"><a href="rna-seq-for-stat736.html#cb156-15" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N50: 2413</span>
<span id="cb156-16"><a href="rna-seq-for-stat736.html#cb156-16" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Median</span> contig length: 413</span>
<span id="cb156-17"><a href="rna-seq-for-stat736.html#cb156-17" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Average</span> contig: 1028.24</span>
<span id="cb156-18"><a href="rna-seq-for-stat736.html#cb156-18" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Total</span> assembled bases: 750921120</span>
<span id="cb156-19"><a href="rna-seq-for-stat736.html#cb156-19" aria-hidden="true" tabindex="-1"></a><span class="co">#####################################################</span></span>
<span id="cb156-20"><a href="rna-seq-for-stat736.html#cb156-20" aria-hidden="true" tabindex="-1"></a><span class="co">## Stats based on ONLY LONGEST ISOFORM per &#39;GENE&#39;:</span></span>
<span id="cb156-21"><a href="rna-seq-for-stat736.html#cb156-21" aria-hidden="true" tabindex="-1"></a><span class="co">#####################################################</span></span>
<span id="cb156-22"><a href="rna-seq-for-stat736.html#cb156-22" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N10: 5470</span>
<span id="cb156-23"><a href="rna-seq-for-stat736.html#cb156-23" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N20: 3434</span>
<span id="cb156-24"><a href="rna-seq-for-stat736.html#cb156-24" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N30: 2391</span>
<span id="cb156-25"><a href="rna-seq-for-stat736.html#cb156-25" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N40: 1741</span>
<span id="cb156-26"><a href="rna-seq-for-stat736.html#cb156-26" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Contig</span> N50: 1285</span>
<span id="cb156-27"><a href="rna-seq-for-stat736.html#cb156-27" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Median</span> contig length: 343</span>
<span id="cb156-28"><a href="rna-seq-for-stat736.html#cb156-28" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Average</span> contig: 703.04</span>
<span id="cb156-29"><a href="rna-seq-for-stat736.html#cb156-29" aria-hidden="true" tabindex="-1"></a>        <span class="ex">Total</span> assembled bases: 423841627</span></code></pre></div>
<div class="sourceCode" id="cb157"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb157-1"><a href="rna-seq-for-stat736.html#cb157-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bowtie2-build</span> trinity_out_dir/Trinity.fasta trinity_out_dir/Trinity.fasta</span>
<span id="cb157-2"><a href="rna-seq-for-stat736.html#cb157-2" aria-hidden="true" tabindex="-1"></a><span class="ex">bowtie2</span> <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out</span></code></pre></div>
<p>##download uniprot swiss-prot db
##<a href="ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz" class="uri">ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz</a></p>
<div class="sourceCode" id="cb158"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb158-1"><a href="rna-seq-for-stat736.html#cb158-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gunzip</span> uniprot_sprot.fasta.gz</span>
<span id="cb158-2"><a href="rna-seq-for-stat736.html#cb158-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> blast_protdb</span>
<span id="cb158-3"><a href="rna-seq-for-stat736.html#cb158-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mv</span> uniprot_sprot.fasta blast_protdb/.</span>
<span id="cb158-4"><a href="rna-seq-for-stat736.html#cb158-4" aria-hidden="true" tabindex="-1"></a><span class="ex">makeblastdb</span> <span class="at">-in</span> uniprot_sprot.fasta <span class="at">-dbtype</span> prot</span>
<span id="cb158-5"><a href="rna-seq-for-stat736.html#cb158-5" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ..</span></code></pre></div>
<div class="sourceCode" id="cb159"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb159-1"><a href="rna-seq-for-stat736.html#cb159-1" aria-hidden="true" tabindex="-1"></a><span class="ex">blastx</span> <span class="at">-query</span> trinity_out_dir/Trinity.fasta <span class="at">-db</span> blast_protdb/uniprot_sprot.fasta <span class="at">-out</span> blastx.outfmt6 <span class="at">-evalue</span> 1e-20 <span class="at">-num_threads</span> 80 <span class="at">-max_target_seqs</span> 1 <span class="at">-outfmt</span> 6</span>
<span id="cb159-2"><a href="rna-seq-for-stat736.html#cb159-2" aria-hidden="true" tabindex="-1"></a><span class="ex">analyze_blastPlus_topHit_coverage.pl</span> blastx.outfmt6 trinity_out_dir/Trinity.fasta blast_protdb/uniprot_sprot.fasta <span class="kw">|</span> <span class="ex">column</span> <span class="at">-t</span></span>
<span id="cb159-3"><a href="rna-seq-for-stat736.html#cb159-3" aria-hidden="true" tabindex="-1"></a><span class="co">#hit_pct_cov_bin  count_in_bin  &gt;bin_below</span></span>
<span id="cb159-4"><a href="rna-seq-for-stat736.html#cb159-4" aria-hidden="true" tabindex="-1"></a><span class="ex">100</span>               7361          7361</span>
<span id="cb159-5"><a href="rna-seq-for-stat736.html#cb159-5" aria-hidden="true" tabindex="-1"></a><span class="ex">90</span>                1117          8478</span>
<span id="cb159-6"><a href="rna-seq-for-stat736.html#cb159-6" aria-hidden="true" tabindex="-1"></a><span class="ex">80</span>                981           9459</span>
<span id="cb159-7"><a href="rna-seq-for-stat736.html#cb159-7" aria-hidden="true" tabindex="-1"></a><span class="ex">70</span>                971           10430</span>
<span id="cb159-8"><a href="rna-seq-for-stat736.html#cb159-8" aria-hidden="true" tabindex="-1"></a><span class="ex">60</span>                1003          11433</span>
<span id="cb159-9"><a href="rna-seq-for-stat736.html#cb159-9" aria-hidden="true" tabindex="-1"></a><span class="ex">50</span>                865           12298</span>
<span id="cb159-10"><a href="rna-seq-for-stat736.html#cb159-10" aria-hidden="true" tabindex="-1"></a><span class="ex">40</span>                924           13222</span>
<span id="cb159-11"><a href="rna-seq-for-stat736.html#cb159-11" aria-hidden="true" tabindex="-1"></a><span class="ex">30</span>                1102          14324</span>
<span id="cb159-12"><a href="rna-seq-for-stat736.html#cb159-12" aria-hidden="true" tabindex="-1"></a><span class="ex">20</span>                1478          15802</span>
<span id="cb159-13"><a href="rna-seq-for-stat736.html#cb159-13" aria-hidden="true" tabindex="-1"></a><span class="ex">10</span>                932           16734</span></code></pre></div>
<div class="sourceCode" id="cb160"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb160-1"><a href="rna-seq-for-stat736.html#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121770_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121770_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121770.coordSorted.bam</span>
<span id="cb160-2"><a href="rna-seq-for-stat736.html#cb160-2" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121771_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121771_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121771.coordSorted.bam</span>
<span id="cb160-3"><a href="rna-seq-for-stat736.html#cb160-3" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121774_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121774_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121774.coordSorted.bam</span>
<span id="cb160-4"><a href="rna-seq-for-stat736.html#cb160-4" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121775_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121775_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121775.coordSorted.bam</span>
<span id="cb160-5"><a href="rna-seq-for-stat736.html#cb160-5" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121778_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121778_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121778.coordSorted.bam</span>
<span id="cb160-6"><a href="rna-seq-for-stat736.html#cb160-6" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121779_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121779_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121779.coordSorted.bam</span>
<span id="cb160-7"><a href="rna-seq-for-stat736.html#cb160-7" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121780_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121780_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121780.coordSorted.bam</span>
<span id="cb160-8"><a href="rna-seq-for-stat736.html#cb160-8" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121781_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121781_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121781.coordSorted.bam</span>
<span id="cb160-9"><a href="rna-seq-for-stat736.html#cb160-9" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121786_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121786_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121786.coordSorted.bam</span>
<span id="cb160-10"><a href="rna-seq-for-stat736.html#cb160-10" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121787_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121787_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121787.coordSorted.bam</span>
<span id="cb160-11"><a href="rna-seq-for-stat736.html#cb160-11" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121788_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121788_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121788.coordSorted.bam</span>
<span id="cb160-12"><a href="rna-seq-for-stat736.html#cb160-12" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> bowtie2 <span class="at">--local</span> <span class="at">--no-unal</span> <span class="at">-x</span> trinity_out_dir/Trinity.fasta <span class="at">-q</span> <span class="at">-1</span> rawReads/SRR2121789_1.fastq.gz <span class="at">-2</span> rawReads/SRR2121789_2.fastq.gz <span class="at">--threads</span> 10 <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">--threads</span> 10 <span class="at">-Sb</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">--threads</span> 10 <span class="at">-o</span> SRR2121789.coordSorted.bam</span></code></pre></div>
<div class="sourceCode" id="cb161"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb161-1"><a href="rna-seq-for-stat736.html#cb161-1" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121770_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121770_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121770.RSEM</span>
<span id="cb161-2"><a href="rna-seq-for-stat736.html#cb161-2" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121771_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121771_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121771.RSEM</span>
<span id="cb161-3"><a href="rna-seq-for-stat736.html#cb161-3" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121774_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121774_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121774.RSEM</span>
<span id="cb161-4"><a href="rna-seq-for-stat736.html#cb161-4" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121775_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121775_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121775.RSEM</span>
<span id="cb161-5"><a href="rna-seq-for-stat736.html#cb161-5" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121778_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121778_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121778.RSEM</span>
<span id="cb161-6"><a href="rna-seq-for-stat736.html#cb161-6" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121779_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121779_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121779.RSEM</span>
<span id="cb161-7"><a href="rna-seq-for-stat736.html#cb161-7" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121780_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121780_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121780.RSEM</span>
<span id="cb161-8"><a href="rna-seq-for-stat736.html#cb161-8" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121781_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121781_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121781.RSEM</span>
<span id="cb161-9"><a href="rna-seq-for-stat736.html#cb161-9" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121786_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121786_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121786.RSEM</span>
<span id="cb161-10"><a href="rna-seq-for-stat736.html#cb161-10" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121787_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121787_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121787.RSEM</span>
<span id="cb161-11"><a href="rna-seq-for-stat736.html#cb161-11" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121788_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121788_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121788.RSEM</span>
<span id="cb161-12"><a href="rna-seq-for-stat736.html#cb161-12" aria-hidden="true" tabindex="-1"></a><span class="ex">$</span> time align_and_estimate_abundance.pl <span class="at">--seqType</span> fq <span class="at">--thread_count</span> <span class="va">$SLURM_NTASKS</span> <span class="at">--left</span> rawReads/SRR2121789_1.fastq.gz <span class="at">--right</span> rawReads/SRR2121789_2.fastq.gz <span class="at">--transcripts</span> trinity_out_dir/Trinity.fasta <span class="at">--est_method</span> RSEM <span class="at">--aln_method</span> bowtie2 <span class="at">--trinity_mode</span> <span class="at">--prep_reference</span> <span class="at">--output_dir</span> SRR2121789.RSEM</span></code></pre></div>
<div class="sourceCode" id="cb162"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb162-1"><a href="rna-seq-for-stat736.html#cb162-1" aria-hidden="true" tabindex="-1"></a><span class="bu">time</span> abundance_estimates_to_matrix.pl <span class="at">--est_method</span> RSEM <span class="at">--out_prefix</span> Trinity_trans SRR2121770.RSEM/SRR2121770.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb162-2"><a href="rna-seq-for-stat736.html#cb162-2" aria-hidden="true" tabindex="-1"></a>SRR2121771.RSEM/SRR2121771.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb162-3"><a href="rna-seq-for-stat736.html#cb162-3" aria-hidden="true" tabindex="-1"></a>SRR2121774.RSEM/SRR2121774.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb162-4"><a href="rna-seq-for-stat736.html#cb162-4" aria-hidden="true" tabindex="-1"></a>SRR2121775.RSEM/SRR2121775.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb162-5"><a href="rna-seq-for-stat736.html#cb162-5" aria-hidden="true" tabindex="-1"></a>SRR2121778.RSEM/SRR2121778.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb162-6"><a href="rna-seq-for-stat736.html#cb162-6" aria-hidden="true" tabindex="-1"></a>SRR2121779.RSEM/SRR2121779.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb162-7"><a href="rna-seq-for-stat736.html#cb162-7" aria-hidden="true" tabindex="-1"></a>SRR2121780.RSEM/SRR2121780.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb162-8"><a href="rna-seq-for-stat736.html#cb162-8" aria-hidden="true" tabindex="-1"></a>SRR2121781.RSEM/SRR2121781.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb162-9"><a href="rna-seq-for-stat736.html#cb162-9" aria-hidden="true" tabindex="-1"></a>SRR2121786.RSEM/SRR2121786.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb162-10"><a href="rna-seq-for-stat736.html#cb162-10" aria-hidden="true" tabindex="-1"></a>SRR2121787.RSEM/SRR2121787.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb162-11"><a href="rna-seq-for-stat736.html#cb162-11" aria-hidden="true" tabindex="-1"></a>SRR2121788.RSEM/SRR2121788.RSEM.trans.results <span class="dt">\</span></span>
<span id="cb162-12"><a href="rna-seq-for-stat736.html#cb162-12" aria-hidden="true" tabindex="-1"></a>SRR2121789.RSEM/SRR2121789.RSEM.trans.results <span class="at">--gene_trans_map</span> none</span></code></pre></div>
<div class="sourceCode" id="cb163"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb163-1"><a href="rna-seq-for-stat736.html#cb163-1" aria-hidden="true" tabindex="-1"></a><span class="bu">time</span> abundance_estimates_to_matrix.pl <span class="at">--est_method</span> RSEM <span class="at">--out_prefix</span> Trinity_genes SRR2121770.RSEM/SRR2121770.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb163-2"><a href="rna-seq-for-stat736.html#cb163-2" aria-hidden="true" tabindex="-1"></a>SRR2121771.RSEM/SRR2121771.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb163-3"><a href="rna-seq-for-stat736.html#cb163-3" aria-hidden="true" tabindex="-1"></a>SRR2121774.RSEM/SRR2121774.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb163-4"><a href="rna-seq-for-stat736.html#cb163-4" aria-hidden="true" tabindex="-1"></a>SRR2121775.RSEM/SRR2121775.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb163-5"><a href="rna-seq-for-stat736.html#cb163-5" aria-hidden="true" tabindex="-1"></a>SRR2121778.RSEM/SRR2121778.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb163-6"><a href="rna-seq-for-stat736.html#cb163-6" aria-hidden="true" tabindex="-1"></a>SRR2121779.RSEM/SRR2121779.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb163-7"><a href="rna-seq-for-stat736.html#cb163-7" aria-hidden="true" tabindex="-1"></a>SRR2121780.RSEM/SRR2121780.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb163-8"><a href="rna-seq-for-stat736.html#cb163-8" aria-hidden="true" tabindex="-1"></a>SRR2121781.RSEM/SRR2121781.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb163-9"><a href="rna-seq-for-stat736.html#cb163-9" aria-hidden="true" tabindex="-1"></a>SRR2121786.RSEM/SRR2121786.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb163-10"><a href="rna-seq-for-stat736.html#cb163-10" aria-hidden="true" tabindex="-1"></a>SRR2121787.RSEM/SRR2121787.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb163-11"><a href="rna-seq-for-stat736.html#cb163-11" aria-hidden="true" tabindex="-1"></a>SRR2121788.RSEM/SRR2121788.RSEM.genes.results <span class="dt">\</span></span>
<span id="cb163-12"><a href="rna-seq-for-stat736.html#cb163-12" aria-hidden="true" tabindex="-1"></a>SRR2121789.RSEM/SRR2121789.RSEM.genes.results <span class="at">--gene_trans_map</span> none</span></code></pre></div>
<div class="sourceCode" id="cb164"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb164-1"><a href="rna-seq-for-stat736.html#cb164-1" aria-hidden="true" tabindex="-1"></a><span class="ex">contig_Exn50_statistic.pl</span> Trinity_trans.TMM.EXPR.matrix trinity_out_dir/Trinity.fasta <span class="op">&gt;</span> ExN50_trans.stats</span></code></pre></div>
<p>Make samples.txt</p>
<div class="sourceCode" id="cb165"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb165-1"><a href="rna-seq-for-stat736.html#cb165-1" aria-hidden="true" tabindex="-1"></a><span class="ex">mock-</span>   SRR2121770.RSEM</span>
<span id="cb165-2"><a href="rna-seq-for-stat736.html#cb165-2" aria-hidden="true" tabindex="-1"></a><span class="ex">mock-</span>   SRR2121771.RSEM</span>
<span id="cb165-3"><a href="rna-seq-for-stat736.html#cb165-3" aria-hidden="true" tabindex="-1"></a><span class="ex">IR-</span> SRR2121774.RSEM</span>
<span id="cb165-4"><a href="rna-seq-for-stat736.html#cb165-4" aria-hidden="true" tabindex="-1"></a><span class="ex">IR-</span> SRR2121775.RSEM</span>
<span id="cb165-5"><a href="rna-seq-for-stat736.html#cb165-5" aria-hidden="true" tabindex="-1"></a><span class="ex">mock+</span>   SRR2121778.RSEM</span>
<span id="cb165-6"><a href="rna-seq-for-stat736.html#cb165-6" aria-hidden="true" tabindex="-1"></a><span class="ex">mock+</span>   SRR2121779.RSEM</span>
<span id="cb165-7"><a href="rna-seq-for-stat736.html#cb165-7" aria-hidden="true" tabindex="-1"></a><span class="ex">mock+</span>   SRR2121780.RSEM</span>
<span id="cb165-8"><a href="rna-seq-for-stat736.html#cb165-8" aria-hidden="true" tabindex="-1"></a><span class="ex">mock+</span>   SRR2121781.RSEM</span>
<span id="cb165-9"><a href="rna-seq-for-stat736.html#cb165-9" aria-hidden="true" tabindex="-1"></a><span class="ex">IR+</span> SRR2121786.RSEM</span>
<span id="cb165-10"><a href="rna-seq-for-stat736.html#cb165-10" aria-hidden="true" tabindex="-1"></a><span class="ex">IR+</span> SRR2121787.RSEM</span>
<span id="cb165-11"><a href="rna-seq-for-stat736.html#cb165-11" aria-hidden="true" tabindex="-1"></a><span class="ex">IR+</span> SRR2121788.RSEM</span>
<span id="cb165-12"><a href="rna-seq-for-stat736.html#cb165-12" aria-hidden="true" tabindex="-1"></a><span class="ex">IR+</span> SRR2121789.RSEM</span></code></pre></div>
<p>The way that the installed version of <em>run_DE_analysis.pl</em> tries to install or check for edgeR doesn’t work for the new versions of R (3.6.1). In the perl script I just had to edit a few lines under the edgeR section for it to run smoothly, as well as install *BiocManager before hand.</p>
<p>I changed the perl script, under <em>sub run_edgeR_sample_pair</em>, when writing the R script to look like:</p>
<div class="sourceCode" id="cb166"><pre class="sourceCode perl"><code class="sourceCode perl"><span id="cb166-1"><a href="rna-seq-for-stat736.html#cb166-1" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span> <span class="dt">$ofh</span> <span class="ot">&quot;</span><span class="st">if (! require(edgeR)) {</span><span class="ch">\n</span><span class="ot">&quot;</span>;</span>
<span id="cb166-2"><a href="rna-seq-for-stat736.html#cb166-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span> <span class="dt">$ofh</span> <span class="ot">&quot;</span><span class="st">   install.packages(\&quot;BiocManager\&quot;)</span><span class="ch">\n</span><span class="ot">&quot;</span>;</span>
<span id="cb166-3"><a href="rna-seq-for-stat736.html#cb166-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span> <span class="dt">$ofh</span> <span class="ot">&quot;</span><span class="st">   BiocManager::install(\&quot;edgeR\&quot;)</span><span class="ch">\n</span><span class="ot">&quot;</span>;</span>
<span id="cb166-4"><a href="rna-seq-for-stat736.html#cb166-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span> <span class="dt">$ofh</span> <span class="ot">&quot;</span><span class="st">   library(edgeR)</span><span class="ch">\n</span><span class="ot">&quot;</span>;</span>
<span id="cb166-5"><a href="rna-seq-for-stat736.html#cb166-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span> <span class="dt">$ofh</span> <span class="ot">&quot;</span><span class="st">}</span><span class="ch">\n\n</span><span class="ot">&quot;</span>;</span></code></pre></div>
<p>To make sure BiocManager and edgeR work, they were installed in the R terminal.</p>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="rna-seq-for-stat736.html#cb167-1" aria-hidden="true" tabindex="-1"></a>R <span class="co">#runs the installed version of R in terminal mode</span></span>
<span id="cb167-2"><a href="rna-seq-for-stat736.html#cb167-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb167-3"><a href="rna-seq-for-stat736.html#cb167-3" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;edgeR&quot;</span>)</span></code></pre></div>
<p>Since we don’t have at least 3 replicates for each of the treatments, edgeR wants a dispersion parameter. 0.035 seems to be pretty common. If the dispersion parameter has the squareroot taken of it we get ~ 0.19, meaning that we are saying the true abundance for each gene can vary up or down by 19% between replicates.</p>
<div class="sourceCode" id="cb168"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb168-1"><a href="rna-seq-for-stat736.html#cb168-1" aria-hidden="true" tabindex="-1"></a><span class="ex">run_DE_analysis.pl</span> <span class="at">--matrix</span> Trinity_trans.counts.matrix <span class="at">--method</span> edgeR <span class="at">--output</span> edgeR_trans <span class="at">--dispersion</span> 0.035 <span class="at">--samples_file</span> samples.txt</span></code></pre></div>
<div class="sourceCode" id="cb169"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb169-1"><a href="rna-seq-for-stat736.html#cb169-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> edgeR_trans/Trinity_trans.counts.matrix.IR-_vs_mock-.edgeR.DE_results <span class="kw">|</span> <span class="ex">column</span> <span class="at">-t</span></span>
<span id="cb169-2"><a href="rna-seq-for-stat736.html#cb169-2" aria-hidden="true" tabindex="-1"></a>                          <span class="ex">sampleA</span>  sampleB  logFC           logCPM             PValue                FDR</span>
<span id="cb169-3"><a href="rna-seq-for-stat736.html#cb169-3" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN2468_c0_g4_i1</span>   IR-      mock-  <span class="at">-14.7753615825754</span>  4.45315041918     5.51021995613131e-115  5.85185359341145e-110</span>
<span id="cb169-4"><a href="rna-seq-for-stat736.html#cb169-4" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN2505_c0_g1_i2</span>   IR-      mock-  <span class="at">-10.1817621187735</span>  4.11815268306374  8.54695600746168e-88   4.53843363996215e-83</span>
<span id="cb169-5"><a href="rna-seq-for-stat736.html#cb169-5" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN3231_c0_g2_i15</span>  IR-      mock-  <span class="at">-13.5922145319911</span>  3.27307743751541  6.29068575237937e-81   2.2269027563423e-76</span>
<span id="cb169-6"><a href="rna-seq-for-stat736.html#cb169-6" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN4975_c0_g1_i7</span>   IR-      mock-  13.4941229058571   3.17132827495398  5.25978730838885e-79   1.39647353037724e-74</span>
<span id="cb169-7"><a href="rna-seq-for-stat736.html#cb169-7" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN4360_c0_g1_i5</span>   IR-      mock-  <span class="at">-13.5215746751431</span>  3.20272600532575  1.0222845072488e-70    2.17133229339645e-66</span>
<span id="cb169-8"><a href="rna-seq-for-stat736.html#cb169-8" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN128_c0_g1_i20</span>   IR-      mock-  13.1582688371709   2.83599676163376  1.8443766792831e-69    3.26454672233108e-65</span>
<span id="cb169-9"><a href="rna-seq-for-stat736.html#cb169-9" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN644_c1_g1_i6</span>    IR-      mock-  <span class="at">-5.91850038396836</span>  4.31456225100214  8.01440426808572e-67   1.21589961895815e-62</span>
<span id="cb169-10"><a href="rna-seq-for-stat736.html#cb169-10" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN2043_c0_g1_i13</span>  IR-      mock-  13.7803059024088   3.45719012794474  3.30519380237759e-66   4.38764477265625e-62</span>
<span id="cb169-11"><a href="rna-seq-for-stat736.html#cb169-11" aria-hidden="true" tabindex="-1"></a><span class="ex">TRINITY_DN3321_c0_g1_i2</span>   IR-      mock-  12.8215942182179   2.49995284585527  9.5059376972195e-61    1.1217006482719e-56</span></code></pre></div>
<div class="sourceCode" id="cb170"><pre class="sourceCode bash"><code class="sourceCode bash"></code></pre></div>
<div class="sourceCode" id="cb171"><pre class="sourceCode bash"><code class="sourceCode bash"></code></pre></div>
<div class="sourceCode" id="cb172"><pre class="sourceCode bash"><code class="sourceCode bash"></code></pre></div>
<div class="sourceCode" id="cb173"><pre class="sourceCode bash"><code class="sourceCode bash"></code></pre></div>
<div class="sourceCode" id="cb174"><pre class="sourceCode bash"><code class="sourceCode bash"></code></pre></div>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb175-1"><a href="rna-seq-for-stat736.html#cb175-1" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install(&quot;EnsDb.Mmusculus.v79&quot;)</span></span>
<span id="cb175-2"><a href="rna-seq-for-stat736.html#cb175-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GO.db)</span>
<span id="cb175-3"><a href="rna-seq-for-stat736.html#cb175-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GOstats)</span>
<span id="cb175-4"><a href="rna-seq-for-stat736.html#cb175-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EnsDb.Mmusculus.v79)</span>
<span id="cb175-5"><a href="rna-seq-for-stat736.html#cb175-5" aria-hidden="true" tabindex="-1"></a>res<span class="fl">.3</span> <span class="ot">=</span> <span class="fu">subset</span>(res<span class="fl">.2</span>, <span class="sc">!</span><span class="fu">is.na</span>(res<span class="fl">.2</span><span class="sc">$</span>entrezgene_id))</span>
<span id="cb175-6"><a href="rna-seq-for-stat736.html#cb175-6" aria-hidden="true" tabindex="-1"></a>selectedGenes <span class="ot">=</span> <span class="fu">unique</span>(resSig[resSig<span class="sc">$</span>log2FoldChange<span class="sc">&gt;</span><span class="dv">0</span>,<span class="st">&#39;entrezgene_id&#39;</span>])  <span class="co"># upregulated genes</span></span>
<span id="cb175-7"><a href="rna-seq-for-stat736.html#cb175-7" aria-hidden="true" tabindex="-1"></a>universeGenes <span class="ot">=</span>  <span class="fu">unique</span>( <span class="fu">mapIds</span>(org.Mm.eg.db,</span>
<span id="cb175-8"><a href="rna-seq-for-stat736.html#cb175-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">keys=</span><span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.[0-9]*$&quot;</span>,<span class="st">&quot;&quot;</span>,res<span class="fl">.3</span><span class="sc">$</span>entrezgene_id),  <span class="co"># this is causing problems for mapping</span></span>
<span id="cb175-9"><a href="rna-seq-for-stat736.html#cb175-9" aria-hidden="true" tabindex="-1"></a>                     <span class="at">column=</span><span class="st">&quot;ENSEMBL&quot;</span>,</span>
<span id="cb175-10"><a href="rna-seq-for-stat736.html#cb175-10" aria-hidden="true" tabindex="-1"></a>                     <span class="at">keytype=</span><span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb175-11"><a href="rna-seq-for-stat736.html#cb175-11" aria-hidden="true" tabindex="-1"></a>                     <span class="at">multiVals=</span><span class="st">&quot;first&quot;</span>)</span>
<span id="cb175-12"><a href="rna-seq-for-stat736.html#cb175-12" aria-hidden="true" tabindex="-1"></a>                    )</span>
<span id="cb175-13"><a href="rna-seq-for-stat736.html#cb175-13" aria-hidden="true" tabindex="-1"></a>hgCutoff <span class="ot">&lt;-</span> <span class="fl">0.001</span></span>
<span id="cb175-14"><a href="rna-seq-for-stat736.html#cb175-14" aria-hidden="true" tabindex="-1"></a>params <span class="ot">&lt;-</span> <span class="fu">new</span>(<span class="st">&quot;GOHyperGParams&quot;</span>,</span>
<span id="cb175-15"><a href="rna-seq-for-stat736.html#cb175-15" aria-hidden="true" tabindex="-1"></a>              <span class="at">geneIds=</span>selectedGenes,</span>
<span id="cb175-16"><a href="rna-seq-for-stat736.html#cb175-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">universeGeneIds=</span>universeGenes,</span>
<span id="cb175-17"><a href="rna-seq-for-stat736.html#cb175-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">annotation=</span><span class="st">&quot;org.Hs.eg.db&quot;</span>,</span>
<span id="cb175-18"><a href="rna-seq-for-stat736.html#cb175-18" aria-hidden="true" tabindex="-1"></a>              <span class="at">ontology=</span><span class="st">&quot;BP&quot;</span>,</span>
<span id="cb175-19"><a href="rna-seq-for-stat736.html#cb175-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff=</span>hgCutoff,</span>
<span id="cb175-20"><a href="rna-seq-for-stat736.html#cb175-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">conditional=</span><span class="cn">FALSE</span>,</span>
<span id="cb175-21"><a href="rna-seq-for-stat736.html#cb175-21" aria-hidden="true" tabindex="-1"></a>              <span class="at">testDirection=</span><span class="st">&quot;over&quot;</span>)</span>
<span id="cb175-22"><a href="rna-seq-for-stat736.html#cb175-22" aria-hidden="true" tabindex="-1"></a>hgOver <span class="ot">&lt;-</span> <span class="fu">hyperGTest</span>(params)</span>
<span id="cb175-23"><a href="rna-seq-for-stat736.html#cb175-23" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(hgOver)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,]</span></code></pre></div>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="rna-seq-for-stat736.html#cb176-1" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install(&quot;AnnotationHub&quot;)</span></span>
<span id="cb176-2"><a href="rna-seq-for-stat736.html#cb176-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationHub)</span>
<span id="cb176-3"><a href="rna-seq-for-stat736.html#cb176-3" aria-hidden="true" tabindex="-1"></a>hub <span class="ot">=</span> <span class="fu">AnnotationHub</span>()</span>
<span id="cb176-4"><a href="rna-seq-for-stat736.html#cb176-4" aria-hidden="true" tabindex="-1"></a><span class="fu">query</span>(hub, <span class="fu">c</span>(<span class="st">&quot;Medicago&quot;</span>, <span class="st">&quot;OrgDb&quot;</span>))</span>
<span id="cb176-5"><a href="rna-seq-for-stat736.html#cb176-5" aria-hidden="true" tabindex="-1"></a>Mt.orgdb <span class="ot">=</span> hub[[<span class="st">&quot;AH72235&quot;</span>]]</span>
<span id="cb176-6"><a href="rna-seq-for-stat736.html#cb176-6" aria-hidden="true" tabindex="-1"></a>Mt.orgdb</span></code></pre></div>
<div class="sourceCode" id="cb177"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb177-1"><a href="rna-seq-for-stat736.html#cb177-1" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install(&quot;AnnotationHub&quot;)</span></span>
<span id="cb177-2"><a href="rna-seq-for-stat736.html#cb177-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationHub)</span>
<span id="cb177-3"><a href="rna-seq-for-stat736.html#cb177-3" aria-hidden="true" tabindex="-1"></a>hub <span class="ot">=</span> <span class="fu">AnnotationHub</span>()</span>
<span id="cb177-4"><a href="rna-seq-for-stat736.html#cb177-4" aria-hidden="true" tabindex="-1"></a><span class="fu">query</span>(hub, <span class="fu">c</span>(<span class="st">&quot;Glycine&quot;</span>, <span class="st">&quot;OrgDb&quot;</span>))</span>
<span id="cb177-5"><a href="rna-seq-for-stat736.html#cb177-5" aria-hidden="true" tabindex="-1"></a>Gm.orgdb <span class="ot">=</span> hub[[<span class="st">&quot;AH72142&quot;</span>]]</span>
<span id="cb177-6"><a href="rna-seq-for-stat736.html#cb177-6" aria-hidden="true" tabindex="-1"></a>Gm.orgdb</span></code></pre></div>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="rna-seq-for-stat736.html#cb178-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationDbi)</span>
<span id="cb178-2"><a href="rna-seq-for-stat736.html#cb178-2" aria-hidden="true" tabindex="-1"></a><span class="fu">keytypes</span>(Mt.orgdb)</span>
<span id="cb178-3"><a href="rna-seq-for-stat736.html#cb178-3" aria-hidden="true" tabindex="-1"></a><span class="co">#key.Gm = keys(Mt.orgdb, keytype=&quot;GENENAME&quot;)</span></span>
<span id="cb178-4"><a href="rna-seq-for-stat736.html#cb178-4" aria-hidden="true" tabindex="-1"></a><span class="co">#head(key.Gm)</span></span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="final-words.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Bioinformatics_RNA-seq.pdf", "Bioinformatics_RNA-seq.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
