<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 RNA-Seq-MKdata | Bioinformatics_RNA-seq</title>
  <meta name="description" content="RNA-seq notes and trials" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 RNA-Seq-MKdata | Bioinformatics_RNA-seq" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="RNA-seq notes and trials" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 RNA-Seq-MKdata | Bioinformatics_RNA-seq" />
  
  <meta name="twitter:description" content="RNA-seq notes and trials" />
  

<meta name="author" content="J. Marcelo Rosales R." />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"/>
<link rel="next" href="r-code-modules-to-be-used-for-stand-alone-analysis-.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to RNA sequencing (RNA_seq)</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#downloading-and-setting-up-conda-environments."><i class="fa fa-check"></i><b>1.1</b> Downloading and setting up conda environments.</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#install-miniconda-or-anaconda."><i class="fa fa-check"></i><b>1.2</b> Install Miniconda, or Anaconda.</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#conda-terminal-commands."><i class="fa fa-check"></i><b>1.2.1</b> Conda terminal commands.</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#anacondaminiconda-install-summary-of-steps."><i class="fa fa-check"></i><b>1.2.2</b> Anaconda/Miniconda install Summary of Steps.</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#conda-activate-error"><i class="fa fa-check"></i><b>1.2.3</b> Conda activate error</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#alignment-procedures-and-files."><i class="fa fa-check"></i><b>1.3</b> Alignment Procedures and files.</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#using-tophat-early-version"><i class="fa fa-check"></i><b>1.3.1</b> Using Tophat (early version)</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#using-tophat"><i class="fa fa-check"></i><b>1.3.2</b> Using Tophat</a></li>
<li class="chapter" data-level="1.3.3" data-path="index.html"><a href="index.html#using-star"><i class="fa fa-check"></i><b>1.3.3</b> Using STAR</a></li>
<li class="chapter" data-level="1.3.4" data-path="index.html"><a href="index.html#Mus-musculus-NCBI-build37-2"><i class="fa fa-check"></i><b>1.3.4</b> Custom/personal way to Download the Mus_musculus_NCBI_build37.2.tar.gz file.</a></li>
<li class="chapter" data-level="1.3.5" data-path="index.html"><a href="index.html#get-the-sample-files."><i class="fa fa-check"></i><b>1.3.5</b> Get the Sample files.</a></li>
<li class="chapter" data-level="1.3.6" data-path="index.html"><a href="index.html#starting-to-analize-fastq.gz-files"><i class="fa fa-check"></i><b>1.3.6</b> Starting to analize fastq.gz files</a></li>
<li class="chapter" data-level="1.3.7" data-path="index.html"><a href="index.html#trimming-with-trimmomatic"><i class="fa fa-check"></i><b>1.3.7</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="1.3.8" data-path="index.html"><a href="index.html#using-star-to-map-genes"><i class="fa fa-check"></i><b>1.3.8</b> Using STAR to map genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html"><i class="fa fa-check"></i><b>2</b> File Download (or SRA), QC, and Trimming</a>
<ul>
<li class="chapter" data-level="2.1" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#trimmomatic"><i class="fa fa-check"></i><b>2.1</b> Trimmomatic</a></li>
<li class="chapter" data-level="2.2" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#indexing-with-star"><i class="fa fa-check"></i><b>2.2</b> Indexing with STAR</a></li>
<li class="chapter" data-level="2.3" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#microbial-contamination"><i class="fa fa-check"></i><b>2.3</b> Microbial contamination</a></li>
<li class="chapter" data-level="2.4" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#feature-counts."><i class="fa fa-check"></i><b>2.4</b> Feature counts.</a></li>
<li class="chapter" data-level="2.5" data-path="file-download-or-sra-qc-and-trimming.html"><a href="file-download-or-sra-qc-and-trimming.html#setting-r-environment."><i class="fa fa-check"></i><b>2.5</b> Setting R environment.</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>3</b> Methods</a></li>
<li class="chapter" data-level="4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><i class="fa fa-check"></i><b>4</b> RNA Seq Differentially Expressed, GO Enrichment, and Pathway analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#background"><i class="fa fa-check"></i><b>4.1</b> <strong>Background</strong></a></li>
<li class="chapter" data-level="4.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#cleaning-data"><i class="fa fa-check"></i><b>4.2</b> <strong>Cleaning Data</strong></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#programs-used"><i class="fa fa-check"></i><b>4.2.1</b> Programs used?</a></li>
<li class="chapter" data-level="4.2.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#acquiring-sequences"><i class="fa fa-check"></i><b>4.2.2</b> Acquiring sequences</a></li>
<li class="chapter" data-level="4.2.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#fastqc"><i class="fa fa-check"></i><b>4.2.3</b> FastQC</a></li>
<li class="chapter" data-level="4.2.4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#trimming-with-trimmomatic-1"><i class="fa fa-check"></i><b>4.2.4</b> Trimming with Trimmomatic</a></li>
<li class="chapter" data-level="4.2.5" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#alignment"><i class="fa fa-check"></i><b>4.2.5</b> Alignment</a></li>
<li class="chapter" data-level="4.2.6" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#assembling-transcripts-with-cufflinks"><i class="fa fa-check"></i><b>4.2.6</b> Assembling transcripts with Cufflinks</a></li>
<li class="chapter" data-level="4.2.7" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#checking-for-contamination"><i class="fa fa-check"></i><b>4.2.7</b> Checking for Contamination</a></li>
<li class="chapter" data-level="4.2.8" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#counting-transcripts"><i class="fa fa-check"></i><b>4.2.8</b> Counting Transcripts</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#differentially-expressed-sequence-identification"><i class="fa fa-check"></i><b>4.3</b> <strong>Differentially Expressed Sequence Identification</strong></a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#analyzing-reads-counts"><i class="fa fa-check"></i><b>4.3.1</b> Analyzing Reads Counts</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#interactions-cause-a-difference-between-the-lfc-betwen-pooled-data-e.g.-p53-control-and-ir-and-p53---control-and-ir"><i class="fa fa-check"></i><b>4.4</b> Interactions cause a difference between the lfc betwen pooled data, e.g. p53+/+ (control and IR) and p53-/- (control and IR)</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>4.4.1</b> Exploratory Data Analysis</a></li>
<li class="chapter" data-level="4.4.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#identification-of-differentially-expressed-genes"><i class="fa fa-check"></i><b>4.4.2</b> Identification of Differentially Expressed Genes</a></li>
<li class="chapter" data-level="4.4.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#gene-annotations"><i class="fa fa-check"></i><b>4.4.3</b> Gene Annotations</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#go-enrichment-analysis-using-gostats"><i class="fa fa-check"></i><b>4.5</b> 7. GO Enrichment analysis using GOstats</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#go-enrichment-analysis-of-downregulated-genes"><i class="fa fa-check"></i><b>4.5.1</b> GO Enrichment analysis of downregulated genes</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-analysis-using-expression-data"><i class="fa fa-check"></i><b>4.6</b> 8. Pathway analysis using expression data</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#prepare-data"><i class="fa fa-check"></i><b>4.6.1</b> Prepare data</a></li>
<li class="chapter" data-level="4.6.2" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#kegg-pathways"><i class="fa fa-check"></i><b>4.6.2</b> KEGG pathways</a></li>
<li class="chapter" data-level="4.6.3" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-oxidative-phosphorylation."><i class="fa fa-check"></i><b>4.6.3</b> Pathway and regulation of genes for Oxidative phosphorylation.</a></li>
<li class="chapter" data-level="4.6.4" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-glycosylphosphatidylinositolgpi-anchor-biosynthesis."><i class="fa fa-check"></i><b>4.6.4</b> Pathway and regulation of genes for Glycosylphosphatidylinositol(GPI)-anchor biosynthesis.</a></li>
<li class="chapter" data-level="4.6.5" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-rna-polymerase."><i class="fa fa-check"></i><b>4.6.5</b> Pathway and regulation of genes for RNA polymerase.</a></li>
<li class="chapter" data-level="4.6.6" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-nucleotide-excision-repair."><i class="fa fa-check"></i><b>4.6.6</b> Pathway and regulation of genes for Nucleotide excision repair.</a></li>
<li class="chapter" data-level="4.6.7" data-path="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html"><a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html#pathway-and-regulation-of-genes-for-non-homologous-end-joining."><i class="fa fa-check"></i><b>4.6.7</b> Pathway and regulation of genes for Non-homologous end-joining.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html"><i class="fa fa-check"></i><b>5</b> RNA-Seq-MKdata</a>
<ul>
<li class="chapter" data-level="5.1" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#pca"><i class="fa fa-check"></i><b>5.1</b> PCA</a></li>
<li class="chapter" data-level="5.2" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#differential-expression"><i class="fa fa-check"></i><b>5.2</b> Differential expression</a></li>
<li class="chapter" data-level="5.3" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#rna-seq-scienceparksg"><i class="fa fa-check"></i><b>5.3</b> RNA-seq (ScienceparkSG)</a></li>
<li class="chapter" data-level="5.4" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#false-discovery-rates-fdr-benjamini-hocheberg-method"><i class="fa fa-check"></i><b>5.4</b> False discovery rates FDR / Benjamini-Hocheberg method</a></li>
<li class="chapter" data-level="5.5" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#volcano-plot"><i class="fa fa-check"></i><b>5.5</b> Volcano plot</a></li>
<li class="chapter" data-level="5.6" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#heatmap-normalization"><i class="fa fa-check"></i><b>5.6</b> Heatmap &amp; normalization</a></li>
<li class="chapter" data-level="5.7" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#ma-plots"><i class="fa fa-check"></i><b>5.7</b> MA plots</a></li>
<li class="chapter" data-level="5.8" data-path="rna-seq-mkdata.html"><a href="rna-seq-mkdata.html#eastern-new-mexico-university-enmu-method"><i class="fa fa-check"></i><b>5.8</b> Eastern New Mexico University ENMU method</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="r-code-modules-to-be-used-for-stand-alone-analysis-.html"><a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html"><i class="fa fa-check"></i><b>6</b> R code modules to be used for stand-alone analysis.</a></li>
<li class="chapter" data-level="7" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html"><i class="fa fa-check"></i><b>7</b> WGCNA RNA-seq Blog</a>
<ul>
<li class="chapter" data-level="7.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#WGCNA"><i class="fa fa-check"></i><b>7.1</b> Introduction to the WGCNA Rpackage</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-rpackage-installation-procedure."><i class="fa fa-check"></i><b>7.1.1</b> WGCNA Rpackage Installation procedure.</a></li>
<li class="chapter" data-level="7.1.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#final-installation-workflow"><i class="fa fa-check"></i><b>7.1.2</b> Final installation workflow:</a></li>
<li class="chapter" data-level="7.1.3" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#wgcna-tutorial."><i class="fa fa-check"></i><b>7.1.3</b> WGCNA tutorial.</a></li>
<li class="chapter" data-level="7.1.4" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#gene-expression-analysis"><i class="fa fa-check"></i><b>7.1.4</b> 3.4 Gene expression analysis</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#rna-seq-analisys-with-kakus-data-sample."><i class="fa fa-check"></i><b>7.2</b> RNA-seq Analisys with Kaku’s Data sample.</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="wgcna-rna-seq-blog.html"><a href="wgcna-rna-seq-blog.html#data-analysis."><i class="fa fa-check"></i><b>7.2.1</b> Data Analysis.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="youtube-references-.html"><a href="youtube-references-.html"><i class="fa fa-check"></i><b>8</b> Youtube references.</a>
<ul>
<li class="chapter" data-level="8.1" data-path="youtube-references-.html"><a href="youtube-references-.html#anaconda-set-up-robust-data-science-environment-with-miniconda-and-conda-forge."><i class="fa fa-check"></i><b>8.1</b> Anaconda set up, Robust Data Science Environment with Miniconda and Conda-Forge.</a></li>
<li class="chapter" data-level="8.2" data-path="youtube-references-.html"><a href="youtube-references-.html#quality-control-preprocessing-of-raw-reads"><i class="fa fa-check"></i><b>8.2</b> Quality control &amp; preprocessing of raw reads</a></li>
<li class="chapter" data-level="8.3" data-path="youtube-references-.html"><a href="youtube-references-.html#trimmomatic."><i class="fa fa-check"></i><b>8.3</b> Trimmomatic.</a></li>
<li class="chapter" data-level="8.4" data-path="youtube-references-.html"><a href="youtube-references-.html#building-genome-index-and-aligning-with-star"><i class="fa fa-check"></i><b>8.4</b> Building Genome Index and Aligning with STAR</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="external-ref.html"><a href="external-ref.html"><i class="fa fa-check"></i><b>9</b> External Ref</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics_RNA-seq</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="rna-seq-mkdata" class="section level1" number="5">
<h1><span class="header-section-number">Chapter 5</span> RNA-Seq-MKdata</h1>
<div id="pca" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> PCA</h2>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="rna-seq-mkdata.html#cb174-1" aria-hidden="true" tabindex="-1"></a><span class="co"># PCA of Raw count file</span></span>
<span id="cb174-2"><a href="rna-seq-mkdata.html#cb174-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the data file (count) --------</span></span>
<span id="cb174-3"><a href="rna-seq-mkdata.html#cb174-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-4"><a href="rna-seq-mkdata.html#cb174-4" aria-hidden="true" tabindex="-1"></a><span class="co"># all_tpmCSV &lt;- read.csv (&quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/all.tpm02.csv&quot;, header = T, row.names=1)</span></span>
<span id="cb174-5"><a href="rna-seq-mkdata.html#cb174-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb174-6"><a href="rna-seq-mkdata.html#cb174-6" aria-hidden="true" tabindex="-1"></a>all_raw_count <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">&quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/all.raw.count.xlsx&quot;</span>)</span>
<span id="cb174-7"><a href="rna-seq-mkdata.html#cb174-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-8"><a href="rna-seq-mkdata.html#cb174-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(all_raw_count)</span>
<span id="cb174-9"><a href="rna-seq-mkdata.html#cb174-9" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(all_raw_count)</span>
<span id="cb174-10"><a href="rna-seq-mkdata.html#cb174-10" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(all_raw_count)</span>
<span id="cb174-11"><a href="rna-seq-mkdata.html#cb174-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-12"><a href="rna-seq-mkdata.html#cb174-12" aria-hidden="true" tabindex="-1"></a>all_raw_count00 <span class="ot">&lt;-</span> all_raw_count <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">&quot;GeneID&quot;</span>) </span>
<span id="cb174-13"><a href="rna-seq-mkdata.html#cb174-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-14"><a href="rna-seq-mkdata.html#cb174-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-15"><a href="rna-seq-mkdata.html#cb174-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract column names.</span></span>
<span id="cb174-16"><a href="rna-seq-mkdata.html#cb174-16" aria-hidden="true" tabindex="-1"></a>colnames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(all_raw_count)</span>
<span id="cb174-17"><a href="rna-seq-mkdata.html#cb174-17" aria-hidden="true" tabindex="-1"></a>colnames</span>
<span id="cb174-18"><a href="rna-seq-mkdata.html#cb174-18" aria-hidden="true" tabindex="-1"></a><span class="co"># “r convert list to comma separated string”</span></span>
<span id="cb174-19"><a href="rna-seq-mkdata.html#cb174-19" aria-hidden="true" tabindex="-1"></a><span class="fu">paste</span>(<span class="fu">shQuote</span>(colnames), <span class="at">collapse=</span><span class="st">&quot;, &quot;</span>) <span class="co"># with quotes.</span></span>
<span id="cb174-20"><a href="rna-seq-mkdata.html#cb174-20" aria-hidden="true" tabindex="-1"></a><span class="fu">paste</span>(colnames, <span class="at">collapse =</span> <span class="st">&quot;,&quot;</span>) <span class="co"># without quotes. </span></span>
<span id="cb174-21"><a href="rna-seq-mkdata.html#cb174-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-22"><a href="rna-seq-mkdata.html#cb174-22" aria-hidden="true" tabindex="-1"></a>excluded_vars <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;GeneName&quot;</span>, <span class="st">&quot;Chr&quot;</span>, <span class="st">&quot;Start&quot;</span>, <span class="st">&quot;End&quot;</span>, <span class="st">&quot;Strand&quot;</span>, <span class="st">&quot;Length&quot;</span> )</span>
<span id="cb174-23"><a href="rna-seq-mkdata.html#cb174-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-24"><a href="rna-seq-mkdata.html#cb174-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr) <span class="co"># for select function</span></span>
<span id="cb174-25"><a href="rna-seq-mkdata.html#cb174-25" aria-hidden="true" tabindex="-1"></a>all_raw_count01 <span class="ot">&lt;-</span> <span class="fu">select</span>(all_raw_count, <span class="sc">-</span>excluded_vars) <span class="co"># it works, but... better use:</span></span>
<span id="cb174-26"><a href="rna-seq-mkdata.html#cb174-26" aria-hidden="true" tabindex="-1"></a>all_raw_count01 <span class="ot">&lt;-</span> <span class="fu">select</span>(all_raw_count, <span class="sc">-</span><span class="fu">all_of</span>(excluded_vars)) <span class="co">#exclude all variables in excluded_vars</span></span>
<span id="cb174-27"><a href="rna-seq-mkdata.html#cb174-27" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(all_raw_count01)</span>
<span id="cb174-28"><a href="rna-seq-mkdata.html#cb174-28" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(all_raw_count01)</span>
<span id="cb174-29"><a href="rna-seq-mkdata.html#cb174-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-30"><a href="rna-seq-mkdata.html#cb174-30" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(all_raw_count01)</span>
<span id="cb174-31"><a href="rna-seq-mkdata.html#cb174-31" aria-hidden="true" tabindex="-1"></a><span class="co"># data.frame</span></span>
<span id="cb174-32"><a href="rna-seq-mkdata.html#cb174-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-33"><a href="rna-seq-mkdata.html#cb174-33" aria-hidden="true" tabindex="-1"></a><span class="co"># df preparation --------</span></span>
<span id="cb174-34"><a href="rna-seq-mkdata.html#cb174-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert values of a column into row names in a df. Source: https://stackoverflow.com/questions/5555408/convert-the-values-in-a-column-into-row-names-in-an-existing-data-frame</span></span>
<span id="cb174-35"><a href="rna-seq-mkdata.html#cb174-35" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb174-36"><a href="rna-seq-mkdata.html#cb174-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-37"><a href="rna-seq-mkdata.html#cb174-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Method01 </span></span>
<span id="cb174-38"><a href="rna-seq-mkdata.html#cb174-38" aria-hidden="true" tabindex="-1"></a><span class="co"># rownames(all_raw_count01) &lt;- all_raw_count01[,1] # Converting first column into row names.* Column will still remain. </span></span>
<span id="cb174-39"><a href="rna-seq-mkdata.html#cb174-39" aria-hidden="true" tabindex="-1"></a><span class="co"># all_raw_count01[,1] &lt;- NULL # Erasing 1st column. </span></span>
<span id="cb174-40"><a href="rna-seq-mkdata.html#cb174-40" aria-hidden="true" tabindex="-1"></a><span class="co"># all_raw_count01</span></span>
<span id="cb174-41"><a href="rna-seq-mkdata.html#cb174-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-42"><a href="rna-seq-mkdata.html#cb174-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Method03: #Worked!!</span></span>
<span id="cb174-43"><a href="rna-seq-mkdata.html#cb174-43" aria-hidden="true" tabindex="-1"></a>all_raw_count01 <span class="ot">&lt;-</span> all_raw_count01 <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">&quot;GeneID&quot;</span>) </span>
<span id="cb174-44"><a href="rna-seq-mkdata.html#cb174-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Source: https://www.geeksforgeeks.org/convert-values-in-column-into-row-names-of-dataframe-in-r/</span></span>
<span id="cb174-45"><a href="rna-seq-mkdata.html#cb174-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-46"><a href="rna-seq-mkdata.html#cb174-46" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(all_raw_count01 [,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span>
<span id="cb174-47"><a href="rna-seq-mkdata.html#cb174-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-48"><a href="rna-seq-mkdata.html#cb174-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Quick exploration of the data ----------</span></span>
<span id="cb174-49"><a href="rna-seq-mkdata.html#cb174-49" aria-hidden="true" tabindex="-1"></a>all_raw_count01 <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(all_raw_count01) <span class="co"># delete all rows with NA</span></span>
<span id="cb174-50"><a href="rna-seq-mkdata.html#cb174-50" aria-hidden="true" tabindex="-1"></a><span class="co"># all_raw_count01_matrix &lt;- as.matrix(all_raw_count01) #convert df into matrix. </span></span>
<span id="cb174-51"><a href="rna-seq-mkdata.html#cb174-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-52"><a href="rna-seq-mkdata.html#cb174-52" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(all_raw_count01, <span class="at">las=</span> <span class="dv">2</span>, <span class="at">main =</span> <span class="st">&quot;Count raw 47&quot;</span>)  <span class="co"># To many outlier and variation, better work with the log scale. </span></span>
<span id="cb174-53"><a href="rna-seq-mkdata.html#cb174-53" aria-hidden="true" tabindex="-1"></a>all_raw_count01_log <span class="ot">&lt;-</span> <span class="fu">log</span>(all_raw_count01<span class="sc">+</span><span class="dv">1</span>) <span class="co">#here log +1 because the log of 0 = infinity and we will not be able to see the data.</span></span>
<span id="cb174-54"><a href="rna-seq-mkdata.html#cb174-54" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(all_raw_count01_log, <span class="at">las =</span> <span class="dv">2</span>, <span class="at">col =</span> <span class="st">&quot;lightblue&quot;</span>, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">15</span>), <span class="at">main =</span> <span class="st">&quot;Count raw log 49&quot;</span>) <span class="co"># las =2 to make the x labels vertically aligned.</span></span>
<span id="cb174-55"><a href="rna-seq-mkdata.html#cb174-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-56"><a href="rna-seq-mkdata.html#cb174-56" aria-hidden="true" tabindex="-1"></a><span class="co"># Histograms of especific samples.</span></span>
<span id="cb174-57"><a href="rna-seq-mkdata.html#cb174-57" aria-hidden="true" tabindex="-1"></a>mPDL_RNA14D_Ko1 <span class="ot">&lt;-</span> all_raw_count01_log[,<span class="dv">1</span>]</span>
<span id="cb174-58"><a href="rna-seq-mkdata.html#cb174-58" aria-hidden="true" tabindex="-1"></a>mPDL_RNA14D_Ko2 <span class="ot">&lt;-</span> all_raw_count01_log[,<span class="dv">2</span>]</span>
<span id="cb174-59"><a href="rna-seq-mkdata.html#cb174-59" aria-hidden="true" tabindex="-1"></a>mPDL_RNA14D_WT1 <span class="ot">&lt;-</span> all_raw_count01_log[,<span class="dv">5</span>]</span>
<span id="cb174-60"><a href="rna-seq-mkdata.html#cb174-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-61"><a href="rna-seq-mkdata.html#cb174-61" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(mPDL_RNA14D_Ko1, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>)</span>
<span id="cb174-62"><a href="rna-seq-mkdata.html#cb174-62" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(mPDL_RNA14D_Ko2, <span class="at">col =</span> <span class="st">&quot;yellow&quot;</span>)</span>
<span id="cb174-63"><a href="rna-seq-mkdata.html#cb174-63" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(mPDL_RNA14D_WT1, <span class="at">col =</span> <span class="st">&quot;green&quot;</span>, <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Count raw 14Ko1, 14Ko2, 14Wt1 54&quot;</span>), <span class="at">add=</span>T)</span>
<span id="cb174-64"><a href="rna-seq-mkdata.html#cb174-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-65"><a href="rna-seq-mkdata.html#cb174-65" aria-hidden="true" tabindex="-1"></a><span class="co"># df clean/filter of non expresed genes ------------------</span></span>
<span id="cb174-66"><a href="rna-seq-mkdata.html#cb174-66" aria-hidden="true" tabindex="-1"></a><span class="co"># If Center and Scale needed, remove all rows containing all 0 values.</span></span>
<span id="cb174-67"><a href="rna-seq-mkdata.html#cb174-67" aria-hidden="true" tabindex="-1"></a><span class="co"># df1[rowSums(df1[])&gt;0,]</span></span>
<span id="cb174-68"><a href="rna-seq-mkdata.html#cb174-68" aria-hidden="true" tabindex="-1"></a>all_raw_count02 <span class="ot">&lt;-</span> all_raw_count01[<span class="fu">rowSums</span>(all_raw_count01[])<span class="sc">&gt;</span><span class="dv">0</span>,] </span>
<span id="cb174-69"><a href="rna-seq-mkdata.html#cb174-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-70"><a href="rna-seq-mkdata.html#cb174-70" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(all_raw_count02 [,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span>
<span id="cb174-71"><a href="rna-seq-mkdata.html#cb174-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-72"><a href="rna-seq-mkdata.html#cb174-72" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(all_raw_count02, <span class="at">las=</span> <span class="dv">2</span>, <span class="at">main =</span> <span class="st">&quot;Count raw no 0 :67&quot;</span>)</span>
<span id="cb174-73"><a href="rna-seq-mkdata.html#cb174-73" aria-hidden="true" tabindex="-1"></a>all_raw_count02_log <span class="ot">&lt;-</span> <span class="fu">log</span>(all_raw_count02<span class="sc">+</span><span class="dv">1</span>) <span class="co">#here log +1 because the log of 0 = infinity and we will not be able to see the data.</span></span>
<span id="cb174-74"><a href="rna-seq-mkdata.html#cb174-74" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(all_raw_count02_log, <span class="at">las =</span> <span class="dv">2</span>, <span class="at">col =</span> <span class="st">&quot;green&quot;</span>, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">15</span>), <span class="at">main =</span> <span class="st">&quot;Count raw no 0 log :69&quot;</span>) <span class="co"># las =2 to make the x labels vertically aligned.</span></span>
<span id="cb174-75"><a href="rna-seq-mkdata.html#cb174-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-76"><a href="rna-seq-mkdata.html#cb174-76" aria-hidden="true" tabindex="-1"></a>mPDL_RNA14D_Ko1_02 <span class="ot">&lt;-</span> all_raw_count02_log[,<span class="dv">1</span>]</span>
<span id="cb174-77"><a href="rna-seq-mkdata.html#cb174-77" aria-hidden="true" tabindex="-1"></a>mPDL_RNA14D_Ko2_02 <span class="ot">&lt;-</span> all_raw_count02_log[,<span class="dv">2</span>]</span>
<span id="cb174-78"><a href="rna-seq-mkdata.html#cb174-78" aria-hidden="true" tabindex="-1"></a>mPDL_RNA14D_WT1_02 <span class="ot">&lt;-</span> all_raw_count02_log[,<span class="dv">5</span>]</span>
<span id="cb174-79"><a href="rna-seq-mkdata.html#cb174-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-80"><a href="rna-seq-mkdata.html#cb174-80" aria-hidden="true" tabindex="-1"></a>col1 <span class="ot">&lt;-</span> <span class="fu">rgb</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>, <span class="fl">0.5</span>) <span class="co"># red</span></span>
<span id="cb174-81"><a href="rna-seq-mkdata.html#cb174-81" aria-hidden="true" tabindex="-1"></a>col2 <span class="ot">&lt;-</span> <span class="fu">rgb</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">0</span>, <span class="fl">0.5</span>) <span class="co"># yellow</span></span>
<span id="cb174-82"><a href="rna-seq-mkdata.html#cb174-82" aria-hidden="true" tabindex="-1"></a>col3 <span class="ot">&lt;-</span> <span class="fu">rgb</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>, <span class="fl">0.5</span>) <span class="co"># blue</span></span>
<span id="cb174-83"><a href="rna-seq-mkdata.html#cb174-83" aria-hidden="true" tabindex="-1"></a>col4 <span class="ot">&lt;-</span> <span class="fu">rgb</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">0</span>, <span class="fl">0.5</span>) <span class="co"># green</span></span>
<span id="cb174-84"><a href="rna-seq-mkdata.html#cb174-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-85"><a href="rna-seq-mkdata.html#cb174-85" aria-hidden="true" tabindex="-1"></a><span class="co"># Histograms</span></span>
<span id="cb174-86"><a href="rna-seq-mkdata.html#cb174-86" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(mPDL_RNA14D_Ko1_02, <span class="at">col =</span> col1, <span class="at">breaks =</span> <span class="dv">40</span>)</span>
<span id="cb174-87"><a href="rna-seq-mkdata.html#cb174-87" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(mPDL_RNA14D_Ko2_02, <span class="at">col =</span> col2, <span class="at">breaks =</span> <span class="dv">40</span>)</span>
<span id="cb174-88"><a href="rna-seq-mkdata.html#cb174-88" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(mPDL_RNA14D_WT1_02, <span class="at">col =</span> col3, <span class="at">breaks =</span> <span class="dv">40</span>, <span class="at">add=</span>T)</span>
<span id="cb174-89"><a href="rna-seq-mkdata.html#cb174-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-90"><a href="rna-seq-mkdata.html#cb174-90" aria-hidden="true" tabindex="-1"></a><span class="co"># Density Plots. </span></span>
<span id="cb174-91"><a href="rna-seq-mkdata.html#cb174-91" aria-hidden="true" tabindex="-1"></a>d1 <span class="ot">&lt;-</span> <span class="fu">density</span>(mPDL_RNA14D_Ko1_02)</span>
<span id="cb174-92"><a href="rna-seq-mkdata.html#cb174-92" aria-hidden="true" tabindex="-1"></a>d2 <span class="ot">&lt;-</span> <span class="fu">density</span>(mPDL_RNA14D_Ko2_02)</span>
<span id="cb174-93"><a href="rna-seq-mkdata.html#cb174-93" aria-hidden="true" tabindex="-1"></a>d3 <span class="ot">&lt;-</span> <span class="fu">density</span>(mPDL_RNA14D_WT1_02)</span>
<span id="cb174-94"><a href="rna-seq-mkdata.html#cb174-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-95"><a href="rna-seq-mkdata.html#cb174-95" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(d1, <span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span>
<span id="cb174-96"><a href="rna-seq-mkdata.html#cb174-96" aria-hidden="true" tabindex="-1"></a><span class="fu">polygon</span>(d1, <span class="at">col =</span> col1, <span class="at">border =</span> <span class="st">&quot;red&quot;</span>) <span class="co"># add color and border to d1</span></span>
<span id="cb174-97"><a href="rna-seq-mkdata.html#cb174-97" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(d2, <span class="at">col=</span><span class="st">&quot;yellow&quot;</span>) <span class="co"># add a new line to the density plot created above. </span></span>
<span id="cb174-98"><a href="rna-seq-mkdata.html#cb174-98" aria-hidden="true" tabindex="-1"></a><span class="fu">polygon</span>(d2, <span class="at">col =</span> col2, <span class="at">border =</span> <span class="st">&quot;green&quot;</span>) <span class="co"># add color and border to d2</span></span>
<span id="cb174-99"><a href="rna-seq-mkdata.html#cb174-99" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(d3, <span class="at">col=</span><span class="st">&quot;blue&quot;</span>) <span class="co"># add one more line to the density plot. </span></span>
<span id="cb174-100"><a href="rna-seq-mkdata.html#cb174-100" aria-hidden="true" tabindex="-1"></a><span class="fu">polygon</span>(d3, <span class="at">col =</span> col3, <span class="at">border =</span> <span class="st">&quot;blue&quot;</span>) <span class="co"># add color and border to d2</span></span>
<span id="cb174-101"><a href="rna-seq-mkdata.html#cb174-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-102"><a href="rna-seq-mkdata.html#cb174-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-103"><a href="rna-seq-mkdata.html#cb174-103" aria-hidden="true" tabindex="-1"></a><span class="co"># PCA using prcomp --------------- </span></span>
<span id="cb174-104"><a href="rna-seq-mkdata.html#cb174-104" aria-hidden="true" tabindex="-1"></a><span class="co"># https://youtu.be/0Jp4gsfOLMs</span></span>
<span id="cb174-105"><a href="rna-seq-mkdata.html#cb174-105" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(all_raw_count02 [,<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>])</span>
<span id="cb174-106"><a href="rna-seq-mkdata.html#cb174-106" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(all_raw_count02), <span class="at">center =</span> T, <span class="at">scale. =</span> T) <span class="co"># 1. Scale = false because there are 0 values. # 2. table was transpose t(), because prcomp() expects samples in rows and genes in columns. 3. The data is not centered... does this make a difference?... No... read here: https://stats.stackexchange.com/questions/189822/how-does-centering-make-a-difference-in-pca-for-svd-and-eigen-decomposition. Searrched with: variables should be shifted to be zero centered. </span></span>
<span id="cb174-107"><a href="rna-seq-mkdata.html#cb174-107" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pca<span class="sc">$</span>x[,<span class="dv">1</span>], pca<span class="sc">$</span>x[,<span class="dv">2</span>], <span class="at">main =</span> <span class="st">&quot;Plot PCA C1 v C2 :101&quot;</span>)</span>
<span id="cb174-108"><a href="rna-seq-mkdata.html#cb174-108" aria-hidden="true" tabindex="-1"></a>pca.var <span class="ot">&lt;-</span>  pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> <span class="co"># To see how much variation PC1 accounts for. </span></span>
<span id="cb174-109"><a href="rna-seq-mkdata.html#cb174-109" aria-hidden="true" tabindex="-1"></a>pca.var.per <span class="ot">&lt;-</span> <span class="fu">round</span>(pca.var<span class="sc">/</span><span class="fu">sum</span>(pca.var)<span class="sc">*</span><span class="dv">100</span>,<span class="dv">1</span>) <span class="co"># % of variations that each PC accounts for is more useful than the value, calculate %.</span></span>
<span id="cb174-110"><a href="rna-seq-mkdata.html#cb174-110" aria-hidden="true" tabindex="-1"></a>pca.var.per</span>
<span id="cb174-111"><a href="rna-seq-mkdata.html#cb174-111" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(pca.var.per, <span class="at">main =</span> <span class="st">&quot;PCA Scree Plot :104&quot;</span>, <span class="at">xlab =</span> <span class="st">&quot;Principal component&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;Percent Variation&quot;</span>)</span>
<span id="cb174-112"><a href="rna-seq-mkdata.html#cb174-112" aria-hidden="true" tabindex="-1"></a><span class="co"># PC1 accounts for 73.8% of the variation, PC2 15.6%, PC3 5.4% and so on..</span></span>
<span id="cb174-113"><a href="rna-seq-mkdata.html#cb174-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-114"><a href="rna-seq-mkdata.html#cb174-114" aria-hidden="true" tabindex="-1"></a><span class="co">#***************************************</span></span>
<span id="cb174-115"><a href="rna-seq-mkdata.html#cb174-115" aria-hidden="true" tabindex="-1"></a><span class="co"># prcomp returns  3 Things:</span></span>
<span id="cb174-116"><a href="rna-seq-mkdata.html#cb174-116" aria-hidden="true" tabindex="-1"></a><span class="co"># x &lt;- Contains all the principal components )PCs) for drawing a graph. We use the first 2 columns [,1] and [,2] or PC1 and PC2.In this analysis we have 16 samples, therfore we obtain 16 PCs. PC1 accounts for the most variation in data, PC2 accounts for the second most variation and so on. </span></span>
<span id="cb174-117"><a href="rna-seq-mkdata.html#cb174-117" aria-hidden="true" tabindex="-1"></a><span class="co"># sdev &lt;- Standard deviation shows how much variation exists between exh sample. </span></span>
<span id="cb174-118"><a href="rna-seq-mkdata.html#cb174-118" aria-hidden="true" tabindex="-1"></a><span class="co"># rotation &lt;- prcom calls &quot;loading&quot; scores &quot;rotation&quot;. There are loading scores for each PC.  </span></span>
<span id="cb174-119"><a href="rna-seq-mkdata.html#cb174-119" aria-hidden="true" tabindex="-1"></a><span class="co">#****************************************</span></span>
<span id="cb174-120"><a href="rna-seq-mkdata.html#cb174-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-121"><a href="rna-seq-mkdata.html#cb174-121" aria-hidden="true" tabindex="-1"></a><span class="co"># In ggplot</span></span>
<span id="cb174-122"><a href="rna-seq-mkdata.html#cb174-122" aria-hidden="true" tabindex="-1"></a>  <span class="co"># construct the df for the ggplot </span></span>
<span id="cb174-123"><a href="rna-seq-mkdata.html#cb174-123" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb174-124"><a href="rna-seq-mkdata.html#cb174-124" aria-hidden="true" tabindex="-1"></a>pca.data_count <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Sample=</span><span class="fu">rownames</span>(pca<span class="sc">$</span>x), </span>
<span id="cb174-125"><a href="rna-seq-mkdata.html#cb174-125" aria-hidden="true" tabindex="-1"></a>                             <span class="at">X=</span>pca<span class="sc">$</span>x[,<span class="dv">1</span>],</span>
<span id="cb174-126"><a href="rna-seq-mkdata.html#cb174-126" aria-hidden="true" tabindex="-1"></a>                             <span class="at">Y=</span>pca<span class="sc">$</span>x[,<span class="dv">2</span>])</span>
<span id="cb174-127"><a href="rna-seq-mkdata.html#cb174-127" aria-hidden="true" tabindex="-1"></a>pca.data_count</span>
<span id="cb174-128"><a href="rna-seq-mkdata.html#cb174-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-129"><a href="rna-seq-mkdata.html#cb174-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-130"><a href="rna-seq-mkdata.html#cb174-130" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel) <span class="co"># for the repel function. </span></span>
<span id="cb174-131"><a href="rna-seq-mkdata.html#cb174-131" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> pca.data_count, <span class="fu">aes</span>(<span class="at">x=</span>X, <span class="at">y=</span>Y, <span class="at">label=</span>Sample, <span class="at">color=</span>Sample, <span class="at">fill=</span>Sample)) <span class="sc">+</span></span>
<span id="cb174-132"><a href="rna-seq-mkdata.html#cb174-132" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_text(size = 3.5) +</span></span>
<span id="cb174-133"><a href="rna-seq-mkdata.html#cb174-133" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb174-134"><a href="rna-seq-mkdata.html#cb174-134" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">rownames</span>(pca.data_count)), <span class="at">size =</span> <span class="fl">3.5</span>) <span class="sc">+</span></span>
<span id="cb174-135"><a href="rna-seq-mkdata.html#cb174-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste</span>(<span class="st">&quot;PC1 - &quot;</span>, pca.var.per[<span class="dv">1</span>], <span class="st">&quot;%&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)) <span class="sc">+</span></span>
<span id="cb174-136"><a href="rna-seq-mkdata.html#cb174-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste</span>(<span class="st">&quot;PC2 - &quot;</span>, pca.var.per[<span class="dv">2</span>], <span class="st">&quot;%&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)) <span class="sc">+</span></span>
<span id="cb174-137"><a href="rna-seq-mkdata.html#cb174-137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;PCA Graph prcomp center scaled :120&quot;</span>)</span></code></pre></div>
</div>
<div id="differential-expression" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Differential expression</h2>
<p>Top 10 -100 genes</p>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb175-1"><a href="rna-seq-mkdata.html#cb175-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Top 10 genes expresion ------------</span></span>
<span id="cb175-2"><a href="rna-seq-mkdata.html#cb175-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Lets look at how to use loading scores to determine which genes have the largest effect on where samples are plotted in the PCA plot</span></span>
<span id="cb175-3"><a href="rna-seq-mkdata.html#cb175-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-4"><a href="rna-seq-mkdata.html#cb175-4" aria-hidden="true" tabindex="-1"></a><span class="co"># PC1</span></span>
<span id="cb175-5"><a href="rna-seq-mkdata.html#cb175-5" aria-hidden="true" tabindex="-1"></a>loading_scores <span class="ot">&lt;-</span> pca<span class="sc">$</span>rotation[,<span class="dv">1</span>]　<span class="co"># Loading score of PC!</span></span>
<span id="cb175-6"><a href="rna-seq-mkdata.html#cb175-6" aria-hidden="true" tabindex="-1"></a>gene_scores <span class="ot">&lt;-</span> <span class="fu">abs</span>(loading_scores) <span class="co"># abs() to sort based on the numbers of magnitude rather than from high to low. </span></span>
<span id="cb175-7"><a href="rna-seq-mkdata.html#cb175-7" aria-hidden="true" tabindex="-1"></a>gene_scores_ranked <span class="ot">&lt;-</span>  <span class="fu">sort</span>(gene_scores, <span class="at">decreasing =</span> T) <span class="co"># Sort the magnitudes of loading scores from high to low. </span></span>
<span id="cb175-8"><a href="rna-seq-mkdata.html#cb175-8" aria-hidden="true" tabindex="-1"></a>top10_genes <span class="ot">&lt;-</span> <span class="fu">names</span>(gene_scores_ranked[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>]) <span class="co"># Get the names of the top 10 genes with the largest loading scores magnitude. </span></span>
<span id="cb175-9"><a href="rna-seq-mkdata.html#cb175-9" aria-hidden="true" tabindex="-1"></a>top10_genes</span>
<span id="cb175-10"><a href="rna-seq-mkdata.html#cb175-10" aria-hidden="true" tabindex="-1"></a>pca<span class="sc">$</span>rotation[top10_genes,<span class="dv">1</span>] <span class="co"># show the scores (and +/-)</span></span>
<span id="cb175-11"><a href="rna-seq-mkdata.html#cb175-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-12"><a href="rna-seq-mkdata.html#cb175-12" aria-hidden="true" tabindex="-1"></a>PC1top10.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC1=</span> pca<span class="sc">$</span>rotation[top10_genes,<span class="dv">1</span>],</span>
<span id="cb175-13"><a href="rna-seq-mkdata.html#cb175-13" aria-hidden="true" tabindex="-1"></a>                       <span class="at">PC2=</span> pca<span class="sc">$</span>rotation[top10_genes,<span class="dv">2</span>])  <span class="co"># Careful This is not top 10 of PC2</span></span>
<span id="cb175-14"><a href="rna-seq-mkdata.html#cb175-14" aria-hidden="true" tabindex="-1"></a>PC1top10.df <span class="co"># Remove PC2</span></span>
<span id="cb175-15"><a href="rna-seq-mkdata.html#cb175-15" aria-hidden="true" tabindex="-1"></a>PC1top10.df[,<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb175-16"><a href="rna-seq-mkdata.html#cb175-16" aria-hidden="true" tabindex="-1"></a>PC1top10.df</span>
<span id="cb175-17"><a href="rna-seq-mkdata.html#cb175-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-18"><a href="rna-seq-mkdata.html#cb175-18" aria-hidden="true" tabindex="-1"></a><span class="co">#PC2</span></span>
<span id="cb175-19"><a href="rna-seq-mkdata.html#cb175-19" aria-hidden="true" tabindex="-1"></a>loading_scores <span class="ot">&lt;-</span> pca<span class="sc">$</span>rotation[,<span class="dv">2</span>]　<span class="co"># Loading score of PC</span></span>
<span id="cb175-20"><a href="rna-seq-mkdata.html#cb175-20" aria-hidden="true" tabindex="-1"></a>gene_scores <span class="ot">&lt;-</span> <span class="fu">abs</span>(loading_scores) <span class="co"># abs() to sort based on the numbers of magnitude rather than from high to low. </span></span>
<span id="cb175-21"><a href="rna-seq-mkdata.html#cb175-21" aria-hidden="true" tabindex="-1"></a>gene_scores_ranked <span class="ot">&lt;-</span>  <span class="fu">sort</span>(gene_scores, <span class="at">decreasing =</span> T) <span class="co"># Sort the magnitudes of loading scores from high to low. </span></span>
<span id="cb175-22"><a href="rna-seq-mkdata.html#cb175-22" aria-hidden="true" tabindex="-1"></a>top10_genes <span class="ot">&lt;-</span> <span class="fu">names</span>(gene_scores_ranked[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>]) <span class="co"># Get the names of the top 10 genes with the largest loading scores magnitude. </span></span>
<span id="cb175-23"><a href="rna-seq-mkdata.html#cb175-23" aria-hidden="true" tabindex="-1"></a>top10_genes <span class="co"># **I am saving using the same name as the others, so this will change for each  component. </span></span>
<span id="cb175-24"><a href="rna-seq-mkdata.html#cb175-24" aria-hidden="true" tabindex="-1"></a>pca<span class="sc">$</span>rotation[top10_genes,<span class="dv">2</span>] <span class="co"># show the scores (and +/-)</span></span>
<span id="cb175-25"><a href="rna-seq-mkdata.html#cb175-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-26"><a href="rna-seq-mkdata.html#cb175-26" aria-hidden="true" tabindex="-1"></a>PC2top10.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC2=</span> pca<span class="sc">$</span>rotation[top10_genes,<span class="dv">2</span>])</span>
<span id="cb175-27"><a href="rna-seq-mkdata.html#cb175-27" aria-hidden="true" tabindex="-1"></a>PC2top10.df</span>
<span id="cb175-28"><a href="rna-seq-mkdata.html#cb175-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-29"><a href="rna-seq-mkdata.html#cb175-29" aria-hidden="true" tabindex="-1"></a>Top10DEgenes_prcomp <span class="ot">&lt;-</span> <span class="fu">merge</span>(PC1top10.df, PC2top10.df, <span class="at">by=</span><span class="dv">0</span>, <span class="at">all=</span>T)</span>
<span id="cb175-30"><a href="rna-seq-mkdata.html#cb175-30" aria-hidden="true" tabindex="-1"></a>Top10DEgenes_prcomp</span>
<span id="cb175-31"><a href="rna-seq-mkdata.html#cb175-31" aria-hidden="true" tabindex="-1"></a>Top100DEgenes_prcomp <span class="ot">&lt;-</span> <span class="fu">merge</span>(PC1top10.df, PC2top10.df, <span class="at">by=</span><span class="dv">0</span>, <span class="at">all=</span>T) <span class="co"># Change code above from 10  to 100.</span></span>
<span id="cb175-32"><a href="rna-seq-mkdata.html#cb175-32" aria-hidden="true" tabindex="-1"></a>Top100DEgenes_prcomp</span>
<span id="cb175-33"><a href="rna-seq-mkdata.html#cb175-33" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(TopDEgenes)</span>
<span id="cb175-34"><a href="rna-seq-mkdata.html#cb175-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-35"><a href="rna-seq-mkdata.html#cb175-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-36"><a href="rna-seq-mkdata.html#cb175-36" aria-hidden="true" tabindex="-1"></a><span class="co"># Save DE genes table in a file. --------</span></span>
<span id="cb175-37"><a href="rna-seq-mkdata.html#cb175-37" aria-hidden="true" tabindex="-1"></a><span class="co"># .csv file. </span></span>
<span id="cb175-38"><a href="rna-seq-mkdata.html#cb175-38" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Top10DEgenes_prcomp, <span class="at">file =</span>  <span class="st">&quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/TopDEgenes_prcomp.csv&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>, <span class="at">row.names =</span> F)</span>
<span id="cb175-39"><a href="rna-seq-mkdata.html#cb175-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-40"><a href="rna-seq-mkdata.html#cb175-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-41"><a href="rna-seq-mkdata.html#cb175-41" aria-hidden="true" tabindex="-1"></a><span class="co">#Excel file. Source: http://www.sthda.com/english/wiki/writing-data-from-r-to-excel-files-xls-xlsx</span></span>
<span id="cb175-42"><a href="rna-seq-mkdata.html#cb175-42" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;xlsx&quot;</span>)</span>
<span id="cb175-43"><a href="rna-seq-mkdata.html#cb175-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Write the first data set in a new workbook</span></span>
<span id="cb175-44"><a href="rna-seq-mkdata.html#cb175-44" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(Top10DEgenes_prcomp, <span class="at">file =</span> <span class="st">&quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/Top10DEgenes_prcomp.xlsx&quot;</span>,</span>
<span id="cb175-45"><a href="rna-seq-mkdata.html#cb175-45" aria-hidden="true" tabindex="-1"></a>           <span class="at">sheetName =</span> <span class="st">&quot;top10&quot;</span>, <span class="at">row.names =</span> F, <span class="at">append =</span> <span class="cn">FALSE</span>)</span>
<span id="cb175-46"><a href="rna-seq-mkdata.html#cb175-46" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a second data set in a new worksheet</span></span>
<span id="cb175-47"><a href="rna-seq-mkdata.html#cb175-47" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(Top100DEgenes_prcomp, <span class="at">file =</span> <span class="st">&quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/Top10DEgenes_prcomp.xlsx&quot;</span>,</span>
<span id="cb175-48"><a href="rna-seq-mkdata.html#cb175-48" aria-hidden="true" tabindex="-1"></a>           <span class="at">sheetName =</span> <span class="st">&quot;top100&quot;</span>, <span class="at">row.names =</span> F, <span class="at">append =</span> <span class="cn">TRUE</span>) <span class="co"># to add as a new sheet.</span></span>
<span id="cb175-49"><a href="rna-seq-mkdata.html#cb175-49" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a third data set.....</span></span>
<span id="cb175-50"><a href="rna-seq-mkdata.html#cb175-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-51"><a href="rna-seq-mkdata.html#cb175-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-52"><a href="rna-seq-mkdata.html#cb175-52" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;writexl&quot;)</span></span>
<span id="cb175-53"><a href="rna-seq-mkdata.html#cb175-53" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;writexl&quot;</span>)</span>
<span id="cb175-54"><a href="rna-seq-mkdata.html#cb175-54" aria-hidden="true" tabindex="-1"></a><span class="fu">write_xlsx</span>(TopDEgenes_prcomp, <span class="st">&quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/all_raw_count01.xlsx&quot;</span>, <span class="at">sheet=</span> top10_genes)</span></code></pre></div>
</div>
<div id="rna-seq-scienceparksg" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> RNA-seq (ScienceparkSG)</h2>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="rna-seq-mkdata.html#cb176-1" aria-hidden="true" tabindex="-1"></a><span class="co">#  RNA-seq (ScienceparkSG) ****** -----------------</span></span>
<span id="cb176-2"><a href="rna-seq-mkdata.html#cb176-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Metadata table construction -----------------</span></span>
<span id="cb176-3"><a href="rna-seq-mkdata.html#cb176-3" aria-hidden="true" tabindex="-1"></a><span class="co"># experimental metadata., metadata, cold data file construction.</span></span>
<span id="cb176-4"><a href="rna-seq-mkdata.html#cb176-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract column names.</span></span>
<span id="cb176-5"><a href="rna-seq-mkdata.html#cb176-5" aria-hidden="true" tabindex="-1"></a>colnames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)</span>
<span id="cb176-6"><a href="rna-seq-mkdata.html#cb176-6" aria-hidden="true" tabindex="-1"></a><span class="co"># “r convert list to comma separated string”</span></span>
<span id="cb176-7"><a href="rna-seq-mkdata.html#cb176-7" aria-hidden="true" tabindex="-1"></a><span class="fu">paste</span>(<span class="fu">shQuote</span>(df), <span class="at">collapse=</span><span class="st">&quot;, &quot;</span>) <span class="co"># with quotes.</span></span>
<span id="cb176-8"><a href="rna-seq-mkdata.html#cb176-8" aria-hidden="true" tabindex="-1"></a><span class="fu">paste</span>(df, <span class="at">collapse =</span> <span class="st">&quot;,&quot;</span>) <span class="co"># without quotes. </span></span>
<span id="cb176-9"><a href="rna-seq-mkdata.html#cb176-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-10"><a href="rna-seq-mkdata.html#cb176-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Excel </span></span>
<span id="cb176-11"><a href="rna-seq-mkdata.html#cb176-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Copy/paste list to excel &gt; Data &gt; Text to columns &gt; coma separated &gt; Transpose. To delete quotes: Find/replace &#39; to nothing. </span></span>
<span id="cb176-12"><a href="rna-seq-mkdata.html#cb176-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Construct rest of the table in excel (it&#39;s faster) then import to R. </span></span>
<span id="cb176-13"><a href="rna-seq-mkdata.html#cb176-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-14"><a href="rna-seq-mkdata.html#cb176-14" aria-hidden="true" tabindex="-1"></a>all_raw_count02_colnames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(all_raw_count02)</span>
<span id="cb176-15"><a href="rna-seq-mkdata.html#cb176-15" aria-hidden="true" tabindex="-1"></a>all_raw_count02_colnames</span>
<span id="cb176-16"><a href="rna-seq-mkdata.html#cb176-16" aria-hidden="true" tabindex="-1"></a>all_raw_count02_colnames2 <span class="ot">&lt;-</span>  <span class="fu">paste</span>(<span class="fu">shQuote</span>(all_raw_count02_colnames), <span class="at">collapse=</span><span class="st">&quot;, &quot;</span>) <span class="co"># with quotes. </span></span>
<span id="cb176-17"><a href="rna-seq-mkdata.html#cb176-17" aria-hidden="true" tabindex="-1"></a>all_raw_count02_colnames2 <span class="ot">&lt;-</span>  <span class="fu">paste</span>(all_raw_count02_colnames, <span class="at">collapse=</span><span class="st">&quot;, &quot;</span>) <span class="co"># without quotes. </span></span>
<span id="cb176-18"><a href="rna-seq-mkdata.html#cb176-18" aria-hidden="true" tabindex="-1"></a>all_raw_count02_colnames2</span>
<span id="cb176-19"><a href="rna-seq-mkdata.html#cb176-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-20"><a href="rna-seq-mkdata.html#cb176-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb176-21"><a href="rna-seq-mkdata.html#cb176-21" aria-hidden="true" tabindex="-1"></a>Metadata <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">&quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/Notes.xlsx&quot;</span>, </span>
<span id="cb176-22"><a href="rna-seq-mkdata.html#cb176-22" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sheet =</span> <span class="st">&quot;MetaData&quot;</span>)</span>
<span id="cb176-23"><a href="rna-seq-mkdata.html#cb176-23" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Metadata)</span>
<span id="cb176-24"><a href="rna-seq-mkdata.html#cb176-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-25"><a href="rna-seq-mkdata.html#cb176-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Order Metadata as you want to appear in the graphs. </span></span>
<span id="cb176-26"><a href="rna-seq-mkdata.html#cb176-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb176-27"><a href="rna-seq-mkdata.html#cb176-27" aria-hidden="true" tabindex="-1"></a>Metadata2 <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">&quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/Notes.xlsx&quot;</span>, </span>
<span id="cb176-28"><a href="rna-seq-mkdata.html#cb176-28" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sheet =</span> <span class="st">&quot;MetaData2&quot;</span>)</span>
<span id="cb176-29"><a href="rna-seq-mkdata.html#cb176-29" aria-hidden="true" tabindex="-1"></a>Metadata2</span>
<span id="cb176-30"><a href="rna-seq-mkdata.html#cb176-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-31"><a href="rna-seq-mkdata.html#cb176-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-32"><a href="rna-seq-mkdata.html#cb176-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Or construct directly in R. </span></span>
<span id="cb176-33"><a href="rna-seq-mkdata.html#cb176-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert data.frame column to a vector?</span></span>
<span id="cb176-34"><a href="rna-seq-mkdata.html#cb176-34" aria-hidden="true" tabindex="-1"></a>vector <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(df[<span class="st">&#39;colName&#39;</span>]) <span class="co"># Op1 As a Tibble. </span></span>
<span id="cb176-35"><a href="rna-seq-mkdata.html#cb176-35" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(vector) </span>
<span id="cb176-36"><a href="rna-seq-mkdata.html#cb176-36" aria-hidden="true" tabindex="-1"></a>vector2 <span class="ot">&lt;-</span> df[[<span class="st">&#39;colName&#39;</span>]] <span class="co">#Opt2 As a Tibble. </span></span>
<span id="cb176-37"><a href="rna-seq-mkdata.html#cb176-37" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(vector)</span>
<span id="cb176-38"><a href="rna-seq-mkdata.html#cb176-38" aria-hidden="true" tabindex="-1"></a>vector3 <span class="ot">&lt;-</span> df[,<span class="dv">2</span>] <span class="co">#Opt.3 As a Tibble. </span></span>
<span id="cb176-39"><a href="rna-seq-mkdata.html#cb176-39" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(avector)</span>
<span id="cb176-40"><a href="rna-seq-mkdata.html#cb176-40" aria-hidden="true" tabindex="-1"></a>vector4 <span class="ot">&lt;-</span> df<span class="sc">$</span>colName   <span class="co">#Opt.4 as string?...</span></span>
<span id="cb176-41"><a href="rna-seq-mkdata.html#cb176-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-42"><a href="rna-seq-mkdata.html#cb176-42" aria-hidden="true" tabindex="-1"></a>GenotypeVec <span class="ot">&lt;-</span> Metadata[,<span class="dv">2</span>]</span>
<span id="cb176-43"><a href="rna-seq-mkdata.html#cb176-43" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(GenotypeVec) </span>
<span id="cb176-44"><a href="rna-seq-mkdata.html#cb176-44" aria-hidden="true" tabindex="-1"></a>GenotypeVec</span>
<span id="cb176-45"><a href="rna-seq-mkdata.html#cb176-45" aria-hidden="true" tabindex="-1"></a>GenotypeVec <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(Metadata[<span class="st">&#39;Genotype&#39;</span>])</span>
<span id="cb176-46"><a href="rna-seq-mkdata.html#cb176-46" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(GenotypeVec) </span>
<span id="cb176-47"><a href="rna-seq-mkdata.html#cb176-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-48"><a href="rna-seq-mkdata.html#cb176-48" aria-hidden="true" tabindex="-1"></a>GenotypeVec <span class="ot">&lt;-</span> Metadata<span class="sc">$</span>Genotype</span>
<span id="cb176-49"><a href="rna-seq-mkdata.html#cb176-49" aria-hidden="true" tabindex="-1"></a>GenotypeVec</span>
<span id="cb176-50"><a href="rna-seq-mkdata.html#cb176-50" aria-hidden="true" tabindex="-1"></a>GenotypeVec <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">shQuote</span>(Genotype), <span class="at">collapse =</span> <span class="st">&quot;,&quot;</span>) <span class="co"># copy/paste result in c().</span></span>
<span id="cb176-51"><a href="rna-seq-mkdata.html#cb176-51" aria-hidden="true" tabindex="-1"></a>Genotype <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;Ko&#39;</span>,<span class="st">&#39;Ko&#39;</span>,<span class="st">&#39;Ko&#39;</span>,<span class="st">&#39;Ko&#39;</span>,<span class="st">&#39;WT&#39;</span>,<span class="st">&#39;WT&#39;</span>,<span class="st">&#39;WT&#39;</span>,<span class="st">&#39;WT&#39;</span>,<span class="st">&#39;Ko&#39;</span>,<span class="st">&#39;Ko&#39;</span>,<span class="st">&#39;Ko&#39;</span>,<span class="st">&#39;Ko&#39;</span>,<span class="st">&#39;WT&#39;</span>,<span class="st">&#39;WT&#39;</span>,<span class="st">&#39;WT&#39;</span>,<span class="st">&#39;WT&#39;</span>)</span>
<span id="cb176-52"><a href="rna-seq-mkdata.html#cb176-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-53"><a href="rna-seq-mkdata.html#cb176-53" aria-hidden="true" tabindex="-1"></a>TimeVect <span class="ot">&lt;-</span> Metadata<span class="sc">$</span>Time</span>
<span id="cb176-54"><a href="rna-seq-mkdata.html#cb176-54" aria-hidden="true" tabindex="-1"></a>TimeVect</span>
<span id="cb176-55"><a href="rna-seq-mkdata.html#cb176-55" aria-hidden="true" tabindex="-1"></a>TimeVect2 <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">shQuote</span>(TimeVect), <span class="at">collapse =</span> <span class="st">&quot;,&quot;</span>) <span class="co"># copy/paste result in c().</span></span>
<span id="cb176-56"><a href="rna-seq-mkdata.html#cb176-56" aria-hidden="true" tabindex="-1"></a>TimeVect2</span>
<span id="cb176-57"><a href="rna-seq-mkdata.html#cb176-57" aria-hidden="true" tabindex="-1"></a>Time <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;14d&#39;</span>,<span class="st">&#39;14d&#39;</span>,<span class="st">&#39;14d&#39;</span>,<span class="st">&#39;14d&#39;</span>,<span class="st">&#39;14d&#39;</span>,<span class="st">&#39;14d&#39;</span>,<span class="st">&#39;14d&#39;</span>,<span class="st">&#39;14d&#39;</span>,<span class="st">&#39;7d&#39;</span>,<span class="st">&#39;7d&#39;</span>,<span class="st">&#39;7d&#39;</span>,<span class="st">&#39;7d&#39;</span>,<span class="st">&#39;7d&#39;</span>,<span class="st">&#39;7d&#39;</span>,<span class="st">&#39;7d&#39;</span>,<span class="st">&#39;7d&#39;</span>)</span>
<span id="cb176-58"><a href="rna-seq-mkdata.html#cb176-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-59"><a href="rna-seq-mkdata.html#cb176-59" aria-hidden="true" tabindex="-1"></a>TissueVect <span class="ot">&lt;-</span> Metadata<span class="sc">$</span>Tissue</span>
<span id="cb176-60"><a href="rna-seq-mkdata.html#cb176-60" aria-hidden="true" tabindex="-1"></a>TissueVect <span class="co"># Instead of new vector name just save in same vector</span></span>
<span id="cb176-61"><a href="rna-seq-mkdata.html#cb176-61" aria-hidden="true" tabindex="-1"></a>TissueVect <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">shQuote</span>(TissueVect), <span class="at">collapse =</span> <span class="st">&quot;,&quot;</span>) <span class="co"># copy/paste result in c().</span></span>
<span id="cb176-62"><a href="rna-seq-mkdata.html#cb176-62" aria-hidden="true" tabindex="-1"></a>TissueVect</span>
<span id="cb176-63"><a href="rna-seq-mkdata.html#cb176-63" aria-hidden="true" tabindex="-1"></a>Tissue <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>)</span>
<span id="cb176-64"><a href="rna-seq-mkdata.html#cb176-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-65"><a href="rna-seq-mkdata.html#cb176-65" aria-hidden="true" tabindex="-1"></a>ShortNVect <span class="ot">&lt;-</span> Metadata<span class="sc">$</span>ShortName</span>
<span id="cb176-66"><a href="rna-seq-mkdata.html#cb176-66" aria-hidden="true" tabindex="-1"></a>ShortNVect <span class="co"># Instead of new vector name just save in same vector</span></span>
<span id="cb176-67"><a href="rna-seq-mkdata.html#cb176-67" aria-hidden="true" tabindex="-1"></a>ShortNVect <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">shQuote</span>(ShortNVect), <span class="at">collapse =</span> <span class="st">&quot;,&quot;</span>) <span class="co"># copy/paste result in c().</span></span>
<span id="cb176-68"><a href="rna-seq-mkdata.html#cb176-68" aria-hidden="true" tabindex="-1"></a>ShortNVect</span>
<span id="cb176-69"><a href="rna-seq-mkdata.html#cb176-69" aria-hidden="true" tabindex="-1"></a>ShortName <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;14_Ko1&#39;</span>,<span class="st">&#39;14_Ko2&#39;</span>,<span class="st">&#39;14_Ko3&#39;</span>,<span class="st">&#39;14_Ko4&#39;</span>,<span class="st">&#39;14_WT1&#39;</span>,<span class="st">&#39;14_WT2&#39;</span>,<span class="st">&#39;14_WT3&#39;</span>,<span class="st">&#39;14_WT4&#39;</span>,<span class="st">&#39;7_Ko1&#39;</span>,<span class="st">&#39;7_Ko2&#39;</span>,<span class="st">&#39;7_Ko3&#39;</span>,<span class="st">&#39;7_Ko4&#39;</span>,<span class="st">&#39;7_WT1&#39;</span>,<span class="st">&#39;7_WT2&#39;</span>,<span class="st">&#39;7_WT3&#39;</span>,<span class="st">&#39;7_WT4&#39;</span>)</span>
<span id="cb176-70"><a href="rna-seq-mkdata.html#cb176-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-71"><a href="rna-seq-mkdata.html#cb176-71" aria-hidden="true" tabindex="-1"></a>SampNVect <span class="ot">&lt;-</span> Metadata<span class="sc">$</span>SampleName</span>
<span id="cb176-72"><a href="rna-seq-mkdata.html#cb176-72" aria-hidden="true" tabindex="-1"></a>SampNVect <span class="co"># Instead of new vector name just save in same vector</span></span>
<span id="cb176-73"><a href="rna-seq-mkdata.html#cb176-73" aria-hidden="true" tabindex="-1"></a>SampNVect <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">shQuote</span>(SampNVect), <span class="at">collapse =</span> <span class="st">&quot;,&quot;</span>) <span class="co"># copy/paste result in c().</span></span>
<span id="cb176-74"><a href="rna-seq-mkdata.html#cb176-74" aria-hidden="true" tabindex="-1"></a>SampNVect</span>
<span id="cb176-75"><a href="rna-seq-mkdata.html#cb176-75" aria-hidden="true" tabindex="-1"></a>SampName <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;mPDL_RNA14D_Ko1&#39;</span>,<span class="st">&#39;mPDL_RNA14D_Ko2&#39;</span>,<span class="st">&#39;mPDL_RNA14D_Ko3&#39;</span>,<span class="st">&#39;mPDL_RNA14D_Ko4&#39;</span>,<span class="st">&#39;mPDL_RNA14D_WT1&#39;</span>,<span class="st">&#39;mPDL_RNA14D_WT2&#39;</span>,<span class="st">&#39;mPDL_RNA14D_WT3&#39;</span>,<span class="st">&#39;mPDL_RNA14D_WT4&#39;</span>,<span class="st">&#39;mPDL_RNA7D_Ko1&#39;</span>,<span class="st">&#39;mPDL_RNA7D_Ko2&#39;</span>,<span class="st">&#39;mPDL_RNA7D_Ko3&#39;</span>,<span class="st">&#39;mPDL_RNA7D_Ko4&#39;</span>,<span class="st">&#39;mPDL_RNA7D_WT1&#39;</span>,<span class="st">&#39;mPDL_RNA7D_WT2&#39;</span>,<span class="st">&#39;mPDL_RNA7D_WT3&#39;</span>,<span class="st">&#39;mPDL_RNA7D_WT4&#39;</span>)</span>
<span id="cb176-76"><a href="rna-seq-mkdata.html#cb176-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-77"><a href="rna-seq-mkdata.html#cb176-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-78"><a href="rna-seq-mkdata.html#cb176-78" aria-hidden="true" tabindex="-1"></a><span class="co"># Or create the Metadata file manually in R. </span></span>
<span id="cb176-79"><a href="rna-seq-mkdata.html#cb176-79" aria-hidden="true" tabindex="-1"></a><span class="co">#Create vectors containing the data. </span></span>
<span id="cb176-80"><a href="rna-seq-mkdata.html#cb176-80" aria-hidden="true" tabindex="-1"></a>SampName <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;mPDL_RNA14D_Ko1&#39;</span>,<span class="st">&#39;mPDL_RNA14D_Ko2&#39;</span>,<span class="st">&#39;mPDL_RNA14D_Ko3&#39;</span>,<span class="st">&#39;mPDL_RNA14D_Ko4&#39;</span>,<span class="st">&#39;mPDL_RNA14D_WT1&#39;</span>,<span class="st">&#39;mPDL_RNA14D_WT2&#39;</span>,<span class="st">&#39;mPDL_RNA14D_WT3&#39;</span>,<span class="st">&#39;mPDL_RNA14D_WT4&#39;</span>,<span class="st">&#39;mPDL_RNA7D_Ko1&#39;</span>,<span class="st">&#39;mPDL_RNA7D_Ko2&#39;</span>,<span class="st">&#39;mPDL_RNA7D_Ko3&#39;</span>,<span class="st">&#39;mPDL_RNA7D_Ko4&#39;</span>,<span class="st">&#39;mPDL_RNA7D_WT1&#39;</span>,<span class="st">&#39;mPDL_RNA7D_WT2&#39;</span>,<span class="st">&#39;mPDL_RNA7D_WT3&#39;</span>,<span class="st">&#39;mPDL_RNA7D_WT4&#39;</span>) <span class="co"># create Samples list as they appear in the df. </span></span>
<span id="cb176-81"><a href="rna-seq-mkdata.html#cb176-81" aria-hidden="true" tabindex="-1"></a>Genotype <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;Ko&#39;</span>,<span class="st">&#39;Ko&#39;</span>,<span class="st">&#39;Ko&#39;</span>,<span class="st">&#39;Ko&#39;</span>,<span class="st">&#39;WT&#39;</span>,<span class="st">&#39;WT&#39;</span>,<span class="st">&#39;WT&#39;</span>,<span class="st">&#39;WT&#39;</span>,<span class="st">&#39;Ko&#39;</span>,<span class="st">&#39;Ko&#39;</span>,<span class="st">&#39;Ko&#39;</span>,<span class="st">&#39;Ko&#39;</span>,<span class="st">&#39;WT&#39;</span>,<span class="st">&#39;WT&#39;</span>,<span class="st">&#39;WT&#39;</span>,<span class="st">&#39;WT&#39;</span>) <span class="co"># Order according to the samples. </span></span>
<span id="cb176-82"><a href="rna-seq-mkdata.html#cb176-82" aria-hidden="true" tabindex="-1"></a>Time <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;14d&#39;</span>,<span class="st">&#39;14d&#39;</span>,<span class="st">&#39;14d&#39;</span>,<span class="st">&#39;14d&#39;</span>,<span class="st">&#39;14d&#39;</span>,<span class="st">&#39;14d&#39;</span>,<span class="st">&#39;14d&#39;</span>,<span class="st">&#39;14d&#39;</span>,<span class="st">&#39;7d&#39;</span>,<span class="st">&#39;7d&#39;</span>,<span class="st">&#39;7d&#39;</span>,<span class="st">&#39;7d&#39;</span>,<span class="st">&#39;7d&#39;</span>,<span class="st">&#39;7d&#39;</span>,<span class="st">&#39;7d&#39;</span>,<span class="st">&#39;7d&#39;</span>)</span>
<span id="cb176-83"><a href="rna-seq-mkdata.html#cb176-83" aria-hidden="true" tabindex="-1"></a>Tissue <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>,<span class="st">&#39;PDL&#39;</span>)</span>
<span id="cb176-84"><a href="rna-seq-mkdata.html#cb176-84" aria-hidden="true" tabindex="-1"></a>ShortName <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;14_Ko1&#39;</span>,<span class="st">&#39;14_Ko2&#39;</span>,<span class="st">&#39;14_Ko3&#39;</span>,<span class="st">&#39;14_Ko4&#39;</span>,<span class="st">&#39;14_WT1&#39;</span>,<span class="st">&#39;14_WT2&#39;</span>,<span class="st">&#39;14_WT3&#39;</span>,<span class="st">&#39;14_WT4&#39;</span>,<span class="st">&#39;7_Ko1&#39;</span>,<span class="st">&#39;7_Ko2&#39;</span>,<span class="st">&#39;7_Ko3&#39;</span>,<span class="st">&#39;7_Ko4&#39;</span>,<span class="st">&#39;7_WT1&#39;</span>,<span class="st">&#39;7_WT2&#39;</span>,<span class="st">&#39;7_WT3&#39;</span>,<span class="st">&#39;7_WT4&#39;</span>)</span>
<span id="cb176-85"><a href="rna-seq-mkdata.html#cb176-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-86"><a href="rna-seq-mkdata.html#cb176-86" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine vectors into a data frame.</span></span>
<span id="cb176-87"><a href="rna-seq-mkdata.html#cb176-87" aria-hidden="true" tabindex="-1"></a>ColdData <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(Genotype, Time, Tissue, ShortName)</span>
<span id="cb176-88"><a href="rna-seq-mkdata.html#cb176-88" aria-hidden="true" tabindex="-1"></a>ColdData</span>
<span id="cb176-89"><a href="rna-seq-mkdata.html#cb176-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-90"><a href="rna-seq-mkdata.html#cb176-90" aria-hidden="true" tabindex="-1"></a><span class="co">#Create row names with the associated sample names.</span></span>
<span id="cb176-91"><a href="rna-seq-mkdata.html#cb176-91" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ColdData) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;mPDL_RNA14D_Ko1&#39;</span>,<span class="st">&#39;mPDL_RNA14D_Ko2&#39;</span>,<span class="st">&#39;mPDL_RNA14D_Ko3&#39;</span>,<span class="st">&#39;mPDL_RNA14D_Ko4&#39;</span>,<span class="st">&#39;mPDL_RNA14D_WT1&#39;</span>,<span class="st">&#39;mPDL_RNA14D_WT2&#39;</span>,<span class="st">&#39;mPDL_RNA14D_WT3&#39;</span>,<span class="st">&#39;mPDL_RNA14D_WT4&#39;</span>,<span class="st">&#39;mPDL_RNA7D_Ko1&#39;</span>,<span class="st">&#39;mPDL_RNA7D_Ko2&#39;</span>,<span class="st">&#39;mPDL_RNA7D_Ko3&#39;</span>,<span class="st">&#39;mPDL_RNA7D_Ko4&#39;</span>,<span class="st">&#39;mPDL_RNA7D_WT1&#39;</span>,<span class="st">&#39;mPDL_RNA7D_WT2&#39;</span>,<span class="st">&#39;mPDL_RNA7D_WT3&#39;</span>,<span class="st">&#39;mPDL_RNA7D_WT4&#39;</span>)</span>
<span id="cb176-92"><a href="rna-seq-mkdata.html#cb176-92" aria-hidden="true" tabindex="-1"></a>ColdData</span>
<span id="cb176-93"><a href="rna-seq-mkdata.html#cb176-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-94"><a href="rna-seq-mkdata.html#cb176-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-95"><a href="rna-seq-mkdata.html#cb176-95" aria-hidden="true" tabindex="-1"></a><span class="co"># If df includes the Sample names as a column.</span></span>
<span id="cb176-96"><a href="rna-seq-mkdata.html#cb176-96" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine vectors into a data frame.</span></span>
<span id="cb176-97"><a href="rna-seq-mkdata.html#cb176-97" aria-hidden="true" tabindex="-1"></a>ColdData2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(SampName, Genotype, Time, Tissue, ShortName)</span>
<span id="cb176-98"><a href="rna-seq-mkdata.html#cb176-98" aria-hidden="true" tabindex="-1"></a>ColdData2</span>
<span id="cb176-99"><a href="rna-seq-mkdata.html#cb176-99" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert values of a column into row names in a df. Source: https://stackoverflow.com/questions/5555408/convert-the-values-in-a-column-into-row-names-in-an-existing-data-frame</span></span>
<span id="cb176-100"><a href="rna-seq-mkdata.html#cb176-100" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb176-101"><a href="rna-seq-mkdata.html#cb176-101" aria-hidden="true" tabindex="-1"></a><span class="co"># Method01</span></span>
<span id="cb176-102"><a href="rna-seq-mkdata.html#cb176-102" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ColdData2) <span class="ot">&lt;-</span> ColdData2[,<span class="dv">1</span>] <span class="co"># Converting first column into row names.* Column will still remain. </span></span>
<span id="cb176-103"><a href="rna-seq-mkdata.html#cb176-103" aria-hidden="true" tabindex="-1"></a>ColdData2</span>
<span id="cb176-104"><a href="rna-seq-mkdata.html#cb176-104" aria-hidden="true" tabindex="-1"></a>ColdData2[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span> <span class="co"># Erasing 1st column. </span></span>
<span id="cb176-105"><a href="rna-seq-mkdata.html#cb176-105" aria-hidden="true" tabindex="-1"></a>ColdData2</span>
<span id="cb176-106"><a href="rna-seq-mkdata.html#cb176-106" aria-hidden="true" tabindex="-1"></a><span class="co"># Method02</span></span>
<span id="cb176-107"><a href="rna-seq-mkdata.html#cb176-107" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(ColdData2) <span class="ot">&lt;-</span> ColdData2<span class="sc">$</span>SampName</span>
<span id="cb176-108"><a href="rna-seq-mkdata.html#cb176-108" aria-hidden="true" tabindex="-1"></a>ColdData2</span>
<span id="cb176-109"><a href="rna-seq-mkdata.html#cb176-109" aria-hidden="true" tabindex="-1"></a>ColdData2[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb176-110"><a href="rna-seq-mkdata.html#cb176-110" aria-hidden="true" tabindex="-1"></a>ColdData2</span>
<span id="cb176-111"><a href="rna-seq-mkdata.html#cb176-111" aria-hidden="true" tabindex="-1"></a><span class="co"># Method03:</span></span>
<span id="cb176-112"><a href="rna-seq-mkdata.html#cb176-112" aria-hidden="true" tabindex="-1"></a>ColdData2 <span class="ot">&lt;-</span> ColdData2 <span class="sc">%&gt;%</span> remove_rownames <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">&quot;SampName&quot;</span>) </span>
<span id="cb176-113"><a href="rna-seq-mkdata.html#cb176-113" aria-hidden="true" tabindex="-1"></a>ColdData2 <span class="ot">&lt;-</span> ColdData2 <span class="sc">%&gt;%</span>  <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">&quot;SampName&quot;</span>) <span class="co"># no need for remove_rowmanes?</span></span>
<span id="cb176-114"><a href="rna-seq-mkdata.html#cb176-114" aria-hidden="true" tabindex="-1"></a>ColdData2</span>
<span id="cb176-115"><a href="rna-seq-mkdata.html#cb176-115" aria-hidden="true" tabindex="-1"></a><span class="co"># Source: https://www.geeksforgeeks.org/convert-values-in-column-into-row-names-of-dataframe-in-r/</span></span>
<span id="cb176-116"><a href="rna-seq-mkdata.html#cb176-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-117"><a href="rna-seq-mkdata.html#cb176-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-118"><a href="rna-seq-mkdata.html#cb176-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-119"><a href="rna-seq-mkdata.html#cb176-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-120"><a href="rna-seq-mkdata.html#cb176-120" aria-hidden="true" tabindex="-1"></a><span class="co"># PCA applied to RNA-seq data --------</span></span>
<span id="cb176-121"><a href="rna-seq-mkdata.html#cb176-121" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb176-122"><a href="rna-seq-mkdata.html#cb176-122" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(magrittr) <span class="co"># for piper %&gt;%</span></span>
<span id="cb176-123"><a href="rna-seq-mkdata.html#cb176-123" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse) <span class="co"># for tibble (column to rowname), and other packages. </span></span>
<span id="cb176-124"><a href="rna-seq-mkdata.html#cb176-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-125"><a href="rna-seq-mkdata.html#cb176-125" aria-hidden="true" tabindex="-1"></a><span class="co">#from the https://scienceparkstudygroup.github.io/rna-seq-lesson/aio/index.html</span></span>
<span id="cb176-126"><a href="rna-seq-mkdata.html#cb176-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-127"><a href="rna-seq-mkdata.html#cb176-127" aria-hidden="true" tabindex="-1"></a><span class="co"># Imported data</span></span>
<span id="cb176-128"><a href="rna-seq-mkdata.html#cb176-128" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(all_raw_count01 [<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>]) <span class="co"># Raw count df</span></span>
<span id="cb176-129"><a href="rna-seq-mkdata.html#cb176-129" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(all_raw_count02 [<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>]) <span class="co"># df  with no 0s for centering and scaling. </span></span>
<span id="cb176-130"><a href="rna-seq-mkdata.html#cb176-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-131"><a href="rna-seq-mkdata.html#cb176-131" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> (all_raw_count02[,<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>]) <span class="co"># Use this for dds construction (no 0)</span></span>
<span id="cb176-132"><a href="rna-seq-mkdata.html#cb176-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-133"><a href="rna-seq-mkdata.html#cb176-133" aria-hidden="true" tabindex="-1"></a><span class="co"># reorder counts columns according to the experimental design file --------</span></span>
<span id="cb176-134"><a href="rna-seq-mkdata.html#cb176-134" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Metadata2)</span>
<span id="cb176-135"><a href="rna-seq-mkdata.html#cb176-135" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> all_raw_count02[,Metadata2<span class="sc">$</span>SampleName]</span>
<span id="cb176-136"><a href="rna-seq-mkdata.html#cb176-136" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(counts [<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>]) </span>
<span id="cb176-137"><a href="rna-seq-mkdata.html#cb176-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-138"><a href="rna-seq-mkdata.html#cb176-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-139"><a href="rna-seq-mkdata.html#cb176-139" aria-hidden="true" tabindex="-1"></a><span class="co"># first five rows and five columns</span></span>
<span id="cb176-140"><a href="rna-seq-mkdata.html#cb176-140" aria-hidden="true" tabindex="-1"></a>counts[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb176-141"><a href="rna-seq-mkdata.html#cb176-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-142"><a href="rna-seq-mkdata.html#cb176-142" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the dds object that will be used through the rest of this episode.</span></span>
<span id="cb176-143"><a href="rna-seq-mkdata.html#cb176-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-144"><a href="rna-seq-mkdata.html#cb176-144" aria-hidden="true" tabindex="-1"></a><span class="co"># Check that sample names match in both files</span></span>
<span id="cb176-145"><a href="rna-seq-mkdata.html#cb176-145" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(counts) <span class="sc">%in%</span> Metadata2<span class="sc">$</span>SampleName)</span>
<span id="cb176-146"><a href="rna-seq-mkdata.html#cb176-146" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(counts) <span class="sc">==</span> Metadata2<span class="sc">$</span>SampleName)</span>
<span id="cb176-147"><a href="rna-seq-mkdata.html#cb176-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-148"><a href="rna-seq-mkdata.html#cb176-148" aria-hidden="true" tabindex="-1"></a><span class="co"># Differential expression analysis.</span></span>
<span id="cb176-149"><a href="rna-seq-mkdata.html#cb176-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-150"><a href="rna-seq-mkdata.html#cb176-150" aria-hidden="true" tabindex="-1"></a><span class="do">## Creation of the DESeqDataSet object</span></span>
<span id="cb176-151"><a href="rna-seq-mkdata.html#cb176-151" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts, </span>
<span id="cb176-152"><a href="rna-seq-mkdata.html#cb176-152" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> Metadata2, </span>
<span id="cb176-153"><a href="rna-seq-mkdata.html#cb176-153" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> Genotype <span class="sc">+</span> Time)</span>
<span id="cb176-154"><a href="rna-seq-mkdata.html#cb176-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-155"><a href="rna-seq-mkdata.html#cb176-155" aria-hidden="true" tabindex="-1"></a><span class="co"># Variance stabilization ---------</span></span>
<span id="cb176-156"><a href="rna-seq-mkdata.html#cb176-156" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot of mean - sd comparison</span></span>
<span id="cb176-157"><a href="rna-seq-mkdata.html#cb176-157" aria-hidden="true" tabindex="-1"></a><span class="co"># Variance - mean plot for all genes</span></span>
<span id="cb176-158"><a href="rna-seq-mkdata.html#cb176-158" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(counts)</span>
<span id="cb176-159"><a href="rna-seq-mkdata.html#cb176-159" aria-hidden="true" tabindex="-1"></a>p_mean_sd_scaled <span class="ot">&lt;-</span> </span>
<span id="cb176-160"><a href="rna-seq-mkdata.html#cb176-160" aria-hidden="true" tabindex="-1"></a>  counts <span class="sc">%&gt;%</span> </span>
<span id="cb176-161"><a href="rna-seq-mkdata.html#cb176-161" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb176-162"><a href="rna-seq-mkdata.html#cb176-162" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb176-163"><a href="rna-seq-mkdata.html#cb176-163" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span> gene, <span class="at">names_to =</span> <span class="st">&quot;sample&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;counts&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb176-164"><a href="rna-seq-mkdata.html#cb176-164" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> </span>
<span id="cb176-165"><a href="rna-seq-mkdata.html#cb176-165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">gene_average =</span> <span class="fu">mean</span>(counts), <span class="at">gene_stdev =</span> <span class="fu">sd</span>(counts)) <span class="sc">%&gt;%</span> </span>
<span id="cb176-166"><a href="rna-seq-mkdata.html#cb176-166" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb176-167"><a href="rna-seq-mkdata.html#cb176-167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">log10</span>(gene_average), <span class="at">y =</span> <span class="fu">log10</span>(gene_stdev))) <span class="sc">+</span></span>
<span id="cb176-168"><a href="rna-seq-mkdata.html#cb176-168" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">fill =</span> <span class="st">&quot;grey&quot;</span>, <span class="at">colour =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb176-169"><a href="rna-seq-mkdata.html#cb176-169" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Gene count average (log10 scale)&quot;</span>,</span>
<span id="cb176-170"><a href="rna-seq-mkdata.html#cb176-170" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Gene count standard deviation (log10 scale)&quot;</span>) <span class="sc">+</span></span>
<span id="cb176-171"><a href="rna-seq-mkdata.html#cb176-171" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Mean - Standard deviation relationship</span><span class="sc">\n</span><span class="st">(no variance stabilisation)&quot;</span>)</span>
<span id="cb176-172"><a href="rna-seq-mkdata.html#cb176-172" aria-hidden="true" tabindex="-1"></a>p_mean_sd_scaled</span>
<span id="cb176-173"><a href="rna-seq-mkdata.html#cb176-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-174"><a href="rna-seq-mkdata.html#cb176-174" aria-hidden="true" tabindex="-1"></a><span class="fu">is.data.frame</span>(counts2)</span>
<span id="cb176-175"><a href="rna-seq-mkdata.html#cb176-175" aria-hidden="true" tabindex="-1"></a>counts2 <span class="ot">&lt;-</span> <span class="fu">rownames_to_column</span>(counts, <span class="st">&quot;gene&quot;</span>)</span>
<span id="cb176-176"><a href="rna-seq-mkdata.html#cb176-176" aria-hidden="true" tabindex="-1"></a>counts2 <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(counts2, <span class="at">cols =</span> <span class="sc">-</span>gene, <span class="at">names_to =</span> <span class="st">&quot;sample&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;counts&quot;</span>)</span>
<span id="cb176-177"><a href="rna-seq-mkdata.html#cb176-177" aria-hidden="true" tabindex="-1"></a>counts2 <span class="ot">&lt;-</span> <span class="fu">group_by</span>(gene)</span>
<span id="cb176-178"><a href="rna-seq-mkdata.html#cb176-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-179"><a href="rna-seq-mkdata.html#cb176-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-180"><a href="rna-seq-mkdata.html#cb176-180" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(counts2[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,])</span>
<span id="cb176-181"><a href="rna-seq-mkdata.html#cb176-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-182"><a href="rna-seq-mkdata.html#cb176-182" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;vsn&quot;</span>)</span>
<span id="cb176-183"><a href="rna-seq-mkdata.html#cb176-183" aria-hidden="true" tabindex="-1"></a><span class="fu">meanSdPlot</span>(cts, <span class="at">ranks =</span> <span class="cn">FALSE</span>)</span>
<span id="cb176-184"><a href="rna-seq-mkdata.html#cb176-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-185"><a href="rna-seq-mkdata.html#cb176-185" aria-hidden="true" tabindex="-1"></a><span class="co"># Variance stabilisation</span></span>
<span id="cb176-186"><a href="rna-seq-mkdata.html#cb176-186" aria-hidden="true" tabindex="-1"></a><span class="co"># dds &lt;- estimateSizeFactors(dds)</span></span>
<span id="cb176-187"><a href="rna-seq-mkdata.html#cb176-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-188"><a href="rna-seq-mkdata.html#cb176-188" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span>  <span class="fu">estimateDispersions</span>(<span class="at">object =</span> dds, </span>
<span id="cb176-189"><a href="rna-seq-mkdata.html#cb176-189" aria-hidden="true" tabindex="-1"></a>                            <span class="at">fitType =</span> <span class="st">&quot;parametric&quot;</span>, </span>
<span id="cb176-190"><a href="rna-seq-mkdata.html#cb176-190" aria-hidden="true" tabindex="-1"></a>                            <span class="at">quiet =</span> <span class="cn">TRUE</span>)</span>
<span id="cb176-191"><a href="rna-seq-mkdata.html#cb176-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-192"><a href="rna-seq-mkdata.html#cb176-192" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds)</span>
<span id="cb176-193"><a href="rna-seq-mkdata.html#cb176-193" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span>  <span class="fu">estimateDispersions</span>(<span class="at">object =</span> dds,</span>
<span id="cb176-194"><a href="rna-seq-mkdata.html#cb176-194" aria-hidden="true" tabindex="-1"></a>                            <span class="at">fitType =</span> <span class="st">&quot;parametric&quot;</span>,</span>
<span id="cb176-195"><a href="rna-seq-mkdata.html#cb176-195" aria-hidden="true" tabindex="-1"></a>                            <span class="at">quiet =</span> <span class="cn">TRUE</span>)</span>
<span id="cb176-196"><a href="rna-seq-mkdata.html#cb176-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-197"><a href="rna-seq-mkdata.html#cb176-197" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">=</span> <span class="fu">varianceStabilizingTransformation</span>(<span class="at">object =</span> dds, </span>
<span id="cb176-198"><a href="rna-seq-mkdata.html#cb176-198" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">blind =</span> <span class="cn">TRUE</span>,           <span class="co"># do not take the design formula into account</span></span>
<span id="cb176-199"><a href="rna-seq-mkdata.html#cb176-199" aria-hidden="true" tabindex="-1"></a>                                        <span class="co"># best practice for sample-level QC</span></span>
<span id="cb176-200"><a href="rna-seq-mkdata.html#cb176-200" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">fitType =</span> <span class="st">&quot;parametric&quot;</span>)</span>
<span id="cb176-201"><a href="rna-seq-mkdata.html#cb176-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-202"><a href="rna-seq-mkdata.html#cb176-202" aria-hidden="true" tabindex="-1"></a><span class="co"># extract the matrix of variance stabilised counts</span></span>
<span id="cb176-203"><a href="rna-seq-mkdata.html#cb176-203" aria-hidden="true" tabindex="-1"></a>variance_stabilised_counts <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd)</span>
<span id="cb176-204"><a href="rna-seq-mkdata.html#cb176-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-205"><a href="rna-seq-mkdata.html#cb176-205" aria-hidden="true" tabindex="-1"></a><span class="co"># create the mean-sd plot</span></span>
<span id="cb176-206"><a href="rna-seq-mkdata.html#cb176-206" aria-hidden="true" tabindex="-1"></a>p_mean_sd_vst <span class="ot">&lt;-</span> </span>
<span id="cb176-207"><a href="rna-seq-mkdata.html#cb176-207" aria-hidden="true" tabindex="-1"></a>  variance_stabilised_counts <span class="sc">%&gt;%</span> </span>
<span id="cb176-208"><a href="rna-seq-mkdata.html#cb176-208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb176-209"><a href="rna-seq-mkdata.html#cb176-209" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb176-210"><a href="rna-seq-mkdata.html#cb176-210" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span> gene, <span class="at">names_to =</span> <span class="st">&quot;sample&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;counts&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb176-211"><a href="rna-seq-mkdata.html#cb176-211" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene) <span class="sc">%&gt;%</span> </span>
<span id="cb176-212"><a href="rna-seq-mkdata.html#cb176-212" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">gene_average =</span> <span class="fu">mean</span>(counts), <span class="at">gene_stdev =</span> <span class="fu">sd</span>(counts)) <span class="sc">%&gt;%</span> </span>
<span id="cb176-213"><a href="rna-seq-mkdata.html#cb176-213" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb176-214"><a href="rna-seq-mkdata.html#cb176-214" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> gene_average, <span class="at">y =</span> gene_stdev)) <span class="sc">+</span></span>
<span id="cb176-215"><a href="rna-seq-mkdata.html#cb176-215" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">fill =</span> <span class="st">&quot;grey&quot;</span>, <span class="at">colour =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb176-216"><a href="rna-seq-mkdata.html#cb176-216" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Gene count average (variance stabilised)&quot;</span>, </span>
<span id="cb176-217"><a href="rna-seq-mkdata.html#cb176-217" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Gene count standard deviation (variance stabilised)&quot;</span>) <span class="sc">+</span></span>
<span id="cb176-218"><a href="rna-seq-mkdata.html#cb176-218" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Mean - Standard deviation relationship</span><span class="sc">\n</span><span class="st">(after variance stabilisation &quot;</span>)</span>
<span id="cb176-219"><a href="rna-seq-mkdata.html#cb176-219" aria-hidden="true" tabindex="-1"></a>p_mean_sd_vst</span>
<span id="cb176-220"><a href="rna-seq-mkdata.html#cb176-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-221"><a href="rna-seq-mkdata.html#cb176-221" aria-hidden="true" tabindex="-1"></a><span class="co"># RNA-seq scree plot -----------</span></span>
<span id="cb176-222"><a href="rna-seq-mkdata.html#cb176-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-223"><a href="rna-seq-mkdata.html#cb176-223" aria-hidden="true" tabindex="-1"></a><span class="co"># Custom PCA function</span></span>
<span id="cb176-224"><a href="rna-seq-mkdata.html#cb176-224" aria-hidden="true" tabindex="-1"></a><span class="co"># define a custom R function called &quot;mypca()&quot;&quot;</span></span>
<span id="cb176-225"><a href="rna-seq-mkdata.html#cb176-225" aria-hidden="true" tabindex="-1"></a>mypca <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">scale =</span> <span class="cn">TRUE</span>){</span>
<span id="cb176-226"><a href="rna-seq-mkdata.html#cb176-226" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Samples should be in rows</span></span>
<span id="cb176-227"><a href="rna-seq-mkdata.html#cb176-227" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Variables in the columns</span></span>
<span id="cb176-228"><a href="rna-seq-mkdata.html#cb176-228" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb176-229"><a href="rna-seq-mkdata.html#cb176-229" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove constant variables</span></span>
<span id="cb176-230"><a href="rna-seq-mkdata.html#cb176-230" aria-hidden="true" tabindex="-1"></a>  constant_val <span class="ot">=</span> <span class="fu">apply</span>(x,<span class="dv">2</span>,<span class="st">&#39;sd&#39;</span>)</span>
<span id="cb176-231"><a href="rna-seq-mkdata.html#cb176-231" aria-hidden="true" tabindex="-1"></a>  x_reduced <span class="ot">=</span> x[,constant_val<span class="sc">&gt;</span><span class="dv">0</span>]</span>
<span id="cb176-232"><a href="rna-seq-mkdata.html#cb176-232" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb176-233"><a href="rna-seq-mkdata.html#cb176-233" aria-hidden="true" tabindex="-1"></a>  <span class="co"># perform SVD</span></span>
<span id="cb176-234"><a href="rna-seq-mkdata.html#cb176-234" aria-hidden="true" tabindex="-1"></a>  SVD <span class="ot">&lt;-</span> <span class="fu">svd</span>(<span class="fu">scale</span>(x_reduced,<span class="at">center =</span> center, <span class="at">scale =</span> scale))</span>
<span id="cb176-235"><a href="rna-seq-mkdata.html#cb176-235" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb176-236"><a href="rna-seq-mkdata.html#cb176-236" aria-hidden="true" tabindex="-1"></a>  <span class="co"># create scores data frame</span></span>
<span id="cb176-237"><a href="rna-seq-mkdata.html#cb176-237" aria-hidden="true" tabindex="-1"></a>  scores <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(SVD<span class="sc">$</span>u <span class="sc">%*%</span> <span class="fu">diag</span>(SVD<span class="sc">$</span>d))</span>
<span id="cb176-238"><a href="rna-seq-mkdata.html#cb176-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(scores) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(x)</span>
<span id="cb176-239"><a href="rna-seq-mkdata.html#cb176-239" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(scores) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;PC&quot;</span>, <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(scores)[<span class="dv">2</span>]))</span>
<span id="cb176-240"><a href="rna-seq-mkdata.html#cb176-240" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb176-241"><a href="rna-seq-mkdata.html#cb176-241" aria-hidden="true" tabindex="-1"></a>  <span class="co"># create loadings data frams</span></span>
<span id="cb176-242"><a href="rna-seq-mkdata.html#cb176-242" aria-hidden="true" tabindex="-1"></a>  loadings <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(SVD<span class="sc">$</span>v)</span>
<span id="cb176-243"><a href="rna-seq-mkdata.html#cb176-243" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(loadings) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;PC&quot;</span>, <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(loadings)[<span class="dv">2</span>]))</span>
<span id="cb176-244"><a href="rna-seq-mkdata.html#cb176-244" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(loadings) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(x_reduced)</span>
<span id="cb176-245"><a href="rna-seq-mkdata.html#cb176-245" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb176-246"><a href="rna-seq-mkdata.html#cb176-246" aria-hidden="true" tabindex="-1"></a>  <span class="co"># create data frame for explained variances</span></span>
<span id="cb176-247"><a href="rna-seq-mkdata.html#cb176-247" aria-hidden="true" tabindex="-1"></a>  explained_var <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">round</span>((SVD<span class="sc">$</span>d<span class="sc">^</span><span class="dv">2</span>) <span class="sc">/</span> <span class="fu">sum</span>(SVD<span class="sc">$</span>d<span class="sc">^</span><span class="dv">2</span>)<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits =</span> <span class="dv">1</span>))</span>
<span id="cb176-248"><a href="rna-seq-mkdata.html#cb176-248" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(explained_var) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;PC&quot;</span>, <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">dim</span>(loadings)[<span class="dv">2</span>]))</span>
<span id="cb176-249"><a href="rna-seq-mkdata.html#cb176-249" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(explained_var) <span class="ot">&lt;-</span> <span class="st">&quot;exp_var&quot;</span></span>
<span id="cb176-250"><a href="rna-seq-mkdata.html#cb176-250" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb176-251"><a href="rna-seq-mkdata.html#cb176-251" aria-hidden="true" tabindex="-1"></a>  <span class="co"># return result</span></span>
<span id="cb176-252"><a href="rna-seq-mkdata.html#cb176-252" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span> (<span class="fu">list</span>(<span class="st">&quot;scores&quot;</span> <span class="ot">=</span> scores, <span class="st">&quot;loadings&quot;</span> <span class="ot">=</span> loadings, <span class="st">&quot;explained_var&quot;</span> <span class="ot">=</span> explained_var))</span>
<span id="cb176-253"><a href="rna-seq-mkdata.html#cb176-253" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb176-254"><a href="rna-seq-mkdata.html#cb176-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-255"><a href="rna-seq-mkdata.html#cb176-255" aria-hidden="true" tabindex="-1"></a><span class="co"># transpose the data because in variance_stabilised_counts the rows are the variables and the columns correspond to the samples</span></span>
<span id="cb176-256"><a href="rna-seq-mkdata.html#cb176-256" aria-hidden="true" tabindex="-1"></a>t_variance_stabilised_counts <span class="ot">&lt;-</span> variance_stabilised_counts <span class="co"># this is not doing anything!!!!</span></span>
<span id="cb176-257"><a href="rna-seq-mkdata.html#cb176-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-258"><a href="rna-seq-mkdata.html#cb176-258" aria-hidden="true" tabindex="-1"></a><span class="co"># before computing the PCA, check that samples are in rows and genes in columns</span></span>
<span id="cb176-259"><a href="rna-seq-mkdata.html#cb176-259" aria-hidden="true" tabindex="-1"></a>pca_results <span class="ot">&lt;-</span> <span class="fu">mypca</span>(<span class="fu">t</span>(t_variance_stabilised_counts), <span class="co"># added the transpose t() funtion here!!!.. now it works.!!!</span></span>
<span id="cb176-260"><a href="rna-seq-mkdata.html#cb176-260" aria-hidden="true" tabindex="-1"></a>                     <span class="at">center =</span> <span class="cn">TRUE</span>, </span>
<span id="cb176-261"><a href="rna-seq-mkdata.html#cb176-261" aria-hidden="true" tabindex="-1"></a>                     <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb176-262"><a href="rna-seq-mkdata.html#cb176-262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-263"><a href="rna-seq-mkdata.html#cb176-263" aria-hidden="true" tabindex="-1"></a><span class="co"># make the plot</span></span>
<span id="cb176-264"><a href="rna-seq-mkdata.html#cb176-264" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pca_results<span class="sc">$</span>explained_var, </span>
<span id="cb176-265"><a href="rna-seq-mkdata.html#cb176-265" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">seq</span>(<span class="at">from =</span> <span class="dv">1</span>, <span class="at">to =</span> <span class="fu">nrow</span>(pca_results<span class="sc">$</span>explained_var)), </span>
<span id="cb176-266"><a href="rna-seq-mkdata.html#cb176-266" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> exp_var)) <span class="sc">+</span></span>
<span id="cb176-267"><a href="rna-seq-mkdata.html#cb176-267" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&#39;explained variance (%)&#39;</span>) <span class="sc">+</span> </span>
<span id="cb176-268"><a href="rna-seq-mkdata.html#cb176-268" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&#39;Explained variance per component&#39;</span>) <span class="sc">+</span> </span>
<span id="cb176-269"><a href="rna-seq-mkdata.html#cb176-269" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb176-270"><a href="rna-seq-mkdata.html#cb176-270" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Principal Component number&quot;</span>) <span class="sc">+</span></span>
<span id="cb176-271"><a href="rna-seq-mkdata.html#cb176-271" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(</span>
<span id="cb176-272"><a href="rna-seq-mkdata.html#cb176-272" aria-hidden="true" tabindex="-1"></a>    <span class="at">from =</span> <span class="dv">1</span>, </span>
<span id="cb176-273"><a href="rna-seq-mkdata.html#cb176-273" aria-hidden="true" tabindex="-1"></a>    <span class="at">to =</span> <span class="fu">nrow</span>(pca_results<span class="sc">$</span>explained_var)))</span>
<span id="cb176-274"><a href="rna-seq-mkdata.html#cb176-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-275"><a href="rna-seq-mkdata.html#cb176-275" aria-hidden="true" tabindex="-1"></a><span class="co">#  How much percentage of the total variance are “caught” by PC1 and PC2?</span></span>
<span id="cb176-276"><a href="rna-seq-mkdata.html#cb176-276" aria-hidden="true" tabindex="-1"></a><span class="co">#  How many PCs are necessary to get 50% of the total variance?</span></span>
<span id="cb176-277"><a href="rna-seq-mkdata.html#cb176-277" aria-hidden="true" tabindex="-1"></a><span class="co">#  Solution</span></span>
<span id="cb176-278"><a href="rna-seq-mkdata.html#cb176-278" aria-hidden="true" tabindex="-1"></a><span class="fu">cumsum</span>(pca_results<span class="sc">$</span>explained_var)[<span class="dv">2</span>,<span class="dv">1</span>] <span class="co"># shows you that 28.8% of the varaince are explained by PC1 and PC2.</span></span>
<span id="cb176-279"><a href="rna-seq-mkdata.html#cb176-279" aria-hidden="true" tabindex="-1"></a><span class="fu">cumsum</span>(pca_results<span class="sc">$</span>explained_var) <span class="sc">%&gt;%</span> </span>
<span id="cb176-280"><a href="rna-seq-mkdata.html#cb176-280" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb176-281"><a href="rna-seq-mkdata.html#cb176-281" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(exp_var <span class="sc">&gt;</span> <span class="dv">50</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb176-282"><a href="rna-seq-mkdata.html#cb176-282" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="at">n =</span> <span class="dv">1</span>) <span class="co"># You need to go up to PC7 to catch 51% of the variance.</span></span>
<span id="cb176-283"><a href="rna-seq-mkdata.html#cb176-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-284"><a href="rna-seq-mkdata.html#cb176-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-285"><a href="rna-seq-mkdata.html#cb176-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-286"><a href="rna-seq-mkdata.html#cb176-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-287"><a href="rna-seq-mkdata.html#cb176-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-288"><a href="rna-seq-mkdata.html#cb176-288" aria-hidden="true" tabindex="-1"></a>scores <span class="ot">&lt;-</span> pca_results<span class="sc">$</span>scores</span>
<span id="cb176-289"><a href="rna-seq-mkdata.html#cb176-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-290"><a href="rna-seq-mkdata.html#cb176-290" aria-hidden="true" tabindex="-1"></a><span class="co"># first 5 rows and columns</span></span>
<span id="cb176-291"><a href="rna-seq-mkdata.html#cb176-291" aria-hidden="true" tabindex="-1"></a>scores[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb176-292"><a href="rna-seq-mkdata.html#cb176-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-293"><a href="rna-seq-mkdata.html#cb176-293" aria-hidden="true" tabindex="-1"></a>scores_with_conditions <span class="ot">&lt;-</span> </span>
<span id="cb176-294"><a href="rna-seq-mkdata.html#cb176-294" aria-hidden="true" tabindex="-1"></a>  scores <span class="sc">%&gt;%</span> </span>
<span id="cb176-295"><a href="rna-seq-mkdata.html#cb176-295" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleName&quot;</span>) <span class="sc">%&gt;%</span> <span class="co"># to prepare to join on the &quot;sample&quot; column</span></span>
<span id="cb176-296"><a href="rna-seq-mkdata.html#cb176-296" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(<span class="at">x =</span> .,                 <span class="co"># this means that we are passing the &#39;scores&#39; dataframe </span></span>
<span id="cb176-297"><a href="rna-seq-mkdata.html#cb176-297" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> Metadata2,         <span class="co"># this dataframe contains the sample to condition correspondence</span></span>
<span id="cb176-298"><a href="rna-seq-mkdata.html#cb176-298" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="st">&quot;SampleName&quot;</span>)</span>
<span id="cb176-299"><a href="rna-seq-mkdata.html#cb176-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-300"><a href="rna-seq-mkdata.html#cb176-300" aria-hidden="true" tabindex="-1"></a><span class="co"># shows the first 5 rows and the last 6 columns. Note that conditions were added to the table.  </span></span>
<span id="cb176-301"><a href="rna-seq-mkdata.html#cb176-301" aria-hidden="true" tabindex="-1"></a>scores_with_conditions[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">15</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb176-302"><a href="rna-seq-mkdata.html#cb176-302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-303"><a href="rna-seq-mkdata.html#cb176-303" aria-hidden="true" tabindex="-1"></a><span class="co"># PC14      PC15          PC16 Genotype Time Tissue</span></span>
<span id="cb176-304"><a href="rna-seq-mkdata.html#cb176-304" aria-hidden="true" tabindex="-1"></a><span class="co"># 1  11.18538  11.34342 -2.053225e-12       WT    7    PDL</span></span>
<span id="cb176-305"><a href="rna-seq-mkdata.html#cb176-305" aria-hidden="true" tabindex="-1"></a><span class="co"># 2 -24.19255  42.96321 -2.053225e-12       WT    7    PDL</span></span>
<span id="cb176-306"><a href="rna-seq-mkdata.html#cb176-306" aria-hidden="true" tabindex="-1"></a><span class="co"># 3  30.91639 -20.58249 -2.053225e-12       WT    7    PDL</span></span>
<span id="cb176-307"><a href="rna-seq-mkdata.html#cb176-307" aria-hidden="true" tabindex="-1"></a><span class="co"># 4 -26.27940 -26.47902 -2.053225e-12       WT    7    PDL</span></span>
<span id="cb176-308"><a href="rna-seq-mkdata.html#cb176-308" aria-hidden="true" tabindex="-1"></a><span class="co"># 5 -67.95911 -14.25815 -2.053225e-12       Ko    7    PDL</span></span>
<span id="cb176-309"><a href="rna-seq-mkdata.html#cb176-309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-310"><a href="rna-seq-mkdata.html#cb176-310" aria-hidden="true" tabindex="-1"></a><span class="co"># explained variance</span></span>
<span id="cb176-311"><a href="rna-seq-mkdata.html#cb176-311" aria-hidden="true" tabindex="-1"></a><span class="co"># one % variance value per PC</span></span>
<span id="cb176-312"><a href="rna-seq-mkdata.html#cb176-312" aria-hidden="true" tabindex="-1"></a>explained_variance <span class="ot">&lt;-</span> </span>
<span id="cb176-313"><a href="rna-seq-mkdata.html#cb176-313" aria-hidden="true" tabindex="-1"></a>  pca_results<span class="sc">$</span>explained_var <span class="sc">%&gt;%</span> </span>
<span id="cb176-314"><a href="rna-seq-mkdata.html#cb176-314" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(<span class="st">&quot;exp_var&quot;</span>)</span>
<span id="cb176-315"><a href="rna-seq-mkdata.html#cb176-315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-316"><a href="rna-seq-mkdata.html#cb176-316" aria-hidden="true" tabindex="-1"></a>explained_variance</span>
<span id="cb176-317"><a href="rna-seq-mkdata.html#cb176-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-318"><a href="rna-seq-mkdata.html#cb176-318" aria-hidden="true" tabindex="-1"></a><span class="co"># RNAseq Ploting ------------------</span></span>
<span id="cb176-319"><a href="rna-seq-mkdata.html#cb176-319" aria-hidden="true" tabindex="-1"></a><span class="co"># Genotype Plot</span></span>
<span id="cb176-320"><a href="rna-seq-mkdata.html#cb176-320" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(scores_with_conditions, </span>
<span id="cb176-321"><a href="rna-seq-mkdata.html#cb176-321" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(PC1, PC2, <span class="at">color =</span> Genotype)) <span class="sc">+</span></span>
<span id="cb176-322"><a href="rna-seq-mkdata.html#cb176-322" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb176-323"><a href="rna-seq-mkdata.html#cb176-323" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">label =</span> SampleName), <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb176-324"><a href="rna-seq-mkdata.html#cb176-324" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC1: &quot;</span>,explained_variance[<span class="dv">1</span>],<span class="st">&quot;% variance&quot;</span>)) <span class="sc">+</span></span>
<span id="cb176-325"><a href="rna-seq-mkdata.html#cb176-325" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC2: &quot;</span>,explained_variance[<span class="dv">2</span>],<span class="st">&quot;% variance&quot;</span>)) <span class="sc">+</span> </span>
<span id="cb176-326"><a href="rna-seq-mkdata.html#cb176-326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>(<span class="at">ratio =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb176-327"><a href="rna-seq-mkdata.html#cb176-327" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;PCA score plot with the Genotype condition overlaid&quot;</span>)</span>
<span id="cb176-328"><a href="rna-seq-mkdata.html#cb176-328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-329"><a href="rna-seq-mkdata.html#cb176-329" aria-hidden="true" tabindex="-1"></a><span class="co"># Time Plot</span></span>
<span id="cb176-330"><a href="rna-seq-mkdata.html#cb176-330" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(scores_with_conditions, </span>
<span id="cb176-331"><a href="rna-seq-mkdata.html#cb176-331" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(PC1, PC2, <span class="at">color =</span> Time)) <span class="sc">+</span></span>
<span id="cb176-332"><a href="rna-seq-mkdata.html#cb176-332" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb176-333"><a href="rna-seq-mkdata.html#cb176-333" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">label =</span> ShortName), <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb176-334"><a href="rna-seq-mkdata.html#cb176-334" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC1: &quot;</span>,explained_variance[<span class="dv">1</span>],<span class="st">&quot;% variance&quot;</span>)) <span class="sc">+</span></span>
<span id="cb176-335"><a href="rna-seq-mkdata.html#cb176-335" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC2: &quot;</span>,explained_variance[<span class="dv">2</span>],<span class="st">&quot;% variance&quot;</span>)) <span class="sc">+</span> </span>
<span id="cb176-336"><a href="rna-seq-mkdata.html#cb176-336" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>(<span class="at">ratio =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb176-337"><a href="rna-seq-mkdata.html#cb176-337" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;PCA score plot with Time condition overlaid&quot;</span>)</span>
<span id="cb176-338"><a href="rna-seq-mkdata.html#cb176-338" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-339"><a href="rna-seq-mkdata.html#cb176-339" aria-hidden="true" tabindex="-1"></a><span class="co"># Other Condition...</span></span>
<span id="cb176-340"><a href="rna-seq-mkdata.html#cb176-340" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(scores_with_conditions, </span>
<span id="cb176-341"><a href="rna-seq-mkdata.html#cb176-341" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(PC1, PC2, <span class="at">color =</span> dpi)) <span class="sc">+</span></span>
<span id="cb176-342"><a href="rna-seq-mkdata.html#cb176-342" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb176-343"><a href="rna-seq-mkdata.html#cb176-343" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">label =</span> sample), <span class="at">size =</span> <span class="fl">3.5</span>) <span class="sc">+</span></span>
<span id="cb176-344"><a href="rna-seq-mkdata.html#cb176-344" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC1: &quot;</span>,explained_variance[<span class="dv">1</span>],<span class="st">&quot;% variance&quot;</span>)) <span class="sc">+</span></span>
<span id="cb176-345"><a href="rna-seq-mkdata.html#cb176-345" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC2: &quot;</span>,explained_variance[<span class="dv">2</span>],<span class="st">&quot;% variance&quot;</span>)) <span class="sc">+</span> </span>
<span id="cb176-346"><a href="rna-seq-mkdata.html#cb176-346" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>(<span class="at">ratio =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb176-347"><a href="rna-seq-mkdata.html#cb176-347" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;PCA score plot with the time after infection (dpi) overlaid&quot;</span>)</span>
<span id="cb176-348"><a href="rna-seq-mkdata.html#cb176-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-349"><a href="rna-seq-mkdata.html#cb176-349" aria-hidden="true" tabindex="-1"></a><span class="co"># ADD on not working ------</span></span>
<span id="cb176-350"><a href="rna-seq-mkdata.html#cb176-350" aria-hidden="true" tabindex="-1"></a><span class="co"># Lets look at how to use loading scores to dermine which genes have the largest effect on where samples are plotted in the PCA plot</span></span>
<span id="cb176-351"><a href="rna-seq-mkdata.html#cb176-351" aria-hidden="true" tabindex="-1"></a><span class="co"># Are Lodings and rotation the same?  Seems like yes!!</span></span>
<span id="cb176-352"><a href="rna-seq-mkdata.html#cb176-352" aria-hidden="true" tabindex="-1"></a>loading_scores2 <span class="ot">&lt;-</span> pca_results<span class="sc">$</span>loadings[,<span class="dv">1</span>]</span>
<span id="cb176-353"><a href="rna-seq-mkdata.html#cb176-353" aria-hidden="true" tabindex="-1"></a>gene_scores2 <span class="ot">&lt;-</span> <span class="fu">abs</span>(loading_scores2)</span>
<span id="cb176-354"><a href="rna-seq-mkdata.html#cb176-354" aria-hidden="true" tabindex="-1"></a>gene_scores_ranked2 <span class="ot">&lt;-</span>  <span class="fu">sort</span>(gene_scores2, <span class="at">decreasing =</span> T)</span>
<span id="cb176-355"><a href="rna-seq-mkdata.html#cb176-355" aria-hidden="true" tabindex="-1"></a>top10_genes2 <span class="ot">&lt;-</span> <span class="fu">row.names</span>(gene_scores_ranked2[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>])</span>
<span id="cb176-356"><a href="rna-seq-mkdata.html#cb176-356" aria-hidden="true" tabindex="-1"></a>top10_genes2</span>
<span id="cb176-357"><a href="rna-seq-mkdata.html#cb176-357" aria-hidden="true" tabindex="-1"></a>pca<span class="sc">$</span>rotation[top10_genes,<span class="dv">1</span>] <span class="co"># show the scores (and +/-)</span></span>
<span id="cb176-358"><a href="rna-seq-mkdata.html#cb176-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-359"><a href="rna-seq-mkdata.html#cb176-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-360"><a href="rna-seq-mkdata.html#cb176-360" aria-hidden="true" tabindex="-1"></a><span class="co"># DESeq2 count normalization ----</span></span>
<span id="cb176-361"><a href="rna-seq-mkdata.html#cb176-361" aria-hidden="true" tabindex="-1"></a><span class="co"># Let’s see how, in practice, we can use DESeq2 median-of-ratios method to normalize the gene counts.</span></span>
<span id="cb176-362"><a href="rna-seq-mkdata.html#cb176-362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-363"><a href="rna-seq-mkdata.html#cb176-363" aria-hidden="true" tabindex="-1"></a><span class="co"># Data Import and  matching to the experimantal design and counts data same as above. </span></span>
<span id="cb176-364"><a href="rna-seq-mkdata.html#cb176-364" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(counts[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span>
<span id="cb176-365"><a href="rna-seq-mkdata.html#cb176-365" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the DESeqDataSet object</span></span>
<span id="cb176-366"><a href="rna-seq-mkdata.html#cb176-366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-367"><a href="rna-seq-mkdata.html#cb176-367" aria-hidden="true" tabindex="-1"></a><span class="co"># suppressPackageStartupMessages(library(DESeq2)) # to load DESeq2 and suppress the long startup message</span></span>
<span id="cb176-368"><a href="rna-seq-mkdata.html#cb176-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-369"><a href="rna-seq-mkdata.html#cb176-369" aria-hidden="true" tabindex="-1"></a><span class="co"># Creation of the DESeqDataSet object</span></span>
<span id="cb176-370"><a href="rna-seq-mkdata.html#cb176-370" aria-hidden="true" tabindex="-1"></a>dds2 <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> all_raw_count02, </span>
<span id="cb176-371"><a href="rna-seq-mkdata.html#cb176-371" aria-hidden="true" tabindex="-1"></a>                               <span class="at">colData =</span> Metadata2, </span>
<span id="cb176-372"><a href="rna-seq-mkdata.html#cb176-372" aria-hidden="true" tabindex="-1"></a>                               <span class="at">design =</span> <span class="sc">~</span> Genotype) </span>
<span id="cb176-373"><a href="rna-seq-mkdata.html#cb176-373" aria-hidden="true" tabindex="-1"></a>dds2</span>
<span id="cb176-374"><a href="rna-seq-mkdata.html#cb176-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-375"><a href="rna-seq-mkdata.html#cb176-375" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate normalized counts</span></span>
<span id="cb176-376"><a href="rna-seq-mkdata.html#cb176-376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-377"><a href="rna-seq-mkdata.html#cb176-377" aria-hidden="true" tabindex="-1"></a>dds2 <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds2)</span>
<span id="cb176-378"><a href="rna-seq-mkdata.html#cb176-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-379"><a href="rna-seq-mkdata.html#cb176-379" aria-hidden="true" tabindex="-1"></a><span class="fu">sizeFactors</span>(dds2)</span>
<span id="cb176-380"><a href="rna-seq-mkdata.html#cb176-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-381"><a href="rna-seq-mkdata.html#cb176-381" aria-hidden="true" tabindex="-1"></a><span class="co"># We can plot these size factors to see how much they differ between samples.</span></span>
<span id="cb176-382"><a href="rna-seq-mkdata.html#cb176-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-383"><a href="rna-seq-mkdata.html#cb176-383" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb176-384"><a href="rna-seq-mkdata.html#cb176-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-385"><a href="rna-seq-mkdata.html#cb176-385" aria-hidden="true" tabindex="-1"></a><span class="co"># create a dplyr tibble</span></span>
<span id="cb176-386"><a href="rna-seq-mkdata.html#cb176-386" aria-hidden="true" tabindex="-1"></a>size_factors_df <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb176-387"><a href="rna-seq-mkdata.html#cb176-387" aria-hidden="true" tabindex="-1"></a>  <span class="at">sample =</span> <span class="fu">names</span>(<span class="fu">sizeFactors</span>(dds2)), </span>
<span id="cb176-388"><a href="rna-seq-mkdata.html#cb176-388" aria-hidden="true" tabindex="-1"></a>  <span class="at">correction_factor =</span> <span class="fu">sizeFactors</span>(dds2)</span>
<span id="cb176-389"><a href="rna-seq-mkdata.html#cb176-389" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb176-390"><a href="rna-seq-mkdata.html#cb176-390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-391"><a href="rna-seq-mkdata.html#cb176-391" aria-hidden="true" tabindex="-1"></a><span class="co"># line plot to connect the different size factor values</span></span>
<span id="cb176-392"><a href="rna-seq-mkdata.html#cb176-392" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(size_factors_df, <span class="fu">aes</span>(<span class="at">x =</span> sample, <span class="at">y =</span> correction_factor, <span class="at">group =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb176-393"><a href="rna-seq-mkdata.html#cb176-393" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb176-394"><a href="rna-seq-mkdata.html#cb176-394" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb176-395"><a href="rna-seq-mkdata.html#cb176-395" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">size =</span> <span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb176-396"><a href="rna-seq-mkdata.html#cb176-396" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="fl">0.5</span>,<span class="dv">2</span>))<span class="sc">+</span></span>
<span id="cb176-397"><a href="rna-seq-mkdata.html#cb176-397" aria-hidden="true" tabindex="-1"></a> <span class="co"># xlab(paste(&quot;PC1 - &quot;, pca.var.per[1], &quot;%&quot;, sep = &quot;&quot;)) +</span></span>
<span id="cb176-398"><a href="rna-seq-mkdata.html#cb176-398" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ylab(paste(&quot;PC2 - &quot;, pca.var.per[2], &quot;%&quot;, sep = &quot;&quot;)) +</span></span>
<span id="cb176-399"><a href="rna-seq-mkdata.html#cb176-399" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Median of ratios method of normalization (size factors) to see how much samples differ between eachother &quot;</span>)</span>
<span id="cb176-400"><a href="rna-seq-mkdata.html#cb176-400" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-401"><a href="rna-seq-mkdata.html#cb176-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-402"><a href="rna-seq-mkdata.html#cb176-402" aria-hidden="true" tabindex="-1"></a><span class="co"># to display the plot</span></span>
<span id="cb176-403"><a href="rna-seq-mkdata.html#cb176-403" aria-hidden="true" tabindex="-1"></a>p</span>
<span id="cb176-404"><a href="rna-seq-mkdata.html#cb176-404" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-405"><a href="rna-seq-mkdata.html#cb176-405" aria-hidden="true" tabindex="-1"></a><span class="co"># extract the normalised counts</span></span>
<span id="cb176-406"><a href="rna-seq-mkdata.html#cb176-406" aria-hidden="true" tabindex="-1"></a>counts_normalised <span class="ot">=</span> <span class="fu">counts</span>(dds2, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb176-407"><a href="rna-seq-mkdata.html#cb176-407" aria-hidden="true" tabindex="-1"></a>counts_normalised[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb176-408"><a href="rna-seq-mkdata.html#cb176-408" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-409"><a href="rna-seq-mkdata.html#cb176-409" aria-hidden="true" tabindex="-1"></a>counts_log10 <span class="ot">&lt;-</span> <span class="fu">log10</span>(counts<span class="sc">+</span><span class="dv">1</span>)</span>
<span id="cb176-410"><a href="rna-seq-mkdata.html#cb176-410" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(counts_log10, <span class="at">las=</span><span class="dv">2</span>, <span class="at">main=</span><span class="st">&quot;Log10 of counts&quot;</span>)</span>
<span id="cb176-411"><a href="rna-seq-mkdata.html#cb176-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-412"><a href="rna-seq-mkdata.html#cb176-412" aria-hidden="true" tabindex="-1"></a>counts_normalised_log10 <span class="ot">&lt;-</span> <span class="fu">log10</span>(counts_normalised<span class="sc">+</span><span class="dv">1</span>)</span>
<span id="cb176-413"><a href="rna-seq-mkdata.html#cb176-413" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(counts_normalised_log10, <span class="at">las=</span><span class="dv">2</span>, <span class="at">col=</span> <span class="st">&quot;green&quot;</span>, <span class="at">main=</span><span class="st">&quot;Normalize log10 counts&quot;</span>)</span>
<span id="cb176-414"><a href="rna-seq-mkdata.html#cb176-414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-415"><a href="rna-seq-mkdata.html#cb176-415" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-416"><a href="rna-seq-mkdata.html#cb176-416" aria-hidden="true" tabindex="-1"></a><span class="co"># Differential expression analysis -----------------------</span></span>
<span id="cb176-417"><a href="rna-seq-mkdata.html#cb176-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-418"><a href="rna-seq-mkdata.html#cb176-418" aria-hidden="true" tabindex="-1"></a><span class="co"># Warning *************************************</span></span>
<span id="cb176-419"><a href="rna-seq-mkdata.html#cb176-419" aria-hidden="true" tabindex="-1"></a><span class="co"># We do not want to work on all comparisons, we will filter out the samples and conditions that we do not need. In this example, only the mock growth and the P. syringae infected condition will remain.</span></span>
<span id="cb176-420"><a href="rna-seq-mkdata.html#cb176-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-421"><a href="rna-seq-mkdata.html#cb176-421" aria-hidden="true" tabindex="-1"></a><span class="co"># You should still have the counts and xp_design objects in your R environment. If not, please run the following code.</span></span>
<span id="cb176-422"><a href="rna-seq-mkdata.html#cb176-422" aria-hidden="true" tabindex="-1"></a><span class="co"># read the xp design file if not available in your environment </span></span>
<span id="cb176-423"><a href="rna-seq-mkdata.html#cb176-423" aria-hidden="true" tabindex="-1"></a><span class="co"># Metadata2 &lt;- read.delim(&quot;experimental_design_modified.txt&quot;, </span></span>
<span id="cb176-424"><a href="rna-seq-mkdata.html#cb176-424" aria-hidden="true" tabindex="-1"></a><span class="co">#                         header = T, </span></span>
<span id="cb176-425"><a href="rna-seq-mkdata.html#cb176-425" aria-hidden="true" tabindex="-1"></a><span class="co">#                         stringsAsFactors = F, </span></span>
<span id="cb176-426"><a href="rna-seq-mkdata.html#cb176-426" aria-hidden="true" tabindex="-1"></a><span class="co">#                         colClasses = rep(&quot;character&quot;,4))</span></span>
<span id="cb176-427"><a href="rna-seq-mkdata.html#cb176-427" aria-hidden="true" tabindex="-1"></a><span class="co"># # change col names</span></span>
<span id="cb176-428"><a href="rna-seq-mkdata.html#cb176-428" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(Metadata2) &lt;- c(&quot;sample&quot;, &quot;seed&quot;, &quot;infected&quot;, &quot;dpi&quot;)</span></span>
<span id="cb176-429"><a href="rna-seq-mkdata.html#cb176-429" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb176-430"><a href="rna-seq-mkdata.html#cb176-430" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb176-431"><a href="rna-seq-mkdata.html#cb176-431" aria-hidden="true" tabindex="-1"></a><span class="co"># counts &lt;- read.delim(&quot;counts.txt&quot;, header = T, stringsAsFactors = F)</span></span>
<span id="cb176-432"><a href="rna-seq-mkdata.html#cb176-432" aria-hidden="true" tabindex="-1"></a><span class="co"># genes &lt;- counts[,1]</span></span>
<span id="cb176-433"><a href="rna-seq-mkdata.html#cb176-433" aria-hidden="true" tabindex="-1"></a><span class="co"># counts &lt;- counts[,-1]</span></span>
<span id="cb176-434"><a href="rna-seq-mkdata.html#cb176-434" aria-hidden="true" tabindex="-1"></a><span class="co"># row.names(counts) &lt;- genes</span></span>
<span id="cb176-435"><a href="rna-seq-mkdata.html#cb176-435" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb176-436"><a href="rna-seq-mkdata.html#cb176-436" aria-hidden="true" tabindex="-1"></a><span class="co"># # reorder counts columns according to the experimental design file</span></span>
<span id="cb176-437"><a href="rna-seq-mkdata.html#cb176-437" aria-hidden="true" tabindex="-1"></a><span class="co"># counts &lt;- counts[, xp_design$sample]</span></span>
<span id="cb176-438"><a href="rna-seq-mkdata.html#cb176-438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-439"><a href="rna-seq-mkdata.html#cb176-439" aria-hidden="true" tabindex="-1"></a><span class="co">#******************************************************************************</span></span>
<span id="cb176-440"><a href="rna-seq-mkdata.html#cb176-440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-441"><a href="rna-seq-mkdata.html#cb176-441" aria-hidden="true" tabindex="-1"></a><span class="co"># We will now filter both the counts and xp_design objects to keep a one-factor comparison to investigate....?</span></span>
<span id="cb176-442"><a href="rna-seq-mkdata.html#cb176-442" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb176-443"><a href="rna-seq-mkdata.html#cb176-443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-444"><a href="rna-seq-mkdata.html#cb176-444" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(Metadata2)</span>
<span id="cb176-445"><a href="rna-seq-mkdata.html#cb176-445" aria-hidden="true" tabindex="-1"></a>Metadata2<span class="sc">$</span>Time <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(Metadata2<span class="sc">$</span>Time)</span>
<span id="cb176-446"><a href="rna-seq-mkdata.html#cb176-446" aria-hidden="true" tabindex="-1"></a>Metadata2<span class="sc">$</span>Genotype <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(Metadata2<span class="sc">$</span>Genotype)</span>
<span id="cb176-447"><a href="rna-seq-mkdata.html#cb176-447" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-448"><a href="rna-seq-mkdata.html#cb176-448" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(Metadata2)</span>
<span id="cb176-449"><a href="rna-seq-mkdata.html#cb176-449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-450"><a href="rna-seq-mkdata.html#cb176-450" aria-hidden="true" tabindex="-1"></a><span class="co"># filter design file (mock versus P. syringae at 7 dpi)</span></span>
<span id="cb176-451"><a href="rna-seq-mkdata.html#cb176-451" aria-hidden="true" tabindex="-1"></a>var_GenetypeKo_vs_Time14 <span class="ot">=</span> Metadata2 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Time <span class="sc">==</span> <span class="st">&quot;14&quot;</span>) <span class="co"># (Genotype == &quot;Ko&quot; &amp; Time == &quot;14&quot;)</span></span>
<span id="cb176-452"><a href="rna-seq-mkdata.html#cb176-452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-453"><a href="rna-seq-mkdata.html#cb176-453" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter count file accordingly (to keep only samples present in the filtered xp_design file)</span></span>
<span id="cb176-454"><a href="rna-seq-mkdata.html#cb176-454" aria-hidden="true" tabindex="-1"></a>counts_filtered <span class="ot">=</span> counts[, <span class="fu">colnames</span>(counts) <span class="sc">%in%</span> var_GenetypeKo_vs_Time14<span class="sc">$</span>SampleName]</span>
<span id="cb176-455"><a href="rna-seq-mkdata.html#cb176-455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-456"><a href="rna-seq-mkdata.html#cb176-456" aria-hidden="true" tabindex="-1"></a><span class="do">## Creation of the DESeqDataSet</span></span>
<span id="cb176-457"><a href="rna-seq-mkdata.html#cb176-457" aria-hidden="true" tabindex="-1"></a>dds3 <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> counts_filtered, </span>
<span id="cb176-458"><a href="rna-seq-mkdata.html#cb176-458" aria-hidden="true" tabindex="-1"></a>                               <span class="at">colData =</span> var_GenetypeKo_vs_Time14, </span>
<span id="cb176-459"><a href="rna-seq-mkdata.html#cb176-459" aria-hidden="true" tabindex="-1"></a>                               <span class="at">design =</span> <span class="sc">~</span> Genotype)</span>
<span id="cb176-460"><a href="rna-seq-mkdata.html#cb176-460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-461"><a href="rna-seq-mkdata.html#cb176-461" aria-hidden="true" tabindex="-1"></a><span class="co"># It is important to make sure that levels are properly ordered so we are indeed using the mock group as our reference level. A positive gene fold change will for instance signify that the gene is upregulated in the P. syringae condition relatively to the mock condition.</span></span>
<span id="cb176-462"><a href="rna-seq-mkdata.html#cb176-462" aria-hidden="true" tabindex="-1"></a><span class="co">#One way to see how levels are interpreted within the DESeqDataSet object is to display the factor levels.</span></span>
<span id="cb176-463"><a href="rna-seq-mkdata.html#cb176-463" aria-hidden="true" tabindex="-1"></a>dds3<span class="sc">$</span>Genotype</span>
<span id="cb176-464"><a href="rna-seq-mkdata.html#cb176-464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-465"><a href="rna-seq-mkdata.html#cb176-465" aria-hidden="true" tabindex="-1"></a>dds3 <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds3)</span>
<span id="cb176-466"><a href="rna-seq-mkdata.html#cb176-466" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds3)</span>
<span id="cb176-467"><a href="rna-seq-mkdata.html#cb176-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-468"><a href="rna-seq-mkdata.html#cb176-468" aria-hidden="true" tabindex="-1"></a><span class="co"># have a peek at the DESeqResults object </span></span>
<span id="cb176-469"><a href="rna-seq-mkdata.html#cb176-469" aria-hidden="true" tabindex="-1"></a>res</span>
<span id="cb176-470"><a href="rna-seq-mkdata.html#cb176-470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-471"><a href="rna-seq-mkdata.html#cb176-471" aria-hidden="true" tabindex="-1"></a>res2 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds3, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">&quot;Genotype&quot;</span>,                      <span class="co"># name of the factor</span></span>
<span id="cb176-472"><a href="rna-seq-mkdata.html#cb176-472" aria-hidden="true" tabindex="-1"></a>                                   <span class="st">&quot;Ko&quot;</span>,    <span class="co"># name of the numerator level for fold change</span></span>
<span id="cb176-473"><a href="rna-seq-mkdata.html#cb176-473" aria-hidden="true" tabindex="-1"></a>                                   <span class="st">&quot;WT&quot;</span>))                          <span class="co"># name of the denominator level    </span></span>
<span id="cb176-474"><a href="rna-seq-mkdata.html#cb176-474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-475"><a href="rna-seq-mkdata.html#cb176-475" aria-hidden="true" tabindex="-1"></a><span class="fu">all_equal</span>(res, res2)</span>
<span id="cb176-476"><a href="rna-seq-mkdata.html#cb176-476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb176-477"><a href="rna-seq-mkdata.html#cb176-477" aria-hidden="true" tabindex="-1"></a><span class="fu">mcols</span>(res)</span></code></pre></div>
</div>
<div id="false-discovery-rates-fdr-benjamini-hocheberg-method" class="section level2" number="5.4">
<h2><span class="header-section-number">5.4</span> False discovery rates FDR / Benjamini-Hocheberg method</h2>
<div class="sourceCode" id="cb177"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb177-1"><a href="rna-seq-mkdata.html#cb177-1" aria-hidden="true" tabindex="-1"></a><span class="co"># False discovery rates FDR / Benjamini-Hocheberg method ----------</span></span>
<span id="cb177-2"><a href="rna-seq-mkdata.html#cb177-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Explanation of this method here: https://youtu.be/K8LQSvtjcEo </span></span>
<span id="cb177-3"><a href="rna-seq-mkdata.html#cb177-3" aria-hidden="true" tabindex="-1"></a><span class="co"># We can count the number of genes that are differentially regulated at a certain \(\alpha\) level.</span></span>
<span id="cb177-4"><a href="rna-seq-mkdata.html#cb177-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-5"><a href="rna-seq-mkdata.html#cb177-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-6"><a href="rna-seq-mkdata.html#cb177-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb177-7"><a href="rna-seq-mkdata.html#cb177-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-8"><a href="rna-seq-mkdata.html#cb177-8" aria-hidden="true" tabindex="-1"></a><span class="co"># threshold of p = 0.01</span></span>
<span id="cb177-9"><a href="rna-seq-mkdata.html#cb177-9" aria-hidden="true" tabindex="-1"></a>res <span class="sc">%&gt;%</span> </span>
<span id="cb177-10"><a href="rna-seq-mkdata.html#cb177-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb177-11"><a href="rna-seq-mkdata.html#cb177-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb177-12"><a href="rna-seq-mkdata.html#cb177-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dim</span>()</span>
<span id="cb177-13"><a href="rna-seq-mkdata.html#cb177-13" aria-hidden="true" tabindex="-1"></a><span class="co"># 2347 genes out of 6 samples</span></span>
<span id="cb177-14"><a href="rna-seq-mkdata.html#cb177-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-15"><a href="rna-seq-mkdata.html#cb177-15" aria-hidden="true" tabindex="-1"></a><span class="dv">2347</span><span class="sc">*</span><span class="dv">100</span><span class="sc">/</span><span class="fu">nrow</span>(counts)   <span class="co"># 6.977643 or around ~7%</span></span>
<span id="cb177-16"><a href="rna-seq-mkdata.html#cb177-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-17"><a href="rna-seq-mkdata.html#cb177-17" aria-hidden="true" tabindex="-1"></a><span class="co"># threshold of p = 0.001</span></span>
<span id="cb177-18"><a href="rna-seq-mkdata.html#cb177-18" aria-hidden="true" tabindex="-1"></a>FDR<span class="fl">.001</span> <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span> </span>
<span id="cb177-19"><a href="rna-seq-mkdata.html#cb177-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb177-20"><a href="rna-seq-mkdata.html#cb177-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.001</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb177-21"><a href="rna-seq-mkdata.html#cb177-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dim</span>()</span>
<span id="cb177-22"><a href="rna-seq-mkdata.html#cb177-22" aria-hidden="true" tabindex="-1"></a>FDR<span class="fl">.001</span></span>
<span id="cb177-23"><a href="rna-seq-mkdata.html#cb177-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-24"><a href="rna-seq-mkdata.html#cb177-24" aria-hidden="true" tabindex="-1"></a>FDR<span class="fl">.001</span><span class="sc">*</span><span class="dv">100</span><span class="sc">/</span><span class="fu">nrow</span>(counts)   <span class="co"># 4.83410631 or around ~5%</span></span>
<span id="cb177-25"><a href="rna-seq-mkdata.html#cb177-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-26"><a href="rna-seq-mkdata.html#cb177-26" aria-hidden="true" tabindex="-1"></a><span class="co">#　You should obtain 2347 differentially expressed genes at 1% and 1626 at 0.1% which are quite important numbers: indeed, it corresponds to respectively ~7% and ~5% of the whole number transcriptome (nrow(counts)　=　total number of mRNA is 33636).</span></span>
<span id="cb177-27"><a href="rna-seq-mkdata.html#cb177-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-28"><a href="rna-seq-mkdata.html#cb177-28" aria-hidden="true" tabindex="-1"></a><span class="co"># distribution of adjusted p-values</span></span>
<span id="cb177-29"><a href="rna-seq-mkdata.html#cb177-29" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(res<span class="sc">$</span>padj, <span class="at">col=</span><span class="st">&quot;lightblue&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Adjusted p-value distribution&quot;</span>)</span>
<span id="cb177-30"><a href="rna-seq-mkdata.html#cb177-30" aria-hidden="true" tabindex="-1"></a><span class="co"># distribution of non-adjusted p-values</span></span>
<span id="cb177-31"><a href="rna-seq-mkdata.html#cb177-31" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(res<span class="sc">$</span>pvalue, <span class="at">col=</span><span class="st">&quot;grey&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Non-adjusted p-value distribution&quot;</span>)</span>
<span id="cb177-32"><a href="rna-seq-mkdata.html#cb177-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-33"><a href="rna-seq-mkdata.html#cb177-33" aria-hidden="true" tabindex="-1"></a><span class="co"># As you can see, the distribution of p-values was already quite similar suggesting that a good proportion of the tests have a significant p-value (inferior to \(\alpha\) = 0.01 for instance). This suggests that a good proportion of these will be true positives (genes truly differentially regulated).</span></span>
<span id="cb177-34"><a href="rna-seq-mkdata.html#cb177-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-35"><a href="rna-seq-mkdata.html#cb177-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Extracting the table of differential genes ------------</span></span>
<span id="cb177-36"><a href="rna-seq-mkdata.html#cb177-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-37"><a href="rna-seq-mkdata.html#cb177-37" aria-hidden="true" tabindex="-1"></a>diff <span class="ot">=</span> res <span class="sc">%&gt;%</span> </span>
<span id="cb177-38"><a href="rna-seq-mkdata.html#cb177-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb177-39"><a href="rna-seq-mkdata.html#cb177-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;genes&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb177-40"><a href="rna-seq-mkdata.html#cb177-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb177-41"><a href="rna-seq-mkdata.html#cb177-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(log2FoldChange), </span>
<span id="cb177-42"><a href="rna-seq-mkdata.html#cb177-42" aria-hidden="true" tabindex="-1"></a>          <span class="fu">desc</span>(padj))</span>
<span id="cb177-43"><a href="rna-seq-mkdata.html#cb177-43" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(diff)</span>
<span id="cb177-44"><a href="rna-seq-mkdata.html#cb177-44" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(diff)</span>
<span id="cb177-45"><a href="rna-seq-mkdata.html#cb177-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-46"><a href="rna-seq-mkdata.html#cb177-46" aria-hidden="true" tabindex="-1"></a><span class="co"># Save differential genes table in a file. --------</span></span>
<span id="cb177-47"><a href="rna-seq-mkdata.html#cb177-47" aria-hidden="true" tabindex="-1"></a><span class="co"># # .csv file. </span></span>
<span id="cb177-48"><a href="rna-seq-mkdata.html#cb177-48" aria-hidden="true" tabindex="-1"></a><span class="co"># write.table(diff, file =  &quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/diff.csv&quot;, sep = &quot;,&quot;, row.names = F)</span></span>
<span id="cb177-49"><a href="rna-seq-mkdata.html#cb177-49" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb177-50"><a href="rna-seq-mkdata.html#cb177-50" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb177-51"><a href="rna-seq-mkdata.html#cb177-51" aria-hidden="true" tabindex="-1"></a><span class="co"># #Excel file. Source: http://www.sthda.com/english/wiki/writing-data-from-r-to-excel-files-xls-xlsx</span></span>
<span id="cb177-52"><a href="rna-seq-mkdata.html#cb177-52" aria-hidden="true" tabindex="-1"></a><span class="co"># library(&quot;xlsx&quot;)</span></span>
<span id="cb177-53"><a href="rna-seq-mkdata.html#cb177-53" aria-hidden="true" tabindex="-1"></a><span class="co"># # Write the first data set in a new workbook</span></span>
<span id="cb177-54"><a href="rna-seq-mkdata.html#cb177-54" aria-hidden="true" tabindex="-1"></a><span class="co"># write.xlsx(diff, file = &quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/diff.xlsx&quot;,</span></span>
<span id="cb177-55"><a href="rna-seq-mkdata.html#cb177-55" aria-hidden="true" tabindex="-1"></a><span class="co">#            sheetName = &quot;diff1&quot;, row.names = F, append = FALSE)</span></span>
<span id="cb177-56"><a href="rna-seq-mkdata.html#cb177-56" aria-hidden="true" tabindex="-1"></a><span class="co"># # Add a second data set in a new worksheet</span></span>
<span id="cb177-57"><a href="rna-seq-mkdata.html#cb177-57" aria-hidden="true" tabindex="-1"></a><span class="co"># write.xlsx(diff2, file = &quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/Top10DEgenes_prcomp.xlsx&quot;,</span></span>
<span id="cb177-58"><a href="rna-seq-mkdata.html#cb177-58" aria-hidden="true" tabindex="-1"></a><span class="co">#            sheetName = &quot;diff2&quot;, row.names = F, append = TRUE) # to add as a new sheet.</span></span>
<span id="cb177-59"><a href="rna-seq-mkdata.html#cb177-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-60"><a href="rna-seq-mkdata.html#cb177-60" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a third data set..... I AM ADDING THIS SHEET TO AN EXISTING FILE FROM THE PREVIOUS PCA ANALISYS.</span></span>
<span id="cb177-61"><a href="rna-seq-mkdata.html#cb177-61" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(diff, <span class="at">file =</span> <span class="st">&quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/Top10DEgenes_prcomp.xlsx&quot;</span>,</span>
<span id="cb177-62"><a href="rna-seq-mkdata.html#cb177-62" aria-hidden="true" tabindex="-1"></a>           <span class="at">sheetName =</span> <span class="st">&quot;diff1&quot;</span>, <span class="at">row.names =</span> F, <span class="at">append =</span> <span class="cn">TRUE</span>) <span class="co"># to add as a new sheet.</span></span>
<span id="cb177-63"><a href="rna-seq-mkdata.html#cb177-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-64"><a href="rna-seq-mkdata.html#cb177-64" aria-hidden="true" tabindex="-1"></a><span class="co"># # Physically opening an excel file within R (maybe Windows only)</span></span>
<span id="cb177-65"><a href="rna-seq-mkdata.html#cb177-65" aria-hidden="true" tabindex="-1"></a><span class="co"># Top10DEgenes_prcomp.xlsx_path &lt;- &quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/Top10DEgenes_prcomp.xlsx&quot;</span></span>
<span id="cb177-66"><a href="rna-seq-mkdata.html#cb177-66" aria-hidden="true" tabindex="-1"></a><span class="co"># command &lt;- paste(&quot;open excel&quot;, Top10DEgenes_prcomp.xlsx_path)</span></span>
<span id="cb177-67"><a href="rna-seq-mkdata.html#cb177-67" aria-hidden="true" tabindex="-1"></a><span class="co"># system(command)</span></span>
<span id="cb177-68"><a href="rna-seq-mkdata.html#cb177-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-69"><a href="rna-seq-mkdata.html#cb177-69" aria-hidden="true" tabindex="-1"></a><span class="co"># # install.packages(&quot;writexl&quot;)</span></span>
<span id="cb177-70"><a href="rna-seq-mkdata.html#cb177-70" aria-hidden="true" tabindex="-1"></a><span class="co"># library(&quot;writexl&quot;)</span></span>
<span id="cb177-71"><a href="rna-seq-mkdata.html#cb177-71" aria-hidden="true" tabindex="-1"></a><span class="co"># write_xlsx(TopDEgenes_prcomp, &quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/all_raw_count01.xlsx&quot;, sheet= top10_genes)</span></span></code></pre></div>
</div>
<div id="volcano-plot" class="section level2" number="5.5">
<h2><span class="header-section-number">5.5</span> Volcano plot</h2>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="rna-seq-mkdata.html#cb178-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Volcano plot -----------</span></span>
<span id="cb178-2"><a href="rna-seq-mkdata.html#cb178-2" aria-hidden="true" tabindex="-1"></a><span class="co"># First, we are going to “shrink” the \(\log2\) fold changes to remove the noise associated with fold changes coming from genes with low count levels. Shrinkage of effect size (LFC estimates) is useful for visualization and ranking of genes. This helps to get more meaningful log2 fold changes for all genes independently of their expression level.</span></span>
<span id="cb178-3"><a href="rna-seq-mkdata.html#cb178-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-4"><a href="rna-seq-mkdata.html#cb178-4" aria-hidden="true" tabindex="-1"></a>resLFC <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(<span class="at">dds =</span> dds3, </span>
<span id="cb178-5"><a href="rna-seq-mkdata.html#cb178-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">res =</span> res,</span>
<span id="cb178-6"><a href="rna-seq-mkdata.html#cb178-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">type =</span> <span class="st">&quot;normal&quot;</span>,</span>
<span id="cb178-7"><a href="rna-seq-mkdata.html#cb178-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">coef =</span> <span class="dv">2</span>) <span class="co"># corresponds to &quot;WT vs Ko&quot; comparison</span></span>
<span id="cb178-8"><a href="rna-seq-mkdata.html#cb178-8" aria-hidden="true" tabindex="-1"></a>resLFC</span>
<span id="cb178-9"><a href="rna-seq-mkdata.html#cb178-9" aria-hidden="true" tabindex="-1"></a><span class="co"># EnhancedVolcano Installation</span></span>
<span id="cb178-10"><a href="rna-seq-mkdata.html#cb178-10" aria-hidden="true" tabindex="-1"></a><span class="co"># if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))</span></span>
<span id="cb178-11"><a href="rna-seq-mkdata.html#cb178-11" aria-hidden="true" tabindex="-1"></a><span class="co">#  install.packages(&quot;BiocManager&quot;)</span></span>
<span id="cb178-12"><a href="rna-seq-mkdata.html#cb178-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-13"><a href="rna-seq-mkdata.html#cb178-13" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install(&quot;EnhancedVolcano&quot;)</span></span>
<span id="cb178-14"><a href="rna-seq-mkdata.html#cb178-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-15"><a href="rna-seq-mkdata.html#cb178-15" aria-hidden="true" tabindex="-1"></a><span class="co"># load the library if not done yet</span></span>
<span id="cb178-16"><a href="rna-seq-mkdata.html#cb178-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;EnhancedVolcano&quot;</span>)</span>
<span id="cb178-17"><a href="rna-seq-mkdata.html#cb178-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-18"><a href="rna-seq-mkdata.html#cb178-18" aria-hidden="true" tabindex="-1"></a><span class="co"># The main function is named after the package</span></span>
<span id="cb178-19"><a href="rna-seq-mkdata.html#cb178-19" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(<span class="at">toptable =</span> resLFC,              <span class="co"># We use the shrunken log2 fold change as noise associated with low count genes is removed </span></span>
<span id="cb178-20"><a href="rna-seq-mkdata.html#cb178-20" aria-hidden="true" tabindex="-1"></a>                <span class="at">x =</span> <span class="st">&quot;log2FoldChange&quot;</span>,           <span class="co"># Name of the column in resLFC that contains the log2 fold changes</span></span>
<span id="cb178-21"><a href="rna-seq-mkdata.html#cb178-21" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span> <span class="st">&quot;padj&quot;</span>,                     <span class="co"># Name of the column in resLFC that contains the p-value</span></span>
<span id="cb178-22"><a href="rna-seq-mkdata.html#cb178-22" aria-hidden="true" tabindex="-1"></a>                <span class="at">lab =</span> <span class="fu">rownames</span>(resLFC)</span>
<span id="cb178-23"><a href="rna-seq-mkdata.html#cb178-23" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb178-24"><a href="rna-seq-mkdata.html#cb178-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-25"><a href="rna-seq-mkdata.html#cb178-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-26"><a href="rna-seq-mkdata.html#cb178-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb178-27"><a href="rna-seq-mkdata.html#cb178-27" aria-hidden="true" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(<span class="at">toptable =</span> resLFC,</span>
<span id="cb178-28"><a href="rna-seq-mkdata.html#cb178-28" aria-hidden="true" tabindex="-1"></a>                <span class="at">x =</span> <span class="st">&quot;log2FoldChange&quot;</span>,</span>
<span id="cb178-29"><a href="rna-seq-mkdata.html#cb178-29" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span> <span class="st">&quot;padj&quot;</span>,</span>
<span id="cb178-30"><a href="rna-seq-mkdata.html#cb178-30" aria-hidden="true" tabindex="-1"></a>                <span class="at">lab =</span> <span class="fu">rownames</span>(resLFC),</span>
<span id="cb178-31"><a href="rna-seq-mkdata.html#cb178-31" aria-hidden="true" tabindex="-1"></a>                <span class="at">labSize =</span> <span class="dv">3</span>,</span>
<span id="cb178-32"><a href="rna-seq-mkdata.html#cb178-32" aria-hidden="true" tabindex="-1"></a>                <span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="sc">+</span><span class="dv">10</span>),</span>
<span id="cb178-33"><a href="rna-seq-mkdata.html#cb178-33" aria-hidden="true" tabindex="-1"></a>                <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>),</span>
<span id="cb178-34"><a href="rna-seq-mkdata.html#cb178-34" aria-hidden="true" tabindex="-1"></a>                <span class="at">pCutoff =</span> <span class="fl">1e-06</span>,</span>
<span id="cb178-35"><a href="rna-seq-mkdata.html#cb178-35" aria-hidden="true" tabindex="-1"></a>                <span class="at">pointSize =</span> <span class="fl">2.0</span>,</span>
<span id="cb178-36"><a href="rna-seq-mkdata.html#cb178-36" aria-hidden="true" tabindex="-1"></a>                <span class="at">FCcutoff =</span> <span class="dv">2</span>, </span>
<span id="cb178-37"><a href="rna-seq-mkdata.html#cb178-37" aria-hidden="true" tabindex="-1"></a>                <span class="at">title =</span> <span class="st">&quot;WT vs Ko comparison (fold change cutoff = 2, p-value cutoff = 1e-06)&quot;</span>,</span>
<span id="cb178-38"><a href="rna-seq-mkdata.html#cb178-38" aria-hidden="true" tabindex="-1"></a>                <span class="at">subtitle =</span> <span class="st">&#39;Differential expression&#39;</span>,</span>
<span id="cb178-39"><a href="rna-seq-mkdata.html#cb178-39" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="heatmap-normalization" class="section level2" number="5.6">
<h2><span class="header-section-number">5.6</span> Heatmap &amp; normalization</h2>
<div class="sourceCode" id="cb179"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb179-1"><a href="rna-seq-mkdata.html#cb179-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Heatmap --------------------</span></span>
<span id="cb179-2"><a href="rna-seq-mkdata.html#cb179-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-3"><a href="rna-seq-mkdata.html#cb179-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb179-4"><a href="rna-seq-mkdata.html#cb179-4" aria-hidden="true" tabindex="-1"></a><span class="co"># df &lt;- scale(mtcars)</span></span>
<span id="cb179-5"><a href="rna-seq-mkdata.html#cb179-5" aria-hidden="true" tabindex="-1"></a><span class="co"># pheatmap(df)</span></span>
<span id="cb179-6"><a href="rna-seq-mkdata.html#cb179-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-7"><a href="rna-seq-mkdata.html#cb179-7" aria-hidden="true" tabindex="-1"></a><span class="co"># import custom function ------</span></span>
<span id="cb179-8"><a href="rna-seq-mkdata.html#cb179-8" aria-hidden="true" tabindex="-1"></a><span class="co"># source(&quot;normalization_function.R&quot;) # cant import, so I pasted below. mor_normalization.</span></span>
<span id="cb179-9"><a href="rna-seq-mkdata.html#cb179-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-10"><a href="rna-seq-mkdata.html#cb179-10" aria-hidden="true" tabindex="-1"></a>mor_normalization <span class="ot">=</span> <span class="cf">function</span>(data){</span>
<span id="cb179-11"><a href="rna-seq-mkdata.html#cb179-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(dplyr)</span>
<span id="cb179-12"><a href="rna-seq-mkdata.html#cb179-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(tibble)</span>
<span id="cb179-13"><a href="rna-seq-mkdata.html#cb179-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb179-14"><a href="rna-seq-mkdata.html#cb179-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># take the log</span></span>
<span id="cb179-15"><a href="rna-seq-mkdata.html#cb179-15" aria-hidden="true" tabindex="-1"></a>  log_data <span class="ot">=</span> <span class="fu">log</span>(data) </span>
<span id="cb179-16"><a href="rna-seq-mkdata.html#cb179-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb179-17"><a href="rna-seq-mkdata.html#cb179-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># find the psuedo-references per sample by taking the geometric mean</span></span>
<span id="cb179-18"><a href="rna-seq-mkdata.html#cb179-18" aria-hidden="true" tabindex="-1"></a>  log_data <span class="ot">=</span> log_data <span class="sc">%&gt;%</span> </span>
<span id="cb179-19"><a href="rna-seq-mkdata.html#cb179-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="st">&#39;gene&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb179-20"><a href="rna-seq-mkdata.html#cb179-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span> (<span class="at">gene_averages =</span> <span class="fu">rowMeans</span>(log_data)) <span class="sc">%&gt;%</span> </span>
<span id="cb179-21"><a href="rna-seq-mkdata.html#cb179-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(gene_averages <span class="sc">!=</span> <span class="st">&quot;-Inf&quot;</span>)</span>
<span id="cb179-22"><a href="rna-seq-mkdata.html#cb179-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb179-23"><a href="rna-seq-mkdata.html#cb179-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># the last columns is the pseudo-reference column </span></span>
<span id="cb179-24"><a href="rna-seq-mkdata.html#cb179-24" aria-hidden="true" tabindex="-1"></a>  pseudo_column <span class="ot">=</span> <span class="fu">ncol</span>(log_data)</span>
<span id="cb179-25"><a href="rna-seq-mkdata.html#cb179-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb179-26"><a href="rna-seq-mkdata.html#cb179-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># where to stop before the pseudo column </span></span>
<span id="cb179-27"><a href="rna-seq-mkdata.html#cb179-27" aria-hidden="true" tabindex="-1"></a>  before_pseduo <span class="ot">=</span> pseudo_column <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb179-28"><a href="rna-seq-mkdata.html#cb179-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb179-29"><a href="rna-seq-mkdata.html#cb179-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># find the ratio of the log data to the pseudo-reference</span></span>
<span id="cb179-30"><a href="rna-seq-mkdata.html#cb179-30" aria-hidden="true" tabindex="-1"></a>  ratios <span class="ot">=</span> <span class="fu">sweep</span>(log_data[,<span class="dv">2</span><span class="sc">:</span>before_pseduo], <span class="dv">1</span>, log_data[,pseudo_column], <span class="st">&quot;-&quot;</span>)</span>
<span id="cb179-31"><a href="rna-seq-mkdata.html#cb179-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb179-32"><a href="rna-seq-mkdata.html#cb179-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># find the median of the ratios</span></span>
<span id="cb179-33"><a href="rna-seq-mkdata.html#cb179-33" aria-hidden="true" tabindex="-1"></a>  sample_medians <span class="ot">=</span> <span class="fu">apply</span>(ratios, <span class="dv">2</span>, median)</span>
<span id="cb179-34"><a href="rna-seq-mkdata.html#cb179-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb179-35"><a href="rna-seq-mkdata.html#cb179-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># convert the median to a scaling factor</span></span>
<span id="cb179-36"><a href="rna-seq-mkdata.html#cb179-36" aria-hidden="true" tabindex="-1"></a>  scaling_factors <span class="ot">=</span> <span class="fu">exp</span>(sample_medians)</span>
<span id="cb179-37"><a href="rna-seq-mkdata.html#cb179-37" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb179-38"><a href="rna-seq-mkdata.html#cb179-38" aria-hidden="true" tabindex="-1"></a>  <span class="co"># use scaling factors to scale the original data</span></span>
<span id="cb179-39"><a href="rna-seq-mkdata.html#cb179-39" aria-hidden="true" tabindex="-1"></a>  manually_normalized <span class="ot">=</span> <span class="fu">sweep</span>(data, <span class="dv">2</span>, scaling_factors, <span class="st">&quot;/&quot;</span>)</span>
<span id="cb179-40"><a href="rna-seq-mkdata.html#cb179-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(manually_normalized)</span>
<span id="cb179-41"><a href="rna-seq-mkdata.html#cb179-41" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb179-42"><a href="rna-seq-mkdata.html#cb179-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-43"><a href="rna-seq-mkdata.html#cb179-43" aria-hidden="true" tabindex="-1"></a><span class="co"># First version</span></span>
<span id="cb179-44"><a href="rna-seq-mkdata.html#cb179-44" aria-hidden="true" tabindex="-1"></a>counts_normalised_only_diff_genes <span class="ot">=</span> </span>
<span id="cb179-45"><a href="rna-seq-mkdata.html#cb179-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mor_normalization</span>(counts) <span class="sc">%&gt;%</span>                 <span class="co"># normalize the counts using our custom function</span></span>
<span id="cb179-46"><a href="rna-seq-mkdata.html#cb179-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;genes&quot;</span>) <span class="sc">%&gt;%</span>               </span>
<span id="cb179-47"><a href="rna-seq-mkdata.html#cb179-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span> genes,                         <span class="co"># pivot_longer() is an update version of gather() since tidyr version 1.0</span></span>
<span id="cb179-48"><a href="rna-seq-mkdata.html#cb179-48" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">&quot;sample&quot;</span>, </span>
<span id="cb179-49"><a href="rna-seq-mkdata.html#cb179-49" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">&quot;counts&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb179-50"><a href="rna-seq-mkdata.html#cb179-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(genes <span class="sc">%in%</span> diff<span class="sc">$</span>genes) <span class="sc">%&gt;%</span>             </span>
<span id="cb179-51"><a href="rna-seq-mkdata.html#cb179-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">&quot;sample&quot;</span>,            <span class="co"># pivot_wider() is an updated approach to spread()</span></span>
<span id="cb179-52"><a href="rna-seq-mkdata.html#cb179-52" aria-hidden="true" tabindex="-1"></a>              <span class="at">values_from =</span> <span class="st">&quot;counts&quot;</span>)  <span class="sc">%&gt;%</span>      </span>
<span id="cb179-53"><a href="rna-seq-mkdata.html#cb179-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;genes&quot;</span>)                   <span class="co"># the gene column is converted back to row names to create a matrix usable with pheatmap</span></span>
<span id="cb179-54"><a href="rna-seq-mkdata.html#cb179-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-55"><a href="rna-seq-mkdata.html#cb179-55" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(counts_normalised_only_diff_genes)          <span class="co"># check that you have the expected number of rows and columns</span></span>
<span id="cb179-56"><a href="rna-seq-mkdata.html#cb179-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-57"><a href="rna-seq-mkdata.html#cb179-57" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(counts_normalised_only_diff_genes, </span>
<span id="cb179-58"><a href="rna-seq-mkdata.html#cb179-58" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>, </span>
<span id="cb179-59"><a href="rna-seq-mkdata.html#cb179-59" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>, </span>
<span id="cb179-60"><a href="rna-seq-mkdata.html#cb179-60" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb179-61"><a href="rna-seq-mkdata.html#cb179-61" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb179-62"><a href="rna-seq-mkdata.html#cb179-62" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">TRUE</span>)</span>
<span id="cb179-63"><a href="rna-seq-mkdata.html#cb179-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-64"><a href="rna-seq-mkdata.html#cb179-64" aria-hidden="true" tabindex="-1"></a><span class="co"># Heatmap is all blue!! WHY?!!!!!!!!!!!!!</span></span>
<span id="cb179-65"><a href="rna-seq-mkdata.html#cb179-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-66"><a href="rna-seq-mkdata.html#cb179-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-67"><a href="rna-seq-mkdata.html#cb179-67" aria-hidden="true" tabindex="-1"></a><span class="co"># distribution of values in the counts_normalised_only_diff_genes table:</span></span>
<span id="cb179-68"><a href="rna-seq-mkdata.html#cb179-68" aria-hidden="true" tabindex="-1"></a>counts_normalised_only_diff_genes <span class="sc">%&gt;%</span> </span>
<span id="cb179-69"><a href="rna-seq-mkdata.html#cb179-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;genes&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb179-70"><a href="rna-seq-mkdata.html#cb179-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span> genes, <span class="at">names_to =</span> <span class="st">&quot;sample&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;counts&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb179-71"><a href="rna-seq-mkdata.html#cb179-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">with</span>(., <span class="fu">hist</span>(counts, <span class="at">col =</span> <span class="st">&quot;dodgerblue&quot;</span>))</span>
<span id="cb179-72"><a href="rna-seq-mkdata.html#cb179-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-73"><a href="rna-seq-mkdata.html#cb179-73" aria-hidden="true" tabindex="-1"></a><span class="co"># The scale on which gene counts are represented is the (main) issue here.</span></span>
<span id="cb179-74"><a href="rna-seq-mkdata.html#cb179-74" aria-hidden="true" tabindex="-1"></a><span class="co"># There are a lot of genes for which the number of counts are very low. If you \(\log2\) transform the counts, you get a nice distribution again where all gene frequency become visible again.</span></span>
<span id="cb179-75"><a href="rna-seq-mkdata.html#cb179-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-76"><a href="rna-seq-mkdata.html#cb179-76" aria-hidden="true" tabindex="-1"></a>counts_normalised_only_diff_genes <span class="sc">%&gt;%</span> </span>
<span id="cb179-77"><a href="rna-seq-mkdata.html#cb179-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;genes&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb179-78"><a href="rna-seq-mkdata.html#cb179-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span> genes,<span class="at">names_to =</span> <span class="st">&quot;sample&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;counts&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb179-79"><a href="rna-seq-mkdata.html#cb179-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">with</span>(., <span class="fu">hist</span>(<span class="fu">log2</span>(counts), <span class="at">col =</span> <span class="st">&quot;dodgerblue&quot;</span>))</span>
<span id="cb179-80"><a href="rna-seq-mkdata.html#cb179-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-81"><a href="rna-seq-mkdata.html#cb179-81" aria-hidden="true" tabindex="-1"></a><span class="co"># If you re-run the code for the first heatmap with a log2 transformation, you will get a simple way to display different gene count levels. We add + 1 to account for genes with count values equal to 0.</span></span>
<span id="cb179-82"><a href="rna-seq-mkdata.html#cb179-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-83"><a href="rna-seq-mkdata.html#cb179-83" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(<span class="fu">log2</span>(counts_normalised_only_diff_genes <span class="sc">+</span> <span class="dv">1</span>), </span>
<span id="cb179-84"><a href="rna-seq-mkdata.html#cb179-84" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>, </span>
<span id="cb179-85"><a href="rna-seq-mkdata.html#cb179-85" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>, </span>
<span id="cb179-86"><a href="rna-seq-mkdata.html#cb179-86" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb179-87"><a href="rna-seq-mkdata.html#cb179-87" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb179-88"><a href="rna-seq-mkdata.html#cb179-88" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">TRUE</span>)</span>
<span id="cb179-89"><a href="rna-seq-mkdata.html#cb179-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-90"><a href="rna-seq-mkdata.html#cb179-90" aria-hidden="true" tabindex="-1"></a><span class="co"># Second version with scaling</span></span>
<span id="cb179-91"><a href="rna-seq-mkdata.html#cb179-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-92"><a href="rna-seq-mkdata.html#cb179-92" aria-hidden="true" tabindex="-1"></a><span class="co">#Scaling trial </span></span>
<span id="cb179-93"><a href="rna-seq-mkdata.html#cb179-93" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb179-94"><a href="rna-seq-mkdata.html#cb179-94" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">runif</span>(<span class="dv">10</span>)</span>
<span id="cb179-95"><a href="rna-seq-mkdata.html#cb179-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-96"><a href="rna-seq-mkdata.html#cb179-96" aria-hidden="true" tabindex="-1"></a><span class="co"># Manually scaling</span></span>
<span id="cb179-97"><a href="rna-seq-mkdata.html#cb179-97" aria-hidden="true" tabindex="-1"></a>(x <span class="sc">-</span> <span class="fu">mean</span>(x)) <span class="sc">/</span> <span class="fu">sd</span>(x)</span>
<span id="cb179-98"><a href="rna-seq-mkdata.html#cb179-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-99"><a href="rna-seq-mkdata.html#cb179-99" aria-hidden="true" tabindex="-1"></a><span class="co"># With scale function</span></span>
<span id="cb179-100"><a href="rna-seq-mkdata.html#cb179-100" aria-hidden="true" tabindex="-1"></a><span class="fu">scale</span>(x)[,<span class="dv">1</span>]</span>
<span id="cb179-101"><a href="rna-seq-mkdata.html#cb179-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-102"><a href="rna-seq-mkdata.html#cb179-102" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(x)</span>
<span id="cb179-103"><a href="rna-seq-mkdata.html#cb179-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-104"><a href="rna-seq-mkdata.html#cb179-104" aria-hidden="true" tabindex="-1"></a>counts_scaled <span class="ot">=</span> </span>
<span id="cb179-105"><a href="rna-seq-mkdata.html#cb179-105" aria-hidden="true" tabindex="-1"></a>  counts_normalised_only_diff_genes <span class="sc">%&gt;%</span> </span>
<span id="cb179-106"><a href="rna-seq-mkdata.html#cb179-106" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>(.) <span class="sc">%&gt;%</span>                              <span class="co"># transpose to have the genes in columns </span></span>
<span id="cb179-107"><a href="rna-seq-mkdata.html#cb179-107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale</span>() <span class="sc">%&gt;%</span>                           <span class="co"># scale(x, center = TRUE, scale = TRUE) </span></span>
<span id="cb179-108"><a href="rna-seq-mkdata.html#cb179-108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>(.)                                  <span class="co"># back in original shape</span></span>
<span id="cb179-109"><a href="rna-seq-mkdata.html#cb179-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-110"><a href="rna-seq-mkdata.html#cb179-110" aria-hidden="true" tabindex="-1"></a><span class="co"># sanity check</span></span>
<span id="cb179-111"><a href="rna-seq-mkdata.html#cb179-111" aria-hidden="true" tabindex="-1"></a><span class="co"># the majority of the values should be around zero</span></span>
<span id="cb179-112"><a href="rna-seq-mkdata.html#cb179-112" aria-hidden="true" tabindex="-1"></a><span class="fu">apply</span>(counts_scaled, <span class="at">MARGIN =</span> <span class="dv">1</span>, mean) <span class="sc">%&gt;%</span>                          <span class="co"># calculate the mean per row</span></span>
<span id="cb179-113"><a href="rna-seq-mkdata.html#cb179-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hist</span>(., <span class="at">main =</span> <span class="st">&quot;&quot;</span>, <span class="at">xlab =</span> <span class="st">&quot;Z-score values&quot;</span>, <span class="at">col =</span> <span class="st">&quot;dodgerblue2&quot;</span>)  </span>
<span id="cb179-114"><a href="rna-seq-mkdata.html#cb179-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-115"><a href="rna-seq-mkdata.html#cb179-115" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(counts_scaled, </span>
<span id="cb179-116"><a href="rna-seq-mkdata.html#cb179-116" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>, </span>
<span id="cb179-117"><a href="rna-seq-mkdata.html#cb179-117" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>, </span>
<span id="cb179-118"><a href="rna-seq-mkdata.html#cb179-118" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb179-119"><a href="rna-seq-mkdata.html#cb179-119" aria-hidden="true" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">&quot;none&quot;</span>,            <span class="co"># already done &quot;manually&quot;</span></span>
<span id="cb179-120"><a href="rna-seq-mkdata.html#cb179-120" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">TRUE</span>)</span>
<span id="cb179-121"><a href="rna-seq-mkdata.html#cb179-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-122"><a href="rna-seq-mkdata.html#cb179-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-123"><a href="rna-seq-mkdata.html#cb179-123" aria-hidden="true" tabindex="-1"></a><span class="co"># Third version with genes and samples grouped by profiles</span></span>
<span id="cb179-124"><a href="rna-seq-mkdata.html#cb179-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-125"><a href="rna-seq-mkdata.html#cb179-125" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(counts_scaled, </span>
<span id="cb179-126"><a href="rna-seq-mkdata.html#cb179-126" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,                      </span>
<span id="cb179-127"><a href="rna-seq-mkdata.html#cb179-127" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>, </span>
<span id="cb179-128"><a href="rna-seq-mkdata.html#cb179-128" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb179-129"><a href="rna-seq-mkdata.html#cb179-129" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">TRUE</span>,</span>
<span id="cb179-130"><a href="rna-seq-mkdata.html#cb179-130" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">&quot;Clustering on&quot;</span>)</span>
<span id="cb179-131"><a href="rna-seq-mkdata.html#cb179-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-132"><a href="rna-seq-mkdata.html#cb179-132" aria-hidden="true" tabindex="-1"></a><span class="co"># Fourth version of our heatmap with the 8 samples being investigated.</span></span>
<span id="cb179-133"><a href="rna-seq-mkdata.html#cb179-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-134"><a href="rna-seq-mkdata.html#cb179-134" aria-hidden="true" tabindex="-1"></a>counts_scaled_filtered <span class="ot">=</span> </span>
<span id="cb179-135"><a href="rna-seq-mkdata.html#cb179-135" aria-hidden="true" tabindex="-1"></a>  counts_scaled <span class="sc">%&gt;%</span> </span>
<span id="cb179-136"><a href="rna-seq-mkdata.html#cb179-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb179-137"><a href="rna-seq-mkdata.html#cb179-137" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(var_GenetypeKo_vs_Time14<span class="sc">$</span>SampleName) <span class="co"># keep the 8 samples</span></span>
<span id="cb179-138"><a href="rna-seq-mkdata.html#cb179-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-139"><a href="rna-seq-mkdata.html#cb179-139" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(counts_scaled_filtered)</span>
<span id="cb179-140"><a href="rna-seq-mkdata.html#cb179-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-141"><a href="rna-seq-mkdata.html#cb179-141" aria-hidden="true" tabindex="-1"></a><span class="co"># Anotations for the map</span></span>
<span id="cb179-142"><a href="rna-seq-mkdata.html#cb179-142" aria-hidden="true" tabindex="-1"></a>anno_col_info <span class="ot">=</span> var_GenetypeKo_vs_Time14 <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">&quot;SampleName&quot;</span>)</span>
<span id="cb179-143"><a href="rna-seq-mkdata.html#cb179-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-144"><a href="rna-seq-mkdata.html#cb179-144" aria-hidden="true" tabindex="-1"></a>anno_info_colors <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb179-145"><a href="rna-seq-mkdata.html#cb179-145" aria-hidden="true" tabindex="-1"></a>  <span class="at">Tissue =</span> <span class="fu">c</span>(<span class="at">PDL =</span> <span class="st">&quot;#d8b365&quot;</span>),</span>
<span id="cb179-146"><a href="rna-seq-mkdata.html#cb179-146" aria-hidden="true" tabindex="-1"></a>  <span class="at">Genotype =</span> <span class="fu">c</span>(<span class="at">WT =</span> <span class="st">&quot;lightgrey&quot;</span>, </span>
<span id="cb179-147"><a href="rna-seq-mkdata.html#cb179-147" aria-hidden="true" tabindex="-1"></a>               <span class="at">Ko =</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb179-148"><a href="rna-seq-mkdata.html#cb179-148" aria-hidden="true" tabindex="-1"></a>  <span class="at">Time =</span> <span class="fu">c</span>(<span class="st">&quot;14&quot;</span> <span class="ot">=</span> <span class="st">&quot;dodgerblue4&quot;</span>)</span>
<span id="cb179-149"><a href="rna-seq-mkdata.html#cb179-149" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb179-150"><a href="rna-seq-mkdata.html#cb179-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb179-151"><a href="rna-seq-mkdata.html#cb179-151" aria-hidden="true" tabindex="-1"></a><span class="co"># Heatmatp with anotations</span></span>
<span id="cb179-152"><a href="rna-seq-mkdata.html#cb179-152" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(counts_scaled_filtered, </span>
<span id="cb179-153"><a href="rna-seq-mkdata.html#cb179-153" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,                       </span>
<span id="cb179-154"><a href="rna-seq-mkdata.html#cb179-154" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>, </span>
<span id="cb179-155"><a href="rna-seq-mkdata.html#cb179-155" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb179-156"><a href="rna-seq-mkdata.html#cb179-156" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">TRUE</span>,</span>
<span id="cb179-157"><a href="rna-seq-mkdata.html#cb179-157" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_col =</span> anno_col_info,</span>
<span id="cb179-158"><a href="rna-seq-mkdata.html#cb179-158" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_colors =</span> anno_info_colors,</span>
<span id="cb179-159"><a href="rna-seq-mkdata.html#cb179-159" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">&quot;Clustering with ward method&quot;</span>)</span></code></pre></div>
</div>
<div id="ma-plots" class="section level2" number="5.7">
<h2><span class="header-section-number">5.7</span> MA plots</h2>
<div class="sourceCode" id="cb180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb180-1"><a href="rna-seq-mkdata.html#cb180-1" aria-hidden="true" tabindex="-1"></a><span class="co"># MA plots -------</span></span>
<span id="cb180-2"><a href="rna-seq-mkdata.html#cb180-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb180-3"><a href="rna-seq-mkdata.html#cb180-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(dds3, <span class="at">alpha =</span> <span class="fl">0.01</span>)</span>
<span id="cb180-4"><a href="rna-seq-mkdata.html#cb180-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb180-5"><a href="rna-seq-mkdata.html#cb180-5" aria-hidden="true" tabindex="-1"></a><span class="do">## S4 method for signature &#39;DESeqDataSet&#39;</span></span>
<span id="cb180-6"><a href="rna-seq-mkdata.html#cb180-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(</span>
<span id="cb180-7"><a href="rna-seq-mkdata.html#cb180-7" aria-hidden="true" tabindex="-1"></a>  dds3,</span>
<span id="cb180-8"><a href="rna-seq-mkdata.html#cb180-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fl">0.01</span>,</span>
<span id="cb180-9"><a href="rna-seq-mkdata.html#cb180-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">&quot;DESeqDataSet plotMA&quot;</span>,</span>
<span id="cb180-10"><a href="rna-seq-mkdata.html#cb180-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">xlab =</span> <span class="st">&quot;mean of normalized counts&quot;</span>,</span>
<span id="cb180-11"><a href="rna-seq-mkdata.html#cb180-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ylim,</span></span>
<span id="cb180-12"><a href="rna-seq-mkdata.html#cb180-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">colNonSig =</span> <span class="st">&quot;gray60&quot;</span>,</span>
<span id="cb180-13"><a href="rna-seq-mkdata.html#cb180-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">colSig =</span> <span class="st">&quot;red&quot;</span>,</span>
<span id="cb180-14"><a href="rna-seq-mkdata.html#cb180-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">colLine =</span> <span class="st">&quot;grey40&quot;</span>,</span>
<span id="cb180-15"><a href="rna-seq-mkdata.html#cb180-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">returnData =</span> <span class="cn">FALSE</span>,</span>
<span id="cb180-16"><a href="rna-seq-mkdata.html#cb180-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">MLE =</span> <span class="cn">FALSE</span>,</span>
<span id="cb180-17"><a href="rna-seq-mkdata.html#cb180-17" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="eastern-new-mexico-university-enmu-method" class="section level2" number="5.8">
<h2><span class="header-section-number">5.8</span> Eastern New Mexico University ENMU method</h2>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="rna-seq-mkdata.html#cb181-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Eastern New Mexico University  ENMU ***** ------</span></span>
<span id="cb181-2"><a href="rna-seq-mkdata.html#cb181-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Load Data</span></span>
<span id="cb181-3"><a href="rna-seq-mkdata.html#cb181-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-4"><a href="rna-seq-mkdata.html#cb181-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb181-5"><a href="rna-seq-mkdata.html#cb181-5" aria-hidden="true" tabindex="-1"></a>all_raw_count <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">&quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/all.raw.count.xlsx&quot;</span>)</span>
<span id="cb181-6"><a href="rna-seq-mkdata.html#cb181-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-7"><a href="rna-seq-mkdata.html#cb181-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(all_raw_count)</span>
<span id="cb181-8"><a href="rna-seq-mkdata.html#cb181-8" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(all_raw_count)</span>
<span id="cb181-9"><a href="rna-seq-mkdata.html#cb181-9" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(all_raw_count)</span>
<span id="cb181-10"><a href="rna-seq-mkdata.html#cb181-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-11"><a href="rna-seq-mkdata.html#cb181-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract column names.</span></span>
<span id="cb181-12"><a href="rna-seq-mkdata.html#cb181-12" aria-hidden="true" tabindex="-1"></a>colnames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(all_raw_count)</span>
<span id="cb181-13"><a href="rna-seq-mkdata.html#cb181-13" aria-hidden="true" tabindex="-1"></a>colnames</span>
<span id="cb181-14"><a href="rna-seq-mkdata.html#cb181-14" aria-hidden="true" tabindex="-1"></a><span class="co"># “r convert list to comma separated string”</span></span>
<span id="cb181-15"><a href="rna-seq-mkdata.html#cb181-15" aria-hidden="true" tabindex="-1"></a><span class="fu">paste</span>(<span class="fu">shQuote</span>(colnames), <span class="at">collapse=</span><span class="st">&quot;, &quot;</span>) <span class="co"># with quotes.</span></span>
<span id="cb181-16"><a href="rna-seq-mkdata.html#cb181-16" aria-hidden="true" tabindex="-1"></a><span class="fu">paste</span>(colnames, <span class="at">collapse =</span> <span class="st">&quot;,&quot;</span>) <span class="co"># without quotes. </span></span>
<span id="cb181-17"><a href="rna-seq-mkdata.html#cb181-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-18"><a href="rna-seq-mkdata.html#cb181-18" aria-hidden="true" tabindex="-1"></a>excluded_vars <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;GeneName&quot;</span>, <span class="st">&quot;Chr&quot;</span>, <span class="st">&quot;Start&quot;</span>, <span class="st">&quot;End&quot;</span>, <span class="st">&quot;Strand&quot;</span>, <span class="st">&quot;Length&quot;</span> )</span>
<span id="cb181-19"><a href="rna-seq-mkdata.html#cb181-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-20"><a href="rna-seq-mkdata.html#cb181-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr) <span class="co"># for select function</span></span>
<span id="cb181-21"><a href="rna-seq-mkdata.html#cb181-21" aria-hidden="true" tabindex="-1"></a>all_raw_count01 <span class="ot">&lt;-</span> <span class="fu">select</span>(all_raw_count, <span class="sc">-</span>excluded_vars) <span class="co"># it works, but... better use:</span></span>
<span id="cb181-22"><a href="rna-seq-mkdata.html#cb181-22" aria-hidden="true" tabindex="-1"></a>all_raw_count01 <span class="ot">&lt;-</span> <span class="fu">select</span>(all_raw_count, <span class="sc">-</span><span class="fu">all_of</span>(excluded_vars)) <span class="co">#exclude all variables in excluded_vars</span></span>
<span id="cb181-23"><a href="rna-seq-mkdata.html#cb181-23" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(all_raw_count01)</span>
<span id="cb181-24"><a href="rna-seq-mkdata.html#cb181-24" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(all_raw_count01)</span>
<span id="cb181-25"><a href="rna-seq-mkdata.html#cb181-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-26"><a href="rna-seq-mkdata.html#cb181-26" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(all_raw_count01)</span>
<span id="cb181-27"><a href="rna-seq-mkdata.html#cb181-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb181-28"><a href="rna-seq-mkdata.html#cb181-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-29"><a href="rna-seq-mkdata.html#cb181-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Converting first column into row names</span></span>
<span id="cb181-30"><a href="rna-seq-mkdata.html#cb181-30" aria-hidden="true" tabindex="-1"></a>all_raw_count01 <span class="ot">&lt;-</span> all_raw_count01 <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">&quot;GeneID&quot;</span>) </span>
<span id="cb181-31"><a href="rna-seq-mkdata.html#cb181-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-32"><a href="rna-seq-mkdata.html#cb181-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean/filter of non expresed genes </span></span>
<span id="cb181-33"><a href="rna-seq-mkdata.html#cb181-33" aria-hidden="true" tabindex="-1"></a><span class="co"># If Center and Scale needed, remove all rows containing all 0 values.</span></span>
<span id="cb181-34"><a href="rna-seq-mkdata.html#cb181-34" aria-hidden="true" tabindex="-1"></a><span class="co"># df1[rowSums(df1[])&gt;0,]</span></span>
<span id="cb181-35"><a href="rna-seq-mkdata.html#cb181-35" aria-hidden="true" tabindex="-1"></a>all_raw_count02 <span class="ot">&lt;-</span> all_raw_count01[<span class="fu">rowSums</span>(all_raw_count01[])<span class="sc">&gt;</span><span class="dv">0</span>,] </span>
<span id="cb181-36"><a href="rna-seq-mkdata.html#cb181-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-37"><a href="rna-seq-mkdata.html#cb181-37" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(all_raw_count02 [,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span>
<span id="cb181-38"><a href="rna-seq-mkdata.html#cb181-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-39"><a href="rna-seq-mkdata.html#cb181-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-40"><a href="rna-seq-mkdata.html#cb181-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Load Metadata</span></span>
<span id="cb181-41"><a href="rna-seq-mkdata.html#cb181-41" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb181-42"><a href="rna-seq-mkdata.html#cb181-42" aria-hidden="true" tabindex="-1"></a>Metadata <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">&quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/Notes.xlsx&quot;</span>, </span>
<span id="cb181-43"><a href="rna-seq-mkdata.html#cb181-43" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sheet =</span> <span class="st">&quot;MetaData&quot;</span>)</span>
<span id="cb181-44"><a href="rna-seq-mkdata.html#cb181-44" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Metadata)</span>
<span id="cb181-45"><a href="rna-seq-mkdata.html#cb181-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-46"><a href="rna-seq-mkdata.html#cb181-46" aria-hidden="true" tabindex="-1"></a><span class="co"># Order Metadata as you want to appear in the graphs. </span></span>
<span id="cb181-47"><a href="rna-seq-mkdata.html#cb181-47" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb181-48"><a href="rna-seq-mkdata.html#cb181-48" aria-hidden="true" tabindex="-1"></a>Metadata2 <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">&quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/Notes.xlsx&quot;</span>, </span>
<span id="cb181-49"><a href="rna-seq-mkdata.html#cb181-49" aria-hidden="true" tabindex="-1"></a>                        <span class="at">sheet =</span> <span class="st">&quot;MetaData2&quot;</span>)</span>
<span id="cb181-50"><a href="rna-seq-mkdata.html#cb181-50" aria-hidden="true" tabindex="-1"></a>Metadata2</span>
<span id="cb181-51"><a href="rna-seq-mkdata.html#cb181-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-52"><a href="rna-seq-mkdata.html#cb181-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-53"><a href="rna-seq-mkdata.html#cb181-53" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(Metadata2)</span>
<span id="cb181-54"><a href="rna-seq-mkdata.html#cb181-54" aria-hidden="true" tabindex="-1"></a>Metadata2<span class="sc">$</span>Time <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(Metadata2<span class="sc">$</span>Time)</span>
<span id="cb181-55"><a href="rna-seq-mkdata.html#cb181-55" aria-hidden="true" tabindex="-1"></a>Metadata2<span class="sc">$</span>Genotype <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(Metadata2<span class="sc">$</span>Genotype)</span>
<span id="cb181-56"><a href="rna-seq-mkdata.html#cb181-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-57"><a href="rna-seq-mkdata.html#cb181-57" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(Metadata2)</span>
<span id="cb181-58"><a href="rna-seq-mkdata.html#cb181-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-59"><a href="rna-seq-mkdata.html#cb181-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-60"><a href="rna-seq-mkdata.html#cb181-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-61"><a href="rna-seq-mkdata.html#cb181-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Create dds file</span></span>
<span id="cb181-62"><a href="rna-seq-mkdata.html#cb181-62" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb181-63"><a href="rna-seq-mkdata.html#cb181-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-64"><a href="rna-seq-mkdata.html#cb181-64" aria-hidden="true" tabindex="-1"></a><span class="do">## Creation of the DESeqDataSet</span></span>
<span id="cb181-65"><a href="rna-seq-mkdata.html#cb181-65" aria-hidden="true" tabindex="-1"></a>dds4 <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> all_raw_count02, </span>
<span id="cb181-66"><a href="rna-seq-mkdata.html#cb181-66" aria-hidden="true" tabindex="-1"></a>                               <span class="at">colData =</span> Metadata2, </span>
<span id="cb181-67"><a href="rna-seq-mkdata.html#cb181-67" aria-hidden="true" tabindex="-1"></a>                               <span class="at">design =</span> <span class="sc">~</span> Genotype) <span class="co"># only one comparison. Time not included. </span></span>
<span id="cb181-68"><a href="rna-seq-mkdata.html#cb181-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-69"><a href="rna-seq-mkdata.html#cb181-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-70"><a href="rna-seq-mkdata.html#cb181-70" aria-hidden="true" tabindex="-1"></a><span class="co">#?`DESeq2-package`</span></span>
<span id="cb181-71"><a href="rna-seq-mkdata.html#cb181-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-72"><a href="rna-seq-mkdata.html#cb181-72" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only the genes with count &gt;10</span></span>
<span id="cb181-73"><a href="rna-seq-mkdata.html#cb181-73" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds4))<span class="sc">&gt;=</span><span class="dv">10</span></span>
<span id="cb181-74"><a href="rna-seq-mkdata.html#cb181-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-75"><a href="rna-seq-mkdata.html#cb181-75" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out keep.</span></span>
<span id="cb181-76"><a href="rna-seq-mkdata.html#cb181-76" aria-hidden="true" tabindex="-1"></a>dds4 <span class="ot">&lt;-</span> dds4[keep,]</span>
<span id="cb181-77"><a href="rna-seq-mkdata.html#cb181-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-78"><a href="rna-seq-mkdata.html#cb181-78" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate Size Factor</span></span>
<span id="cb181-79"><a href="rna-seq-mkdata.html#cb181-79" aria-hidden="true" tabindex="-1"></a>ddsN <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds4)</span>
<span id="cb181-80"><a href="rna-seq-mkdata.html#cb181-80" aria-hidden="true" tabindex="-1"></a>ddsN <span class="ot">&lt;-</span> <span class="fu">estimateDispersions</span>(ddsN)</span>
<span id="cb181-81"><a href="rna-seq-mkdata.html#cb181-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-82"><a href="rna-seq-mkdata.html#cb181-82" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize Data</span></span>
<span id="cb181-83"><a href="rna-seq-mkdata.html#cb181-83" aria-hidden="true" tabindex="-1"></a>Ndds <span class="ot">&lt;-</span> <span class="fu">counts</span>(ddsN, <span class="at">normalized=</span>T)</span>
<span id="cb181-84"><a href="rna-seq-mkdata.html#cb181-84" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Ndds)</span>
<span id="cb181-85"><a href="rna-seq-mkdata.html#cb181-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-86"><a href="rna-seq-mkdata.html#cb181-86" aria-hidden="true" tabindex="-1"></a><span class="co"># Save into a file.</span></span>
<span id="cb181-87"><a href="rna-seq-mkdata.html#cb181-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-88"><a href="rna-seq-mkdata.html#cb181-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-89"><a href="rna-seq-mkdata.html#cb181-89" aria-hidden="true" tabindex="-1"></a><span class="co"># Make the differentiation Expression Table.</span></span>
<span id="cb181-90"><a href="rna-seq-mkdata.html#cb181-90" aria-hidden="true" tabindex="-1"></a>ddsDE <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds4) <span class="co">#Why dds4 and not Ndds?</span></span>
<span id="cb181-91"><a href="rna-seq-mkdata.html#cb181-91" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ddsDE)</span>
<span id="cb181-92"><a href="rna-seq-mkdata.html#cb181-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-93"><a href="rna-seq-mkdata.html#cb181-93" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the results of the differentiation expression analysis and define alpha.</span></span>
<span id="cb181-94"><a href="rna-seq-mkdata.html#cb181-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-95"><a href="rna-seq-mkdata.html#cb181-95" aria-hidden="true" tabindex="-1"></a>res4 <span class="ot">&lt;-</span> <span class="fu">results</span>(ddsDE, <span class="at">alpha =</span> <span class="fl">0.001</span>)</span>
<span id="cb181-96"><a href="rna-seq-mkdata.html#cb181-96" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res4)</span>
<span id="cb181-97"><a href="rna-seq-mkdata.html#cb181-97" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(res4)</span>
<span id="cb181-98"><a href="rna-seq-mkdata.html#cb181-98" aria-hidden="true" tabindex="-1"></a><span class="co"># Save as file.</span></span>
<span id="cb181-99"><a href="rna-seq-mkdata.html#cb181-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-100"><a href="rna-seq-mkdata.html#cb181-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-101"><a href="rna-seq-mkdata.html#cb181-101" aria-hidden="true" tabindex="-1"></a><span class="co"># Order padj values from smallest to largest</span></span>
<span id="cb181-102"><a href="rna-seq-mkdata.html#cb181-102" aria-hidden="true" tabindex="-1"></a>res4Ordered <span class="ot">&lt;-</span> res4[<span class="fu">order</span>(res4<span class="sc">$</span>padj),]</span>
<span id="cb181-103"><a href="rna-seq-mkdata.html#cb181-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-104"><a href="rna-seq-mkdata.html#cb181-104" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res4Ordered)</span>
<span id="cb181-105"><a href="rna-seq-mkdata.html#cb181-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-106"><a href="rna-seq-mkdata.html#cb181-106" aria-hidden="true" tabindex="-1"></a><span class="co"># Save values in a file. </span></span>
<span id="cb181-107"><a href="rna-seq-mkdata.html#cb181-107" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;xlsx&quot;</span>)</span>
<span id="cb181-108"><a href="rna-seq-mkdata.html#cb181-108" aria-hidden="true" tabindex="-1"></a><span class="co"># Add another data set..... I AM ADDING THIS AS A NEW SHEET TO AN EXISTING FILE FROM THE PREVIOUS PCA ANALISYS.</span></span>
<span id="cb181-109"><a href="rna-seq-mkdata.html#cb181-109" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(res4Ordered, <span class="at">file =</span> <span class="st">&quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/Top10DEgenes_prcomp.xlsx&quot;</span>,</span>
<span id="cb181-110"><a href="rna-seq-mkdata.html#cb181-110" aria-hidden="true" tabindex="-1"></a>           <span class="at">sheetName =</span> <span class="st">&quot;diff2ENMU.001&quot;</span>, <span class="at">row.names =</span> T, <span class="at">append =</span> <span class="cn">TRUE</span>) <span class="co"># to add as a new sheet.</span></span>
<span id="cb181-111"><a href="rna-seq-mkdata.html#cb181-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-112"><a href="rna-seq-mkdata.html#cb181-112" aria-hidden="true" tabindex="-1"></a><span class="co"># TAKES A LONG TIME TO SAVE... BE PATIENT. *****************************************</span></span>
<span id="cb181-113"><a href="rna-seq-mkdata.html#cb181-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-114"><a href="rna-seq-mkdata.html#cb181-114" aria-hidden="true" tabindex="-1"></a><span class="co"># PlotMA</span></span>
<span id="cb181-115"><a href="rna-seq-mkdata.html#cb181-115" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(ddsDE,</span>
<span id="cb181-116"><a href="rna-seq-mkdata.html#cb181-116" aria-hidden="true" tabindex="-1"></a>       <span class="at">colSig =</span> <span class="st">&quot;red&quot;</span>,)</span>
<span id="cb181-117"><a href="rna-seq-mkdata.html#cb181-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-118"><a href="rna-seq-mkdata.html#cb181-118" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot PCA</span></span>
<span id="cb181-119"><a href="rna-seq-mkdata.html#cb181-119" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply a variance stabilizing transformation (VST) to the count data</span></span>
<span id="cb181-120"><a href="rna-seq-mkdata.html#cb181-120" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds4, <span class="at">blind =</span> F)</span>
<span id="cb181-121"><a href="rna-seq-mkdata.html#cb181-121" aria-hidden="true" tabindex="-1"></a>vsd</span>
<span id="cb181-122"><a href="rna-seq-mkdata.html#cb181-122" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(vsd)</span>
<span id="cb181-123"><a href="rna-seq-mkdata.html#cb181-123" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd, <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">&quot;Time&quot;</span>, <span class="st">&quot;Genotype&quot;</span>))</span>
<span id="cb181-124"><a href="rna-seq-mkdata.html#cb181-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-125"><a href="rna-seq-mkdata.html#cb181-125" aria-hidden="true" tabindex="-1"></a><span class="co"># Sample to sample distribution</span></span>
<span id="cb181-126"><a href="rna-seq-mkdata.html#cb181-126" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb181-127"><a href="rna-seq-mkdata.html#cb181-127" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb181-128"><a href="rna-seq-mkdata.html#cb181-128" aria-hidden="true" tabindex="-1"></a>sampleDist <span class="ot">&lt;-</span> <span class="fu">dist</span>(<span class="fu">t</span>(<span class="fu">assay</span>(vsd)))</span>
<span id="cb181-129"><a href="rna-seq-mkdata.html#cb181-129" aria-hidden="true" tabindex="-1"></a>sampleDist</span>
<span id="cb181-130"><a href="rna-seq-mkdata.html#cb181-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-131"><a href="rna-seq-mkdata.html#cb181-131" aria-hidden="true" tabindex="-1"></a>sampleDistMatrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(sampleDist)</span>
<span id="cb181-132"><a href="rna-seq-mkdata.html#cb181-132" aria-hidden="true" tabindex="-1"></a>sampleDistMatrix</span>
<span id="cb181-133"><a href="rna-seq-mkdata.html#cb181-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-134"><a href="rna-seq-mkdata.html#cb181-134" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sampleDistMatrix) <span class="ot">&lt;-</span> vsd<span class="sc">$</span>Genotype</span>
<span id="cb181-135"><a href="rna-seq-mkdata.html#cb181-135" aria-hidden="true" tabindex="-1"></a>sampleDistMatrix</span>
<span id="cb181-136"><a href="rna-seq-mkdata.html#cb181-136" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sampleDistMatrix) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb181-137"><a href="rna-seq-mkdata.html#cb181-137" aria-hidden="true" tabindex="-1"></a>sampleDistMatrix</span>
<span id="cb181-138"><a href="rna-seq-mkdata.html#cb181-138" aria-hidden="true" tabindex="-1"></a><span class="co">#colors &lt;- colorRampPalette(rev(brewer.pal(9, &quot;Blues&quot;))), (225))</span></span>
<span id="cb181-139"><a href="rna-seq-mkdata.html#cb181-139" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(sampleDistMatrix,</span>
<span id="cb181-140"><a href="rna-seq-mkdata.html#cb181-140" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> sampleDist,</span>
<span id="cb181-141"><a href="rna-seq-mkdata.html#cb181-141" aria-hidden="true" tabindex="-1"></a>         <span class="at">clustering_distance_cols =</span> sampleDist,</span>
<span id="cb181-142"><a href="rna-seq-mkdata.html#cb181-142" aria-hidden="true" tabindex="-1"></a>         <span class="co">#col=colors</span></span>
<span id="cb181-143"><a href="rna-seq-mkdata.html#cb181-143" aria-hidden="true" tabindex="-1"></a>         )</span>
<span id="cb181-144"><a href="rna-seq-mkdata.html#cb181-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-145"><a href="rna-seq-mkdata.html#cb181-145" aria-hidden="true" tabindex="-1"></a><span class="co"># Gene expression plotting</span></span>
<span id="cb181-146"><a href="rna-seq-mkdata.html#cb181-146" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb181-147"><a href="rna-seq-mkdata.html#cb181-147" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages(&quot;meltt&quot;)</span></span>
<span id="cb181-148"><a href="rna-seq-mkdata.html#cb181-148" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape)</span>
<span id="cb181-149"><a href="rna-seq-mkdata.html#cb181-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-150"><a href="rna-seq-mkdata.html#cb181-150" aria-hidden="true" tabindex="-1"></a><span class="co"># Expression file (Ndds)</span></span>
<span id="cb181-151"><a href="rna-seq-mkdata.html#cb181-151" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Ndds)  <span class="co"># Is a matrix</span></span>
<span id="cb181-152"><a href="rna-seq-mkdata.html#cb181-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-153"><a href="rna-seq-mkdata.html#cb181-153" aria-hidden="true" tabindex="-1"></a><span class="co"># DE results file</span></span>
<span id="cb181-154"><a href="rna-seq-mkdata.html#cb181-154" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res4Ordered) <span class="co"># Is not a matrix. but was saved in a file </span></span>
<span id="cb181-155"><a href="rna-seq-mkdata.html#cb181-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-156"><a href="rna-seq-mkdata.html#cb181-156" aria-hidden="true" tabindex="-1"></a><span class="co"># Load DE results file </span></span>
<span id="cb181-157"><a href="rna-seq-mkdata.html#cb181-157" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb181-158"><a href="rna-seq-mkdata.html#cb181-158" aria-hidden="true" tabindex="-1"></a>statsDE <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">&quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/Top10DEgenes_prcomp.xlsx&quot;</span>, </span>
<span id="cb181-159"><a href="rna-seq-mkdata.html#cb181-159" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">sheet =</span> <span class="st">&quot;diff2ENMU.01&quot;</span>)</span>
<span id="cb181-160"><a href="rna-seq-mkdata.html#cb181-160" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(statsDE)</span>
<span id="cb181-161"><a href="rna-seq-mkdata.html#cb181-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-162"><a href="rna-seq-mkdata.html#cb181-162" aria-hidden="true" tabindex="-1"></a><span class="co"># Change column of GeneID to row names Method 3 </span></span>
<span id="cb181-163"><a href="rna-seq-mkdata.html#cb181-163" aria-hidden="true" tabindex="-1"></a>statsDE <span class="ot">&lt;-</span> statsDE <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">&quot;...1&quot;</span>) </span>
<span id="cb181-164"><a href="rna-seq-mkdata.html#cb181-164" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(statsDE)</span>
<span id="cb181-165"><a href="rna-seq-mkdata.html#cb181-165" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(statsDE)</span>
<span id="cb181-166"><a href="rna-seq-mkdata.html#cb181-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-167"><a href="rna-seq-mkdata.html#cb181-167" aria-hidden="true" tabindex="-1"></a><span class="co"># Check if there are NA values in the df. If true then omit all NA rows. </span></span>
<span id="cb181-168"><a href="rna-seq-mkdata.html#cb181-168" aria-hidden="true" tabindex="-1"></a><span class="fu">apply</span>(statsDE, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">any</span>(<span class="fu">is.na</span>(x))) <span class="co"># check by columns</span></span>
<span id="cb181-169"><a href="rna-seq-mkdata.html#cb181-169" aria-hidden="true" tabindex="-1"></a><span class="co"># baseMean log2FoldChange          lfcSE           stat         pvalue           padj </span></span>
<span id="cb181-170"><a href="rna-seq-mkdata.html#cb181-170" aria-hidden="true" tabindex="-1"></a><span class="co"># FALSE          FALSE          FALSE          FALSE          FALSE           TRUE </span></span>
<span id="cb181-171"><a href="rna-seq-mkdata.html#cb181-171" aria-hidden="true" tabindex="-1"></a>statsDE <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(statsDE)</span>
<span id="cb181-172"><a href="rna-seq-mkdata.html#cb181-172" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(statsDE)</span>
<span id="cb181-173"><a href="rna-seq-mkdata.html#cb181-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-174"><a href="rna-seq-mkdata.html#cb181-174" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter Significant values. 0.05 or 0.01 or 0.001</span></span>
<span id="cb181-175"><a href="rna-seq-mkdata.html#cb181-175" aria-hidden="true" tabindex="-1"></a>statsDE<span class="sc">$</span>sig <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(statsDE<span class="sc">$</span>padj <span class="sc">&lt;=</span> <span class="fl">0.01</span>, <span class="st">&quot;sig&quot;</span>, <span class="st">&quot;not&quot;</span>)</span>
<span id="cb181-176"><a href="rna-seq-mkdata.html#cb181-176" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(statsDE)</span>
<span id="cb181-177"><a href="rna-seq-mkdata.html#cb181-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-178"><a href="rna-seq-mkdata.html#cb181-178" aria-hidden="true" tabindex="-1"></a>plotMA <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(statsDE, <span class="fu">aes</span>(<span class="at">x=</span> <span class="fu">log10</span>(baseMean), <span class="at">y=</span> log2FoldChange, <span class="at">color=</span>sig)) <span class="sc">+</span> </span>
<span id="cb181-179"><a href="rna-seq-mkdata.html#cb181-179" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()</span>
<span id="cb181-180"><a href="rna-seq-mkdata.html#cb181-180" aria-hidden="true" tabindex="-1"></a>plotMA</span>
<span id="cb181-181"><a href="rna-seq-mkdata.html#cb181-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-182"><a href="rna-seq-mkdata.html#cb181-182" aria-hidden="true" tabindex="-1"></a><span class="co"># Volcano Plot</span></span>
<span id="cb181-183"><a href="rna-seq-mkdata.html#cb181-183" aria-hidden="true" tabindex="-1"></a>volcanoplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(statsDE, <span class="fu">aes</span>(<span class="at">x=</span> log2FoldChange, <span class="at">y=</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">color =</span> sig)) <span class="sc">+</span></span>
<span id="cb181-184"><a href="rna-seq-mkdata.html#cb181-184" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()</span>
<span id="cb181-185"><a href="rna-seq-mkdata.html#cb181-185" aria-hidden="true" tabindex="-1"></a>volcanoplot</span>
<span id="cb181-186"><a href="rna-seq-mkdata.html#cb181-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-187"><a href="rna-seq-mkdata.html#cb181-187" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the top 10  differentially expressed genes. </span></span>
<span id="cb181-188"><a href="rna-seq-mkdata.html#cb181-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-189"><a href="rna-seq-mkdata.html#cb181-189" aria-hidden="true" tabindex="-1"></a>topTen <span class="ot">&lt;-</span> statsDE[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>,]</span>
<span id="cb181-190"><a href="rna-seq-mkdata.html#cb181-190" aria-hidden="true" tabindex="-1"></a>topTen</span>
<span id="cb181-191"><a href="rna-seq-mkdata.html#cb181-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-192"><a href="rna-seq-mkdata.html#cb181-192" aria-hidden="true" tabindex="-1"></a>all_raw_count00 <span class="ot">&lt;-</span> all_raw_count <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">&quot;GeneID&quot;</span>) <span class="co"># For all data columns</span></span>
<span id="cb181-193"><a href="rna-seq-mkdata.html#cb181-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-194"><a href="rna-seq-mkdata.html#cb181-194" aria-hidden="true" tabindex="-1"></a>all_raw_count00[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,]</span>
<span id="cb181-195"><a href="rna-seq-mkdata.html#cb181-195" aria-hidden="true" tabindex="-1"></a>topTenMerg <span class="ot">&lt;-</span> <span class="fu">merge</span>(topTen, all_raw_count00, <span class="at">by=</span><span class="dv">0</span>) </span>
<span id="cb181-196"><a href="rna-seq-mkdata.html#cb181-196" aria-hidden="true" tabindex="-1"></a>topTenMerg [<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,]</span>
<span id="cb181-197"><a href="rna-seq-mkdata.html#cb181-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-198"><a href="rna-seq-mkdata.html#cb181-198" aria-hidden="true" tabindex="-1"></a><span class="co"># Save values in a file. </span></span>
<span id="cb181-199"><a href="rna-seq-mkdata.html#cb181-199" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;xlsx&quot;</span>)</span>
<span id="cb181-200"><a href="rna-seq-mkdata.html#cb181-200" aria-hidden="true" tabindex="-1"></a><span class="co"># Add another data set..... I AM ADDING THIS AS A NEW SHEET TO AN EXISTING FILE FROM THE PREVIOUS PCA ANALISYS.</span></span>
<span id="cb181-201"><a href="rna-seq-mkdata.html#cb181-201" aria-hidden="true" tabindex="-1"></a><span class="fu">write.xlsx</span>(topTenMerg, <span class="at">file =</span> <span class="st">&quot;~/Box Sync/Niigata Uni Box/Kakenhi/2020-22 Kakenhi/Raw data/Top10DEgenes_prcomp.xlsx&quot;</span>,</span>
<span id="cb181-202"><a href="rna-seq-mkdata.html#cb181-202" aria-hidden="true" tabindex="-1"></a>           <span class="at">sheetName =</span> <span class="st">&quot;top100MergENMU&quot;</span>, <span class="at">row.names =</span> T, <span class="at">append =</span> <span class="cn">TRUE</span>) <span class="co"># to add as a new sheet.</span></span>
<span id="cb181-203"><a href="rna-seq-mkdata.html#cb181-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-204"><a href="rna-seq-mkdata.html#cb181-204" aria-hidden="true" tabindex="-1"></a><span class="co"># Heat Map of top10 genes.</span></span>
<span id="cb181-205"><a href="rna-seq-mkdata.html#cb181-205" aria-hidden="true" tabindex="-1"></a><span class="co"># Cunstruct Matrix.</span></span>
<span id="cb181-206"><a href="rna-seq-mkdata.html#cb181-206" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract column names.</span></span>
<span id="cb181-207"><a href="rna-seq-mkdata.html#cb181-207" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(topTenMerg)</span>
<span id="cb181-208"><a href="rna-seq-mkdata.html#cb181-208" aria-hidden="true" tabindex="-1"></a><span class="fu">paste</span>(<span class="fu">shQuote</span>(<span class="fu">colnames</span>(topTenMerg)), <span class="at">collapse=</span><span class="st">&quot;, &quot;</span>) <span class="co"># with quotes.</span></span>
<span id="cb181-209"><a href="rna-seq-mkdata.html#cb181-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-210"><a href="rna-seq-mkdata.html#cb181-210" aria-hidden="true" tabindex="-1"></a>include_var <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;Row.names&#39;</span>, <span class="st">&#39;mPDL_RNA14D_Ko1&#39;</span>, <span class="st">&#39;mPDL_RNA14D_Ko2&#39;</span>, <span class="st">&#39;mPDL_RNA14D_Ko3&#39;</span>, <span class="st">&#39;mPDL_RNA14D_Ko4&#39;</span>, <span class="st">&#39;mPDL_RNA14D_WT1&#39;</span>, <span class="st">&#39;mPDL_RNA14D_WT2&#39;</span>, <span class="st">&#39;mPDL_RNA14D_WT3&#39;</span>, <span class="st">&#39;mPDL_RNA14D_WT4&#39;</span>, <span class="st">&#39;mPDL_RNA7D_Ko1&#39;</span>, <span class="st">&#39;mPDL_RNA7D_Ko2&#39;</span>, <span class="st">&#39;mPDL_RNA7D_Ko3&#39;</span>, <span class="st">&#39;mPDL_RNA7D_Ko4&#39;</span>, <span class="st">&#39;mPDL_RNA7D_WT1&#39;</span>, <span class="st">&#39;mPDL_RNA7D_WT2&#39;</span>, <span class="st">&#39;mPDL_RNA7D_WT3&#39;</span>, <span class="st">&#39;mPDL_RNA7D_WT4&#39;</span> )</span>
<span id="cb181-211"><a href="rna-seq-mkdata.html#cb181-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-212"><a href="rna-seq-mkdata.html#cb181-212" aria-hidden="true" tabindex="-1"></a><span class="co"># Top10heat &lt;- topTenMerg[,names(topTenMerg)] %in% include_var #Not working, gives T/F vector instead of matrix.</span></span>
<span id="cb181-213"><a href="rna-seq-mkdata.html#cb181-213" aria-hidden="true" tabindex="-1"></a><span class="co"># Top10heat</span></span>
<span id="cb181-214"><a href="rna-seq-mkdata.html#cb181-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-215"><a href="rna-seq-mkdata.html#cb181-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-216"><a href="rna-seq-mkdata.html#cb181-216" aria-hidden="true" tabindex="-1"></a>Top10heat <span class="ot">&lt;-</span> <span class="fu">select</span>(topTenMerg, <span class="fu">all_of</span>(include_var)) <span class="co">#exclude all variables in excluded_vars</span></span>
<span id="cb181-217"><a href="rna-seq-mkdata.html#cb181-217" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Top10heat)</span>
<span id="cb181-218"><a href="rna-seq-mkdata.html#cb181-218" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(Top10heat)</span>
<span id="cb181-219"><a href="rna-seq-mkdata.html#cb181-219" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(Top10heat)</span>
<span id="cb181-220"><a href="rna-seq-mkdata.html#cb181-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-221"><a href="rna-seq-mkdata.html#cb181-221" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb181-222"><a href="rna-seq-mkdata.html#cb181-222" aria-hidden="true" tabindex="-1"></a><span class="co"># col1 to rownames</span></span>
<span id="cb181-223"><a href="rna-seq-mkdata.html#cb181-223" aria-hidden="true" tabindex="-1"></a>Top10heat <span class="ot">&lt;-</span> Top10heat <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">&quot;Row.names&quot;</span>) </span>
<span id="cb181-224"><a href="rna-seq-mkdata.html#cb181-224" aria-hidden="true" tabindex="-1"></a><span class="co"># Source: https://www.geeksforgeeks.org/convert-values-in-column-into-row-names-of-dataframe-in-r/</span></span>
<span id="cb181-225"><a href="rna-seq-mkdata.html#cb181-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-226"><a href="rna-seq-mkdata.html#cb181-226" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Top10heat)</span>
<span id="cb181-227"><a href="rna-seq-mkdata.html#cb181-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-228"><a href="rna-seq-mkdata.html#cb181-228" aria-hidden="true" tabindex="-1"></a><span class="co"># reorder  columns according to the experimental design file</span></span>
<span id="cb181-229"><a href="rna-seq-mkdata.html#cb181-229" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Metadata2)</span>
<span id="cb181-230"><a href="rna-seq-mkdata.html#cb181-230" aria-hidden="true" tabindex="-1"></a>Top10heatord <span class="ot">&lt;-</span> Top10heat[,Metadata2<span class="sc">$</span>SampleName]</span>
<span id="cb181-231"><a href="rna-seq-mkdata.html#cb181-231" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Top10heatord [<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>]) </span>
<span id="cb181-232"><a href="rna-seq-mkdata.html#cb181-232" aria-hidden="true" tabindex="-1"></a>Top10heatord</span>
<span id="cb181-233"><a href="rna-seq-mkdata.html#cb181-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-234"><a href="rna-seq-mkdata.html#cb181-234" aria-hidden="true" tabindex="-1"></a><span class="co"># Heat plot</span></span>
<span id="cb181-235"><a href="rna-seq-mkdata.html#cb181-235" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(Top10heatord, <span class="at">main =</span> <span class="st">&quot;heatmap top 10 genes try1&quot;</span>)</span>
<span id="cb181-236"><a href="rna-seq-mkdata.html#cb181-236" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(<span class="fu">log2</span>(Top10heatord<span class="sc">+</span><span class="dv">1</span>), <span class="at">main =</span> <span class="st">&quot;heatmap top 100 genes try2&quot;</span>)</span>
<span id="cb181-237"><a href="rna-seq-mkdata.html#cb181-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-238"><a href="rna-seq-mkdata.html#cb181-238" aria-hidden="true" tabindex="-1"></a><span class="co"># Individual expression values of the DE genes</span></span>
<span id="cb181-239"><a href="rna-seq-mkdata.html#cb181-239" aria-hidden="true" tabindex="-1"></a><span class="co"># Only for the top 10 genes. </span></span>
<span id="cb181-240"><a href="rna-seq-mkdata.html#cb181-240" aria-hidden="true" tabindex="-1"></a>newtop10 <span class="ot">&lt;-</span> Top10heatord[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,]</span>
<span id="cb181-241"><a href="rna-seq-mkdata.html#cb181-241" aria-hidden="true" tabindex="-1"></a>newtop10</span>
<span id="cb181-242"><a href="rna-seq-mkdata.html#cb181-242" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(newtop10)</span>
<span id="cb181-243"><a href="rna-seq-mkdata.html#cb181-243" aria-hidden="true" tabindex="-1"></a><span class="co"># construct  appropiate df</span></span>
<span id="cb181-244"><a href="rna-seq-mkdata.html#cb181-244" aria-hidden="true" tabindex="-1"></a>newtop10 <span class="ot">&lt;-</span> <span class="fu">melt</span>(<span class="fu">as.matrix</span>(newtop10))</span>
<span id="cb181-245"><a href="rna-seq-mkdata.html#cb181-245" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename columns</span></span>
<span id="cb181-246"><a href="rna-seq-mkdata.html#cb181-246" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(newtop10) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;GeneID&quot;</span>, <span class="st">&quot;Sample&quot;</span>, <span class="st">&quot;count&quot;</span>) <span class="co"># maybe better use exp (expression) instead of count. </span></span>
<span id="cb181-247"><a href="rna-seq-mkdata.html#cb181-247" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(newtop10)</span>
<span id="cb181-248"><a href="rna-seq-mkdata.html#cb181-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-249"><a href="rna-seq-mkdata.html#cb181-249" aria-hidden="true" tabindex="-1"></a><span class="co"># Add new columns Genetype and Time</span></span>
<span id="cb181-250"><a href="rna-seq-mkdata.html#cb181-250" aria-hidden="true" tabindex="-1"></a>newtop10<span class="sc">$</span>Genetype <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;WT&quot;</span>, newtop10<span class="sc">$</span>Sample), <span class="st">&quot;WT&quot;</span>, <span class="st">&quot;Ko&quot;</span>)</span>
<span id="cb181-251"><a href="rna-seq-mkdata.html#cb181-251" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(newtop10)</span>
<span id="cb181-252"><a href="rna-seq-mkdata.html#cb181-252" aria-hidden="true" tabindex="-1"></a>newtop10<span class="sc">$</span>Time <span class="ot">&lt;-</span> <span class="fu">as.factor</span>( <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;7&quot;</span>, newtop10<span class="sc">$</span>Sample), <span class="st">&quot;7&quot;</span>, <span class="st">&quot;14&quot;</span>))</span>
<span id="cb181-253"><a href="rna-seq-mkdata.html#cb181-253" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(newtop10)</span>
<span id="cb181-254"><a href="rna-seq-mkdata.html#cb181-254" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(newtop10)</span>
<span id="cb181-255"><a href="rna-seq-mkdata.html#cb181-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-256"><a href="rna-seq-mkdata.html#cb181-256" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(newtop10, <span class="fu">aes</span>(<span class="at">x=</span> Genetype, <span class="at">y=</span> <span class="fu">log2</span>(count<span class="sc">+</span><span class="dv">1</span>), <span class="at">color=</span> Genetype)) <span class="sc">+</span></span>
<span id="cb181-257"><a href="rna-seq-mkdata.html#cb181-257" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb181-258"><a href="rna-seq-mkdata.html#cb181-258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(Time<span class="sc">~</span>GeneID) <span class="sc">+</span></span>
<span id="cb181-259"><a href="rna-seq-mkdata.html#cb181-259" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>()</span>
<span id="cb181-260"><a href="rna-seq-mkdata.html#cb181-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-261"><a href="rna-seq-mkdata.html#cb181-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-262"><a href="rna-seq-mkdata.html#cb181-262" aria-hidden="true" tabindex="-1"></a><span class="co"># Extremely quick tutorial on DESequ analysis  ***** -----------------------------</span></span>
<span id="cb181-263"><a href="rna-seq-mkdata.html#cb181-263" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(htmltools)</span>
<span id="cb181-264"><a href="rna-seq-mkdata.html#cb181-264" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb181-265"><a href="rna-seq-mkdata.html#cb181-265" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb181-266"><a href="rna-seq-mkdata.html#cb181-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-267"><a href="rna-seq-mkdata.html#cb181-267" aria-hidden="true" tabindex="-1"></a><span class="co">#setwd()</span></span>
<span id="cb181-268"><a href="rna-seq-mkdata.html#cb181-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-269"><a href="rna-seq-mkdata.html#cb181-269" aria-hidden="true" tabindex="-1"></a><span class="co"># Load Data -&gt; construct appropiate df.</span></span>
<span id="cb181-270"><a href="rna-seq-mkdata.html#cb181-270" aria-hidden="true" tabindex="-1"></a>all_raw_count02</span>
<span id="cb181-271"><a href="rna-seq-mkdata.html#cb181-271" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(all_raw_count02)</span>
<span id="cb181-272"><a href="rna-seq-mkdata.html#cb181-272" aria-hidden="true" tabindex="-1"></a><span class="co"># Load Metadata -&gt; construct appropiate df. </span></span>
<span id="cb181-273"><a href="rna-seq-mkdata.html#cb181-273" aria-hidden="true" tabindex="-1"></a>Metadata2</span>
<span id="cb181-274"><a href="rna-seq-mkdata.html#cb181-274" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Metadata2)</span>
<span id="cb181-275"><a href="rna-seq-mkdata.html#cb181-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-276"><a href="rna-seq-mkdata.html#cb181-276" aria-hidden="true" tabindex="-1"></a><span class="co"># Contruct DESeqDataSet object</span></span>
<span id="cb181-277"><a href="rna-seq-mkdata.html#cb181-277" aria-hidden="true" tabindex="-1"></a>dds210721 <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> all_raw_count02,</span>
<span id="cb181-278"><a href="rna-seq-mkdata.html#cb181-278" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">colData =</span> Metadata2,</span>
<span id="cb181-279"><a href="rna-seq-mkdata.html#cb181-279" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">design =</span> <span class="sc">~</span> Genotype <span class="sc">+</span>Time, </span>
<span id="cb181-280"><a href="rna-seq-mkdata.html#cb181-280" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">tidy =</span> F) <span class="co">#If the 1st column of countData is the rownames for the count matrix then TRUE</span></span>
<span id="cb181-281"><a href="rna-seq-mkdata.html#cb181-281" aria-hidden="true" tabindex="-1"></a>dds210721</span>
<span id="cb181-282"><a href="rna-seq-mkdata.html#cb181-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-283"><a href="rna-seq-mkdata.html#cb181-283" aria-hidden="true" tabindex="-1"></a><span class="co">#Run DESeq function for the differentiation Expression Table</span></span>
<span id="cb181-284"><a href="rna-seq-mkdata.html#cb181-284" aria-hidden="true" tabindex="-1"></a>dds210721 <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds210721)</span>
<span id="cb181-285"><a href="rna-seq-mkdata.html#cb181-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-286"><a href="rna-seq-mkdata.html#cb181-286" aria-hidden="true" tabindex="-1"></a><span class="co"># Call the results of the differentiation expression analysis and define alpha.</span></span>
<span id="cb181-287"><a href="rna-seq-mkdata.html#cb181-287" aria-hidden="true" tabindex="-1"></a>res210721 <span class="ot">&lt;-</span> <span class="fu">results</span>(dds210721, <span class="at">alpha =</span> <span class="fl">0.001</span>)</span>
<span id="cb181-288"><a href="rna-seq-mkdata.html#cb181-288" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res210721, <span class="at">tidy=</span>F)</span>
<span id="cb181-289"><a href="rna-seq-mkdata.html#cb181-289" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(res210721)</span>
<span id="cb181-290"><a href="rna-seq-mkdata.html#cb181-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-291"><a href="rna-seq-mkdata.html#cb181-291" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res210721)</span>
<span id="cb181-292"><a href="rna-seq-mkdata.html#cb181-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-293"><a href="rna-seq-mkdata.html#cb181-293" aria-hidden="true" tabindex="-1"></a><span class="co"># Sort/order summary list with  padj values from smallest to largest</span></span>
<span id="cb181-294"><a href="rna-seq-mkdata.html#cb181-294" aria-hidden="true" tabindex="-1"></a>res210721Rank <span class="ot">&lt;-</span> res210721[<span class="fu">order</span>(res210721<span class="sc">$</span>padj),]</span>
<span id="cb181-295"><a href="rna-seq-mkdata.html#cb181-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-296"><a href="rna-seq-mkdata.html#cb181-296" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res210721Rank)</span>
<span id="cb181-297"><a href="rna-seq-mkdata.html#cb181-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-298"><a href="rna-seq-mkdata.html#cb181-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb181-299"><a href="rna-seq-mkdata.html#cb181-299" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot PCA</span></span>
<span id="cb181-300"><a href="rna-seq-mkdata.html#cb181-300" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply a variance stabilizing transformation (VST) to the count data</span></span>
<span id="cb181-301"><a href="rna-seq-mkdata.html#cb181-301" aria-hidden="true" tabindex="-1"></a>vsd210721 <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds210721, <span class="at">blind =</span> F)</span>
<span id="cb181-302"><a href="rna-seq-mkdata.html#cb181-302" aria-hidden="true" tabindex="-1"></a>vsd210721</span>
<span id="cb181-303"><a href="rna-seq-mkdata.html#cb181-303" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(vsd210721)</span>
<span id="cb181-304"><a href="rna-seq-mkdata.html#cb181-304" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd210721, <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">&quot;Time&quot;</span>, <span class="st">&quot;Genotype&quot;</span>))</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="rna-seq-differentially-expressed-go-enrichment-and-pathway-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="r-code-modules-to-be-used-for-stand-alone-analysis-.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Bioinformatics_RNA-seq.pdf", "Bioinformatics_RNA-seq.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
